{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "a077d86b",
   "metadata": {},
   "source": [
    "Old Analysis of operator domain, with different approach and different bio tools"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "134afa9f",
   "metadata": {},
   "source": [
    "# Operator domain data analysis\n",
    "- Tables 3 and 4 in the paper number of tools used in the workflows \n",
    "- (e.g., bioinformatics tools such as samtools, bcftools)\n",
    "\n",
    "In this seperate notebook, we want to analyze the operators in the nf-core git-repos. \n",
    "The target operators are provided in two excel sheets and can be found in the folder 'data'. Also we want to consider the operators from the Nexflow documentation here: https://www.nextflow.io/docs/latest/operator.html"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e969fef3",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "\n",
    "df_operators = pd.read_excel('./data/all_operator_dataset.xlsx')\n",
    "#df_bioinformatic_tools = pd.read_excel('./data/bioinformatics_tools.xlsx')\n",
    "\n",
    "pattern_list = df_operators['operator'].tolist()\n",
    "\n",
    "print(pattern_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fd0a6e2a",
   "metadata": {},
   "outputs": [],
   "source": [
    "len(pattern_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "61045922",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_operators_clustured = pd.read_csv('./data/operator_dataset_clustered.csv')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1a631a34",
   "metadata": {},
   "source": [
    "We have 6 unique domain groups: [\"Bioinformatics\", \"Control flow\", \"File processing\", \"File system\", \"Code execution\", \"Data handling\", \"Misc\", \"Noise\", \"under 10\"]\n",
    "\n",
    "\n",
    "e.g.: Assigned Group 0 = \"Bioinformatics\"\n",
    "if domain >= 6: \"Other\" domain"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4d2f4054",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_operators_clustured.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e6355214",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_operators_clustured['Assigned Group'].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5f34bbac",
   "metadata": {},
   "outputs": [],
   "source": [
    "df"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "99f0d820",
   "metadata": {},
   "source": [
    "# Operator domain data (all operators from csv) in processes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d6a396f7",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import re\n",
    "import csv\n",
    "import pandas as pd\n",
    "\n",
    "def find_operators_in_processes(directory, pattern_list, output_file_name):\n",
    "    result_list = []\n",
    "\n",
    "    for pipeline in os.listdir(directory):\n",
    "        subdirectory_path = os.path.join(directory, pipeline)\n",
    "        folder_name = os.path.basename(subdirectory_path)\n",
    "\n",
    "        for root, dirs, files in os.walk(subdirectory_path):\n",
    "            for filename in files:\n",
    "                file_path = os.path.join(root, filename)\n",
    "                \n",
    "                if filename.endswith('.nf') or filename.endswith('.config'):\n",
    "                    with open(file_path, 'r', encoding='utf-8', errors='ignore') as file:\n",
    "                        lines = file.readlines()\n",
    "                    \n",
    "                    process_flag = False\n",
    "                    process_name = None\n",
    "                    brace_stack = []\n",
    "\n",
    "                    process_pattern = r'process\\s+(\\w+)\\s*{'\n",
    "\n",
    "                    for line_num, line in enumerate(lines):\n",
    "                        stripped_line = line.strip()\n",
    "                        \n",
    "                        if line.strip().startswith(('#', '//', '*')):\n",
    "                                continue\n",
    "\n",
    "                        if re.search(process_pattern, stripped_line):\n",
    "                            process_flag = True\n",
    "                            process_match = re.search(process_pattern, stripped_line)\n",
    "                            if process_match:\n",
    "                                process_name = process_match.group(1)\n",
    "                            print(f'Process {process_name} in filepath {file_path}')\n",
    "                            brace_stack.append('{')\n",
    "\n",
    "                        if '{' in stripped_line:\n",
    "                            brace_stack.append('{')\n",
    "\n",
    "                        if '}' in stripped_line:\n",
    "                            if brace_stack:\n",
    "                                brace_stack.pop()\n",
    "                                if not brace_stack and process_flag:\n",
    "                                    process_flag = False\n",
    "\n",
    "                        if process_flag:\n",
    "                            pattern_found = False\n",
    "                            for pattern in pattern_list:\n",
    "\n",
    "                                regex_pattern = re.compile(re.escape(pattern))\n",
    "                                #regex_pattern = re.compile(r'(\\.\\s*|\\s+)' + re.escape(pattern) + r'(\\s+|\\.\\s*)')\n",
    "                                if regex_pattern.search(stripped_line):\n",
    "                                    result_list.append([folder_name, process_name, pattern, file_path, line_num+1, stripped_line])\n",
    "                                    print(f'Operator found {pattern} with content: {stripped_line}')\n",
    "                                    pattern_found = True\n",
    "\n",
    "                    if brace_stack:\n",
    "                        print(\"Error: Unclosed process block in file:\", file_path)\n",
    "\n",
    "    with open(f'./results/{output_file_name}.csv', 'w', newline='', encoding='utf-8') as csvfile:\n",
    "        writer = csv.writer(csvfile)\n",
    "        writer.writerow(['pipeline', 'process_name', 'operator', 'file_path', 'line_number', 'line_content'])\n",
    "        writer.writerows(result_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e912f6e3",
   "metadata": {},
   "outputs": [],
   "source": [
    "directory_path = './git_repos'\n",
    "\n",
    "df_operators = pd.read_excel('./data/all_operator_dataset.xlsx')\n",
    "operator_list = [str(operator) for operator in df_operators['operator']]\n",
    "\n",
    "output_file_name = 'all_operators_in_processes'\n",
    "\n",
    "find_operators_in_processes(directory_path, operator_list, output_file_name)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0508037b",
   "metadata": {},
   "source": [
    "Bioinformatic tools in processes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f7036573",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_operators = pd.read_excel('./data/bioinformatics_tools.xlsx')\n",
    "pattern_list = [str(pattern) for pattern in df_operators['operator']]\n",
    "\n",
    "output_file_name = 'all_operators_in_processes'\n",
    "\n",
    "find_operators_in_processes(directory_path, pattern_list, output_file_name)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3d8a612b",
   "metadata": {},
   "source": [
    "# Parser for operator domain data occurences in config files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "7ed6af1c",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import re\n",
    "import csv\n",
    "import pandas as pd\n",
    "\n",
    "def find_operators_in_config(directory, pattern_list, output_file_name):\n",
    "    result_list = []\n",
    "\n",
    "    for pipeline in os.listdir(directory):\n",
    "        subdirectory_path = os.path.join(directory, pipeline)\n",
    "        folder_name = os.path.basename(subdirectory_path)\n",
    "\n",
    "        for root, dirs, files in os.walk(subdirectory_path):\n",
    "            for filename in files:\n",
    "                file_path = os.path.join(root, filename)\n",
    "                \n",
    "                if filename.endswith('.config'):\n",
    "                #if filename.endswith('.nf'):\n",
    "                    with open(file_path, 'r', encoding='utf-8', errors='ignore') as file:\n",
    "                        lines = file.readlines()\n",
    "\n",
    "                    for line_num, line in enumerate(lines):\n",
    "                        stripped_line = line.strip()\n",
    "                        \n",
    "                        if line.strip().startswith(('#', '//', '*')):\n",
    "                            continue\n",
    "                        \n",
    "                        for pattern in pattern_list:\n",
    "\n",
    "                            regex_pattern = re.compile(re.escape(pattern))\n",
    "                            if regex_pattern.search(stripped_line):\n",
    "                                result_list.append([folder_name, pattern, file_path, line_num+1, stripped_line])\n",
    "                                print(f'Operator found {pattern} with content: {stripped_line}')\n",
    "\n",
    "    with open(f'./results/{output_file_name}.csv', 'w', newline='', encoding='utf-8') as csvfile:\n",
    "        writer = csv.writer(csvfile)\n",
    "        writer.writerow(['pipeline', 'operator', 'file_path', 'line_number', 'line_content'])\n",
    "        writer.writerows(result_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "9379f32e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\"directory_path = './git_repos'\\n\\ndf_operators = pd.read_excel('./data/all_operator_dataset.xlsx')\\noperator_list = [str(pattern) for pattern in df_operators['operator']]\\n\\noutput_file_name = 'all_operators_in_config'\\n\\nfind_operators_in_config(directory_path, operator_list, output_file_name)\""
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "'''directory_path = './git_repos'\n",
    "\n",
    "df_operators = pd.read_excel('./data/all_operator_dataset.xlsx')\n",
    "operator_list = [str(pattern) for pattern in df_operators['operator']]\n",
    "\n",
    "output_file_name = 'all_operators_in_config'\n",
    "\n",
    "find_operators_in_config(directory_path, operator_list, output_file_name)'''"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1960c103",
   "metadata": {},
   "source": [
    "Bioinformatic tools"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "77bf1de4",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Operator found multiqc with content: skip_multiqc               = false\n",
      "Operator found multiqc with content: multiqc_config             = null\n",
      "Operator found multiqc with content: multiqc_title              = null\n",
      "Operator found multiqc with content: multiqc_logo               = null\n",
      "Operator found multiqc with content: max_multiqc_email_size     = '25.MB'\n",
      "Operator found multiqc with content: multiqc_methods_description = null\n",
      "Operator found fastp with content: path: { \"${params.outdir}/fastp/${meta.id}\" },\n",
      "Operator found fastp with content: path: { \"${params.outdir}/fastp/${meta.id}/\" },\n",
      "Operator found fastp with content: pattern: \"*.fastp.fastq.gz\"\n",
      "Operator found fastqc with content: path: { \"${params.outdir}/fastqc/postassembly\" },\n",
      "Operator found multiqc with content: ext.args   = params.multiqc_title ? \"--title \\\"$params.multiqc_title\\\"\" : ''\n",
      "Operator found multiqc with content: path: { \"${params.outdir}/multiqc\" },\n",
      "Operator found cutadapt with content: cutadapt_min_overlap = 3\n",
      "Operator found cutadapt with content: cutadapt_max_error_rate = 0.1\n",
      "Operator found qiime with content: qiime_adonis_formula = null\n",
      "Operator found cutadapt with content: skip_cutadapt          = false\n",
      "Operator found qiime with content: skip_qiime             = false\n",
      "Operator found qiime with content: skip_qiime_downstream  = false\n",
      "Operator found fastqc with content: skip_fastqc            = false\n",
      "Operator found multiqc with content: skip_multiqc           = false\n",
      "Operator found qiime with content: qiime_ref_taxonomy       = null\n",
      "Operator found qiime with content: qiime_ref_tax_custom     = null\n",
      "Operator found kraken2 with content: kraken2_ref_taxonomy     = null\n",
      "Operator found kraken2 with content: kraken2_assign_taxlevels = null\n",
      "Operator found kraken2 with content: kraken2_ref_tax_custom   = null\n",
      "Operator found kraken2 with content: kraken2_confidence       = 0.0\n",
      "Operator found multiqc with content: multiqc_config             = null\n",
      "Operator found multiqc with content: multiqc_title              = null\n",
      "Operator found multiqc with content: multiqc_logo               = null\n",
      "Operator found multiqc with content: max_multiqc_email_size     = '25.MB'\n",
      "Operator found multiqc with content: multiqc_methods_description = null\n",
      "Operator found qiime with content: validationSchemaIgnoreParams     = 'dada_ref_databases,qiime_ref_databases,sintax_ref_databases,kraken2_ref_databases,genomes,igenomes_base'\n",
      "Operator found kraken2 with content: validationSchemaIgnoreParams     = 'dada_ref_databases,qiime_ref_databases,sintax_ref_databases,kraken2_ref_databases,genomes,igenomes_base'\n",
      "Operator found qiime with content: test_qiimecustom   { includeConfig 'conf/test_qiimecustom.config'   }\n",
      "Operator found cutadapt with content: ext.args = params.skip_cutadapt ? 'cp' : 'ln -s'\n",
      "Operator found cutadapt with content: \"-O ${params.cutadapt_min_overlap}\",\n",
      "Operator found cutadapt with content: \"-e ${params.cutadapt_max_error_rate}\",\n",
      "Operator found cutadapt with content: path: { \"${params.outdir}/cutadapt\" },\n",
      "Operator found cutadapt with content: \"-O ${params.cutadapt_min_overlap}\",\n",
      "Operator found cutadapt with content: \"-e ${params.cutadapt_max_error_rate}\",\n",
      "Operator found cutadapt with content: path: { \"${params.outdir}/cutadapt\" },\n",
      "Operator found cutadapt with content: \"-O ${params.cutadapt_min_overlap}\",\n",
      "Operator found cutadapt with content: \"-e ${params.cutadapt_max_error_rate}\",\n",
      "Operator found cutadapt with content: path: { \"${params.outdir}/cutadapt\" },\n",
      "Operator found cutadapt with content: path: { \"${params.outdir}/cutadapt\" },\n",
      "Operator found cutadapt with content: path: { \"${params.outdir}/cutadapt\" },\n",
      "Operator found kraken2 with content: ext.args = \"--use-names --confidence ${params.kraken2_confidence}\"\n",
      "Operator found kraken2 with content: path: { \"${params.outdir}/kraken2\" },\n",
      "Operator found kraken2 with content: path: { \"${params.outdir}/kraken2\" },\n",
      "Operator found qiime with content: path: { \"${params.outdir}/qiime2/taxonomy\" },\n",
      "Operator found qiime with content: path: { \"${params.outdir}/qiime2/taxonomy\" },\n",
      "Operator found qiime with content: path: { \"${params.outdir}/qiime2/input\" },\n",
      "Operator found qiime with content: path: { \"${params.outdir}/qiime2/input\" },\n",
      "Operator found qiime with content: path: { \"${params.outdir}/qiime2/abundance_tables\" },\n",
      "Operator found qiime with content: path: { \"${params.outdir}/qiime2/representative_sequences\" },\n",
      "Operator found qiime with content: path: { \"${params.outdir}/qiime2/abundance_tables\" },\n",
      "Operator found qiime with content: path: { \"${params.outdir}/qiime2\" },\n",
      "Operator found qiime with content: path: { \"${params.outdir}/qiime2/barplot_average\" },\n",
      "Operator found qiime with content: path: { \"${params.outdir}/qiime2/representative_sequences\" },\n",
      "Operator found qiime with content: path: { \"${params.outdir}/qiime2/abundance_tables\" },\n",
      "Operator found qiime with content: path: { \"${params.outdir}/qiime2/abundance_tables\" },\n",
      "Operator found qiime with content: path: { \"${params.outdir}/qiime2/representative_sequences\" },\n",
      "Operator found qiime with content: path: { \"${params.outdir}/qiime2/rel_abundance_tables\" },\n",
      "Operator found qiime with content: path: { \"${params.outdir}/qiime2/phylogenetic_tree\" },\n",
      "Operator found qiime with content: path: { \"${params.outdir}/qiime2\" },\n",
      "Operator found qiime with content: path: { \"${params.outdir}/qiime2/diversity\" },\n",
      "Operator found qiime with content: path: { \"${params.outdir}/qiime2/diversity/beta_diversity\" },\n",
      "Operator found qiime with content: path: { \"${params.outdir}/qiime2\" },\n",
      "Operator found multiqc with content: ext.args   = { params.multiqc_title ? \"--title \\\"$params.multiqc_title\\\"\" : '' }\n",
      "Operator found multiqc with content: path: { \"${params.outdir}/multiqc\" },\n",
      "Operator found qiime with content: qiime_ref_databases {\n",
      "Operator found qiime with content: file = [ \"https://data.qiime2.org/2023.7/common/silva-138-99-seqs.qza\", \"https://data.qiime2.org/2023.7/common/silva-138-99-tax.qza\" ]\n",
      "Operator found qiime with content: fmtscript = \"taxref_reformat_qiime_silva138.sh\"\n",
      "Operator found qiime with content: file = [ \"https://data.qiime2.org/2023.7/common/silva-138-99-seqs.qza\", \"https://data.qiime2.org/2023.7/common/silva-138-99-tax.qza\" ]\n",
      "Operator found qiime with content: fmtscript = \"taxref_reformat_qiime_silva138.sh\"\n",
      "Operator found qiime with content: fmtscript = \"taxref_reformat_qiime_unite.sh\"\n",
      "Operator found qiime with content: fmtscript = \"taxref_reformat_qiime_unite.sh\"\n",
      "Operator found qiime with content: fmtscript = \"taxref_reformat_qiime_unite.sh\"\n",
      "Operator found qiime with content: fmtscript = \"taxref_reformat_qiime_unite.sh\"\n",
      "Operator found qiime with content: fmtscript = \"taxref_reformat_qiime_unite.sh\"\n",
      "Operator found qiime with content: fmtscript = \"taxref_reformat_qiime_unite.sh\"\n",
      "Operator found qiime with content: fmtscript = \"taxref_reformat_qiime_unite.sh\"\n",
      "Operator found qiime with content: file = [ \"https://data.qiime2.org/2023.7/tutorials/training-feature-classifiers/85_otus.fasta\", \"https://data.qiime2.org/2023.7/tutorials/training-feature-classifiers/85_otu_taxonomy.txt\" ]\n",
      "Operator found qiime with content: fmtscript = \"taxref_reformat_qiime_greengenes85.sh\"\n",
      "Operator found qiime with content: fmtscript = \"taxref_reformat_qiime_greengenes2022.sh\"\n",
      "Operator found qiime with content: fmtscript = \"taxref_reformat_qiime_greengenes2022.sh\"\n",
      "Operator found kraken2 with content: kraken2_ref_databases {\n",
      "Operator found qiime with content: qiime_ref_taxonomy = \"greengenes85\"\n",
      "Operator found qiime with content: qiime_adonis_formula = \"treatment1,mix8\"\n",
      "Operator found kraken2 with content: kraken2_ref_taxonomy = \"greengenes\"\n",
      "Operator found fastqc with content: skip_fastqc = true\n",
      "Operator found fastqc with content: skip_fastqc = true\n",
      "Operator found qiime with content: skip_qiime = true\n",
      "Operator found qiime with content: qiime_ref_taxonomy = \"greengenes85\"\n",
      "Operator found qiime with content: qiime_adonis_formula = \"habitat\"\n",
      "Operator found qiime with content: skip_qiime = true\n",
      "Operator found cutadapt with content: skip_cutadapt = true\n",
      "Operator found cutadapt with content: skip_cutadapt = true\n",
      "Operator found qiime with content: skip_qiime = true\n",
      "Operator found qiime with content: skip_qiime = true\n",
      "Operator found qiime with content: qiime_ref_taxonomy = \"greengenes85\"\n",
      "Operator found fastqc with content: skip_fastqc = true\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Operator found qiime with content: nextflow run nf-core/ampliseq -profile test_qiimecustom,<docker/singularity> --outdir <OUTDIR>\n",
      "Operator found qiime with content: config_profile_description = 'Minimal test dataset to check --qiime_ref_tax_custom'\n",
      "Operator found qiime with content: qiime_ref_tax_custom = \"https://raw.githubusercontent.com/nf-core/test-datasets/ampliseq/testdata/85_greengenes.fna.gz,https://raw.githubusercontent.com/nf-core/test-datasets/ampliseq/testdata/85_greengenes.tax.gz\"\n",
      "Operator found qiime with content: skip_qiime_downstream = true\n",
      "Operator found kraken2 with content: kraken2_ref_tax_custom = \"https://genome-idx.s3.amazonaws.com/kraken/16S_Greengenes13.5_20200326.tgz\"\n",
      "Operator found kraken2 with content: kraken2_assign_taxlevels = \"D,P,C,O\"\n",
      "Operator found qiime with content: qiime_ref_tax_custom = \"https://raw.githubusercontent.com/nf-core/test-datasets/ampliseq/testdata/85_greengenes.tar.gz\"\n",
      "Operator found qiime with content: skip_qiime_downstream = true\n",
      "Operator found qiime with content: skip_qiime = true\n",
      "Operator found qiime with content: qiime_adonis_formula = \"var2\"\n",
      "Operator found bwa with content: aligner                    = 'bwa'\n",
      "Operator found bwa with content: bwa_min_score              = null\n",
      "Operator found fastqc with content: skip_fastqc                = false\n",
      "Operator found picard with content: skip_picard_metrics        = false\n",
      "Operator found multiqc with content: skip_multiqc               = false\n",
      "Operator found multiqc with content: multiqc_config             = null\n",
      "Operator found multiqc with content: multiqc_title              = null\n",
      "Operator found multiqc with content: multiqc_logo               = null\n",
      "Operator found multiqc with content: max_multiqc_email_size     = '25.MB'\n",
      "Operator found multiqc with content: multiqc_methods_description = null\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/CHM13/Sequence/BWAIndex/\"\n",
      "Operator found bwa with content: bwamem2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/CHM13/Sequence/BWAmem2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BismarkIndex/\"\n",
      "Operator found fastqc with content: if (!(params.skip_fastqc || params.skip_qc)) {\n",
      "Operator found fastqc with content: path: { \"${params.outdir}/fastqc\" },\n",
      "Operator found fastqc with content: path: { \"${params.outdir}/fastqc/zips\" },\n",
      "Operator found fastqc with content: '--fastqc',\n",
      "Operator found fastqc with content: path: { \"${params.outdir}/trimgalore/fastqc\" },\n",
      "Operator found fastqc with content: path: { \"${params.outdir}/trimgalore/fastqc/zips\" },\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Operator found samtools with content: path: { \"${params.outdir}/${params.aligner}/library/samtools_stats\" },\n",
      "Operator found bwa with content: if (params.aligner == 'bwa') {\n",
      "Operator found bwa with content: params.bwa_min_score ? \" -T ${params.bwa_min_score}\" : '',\n",
      "Operator found bowtie2 with content: if (params.aligner == 'bowtie2') {\n",
      "Operator found picard with content: path: { \"${params.outdir}/${params.aligner}/merged_library/picard_metrics\" },\n",
      "Operator found samtools with content: path: { \"${params.outdir}/${params.aligner}/merged_library/samtools_stats\" },\n",
      "Operator found samtools with content: path: { \"${params.outdir}/${params.aligner}/merged_library/samtools_stats\" },\n",
      "Operator found samtools with content: path: { \"${params.outdir}/${params.aligner}/merged_library/samtools_stats\" },\n",
      "Operator found picard with content: if (!params.skip_picard_metrics) {\n",
      "Operator found picard with content: path: { \"${params.outdir}/${params.aligner}/merged_library/picard_metrics\" },\n",
      "Operator found picard with content: path: { \"${params.outdir}/${params.aligner}/merged_library/picard_metrics/pdf\" },\n",
      "Operator found macs2 with content: path: { \"${params.outdir}/${params.aligner}/merged_library/macs2/${params.narrow_peak ? '/narrow_peak' : '/broad_peak'}\" },\n",
      "Operator found macs2 with content: path: { \"${params.outdir}/${params.aligner}/merged_library/macs2/${params.narrow_peak ? '/narrow_peak' : '/broad_peak'}/qc\" },\n",
      "Operator found macs2 with content: path: { \"${params.outdir}/${params.aligner}/merged_library/macs2/${params.narrow_peak ? '/narrow_peak' : '/broad_peak'}/qc\" },\n",
      "Operator found macs2 with content: path: { \"${params.outdir}/${params.aligner}/merged_library/macs2/${params.narrow_peak ? '/narrow_peak' : '/broad_peak'}\" },\n",
      "Operator found macs2 with content: ext.args   = '-o ./ -p macs2_peak.mLb.clN'\n",
      "Operator found macs2 with content: path: { \"${params.outdir}/${params.aligner}/merged_library/macs2/${params.narrow_peak ? '/narrow_peak' : '/broad_peak'}/qc\" },\n",
      "Operator found macs2 with content: ext.prefix = 'macs2_annotatePeaks.mLb.clN'\n",
      "Operator found macs2 with content: path: { \"${params.outdir}/${params.aligner}/merged_library/macs2/${params.narrow_peak ? '/narrow_peak' : '/broad_peak'}/qc\" },\n",
      "Operator found macs2 with content: path: { \"${params.outdir}/${params.aligner}/merged_library/macs2/${params.narrow_peak ? '/narrow_peak' : '/broad_peak'}/consensus\" },\n",
      "Operator found macs2 with content: path: { \"${params.outdir}/${params.aligner}/merged_library/macs2/${params.narrow_peak ? '/narrow_peak' : '/broad_peak'}/consensus\" },\n",
      "Operator found macs2 with content: path: { \"${params.outdir}/${params.aligner}/merged_library/macs2/${params.narrow_peak ? '/narrow_peak' : '/broad_peak'}/consensus/deseq2\" },\n",
      "Operator found macs2 with content: path: { \"${params.outdir}/${params.aligner}/merged_library/macs2/${params.narrow_peak ? '/narrow_peak' : '/broad_peak'}/consensus\" },\n",
      "Operator found picard with content: path: { \"${params.outdir}/${params.aligner}/merged_replicate/picard_metrics\" },\n",
      "Operator found samtools with content: path: { \"${params.outdir}/${params.aligner}/merged_replicate/samtools_stats\" },\n",
      "Operator found macs2 with content: path: { \"${params.outdir}/${params.aligner}/merged_replicate/macs2/${params.narrow_peak ? '/narrow_peak' : '/broad_peak'}\" },\n",
      "Operator found macs2 with content: path: { \"${params.outdir}/${params.aligner}/merged_replicate/macs2/${params.narrow_peak ? '/narrow_peak' : '/broad_peak'}/qc\" },\n",
      "Operator found macs2 with content: path: { \"${params.outdir}/${params.aligner}/merged_replicate/macs2/${params.narrow_peak ? '/narrow_peak' : '/broad_peak'}/qc\" },\n",
      "Operator found macs2 with content: path: { \"${params.outdir}/${params.aligner}/merged_replicate/macs2/${params.narrow_peak ? '/narrow_peak' : '/broad_peak'}\" },\n",
      "Operator found macs2 with content: ext.args   = '-o ./ -p macs2_peak.mRp.clN'\n",
      "Operator found macs2 with content: path: { \"${params.outdir}/${params.aligner}/merged_replicate/macs2/${params.narrow_peak ? '/narrow_peak' : '/broad_peak'}/qc\" },\n",
      "Operator found macs2 with content: ext.prefix = 'macs2_annotatePeaks.mRp.clN'\n",
      "Operator found macs2 with content: path: { \"${params.outdir}/${params.aligner}/merged_replicate/macs2/${params.narrow_peak ? '/narrow_peak' : '/broad_peak'}/qc\" },\n",
      "Operator found macs2 with content: path: { \"${params.outdir}/${params.aligner}/merged_replicate/macs2/${params.narrow_peak ? '/narrow_peak' : '/broad_peak'}/consensus\" },\n",
      "Operator found macs2 with content: path: { \"${params.outdir}/${params.aligner}/merged_replicate/macs2/${params.narrow_peak ? '/narrow_peak' : '/broad_peak'}/consensus\" },\n",
      "Operator found macs2 with content: path: { \"${params.outdir}/${params.aligner}/merged_replicate/macs2/${params.narrow_peak ? '/narrow_peak' : '/broad_peak'}/consensus/deseq2\" },\n",
      "Operator found macs2 with content: path: { \"${params.outdir}/${params.aligner}/merged_replicate/macs2/${params.narrow_peak ? '/narrow_peak' : '/broad_peak'}/consensus\" },\n",
      "Operator found multiqc with content: if (!params.skip_multiqc) {\n",
      "Operator found multiqc with content: ext.args   = params.multiqc_title ? \"--title \\\"$params.multiqc_title\\\"\" : ''\n",
      "Operator found multiqc with content: path: { \"${params.outdir}/multiqc/${params.narrow_peak ? '/narrow_peak' : '/broad_peak'}\" },\n",
      "Operator found kraken2 with content: kraken2db                       = \"\"\n",
      "Operator found fastqc with content: skip_fastqc                     = false\n",
      "Operator found fastp with content: skip_fastp                      = false\n",
      "Operator found kraken2 with content: skip_kraken2                    = false\n",
      "Operator found multiqc with content: skip_multiqc                    = false\n",
      "Operator found multiqc with content: multiqc_config                  = null\n",
      "Operator found multiqc with content: multiqc_title                   = null\n",
      "Operator found multiqc with content: multiqc_logo                    = null\n",
      "Operator found multiqc with content: max_multiqc_email_size          = '25.MB'\n",
      "Operator found multiqc with content: multiqc_methods_description     = null\n",
      "Operator found NanoPlot with content: path: { \"${params.outdir}/QC_longreads/NanoPlot\" },\n",
      "Operator found kraken2 with content: path: { \"${params.outdir}/kraken2\" },\n",
      "Operator found multiqc with content: path: { \"${params.outdir}/multiqc\" },\n",
      "Operator found fastqc with content: if (!params.skip_fastqc) {\n",
      "Operator found fastp with content: if (!params.skip_fastp) {\n",
      "Operator found fastp with content: pattern: \"*.fastp.fastq.gz\",\n",
      "Operator found fastqc with content: if (!params.skip_fastqc) {\n",
      "Operator found kraken2 with content: skip_kraken2 = true\n",
      "Operator found kraken2 with content: skip_kraken2 = true\n",
      "Operator found kraken2 with content: kraken2db = 'https://genome-idx.s3.amazonaws.com/kraken/k2_standard_8gb_20210517.tar.gz'\n",
      "Operator found kraken2 with content: skip_kraken2 = true\n",
      "Operator found kraken2 with content: skip_kraken2 = true\n",
      "Operator found kraken2 with content: kraken2db = \"https://genome-idx.s3.amazonaws.com/kraken/16S_Greengenes13.5_20200326.tgz\"\n",
      "Operator found multiqc with content: multiqc_config             = null\n",
      "Operator found multiqc with content: multiqc_title              = null\n",
      "Operator found multiqc with content: max_multiqc_email_size            = '25.MB'\n",
      "Operator found multiqc with content: skip_multiqc                      = false\n",
      "Operator found fastp with content: 'fastp' {\n",
      "Operator found bwa with content: 'bwa_index' {\n",
      "Operator found bwa with content: 'bwa_mem' {\n",
      "Operator found samtools with content: args2 = '-F 4' // samtools view options discarding unmapped reads\n",
      "Operator found samtools with content: 'samtools_sort' {\n",
      "Operator found samtools with content: 'samtools_index' {\n",
      "Operator found bcftools with content: 'bcftools_mpileup' {\n",
      "Operator found bcftools with content: 'bcftools_filter' {\n",
      "Operator found multiqc with content: 'multiqc' {\n",
      "Operator found multiqc with content: publish_dir     = 'multiqc'\n",
      "Operator found bcftools with content: 'bcftools_filter' {\n",
      "Operator found bcftools with content: 'bcftools_filter' {\n",
      "Operator found samtools with content: samtools_collate_fast      = false\n",
      "Operator found multiqc with content: multiqc_config             = null\n",
      "Operator found multiqc with content: multiqc_title              = null\n",
      "Operator found multiqc with content: multiqc_logo               = null\n",
      "Operator found multiqc with content: max_multiqc_email_size     = '25.MB'\n",
      "Operator found multiqc with content: multiqc_methods_description = null\n",
      "Operator found samtools with content: ext.args         = { params.samtools_collate_fast ? \"-f -r \" + params.reads_in_memory : \"\" }\n",
      "Operator found samtools with content: ext.args         = { params.samtools_collate_fast ? \"-f -r \" + params.reads_in_memory : \"\" }\n",
      "Operator found samtools with content: ext.args          = { params.samtools_collate_fast ? \"-f -r \" + params.reads_in_memory : \"\" }\n",
      "Operator found samtools with content: path: { \"${params.outdir}/samtools\" },\n",
      "Operator found samtools with content: path: { \"${params.outdir}/samtools\" },\n",
      "Operator found samtools with content: path: { \"${params.outdir}/samtools\" },\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Operator found samtools with content: samtools_collate_fast = true\n",
      "Operator found fastqc with content: skip_initial_fastqc = false\n",
      "Operator found fastqc with content: skip_trimming_fastqc = false\n",
      "Operator found samtools with content: skip_samtools_stats = false\n",
      "Operator found multiqc with content: multiqc_config = false\n",
      "Operator found multiqc with content: max_multiqc_email_size = 25.MB\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: aligner                     = 'bwamem2'\n",
      "Operator found multiqc with content: multiqc_config              = null\n",
      "Operator found multiqc with content: multiqc_title               = null\n",
      "Operator found multiqc with content: multiqc_logo                = null\n",
      "Operator found multiqc with content: max_multiqc_email_size      = '25.MB'\n",
      "Operator found multiqc with content: multiqc_methods_description = null\n",
      "Operator found bwa with content: aligner                    = 'bwamem2'\n",
      "Operator found bwa with content: aligner                    = 'bwamem2'\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/CHM13/Sequence/BWAIndex/\"\n",
      "Operator found bwa with content: bwamem2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/CHM13/Sequence/BWAmem2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BismarkIndex/\"\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BismarkIndex/\"\n",
      "Operator found multiqc with content: path: { \"${params.outdir}/multiqc\" },\n",
      "Operator found bwa with content: aligner                     = 'bwamem2'\n",
      "Operator found bwa with content: aligner                    = 'bwa'\n",
      "Operator found bwa with content: aligner                    = 'bwamem2'\n",
      "Operator found bwa with content: aligner                    = 'bwa'\n",
      "Operator found bwa with content: bwa_min_score              = null\n",
      "Operator found fastqc with content: skip_fastqc                = false\n",
      "Operator found picard with content: skip_picard_metrics        = false\n",
      "Operator found multiqc with content: skip_multiqc               = false\n",
      "Operator found multiqc with content: multiqc_config             = null\n",
      "Operator found multiqc with content: multiqc_title              = null\n",
      "Operator found multiqc with content: max_multiqc_email_size     = '25.MB'\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/Bowtie2Index/\"\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BismarkIndex/\"\n",
      "Operator found fastqc with content: if (!(params.skip_fastqc || params.skip_qc)) {\n",
      "Operator found fastqc with content: path: { \"${params.outdir}/fastqc\" },\n",
      "Operator found fastqc with content: path: { \"${params.outdir}/fastqc/zips\" },\n",
      "Operator found fastqc with content: '--fastqc',\n",
      "Operator found fastqc with content: path: { \"${params.outdir}/trimgalore/fastqc\" },\n",
      "Operator found fastqc with content: path: { \"${params.outdir}/trimgalore/fastqc/zips\" },\n",
      "Operator found samtools with content: path: { \"${params.outdir}/${params.aligner}/library/samtools_stats/\" },\n",
      "Operator found bwa with content: if (params.aligner == 'bwa') {\n",
      "Operator found bwa with content: params.bwa_min_score ? \" -T ${params.bwa_min_score}\" : '',\n",
      "Operator found bowtie2 with content: if (params.aligner == 'bowtie2') {\n",
      "Operator found picard with content: path: { \"${params.outdir}/${params.aligner}/mergedLibrary/picard_metrics\" },\n",
      "Operator found samtools with content: path: { \"${params.outdir}/${params.aligner}/mergedLibrary/samtools_stats\" },\n",
      "Operator found samtools with content: path: { \"${params.outdir}/${params.aligner}/mergedLibrary/samtools_stats\" },\n",
      "Operator found picard with content: if (!params.skip_picard_metrics) {\n",
      "Operator found picard with content: path: { \"${params.outdir}/${params.aligner}/mergedLibrary/picard_metrics\" },\n",
      "Operator found picard with content: path: { \"${params.outdir}/${params.aligner}/mergedLibrary/picard_metrics/pdf\" },\n",
      "Operator found macs2 with content: \"${params.outdir}/${params.aligner}/mergedLibrary/macs2\",\n",
      "Operator found macs2 with content: \"${params.outdir}/${params.aligner}/mergedLibrary/macs2\",\n",
      "Operator found macs2 with content: \"${params.outdir}/${params.aligner}/mergedLibrary/macs2\",\n",
      "Operator found macs2 with content: \"${params.outdir}/${params.aligner}/mergedLibrary/macs2\",\n",
      "Operator found macs2 with content: ext.args   = '-o ./ -p macs2_peak'\n",
      "Operator found macs2 with content: \"${params.outdir}/${params.aligner}/mergedLibrary/macs2\",\n",
      "Operator found macs2 with content: ext.prefix = 'macs2_annotatePeaks'\n",
      "Operator found macs2 with content: \"${params.outdir}/${params.aligner}/mergedLibrary/macs2\",\n",
      "Operator found macs2 with content: \"${params.outdir}/${params.aligner}/mergedLibrary/macs2\",\n",
      "Operator found macs2 with content: \"${params.outdir}/${params.aligner}/mergedLibrary/macs2\",\n",
      "Operator found macs2 with content: \"${params.outdir}/${params.aligner}/mergedLibrary/macs2\",\n",
      "Operator found macs2 with content: \"${params.outdir}/${params.aligner}/mergedLibrary/macs2\",\n",
      "Operator found macs2 with content: \"${params.outdir}/${params.aligner}/mergedLibrary/macs2\",\n",
      "Operator found multiqc with content: if (!params.skip_multiqc) {\n",
      "Operator found multiqc with content: ext.args   = params.multiqc_title ? \"--title \\\"$params.multiqc_title\\\"\" : ''\n",
      "Operator found multiqc with content: \"${params.outdir}/multiqc\",\n",
      "Operator found bwa with content: bwa_index                   = null\n",
      "Operator found multiqc with content: skip_multiqc               = false\n",
      "Operator found multiqc with content: multiqc_config             = null\n",
      "Operator found multiqc with content: multiqc_title              = null\n",
      "Operator found multiqc with content: multiqc_logo               = null\n",
      "Operator found multiqc with content: max_multiqc_email_size     = '25.MB'\n",
      "Operator found multiqc with content: multiqc_methods_description = null\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/CHM13/Sequence/BWAIndex/\"\n",
      "Operator found bwa with content: bwamem2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/CHM13/Sequence/BWAmem2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BismarkIndex/\"\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BismarkIndex/\"\n",
      "Operator found fastqc with content: path: { \"${params.outdir}/reports/fastqc\" },\n",
      "Operator found fastqc with content: '--fastqc',\n",
      "Operator found fastqc with content: path: { \"${params.outdir}/reports/trimgalore_fastqc\" },\n",
      "Operator found bwa with content: path: { \"${params.outdir}/bwa\" },\n",
      "Operator found bwa with content: path: { \"${params.outdir}/bwa\" },\n",
      "Operator found bwa with content: path: { \"${params.outdir}/bwa\" },\n",
      "Operator found samtools with content: path: { \"${params.outdir}/reports/samtools_stats/${meta.id}\" },\n",
      "Operator found samtools with content: path: { \"${params.outdir}/reports/samtools_stats/${meta.id}/md/\" },\n",
      "Operator found samtools with content: path: { \"${params.outdir}/circlefinder/samtools\" },\n",
      "Operator found samtools with content: path: { \"${params.outdir}/circlemap/samtools\" },\n",
      "Operator found seqtk with content: path: { \"${params.outdir}/unicycler/seqtk\" },\n",
      "Operator found minimap2 with content: path: { \"${params.outdir}/unicycler/minimap2\" },\n",
      "Operator found multiqc with content: if (!params.skip_multiqc) {\n",
      "Operator found multiqc with content: ext.args   = params.multiqc_title ? \"--title \\\"$params.multiqc_title\\\"\" : ''\n",
      "Operator found multiqc with content: path: { \"${params.outdir}/multiqc\" },\n",
      "Operator found hisat2 with content: hisat2                     = null\n",
      "Operator found hisat2 with content: hisat2_build_memory        = '200.GB'\n",
      "Operator found fastqc with content: skip_fastqc                = false\n",
      "Operator found multiqc with content: multiqc_config             = null\n",
      "Operator found multiqc with content: multiqc_title              = null\n",
      "Operator found multiqc with content: multiqc_logo               = null\n",
      "Operator found multiqc with content: max_multiqc_email_size     = '25.MB'\n",
      "Operator found multiqc with content: multiqc_methods_description = null\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/Bowtie2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/Bowtie2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/CHM13/Sequence/BWAIndex/\"\n",
      "Operator found bwa with content: bwamem2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/CHM13/Sequence/BWAmem2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/CHM13/Sequence/BWAIndex/\"\n",
      "Operator found bwa with content: bwamem2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/CHM13/Sequence/BWAmem2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/Bowtie2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/Bowtie2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/Bowtie2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BWAIndex/version0.6.0/\"\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/Bowtie2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/Bowtie2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/Bowtie2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/Bowtie2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/Bowtie2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/Bowtie2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/Bowtie2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/Bowtie2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/Bowtie2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/Bowtie2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/Bowtie2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/Bowtie2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/Bowtie2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/Bowtie2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/Bowtie2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/Bowtie2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/Bowtie2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/Bowtie2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/Bowtie2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/Bowtie2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/Bowtie2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/Bowtie2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/Bowtie2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/Bowtie2Index/\"\n",
      "Operator found fastqc with content: \"--fastqc_args '-t ${task.cpus}' \",\n",
      "Operator found fastqc with content: path: { \"${params.outdir}/quality_control/trimgalore/fastqc\" },\n",
      "Operator found bowtie2 with content: ext.when = { params.fasta && !params.bowtie2 && params.tool.split(',').contains('find_circ') && params.module.split(',').contains('circrna_discovery') }\n",
      "Operator found bwa with content: ext.when = { params.fasta && !params.bwa && params.tool.split(',').contains('ciriquant') && params.module.split(',').contains('circrna_discovery') }\n",
      "Operator found hisat2 with content: path: { \"${params.outdir}/references/index/hisat2\" },\n",
      "Operator found hisat2 with content: path: { \"${params.outdir}/differential_expression/intermediates/hisat2/\" },\n",
      "Operator found hisat2 with content: path: { \"${params.outdir}/differential_expression/intermediates/hisat2/\" },\n",
      "Operator found stringtie with content: path: { \"${params.outdir}/differential_expression/intermediates/stringtie/\" },\n",
      "Operator found multiqc with content: multiqc_config = null\n",
      "Operator found multiqc with content: max_multiqc_email_size = '25.MB'\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BismarkIndex/\"\n",
      "Operator found multiqc with content: multiqc_config = \"$baseDir/assets/multiqc_config.yaml\"\n",
      "Operator found multiqc with content: max_multiqc_email_size = 25.MB\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BismarkIndex/\"\n",
      "Operator found multiqc with content: multiqc_config             = null\n",
      "Operator found multiqc with content: multiqc_title              = null\n",
      "Operator found multiqc with content: multiqc_logo               = null\n",
      "Operator found multiqc with content: max_multiqc_email_size     = '25.MB'\n",
      "Operator found multiqc with content: multiqc_methods_description = null\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BismarkIndex/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/Bowtie2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BismarkIndex/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/Bowtie2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BismarkIndex/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/Bowtie2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BismarkIndex/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/Bowtie2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BismarkIndex/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/Bowtie2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BismarkIndex/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/Bowtie2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BismarkIndex/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BismarkIndex/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/Bowtie2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BismarkIndex/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/Bowtie2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bismark with content: bismark          = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BismarkIndex/\"\n",
      "Operator found bowtie2 with content: bowtie2          = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/Bowtie2Index/\"\n",
      "Operator found bwa with content: bwa              = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BismarkIndex/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/Bowtie2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BismarkIndex/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BismarkIndex/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/Bowtie2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BismarkIndex/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/Bowtie2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BismarkIndex/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BismarkIndex/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BismarkIndex/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/Bowtie2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BismarkIndex/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/Bowtie2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BismarkIndex/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/Bowtie2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BismarkIndex/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/Bowtie2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bwa with content: bwa              = \"${params.igenomes_base}/Homo_sapiens/GATK/GRCh37/Sequence/BWAIndex/\"\n",
      "Operator found bwa with content: bwa              = \"${params.igenomes_base}/Homo_sapiens/GATK/GRCh38/Sequence/BWAIndex/\"\n",
      "Operator found bwa with content: bwamem2          = \"${params.igenomes_base}/Homo_sapiens/GATK/GRCh38/Sequence/BWAmem2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BismarkIndex/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/Bowtie2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BismarkIndex/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/Bowtie2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BismarkIndex/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/Bowtie2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BismarkIndex/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/Bowtie2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/CHM13/Sequence/BWAIndex/\"\n",
      "Operator found bwa with content: bwamem2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/CHM13/Sequence/BWAmem2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BismarkIndex/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/Bowtie2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BismarkIndex/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/Bowtie2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BismarkIndex/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BismarkIndex/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/Bowtie2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BismarkIndex/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/Bowtie2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BismarkIndex/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/Bowtie2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BismarkIndex/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/Bowtie2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BismarkIndex/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/Bowtie2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BismarkIndex/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/Bowtie2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BismarkIndex/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/Bowtie2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BWAIndex/version0.6.0/\"\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Operator found gatk with content: ploidy_priors     = \"https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/genomics/homo_sapiens/illumina/gatk/contig_ploidy_priors_table.tsv\"\n",
      "Operator found multiqc with content: ext.args   = { params.multiqc_title ? \"--title \\\"$params.multiqc_title\\\"\" : '' }\n",
      "Operator found multiqc with content: path: { \"${params.outdir}/multiqc\" },\n",
      "Operator found multiqc with content: multiqc_config             = null\n",
      "Operator found multiqc with content: multiqc_title              = null\n",
      "Operator found multiqc with content: multiqc_logo               = null\n",
      "Operator found multiqc with content: max_multiqc_email_size     = '25.MB'\n",
      "Operator found multiqc with content: multiqc_methods_description = null\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/CHM13/Sequence/BWAIndex/\"\n",
      "Operator found bwa with content: bwamem2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/CHM13/Sequence/BWAmem2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BismarkIndex/\"\n",
      "Operator found multiqc with content: ext.args   = { params.multiqc_title ? \"--title \\\"$params.multiqc_title\\\"\" : '' }\n",
      "Operator found multiqc with content: path: { \"${params.outdir}/multiqc\" },\n",
      "Operator found minimap2 with content: aligner                    = 'minimap2'\n",
      "Operator found cutadapt with content: cutadapt                   = null\n",
      "Operator found multiqc with content: multiqc_config             = null\n",
      "Operator found multiqc with content: multiqc_title              = null\n",
      "Operator found multiqc with content: multiqc_logo               = null\n",
      "Operator found multiqc with content: max_multiqc_email_size     = '25.MB'\n",
      "Operator found multiqc with content: multiqc_methods_description = null\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/CHM13/Sequence/BWAIndex/\"\n",
      "Operator found bwa with content: bwamem2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/CHM13/Sequence/BWAmem2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/Bowtie2Index/\"\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BismarkIndex/\"\n",
      "Operator found cutadapt with content: if(params.analysis == 'screening' && params.cutadapt) {\n",
      "Operator found cutadapt with content: ext.args = \"-a ${params.cutadapt}\"\n",
      "Operator found minimap2 with content: path: { \"${params.outdir}/minimap2_umi\" },\n",
      "Operator found minimap2 with content: path: { \"${params.outdir}/minimap2_umi\" },\n",
      "Operator found multiqc with content: ext.args   = params.multiqc_title ? \"--title \\\"$params.multiqc_title\\\"\" : ''\n",
      "Operator found multiqc with content: path: { \"${params.outdir}/multiqc\" },\n",
      "Operator found minimap2 with content: aligner = 'minimap2'\n",
      "Operator found minimap2 with content: aligner = 'minimap2'\n",
      "Operator found minimap2 with content: aligner = 'minimap2'\n",
      "Operator found fastqc with content: withName: fastqc {\n",
      "Operator found multiqc with content: withName: multiqc {\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Operator found fastqc with content: skip_fastqc                = false\n",
      "Operator found bowtie2 with content: aligner                    = \"bowtie2\"\n",
      "Operator found macs2 with content: macs2_pvalue               = null\n",
      "Operator found macs2 with content: macs2_qvalue               = 0.01\n",
      "Operator found macs2 with content: macs2_narrow_peak          = true\n",
      "Operator found macs2 with content: macs2_broad_cutoff         = 0.1\n",
      "Operator found multiqc with content: skip_multiqc               = false\n",
      "Operator found multiqc with content: multiqc_config                    = null\n",
      "Operator found multiqc with content: multiqc_title                     = null\n",
      "Operator found multiqc with content: multiqc_logo                      = null\n",
      "Operator found multiqc with content: multiqc_methods_description       = null\n",
      "Operator found multiqc with content: max_multiqc_email_size            = \"25.MB\"\n",
      "Operator found fastqc with content: validationSchemaIgnoreParams     = 'igenomes_base,genomes,callers,dedup_control_only,fragment_size,run_igv,run_multiqc,run_reporting,run_consensus_all,run_peak_calling,run_remove_dups,run_remove_linear_dups,run_mark_dups,run_read_filter,run_alignment,run_trim_galore_fastqc,run_cat_fastq,run_input_check,run_genome_prep,run_peak_qc,run_deeptools_qc,run_deeptools_heatmaps,run_preseq'\n",
      "Operator found multiqc with content: validationSchemaIgnoreParams     = 'igenomes_base,genomes,callers,dedup_control_only,fragment_size,run_igv,run_multiqc,run_reporting,run_consensus_all,run_peak_calling,run_remove_dups,run_remove_linear_dups,run_mark_dups,run_read_filter,run_alignment,run_trim_galore_fastqc,run_cat_fastq,run_input_check,run_genome_prep,run_peak_qc,run_deeptools_qc,run_deeptools_heatmaps,run_preseq'\n",
      "Operator found trim_galore with content: validationSchemaIgnoreParams     = 'igenomes_base,genomes,callers,dedup_control_only,fragment_size,run_igv,run_multiqc,run_reporting,run_consensus_all,run_peak_calling,run_remove_dups,run_remove_linear_dups,run_mark_dups,run_read_filter,run_alignment,run_trim_galore_fastqc,run_cat_fastq,run_input_check,run_genome_prep,run_peak_qc,run_deeptools_qc,run_deeptools_heatmaps,run_preseq'\n",
      "Operator found fastqc with content: run_trim_galore_fastqc = true\n",
      "Operator found trim_galore with content: run_trim_galore_fastqc = true\n",
      "Operator found multiqc with content: run_multiqc            = true\n",
      "Operator found multiqc with content: params.run_multiqc       = false\n",
      "Operator found fastqc with content: params.run_trim_galore_fastqc = false\n",
      "Operator found trim_galore with content: params.run_trim_galore_fastqc = false\n",
      "Operator found multiqc with content: params.run_multiqc            = false\n",
      "Operator found fastqc with content: params.run_trim_galore_fastqc = false\n",
      "Operator found trim_galore with content: params.run_trim_galore_fastqc = false\n",
      "Operator found multiqc with content: params.run_multiqc            = false\n",
      "Operator found multiqc with content: params.run_multiqc      = true\n",
      "Operator found multiqc with content: params.run_multiqc      = true\n",
      "Operator found multiqc with content: params.run_multiqc      = true\n",
      "Operator found multiqc with content: params.run_multiqc   = true\n",
      "Operator found multiqc with content: if(params.skip_multiqc) { params.run_multiqc = false }\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/CHM13/Sequence/BWAIndex/\"\n",
      "Operator found bwa with content: bwamem2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/CHM13/Sequence/BWAmem2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/CHM13/Sequence/BWAIndex/\"\n",
      "Operator found bwa with content: bwamem2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/CHM13/Sequence/BWAmem2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Escherichia_coli_K_12_MG1655/NCBI/2001-10-15/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Escherichia_coli_K_12_MG1655/NCBI/2001-10-15/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Escherichia_coli_K_12_MG1655/NCBI/2001-10-15/Sequence/BismarkIndex/\"\n",
      "Operator found fastqc with content: if(params.run_trim_galore_fastqc && !params.skip_fastqc) {\n",
      "Operator found trim_galore with content: if(params.run_trim_galore_fastqc && !params.skip_fastqc) {\n",
      "Operator found fastqc with content: path: { \"${params.outdir}/01_prealign/pretrim_fastqc\" },\n",
      "Operator found fastqc with content: if(params.run_trim_galore_fastqc && !params.skip_trimming) {\n",
      "Operator found trim_galore with content: if(params.run_trim_galore_fastqc && !params.skip_trimming) {\n",
      "Operator found fastqc with content: '--fastqc',\n",
      "Operator found fastqc with content: path: { \"${params.outdir}/01_prealign/trimgalore/fastqc\" },\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Operator found picard with content: ext.suffix = \"_meta_picard_dups\"\n",
      "Operator found macs2 with content: if(params.run_peak_calling && 'macs2' in params.callers) {\n",
      "Operator found macs2 with content: params.macs2_pvalue ? \"-p ${params.macs2_pvalue}\" : \"-q ${params.macs2_qvalue}\",\n",
      "Operator found macs2 with content: params.macs2_narrow_peak ? '' : \"--broad --broad-cutoff ${params.macs2_broad_cutoff}\"\n",
      "Operator found macs2 with content: ext.prefix = { \"${meta.id}.macs2\" }\n",
      "Operator found macs2 with content: path: { \"${params.outdir}/03_peak_calling/04_called_peaks/macs2\" },\n",
      "Operator found macs2 with content: if(params.run_peak_calling && 'macs2' in params.callers) {\n",
      "Operator found macs2 with content: ext.suffix  = \".macs2.peaks\"\n",
      "Operator found macs2 with content: path: { \"${params.outdir}/03_peak_calling/04_called_peaks/macs2\" },\n",
      "Operator found multiqc with content: if (params.run_multiqc) {\n",
      "Operator found multiqc with content: ext.args   = params.multiqc_title ? \"-v --title \\\"$params.multiqc_title\\\"\" : '-v'\n",
      "Operator found multiqc with content: path: { \"${params.outdir}/04_reporting/multiqc\" },\n",
      "Operator found bowtie2 with content: bowtie2                    = 'https://raw.githubusercontent.com/nf-core/test-datasets/cutandrun/reference/genomes/hg38-chr20-bowtie2.tar.gz'\n",
      "Operator found bowtie2 with content: spikein_bowtie2            = 'https://raw.githubusercontent.com/nf-core/test-datasets/cutandrun/reference/genomes/e_coli_U00096_3.tar.gz'\n",
      "Operator found bowtie2 with content: bowtie2            = 'https://raw.githubusercontent.com/nf-core/test-datasets/cutandrun/reference/genomes/hg38-chr20-bowtie2.tar.gz'\n",
      "Operator found bowtie2 with content: spikein_bowtie2    = 'https://raw.githubusercontent.com/nf-core/test-datasets/cutandrun/reference/genomes/e_coli_U00096_3.tar.gz'\n",
      "Operator found bowtie2 with content: bowtie2            = 'https://raw.githubusercontent.com/nf-core/test-datasets/cutandrun/reference/genomes/hg38-chr20-bowtie2.tar.gz'\n",
      "Operator found bowtie2 with content: spikein_bowtie2    = 'https://raw.githubusercontent.com/nf-core/test-datasets/cutandrun/reference/genomes/e_coli_U00096_3.tar.gz'\n",
      "Operator found bowtie2 with content: bowtie2            = '/home/runner/work/cutandrun/cutandrun/hg38-chr20-bowtie2.tar.gz'\n",
      "Operator found bowtie2 with content: spikein_bowtie2    = '/home/runner/work/cutandrun/cutandrun/e_coli_U00096_3.tar.gz'\n",
      "Operator found bowtie2 with content: bowtie2            = '/home/runner/work/cutandrun/cutandrun/hg38-chr20-bowtie2.tar.gz'\n",
      "Operator found bowtie2 with content: spikein_bowtie2    = '/home/runner/work/cutandrun/cutandrun/e_coli_U00096_3.tar.gz'\n",
      "Operator found bowtie2 with content: bowtie2            = 'https://raw.githubusercontent.com/nf-core/test-datasets/cutandrun/reference/genomes/hg38-chr20-bowtie2.tar.gz'\n",
      "Operator found bowtie2 with content: spikein_bowtie2    = 'https://raw.githubusercontent.com/nf-core/test-datasets/cutandrun/reference/genomes/e_coli_U00096_3.tar.gz'\n",
      "Operator found bowtie2 with content: bowtie2            = 'https://raw.githubusercontent.com/nf-core/test-datasets/cutandrun/reference/genomes/hg38-chr20-bowtie2.tar.gz'\n",
      "Operator found bowtie2 with content: spikein_bowtie2    = 'https://raw.githubusercontent.com/nf-core/test-datasets/cutandrun/reference/genomes/e_coli_U00096_3.tar.gz'\n",
      "Operator found fastqc with content: skip_tools                  = []            // list [fastp, fastqc]\n",
      "Operator found fastp with content: skip_tools                  = []            // list [fastp, fastqc]\n",
      "Operator found multiqc with content: multiqc_config             = null\n",
      "Operator found multiqc with content: multiqc_title              = null\n",
      "Operator found multiqc with content: multiqc_logo               = null\n",
      "Operator found multiqc with content: max_multiqc_email_size     = '25.MB'\n",
      "Operator found multiqc with content: multiqc_methods_description = null\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/CHM13/Sequence/BWAIndex/\"\n",
      "Operator found bwa with content: bwamem2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/CHM13/Sequence/BWAmem2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BismarkIndex/\"\n",
      "Operator found multiqc with content: ext.args   = params.multiqc_title ? \"--title \\\"$params.multiqc_title\\\"\" : ''\n",
      "Operator found multiqc with content: path: { \"${params.outdir}/multiqc\" },\n",
      "Operator found multiqc with content: multiqc_config = \"$baseDir/assets/multiqc_config.yaml\"\n",
      "Operator found kraken2 with content: filter_with_kraken2        = (params.enable_filter && params.skip_blastn) ? true : false\n",
      "Operator found fastqc with content: save_fastqc                = false\n",
      "Operator found fastp with content: fastp_save_trimmed_fail    = false\n",
      "Operator found fastp with content: fastp_qualified_quality    = 0\n",
      "Operator found fastp with content: fastp_cut_mean_quality     = 15\n",
      "Operator found kraken2 with content: kraken2db                  = 'https://genome-idx.s3.amazonaws.com/kraken/k2_standard_08gb_20231009.tar.gz'\n",
      "Operator found kraken2 with content: kraken2confidence          = 0.05\n",
      "Operator found multiqc with content: multiqc_config             = null\n",
      "Operator found multiqc with content: multiqc_title              = null\n",
      "Operator found multiqc with content: multiqc_logo               = null\n",
      "Operator found multiqc with content: max_multiqc_email_size     = '25.MB'\n",
      "Operator found multiqc with content: multiqc_methods_description = null\n",
      "Operator found kraken2 with content: description     = \"\"\"A pipeline to identify (and remove) certain sequences from raw genomic data using kraken2 and blast. Default taxon to identify (and remove) is Homo. Removal is optional.\"\"\"\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/CHM13/Sequence/BWAIndex/\"\n",
      "Operator found bwa with content: bwamem2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/CHM13/Sequence/BWAmem2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BismarkIndex/\"\n",
      "Operator found fastp with content: \"-q ${params.fastp_qualified_quality}\",\n",
      "Operator found fastp with content: \"--cut_mean_quality ${params.fastp_cut_mean_quality}\",\n",
      "Operator found fastp with content: path: { \"${params.outdir}/fastp/${meta.id}\" },\n",
      "Operator found fastp with content: path: { \"${params.outdir}/fastp/${meta.id}\" },\n",
      "Operator found fastqc with content: enabled: params.save_fastqc\n",
      "Operator found kraken2 with content: \"--confidence ${params.kraken2confidence}\",\n",
      "Operator found kraken2 with content: path: { \"${params.outdir}/kraken2\" },\n",
      "Operator found kraken2 with content: pattern: '*kraken2*',\n",
      "Operator found kraken2 with content: path: { \"${params.outdir}/kraken2/taxonomy\" },\n",
      "Operator found kraken2 with content: path: { \"${params.outdir}/kraken2/isolated\" },\n",
      "Operator found kraken2 with content: path: { \"${params.outdir}/kraken2/summary\" },\n",
      "Operator found kraken2 with content: pattern: '*.kraken2_summary.tsv',\n",
      "Operator found multiqc with content: ext.args   = { params.multiqc_title ? \"--title \\\"$params.multiqc_title\\\"\" : '' }\n",
      "Operator found multiqc with content: path: { \"${params.outdir}/multiqc\" },\n",
      "Operator found kraken2 with content: kraken2db           = \"https://github.com/nf-core/test-datasets/raw/mag/test_data/minigut_kraken.tgz\"\n",
      "Operator found kraken2 with content: kraken2confidence   = 0.00\n",
      "Operator found kraken2 with content: kraken2db           = \"https://genome-idx.s3.amazonaws.com/kraken/k2_standard_08gb_20231009.tar.gz\"\n",
      "Operator found kraken2 with content: kraken2confidence   = 0.00\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BismarkIndex/\"\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/CHM13/Sequence/BWAIndex/\"\n",
      "Operator found bwa with content: bwamem2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/CHM13/Sequence/BWAmem2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BismarkIndex/\"\n",
      "Operator found salmon with content: matrix = 'https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/genomics/mus_musculus/rnaseq_expression/SRP254919.salmon.merged.gene_counts.top1000cov.tsv'\n",
      "Operator found salmon with content: matrix =  'https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/genomics/mus_musculus/rnaseq_expression/SRP254919.salmon.merged.gene_counts.top1000cov.tsv'\n",
      "Operator found cutadapt with content: run_cutadapt = false\n",
      "Operator found cutadapt with content: cutadapt_params = \"\"\n",
      "Operator found fastqc with content: skip_fastqc = false\n",
      "Operator found fastqc with content: fastqc_params = \"\"\n",
      "Operator found salmon with content: generate_salmon_uniq_ambig = false\n",
      "Operator found salmon with content: run_salmon_selective_alignment = false\n",
      "Operator found salmon with content: salmon_sa_params_index = \"\"\n",
      "Operator found salmon with content: salmon_sa_params_mapping = \"\"\n",
      "Operator found salmon with content: star_salmon_index_params = \"\"\n",
      "Operator found salmon with content: star_salmon_alignment_params = \"\"\n",
      "Operator found salmon with content: run_salmon_alignment_based_mode = false\n",
      "Operator found salmon with content: salmon_alignment_based_params = \"\"\n",
      "Operator found multiqc with content: multiqc_config = false\n",
      "Operator found multiqc with content: max_multiqc_email_size = 25.MB\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Operator found salmon with content: run_salmon_selective_alignment = true\n",
      "Operator found salmon with content: run_salmon_selective_alignment = true\n",
      "Operator found bwa with content: bwa_index = null\n",
      "Operator found fastqc with content: skip_fastqc = false\n",
      "Operator found bwa with content: mapper = 'bwaaln'\n",
      "Operator found bwa with content: bwaalnn = 0.01 // From Oliva et al. 2021 (10.1093/bib/bbab076)\n",
      "Operator found bwa with content: bwaalnk = 2\n",
      "Operator found bwa with content: bwaalnl = 1024 // From Oliva et al. 2021 (10.1093/bib/bbab076)\n",
      "Operator found bwa with content: bwaalno = 2 // From Oliva et al. 2021 (10.1093/bib/bbab076)\n",
      "Operator found bowtie2 with content: bt2n = 0 // Do not set Cahill 2018 recommendation of 1 here, so not to 'hide' overriding bowtie2 presets\n",
      "Operator found bedtools with content: run_bedtools_coverage = false\n",
      "Operator found gatk with content: gatk_call_conf = 30\n",
      "Operator found gatk with content: gatk_ploidy = 2\n",
      "Operator found gatk with content: gatk_downsample = 250\n",
      "Operator found gatk with content: gatk_dbsnp = null\n",
      "Operator found gatk with content: gatk_hc_out_mode = 'EMIT_VARIANTS_ONLY'\n",
      "Operator found gatk with content: gatk_hc_emitrefconf = 'GVCF'\n",
      "Operator found gatk with content: gatk_ug_genotype_model = 'SNP'\n",
      "Operator found gatk with content: gatk_ug_out_mode = 'EMIT_VARIANTS_ONLY'\n",
      "Operator found gatk with content: gatk_ug_keep_realign_bam = false\n",
      "Operator found gatk with content: gatk_ug_defaultbasequalities = null\n",
      "Operator found samtools with content: angsd_glmodel = 'samtools'\n",
      "Operator found bcftools with content: run_bcftools_stats = true\n",
      "Operator found multiqc with content: multiqc_config = false\n",
      "Operator found multiqc with content: max_multiqc_email_size = 25.MB\n",
      "Operator found gatk with content: gatk_call_conf = 20\n",
      "Operator found bwa with content: withName:'bwa'{\n",
      "Operator found bwa with content: bwaalnn = 0.04\n",
      "Operator found bwa with content: bwaalnl = 1024\n",
      "Operator found gatk with content: gatk_ploidy = 2\n",
      "Operator found bwa with content: withName:'bwa'{\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwaalnn = 0.04\n",
      "Operator found bwa with content: bwaalnl = 1024\n",
      "Operator found gatk with content: gatk_ploidy = 2\n",
      "Operator found bwa with content: withName:'bwa'{\n",
      "Operator found bwa with content: withName:'bwa'{\n",
      "Operator found multiqc with content: multiqc_config             = null\n",
      "Operator found multiqc with content: multiqc_title              = null\n",
      "Operator found multiqc with content: multiqc_logo               = \"$projectDir/assets/nf-core-epitopeprediction_logo_light.png\"\n",
      "Operator found multiqc with content: max_multiqc_email_size     = '25.MB'\n",
      "Operator found multiqc with content: multiqc_methods_description = null\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BismarkIndex/\"\n",
      "Operator found fastqc with content: withName:fastqc {\n",
      "Operator found trim_galore with content: withName:trim_galore {\n",
      "Operator found bwa with content: withName:bwamem {\n",
      "Operator found samtools with content: container = 'shub://qbicsoftware/qbic-singularity-samtools'\n",
      "Operator found bwa with content: bwa_index = '/beegfs/work/zxmai83/Reference/genome/b37/human_g1k_v37.fasta.bwt'\n",
      "Operator found bwa with content: bwa_index = '${params.metafiles}/b37/human_g1k_v37.fasta.bwt'\n",
      "Operator found bwa with content: $bwaAlign {\n",
      "Operator found bwa with content: module = ['bioinfo-tools', 'bwa/0.7.15']\n",
      "Operator found picard with content: module = ['bioinfo-tools', 'picard/2.0.1']\n",
      "Operator found picard with content: module = ['bioinfo-tools', 'picard/2.0.1']\n",
      "Operator found picard with content: module = ['bioinfo-tools', 'picard/2.0.1']\n",
      "Operator found picard with content: module = ['bioinfo-tools', 'picard/2.0.1']\n",
      "Operator found picard with content: module = ['bioinfo-tools', 'picard/2.0.1']\n",
      "Operator found picard with content: module = ['bioinfo-tools', 'picard/2.0.1']\n",
      "Operator found bwa with content: bwa_index = '/proj/ngi2016003/nobackup/senthil/seqcap_test/resource/bwa_genome_hg19/hg19.ga'\n",
      "Operator found multiqc with content: multiqc_config             = null\n",
      "Operator found multiqc with content: multiqc_title              = null\n",
      "Operator found multiqc with content: max_multiqc_email_size     = '25.MB'\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BismarkIndex/\"\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Operator found multiqc with content: multiqc_mappings_config             = \"${params.test_data_base}/modules/local/multiqc_mappings_config/SRX9626017_SRR13191702.mappings.csv\"\n",
      "Operator found multiqc with content: includeConfig \"../../modules/local/multiqc_mappings_config/nextflow.config\"\n",
      "Operator found abricate with content: arg_skip_abricate                       = false\n",
      "Operator found abricate with content: arg_abricate_db                         = 'ncbi'\n",
      "Operator found abricate with content: arg_abricate_minid                      = 80\n",
      "Operator found abricate with content: arg_abricate_mincov                     = 80\n",
      "Operator found multiqc with content: multiqc_config             = null\n",
      "Operator found multiqc with content: multiqc_title              = null\n",
      "Operator found multiqc with content: multiqc_logo               = null\n",
      "Operator found multiqc with content: max_multiqc_email_size     = '25.MB'\n",
      "Operator found multiqc with content: multiqc_methods_description = null\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/CHM13/Sequence/BWAIndex/\"\n",
      "Operator found bwa with content: bwamem2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/CHM13/Sequence/BWAmem2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BismarkIndex/\"\n",
      "Operator found multiqc with content: ext.args   = params.multiqc_title ? \"--title \\\"$params.multiqc_title\\\"\" : ''\n",
      "Operator found multiqc with content: path: { \"${params.outdir}/multiqc\" },\n",
      "Operator found abricate with content: path: { \"${params.outdir}/arg/abricate/${meta.id}\" },\n",
      "Operator found abricate with content: \"--db ${params.arg_abricate_db}\",\n",
      "Operator found abricate with content: \"--minid ${params.arg_abricate_minid}\",\n",
      "Operator found abricate with content: \"--mincov ${params.arg_abricate_mincov}\"\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Operator found abricate with content: path: { \"${params.outdir}/arg/hamronization/abricate\" },\n",
      "Operator found abricate with content: ext.prefix = { \"${report}.abricate\" }\n",
      "Operator found abricate with content: arg_skip_abricate      = true\n",
      "Operator found multiqc with content: multiqc_config             = null\n",
      "Operator found multiqc with content: multiqc_title              = null\n",
      "Operator found multiqc with content: max_multiqc_email_size     = '25.MB'\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BismarkIndex/\"\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Operator found multiqc with content: multiqc_config             = null\n",
      "Operator found multiqc with content: multiqc_title              = null\n",
      "Operator found multiqc with content: multiqc_logo               = null\n",
      "Operator found multiqc with content: max_multiqc_email_size     = '25.MB'\n",
      "Operator found multiqc with content: multiqc_methods_description = null\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/CHM13/Sequence/BWAIndex/\"\n",
      "Operator found bwa with content: bwamem2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/CHM13/Sequence/BWAmem2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BismarkIndex/\"\n",
      "Operator found multiqc with content: multiqc_config             = null\n",
      "Operator found multiqc with content: multiqc_title              = null\n",
      "Operator found multiqc with content: max_multiqc_email_size     = '25.MB'\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BismarkIndex/\"\n",
      "Operator found multiqc with content: multiqc_config             = null\n",
      "Operator found multiqc with content: multiqc_title              = null\n",
      "Operator found multiqc with content: multiqc_logo               = null\n",
      "Operator found multiqc with content: max_multiqc_email_size     = '25.MB'\n",
      "Operator found multiqc with content: multiqc_methods_description = null\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/CHM13/Sequence/BWAIndex/\"\n",
      "Operator found bwa with content: bwamem2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/CHM13/Sequence/BWAmem2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BismarkIndex/\"\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Operator found bwa with content: aligner                    = \"bwa-mem\"\n",
      "Operator found multiqc with content: multiqc_config             = null\n",
      "Operator found multiqc with content: multiqc_title              = null\n",
      "Operator found multiqc with content: multiqc_logo               = null\n",
      "Operator found multiqc with content: max_multiqc_email_size     = '25.MB'\n",
      "Operator found multiqc with content: multiqc_runkraken          = true\n",
      "Operator found multiqc with content: multiqc_methods_description = null\n",
      "Operator found multiqc with content: params.multiqc_runkraken = false\n",
      "Operator found multiqc with content: params.multiqc_runkraken = false\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/CHM13/Sequence/BWAIndex/\"\n",
      "Operator found bwa with content: bwamem2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/CHM13/Sequence/BWAmem2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BismarkIndex/\"\n",
      "Operator found fastqc with content: path: { \"${params.outdir}/QC/fastqc_raw\" },\n",
      "Operator found fastqc with content: path: { \"${params.outdir}/QC/fastqc_trimmed\" },\n",
      "Operator found bwa with content: ext.when = { !params.bwaindex && params.aligner == \"bwa-mem\" }\n",
      "Operator found bwa with content: ext.when = { !params.bwamem2index && params.aligner == \"bwa-mem2\" }\n",
      "Operator found bwa-mem2 with content: ext.when = { !params.bwamem2index && params.aligner == \"bwa-mem2\" }\n",
      "Operator found multiqc with content: path: { \"${params.outdir}/QC/multiqc\" },\n",
      "Operator found kraken2 with content: krakendb = 'https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/genomics/sarscov2/genome/db/kraken2.tar.gz'\n",
      "Operator found kraken2 with content: krakendb = 'https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/genomics/sarscov2/genome/db/kraken2.tar.gz'\n",
      "Operator found multiqc with content: skip_multiqc = false\n",
      "Operator found multiqc with content: multiqc_config             = null\n",
      "Operator found multiqc with content: multiqc_title              = null\n",
      "Operator found multiqc with content: multiqc_logo               = null\n",
      "Operator found multiqc with content: max_multiqc_email_size     = '25.MB'\n",
      "Operator found multiqc with content: multiqc_methods_description = null\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/Bowtie2Index/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/Bowtie2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/CHM13/Sequence/BWAIndex/\"\n",
      "Operator found bwa with content: bwamem2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/CHM13/Sequence/BWAmem2Index/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/Bowtie2Index/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/Bowtie2Index/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/Bowtie2Index/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/Bowtie2Index/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/Bowtie2Index/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/Bowtie2Index/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/Bowtie2Index/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/Bowtie2Index/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/Bowtie2Index/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/Bowtie2Index/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/Bowtie2Index/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/Bowtie2Index/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/Bowtie2Index/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/Bowtie2Index/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/Bowtie2Index/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/Bowtie2Index/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/Bowtie2Index/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/Bowtie2Index/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/Bowtie2Index/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/Bowtie2Index/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/Bowtie2Index/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/Bowtie2Index/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/Bowtie2Index/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/Bowtie2Index/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/Bowtie2Index/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/Bowtie2Index/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/Bowtie2Index/\"\n",
      "Operator found bowtie2 with content: path: { \"${params.outdir}/genome/bowtie2\" },\n",
      "Operator found cutadapt with content: skip_cutadapt              = false\n",
      "Operator found cutadapt with content: cutadapt_5end              = '^TAC'\n",
      "Operator found fastqc with content: skip_fastqc                = false\n",
      "Operator found multiqc with content: skip_multiqc               = false\n",
      "Operator found multiqc with content: multiqc_config             = null\n",
      "Operator found multiqc with content: multiqc_title              = null\n",
      "Operator found multiqc with content: multiqc_logo               = null\n",
      "Operator found multiqc with content: max_multiqc_email_size     = '25.MB'\n",
      "Operator found multiqc with content: multiqc_methods_description = null\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/CHM13/Sequence/BWAIndex/\"\n",
      "Operator found bwa with content: bwamem2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/CHM13/Sequence/BWAmem2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BismarkIndex/\"\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Operator found fastqc with content: path: { \"${params.outdir}/fastqc\" },\n",
      "Operator found multiqc with content: ext.args   = { params.multiqc_config ? \"--config $multiqc_custom_config\" : '' }\n",
      "Operator found multiqc with content: path: { \"${params.outdir}/multiqc\" },\n",
      "Operator found bwa with content: path: { \"${params.outdir}/genome/bwa_index\" },\n",
      "Operator found cutadapt with content: if(!params.skip_cutadapt){\n",
      "Operator found cutadapt with content: ext.args    = { \"-Z -e 0.15 --no-indels --action none --discard-untrimmed -g ${params.cutadapt_5end}\" }\n",
      "Operator found samtools with content: ext.args2   = '-bhS' // for samtools view\n",
      "Operator found bwa with content: path: { \"${params.outdir}/bwa/mapped/bam\" },\n",
      "Operator found bwa with content: path: { \"${params.outdir}/bwa/mapped/bam\" },\n",
      "Operator found bwa with content: path: { \"${params.outdir}/bwa/mapped/QC\" },\n",
      "Operator found bwa with content: path: { \"${params.outdir}/bwa/mapped/QC\" },\n",
      "Operator found bwa with content: path: { \"${params.outdir}/bwa/mapped/QC\" },\n",
      "Operator found macs2 with content: withName: 'BEDFILES_SORT_PER_GROUP' {// sort the bedgraph output from macs2\n",
      "Operator found macs2 with content: withName: 'UCSC_BEDCLIP' {// clip the bed file output from macs2\n",
      "Operator found bedtools with content: withName: 'MAPS_FEND' {//fix end, by bedtools slop, here are fixed -l 0 -r fragment size;\n",
      "Operator found bedtools with content: ext.args    = '-s -l 0' // parameters for bedtools/slop, please do not modify.\n",
      "Operator found multiqc with content: multiqc_config             = \"$projectDir/assets/multiqc_optitype_config.yml\"\n",
      "Operator found multiqc with content: multiqc_title              = null\n",
      "Operator found multiqc with content: multiqc_logo               = \"${projectDir}/assets/nf-core-hlatyping_logo_light.png\"\n",
      "Operator found multiqc with content: max_multiqc_email_size     = '25.MB'\n",
      "Operator found multiqc with content: multiqc_methods_description = null\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BismarkIndex/\"\n",
      "Operator found multiqc with content: multiqc_config             = null\n",
      "Operator found multiqc with content: multiqc_title              = null\n",
      "Operator found multiqc with content: multiqc_logo               = null\n",
      "Operator found multiqc with content: max_multiqc_email_size     = '25.MB'\n",
      "Operator found multiqc with content: multiqc_methods_description = null\n",
      "Operator found minimap2 with content: test_minimap2 { includeConfig 'conf/test_minimap2.config' }\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/CHM13/Sequence/BWAIndex/\"\n",
      "Operator found bwa with content: bwamem2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/CHM13/Sequence/BWAmem2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BismarkIndex/\"\n",
      "Operator found minimap2 with content: aligner         = 'minimap2'\n",
      "Operator found multiqc with content: skip_multiqc = false\n",
      "Operator found multiqc with content: multiqc_config = \"$baseDir/assets/multiqc_config.yaml\"\n",
      "Operator found multiqc with content: max_multiqc_email_size = 25.MB\n",
      "Operator found multiqc with content: withName: 'multiqc|get_software_versions' {\n",
      "Operator found sourmash with content: withName: 'sourmash_compute_sketch_fastx_nucleotide|sourmash_compute_sketch_fastx_peptide' {\n",
      "Operator found multiqc with content: multiqc_config             = null\n",
      "Operator found multiqc with content: multiqc_title              = null\n",
      "Operator found multiqc with content: max_multiqc_email_size     = '25.MB'\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BismarkIndex/\"\n",
      "Operator found fastqc with content: 'fastqc' {\n",
      "Operator found multiqc with content: 'multiqc' {\n",
      "Operator found hisat2 with content: hisat2_index = '/data/database/hg38/hisatIndex/grch38_snp_tran/genome_snp_tran'\n",
      "Operator found bowtie2 with content: bowtie2_index = ''//set if tophat used\n",
      "Operator found samtools with content: sam_processor=\"sambamba\"//or \"samtools(deprecated)\"\n",
      "Operator found fastqc with content: qctools =\"fastp\"  // or \"afterqc\",\"fastp\",\"fastqc\",\"none\" to skip qc step\n",
      "Operator found fastp with content: qctools =\"fastp\"  // or \"afterqc\",\"fastp\",\"fastqc\",\"none\" to skip qc step\n",
      "Operator found kallisto with content: quant = \"kallisto\"// or 'htseq'\n",
      "Operator found hisat2 with content: withName:fastq_hisat2_alignment_For_discovery {\n",
      "Operator found kallisto with content: withName:Build_kallisto_index_of_GTF_for_quantification {\n",
      "Operator found kallisto with content: withName:Run_kallisto_for_quantification {\n",
      "Operator found hisat2 with content: hisat2_index = \"./Genome/hisat_index/chr22\"\n",
      "Operator found fastp with content: clip_tool                            = 'fastp'\n",
      "Operator found fastp with content: fastp_save_trimmed_fail              = false\n",
      "Operator found fastp with content: fastp_qualified_quality              = 15\n",
      "Operator found fastp with content: fastp_cut_mean_quality               = 15\n",
      "Operator found bowtie2 with content: bowtie2_mode                         = null\n",
      "Operator found megahit with content: megahit_options                      = null\n",
      "Operator found megahit with content: skip_megahit                         = false\n",
      "Operator found bcftools with content: bcftools_view_high_variant_quality   = 30\n",
      "Operator found bcftools with content: bcftools_view_medium_variant_quality = 20\n",
      "Operator found bcftools with content: bcftools_view_minimal_allelesupport  = 3\n",
      "Operator found kraken2 with content: kraken2_db                           = null\n",
      "Operator found gtdbtk with content: skip_gtdbtk                          = false\n",
      "Operator found gtdbtk with content: gtdb_db                              = \"https://data.ace.uq.edu.au/public/gtdb/data/releases/release214/214.1/auxillary_files/gtdbtk_r214_data.tar.gz\"\n",
      "Operator found gtdbtk with content: gtdbtk_min_completeness              = 50.0\n",
      "Operator found gtdbtk with content: gtdbtk_max_contamination             = 10.0\n",
      "Operator found gtdbtk with content: gtdbtk_min_perc_aa                   = 10\n",
      "Operator found gtdbtk with content: gtdbtk_min_af                        = 0.65\n",
      "Operator found gtdbtk with content: gtdbtk_pplacer_cpus                  = 1\n",
      "Operator found gtdbtk with content: gtdbtk_pplacer_scratch               = true\n",
      "Operator found checkm with content: checkm_download_url                  = \"https://data.ace.uq.edu.au/public/CheckM_databases/checkm_data_2015_01_16.tar.gz\"\n",
      "Operator found checkm with content: checkm_db                            = null\n",
      "Operator found checkm with content: save_checkm_data                     = false\n",
      "Operator found megahit with content: megahit_fix_cpu_1                    = false\n",
      "Operator found mmseqs with content: metaeuk_mmseqs_db                    = null\n",
      "Operator found mmseqs with content: save_mmseqs_db                       = false\n",
      "Operator found multiqc with content: multiqc_config             = null\n",
      "Operator found multiqc with content: multiqc_title              = null\n",
      "Operator found multiqc with content: multiqc_logo               = null\n",
      "Operator found multiqc with content: max_multiqc_email_size     = '25.MB'\n",
      "Operator found multiqc with content: multiqc_methods_description = null\n",
      "Operator found megahit with content: def check_megahit_cpus (x, attempt ) {\n",
      "Operator found megahit with content: if (params.megahit_fix_cpu_1) return 1\n",
      "Operator found megahit with content: cpus          = { check_megahit_cpus (8, task.attempt  ) }\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/CHM13/Sequence/BWAIndex/\"\n",
      "Operator found bwa with content: bwamem2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/CHM13/Sequence/BWAmem2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BismarkIndex/\"\n",
      "Operator found fastqc with content: path: { \"${params.outdir}/QC_shortreads/fastqc\" },\n",
      "Operator found fastp with content: \"-q ${params.fastp_qualified_quality}\",\n",
      "Operator found fastp with content: \"--cut_mean_quality ${params.fastp_cut_mean_quality}\",\n",
      "Operator found fastp with content: path: { \"${params.outdir}/QC_shortreads/fastp/${meta.id}\" },\n",
      "Operator found fastp with content: path: { \"${params.outdir}/QC_shortreads/fastp/${meta.id}\" },\n",
      "Operator found fastp with content: ext.prefix = { \"${meta.id}_run${meta.run}_fastp\" }\n",
      "Operator found fastqc with content: path: { \"${params.outdir}/QC_shortreads/fastqc\" },\n",
      "Operator found NanoPlot with content: path: { \"${params.outdir}/QC_longreads/NanoPlot/${meta.id}\" },\n",
      "Operator found NanoPlot with content: path: { \"${params.outdir}/QC_longreads/NanoPlot/${meta.id}\" },\n",
      "Operator found kraken2 with content: path: { \"${params.outdir}/Taxonomy/kraken2/${meta.id}\" },\n",
      "Operator found megahit with content: ext.args = params.megahit_options ?: ''\n",
      "Operator found bowtie2 with content: ext.args = params.bowtie2_mode ? params.bowtie2_mode : params.ancient_dna ? '--very-sensitive -N 1' : ''\n",
      "Operator found checkm with content: path: { \"${params.outdir}/GenomeBinning/QC/CheckM/checkm_downloads\" },\n",
      "Operator found checkm with content: enabled: params.save_checkm_data\n",
      "Operator found checkm with content: ext.prefix = { \"checkm_summary\" }\n",
      "Operator found checkm with content: path: { \"${params.outdir}/GenomeBinning/QC/GUNC/checkmmerged/${meta.assembler}-${meta.binner}-${meta.domain}-${meta.refinement}-${meta.id}\" },\n",
      "Operator found bcftools with content: ext.args   = \"-v snps,mnps -i 'QUAL>=${params.bcftools_view_high_variant_quality} || (QUAL>=${params.bcftools_view_medium_variant_quality} && FORMAT/AO>=${params.bcftools_view_minimal_allelesupport})'\"\n",
      "Operator found samtools with content: path: {\"${params.outdir}/Ancient_DNA/samtools/faidx\" },\n",
      "Operator found mmseqs with content: ext.prefix = { \"${params.metaeuk_mmseqs_db.replaceAll(\"/\", \"-\")}\" }\n",
      "Operator found mmseqs with content: path: { \"${params.outdir}/Annotation/mmseqs_db/\" },\n",
      "Operator found mmseqs with content: enabled: params.save_mmseqs_db\n",
      "Operator found multiqc with content: ext.args   = { params.multiqc_title ? \"--title \\\"$params.multiqc_title\\\"\" : '' }\n",
      "Operator found multiqc with content: path: { \"${params.outdir}/multiqc\" },\n",
      "Operator found kraken2 with content: kraken2_db                    = \"https://raw.githubusercontent.com/nf-core/test-datasets/mag/test_data/minigut_kraken.tgz\"\n",
      "Operator found gtdbtk with content: skip_gtdbtk                   = true\n",
      "Operator found kraken2 with content: kraken2_db                  = \"https://raw.githubusercontent.com/nf-core/test-datasets/mag/test_data/minigut_kraken.tgz\"\n",
      "Operator found gtdbtk with content: skip_gtdbtk                 = true\n",
      "Operator found kraken2 with content: kraken2_db                           = \"https://raw.githubusercontent.com/nf-core/test-datasets/mag/test_data/minigut_kraken.tgz\"\n",
      "Operator found gtdbtk with content: skip_gtdbtk                          = true\n",
      "Operator found bcftools with content: bcftools_view_high_variant_quality   = 0\n",
      "Operator found bcftools with content: bcftools_view_medium_variant_quality = 0\n",
      "Operator found bcftools with content: bcftools_view_minimal_allelesupport  = 3\n",
      "Operator found kraken2 with content: kraken2_db                    = \"https://raw.githubusercontent.com/nf-core/test-datasets/mag/test_data/minigut_kraken.tgz\"\n",
      "Operator found gtdbtk with content: skip_gtdbtk                   = true\n",
      "Operator found kraken2 with content: kraken2_db                    = \"https://raw.githubusercontent.com/nf-core/test-datasets/mag/test_data/minigut_kraken.tgz\"\n",
      "Operator found gtdbtk with content: skip_gtdbtk                   = true\n",
      "Operator found gtdbtk with content: skip_gtdbtk                 = true\n",
      "Operator found kraken2 with content: kraken2_db    = \"s3://ngi-igenomes/test-data/mag/minikraken_8GB_202003.tgz\"\n",
      "Operator found gtdbtk with content: skip_gtdbtk   = true\n",
      "Operator found megahit with content: megahit_fix_cpu_1     = true\n",
      "Operator found gtdbtk with content: skip_gtdbtk                 = true\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Operator found gtdbtk with content: skip_gtdbtk                 = true\n",
      "Operator found gtdbtk with content: skip_gtdbtk                 = true\n",
      "Operator found kraken2 with content: kraken2_db                    = null\n",
      "Operator found megahit with content: skip_megahit                  = true\n",
      "Operator found gtdbtk with content: skip_gtdbtk                   = true\n",
      "Operator found gtdbtk with content: skip_gtdbtk                 = true\n",
      "Operator found multiqc with content: multiqc_config             = null\n",
      "Operator found multiqc with content: multiqc_title              = null\n",
      "Operator found multiqc with content: multiqc_logo               = null\n",
      "Operator found multiqc with content: max_multiqc_email_size     = '25.MB'\n",
      "Operator found multiqc with content: multiqc_methods_description = null\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.genomes_base}/mm10_v32/bowtie2/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.genomes_base}/mm10/bowtie2/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.genomes_base}/mm9/bowtie2\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.genomes_base}/GRCh38_v43/bowtie2/\"\n",
      "Operator found fastqc with content: path: { \"${params.outdir}/${meta.id}/fastqc/\" },\n",
      "Operator found multiqc with content: path: { \"${params.outdir}/multiqc\" },\n",
      "Operator found multiqc with content: multiqc_config             = null\n",
      "Operator found multiqc with content: multiqc_title              = null\n",
      "Operator found multiqc with content: multiqc_logo               = null\n",
      "Operator found multiqc with content: max_multiqc_email_size     = '25.MB'\n",
      "Operator found multiqc with content: multiqc_methods_description = null\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/CHM13/Sequence/BWAIndex/\"\n",
      "Operator found bwa with content: bwamem2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/CHM13/Sequence/BWAmem2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BismarkIndex/\"\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BismarkIndex/\"\n",
      "Operator found multiqc with content: ext.args   = { params.multiqc_title ? \"--title \\\"$params.multiqc_title\\\"\" : '' }\n",
      "Operator found multiqc with content: path: { \"${params.outdir}/multiqc\" },\n",
      "Operator found multiqc with content: multiqc_config             = null\n",
      "Operator found multiqc with content: multiqc_title              = null\n",
      "Operator found multiqc with content: multiqc_logo               = null\n",
      "Operator found multiqc with content: max_multiqc_email_size     = '25.MB'\n",
      "Operator found multiqc with content: multiqc_methods_description = null\n",
      "Operator found multiqc with content: ext.args   = params.multiqc_title ? \"--title \\\"$params.multiqc_title\\\"\" : ''\n",
      "Operator found multiqc with content: path: { \"${params.outdir}/multiqc\" },\n",
      "Operator found fastqc with content: skip_fastqc                 = false\n",
      "Operator found megahit with content: assembler                   = 'megahit'\n",
      "Operator found samtools with content: save_samtools               = false\n",
      "Operator found multiqc with content: multiqc_config             = null\n",
      "Operator found multiqc with content: multiqc_title              = null\n",
      "Operator found multiqc with content: multiqc_logo               = null\n",
      "Operator found multiqc with content: max_multiqc_email_size     = '25.MB'\n",
      "Operator found multiqc with content: multiqc_methods_description = null\n",
      "Operator found fastqc with content: '--fastqc',\n",
      "Operator found fastqc with content: path: { \"${params.outdir}/trimgalore/fastqc\" },\n",
      "Operator found megahit with content: path: { \"${params.outdir}/megahit\" },\n",
      "Operator found samtools with content: path: { \"${params.outdir}/samtools/\" },\n",
      "Operator found samtools with content: path: { \"${params.outdir}/samtools/\" },\n",
      "Operator found samtools with content: path: { \"${params.outdir}/samtools/\" },\n",
      "Operator found samtools with content: enabled: params.save_samtools,\n",
      "Operator found featureCounts with content: pattern: \"*.featureCounts.*\"\n",
      "Operator found multiqc with content: ext.args   = { params.multiqc_title ? \"--title \\\"$params.multiqc_title\\\"\" : '' }\n",
      "Operator found multiqc with content: path: \"${params.outdir}/multiqc\",\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Operator found multiqc with content: multiqc_config             = null\n",
      "Operator found multiqc with content: multiqc_title              = null\n",
      "Operator found multiqc with content: multiqc_logo               = null\n",
      "Operator found multiqc with content: max_multiqc_email_size     = '25.MB'\n",
      "Operator found multiqc with content: multiqc_methods_description = null\n",
      "Operator found bismark with content: aligner                    = 'bismark'\n",
      "Operator found multiqc with content: skip_multiqc               = false\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/CHM13/Sequence/BWAIndex/\"\n",
      "Operator found bwa with content: bwamem2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/CHM13/Sequence/BWAmem2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BismarkIndex/\"\n",
      "Operator found fastqc with content: path: { \"${params.outdir}/fastqc\" },\n",
      "Operator found fastqc with content: path: { \"${params.outdir}/fastqc/zips\" },\n",
      "Operator found fastqc with content: '--fastqc',\n",
      "Operator found fastqc with content: path: { \"${params.outdir}/trimgalore/fastqc\" },\n",
      "Operator found fastqc with content: path: { \"${params.outdir}/trimgalore/fastqc/zips\" },\n",
      "Operator found bismark with content: params.aligner == 'bismark_hisat' ? ' --hisat2' : ' --bowtie2',\n",
      "Operator found bowtie2 with content: params.aligner == 'bismark_hisat' ? ' --hisat2' : ' --bowtie2',\n",
      "Operator found hisat2 with content: params.aligner == 'bismark_hisat' ? ' --hisat2' : ' --bowtie2',\n",
      "Operator found bismark with content: params.aligner == 'bismark_hisat' ? ' --hisat2' : ' --bowtie2',\n",
      "Operator found bowtie2 with content: params.aligner == 'bismark_hisat' ? ' --hisat2' : ' --bowtie2',\n",
      "Operator found hisat2 with content: params.aligner == 'bismark_hisat' ? ' --hisat2' : ' --bowtie2',\n",
      "Operator found bismark with content: params.aligner == 'bismark_hisat' && params.known_splices ? \" --known-splicesite-infile <(hisat2_extract_splice_sites.py ${known_splices})\" : '',\n",
      "Operator found hisat2 with content: params.aligner == 'bismark_hisat' && params.known_splices ? \" --known-splicesite-infile <(hisat2_extract_splice_sites.py ${known_splices})\" : '',\n",
      "Operator found bismark with content: path: { \"${params.outdir}/bismark/coverage2cytosine/summaries\" },\n",
      "Operator found bismark with content: path: { \"${params.outdir}/bismark/coverage2cytosine/reports\" },\n",
      "Operator found bismark with content: path: { \"${params.outdir}/bismark/coverage2cytosine/coverage\" },\n",
      "Operator found picard with content: path: { \"${params.outdir}/${params.aligner}/deduplicated/picard_metrics\" },\n",
      "Operator found samtools with content: path: { \"${params.outdir}/${params.aligner}/alignments/samtools_stats/\" },\n",
      "Operator found samtools with content: path: { \"${params.outdir}/${params.aligner}/alignments/samtools_stats/\" },\n",
      "Operator found multiqc with content: ext.args   = { params.multiqc_title ? \"--title \\\"$params.multiqc_title\\\"\" : '' }\n",
      "Operator found multiqc with content: path: { \"${params.outdir}/multiqc/${params.aligner}\" },\n",
      "Operator found fastqc with content: test_1_fastq_gz_fastqc_zip = \"https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/genomics/sarscov2/illumina/fastqc/test_fastqc.zip\"\n",
      "Operator found kraken2 with content: kraken2_tar_gz          = \"https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/genomics/sarscov2/genome/db/kraken2.tar.gz\"\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Operator found multiqc with content: skip_multiqc                    = false\n",
      "Operator found multiqc with content: multiqc_config                  = null\n",
      "Operator found multiqc with content: multiqc_title                   = null\n",
      "Operator found multiqc with content: multiqc_logo                    = null\n",
      "Operator found multiqc with content: max_multiqc_email_size          = '25.MB'\n",
      "Operator found multiqc with content: multiqc_methods_description     = null\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/CHM13/Sequence/BWAIndex/\"\n",
      "Operator found bwa with content: bwamem2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/CHM13/Sequence/BWAmem2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BismarkIndex/\"\n",
      "Operator found multiqc with content: ext.args   = params.multiqc_title ? \"--title \\\"$params.multiqc_title\\\"\" : ''\n",
      "Operator found multiqc with content: path: { \"${params.outdir}/multiqc\" },\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Operator found fastqc with content: skip_fastqc = false\n",
      "Operator found picard with content: skip_picard_metrics = false\n",
      "Operator found multiqc with content: skip_multiqc = false\n",
      "Operator found multiqc with content: multiqc_config = false\n",
      "Operator found multiqc with content: max_multiqc_email_size = 25.MB\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BismarkIndex/\"\n",
      "Operator found multiqc with content: multiqc_config             = null\n",
      "Operator found multiqc with content: multiqc_title              = null\n",
      "Operator found multiqc with content: multiqc_logo               = null\n",
      "Operator found multiqc with content: max_multiqc_email_size     = '25.MB'\n",
      "Operator found multiqc with content: multiqc_methods_description = null\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/CHM13/Sequence/BWAIndex/\"\n",
      "Operator found bwa with content: bwamem2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/CHM13/Sequence/BWAmem2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BismarkIndex/\"\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Operator found multiqc with content: multiqc_config             = null\n",
      "Operator found multiqc with content: multiqc_title              = null\n",
      "Operator found multiqc with content: multiqc_logo               = null\n",
      "Operator found multiqc with content: max_multiqc_email_size     = '25.MB'\n",
      "Operator found multiqc with content: multiqc_methods_description = null\n",
      "Operator found multiqc with content: skip_multiqc               = false\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/CHM13/Sequence/BWAIndex/\"\n",
      "Operator found bwa with content: bwamem2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/CHM13/Sequence/BWAmem2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BismarkIndex/\"\n",
      "Operator found multiqc with content: ext.args   = { params.multiqc_title ? \"--title \\\"$params.multiqc_title\\\"\" : '' }\n",
      "Operator found multiqc with content: path: { \"${params.outdir}/multiqc\" },\n",
      "Operator found multiqc with content: skip_multiqc = true\n",
      "Operator found minimap2 with content: aligner                    = 'minimap2'\n",
      "Operator found fastqc with content: skip_fastqc                = false\n",
      "Operator found multiqc with content: skip_multiqc               = false\n",
      "Operator found multiqc with content: multiqc_config             = null\n",
      "Operator found multiqc with content: max_multiqc_email_size     = '25.MB' //note\n",
      "Operator found multiqc with content: multiqc_title              = null //note\n",
      "Operator found stringtie with content: test_nodx_stringtie { includeConfig 'conf/test_nodx_stringtie.config' }\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BismarkIndex/\"\n",
      "Operator found fastqc with content: if (!params.skip_fastqc) {\n",
      "Operator found fastqc with content: path: { \"${params.outdir}/fastqc\" },\n",
      "Operator found samtools with content: path: { \"${params.outdir}/graphmap2/samtools_stats\" },\n",
      "Operator found samtools with content: path: { \"${params.outdir}/graphmap2/samtools_stats\" },\n",
      "Operator found samtools with content: path: { \"${params.outdir}/graphmap2/samtools_stats\" },\n",
      "Operator found minimap2 with content: if (params.aligner == \"minimap2\") {\n",
      "Operator found minimap2 with content: path: { \"${params.outdir}/minimap2\" },\n",
      "Operator found minimap2 with content: path: { \"${params.outdir}/minimap2\" },\n",
      "Operator found minimap2 with content: path: { \"${params.outdir}/minimap2\" },\n",
      "Operator found minimap2 with content: path: { \"${params.outdir}/minimap2\" },\n",
      "Operator found minimap2 with content: path: { \"${params.outdir}/minimap2\" },\n",
      "Operator found minimap2 with content: path: { \"${params.outdir}/minimap2\" },\n",
      "Operator found samtools with content: path: { \"${params.outdir}/minimap2/samtools_stats\" },\n",
      "Operator found minimap2 with content: path: { \"${params.outdir}/minimap2/samtools_stats\" },\n",
      "Operator found samtools with content: path: { \"${params.outdir}/minimap2/samtools_stats\" },\n",
      "Operator found minimap2 with content: path: { \"${params.outdir}/minimap2/samtools_stats\" },\n",
      "Operator found samtools with content: path: { \"${params.outdir}/minimap2/samtools_stats\" },\n",
      "Operator found minimap2 with content: path: { \"${params.outdir}/minimap2/samtools_stats\" },\n",
      "Operator found minimap2 with content: path: { \"${params.outdir}/minimap2/bigbed\" },\n",
      "Operator found minimap2 with content: path: { \"${params.outdir}/minimap2/bigbed\" },\n",
      "Operator found minimap2 with content: path: { \"${params.outdir}/minimap2/bigwig\" },\n",
      "Operator found minimap2 with content: path: { \"${params.outdir}/minimap2/bigwig\" },\n",
      "Operator found stringtie with content: if (params.quantification_method == \"stringtie2\") {\n",
      "Operator found stringtie with content: path: { \"${params.outdir}/stringtie2\" },\n",
      "Operator found featureCounts with content: path: { \"${params.outdir}/stringtie2/featureCounts\" },\n",
      "Operator found stringtie with content: path: { \"${params.outdir}/stringtie2/featureCounts\" },\n",
      "Operator found multiqc with content: if (!params.skip_multiqc) {\n",
      "Operator found multiqc with content: ext.args   = params.multiqc_title ? \"--title \\\"$params.multiqc_title\\\"\" : ''\n",
      "Operator found multiqc with content: \"${params.outdir}/multiqc\",\n",
      "Operator found stringtie with content: input               = 'https://raw.githubusercontent.com/nf-core/test-datasets/nanoseq/3.0/samplesheet/samplesheet_nobc_nodx_stringtie.csv'\n",
      "Operator found stringtie with content: quantification_method = 'stringtie2'\n",
      "Operator found multiqc with content: multiqc_config             = null\n",
      "Operator found multiqc with content: multiqc_title              = null\n",
      "Operator found multiqc with content: multiqc_logo               = null\n",
      "Operator found multiqc with content: max_multiqc_email_size     = '25.MB'\n",
      "Operator found multiqc with content: multiqc_methods_description = null\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/CHM13/Sequence/BWAIndex/\"\n",
      "Operator found bwa with content: bwamem2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/CHM13/Sequence/BWAmem2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BismarkIndex/\"\n",
      "Operator found multiqc with content: ext.args   = { params.multiqc_title ? \"--title \\\"$params.multiqc_title\\\"\" : '' }\n",
      "Operator found multiqc with content: path: { \"${params.outdir}/multiqc\" },\n",
      "Operator found bwa with content: aligner                    = 'bwa'\n",
      "Operator found multiqc with content: multiqc_config             = null\n",
      "Operator found multiqc with content: multiqc_title              = null\n",
      "Operator found multiqc with content: multiqc_logo               = null\n",
      "Operator found multiqc with content: max_multiqc_email_size     = '25.MB'\n",
      "Operator found multiqc with content: multiqc_methods_description = null\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/CHM13/Sequence/BWAIndex/\"\n",
      "Operator found bwa with content: bwamem2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/CHM13/Sequence/BWAmem2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BismarkIndex/\"\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BismarkIndex/\"\n",
      "Operator found multiqc with content: multiqc_config = \"$baseDir/assets/multiqc_config.yaml\"\n",
      "Operator found multiqc with content: withName: multiqc {\n",
      "Operator found multiqc with content: multiqc_config              = null\n",
      "Operator found multiqc with content: multiqc_title               = null\n",
      "Operator found multiqc with content: multiqc_logo                = null\n",
      "Operator found multiqc with content: max_multiqc_email_size      = '25.MB'\n",
      "Operator found multiqc with content: multiqc_methods_description = null\n",
      "Operator found multiqc with content: ext.args = \"-P --multiqc\"\n",
      "Operator found multiqc with content: ext.prefix = {\"${meta.id}_multiqc\"}\n",
      "Operator found multiqc with content: ext.prefix = {\"${meta.id}_multiqc\"}\n",
      "Operator found multiqc with content: ext.prefix = {\"${meta.id}_multiqc\"}\n",
      "Operator found multiqc with content: ext.prefix = {\"${meta.id}_multiqc\"}\n",
      "Operator found multiqc with content: ext.prefix = {\"${meta.id}_multiqc\"}\n",
      "Operator found multiqc with content: ext.prefix = {\"${meta.id}_multiqc\"}\n",
      "Operator found multiqc with content: ext.prefix = {\"${meta.id}.draw_multiqc\"}\n",
      "Operator found tabix with content: path: { \"${params.outdir}/tabix_bgzip\" },\n",
      "Operator found samtools with content: path: { \"${params.outdir}/samtools_faidx\" },\n",
      "Operator found multiqc with content: ext.args = { \"--outdir ${meta.id}_multiqc\" }\n",
      "Operator found multiqc with content: multiqc_config             = null\n",
      "Operator found multiqc with content: multiqc_title              = null\n",
      "Operator found multiqc with content: multiqc_logo               = null\n",
      "Operator found multiqc with content: max_multiqc_email_size     = '25.MB'\n",
      "Operator found multiqc with content: multiqc_methods_description = null\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BismarkIndex/\"\n",
      "Operator found multiqc with content: max_multiqc_email_size = 25.MB\n",
      "Operator found gatk with content: description = 'Proteogenomics database creation workflow using pypgatk framework. '\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BismarkIndex/\"\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Operator found multiqc with content: multiqc_config             = null\n",
      "Operator found multiqc with content: multiqc_title              = null\n",
      "Operator found multiqc with content: multiqc_logo               = null\n",
      "Operator found multiqc with content: max_multiqc_email_size     = '25.MB'\n",
      "Operator found multiqc with content: multiqc_methods_description = null\n",
      "Operator found seqkit with content: assembly_fasta_gz       = \"${params.test_data_base}/modules/local/seqkit/seq/assembly.fasta.gz\"\n",
      "Operator found fastqc with content: test_1_fastq_gz_fastqc_zip = \"https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/genomics/sarscov2/illumina/fastqc/test_fastqc.zip\"\n",
      "Operator found kraken2 with content: kraken2_tar_gz          = \"https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/genomics/sarscov2/genome/db/kraken2.tar.gz\"\n",
      "Operator found seqkit with content: path: { \"${params.outdir}/AssemblyFilter/seqkit/seq\" },\n",
      "Operator found bowtie2 with content: path: { \"${params.outdir}/VirusAbundance/bowtie2/align\" },\n",
      "Operator found bowtie2 with content: path: { \"${params.outdir}/VirusAbundance/bowtie2/align\" },\n",
      "Operator found bowtie2 with content: path: { \"${params.outdir}/VirusAbundance/bowtie2/build\" },\n",
      "Operator found bowtie2 with content: pattern: 'bowtie2'\n",
      "Operator found fastqc with content: path: { \"${params.outdir}/ReadQuality/fastqc\" },\n",
      "Operator found multiqc with content: ext.args   = params.multiqc_title ? \"--title \\\"$params.multiqc_title\\\"\" : ''\n",
      "Operator found multiqc with content: path: { \"${params.outdir}/multiqc\" },\n",
      "Operator found samtools with content: includeConfig '../../../modules/nf-core/samtools/sort/nextflow.config'\n",
      "Operator found samtools with content: includeConfig '../../../modules/nf-core/samtools/index/nextflow.config'\n",
      "Operator found samtools with content: includeConfig '../../../modules/nf-core/samtools/stats/nextflow.config'\n",
      "Operator found samtools with content: includeConfig '../../../modules/nf-core/samtools/idxstats/nextflow.config'\n",
      "Operator found samtools with content: includeConfig '../../../modules/nf-core/samtools/flagstat/nextflow.config'\n",
      "Operator found bowtie2 with content: includeConfig '../../../modules/nf-core/bowtie2/align/nextflow.config'\n",
      "Operator found bowtie2 with content: includeConfig '../../../modules/nf-core/bowtie2/build/nextflow.config'\n",
      "Operator found fastqc with content: includeConfig '../../modules/nf-core/fastqc/nextflow.config'\n",
      "Operator found seqkit with content: includeConfig '../../modules/local/seqkit/seq/nextflow.config'\n",
      "Operator found multiqc with content: includeConfig '../../modules/nf-core/multiqc/nextflow.config'\n",
      "Operator found samtools with content: includeConfig '../../subworkflows/nf-core/bam_sort_stats_samtools/nextflow.config'\n",
      "Operator found samtools with content: includeConfig '../../subworkflows/nf-core/bam_stats_samtools/nextflow.config'\n",
      "Operator found bowtie2 with content: includeConfig '../../subworkflows/nf-core/fastq_align_bowtie2/nextflow.config'\n",
      "Operator found multiqc with content: multiqc_config             = null\n",
      "Operator found multiqc with content: multiqc_title              = null\n",
      "Operator found multiqc with content: multiqc_logo               = null\n",
      "Operator found multiqc with content: max_multiqc_email_size     = '25.MB'\n",
      "Operator found multiqc with content: multiqc_methods_description = null\n",
      "Operator found multiqc with content: multiqc_config              = null\n",
      "Operator found multiqc with content: multiqc_title               = null\n",
      "Operator found multiqc with content: multiqc_logo                = null\n",
      "Operator found multiqc with content: max_multiqc_email_size      = '25.MB'\n",
      "Operator found multiqc with content: multiqc_methods_description = null\n",
      "Operator found multiqc with content: max_multiqc_email_size = 25.MB\n",
      "Operator found multiqc with content: enable_pmultiqc          = true\n",
      "Operator found multiqc with content: pmultiqc_idxml_skip      = true\n",
      "Operator found multiqc with content: multiqc_config             = null\n",
      "Operator found multiqc with content: multiqc_title              = null\n",
      "Operator found multiqc with content: multiqc_logo               = null\n",
      "Operator found multiqc with content: max_multiqc_email_size     = '25.MB'\n",
      "Operator found multiqc with content: multiqc_methods_description = null\n",
      "Operator found multiqc with content: ext.args   = params.multiqc_title ? \"--title \\\"$params.multiqc_title\\\"\" : ''\n",
      "Operator found multiqc with content: path: { \"${params.outdir}/multiqc\" },\n",
      "Operator found bwa with content: aligner                        = 'bwamem2'\n",
      "Operator found multiqc with content: multiqc_config                 = null\n",
      "Operator found multiqc with content: multiqc_title                  = null\n",
      "Operator found multiqc with content: max_multiqc_email_size         = '25.MB'\n",
      "Operator found seqtk with content: schema_ignore_params       = 'aligner_options,genomes,method_options,publish_dir_mode,save_reference_indices,denovo_intermediate_file,method_options,publish_dir_mode,save_freebayes_intervals,save_intervals,save_reference_fai,save_trim_adapters_fastp,save_cdhit_clstr,save_seqtk_seq_fasta,save_uniq_full_fasta,save_uniqseq,save_trimmed,min_repeat_entropy,min_alt_fraction,use_best_n_alleles,complex_gap,min_base_qual,min_map_qual,max_read_coverage_to_split,subset_intervals_channel,quality_score,gap_penalty,mismatch_score,matching_score,min_aln_quality,mark_short_as_sec,output_secondary,clipping_penalty,aligner,min_similarity_fraction,min_overlap,max_reads_for_assembly,max_clusters_for_merge,min_reads,max_variants,similarity_fraction,sequence_simularity,description_length,cluster_algorithm,minreaddepth_betweenindividual,minreaddepth_withinindividual,umi_read_structure,dont_eval_duplicates,trim_polyg,clip_r2,clip_r1,overlap_dif_limit,pairedend_bp_corr,mean_min_quality,window_size,cut_right,sequence_type,method,popmap'\n",
      "Operator found fastp with content: schema_ignore_params       = 'aligner_options,genomes,method_options,publish_dir_mode,save_reference_indices,denovo_intermediate_file,method_options,publish_dir_mode,save_freebayes_intervals,save_intervals,save_reference_fai,save_trim_adapters_fastp,save_cdhit_clstr,save_seqtk_seq_fasta,save_uniq_full_fasta,save_uniqseq,save_trimmed,min_repeat_entropy,min_alt_fraction,use_best_n_alleles,complex_gap,min_base_qual,min_map_qual,max_read_coverage_to_split,subset_intervals_channel,quality_score,gap_penalty,mismatch_score,matching_score,min_aln_quality,mark_short_as_sec,output_secondary,clipping_penalty,aligner,min_similarity_fraction,min_overlap,max_reads_for_assembly,max_clusters_for_merge,min_reads,max_variants,similarity_fraction,sequence_simularity,description_length,cluster_algorithm,minreaddepth_betweenindividual,minreaddepth_withinindividual,umi_read_structure,dont_eval_duplicates,trim_polyg,clip_r2,clip_r1,overlap_dif_limit,pairedend_bp_corr,mean_min_quality,window_size,cut_right,sequence_type,method,popmap'\n",
      "Operator found bwa with content: aligner_options            = ' bwa, bwamem2'\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BismarkIndex/\"\n",
      "Operator found fastp with content: path: { \"${params.outdir}/fastp\" },\n",
      "Operator found fastp with content: path: { \"${params.outdir}/fastp\" },\n",
      "Operator found fastp with content: path: { \"${params.outdir}/fastp\" },\n",
      "Operator found fastp with content: path: { \"${params.outdir}/denovo/trim_fastp/\" },\n",
      "Operator found fastp with content: path: { \"${params.outdir}/denovo/trim_fastp/\" },\n",
      "Operator found fastp with content: path: { \"${params.outdir}/denovo/trim_fastp/\" },\n",
      "Operator found seqtk with content: path: { \"${params.outdir}/${params.method}/reference/seqtk/\" },\n",
      "Operator found seqtk with content: pattern: \"*.seqtk-seq.*.gz\",\n",
      "Operator found fastp with content: path: { \"${params.outdir}/${params.method}/reference/fastp/\" },\n",
      "Operator found fastp with content: path: { \"${params.outdir}/${params.method}/reference/fastp/\" },\n",
      "Operator found samtools with content: path: { ( \"${params.method}\" == 'denovo' ? \"${params.outdir}/${params.method}/reference/samtools/index\" : \"${params.outdir}/${params.method}/samtools/index\" ) },\n",
      "Operator found samtools with content: path: { \"${params.outdir}/${params.method}/alignments/${params.aligner}/samtools_index\" },\n",
      "Operator found samtools with content: path: { \"${params.outdir}/${params.method}/alignments/${params.aligner}/samtools_merge/\" },\n",
      "Operator found samtools with content: path: { \"${params.outdir}/${params.method}/alignments/${params.aligner}/samtools_stats/\" },\n",
      "Operator found samtools with content: path: { \"${params.outdir}/${params.method}/alignments/${params.aligner}/samtools_stats/\" },\n",
      "Operator found samtools with content: path: { \"${params.outdir}/${params.method}/alignments/${params.aligner}/samtools_stats/\" },\n",
      "Operator found bedtools with content: path: { \"${params.outdir}/${params.method}/alignments/${params.aligner}/intervals/bedtools_bamtobed/\" },\n",
      "Operator found bedtools with content: path: { \"${params.outdir}/${params.method}/alignments/${params.aligner}/intervals/bedtools_sort/\" },\n",
      "Operator found bedtools with content: path: { \"${params.outdir}/${params.method}/alignments/${params.aligner}/intervals/bedtools_coverage/\" },\n",
      "Operator found bedtools with content: path: { \"${params.outdir}/${params.method}/alignments/${params.aligner}/intervals/bedtools_merge/\" },\n",
      "Operator found bedtools with content: path: { \"${params.outdir}/${params.method}/alignments/${params.aligner}/intervals/bedtools_makewindows/\" },\n",
      "Operator found bedtools with content: path: { \"${params.outdir}/${params.method}/alignments/${params.aligner}/intervals/bedtools_makewindows/\" },\n",
      "Operator found bedtools with content: path: { \"${params.outdir}/${params.method}/alignments/${params.aligner}/intervals/bedtools_makewindows/\" },\n",
      "Operator found bedtools with content: path: { \"${params.outdir}/${params.method}/alignments/${params.aligner}/intervals/bedtools_intersect/\" },\n",
      "Operator found bwa with content: aligner                    = 'bwamem2'\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Operator found multiqc with content: multiqc_config             = null\n",
      "Operator found multiqc with content: multiqc_title              = null\n",
      "Operator found multiqc with content: multiqc_logo               = null\n",
      "Operator found multiqc with content: max_multiqc_email_size     = '25.MB'\n",
      "Operator found multiqc with content: multiqc_methods_description = null\n",
      "Operator found multiqc with content: ext.args   = params.multiqc_title ? \"--title \\\"$params.multiqc_title\\\"\" : ''\n",
      "Operator found multiqc with content: path:   { \"${params.outdir}/multiqc\" },\n",
      "Operator found bwa with content: aligner                    = 'bwamem2'\n",
      "Operator found multiqc with content: multiqc_config             = null\n",
      "Operator found multiqc with content: multiqc_title              = null\n",
      "Operator found multiqc with content: multiqc_logo               = null\n",
      "Operator found multiqc with content: max_multiqc_email_size     = '25.MB'\n",
      "Operator found multiqc with content: multiqc_methods_description = null\n",
      "Operator found bwa with content: includeConfig 'conf/modules/align_bwamem2.config'\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/CHM13/Sequence/BWAIndex/\"\n",
      "Operator found bwa with content: bwamem2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/CHM13/Sequence/BWAmem2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BismarkIndex/\"\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: ext.when = { !params.skip_mt_analysis && !(params.analysis_type == \"wes\") && params.aligner == \"bwamem2\" }\n",
      "Operator found bwa with content: ext.when = { !params.skip_mt_analysis && !(params.analysis_type == \"wes\") && params.aligner == \"bwamem2\" }\n",
      "Operator found bwa with content: ext.when = params.aligner.equals(\"bwamem2\")\n",
      "Operator found bwa with content: ext.when = {!params.bwamem2 && params.aligner == \"bwamem2\"}\n",
      "Operator found bwa with content: ext.when = { !params.skip_mt_analysis && !(params.analysis_type == \"wes\") && params.aligner == \"bwamem2\"}\n",
      "Operator found bwa with content: ext.when = {!params.bwa && params.aligner == \"sentieon\"}\n",
      "Operator found bwa with content: ext.when = {!params.bwa && !(params.aligner == \"sentieon\")}\n",
      "Operator found bwa with content: ext.when = params.aligner.equals(\"bwamem2\")\n",
      "Operator found bwa with content: ext.when = params.aligner.equals(\"bwamem2\")\n",
      "Operator found fastqc with content: path: { \"${params.outdir}/fastqc/${meta.id}\" },\n",
      "Operator found multiqc with content: path: { \"${params.outdir}/multiqc/\" },\n",
      "Operator found multiqc with content: multiqc_config             = null\n",
      "Operator found multiqc with content: multiqc_title              = null\n",
      "Operator found multiqc with content: multiqc_logo               = null\n",
      "Operator found multiqc with content: max_multiqc_email_size     = '25.MB'\n",
      "Operator found multiqc with content: multiqc_methods_description = null\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/CHM13/Sequence/BWAIndex/\"\n",
      "Operator found bwa with content: bwamem2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/CHM13/Sequence/BWAmem2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BismarkIndex/\"\n",
      "Operator found samtools with content: ext.args2 = \"-b | samtools sort\"\n",
      "Operator found bowtie2 with content: path: { \"${params.outdir}/bowtie2\" },\n",
      "Operator found bowtie2 with content: path: { \"${params.outdir}/bowtie2\" },\n",
      "Operator found multiqc with content: ext.args   = { params.multiqc_title ? \"--title \\\"$params.multiqc_title\\\"\" : '' }\n",
      "Operator found multiqc with content: path: { \"${params.outdir}/multiqc\" },\n",
      "Operator found bowtie2 with content: path: { \"${params.outdir}/bowtie2\" },\n",
      "Operator found kraken2 with content: kraken2                                        = \"${params.test_data_base}/data/genomics/sarscov2/genome/db/kraken2\"\n",
      "Operator found kraken2 with content: kraken2_tar_gz                                 = \"${params.test_data_base}/data/genomics/sarscov2/genome/db/kraken2.tar.gz\"\n",
      "Operator found kraken2 with content: kraken2_bracken                                = \"${params.test_data_base}/data/genomics/sarscov2/genome/db/kraken2_bracken\"\n",
      "Operator found kraken2 with content: kraken2_bracken_tar_gz                         = \"${params.test_data_base}/data/genomics/sarscov2/genome/db/kraken2_bracken.tar.gz\"\n",
      "Operator found mmseqs with content: mmseqs_tar_gz                                  = \"${params.test_data_base}/data/genomics/sarscov2/genome/db/mmseqs.tar.gz\"\n",
      "Operator found picard with content: baits_interval_list                            = \"${params.test_data_base}/data/genomics/sarscov2/genome/picard/baits.interval_list\"\n",
      "Operator found picard with content: targets_interval_list                          = \"${params.test_data_base}/data/genomics/sarscov2/genome/picard/targets.interval_list\"\n",
      "Operator found fastqc with content: test_1_fastq_gz_fastqc_html                    = \"${params.test_data_base}/data/genomics/sarscov2/illumina/fastqc/test_fastqc.html\"\n",
      "Operator found fastqc with content: test_1_fastq_gz_fastqc_zip                     = \"${params.test_data_base}/data/genomics/sarscov2/illumina/fastqc/test_fastqc.zip\"\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Operator found gatk with content: test_baserecalibrator_table                    = \"${params.test_data_base}/data/genomics/sarscov2/illumina/gatk/test.baserecalibrator.table\"\n",
      "Operator found picard with content: test_single_end_bam_readlist_txt               = \"${params.test_data_base}/data/genomics/sarscov2/illumina/picard/test.single_end.bam.readlist.txt\"\n",
      "Operator found kraken2 with content: classified_reads_assignment                    = \"${params.test_data_base}/data/genomics/sarscov2/metagenome/test_1.kraken2.reads.txt\"\n",
      "Operator found kraken2 with content: kraken_report                                  = \"${params.test_data_base}/data/genomics/sarscov2/metagenome/test_1.kraken2.report.txt\"\n",
      "Operator found salmon with content: rnaseq_matrix                                  = \"${params.test_data_base}/data/genomics/mus_musculus/rnaseq_expression/SRP254919.salmon.merged.gene_counts.top1000cov.tsv\"\n",
      "Operator found salmon with content: deseq_results                                  = \"${params.test_data_base}/data/genomics/mus_musculus/rnaseq_expression/SRP254919.salmon.merged.deseq2.results.tsv\"\n",
      "Operator found checkm with content: ngscheckmate_bed                               = \"${params.test_data_base}/data/genomics/homo_sapiens/genome/chr21/germlineresources/SNP_GRCh38_hg38_wChr.bed\"\n",
      "Operator found salmon with content: index_salmon                                   = \"${params.test_data_base}/data/genomics/homo_sapiens/genome/index/salmon\"\n",
      "Operator found gatk with content: test_baserecalibrator_table                             = \"${params.test_data_base}/data/genomics/homo_sapiens/illumina/gatk/test.baserecalibrator.table\"\n",
      "Operator found gatk with content: test2_baserecalibrator_table                            = \"${params.test_data_base}/data/genomics/homo_sapiens/illumina/gatk/test2.baserecalibrator.table\"\n",
      "Operator found gatk with content: test_pileups_table                                      = \"${params.test_data_base}/data/genomics/homo_sapiens/illumina/gatk/test.pileups.table\"\n",
      "Operator found gatk with content: test2_pileups_table                                     = \"${params.test_data_base}/data/genomics/homo_sapiens/illumina/gatk/test2.pileups.table\"\n",
      "Operator found gatk with content: test_paired_end_sorted_dragstrmodel                     = \"${params.test_data_base}/data/genomics/homo_sapiens/illumina/gatk/test_paired_end_sorted_dragstrmodel.txt\"\n",
      "Operator found gatk with content: test_genomicsdb_tar_gz                                  = \"${params.test_data_base}/data/genomics/homo_sapiens/illumina/gatk/test_genomicsdb.tar.gz\"\n",
      "Operator found gatk with content: test_pon_genomicsdb_tar_gz                              = \"${params.test_data_base}/data/genomics/homo_sapiens/illumina/gatk/test_pon_genomicsdb.tar.gz\"\n",
      "Operator found gatk with content: test2_haplotc_ann_vcf_gz                                = \"${params.test_data_base}/data/genomics/homo_sapiens/illumina/gatk/haplotypecaller_calls/test2_haplotc.ann.vcf.gz\"\n",
      "Operator found gatk with content: test2_haplotc_ann_vcf_gz_tbi                            = \"${params.test_data_base}/data/genomics/homo_sapiens/illumina/gatk/haplotypecaller_calls/test2_haplotc.ann.vcf.gz.tbi\"\n",
      "Operator found gatk with content: test_haplotc_cnn_vcf_gz                                 = \"${params.test_data_base}/data/genomics/homo_sapiens/illumina/gatk/haplotypecaller_calls/test_haplotcaller.cnn.vcf.gz\"\n",
      "Operator found gatk with content: test_haplotc_cnn_vcf_gz_tbi                             = \"${params.test_data_base}/data/genomics/homo_sapiens/illumina/gatk/haplotypecaller_calls/test_haplotcaller.cnn.vcf.gz.tbi\"\n",
      "Operator found gatk with content: test2_haplotc_vcf_gz                                    = \"${params.test_data_base}/data/genomics/homo_sapiens/illumina/gatk/haplotypecaller_calls/test2_haplotc.vcf.gz\"\n",
      "Operator found gatk with content: test2_haplotc_vcf_gz_tbi                                = \"${params.test_data_base}/data/genomics/homo_sapiens/illumina/gatk/haplotypecaller_calls/test2_haplotc.vcf.gz.tbi\"\n",
      "Operator found gatk with content: test2_recal                                             = \"${params.test_data_base}/data/genomics/homo_sapiens/illumina/gatk/variantrecalibrator/test2.recal\"\n",
      "Operator found gatk with content: test2_recal_idx                                         = \"${params.test_data_base}/data/genomics/homo_sapiens/illumina/gatk/variantrecalibrator/test2.recal.idx\"\n",
      "Operator found gatk with content: test2_tranches                                          = \"${params.test_data_base}/data/genomics/homo_sapiens/illumina/gatk/variantrecalibrator/test2.tranches\"\n",
      "Operator found gatk with content: test2_allele_specific_recal                             = \"${params.test_data_base}/data/genomics/homo_sapiens/illumina/gatk/variantrecalibrator/test2_allele_specific.recal\"\n",
      "Operator found gatk with content: test2_allele_specific_recal_idx                         = \"${params.test_data_base}/data/genomics/homo_sapiens/illumina/gatk/variantrecalibrator/test2_allele_specific.recal.idx\"\n",
      "Operator found gatk with content: test2_allele_specific_tranches                          = \"${params.test_data_base}/data/genomics/homo_sapiens/illumina/gatk/variantrecalibrator/test2_allele_specific.tranches\"\n",
      "Operator found gatk with content: test_test2_paired_mutect2_calls_vcf_gz                  = \"${params.test_data_base}/data/genomics/homo_sapiens/illumina/gatk/paired_mutect2_calls/test_test2_paired_mutect2_calls.vcf.gz\"\n",
      "Operator found gatk with content: test_test2_paired_mutect2_calls_vcf_gz_tbi              = \"${params.test_data_base}/data/genomics/homo_sapiens/illumina/gatk/paired_mutect2_calls/test_test2_paired_mutect2_calls.vcf.gz.tbi\"\n",
      "Operator found gatk with content: test_test2_paired_mutect2_calls_vcf_gz_stats            = \"${params.test_data_base}/data/genomics/homo_sapiens/illumina/gatk/paired_mutect2_calls/test_test2_paired_mutect2_calls.vcf.gz.stats\"\n",
      "Operator found gatk with content: test_test2_paired_mutect2_calls_f1r2_tar_gz             = \"${params.test_data_base}/data/genomics/homo_sapiens/illumina/gatk/paired_mutect2_calls/test_test2_paired_mutect2_calls.f1r2.tar.gz\"\n",
      "Operator found gatk with content: test_test2_paired_mutect2_calls_artifact_prior_tar_gz   = \"${params.test_data_base}/data/genomics/homo_sapiens/illumina/gatk/test_test2_paired_mutect2_calls.artifact-prior.tar.gz\"\n",
      "Operator found gatk with content: test_test2_paired_segmentation_table                    = \"${params.test_data_base}/data/genomics/homo_sapiens/illumina/gatk/test_test2_paired.segmentation.table\"\n",
      "Operator found gatk with content: test_test2_paired_contamination_table                   = \"${params.test_data_base}/data/genomics/homo_sapiens/illumina/gatk/test_test2_paired.contamination.table\"\n",
      "Operator found gatk with content: contig_ploidy_priors_table                              = \"${params.test_data_base}/data/genomics/homo_sapiens/illumina/gatk/contig_ploidy_priors_table.tsv\"\n",
      "Operator found abricate with content: genome_abricate_tsv             = \"${params.test_data_base}/data/genomics/prokaryotes/bacteroides_fragilis/hamronization/genome.abricate.tsv\"\n",
      "Operator found tabix with content: test_tabix_pair_gz              = \"${params.test_data_base}/data/genomics/homo_sapiens/cooler/cload/hg19/hg19.GM12878-MboI.pairs.subsample.sorted.possrt.txt.gz\"\n",
      "Operator found tabix with content: test_tabix_pair_gz_tbi          = \"${params.test_data_base}/data/genomics/homo_sapiens/cooler/cload/hg19/hg19.GM12878-MboI.pairs.subsample.sorted.possrt.txt.gz.tbi\"\n",
      "Operator found hisat2 with content: aligner                    = 'hisat2'\n",
      "Operator found multiqc with content: multiqc_config             = null\n",
      "Operator found multiqc with content: multiqc_title              = null\n",
      "Operator found multiqc with content: multiqc_logo               = null\n",
      "Operator found multiqc with content: max_multiqc_email_size     = '25.MB'\n",
      "Operator found multiqc with content: multiqc_methods_description = null\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/CHM13/Sequence/BWAIndex/\"\n",
      "Operator found bwa with content: bwamem2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/CHM13/Sequence/BWAmem2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BWAIndex/version0.6.0/\"\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: aligner                         = 'bwa-mem'    // Only STAR is currently supported.\n",
      "Operator found hisat2 with content: hisat2_build_memory             = null\n",
      "Operator found hisat2 with content: hisat2_index                    = null\n",
      "Operator found gatk with content: gatk_interval_scatter_count     = 25\n",
      "Operator found multiqc with content: multiqc_config             = null\n",
      "Operator found multiqc with content: multiqc_title              = null\n",
      "Operator found multiqc with content: multiqc_logo               = null\n",
      "Operator found multiqc with content: max_multiqc_email_size     = '25.MB'\n",
      "Operator found multiqc with content: multiqc_methods_description = null\n",
      "Operator found gatk with content: includeConfig 'conf/modules/gatk4_preprocessing/markduplicates.config'\n",
      "Operator found gatk with content: includeConfig 'conf/modules/gatk4_preprocessing/splitncigarreads.config'\n",
      "Operator found gatk with content: includeConfig 'conf/modules/gatk4_preprocessing/prepare_recalibration.config'\n",
      "Operator found gatk with content: includeConfig 'conf/modules/gatk4_preprocessing/recalibrate.config'\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/CHM13/Sequence/BWAIndex/\"\n",
      "Operator found bwa with content: bwamem2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/CHM13/Sequence/BWAmem2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BismarkIndex/\"\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BismarkIndex/\"\n",
      "Operator found gatk with content: docker.userEmulation                     = { params.use_gatk_spark ? false : true }.call()\n",
      "Operator found bwa with content: ext.when         = { params.aligner == \"bwa-mem\" }\n",
      "Operator found bwa with content: ext.when         = { params.aligner == \"bwa-mem2\" }\n",
      "Operator found bwa-mem2 with content: ext.when         = { params.aligner == \"bwa-mem2\" }\n",
      "Operator found hisat2 with content: path: { \"${params.outdir}/report/hisat2/${meta.patient}/${meta.id}\" },\n",
      "Operator found hisat2 with content: path: { \"${params.outdir}/preprocessing/hisat2/${meta.patient}/${meta.id}/\" },\n",
      "Operator found hisat2 with content: path: { \"${params.outdir}/preprocessing/hisat2/${meta.patient}/${meta.id}/unmapped\" },\n",
      "Operator found hisat2 with content: path: { \"${params.outdir}/preprocessing/hisat2/\" },\n",
      "Operator found hisat2 with content: path: { \"${params.outdir}/preprocessing/hisat2/${meta.patient}/${meta.id}/\" },\n",
      "Operator found samtools with content: path: { \"${params.outdir}/reports/samtools/\" },\n",
      "Operator found bwa with content: ext.when         = { !params.bwa && params.step == \"mapping\" && (params.aligner == \"bwa-mem\" || params.aligner == \"sentieon-bwamem\")}\n",
      "Operator found bwa with content: pattern: \"bwa\"\n",
      "Operator found bwa with content: ext.when         = { !params.bwamem2 && params.step == \"mapping\" && params.aligner == \"bwa-mem2\" }\n",
      "Operator found bwa-mem2 with content: ext.when         = { !params.bwamem2 && params.step == \"mapping\" && params.aligner == \"bwa-mem2\" }\n",
      "Operator found bwa with content: pattern: \"bwamem2\"\n",
      "Operator found fastqc with content: ext.when         = { !(params.skip_tools && params.skip_tools.split(',').contains('fastqc')) }\n",
      "Operator found fastqc with content: path: { \"${params.outdir}/reports/fastqc/${meta.id}\" },\n",
      "Operator found multiqc with content: ext.args   = params.multiqc_title ? \"--title \\\"$params.multiqc_title\\\"\" : ''\n",
      "Operator found multiqc with content: enabled: !(params.skip_tools && (params.skip_tools.split(',').contains('multiqc')))\n",
      "Operator found samtools with content: ext.when         = { !(params.skip_tools && params.skip_tools.split(',').contains('samtools')) }\n",
      "Operator found samtools with content: path: { \"${params.outdir}/reports/samtools/${meta.id}\" },\n",
      "Operator found samtools with content: ext.when         = { !(params.skip_tools && params.skip_tools.split(',').contains('samtools')) }\n",
      "Operator found samtools with content: path: { \"${params.outdir}/reports/samtools/${meta.id}\" },\n",
      "Operator found bcftools with content: ext.when         = { !(params.skip_tools && params.skip_tools.split(',').contains('bcftools')) }\n",
      "Operator found bcftools with content: path: { \"${params.outdir}/reports/bcftools/${meta.variantcaller}/${meta.id}/\" },\n",
      "Operator found vcftools with content: ext.when         = { !(params.skip_tools && params.skip_tools.split(',').contains('vcftools')) }\n",
      "Operator found vcftools with content: path: { \"${params.outdir}/reports/vcftools/${meta.variantcaller}/${meta.id}/\" },\n",
      "Operator found samtools with content: ext.when    = { !(params.skip_tools && params.skip_tools.split(',').contains('samtools')) }\n",
      "Operator found samtools with content: path: { \"${params.outdir}/reports/samtools/samtools_stats\" },\n",
      "Operator found samtools with content: ext.when    = { !(params.skip_tools && params.skip_tools.split(',').contains('samtools')) }\n",
      "Operator found samtools with content: path: { \"${params.outdir}/reports/samtools/hisat2/samtools_stats\" },\n",
      "Operator found hisat2 with content: path: { \"${params.outdir}/reports/samtools/hisat2/samtools_stats\" },\n",
      "Operator found samtools with content: ext.when    = { !(params.skip_tools && params.skip_tools.split(',').contains('samtools')) }\n",
      "Operator found samtools with content: path: { \"${params.outdir}/reports/samtools/markduplicates/samtools_stats\" },\n",
      "Operator found fastp with content: path: { \"${params.outdir}/reports/fastp/${meta.sample}\" },\n",
      "Operator found fastp with content: path: { \"${params.outdir}/preprocessing/fastp/${meta.sample}/\" },\n",
      "Operator found fastp with content: pattern: \"*.fastp.fastq.gz\"\n",
      "Operator found multiqc with content: multiqc_config             = null\n",
      "Operator found multiqc with content: multiqc_title              = null\n",
      "Operator found multiqc with content: multiqc_logo               = null\n",
      "Operator found multiqc with content: max_multiqc_email_size     = '25.MB'\n",
      "Operator found multiqc with content: multiqc_methods_description = null\n",
      "Operator found fastp with content: fastp_trim                 = false\n",
      "Operator found stringtie with content: stringtie                  = false\n",
      "Operator found fastqc with content: path: { \"${params.outdir}/fastqc\" },\n",
      "Operator found fastqc with content: path: { \"${params.outdir}/fastqc_for_fastp\" },\n",
      "Operator found fastp with content: path: { \"${params.outdir}/fastqc_for_fastp\" },\n",
      "Operator found multiqc with content: ext.args   = params.multiqc_title ? \"--title \\\"$params.multiqc_title\\\"\" : ''\n",
      "Operator found multiqc with content: path: { \"${params.outdir}/multiqc\" },\n",
      "Operator found picard with content: path: { \"${params.outdir}/picard\" },\n",
      "Operator found picard with content: path: { \"${params.outdir}/picard\" },\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Operator found stringtie with content: path: { \"${params.outdir}/stringtie/${meta.id}\" },\n",
      "Operator found fastp with content: extra_fastp_args           = null\n",
      "Operator found salmon with content: aligner                    = 'star_salmon'\n",
      "Operator found salmon with content: salmon_quant_libtype       = null\n",
      "Operator found hisat2 with content: hisat2_build_memory        = '200.GB'  // Amount of memory required to build HISAT2 index with splice sites\n",
      "Operator found stringtie with content: stringtie_ignore_gtf       = false\n",
      "Operator found salmon with content: extra_salmon_quant_args    = null\n",
      "Operator found kallisto with content: extra_kallisto_quant_args  = null\n",
      "Operator found kallisto with content: kallisto_quant_fraglen     = 200\n",
      "Operator found kallisto with content: kallisto_quant_fraglen_sd  = 200\n",
      "Operator found stringtie with content: skip_stringtie             = false\n",
      "Operator found fastqc with content: skip_fastqc                = false\n",
      "Operator found multiqc with content: skip_multiqc               = false\n",
      "Operator found multiqc with content: multiqc_config             = null\n",
      "Operator found multiqc with content: multiqc_title              = null\n",
      "Operator found multiqc with content: multiqc_logo               = null\n",
      "Operator found multiqc with content: max_multiqc_email_size     = '25.MB'\n",
      "Operator found multiqc with content: multiqc_methods_description = null\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/CHM13/Sequence/BWAIndex/\"\n",
      "Operator found bwa with content: bwamem2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/CHM13/Sequence/BWAmem2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BismarkIndex/\"\n",
      "Operator found fastqc with content: if (!(params.skip_fastqc || params.skip_qc)) {\n",
      "Operator found fastp with content: if (params.trimmer == 'fastp') {\n",
      "Operator found fastqc with content: path: { \"${params.outdir}/${params.trimmer}/fastqc\" },\n",
      "Operator found fastqc with content: \"--fastqc_args '-t ${task.cpus}'\",\n",
      "Operator found fastqc with content: path: { \"${params.outdir}/${params.trimmer}/fastqc\" },\n",
      "Operator found fastp with content: if (params.trimmer == 'fastp') {\n",
      "Operator found fastp with content: ext.args   = { params.extra_fastp_args ?: '' }\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Operator found samtools with content: path: { \"${params.outdir}/${params.aligner}/samtools_stats\" },\n",
      "Operator found salmon with content: path: { ( ['star_salmon','hisat2'].contains(params.aligner) &&\n",
      "Operator found hisat2 with content: path: { ( ['star_salmon','hisat2'].contains(params.aligner) &&\n",
      "Operator found salmon with content: saveAs: { ( ['star_salmon','hisat2'].contains(params.aligner) &&\n",
      "Operator found hisat2 with content: saveAs: { ( ['star_salmon','hisat2'].contains(params.aligner) &&\n",
      "Operator found salmon with content: path: { ( ['star_salmon','hisat2'].contains(params.aligner) &&\n",
      "Operator found hisat2 with content: path: { ( ['star_salmon','hisat2'].contains(params.aligner) &&\n",
      "Operator found salmon with content: saveAs: { ( ['star_salmon','hisat2'].contains(params.aligner) &&\n",
      "Operator found hisat2 with content: saveAs: { ( ['star_salmon','hisat2'].contains(params.aligner) &&\n",
      "Operator found picard with content: path: { \"${params.outdir}/${params.aligner}/picard_metrics\" },\n",
      "Operator found samtools with content: path: { \"${params.outdir}/${params.aligner}/samtools_stats\" },\n",
      "Operator found salmon with content: if (params.with_umi && ['star_salmon','hisat2'].contains(params.aligner)) {\n",
      "Operator found hisat2 with content: if (params.with_umi && ['star_salmon','hisat2'].contains(params.aligner)) {\n",
      "Operator found samtools with content: path: { \"${params.outdir}/${params.aligner}/samtools_stats\" },\n",
      "Operator found stringtie with content: if (!params.skip_stringtie) {\n",
      "Operator found stringtie with content: params.stringtie_ignore_gtf ? '' : '-e'\n",
      "Operator found stringtie with content: path: { \"${params.outdir}/${params.aligner}/stringtie\" },\n",
      "Operator found salmon with content: if (!params.skip_alignment && params.aligner == 'star_salmon') {\n",
      "Operator found salmon with content: ext.args   = { params.extra_salmon_quant_args ?: '' }\n",
      "Operator found samtools with content: path: { params.save_align_intermeds || params.save_umi_intermeds ? \"${params.outdir}/${params.aligner}/samtools_stats\" : params.outdir },\n",
      "Operator found samtools with content: path: { \"${params.outdir}/${params.aligner}/samtools_stats\" },\n",
      "Operator found salmon with content: ext.args2  = 'star_salmon'\n",
      "Operator found hisat2 with content: if (!params.skip_alignment && params.aligner == 'hisat2') {\n",
      "Operator found multiqc with content: if (!params.skip_multiqc) {\n",
      "Operator found multiqc with content: ext.args   = { params.multiqc_title ? \"--title \\\"$params.multiqc_title\\\"\" : '' }\n",
      "Operator found multiqc with content: ext.prefix = \"multiqc_report\"\n",
      "Operator found multiqc with content: \"${params.outdir}/multiqc\",\n",
      "Operator found salmon with content: if (!params.skip_pseudo_alignment && params.pseudo_aligner == 'salmon') {\n",
      "Operator found salmon with content: ext.args   = { params.extra_salmon_quant_args ?: '' }\n",
      "Operator found kallisto with content: if (!params.skip_pseudo_alignment && params.pseudo_aligner == 'kallisto') {\n",
      "Operator found kallisto with content: ext.args = params.extra_kallisto_quant_args ?: ''\n",
      "Operator found hisat2 with content: hisat2_index       = \"https://raw.githubusercontent.com/nf-core/test-datasets/7f1614baeb0ddf66e60be78c3d9fa55440465ac8/reference/hisat2.tar.gz\"\n",
      "Operator found salmon with content: salmon_index       = \"https://raw.githubusercontent.com/nf-core/test-datasets/7f1614baeb0ddf66e60be78c3d9fa55440465ac8/reference/salmon.tar.gz\"\n",
      "Operator found salmon with content: pseudo_aligner      = 'salmon'\n",
      "Operator found hisat2 with content: hisat2_index       = \"${params.test_data_base}/reference/hisat2.tar.gz\"\n",
      "Operator found salmon with content: salmon_index       = \"${params.test_data_base}/reference/salmon.tar.gz\"\n",
      "Operator found salmon with content: pseudo_aligner      = 'salmon'\n",
      "Operator found salmon with content: pseudo_aligner = 'salmon'\n",
      "Operator found fastqc with content: skip_trimgalore_fastqc     = false\n",
      "Operator found salmon with content: aligner                    = 'star_salmon'\n",
      "Operator found salmon with content: pseudo_aligner             = 'salmon'\n",
      "Operator found salmon with content: salmon_quant_libtype       = null\n",
      "Operator found fastqc with content: skip_fastqc                = false\n",
      "Operator found multiqc with content: multiqc_config             = null\n",
      "Operator found multiqc with content: multiqc_title              = null\n",
      "Operator found multiqc with content: multiqc_logo               = null\n",
      "Operator found multiqc with content: max_multiqc_email_size     = '25.MB'\n",
      "Operator found multiqc with content: multiqc_methods_description = null\n",
      "Operator found salmon with content: test_salmon_results    { includeConfig 'conf/test_salmon_results.config'    }\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/CHM13/Sequence/BWAIndex/\"\n",
      "Operator found bwa with content: bwamem2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/CHM13/Sequence/BWAmem2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BismarkIndex/\"\n",
      "Operator found fastqc with content: if (!(params.skip_fastqc)) {\n",
      "Operator found fastqc with content: params.skip_trimgalore_fastqc ? \"\" : '--fastqc',\n",
      "Operator found fastqc with content: path: { \"${params.outdir}/trimgalore/fastqc\" },\n",
      "Operator found bedtools with content: path: { \"${params.outdir}/bedtools/${meta.id}\" },\n",
      "Operator found salmon with content: if (!params.skip_alignment && (params.aligner == 'star_salmon' || params.aligner == \"star\")) {\n",
      "Operator found salmon with content: if (params.source != \"salmon_results\") {\n",
      "Operator found samtools with content: path: { \"${params.outdir}/${params.aligner}/samtools_stats\" },\n",
      "Operator found salmon with content: enabled: ( ['star_salmon','star'].contains(params.aligner))\n",
      "Operator found salmon with content: enabled: ( ['star_salmon','star'].contains(params.aligner))\n",
      "Operator found salmon with content: if (!params.skip_alignment && params.aligner == 'star_salmon') {\n",
      "Operator found salmon with content: if (params.pseudo_aligner == 'salmon') {\n",
      "Operator found salmon with content: if (params.aligner == \"star_salmon\" && params.dexseq_dtu) {\n",
      "Operator found salmon with content: if (params.source == \"fastq\" | params.source == \"salmon_results\") {\n",
      "Operator found salmon with content: if (params.pseudo_aligner == \"salmon\" && params.dexseq_dtu) {\n",
      "Operator found salmon with content: if (params.aligner == \"star_salmon\" && params.suppa) {\n",
      "Operator found salmon with content: if (params.source == \"fastq\" | params.source == \"salmon_results\") {\n",
      "Operator found salmon with content: if (params.pseudo_aligner == \"salmon\" && params.suppa) {\n",
      "Operator found salmon with content: if (params.aligner == \"star_salmon\") {\n",
      "Operator found salmon with content: if (params.source == \"fastq\" | params.source == \"salmon_results\") {\n",
      "Operator found salmon with content: if (params.pseudo_aligner == \"salmon\") {\n",
      "Operator found multiqc with content: ext.args   = { params.multiqc_title ? \"--title \\\"$params.multiqc_title\\\"\" : '' }\n",
      "Operator found multiqc with content: path: { \"${params.outdir}/multiqc\" },\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Operator found salmon with content: input     = 'https://raw.githubusercontent.com/nf-core/test-datasets/rnasplice/samplesheet/samplesheet_salmon.csv'\n",
      "Operator found salmon with content: source    = 'salmon_results'\n",
      "Operator found salmon with content: aligner = 'star_salmon'\n",
      "Operator found gatk with content: gatk_interval_scatter_count     = 25\n",
      "Operator found gatk with content: gatk_hc_call_conf               = 20\n",
      "Operator found gatk with content: gatk_vf_window_size             = 35\n",
      "Operator found gatk with content: gatk_vf_cluster_size            = 3\n",
      "Operator found gatk with content: gatk_vf_fs_filter               = 30.0\n",
      "Operator found gatk with content: gatk_vf_qd_filter               = 2.0\n",
      "Operator found multiqc with content: skip_multiqc                    = false\n",
      "Operator found multiqc with content: multiqc_config             = null\n",
      "Operator found multiqc with content: multiqc_title              = null\n",
      "Operator found multiqc with content: max_multiqc_email_size     = '25.MB'\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BismarkIndex/\"\n",
      "Operator found multiqc with content: ext.args   = params.multiqc_title ? \"--title \\\"$params.multiqc_title\\\"\" : ''\n",
      "Operator found multiqc with content: enabled: !params.skip_multiqc\n",
      "Operator found gatk with content: params.gatk_interval_scatter_count ? \"--SCATTER_COUNT $params.gatk_interval_scatter_count\" : ''\n",
      "Operator found multiqc with content: enabled: !params.skip_multiqc,\n",
      "Operator found gatk with content: params.gatk_hc_call_conf ? \"--standard-min-confidence-threshold-for-calling $params.gatk_hc_call_conf\" : '',\n",
      "Operator found gatk with content: params.gatk_vf_window_size ? \"--window $params.gatk_vf_window_size\" : '',\n",
      "Operator found gatk with content: params.gatk_vf_cluster_size ? \"--cluster $params.gatk_vf_cluster_size\" : '',\n",
      "Operator found gatk with content: params.gatk_vf_fs_filter ? \"--filter-name \\\"FS\\\" --filter \\\"FS > $params.gatk_vf_fs_filter\\\" \" : '',\n",
      "Operator found gatk with content: params.gatk_vf_qd_filter ? \"--filter-name \\\"QD\\\" --filter \\\"QD < $params.gatk_vf_qd_filter\\\" \" : '',\n",
      "Operator found multiqc with content: enabled: !params.skip_multiqc\n",
      "Operator found multiqc with content: enabled: !params.skip_multiqc\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Operator found bwa with content: bwa                  = null\n",
      "Operator found multiqc with content: multiqc_config             = null\n",
      "Operator found multiqc with content: multiqc_title              = null\n",
      "Operator found multiqc with content: multiqc_logo               = null\n",
      "Operator found multiqc with content: max_multiqc_email_size     = '25.MB'\n",
      "Operator found multiqc with content: multiqc_methods_description = null\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/CHM13/Sequence/BWAIndex/\"\n",
      "Operator found bwa with content: bwamem2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/CHM13/Sequence/BWAmem2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BismarkIndex/\"\n",
      "Operator found multiqc with content: ext.args   = params.multiqc_title ? \"--title \\\"$params.multiqc_title\\\"\" : ''\n",
      "Operator found multiqc with content: path: { \"${params.outdir}/multiqc\" },\n",
      "Operator found bwa with content: path: { \"${params.outdir}/alignment/bwa\" },\n",
      "Operator found samtools with content: path: { \"${params.outdir}/reports/samtools_stats/${meta.id}/md/\" },\n",
      "Operator found bwa with content: aligner            = 'bwa-mem'  // Default is bwa-mem, bwa-mem2 and dragmap can be used too\n",
      "Operator found bwa-mem2 with content: aligner            = 'bwa-mem'  // Default is bwa-mem, bwa-mem2 and dragmap can be used too\n",
      "Operator found gatk with content: use_gatk_spark     = null       // GATK Spark implementation of their tools in local mode not used by default\n",
      "Operator found bcftools with content: bcftools_annotations       = null  // No extra annotation file\n",
      "Operator found bcftools with content: bcftools_annotations_index = null  // No extra annotation file index\n",
      "Operator found bcftools with content: bcftools_header_lines      = null  // No header lines to be added to the VCF file\n",
      "Operator found multiqc with content: multiqc_config             = null\n",
      "Operator found multiqc with content: multiqc_title              = null\n",
      "Operator found multiqc with content: multiqc_logo               = null\n",
      "Operator found multiqc with content: max_multiqc_email_size     = '25.MB'\n",
      "Operator found multiqc with content: multiqc_methods_description = null\n",
      "Operator found gatk with content: docker.userEmulation   = { params.use_gatk_spark ? false : true }.call()\n",
      "Operator found gatk with content: use_gatk_spark             { includeConfig 'conf/test/use_gatk_spark.config' }\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Operator found checkm with content: includeConfig 'conf/modules/ngscheckmate.config'\n",
      "Operator found bwa with content: bwa                   = \"${params.igenomes_base}/Homo_sapiens/GATK/GRCh37/Sequence/BWAIndex/\"\n",
      "Operator found checkm with content: ngscheckmate_bed      = \"${params.igenomes_base}/Homo_sapiens/GATK/GRCh37/Annotation/NGSCheckMate/SNP_GRCh37_hg19_wChr.bed\"\n",
      "Operator found bwa with content: bwa                     = \"${params.igenomes_base}/Homo_sapiens/GATK/GRCh38/Sequence/BWAIndex/\"\n",
      "Operator found bwa with content: bwamem2                 = \"${params.igenomes_base}/Homo_sapiens/GATK/GRCh38/Sequence/BWAmem2Index/\"\n",
      "Operator found gatk with content: known_indels_vqsr       = '--resource:gatk,known=false,training=true,truth=true,prior=10.0 Homo_sapiens_assembly38.known_indels.vcf.gz --resource:mills,known=false,training=true,truth=true,prior=10.0 Mills_and_1000G_gold_standard.indels.hg38.vcf.gz'\n",
      "Operator found checkm with content: ngscheckmate_bed        = \"${params.igenomes_base}/Homo_sapiens/GATK/GRCh38/Annotation/NGSCheckMate/SNP_GRCh38_hg38_wChr.bed\"\n",
      "Operator found bwa with content: bwa                   = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found checkm with content: ngscheckmate_bed      = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Annotation/NGSCheckMate/SNP_GRCh37_hg19_woChr.bed\"\n",
      "Operator found bwa with content: bwa                   = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found checkm with content: ngscheckmate_bed      =\"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Annotation/NGSCheckMate/SNP_GRCh38_hg38_wChr.bed\"\n",
      "Operator found bwa with content: bwa                   = \"${params.igenomes_base}/Homo_sapiens/UCSC/CHM13/Sequence/BWAIndex/\"\n",
      "Operator found bwa with content: bwamem2               = \"${params.igenomes_base}/Homo_sapiens/UCSC/CHM13/Sequence/BWAmem2Index/\"\n",
      "Operator found bwa with content: bwa                   = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bwa with content: bwa                   = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bwa with content: bwa                   = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bwa with content: bwa                   = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bwa with content: bwa                   = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bwa with content: bwa                   = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bwa with content: bwa                   = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bwa with content: bwa                   = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bwa with content: bwa                   = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bwa with content: bwa                   = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bwa with content: bwa                   = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bwa with content: bwa                   = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bwa with content: bwa                   = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bwa with content: bwa                   = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bwa with content: bwa                   = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bwa with content: bwa                   = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bwa with content: bwa                   = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bwa with content: bwa                   = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bwa with content: bwa                   = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bwa with content: bwa                   = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bwa with content: bwa                   = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bwa with content: bwa                   = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bwa with content: bwa                   = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bwa with content: bwa                   = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bwa with content: bwa                   = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bwa with content: bwa                   = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bwa with content: bwa                   = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bwa with content: bwa                   = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bwa with content: bwa                   = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bwa with content: bwa                   = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bwa with content: bwa                   = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bwa with content: bwa                   = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bwa with content: bwa                   = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bwa with content: bwa                   = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bwa with content: bwa                   = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bwa with content: bwa                   = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bcftools with content: bcftools_annotations       = \"https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/genomics/sarscov2/illumina/vcf/test2.vcf.gz\"\n",
      "Operator found bcftools with content: bcftools_annotations_index = \"https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/genomics/sarscov2/illumina/vcf/test2.vcf.gz.tbi\"\n",
      "Operator found bcftools with content: bcftools_header_lines      = \"${projectDir}/tests/config/bcfann_test_header.txt\"\n",
      "Operator found checkm with content: ngscheckmate_bed           = \"https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/genomics/homo_sapiens/genome/chr21/germlineresources/SNP_GRCh38_hg38_wChr.bed\"\n",
      "Operator found gatk with content: docker.userEmulation                     = { params.use_gatk_spark ? false : true }.call()\n",
      "Operator found checkm with content: tools       = 'ngscheckmate,strelka,mutect2,freebayes,ascat,manta,cnvkit,tiddit,controlfreec,vep,snpeff'\n",
      "Operator found bwa with content: ext.when   = { params.aligner == 'bwa-mem' }\n",
      "Operator found bwa with content: ext.when   = { params.aligner == 'bwa-mem2' }\n",
      "Operator found bwa-mem2 with content: ext.when   = { params.aligner == 'bwa-mem2' }\n",
      "Operator found bwa with content: ext.when   = { params.aligner == 'sentieon-bwamem' }\n",
      "Operator found gatk with content: ext.args2  = { params.use_gatk_spark && params.use_gatk_spark.contains('markduplicates') && (!params.skip_tools || (params.skip_tools && !params.skip_tools.split(',').contains('markduplicates'))) ? '-n' : '' }\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Operator found samtools with content: ext.when   = { !(params.skip_tools && params.skip_tools.split(',').contains('samtools')) }\n",
      "Operator found samtools with content: path: { \"${params.outdir}/reports/samtools/${meta.id}\" },\n",
      "Operator found fastqc with content: ext.when   = { !(params.skip_tools && params.skip_tools.split(',').contains('fastqc')) }\n",
      "Operator found fastqc with content: path: { \"${params.outdir}/reports/fastqc/${meta.id}\" },\n",
      "Operator found multiqc with content: ext.args   = { params.multiqc_title ? \"--title \\\"$params.multiqc_title\\\"\" : '' }\n",
      "Operator found multiqc with content: path: { \"${params.outdir}/multiqc\" },\n",
      "Operator found samtools with content: ext.when   = { !(params.skip_tools && params.skip_tools.split(',').contains('samtools')) }\n",
      "Operator found samtools with content: path: { \"${params.outdir}/reports/samtools/${meta.id}\" },\n",
      "Operator found samtools with content: ext.when   = { !(params.skip_tools && params.skip_tools.split(',').contains('samtools')) }\n",
      "Operator found samtools with content: path: { \"${params.outdir}/reports/samtools/${meta.id}\" },\n",
      "Operator found bcftools with content: ext.when   = { !(params.skip_tools && params.skip_tools.split(',').contains('bcftools')) }\n",
      "Operator found bcftools with content: path: { \"${params.outdir}/reports/bcftools/${meta.variantcaller}/${meta.id}/\" },\n",
      "Operator found vcftools with content: ext.when   = { !(params.skip_tools && params.skip_tools.split(',').contains('vcftools')) }\n",
      "Operator found vcftools with content: path: { \"${params.outdir}/reports/vcftools/${meta.variantcaller}/${meta.id}/\" },\n",
      "Operator found bcftools with content: ext.prefix = { meta.num_intervals <= 1 ? \"${meta.id}.bcftools\" : \"${meta.id}_${intervals.simpleName}.bcftools\" }\n",
      "Operator found bcftools with content: path: { \"${params.outdir}/variant_calling/bcftools/${meta.id}/\" },\n",
      "Operator found bcftools with content: ext.prefix = {\"${meta.id}.bcftools\"}\n",
      "Operator found bcftools with content: path: { \"${params.outdir}/variant_calling/bcftools/${meta.id}/\" },\n",
      "Operator found checkm with content: ext.prefix = { \"${meta.id}.ngscheckmate\" }\n",
      "Operator found checkm with content: ext.when   = { params.tools && params.tools.split(',').contains('ngscheckmate') }\n",
      "Operator found checkm with content: path: { \"${params.outdir}/reports/ngscheckmate/vcfs\" },\n",
      "Operator found checkm with content: path: { \"${params.outdir}/reports/ngscheckmate/\" },\n",
      "Operator found bwa with content: ext.when         = { !params.bwa && params.step == \"mapping\" && (params.aligner == \"bwa-mem\" || params.aligner == \"sentieon-bwamem\")}\n",
      "Operator found bwa with content: pattern: \"bwa\",\n",
      "Operator found bwa with content: ext.when         = { !params.bwamem2 && params.step == \"mapping\" && params.aligner == \"bwa-mem2\" }\n",
      "Operator found bwa-mem2 with content: ext.when         = { !params.bwamem2 && params.step == \"mapping\" && params.aligner == \"bwa-mem2\" }\n",
      "Operator found bwa with content: pattern: \"bwamem2\",\n",
      "Operator found multiqc with content: pattern: \"*{metrics,metrics.multiqc.tsv}\",\n",
      "Operator found samtools with content: ext.when         = { !(params.skip_tools && params.skip_tools.split(',').contains('samtools')) }\n",
      "Operator found samtools with content: path: { \"${params.outdir}/reports/samtools/${meta.id}\" },\n",
      "Operator found bwa with content: ext.args   = { bwa_index ? '' : '--skip_assembly' }\n",
      "Operator found fastp with content: path: { \"${params.outdir}/reports/fastp/${meta.sample}\" },\n",
      "Operator found fastp with content: path: { \"${params.outdir}/preprocessing/fastp/${meta.sample}/\" },\n",
      "Operator found fastp with content: pattern: \"*.fastp.fastq.gz\",\n",
      "Operator found bcftools with content: bcftools_annotations       = params.test_data['sarscov2']['illumina']['test2_vcf_gz']\n",
      "Operator found bcftools with content: bcftools_annotations_index = params.test_data['sarscov2']['illumina']['test2_vcf_gz_tbi']\n",
      "Operator found bcftools with content: bcftools_header_lines      = \"${projectDir}/tests/config/bcfann_test_header.txt\"\n",
      "Operator found checkm with content: ngscheckmate_bed           = params.test_data['homo_sapiens']['genome']['ngscheckmate_bed']\n",
      "Operator found gatk with content: docker.userEmulation                     = { params.use_gatk_spark ? false : true }.call()\n",
      "Operator found gatk with content: use_gatk_spark = 'baserecalibrator,markduplicates'\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BismarkIndex/\"\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BismarkIndex/\"\n",
      "Operator found salmon with content: salmon_index      = null\n",
      "Operator found kallisto with content: kallisto_gene_map = null\n",
      "Operator found kallisto with content: kallisto_index    = null\n",
      "Operator found multiqc with content: skip_multiqc               = false\n",
      "Operator found fastqc with content: skip_fastqc                = false\n",
      "Operator found multiqc with content: multiqc_config             = null\n",
      "Operator found multiqc with content: multiqc_title              = null\n",
      "Operator found multiqc with content: multiqc_logo               = null\n",
      "Operator found multiqc with content: max_multiqc_email_size     = '25.MB'\n",
      "Operator found multiqc with content: multiqc_methods_description = null\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/CHM13/Sequence/BWAIndex/\"\n",
      "Operator found bwa with content: bwamem2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/CHM13/Sequence/BWAmem2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BismarkIndex/\"\n",
      "Operator found multiqc with content: ext.args   = { params.multiqc_title ? \"--title \\\"$params.multiqc_title\\\"\" : '' }\n",
      "Operator found multiqc with content: path: { \"${params.outdir}/multiqc\" },\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Operator found gffread with content: ext.prefix = { \"${gff.baseName}_gffread\" }\n",
      "Operator found kallisto with content: if (params.aligner == 'kallisto') {\n",
      "Operator found multiqc with content: multiqc_config = false\n",
      "Operator found multiqc with content: max_multiqc_email_size = 25.MB\n",
      "Operator found multiqc with content: withName: multiqc {\n",
      "Operator found fastqc with content: skip_fastqc = false\n",
      "Operator found multiqc with content: multiqc_config = false\n",
      "Operator found multiqc with content: max_multiqc_email_size = 25.MB\n",
      "Operator found multiqc with content: withName: 'multiqc' {\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BismarkIndex/\"\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Operator found fastp with content: fastp_min_length            = 17\n",
      "Operator found fastp with content: fastp_known_mirna_adapters  = \"$projectDir/assets/known_adapters.fa\"\n",
      "Operator found fastqc with content: skip_fastqc                 = false\n",
      "Operator found multiqc with content: skip_multiqc                = false\n",
      "Operator found fastp with content: skip_fastp                  = false\n",
      "Operator found fastp with content: fastp_max_length            = 40\n",
      "Operator found multiqc with content: multiqc_config              = null\n",
      "Operator found multiqc with content: multiqc_title               = null\n",
      "Operator found multiqc with content: multiqc_logo                = null\n",
      "Operator found multiqc with content: max_multiqc_email_size      = '25.MB'\n",
      "Operator found multiqc with content: multiqc_methods_description = null\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/CHM13/Sequence/BWAIndex/\"\n",
      "Operator found bwa with content: bwamem2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/CHM13/Sequence/BWAmem2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BismarkIndex/\"\n",
      "Operator found fastqc with content: if (!(params.skip_fastqc)) {\n",
      "Operator found fastp with content: if (!params.skip_fastp) {\n",
      "Operator found fastp with content: params.fastp_min_length > 0    ? \"-l ${params.fastp_min_length}\" : \"\",\n",
      "Operator found fastp with content: params.fastp_max_length > 0    ? \"--max_len1 ${params.fastp_max_length}\" : \"\",\n",
      "Operator found fastp with content: path: { \"${params.outdir}/fastp\" },\n",
      "Operator found fastp with content: path: { \"${params.outdir}/fastp/log\" },\n",
      "Operator found fastp with content: path: { \"${params.outdir}/fastp\" },\n",
      "Operator found fastqc with content: if (!params.skip_fastqc) {\n",
      "Operator found fastqc with content: path: { \"${params.outdir}/fastqc/trim\" },\n",
      "Operator found samtools with content: path: { \"${params.outdir}/samtools\" },\n",
      "Operator found samtools with content: path: { \"${params.outdir}/samtools/samtools_stats\" },\n",
      "Operator found samtools with content: path: { \"${params.outdir}/samtools\" },\n",
      "Operator found samtools with content: path: { \"${params.outdir}/samtools/samtools_stats\" },\n",
      "Operator found multiqc with content: if (!params.skip_multiqc) {\n",
      "Operator found multiqc with content: ext.args   = params.multiqc_title ? \"--title \\\"$params.multiqc_title\\\"\" : ''\n",
      "Operator found multiqc with content: path: { \"${params.outdir}/multiqc\" },\n",
      "Operator found multiqc with content: multiqc_config               = null\n",
      "Operator found multiqc with content: multiqc_title                = null\n",
      "Operator found multiqc with content: multiqc_logo                 = null\n",
      "Operator found multiqc with content: max_multiqc_email_size       = '25.MB'\n",
      "Operator found multiqc with content: multiqc_methods_description  = null\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Operator found fastqc with content: path: { \"${params.outdir}/${meta.id}/fastqc\" },\n",
      "Operator found multiqc with content: multiqc_config             = null\n",
      "Operator found multiqc with content: multiqc_title              = null\n",
      "Operator found multiqc with content: multiqc_logo               = null\n",
      "Operator found multiqc with content: max_multiqc_email_size     = '25.MB'\n",
      "Operator found multiqc with content: multiqc_methods_description = null\n",
      "Operator found multiqc with content: multiqc_config             = \"${projectDir}/conf/multiqc_ssds_config.yaml\"\n",
      "Operator found multiqc with content: multiqc_title              = 'SSDS'\n",
      "Operator found multiqc with content: max_multiqc_email_size     = '25.MB'\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BismarkIndex/\"\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BWAIndex/genome.fa\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BismarkIndex/\"\n",
      "Operator found fastqc with content: 'fastqc' {\n",
      "Operator found multiqc with content: 'multiqc' {\n",
      "Operator found picard with content: 'picard/sortsam_parsed' {\n",
      "Operator found picard with content: 'picard/sortsam_co' {\n",
      "Operator found picard with content: 'picard/sortsam_qry' {\n",
      "Operator found picard with content: 'picard/markduplicates' {\n",
      "Operator found picard with content: 'picard/collectmultiplemetrics' {\n",
      "Operator found bwa with content: 'bwa/mem' {\n",
      "Operator found bwa with content: 'bwa/index' {\n",
      "Operator found bedtools with content: 'bedtools/sort' {\n",
      "Operator found bedtools with content: 'bedtools/makewindows' {\n",
      "Operator found samtools with content: 'samtools/index' {\n",
      "Operator found samtools with content: 'samtools/idxstats' {\n",
      "Operator found samtools with content: 'samtools/view' {\n",
      "Operator found bwa with content: ext.prefix = { \"${meta.id}.bwa_initial_alignment\" }\n",
      "Operator found multiqc with content: params.multiqc_title ? \"--title \\\"${params.multiqc_title}\\\"\" : \"SSDS\",\n",
      "Operator found multiqc with content: params.multiqc_config ? \"-c multiqc_ssds_config.yaml\" : \"\"\n",
      "Operator found multiqc with content: path: { \"${params.outdir}/multiqc\" },\n",
      "Operator found multiqc with content: multiqc_config             = null\n",
      "Operator found multiqc with content: multiqc_title              = null\n",
      "Operator found multiqc with content: multiqc_logo               = null\n",
      "Operator found multiqc with content: max_multiqc_email_size     = '25.MB'\n",
      "Operator found multiqc with content: multiqc_methods_description = null\n",
      "Operator found fastqc with content: preprocessing_qc_tool            = 'fastqc'\n",
      "Operator found fastp with content: shortread_qc_tool                = 'fastp'\n",
      "Operator found fastp with content: shortread_complexityfilter_fastp_threshold           = 30\n",
      "Operator found kraken2 with content: run_kraken2                      = false\n",
      "Operator found kraken2 with content: kraken2_save_reads               = false // added directly to module in profiling.nf\n",
      "Operator found kraken2 with content: kraken2_save_readclassifications = false // added directly to module in profiling.nf\n",
      "Operator found kraken2 with content: kraken2_save_minimizers          = false\n",
      "Operator found fastp with content: test_fastp              { includeConfig 'conf/test_fastp.config' }\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/CHM13/Sequence/BWAIndex/\"\n",
      "Operator found bwa with content: bwamem2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/CHM13/Sequence/BWAmem2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BismarkIndex/\"\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BismarkIndex/\"\n",
      "Operator found fastqc with content: path: { \"${params.outdir}/fastqc/raw\" },\n",
      "Operator found fastqc with content: path: { \"${params.outdir}/fastqc/processed\" },\n",
      "Operator found fastp with content: (params.perform_shortread_complexityfilter && params.shortread_complexityfilter_tool == 'fastp') ? \"--low_complexity_filter --complexity_threshold ${params.shortread_complexityfilter_fastp_threshold}\" : ''\n",
      "Operator found fastp with content: path: { \"${params.outdir}/fastp\" },\n",
      "Operator found fastp with content: path: { \"${params.outdir}/fastp\" },\n",
      "Operator found fastp with content: params.perform_shortread_complexityfilter && params.shortread_complexityfilter_tool == 'fastp' ? \"--low_complexity_filter --complexity_threshold ${params.shortread_complexityfilter_fastp_threshold}\" : ''\n",
      "Operator found fastp with content: path: { \"${params.outdir}/fastp\" },\n",
      "Operator found fastp with content: path: { \"${params.outdir}/fastp\" },\n",
      "Operator found fastp with content: pattern: params.shortread_qc_mergepairs ? '*merged.fastq.gz' : '*.fastp.fastq.gz',\n",
      "Operator found bowtie2 with content: path: { \"${params.outdir}/bowtie2/build\" },\n",
      "Operator found bowtie2 with content: pattern: 'bowtie2',\n",
      "Operator found bowtie2 with content: path: { \"${params.outdir}/bowtie2/align\" },\n",
      "Operator found bowtie2 with content: path: { \"${params.outdir}/bowtie2/align\" },\n",
      "Operator found bowtie2 with content: path: { \"${params.outdir}/bowtie2/align\" },\n",
      "Operator found minimap2 with content: path: { \"${params.outdir}/minimap2/index\" },\n",
      "Operator found minimap2 with content: path: { \"${params.outdir}/minimap2/align\" },\n",
      "Operator found samtools with content: path: { \"${params.outdir}/samtools/fastq\" },\n",
      "Operator found samtools with content: path: { \"${params.outdir}/samtools/stats\" },\n",
      "Operator found kraken2 with content: ext.args = params.kraken2_save_minimizers ? { \"${meta.db_params} --report-minimizer-data\" } : { \"${meta.db_params}\" }\n",
      "Operator found kraken2 with content: ext.prefix = params.perform_runmerging ? { meta.tool == \"bracken\" ? \"${meta.id}_${meta.db_name}.bracken\" : \"${meta.id}_${meta.db_name}.kraken2\" } : { meta.tool == \"bracken\" ? \"${meta.id}_${meta.run_accession}_${meta.db_name}.bracken\" : \"${meta.id}_${meta.run_accession}_${meta.db_name}.kraken2\" }\n",
      "Operator found kraken2 with content: path: { \"${params.outdir}/kraken2/${meta.db_name}/\" },\n",
      "Operator found kraken2 with content: ext.prefix = params.perform_runmerging ? { meta.tool == \"bracken\" ? \"${meta.id}_${meta.db_name}.bracken\" : \"${meta.id}_${meta.db_name}.kraken2\" } : { meta.tool == \"bracken\" ? \"${meta.id}_${meta.run_accession}_${meta.db_name}.bracken\" : \"${meta.id}_${meta.run_accession}_${meta.db_name}.kraken2\" }\n",
      "Operator found kraken2 with content: path: { \"${params.outdir}/kraken2/${meta.db_name}/\" },\n",
      "Operator found kraken2 with content: ext.prefix = { \"kraken2_${meta.db_name}_combined_reports\" }\n",
      "Operator found kraken2 with content: path: { \"${params.outdir}/kraken2/\" },\n",
      "Operator found multiqc with content: ext.args   = { params.multiqc_title ? \"--title \\\"$params.multiqc_title\\\"\" : '' }\n",
      "Operator found multiqc with content: path: { \"${params.outdir}/multiqc\" },\n",
      "Operator found kraken2 with content: run_kraken2                           = true\n",
      "Operator found kraken2 with content: kraken2_save_reads                    = true\n",
      "Operator found kraken2 with content: run_kraken2                           = true\n",
      "Operator found kraken2 with content: run_kraken2                           = true\n",
      "Operator found fastqc with content: config_profile_description = \"Minimal test dataset without performing any preprocessing nor profiling to check pipeline function but running falco instead of fastqc. Useful when you only wish to test a single profiler without having to 'opt-out' of all the others\"\n",
      "Operator found kraken2 with content: run_kraken2                           = false\n",
      "Operator found fastp with content: config_profile_name        = 'Test profile for fastp'\n",
      "Operator found fastp with content: config_profile_description = \"Minimal test to check the default short-read QC function, fastp\"\n",
      "Operator found fastp with content: shortread_qc_tool                     = 'fastp'\n",
      "Operator found fastp with content: shortread_complexityfilter_tool       = 'fastp'\n",
      "Operator found kraken2 with content: run_kraken2                           = true\n",
      "Operator found kraken2 with content: run_kraken2                         = true\n",
      "Operator found kraken2 with content: kraken2_save_reads                  = false\n",
      "Operator found kraken2 with content: kraken2_save_readclassifications    = false\n",
      "Operator found kraken2 with content: kraken2_save_minimizers             = false\n",
      "Operator found kraken2 with content: run_kraken2                           = false\n",
      "Operator found kraken2 with content: kraken2_save_reads                    = false\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Operator found kraken2 with content: run_kraken2                           = false\n",
      "Operator found kraken2 with content: run_kraken2                           = false\n",
      "Operator found kraken2 with content: run_kraken2                           = true\n",
      "Operator found kraken2 with content: run_kraken2                           = false\n",
      "Operator found kraken2 with content: run_kraken2                           = false\n",
      "Operator found kraken2 with content: run_kraken2                           = true\n",
      "Operator found multiqc with content: multiqc_config             = null\n",
      "Operator found multiqc with content: multiqc_title              = null\n",
      "Operator found multiqc with content: multiqc_logo               = null\n",
      "Operator found multiqc with content: max_multiqc_email_size     = '25.MB'\n",
      "Operator found multiqc with content: multiqc_methods_description = null\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/CHM13/Sequence/BWAIndex/\"\n",
      "Operator found bwa with content: bwamem2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/CHM13/Sequence/BWAmem2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BismarkIndex/\"\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Operator found multiqc with content: multiqc_config             = null\n",
      "Operator found multiqc with content: multiqc_title              = null\n",
      "Operator found multiqc with content: multiqc_logo               = null\n",
      "Operator found multiqc with content: max_multiqc_email_size     = '25.MB'\n",
      "Operator found multiqc with content: multiqc_methods_description = null\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/Ensembl/GRCh37/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/CHM13/Sequence/BWAIndex/\"\n",
      "Operator found bwa with content: bwamem2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/CHM13/Sequence/BWAmem2Index/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/Ensembl/GRCm38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Arabidopsis_thaliana/Ensembl/TAIR10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bacillus_subtilis_168/Ensembl/EB2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/Ensembl/UMD3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/Ensembl/WBcel235/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/Ensembl/CanFam3.1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/Ensembl/GRCz10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/Ensembl/BDGP6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/Ensembl/EquCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Escherichia_coli_K_12_DH10B/Ensembl/EB1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/Ensembl/Galgal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Glycine_max/Ensembl/Gm01/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Macaca_mulatta/Ensembl/Mmul_1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Oryza_sativa_japonica/Ensembl/IRGSP-1.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/Ensembl/CHIMP2.1.4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_5.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/Ensembl/Rnor_6.0/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Schizosaccharomyces_pombe/Ensembl/EF2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sorghum_bicolor/Ensembl/Sbi1/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/Ensembl/Sscrofa10.2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Zea_mays/Ensembl/AGPv3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg38/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Homo_sapiens/UCSC/hg19/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Mus_musculus/UCSC/mm10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Bos_taurus/UCSC/bosTau8/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Caenorhabditis_elegans/UCSC/ce10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Canis_familiaris/UCSC/canFam3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Danio_rerio/UCSC/danRer10/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Drosophila_melanogaster/UCSC/dm6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Equus_caballus/UCSC/equCab2/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Gallus_gallus/UCSC/galGal4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Pan_troglodytes/UCSC/panTro4/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Rattus_norvegicus/UCSC/rn6/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Saccharomyces_cerevisiae/UCSC/sacCer3/Sequence/BismarkIndex/\"\n",
      "Operator found bwa with content: bwa         = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BWAIndex/version0.6.0/\"\n",
      "Operator found bowtie2 with content: bowtie2     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/Bowtie2Index/\"\n",
      "Operator found bismark with content: bismark     = \"${params.igenomes_base}/Sus_scrofa/UCSC/susScr3/Sequence/BismarkIndex/\"\n",
      "Operator found minimap2 with content: artic_minion_aligner       = 'minimap2'\n",
      "Operator found multiqc with content: multiqc_title              = null\n",
      "Operator found multiqc with content: multiqc_config             = null\n",
      "Operator found multiqc with content: max_multiqc_email_size     = '25.MB'\n",
      "Operator found multiqc with content: skip_multiqc               = false\n",
      "Operator found kraken2 with content: kraken2_db                 = 's3://ngi-igenomes/test-data/viralrecon/kraken2_human.tar.gz'\n",
      "Operator found kraken2 with content: kraken2_db_name            = 'human'\n",
      "Operator found kraken2 with content: kraken2_variants_host_filter = false\n",
      "Operator found kraken2 with content: kraken2_assembly_host_filter = true\n",
      "Operator found fastqc with content: skip_fastqc                = false\n",
      "Operator found kraken2 with content: skip_kraken2               = false\n",
      "Operator found fastp with content: skip_fastp                 = false\n",
      "Operator found cutadapt with content: skip_cutadapt              = false\n",
      "Operator found bcftools with content: consensus_caller           = 'bcftools'\n",
      "Operator found picard with content: skip_picard_metrics        = false\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Operator found bcftools with content: if (!variant_caller) { variant_caller = params.protocol == 'amplicon' ? 'ivar' : 'bcftools' }\n",
      "Operator found fastqc with content: if (!params.skip_fastqc) {\n",
      "Operator found fastqc with content: path: { \"${params.outdir}/fastqc/raw\" },\n",
      "Operator found fastp with content: if (!params.skip_fastp) {\n",
      "Operator found fastp with content: path: { \"${params.outdir}/fastp\" },\n",
      "Operator found fastp with content: path: { \"${params.outdir}/fastp/log\" },\n",
      "Operator found fastp with content: path: { \"${params.outdir}/fastp\" },\n",
      "Operator found fastqc with content: if (!params.skip_fastqc) {\n",
      "Operator found fastqc with content: path: { \"${params.outdir}/fastqc/trim\" },\n",
      "Operator found kraken2 with content: if (!params.skip_kraken2) {\n",
      "Operator found kraken2 with content: path: { \"${params.outdir}/kraken2\" },\n",
      "Operator found bowtie2 with content: path: { \"${params.outdir}/variants/bowtie2/log\" },\n",
      "Operator found bowtie2 with content: path: { \"${params.outdir}/variants/bowtie2/unmapped\" },\n",
      "Operator found bowtie2 with content: path: { \"${params.outdir}/variants/bowtie2\" },\n",
      "Operator found bowtie2 with content: path: { \"${params.outdir}/variants/bowtie2\" },\n",
      "Operator found samtools with content: path: { \"${params.outdir}/variants/bowtie2/samtools_stats\" },\n",
      "Operator found bowtie2 with content: path: { \"${params.outdir}/variants/bowtie2/samtools_stats\" },\n",
      "Operator found bowtie2 with content: path: { \"${params.outdir}/variants/bowtie2/log\" },\n",
      "Operator found bowtie2 with content: path: { \"${params.outdir}/variants/bowtie2\" },\n",
      "Operator found bowtie2 with content: path: { \"${params.outdir}/variants/bowtie2\" },\n",
      "Operator found samtools with content: path: { \"${params.outdir}/variants/bowtie2/samtools_stats\" },\n",
      "Operator found bowtie2 with content: path: { \"${params.outdir}/variants/bowtie2/samtools_stats\" },\n",
      "Operator found picard with content: path: { \"${params.outdir}/variants/bowtie2/picard_metrics\" },\n",
      "Operator found bowtie2 with content: path: { \"${params.outdir}/variants/bowtie2/picard_metrics\" },\n",
      "Operator found bowtie2 with content: path: { \"${params.outdir}/variants/bowtie2\" },\n",
      "Operator found bowtie2 with content: path: { \"${params.outdir}/variants/bowtie2\" },\n",
      "Operator found samtools with content: path: { \"${params.outdir}/variants/bowtie2/samtools_stats\" },\n",
      "Operator found bowtie2 with content: path: { \"${params.outdir}/variants/bowtie2/samtools_stats\" },\n",
      "Operator found picard with content: if (!params.skip_picard_metrics) {\n",
      "Operator found picard with content: path: { \"${params.outdir}/variants/bowtie2/picard_metrics\" },\n",
      "Operator found bowtie2 with content: path: { \"${params.outdir}/variants/bowtie2/picard_metrics\" },\n",
      "Operator found picard with content: path: { \"${params.outdir}/variants/bowtie2/picard_metrics/pdf\" },\n",
      "Operator found bowtie2 with content: path: { \"${params.outdir}/variants/bowtie2/picard_metrics/pdf\" },\n",
      "Operator found bowtie2 with content: path: { \"${params.outdir}/variants/bowtie2/mosdepth/genome\" },\n",
      "Operator found bowtie2 with content: path: { \"${params.outdir}/variants/bowtie2/mosdepth/genome\" },\n",
      "Operator found bowtie2 with content: path: { \"${params.outdir}/variants/bowtie2/mosdepth/amplicon\" },\n",
      "Operator found bowtie2 with content: path: { \"${params.outdir}/variants/bowtie2/mosdepth/amplicon\" },\n",
      "Operator found bcftools with content: path: { \"${params.outdir}/variants/ivar/bcftools_stats\" },\n",
      "Operator found bcftools with content: if (variant_caller == 'bcftools') {\n",
      "Operator found bcftools with content: path: { \"${params.outdir}/variants/bcftools\" },\n",
      "Operator found bcftools with content: path: { \"${params.outdir}/variants/bcftools\" },\n",
      "Operator found bcftools with content: path: { \"${params.outdir}/variants/bcftools\" },\n",
      "Operator found bcftools with content: path: { \"${params.outdir}/variants/bcftools/bcftools_stats\" },\n",
      "Operator found bcftools with content: path: { \"${params.outdir}/variants/${variant_caller}/snpeff/bcftools_stats\" },\n",
      "Operator found bcftools with content: variant_caller == 'bcftools' ? \"-H -f '%CHROM\\\\t%POS\\\\t%REF\\\\t%ALT\\\\t%FILTER\\\\t[%DP\\\\t]\\\\t[%AD\\\\t]\\\\n'\" : '',\n",
      "Operator found bcftools with content: ext.prefix = { \"${meta.id}.bcftools_query\" }\n",
      "Operator found bcftools with content: if (!params.skip_consensus && params.consensus_caller == 'bcftools') {\n",
      "Operator found bcftools with content: variant_caller == 'bcftools' ? \"--include 'FORMAT/AD[:1] / FORMAT/DP >= 0.75'\" : '',\n",
      "Operator found bcftools with content: path: { \"${params.outdir}/variants/${variant_caller}/consensus/bcftools\" },\n",
      "Operator found bcftools with content: path: { \"${params.outdir}/variants/${variant_caller}/consensus/bcftools\" },\n",
      "Operator found bcftools with content: path: { \"${params.outdir}/variants/${variant_caller}/consensus/bcftools\" },\n",
      "Operator found bcftools with content: path: { \"${params.outdir}/variants/${variant_caller}/consensus/bcftools\" },\n",
      "Operator found bcftools with content: path: { \"${params.outdir}/variants/${variant_caller}/consensus/bcftools\" },\n",
      "Operator found bcftools with content: path: { \"${params.outdir}/variants/${variant_caller}/consensus/bcftools\" },\n",
      "Operator found bcftools with content: path: { \"${params.outdir}/variants/${variant_caller}/consensus/bcftools\" },\n",
      "Operator found cutadapt with content: if (params.protocol == 'amplicon' && !params.skip_cutadapt) {\n",
      "Operator found cutadapt with content: path: { \"${params.outdir}/assembly/cutadapt/log\" },\n",
      "Operator found fastqc with content: if (!params.skip_fastqc) {\n",
      "Operator found fastqc with content: path: { \"${params.outdir}/assembly/cutadapt/fastqc\" },\n",
      "Operator found cutadapt with content: path: { \"${params.outdir}/assembly/cutadapt/fastqc\" },\n",
      "Operator found multiqc with content: if (!params.skip_multiqc) {\n",
      "Operator found multiqc with content: params.multiqc_title ? \"--title \\\"$params.multiqc_title\\\"\" : ''\n",
      "Operator found multiqc with content: path: { \"${params.outdir}/multiqc\" },\n",
      "Operator found multiqc with content: pattern: 'multiqc*'\n",
      "Operator found multiqc with content: path: { \"${params.outdir}/multiqc\" },\n",
      "Operator found multiqc with content: path: { \"${params.outdir}/multiqc\" },\n",
      "Operator found bwa with content: params.artic_minion_aligner == 'bwa'    ? '--bwa'    : '--minimap2'\n",
      "Operator found minimap2 with content: params.artic_minion_aligner == 'bwa'    ? '--bwa'    : '--minimap2'\n",
      "Operator found samtools with content: path: { \"${params.outdir}/${params.artic_minion_caller}/samtools_stats\" },\n",
      "Operator found bcftools with content: path: { \"${params.outdir}/${params.artic_minion_caller}/bcftools_stats\" },\n",
      "Operator found bcftools with content: path: { \"${params.outdir}/${params.artic_minion_caller}/snpeff/bcftools_stats\" },\n",
      "Operator found bcftools with content: ext.prefix = { \"${meta.id}.bcftools_query\" }\n",
      "Operator found multiqc with content: if (!params.skip_multiqc) {\n",
      "Operator found multiqc with content: params.multiqc_title ? \"--title \\\"$params.multiqc_title\\\"\" : ''\n",
      "Operator found multiqc with content: path: { \"${params.outdir}/multiqc/${params.artic_minion_caller}\" },\n",
      "Operator found kraken2 with content: kraken2_db = 'https://raw.githubusercontent.com/nf-core/test-datasets/viralrecon/genome/kraken2/kraken2_hs22.tar.gz'\n",
      "Operator found bcftools with content: variant_caller = 'bcftools'\n",
      "Operator found kraken2 with content: kraken2_db = 'https://raw.githubusercontent.com/nf-core/test-datasets/viralrecon/genome/kraken2/kraken2_hs22.tar.gz'\n",
      "Operator found bcftools with content: variant_caller = 'bcftools'\n"
     ]
    }
   ],
   "source": [
    "directory_path = './git_repos'\n",
    "\n",
    "df_operators = pd.read_excel('./data/bioinformatics_tools.xlsx')\n",
    "operator_list = [str(pattern) for pattern in df_operators['operator']]\n",
    "\n",
    "output_file_name = 'bioinformatic_tools_in_config'\n",
    "\n",
    "find_operators_in_config(directory_path, operator_list, output_file_name)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "10cbb88c",
   "metadata": {},
   "source": [
    "# Bioinformatic tools in configuration files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "b7275dd7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pipeline</th>\n",
       "      <th>operator</th>\n",
       "      <th>file_path</th>\n",
       "      <th>line_number</th>\n",
       "      <th>line_content</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>airrflow</td>\n",
       "      <td>multiqc</td>\n",
       "      <td>./git_repos\\airrflow\\nextflow.config</td>\n",
       "      <td>107</td>\n",
       "      <td>skip_multiqc               = false</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>airrflow</td>\n",
       "      <td>multiqc</td>\n",
       "      <td>./git_repos\\airrflow\\nextflow.config</td>\n",
       "      <td>108</td>\n",
       "      <td>multiqc_config             = null</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>airrflow</td>\n",
       "      <td>multiqc</td>\n",
       "      <td>./git_repos\\airrflow\\nextflow.config</td>\n",
       "      <td>109</td>\n",
       "      <td>multiqc_title              = null</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>airrflow</td>\n",
       "      <td>multiqc</td>\n",
       "      <td>./git_repos\\airrflow\\nextflow.config</td>\n",
       "      <td>110</td>\n",
       "      <td>multiqc_logo               = null</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>airrflow</td>\n",
       "      <td>multiqc</td>\n",
       "      <td>./git_repos\\airrflow\\nextflow.config</td>\n",
       "      <td>111</td>\n",
       "      <td>max_multiqc_email_size     = '25.MB'</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8299</th>\n",
       "      <td>viralrecon</td>\n",
       "      <td>multiqc</td>\n",
       "      <td>./git_repos\\viralrecon\\conf\\modules_nanopore.c...</td>\n",
       "      <td>345</td>\n",
       "      <td>path: { \"${params.outdir}/multiqc/${params.art...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8300</th>\n",
       "      <td>viralrecon</td>\n",
       "      <td>kraken2</td>\n",
       "      <td>./git_repos\\viralrecon\\conf\\test.config</td>\n",
       "      <td>31</td>\n",
       "      <td>kraken2_db = 'https://raw.githubusercontent.co...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8301</th>\n",
       "      <td>viralrecon</td>\n",
       "      <td>bcftools</td>\n",
       "      <td>./git_repos\\viralrecon\\conf\\test_full_sispa.co...</td>\n",
       "      <td>26</td>\n",
       "      <td>variant_caller = 'bcftools'</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8302</th>\n",
       "      <td>viralrecon</td>\n",
       "      <td>kraken2</td>\n",
       "      <td>./git_repos\\viralrecon\\conf\\test_sispa.config</td>\n",
       "      <td>29</td>\n",
       "      <td>kraken2_db = 'https://raw.githubusercontent.co...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8303</th>\n",
       "      <td>viralrecon</td>\n",
       "      <td>bcftools</td>\n",
       "      <td>./git_repos\\viralrecon\\conf\\test_sispa.config</td>\n",
       "      <td>32</td>\n",
       "      <td>variant_caller = 'bcftools'</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>8304 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        pipeline  operator                                          file_path  \\\n",
       "0       airrflow   multiqc               ./git_repos\\airrflow\\nextflow.config   \n",
       "1       airrflow   multiqc               ./git_repos\\airrflow\\nextflow.config   \n",
       "2       airrflow   multiqc               ./git_repos\\airrflow\\nextflow.config   \n",
       "3       airrflow   multiqc               ./git_repos\\airrflow\\nextflow.config   \n",
       "4       airrflow   multiqc               ./git_repos\\airrflow\\nextflow.config   \n",
       "...          ...       ...                                                ...   \n",
       "8299  viralrecon   multiqc  ./git_repos\\viralrecon\\conf\\modules_nanopore.c...   \n",
       "8300  viralrecon   kraken2            ./git_repos\\viralrecon\\conf\\test.config   \n",
       "8301  viralrecon  bcftools  ./git_repos\\viralrecon\\conf\\test_full_sispa.co...   \n",
       "8302  viralrecon   kraken2      ./git_repos\\viralrecon\\conf\\test_sispa.config   \n",
       "8303  viralrecon  bcftools      ./git_repos\\viralrecon\\conf\\test_sispa.config   \n",
       "\n",
       "      line_number                                       line_content  \n",
       "0             107                 skip_multiqc               = false  \n",
       "1             108                  multiqc_config             = null  \n",
       "2             109                  multiqc_title              = null  \n",
       "3             110                  multiqc_logo               = null  \n",
       "4             111               max_multiqc_email_size     = '25.MB'  \n",
       "...           ...                                                ...  \n",
       "8299          345  path: { \"${params.outdir}/multiqc/${params.art...  \n",
       "8300           31  kraken2_db = 'https://raw.githubusercontent.co...  \n",
       "8301           26                        variant_caller = 'bcftools'  \n",
       "8302           29  kraken2_db = 'https://raw.githubusercontent.co...  \n",
       "8303           32                        variant_caller = 'bcftools'  \n",
       "\n",
       "[8304 rows x 5 columns]"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "\n",
    "df_bioinformatics = pd.read_csv('./results/bioinformatic_tools_in_config.csv')\n",
    "df_bioinformatics"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "cdbbc403",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Matched in 88 pipelines and 332 config files\n"
     ]
    }
   ],
   "source": [
    "pipelines_total = df_bioinformatics.groupby('pipeline').size().reset_index(name='bioinformatic_occur')\n",
    "config_count = df_bioinformatics.groupby('file_path').size().reset_index(name='bioinformatic_occur')\n",
    "\n",
    "print(f'Matched in {len(pipelines_total)} pipelines and {len(config_count)} config files')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "95923af2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pipeline</th>\n",
       "      <th>operator</th>\n",
       "      <th>file_path</th>\n",
       "      <th>line_number</th>\n",
       "      <th>line_content</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>airrflow</td>\n",
       "      <td>multiqc</td>\n",
       "      <td>./git_repos\\airrflow\\nextflow.config</td>\n",
       "      <td>107</td>\n",
       "      <td>skip_multiqc               = false</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>airrflow</td>\n",
       "      <td>multiqc</td>\n",
       "      <td>./git_repos\\airrflow\\nextflow.config</td>\n",
       "      <td>108</td>\n",
       "      <td>multiqc_config             = null</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>airrflow</td>\n",
       "      <td>multiqc</td>\n",
       "      <td>./git_repos\\airrflow\\nextflow.config</td>\n",
       "      <td>109</td>\n",
       "      <td>multiqc_title              = null</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>airrflow</td>\n",
       "      <td>multiqc</td>\n",
       "      <td>./git_repos\\airrflow\\nextflow.config</td>\n",
       "      <td>110</td>\n",
       "      <td>multiqc_logo               = null</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>airrflow</td>\n",
       "      <td>multiqc</td>\n",
       "      <td>./git_repos\\airrflow\\nextflow.config</td>\n",
       "      <td>111</td>\n",
       "      <td>max_multiqc_email_size     = '25.MB'</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8299</th>\n",
       "      <td>viralrecon</td>\n",
       "      <td>multiqc</td>\n",
       "      <td>./git_repos\\viralrecon\\conf\\modules_nanopore.c...</td>\n",
       "      <td>345</td>\n",
       "      <td>path: { \"${params.outdir}/multiqc/${params.art...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8300</th>\n",
       "      <td>viralrecon</td>\n",
       "      <td>kraken2</td>\n",
       "      <td>./git_repos\\viralrecon\\conf\\test.config</td>\n",
       "      <td>31</td>\n",
       "      <td>kraken2_db = 'https://raw.githubusercontent.co...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8301</th>\n",
       "      <td>viralrecon</td>\n",
       "      <td>bcftools</td>\n",
       "      <td>./git_repos\\viralrecon\\conf\\test_full_sispa.co...</td>\n",
       "      <td>26</td>\n",
       "      <td>variant_caller = 'bcftools'</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8302</th>\n",
       "      <td>viralrecon</td>\n",
       "      <td>kraken2</td>\n",
       "      <td>./git_repos\\viralrecon\\conf\\test_sispa.config</td>\n",
       "      <td>29</td>\n",
       "      <td>kraken2_db = 'https://raw.githubusercontent.co...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8303</th>\n",
       "      <td>viralrecon</td>\n",
       "      <td>bcftools</td>\n",
       "      <td>./git_repos\\viralrecon\\conf\\test_sispa.config</td>\n",
       "      <td>32</td>\n",
       "      <td>variant_caller = 'bcftools'</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>8304 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        pipeline  operator                                          file_path  \\\n",
       "0       airrflow   multiqc               ./git_repos\\airrflow\\nextflow.config   \n",
       "1       airrflow   multiqc               ./git_repos\\airrflow\\nextflow.config   \n",
       "2       airrflow   multiqc               ./git_repos\\airrflow\\nextflow.config   \n",
       "3       airrflow   multiqc               ./git_repos\\airrflow\\nextflow.config   \n",
       "4       airrflow   multiqc               ./git_repos\\airrflow\\nextflow.config   \n",
       "...          ...       ...                                                ...   \n",
       "8299  viralrecon   multiqc  ./git_repos\\viralrecon\\conf\\modules_nanopore.c...   \n",
       "8300  viralrecon   kraken2            ./git_repos\\viralrecon\\conf\\test.config   \n",
       "8301  viralrecon  bcftools  ./git_repos\\viralrecon\\conf\\test_full_sispa.co...   \n",
       "8302  viralrecon   kraken2      ./git_repos\\viralrecon\\conf\\test_sispa.config   \n",
       "8303  viralrecon  bcftools      ./git_repos\\viralrecon\\conf\\test_sispa.config   \n",
       "\n",
       "      line_number                                       line_content  \n",
       "0             107                 skip_multiqc               = false  \n",
       "1             108                  multiqc_config             = null  \n",
       "2             109                  multiqc_title              = null  \n",
       "3             110                  multiqc_logo               = null  \n",
       "4             111               max_multiqc_email_size     = '25.MB'  \n",
       "...           ...                                                ...  \n",
       "8299          345  path: { \"${params.outdir}/multiqc/${params.art...  \n",
       "8300           31  kraken2_db = 'https://raw.githubusercontent.co...  \n",
       "8301           26                        variant_caller = 'bcftools'  \n",
       "8302           29  kraken2_db = 'https://raw.githubusercontent.co...  \n",
       "8303           32                        variant_caller = 'bcftools'  \n",
       "\n",
       "[8304 rows x 5 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_bioinformatics"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "0412edd9",
   "metadata": {},
   "outputs": [],
   "source": [
    "bio_df = df_bioinformatics.groupby('operator').agg({'pipeline': 'nunique', 'file_path': 'nunique'}).reset_index().sort_values(by='pipeline', ascending=False)\n",
    "#grouped_df = merged_df.groupby('Assigned Group').agg({'config_file': 'nunique', 'operator_in_file': 'nunique'}).reset_index()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "c2a6965d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>operator</th>\n",
       "      <th>pipeline</th>\n",
       "      <th>file_path</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>multiqc</td>\n",
       "      <td>83</td>\n",
       "      <td>140</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>bowtie2</td>\n",
       "      <td>63</td>\n",
       "      <td>84</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>bwa</td>\n",
       "      <td>63</td>\n",
       "      <td>106</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>bismark</td>\n",
       "      <td>57</td>\n",
       "      <td>59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>fastqc</td>\n",
       "      <td>40</td>\n",
       "      <td>62</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   operator  pipeline  file_path\n",
       "23  multiqc        83        140\n",
       "5   bowtie2        63         84\n",
       "6       bwa        63        106\n",
       "4   bismark        57         59\n",
       "11   fastqc        40         62"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bio_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "61eb7301",
   "metadata": {},
   "source": [
    "# Operator domain data analysis\n",
    "\n",
    "- we get all the lines with occuring operators, filter by length of operator due to noisy data!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "0dcac7cc",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "\n",
    "df = pd.read_csv('./results/all_operators_in_config.csv')\n",
    "df_operator_clustered = pd.read_csv('./data/operator_dataset_clustered.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "fea6ae51",
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv('./results/all_operators_in_config.csv')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c3d3af8a",
   "metadata": {},
   "source": [
    "Here we filter by the length of the operator, due to noise in data results similar approach to GitLab Snakemake (https://gitlab.informatik.hu-berlin.de/pohlseba/github_scraping/-/blob/main/analysis/operators/operators.py, line 222)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "62b078a9",
   "metadata": {},
   "outputs": [],
   "source": [
    "df = df[df['operator'].str.len() > 3]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e1c476b1",
   "metadata": {},
   "source": [
    "By applying that, we can make sure that every operator occures just once in a pipline and we dont count multiple occurences"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "1a3d0c0d",
   "metadata": {},
   "outputs": [],
   "source": [
    "# we get each operator occurence once per file from 76137 to 12132  entries\n",
    "\n",
    "df = df.groupby(['pipeline', 'operator', 'file_path']).size().reset_index(name='operator_in_file')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "07988345",
   "metadata": {},
   "outputs": [],
   "source": [
    "df = df.rename(columns={'operator': 'Operator', 'file_path': 'config_file'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "5f799854",
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_df = pd.merge(df, df_operator_clustered, on='Operator')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "e293d13d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1022"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# 1022 configuration files\n",
    "merged_df['config_file'].nunique()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2a9782a0",
   "metadata": {},
   "source": [
    "Group by Assigned Group"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "2348ee42",
   "metadata": {},
   "outputs": [],
   "source": [
    "# n = number of operators to be shown\n",
    "def most_common_operators(x, n=4):\n",
    "    return list(x.value_counts().head(n).index)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "9bc848fc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Assigned Group</th>\n",
       "      <th>pipeline</th>\n",
       "      <th>config_file</th>\n",
       "      <th>operator_in_file</th>\n",
       "      <th>Operator</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>87</td>\n",
       "      <td>406</td>\n",
       "      <td>680</td>\n",
       "      <td>[fasta, fastq, fastqc, fastq.gz]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>94</td>\n",
       "      <td>446</td>\n",
       "      <td>580</td>\n",
       "      <td>[docker, else, centrifuge, classifier]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>94</td>\n",
       "      <td>838</td>\n",
       "      <td>1113</td>\n",
       "      <td>[params, print, echo, paired]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>13</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>[zcat, coverm, cpan, sourmash]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>94</td>\n",
       "      <td>715</td>\n",
       "      <td>2060</td>\n",
       "      <td>[script, singularity, report, conda]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>5</td>\n",
       "      <td>7</td>\n",
       "      <td>11</td>\n",
       "      <td>20</td>\n",
       "      <td>[gzip, bgzip, gunzip, unzip]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>6</td>\n",
       "      <td>94</td>\n",
       "      <td>1013</td>\n",
       "      <td>7676</td>\n",
       "      <td>[file, test, time, input]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Assigned Group  pipeline  config_file  operator_in_file  \\\n",
       "0               0        87          406               680   \n",
       "1               1        94          446               580   \n",
       "2               2        94          838              1113   \n",
       "3               3        13           22                22   \n",
       "4               4        94          715              2060   \n",
       "5               5         7           11                20   \n",
       "6               6        94         1013              7676   \n",
       "\n",
       "                                 Operator  \n",
       "0        [fasta, fastq, fastqc, fastq.gz]  \n",
       "1  [docker, else, centrifuge, classifier]  \n",
       "2           [params, print, echo, paired]  \n",
       "3          [zcat, coverm, cpan, sourmash]  \n",
       "4    [script, singularity, report, conda]  \n",
       "5            [gzip, bgzip, gunzip, unzip]  \n",
       "6               [file, test, time, input]  "
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "agg_bio = merged_df.groupby('Assigned Group').agg({'pipeline': 'nunique', 'config_file': 'nunique', 'operator_in_file': 'count', 'Operator': lambda x: most_common_operators(x)}).reset_index()\n",
    "agg_bio"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "ca09beb7",
   "metadata": {},
   "outputs": [
    {
     "ename": "ValueError",
     "evalue": "No axis named config_file for object type DataFrame",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "\u001b[1;32mC:\\ProgramData\\anaconda3\\Lib\\site-packages\\pandas\\core\\generic.py\u001b[0m in \u001b[0;36m?\u001b[1;34m(cls, axis)\u001b[0m\n\u001b[0;32m    513\u001b[0m             \u001b[1;32mreturn\u001b[0m \u001b[0mcls\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_AXIS_TO_AXIS_NUMBER\u001b[0m\u001b[1;33m[\u001b[0m\u001b[0maxis\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    514\u001b[0m         \u001b[1;32mexcept\u001b[0m \u001b[0mKeyError\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 515\u001b[1;33m             \u001b[1;32mraise\u001b[0m \u001b[0mValueError\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34mf\"No axis named {axis} for object type {cls.__name__}\"\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[1;31mKeyError\u001b[0m: 'config_file'",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[1;31mValueError\u001b[0m                                Traceback (most recent call last)",
      "\u001b[1;32m~\\AppData\\Local\\Temp\\ipykernel_16932\\3853819567.py\u001b[0m in \u001b[0;36m?\u001b[1;34m()\u001b[0m\n\u001b[1;32m----> 1\u001b[1;33m \u001b[0mplot\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mmerged_df\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mgroupby\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m'Assigned Group'\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m'config_file'\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0magg\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m{\u001b[0m\u001b[1;34m'operator_in_file'\u001b[0m\u001b[1;33m:\u001b[0m \u001b[1;34m'nunique'\u001b[0m\u001b[1;33m}\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mreset_index\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m      2\u001b[0m \u001b[0mplot\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32mC:\\ProgramData\\anaconda3\\Lib\\site-packages\\pandas\\core\\frame.py\u001b[0m in \u001b[0;36m?\u001b[1;34m(self, by, axis, level, as_index, sort, group_keys, observed, dropna)\u001b[0m\n\u001b[0;32m   8246\u001b[0m         \u001b[1;32mfrom\u001b[0m \u001b[0mpandas\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mcore\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mgroupby\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mgeneric\u001b[0m \u001b[1;32mimport\u001b[0m \u001b[0mDataFrameGroupBy\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   8247\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   8248\u001b[0m         \u001b[1;32mif\u001b[0m \u001b[0mlevel\u001b[0m \u001b[1;32mis\u001b[0m \u001b[1;32mNone\u001b[0m \u001b[1;32mand\u001b[0m \u001b[0mby\u001b[0m \u001b[1;32mis\u001b[0m \u001b[1;32mNone\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   8249\u001b[0m             \u001b[1;32mraise\u001b[0m \u001b[0mTypeError\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m\"You have to supply one of 'by' and 'level'\"\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m-> 8250\u001b[1;33m         \u001b[0maxis\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_get_axis_number\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0maxis\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m   8251\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   8252\u001b[0m         return DataFrameGroupBy(\n\u001b[0;32m   8253\u001b[0m             \u001b[0mobj\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32mC:\\ProgramData\\anaconda3\\Lib\\site-packages\\pandas\\core\\generic.py\u001b[0m in \u001b[0;36m?\u001b[1;34m(cls, axis)\u001b[0m\n\u001b[0;32m    511\u001b[0m     \u001b[1;32mdef\u001b[0m \u001b[0m_get_axis_number\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mcls\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0maxis\u001b[0m\u001b[1;33m:\u001b[0m \u001b[0mAxis\u001b[0m\u001b[1;33m)\u001b[0m \u001b[1;33m->\u001b[0m \u001b[0mAxisInt\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    512\u001b[0m         \u001b[1;32mtry\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    513\u001b[0m             \u001b[1;32mreturn\u001b[0m \u001b[0mcls\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_AXIS_TO_AXIS_NUMBER\u001b[0m\u001b[1;33m[\u001b[0m\u001b[0maxis\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    514\u001b[0m         \u001b[1;32mexcept\u001b[0m \u001b[0mKeyError\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 515\u001b[1;33m             \u001b[1;32mraise\u001b[0m \u001b[0mValueError\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34mf\"No axis named {axis} for object type {cls.__name__}\"\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[1;31mValueError\u001b[0m: No axis named config_file for object type DataFrame"
     ]
    }
   ],
   "source": [
    "plot = merged_df.groupby('Assigned Group', 'config_file').agg({'operator_in_file': 'nunique'}).reset_index()\n",
    "plot"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "60dcc419",
   "metadata": {},
   "source": [
    "- Assigned Group 0:    bioinformatics tool\n",
    "- Assigned Group 1:    control flow\n",
    "- Assigned Group 2:    file processing\n",
    "- Assigned Group 3:    file system\n",
    "- Assigned Group 4:    code execution\n",
    "- Assigned Group 5:    data handling\n",
    "- Assigned Group 6:    other"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4a2cedff",
   "metadata": {},
   "source": [
    "# Analysis of operator domain data in processes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "id": "96525dd9",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "\n",
    "df = pd.read_csv('./results/all_operators_in_processes.csv')\n",
    "\n",
    "df_operator_clustered = pd.read_csv('./data/operator_dataset_clustered.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "id": "80740074",
   "metadata": {},
   "outputs": [],
   "source": [
    "df = df[df['operator'].str.len() > 3]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "id": "1043b2b4",
   "metadata": {},
   "outputs": [],
   "source": [
    "df = df.rename(columns={'operator': 'Operator'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "id": "c8062f7a",
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_df = pd.merge(df, df_operator_clustered, on='Operator')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "id": "1dfe4858",
   "metadata": {},
   "outputs": [],
   "source": [
    "grouped_df = merged_df.groupby('Assigned Group').agg({'pipeline': 'nunique', 'process_name': 'nunique', 'Operator': 'count', 'file_path': 'nunique'}).reset_index()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d9ac098b",
   "metadata": {},
   "source": [
    "Add ratio = # occuring processes /# total rules"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "id": "ba17677a",
   "metadata": {},
   "outputs": [],
   "source": [
    "sum_processes = merged_df['process_name'].nunique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "id": "adafc65b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1982"
      ]
     },
     "execution_count": 77,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sum_processes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "id": "5d265a65",
   "metadata": {},
   "outputs": [],
   "source": [
    "grouped_df['ratio'] = grouped_df['process_name']/sum_processes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "id": "d4c0e864",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Assigned Group</th>\n",
       "      <th>pipeline</th>\n",
       "      <th>process_name</th>\n",
       "      <th>Operator</th>\n",
       "      <th>file_path</th>\n",
       "      <th>ratio</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>88</td>\n",
       "      <td>863</td>\n",
       "      <td>9315</td>\n",
       "      <td>1269</td>\n",
       "      <td>0.435419</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>93</td>\n",
       "      <td>1588</td>\n",
       "      <td>4045</td>\n",
       "      <td>2463</td>\n",
       "      <td>0.801211</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>94</td>\n",
       "      <td>1622</td>\n",
       "      <td>8498</td>\n",
       "      <td>1943</td>\n",
       "      <td>0.818365</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>36</td>\n",
       "      <td>35</td>\n",
       "      <td>120</td>\n",
       "      <td>42</td>\n",
       "      <td>0.017659</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>94</td>\n",
       "      <td>1979</td>\n",
       "      <td>18899</td>\n",
       "      <td>2573</td>\n",
       "      <td>0.998486</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>5</td>\n",
       "      <td>70</td>\n",
       "      <td>187</td>\n",
       "      <td>1120</td>\n",
       "      <td>283</td>\n",
       "      <td>0.094349</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>6</td>\n",
       "      <td>94</td>\n",
       "      <td>1982</td>\n",
       "      <td>33788</td>\n",
       "      <td>2576</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Assigned Group  pipeline  process_name  Operator  file_path     ratio\n",
       "0               0        88           863      9315       1269  0.435419\n",
       "1               1        93          1588      4045       2463  0.801211\n",
       "2               2        94          1622      8498       1943  0.818365\n",
       "3               3        36            35       120         42  0.017659\n",
       "4               4        94          1979     18899       2573  0.998486\n",
       "5               5        70           187      1120        283  0.094349\n",
       "6               6        94          1982     33788       2576  1.000000"
      ]
     },
     "execution_count": 79,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "grouped_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "80692fd1",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
