{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "2a5d0328",
   "metadata": {},
   "source": [
    "Check for all operators in script block of a process:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 338,
   "id": "8f43fb25",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import re\n",
    "import csv\n",
    "import pandas as pd\n",
    "\n",
    "def find_operators_in_script(directory, pattern_list, output_file_name):\n",
    "    result_list = []\n",
    "\n",
    "    for pipeline in os.listdir(directory):\n",
    "        subdirectory_path = os.path.join(directory, pipeline)\n",
    "        folder_name = os.path.basename(subdirectory_path)\n",
    "        #print(pipeline)\n",
    "\n",
    "        for root, dirs, files in os.walk(subdirectory_path):\n",
    "            for filename in files:\n",
    "                file_path = os.path.join(root, filename)\n",
    "                \n",
    "                if filename.endswith('.nf') or filename.endswith('.config'):\n",
    "                    with open(file_path, 'r', encoding='utf-8', errors='ignore') as file:\n",
    "                        lines = file.readlines()\n",
    "                    \n",
    "                    inside_block = False\n",
    "                    brace_stack = []\n",
    "                    process_flag = False\n",
    "                    process_name = None\n",
    "\n",
    "                    process_pattern = r'process\\s+(\\w+)\\s*{'\n",
    "                    \n",
    "                    for line_num, line in enumerate(lines):\n",
    "                        stripped_line = line.strip()\n",
    "                        \n",
    "                        if line.strip().startswith(('#', '//', '*')):\n",
    "                            continue\n",
    "\n",
    "                        if re.search(process_pattern, stripped_line):\n",
    "                            process_flag = True\n",
    "                            process_match = re.search(process_pattern, stripped_line)\n",
    "                            if process_match:\n",
    "                                process_name = process_match.group(1)\n",
    "                            brace_stack.append('{')\n",
    "\n",
    "                        if '{' in stripped_line:\n",
    "                            brace_stack.append('{')\n",
    "\n",
    "                        if '}' in stripped_line:\n",
    "                            if brace_stack:\n",
    "                                brace_stack.pop()\n",
    "                                if not brace_stack and process_flag:\n",
    "                                    process_flag = False\n",
    "\n",
    "                        if '\"\"\"' in stripped_line:\n",
    "                            inside_block = not inside_block\n",
    "\n",
    "                        if process_flag and inside_block:\n",
    "                            operator_pattern = \"(\" + \"|\".join(map(re.escape, pattern_list)) + \")\"\n",
    "                            if re.search(operator_pattern, stripped_line):\n",
    "                                for pattern in pattern_list:\n",
    "                                    if re.search(re.escape(pattern), stripped_line):\n",
    "                                        result_list.append([folder_name, process_name, pattern, file_path, line_num+1, stripped_line])\n",
    "\n",
    "    write_to_csv(result_list, output_file_name)\n",
    "\n",
    "def write_to_csv(result_list, output_file_name):\n",
    "    with open(f'./results/{output_file_name}.csv', 'w', newline='', encoding='utf-8') as csvfile:\n",
    "        writer = csv.writer(csvfile)\n",
    "        writer.writerow(['pipeline', 'process_name', 'operator', 'file_path', 'line_number', 'line_content'])\n",
    "        writer.writerows(result_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 339,
   "id": "f03c75d1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\"directory_path = './git_repos'\\n\\ndf_operators = pd.read_excel('./data/all_operator_dataset.xlsx')\\noperator_list = [str(operator) for operator in df_operators['operator']]\\n\\noutput_file_name = 'all_operators_in_scripts_new'\\n\\nfind_operators_in_script(directory_path, operator_list, output_file_name)\""
      ]
     },
     "execution_count": 339,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "'''directory_path = './git_repos'\n",
    "\n",
    "df_operators = pd.read_excel('./data/all_operator_dataset.xlsx')\n",
    "operator_list = [str(operator) for operator in df_operators['operator']]\n",
    "\n",
    "output_file_name = 'all_operators_in_scripts_new'\n",
    "\n",
    "find_operators_in_script(directory_path, operator_list, output_file_name)'''"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 340,
   "id": "34a5eda1",
   "metadata": {},
   "outputs": [],
   "source": [
    "'''directory_path = './git_repos'\n",
    "\n",
    "df_operators = pd.read_excel('./data/bioinformatics_tools.xlsx')\n",
    "operator_list = [str(operator) for operator in df_operators['operator']]\n",
    "\n",
    "output_file_name = 'bioinformatic_tools_in_scripts'\n",
    "\n",
    "find_operators_in_script(directory_path, operator_list, output_file_name)'''"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "e94fbb27",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pipeline</th>\n",
       "      <th>process_name</th>\n",
       "      <th>operator</th>\n",
       "      <th>file_path</th>\n",
       "      <th>line_number</th>\n",
       "      <th>line_content</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>airrflow</td>\n",
       "      <td>FASTQC_POSTASSEMBLY</td>\n",
       "      <td>fastqc</td>\n",
       "      <td>./git_repos\\airrflow\\modules\\local\\fastqc_postassembly.nf</td>\n",
       "      <td>24</td>\n",
       "      <td>fastqc $args --threads $task.cpus ${prefix}_ASSEMBLED.fastq</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>airrflow</td>\n",
       "      <td>FASTQC_POSTASSEMBLY</td>\n",
       "      <td>fastqc</td>\n",
       "      <td>./git_repos\\airrflow\\modules\\local\\fastqc_postassembly.nf</td>\n",
       "      <td>28</td>\n",
       "      <td>fastqc: \\$( fastqc --version | sed -e \"s/FastQC v//g\" )</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>airrflow</td>\n",
       "      <td>FASTP</td>\n",
       "      <td>fastp</td>\n",
       "      <td>./git_repos\\airrflow\\modules\\nf-core\\fastp\\main.nf</td>\n",
       "      <td>39</td>\n",
       "      <td>fastp \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>airrflow</td>\n",
       "      <td>FASTP</td>\n",
       "      <td>fastp</td>\n",
       "      <td>./git_repos\\airrflow\\modules\\nf-core\\fastp\\main.nf</td>\n",
       "      <td>43</td>\n",
       "      <td>--json ${prefix}.fastp.json \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>airrflow</td>\n",
       "      <td>FASTP</td>\n",
       "      <td>fastp</td>\n",
       "      <td>./git_repos\\airrflow\\modules\\nf-core\\fastp\\main.nf</td>\n",
       "      <td>44</td>\n",
       "      <td>--html ${prefix}.fastp.html \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>airrflow</td>\n",
       "      <td>FASTP</td>\n",
       "      <td>fastp</td>\n",
       "      <td>./git_repos\\airrflow\\modules\\nf-core\\fastp\\main.nf</td>\n",
       "      <td>48</td>\n",
       "      <td>2&gt; ${prefix}.fastp.log \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>airrflow</td>\n",
       "      <td>FASTP</td>\n",
       "      <td>fastp</td>\n",
       "      <td>./git_repos\\airrflow\\modules\\nf-core\\fastp\\main.nf</td>\n",
       "      <td>49</td>\n",
       "      <td>| gzip -c &gt; ${prefix}.fastp.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>airrflow</td>\n",
       "      <td>FASTP</td>\n",
       "      <td>fastp</td>\n",
       "      <td>./git_repos\\airrflow\\modules\\nf-core\\fastp\\main.nf</td>\n",
       "      <td>53</td>\n",
       "      <td>fastp: \\$(fastp --version 2&gt;&amp;1 | sed -e \"s/fastp //g\")</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>airrflow</td>\n",
       "      <td>FASTP</td>\n",
       "      <td>fastp</td>\n",
       "      <td>./git_repos\\airrflow\\modules\\nf-core\\fastp\\main.nf</td>\n",
       "      <td>60</td>\n",
       "      <td>fastp \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>airrflow</td>\n",
       "      <td>FASTP</td>\n",
       "      <td>fastp</td>\n",
       "      <td>./git_repos\\airrflow\\modules\\nf-core\\fastp\\main.nf</td>\n",
       "      <td>62</td>\n",
       "      <td>--out1  ${prefix}.fastp.fastq.gz \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>airrflow</td>\n",
       "      <td>FASTP</td>\n",
       "      <td>fastp</td>\n",
       "      <td>./git_repos\\airrflow\\modules\\nf-core\\fastp\\main.nf</td>\n",
       "      <td>64</td>\n",
       "      <td>--json ${prefix}.fastp.json \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>airrflow</td>\n",
       "      <td>FASTP</td>\n",
       "      <td>fastp</td>\n",
       "      <td>./git_repos\\airrflow\\modules\\nf-core\\fastp\\main.nf</td>\n",
       "      <td>65</td>\n",
       "      <td>--html ${prefix}.fastp.html \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>airrflow</td>\n",
       "      <td>FASTP</td>\n",
       "      <td>fastp</td>\n",
       "      <td>./git_repos\\airrflow\\modules\\nf-core\\fastp\\main.nf</td>\n",
       "      <td>69</td>\n",
       "      <td>2&gt; ${prefix}.fastp.log</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>airrflow</td>\n",
       "      <td>FASTP</td>\n",
       "      <td>fastp</td>\n",
       "      <td>./git_repos\\airrflow\\modules\\nf-core\\fastp\\main.nf</td>\n",
       "      <td>73</td>\n",
       "      <td>fastp: \\$(fastp --version 2&gt;&amp;1 | sed -e \"s/fastp //g\")</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>airrflow</td>\n",
       "      <td>FASTP</td>\n",
       "      <td>fastp</td>\n",
       "      <td>./git_repos\\airrflow\\modules\\nf-core\\fastp\\main.nf</td>\n",
       "      <td>81</td>\n",
       "      <td>fastp \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>airrflow</td>\n",
       "      <td>FASTP</td>\n",
       "      <td>fastp</td>\n",
       "      <td>./git_repos\\airrflow\\modules\\nf-core\\fastp\\main.nf</td>\n",
       "      <td>84</td>\n",
       "      <td>--out1 ${prefix}_1.fastp.fastq.gz \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>airrflow</td>\n",
       "      <td>FASTP</td>\n",
       "      <td>fastp</td>\n",
       "      <td>./git_repos\\airrflow\\modules\\nf-core\\fastp\\main.nf</td>\n",
       "      <td>85</td>\n",
       "      <td>--out2 ${prefix}_2.fastp.fastq.gz \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>airrflow</td>\n",
       "      <td>FASTP</td>\n",
       "      <td>fastp</td>\n",
       "      <td>./git_repos\\airrflow\\modules\\nf-core\\fastp\\main.nf</td>\n",
       "      <td>86</td>\n",
       "      <td>--json ${prefix}.fastp.json \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>airrflow</td>\n",
       "      <td>FASTP</td>\n",
       "      <td>fastp</td>\n",
       "      <td>./git_repos\\airrflow\\modules\\nf-core\\fastp\\main.nf</td>\n",
       "      <td>87</td>\n",
       "      <td>--html ${prefix}.fastp.html \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>airrflow</td>\n",
       "      <td>FASTP</td>\n",
       "      <td>fastp</td>\n",
       "      <td>./git_repos\\airrflow\\modules\\nf-core\\fastp\\main.nf</td>\n",
       "      <td>94</td>\n",
       "      <td>2&gt; ${prefix}.fastp.log</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>airrflow</td>\n",
       "      <td>FASTP</td>\n",
       "      <td>fastp</td>\n",
       "      <td>./git_repos\\airrflow\\modules\\nf-core\\fastp\\main.nf</td>\n",
       "      <td>98</td>\n",
       "      <td>fastp: \\$(fastp --version 2&gt;&amp;1 | sed -e \"s/fastp //g\")</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>airrflow</td>\n",
       "      <td>FASTQC</td>\n",
       "      <td>fastqc</td>\n",
       "      <td>./git_repos\\airrflow\\modules\\nf-core\\fastqc\\main.nf</td>\n",
       "      <td>33</td>\n",
       "      <td>fastqc \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>airrflow</td>\n",
       "      <td>FASTQC</td>\n",
       "      <td>fastqc</td>\n",
       "      <td>./git_repos\\airrflow\\modules\\nf-core\\fastqc\\main.nf</td>\n",
       "      <td>40</td>\n",
       "      <td>fastqc: \\$( fastqc --version | sed -e \"s/FastQC v//g\" )</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>airrflow</td>\n",
       "      <td>FASTQC</td>\n",
       "      <td>fastqc</td>\n",
       "      <td>./git_repos\\airrflow\\modules\\nf-core\\fastqc\\main.nf</td>\n",
       "      <td>52</td>\n",
       "      <td>fastqc: \\$( fastqc --version | sed -e \"s/FastQC v//g\" )</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>airrflow</td>\n",
       "      <td>MULTIQC</td>\n",
       "      <td>multiqc</td>\n",
       "      <td>./git_repos\\airrflow\\modules\\nf-core\\multiqc\\main.nf</td>\n",
       "      <td>29</td>\n",
       "      <td>multiqc \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4223</th>\n",
       "      <td>viralrecon</td>\n",
       "      <td>PICARD_COLLECTMULTIPLEMETRICS</td>\n",
       "      <td>picard</td>\n",
       "      <td>./git_repos\\viralrecon\\modules\\nf-core\\picard\\collectmultiplemetrics\\main.nf</td>\n",
       "      <td>64</td>\n",
       "      <td>picard: \\$(echo \\$(picard CollectMultipleMetrics --version 2&gt;&amp;1) | grep -o 'Version:.*' | cut -f2- -d:)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4224</th>\n",
       "      <td>viralrecon</td>\n",
       "      <td>PICARD_MARKDUPLICATES</td>\n",
       "      <td>picard</td>\n",
       "      <td>./git_repos\\viralrecon\\modules\\nf-core\\picard\\markduplicates\\main.nf</td>\n",
       "      <td>34</td>\n",
       "      <td>picard \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4225</th>\n",
       "      <td>viralrecon</td>\n",
       "      <td>PICARD_MARKDUPLICATES</td>\n",
       "      <td>picard</td>\n",
       "      <td>./git_repos\\viralrecon\\modules\\nf-core\\picard\\markduplicates\\main.nf</td>\n",
       "      <td>45</td>\n",
       "      <td>picard: \\$(echo \\$(picard MarkDuplicates --version 2&gt;&amp;1) | grep -o 'Version:.*' | cut -f2- -d:)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4226</th>\n",
       "      <td>viralrecon</td>\n",
       "      <td>PICARD_MARKDUPLICATES</td>\n",
       "      <td>picard</td>\n",
       "      <td>./git_repos\\viralrecon\\modules\\nf-core\\picard\\markduplicates\\main.nf</td>\n",
       "      <td>58</td>\n",
       "      <td>picard: \\$(echo \\$(picard MarkDuplicates --version 2&gt;&amp;1) | grep -o 'Version:.*' | cut -f2- -d:)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4227</th>\n",
       "      <td>viralrecon</td>\n",
       "      <td>SAMTOOLS_FLAGSTAT</td>\n",
       "      <td>samtools</td>\n",
       "      <td>./git_repos\\viralrecon\\modules\\nf-core\\samtools\\flagstat\\main.nf</td>\n",
       "      <td>24</td>\n",
       "      <td>samtools \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4228</th>\n",
       "      <td>viralrecon</td>\n",
       "      <td>SAMTOOLS_FLAGSTAT</td>\n",
       "      <td>samtools</td>\n",
       "      <td>./git_repos\\viralrecon\\modules\\nf-core\\samtools\\flagstat\\main.nf</td>\n",
       "      <td>32</td>\n",
       "      <td>samtools: \\$(echo \\$(samtools --version 2&gt;&amp;1) | sed 's/^.*samtools //; s/Using.*\\$//')</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4229</th>\n",
       "      <td>viralrecon</td>\n",
       "      <td>SAMTOOLS_IDXSTATS</td>\n",
       "      <td>samtools</td>\n",
       "      <td>./git_repos\\viralrecon\\modules\\nf-core\\samtools\\idxstats\\main.nf</td>\n",
       "      <td>25</td>\n",
       "      <td>samtools \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4230</th>\n",
       "      <td>viralrecon</td>\n",
       "      <td>SAMTOOLS_IDXSTATS</td>\n",
       "      <td>samtools</td>\n",
       "      <td>./git_repos\\viralrecon\\modules\\nf-core\\samtools\\idxstats\\main.nf</td>\n",
       "      <td>33</td>\n",
       "      <td>samtools: \\$(echo \\$(samtools --version 2&gt;&amp;1) | sed 's/^.*samtools //; s/Using.*\\$//')</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4231</th>\n",
       "      <td>viralrecon</td>\n",
       "      <td>SAMTOOLS_INDEX</td>\n",
       "      <td>samtools</td>\n",
       "      <td>./git_repos\\viralrecon\\modules\\nf-core\\samtools\\index\\main.nf</td>\n",
       "      <td>25</td>\n",
       "      <td>samtools \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4232</th>\n",
       "      <td>viralrecon</td>\n",
       "      <td>SAMTOOLS_INDEX</td>\n",
       "      <td>samtools</td>\n",
       "      <td>./git_repos\\viralrecon\\modules\\nf-core\\samtools\\index\\main.nf</td>\n",
       "      <td>33</td>\n",
       "      <td>samtools: \\$(echo \\$(samtools --version 2&gt;&amp;1) | sed 's/^.*samtools //; s/Using.*\\$//')</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4233</th>\n",
       "      <td>viralrecon</td>\n",
       "      <td>SAMTOOLS_INDEX</td>\n",
       "      <td>samtools</td>\n",
       "      <td>./git_repos\\viralrecon\\modules\\nf-core\\samtools\\index\\main.nf</td>\n",
       "      <td>45</td>\n",
       "      <td>samtools: \\$(echo \\$(samtools --version 2&gt;&amp;1) | sed 's/^.*samtools //; s/Using.*\\$//')</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4234</th>\n",
       "      <td>viralrecon</td>\n",
       "      <td>SAMTOOLS_SORT</td>\n",
       "      <td>samtools</td>\n",
       "      <td>./git_repos\\viralrecon\\modules\\nf-core\\samtools\\sort\\main.nf</td>\n",
       "      <td>26</td>\n",
       "      <td>samtools sort $args -@ $task.cpus -o ${prefix}.bam -T $prefix $bam</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4235</th>\n",
       "      <td>viralrecon</td>\n",
       "      <td>SAMTOOLS_SORT</td>\n",
       "      <td>samtools</td>\n",
       "      <td>./git_repos\\viralrecon\\modules\\nf-core\\samtools\\sort\\main.nf</td>\n",
       "      <td>29</td>\n",
       "      <td>samtools: \\$(echo \\$(samtools --version 2&gt;&amp;1) | sed 's/^.*samtools //; s/Using.*\\$//')</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4236</th>\n",
       "      <td>viralrecon</td>\n",
       "      <td>SAMTOOLS_SORT</td>\n",
       "      <td>samtools</td>\n",
       "      <td>./git_repos\\viralrecon\\modules\\nf-core\\samtools\\sort\\main.nf</td>\n",
       "      <td>40</td>\n",
       "      <td>samtools: \\$(echo \\$(samtools --version 2&gt;&amp;1) | sed 's/^.*samtools //; s/Using.*\\$//')</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4237</th>\n",
       "      <td>viralrecon</td>\n",
       "      <td>SAMTOOLS_STATS</td>\n",
       "      <td>samtools</td>\n",
       "      <td>./git_repos\\viralrecon\\modules\\nf-core\\samtools\\stats\\main.nf</td>\n",
       "      <td>26</td>\n",
       "      <td>samtools \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4238</th>\n",
       "      <td>viralrecon</td>\n",
       "      <td>SAMTOOLS_STATS</td>\n",
       "      <td>samtools</td>\n",
       "      <td>./git_repos\\viralrecon\\modules\\nf-core\\samtools\\stats\\main.nf</td>\n",
       "      <td>35</td>\n",
       "      <td>samtools: \\$(echo \\$(samtools --version 2&gt;&amp;1) | sed 's/^.*samtools //; s/Using.*\\$//')</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4239</th>\n",
       "      <td>viralrecon</td>\n",
       "      <td>SAMTOOLS_STATS</td>\n",
       "      <td>samtools</td>\n",
       "      <td>./git_repos\\viralrecon\\modules\\nf-core\\samtools\\stats\\main.nf</td>\n",
       "      <td>46</td>\n",
       "      <td>samtools: \\$(echo \\$(samtools --version 2&gt;&amp;1) | sed 's/^.*samtools //; s/Using.*\\$//')</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4240</th>\n",
       "      <td>viralrecon</td>\n",
       "      <td>SAMTOOLS_VIEW</td>\n",
       "      <td>samtools</td>\n",
       "      <td>./git_repos\\viralrecon\\modules\\nf-core\\samtools\\view\\main.nf</td>\n",
       "      <td>39</td>\n",
       "      <td>samtools \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4241</th>\n",
       "      <td>viralrecon</td>\n",
       "      <td>SAMTOOLS_VIEW</td>\n",
       "      <td>samtools</td>\n",
       "      <td>./git_repos\\viralrecon\\modules\\nf-core\\samtools\\view\\main.nf</td>\n",
       "      <td>51</td>\n",
       "      <td>samtools: \\$(echo \\$(samtools --version 2&gt;&amp;1) | sed 's/^.*samtools //; s/Using.*\\$//')</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4242</th>\n",
       "      <td>viralrecon</td>\n",
       "      <td>SAMTOOLS_VIEW</td>\n",
       "      <td>samtools</td>\n",
       "      <td>./git_repos\\viralrecon\\modules\\nf-core\\samtools\\view\\main.nf</td>\n",
       "      <td>63</td>\n",
       "      <td>samtools: \\$(echo \\$(samtools --version 2&gt;&amp;1) | sed 's/^.*samtools //; s/Using.*\\$//')</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4243</th>\n",
       "      <td>viralrecon</td>\n",
       "      <td>TABIX_BGZIP</td>\n",
       "      <td>tabix</td>\n",
       "      <td>./git_repos\\viralrecon\\modules\\nf-core\\tabix\\bgzip\\main.nf</td>\n",
       "      <td>37</td>\n",
       "      <td>tabix: \\$(echo \\$(tabix -h 2&gt;&amp;1) | sed 's/^.*Version: //; s/ .*\\$//')</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4244</th>\n",
       "      <td>viralrecon</td>\n",
       "      <td>TABIX_BGZIP</td>\n",
       "      <td>tabix</td>\n",
       "      <td>./git_repos\\viralrecon\\modules\\nf-core\\tabix\\bgzip\\main.nf</td>\n",
       "      <td>51</td>\n",
       "      <td>tabix: \\$(echo \\$(tabix -h 2&gt;&amp;1) | sed 's/^.*Version: //; s/ .*\\$//')</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4245</th>\n",
       "      <td>viralrecon</td>\n",
       "      <td>TABIX_TABIX</td>\n",
       "      <td>tabix</td>\n",
       "      <td>./git_repos\\viralrecon\\modules\\nf-core\\tabix\\tabix\\main.nf</td>\n",
       "      <td>24</td>\n",
       "      <td>tabix $args $tab</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4246</th>\n",
       "      <td>viralrecon</td>\n",
       "      <td>TABIX_TABIX</td>\n",
       "      <td>tabix</td>\n",
       "      <td>./git_repos\\viralrecon\\modules\\nf-core\\tabix\\tabix\\main.nf</td>\n",
       "      <td>28</td>\n",
       "      <td>tabix: \\$(echo \\$(tabix -h 2&gt;&amp;1) | sed 's/^.*Version: //; s/ .*\\$//')</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4247</th>\n",
       "      <td>viralrecon</td>\n",
       "      <td>TABIX_TABIX</td>\n",
       "      <td>tabix</td>\n",
       "      <td>./git_repos\\viralrecon\\modules\\nf-core\\tabix\\tabix\\main.nf</td>\n",
       "      <td>39</td>\n",
       "      <td>tabix: \\$(echo \\$(tabix -h 2&gt;&amp;1) | sed 's/^.*Version: //; s/ .*\\$//')</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4248 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        pipeline                   process_name  operator  \\\n",
       "0       airrflow            FASTQC_POSTASSEMBLY    fastqc   \n",
       "1       airrflow            FASTQC_POSTASSEMBLY    fastqc   \n",
       "2       airrflow                          FASTP     fastp   \n",
       "3       airrflow                          FASTP     fastp   \n",
       "4       airrflow                          FASTP     fastp   \n",
       "5       airrflow                          FASTP     fastp   \n",
       "6       airrflow                          FASTP     fastp   \n",
       "7       airrflow                          FASTP     fastp   \n",
       "8       airrflow                          FASTP     fastp   \n",
       "9       airrflow                          FASTP     fastp   \n",
       "10      airrflow                          FASTP     fastp   \n",
       "11      airrflow                          FASTP     fastp   \n",
       "12      airrflow                          FASTP     fastp   \n",
       "13      airrflow                          FASTP     fastp   \n",
       "14      airrflow                          FASTP     fastp   \n",
       "15      airrflow                          FASTP     fastp   \n",
       "16      airrflow                          FASTP     fastp   \n",
       "17      airrflow                          FASTP     fastp   \n",
       "18      airrflow                          FASTP     fastp   \n",
       "19      airrflow                          FASTP     fastp   \n",
       "20      airrflow                          FASTP     fastp   \n",
       "21      airrflow                         FASTQC    fastqc   \n",
       "22      airrflow                         FASTQC    fastqc   \n",
       "23      airrflow                         FASTQC    fastqc   \n",
       "24      airrflow                        MULTIQC   multiqc   \n",
       "...          ...                            ...       ...   \n",
       "4223  viralrecon  PICARD_COLLECTMULTIPLEMETRICS    picard   \n",
       "4224  viralrecon          PICARD_MARKDUPLICATES    picard   \n",
       "4225  viralrecon          PICARD_MARKDUPLICATES    picard   \n",
       "4226  viralrecon          PICARD_MARKDUPLICATES    picard   \n",
       "4227  viralrecon              SAMTOOLS_FLAGSTAT  samtools   \n",
       "4228  viralrecon              SAMTOOLS_FLAGSTAT  samtools   \n",
       "4229  viralrecon              SAMTOOLS_IDXSTATS  samtools   \n",
       "4230  viralrecon              SAMTOOLS_IDXSTATS  samtools   \n",
       "4231  viralrecon                 SAMTOOLS_INDEX  samtools   \n",
       "4232  viralrecon                 SAMTOOLS_INDEX  samtools   \n",
       "4233  viralrecon                 SAMTOOLS_INDEX  samtools   \n",
       "4234  viralrecon                  SAMTOOLS_SORT  samtools   \n",
       "4235  viralrecon                  SAMTOOLS_SORT  samtools   \n",
       "4236  viralrecon                  SAMTOOLS_SORT  samtools   \n",
       "4237  viralrecon                 SAMTOOLS_STATS  samtools   \n",
       "4238  viralrecon                 SAMTOOLS_STATS  samtools   \n",
       "4239  viralrecon                 SAMTOOLS_STATS  samtools   \n",
       "4240  viralrecon                  SAMTOOLS_VIEW  samtools   \n",
       "4241  viralrecon                  SAMTOOLS_VIEW  samtools   \n",
       "4242  viralrecon                  SAMTOOLS_VIEW  samtools   \n",
       "4243  viralrecon                    TABIX_BGZIP     tabix   \n",
       "4244  viralrecon                    TABIX_BGZIP     tabix   \n",
       "4245  viralrecon                    TABIX_TABIX     tabix   \n",
       "4246  viralrecon                    TABIX_TABIX     tabix   \n",
       "4247  viralrecon                    TABIX_TABIX     tabix   \n",
       "\n",
       "                                                                         file_path  \\\n",
       "0                        ./git_repos\\airrflow\\modules\\local\\fastqc_postassembly.nf   \n",
       "1                        ./git_repos\\airrflow\\modules\\local\\fastqc_postassembly.nf   \n",
       "2                               ./git_repos\\airrflow\\modules\\nf-core\\fastp\\main.nf   \n",
       "3                               ./git_repos\\airrflow\\modules\\nf-core\\fastp\\main.nf   \n",
       "4                               ./git_repos\\airrflow\\modules\\nf-core\\fastp\\main.nf   \n",
       "5                               ./git_repos\\airrflow\\modules\\nf-core\\fastp\\main.nf   \n",
       "6                               ./git_repos\\airrflow\\modules\\nf-core\\fastp\\main.nf   \n",
       "7                               ./git_repos\\airrflow\\modules\\nf-core\\fastp\\main.nf   \n",
       "8                               ./git_repos\\airrflow\\modules\\nf-core\\fastp\\main.nf   \n",
       "9                               ./git_repos\\airrflow\\modules\\nf-core\\fastp\\main.nf   \n",
       "10                              ./git_repos\\airrflow\\modules\\nf-core\\fastp\\main.nf   \n",
       "11                              ./git_repos\\airrflow\\modules\\nf-core\\fastp\\main.nf   \n",
       "12                              ./git_repos\\airrflow\\modules\\nf-core\\fastp\\main.nf   \n",
       "13                              ./git_repos\\airrflow\\modules\\nf-core\\fastp\\main.nf   \n",
       "14                              ./git_repos\\airrflow\\modules\\nf-core\\fastp\\main.nf   \n",
       "15                              ./git_repos\\airrflow\\modules\\nf-core\\fastp\\main.nf   \n",
       "16                              ./git_repos\\airrflow\\modules\\nf-core\\fastp\\main.nf   \n",
       "17                              ./git_repos\\airrflow\\modules\\nf-core\\fastp\\main.nf   \n",
       "18                              ./git_repos\\airrflow\\modules\\nf-core\\fastp\\main.nf   \n",
       "19                              ./git_repos\\airrflow\\modules\\nf-core\\fastp\\main.nf   \n",
       "20                              ./git_repos\\airrflow\\modules\\nf-core\\fastp\\main.nf   \n",
       "21                             ./git_repos\\airrflow\\modules\\nf-core\\fastqc\\main.nf   \n",
       "22                             ./git_repos\\airrflow\\modules\\nf-core\\fastqc\\main.nf   \n",
       "23                             ./git_repos\\airrflow\\modules\\nf-core\\fastqc\\main.nf   \n",
       "24                            ./git_repos\\airrflow\\modules\\nf-core\\multiqc\\main.nf   \n",
       "...                                                                            ...   \n",
       "4223  ./git_repos\\viralrecon\\modules\\nf-core\\picard\\collectmultiplemetrics\\main.nf   \n",
       "4224          ./git_repos\\viralrecon\\modules\\nf-core\\picard\\markduplicates\\main.nf   \n",
       "4225          ./git_repos\\viralrecon\\modules\\nf-core\\picard\\markduplicates\\main.nf   \n",
       "4226          ./git_repos\\viralrecon\\modules\\nf-core\\picard\\markduplicates\\main.nf   \n",
       "4227              ./git_repos\\viralrecon\\modules\\nf-core\\samtools\\flagstat\\main.nf   \n",
       "4228              ./git_repos\\viralrecon\\modules\\nf-core\\samtools\\flagstat\\main.nf   \n",
       "4229              ./git_repos\\viralrecon\\modules\\nf-core\\samtools\\idxstats\\main.nf   \n",
       "4230              ./git_repos\\viralrecon\\modules\\nf-core\\samtools\\idxstats\\main.nf   \n",
       "4231                 ./git_repos\\viralrecon\\modules\\nf-core\\samtools\\index\\main.nf   \n",
       "4232                 ./git_repos\\viralrecon\\modules\\nf-core\\samtools\\index\\main.nf   \n",
       "4233                 ./git_repos\\viralrecon\\modules\\nf-core\\samtools\\index\\main.nf   \n",
       "4234                  ./git_repos\\viralrecon\\modules\\nf-core\\samtools\\sort\\main.nf   \n",
       "4235                  ./git_repos\\viralrecon\\modules\\nf-core\\samtools\\sort\\main.nf   \n",
       "4236                  ./git_repos\\viralrecon\\modules\\nf-core\\samtools\\sort\\main.nf   \n",
       "4237                 ./git_repos\\viralrecon\\modules\\nf-core\\samtools\\stats\\main.nf   \n",
       "4238                 ./git_repos\\viralrecon\\modules\\nf-core\\samtools\\stats\\main.nf   \n",
       "4239                 ./git_repos\\viralrecon\\modules\\nf-core\\samtools\\stats\\main.nf   \n",
       "4240                  ./git_repos\\viralrecon\\modules\\nf-core\\samtools\\view\\main.nf   \n",
       "4241                  ./git_repos\\viralrecon\\modules\\nf-core\\samtools\\view\\main.nf   \n",
       "4242                  ./git_repos\\viralrecon\\modules\\nf-core\\samtools\\view\\main.nf   \n",
       "4243                    ./git_repos\\viralrecon\\modules\\nf-core\\tabix\\bgzip\\main.nf   \n",
       "4244                    ./git_repos\\viralrecon\\modules\\nf-core\\tabix\\bgzip\\main.nf   \n",
       "4245                    ./git_repos\\viralrecon\\modules\\nf-core\\tabix\\tabix\\main.nf   \n",
       "4246                    ./git_repos\\viralrecon\\modules\\nf-core\\tabix\\tabix\\main.nf   \n",
       "4247                    ./git_repos\\viralrecon\\modules\\nf-core\\tabix\\tabix\\main.nf   \n",
       "\n",
       "      line_number  \\\n",
       "0              24   \n",
       "1              28   \n",
       "2              39   \n",
       "3              43   \n",
       "4              44   \n",
       "5              48   \n",
       "6              49   \n",
       "7              53   \n",
       "8              60   \n",
       "9              62   \n",
       "10             64   \n",
       "11             65   \n",
       "12             69   \n",
       "13             73   \n",
       "14             81   \n",
       "15             84   \n",
       "16             85   \n",
       "17             86   \n",
       "18             87   \n",
       "19             94   \n",
       "20             98   \n",
       "21             33   \n",
       "22             40   \n",
       "23             52   \n",
       "24             29   \n",
       "...           ...   \n",
       "4223           64   \n",
       "4224           34   \n",
       "4225           45   \n",
       "4226           58   \n",
       "4227           24   \n",
       "4228           32   \n",
       "4229           25   \n",
       "4230           33   \n",
       "4231           25   \n",
       "4232           33   \n",
       "4233           45   \n",
       "4234           26   \n",
       "4235           29   \n",
       "4236           40   \n",
       "4237           26   \n",
       "4238           35   \n",
       "4239           46   \n",
       "4240           39   \n",
       "4241           51   \n",
       "4242           63   \n",
       "4243           37   \n",
       "4244           51   \n",
       "4245           24   \n",
       "4246           28   \n",
       "4247           39   \n",
       "\n",
       "                                                                                                 line_content  \n",
       "0                                                 fastqc $args --threads $task.cpus ${prefix}_ASSEMBLED.fastq  \n",
       "1                                                     fastqc: \\$( fastqc --version | sed -e \"s/FastQC v//g\" )  \n",
       "2                                                                                                    fastp \\\\  \n",
       "3                                                                              --json ${prefix}.fastp.json \\\\  \n",
       "4                                                                              --html ${prefix}.fastp.html \\\\  \n",
       "5                                                                                   2> ${prefix}.fastp.log \\\\  \n",
       "6                                                                        | gzip -c > ${prefix}.fastp.fastq.gz  \n",
       "7                                                      fastp: \\$(fastp --version 2>&1 | sed -e \"s/fastp //g\")  \n",
       "8                                                                                                    fastp \\\\  \n",
       "9                                                                         --out1  ${prefix}.fastp.fastq.gz \\\\  \n",
       "10                                                                             --json ${prefix}.fastp.json \\\\  \n",
       "11                                                                             --html ${prefix}.fastp.html \\\\  \n",
       "12                                                                                     2> ${prefix}.fastp.log  \n",
       "13                                                     fastp: \\$(fastp --version 2>&1 | sed -e \"s/fastp //g\")  \n",
       "14                                                                                                   fastp \\\\  \n",
       "15                                                                       --out1 ${prefix}_1.fastp.fastq.gz \\\\  \n",
       "16                                                                       --out2 ${prefix}_2.fastp.fastq.gz \\\\  \n",
       "17                                                                             --json ${prefix}.fastp.json \\\\  \n",
       "18                                                                             --html ${prefix}.fastp.html \\\\  \n",
       "19                                                                                     2> ${prefix}.fastp.log  \n",
       "20                                                     fastp: \\$(fastp --version 2>&1 | sed -e \"s/fastp //g\")  \n",
       "21                                                                                                  fastqc \\\\  \n",
       "22                                                    fastqc: \\$( fastqc --version | sed -e \"s/FastQC v//g\" )  \n",
       "23                                                    fastqc: \\$( fastqc --version | sed -e \"s/FastQC v//g\" )  \n",
       "24                                                                                                 multiqc \\\\  \n",
       "...                                                                                                       ...  \n",
       "4223  picard: \\$(echo \\$(picard CollectMultipleMetrics --version 2>&1) | grep -o 'Version:.*' | cut -f2- -d:)  \n",
       "4224                                                                                                picard \\\\  \n",
       "4225          picard: \\$(echo \\$(picard MarkDuplicates --version 2>&1) | grep -o 'Version:.*' | cut -f2- -d:)  \n",
       "4226          picard: \\$(echo \\$(picard MarkDuplicates --version 2>&1) | grep -o 'Version:.*' | cut -f2- -d:)  \n",
       "4227                                                                                              samtools \\\\  \n",
       "4228                   samtools: \\$(echo \\$(samtools --version 2>&1) | sed 's/^.*samtools //; s/Using.*\\$//')  \n",
       "4229                                                                                              samtools \\\\  \n",
       "4230                   samtools: \\$(echo \\$(samtools --version 2>&1) | sed 's/^.*samtools //; s/Using.*\\$//')  \n",
       "4231                                                                                              samtools \\\\  \n",
       "4232                   samtools: \\$(echo \\$(samtools --version 2>&1) | sed 's/^.*samtools //; s/Using.*\\$//')  \n",
       "4233                   samtools: \\$(echo \\$(samtools --version 2>&1) | sed 's/^.*samtools //; s/Using.*\\$//')  \n",
       "4234                                       samtools sort $args -@ $task.cpus -o ${prefix}.bam -T $prefix $bam  \n",
       "4235                   samtools: \\$(echo \\$(samtools --version 2>&1) | sed 's/^.*samtools //; s/Using.*\\$//')  \n",
       "4236                   samtools: \\$(echo \\$(samtools --version 2>&1) | sed 's/^.*samtools //; s/Using.*\\$//')  \n",
       "4237                                                                                              samtools \\\\  \n",
       "4238                   samtools: \\$(echo \\$(samtools --version 2>&1) | sed 's/^.*samtools //; s/Using.*\\$//')  \n",
       "4239                   samtools: \\$(echo \\$(samtools --version 2>&1) | sed 's/^.*samtools //; s/Using.*\\$//')  \n",
       "4240                                                                                              samtools \\\\  \n",
       "4241                   samtools: \\$(echo \\$(samtools --version 2>&1) | sed 's/^.*samtools //; s/Using.*\\$//')  \n",
       "4242                   samtools: \\$(echo \\$(samtools --version 2>&1) | sed 's/^.*samtools //; s/Using.*\\$//')  \n",
       "4243                                    tabix: \\$(echo \\$(tabix -h 2>&1) | sed 's/^.*Version: //; s/ .*\\$//')  \n",
       "4244                                    tabix: \\$(echo \\$(tabix -h 2>&1) | sed 's/^.*Version: //; s/ .*\\$//')  \n",
       "4245                                                                                         tabix $args $tab  \n",
       "4246                                    tabix: \\$(echo \\$(tabix -h 2>&1) | sed 's/^.*Version: //; s/ .*\\$//')  \n",
       "4247                                    tabix: \\$(echo \\$(tabix -h 2>&1) | sed 's/^.*Version: //; s/ .*\\$//')  \n",
       "\n",
       "[4248 rows x 6 columns]"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "bioinformatics_script = pd.read_csv('./results/bioinformatic_tools_in_scripts.csv')\n",
    "bioinformatics_script"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "074cd7bf",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "operator\n",
       "samtools                         1098\n",
       "multiqc                           447\n",
       "bwa                               385\n",
       "fastp                             348\n",
       "gatk                              336\n",
       "fastqc                            236\n",
       "bowtie2                           169\n",
       "bedtools                          154\n",
       "bcftools                          125\n",
       "picard                            122\n",
       "qiime                              87\n",
       "cutadapt                           69\n",
       "kraken2                            66\n",
       "tabix                              63\n",
       "salmon                             55\n",
       "trim_galore                        54\n",
       "hisat2                             52\n",
       "bwa-mem2                           44\n",
       "minimap2                           31\n",
       "featureCounts                      29\n",
       "gffread                            29\n",
       "stringtie                          28\n",
       "seqtk                              26\n",
       "gtdbtk                             22\n",
       "bowtie2-build                      22\n",
       "seqkit                             20\n",
       "macs2                              18\n",
       "kallisto                           13\n",
       "bismark                            13\n",
       "checkm                             12\n",
       "sourmash                           11\n",
       "NanoPlot                           11\n",
       "bamCoverage                        10\n",
       "vcftools                            9\n",
       "trimmomatic                         9\n",
       "mmseqs                              9\n",
       "megahit                             6\n",
       "bam_stat.py                         4\n",
       "abricate                            3\n",
       "multiBamSummary                     2\n",
       "bismark_methylation_extractor       1\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_tools = bioinformatics_script['operator'].value_counts()\n",
    "all_tools"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "30327de7",
   "metadata": {},
   "source": [
    "Add assigned Group == 0 "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "3be17435",
   "metadata": {},
   "outputs": [],
   "source": [
    "assigned_group = 0\n",
    "bioinformatics_script['Assigned Group'] = assigned_group"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "b523a7a6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pipeline</th>\n",
       "      <th>process_name</th>\n",
       "      <th>operator</th>\n",
       "      <th>file_path</th>\n",
       "      <th>line_number</th>\n",
       "      <th>line_content</th>\n",
       "      <th>Assigned Group</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>airrflow</td>\n",
       "      <td>FASTQC_POSTASSEMBLY</td>\n",
       "      <td>fastqc</td>\n",
       "      <td>./git_repos\\airrflow\\modules\\local\\fastqc_postassembly.nf</td>\n",
       "      <td>24</td>\n",
       "      <td>fastqc $args --threads $task.cpus ${prefix}_ASSEMBLED.fastq</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>airrflow</td>\n",
       "      <td>FASTQC_POSTASSEMBLY</td>\n",
       "      <td>fastqc</td>\n",
       "      <td>./git_repos\\airrflow\\modules\\local\\fastqc_postassembly.nf</td>\n",
       "      <td>28</td>\n",
       "      <td>fastqc: \\$( fastqc --version | sed -e \"s/FastQC v//g\" )</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>airrflow</td>\n",
       "      <td>FASTP</td>\n",
       "      <td>fastp</td>\n",
       "      <td>./git_repos\\airrflow\\modules\\nf-core\\fastp\\main.nf</td>\n",
       "      <td>39</td>\n",
       "      <td>fastp \\\\</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>airrflow</td>\n",
       "      <td>FASTP</td>\n",
       "      <td>fastp</td>\n",
       "      <td>./git_repos\\airrflow\\modules\\nf-core\\fastp\\main.nf</td>\n",
       "      <td>43</td>\n",
       "      <td>--json ${prefix}.fastp.json \\\\</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>airrflow</td>\n",
       "      <td>FASTP</td>\n",
       "      <td>fastp</td>\n",
       "      <td>./git_repos\\airrflow\\modules\\nf-core\\fastp\\main.nf</td>\n",
       "      <td>44</td>\n",
       "      <td>--html ${prefix}.fastp.html \\\\</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   pipeline         process_name operator  \\\n",
       "0  airrflow  FASTQC_POSTASSEMBLY   fastqc   \n",
       "1  airrflow  FASTQC_POSTASSEMBLY   fastqc   \n",
       "2  airrflow                FASTP    fastp   \n",
       "3  airrflow                FASTP    fastp   \n",
       "4  airrflow                FASTP    fastp   \n",
       "\n",
       "                                                   file_path  line_number  \\\n",
       "0  ./git_repos\\airrflow\\modules\\local\\fastqc_postassembly.nf           24   \n",
       "1  ./git_repos\\airrflow\\modules\\local\\fastqc_postassembly.nf           28   \n",
       "2         ./git_repos\\airrflow\\modules\\nf-core\\fastp\\main.nf           39   \n",
       "3         ./git_repos\\airrflow\\modules\\nf-core\\fastp\\main.nf           43   \n",
       "4         ./git_repos\\airrflow\\modules\\nf-core\\fastp\\main.nf           44   \n",
       "\n",
       "                                                  line_content  Assigned Group  \n",
       "0  fastqc $args --threads $task.cpus ${prefix}_ASSEMBLED.fastq               0  \n",
       "1      fastqc: \\$( fastqc --version | sed -e \"s/FastQC v//g\" )               0  \n",
       "2                                                     fastp \\\\               0  \n",
       "3                               --json ${prefix}.fastp.json \\\\               0  \n",
       "4                               --html ${prefix}.fastp.html \\\\               0  "
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bioinformatics_script = bioinformatics_script.rename(columns={'Operator': 'operator'})\n",
    "bioinformatics_script.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "46345c37",
   "metadata": {},
   "source": [
    "# Analysis of operators in operator domain dataset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 143,
   "id": "811318b7",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "\n",
    "operator_script = pd.read_csv('./results/all_operators_in_scripts_new.csv')\n",
    "df_operator_clustered = pd.read_csv('./data/operator_dataset_clustered.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 144,
   "id": "3198e641",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_operator_clustered_assigned_group_one = df_operator_clustered[df_operator_clustered['Assigned Group'] == 0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 145,
   "id": "2e2aef8b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Operator</th>\n",
       "      <th>Assigned Group</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>samtools</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>bcftools</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>bedtools</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>fastqc</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>bwa</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>bedextract</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>81</th>\n",
       "      <td>seqtk</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83</th>\n",
       "      <td>{output.bed}</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>99</th>\n",
       "      <td>{output.bam}</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>112</th>\n",
       "      <td>{input.bam}</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>115</th>\n",
       "      <td>{input.bed}</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121</th>\n",
       "      <td>bamCoverage</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>122</th>\n",
       "      <td>seqkit</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>136</th>\n",
       "      <td>mmseqs</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>150</th>\n",
       "      <td>{params.bcftools}</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>156</th>\n",
       "      <td>{input.fasta}</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>167</th>\n",
       "      <td>bedGraphToBigWig</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>186</th>\n",
       "      <td>{output.fasta}</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>194</th>\n",
       "      <td>{output.fname_bam}</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>203</th>\n",
       "      <td>{input.fastq}</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>209</th>\n",
       "      <td>{output.fastq}</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>214</th>\n",
       "      <td>biom</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>222</th>\n",
       "      <td>sambamba</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>246</th>\n",
       "      <td>INPUT={input.bam}</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>274</th>\n",
       "      <td>{params.fasta}</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4747</th>\n",
       "      <td>trim=${{fq%_[12].fastq.gz}}</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4748</th>\n",
       "      <td>numLines=$fastq-dump</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4749</th>\n",
       "      <td>fname=${{trim}}_1.fastq</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4755</th>\n",
       "      <td>{output.tbed}</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4763</th>\n",
       "      <td>{params.execdir}/scripts/make_bed_from_gubbins.py</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4764</th>\n",
       "      <td>{params.execdir}/resources/alignment_references/{wildcards.ref}.fasta</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4777</th>\n",
       "      <td>{params.execdir}/scripts/make_vcf_density_bed.py</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4778</th>\n",
       "      <td>{params.execdir}/scripts/make_PE_PPE_BED.py</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4785</th>\n",
       "      <td>{params.execdir}/resources/alignment_references/mabcessus.fasta</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4790</th>\n",
       "      <td>{input.mabs_bam}</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4800</th>\n",
       "      <td>REFERENCE_SEQUENCE={params.genome}</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4806</th>\n",
       "      <td>{params.cbam}</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4889</th>\n",
       "      <td>{output.bam};</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4894</th>\n",
       "      <td>fastqsplitter</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4916</th>\n",
       "      <td>{output.bwa1}</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4918</th>\n",
       "      <td>{output.bwa2}</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4920</th>\n",
       "      <td>{input.bwa1}</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4931</th>\n",
       "      <td>1.bed</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4932</th>\n",
       "      <td>2.bed</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4933</th>\n",
       "      <td>3.bed</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4934</th>\n",
       "      <td>4.bed</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4935</th>\n",
       "      <td>frags.bed</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4936</th>\n",
       "      <td>fragments.bed</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4944</th>\n",
       "      <td>{params.base_dir}/temp.fastq</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4945</th>\n",
       "      <td>{params.base_dir}/{wildcards.sample}.ribosomes.fastq</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>467 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                   Operator  \\\n",
       "2                                                                  samtools   \n",
       "9                                                                  bcftools   \n",
       "22                                                                 bedtools   \n",
       "39                                                                   fastqc   \n",
       "46                                                                      bwa   \n",
       "47                                                               bedextract   \n",
       "81                                                                    seqtk   \n",
       "83                                                             {output.bed}   \n",
       "99                                                             {output.bam}   \n",
       "112                                                             {input.bam}   \n",
       "115                                                             {input.bed}   \n",
       "121                                                             bamCoverage   \n",
       "122                                                                  seqkit   \n",
       "136                                                                  mmseqs   \n",
       "150                                                       {params.bcftools}   \n",
       "156                                                           {input.fasta}   \n",
       "167                                                        bedGraphToBigWig   \n",
       "186                                                          {output.fasta}   \n",
       "194                                                      {output.fname_bam}   \n",
       "203                                                           {input.fastq}   \n",
       "209                                                          {output.fastq}   \n",
       "214                                                                    biom   \n",
       "222                                                                sambamba   \n",
       "246                                                       INPUT={input.bam}   \n",
       "274                                                          {params.fasta}   \n",
       "...                                                                     ...   \n",
       "4747                                            trim=${{fq%_[12].fastq.gz}}   \n",
       "4748                                                   numLines=$fastq-dump   \n",
       "4749                                                fname=${{trim}}_1.fastq   \n",
       "4755                                                          {output.tbed}   \n",
       "4763                      {params.execdir}/scripts/make_bed_from_gubbins.py   \n",
       "4764  {params.execdir}/resources/alignment_references/{wildcards.ref}.fasta   \n",
       "4777                       {params.execdir}/scripts/make_vcf_density_bed.py   \n",
       "4778                            {params.execdir}/scripts/make_PE_PPE_BED.py   \n",
       "4785        {params.execdir}/resources/alignment_references/mabcessus.fasta   \n",
       "4790                                                       {input.mabs_bam}   \n",
       "4800                                     REFERENCE_SEQUENCE={params.genome}   \n",
       "4806                                                          {params.cbam}   \n",
       "4889                                                          {output.bam};   \n",
       "4894                                                          fastqsplitter   \n",
       "4916                                                          {output.bwa1}   \n",
       "4918                                                          {output.bwa2}   \n",
       "4920                                                           {input.bwa1}   \n",
       "4931                                                                  1.bed   \n",
       "4932                                                                  2.bed   \n",
       "4933                                                                  3.bed   \n",
       "4934                                                                  4.bed   \n",
       "4935                                                              frags.bed   \n",
       "4936                                                          fragments.bed   \n",
       "4944                                           {params.base_dir}/temp.fastq   \n",
       "4945                   {params.base_dir}/{wildcards.sample}.ribosomes.fastq   \n",
       "\n",
       "      Assigned Group  \n",
       "2                  0  \n",
       "9                  0  \n",
       "22                 0  \n",
       "39                 0  \n",
       "46                 0  \n",
       "47                 0  \n",
       "81                 0  \n",
       "83                 0  \n",
       "99                 0  \n",
       "112                0  \n",
       "115                0  \n",
       "121                0  \n",
       "122                0  \n",
       "136                0  \n",
       "150                0  \n",
       "156                0  \n",
       "167                0  \n",
       "186                0  \n",
       "194                0  \n",
       "203                0  \n",
       "209                0  \n",
       "214                0  \n",
       "222                0  \n",
       "246                0  \n",
       "274                0  \n",
       "...              ...  \n",
       "4747               0  \n",
       "4748               0  \n",
       "4749               0  \n",
       "4755               0  \n",
       "4763               0  \n",
       "4764               0  \n",
       "4777               0  \n",
       "4778               0  \n",
       "4785               0  \n",
       "4790               0  \n",
       "4800               0  \n",
       "4806               0  \n",
       "4889               0  \n",
       "4894               0  \n",
       "4916               0  \n",
       "4918               0  \n",
       "4920               0  \n",
       "4931               0  \n",
       "4932               0  \n",
       "4933               0  \n",
       "4934               0  \n",
       "4935               0  \n",
       "4936               0  \n",
       "4944               0  \n",
       "4945               0  \n",
       "\n",
       "[467 rows x 2 columns]"
      ]
     },
     "execution_count": 145,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_operator_clustered_assigned_group_one"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 146,
   "id": "2a72a951",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Operator</th>\n",
       "      <th>Assigned Group</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>{log}</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>{output}</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>samtools</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>echo</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>shell:</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Operator  Assigned Group\n",
       "0     {log}               6\n",
       "1  {output}               6\n",
       "2  samtools               0\n",
       "3      echo               2\n",
       "4    shell:               4"
      ]
     },
     "execution_count": 146,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_operator_clustered.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a44f5c99",
   "metadata": {},
   "source": [
    "Add Assigned Group == 0  to bioinformatic tools"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 147,
   "id": "7b60eae2",
   "metadata": {},
   "outputs": [],
   "source": [
    "all_process = operator_script.groupby(['pipeline', 'process_name']).size().reset_index(name='operator_count')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 148,
   "id": "1d1c326e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "2941"
      ]
     },
     "execution_count": 148,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sum_all_process = len(all_process)\n",
    "sum_all_process"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 187,
   "id": "b62e0e42",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "29532\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pipeline</th>\n",
       "      <th>process_name</th>\n",
       "      <th>operator</th>\n",
       "      <th>file_path</th>\n",
       "      <th>line_number</th>\n",
       "      <th>line_content</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>62</th>\n",
       "      <td>diaproteomics</td>\n",
       "      <td>get_software_versions</td>\n",
       "      <td>echo</td>\n",
       "      <td>./git_repos/diaproteomics/main.nf</td>\n",
       "      <td>335</td>\n",
       "      <td>echo $workflow.manifest.version &gt; v_pipeline.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>72</th>\n",
       "      <td>diaproteomics</td>\n",
       "      <td>get_software_versions</td>\n",
       "      <td>echo</td>\n",
       "      <td>./git_repos/diaproteomics/main.nf</td>\n",
       "      <td>336</td>\n",
       "      <td>echo $workflow.nextflow.version &gt; v_nextflow.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>76</th>\n",
       "      <td>diaproteomics</td>\n",
       "      <td>get_software_versions</td>\n",
       "      <td>nextflow</td>\n",
       "      <td>./git_repos/diaproteomics/main.nf</td>\n",
       "      <td>336</td>\n",
       "      <td>echo $workflow.nextflow.version &gt; v_nextflow.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>117</th>\n",
       "      <td>diaproteomics</td>\n",
       "      <td>dda_raw_file_conversion</td>\n",
       "      <td>Name</td>\n",
       "      <td>./git_repos/diaproteomics/main.nf</td>\n",
       "      <td>359</td>\n",
       "      <td>ThermoRawFileParser.sh -i=${raw_file} -f=2 -b=${raw_file.baseName}.mzML</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>119</th>\n",
       "      <td>diaproteomics</td>\n",
       "      <td>dda_raw_file_conversion</td>\n",
       "      <td>file</td>\n",
       "      <td>./git_repos/diaproteomics/main.nf</td>\n",
       "      <td>359</td>\n",
       "      <td>ThermoRawFileParser.sh -i=${raw_file} -f=2 -b=${raw_file.baseName}.mzML</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>129</th>\n",
       "      <td>diaproteomics</td>\n",
       "      <td>dda_id_format_conversion</td>\n",
       "      <td>FileConverter</td>\n",
       "      <td>./git_repos/diaproteomics/main.nf</td>\n",
       "      <td>380</td>\n",
       "      <td>IDFileConverter -in ${dda_id_file} -out ${id}_${sample}_peptide_ids.idXML -threads ${task.cpus}</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>132</th>\n",
       "      <td>diaproteomics</td>\n",
       "      <td>dda_id_format_conversion</td>\n",
       "      <td>file</td>\n",
       "      <td>./git_repos/diaproteomics/main.nf</td>\n",
       "      <td>380</td>\n",
       "      <td>IDFileConverter -in ${dda_id_file} -out ${id}_${sample}_peptide_ids.idXML -threads ${task.cpus}</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>135</th>\n",
       "      <td>diaproteomics</td>\n",
       "      <td>dda_library_generation</td>\n",
       "      <td>convert</td>\n",
       "      <td>./git_repos/diaproteomics/main.nf</td>\n",
       "      <td>402</td>\n",
       "      <td>easypqp convert \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>143</th>\n",
       "      <td>diaproteomics</td>\n",
       "      <td>dda_library_generation</td>\n",
       "      <td>file</td>\n",
       "      <td>./git_repos/diaproteomics/main.nf</td>\n",
       "      <td>403</td>\n",
       "      <td>--unimod ${unimod_file} \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>150</th>\n",
       "      <td>diaproteomics</td>\n",
       "      <td>dda_library_generation</td>\n",
       "      <td>file</td>\n",
       "      <td>./git_repos/diaproteomics/main.nf</td>\n",
       "      <td>404</td>\n",
       "      <td>--pepxml ${idxml_file} \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>159</th>\n",
       "      <td>diaproteomics</td>\n",
       "      <td>dda_library_generation</td>\n",
       "      <td>file</td>\n",
       "      <td>./git_repos/diaproteomics/main.nf</td>\n",
       "      <td>405</td>\n",
       "      <td>--spectra ${dda_mzml_file}</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>174</th>\n",
       "      <td>diaproteomics</td>\n",
       "      <td>dda_library_generation</td>\n",
       "      <td>Name</td>\n",
       "      <td>./git_repos/diaproteomics/main.nf</td>\n",
       "      <td>408</td>\n",
       "      <td>--out ${dda_mzml_file.baseName}_run_peaks.tsv \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>176</th>\n",
       "      <td>diaproteomics</td>\n",
       "      <td>dda_library_generation</td>\n",
       "      <td>file</td>\n",
       "      <td>./git_repos/diaproteomics/main.nf</td>\n",
       "      <td>408</td>\n",
       "      <td>--out ${dda_mzml_file.baseName}_run_peaks.tsv \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>186</th>\n",
       "      <td>diaproteomics</td>\n",
       "      <td>dda_library_generation</td>\n",
       "      <td>params</td>\n",
       "      <td>./git_repos/diaproteomics/main.nf</td>\n",
       "      <td>409</td>\n",
       "      <td>--rt_psm_fdr_threshold ${params.library_rt_fdr} \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>197</th>\n",
       "      <td>diaproteomics</td>\n",
       "      <td>dda_library_generation</td>\n",
       "      <td>Name</td>\n",
       "      <td>./git_repos/diaproteomics/main.nf</td>\n",
       "      <td>411</td>\n",
       "      <td>${dda_mzml_file.baseName}.psmpkl \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>198</th>\n",
       "      <td>diaproteomics</td>\n",
       "      <td>dda_library_generation</td>\n",
       "      <td>file</td>\n",
       "      <td>./git_repos/diaproteomics/main.nf</td>\n",
       "      <td>411</td>\n",
       "      <td>${dda_mzml_file.baseName}.psmpkl \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>206</th>\n",
       "      <td>diaproteomics</td>\n",
       "      <td>dda_library_generation</td>\n",
       "      <td>Name</td>\n",
       "      <td>./git_repos/diaproteomics/main.nf</td>\n",
       "      <td>412</td>\n",
       "      <td>${dda_mzml_file.baseName}.peakpkl</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>207</th>\n",
       "      <td>diaproteomics</td>\n",
       "      <td>dda_library_generation</td>\n",
       "      <td>file</td>\n",
       "      <td>./git_repos/diaproteomics/main.nf</td>\n",
       "      <td>412</td>\n",
       "      <td>${dda_mzml_file.baseName}.peakpkl</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>219</th>\n",
       "      <td>diaproteomics</td>\n",
       "      <td>dda_library_generation</td>\n",
       "      <td>Name</td>\n",
       "      <td>./git_repos/diaproteomics/main.nf</td>\n",
       "      <td>414</td>\n",
       "      <td>mv ${dda_mzml_file.baseName}_run_peaks.tsv ${id}_${sample}_library.tsv</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>221</th>\n",
       "      <td>diaproteomics</td>\n",
       "      <td>dda_library_generation</td>\n",
       "      <td>file</td>\n",
       "      <td>./git_repos/diaproteomics/main.nf</td>\n",
       "      <td>414</td>\n",
       "      <td>mv ${dda_mzml_file.baseName}_run_peaks.tsv ${id}_${sample}_library.tsv</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>225</th>\n",
       "      <td>diaproteomics</td>\n",
       "      <td>assay_generation</td>\n",
       "      <td>FileConverter</td>\n",
       "      <td>./git_repos/diaproteomics/main.nf</td>\n",
       "      <td>435</td>\n",
       "      <td>TargetedFileConverter \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>231</th>\n",
       "      <td>diaproteomics</td>\n",
       "      <td>assay_generation</td>\n",
       "      <td>file</td>\n",
       "      <td>./git_repos/diaproteomics/main.nf</td>\n",
       "      <td>436</td>\n",
       "      <td>-in ${lib_file_na} \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>238</th>\n",
       "      <td>diaproteomics</td>\n",
       "      <td>assay_generation</td>\n",
       "      <td>Name</td>\n",
       "      <td>./git_repos/diaproteomics/main.nf</td>\n",
       "      <td>437</td>\n",
       "      <td>-out ${lib_file_na.baseName}.tsv \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>240</th>\n",
       "      <td>diaproteomics</td>\n",
       "      <td>assay_generation</td>\n",
       "      <td>file</td>\n",
       "      <td>./git_repos/diaproteomics/main.nf</td>\n",
       "      <td>437</td>\n",
       "      <td>-out ${lib_file_na.baseName}.tsv \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>259</th>\n",
       "      <td>diaproteomics</td>\n",
       "      <td>assay_generation</td>\n",
       "      <td>Name</td>\n",
       "      <td>./git_repos/diaproteomics/main.nf</td>\n",
       "      <td>441</td>\n",
       "      <td>-in ${lib_file_na.baseName}.tsv \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5988</th>\n",
       "      <td>callingcards</td>\n",
       "      <td>SAMTOOLS_INDEX</td>\n",
       "      <td>touch</td>\n",
       "      <td>./git_repos/callingcards/modules/nf-core/samtools/index/main.nf</td>\n",
       "      <td>40</td>\n",
       "      <td>touch ${input}.crai</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5992</th>\n",
       "      <td>callingcards</td>\n",
       "      <td>SAMTOOLS_INDEX</td>\n",
       "      <td>input</td>\n",
       "      <td>./git_repos/callingcards/modules/nf-core/samtools/index/main.nf</td>\n",
       "      <td>40</td>\n",
       "      <td>touch ${input}.crai</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5993</th>\n",
       "      <td>callingcards</td>\n",
       "      <td>SAMTOOLS_INDEX</td>\n",
       "      <td>{input}</td>\n",
       "      <td>./git_repos/callingcards/modules/nf-core/samtools/index/main.nf</td>\n",
       "      <td>41</td>\n",
       "      <td>touch ${input}.csi</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5994</th>\n",
       "      <td>callingcards</td>\n",
       "      <td>SAMTOOLS_INDEX</td>\n",
       "      <td>touch</td>\n",
       "      <td>./git_repos/callingcards/modules/nf-core/samtools/index/main.nf</td>\n",
       "      <td>41</td>\n",
       "      <td>touch ${input}.csi</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5998</th>\n",
       "      <td>callingcards</td>\n",
       "      <td>SAMTOOLS_INDEX</td>\n",
       "      <td>input</td>\n",
       "      <td>./git_repos/callingcards/modules/nf-core/samtools/index/main.nf</td>\n",
       "      <td>41</td>\n",
       "      <td>touch ${input}.csi</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6014</th>\n",
       "      <td>callingcards</td>\n",
       "      <td>SAMTOOLS_INDEX</td>\n",
       "      <td>samtools</td>\n",
       "      <td>./git_repos/callingcards/modules/nf-core/samtools/index/main.nf</td>\n",
       "      <td>45</td>\n",
       "      <td>samtools: \\$(echo \\$(samtools --version 2&gt;&amp;1) | sed 's/^.*samtools //; s/Using.*\\$//')</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6015</th>\n",
       "      <td>callingcards</td>\n",
       "      <td>SAMTOOLS_INDEX</td>\n",
       "      <td>echo</td>\n",
       "      <td>./git_repos/callingcards/modules/nf-core/samtools/index/main.nf</td>\n",
       "      <td>45</td>\n",
       "      <td>samtools: \\$(echo \\$(samtools --version 2&gt;&amp;1) | sed 's/^.*samtools //; s/Using.*\\$//')</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6036</th>\n",
       "      <td>callingcards</td>\n",
       "      <td>SAMTOOLS_FAIDX</td>\n",
       "      <td>samtools</td>\n",
       "      <td>./git_repos/callingcards/modules/nf-core/samtools/faidx/main.nf</td>\n",
       "      <td>26</td>\n",
       "      <td>samtools \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6047</th>\n",
       "      <td>callingcards</td>\n",
       "      <td>SAMTOOLS_FAIDX</td>\n",
       "      <td>fasta</td>\n",
       "      <td>./git_repos/callingcards/modules/nf-core/samtools/faidx/main.nf</td>\n",
       "      <td>28</td>\n",
       "      <td>$fasta \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6066</th>\n",
       "      <td>callingcards</td>\n",
       "      <td>SAMTOOLS_FAIDX</td>\n",
       "      <td>samtools</td>\n",
       "      <td>./git_repos/callingcards/modules/nf-core/samtools/faidx/main.nf</td>\n",
       "      <td>33</td>\n",
       "      <td>samtools: \\$(echo \\$(samtools --version 2&gt;&amp;1) | sed 's/^.*samtools //; s/Using.*\\$//')</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6067</th>\n",
       "      <td>callingcards</td>\n",
       "      <td>SAMTOOLS_FAIDX</td>\n",
       "      <td>echo</td>\n",
       "      <td>./git_repos/callingcards/modules/nf-core/samtools/faidx/main.nf</td>\n",
       "      <td>33</td>\n",
       "      <td>samtools: \\$(echo \\$(samtools --version 2&gt;&amp;1) | sed 's/^.*samtools //; s/Using.*\\$//')</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6091</th>\n",
       "      <td>callingcards</td>\n",
       "      <td>SAMTOOLS_FAIDX</td>\n",
       "      <td>fasta</td>\n",
       "      <td>./git_repos/callingcards/modules/nf-core/samtools/faidx/main.nf</td>\n",
       "      <td>41</td>\n",
       "      <td>${fastacmd}</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6093</th>\n",
       "      <td>callingcards</td>\n",
       "      <td>SAMTOOLS_FAIDX</td>\n",
       "      <td>touch</td>\n",
       "      <td>./git_repos/callingcards/modules/nf-core/samtools/faidx/main.nf</td>\n",
       "      <td>42</td>\n",
       "      <td>touch ${fasta}.fai</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6098</th>\n",
       "      <td>callingcards</td>\n",
       "      <td>SAMTOOLS_FAIDX</td>\n",
       "      <td>fasta</td>\n",
       "      <td>./git_repos/callingcards/modules/nf-core/samtools/faidx/main.nf</td>\n",
       "      <td>42</td>\n",
       "      <td>touch ${fasta}.fai</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6114</th>\n",
       "      <td>callingcards</td>\n",
       "      <td>SAMTOOLS_FAIDX</td>\n",
       "      <td>samtools</td>\n",
       "      <td>./git_repos/callingcards/modules/nf-core/samtools/faidx/main.nf</td>\n",
       "      <td>47</td>\n",
       "      <td>samtools: \\$(echo \\$(samtools --version 2&gt;&amp;1) | sed 's/^.*samtools //; s/Using.*\\$//')</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6115</th>\n",
       "      <td>callingcards</td>\n",
       "      <td>SAMTOOLS_FAIDX</td>\n",
       "      <td>echo</td>\n",
       "      <td>./git_repos/callingcards/modules/nf-core/samtools/faidx/main.nf</td>\n",
       "      <td>47</td>\n",
       "      <td>samtools: \\$(echo \\$(samtools --version 2&gt;&amp;1) | sed 's/^.*samtools //; s/Using.*\\$//')</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6136</th>\n",
       "      <td>callingcards</td>\n",
       "      <td>SEQKIT_SPLIT2</td>\n",
       "      <td>seqkit</td>\n",
       "      <td>./git_repos/callingcards/modules/nf-core/seqkit/split2/main.nf</td>\n",
       "      <td>25</td>\n",
       "      <td>seqkit \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6140</th>\n",
       "      <td>callingcards</td>\n",
       "      <td>SEQKIT_SPLIT2</td>\n",
       "      <td>split</td>\n",
       "      <td>./git_repos/callingcards/modules/nf-core/seqkit/split2/main.nf</td>\n",
       "      <td>26</td>\n",
       "      <td>split2 \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6180</th>\n",
       "      <td>callingcards</td>\n",
       "      <td>SEQKIT_SPLIT2</td>\n",
       "      <td>echo</td>\n",
       "      <td>./git_repos/callingcards/modules/nf-core/seqkit/split2/main.nf</td>\n",
       "      <td>34</td>\n",
       "      <td>seqkit: \\$(echo \\$(seqkit 2&gt;&amp;1) | sed 's/^.*Version: //; s/ .*\\$//')</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6186</th>\n",
       "      <td>callingcards</td>\n",
       "      <td>SEQKIT_SPLIT2</td>\n",
       "      <td>seqkit</td>\n",
       "      <td>./git_repos/callingcards/modules/nf-core/seqkit/split2/main.nf</td>\n",
       "      <td>34</td>\n",
       "      <td>seqkit: \\$(echo \\$(seqkit 2&gt;&amp;1) | sed 's/^.*Version: //; s/ .*\\$//')</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6199</th>\n",
       "      <td>callingcards</td>\n",
       "      <td>SEQKIT_SPLIT2</td>\n",
       "      <td>seqkit</td>\n",
       "      <td>./git_repos/callingcards/modules/nf-core/seqkit/split2/main.nf</td>\n",
       "      <td>39</td>\n",
       "      <td>seqkit \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6203</th>\n",
       "      <td>callingcards</td>\n",
       "      <td>SEQKIT_SPLIT2</td>\n",
       "      <td>split</td>\n",
       "      <td>./git_repos/callingcards/modules/nf-core/seqkit/split2/main.nf</td>\n",
       "      <td>40</td>\n",
       "      <td>split2 \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6252</th>\n",
       "      <td>callingcards</td>\n",
       "      <td>SEQKIT_SPLIT2</td>\n",
       "      <td>echo</td>\n",
       "      <td>./git_repos/callingcards/modules/nf-core/seqkit/split2/main.nf</td>\n",
       "      <td>49</td>\n",
       "      <td>seqkit: \\$(echo \\$(seqkit 2&gt;&amp;1) | sed 's/^.*Version: //; s/ .*\\$//')</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6258</th>\n",
       "      <td>callingcards</td>\n",
       "      <td>SEQKIT_SPLIT2</td>\n",
       "      <td>seqkit</td>\n",
       "      <td>./git_repos/callingcards/modules/nf-core/seqkit/split2/main.nf</td>\n",
       "      <td>49</td>\n",
       "      <td>seqkit: \\$(echo \\$(seqkit 2&gt;&amp;1) | sed 's/^.*Version: //; s/ .*\\$//')</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6274</th>\n",
       "      <td>callingcards</td>\n",
       "      <td>BWAMEM2_MEM</td>\n",
       "      <td>find</td>\n",
       "      <td>./git_repos/callingcards/modules/nf-core/bwamem2/mem/main.nf</td>\n",
       "      <td>28</td>\n",
       "      <td>INDEX=`find -L ./ -name \"*.amb\" | sed 's/\\\\.amb\\$//'`</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>600 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           pipeline              process_name       operator  \\\n",
       "62    diaproteomics     get_software_versions           echo   \n",
       "72    diaproteomics     get_software_versions           echo   \n",
       "76    diaproteomics     get_software_versions       nextflow   \n",
       "117   diaproteomics   dda_raw_file_conversion           Name   \n",
       "119   diaproteomics   dda_raw_file_conversion           file   \n",
       "129   diaproteomics  dda_id_format_conversion  FileConverter   \n",
       "132   diaproteomics  dda_id_format_conversion           file   \n",
       "135   diaproteomics    dda_library_generation        convert   \n",
       "143   diaproteomics    dda_library_generation           file   \n",
       "150   diaproteomics    dda_library_generation           file   \n",
       "159   diaproteomics    dda_library_generation           file   \n",
       "174   diaproteomics    dda_library_generation           Name   \n",
       "176   diaproteomics    dda_library_generation           file   \n",
       "186   diaproteomics    dda_library_generation         params   \n",
       "197   diaproteomics    dda_library_generation           Name   \n",
       "198   diaproteomics    dda_library_generation           file   \n",
       "206   diaproteomics    dda_library_generation           Name   \n",
       "207   diaproteomics    dda_library_generation           file   \n",
       "219   diaproteomics    dda_library_generation           Name   \n",
       "221   diaproteomics    dda_library_generation           file   \n",
       "225   diaproteomics          assay_generation  FileConverter   \n",
       "231   diaproteomics          assay_generation           file   \n",
       "238   diaproteomics          assay_generation           Name   \n",
       "240   diaproteomics          assay_generation           file   \n",
       "259   diaproteomics          assay_generation           Name   \n",
       "...             ...                       ...            ...   \n",
       "5988   callingcards            SAMTOOLS_INDEX          touch   \n",
       "5992   callingcards            SAMTOOLS_INDEX          input   \n",
       "5993   callingcards            SAMTOOLS_INDEX        {input}   \n",
       "5994   callingcards            SAMTOOLS_INDEX          touch   \n",
       "5998   callingcards            SAMTOOLS_INDEX          input   \n",
       "6014   callingcards            SAMTOOLS_INDEX       samtools   \n",
       "6015   callingcards            SAMTOOLS_INDEX           echo   \n",
       "6036   callingcards            SAMTOOLS_FAIDX       samtools   \n",
       "6047   callingcards            SAMTOOLS_FAIDX          fasta   \n",
       "6066   callingcards            SAMTOOLS_FAIDX       samtools   \n",
       "6067   callingcards            SAMTOOLS_FAIDX           echo   \n",
       "6091   callingcards            SAMTOOLS_FAIDX          fasta   \n",
       "6093   callingcards            SAMTOOLS_FAIDX          touch   \n",
       "6098   callingcards            SAMTOOLS_FAIDX          fasta   \n",
       "6114   callingcards            SAMTOOLS_FAIDX       samtools   \n",
       "6115   callingcards            SAMTOOLS_FAIDX           echo   \n",
       "6136   callingcards             SEQKIT_SPLIT2         seqkit   \n",
       "6140   callingcards             SEQKIT_SPLIT2          split   \n",
       "6180   callingcards             SEQKIT_SPLIT2           echo   \n",
       "6186   callingcards             SEQKIT_SPLIT2         seqkit   \n",
       "6199   callingcards             SEQKIT_SPLIT2         seqkit   \n",
       "6203   callingcards             SEQKIT_SPLIT2          split   \n",
       "6252   callingcards             SEQKIT_SPLIT2           echo   \n",
       "6258   callingcards             SEQKIT_SPLIT2         seqkit   \n",
       "6274   callingcards               BWAMEM2_MEM           find   \n",
       "\n",
       "                                                            file_path  \\\n",
       "62                                  ./git_repos/diaproteomics/main.nf   \n",
       "72                                  ./git_repos/diaproteomics/main.nf   \n",
       "76                                  ./git_repos/diaproteomics/main.nf   \n",
       "117                                 ./git_repos/diaproteomics/main.nf   \n",
       "119                                 ./git_repos/diaproteomics/main.nf   \n",
       "129                                 ./git_repos/diaproteomics/main.nf   \n",
       "132                                 ./git_repos/diaproteomics/main.nf   \n",
       "135                                 ./git_repos/diaproteomics/main.nf   \n",
       "143                                 ./git_repos/diaproteomics/main.nf   \n",
       "150                                 ./git_repos/diaproteomics/main.nf   \n",
       "159                                 ./git_repos/diaproteomics/main.nf   \n",
       "174                                 ./git_repos/diaproteomics/main.nf   \n",
       "176                                 ./git_repos/diaproteomics/main.nf   \n",
       "186                                 ./git_repos/diaproteomics/main.nf   \n",
       "197                                 ./git_repos/diaproteomics/main.nf   \n",
       "198                                 ./git_repos/diaproteomics/main.nf   \n",
       "206                                 ./git_repos/diaproteomics/main.nf   \n",
       "207                                 ./git_repos/diaproteomics/main.nf   \n",
       "219                                 ./git_repos/diaproteomics/main.nf   \n",
       "221                                 ./git_repos/diaproteomics/main.nf   \n",
       "225                                 ./git_repos/diaproteomics/main.nf   \n",
       "231                                 ./git_repos/diaproteomics/main.nf   \n",
       "238                                 ./git_repos/diaproteomics/main.nf   \n",
       "240                                 ./git_repos/diaproteomics/main.nf   \n",
       "259                                 ./git_repos/diaproteomics/main.nf   \n",
       "...                                                               ...   \n",
       "5988  ./git_repos/callingcards/modules/nf-core/samtools/index/main.nf   \n",
       "5992  ./git_repos/callingcards/modules/nf-core/samtools/index/main.nf   \n",
       "5993  ./git_repos/callingcards/modules/nf-core/samtools/index/main.nf   \n",
       "5994  ./git_repos/callingcards/modules/nf-core/samtools/index/main.nf   \n",
       "5998  ./git_repos/callingcards/modules/nf-core/samtools/index/main.nf   \n",
       "6014  ./git_repos/callingcards/modules/nf-core/samtools/index/main.nf   \n",
       "6015  ./git_repos/callingcards/modules/nf-core/samtools/index/main.nf   \n",
       "6036  ./git_repos/callingcards/modules/nf-core/samtools/faidx/main.nf   \n",
       "6047  ./git_repos/callingcards/modules/nf-core/samtools/faidx/main.nf   \n",
       "6066  ./git_repos/callingcards/modules/nf-core/samtools/faidx/main.nf   \n",
       "6067  ./git_repos/callingcards/modules/nf-core/samtools/faidx/main.nf   \n",
       "6091  ./git_repos/callingcards/modules/nf-core/samtools/faidx/main.nf   \n",
       "6093  ./git_repos/callingcards/modules/nf-core/samtools/faidx/main.nf   \n",
       "6098  ./git_repos/callingcards/modules/nf-core/samtools/faidx/main.nf   \n",
       "6114  ./git_repos/callingcards/modules/nf-core/samtools/faidx/main.nf   \n",
       "6115  ./git_repos/callingcards/modules/nf-core/samtools/faidx/main.nf   \n",
       "6136   ./git_repos/callingcards/modules/nf-core/seqkit/split2/main.nf   \n",
       "6140   ./git_repos/callingcards/modules/nf-core/seqkit/split2/main.nf   \n",
       "6180   ./git_repos/callingcards/modules/nf-core/seqkit/split2/main.nf   \n",
       "6186   ./git_repos/callingcards/modules/nf-core/seqkit/split2/main.nf   \n",
       "6199   ./git_repos/callingcards/modules/nf-core/seqkit/split2/main.nf   \n",
       "6203   ./git_repos/callingcards/modules/nf-core/seqkit/split2/main.nf   \n",
       "6252   ./git_repos/callingcards/modules/nf-core/seqkit/split2/main.nf   \n",
       "6258   ./git_repos/callingcards/modules/nf-core/seqkit/split2/main.nf   \n",
       "6274     ./git_repos/callingcards/modules/nf-core/bwamem2/mem/main.nf   \n",
       "\n",
       "      line_number  \\\n",
       "62            335   \n",
       "72            336   \n",
       "76            336   \n",
       "117           359   \n",
       "119           359   \n",
       "129           380   \n",
       "132           380   \n",
       "135           402   \n",
       "143           403   \n",
       "150           404   \n",
       "159           405   \n",
       "174           408   \n",
       "176           408   \n",
       "186           409   \n",
       "197           411   \n",
       "198           411   \n",
       "206           412   \n",
       "207           412   \n",
       "219           414   \n",
       "221           414   \n",
       "225           435   \n",
       "231           436   \n",
       "238           437   \n",
       "240           437   \n",
       "259           441   \n",
       "...           ...   \n",
       "5988           40   \n",
       "5992           40   \n",
       "5993           41   \n",
       "5994           41   \n",
       "5998           41   \n",
       "6014           45   \n",
       "6015           45   \n",
       "6036           26   \n",
       "6047           28   \n",
       "6066           33   \n",
       "6067           33   \n",
       "6091           41   \n",
       "6093           42   \n",
       "6098           42   \n",
       "6114           47   \n",
       "6115           47   \n",
       "6136           25   \n",
       "6140           26   \n",
       "6180           34   \n",
       "6186           34   \n",
       "6199           39   \n",
       "6203           40   \n",
       "6252           49   \n",
       "6258           49   \n",
       "6274           28   \n",
       "\n",
       "                                                                                         line_content  \n",
       "62                                                   echo $workflow.manifest.version > v_pipeline.txt  \n",
       "72                                                   echo $workflow.nextflow.version > v_nextflow.txt  \n",
       "76                                                   echo $workflow.nextflow.version > v_nextflow.txt  \n",
       "117                           ThermoRawFileParser.sh -i=${raw_file} -f=2 -b=${raw_file.baseName}.mzML  \n",
       "119                           ThermoRawFileParser.sh -i=${raw_file} -f=2 -b=${raw_file.baseName}.mzML  \n",
       "129   IDFileConverter -in ${dda_id_file} -out ${id}_${sample}_peptide_ids.idXML -threads ${task.cpus}  \n",
       "132   IDFileConverter -in ${dda_id_file} -out ${id}_${sample}_peptide_ids.idXML -threads ${task.cpus}  \n",
       "135                                                                                easypqp convert \\\\  \n",
       "143                                                                        --unimod ${unimod_file} \\\\  \n",
       "150                                                                         --pepxml ${idxml_file} \\\\  \n",
       "159                                                                        --spectra ${dda_mzml_file}  \n",
       "174                                                  --out ${dda_mzml_file.baseName}_run_peaks.tsv \\\\  \n",
       "176                                                  --out ${dda_mzml_file.baseName}_run_peaks.tsv \\\\  \n",
       "186                                                --rt_psm_fdr_threshold ${params.library_rt_fdr} \\\\  \n",
       "197                                                               ${dda_mzml_file.baseName}.psmpkl \\\\  \n",
       "198                                                               ${dda_mzml_file.baseName}.psmpkl \\\\  \n",
       "206                                                                 ${dda_mzml_file.baseName}.peakpkl  \n",
       "207                                                                 ${dda_mzml_file.baseName}.peakpkl  \n",
       "219                            mv ${dda_mzml_file.baseName}_run_peaks.tsv ${id}_${sample}_library.tsv  \n",
       "221                            mv ${dda_mzml_file.baseName}_run_peaks.tsv ${id}_${sample}_library.tsv  \n",
       "225                                                                          TargetedFileConverter \\\\  \n",
       "231                                                                             -in ${lib_file_na} \\\\  \n",
       "238                                                               -out ${lib_file_na.baseName}.tsv \\\\  \n",
       "240                                                               -out ${lib_file_na.baseName}.tsv \\\\  \n",
       "259                                                                -in ${lib_file_na.baseName}.tsv \\\\  \n",
       "...                                                                                               ...  \n",
       "5988                                                                              touch ${input}.crai  \n",
       "5992                                                                              touch ${input}.crai  \n",
       "5993                                                                               touch ${input}.csi  \n",
       "5994                                                                               touch ${input}.csi  \n",
       "5998                                                                               touch ${input}.csi  \n",
       "6014           samtools: \\$(echo \\$(samtools --version 2>&1) | sed 's/^.*samtools //; s/Using.*\\$//')  \n",
       "6015           samtools: \\$(echo \\$(samtools --version 2>&1) | sed 's/^.*samtools //; s/Using.*\\$//')  \n",
       "6036                                                                                      samtools \\\\  \n",
       "6047                                                                                        $fasta \\\\  \n",
       "6066           samtools: \\$(echo \\$(samtools --version 2>&1) | sed 's/^.*samtools //; s/Using.*\\$//')  \n",
       "6067           samtools: \\$(echo \\$(samtools --version 2>&1) | sed 's/^.*samtools //; s/Using.*\\$//')  \n",
       "6091                                                                                      ${fastacmd}  \n",
       "6093                                                                               touch ${fasta}.fai  \n",
       "6098                                                                               touch ${fasta}.fai  \n",
       "6114           samtools: \\$(echo \\$(samtools --version 2>&1) | sed 's/^.*samtools //; s/Using.*\\$//')  \n",
       "6115           samtools: \\$(echo \\$(samtools --version 2>&1) | sed 's/^.*samtools //; s/Using.*\\$//')  \n",
       "6136                                                                                        seqkit \\\\  \n",
       "6140                                                                                        split2 \\\\  \n",
       "6180                             seqkit: \\$(echo \\$(seqkit 2>&1) | sed 's/^.*Version: //; s/ .*\\$//')  \n",
       "6186                             seqkit: \\$(echo \\$(seqkit 2>&1) | sed 's/^.*Version: //; s/ .*\\$//')  \n",
       "6199                                                                                        seqkit \\\\  \n",
       "6203                                                                                        split2 \\\\  \n",
       "6252                             seqkit: \\$(echo \\$(seqkit 2>&1) | sed 's/^.*Version: //; s/ .*\\$//')  \n",
       "6258                             seqkit: \\$(echo \\$(seqkit 2>&1) | sed 's/^.*Version: //; s/ .*\\$//')  \n",
       "6274                                            INDEX=`find -L ./ -name \"*.amb\" | sed 's/\\\\.amb\\$//'`  \n",
       "\n",
       "[600 rows x 6 columns]"
      ]
     },
     "execution_count": 187,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "operator_script = operator_script[operator_script['operator'].str.len() >= 4]\n",
    "#pd.set_option('display.max_colwidth', None)\n",
    "#pd.set_option('display.max_rows', None)\n",
    "print(len(operator_script))\n",
    "operator_script.head(600)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 188,
   "id": "efce58c5",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\Admin\\AppData\\Local\\Temp\\ipykernel_12680\\2951588785.py:1: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  operator_script.dropna(inplace=True)\n"
     ]
    }
   ],
   "source": [
    "operator_script.dropna(inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 189,
   "id": "ae8e3944",
   "metadata": {},
   "outputs": [],
   "source": [
    "operator_script_total = operator_script"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 190,
   "id": "b12243e2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pipeline</th>\n",
       "      <th>process_name</th>\n",
       "      <th>operator</th>\n",
       "      <th>file_path</th>\n",
       "      <th>line_number</th>\n",
       "      <th>line_content</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>62</th>\n",
       "      <td>diaproteomics</td>\n",
       "      <td>get_software_versions</td>\n",
       "      <td>echo</td>\n",
       "      <td>./git_repos/diaproteomics/main.nf</td>\n",
       "      <td>335</td>\n",
       "      <td>echo $workflow.manifest.version &gt; v_pipeline.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>72</th>\n",
       "      <td>diaproteomics</td>\n",
       "      <td>get_software_versions</td>\n",
       "      <td>echo</td>\n",
       "      <td>./git_repos/diaproteomics/main.nf</td>\n",
       "      <td>336</td>\n",
       "      <td>echo $workflow.nextflow.version &gt; v_nextflow.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>76</th>\n",
       "      <td>diaproteomics</td>\n",
       "      <td>get_software_versions</td>\n",
       "      <td>nextflow</td>\n",
       "      <td>./git_repos/diaproteomics/main.nf</td>\n",
       "      <td>336</td>\n",
       "      <td>echo $workflow.nextflow.version &gt; v_nextflow.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>117</th>\n",
       "      <td>diaproteomics</td>\n",
       "      <td>dda_raw_file_conversion</td>\n",
       "      <td>Name</td>\n",
       "      <td>./git_repos/diaproteomics/main.nf</td>\n",
       "      <td>359</td>\n",
       "      <td>ThermoRawFileParser.sh -i=${raw_file} -f=2 -b=${raw_file.baseName}.mzML</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>119</th>\n",
       "      <td>diaproteomics</td>\n",
       "      <td>dda_raw_file_conversion</td>\n",
       "      <td>file</td>\n",
       "      <td>./git_repos/diaproteomics/main.nf</td>\n",
       "      <td>359</td>\n",
       "      <td>ThermoRawFileParser.sh -i=${raw_file} -f=2 -b=${raw_file.baseName}.mzML</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>129</th>\n",
       "      <td>diaproteomics</td>\n",
       "      <td>dda_id_format_conversion</td>\n",
       "      <td>FileConverter</td>\n",
       "      <td>./git_repos/diaproteomics/main.nf</td>\n",
       "      <td>380</td>\n",
       "      <td>IDFileConverter -in ${dda_id_file} -out ${id}_${sample}_peptide_ids.idXML -threads ${task.cpus}</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>132</th>\n",
       "      <td>diaproteomics</td>\n",
       "      <td>dda_id_format_conversion</td>\n",
       "      <td>file</td>\n",
       "      <td>./git_repos/diaproteomics/main.nf</td>\n",
       "      <td>380</td>\n",
       "      <td>IDFileConverter -in ${dda_id_file} -out ${id}_${sample}_peptide_ids.idXML -threads ${task.cpus}</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>135</th>\n",
       "      <td>diaproteomics</td>\n",
       "      <td>dda_library_generation</td>\n",
       "      <td>convert</td>\n",
       "      <td>./git_repos/diaproteomics/main.nf</td>\n",
       "      <td>402</td>\n",
       "      <td>easypqp convert \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>143</th>\n",
       "      <td>diaproteomics</td>\n",
       "      <td>dda_library_generation</td>\n",
       "      <td>file</td>\n",
       "      <td>./git_repos/diaproteomics/main.nf</td>\n",
       "      <td>403</td>\n",
       "      <td>--unimod ${unimod_file} \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>150</th>\n",
       "      <td>diaproteomics</td>\n",
       "      <td>dda_library_generation</td>\n",
       "      <td>file</td>\n",
       "      <td>./git_repos/diaproteomics/main.nf</td>\n",
       "      <td>404</td>\n",
       "      <td>--pepxml ${idxml_file} \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>159</th>\n",
       "      <td>diaproteomics</td>\n",
       "      <td>dda_library_generation</td>\n",
       "      <td>file</td>\n",
       "      <td>./git_repos/diaproteomics/main.nf</td>\n",
       "      <td>405</td>\n",
       "      <td>--spectra ${dda_mzml_file}</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>174</th>\n",
       "      <td>diaproteomics</td>\n",
       "      <td>dda_library_generation</td>\n",
       "      <td>Name</td>\n",
       "      <td>./git_repos/diaproteomics/main.nf</td>\n",
       "      <td>408</td>\n",
       "      <td>--out ${dda_mzml_file.baseName}_run_peaks.tsv \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>176</th>\n",
       "      <td>diaproteomics</td>\n",
       "      <td>dda_library_generation</td>\n",
       "      <td>file</td>\n",
       "      <td>./git_repos/diaproteomics/main.nf</td>\n",
       "      <td>408</td>\n",
       "      <td>--out ${dda_mzml_file.baseName}_run_peaks.tsv \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>186</th>\n",
       "      <td>diaproteomics</td>\n",
       "      <td>dda_library_generation</td>\n",
       "      <td>params</td>\n",
       "      <td>./git_repos/diaproteomics/main.nf</td>\n",
       "      <td>409</td>\n",
       "      <td>--rt_psm_fdr_threshold ${params.library_rt_fdr} \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>197</th>\n",
       "      <td>diaproteomics</td>\n",
       "      <td>dda_library_generation</td>\n",
       "      <td>Name</td>\n",
       "      <td>./git_repos/diaproteomics/main.nf</td>\n",
       "      <td>411</td>\n",
       "      <td>${dda_mzml_file.baseName}.psmpkl \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>198</th>\n",
       "      <td>diaproteomics</td>\n",
       "      <td>dda_library_generation</td>\n",
       "      <td>file</td>\n",
       "      <td>./git_repos/diaproteomics/main.nf</td>\n",
       "      <td>411</td>\n",
       "      <td>${dda_mzml_file.baseName}.psmpkl \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>206</th>\n",
       "      <td>diaproteomics</td>\n",
       "      <td>dda_library_generation</td>\n",
       "      <td>Name</td>\n",
       "      <td>./git_repos/diaproteomics/main.nf</td>\n",
       "      <td>412</td>\n",
       "      <td>${dda_mzml_file.baseName}.peakpkl</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>207</th>\n",
       "      <td>diaproteomics</td>\n",
       "      <td>dda_library_generation</td>\n",
       "      <td>file</td>\n",
       "      <td>./git_repos/diaproteomics/main.nf</td>\n",
       "      <td>412</td>\n",
       "      <td>${dda_mzml_file.baseName}.peakpkl</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>219</th>\n",
       "      <td>diaproteomics</td>\n",
       "      <td>dda_library_generation</td>\n",
       "      <td>Name</td>\n",
       "      <td>./git_repos/diaproteomics/main.nf</td>\n",
       "      <td>414</td>\n",
       "      <td>mv ${dda_mzml_file.baseName}_run_peaks.tsv ${id}_${sample}_library.tsv</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>221</th>\n",
       "      <td>diaproteomics</td>\n",
       "      <td>dda_library_generation</td>\n",
       "      <td>file</td>\n",
       "      <td>./git_repos/diaproteomics/main.nf</td>\n",
       "      <td>414</td>\n",
       "      <td>mv ${dda_mzml_file.baseName}_run_peaks.tsv ${id}_${sample}_library.tsv</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>225</th>\n",
       "      <td>diaproteomics</td>\n",
       "      <td>assay_generation</td>\n",
       "      <td>FileConverter</td>\n",
       "      <td>./git_repos/diaproteomics/main.nf</td>\n",
       "      <td>435</td>\n",
       "      <td>TargetedFileConverter \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>231</th>\n",
       "      <td>diaproteomics</td>\n",
       "      <td>assay_generation</td>\n",
       "      <td>file</td>\n",
       "      <td>./git_repos/diaproteomics/main.nf</td>\n",
       "      <td>436</td>\n",
       "      <td>-in ${lib_file_na} \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>238</th>\n",
       "      <td>diaproteomics</td>\n",
       "      <td>assay_generation</td>\n",
       "      <td>Name</td>\n",
       "      <td>./git_repos/diaproteomics/main.nf</td>\n",
       "      <td>437</td>\n",
       "      <td>-out ${lib_file_na.baseName}.tsv \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>240</th>\n",
       "      <td>diaproteomics</td>\n",
       "      <td>assay_generation</td>\n",
       "      <td>file</td>\n",
       "      <td>./git_repos/diaproteomics/main.nf</td>\n",
       "      <td>437</td>\n",
       "      <td>-out ${lib_file_na.baseName}.tsv \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>259</th>\n",
       "      <td>diaproteomics</td>\n",
       "      <td>assay_generation</td>\n",
       "      <td>Name</td>\n",
       "      <td>./git_repos/diaproteomics/main.nf</td>\n",
       "      <td>441</td>\n",
       "      <td>-in ${lib_file_na.baseName}.tsv \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>345117</th>\n",
       "      <td>hic</td>\n",
       "      <td>FASTQC</td>\n",
       "      <td>printf</td>\n",
       "      <td>./git_repos/hic/modules/nf-core/fastqc/main.nf</td>\n",
       "      <td>29</td>\n",
       "      <td>printf \"%s %s\\\\n\" $rename_to | while read old_name new_name; do</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>345126</th>\n",
       "      <td>hic</td>\n",
       "      <td>FASTQC</td>\n",
       "      <td>done</td>\n",
       "      <td>./git_repos/hic/modules/nf-core/fastqc/main.nf</td>\n",
       "      <td>31</td>\n",
       "      <td>done</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>345132</th>\n",
       "      <td>hic</td>\n",
       "      <td>FASTQC</td>\n",
       "      <td>fastqc</td>\n",
       "      <td>./git_repos/hic/modules/nf-core/fastqc/main.nf</td>\n",
       "      <td>32</td>\n",
       "      <td>fastqc $args --threads $task.cpus $renamed_files</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>345137</th>\n",
       "      <td>hic</td>\n",
       "      <td>FASTQC</td>\n",
       "      <td>rename</td>\n",
       "      <td>./git_repos/hic/modules/nf-core/fastqc/main.nf</td>\n",
       "      <td>32</td>\n",
       "      <td>fastqc $args --threads $task.cpus $renamed_files</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>345139</th>\n",
       "      <td>hic</td>\n",
       "      <td>FASTQC</td>\n",
       "      <td>fastq</td>\n",
       "      <td>./git_repos/hic/modules/nf-core/fastqc/main.nf</td>\n",
       "      <td>32</td>\n",
       "      <td>fastqc $args --threads $task.cpus $renamed_files</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>345145</th>\n",
       "      <td>hic</td>\n",
       "      <td>FASTQC</td>\n",
       "      <td>file</td>\n",
       "      <td>./git_repos/hic/modules/nf-core/fastqc/main.nf</td>\n",
       "      <td>32</td>\n",
       "      <td>fastqc $args --threads $task.cpus $renamed_files</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>345162</th>\n",
       "      <td>hic</td>\n",
       "      <td>FASTQC</td>\n",
       "      <td>fastqc</td>\n",
       "      <td>./git_repos/hic/modules/nf-core/fastqc/main.nf</td>\n",
       "      <td>36</td>\n",
       "      <td>fastqc: \\$( fastqc --version | sed -e \"s/FastQC v//g\" )</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>345171</th>\n",
       "      <td>hic</td>\n",
       "      <td>FASTQC</td>\n",
       "      <td>fastq</td>\n",
       "      <td>./git_repos/hic/modules/nf-core/fastqc/main.nf</td>\n",
       "      <td>36</td>\n",
       "      <td>fastqc: \\$( fastqc --version | sed -e \"s/FastQC v//g\" )</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>345180</th>\n",
       "      <td>hic</td>\n",
       "      <td>FASTQC</td>\n",
       "      <td>touch</td>\n",
       "      <td>./git_repos/hic/modules/nf-core/fastqc/main.nf</td>\n",
       "      <td>43</td>\n",
       "      <td>touch ${prefix}.html</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>345188</th>\n",
       "      <td>hic</td>\n",
       "      <td>FASTQC</td>\n",
       "      <td>html</td>\n",
       "      <td>./git_repos/hic/modules/nf-core/fastqc/main.nf</td>\n",
       "      <td>43</td>\n",
       "      <td>touch ${prefix}.html</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>345191</th>\n",
       "      <td>hic</td>\n",
       "      <td>FASTQC</td>\n",
       "      <td>touch</td>\n",
       "      <td>./git_repos/hic/modules/nf-core/fastqc/main.nf</td>\n",
       "      <td>44</td>\n",
       "      <td>touch ${prefix}.zip</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>345217</th>\n",
       "      <td>hic</td>\n",
       "      <td>FASTQC</td>\n",
       "      <td>fastqc</td>\n",
       "      <td>./git_repos/hic/modules/nf-core/fastqc/main.nf</td>\n",
       "      <td>48</td>\n",
       "      <td>fastqc: \\$( fastqc --version | sed -e \"s/FastQC v//g\" )</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>345226</th>\n",
       "      <td>hic</td>\n",
       "      <td>FASTQC</td>\n",
       "      <td>fastq</td>\n",
       "      <td>./git_repos/hic/modules/nf-core/fastqc/main.nf</td>\n",
       "      <td>48</td>\n",
       "      <td>fastqc: \\$( fastqc --version | sed -e \"s/FastQC v//g\" )</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>345246</th>\n",
       "      <td>hic</td>\n",
       "      <td>COOLER_BALANCE</td>\n",
       "      <td>cooler</td>\n",
       "      <td>./git_repos/hic/modules/nf-core/cooler/balance/main.nf</td>\n",
       "      <td>29</td>\n",
       "      <td>cooler balance \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>345287</th>\n",
       "      <td>hic</td>\n",
       "      <td>COOLER_BALANCE</td>\n",
       "      <td>cooler</td>\n",
       "      <td>./git_repos/hic/modules/nf-core/cooler/balance/main.nf</td>\n",
       "      <td>36</td>\n",
       "      <td>cooler: \\$(cooler --version 2&gt;&amp;1 | sed 's/cooler, version //')</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>345297</th>\n",
       "      <td>hic</td>\n",
       "      <td>COOLER_ZOOMIFY</td>\n",
       "      <td>cooler</td>\n",
       "      <td>./git_repos/hic/modules/nf-core/cooler/zoomify/main.nf</td>\n",
       "      <td>24</td>\n",
       "      <td>cooler zoomify \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>345338</th>\n",
       "      <td>hic</td>\n",
       "      <td>COOLER_ZOOMIFY</td>\n",
       "      <td>cooler</td>\n",
       "      <td>./git_repos/hic/modules/nf-core/cooler/zoomify/main.nf</td>\n",
       "      <td>32</td>\n",
       "      <td>cooler: \\$(cooler --version 2&gt;&amp;1 | sed 's/cooler, version //')</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>345347</th>\n",
       "      <td>hic</td>\n",
       "      <td>COOLER_DUMP</td>\n",
       "      <td>cooler</td>\n",
       "      <td>./git_repos/hic/modules/nf-core/cooler/dump/main.nf</td>\n",
       "      <td>25</td>\n",
       "      <td>cooler dump \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>345348</th>\n",
       "      <td>hic</td>\n",
       "      <td>COOLER_DUMP</td>\n",
       "      <td>dump</td>\n",
       "      <td>./git_repos/hic/modules/nf-core/cooler/dump/main.nf</td>\n",
       "      <td>25</td>\n",
       "      <td>cooler dump \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>345389</th>\n",
       "      <td>hic</td>\n",
       "      <td>COOLER_DUMP</td>\n",
       "      <td>cooler</td>\n",
       "      <td>./git_repos/hic/modules/nf-core/cooler/dump/main.nf</td>\n",
       "      <td>32</td>\n",
       "      <td>cooler: \\$(cooler --version 2&gt;&amp;1 | sed 's/cooler, version //')</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>345398</th>\n",
       "      <td>hic</td>\n",
       "      <td>COOLER_CLOAD</td>\n",
       "      <td>cooler</td>\n",
       "      <td>./git_repos/hic/modules/nf-core/cooler/cload/main.nf</td>\n",
       "      <td>27</td>\n",
       "      <td>cooler cload \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>345444</th>\n",
       "      <td>hic</td>\n",
       "      <td>COOLER_CLOAD</td>\n",
       "      <td>cooler</td>\n",
       "      <td>./git_repos/hic/modules/nf-core/cooler/cload/main.nf</td>\n",
       "      <td>36</td>\n",
       "      <td>cooler: \\$(cooler --version 2&gt;&amp;1 | sed 's/cooler, version //')</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>345453</th>\n",
       "      <td>hic</td>\n",
       "      <td>COOLER_MAKEBINS</td>\n",
       "      <td>make</td>\n",
       "      <td>./git_repos/hic/modules/nf-core/cooler/makebins/main.nf</td>\n",
       "      <td>24</td>\n",
       "      <td>cooler makebins \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>345454</th>\n",
       "      <td>hic</td>\n",
       "      <td>COOLER_MAKEBINS</td>\n",
       "      <td>cooler</td>\n",
       "      <td>./git_repos/hic/modules/nf-core/cooler/makebins/main.nf</td>\n",
       "      <td>24</td>\n",
       "      <td>cooler makebins \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>345495</th>\n",
       "      <td>hic</td>\n",
       "      <td>COOLER_MAKEBINS</td>\n",
       "      <td>cooler</td>\n",
       "      <td>./git_repos/hic/modules/nf-core/cooler/makebins/main.nf</td>\n",
       "      <td>31</td>\n",
       "      <td>cooler: \\$(cooler --version 2&gt;&amp;1 | sed 's/cooler, version //')</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>29532 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             pipeline              process_name       operator  \\\n",
       "62      diaproteomics     get_software_versions           echo   \n",
       "72      diaproteomics     get_software_versions           echo   \n",
       "76      diaproteomics     get_software_versions       nextflow   \n",
       "117     diaproteomics   dda_raw_file_conversion           Name   \n",
       "119     diaproteomics   dda_raw_file_conversion           file   \n",
       "129     diaproteomics  dda_id_format_conversion  FileConverter   \n",
       "132     diaproteomics  dda_id_format_conversion           file   \n",
       "135     diaproteomics    dda_library_generation        convert   \n",
       "143     diaproteomics    dda_library_generation           file   \n",
       "150     diaproteomics    dda_library_generation           file   \n",
       "159     diaproteomics    dda_library_generation           file   \n",
       "174     diaproteomics    dda_library_generation           Name   \n",
       "176     diaproteomics    dda_library_generation           file   \n",
       "186     diaproteomics    dda_library_generation         params   \n",
       "197     diaproteomics    dda_library_generation           Name   \n",
       "198     diaproteomics    dda_library_generation           file   \n",
       "206     diaproteomics    dda_library_generation           Name   \n",
       "207     diaproteomics    dda_library_generation           file   \n",
       "219     diaproteomics    dda_library_generation           Name   \n",
       "221     diaproteomics    dda_library_generation           file   \n",
       "225     diaproteomics          assay_generation  FileConverter   \n",
       "231     diaproteomics          assay_generation           file   \n",
       "238     diaproteomics          assay_generation           Name   \n",
       "240     diaproteomics          assay_generation           file   \n",
       "259     diaproteomics          assay_generation           Name   \n",
       "...               ...                       ...            ...   \n",
       "345117            hic                    FASTQC         printf   \n",
       "345126            hic                    FASTQC           done   \n",
       "345132            hic                    FASTQC         fastqc   \n",
       "345137            hic                    FASTQC         rename   \n",
       "345139            hic                    FASTQC          fastq   \n",
       "345145            hic                    FASTQC           file   \n",
       "345162            hic                    FASTQC         fastqc   \n",
       "345171            hic                    FASTQC          fastq   \n",
       "345180            hic                    FASTQC          touch   \n",
       "345188            hic                    FASTQC           html   \n",
       "345191            hic                    FASTQC          touch   \n",
       "345217            hic                    FASTQC         fastqc   \n",
       "345226            hic                    FASTQC          fastq   \n",
       "345246            hic            COOLER_BALANCE         cooler   \n",
       "345287            hic            COOLER_BALANCE         cooler   \n",
       "345297            hic            COOLER_ZOOMIFY         cooler   \n",
       "345338            hic            COOLER_ZOOMIFY         cooler   \n",
       "345347            hic               COOLER_DUMP         cooler   \n",
       "345348            hic               COOLER_DUMP           dump   \n",
       "345389            hic               COOLER_DUMP         cooler   \n",
       "345398            hic              COOLER_CLOAD         cooler   \n",
       "345444            hic              COOLER_CLOAD         cooler   \n",
       "345453            hic           COOLER_MAKEBINS           make   \n",
       "345454            hic           COOLER_MAKEBINS         cooler   \n",
       "345495            hic           COOLER_MAKEBINS         cooler   \n",
       "\n",
       "                                                      file_path  line_number  \\\n",
       "62                            ./git_repos/diaproteomics/main.nf          335   \n",
       "72                            ./git_repos/diaproteomics/main.nf          336   \n",
       "76                            ./git_repos/diaproteomics/main.nf          336   \n",
       "117                           ./git_repos/diaproteomics/main.nf          359   \n",
       "119                           ./git_repos/diaproteomics/main.nf          359   \n",
       "129                           ./git_repos/diaproteomics/main.nf          380   \n",
       "132                           ./git_repos/diaproteomics/main.nf          380   \n",
       "135                           ./git_repos/diaproteomics/main.nf          402   \n",
       "143                           ./git_repos/diaproteomics/main.nf          403   \n",
       "150                           ./git_repos/diaproteomics/main.nf          404   \n",
       "159                           ./git_repos/diaproteomics/main.nf          405   \n",
       "174                           ./git_repos/diaproteomics/main.nf          408   \n",
       "176                           ./git_repos/diaproteomics/main.nf          408   \n",
       "186                           ./git_repos/diaproteomics/main.nf          409   \n",
       "197                           ./git_repos/diaproteomics/main.nf          411   \n",
       "198                           ./git_repos/diaproteomics/main.nf          411   \n",
       "206                           ./git_repos/diaproteomics/main.nf          412   \n",
       "207                           ./git_repos/diaproteomics/main.nf          412   \n",
       "219                           ./git_repos/diaproteomics/main.nf          414   \n",
       "221                           ./git_repos/diaproteomics/main.nf          414   \n",
       "225                           ./git_repos/diaproteomics/main.nf          435   \n",
       "231                           ./git_repos/diaproteomics/main.nf          436   \n",
       "238                           ./git_repos/diaproteomics/main.nf          437   \n",
       "240                           ./git_repos/diaproteomics/main.nf          437   \n",
       "259                           ./git_repos/diaproteomics/main.nf          441   \n",
       "...                                                         ...          ...   \n",
       "345117           ./git_repos/hic/modules/nf-core/fastqc/main.nf           29   \n",
       "345126           ./git_repos/hic/modules/nf-core/fastqc/main.nf           31   \n",
       "345132           ./git_repos/hic/modules/nf-core/fastqc/main.nf           32   \n",
       "345137           ./git_repos/hic/modules/nf-core/fastqc/main.nf           32   \n",
       "345139           ./git_repos/hic/modules/nf-core/fastqc/main.nf           32   \n",
       "345145           ./git_repos/hic/modules/nf-core/fastqc/main.nf           32   \n",
       "345162           ./git_repos/hic/modules/nf-core/fastqc/main.nf           36   \n",
       "345171           ./git_repos/hic/modules/nf-core/fastqc/main.nf           36   \n",
       "345180           ./git_repos/hic/modules/nf-core/fastqc/main.nf           43   \n",
       "345188           ./git_repos/hic/modules/nf-core/fastqc/main.nf           43   \n",
       "345191           ./git_repos/hic/modules/nf-core/fastqc/main.nf           44   \n",
       "345217           ./git_repos/hic/modules/nf-core/fastqc/main.nf           48   \n",
       "345226           ./git_repos/hic/modules/nf-core/fastqc/main.nf           48   \n",
       "345246   ./git_repos/hic/modules/nf-core/cooler/balance/main.nf           29   \n",
       "345287   ./git_repos/hic/modules/nf-core/cooler/balance/main.nf           36   \n",
       "345297   ./git_repos/hic/modules/nf-core/cooler/zoomify/main.nf           24   \n",
       "345338   ./git_repos/hic/modules/nf-core/cooler/zoomify/main.nf           32   \n",
       "345347      ./git_repos/hic/modules/nf-core/cooler/dump/main.nf           25   \n",
       "345348      ./git_repos/hic/modules/nf-core/cooler/dump/main.nf           25   \n",
       "345389      ./git_repos/hic/modules/nf-core/cooler/dump/main.nf           32   \n",
       "345398     ./git_repos/hic/modules/nf-core/cooler/cload/main.nf           27   \n",
       "345444     ./git_repos/hic/modules/nf-core/cooler/cload/main.nf           36   \n",
       "345453  ./git_repos/hic/modules/nf-core/cooler/makebins/main.nf           24   \n",
       "345454  ./git_repos/hic/modules/nf-core/cooler/makebins/main.nf           24   \n",
       "345495  ./git_repos/hic/modules/nf-core/cooler/makebins/main.nf           31   \n",
       "\n",
       "                                                                                           line_content  \n",
       "62                                                     echo $workflow.manifest.version > v_pipeline.txt  \n",
       "72                                                     echo $workflow.nextflow.version > v_nextflow.txt  \n",
       "76                                                     echo $workflow.nextflow.version > v_nextflow.txt  \n",
       "117                             ThermoRawFileParser.sh -i=${raw_file} -f=2 -b=${raw_file.baseName}.mzML  \n",
       "119                             ThermoRawFileParser.sh -i=${raw_file} -f=2 -b=${raw_file.baseName}.mzML  \n",
       "129     IDFileConverter -in ${dda_id_file} -out ${id}_${sample}_peptide_ids.idXML -threads ${task.cpus}  \n",
       "132     IDFileConverter -in ${dda_id_file} -out ${id}_${sample}_peptide_ids.idXML -threads ${task.cpus}  \n",
       "135                                                                                  easypqp convert \\\\  \n",
       "143                                                                          --unimod ${unimod_file} \\\\  \n",
       "150                                                                           --pepxml ${idxml_file} \\\\  \n",
       "159                                                                          --spectra ${dda_mzml_file}  \n",
       "174                                                    --out ${dda_mzml_file.baseName}_run_peaks.tsv \\\\  \n",
       "176                                                    --out ${dda_mzml_file.baseName}_run_peaks.tsv \\\\  \n",
       "186                                                  --rt_psm_fdr_threshold ${params.library_rt_fdr} \\\\  \n",
       "197                                                                 ${dda_mzml_file.baseName}.psmpkl \\\\  \n",
       "198                                                                 ${dda_mzml_file.baseName}.psmpkl \\\\  \n",
       "206                                                                   ${dda_mzml_file.baseName}.peakpkl  \n",
       "207                                                                   ${dda_mzml_file.baseName}.peakpkl  \n",
       "219                              mv ${dda_mzml_file.baseName}_run_peaks.tsv ${id}_${sample}_library.tsv  \n",
       "221                              mv ${dda_mzml_file.baseName}_run_peaks.tsv ${id}_${sample}_library.tsv  \n",
       "225                                                                            TargetedFileConverter \\\\  \n",
       "231                                                                               -in ${lib_file_na} \\\\  \n",
       "238                                                                 -out ${lib_file_na.baseName}.tsv \\\\  \n",
       "240                                                                 -out ${lib_file_na.baseName}.tsv \\\\  \n",
       "259                                                                  -in ${lib_file_na.baseName}.tsv \\\\  \n",
       "...                                                                                                 ...  \n",
       "345117                                  printf \"%s %s\\\\n\" $rename_to | while read old_name new_name; do  \n",
       "345126                                                                                             done  \n",
       "345132                                                 fastqc $args --threads $task.cpus $renamed_files  \n",
       "345137                                                 fastqc $args --threads $task.cpus $renamed_files  \n",
       "345139                                                 fastqc $args --threads $task.cpus $renamed_files  \n",
       "345145                                                 fastqc $args --threads $task.cpus $renamed_files  \n",
       "345162                                          fastqc: \\$( fastqc --version | sed -e \"s/FastQC v//g\" )  \n",
       "345171                                          fastqc: \\$( fastqc --version | sed -e \"s/FastQC v//g\" )  \n",
       "345180                                                                             touch ${prefix}.html  \n",
       "345188                                                                             touch ${prefix}.html  \n",
       "345191                                                                              touch ${prefix}.zip  \n",
       "345217                                          fastqc: \\$( fastqc --version | sed -e \"s/FastQC v//g\" )  \n",
       "345226                                          fastqc: \\$( fastqc --version | sed -e \"s/FastQC v//g\" )  \n",
       "345246                                                                                cooler balance \\\\  \n",
       "345287                                   cooler: \\$(cooler --version 2>&1 | sed 's/cooler, version //')  \n",
       "345297                                                                                cooler zoomify \\\\  \n",
       "345338                                   cooler: \\$(cooler --version 2>&1 | sed 's/cooler, version //')  \n",
       "345347                                                                                   cooler dump \\\\  \n",
       "345348                                                                                   cooler dump \\\\  \n",
       "345389                                   cooler: \\$(cooler --version 2>&1 | sed 's/cooler, version //')  \n",
       "345398                                                                                  cooler cload \\\\  \n",
       "345444                                   cooler: \\$(cooler --version 2>&1 | sed 's/cooler, version //')  \n",
       "345453                                                                               cooler makebins \\\\  \n",
       "345454                                                                               cooler makebins \\\\  \n",
       "345495                                   cooler: \\$(cooler --version 2>&1 | sed 's/cooler, version //')  \n",
       "\n",
       "[29532 rows x 6 columns]"
      ]
     },
     "execution_count": 190,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "operator_script = operator_script[~operator_script['file_path'].str.endswith('.config')]\n",
    "operator_script"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 191,
   "id": "b282a47e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pipeline</th>\n",
       "      <th>process_name</th>\n",
       "      <th>operator</th>\n",
       "      <th>file_path</th>\n",
       "      <th>line_number</th>\n",
       "      <th>line_content</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>62</th>\n",
       "      <td>diaproteomics</td>\n",
       "      <td>get_software_versions</td>\n",
       "      <td>echo</td>\n",
       "      <td>./git_repos/diaproteomics/main.nf</td>\n",
       "      <td>335</td>\n",
       "      <td>echo $workflow.manifest.version &gt; v_pipeline.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>72</th>\n",
       "      <td>diaproteomics</td>\n",
       "      <td>get_software_versions</td>\n",
       "      <td>echo</td>\n",
       "      <td>./git_repos/diaproteomics/main.nf</td>\n",
       "      <td>336</td>\n",
       "      <td>echo $workflow.nextflow.version &gt; v_nextflow.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>76</th>\n",
       "      <td>diaproteomics</td>\n",
       "      <td>get_software_versions</td>\n",
       "      <td>nextflow</td>\n",
       "      <td>./git_repos/diaproteomics/main.nf</td>\n",
       "      <td>336</td>\n",
       "      <td>echo $workflow.nextflow.version &gt; v_nextflow.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>117</th>\n",
       "      <td>diaproteomics</td>\n",
       "      <td>dda_raw_file_conversion</td>\n",
       "      <td>Name</td>\n",
       "      <td>./git_repos/diaproteomics/main.nf</td>\n",
       "      <td>359</td>\n",
       "      <td>ThermoRawFileParser.sh -i=${raw_file} -f=2 -b=${raw_file.baseName}.mzML</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>119</th>\n",
       "      <td>diaproteomics</td>\n",
       "      <td>dda_raw_file_conversion</td>\n",
       "      <td>file</td>\n",
       "      <td>./git_repos/diaproteomics/main.nf</td>\n",
       "      <td>359</td>\n",
       "      <td>ThermoRawFileParser.sh -i=${raw_file} -f=2 -b=${raw_file.baseName}.mzML</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>129</th>\n",
       "      <td>diaproteomics</td>\n",
       "      <td>dda_id_format_conversion</td>\n",
       "      <td>FileConverter</td>\n",
       "      <td>./git_repos/diaproteomics/main.nf</td>\n",
       "      <td>380</td>\n",
       "      <td>IDFileConverter -in ${dda_id_file} -out ${id}_${sample}_peptide_ids.idXML -threads ${task.cpus}</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>132</th>\n",
       "      <td>diaproteomics</td>\n",
       "      <td>dda_id_format_conversion</td>\n",
       "      <td>file</td>\n",
       "      <td>./git_repos/diaproteomics/main.nf</td>\n",
       "      <td>380</td>\n",
       "      <td>IDFileConverter -in ${dda_id_file} -out ${id}_${sample}_peptide_ids.idXML -threads ${task.cpus}</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>135</th>\n",
       "      <td>diaproteomics</td>\n",
       "      <td>dda_library_generation</td>\n",
       "      <td>convert</td>\n",
       "      <td>./git_repos/diaproteomics/main.nf</td>\n",
       "      <td>402</td>\n",
       "      <td>easypqp convert \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>143</th>\n",
       "      <td>diaproteomics</td>\n",
       "      <td>dda_library_generation</td>\n",
       "      <td>file</td>\n",
       "      <td>./git_repos/diaproteomics/main.nf</td>\n",
       "      <td>403</td>\n",
       "      <td>--unimod ${unimod_file} \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>150</th>\n",
       "      <td>diaproteomics</td>\n",
       "      <td>dda_library_generation</td>\n",
       "      <td>file</td>\n",
       "      <td>./git_repos/diaproteomics/main.nf</td>\n",
       "      <td>404</td>\n",
       "      <td>--pepxml ${idxml_file} \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>159</th>\n",
       "      <td>diaproteomics</td>\n",
       "      <td>dda_library_generation</td>\n",
       "      <td>file</td>\n",
       "      <td>./git_repos/diaproteomics/main.nf</td>\n",
       "      <td>405</td>\n",
       "      <td>--spectra ${dda_mzml_file}</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>174</th>\n",
       "      <td>diaproteomics</td>\n",
       "      <td>dda_library_generation</td>\n",
       "      <td>Name</td>\n",
       "      <td>./git_repos/diaproteomics/main.nf</td>\n",
       "      <td>408</td>\n",
       "      <td>--out ${dda_mzml_file.baseName}_run_peaks.tsv \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>176</th>\n",
       "      <td>diaproteomics</td>\n",
       "      <td>dda_library_generation</td>\n",
       "      <td>file</td>\n",
       "      <td>./git_repos/diaproteomics/main.nf</td>\n",
       "      <td>408</td>\n",
       "      <td>--out ${dda_mzml_file.baseName}_run_peaks.tsv \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>186</th>\n",
       "      <td>diaproteomics</td>\n",
       "      <td>dda_library_generation</td>\n",
       "      <td>params</td>\n",
       "      <td>./git_repos/diaproteomics/main.nf</td>\n",
       "      <td>409</td>\n",
       "      <td>--rt_psm_fdr_threshold ${params.library_rt_fdr} \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>197</th>\n",
       "      <td>diaproteomics</td>\n",
       "      <td>dda_library_generation</td>\n",
       "      <td>Name</td>\n",
       "      <td>./git_repos/diaproteomics/main.nf</td>\n",
       "      <td>411</td>\n",
       "      <td>${dda_mzml_file.baseName}.psmpkl \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>198</th>\n",
       "      <td>diaproteomics</td>\n",
       "      <td>dda_library_generation</td>\n",
       "      <td>file</td>\n",
       "      <td>./git_repos/diaproteomics/main.nf</td>\n",
       "      <td>411</td>\n",
       "      <td>${dda_mzml_file.baseName}.psmpkl \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>206</th>\n",
       "      <td>diaproteomics</td>\n",
       "      <td>dda_library_generation</td>\n",
       "      <td>Name</td>\n",
       "      <td>./git_repos/diaproteomics/main.nf</td>\n",
       "      <td>412</td>\n",
       "      <td>${dda_mzml_file.baseName}.peakpkl</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>207</th>\n",
       "      <td>diaproteomics</td>\n",
       "      <td>dda_library_generation</td>\n",
       "      <td>file</td>\n",
       "      <td>./git_repos/diaproteomics/main.nf</td>\n",
       "      <td>412</td>\n",
       "      <td>${dda_mzml_file.baseName}.peakpkl</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>219</th>\n",
       "      <td>diaproteomics</td>\n",
       "      <td>dda_library_generation</td>\n",
       "      <td>Name</td>\n",
       "      <td>./git_repos/diaproteomics/main.nf</td>\n",
       "      <td>414</td>\n",
       "      <td>mv ${dda_mzml_file.baseName}_run_peaks.tsv ${id}_${sample}_library.tsv</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>221</th>\n",
       "      <td>diaproteomics</td>\n",
       "      <td>dda_library_generation</td>\n",
       "      <td>file</td>\n",
       "      <td>./git_repos/diaproteomics/main.nf</td>\n",
       "      <td>414</td>\n",
       "      <td>mv ${dda_mzml_file.baseName}_run_peaks.tsv ${id}_${sample}_library.tsv</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>225</th>\n",
       "      <td>diaproteomics</td>\n",
       "      <td>assay_generation</td>\n",
       "      <td>FileConverter</td>\n",
       "      <td>./git_repos/diaproteomics/main.nf</td>\n",
       "      <td>435</td>\n",
       "      <td>TargetedFileConverter \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>231</th>\n",
       "      <td>diaproteomics</td>\n",
       "      <td>assay_generation</td>\n",
       "      <td>file</td>\n",
       "      <td>./git_repos/diaproteomics/main.nf</td>\n",
       "      <td>436</td>\n",
       "      <td>-in ${lib_file_na} \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>238</th>\n",
       "      <td>diaproteomics</td>\n",
       "      <td>assay_generation</td>\n",
       "      <td>Name</td>\n",
       "      <td>./git_repos/diaproteomics/main.nf</td>\n",
       "      <td>437</td>\n",
       "      <td>-out ${lib_file_na.baseName}.tsv \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>240</th>\n",
       "      <td>diaproteomics</td>\n",
       "      <td>assay_generation</td>\n",
       "      <td>file</td>\n",
       "      <td>./git_repos/diaproteomics/main.nf</td>\n",
       "      <td>437</td>\n",
       "      <td>-out ${lib_file_na.baseName}.tsv \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>259</th>\n",
       "      <td>diaproteomics</td>\n",
       "      <td>assay_generation</td>\n",
       "      <td>Name</td>\n",
       "      <td>./git_repos/diaproteomics/main.nf</td>\n",
       "      <td>441</td>\n",
       "      <td>-in ${lib_file_na.baseName}.tsv \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>345117</th>\n",
       "      <td>hic</td>\n",
       "      <td>FASTQC</td>\n",
       "      <td>printf</td>\n",
       "      <td>./git_repos/hic/modules/nf-core/fastqc/main.nf</td>\n",
       "      <td>29</td>\n",
       "      <td>printf \"%s %s\\\\n\" $rename_to | while read old_name new_name; do</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>345126</th>\n",
       "      <td>hic</td>\n",
       "      <td>FASTQC</td>\n",
       "      <td>done</td>\n",
       "      <td>./git_repos/hic/modules/nf-core/fastqc/main.nf</td>\n",
       "      <td>31</td>\n",
       "      <td>done</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>345132</th>\n",
       "      <td>hic</td>\n",
       "      <td>FASTQC</td>\n",
       "      <td>fastqc</td>\n",
       "      <td>./git_repos/hic/modules/nf-core/fastqc/main.nf</td>\n",
       "      <td>32</td>\n",
       "      <td>fastqc $args --threads $task.cpus $renamed_files</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>345137</th>\n",
       "      <td>hic</td>\n",
       "      <td>FASTQC</td>\n",
       "      <td>rename</td>\n",
       "      <td>./git_repos/hic/modules/nf-core/fastqc/main.nf</td>\n",
       "      <td>32</td>\n",
       "      <td>fastqc $args --threads $task.cpus $renamed_files</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>345139</th>\n",
       "      <td>hic</td>\n",
       "      <td>FASTQC</td>\n",
       "      <td>fastq</td>\n",
       "      <td>./git_repos/hic/modules/nf-core/fastqc/main.nf</td>\n",
       "      <td>32</td>\n",
       "      <td>fastqc $args --threads $task.cpus $renamed_files</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>345145</th>\n",
       "      <td>hic</td>\n",
       "      <td>FASTQC</td>\n",
       "      <td>file</td>\n",
       "      <td>./git_repos/hic/modules/nf-core/fastqc/main.nf</td>\n",
       "      <td>32</td>\n",
       "      <td>fastqc $args --threads $task.cpus $renamed_files</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>345162</th>\n",
       "      <td>hic</td>\n",
       "      <td>FASTQC</td>\n",
       "      <td>fastqc</td>\n",
       "      <td>./git_repos/hic/modules/nf-core/fastqc/main.nf</td>\n",
       "      <td>36</td>\n",
       "      <td>fastqc: \\$( fastqc --version | sed -e \"s/FastQC v//g\" )</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>345171</th>\n",
       "      <td>hic</td>\n",
       "      <td>FASTQC</td>\n",
       "      <td>fastq</td>\n",
       "      <td>./git_repos/hic/modules/nf-core/fastqc/main.nf</td>\n",
       "      <td>36</td>\n",
       "      <td>fastqc: \\$( fastqc --version | sed -e \"s/FastQC v//g\" )</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>345180</th>\n",
       "      <td>hic</td>\n",
       "      <td>FASTQC</td>\n",
       "      <td>touch</td>\n",
       "      <td>./git_repos/hic/modules/nf-core/fastqc/main.nf</td>\n",
       "      <td>43</td>\n",
       "      <td>touch ${prefix}.html</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>345188</th>\n",
       "      <td>hic</td>\n",
       "      <td>FASTQC</td>\n",
       "      <td>html</td>\n",
       "      <td>./git_repos/hic/modules/nf-core/fastqc/main.nf</td>\n",
       "      <td>43</td>\n",
       "      <td>touch ${prefix}.html</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>345191</th>\n",
       "      <td>hic</td>\n",
       "      <td>FASTQC</td>\n",
       "      <td>touch</td>\n",
       "      <td>./git_repos/hic/modules/nf-core/fastqc/main.nf</td>\n",
       "      <td>44</td>\n",
       "      <td>touch ${prefix}.zip</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>345217</th>\n",
       "      <td>hic</td>\n",
       "      <td>FASTQC</td>\n",
       "      <td>fastqc</td>\n",
       "      <td>./git_repos/hic/modules/nf-core/fastqc/main.nf</td>\n",
       "      <td>48</td>\n",
       "      <td>fastqc: \\$( fastqc --version | sed -e \"s/FastQC v//g\" )</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>345226</th>\n",
       "      <td>hic</td>\n",
       "      <td>FASTQC</td>\n",
       "      <td>fastq</td>\n",
       "      <td>./git_repos/hic/modules/nf-core/fastqc/main.nf</td>\n",
       "      <td>48</td>\n",
       "      <td>fastqc: \\$( fastqc --version | sed -e \"s/FastQC v//g\" )</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>345246</th>\n",
       "      <td>hic</td>\n",
       "      <td>COOLER_BALANCE</td>\n",
       "      <td>cooler</td>\n",
       "      <td>./git_repos/hic/modules/nf-core/cooler/balance/main.nf</td>\n",
       "      <td>29</td>\n",
       "      <td>cooler balance \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>345287</th>\n",
       "      <td>hic</td>\n",
       "      <td>COOLER_BALANCE</td>\n",
       "      <td>cooler</td>\n",
       "      <td>./git_repos/hic/modules/nf-core/cooler/balance/main.nf</td>\n",
       "      <td>36</td>\n",
       "      <td>cooler: \\$(cooler --version 2&gt;&amp;1 | sed 's/cooler, version //')</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>345297</th>\n",
       "      <td>hic</td>\n",
       "      <td>COOLER_ZOOMIFY</td>\n",
       "      <td>cooler</td>\n",
       "      <td>./git_repos/hic/modules/nf-core/cooler/zoomify/main.nf</td>\n",
       "      <td>24</td>\n",
       "      <td>cooler zoomify \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>345338</th>\n",
       "      <td>hic</td>\n",
       "      <td>COOLER_ZOOMIFY</td>\n",
       "      <td>cooler</td>\n",
       "      <td>./git_repos/hic/modules/nf-core/cooler/zoomify/main.nf</td>\n",
       "      <td>32</td>\n",
       "      <td>cooler: \\$(cooler --version 2&gt;&amp;1 | sed 's/cooler, version //')</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>345347</th>\n",
       "      <td>hic</td>\n",
       "      <td>COOLER_DUMP</td>\n",
       "      <td>cooler</td>\n",
       "      <td>./git_repos/hic/modules/nf-core/cooler/dump/main.nf</td>\n",
       "      <td>25</td>\n",
       "      <td>cooler dump \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>345348</th>\n",
       "      <td>hic</td>\n",
       "      <td>COOLER_DUMP</td>\n",
       "      <td>dump</td>\n",
       "      <td>./git_repos/hic/modules/nf-core/cooler/dump/main.nf</td>\n",
       "      <td>25</td>\n",
       "      <td>cooler dump \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>345389</th>\n",
       "      <td>hic</td>\n",
       "      <td>COOLER_DUMP</td>\n",
       "      <td>cooler</td>\n",
       "      <td>./git_repos/hic/modules/nf-core/cooler/dump/main.nf</td>\n",
       "      <td>32</td>\n",
       "      <td>cooler: \\$(cooler --version 2&gt;&amp;1 | sed 's/cooler, version //')</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>345398</th>\n",
       "      <td>hic</td>\n",
       "      <td>COOLER_CLOAD</td>\n",
       "      <td>cooler</td>\n",
       "      <td>./git_repos/hic/modules/nf-core/cooler/cload/main.nf</td>\n",
       "      <td>27</td>\n",
       "      <td>cooler cload \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>345444</th>\n",
       "      <td>hic</td>\n",
       "      <td>COOLER_CLOAD</td>\n",
       "      <td>cooler</td>\n",
       "      <td>./git_repos/hic/modules/nf-core/cooler/cload/main.nf</td>\n",
       "      <td>36</td>\n",
       "      <td>cooler: \\$(cooler --version 2&gt;&amp;1 | sed 's/cooler, version //')</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>345453</th>\n",
       "      <td>hic</td>\n",
       "      <td>COOLER_MAKEBINS</td>\n",
       "      <td>make</td>\n",
       "      <td>./git_repos/hic/modules/nf-core/cooler/makebins/main.nf</td>\n",
       "      <td>24</td>\n",
       "      <td>cooler makebins \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>345454</th>\n",
       "      <td>hic</td>\n",
       "      <td>COOLER_MAKEBINS</td>\n",
       "      <td>cooler</td>\n",
       "      <td>./git_repos/hic/modules/nf-core/cooler/makebins/main.nf</td>\n",
       "      <td>24</td>\n",
       "      <td>cooler makebins \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>345495</th>\n",
       "      <td>hic</td>\n",
       "      <td>COOLER_MAKEBINS</td>\n",
       "      <td>cooler</td>\n",
       "      <td>./git_repos/hic/modules/nf-core/cooler/makebins/main.nf</td>\n",
       "      <td>31</td>\n",
       "      <td>cooler: \\$(cooler --version 2&gt;&amp;1 | sed 's/cooler, version //')</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>29532 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             pipeline              process_name       operator  \\\n",
       "62      diaproteomics     get_software_versions           echo   \n",
       "72      diaproteomics     get_software_versions           echo   \n",
       "76      diaproteomics     get_software_versions       nextflow   \n",
       "117     diaproteomics   dda_raw_file_conversion           Name   \n",
       "119     diaproteomics   dda_raw_file_conversion           file   \n",
       "129     diaproteomics  dda_id_format_conversion  FileConverter   \n",
       "132     diaproteomics  dda_id_format_conversion           file   \n",
       "135     diaproteomics    dda_library_generation        convert   \n",
       "143     diaproteomics    dda_library_generation           file   \n",
       "150     diaproteomics    dda_library_generation           file   \n",
       "159     diaproteomics    dda_library_generation           file   \n",
       "174     diaproteomics    dda_library_generation           Name   \n",
       "176     diaproteomics    dda_library_generation           file   \n",
       "186     diaproteomics    dda_library_generation         params   \n",
       "197     diaproteomics    dda_library_generation           Name   \n",
       "198     diaproteomics    dda_library_generation           file   \n",
       "206     diaproteomics    dda_library_generation           Name   \n",
       "207     diaproteomics    dda_library_generation           file   \n",
       "219     diaproteomics    dda_library_generation           Name   \n",
       "221     diaproteomics    dda_library_generation           file   \n",
       "225     diaproteomics          assay_generation  FileConverter   \n",
       "231     diaproteomics          assay_generation           file   \n",
       "238     diaproteomics          assay_generation           Name   \n",
       "240     diaproteomics          assay_generation           file   \n",
       "259     diaproteomics          assay_generation           Name   \n",
       "...               ...                       ...            ...   \n",
       "345117            hic                    FASTQC         printf   \n",
       "345126            hic                    FASTQC           done   \n",
       "345132            hic                    FASTQC         fastqc   \n",
       "345137            hic                    FASTQC         rename   \n",
       "345139            hic                    FASTQC          fastq   \n",
       "345145            hic                    FASTQC           file   \n",
       "345162            hic                    FASTQC         fastqc   \n",
       "345171            hic                    FASTQC          fastq   \n",
       "345180            hic                    FASTQC          touch   \n",
       "345188            hic                    FASTQC           html   \n",
       "345191            hic                    FASTQC          touch   \n",
       "345217            hic                    FASTQC         fastqc   \n",
       "345226            hic                    FASTQC          fastq   \n",
       "345246            hic            COOLER_BALANCE         cooler   \n",
       "345287            hic            COOLER_BALANCE         cooler   \n",
       "345297            hic            COOLER_ZOOMIFY         cooler   \n",
       "345338            hic            COOLER_ZOOMIFY         cooler   \n",
       "345347            hic               COOLER_DUMP         cooler   \n",
       "345348            hic               COOLER_DUMP           dump   \n",
       "345389            hic               COOLER_DUMP         cooler   \n",
       "345398            hic              COOLER_CLOAD         cooler   \n",
       "345444            hic              COOLER_CLOAD         cooler   \n",
       "345453            hic           COOLER_MAKEBINS           make   \n",
       "345454            hic           COOLER_MAKEBINS         cooler   \n",
       "345495            hic           COOLER_MAKEBINS         cooler   \n",
       "\n",
       "                                                      file_path  line_number  \\\n",
       "62                            ./git_repos/diaproteomics/main.nf          335   \n",
       "72                            ./git_repos/diaproteomics/main.nf          336   \n",
       "76                            ./git_repos/diaproteomics/main.nf          336   \n",
       "117                           ./git_repos/diaproteomics/main.nf          359   \n",
       "119                           ./git_repos/diaproteomics/main.nf          359   \n",
       "129                           ./git_repos/diaproteomics/main.nf          380   \n",
       "132                           ./git_repos/diaproteomics/main.nf          380   \n",
       "135                           ./git_repos/diaproteomics/main.nf          402   \n",
       "143                           ./git_repos/diaproteomics/main.nf          403   \n",
       "150                           ./git_repos/diaproteomics/main.nf          404   \n",
       "159                           ./git_repos/diaproteomics/main.nf          405   \n",
       "174                           ./git_repos/diaproteomics/main.nf          408   \n",
       "176                           ./git_repos/diaproteomics/main.nf          408   \n",
       "186                           ./git_repos/diaproteomics/main.nf          409   \n",
       "197                           ./git_repos/diaproteomics/main.nf          411   \n",
       "198                           ./git_repos/diaproteomics/main.nf          411   \n",
       "206                           ./git_repos/diaproteomics/main.nf          412   \n",
       "207                           ./git_repos/diaproteomics/main.nf          412   \n",
       "219                           ./git_repos/diaproteomics/main.nf          414   \n",
       "221                           ./git_repos/diaproteomics/main.nf          414   \n",
       "225                           ./git_repos/diaproteomics/main.nf          435   \n",
       "231                           ./git_repos/diaproteomics/main.nf          436   \n",
       "238                           ./git_repos/diaproteomics/main.nf          437   \n",
       "240                           ./git_repos/diaproteomics/main.nf          437   \n",
       "259                           ./git_repos/diaproteomics/main.nf          441   \n",
       "...                                                         ...          ...   \n",
       "345117           ./git_repos/hic/modules/nf-core/fastqc/main.nf           29   \n",
       "345126           ./git_repos/hic/modules/nf-core/fastqc/main.nf           31   \n",
       "345132           ./git_repos/hic/modules/nf-core/fastqc/main.nf           32   \n",
       "345137           ./git_repos/hic/modules/nf-core/fastqc/main.nf           32   \n",
       "345139           ./git_repos/hic/modules/nf-core/fastqc/main.nf           32   \n",
       "345145           ./git_repos/hic/modules/nf-core/fastqc/main.nf           32   \n",
       "345162           ./git_repos/hic/modules/nf-core/fastqc/main.nf           36   \n",
       "345171           ./git_repos/hic/modules/nf-core/fastqc/main.nf           36   \n",
       "345180           ./git_repos/hic/modules/nf-core/fastqc/main.nf           43   \n",
       "345188           ./git_repos/hic/modules/nf-core/fastqc/main.nf           43   \n",
       "345191           ./git_repos/hic/modules/nf-core/fastqc/main.nf           44   \n",
       "345217           ./git_repos/hic/modules/nf-core/fastqc/main.nf           48   \n",
       "345226           ./git_repos/hic/modules/nf-core/fastqc/main.nf           48   \n",
       "345246   ./git_repos/hic/modules/nf-core/cooler/balance/main.nf           29   \n",
       "345287   ./git_repos/hic/modules/nf-core/cooler/balance/main.nf           36   \n",
       "345297   ./git_repos/hic/modules/nf-core/cooler/zoomify/main.nf           24   \n",
       "345338   ./git_repos/hic/modules/nf-core/cooler/zoomify/main.nf           32   \n",
       "345347      ./git_repos/hic/modules/nf-core/cooler/dump/main.nf           25   \n",
       "345348      ./git_repos/hic/modules/nf-core/cooler/dump/main.nf           25   \n",
       "345389      ./git_repos/hic/modules/nf-core/cooler/dump/main.nf           32   \n",
       "345398     ./git_repos/hic/modules/nf-core/cooler/cload/main.nf           27   \n",
       "345444     ./git_repos/hic/modules/nf-core/cooler/cload/main.nf           36   \n",
       "345453  ./git_repos/hic/modules/nf-core/cooler/makebins/main.nf           24   \n",
       "345454  ./git_repos/hic/modules/nf-core/cooler/makebins/main.nf           24   \n",
       "345495  ./git_repos/hic/modules/nf-core/cooler/makebins/main.nf           31   \n",
       "\n",
       "                                                                                           line_content  \n",
       "62                                                     echo $workflow.manifest.version > v_pipeline.txt  \n",
       "72                                                     echo $workflow.nextflow.version > v_nextflow.txt  \n",
       "76                                                     echo $workflow.nextflow.version > v_nextflow.txt  \n",
       "117                             ThermoRawFileParser.sh -i=${raw_file} -f=2 -b=${raw_file.baseName}.mzML  \n",
       "119                             ThermoRawFileParser.sh -i=${raw_file} -f=2 -b=${raw_file.baseName}.mzML  \n",
       "129     IDFileConverter -in ${dda_id_file} -out ${id}_${sample}_peptide_ids.idXML -threads ${task.cpus}  \n",
       "132     IDFileConverter -in ${dda_id_file} -out ${id}_${sample}_peptide_ids.idXML -threads ${task.cpus}  \n",
       "135                                                                                  easypqp convert \\\\  \n",
       "143                                                                          --unimod ${unimod_file} \\\\  \n",
       "150                                                                           --pepxml ${idxml_file} \\\\  \n",
       "159                                                                          --spectra ${dda_mzml_file}  \n",
       "174                                                    --out ${dda_mzml_file.baseName}_run_peaks.tsv \\\\  \n",
       "176                                                    --out ${dda_mzml_file.baseName}_run_peaks.tsv \\\\  \n",
       "186                                                  --rt_psm_fdr_threshold ${params.library_rt_fdr} \\\\  \n",
       "197                                                                 ${dda_mzml_file.baseName}.psmpkl \\\\  \n",
       "198                                                                 ${dda_mzml_file.baseName}.psmpkl \\\\  \n",
       "206                                                                   ${dda_mzml_file.baseName}.peakpkl  \n",
       "207                                                                   ${dda_mzml_file.baseName}.peakpkl  \n",
       "219                              mv ${dda_mzml_file.baseName}_run_peaks.tsv ${id}_${sample}_library.tsv  \n",
       "221                              mv ${dda_mzml_file.baseName}_run_peaks.tsv ${id}_${sample}_library.tsv  \n",
       "225                                                                            TargetedFileConverter \\\\  \n",
       "231                                                                               -in ${lib_file_na} \\\\  \n",
       "238                                                                 -out ${lib_file_na.baseName}.tsv \\\\  \n",
       "240                                                                 -out ${lib_file_na.baseName}.tsv \\\\  \n",
       "259                                                                  -in ${lib_file_na.baseName}.tsv \\\\  \n",
       "...                                                                                                 ...  \n",
       "345117                                  printf \"%s %s\\\\n\" $rename_to | while read old_name new_name; do  \n",
       "345126                                                                                             done  \n",
       "345132                                                 fastqc $args --threads $task.cpus $renamed_files  \n",
       "345137                                                 fastqc $args --threads $task.cpus $renamed_files  \n",
       "345139                                                 fastqc $args --threads $task.cpus $renamed_files  \n",
       "345145                                                 fastqc $args --threads $task.cpus $renamed_files  \n",
       "345162                                          fastqc: \\$( fastqc --version | sed -e \"s/FastQC v//g\" )  \n",
       "345171                                          fastqc: \\$( fastqc --version | sed -e \"s/FastQC v//g\" )  \n",
       "345180                                                                             touch ${prefix}.html  \n",
       "345188                                                                             touch ${prefix}.html  \n",
       "345191                                                                              touch ${prefix}.zip  \n",
       "345217                                          fastqc: \\$( fastqc --version | sed -e \"s/FastQC v//g\" )  \n",
       "345226                                          fastqc: \\$( fastqc --version | sed -e \"s/FastQC v//g\" )  \n",
       "345246                                                                                cooler balance \\\\  \n",
       "345287                                   cooler: \\$(cooler --version 2>&1 | sed 's/cooler, version //')  \n",
       "345297                                                                                cooler zoomify \\\\  \n",
       "345338                                   cooler: \\$(cooler --version 2>&1 | sed 's/cooler, version //')  \n",
       "345347                                                                                   cooler dump \\\\  \n",
       "345348                                                                                   cooler dump \\\\  \n",
       "345389                                   cooler: \\$(cooler --version 2>&1 | sed 's/cooler, version //')  \n",
       "345398                                                                                  cooler cload \\\\  \n",
       "345444                                   cooler: \\$(cooler --version 2>&1 | sed 's/cooler, version //')  \n",
       "345453                                                                               cooler makebins \\\\  \n",
       "345454                                                                               cooler makebins \\\\  \n",
       "345495                                   cooler: \\$(cooler --version 2>&1 | sed 's/cooler, version //')  \n",
       "\n",
       "[29532 rows x 6 columns]"
      ]
     },
     "execution_count": 191,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "operator_script_total"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 192,
   "id": "68cb6fc0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pipeline</th>\n",
       "      <th>process_name</th>\n",
       "      <th>operator_count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>airrflow</td>\n",
       "      <td>ADD_META_TO_TAB</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>airrflow</td>\n",
       "      <td>AIRRFLOW_REPORT</td>\n",
       "      <td>21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>airrflow</td>\n",
       "      <td>CHANGEO_ASSIGNGENES</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>airrflow</td>\n",
       "      <td>CHANGEO_CONVERTDB_FASTA</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>airrflow</td>\n",
       "      <td>CHANGEO_CREATEGERMLINES</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>airrflow</td>\n",
       "      <td>CHANGEO_MAKEDB</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>airrflow</td>\n",
       "      <td>CHANGEO_PARSEDB_SELECT</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>airrflow</td>\n",
       "      <td>CHANGEO_PARSEDB_SPLIT</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>airrflow</td>\n",
       "      <td>COLLAPSE_DUPLICATES</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>airrflow</td>\n",
       "      <td>DEFINE_CLONES</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>airrflow</td>\n",
       "      <td>DETECT_CONTAMINATION</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>airrflow</td>\n",
       "      <td>DOWSER_LINEAGES</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>airrflow</td>\n",
       "      <td>FASTP</td>\n",
       "      <td>51</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>airrflow</td>\n",
       "      <td>FASTQC</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>airrflow</td>\n",
       "      <td>FASTQC_POSTASSEMBLY</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>airrflow</td>\n",
       "      <td>FETCH_DATABASES</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>airrflow</td>\n",
       "      <td>FILTER_JUNCTION_MOD3</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>airrflow</td>\n",
       "      <td>FILTER_QUALITY</td>\n",
       "      <td>21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>airrflow</td>\n",
       "      <td>FIND_THRESHOLD</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>airrflow</td>\n",
       "      <td>GUNZIP</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>airrflow</td>\n",
       "      <td>MERGE_UMI</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>airrflow</td>\n",
       "      <td>MULTIQC</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>airrflow</td>\n",
       "      <td>PARSE_LOGS</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>airrflow</td>\n",
       "      <td>PRESTO_ASSEMBLEPAIRS</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>airrflow</td>\n",
       "      <td>PRESTO_BUILDCONSENSUS</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2839</th>\n",
       "      <td>viralrecon</td>\n",
       "      <td>PANGOLIN</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2840</th>\n",
       "      <td>viralrecon</td>\n",
       "      <td>PICARD_COLLECTMULTIPLEMETRICS</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2841</th>\n",
       "      <td>viralrecon</td>\n",
       "      <td>PICARD_MARKDUPLICATES</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2842</th>\n",
       "      <td>viralrecon</td>\n",
       "      <td>PLASMIDID</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2843</th>\n",
       "      <td>viralrecon</td>\n",
       "      <td>PLOT_BASE_DENSITY</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2844</th>\n",
       "      <td>viralrecon</td>\n",
       "      <td>PLOT_MOSDEPTH_REGIONS</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2845</th>\n",
       "      <td>viralrecon</td>\n",
       "      <td>PYCOQC</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2846</th>\n",
       "      <td>viralrecon</td>\n",
       "      <td>QUAST</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2847</th>\n",
       "      <td>viralrecon</td>\n",
       "      <td>RENAME_FASTA_HEADER</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2848</th>\n",
       "      <td>viralrecon</td>\n",
       "      <td>SAMPLESHEET_CHECK</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2849</th>\n",
       "      <td>viralrecon</td>\n",
       "      <td>SAMTOOLS_FLAGSTAT</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2850</th>\n",
       "      <td>viralrecon</td>\n",
       "      <td>SAMTOOLS_IDXSTATS</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2851</th>\n",
       "      <td>viralrecon</td>\n",
       "      <td>SAMTOOLS_INDEX</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2852</th>\n",
       "      <td>viralrecon</td>\n",
       "      <td>SAMTOOLS_SORT</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2853</th>\n",
       "      <td>viralrecon</td>\n",
       "      <td>SAMTOOLS_STATS</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2854</th>\n",
       "      <td>viralrecon</td>\n",
       "      <td>SAMTOOLS_VIEW</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2855</th>\n",
       "      <td>viralrecon</td>\n",
       "      <td>SNPEFF_ANN</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2856</th>\n",
       "      <td>viralrecon</td>\n",
       "      <td>SNPEFF_BUILD</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2857</th>\n",
       "      <td>viralrecon</td>\n",
       "      <td>SNPSIFT_EXTRACTFIELDS</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2858</th>\n",
       "      <td>viralrecon</td>\n",
       "      <td>SPADES</td>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2859</th>\n",
       "      <td>viralrecon</td>\n",
       "      <td>TABIX_BGZIP</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2860</th>\n",
       "      <td>viralrecon</td>\n",
       "      <td>TABIX_TABIX</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2861</th>\n",
       "      <td>viralrecon</td>\n",
       "      <td>UNICYCLER</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2862</th>\n",
       "      <td>viralrecon</td>\n",
       "      <td>UNTAR</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2863</th>\n",
       "      <td>viralrecon</td>\n",
       "      <td>VCFLIB_VCFUNIQ</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2864 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        pipeline                   process_name  operator_count\n",
       "0       airrflow                ADD_META_TO_TAB              11\n",
       "1       airrflow                AIRRFLOW_REPORT              21\n",
       "2       airrflow            CHANGEO_ASSIGNGENES               4\n",
       "3       airrflow        CHANGEO_CONVERTDB_FASTA               6\n",
       "4       airrflow        CHANGEO_CREATEGERMLINES               4\n",
       "5       airrflow                 CHANGEO_MAKEDB               5\n",
       "6       airrflow         CHANGEO_PARSEDB_SELECT               8\n",
       "7       airrflow          CHANGEO_PARSEDB_SPLIT               3\n",
       "8       airrflow            COLLAPSE_DUPLICATES              16\n",
       "9       airrflow                  DEFINE_CLONES              17\n",
       "10      airrflow           DETECT_CONTAMINATION              16\n",
       "11      airrflow                DOWSER_LINEAGES              14\n",
       "12      airrflow                          FASTP              51\n",
       "13      airrflow                         FASTQC              17\n",
       "14      airrflow            FASTQC_POSTASSEMBLY               5\n",
       "15      airrflow                FETCH_DATABASES               5\n",
       "16      airrflow           FILTER_JUNCTION_MOD3              12\n",
       "17      airrflow                 FILTER_QUALITY              21\n",
       "18      airrflow                 FIND_THRESHOLD              16\n",
       "19      airrflow                         GUNZIP               8\n",
       "20      airrflow                      MERGE_UMI              14\n",
       "21      airrflow                        MULTIQC              12\n",
       "22      airrflow                     PARSE_LOGS              14\n",
       "23      airrflow           PRESTO_ASSEMBLEPAIRS               4\n",
       "24      airrflow          PRESTO_BUILDCONSENSUS               5\n",
       "...          ...                            ...             ...\n",
       "2839  viralrecon                       PANGOLIN               5\n",
       "2840  viralrecon  PICARD_COLLECTMULTIPLEMETRICS              19\n",
       "2841  viralrecon          PICARD_MARKDUPLICATES              16\n",
       "2842  viralrecon                      PLASMIDID               3\n",
       "2843  viralrecon              PLOT_BASE_DENSITY               5\n",
       "2844  viralrecon          PLOT_MOSDEPTH_REGIONS               9\n",
       "2845  viralrecon                         PYCOQC               4\n",
       "2846  viralrecon                          QUAST               9\n",
       "2847  viralrecon            RENAME_FASTA_HEADER               2\n",
       "2848  viralrecon              SAMPLESHEET_CHECK               1\n",
       "2849  viralrecon              SAMTOOLS_FLAGSTAT               5\n",
       "2850  viralrecon              SAMTOOLS_IDXSTATS               3\n",
       "2851  viralrecon                 SAMTOOLS_INDEX              15\n",
       "2852  viralrecon                  SAMTOOLS_SORT               7\n",
       "2853  viralrecon                 SAMTOOLS_STATS               8\n",
       "2854  viralrecon                  SAMTOOLS_VIEW              10\n",
       "2855  viralrecon                     SNPEFF_ANN               8\n",
       "2856  viralrecon                   SNPEFF_BUILD              12\n",
       "2857  viralrecon          SNPSIFT_EXTRACTFIELDS               6\n",
       "2858  viralrecon                         SPADES              32\n",
       "2859  viralrecon                    TABIX_BGZIP              13\n",
       "2860  viralrecon                    TABIX_TABIX               6\n",
       "2861  viralrecon                      UNICYCLER               6\n",
       "2862  viralrecon                          UNTAR              10\n",
       "2863  viralrecon                 VCFLIB_VCFUNIQ               3\n",
       "\n",
       "[2864 rows x 3 columns]"
      ]
     },
     "execution_count": 192,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "operators_in_all_process = operator_script.groupby(['pipeline', 'process_name']).size().reset_index(name='operator_count')\n",
    "operators_in_all_process"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 193,
   "id": "1c67385a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pipeline</th>\n",
       "      <th>process_name</th>\n",
       "      <th>operator</th>\n",
       "      <th>file_path</th>\n",
       "      <th>line_number</th>\n",
       "      <th>line_content</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>13557</th>\n",
       "      <td>methylseq</td>\n",
       "      <td>QUALIMAP_BAMQC</td>\n",
       "      <td>java</td>\n",
       "      <td>./git_repos/methylseq/modules/nf-core/qualimap/bamqc/main.nf</td>\n",
       "      <td>38</td>\n",
       "      <td>export _JAVA_OPTIONS=-Djava.io.tmpdir=./tmp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13574</th>\n",
       "      <td>methylseq</td>\n",
       "      <td>QUALIMAP_BAMQC</td>\n",
       "      <td>java</td>\n",
       "      <td>./git_repos/methylseq/modules/nf-core/qualimap/bamqc/main.nf</td>\n",
       "      <td>40</td>\n",
       "      <td>--java-mem-size=$memory \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14201</th>\n",
       "      <td>methylseq</td>\n",
       "      <td>QUALIMAP_BAMQCCRAM</td>\n",
       "      <td>java</td>\n",
       "      <td>./git_repos/methylseq/modules/nf-core/qualimap/bamqccram/main.nf</td>\n",
       "      <td>40</td>\n",
       "      <td>export _JAVA_OPTIONS=-Djava.io.tmpdir=./tmp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14229</th>\n",
       "      <td>methylseq</td>\n",
       "      <td>QUALIMAP_BAMQCCRAM</td>\n",
       "      <td>java</td>\n",
       "      <td>./git_repos/methylseq/modules/nf-core/qualimap/bamqccram/main.nf</td>\n",
       "      <td>44</td>\n",
       "      <td>--java-mem-size=$memory \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27071</th>\n",
       "      <td>createpanelrefs</td>\n",
       "      <td>GATK4_PREPROCESSINTERVALS</td>\n",
       "      <td>java</td>\n",
       "      <td>./git_repos/createpanelrefs/modules/nf-core/gatk4/preprocessintervals/main.nf</td>\n",
       "      <td>38</td>\n",
       "      <td>gatk --java-options \"-Xmx${avail_mem}M -XX:-UsePerfData\" \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27204</th>\n",
       "      <td>createpanelrefs</td>\n",
       "      <td>GATK4_GERMLINECNVCALLER</td>\n",
       "      <td>java</td>\n",
       "      <td>./git_repos/createpanelrefs/modules/nf-core/gatk4/germlinecnvcaller/main.nf</td>\n",
       "      <td>40</td>\n",
       "      <td>gatk --java-options \"-Xmx${avail_mem}g -XX:-UsePerfData\" \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27377</th>\n",
       "      <td>createpanelrefs</td>\n",
       "      <td>GATK4_INTERVALLISTTOOLS</td>\n",
       "      <td>java</td>\n",
       "      <td>./git_repos/createpanelrefs/modules/nf-core/gatk4/intervallisttools/main.nf</td>\n",
       "      <td>34</td>\n",
       "      <td>gatk --java-options \"-Xmx${avail_mem}M -XX:-UsePerfData\" \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27692</th>\n",
       "      <td>createpanelrefs</td>\n",
       "      <td>GATK4_FILTERINTERVALS</td>\n",
       "      <td>java</td>\n",
       "      <td>./git_repos/createpanelrefs/modules/nf-core/gatk4/filterintervals/main.nf</td>\n",
       "      <td>35</td>\n",
       "      <td>gatk --java-options \"-Xmx${avail_mem}M -XX:-UsePerfData\" \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27823</th>\n",
       "      <td>createpanelrefs</td>\n",
       "      <td>GATK4_ANNOTATEINTERVALS</td>\n",
       "      <td>java</td>\n",
       "      <td>./git_repos/createpanelrefs/modules/nf-core/gatk4/annotateintervals/main.nf</td>\n",
       "      <td>43</td>\n",
       "      <td>gatk --java-options \"-Xmx${avail_mem}M -XX:-UsePerfData\" \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27966</th>\n",
       "      <td>createpanelrefs</td>\n",
       "      <td>GATK4_COLLECTREADCOUNTS</td>\n",
       "      <td>java</td>\n",
       "      <td>./git_repos/createpanelrefs/modules/nf-core/gatk4/collectreadcounts/main.nf</td>\n",
       "      <td>40</td>\n",
       "      <td>gatk --java-options \"-Xmx${avail_mem}M -XX:-UsePerfData\" \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28099</th>\n",
       "      <td>createpanelrefs</td>\n",
       "      <td>GATK4_DETERMINEGERMLINECONTIGPLOIDY</td>\n",
       "      <td>java</td>\n",
       "      <td>./git_repos/createpanelrefs/modules/nf-core/gatk4/determinegermlinecontigploidy/main.nf</td>\n",
       "      <td>42</td>\n",
       "      <td>gatk --java-options \"-Xmx${avail_mem}M -XX:-UsePerfData\" \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39353</th>\n",
       "      <td>ddamsproteomics</td>\n",
       "      <td>msgfPlus</td>\n",
       "      <td>java</td>\n",
       "      <td>./git_repos/ddamsproteomics/main.nf</td>\n",
       "      <td>515</td>\n",
       "      <td>msgf_plus -Xmx3500M edu.ucsd.msjava.ui.MzIDToTsv -i \"${sample}.mzid\" -o out.mzid.tsv</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40507</th>\n",
       "      <td>hgtseq</td>\n",
       "      <td>QUALIMAP_BAMQC</td>\n",
       "      <td>java</td>\n",
       "      <td>./git_repos/hgtseq/modules/nf-core/qualimap/bamqc/main.nf</td>\n",
       "      <td>38</td>\n",
       "      <td>export _JAVA_OPTIONS=-Djava.io.tmpdir=./tmp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40524</th>\n",
       "      <td>hgtseq</td>\n",
       "      <td>QUALIMAP_BAMQC</td>\n",
       "      <td>java</td>\n",
       "      <td>./git_repos/hgtseq/modules/nf-core/qualimap/bamqc/main.nf</td>\n",
       "      <td>40</td>\n",
       "      <td>--java-mem-size=$memory \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89722</th>\n",
       "      <td>sarek</td>\n",
       "      <td>GATK4_CALCULATECONTAMINATION</td>\n",
       "      <td>java</td>\n",
       "      <td>./git_repos/sarek/modules/nf-core/gatk4/calculatecontamination/main.nf</td>\n",
       "      <td>33</td>\n",
       "      <td>gatk --java-options \"-Xmx${avail_mem}M -XX:-UsePerfData\" \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89797</th>\n",
       "      <td>sarek</td>\n",
       "      <td>GATK4_BASERECALIBRATOR</td>\n",
       "      <td>java</td>\n",
       "      <td>./git_repos/sarek/modules/nf-core/gatk4/baserecalibrator/main.nf</td>\n",
       "      <td>38</td>\n",
       "      <td>gatk --java-options \"-Xmx${avail_mem}M -XX:-UsePerfData\" \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89889</th>\n",
       "      <td>sarek</td>\n",
       "      <td>GATK4_GETPILEUPSUMMARIES</td>\n",
       "      <td>java</td>\n",
       "      <td>./git_repos/sarek/modules/nf-core/gatk4/getpileupsummaries/main.nf</td>\n",
       "      <td>38</td>\n",
       "      <td>gatk --java-options \"-Xmx${avail_mem}M -XX:-UsePerfData\" \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89980</th>\n",
       "      <td>sarek</td>\n",
       "      <td>GATK4_HAPLOTYPECALLER</td>\n",
       "      <td>java</td>\n",
       "      <td>./git_repos/sarek/modules/nf-core/gatk4/haplotypecaller/main.nf</td>\n",
       "      <td>42</td>\n",
       "      <td>gatk --java-options \"-Xmx${avail_mem}M -XX:-UsePerfData\" \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>90145</th>\n",
       "      <td>sarek</td>\n",
       "      <td>GATK4_MUTECT2</td>\n",
       "      <td>java</td>\n",
       "      <td>./git_repos/sarek/modules/nf-core/gatk4/mutect2/main.nf</td>\n",
       "      <td>45</td>\n",
       "      <td>gatk --java-options \"-Xmx${avail_mem}M -XX:-UsePerfData\" \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>90326</th>\n",
       "      <td>sarek</td>\n",
       "      <td>GATK4_GATHERBQSRREPORTS</td>\n",
       "      <td>java</td>\n",
       "      <td>./git_repos/sarek/modules/nf-core/gatk4/gatherbqsrreports/main.nf</td>\n",
       "      <td>32</td>\n",
       "      <td>gatk --java-options \"-Xmx${avail_mem}M -XX:-UsePerfData\" \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>90403</th>\n",
       "      <td>sarek</td>\n",
       "      <td>GATK4_ESTIMATELIBRARYCOMPLEXITY</td>\n",
       "      <td>java</td>\n",
       "      <td>./git_repos/sarek/modules/nf-core/gatk4/estimatelibrarycomplexity/main.nf</td>\n",
       "      <td>35</td>\n",
       "      <td>gatk --java-options \"-Xmx${avail_mem}M -XX:-UsePerfData\" \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>90487</th>\n",
       "      <td>sarek</td>\n",
       "      <td>GATK4_LEARNREADORIENTATIONMODEL</td>\n",
       "      <td>java</td>\n",
       "      <td>./git_repos/sarek/modules/nf-core/gatk4/learnreadorientationmodel/main.nf</td>\n",
       "      <td>32</td>\n",
       "      <td>gatk --java-options \"-Xmx${avail_mem}M -XX:-UsePerfData\" \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>90564</th>\n",
       "      <td>sarek</td>\n",
       "      <td>GATK4_GENOTYPEGVCFS</td>\n",
       "      <td>java</td>\n",
       "      <td>./git_repos/sarek/modules/nf-core/gatk4/genotypegvcfs/main.nf</td>\n",
       "      <td>40</td>\n",
       "      <td>gatk --java-options \"-Xmx${avail_mem}M -XX:-UsePerfData\" \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>90717</th>\n",
       "      <td>sarek</td>\n",
       "      <td>GATK4_MARKDUPLICATES</td>\n",
       "      <td>java</td>\n",
       "      <td>./git_repos/sarek/modules/nf-core/gatk4/markduplicates/main.nf</td>\n",
       "      <td>46</td>\n",
       "      <td>gatk --java-options \"-Xmx${avail_mem}M -XX:-UsePerfData\" \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>90999</th>\n",
       "      <td>sarek</td>\n",
       "      <td>GATK4_GATHERPILEUPSUMMARIES</td>\n",
       "      <td>java</td>\n",
       "      <td>./git_repos/sarek/modules/nf-core/gatk4/gatherpileupsummaries/main.nf</td>\n",
       "      <td>34</td>\n",
       "      <td>gatk --java-options \"-Xmx${avail_mem}M -XX:-UsePerfData\" \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>266291</th>\n",
       "      <td>raredisease</td>\n",
       "      <td>GATK4_GERMLINECNVCALLER</td>\n",
       "      <td>java</td>\n",
       "      <td>./git_repos/raredisease/modules/nf-core/gatk4/germlinecnvcaller/main.nf</td>\n",
       "      <td>40</td>\n",
       "      <td>gatk --java-options \"-Xmx${avail_mem}g\" GermlineCNVCaller \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>266430</th>\n",
       "      <td>raredisease</td>\n",
       "      <td>GATK4_SPLITINTERVALS</td>\n",
       "      <td>java</td>\n",
       "      <td>./git_repos/raredisease/modules/nf-core/gatk4/splitintervals/main.nf</td>\n",
       "      <td>35</td>\n",
       "      <td>gatk --java-options \"-Xmx${avail_mem}M\" SplitIntervals \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>266588</th>\n",
       "      <td>raredisease</td>\n",
       "      <td>GATK4_INTERVALLISTTOOLS</td>\n",
       "      <td>java</td>\n",
       "      <td>./git_repos/raredisease/modules/nf-core/gatk4/intervallisttools/main.nf</td>\n",
       "      <td>34</td>\n",
       "      <td>gatk --java-options \"-Xmx${avail_mem}M\" IntervalListTools \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>266895</th>\n",
       "      <td>raredisease</td>\n",
       "      <td>GATK4_SELECTVARIANTS</td>\n",
       "      <td>java</td>\n",
       "      <td>./git_repos/raredisease/modules/nf-core/gatk4/selectvariants/main.nf</td>\n",
       "      <td>33</td>\n",
       "      <td>gatk --java-options \"-Xmx${avail_mem}M\" SelectVariants \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>267032</th>\n",
       "      <td>raredisease</td>\n",
       "      <td>GATK4_REVERTSAM</td>\n",
       "      <td>java</td>\n",
       "      <td>./git_repos/raredisease/modules/nf-core/gatk4/revertsam/main.nf</td>\n",
       "      <td>31</td>\n",
       "      <td>gatk --java-options \"-Xmx${avail_mem}M\" RevertSam \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>269238</th>\n",
       "      <td>raredisease</td>\n",
       "      <td>QUALIMAP_BAMQC</td>\n",
       "      <td>java</td>\n",
       "      <td>./git_repos/raredisease/modules/nf-core/qualimap/bamqc/main.nf</td>\n",
       "      <td>38</td>\n",
       "      <td>export _JAVA_OPTIONS=-Djava.io.tmpdir=./tmp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>269255</th>\n",
       "      <td>raredisease</td>\n",
       "      <td>QUALIMAP_BAMQC</td>\n",
       "      <td>java</td>\n",
       "      <td>./git_repos/raredisease/modules/nf-core/qualimap/bamqc/main.nf</td>\n",
       "      <td>40</td>\n",
       "      <td>--java-mem-size=$memory \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>275674</th>\n",
       "      <td>smrnaseq</td>\n",
       "      <td>MIRTRACE_RUN</td>\n",
       "      <td>java</td>\n",
       "      <td>./git_repos/smrnaseq/modules/local/mirtrace.nf</td>\n",
       "      <td>36</td>\n",
       "      <td>java $java_mem -jar \\$mirtracejar/mirtrace.jar --mirtrace-wrapper-name mirtrace qc  \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>282762</th>\n",
       "      <td>rnaseq</td>\n",
       "      <td>QUALIMAP_RNASEQ</td>\n",
       "      <td>java</td>\n",
       "      <td>./git_repos/rnaseq/modules/nf-core/qualimap/rnaseq/main.nf</td>\n",
       "      <td>36</td>\n",
       "      <td>export _JAVA_OPTIONS=-Djava.io.tmpdir=./tmp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>282779</th>\n",
       "      <td>rnaseq</td>\n",
       "      <td>QUALIMAP_RNASEQ</td>\n",
       "      <td>java</td>\n",
       "      <td>./git_repos/rnaseq/modules/nf-core/qualimap/rnaseq/main.nf</td>\n",
       "      <td>38</td>\n",
       "      <td>--java-mem-size=$memory \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>310521</th>\n",
       "      <td>variantcatalogue</td>\n",
       "      <td>GATK4_VARIANTFILTRATION</td>\n",
       "      <td>java</td>\n",
       "      <td>./git_repos/variantcatalogue/modules/nf-core/gatk4/variantfiltration/main.nf</td>\n",
       "      <td>35</td>\n",
       "      <td>gatk --java-options \"-Xmx${avail_mem}M\" VariantFiltration \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>310655</th>\n",
       "      <td>variantcatalogue</td>\n",
       "      <td>GATK4_LEFTALIGNANDTRIMVARIANTS</td>\n",
       "      <td>java</td>\n",
       "      <td>./git_repos/variantcatalogue/modules/nf-core/gatk4/leftalignandtrimvariants/main.nf</td>\n",
       "      <td>35</td>\n",
       "      <td>gatk --java-options \"-Xmx${avail_mem}M\" \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>310741</th>\n",
       "      <td>variantcatalogue</td>\n",
       "      <td>GATK4_MUTECT2</td>\n",
       "      <td>java</td>\n",
       "      <td>./git_repos/variantcatalogue/modules/nf-core/gatk4/mutect2/main.nf</td>\n",
       "      <td>45</td>\n",
       "      <td>gatk --java-options \"-Xmx${avail_mem}M\" Mutect2 \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>310916</th>\n",
       "      <td>variantcatalogue</td>\n",
       "      <td>GATK4_MERGEVCFS</td>\n",
       "      <td>java</td>\n",
       "      <td>./git_repos/variantcatalogue/modules/nf-core/gatk4/mergevcfs/main.nf</td>\n",
       "      <td>35</td>\n",
       "      <td>gatk --java-options \"-Xmx${avail_mem}M\" MergeVcfs \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>311049</th>\n",
       "      <td>variantcatalogue</td>\n",
       "      <td>GATK4_SAMTOFASTQ</td>\n",
       "      <td>java</td>\n",
       "      <td>./git_repos/variantcatalogue/modules/nf-core/gatk4/samtofastq/main.nf</td>\n",
       "      <td>32</td>\n",
       "      <td>gatk --java-options \"-Xmx${avail_mem}M\" SamToFastq \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>311188</th>\n",
       "      <td>variantcatalogue</td>\n",
       "      <td>GATK4_MERGEMUTECTSTATS</td>\n",
       "      <td>java</td>\n",
       "      <td>./git_repos/variantcatalogue/modules/nf-core/gatk4/mergemutectstats/main.nf</td>\n",
       "      <td>32</td>\n",
       "      <td>gatk --java-options \"-Xmx${avail_mem}M\" MergeMutectStats \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>311260</th>\n",
       "      <td>variantcatalogue</td>\n",
       "      <td>GATK4_MARKDUPLICATES</td>\n",
       "      <td>java</td>\n",
       "      <td>./git_repos/variantcatalogue/modules/nf-core/gatk4/markduplicates/main.nf</td>\n",
       "      <td>39</td>\n",
       "      <td>gatk --java-options \"-Xmx${avail_mem}M\" MarkDuplicates \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>311375</th>\n",
       "      <td>variantcatalogue</td>\n",
       "      <td>GATK4_PRINTREADS</td>\n",
       "      <td>java</td>\n",
       "      <td>./git_repos/variantcatalogue/modules/nf-core/gatk4/printreads/main.nf</td>\n",
       "      <td>39</td>\n",
       "      <td>gatk --java-options \"-Xmx${avail_mem}M\" PrintReads \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>311528</th>\n",
       "      <td>variantcatalogue</td>\n",
       "      <td>GATK4_FILTERMUTECTCALLS</td>\n",
       "      <td>java</td>\n",
       "      <td>./git_repos/variantcatalogue/modules/nf-core/gatk4/filtermutectcalls/main.nf</td>\n",
       "      <td>41</td>\n",
       "      <td>gatk --java-options \"-Xmx${avail_mem}M\" FilterMutectCalls \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>312799</th>\n",
       "      <td>variantcatalogue</td>\n",
       "      <td>Picard_CollectAlignmentSummaryMetrics</td>\n",
       "      <td>java</td>\n",
       "      <td>./git_repos/variantcatalogue/modules/local/Picard_CollectAlignmentSummaryMetrics.nf</td>\n",
       "      <td>19</td>\n",
       "      <td>--java-options \"-Xmx2000M\" \\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>313117</th>\n",
       "      <td>variantcatalogue</td>\n",
       "      <td>GATK4_VARIANTFILTRATION</td>\n",
       "      <td>java</td>\n",
       "      <td>./git_repos/variantcatalogue/modules/local/variantfiltration/main.nf</td>\n",
       "      <td>37</td>\n",
       "      <td>gatk --java-options \"-Xmx${avail_mem}M\" VariantFiltration \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>313293</th>\n",
       "      <td>variantcatalogue</td>\n",
       "      <td>GATK4_LEFTALIGNANDTRIMVARIANTS</td>\n",
       "      <td>java</td>\n",
       "      <td>./git_repos/variantcatalogue/modules/local/gatk4/leftalignandtrimvariants/main.nf</td>\n",
       "      <td>35</td>\n",
       "      <td>gatk --java-options \"-Xmx${avail_mem}M\" \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>313380</th>\n",
       "      <td>variantcatalogue</td>\n",
       "      <td>GATK4_MERGEMUTECTSTATS</td>\n",
       "      <td>java</td>\n",
       "      <td>./git_repos/variantcatalogue/modules/local/gatk4/mergemutectstats/main.nf</td>\n",
       "      <td>33</td>\n",
       "      <td>gatk --java-options \"-Xmx${avail_mem}M\" MergeMutectStats \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>313457</th>\n",
       "      <td>variantcatalogue</td>\n",
       "      <td>GATK4_SAMTOFASTQ</td>\n",
       "      <td>java</td>\n",
       "      <td>./git_repos/variantcatalogue/modules/local/gatk4/samtofastq/main.nf</td>\n",
       "      <td>30</td>\n",
       "      <td>gatk --java-options \"-Xmx${avail_mem}M\" SamToFastq \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>313606</th>\n",
       "      <td>variantcatalogue</td>\n",
       "      <td>GATK4_PRINTREADS</td>\n",
       "      <td>java</td>\n",
       "      <td>./git_repos/variantcatalogue/modules/local/gatk4/printreads/main.nf</td>\n",
       "      <td>39</td>\n",
       "      <td>gatk --java-options \"-Xmx${avail_mem}M\" PrintReads \\\\</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>148 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                pipeline                           process_name operator  \\\n",
       "13557          methylseq                         QUALIMAP_BAMQC     java   \n",
       "13574          methylseq                         QUALIMAP_BAMQC     java   \n",
       "14201          methylseq                     QUALIMAP_BAMQCCRAM     java   \n",
       "14229          methylseq                     QUALIMAP_BAMQCCRAM     java   \n",
       "27071    createpanelrefs              GATK4_PREPROCESSINTERVALS     java   \n",
       "27204    createpanelrefs                GATK4_GERMLINECNVCALLER     java   \n",
       "27377    createpanelrefs                GATK4_INTERVALLISTTOOLS     java   \n",
       "27692    createpanelrefs                  GATK4_FILTERINTERVALS     java   \n",
       "27823    createpanelrefs                GATK4_ANNOTATEINTERVALS     java   \n",
       "27966    createpanelrefs                GATK4_COLLECTREADCOUNTS     java   \n",
       "28099    createpanelrefs    GATK4_DETERMINEGERMLINECONTIGPLOIDY     java   \n",
       "39353    ddamsproteomics                               msgfPlus     java   \n",
       "40507             hgtseq                         QUALIMAP_BAMQC     java   \n",
       "40524             hgtseq                         QUALIMAP_BAMQC     java   \n",
       "89722              sarek           GATK4_CALCULATECONTAMINATION     java   \n",
       "89797              sarek                 GATK4_BASERECALIBRATOR     java   \n",
       "89889              sarek               GATK4_GETPILEUPSUMMARIES     java   \n",
       "89980              sarek                  GATK4_HAPLOTYPECALLER     java   \n",
       "90145              sarek                          GATK4_MUTECT2     java   \n",
       "90326              sarek                GATK4_GATHERBQSRREPORTS     java   \n",
       "90403              sarek        GATK4_ESTIMATELIBRARYCOMPLEXITY     java   \n",
       "90487              sarek        GATK4_LEARNREADORIENTATIONMODEL     java   \n",
       "90564              sarek                    GATK4_GENOTYPEGVCFS     java   \n",
       "90717              sarek                   GATK4_MARKDUPLICATES     java   \n",
       "90999              sarek            GATK4_GATHERPILEUPSUMMARIES     java   \n",
       "...                  ...                                    ...      ...   \n",
       "266291       raredisease                GATK4_GERMLINECNVCALLER     java   \n",
       "266430       raredisease                   GATK4_SPLITINTERVALS     java   \n",
       "266588       raredisease                GATK4_INTERVALLISTTOOLS     java   \n",
       "266895       raredisease                   GATK4_SELECTVARIANTS     java   \n",
       "267032       raredisease                        GATK4_REVERTSAM     java   \n",
       "269238       raredisease                         QUALIMAP_BAMQC     java   \n",
       "269255       raredisease                         QUALIMAP_BAMQC     java   \n",
       "275674          smrnaseq                           MIRTRACE_RUN     java   \n",
       "282762            rnaseq                        QUALIMAP_RNASEQ     java   \n",
       "282779            rnaseq                        QUALIMAP_RNASEQ     java   \n",
       "310521  variantcatalogue                GATK4_VARIANTFILTRATION     java   \n",
       "310655  variantcatalogue         GATK4_LEFTALIGNANDTRIMVARIANTS     java   \n",
       "310741  variantcatalogue                          GATK4_MUTECT2     java   \n",
       "310916  variantcatalogue                        GATK4_MERGEVCFS     java   \n",
       "311049  variantcatalogue                       GATK4_SAMTOFASTQ     java   \n",
       "311188  variantcatalogue                 GATK4_MERGEMUTECTSTATS     java   \n",
       "311260  variantcatalogue                   GATK4_MARKDUPLICATES     java   \n",
       "311375  variantcatalogue                       GATK4_PRINTREADS     java   \n",
       "311528  variantcatalogue                GATK4_FILTERMUTECTCALLS     java   \n",
       "312799  variantcatalogue  Picard_CollectAlignmentSummaryMetrics     java   \n",
       "313117  variantcatalogue                GATK4_VARIANTFILTRATION     java   \n",
       "313293  variantcatalogue         GATK4_LEFTALIGNANDTRIMVARIANTS     java   \n",
       "313380  variantcatalogue                 GATK4_MERGEMUTECTSTATS     java   \n",
       "313457  variantcatalogue                       GATK4_SAMTOFASTQ     java   \n",
       "313606  variantcatalogue                       GATK4_PRINTREADS     java   \n",
       "\n",
       "                                                                                      file_path  \\\n",
       "13557                              ./git_repos/methylseq/modules/nf-core/qualimap/bamqc/main.nf   \n",
       "13574                              ./git_repos/methylseq/modules/nf-core/qualimap/bamqc/main.nf   \n",
       "14201                          ./git_repos/methylseq/modules/nf-core/qualimap/bamqccram/main.nf   \n",
       "14229                          ./git_repos/methylseq/modules/nf-core/qualimap/bamqccram/main.nf   \n",
       "27071             ./git_repos/createpanelrefs/modules/nf-core/gatk4/preprocessintervals/main.nf   \n",
       "27204               ./git_repos/createpanelrefs/modules/nf-core/gatk4/germlinecnvcaller/main.nf   \n",
       "27377               ./git_repos/createpanelrefs/modules/nf-core/gatk4/intervallisttools/main.nf   \n",
       "27692                 ./git_repos/createpanelrefs/modules/nf-core/gatk4/filterintervals/main.nf   \n",
       "27823               ./git_repos/createpanelrefs/modules/nf-core/gatk4/annotateintervals/main.nf   \n",
       "27966               ./git_repos/createpanelrefs/modules/nf-core/gatk4/collectreadcounts/main.nf   \n",
       "28099   ./git_repos/createpanelrefs/modules/nf-core/gatk4/determinegermlinecontigploidy/main.nf   \n",
       "39353                                                       ./git_repos/ddamsproteomics/main.nf   \n",
       "40507                                 ./git_repos/hgtseq/modules/nf-core/qualimap/bamqc/main.nf   \n",
       "40524                                 ./git_repos/hgtseq/modules/nf-core/qualimap/bamqc/main.nf   \n",
       "89722                    ./git_repos/sarek/modules/nf-core/gatk4/calculatecontamination/main.nf   \n",
       "89797                          ./git_repos/sarek/modules/nf-core/gatk4/baserecalibrator/main.nf   \n",
       "89889                        ./git_repos/sarek/modules/nf-core/gatk4/getpileupsummaries/main.nf   \n",
       "89980                           ./git_repos/sarek/modules/nf-core/gatk4/haplotypecaller/main.nf   \n",
       "90145                                   ./git_repos/sarek/modules/nf-core/gatk4/mutect2/main.nf   \n",
       "90326                         ./git_repos/sarek/modules/nf-core/gatk4/gatherbqsrreports/main.nf   \n",
       "90403                 ./git_repos/sarek/modules/nf-core/gatk4/estimatelibrarycomplexity/main.nf   \n",
       "90487                 ./git_repos/sarek/modules/nf-core/gatk4/learnreadorientationmodel/main.nf   \n",
       "90564                             ./git_repos/sarek/modules/nf-core/gatk4/genotypegvcfs/main.nf   \n",
       "90717                            ./git_repos/sarek/modules/nf-core/gatk4/markduplicates/main.nf   \n",
       "90999                     ./git_repos/sarek/modules/nf-core/gatk4/gatherpileupsummaries/main.nf   \n",
       "...                                                                                         ...   \n",
       "266291                  ./git_repos/raredisease/modules/nf-core/gatk4/germlinecnvcaller/main.nf   \n",
       "266430                     ./git_repos/raredisease/modules/nf-core/gatk4/splitintervals/main.nf   \n",
       "266588                  ./git_repos/raredisease/modules/nf-core/gatk4/intervallisttools/main.nf   \n",
       "266895                     ./git_repos/raredisease/modules/nf-core/gatk4/selectvariants/main.nf   \n",
       "267032                          ./git_repos/raredisease/modules/nf-core/gatk4/revertsam/main.nf   \n",
       "269238                           ./git_repos/raredisease/modules/nf-core/qualimap/bamqc/main.nf   \n",
       "269255                           ./git_repos/raredisease/modules/nf-core/qualimap/bamqc/main.nf   \n",
       "275674                                           ./git_repos/smrnaseq/modules/local/mirtrace.nf   \n",
       "282762                               ./git_repos/rnaseq/modules/nf-core/qualimap/rnaseq/main.nf   \n",
       "282779                               ./git_repos/rnaseq/modules/nf-core/qualimap/rnaseq/main.nf   \n",
       "310521             ./git_repos/variantcatalogue/modules/nf-core/gatk4/variantfiltration/main.nf   \n",
       "310655      ./git_repos/variantcatalogue/modules/nf-core/gatk4/leftalignandtrimvariants/main.nf   \n",
       "310741                       ./git_repos/variantcatalogue/modules/nf-core/gatk4/mutect2/main.nf   \n",
       "310916                     ./git_repos/variantcatalogue/modules/nf-core/gatk4/mergevcfs/main.nf   \n",
       "311049                    ./git_repos/variantcatalogue/modules/nf-core/gatk4/samtofastq/main.nf   \n",
       "311188              ./git_repos/variantcatalogue/modules/nf-core/gatk4/mergemutectstats/main.nf   \n",
       "311260                ./git_repos/variantcatalogue/modules/nf-core/gatk4/markduplicates/main.nf   \n",
       "311375                    ./git_repos/variantcatalogue/modules/nf-core/gatk4/printreads/main.nf   \n",
       "311528             ./git_repos/variantcatalogue/modules/nf-core/gatk4/filtermutectcalls/main.nf   \n",
       "312799      ./git_repos/variantcatalogue/modules/local/Picard_CollectAlignmentSummaryMetrics.nf   \n",
       "313117                     ./git_repos/variantcatalogue/modules/local/variantfiltration/main.nf   \n",
       "313293        ./git_repos/variantcatalogue/modules/local/gatk4/leftalignandtrimvariants/main.nf   \n",
       "313380                ./git_repos/variantcatalogue/modules/local/gatk4/mergemutectstats/main.nf   \n",
       "313457                      ./git_repos/variantcatalogue/modules/local/gatk4/samtofastq/main.nf   \n",
       "313606                      ./git_repos/variantcatalogue/modules/local/gatk4/printreads/main.nf   \n",
       "\n",
       "        line_number  \\\n",
       "13557            38   \n",
       "13574            40   \n",
       "14201            40   \n",
       "14229            44   \n",
       "27071            38   \n",
       "27204            40   \n",
       "27377            34   \n",
       "27692            35   \n",
       "27823            43   \n",
       "27966            40   \n",
       "28099            42   \n",
       "39353           515   \n",
       "40507            38   \n",
       "40524            40   \n",
       "89722            33   \n",
       "89797            38   \n",
       "89889            38   \n",
       "89980            42   \n",
       "90145            45   \n",
       "90326            32   \n",
       "90403            35   \n",
       "90487            32   \n",
       "90564            40   \n",
       "90717            46   \n",
       "90999            34   \n",
       "...             ...   \n",
       "266291           40   \n",
       "266430           35   \n",
       "266588           34   \n",
       "266895           33   \n",
       "267032           31   \n",
       "269238           38   \n",
       "269255           40   \n",
       "275674           36   \n",
       "282762           36   \n",
       "282779           38   \n",
       "310521           35   \n",
       "310655           35   \n",
       "310741           45   \n",
       "310916           35   \n",
       "311049           32   \n",
       "311188           32   \n",
       "311260           39   \n",
       "311375           39   \n",
       "311528           41   \n",
       "312799           19   \n",
       "313117           37   \n",
       "313293           35   \n",
       "313380           33   \n",
       "313457           30   \n",
       "313606           39   \n",
       "\n",
       "                                                                                  line_content  \n",
       "13557                                              export _JAVA_OPTIONS=-Djava.io.tmpdir=./tmp  \n",
       "13574                                                               --java-mem-size=$memory \\\\  \n",
       "14201                                              export _JAVA_OPTIONS=-Djava.io.tmpdir=./tmp  \n",
       "14229                                                               --java-mem-size=$memory \\\\  \n",
       "27071                              gatk --java-options \"-Xmx${avail_mem}M -XX:-UsePerfData\" \\\\  \n",
       "27204                              gatk --java-options \"-Xmx${avail_mem}g -XX:-UsePerfData\" \\\\  \n",
       "27377                              gatk --java-options \"-Xmx${avail_mem}M -XX:-UsePerfData\" \\\\  \n",
       "27692                              gatk --java-options \"-Xmx${avail_mem}M -XX:-UsePerfData\" \\\\  \n",
       "27823                              gatk --java-options \"-Xmx${avail_mem}M -XX:-UsePerfData\" \\\\  \n",
       "27966                              gatk --java-options \"-Xmx${avail_mem}M -XX:-UsePerfData\" \\\\  \n",
       "28099                              gatk --java-options \"-Xmx${avail_mem}M -XX:-UsePerfData\" \\\\  \n",
       "39353     msgf_plus -Xmx3500M edu.ucsd.msjava.ui.MzIDToTsv -i \"${sample}.mzid\" -o out.mzid.tsv  \n",
       "40507                                              export _JAVA_OPTIONS=-Djava.io.tmpdir=./tmp  \n",
       "40524                                                               --java-mem-size=$memory \\\\  \n",
       "89722                              gatk --java-options \"-Xmx${avail_mem}M -XX:-UsePerfData\" \\\\  \n",
       "89797                              gatk --java-options \"-Xmx${avail_mem}M -XX:-UsePerfData\" \\\\  \n",
       "89889                              gatk --java-options \"-Xmx${avail_mem}M -XX:-UsePerfData\" \\\\  \n",
       "89980                              gatk --java-options \"-Xmx${avail_mem}M -XX:-UsePerfData\" \\\\  \n",
       "90145                              gatk --java-options \"-Xmx${avail_mem}M -XX:-UsePerfData\" \\\\  \n",
       "90326                              gatk --java-options \"-Xmx${avail_mem}M -XX:-UsePerfData\" \\\\  \n",
       "90403                              gatk --java-options \"-Xmx${avail_mem}M -XX:-UsePerfData\" \\\\  \n",
       "90487                              gatk --java-options \"-Xmx${avail_mem}M -XX:-UsePerfData\" \\\\  \n",
       "90564                              gatk --java-options \"-Xmx${avail_mem}M -XX:-UsePerfData\" \\\\  \n",
       "90717                              gatk --java-options \"-Xmx${avail_mem}M -XX:-UsePerfData\" \\\\  \n",
       "90999                              gatk --java-options \"-Xmx${avail_mem}M -XX:-UsePerfData\" \\\\  \n",
       "...                                                                                        ...  \n",
       "266291                            gatk --java-options \"-Xmx${avail_mem}g\" GermlineCNVCaller \\\\  \n",
       "266430                               gatk --java-options \"-Xmx${avail_mem}M\" SplitIntervals \\\\  \n",
       "266588                            gatk --java-options \"-Xmx${avail_mem}M\" IntervalListTools \\\\  \n",
       "266895                               gatk --java-options \"-Xmx${avail_mem}M\" SelectVariants \\\\  \n",
       "267032                                    gatk --java-options \"-Xmx${avail_mem}M\" RevertSam \\\\  \n",
       "269238                                             export _JAVA_OPTIONS=-Djava.io.tmpdir=./tmp  \n",
       "269255                                                              --java-mem-size=$memory \\\\  \n",
       "275674  java $java_mem -jar \\$mirtracejar/mirtrace.jar --mirtrace-wrapper-name mirtrace qc  \\\\  \n",
       "282762                                             export _JAVA_OPTIONS=-Djava.io.tmpdir=./tmp  \n",
       "282779                                                              --java-mem-size=$memory \\\\  \n",
       "310521                            gatk --java-options \"-Xmx${avail_mem}M\" VariantFiltration \\\\  \n",
       "310655                                              gatk --java-options \"-Xmx${avail_mem}M\" \\\\  \n",
       "310741                                      gatk --java-options \"-Xmx${avail_mem}M\" Mutect2 \\\\  \n",
       "310916                                    gatk --java-options \"-Xmx${avail_mem}M\" MergeVcfs \\\\  \n",
       "311049                                   gatk --java-options \"-Xmx${avail_mem}M\" SamToFastq \\\\  \n",
       "311188                             gatk --java-options \"-Xmx${avail_mem}M\" MergeMutectStats \\\\  \n",
       "311260                               gatk --java-options \"-Xmx${avail_mem}M\" MarkDuplicates \\\\  \n",
       "311375                                   gatk --java-options \"-Xmx${avail_mem}M\" PrintReads \\\\  \n",
       "311528                            gatk --java-options \"-Xmx${avail_mem}M\" FilterMutectCalls \\\\  \n",
       "312799                                                            --java-options \"-Xmx2000M\" \\  \n",
       "313117                            gatk --java-options \"-Xmx${avail_mem}M\" VariantFiltration \\\\  \n",
       "313293                                              gatk --java-options \"-Xmx${avail_mem}M\" \\\\  \n",
       "313380                             gatk --java-options \"-Xmx${avail_mem}M\" MergeMutectStats \\\\  \n",
       "313457                                   gatk --java-options \"-Xmx${avail_mem}M\" SamToFastq \\\\  \n",
       "313606                                   gatk --java-options \"-Xmx${avail_mem}M\" PrintReads \\\\  \n",
       "\n",
       "[148 rows x 6 columns]"
      ]
     },
     "execution_count": 193,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "java = operator_script[operator_script['operator'] == 'java']\n",
    "java"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 194,
   "id": "ff913d6e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "4970"
      ]
     },
     "execution_count": 194,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(df_operator_clustered)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 195,
   "id": "483409a1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pipeline</th>\n",
       "      <th>process_name</th>\n",
       "      <th>operator</th>\n",
       "      <th>file_path</th>\n",
       "      <th>line_number</th>\n",
       "      <th>line_content</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1836</th>\n",
       "      <td>readsimulator</td>\n",
       "      <td>MULTIQC</td>\n",
       "      <td>multiqc</td>\n",
       "      <td>./git_repos/readsimulator/modules/nf-core/multiqc/main.nf</td>\n",
       "      <td>30</td>\n",
       "      <td>multiqc \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1877</th>\n",
       "      <td>readsimulator</td>\n",
       "      <td>MULTIQC</td>\n",
       "      <td>multiqc</td>\n",
       "      <td>./git_repos/readsimulator/modules/nf-core/multiqc/main.nf</td>\n",
       "      <td>40</td>\n",
       "      <td>multiqc: \\$( multiqc --version | sed -e \"s/multiqc, version //g\" )</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1892</th>\n",
       "      <td>readsimulator</td>\n",
       "      <td>MULTIQC</td>\n",
       "      <td>multiqc</td>\n",
       "      <td>./git_repos/readsimulator/modules/nf-core/multiqc/main.nf</td>\n",
       "      <td>46</td>\n",
       "      <td>mkdir multiqc_data</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1895</th>\n",
       "      <td>readsimulator</td>\n",
       "      <td>MULTIQC</td>\n",
       "      <td>multiqc</td>\n",
       "      <td>./git_repos/readsimulator/modules/nf-core/multiqc/main.nf</td>\n",
       "      <td>47</td>\n",
       "      <td>touch multiqc_plots</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1903</th>\n",
       "      <td>readsimulator</td>\n",
       "      <td>MULTIQC</td>\n",
       "      <td>multiqc</td>\n",
       "      <td>./git_repos/readsimulator/modules/nf-core/multiqc/main.nf</td>\n",
       "      <td>48</td>\n",
       "      <td>touch multiqc_report.html</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1928</th>\n",
       "      <td>readsimulator</td>\n",
       "      <td>MULTIQC</td>\n",
       "      <td>multiqc</td>\n",
       "      <td>./git_repos/readsimulator/modules/nf-core/multiqc/main.nf</td>\n",
       "      <td>52</td>\n",
       "      <td>multiqc: \\$( multiqc --version | sed -e \"s/multiqc, version //g\" )</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7325</th>\n",
       "      <td>callingcards</td>\n",
       "      <td>MULTIQC</td>\n",
       "      <td>multiqc</td>\n",
       "      <td>./git_repos/callingcards/modules/nf-core/multiqc/main.nf</td>\n",
       "      <td>30</td>\n",
       "      <td>multiqc \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7366</th>\n",
       "      <td>callingcards</td>\n",
       "      <td>MULTIQC</td>\n",
       "      <td>multiqc</td>\n",
       "      <td>./git_repos/callingcards/modules/nf-core/multiqc/main.nf</td>\n",
       "      <td>40</td>\n",
       "      <td>multiqc: \\$( multiqc --version | sed -e \"s/multiqc, version //g\" )</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7380</th>\n",
       "      <td>callingcards</td>\n",
       "      <td>MULTIQC</td>\n",
       "      <td>multiqc</td>\n",
       "      <td>./git_repos/callingcards/modules/nf-core/multiqc/main.nf</td>\n",
       "      <td>46</td>\n",
       "      <td>touch multiqc_data</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7383</th>\n",
       "      <td>callingcards</td>\n",
       "      <td>MULTIQC</td>\n",
       "      <td>multiqc</td>\n",
       "      <td>./git_repos/callingcards/modules/nf-core/multiqc/main.nf</td>\n",
       "      <td>47</td>\n",
       "      <td>touch multiqc_plots</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7391</th>\n",
       "      <td>callingcards</td>\n",
       "      <td>MULTIQC</td>\n",
       "      <td>multiqc</td>\n",
       "      <td>./git_repos/callingcards/modules/nf-core/multiqc/main.nf</td>\n",
       "      <td>48</td>\n",
       "      <td>touch multiqc_report.html</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7416</th>\n",
       "      <td>callingcards</td>\n",
       "      <td>MULTIQC</td>\n",
       "      <td>multiqc</td>\n",
       "      <td>./git_repos/callingcards/modules/nf-core/multiqc/main.nf</td>\n",
       "      <td>52</td>\n",
       "      <td>multiqc: \\$( multiqc --version | sed -e \"s/multiqc, version //g\" )</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8481</th>\n",
       "      <td>spinningjenny</td>\n",
       "      <td>MULTIQC</td>\n",
       "      <td>multiqc</td>\n",
       "      <td>./git_repos/spinningjenny/modules/nf-core/multiqc/main.nf</td>\n",
       "      <td>29</td>\n",
       "      <td>multiqc \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8520</th>\n",
       "      <td>spinningjenny</td>\n",
       "      <td>MULTIQC</td>\n",
       "      <td>multiqc</td>\n",
       "      <td>./git_repos/spinningjenny/modules/nf-core/multiqc/main.nf</td>\n",
       "      <td>38</td>\n",
       "      <td>multiqc: \\$( multiqc --version | sed -e \"s/multiqc, version //g\" )</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8534</th>\n",
       "      <td>spinningjenny</td>\n",
       "      <td>MULTIQC</td>\n",
       "      <td>multiqc</td>\n",
       "      <td>./git_repos/spinningjenny/modules/nf-core/multiqc/main.nf</td>\n",
       "      <td>44</td>\n",
       "      <td>touch multiqc_data</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8537</th>\n",
       "      <td>spinningjenny</td>\n",
       "      <td>MULTIQC</td>\n",
       "      <td>multiqc</td>\n",
       "      <td>./git_repos/spinningjenny/modules/nf-core/multiqc/main.nf</td>\n",
       "      <td>45</td>\n",
       "      <td>touch multiqc_plots</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8545</th>\n",
       "      <td>spinningjenny</td>\n",
       "      <td>MULTIQC</td>\n",
       "      <td>multiqc</td>\n",
       "      <td>./git_repos/spinningjenny/modules/nf-core/multiqc/main.nf</td>\n",
       "      <td>46</td>\n",
       "      <td>touch multiqc_report.html</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8570</th>\n",
       "      <td>spinningjenny</td>\n",
       "      <td>MULTIQC</td>\n",
       "      <td>multiqc</td>\n",
       "      <td>./git_repos/spinningjenny/modules/nf-core/multiqc/main.nf</td>\n",
       "      <td>50</td>\n",
       "      <td>multiqc: \\$( multiqc --version | sed -e \"s/multiqc, version //g\" )</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12959</th>\n",
       "      <td>mcmicro</td>\n",
       "      <td>MULTIQC</td>\n",
       "      <td>multiqc</td>\n",
       "      <td>./git_repos/mcmicro/modules/nf-core/multiqc/main.nf</td>\n",
       "      <td>30</td>\n",
       "      <td>multiqc \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13000</th>\n",
       "      <td>mcmicro</td>\n",
       "      <td>MULTIQC</td>\n",
       "      <td>multiqc</td>\n",
       "      <td>./git_repos/mcmicro/modules/nf-core/multiqc/main.nf</td>\n",
       "      <td>40</td>\n",
       "      <td>multiqc: \\$( multiqc --version | sed -e \"s/multiqc, version //g\" )</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13014</th>\n",
       "      <td>mcmicro</td>\n",
       "      <td>MULTIQC</td>\n",
       "      <td>multiqc</td>\n",
       "      <td>./git_repos/mcmicro/modules/nf-core/multiqc/main.nf</td>\n",
       "      <td>46</td>\n",
       "      <td>touch multiqc_data</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13017</th>\n",
       "      <td>mcmicro</td>\n",
       "      <td>MULTIQC</td>\n",
       "      <td>multiqc</td>\n",
       "      <td>./git_repos/mcmicro/modules/nf-core/multiqc/main.nf</td>\n",
       "      <td>47</td>\n",
       "      <td>touch multiqc_plots</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13025</th>\n",
       "      <td>mcmicro</td>\n",
       "      <td>MULTIQC</td>\n",
       "      <td>multiqc</td>\n",
       "      <td>./git_repos/mcmicro/modules/nf-core/multiqc/main.nf</td>\n",
       "      <td>48</td>\n",
       "      <td>touch multiqc_report.html</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13050</th>\n",
       "      <td>mcmicro</td>\n",
       "      <td>MULTIQC</td>\n",
       "      <td>multiqc</td>\n",
       "      <td>./git_repos/mcmicro/modules/nf-core/multiqc/main.nf</td>\n",
       "      <td>52</td>\n",
       "      <td>multiqc: \\$( multiqc --version | sed -e \"s/multiqc, version //g\" )</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15634</th>\n",
       "      <td>methylseq</td>\n",
       "      <td>MULTIQC</td>\n",
       "      <td>multiqc</td>\n",
       "      <td>./git_repos/methylseq/modules/nf-core/multiqc/main.nf</td>\n",
       "      <td>30</td>\n",
       "      <td>multiqc \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334227</th>\n",
       "      <td>bacass</td>\n",
       "      <td>MULTIQC</td>\n",
       "      <td>multiqc</td>\n",
       "      <td>./git_repos/bacass/modules/nf-core/multiqc/main.nf</td>\n",
       "      <td>45</td>\n",
       "      <td>touch multiqc_plots</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334235</th>\n",
       "      <td>bacass</td>\n",
       "      <td>MULTIQC</td>\n",
       "      <td>multiqc</td>\n",
       "      <td>./git_repos/bacass/modules/nf-core/multiqc/main.nf</td>\n",
       "      <td>46</td>\n",
       "      <td>touch multiqc_report.html</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334260</th>\n",
       "      <td>bacass</td>\n",
       "      <td>MULTIQC</td>\n",
       "      <td>multiqc</td>\n",
       "      <td>./git_repos/bacass/modules/nf-core/multiqc/main.nf</td>\n",
       "      <td>50</td>\n",
       "      <td>multiqc: \\$( multiqc --version | sed -e \"s/multiqc, version //g\" )</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334682</th>\n",
       "      <td>molkart</td>\n",
       "      <td>MULTIQC</td>\n",
       "      <td>multiqc</td>\n",
       "      <td>./git_repos/molkart/modules/nf-core/multiqc/main.nf</td>\n",
       "      <td>29</td>\n",
       "      <td>multiqc \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334721</th>\n",
       "      <td>molkart</td>\n",
       "      <td>MULTIQC</td>\n",
       "      <td>multiqc</td>\n",
       "      <td>./git_repos/molkart/modules/nf-core/multiqc/main.nf</td>\n",
       "      <td>38</td>\n",
       "      <td>multiqc: \\$( multiqc --version | sed -e \"s/multiqc, version //g\" )</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334735</th>\n",
       "      <td>molkart</td>\n",
       "      <td>MULTIQC</td>\n",
       "      <td>multiqc</td>\n",
       "      <td>./git_repos/molkart/modules/nf-core/multiqc/main.nf</td>\n",
       "      <td>44</td>\n",
       "      <td>touch multiqc_data</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334738</th>\n",
       "      <td>molkart</td>\n",
       "      <td>MULTIQC</td>\n",
       "      <td>multiqc</td>\n",
       "      <td>./git_repos/molkart/modules/nf-core/multiqc/main.nf</td>\n",
       "      <td>45</td>\n",
       "      <td>touch multiqc_plots</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334746</th>\n",
       "      <td>molkart</td>\n",
       "      <td>MULTIQC</td>\n",
       "      <td>multiqc</td>\n",
       "      <td>./git_repos/molkart/modules/nf-core/multiqc/main.nf</td>\n",
       "      <td>46</td>\n",
       "      <td>touch multiqc_report.html</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334771</th>\n",
       "      <td>molkart</td>\n",
       "      <td>MULTIQC</td>\n",
       "      <td>multiqc</td>\n",
       "      <td>./git_repos/molkart/modules/nf-core/multiqc/main.nf</td>\n",
       "      <td>50</td>\n",
       "      <td>multiqc: \\$( multiqc --version | sed -e \"s/multiqc, version //g\" )</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>335050</th>\n",
       "      <td>liverctanalysis</td>\n",
       "      <td>MULTIQC</td>\n",
       "      <td>multiqc</td>\n",
       "      <td>./git_repos/liverctanalysis/modules/nf-core/modules/multiqc/main.nf</td>\n",
       "      <td>32</td>\n",
       "      <td>multiqc -f $options.args .</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>335060</th>\n",
       "      <td>liverctanalysis</td>\n",
       "      <td>MULTIQC</td>\n",
       "      <td>multiqc</td>\n",
       "      <td>./git_repos/liverctanalysis/modules/nf-core/modules/multiqc/main.nf</td>\n",
       "      <td>33</td>\n",
       "      <td>multiqc --version | sed -e \"s/multiqc, version //g\" &gt; ${software}.version.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>335275</th>\n",
       "      <td>demultiplex</td>\n",
       "      <td>MULTIQC</td>\n",
       "      <td>multiqc</td>\n",
       "      <td>./git_repos/demultiplex/modules/nf-core/multiqc/main.nf</td>\n",
       "      <td>30</td>\n",
       "      <td>multiqc \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>335316</th>\n",
       "      <td>demultiplex</td>\n",
       "      <td>MULTIQC</td>\n",
       "      <td>multiqc</td>\n",
       "      <td>./git_repos/demultiplex/modules/nf-core/multiqc/main.nf</td>\n",
       "      <td>40</td>\n",
       "      <td>multiqc: \\$( multiqc --version | sed -e \"s/multiqc, version //g\" )</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>335330</th>\n",
       "      <td>demultiplex</td>\n",
       "      <td>MULTIQC</td>\n",
       "      <td>multiqc</td>\n",
       "      <td>./git_repos/demultiplex/modules/nf-core/multiqc/main.nf</td>\n",
       "      <td>46</td>\n",
       "      <td>touch multiqc_data</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>335333</th>\n",
       "      <td>demultiplex</td>\n",
       "      <td>MULTIQC</td>\n",
       "      <td>multiqc</td>\n",
       "      <td>./git_repos/demultiplex/modules/nf-core/multiqc/main.nf</td>\n",
       "      <td>47</td>\n",
       "      <td>touch multiqc_plots</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>335341</th>\n",
       "      <td>demultiplex</td>\n",
       "      <td>MULTIQC</td>\n",
       "      <td>multiqc</td>\n",
       "      <td>./git_repos/demultiplex/modules/nf-core/multiqc/main.nf</td>\n",
       "      <td>48</td>\n",
       "      <td>touch multiqc_report.html</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>335366</th>\n",
       "      <td>demultiplex</td>\n",
       "      <td>MULTIQC</td>\n",
       "      <td>multiqc</td>\n",
       "      <td>./git_repos/demultiplex/modules/nf-core/multiqc/main.nf</td>\n",
       "      <td>52</td>\n",
       "      <td>multiqc: \\$( multiqc --version | sed -e \"s/multiqc, version //g\" )</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>342860</th>\n",
       "      <td>metatdenovo</td>\n",
       "      <td>MULTIQC</td>\n",
       "      <td>multiqc</td>\n",
       "      <td>./git_repos/metatdenovo/modules/nf-core/multiqc/main.nf</td>\n",
       "      <td>30</td>\n",
       "      <td>multiqc \\\\</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>342901</th>\n",
       "      <td>metatdenovo</td>\n",
       "      <td>MULTIQC</td>\n",
       "      <td>multiqc</td>\n",
       "      <td>./git_repos/metatdenovo/modules/nf-core/multiqc/main.nf</td>\n",
       "      <td>40</td>\n",
       "      <td>multiqc: \\$( multiqc --version | sed -e \"s/multiqc, version //g\" )</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>342916</th>\n",
       "      <td>metatdenovo</td>\n",
       "      <td>MULTIQC</td>\n",
       "      <td>multiqc</td>\n",
       "      <td>./git_repos/metatdenovo/modules/nf-core/multiqc/main.nf</td>\n",
       "      <td>46</td>\n",
       "      <td>mkdir multiqc_data</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>342919</th>\n",
       "      <td>metatdenovo</td>\n",
       "      <td>MULTIQC</td>\n",
       "      <td>multiqc</td>\n",
       "      <td>./git_repos/metatdenovo/modules/nf-core/multiqc/main.nf</td>\n",
       "      <td>47</td>\n",
       "      <td>touch multiqc_plots</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>342927</th>\n",
       "      <td>metatdenovo</td>\n",
       "      <td>MULTIQC</td>\n",
       "      <td>multiqc</td>\n",
       "      <td>./git_repos/metatdenovo/modules/nf-core/multiqc/main.nf</td>\n",
       "      <td>48</td>\n",
       "      <td>touch multiqc_report.html</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>342952</th>\n",
       "      <td>metatdenovo</td>\n",
       "      <td>MULTIQC</td>\n",
       "      <td>multiqc</td>\n",
       "      <td>./git_repos/metatdenovo/modules/nf-core/multiqc/main.nf</td>\n",
       "      <td>52</td>\n",
       "      <td>multiqc: \\$( multiqc --version | sed -e \"s/multiqc, version //g\" )</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>343370</th>\n",
       "      <td>hic</td>\n",
       "      <td>MULTIQC</td>\n",
       "      <td>multiqc</td>\n",
       "      <td>./git_repos/hic/modules/local/multiqc.nf</td>\n",
       "      <td>28</td>\n",
       "      <td>multiqc -f $args .</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>343393</th>\n",
       "      <td>hic</td>\n",
       "      <td>MULTIQC</td>\n",
       "      <td>multiqc</td>\n",
       "      <td>./git_repos/hic/modules/local/multiqc.nf</td>\n",
       "      <td>32</td>\n",
       "      <td>multiqc: \\$( multiqc --version | sed -e \"s/multiqc, version //g\" )</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>447 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               pipeline process_name operator  \\\n",
       "1836      readsimulator      MULTIQC  multiqc   \n",
       "1877      readsimulator      MULTIQC  multiqc   \n",
       "1892      readsimulator      MULTIQC  multiqc   \n",
       "1895      readsimulator      MULTIQC  multiqc   \n",
       "1903      readsimulator      MULTIQC  multiqc   \n",
       "1928      readsimulator      MULTIQC  multiqc   \n",
       "7325       callingcards      MULTIQC  multiqc   \n",
       "7366       callingcards      MULTIQC  multiqc   \n",
       "7380       callingcards      MULTIQC  multiqc   \n",
       "7383       callingcards      MULTIQC  multiqc   \n",
       "7391       callingcards      MULTIQC  multiqc   \n",
       "7416       callingcards      MULTIQC  multiqc   \n",
       "8481      spinningjenny      MULTIQC  multiqc   \n",
       "8520      spinningjenny      MULTIQC  multiqc   \n",
       "8534      spinningjenny      MULTIQC  multiqc   \n",
       "8537      spinningjenny      MULTIQC  multiqc   \n",
       "8545      spinningjenny      MULTIQC  multiqc   \n",
       "8570      spinningjenny      MULTIQC  multiqc   \n",
       "12959           mcmicro      MULTIQC  multiqc   \n",
       "13000           mcmicro      MULTIQC  multiqc   \n",
       "13014           mcmicro      MULTIQC  multiqc   \n",
       "13017           mcmicro      MULTIQC  multiqc   \n",
       "13025           mcmicro      MULTIQC  multiqc   \n",
       "13050           mcmicro      MULTIQC  multiqc   \n",
       "15634         methylseq      MULTIQC  multiqc   \n",
       "...                 ...          ...      ...   \n",
       "334227           bacass      MULTIQC  multiqc   \n",
       "334235           bacass      MULTIQC  multiqc   \n",
       "334260           bacass      MULTIQC  multiqc   \n",
       "334682          molkart      MULTIQC  multiqc   \n",
       "334721          molkart      MULTIQC  multiqc   \n",
       "334735          molkart      MULTIQC  multiqc   \n",
       "334738          molkart      MULTIQC  multiqc   \n",
       "334746          molkart      MULTIQC  multiqc   \n",
       "334771          molkart      MULTIQC  multiqc   \n",
       "335050  liverctanalysis      MULTIQC  multiqc   \n",
       "335060  liverctanalysis      MULTIQC  multiqc   \n",
       "335275      demultiplex      MULTIQC  multiqc   \n",
       "335316      demultiplex      MULTIQC  multiqc   \n",
       "335330      demultiplex      MULTIQC  multiqc   \n",
       "335333      demultiplex      MULTIQC  multiqc   \n",
       "335341      demultiplex      MULTIQC  multiqc   \n",
       "335366      demultiplex      MULTIQC  multiqc   \n",
       "342860      metatdenovo      MULTIQC  multiqc   \n",
       "342901      metatdenovo      MULTIQC  multiqc   \n",
       "342916      metatdenovo      MULTIQC  multiqc   \n",
       "342919      metatdenovo      MULTIQC  multiqc   \n",
       "342927      metatdenovo      MULTIQC  multiqc   \n",
       "342952      metatdenovo      MULTIQC  multiqc   \n",
       "343370              hic      MULTIQC  multiqc   \n",
       "343393              hic      MULTIQC  multiqc   \n",
       "\n",
       "                                                                  file_path  \\\n",
       "1836              ./git_repos/readsimulator/modules/nf-core/multiqc/main.nf   \n",
       "1877              ./git_repos/readsimulator/modules/nf-core/multiqc/main.nf   \n",
       "1892              ./git_repos/readsimulator/modules/nf-core/multiqc/main.nf   \n",
       "1895              ./git_repos/readsimulator/modules/nf-core/multiqc/main.nf   \n",
       "1903              ./git_repos/readsimulator/modules/nf-core/multiqc/main.nf   \n",
       "1928              ./git_repos/readsimulator/modules/nf-core/multiqc/main.nf   \n",
       "7325               ./git_repos/callingcards/modules/nf-core/multiqc/main.nf   \n",
       "7366               ./git_repos/callingcards/modules/nf-core/multiqc/main.nf   \n",
       "7380               ./git_repos/callingcards/modules/nf-core/multiqc/main.nf   \n",
       "7383               ./git_repos/callingcards/modules/nf-core/multiqc/main.nf   \n",
       "7391               ./git_repos/callingcards/modules/nf-core/multiqc/main.nf   \n",
       "7416               ./git_repos/callingcards/modules/nf-core/multiqc/main.nf   \n",
       "8481              ./git_repos/spinningjenny/modules/nf-core/multiqc/main.nf   \n",
       "8520              ./git_repos/spinningjenny/modules/nf-core/multiqc/main.nf   \n",
       "8534              ./git_repos/spinningjenny/modules/nf-core/multiqc/main.nf   \n",
       "8537              ./git_repos/spinningjenny/modules/nf-core/multiqc/main.nf   \n",
       "8545              ./git_repos/spinningjenny/modules/nf-core/multiqc/main.nf   \n",
       "8570              ./git_repos/spinningjenny/modules/nf-core/multiqc/main.nf   \n",
       "12959                   ./git_repos/mcmicro/modules/nf-core/multiqc/main.nf   \n",
       "13000                   ./git_repos/mcmicro/modules/nf-core/multiqc/main.nf   \n",
       "13014                   ./git_repos/mcmicro/modules/nf-core/multiqc/main.nf   \n",
       "13017                   ./git_repos/mcmicro/modules/nf-core/multiqc/main.nf   \n",
       "13025                   ./git_repos/mcmicro/modules/nf-core/multiqc/main.nf   \n",
       "13050                   ./git_repos/mcmicro/modules/nf-core/multiqc/main.nf   \n",
       "15634                 ./git_repos/methylseq/modules/nf-core/multiqc/main.nf   \n",
       "...                                                                     ...   \n",
       "334227                   ./git_repos/bacass/modules/nf-core/multiqc/main.nf   \n",
       "334235                   ./git_repos/bacass/modules/nf-core/multiqc/main.nf   \n",
       "334260                   ./git_repos/bacass/modules/nf-core/multiqc/main.nf   \n",
       "334682                  ./git_repos/molkart/modules/nf-core/multiqc/main.nf   \n",
       "334721                  ./git_repos/molkart/modules/nf-core/multiqc/main.nf   \n",
       "334735                  ./git_repos/molkart/modules/nf-core/multiqc/main.nf   \n",
       "334738                  ./git_repos/molkart/modules/nf-core/multiqc/main.nf   \n",
       "334746                  ./git_repos/molkart/modules/nf-core/multiqc/main.nf   \n",
       "334771                  ./git_repos/molkart/modules/nf-core/multiqc/main.nf   \n",
       "335050  ./git_repos/liverctanalysis/modules/nf-core/modules/multiqc/main.nf   \n",
       "335060  ./git_repos/liverctanalysis/modules/nf-core/modules/multiqc/main.nf   \n",
       "335275              ./git_repos/demultiplex/modules/nf-core/multiqc/main.nf   \n",
       "335316              ./git_repos/demultiplex/modules/nf-core/multiqc/main.nf   \n",
       "335330              ./git_repos/demultiplex/modules/nf-core/multiqc/main.nf   \n",
       "335333              ./git_repos/demultiplex/modules/nf-core/multiqc/main.nf   \n",
       "335341              ./git_repos/demultiplex/modules/nf-core/multiqc/main.nf   \n",
       "335366              ./git_repos/demultiplex/modules/nf-core/multiqc/main.nf   \n",
       "342860              ./git_repos/metatdenovo/modules/nf-core/multiqc/main.nf   \n",
       "342901              ./git_repos/metatdenovo/modules/nf-core/multiqc/main.nf   \n",
       "342916              ./git_repos/metatdenovo/modules/nf-core/multiqc/main.nf   \n",
       "342919              ./git_repos/metatdenovo/modules/nf-core/multiqc/main.nf   \n",
       "342927              ./git_repos/metatdenovo/modules/nf-core/multiqc/main.nf   \n",
       "342952              ./git_repos/metatdenovo/modules/nf-core/multiqc/main.nf   \n",
       "343370                             ./git_repos/hic/modules/local/multiqc.nf   \n",
       "343393                             ./git_repos/hic/modules/local/multiqc.nf   \n",
       "\n",
       "        line_number  \\\n",
       "1836             30   \n",
       "1877             40   \n",
       "1892             46   \n",
       "1895             47   \n",
       "1903             48   \n",
       "1928             52   \n",
       "7325             30   \n",
       "7366             40   \n",
       "7380             46   \n",
       "7383             47   \n",
       "7391             48   \n",
       "7416             52   \n",
       "8481             29   \n",
       "8520             38   \n",
       "8534             44   \n",
       "8537             45   \n",
       "8545             46   \n",
       "8570             50   \n",
       "12959            30   \n",
       "13000            40   \n",
       "13014            46   \n",
       "13017            47   \n",
       "13025            48   \n",
       "13050            52   \n",
       "15634            30   \n",
       "...             ...   \n",
       "334227           45   \n",
       "334235           46   \n",
       "334260           50   \n",
       "334682           29   \n",
       "334721           38   \n",
       "334735           44   \n",
       "334738           45   \n",
       "334746           46   \n",
       "334771           50   \n",
       "335050           32   \n",
       "335060           33   \n",
       "335275           30   \n",
       "335316           40   \n",
       "335330           46   \n",
       "335333           47   \n",
       "335341           48   \n",
       "335366           52   \n",
       "342860           30   \n",
       "342901           40   \n",
       "342916           46   \n",
       "342919           47   \n",
       "342927           48   \n",
       "342952           52   \n",
       "343370           28   \n",
       "343393           32   \n",
       "\n",
       "                                                                         line_content  \n",
       "1836                                                                       multiqc \\\\  \n",
       "1877               multiqc: \\$( multiqc --version | sed -e \"s/multiqc, version //g\" )  \n",
       "1892                                                               mkdir multiqc_data  \n",
       "1895                                                              touch multiqc_plots  \n",
       "1903                                                        touch multiqc_report.html  \n",
       "1928               multiqc: \\$( multiqc --version | sed -e \"s/multiqc, version //g\" )  \n",
       "7325                                                                       multiqc \\\\  \n",
       "7366               multiqc: \\$( multiqc --version | sed -e \"s/multiqc, version //g\" )  \n",
       "7380                                                               touch multiqc_data  \n",
       "7383                                                              touch multiqc_plots  \n",
       "7391                                                        touch multiqc_report.html  \n",
       "7416               multiqc: \\$( multiqc --version | sed -e \"s/multiqc, version //g\" )  \n",
       "8481                                                                       multiqc \\\\  \n",
       "8520               multiqc: \\$( multiqc --version | sed -e \"s/multiqc, version //g\" )  \n",
       "8534                                                               touch multiqc_data  \n",
       "8537                                                              touch multiqc_plots  \n",
       "8545                                                        touch multiqc_report.html  \n",
       "8570               multiqc: \\$( multiqc --version | sed -e \"s/multiqc, version //g\" )  \n",
       "12959                                                                      multiqc \\\\  \n",
       "13000              multiqc: \\$( multiqc --version | sed -e \"s/multiqc, version //g\" )  \n",
       "13014                                                              touch multiqc_data  \n",
       "13017                                                             touch multiqc_plots  \n",
       "13025                                                       touch multiqc_report.html  \n",
       "13050              multiqc: \\$( multiqc --version | sed -e \"s/multiqc, version //g\" )  \n",
       "15634                                                                      multiqc \\\\  \n",
       "...                                                                               ...  \n",
       "334227                                                            touch multiqc_plots  \n",
       "334235                                                      touch multiqc_report.html  \n",
       "334260             multiqc: \\$( multiqc --version | sed -e \"s/multiqc, version //g\" )  \n",
       "334682                                                                     multiqc \\\\  \n",
       "334721             multiqc: \\$( multiqc --version | sed -e \"s/multiqc, version //g\" )  \n",
       "334735                                                             touch multiqc_data  \n",
       "334738                                                            touch multiqc_plots  \n",
       "334746                                                      touch multiqc_report.html  \n",
       "334771             multiqc: \\$( multiqc --version | sed -e \"s/multiqc, version //g\" )  \n",
       "335050                                                     multiqc -f $options.args .  \n",
       "335060  multiqc --version | sed -e \"s/multiqc, version //g\" > ${software}.version.txt  \n",
       "335275                                                                     multiqc \\\\  \n",
       "335316             multiqc: \\$( multiqc --version | sed -e \"s/multiqc, version //g\" )  \n",
       "335330                                                             touch multiqc_data  \n",
       "335333                                                            touch multiqc_plots  \n",
       "335341                                                      touch multiqc_report.html  \n",
       "335366             multiqc: \\$( multiqc --version | sed -e \"s/multiqc, version //g\" )  \n",
       "342860                                                                     multiqc \\\\  \n",
       "342901             multiqc: \\$( multiqc --version | sed -e \"s/multiqc, version //g\" )  \n",
       "342916                                                             mkdir multiqc_data  \n",
       "342919                                                            touch multiqc_plots  \n",
       "342927                                                      touch multiqc_report.html  \n",
       "342952             multiqc: \\$( multiqc --version | sed -e \"s/multiqc, version //g\" )  \n",
       "343370                                                             multiqc -f $args .  \n",
       "343393             multiqc: \\$( multiqc --version | sed -e \"s/multiqc, version //g\" )  \n",
       "\n",
       "[447 rows x 6 columns]"
      ]
     },
     "execution_count": 195,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "multiqc = operator_script[operator_script['operator'] == 'multiqc']\n",
    "pd.set_option('display.max_colwidth', None)\n",
    "multiqc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 196,
   "id": "d3cbd4f9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>operator</th>\n",
       "      <th>Assigned Group</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>{log}</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>{output}</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>samtools</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>echo</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>shell:</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>python</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>mkdir</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>&amp;1</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>rm</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>bcftools</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>mv</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>cut</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>set</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>cat</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>sed</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>awk</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>if</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>fi</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>grep</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>Rscript</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>cp</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>cd</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>bedtools</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>{input}</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>touch</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4945</th>\n",
       "      <td>{params.base_dir}/{wildcards.sample}.ribosomes.fastq</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4946</th>\n",
       "      <td>INDIR=$dirname</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4947</th>\n",
       "      <td>{params.export_dir}/without_names/humann2_${{group}}_unstratified.spf</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4948</th>\n",
       "      <td>{params.export_dir}/with_names/humann2_${{group}}_named_unstratified.spf</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4949</th>\n",
       "      <td>{params.export_dir}/without_names/humann2_${{group}}_relab_unstratified.spf</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4950</th>\n",
       "      <td>{params.export_dir}/with_names/humann2_${{group}}_named_relab_unstratified.spf</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4951</th>\n",
       "      <td>{input.isoforms}</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4952</th>\n",
       "      <td>{params.database}</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4953</th>\n",
       "      <td>{params.parallel}</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4954</th>\n",
       "      <td>init</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4955</th>\n",
       "      <td>{input.swissprot_blastp}</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4956</th>\n",
       "      <td>{input.swissprot_blastx}</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4957</th>\n",
       "      <td>LOAD_rnammer</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4958</th>\n",
       "      <td>{input.rRNA}</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4959</th>\n",
       "      <td>report</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4960</th>\n",
       "      <td>{output.annotation_table}</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4961</th>\n",
       "      <td>rwc</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4962</th>\n",
       "      <td>sizes.txt</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4963</th>\n",
       "      <td>get_parent_dir={input.sa};</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4964</th>\n",
       "      <td>drop</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4965</th>\n",
       "      <td>{output.done}</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4966</th>\n",
       "      <td>format1=$echo</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4967</th>\n",
       "      <td>format2=$echo</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4968</th>\n",
       "      <td>tmpdir=$mktemp</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4969</th>\n",
       "      <td>{input.star_aln}</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4970 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                            operator  \\\n",
       "0                                                                              {log}   \n",
       "1                                                                           {output}   \n",
       "2                                                                           samtools   \n",
       "3                                                                               echo   \n",
       "4                                                                             shell:   \n",
       "5                                                                             python   \n",
       "6                                                                              mkdir   \n",
       "7                                                                                 &1   \n",
       "8                                                                                 rm   \n",
       "9                                                                           bcftools   \n",
       "10                                                                                mv   \n",
       "11                                                                               cut   \n",
       "12                                                                               set   \n",
       "13                                                                               cat   \n",
       "14                                                                               sed   \n",
       "15                                                                               awk   \n",
       "16                                                                                if   \n",
       "17                                                                                fi   \n",
       "18                                                                              grep   \n",
       "19                                                                           Rscript   \n",
       "20                                                                                cp   \n",
       "21                                                                                cd   \n",
       "22                                                                          bedtools   \n",
       "23                                                                           {input}   \n",
       "24                                                                             touch   \n",
       "...                                                                              ...   \n",
       "4945                            {params.base_dir}/{wildcards.sample}.ribosomes.fastq   \n",
       "4946                                                                  INDIR=$dirname   \n",
       "4947           {params.export_dir}/without_names/humann2_${{group}}_unstratified.spf   \n",
       "4948        {params.export_dir}/with_names/humann2_${{group}}_named_unstratified.spf   \n",
       "4949     {params.export_dir}/without_names/humann2_${{group}}_relab_unstratified.spf   \n",
       "4950  {params.export_dir}/with_names/humann2_${{group}}_named_relab_unstratified.spf   \n",
       "4951                                                                {input.isoforms}   \n",
       "4952                                                               {params.database}   \n",
       "4953                                                               {params.parallel}   \n",
       "4954                                                                            init   \n",
       "4955                                                        {input.swissprot_blastp}   \n",
       "4956                                                        {input.swissprot_blastx}   \n",
       "4957                                                                    LOAD_rnammer   \n",
       "4958                                                                    {input.rRNA}   \n",
       "4959                                                                          report   \n",
       "4960                                                       {output.annotation_table}   \n",
       "4961                                                                             rwc   \n",
       "4962                                                                       sizes.txt   \n",
       "4963                                                      get_parent_dir={input.sa};   \n",
       "4964                                                                            drop   \n",
       "4965                                                                   {output.done}   \n",
       "4966                                                                   format1=$echo   \n",
       "4967                                                                   format2=$echo   \n",
       "4968                                                                  tmpdir=$mktemp   \n",
       "4969                                                                {input.star_aln}   \n",
       "\n",
       "      Assigned Group  \n",
       "0                  6  \n",
       "1                  6  \n",
       "2                  0  \n",
       "3                  2  \n",
       "4                  4  \n",
       "5                  4  \n",
       "6                  4  \n",
       "7                  6  \n",
       "8                  3  \n",
       "9                  0  \n",
       "10                 3  \n",
       "11                 2  \n",
       "12                 6  \n",
       "13                 3  \n",
       "14                 6  \n",
       "15                 2  \n",
       "16                 1  \n",
       "17                 1  \n",
       "18                 2  \n",
       "19                 4  \n",
       "20                 3  \n",
       "21                 3  \n",
       "22                 0  \n",
       "23                 6  \n",
       "24                 6  \n",
       "...              ...  \n",
       "4945               0  \n",
       "4946               1  \n",
       "4947               1  \n",
       "4948               1  \n",
       "4949               1  \n",
       "4950               1  \n",
       "4951               3  \n",
       "4952               2  \n",
       "4953               2  \n",
       "4954               6  \n",
       "4955               4  \n",
       "4956               4  \n",
       "4957               4  \n",
       "4958               4  \n",
       "4959               4  \n",
       "4960               6  \n",
       "4961               4  \n",
       "4962               6  \n",
       "4963               1  \n",
       "4964               4  \n",
       "4965               1  \n",
       "4966               2  \n",
       "4967               2  \n",
       "4968               1  \n",
       "4969               4  \n",
       "\n",
       "[4970 rows x 2 columns]"
      ]
     },
     "execution_count": 196,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_operator_clustered"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 197,
   "id": "0e51ae7e",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_operator_clustered = df_operator_clustered.rename(columns={'Operator': 'operator'})\n",
    "\n",
    "merged_df = pd.merge(operator_script, df_operator_clustered, on='operator')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 198,
   "id": "c750294c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pipeline</th>\n",
       "      <th>process_name</th>\n",
       "      <th>operator</th>\n",
       "      <th>file_path</th>\n",
       "      <th>line_number</th>\n",
       "      <th>line_content</th>\n",
       "      <th>Assigned Group</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>diaproteomics</td>\n",
       "      <td>get_software_versions</td>\n",
       "      <td>echo</td>\n",
       "      <td>./git_repos/diaproteomics/main.nf</td>\n",
       "      <td>335</td>\n",
       "      <td>echo $workflow.manifest.version &gt; v_pipeline.txt</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>diaproteomics</td>\n",
       "      <td>get_software_versions</td>\n",
       "      <td>echo</td>\n",
       "      <td>./git_repos/diaproteomics/main.nf</td>\n",
       "      <td>336</td>\n",
       "      <td>echo $workflow.nextflow.version &gt; v_nextflow.txt</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>diaproteomics</td>\n",
       "      <td>statistical_post_processing</td>\n",
       "      <td>echo</td>\n",
       "      <td>./git_repos/diaproteomics/main.nf</td>\n",
       "      <td>971</td>\n",
       "      <td>msstats.R &gt; msstats.log || echo \"Optional MSstats step failed. Please check logs and re-run or do a manual statistical analysis.\"</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>readsimulator</td>\n",
       "      <td>BOWTIE2_ALIGN</td>\n",
       "      <td>echo</td>\n",
       "      <td>./git_repos/readsimulator/modules/nf-core/bowtie2/align/main.nf</td>\n",
       "      <td>47</td>\n",
       "      <td>[ -z \"\\$INDEX\" ] &amp;&amp; echo \"Bowtie2 index files not found\" 1&gt;&amp;2 &amp;&amp; exit 1</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>readsimulator</td>\n",
       "      <td>BOWTIE2_ALIGN</td>\n",
       "      <td>echo</td>\n",
       "      <td>./git_repos/readsimulator/modules/nf-core/bowtie2/align/main.nf</td>\n",
       "      <td>68</td>\n",
       "      <td>bowtie2: \\$(echo \\$(bowtie2 --version 2&gt;&amp;1) | sed 's/^.*bowtie2-align-s version //; s/ .*\\$//')</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>readsimulator</td>\n",
       "      <td>BOWTIE2_ALIGN</td>\n",
       "      <td>echo</td>\n",
       "      <td>./git_repos/readsimulator/modules/nf-core/bowtie2/align/main.nf</td>\n",
       "      <td>69</td>\n",
       "      <td>samtools: \\$(echo \\$(samtools --version 2&gt;&amp;1) | sed 's/^.*samtools //; s/Using.*\\$//')</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>readsimulator</td>\n",
       "      <td>BOWTIE2_ALIGN</td>\n",
       "      <td>echo</td>\n",
       "      <td>./git_repos/readsimulator/modules/nf-core/bowtie2/align/main.nf</td>\n",
       "      <td>88</td>\n",
       "      <td>bowtie2: \\$(echo \\$(bowtie2 --version 2&gt;&amp;1) | sed 's/^.*bowtie2-align-s version //; s/ .*\\$//')</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>readsimulator</td>\n",
       "      <td>BOWTIE2_ALIGN</td>\n",
       "      <td>echo</td>\n",
       "      <td>./git_repos/readsimulator/modules/nf-core/bowtie2/align/main.nf</td>\n",
       "      <td>89</td>\n",
       "      <td>samtools: \\$(echo \\$(samtools --version 2&gt;&amp;1) | sed 's/^.*samtools //; s/Using.*\\$//')</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>readsimulator</td>\n",
       "      <td>BOWTIE2_BUILD</td>\n",
       "      <td>echo</td>\n",
       "      <td>./git_repos/readsimulator/modules/nf-core/bowtie2/build/main.nf</td>\n",
       "      <td>27</td>\n",
       "      <td>bowtie2: \\$(echo \\$(bowtie2 --version 2&gt;&amp;1) | sed 's/^.*bowtie2-align-s version //; s/ .*\\$//')</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>readsimulator</td>\n",
       "      <td>BOWTIE2_BUILD</td>\n",
       "      <td>echo</td>\n",
       "      <td>./git_repos/readsimulator/modules/nf-core/bowtie2/build/main.nf</td>\n",
       "      <td>39</td>\n",
       "      <td>bowtie2: \\$(echo \\$(bowtie2 --version 2&gt;&amp;1) | sed 's/^.*bowtie2-align-s version //; s/ .*\\$//')</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>readsimulator</td>\n",
       "      <td>SAMTOOLS_INDEX</td>\n",
       "      <td>echo</td>\n",
       "      <td>./git_repos/readsimulator/modules/nf-core/samtools/index/main.nf</td>\n",
       "      <td>33</td>\n",
       "      <td>samtools: \\$(echo \\$(samtools --version 2&gt;&amp;1) | sed 's/^.*samtools //; s/Using.*\\$//')</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>readsimulator</td>\n",
       "      <td>SAMTOOLS_INDEX</td>\n",
       "      <td>echo</td>\n",
       "      <td>./git_repos/readsimulator/modules/nf-core/samtools/index/main.nf</td>\n",
       "      <td>45</td>\n",
       "      <td>samtools: \\$(echo \\$(samtools --version 2&gt;&amp;1) | sed 's/^.*samtools //; s/Using.*\\$//')</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>readsimulator</td>\n",
       "      <td>INSILICOSEQ_GENERATE</td>\n",
       "      <td>echo</td>\n",
       "      <td>./git_repos/readsimulator/modules/local/insilicoseq/generate/main.nf</td>\n",
       "      <td>29</td>\n",
       "      <td>seed=\\$(echo $seed | sed 's/\\\\[//g' | sed 's/\\\\]//g')</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>readsimulator</td>\n",
       "      <td>INSILICOSEQ_GENERATE</td>\n",
       "      <td>echo</td>\n",
       "      <td>./git_repos/readsimulator/modules/local/insilicoseq/generate/main.nf</td>\n",
       "      <td>30</td>\n",
       "      <td>prefix=\\$(echo $prefix | sed 's/\\\\[//g' | sed 's/\\\\]//g')</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>readsimulator</td>\n",
       "      <td>INSILICOSEQ_GENERATE</td>\n",
       "      <td>echo</td>\n",
       "      <td>./git_repos/readsimulator/modules/local/insilicoseq/generate/main.nf</td>\n",
       "      <td>53</td>\n",
       "      <td>seed=\\$(echo $seed | sed 's/\\\\[//g' | sed 's/\\\\]//g')</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>readsimulator</td>\n",
       "      <td>INSILICOSEQ_GENERATE</td>\n",
       "      <td>echo</td>\n",
       "      <td>./git_repos/readsimulator/modules/local/insilicoseq/generate/main.nf</td>\n",
       "      <td>54</td>\n",
       "      <td>prefix=\\$(echo $prefix | sed 's/\\\\[//g' | sed 's/\\\\]//g')</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>readsimulator</td>\n",
       "      <td>MERGE_SAMPLESHEETS</td>\n",
       "      <td>echo</td>\n",
       "      <td>./git_repos/readsimulator/modules/local/custom/merge_samplesheets/main.nf</td>\n",
       "      <td>23</td>\n",
       "      <td>echo \\\\\"sample\\\\\",\\\\\"fastq_1\\\\\",\\\\\"fastq_2\\\\\" &gt; \"${prefix}_samplesheet.csv\"</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>readsimulator</td>\n",
       "      <td>MERGE_SAMPLESHEETS</td>\n",
       "      <td>echo</td>\n",
       "      <td>./git_repos/readsimulator/modules/local/custom/merge_samplesheets/main.nf</td>\n",
       "      <td>26</td>\n",
       "      <td>echo &gt;&gt; \"${prefix}_samplesheet.csv\"</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>callingcards</td>\n",
       "      <td>GTF2BED</td>\n",
       "      <td>echo</td>\n",
       "      <td>./git_repos/callingcards/modules/local/gtf2bed/main.nf</td>\n",
       "      <td>28</td>\n",
       "      <td>perl: \\$(echo \\$(perl --version 2&gt;&amp;1) | sed 's/.*v\\\\(.*\\\\)) built.*/\\\\1/')</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>callingcards</td>\n",
       "      <td>DEMULTIPLEX</td>\n",
       "      <td>echo</td>\n",
       "      <td>./git_repos/callingcards/modules/local/callingcardstools/yeast/demultiplex/main.nf</td>\n",
       "      <td>36</td>\n",
       "      <td>callingcardstools: \\$(echo \\$(callingcardstools --version 2&gt;&amp;1) | sed 's/.* //')</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>callingcards</td>\n",
       "      <td>COUNT_HOPS</td>\n",
       "      <td>echo</td>\n",
       "      <td>./git_repos/callingcards/modules/local/callingcardstools/yeast/count_hops/main.nf</td>\n",
       "      <td>39</td>\n",
       "      <td>callingcardstools: \\$(echo \\$(callingcardstools --version 2&gt;&amp;1) | sed 's/.* //')</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>callingcards</td>\n",
       "      <td>CONCATQC</td>\n",
       "      <td>echo</td>\n",
       "      <td>./git_repos/callingcards/modules/local/callingcardstools/yeast/concatQC/main.nf</td>\n",
       "      <td>33</td>\n",
       "      <td>callingcardstools: \\$(echo \\$(callingcardstools --version 2&gt;&amp;1) | sed 's/.* //')</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>callingcards</td>\n",
       "      <td>CONCATQC</td>\n",
       "      <td>echo</td>\n",
       "      <td>./git_repos/callingcards/modules/local/callingcardstools/mammals/concatQC/main.nf</td>\n",
       "      <td>38</td>\n",
       "      <td>callingcardstools: \\$(echo \\$(callingcardstools --version 2&gt;&amp;1) | sed 's/.* //')</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>callingcards</td>\n",
       "      <td>BOWTIE2_ALIGN</td>\n",
       "      <td>echo</td>\n",
       "      <td>./git_repos/callingcards/modules/nf-core/bowtie2/align/main.nf</td>\n",
       "      <td>47</td>\n",
       "      <td>[ -z \"\\$INDEX\" ] &amp;&amp; echo \"Bowtie2 index files not found\" 1&gt;&amp;2 &amp;&amp; exit 1</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>callingcards</td>\n",
       "      <td>BOWTIE2_ALIGN</td>\n",
       "      <td>echo</td>\n",
       "      <td>./git_repos/callingcards/modules/nf-core/bowtie2/align/main.nf</td>\n",
       "      <td>68</td>\n",
       "      <td>bowtie2: \\$(echo \\$(bowtie2 --version 2&gt;&amp;1) | sed 's/^.*bowtie2-align-s version //; s/ .*\\$//')</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29743</th>\n",
       "      <td>circrna</td>\n",
       "      <td>PARENT_GENE</td>\n",
       "      <td>2.bed</td>\n",
       "      <td>./git_repos/circrna/modules/local/annotation/parent_gene/main.nf</td>\n",
       "      <td>45</td>\n",
       "      <td>awk -v OFS=\"\\\\t\" '{print \\$4, \\$14}' master_bed12.bed &gt; circrna_host-gene.txt</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29744</th>\n",
       "      <td>circrna</td>\n",
       "      <td>ANNOTATION</td>\n",
       "      <td>2.bed</td>\n",
       "      <td>./git_repos/circrna/modules/local/annotation/full_annotation/main.nf</td>\n",
       "      <td>34</td>\n",
       "      <td>mv master_bed12.bed ${prefix}.bed.tmp</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29745</th>\n",
       "      <td>circrna</td>\n",
       "      <td>CIRIQUANT_YML</td>\n",
       "      <td>$STR</td>\n",
       "      <td>./git_repos/circrna/modules/local/ciriquant/yml/main.nf</td>\n",
       "      <td>38</td>\n",
       "      <td>printf \"name: ciriquant\\ntools:\\n  bwa: \\$BWA\\n  hisat2: \\$HISAT2\\n  stringtie: \\$STRINGTIE\\n  samtools: \\$SAMTOOLS\\n\\nreference:\\n  fasta: ${fasta_path}\\n  gtf: ${gtf_path}\\n  bwa_index: ${bwa_path}/\\$BWA_FILE\\n  hisat_index: ${hisat2_path}/${hisat2_prefix}\" &gt;&gt; travis.yml</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29746</th>\n",
       "      <td>circrna</td>\n",
       "      <td>DCC</td>\n",
       "      <td>chr/</td>\n",
       "      <td>./git_repos/circrna/modules/local/dcc/dcc/main.nf</td>\n",
       "      <td>27</td>\n",
       "      <td>sed -i 's/^chr//g' $gtf</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29747</th>\n",
       "      <td>circrna</td>\n",
       "      <td>DCC</td>\n",
       "      <td>chr/</td>\n",
       "      <td>./git_repos/circrna/modules/local/dcc/dcc/main.nf</td>\n",
       "      <td>43</td>\n",
       "      <td>sed -i 's/^chr//g' $gtf</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29748</th>\n",
       "      <td>bacass</td>\n",
       "      <td>CANU</td>\n",
       "      <td>canu</td>\n",
       "      <td>./git_repos/bacass/modules/nf-core/canu/main.nf</td>\n",
       "      <td>35</td>\n",
       "      <td>canu \\\\</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29749</th>\n",
       "      <td>bacass</td>\n",
       "      <td>CANU</td>\n",
       "      <td>canu</td>\n",
       "      <td>./git_repos/bacass/modules/nf-core/canu/main.nf</td>\n",
       "      <td>47</td>\n",
       "      <td>canu: \\$(echo \\$(canu --version 2&gt;&amp;1) | sed 's/^.*canu //; s/Using.*\\$//' )</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29750</th>\n",
       "      <td>bacass</td>\n",
       "      <td>NANOPLOT</td>\n",
       "      <td>NanoStat</td>\n",
       "      <td>./git_repos/bacass/modules/nf-core/nanoplot/main.nf</td>\n",
       "      <td>33</td>\n",
       "      <td>mv NanoStats.txt ${prefix}.txt</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29751</th>\n",
       "      <td>demultiplex</td>\n",
       "      <td>BCLCONVERT</td>\n",
       "      <td>bcl-convert</td>\n",
       "      <td>./git_repos/demultiplex/modules/nf-core/bclconvert/main.nf</td>\n",
       "      <td>59</td>\n",
       "      <td>bcl-convert \\\\</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29752</th>\n",
       "      <td>demultiplex</td>\n",
       "      <td>BCLCONVERT</td>\n",
       "      <td>bcl-convert</td>\n",
       "      <td>./git_repos/demultiplex/modules/nf-core/bclconvert/main.nf</td>\n",
       "      <td>69</td>\n",
       "      <td>bclconvert: \\$(bcl-convert -V 2&gt;&amp;1 | head -n 1 | sed 's/^.*Version //')</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29753</th>\n",
       "      <td>metatdenovo</td>\n",
       "      <td>EGGNOG_DOWNLOAD</td>\n",
       "      <td>download_eggnog_data.py</td>\n",
       "      <td>./git_repos/metatdenovo/modules/local/eggnog/download.nf</td>\n",
       "      <td>21</td>\n",
       "      <td>download_eggnog_data.py \\\\</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29754</th>\n",
       "      <td>metatdenovo</td>\n",
       "      <td>EGGNOG_DOWNLOAD</td>\n",
       "      <td>emapper.py</td>\n",
       "      <td>./git_repos/metatdenovo/modules/local/eggnog/download.nf</td>\n",
       "      <td>28</td>\n",
       "      <td>eggnog: \\$( echo \\$(emapper.py --version 2&gt;&amp;1)| sed 's/.* emapper-//' | sed 's/ \\\\/ Expected.*//')</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29755</th>\n",
       "      <td>metatdenovo</td>\n",
       "      <td>EGGNOG_DOWNLOAD</td>\n",
       "      <td>emapper.py</td>\n",
       "      <td>./git_repos/metatdenovo/modules/local/eggnog/download.nf</td>\n",
       "      <td>45</td>\n",
       "      <td>eggnog: \\$( echo \\$(emapper.py --version 2&gt;&amp;1)| sed 's/.* emapper-//' )</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29756</th>\n",
       "      <td>metatdenovo</td>\n",
       "      <td>EGGNOG_MAPPER</td>\n",
       "      <td>emapper.py</td>\n",
       "      <td>./git_repos/metatdenovo/modules/local/eggnog/mapper.nf</td>\n",
       "      <td>36</td>\n",
       "      <td>emapper.py \\\\</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29757</th>\n",
       "      <td>metatdenovo</td>\n",
       "      <td>EGGNOG_MAPPER</td>\n",
       "      <td>emapper.py</td>\n",
       "      <td>./git_repos/metatdenovo/modules/local/eggnog/mapper.nf</td>\n",
       "      <td>50</td>\n",
       "      <td>eggnog: \\$( echo \\$(emapper.py --version 2&gt;&amp;1)| sed 's/.* emapper-//' | sed 's/ \\\\/ Expected.*//')</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29758</th>\n",
       "      <td>metatdenovo</td>\n",
       "      <td>EGGNOG_MAPPER</td>\n",
       "      <td>emapper.py</td>\n",
       "      <td>./git_repos/metatdenovo/modules/local/eggnog/mapper.nf</td>\n",
       "      <td>62</td>\n",
       "      <td>eggnog: \\$( echo \\$(emapper.py --version 2&gt;&amp;1)| sed 's/.* emapper-//' | sed 's/ \\\\/ Expected.*//')</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29759</th>\n",
       "      <td>metatdenovo</td>\n",
       "      <td>TRANSDECODER_LONGORF</td>\n",
       "      <td>TransDecoder.LongOrfs</td>\n",
       "      <td>./git_repos/metatdenovo/modules/nf-core/transdecoder/longorf/main.nf</td>\n",
       "      <td>29</td>\n",
       "      <td>TransDecoder.LongOrfs \\\\</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29760</th>\n",
       "      <td>metatdenovo</td>\n",
       "      <td>TRANSDECODER_LONGORF</td>\n",
       "      <td>TransDecoder.LongOrfs</td>\n",
       "      <td>./git_repos/metatdenovo/modules/nf-core/transdecoder/longorf/main.nf</td>\n",
       "      <td>37</td>\n",
       "      <td>transdecoder: \\$(echo \\$(TransDecoder.LongOrfs --version) | sed -e \"s/TransDecoder.LongOrfs //g\")</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29761</th>\n",
       "      <td>metatdenovo</td>\n",
       "      <td>TRANSDECODER_PREDICT</td>\n",
       "      <td>TransDecoder.Predict</td>\n",
       "      <td>./git_repos/metatdenovo/modules/nf-core/transdecoder/predict/main.nf</td>\n",
       "      <td>29</td>\n",
       "      <td>TransDecoder.Predict \\\\</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29762</th>\n",
       "      <td>metatdenovo</td>\n",
       "      <td>TRANSDECODER_PREDICT</td>\n",
       "      <td>TransDecoder.Predict</td>\n",
       "      <td>./git_repos/metatdenovo/modules/nf-core/transdecoder/predict/main.nf</td>\n",
       "      <td>37</td>\n",
       "      <td>transdecoder: \\$(echo \\$(TransDecoder.Predict --version) | sed -e \"s/TransDecoder.Predict //g\")</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29763</th>\n",
       "      <td>metatdenovo</td>\n",
       "      <td>BBMAP_ALIGN</td>\n",
       "      <td>bbmap.sh</td>\n",
       "      <td>./git_repos/metatdenovo/modules/nf-core/bbmap/align/main.nf</td>\n",
       "      <td>42</td>\n",
       "      <td>bbmap.sh \\\\</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29764</th>\n",
       "      <td>metatdenovo</td>\n",
       "      <td>BBMAP_INDEX</td>\n",
       "      <td>bbmap.sh</td>\n",
       "      <td>./git_repos/metatdenovo/modules/nf-core/bbmap/index/main.nf</td>\n",
       "      <td>23</td>\n",
       "      <td>bbmap.sh \\\\</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29765</th>\n",
       "      <td>hic</td>\n",
       "      <td>GET_RESTRICTION_FRAGMENTS</td>\n",
       "      <td>fragments.bed</td>\n",
       "      <td>./git_repos/hic/modules/local/hicpro/get_restriction_fragments.nf</td>\n",
       "      <td>20</td>\n",
       "      <td>digest_genome.py -r ${res_site} -o restriction_fragments.bed ${fasta}</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29766</th>\n",
       "      <td>hic</td>\n",
       "      <td>HICPRO2PAIRS</td>\n",
       "      <td>contacts</td>\n",
       "      <td>./git_repos/hic/modules/local/hicpro/hicpro2pairs.nf</td>\n",
       "      <td>22</td>\n",
       "      <td>awk '{OFS=\"\\t\";print \\$1,\\$2,\\$3,\\$5,\\$6,\\$4,\\$7}' $vpairs | bgzip -c &gt; ${prefix}_contacts.pairs.gz</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29767</th>\n",
       "      <td>hic</td>\n",
       "      <td>HICPRO2PAIRS</td>\n",
       "      <td>contacts</td>\n",
       "      <td>./git_repos/hic/modules/local/hicpro/hicpro2pairs.nf</td>\n",
       "      <td>24</td>\n",
       "      <td>pairix -f ${prefix}_contacts.pairs.gz</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>29768 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            pipeline                 process_name                 operator  \\\n",
       "0      diaproteomics        get_software_versions                     echo   \n",
       "1      diaproteomics        get_software_versions                     echo   \n",
       "2      diaproteomics  statistical_post_processing                     echo   \n",
       "3      readsimulator                BOWTIE2_ALIGN                     echo   \n",
       "4      readsimulator                BOWTIE2_ALIGN                     echo   \n",
       "5      readsimulator                BOWTIE2_ALIGN                     echo   \n",
       "6      readsimulator                BOWTIE2_ALIGN                     echo   \n",
       "7      readsimulator                BOWTIE2_ALIGN                     echo   \n",
       "8      readsimulator                BOWTIE2_BUILD                     echo   \n",
       "9      readsimulator                BOWTIE2_BUILD                     echo   \n",
       "10     readsimulator               SAMTOOLS_INDEX                     echo   \n",
       "11     readsimulator               SAMTOOLS_INDEX                     echo   \n",
       "12     readsimulator         INSILICOSEQ_GENERATE                     echo   \n",
       "13     readsimulator         INSILICOSEQ_GENERATE                     echo   \n",
       "14     readsimulator         INSILICOSEQ_GENERATE                     echo   \n",
       "15     readsimulator         INSILICOSEQ_GENERATE                     echo   \n",
       "16     readsimulator           MERGE_SAMPLESHEETS                     echo   \n",
       "17     readsimulator           MERGE_SAMPLESHEETS                     echo   \n",
       "18      callingcards                      GTF2BED                     echo   \n",
       "19      callingcards                  DEMULTIPLEX                     echo   \n",
       "20      callingcards                   COUNT_HOPS                     echo   \n",
       "21      callingcards                     CONCATQC                     echo   \n",
       "22      callingcards                     CONCATQC                     echo   \n",
       "23      callingcards                BOWTIE2_ALIGN                     echo   \n",
       "24      callingcards                BOWTIE2_ALIGN                     echo   \n",
       "...              ...                          ...                      ...   \n",
       "29743        circrna                  PARENT_GENE                    2.bed   \n",
       "29744        circrna                   ANNOTATION                    2.bed   \n",
       "29745        circrna                CIRIQUANT_YML                     $STR   \n",
       "29746        circrna                          DCC                     chr/   \n",
       "29747        circrna                          DCC                     chr/   \n",
       "29748         bacass                         CANU                     canu   \n",
       "29749         bacass                         CANU                     canu   \n",
       "29750         bacass                     NANOPLOT                 NanoStat   \n",
       "29751    demultiplex                   BCLCONVERT              bcl-convert   \n",
       "29752    demultiplex                   BCLCONVERT              bcl-convert   \n",
       "29753    metatdenovo              EGGNOG_DOWNLOAD  download_eggnog_data.py   \n",
       "29754    metatdenovo              EGGNOG_DOWNLOAD               emapper.py   \n",
       "29755    metatdenovo              EGGNOG_DOWNLOAD               emapper.py   \n",
       "29756    metatdenovo                EGGNOG_MAPPER               emapper.py   \n",
       "29757    metatdenovo                EGGNOG_MAPPER               emapper.py   \n",
       "29758    metatdenovo                EGGNOG_MAPPER               emapper.py   \n",
       "29759    metatdenovo         TRANSDECODER_LONGORF    TransDecoder.LongOrfs   \n",
       "29760    metatdenovo         TRANSDECODER_LONGORF    TransDecoder.LongOrfs   \n",
       "29761    metatdenovo         TRANSDECODER_PREDICT     TransDecoder.Predict   \n",
       "29762    metatdenovo         TRANSDECODER_PREDICT     TransDecoder.Predict   \n",
       "29763    metatdenovo                  BBMAP_ALIGN                 bbmap.sh   \n",
       "29764    metatdenovo                  BBMAP_INDEX                 bbmap.sh   \n",
       "29765            hic    GET_RESTRICTION_FRAGMENTS            fragments.bed   \n",
       "29766            hic                 HICPRO2PAIRS                 contacts   \n",
       "29767            hic                 HICPRO2PAIRS                 contacts   \n",
       "\n",
       "                                                                                file_path  \\\n",
       "0                                                       ./git_repos/diaproteomics/main.nf   \n",
       "1                                                       ./git_repos/diaproteomics/main.nf   \n",
       "2                                                       ./git_repos/diaproteomics/main.nf   \n",
       "3                         ./git_repos/readsimulator/modules/nf-core/bowtie2/align/main.nf   \n",
       "4                         ./git_repos/readsimulator/modules/nf-core/bowtie2/align/main.nf   \n",
       "5                         ./git_repos/readsimulator/modules/nf-core/bowtie2/align/main.nf   \n",
       "6                         ./git_repos/readsimulator/modules/nf-core/bowtie2/align/main.nf   \n",
       "7                         ./git_repos/readsimulator/modules/nf-core/bowtie2/align/main.nf   \n",
       "8                         ./git_repos/readsimulator/modules/nf-core/bowtie2/build/main.nf   \n",
       "9                         ./git_repos/readsimulator/modules/nf-core/bowtie2/build/main.nf   \n",
       "10                       ./git_repos/readsimulator/modules/nf-core/samtools/index/main.nf   \n",
       "11                       ./git_repos/readsimulator/modules/nf-core/samtools/index/main.nf   \n",
       "12                   ./git_repos/readsimulator/modules/local/insilicoseq/generate/main.nf   \n",
       "13                   ./git_repos/readsimulator/modules/local/insilicoseq/generate/main.nf   \n",
       "14                   ./git_repos/readsimulator/modules/local/insilicoseq/generate/main.nf   \n",
       "15                   ./git_repos/readsimulator/modules/local/insilicoseq/generate/main.nf   \n",
       "16              ./git_repos/readsimulator/modules/local/custom/merge_samplesheets/main.nf   \n",
       "17              ./git_repos/readsimulator/modules/local/custom/merge_samplesheets/main.nf   \n",
       "18                                 ./git_repos/callingcards/modules/local/gtf2bed/main.nf   \n",
       "19     ./git_repos/callingcards/modules/local/callingcardstools/yeast/demultiplex/main.nf   \n",
       "20      ./git_repos/callingcards/modules/local/callingcardstools/yeast/count_hops/main.nf   \n",
       "21        ./git_repos/callingcards/modules/local/callingcardstools/yeast/concatQC/main.nf   \n",
       "22      ./git_repos/callingcards/modules/local/callingcardstools/mammals/concatQC/main.nf   \n",
       "23                         ./git_repos/callingcards/modules/nf-core/bowtie2/align/main.nf   \n",
       "24                         ./git_repos/callingcards/modules/nf-core/bowtie2/align/main.nf   \n",
       "...                                                                                   ...   \n",
       "29743                    ./git_repos/circrna/modules/local/annotation/parent_gene/main.nf   \n",
       "29744                ./git_repos/circrna/modules/local/annotation/full_annotation/main.nf   \n",
       "29745                             ./git_repos/circrna/modules/local/ciriquant/yml/main.nf   \n",
       "29746                                   ./git_repos/circrna/modules/local/dcc/dcc/main.nf   \n",
       "29747                                   ./git_repos/circrna/modules/local/dcc/dcc/main.nf   \n",
       "29748                                     ./git_repos/bacass/modules/nf-core/canu/main.nf   \n",
       "29749                                     ./git_repos/bacass/modules/nf-core/canu/main.nf   \n",
       "29750                                 ./git_repos/bacass/modules/nf-core/nanoplot/main.nf   \n",
       "29751                          ./git_repos/demultiplex/modules/nf-core/bclconvert/main.nf   \n",
       "29752                          ./git_repos/demultiplex/modules/nf-core/bclconvert/main.nf   \n",
       "29753                            ./git_repos/metatdenovo/modules/local/eggnog/download.nf   \n",
       "29754                            ./git_repos/metatdenovo/modules/local/eggnog/download.nf   \n",
       "29755                            ./git_repos/metatdenovo/modules/local/eggnog/download.nf   \n",
       "29756                              ./git_repos/metatdenovo/modules/local/eggnog/mapper.nf   \n",
       "29757                              ./git_repos/metatdenovo/modules/local/eggnog/mapper.nf   \n",
       "29758                              ./git_repos/metatdenovo/modules/local/eggnog/mapper.nf   \n",
       "29759                ./git_repos/metatdenovo/modules/nf-core/transdecoder/longorf/main.nf   \n",
       "29760                ./git_repos/metatdenovo/modules/nf-core/transdecoder/longorf/main.nf   \n",
       "29761                ./git_repos/metatdenovo/modules/nf-core/transdecoder/predict/main.nf   \n",
       "29762                ./git_repos/metatdenovo/modules/nf-core/transdecoder/predict/main.nf   \n",
       "29763                         ./git_repos/metatdenovo/modules/nf-core/bbmap/align/main.nf   \n",
       "29764                         ./git_repos/metatdenovo/modules/nf-core/bbmap/index/main.nf   \n",
       "29765                   ./git_repos/hic/modules/local/hicpro/get_restriction_fragments.nf   \n",
       "29766                                ./git_repos/hic/modules/local/hicpro/hicpro2pairs.nf   \n",
       "29767                                ./git_repos/hic/modules/local/hicpro/hicpro2pairs.nf   \n",
       "\n",
       "       line_number  \\\n",
       "0              335   \n",
       "1              336   \n",
       "2              971   \n",
       "3               47   \n",
       "4               68   \n",
       "5               69   \n",
       "6               88   \n",
       "7               89   \n",
       "8               27   \n",
       "9               39   \n",
       "10              33   \n",
       "11              45   \n",
       "12              29   \n",
       "13              30   \n",
       "14              53   \n",
       "15              54   \n",
       "16              23   \n",
       "17              26   \n",
       "18              28   \n",
       "19              36   \n",
       "20              39   \n",
       "21              33   \n",
       "22              38   \n",
       "23              47   \n",
       "24              68   \n",
       "...            ...   \n",
       "29743           45   \n",
       "29744           34   \n",
       "29745           38   \n",
       "29746           27   \n",
       "29747           43   \n",
       "29748           35   \n",
       "29749           47   \n",
       "29750           33   \n",
       "29751           59   \n",
       "29752           69   \n",
       "29753           21   \n",
       "29754           28   \n",
       "29755           45   \n",
       "29756           36   \n",
       "29757           50   \n",
       "29758           62   \n",
       "29759           29   \n",
       "29760           37   \n",
       "29761           29   \n",
       "29762           37   \n",
       "29763           42   \n",
       "29764           23   \n",
       "29765           20   \n",
       "29766           22   \n",
       "29767           24   \n",
       "\n",
       "                                                                                                                                                                                                                                                                            line_content  \\\n",
       "0                                                                                                                                                                                                                                       echo $workflow.manifest.version > v_pipeline.txt   \n",
       "1                                                                                                                                                                                                                                       echo $workflow.nextflow.version > v_nextflow.txt   \n",
       "2                                                                                                                                                      msstats.R > msstats.log || echo \"Optional MSstats step failed. Please check logs and re-run or do a manual statistical analysis.\"   \n",
       "3                                                                                                                                                                                                                [ -z \"\\$INDEX\" ] && echo \"Bowtie2 index files not found\" 1>&2 && exit 1   \n",
       "4                                                                                                                                                                                        bowtie2: \\$(echo \\$(bowtie2 --version 2>&1) | sed 's/^.*bowtie2-align-s version //; s/ .*\\$//')   \n",
       "5                                                                                                                                                                                                 samtools: \\$(echo \\$(samtools --version 2>&1) | sed 's/^.*samtools //; s/Using.*\\$//')   \n",
       "6                                                                                                                                                                                        bowtie2: \\$(echo \\$(bowtie2 --version 2>&1) | sed 's/^.*bowtie2-align-s version //; s/ .*\\$//')   \n",
       "7                                                                                                                                                                                                 samtools: \\$(echo \\$(samtools --version 2>&1) | sed 's/^.*samtools //; s/Using.*\\$//')   \n",
       "8                                                                                                                                                                                        bowtie2: \\$(echo \\$(bowtie2 --version 2>&1) | sed 's/^.*bowtie2-align-s version //; s/ .*\\$//')   \n",
       "9                                                                                                                                                                                        bowtie2: \\$(echo \\$(bowtie2 --version 2>&1) | sed 's/^.*bowtie2-align-s version //; s/ .*\\$//')   \n",
       "10                                                                                                                                                                                                samtools: \\$(echo \\$(samtools --version 2>&1) | sed 's/^.*samtools //; s/Using.*\\$//')   \n",
       "11                                                                                                                                                                                                samtools: \\$(echo \\$(samtools --version 2>&1) | sed 's/^.*samtools //; s/Using.*\\$//')   \n",
       "12                                                                                                                                                                                                                                 seed=\\$(echo $seed | sed 's/\\\\[//g' | sed 's/\\\\]//g')   \n",
       "13                                                                                                                                                                                                                             prefix=\\$(echo $prefix | sed 's/\\\\[//g' | sed 's/\\\\]//g')   \n",
       "14                                                                                                                                                                                                                                 seed=\\$(echo $seed | sed 's/\\\\[//g' | sed 's/\\\\]//g')   \n",
       "15                                                                                                                                                                                                                             prefix=\\$(echo $prefix | sed 's/\\\\[//g' | sed 's/\\\\]//g')   \n",
       "16                                                                                                                                                                                                           echo \\\\\"sample\\\\\",\\\\\"fastq_1\\\\\",\\\\\"fastq_2\\\\\" > \"${prefix}_samplesheet.csv\"   \n",
       "17                                                                                                                                                                                                                                                   echo >> \"${prefix}_samplesheet.csv\"   \n",
       "18                                                                                                                                                                                                            perl: \\$(echo \\$(perl --version 2>&1) | sed 's/.*v\\\\(.*\\\\)) built.*/\\\\1/')   \n",
       "19                                                                                                                                                                                                      callingcardstools: \\$(echo \\$(callingcardstools --version 2>&1) | sed 's/.* //')   \n",
       "20                                                                                                                                                                                                      callingcardstools: \\$(echo \\$(callingcardstools --version 2>&1) | sed 's/.* //')   \n",
       "21                                                                                                                                                                                                      callingcardstools: \\$(echo \\$(callingcardstools --version 2>&1) | sed 's/.* //')   \n",
       "22                                                                                                                                                                                                      callingcardstools: \\$(echo \\$(callingcardstools --version 2>&1) | sed 's/.* //')   \n",
       "23                                                                                                                                                                                                               [ -z \"\\$INDEX\" ] && echo \"Bowtie2 index files not found\" 1>&2 && exit 1   \n",
       "24                                                                                                                                                                                       bowtie2: \\$(echo \\$(bowtie2 --version 2>&1) | sed 's/^.*bowtie2-align-s version //; s/ .*\\$//')   \n",
       "...                                                                                                                                                                                                                                                                                  ...   \n",
       "29743                                                                                                                                                                                                      awk -v OFS=\"\\\\t\" '{print \\$4, \\$14}' master_bed12.bed > circrna_host-gene.txt   \n",
       "29744                                                                                                                                                                                                                                              mv master_bed12.bed ${prefix}.bed.tmp   \n",
       "29745  printf \"name: ciriquant\\ntools:\\n  bwa: \\$BWA\\n  hisat2: \\$HISAT2\\n  stringtie: \\$STRINGTIE\\n  samtools: \\$SAMTOOLS\\n\\nreference:\\n  fasta: ${fasta_path}\\n  gtf: ${gtf_path}\\n  bwa_index: ${bwa_path}/\\$BWA_FILE\\n  hisat_index: ${hisat2_path}/${hisat2_prefix}\" >> travis.yml   \n",
       "29746                                                                                                                                                                                                                                                            sed -i 's/^chr//g' $gtf   \n",
       "29747                                                                                                                                                                                                                                                            sed -i 's/^chr//g' $gtf   \n",
       "29748                                                                                                                                                                                                                                                                            canu \\\\   \n",
       "29749                                                                                                                                                                                                        canu: \\$(echo \\$(canu --version 2>&1) | sed 's/^.*canu //; s/Using.*\\$//' )   \n",
       "29750                                                                                                                                                                                                                                                     mv NanoStats.txt ${prefix}.txt   \n",
       "29751                                                                                                                                                                                                                                                                     bcl-convert \\\\   \n",
       "29752                                                                                                                                                                                                            bclconvert: \\$(bcl-convert -V 2>&1 | head -n 1 | sed 's/^.*Version //')   \n",
       "29753                                                                                                                                                                                                                                                         download_eggnog_data.py \\\\   \n",
       "29754                                                                                                                                                                                 eggnog: \\$( echo \\$(emapper.py --version 2>&1)| sed 's/.* emapper-//' | sed 's/ \\\\/ Expected.*//')   \n",
       "29755                                                                                                                                                                                                            eggnog: \\$( echo \\$(emapper.py --version 2>&1)| sed 's/.* emapper-//' )   \n",
       "29756                                                                                                                                                                                                                                                                      emapper.py \\\\   \n",
       "29757                                                                                                                                                                                 eggnog: \\$( echo \\$(emapper.py --version 2>&1)| sed 's/.* emapper-//' | sed 's/ \\\\/ Expected.*//')   \n",
       "29758                                                                                                                                                                                 eggnog: \\$( echo \\$(emapper.py --version 2>&1)| sed 's/.* emapper-//' | sed 's/ \\\\/ Expected.*//')   \n",
       "29759                                                                                                                                                                                                                                                           TransDecoder.LongOrfs \\\\   \n",
       "29760                                                                                                                                                                                  transdecoder: \\$(echo \\$(TransDecoder.LongOrfs --version) | sed -e \"s/TransDecoder.LongOrfs //g\")   \n",
       "29761                                                                                                                                                                                                                                                            TransDecoder.Predict \\\\   \n",
       "29762                                                                                                                                                                                    transdecoder: \\$(echo \\$(TransDecoder.Predict --version) | sed -e \"s/TransDecoder.Predict //g\")   \n",
       "29763                                                                                                                                                                                                                                                                        bbmap.sh \\\\   \n",
       "29764                                                                                                                                                                                                                                                                        bbmap.sh \\\\   \n",
       "29765                                                                                                                                                                                                              digest_genome.py -r ${res_site} -o restriction_fragments.bed ${fasta}   \n",
       "29766                                                                                                                                                                                awk '{OFS=\"\\t\";print \\$1,\\$2,\\$3,\\$5,\\$6,\\$4,\\$7}' $vpairs | bgzip -c > ${prefix}_contacts.pairs.gz   \n",
       "29767                                                                                                                                                                                                                                              pairix -f ${prefix}_contacts.pairs.gz   \n",
       "\n",
       "       Assigned Group  \n",
       "0                   2  \n",
       "1                   2  \n",
       "2                   2  \n",
       "3                   2  \n",
       "4                   2  \n",
       "5                   2  \n",
       "6                   2  \n",
       "7                   2  \n",
       "8                   2  \n",
       "9                   2  \n",
       "10                  2  \n",
       "11                  2  \n",
       "12                  2  \n",
       "13                  2  \n",
       "14                  2  \n",
       "15                  2  \n",
       "16                  2  \n",
       "17                  2  \n",
       "18                  2  \n",
       "19                  2  \n",
       "20                  2  \n",
       "21                  2  \n",
       "22                  2  \n",
       "23                  2  \n",
       "24                  2  \n",
       "...               ...  \n",
       "29743               0  \n",
       "29744               0  \n",
       "29745               6  \n",
       "29746               4  \n",
       "29747               4  \n",
       "29748               6  \n",
       "29749               6  \n",
       "29750               6  \n",
       "29751               4  \n",
       "29752               4  \n",
       "29753               4  \n",
       "29754               4  \n",
       "29755               4  \n",
       "29756               4  \n",
       "29757               4  \n",
       "29758               4  \n",
       "29759               4  \n",
       "29760               4  \n",
       "29761               4  \n",
       "29762               4  \n",
       "29763               4  \n",
       "29764               4  \n",
       "29765               0  \n",
       "29766               6  \n",
       "29767               6  \n",
       "\n",
       "[29768 rows x 7 columns]"
      ]
     },
     "execution_count": 198,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_df"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7bd62253",
   "metadata": {},
   "source": [
    "We merge the biotool and the all operator df here"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 199,
   "id": "8b0b6ec9",
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'bioinformatics_script' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "Cell \u001b[1;32mIn[199], line 1\u001b[0m\n\u001b[1;32m----> 1\u001b[0m bioinformatics_script\n",
      "\u001b[1;31mNameError\u001b[0m: name 'bioinformatics_script' is not defined"
     ]
    }
   ],
   "source": [
    "bioinformatics_script"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 200,
   "id": "878bed67",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pipeline</th>\n",
       "      <th>process_name</th>\n",
       "      <th>operator</th>\n",
       "      <th>file_path</th>\n",
       "      <th>line_number</th>\n",
       "      <th>line_content</th>\n",
       "      <th>Assigned Group</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>25669</th>\n",
       "      <td>detaxizer</td>\n",
       "      <td>FASTP</td>\n",
       "      <td>fastp</td>\n",
       "      <td>./git_repos/detaxizer/modules/nf-core/fastp/main.nf</td>\n",
       "      <td>39</td>\n",
       "      <td>fastp \\\\</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25670</th>\n",
       "      <td>detaxizer</td>\n",
       "      <td>FASTP</td>\n",
       "      <td>fastp</td>\n",
       "      <td>./git_repos/detaxizer/modules/nf-core/fastp/main.nf</td>\n",
       "      <td>43</td>\n",
       "      <td>--json ${prefix}.fastp.json \\\\</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25671</th>\n",
       "      <td>detaxizer</td>\n",
       "      <td>FASTP</td>\n",
       "      <td>fastp</td>\n",
       "      <td>./git_repos/detaxizer/modules/nf-core/fastp/main.nf</td>\n",
       "      <td>44</td>\n",
       "      <td>--html ${prefix}.fastp.html \\\\</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25672</th>\n",
       "      <td>detaxizer</td>\n",
       "      <td>FASTP</td>\n",
       "      <td>fastp</td>\n",
       "      <td>./git_repos/detaxizer/modules/nf-core/fastp/main.nf</td>\n",
       "      <td>48</td>\n",
       "      <td>2&gt; &gt;(tee ${prefix}.fastp.log &gt;&amp;2) \\\\</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25673</th>\n",
       "      <td>detaxizer</td>\n",
       "      <td>FASTP</td>\n",
       "      <td>fastp</td>\n",
       "      <td>./git_repos/detaxizer/modules/nf-core/fastp/main.nf</td>\n",
       "      <td>49</td>\n",
       "      <td>| gzip -c &gt; ${prefix}.fastp.fastq.gz</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25674</th>\n",
       "      <td>detaxizer</td>\n",
       "      <td>FASTP</td>\n",
       "      <td>fastp</td>\n",
       "      <td>./git_repos/detaxizer/modules/nf-core/fastp/main.nf</td>\n",
       "      <td>53</td>\n",
       "      <td>fastp: \\$(fastp --version 2&gt;&amp;1 | sed -e \"s/fastp //g\")</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25675</th>\n",
       "      <td>detaxizer</td>\n",
       "      <td>FASTP</td>\n",
       "      <td>fastp</td>\n",
       "      <td>./git_repos/detaxizer/modules/nf-core/fastp/main.nf</td>\n",
       "      <td>60</td>\n",
       "      <td>fastp \\\\</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25676</th>\n",
       "      <td>detaxizer</td>\n",
       "      <td>FASTP</td>\n",
       "      <td>fastp</td>\n",
       "      <td>./git_repos/detaxizer/modules/nf-core/fastp/main.nf</td>\n",
       "      <td>62</td>\n",
       "      <td>--out1  ${prefix}.fastp.fastq.gz \\\\</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25677</th>\n",
       "      <td>detaxizer</td>\n",
       "      <td>FASTP</td>\n",
       "      <td>fastp</td>\n",
       "      <td>./git_repos/detaxizer/modules/nf-core/fastp/main.nf</td>\n",
       "      <td>64</td>\n",
       "      <td>--json ${prefix}.fastp.json \\\\</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25678</th>\n",
       "      <td>detaxizer</td>\n",
       "      <td>FASTP</td>\n",
       "      <td>fastp</td>\n",
       "      <td>./git_repos/detaxizer/modules/nf-core/fastp/main.nf</td>\n",
       "      <td>65</td>\n",
       "      <td>--html ${prefix}.fastp.html \\\\</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25679</th>\n",
       "      <td>detaxizer</td>\n",
       "      <td>FASTP</td>\n",
       "      <td>fastp</td>\n",
       "      <td>./git_repos/detaxizer/modules/nf-core/fastp/main.nf</td>\n",
       "      <td>69</td>\n",
       "      <td>2&gt; &gt;(tee ${prefix}.fastp.log &gt;&amp;2)</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25680</th>\n",
       "      <td>detaxizer</td>\n",
       "      <td>FASTP</td>\n",
       "      <td>fastp</td>\n",
       "      <td>./git_repos/detaxizer/modules/nf-core/fastp/main.nf</td>\n",
       "      <td>73</td>\n",
       "      <td>fastp: \\$(fastp --version 2&gt;&amp;1 | sed -e \"s/fastp //g\")</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25681</th>\n",
       "      <td>detaxizer</td>\n",
       "      <td>FASTP</td>\n",
       "      <td>fastp</td>\n",
       "      <td>./git_repos/detaxizer/modules/nf-core/fastp/main.nf</td>\n",
       "      <td>81</td>\n",
       "      <td>fastp \\\\</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25682</th>\n",
       "      <td>detaxizer</td>\n",
       "      <td>FASTP</td>\n",
       "      <td>fastp</td>\n",
       "      <td>./git_repos/detaxizer/modules/nf-core/fastp/main.nf</td>\n",
       "      <td>84</td>\n",
       "      <td>--out1 ${prefix}_1.fastp.fastq.gz \\\\</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25683</th>\n",
       "      <td>detaxizer</td>\n",
       "      <td>FASTP</td>\n",
       "      <td>fastp</td>\n",
       "      <td>./git_repos/detaxizer/modules/nf-core/fastp/main.nf</td>\n",
       "      <td>85</td>\n",
       "      <td>--out2 ${prefix}_2.fastp.fastq.gz \\\\</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25684</th>\n",
       "      <td>detaxizer</td>\n",
       "      <td>FASTP</td>\n",
       "      <td>fastp</td>\n",
       "      <td>./git_repos/detaxizer/modules/nf-core/fastp/main.nf</td>\n",
       "      <td>86</td>\n",
       "      <td>--json ${prefix}.fastp.json \\\\</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25685</th>\n",
       "      <td>detaxizer</td>\n",
       "      <td>FASTP</td>\n",
       "      <td>fastp</td>\n",
       "      <td>./git_repos/detaxizer/modules/nf-core/fastp/main.nf</td>\n",
       "      <td>87</td>\n",
       "      <td>--html ${prefix}.fastp.html \\\\</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25686</th>\n",
       "      <td>detaxizer</td>\n",
       "      <td>FASTP</td>\n",
       "      <td>fastp</td>\n",
       "      <td>./git_repos/detaxizer/modules/nf-core/fastp/main.nf</td>\n",
       "      <td>94</td>\n",
       "      <td>2&gt; &gt;(tee ${prefix}.fastp.log &gt;&amp;2)</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25687</th>\n",
       "      <td>detaxizer</td>\n",
       "      <td>FASTP</td>\n",
       "      <td>fastp</td>\n",
       "      <td>./git_repos/detaxizer/modules/nf-core/fastp/main.nf</td>\n",
       "      <td>98</td>\n",
       "      <td>fastp: \\$(fastp --version 2&gt;&amp;1 | sed -e \"s/fastp //g\")</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25688</th>\n",
       "      <td>detaxizer</td>\n",
       "      <td>FASTP</td>\n",
       "      <td>fastp</td>\n",
       "      <td>./git_repos/detaxizer/modules/nf-core/fastp/main.nf</td>\n",
       "      <td>110</td>\n",
       "      <td>touch \"${prefix}.fastp.json\"</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25689</th>\n",
       "      <td>detaxizer</td>\n",
       "      <td>FASTP</td>\n",
       "      <td>fastp</td>\n",
       "      <td>./git_repos/detaxizer/modules/nf-core/fastp/main.nf</td>\n",
       "      <td>111</td>\n",
       "      <td>touch \"${prefix}.fastp.html\"</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25690</th>\n",
       "      <td>detaxizer</td>\n",
       "      <td>FASTP</td>\n",
       "      <td>fastp</td>\n",
       "      <td>./git_repos/detaxizer/modules/nf-core/fastp/main.nf</td>\n",
       "      <td>112</td>\n",
       "      <td>touch \"${prefix}.fastp.log\"</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25691</th>\n",
       "      <td>detaxizer</td>\n",
       "      <td>FASTP</td>\n",
       "      <td>fastp</td>\n",
       "      <td>./git_repos/detaxizer/modules/nf-core/fastp/main.nf</td>\n",
       "      <td>117</td>\n",
       "      <td>fastp: \\$(fastp --version 2&gt;&amp;1 | sed -e \"s/fastp //g\")</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25692</th>\n",
       "      <td>kmermaid</td>\n",
       "      <td>get_software_versions</td>\n",
       "      <td>fastp</td>\n",
       "      <td>./git_repos/kmermaid/main.nf</td>\n",
       "      <td>670</td>\n",
       "      <td>fastp --version &amp;&gt; v_fastp.txt</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25693</th>\n",
       "      <td>kmermaid</td>\n",
       "      <td>fastp</td>\n",
       "      <td>fastp</td>\n",
       "      <td>./git_repos/kmermaid/main.nf</td>\n",
       "      <td>998</td>\n",
       "      <td>fastp \\\\</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25992</th>\n",
       "      <td>bacass</td>\n",
       "      <td>FASTP</td>\n",
       "      <td>fastp</td>\n",
       "      <td>./git_repos/bacass/modules/nf-core/fastp/main.nf</td>\n",
       "      <td>84</td>\n",
       "      <td>--out1 ${prefix}_1.fastp.fastq.gz \\\\</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25993</th>\n",
       "      <td>bacass</td>\n",
       "      <td>FASTP</td>\n",
       "      <td>fastp</td>\n",
       "      <td>./git_repos/bacass/modules/nf-core/fastp/main.nf</td>\n",
       "      <td>85</td>\n",
       "      <td>--out2 ${prefix}_2.fastp.fastq.gz \\\\</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25994</th>\n",
       "      <td>bacass</td>\n",
       "      <td>FASTP</td>\n",
       "      <td>fastp</td>\n",
       "      <td>./git_repos/bacass/modules/nf-core/fastp/main.nf</td>\n",
       "      <td>86</td>\n",
       "      <td>--json ${prefix}.fastp.json \\\\</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25995</th>\n",
       "      <td>bacass</td>\n",
       "      <td>FASTP</td>\n",
       "      <td>fastp</td>\n",
       "      <td>./git_repos/bacass/modules/nf-core/fastp/main.nf</td>\n",
       "      <td>87</td>\n",
       "      <td>--html ${prefix}.fastp.html \\\\</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25996</th>\n",
       "      <td>bacass</td>\n",
       "      <td>FASTP</td>\n",
       "      <td>fastp</td>\n",
       "      <td>./git_repos/bacass/modules/nf-core/fastp/main.nf</td>\n",
       "      <td>94</td>\n",
       "      <td>2&gt; ${prefix}.fastp.log</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25997</th>\n",
       "      <td>bacass</td>\n",
       "      <td>FASTP</td>\n",
       "      <td>fastp</td>\n",
       "      <td>./git_repos/bacass/modules/nf-core/fastp/main.nf</td>\n",
       "      <td>98</td>\n",
       "      <td>fastp: \\$(fastp --version 2&gt;&amp;1 | sed -e \"s/fastp //g\")</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25998</th>\n",
       "      <td>demultiplex</td>\n",
       "      <td>FASTP</td>\n",
       "      <td>fastp</td>\n",
       "      <td>./git_repos/demultiplex/modules/nf-core/fastp/main.nf</td>\n",
       "      <td>39</td>\n",
       "      <td>fastp \\\\</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25999</th>\n",
       "      <td>demultiplex</td>\n",
       "      <td>FASTP</td>\n",
       "      <td>fastp</td>\n",
       "      <td>./git_repos/demultiplex/modules/nf-core/fastp/main.nf</td>\n",
       "      <td>43</td>\n",
       "      <td>--json ${prefix}.fastp.json \\\\</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26000</th>\n",
       "      <td>demultiplex</td>\n",
       "      <td>FASTP</td>\n",
       "      <td>fastp</td>\n",
       "      <td>./git_repos/demultiplex/modules/nf-core/fastp/main.nf</td>\n",
       "      <td>44</td>\n",
       "      <td>--html ${prefix}.fastp.html \\\\</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26001</th>\n",
       "      <td>demultiplex</td>\n",
       "      <td>FASTP</td>\n",
       "      <td>fastp</td>\n",
       "      <td>./git_repos/demultiplex/modules/nf-core/fastp/main.nf</td>\n",
       "      <td>48</td>\n",
       "      <td>2&gt; &gt;(tee ${prefix}.fastp.log &gt;&amp;2) \\\\</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26002</th>\n",
       "      <td>demultiplex</td>\n",
       "      <td>FASTP</td>\n",
       "      <td>fastp</td>\n",
       "      <td>./git_repos/demultiplex/modules/nf-core/fastp/main.nf</td>\n",
       "      <td>49</td>\n",
       "      <td>| gzip -c &gt; ${prefix}.fastp.fastq.gz</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26003</th>\n",
       "      <td>demultiplex</td>\n",
       "      <td>FASTP</td>\n",
       "      <td>fastp</td>\n",
       "      <td>./git_repos/demultiplex/modules/nf-core/fastp/main.nf</td>\n",
       "      <td>53</td>\n",
       "      <td>fastp: \\$(fastp --version 2&gt;&amp;1 | sed -e \"s/fastp //g\")</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26004</th>\n",
       "      <td>demultiplex</td>\n",
       "      <td>FASTP</td>\n",
       "      <td>fastp</td>\n",
       "      <td>./git_repos/demultiplex/modules/nf-core/fastp/main.nf</td>\n",
       "      <td>60</td>\n",
       "      <td>fastp \\\\</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26005</th>\n",
       "      <td>demultiplex</td>\n",
       "      <td>FASTP</td>\n",
       "      <td>fastp</td>\n",
       "      <td>./git_repos/demultiplex/modules/nf-core/fastp/main.nf</td>\n",
       "      <td>62</td>\n",
       "      <td>--out1  ${prefix}.fastp.fastq.gz \\\\</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26006</th>\n",
       "      <td>demultiplex</td>\n",
       "      <td>FASTP</td>\n",
       "      <td>fastp</td>\n",
       "      <td>./git_repos/demultiplex/modules/nf-core/fastp/main.nf</td>\n",
       "      <td>64</td>\n",
       "      <td>--json ${prefix}.fastp.json \\\\</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26007</th>\n",
       "      <td>demultiplex</td>\n",
       "      <td>FASTP</td>\n",
       "      <td>fastp</td>\n",
       "      <td>./git_repos/demultiplex/modules/nf-core/fastp/main.nf</td>\n",
       "      <td>65</td>\n",
       "      <td>--html ${prefix}.fastp.html \\\\</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26008</th>\n",
       "      <td>demultiplex</td>\n",
       "      <td>FASTP</td>\n",
       "      <td>fastp</td>\n",
       "      <td>./git_repos/demultiplex/modules/nf-core/fastp/main.nf</td>\n",
       "      <td>69</td>\n",
       "      <td>2&gt; &gt;(tee ${prefix}.fastp.log &gt;&amp;2)</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26009</th>\n",
       "      <td>demultiplex</td>\n",
       "      <td>FASTP</td>\n",
       "      <td>fastp</td>\n",
       "      <td>./git_repos/demultiplex/modules/nf-core/fastp/main.nf</td>\n",
       "      <td>73</td>\n",
       "      <td>fastp: \\$(fastp --version 2&gt;&amp;1 | sed -e \"s/fastp //g\")</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26010</th>\n",
       "      <td>demultiplex</td>\n",
       "      <td>FASTP</td>\n",
       "      <td>fastp</td>\n",
       "      <td>./git_repos/demultiplex/modules/nf-core/fastp/main.nf</td>\n",
       "      <td>81</td>\n",
       "      <td>fastp \\\\</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26011</th>\n",
       "      <td>demultiplex</td>\n",
       "      <td>FASTP</td>\n",
       "      <td>fastp</td>\n",
       "      <td>./git_repos/demultiplex/modules/nf-core/fastp/main.nf</td>\n",
       "      <td>84</td>\n",
       "      <td>--out1 ${prefix}_1.fastp.fastq.gz \\\\</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26012</th>\n",
       "      <td>demultiplex</td>\n",
       "      <td>FASTP</td>\n",
       "      <td>fastp</td>\n",
       "      <td>./git_repos/demultiplex/modules/nf-core/fastp/main.nf</td>\n",
       "      <td>85</td>\n",
       "      <td>--out2 ${prefix}_2.fastp.fastq.gz \\\\</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26013</th>\n",
       "      <td>demultiplex</td>\n",
       "      <td>FASTP</td>\n",
       "      <td>fastp</td>\n",
       "      <td>./git_repos/demultiplex/modules/nf-core/fastp/main.nf</td>\n",
       "      <td>86</td>\n",
       "      <td>--json ${prefix}.fastp.json \\\\</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26014</th>\n",
       "      <td>demultiplex</td>\n",
       "      <td>FASTP</td>\n",
       "      <td>fastp</td>\n",
       "      <td>./git_repos/demultiplex/modules/nf-core/fastp/main.nf</td>\n",
       "      <td>87</td>\n",
       "      <td>--html ${prefix}.fastp.html \\\\</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26015</th>\n",
       "      <td>demultiplex</td>\n",
       "      <td>FASTP</td>\n",
       "      <td>fastp</td>\n",
       "      <td>./git_repos/demultiplex/modules/nf-core/fastp/main.nf</td>\n",
       "      <td>94</td>\n",
       "      <td>2&gt; &gt;(tee ${prefix}.fastp.log &gt;&amp;2)</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26016</th>\n",
       "      <td>demultiplex</td>\n",
       "      <td>FASTP</td>\n",
       "      <td>fastp</td>\n",
       "      <td>./git_repos/demultiplex/modules/nf-core/fastp/main.nf</td>\n",
       "      <td>98</td>\n",
       "      <td>fastp: \\$(fastp --version 2&gt;&amp;1 | sed -e \"s/fastp //g\")</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>348 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          pipeline           process_name operator  \\\n",
       "25669    detaxizer                  FASTP    fastp   \n",
       "25670    detaxizer                  FASTP    fastp   \n",
       "25671    detaxizer                  FASTP    fastp   \n",
       "25672    detaxizer                  FASTP    fastp   \n",
       "25673    detaxizer                  FASTP    fastp   \n",
       "25674    detaxizer                  FASTP    fastp   \n",
       "25675    detaxizer                  FASTP    fastp   \n",
       "25676    detaxizer                  FASTP    fastp   \n",
       "25677    detaxizer                  FASTP    fastp   \n",
       "25678    detaxizer                  FASTP    fastp   \n",
       "25679    detaxizer                  FASTP    fastp   \n",
       "25680    detaxizer                  FASTP    fastp   \n",
       "25681    detaxizer                  FASTP    fastp   \n",
       "25682    detaxizer                  FASTP    fastp   \n",
       "25683    detaxizer                  FASTP    fastp   \n",
       "25684    detaxizer                  FASTP    fastp   \n",
       "25685    detaxizer                  FASTP    fastp   \n",
       "25686    detaxizer                  FASTP    fastp   \n",
       "25687    detaxizer                  FASTP    fastp   \n",
       "25688    detaxizer                  FASTP    fastp   \n",
       "25689    detaxizer                  FASTP    fastp   \n",
       "25690    detaxizer                  FASTP    fastp   \n",
       "25691    detaxizer                  FASTP    fastp   \n",
       "25692     kmermaid  get_software_versions    fastp   \n",
       "25693     kmermaid                  fastp    fastp   \n",
       "...            ...                    ...      ...   \n",
       "25992       bacass                  FASTP    fastp   \n",
       "25993       bacass                  FASTP    fastp   \n",
       "25994       bacass                  FASTP    fastp   \n",
       "25995       bacass                  FASTP    fastp   \n",
       "25996       bacass                  FASTP    fastp   \n",
       "25997       bacass                  FASTP    fastp   \n",
       "25998  demultiplex                  FASTP    fastp   \n",
       "25999  demultiplex                  FASTP    fastp   \n",
       "26000  demultiplex                  FASTP    fastp   \n",
       "26001  demultiplex                  FASTP    fastp   \n",
       "26002  demultiplex                  FASTP    fastp   \n",
       "26003  demultiplex                  FASTP    fastp   \n",
       "26004  demultiplex                  FASTP    fastp   \n",
       "26005  demultiplex                  FASTP    fastp   \n",
       "26006  demultiplex                  FASTP    fastp   \n",
       "26007  demultiplex                  FASTP    fastp   \n",
       "26008  demultiplex                  FASTP    fastp   \n",
       "26009  demultiplex                  FASTP    fastp   \n",
       "26010  demultiplex                  FASTP    fastp   \n",
       "26011  demultiplex                  FASTP    fastp   \n",
       "26012  demultiplex                  FASTP    fastp   \n",
       "26013  demultiplex                  FASTP    fastp   \n",
       "26014  demultiplex                  FASTP    fastp   \n",
       "26015  demultiplex                  FASTP    fastp   \n",
       "26016  demultiplex                  FASTP    fastp   \n",
       "\n",
       "                                                   file_path  line_number  \\\n",
       "25669    ./git_repos/detaxizer/modules/nf-core/fastp/main.nf           39   \n",
       "25670    ./git_repos/detaxizer/modules/nf-core/fastp/main.nf           43   \n",
       "25671    ./git_repos/detaxizer/modules/nf-core/fastp/main.nf           44   \n",
       "25672    ./git_repos/detaxizer/modules/nf-core/fastp/main.nf           48   \n",
       "25673    ./git_repos/detaxizer/modules/nf-core/fastp/main.nf           49   \n",
       "25674    ./git_repos/detaxizer/modules/nf-core/fastp/main.nf           53   \n",
       "25675    ./git_repos/detaxizer/modules/nf-core/fastp/main.nf           60   \n",
       "25676    ./git_repos/detaxizer/modules/nf-core/fastp/main.nf           62   \n",
       "25677    ./git_repos/detaxizer/modules/nf-core/fastp/main.nf           64   \n",
       "25678    ./git_repos/detaxizer/modules/nf-core/fastp/main.nf           65   \n",
       "25679    ./git_repos/detaxizer/modules/nf-core/fastp/main.nf           69   \n",
       "25680    ./git_repos/detaxizer/modules/nf-core/fastp/main.nf           73   \n",
       "25681    ./git_repos/detaxizer/modules/nf-core/fastp/main.nf           81   \n",
       "25682    ./git_repos/detaxizer/modules/nf-core/fastp/main.nf           84   \n",
       "25683    ./git_repos/detaxizer/modules/nf-core/fastp/main.nf           85   \n",
       "25684    ./git_repos/detaxizer/modules/nf-core/fastp/main.nf           86   \n",
       "25685    ./git_repos/detaxizer/modules/nf-core/fastp/main.nf           87   \n",
       "25686    ./git_repos/detaxizer/modules/nf-core/fastp/main.nf           94   \n",
       "25687    ./git_repos/detaxizer/modules/nf-core/fastp/main.nf           98   \n",
       "25688    ./git_repos/detaxizer/modules/nf-core/fastp/main.nf          110   \n",
       "25689    ./git_repos/detaxizer/modules/nf-core/fastp/main.nf          111   \n",
       "25690    ./git_repos/detaxizer/modules/nf-core/fastp/main.nf          112   \n",
       "25691    ./git_repos/detaxizer/modules/nf-core/fastp/main.nf          117   \n",
       "25692                           ./git_repos/kmermaid/main.nf          670   \n",
       "25693                           ./git_repos/kmermaid/main.nf          998   \n",
       "...                                                      ...          ...   \n",
       "25992       ./git_repos/bacass/modules/nf-core/fastp/main.nf           84   \n",
       "25993       ./git_repos/bacass/modules/nf-core/fastp/main.nf           85   \n",
       "25994       ./git_repos/bacass/modules/nf-core/fastp/main.nf           86   \n",
       "25995       ./git_repos/bacass/modules/nf-core/fastp/main.nf           87   \n",
       "25996       ./git_repos/bacass/modules/nf-core/fastp/main.nf           94   \n",
       "25997       ./git_repos/bacass/modules/nf-core/fastp/main.nf           98   \n",
       "25998  ./git_repos/demultiplex/modules/nf-core/fastp/main.nf           39   \n",
       "25999  ./git_repos/demultiplex/modules/nf-core/fastp/main.nf           43   \n",
       "26000  ./git_repos/demultiplex/modules/nf-core/fastp/main.nf           44   \n",
       "26001  ./git_repos/demultiplex/modules/nf-core/fastp/main.nf           48   \n",
       "26002  ./git_repos/demultiplex/modules/nf-core/fastp/main.nf           49   \n",
       "26003  ./git_repos/demultiplex/modules/nf-core/fastp/main.nf           53   \n",
       "26004  ./git_repos/demultiplex/modules/nf-core/fastp/main.nf           60   \n",
       "26005  ./git_repos/demultiplex/modules/nf-core/fastp/main.nf           62   \n",
       "26006  ./git_repos/demultiplex/modules/nf-core/fastp/main.nf           64   \n",
       "26007  ./git_repos/demultiplex/modules/nf-core/fastp/main.nf           65   \n",
       "26008  ./git_repos/demultiplex/modules/nf-core/fastp/main.nf           69   \n",
       "26009  ./git_repos/demultiplex/modules/nf-core/fastp/main.nf           73   \n",
       "26010  ./git_repos/demultiplex/modules/nf-core/fastp/main.nf           81   \n",
       "26011  ./git_repos/demultiplex/modules/nf-core/fastp/main.nf           84   \n",
       "26012  ./git_repos/demultiplex/modules/nf-core/fastp/main.nf           85   \n",
       "26013  ./git_repos/demultiplex/modules/nf-core/fastp/main.nf           86   \n",
       "26014  ./git_repos/demultiplex/modules/nf-core/fastp/main.nf           87   \n",
       "26015  ./git_repos/demultiplex/modules/nf-core/fastp/main.nf           94   \n",
       "26016  ./git_repos/demultiplex/modules/nf-core/fastp/main.nf           98   \n",
       "\n",
       "                                                 line_content  Assigned Group  \n",
       "25669                                                fastp \\\\               6  \n",
       "25670                          --json ${prefix}.fastp.json \\\\               6  \n",
       "25671                          --html ${prefix}.fastp.html \\\\               6  \n",
       "25672                    2> >(tee ${prefix}.fastp.log >&2) \\\\               6  \n",
       "25673                    | gzip -c > ${prefix}.fastp.fastq.gz               6  \n",
       "25674  fastp: \\$(fastp --version 2>&1 | sed -e \"s/fastp //g\")               6  \n",
       "25675                                                fastp \\\\               6  \n",
       "25676                     --out1  ${prefix}.fastp.fastq.gz \\\\               6  \n",
       "25677                          --json ${prefix}.fastp.json \\\\               6  \n",
       "25678                          --html ${prefix}.fastp.html \\\\               6  \n",
       "25679                       2> >(tee ${prefix}.fastp.log >&2)               6  \n",
       "25680  fastp: \\$(fastp --version 2>&1 | sed -e \"s/fastp //g\")               6  \n",
       "25681                                                fastp \\\\               6  \n",
       "25682                    --out1 ${prefix}_1.fastp.fastq.gz \\\\               6  \n",
       "25683                    --out2 ${prefix}_2.fastp.fastq.gz \\\\               6  \n",
       "25684                          --json ${prefix}.fastp.json \\\\               6  \n",
       "25685                          --html ${prefix}.fastp.html \\\\               6  \n",
       "25686                       2> >(tee ${prefix}.fastp.log >&2)               6  \n",
       "25687  fastp: \\$(fastp --version 2>&1 | sed -e \"s/fastp //g\")               6  \n",
       "25688                            touch \"${prefix}.fastp.json\"               6  \n",
       "25689                            touch \"${prefix}.fastp.html\"               6  \n",
       "25690                             touch \"${prefix}.fastp.log\"               6  \n",
       "25691  fastp: \\$(fastp --version 2>&1 | sed -e \"s/fastp //g\")               6  \n",
       "25692                          fastp --version &> v_fastp.txt               6  \n",
       "25693                                                fastp \\\\               6  \n",
       "...                                                       ...             ...  \n",
       "25992                    --out1 ${prefix}_1.fastp.fastq.gz \\\\               6  \n",
       "25993                    --out2 ${prefix}_2.fastp.fastq.gz \\\\               6  \n",
       "25994                          --json ${prefix}.fastp.json \\\\               6  \n",
       "25995                          --html ${prefix}.fastp.html \\\\               6  \n",
       "25996                                  2> ${prefix}.fastp.log               6  \n",
       "25997  fastp: \\$(fastp --version 2>&1 | sed -e \"s/fastp //g\")               6  \n",
       "25998                                                fastp \\\\               6  \n",
       "25999                          --json ${prefix}.fastp.json \\\\               6  \n",
       "26000                          --html ${prefix}.fastp.html \\\\               6  \n",
       "26001                    2> >(tee ${prefix}.fastp.log >&2) \\\\               6  \n",
       "26002                    | gzip -c > ${prefix}.fastp.fastq.gz               6  \n",
       "26003  fastp: \\$(fastp --version 2>&1 | sed -e \"s/fastp //g\")               6  \n",
       "26004                                                fastp \\\\               6  \n",
       "26005                     --out1  ${prefix}.fastp.fastq.gz \\\\               6  \n",
       "26006                          --json ${prefix}.fastp.json \\\\               6  \n",
       "26007                          --html ${prefix}.fastp.html \\\\               6  \n",
       "26008                       2> >(tee ${prefix}.fastp.log >&2)               6  \n",
       "26009  fastp: \\$(fastp --version 2>&1 | sed -e \"s/fastp //g\")               6  \n",
       "26010                                                fastp \\\\               6  \n",
       "26011                    --out1 ${prefix}_1.fastp.fastq.gz \\\\               6  \n",
       "26012                    --out2 ${prefix}_2.fastp.fastq.gz \\\\               6  \n",
       "26013                          --json ${prefix}.fastp.json \\\\               6  \n",
       "26014                          --html ${prefix}.fastp.html \\\\               6  \n",
       "26015                       2> >(tee ${prefix}.fastp.log >&2)               6  \n",
       "26016  fastp: \\$(fastp --version 2>&1 | sed -e \"s/fastp //g\")               6  \n",
       "\n",
       "[348 rows x 7 columns]"
      ]
     },
     "execution_count": 200,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_df[merged_df[\"operator\"]==\"fastp\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 201,
   "id": "6083dc6c",
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'bioinformatics_script' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "Cell \u001b[1;32mIn[201], line 1\u001b[0m\n\u001b[1;32m----> 1\u001b[0m bioinformatics_script[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124moperator\u001b[39m\u001b[38;5;124m'\u001b[39m]\n",
      "\u001b[1;31mNameError\u001b[0m: name 'bioinformatics_script' is not defined"
     ]
    }
   ],
   "source": [
    "bioinformatics_script['operator']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 202,
   "id": "44cb26ec",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0                           echo\n",
       "1                           echo\n",
       "2                           echo\n",
       "3                           echo\n",
       "4                           echo\n",
       "5                           echo\n",
       "6                           echo\n",
       "7                           echo\n",
       "8                           echo\n",
       "9                           echo\n",
       "10                          echo\n",
       "11                          echo\n",
       "12                          echo\n",
       "13                          echo\n",
       "14                          echo\n",
       "15                          echo\n",
       "16                          echo\n",
       "17                          echo\n",
       "18                          echo\n",
       "19                          echo\n",
       "20                          echo\n",
       "21                          echo\n",
       "22                          echo\n",
       "23                          echo\n",
       "24                          echo\n",
       "                  ...           \n",
       "29743                      2.bed\n",
       "29744                      2.bed\n",
       "29745                       $STR\n",
       "29746                       chr/\n",
       "29747                       chr/\n",
       "29748                       canu\n",
       "29749                       canu\n",
       "29750                   NanoStat\n",
       "29751                bcl-convert\n",
       "29752                bcl-convert\n",
       "29753    download_eggnog_data.py\n",
       "29754                 emapper.py\n",
       "29755                 emapper.py\n",
       "29756                 emapper.py\n",
       "29757                 emapper.py\n",
       "29758                 emapper.py\n",
       "29759      TransDecoder.LongOrfs\n",
       "29760      TransDecoder.LongOrfs\n",
       "29761       TransDecoder.Predict\n",
       "29762       TransDecoder.Predict\n",
       "29763                   bbmap.sh\n",
       "29764                   bbmap.sh\n",
       "29765              fragments.bed\n",
       "29766                   contacts\n",
       "29767                   contacts\n",
       "Name: operator, Length: 29768, dtype: object"
      ]
     },
     "execution_count": 202,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_df['operator']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 203,
   "id": "97dc9e78",
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'bioinformatics_script' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "Cell \u001b[1;32mIn[203], line 1\u001b[0m\n\u001b[1;32m----> 1\u001b[0m new_rows \u001b[38;5;241m=\u001b[39m bioinformatics_script[\u001b[38;5;241m~\u001b[39mbioinformatics_script[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124moperator\u001b[39m\u001b[38;5;124m'\u001b[39m]\u001b[38;5;241m.\u001b[39misin(df_operator_clustered[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124moperator\u001b[39m\u001b[38;5;124m'\u001b[39m])]\n",
      "\u001b[1;31mNameError\u001b[0m: name 'bioinformatics_script' is not defined"
     ]
    }
   ],
   "source": [
    "new_rows = bioinformatics_script[~bioinformatics_script['operator'].isin(df_operator_clustered['operator'])]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 204,
   "id": "ba9bad2d",
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'bioinformatics_script' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "Cell \u001b[1;32mIn[204], line 1\u001b[0m\n\u001b[1;32m----> 1\u001b[0m bioinformatics_script\n",
      "\u001b[1;31mNameError\u001b[0m: name 'bioinformatics_script' is not defined"
     ]
    }
   ],
   "source": [
    "bioinformatics_script"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 205,
   "id": "7e296a8e",
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'bioinformatics_script' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "Cell \u001b[1;32mIn[205], line 1\u001b[0m\n\u001b[1;32m----> 1\u001b[0m new_rows \u001b[38;5;241m=\u001b[39m bioinformatics_script[bioinformatics_script[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124moperator\u001b[39m\u001b[38;5;124m'\u001b[39m]\u001b[38;5;241m.\u001b[39misin(df_operator_clustered[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124moperator\u001b[39m\u001b[38;5;124m'\u001b[39m])]\n\u001b[0;32m      2\u001b[0m new_rows\n",
      "\u001b[1;31mNameError\u001b[0m: name 'bioinformatics_script' is not defined"
     ]
    }
   ],
   "source": [
    "new_rows = bioinformatics_script[bioinformatics_script['operator'].isin(df_operator_clustered['operator'])]\n",
    "new_rows"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 206,
   "id": "abd61462",
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'new_rows' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "Cell \u001b[1;32mIn[206], line 2\u001b[0m\n\u001b[0;32m      1\u001b[0m \u001b[38;5;66;03m# Concatenate df_operator_clustered and the filtered rows from bioinformatics_script\u001b[39;00m\n\u001b[1;32m----> 2\u001b[0m extended_df \u001b[38;5;241m=\u001b[39m pd\u001b[38;5;241m.\u001b[39mconcat([df_operator_clustered, new_rows], ignore_index\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mTrue\u001b[39;00m)\n",
      "\u001b[1;31mNameError\u001b[0m: name 'new_rows' is not defined"
     ]
    }
   ],
   "source": [
    "# Concatenate df_operator_clustered and the filtered rows from bioinformatics_script\n",
    "extended_df = pd.concat([df_operator_clustered, new_rows], ignore_index=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 207,
   "id": "9f1c6edb",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>operator</th>\n",
       "      <th>Assigned Group</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>{log}</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>{output}</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>samtools</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>echo</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>shell:</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>python</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>mkdir</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>&amp;1</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>rm</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>bcftools</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>mv</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>cut</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>set</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>cat</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>sed</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>awk</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>if</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>fi</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>grep</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>Rscript</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>cp</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>cd</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>bedtools</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>{input}</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>touch</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4945</th>\n",
       "      <td>{params.base_dir}/{wildcards.sample}.ribosomes.fastq</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4946</th>\n",
       "      <td>INDIR=$dirname</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4947</th>\n",
       "      <td>{params.export_dir}/without_names/humann2_${{group}}_unstratified.spf</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4948</th>\n",
       "      <td>{params.export_dir}/with_names/humann2_${{group}}_named_unstratified.spf</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4949</th>\n",
       "      <td>{params.export_dir}/without_names/humann2_${{group}}_relab_unstratified.spf</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4950</th>\n",
       "      <td>{params.export_dir}/with_names/humann2_${{group}}_named_relab_unstratified.spf</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4951</th>\n",
       "      <td>{input.isoforms}</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4952</th>\n",
       "      <td>{params.database}</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4953</th>\n",
       "      <td>{params.parallel}</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4954</th>\n",
       "      <td>init</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4955</th>\n",
       "      <td>{input.swissprot_blastp}</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4956</th>\n",
       "      <td>{input.swissprot_blastx}</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4957</th>\n",
       "      <td>LOAD_rnammer</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4958</th>\n",
       "      <td>{input.rRNA}</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4959</th>\n",
       "      <td>report</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4960</th>\n",
       "      <td>{output.annotation_table}</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4961</th>\n",
       "      <td>rwc</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4962</th>\n",
       "      <td>sizes.txt</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4963</th>\n",
       "      <td>get_parent_dir={input.sa};</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4964</th>\n",
       "      <td>drop</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4965</th>\n",
       "      <td>{output.done}</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4966</th>\n",
       "      <td>format1=$echo</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4967</th>\n",
       "      <td>format2=$echo</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4968</th>\n",
       "      <td>tmpdir=$mktemp</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4969</th>\n",
       "      <td>{input.star_aln}</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4970 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                            operator  \\\n",
       "0                                                                              {log}   \n",
       "1                                                                           {output}   \n",
       "2                                                                           samtools   \n",
       "3                                                                               echo   \n",
       "4                                                                             shell:   \n",
       "5                                                                             python   \n",
       "6                                                                              mkdir   \n",
       "7                                                                                 &1   \n",
       "8                                                                                 rm   \n",
       "9                                                                           bcftools   \n",
       "10                                                                                mv   \n",
       "11                                                                               cut   \n",
       "12                                                                               set   \n",
       "13                                                                               cat   \n",
       "14                                                                               sed   \n",
       "15                                                                               awk   \n",
       "16                                                                                if   \n",
       "17                                                                                fi   \n",
       "18                                                                              grep   \n",
       "19                                                                           Rscript   \n",
       "20                                                                                cp   \n",
       "21                                                                                cd   \n",
       "22                                                                          bedtools   \n",
       "23                                                                           {input}   \n",
       "24                                                                             touch   \n",
       "...                                                                              ...   \n",
       "4945                            {params.base_dir}/{wildcards.sample}.ribosomes.fastq   \n",
       "4946                                                                  INDIR=$dirname   \n",
       "4947           {params.export_dir}/without_names/humann2_${{group}}_unstratified.spf   \n",
       "4948        {params.export_dir}/with_names/humann2_${{group}}_named_unstratified.spf   \n",
       "4949     {params.export_dir}/without_names/humann2_${{group}}_relab_unstratified.spf   \n",
       "4950  {params.export_dir}/with_names/humann2_${{group}}_named_relab_unstratified.spf   \n",
       "4951                                                                {input.isoforms}   \n",
       "4952                                                               {params.database}   \n",
       "4953                                                               {params.parallel}   \n",
       "4954                                                                            init   \n",
       "4955                                                        {input.swissprot_blastp}   \n",
       "4956                                                        {input.swissprot_blastx}   \n",
       "4957                                                                    LOAD_rnammer   \n",
       "4958                                                                    {input.rRNA}   \n",
       "4959                                                                          report   \n",
       "4960                                                       {output.annotation_table}   \n",
       "4961                                                                             rwc   \n",
       "4962                                                                       sizes.txt   \n",
       "4963                                                      get_parent_dir={input.sa};   \n",
       "4964                                                                            drop   \n",
       "4965                                                                   {output.done}   \n",
       "4966                                                                   format1=$echo   \n",
       "4967                                                                   format2=$echo   \n",
       "4968                                                                  tmpdir=$mktemp   \n",
       "4969                                                                {input.star_aln}   \n",
       "\n",
       "      Assigned Group  \n",
       "0                  6  \n",
       "1                  6  \n",
       "2                  0  \n",
       "3                  2  \n",
       "4                  4  \n",
       "5                  4  \n",
       "6                  4  \n",
       "7                  6  \n",
       "8                  3  \n",
       "9                  0  \n",
       "10                 3  \n",
       "11                 2  \n",
       "12                 6  \n",
       "13                 3  \n",
       "14                 6  \n",
       "15                 2  \n",
       "16                 1  \n",
       "17                 1  \n",
       "18                 2  \n",
       "19                 4  \n",
       "20                 3  \n",
       "21                 3  \n",
       "22                 0  \n",
       "23                 6  \n",
       "24                 6  \n",
       "...              ...  \n",
       "4945               0  \n",
       "4946               1  \n",
       "4947               1  \n",
       "4948               1  \n",
       "4949               1  \n",
       "4950               1  \n",
       "4951               3  \n",
       "4952               2  \n",
       "4953               2  \n",
       "4954               6  \n",
       "4955               4  \n",
       "4956               4  \n",
       "4957               4  \n",
       "4958               4  \n",
       "4959               4  \n",
       "4960               6  \n",
       "4961               4  \n",
       "4962               6  \n",
       "4963               1  \n",
       "4964               4  \n",
       "4965               1  \n",
       "4966               2  \n",
       "4967               2  \n",
       "4968               1  \n",
       "4969               4  \n",
       "\n",
       "[4970 rows x 2 columns]"
      ]
     },
     "execution_count": 207,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_operator_clustered"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 208,
   "id": "9dfa20d6",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "operator\n",
      "fastq                              1166\n",
      "samtools                           1098\n",
      "fasta                              1021\n",
      "fastq.gz                            645\n",
      "fastqc                              236\n",
      "bedtools                            154\n",
      "bcftools                            125\n",
      "bwa-mem2                             44\n",
      "biom                                 40\n",
      "$samtools                            37\n",
      "seqtk                                26\n",
      "fgbio                                24\n",
      "seqkit                               20\n",
      "fasta.gz                             18\n",
      "bamtools                             14\n",
      "gtf2bed                              12\n",
      "biotype                              11\n",
      "bedGraphToBigWig                     11\n",
      "bamCoverage                          10\n",
      "mmseqs                                9\n",
      "{params.genome}                       8\n",
      "transcripts.fasta                     6\n",
      "bedClip                               4\n",
      "complementBed                         3\n",
      "sortBed                               3\n",
      "mergeBed                              2\n",
      "intersectBed                          2\n",
      "bamToBed                              2\n",
      "genomeCoverageBed                     2\n",
      "fasta-splitter                        2\n",
      "bedops                                2\n",
      "isoseq3                               2\n",
      "2.bed                                 2\n",
      "jgi_summarize_bam_contig_depths       1\n",
      "gtf2bed.pl                            1\n",
      "findMotifsGenome.pl                   1\n",
      "bismark_genome_preparation            1\n",
      "bedToBigBed                           1\n",
      "fragments.bed                         1\n",
      "Name: count, dtype: int64\n"
     ]
    }
   ],
   "source": [
    "filtered_df = merged_df[merged_df['Assigned Group'] == 0]\n",
    "\n",
    "bio_tools = filtered_df['operator'].value_counts()\n",
    "\n",
    "print(bio_tools)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d23e0181",
   "metadata": {},
   "source": [
    "Discover the Assigned Groups"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 209,
   "id": "5df90ce8",
   "metadata": {},
   "outputs": [],
   "source": [
    "bioinformatic_tools = merged_df[merged_df['Assigned Group'] == 0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 210,
   "id": "490ba30a",
   "metadata": {},
   "outputs": [],
   "source": [
    "count_operators = merged_df[merged_df['Assigned Group'] == 0].groupby('operator')['operator'].count()\n",
    "nunique_pipeline = merged_df[merged_df['Assigned Group'] == 0].groupby('operator')['pipeline'].nunique()\n",
    "nunique_file_path = merged_df[merged_df['Assigned Group'] == 0].groupby('operator')['file_path'].nunique()\n",
    "\n",
    "bio_tools = pd.DataFrame({\n",
    "    'pipeline': nunique_pipeline,\n",
    "    'file_path': nunique_file_path,\n",
    "    'sum_operators': count_operators\n",
    "})\n",
    "\n",
    "bio_tools = bio_tools.sort_values(by='pipeline', ascending=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 211,
   "id": "011f28f2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pipeline</th>\n",
       "      <th>file_path</th>\n",
       "      <th>sum_operators</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>operator</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>fasta</th>\n",
       "      <td>73</td>\n",
       "      <td>527</td>\n",
       "      <td>1021</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>fastq</th>\n",
       "      <td>72</td>\n",
       "      <td>241</td>\n",
       "      <td>1166</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>fastqc</th>\n",
       "      <td>67</td>\n",
       "      <td>71</td>\n",
       "      <td>236</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>samtools</th>\n",
       "      <td>58</td>\n",
       "      <td>356</td>\n",
       "      <td>1098</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>fastq.gz</th>\n",
       "      <td>57</td>\n",
       "      <td>132</td>\n",
       "      <td>645</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>$samtools</th>\n",
       "      <td>24</td>\n",
       "      <td>37</td>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>bedtools</th>\n",
       "      <td>21</td>\n",
       "      <td>58</td>\n",
       "      <td>154</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gtf2bed</th>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>bcftools</th>\n",
       "      <td>12</td>\n",
       "      <td>39</td>\n",
       "      <td>125</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>bedGraphToBigWig</th>\n",
       "      <td>10</td>\n",
       "      <td>10</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>seqtk</th>\n",
       "      <td>9</td>\n",
       "      <td>11</td>\n",
       "      <td>26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>bwa-mem2</th>\n",
       "      <td>7</td>\n",
       "      <td>14</td>\n",
       "      <td>44</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>fasta.gz</th>\n",
       "      <td>6</td>\n",
       "      <td>10</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>bamtools</th>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>bedClip</th>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>biom</th>\n",
       "      <td>4</td>\n",
       "      <td>19</td>\n",
       "      <td>40</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>biotype</th>\n",
       "      <td>4</td>\n",
       "      <td>6</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>seqkit</th>\n",
       "      <td>4</td>\n",
       "      <td>5</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>bamCoverage</th>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>complementBed</th>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>sortBed</th>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>transcripts.fasta</th>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mmseqs</th>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mergeBed</th>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>intersectBed</th>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>fgbio</th>\n",
       "      <td>2</td>\n",
       "      <td>10</td>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>genomeCoverageBed</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gtf2bed.pl</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>fragments.bed</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>isoseq3</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>jgi_summarize_bam_contig_depths</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>findMotifsGenome.pl</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>bamToBed</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2.bed</th>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>fasta-splitter</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>bismark_genome_preparation</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>bedops</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>bedToBigBed</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>{params.genome}</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                 pipeline  file_path  sum_operators\n",
       "operator                                                           \n",
       "fasta                                  73        527           1021\n",
       "fastq                                  72        241           1166\n",
       "fastqc                                 67         71            236\n",
       "samtools                               58        356           1098\n",
       "fastq.gz                               57        132            645\n",
       "$samtools                              24         37             37\n",
       "bedtools                               21         58            154\n",
       "gtf2bed                                12         12             12\n",
       "bcftools                               12         39            125\n",
       "bedGraphToBigWig                       10         10             11\n",
       "seqtk                                   9         11             26\n",
       "bwa-mem2                                7         14             44\n",
       "fasta.gz                                6         10             18\n",
       "bamtools                                5          5             14\n",
       "bedClip                                 4          4              4\n",
       "biom                                    4         19             40\n",
       "biotype                                 4          6             11\n",
       "seqkit                                  4          5             20\n",
       "bamCoverage                             3          3             10\n",
       "complementBed                           3          3              3\n",
       "sortBed                                 3          3              3\n",
       "transcripts.fasta                       3          3              6\n",
       "mmseqs                                  2          3              9\n",
       "mergeBed                                2          2              2\n",
       "intersectBed                            2          2              2\n",
       "fgbio                                   2         10             24\n",
       "genomeCoverageBed                       1          1              2\n",
       "gtf2bed.pl                              1          1              1\n",
       "fragments.bed                           1          1              1\n",
       "isoseq3                                 1          1              2\n",
       "jgi_summarize_bam_contig_depths         1          1              1\n",
       "findMotifsGenome.pl                     1          1              1\n",
       "bamToBed                                1          1              2\n",
       "2.bed                                   1          2              2\n",
       "fasta-splitter                          1          1              2\n",
       "bismark_genome_preparation              1          1              1\n",
       "bedops                                  1          1              2\n",
       "bedToBigBed                             1          1              1\n",
       "{params.genome}                         1          1              8"
      ]
     },
     "execution_count": 211,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bio_tools.head(60)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 212,
   "id": "cd088093",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "operator\n",
       "else                             232\n",
       "done                             125\n",
       "while                             70\n",
       "classifier                        25\n",
       "elif                              16\n",
       "prefix=                           10\n",
       "VALIDATION_STRINGENCY=LENIENT      8\n",
       "NR==1                              7\n",
       "centrifuge                         7\n",
       "done;                              6\n",
       "SnpSift                            4\n",
       "LC_ALL=C                           4\n",
       "ASSUME_SORTED=true                 3\n",
       "sniffles                           3\n",
       "SORT_ORDER=coordinate              2\n",
       "centrifuge-kreport                 2\n",
       "=NF;i++                            2\n",
       "=NF;                               2\n",
       "download-antismash-databases       2\n",
       "VALIDATION_STRINGENCY=SILENT       2\n",
       "REMOVE_DUPLICATES=TRUE             2\n",
       "REMOVE_DUPLICATES=true             1\n",
       "RGPL=illumina                      1\n",
       "LC_COLLATE=C                       1\n",
       "ORS=                               1\n",
       "RGID=1                             1\n",
       "FNR==1                             1\n",
       "dos2unix                           1\n",
       "NR==2                              1\n",
       "chr=                               1\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 212,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_df['operator'][merged_df['Assigned Group'] == 1].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 213,
   "id": "0e11c772",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "operator\n",
       "echo                        2515\n",
       "params                       632\n",
       "print                        575\n",
       "printf                       468\n",
       "grep                         439\n",
       "head                         353\n",
       "gawk                          80\n",
       "MarkDuplicates                73\n",
       "cutadapt                      69\n",
       "paired                        62\n",
       "plotFingerprint               25\n",
       "fusioncatcher                 15\n",
       "porechop                       9\n",
       "mawk                           6\n",
       "abricate                       3\n",
       "zgrep                          3\n",
       "cuteSV                         3\n",
       "/head                          2\n",
       "{params.aligner}               2\n",
       "egrep                          2\n",
       "{params.min_contig_size}       1\n",
       "{params.dbsnp}                 1\n",
       "{params.mills}                 1\n",
       "deduplicate_bismark            1\n",
       "DB=`echo                       1\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 213,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_df['operator'][merged_df['Assigned Group'] == 2].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 214,
   "id": "44c0f80d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "operator\n",
       "zcat                   13\n",
       "sourmash               11\n",
       "/bin/rm                 6\n",
       "coverm                  3\n",
       "MapRTTransformer        2\n",
       "thermorawfileparser     2\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 214,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_df['operator'][merged_df['Assigned Group'] == 3].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 215,
   "id": "4c2b0b66",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "operator\n",
       "mkdir                            473\n",
       "python                           415\n",
       "sort                             374\n",
       "script                           341\n",
       "report                           273\n",
       "export                           246\n",
       "Rscript                          195\n",
       "summary                          186\n",
       "feature                          144\n",
       "results                          138\n",
       "source                           130\n",
       "picard                           122\n",
       "rename                           111\n",
       "graph                             78\n",
       "kraken2                           66\n",
       "return                            64\n",
       "trim_galore                       54\n",
       "trimmed                           51\n",
       "Filter                            51\n",
       "histogram                         43\n",
       "drop                              43\n",
       "convert                           37\n",
       "gffread                           29\n",
       "featureCounts                     29\n",
       "cram                              28\n",
       "                                ... \n",
       "coverage2cytosine                  1\n",
       "sort}                              1\n",
       "dv_make_examples.py                1\n",
       "art_illumina                       1\n",
       "Attribute                          1\n",
       "FeatureFinderMetaboIdent           1\n",
       "hmmpress                           1\n",
       "bismark2summary                    1\n",
       "draw_fusions.R                     1\n",
       "integrated                         1\n",
       "FilterConsensusReads               1\n",
       "amrfinder_update                   1\n",
       "run_busco                          1\n",
       "indexdb_rna                        1\n",
       "roary                              1\n",
       "MergeBamAlignment                  1\n",
       "sketch.sh                          1\n",
       "true;                              1\n",
       "/ilastik-release/bin/python        1\n",
       "concoct_coverage_table.py          1\n",
       "extract_fasta_bins.py              1\n",
       "cut_up_fasta.py                    1\n",
       "usearch                            1\n",
       "merge_cutup_clustering.py          1\n",
       "bismark_methylation_extractor      1\n",
       "Name: count, Length: 186, dtype: int64"
      ]
     },
     "execution_count": 215,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_df['operator'][merged_df['Assigned Group'] == 4].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 216,
   "id": "2011b15f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "operator\n",
       "mkdir                            473\n",
       "python                           415\n",
       "sort                             374\n",
       "script                           341\n",
       "report                           273\n",
       "export                           246\n",
       "Rscript                          195\n",
       "summary                          186\n",
       "feature                          144\n",
       "results                          138\n",
       "source                           130\n",
       "picard                           122\n",
       "rename                           111\n",
       "graph                             78\n",
       "kraken2                           66\n",
       "return                            64\n",
       "trim_galore                       54\n",
       "trimmed                           51\n",
       "Filter                            51\n",
       "histogram                         43\n",
       "drop                              43\n",
       "convert                           37\n",
       "gffread                           29\n",
       "featureCounts                     29\n",
       "cram                              28\n",
       "                                ... \n",
       "coverage2cytosine                  1\n",
       "sort}                              1\n",
       "dv_make_examples.py                1\n",
       "art_illumina                       1\n",
       "Attribute                          1\n",
       "FeatureFinderMetaboIdent           1\n",
       "hmmpress                           1\n",
       "bismark2summary                    1\n",
       "draw_fusions.R                     1\n",
       "integrated                         1\n",
       "FilterConsensusReads               1\n",
       "amrfinder_update                   1\n",
       "run_busco                          1\n",
       "indexdb_rna                        1\n",
       "roary                              1\n",
       "MergeBamAlignment                  1\n",
       "sketch.sh                          1\n",
       "true;                              1\n",
       "/ilastik-release/bin/python        1\n",
       "concoct_coverage_table.py          1\n",
       "extract_fasta_bins.py              1\n",
       "cut_up_fasta.py                    1\n",
       "usearch                            1\n",
       "merge_cutup_clustering.py          1\n",
       "bismark_methylation_extractor      1\n",
       "Name: count, Length: 186, dtype: int64"
      ]
     },
     "execution_count": 216,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_df['operator'][merged_df['Assigned Group'] == 4].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 217,
   "id": "3c2fc42a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "operator\n",
       "fastq                              1166\n",
       "samtools                           1098\n",
       "fasta                              1021\n",
       "fastq.gz                            645\n",
       "fastqc                              236\n",
       "bedtools                            154\n",
       "bcftools                            125\n",
       "bwa-mem2                             44\n",
       "biom                                 40\n",
       "$samtools                            37\n",
       "seqtk                                26\n",
       "fgbio                                24\n",
       "seqkit                               20\n",
       "fasta.gz                             18\n",
       "bamtools                             14\n",
       "gtf2bed                              12\n",
       "biotype                              11\n",
       "bedGraphToBigWig                     11\n",
       "bamCoverage                          10\n",
       "mmseqs                                9\n",
       "{params.genome}                       8\n",
       "transcripts.fasta                     6\n",
       "bedClip                               4\n",
       "complementBed                         3\n",
       "sortBed                               3\n",
       "mergeBed                              2\n",
       "intersectBed                          2\n",
       "bamToBed                              2\n",
       "genomeCoverageBed                     2\n",
       "fasta-splitter                        2\n",
       "bedops                                2\n",
       "isoseq3                               2\n",
       "2.bed                                 2\n",
       "jgi_summarize_bam_contig_depths       1\n",
       "gtf2bed.pl                            1\n",
       "findMotifsGenome.pl                   1\n",
       "bismark_genome_preparation            1\n",
       "bedToBigBed                           1\n",
       "fragments.bed                         1\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 217,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.set_option(\"display.min_rows\", 50)\n",
    "merged_df['operator'][merged_df['Assigned Group'] == 0].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 218,
   "id": "93effa9a",
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_df['pipeline_process_name'] = merged_df['pipeline'] + \".\" + merged_df['process_name']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 219,
   "id": "b8c112ef",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "pipeline                  482\n",
       "pipeline_process_name    1831\n",
       "dtype: int64"
      ]
     },
     "execution_count": 219,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bio_tools = merged_df[merged_df['Assigned Group'] == 0]\n",
    "\n",
    "bio_tools = bio_tools.groupby('operator').agg({'pipeline': 'nunique', 'pipeline_process_name': 'nunique'}).sort_values(by='pipeline', ascending=False)\n",
    "bio_tools.sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 220,
   "id": "11c504f7",
   "metadata": {},
   "outputs": [],
   "source": [
    "grouped_df = merged_df.groupby('Assigned Group').agg({'pipeline': 'nunique', 'pipeline_process_name': 'nunique', 'operator': 'count'}).reset_index()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 221,
   "id": "eda5b7e5",
   "metadata": {},
   "outputs": [],
   "source": [
    "sum_processes = 3081"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 222,
   "id": "f3622095",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pipeline</th>\n",
       "      <th>process_name</th>\n",
       "      <th>operator</th>\n",
       "      <th>file_path</th>\n",
       "      <th>line_number</th>\n",
       "      <th>line_content</th>\n",
       "      <th>Assigned Group</th>\n",
       "      <th>pipeline_process_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>diaproteomics</td>\n",
       "      <td>get_software_versions</td>\n",
       "      <td>echo</td>\n",
       "      <td>./git_repos/diaproteomics/main.nf</td>\n",
       "      <td>335</td>\n",
       "      <td>echo $workflow.manifest.version &gt; v_pipeline.txt</td>\n",
       "      <td>2</td>\n",
       "      <td>diaproteomics.get_software_versions</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>diaproteomics</td>\n",
       "      <td>get_software_versions</td>\n",
       "      <td>echo</td>\n",
       "      <td>./git_repos/diaproteomics/main.nf</td>\n",
       "      <td>336</td>\n",
       "      <td>echo $workflow.nextflow.version &gt; v_nextflow.txt</td>\n",
       "      <td>2</td>\n",
       "      <td>diaproteomics.get_software_versions</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>diaproteomics</td>\n",
       "      <td>statistical_post_processing</td>\n",
       "      <td>echo</td>\n",
       "      <td>./git_repos/diaproteomics/main.nf</td>\n",
       "      <td>971</td>\n",
       "      <td>msstats.R &gt; msstats.log || echo \"Optional MSstats step failed. Please check logs and re-run or do a manual statistical analysis.\"</td>\n",
       "      <td>2</td>\n",
       "      <td>diaproteomics.statistical_post_processing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>readsimulator</td>\n",
       "      <td>BOWTIE2_ALIGN</td>\n",
       "      <td>echo</td>\n",
       "      <td>./git_repos/readsimulator/modules/nf-core/bowtie2/align/main.nf</td>\n",
       "      <td>47</td>\n",
       "      <td>[ -z \"\\$INDEX\" ] &amp;&amp; echo \"Bowtie2 index files not found\" 1&gt;&amp;2 &amp;&amp; exit 1</td>\n",
       "      <td>2</td>\n",
       "      <td>readsimulator.BOWTIE2_ALIGN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>readsimulator</td>\n",
       "      <td>BOWTIE2_ALIGN</td>\n",
       "      <td>echo</td>\n",
       "      <td>./git_repos/readsimulator/modules/nf-core/bowtie2/align/main.nf</td>\n",
       "      <td>68</td>\n",
       "      <td>bowtie2: \\$(echo \\$(bowtie2 --version 2&gt;&amp;1) | sed 's/^.*bowtie2-align-s version //; s/ .*\\$//')</td>\n",
       "      <td>2</td>\n",
       "      <td>readsimulator.BOWTIE2_ALIGN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>readsimulator</td>\n",
       "      <td>BOWTIE2_ALIGN</td>\n",
       "      <td>echo</td>\n",
       "      <td>./git_repos/readsimulator/modules/nf-core/bowtie2/align/main.nf</td>\n",
       "      <td>69</td>\n",
       "      <td>samtools: \\$(echo \\$(samtools --version 2&gt;&amp;1) | sed 's/^.*samtools //; s/Using.*\\$//')</td>\n",
       "      <td>2</td>\n",
       "      <td>readsimulator.BOWTIE2_ALIGN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>readsimulator</td>\n",
       "      <td>BOWTIE2_ALIGN</td>\n",
       "      <td>echo</td>\n",
       "      <td>./git_repos/readsimulator/modules/nf-core/bowtie2/align/main.nf</td>\n",
       "      <td>88</td>\n",
       "      <td>bowtie2: \\$(echo \\$(bowtie2 --version 2&gt;&amp;1) | sed 's/^.*bowtie2-align-s version //; s/ .*\\$//')</td>\n",
       "      <td>2</td>\n",
       "      <td>readsimulator.BOWTIE2_ALIGN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>readsimulator</td>\n",
       "      <td>BOWTIE2_ALIGN</td>\n",
       "      <td>echo</td>\n",
       "      <td>./git_repos/readsimulator/modules/nf-core/bowtie2/align/main.nf</td>\n",
       "      <td>89</td>\n",
       "      <td>samtools: \\$(echo \\$(samtools --version 2&gt;&amp;1) | sed 's/^.*samtools //; s/Using.*\\$//')</td>\n",
       "      <td>2</td>\n",
       "      <td>readsimulator.BOWTIE2_ALIGN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>readsimulator</td>\n",
       "      <td>BOWTIE2_BUILD</td>\n",
       "      <td>echo</td>\n",
       "      <td>./git_repos/readsimulator/modules/nf-core/bowtie2/build/main.nf</td>\n",
       "      <td>27</td>\n",
       "      <td>bowtie2: \\$(echo \\$(bowtie2 --version 2&gt;&amp;1) | sed 's/^.*bowtie2-align-s version //; s/ .*\\$//')</td>\n",
       "      <td>2</td>\n",
       "      <td>readsimulator.BOWTIE2_BUILD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>readsimulator</td>\n",
       "      <td>BOWTIE2_BUILD</td>\n",
       "      <td>echo</td>\n",
       "      <td>./git_repos/readsimulator/modules/nf-core/bowtie2/build/main.nf</td>\n",
       "      <td>39</td>\n",
       "      <td>bowtie2: \\$(echo \\$(bowtie2 --version 2&gt;&amp;1) | sed 's/^.*bowtie2-align-s version //; s/ .*\\$//')</td>\n",
       "      <td>2</td>\n",
       "      <td>readsimulator.BOWTIE2_BUILD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>readsimulator</td>\n",
       "      <td>SAMTOOLS_INDEX</td>\n",
       "      <td>echo</td>\n",
       "      <td>./git_repos/readsimulator/modules/nf-core/samtools/index/main.nf</td>\n",
       "      <td>33</td>\n",
       "      <td>samtools: \\$(echo \\$(samtools --version 2&gt;&amp;1) | sed 's/^.*samtools //; s/Using.*\\$//')</td>\n",
       "      <td>2</td>\n",
       "      <td>readsimulator.SAMTOOLS_INDEX</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>readsimulator</td>\n",
       "      <td>SAMTOOLS_INDEX</td>\n",
       "      <td>echo</td>\n",
       "      <td>./git_repos/readsimulator/modules/nf-core/samtools/index/main.nf</td>\n",
       "      <td>45</td>\n",
       "      <td>samtools: \\$(echo \\$(samtools --version 2&gt;&amp;1) | sed 's/^.*samtools //; s/Using.*\\$//')</td>\n",
       "      <td>2</td>\n",
       "      <td>readsimulator.SAMTOOLS_INDEX</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>readsimulator</td>\n",
       "      <td>INSILICOSEQ_GENERATE</td>\n",
       "      <td>echo</td>\n",
       "      <td>./git_repos/readsimulator/modules/local/insilicoseq/generate/main.nf</td>\n",
       "      <td>29</td>\n",
       "      <td>seed=\\$(echo $seed | sed 's/\\\\[//g' | sed 's/\\\\]//g')</td>\n",
       "      <td>2</td>\n",
       "      <td>readsimulator.INSILICOSEQ_GENERATE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>readsimulator</td>\n",
       "      <td>INSILICOSEQ_GENERATE</td>\n",
       "      <td>echo</td>\n",
       "      <td>./git_repos/readsimulator/modules/local/insilicoseq/generate/main.nf</td>\n",
       "      <td>30</td>\n",
       "      <td>prefix=\\$(echo $prefix | sed 's/\\\\[//g' | sed 's/\\\\]//g')</td>\n",
       "      <td>2</td>\n",
       "      <td>readsimulator.INSILICOSEQ_GENERATE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>readsimulator</td>\n",
       "      <td>INSILICOSEQ_GENERATE</td>\n",
       "      <td>echo</td>\n",
       "      <td>./git_repos/readsimulator/modules/local/insilicoseq/generate/main.nf</td>\n",
       "      <td>53</td>\n",
       "      <td>seed=\\$(echo $seed | sed 's/\\\\[//g' | sed 's/\\\\]//g')</td>\n",
       "      <td>2</td>\n",
       "      <td>readsimulator.INSILICOSEQ_GENERATE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>readsimulator</td>\n",
       "      <td>INSILICOSEQ_GENERATE</td>\n",
       "      <td>echo</td>\n",
       "      <td>./git_repos/readsimulator/modules/local/insilicoseq/generate/main.nf</td>\n",
       "      <td>54</td>\n",
       "      <td>prefix=\\$(echo $prefix | sed 's/\\\\[//g' | sed 's/\\\\]//g')</td>\n",
       "      <td>2</td>\n",
       "      <td>readsimulator.INSILICOSEQ_GENERATE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>readsimulator</td>\n",
       "      <td>MERGE_SAMPLESHEETS</td>\n",
       "      <td>echo</td>\n",
       "      <td>./git_repos/readsimulator/modules/local/custom/merge_samplesheets/main.nf</td>\n",
       "      <td>23</td>\n",
       "      <td>echo \\\\\"sample\\\\\",\\\\\"fastq_1\\\\\",\\\\\"fastq_2\\\\\" &gt; \"${prefix}_samplesheet.csv\"</td>\n",
       "      <td>2</td>\n",
       "      <td>readsimulator.MERGE_SAMPLESHEETS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>readsimulator</td>\n",
       "      <td>MERGE_SAMPLESHEETS</td>\n",
       "      <td>echo</td>\n",
       "      <td>./git_repos/readsimulator/modules/local/custom/merge_samplesheets/main.nf</td>\n",
       "      <td>26</td>\n",
       "      <td>echo &gt;&gt; \"${prefix}_samplesheet.csv\"</td>\n",
       "      <td>2</td>\n",
       "      <td>readsimulator.MERGE_SAMPLESHEETS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>callingcards</td>\n",
       "      <td>GTF2BED</td>\n",
       "      <td>echo</td>\n",
       "      <td>./git_repos/callingcards/modules/local/gtf2bed/main.nf</td>\n",
       "      <td>28</td>\n",
       "      <td>perl: \\$(echo \\$(perl --version 2&gt;&amp;1) | sed 's/.*v\\\\(.*\\\\)) built.*/\\\\1/')</td>\n",
       "      <td>2</td>\n",
       "      <td>callingcards.GTF2BED</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>callingcards</td>\n",
       "      <td>DEMULTIPLEX</td>\n",
       "      <td>echo</td>\n",
       "      <td>./git_repos/callingcards/modules/local/callingcardstools/yeast/demultiplex/main.nf</td>\n",
       "      <td>36</td>\n",
       "      <td>callingcardstools: \\$(echo \\$(callingcardstools --version 2&gt;&amp;1) | sed 's/.* //')</td>\n",
       "      <td>2</td>\n",
       "      <td>callingcards.DEMULTIPLEX</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>callingcards</td>\n",
       "      <td>COUNT_HOPS</td>\n",
       "      <td>echo</td>\n",
       "      <td>./git_repos/callingcards/modules/local/callingcardstools/yeast/count_hops/main.nf</td>\n",
       "      <td>39</td>\n",
       "      <td>callingcardstools: \\$(echo \\$(callingcardstools --version 2&gt;&amp;1) | sed 's/.* //')</td>\n",
       "      <td>2</td>\n",
       "      <td>callingcards.COUNT_HOPS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>callingcards</td>\n",
       "      <td>CONCATQC</td>\n",
       "      <td>echo</td>\n",
       "      <td>./git_repos/callingcards/modules/local/callingcardstools/yeast/concatQC/main.nf</td>\n",
       "      <td>33</td>\n",
       "      <td>callingcardstools: \\$(echo \\$(callingcardstools --version 2&gt;&amp;1) | sed 's/.* //')</td>\n",
       "      <td>2</td>\n",
       "      <td>callingcards.CONCATQC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>callingcards</td>\n",
       "      <td>CONCATQC</td>\n",
       "      <td>echo</td>\n",
       "      <td>./git_repos/callingcards/modules/local/callingcardstools/mammals/concatQC/main.nf</td>\n",
       "      <td>38</td>\n",
       "      <td>callingcardstools: \\$(echo \\$(callingcardstools --version 2&gt;&amp;1) | sed 's/.* //')</td>\n",
       "      <td>2</td>\n",
       "      <td>callingcards.CONCATQC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>callingcards</td>\n",
       "      <td>BOWTIE2_ALIGN</td>\n",
       "      <td>echo</td>\n",
       "      <td>./git_repos/callingcards/modules/nf-core/bowtie2/align/main.nf</td>\n",
       "      <td>47</td>\n",
       "      <td>[ -z \"\\$INDEX\" ] &amp;&amp; echo \"Bowtie2 index files not found\" 1&gt;&amp;2 &amp;&amp; exit 1</td>\n",
       "      <td>2</td>\n",
       "      <td>callingcards.BOWTIE2_ALIGN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>callingcards</td>\n",
       "      <td>BOWTIE2_ALIGN</td>\n",
       "      <td>echo</td>\n",
       "      <td>./git_repos/callingcards/modules/nf-core/bowtie2/align/main.nf</td>\n",
       "      <td>68</td>\n",
       "      <td>bowtie2: \\$(echo \\$(bowtie2 --version 2&gt;&amp;1) | sed 's/^.*bowtie2-align-s version //; s/ .*\\$//')</td>\n",
       "      <td>2</td>\n",
       "      <td>callingcards.BOWTIE2_ALIGN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29743</th>\n",
       "      <td>circrna</td>\n",
       "      <td>PARENT_GENE</td>\n",
       "      <td>2.bed</td>\n",
       "      <td>./git_repos/circrna/modules/local/annotation/parent_gene/main.nf</td>\n",
       "      <td>45</td>\n",
       "      <td>awk -v OFS=\"\\\\t\" '{print \\$4, \\$14}' master_bed12.bed &gt; circrna_host-gene.txt</td>\n",
       "      <td>0</td>\n",
       "      <td>circrna.PARENT_GENE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29744</th>\n",
       "      <td>circrna</td>\n",
       "      <td>ANNOTATION</td>\n",
       "      <td>2.bed</td>\n",
       "      <td>./git_repos/circrna/modules/local/annotation/full_annotation/main.nf</td>\n",
       "      <td>34</td>\n",
       "      <td>mv master_bed12.bed ${prefix}.bed.tmp</td>\n",
       "      <td>0</td>\n",
       "      <td>circrna.ANNOTATION</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29745</th>\n",
       "      <td>circrna</td>\n",
       "      <td>CIRIQUANT_YML</td>\n",
       "      <td>$STR</td>\n",
       "      <td>./git_repos/circrna/modules/local/ciriquant/yml/main.nf</td>\n",
       "      <td>38</td>\n",
       "      <td>printf \"name: ciriquant\\ntools:\\n  bwa: \\$BWA\\n  hisat2: \\$HISAT2\\n  stringtie: \\$STRINGTIE\\n  samtools: \\$SAMTOOLS\\n\\nreference:\\n  fasta: ${fasta_path}\\n  gtf: ${gtf_path}\\n  bwa_index: ${bwa_path}/\\$BWA_FILE\\n  hisat_index: ${hisat2_path}/${hisat2_prefix}\" &gt;&gt; travis.yml</td>\n",
       "      <td>6</td>\n",
       "      <td>circrna.CIRIQUANT_YML</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29746</th>\n",
       "      <td>circrna</td>\n",
       "      <td>DCC</td>\n",
       "      <td>chr/</td>\n",
       "      <td>./git_repos/circrna/modules/local/dcc/dcc/main.nf</td>\n",
       "      <td>27</td>\n",
       "      <td>sed -i 's/^chr//g' $gtf</td>\n",
       "      <td>4</td>\n",
       "      <td>circrna.DCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29747</th>\n",
       "      <td>circrna</td>\n",
       "      <td>DCC</td>\n",
       "      <td>chr/</td>\n",
       "      <td>./git_repos/circrna/modules/local/dcc/dcc/main.nf</td>\n",
       "      <td>43</td>\n",
       "      <td>sed -i 's/^chr//g' $gtf</td>\n",
       "      <td>4</td>\n",
       "      <td>circrna.DCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29748</th>\n",
       "      <td>bacass</td>\n",
       "      <td>CANU</td>\n",
       "      <td>canu</td>\n",
       "      <td>./git_repos/bacass/modules/nf-core/canu/main.nf</td>\n",
       "      <td>35</td>\n",
       "      <td>canu \\\\</td>\n",
       "      <td>6</td>\n",
       "      <td>bacass.CANU</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29749</th>\n",
       "      <td>bacass</td>\n",
       "      <td>CANU</td>\n",
       "      <td>canu</td>\n",
       "      <td>./git_repos/bacass/modules/nf-core/canu/main.nf</td>\n",
       "      <td>47</td>\n",
       "      <td>canu: \\$(echo \\$(canu --version 2&gt;&amp;1) | sed 's/^.*canu //; s/Using.*\\$//' )</td>\n",
       "      <td>6</td>\n",
       "      <td>bacass.CANU</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29750</th>\n",
       "      <td>bacass</td>\n",
       "      <td>NANOPLOT</td>\n",
       "      <td>NanoStat</td>\n",
       "      <td>./git_repos/bacass/modules/nf-core/nanoplot/main.nf</td>\n",
       "      <td>33</td>\n",
       "      <td>mv NanoStats.txt ${prefix}.txt</td>\n",
       "      <td>6</td>\n",
       "      <td>bacass.NANOPLOT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29751</th>\n",
       "      <td>demultiplex</td>\n",
       "      <td>BCLCONVERT</td>\n",
       "      <td>bcl-convert</td>\n",
       "      <td>./git_repos/demultiplex/modules/nf-core/bclconvert/main.nf</td>\n",
       "      <td>59</td>\n",
       "      <td>bcl-convert \\\\</td>\n",
       "      <td>4</td>\n",
       "      <td>demultiplex.BCLCONVERT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29752</th>\n",
       "      <td>demultiplex</td>\n",
       "      <td>BCLCONVERT</td>\n",
       "      <td>bcl-convert</td>\n",
       "      <td>./git_repos/demultiplex/modules/nf-core/bclconvert/main.nf</td>\n",
       "      <td>69</td>\n",
       "      <td>bclconvert: \\$(bcl-convert -V 2&gt;&amp;1 | head -n 1 | sed 's/^.*Version //')</td>\n",
       "      <td>4</td>\n",
       "      <td>demultiplex.BCLCONVERT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29753</th>\n",
       "      <td>metatdenovo</td>\n",
       "      <td>EGGNOG_DOWNLOAD</td>\n",
       "      <td>download_eggnog_data.py</td>\n",
       "      <td>./git_repos/metatdenovo/modules/local/eggnog/download.nf</td>\n",
       "      <td>21</td>\n",
       "      <td>download_eggnog_data.py \\\\</td>\n",
       "      <td>4</td>\n",
       "      <td>metatdenovo.EGGNOG_DOWNLOAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29754</th>\n",
       "      <td>metatdenovo</td>\n",
       "      <td>EGGNOG_DOWNLOAD</td>\n",
       "      <td>emapper.py</td>\n",
       "      <td>./git_repos/metatdenovo/modules/local/eggnog/download.nf</td>\n",
       "      <td>28</td>\n",
       "      <td>eggnog: \\$( echo \\$(emapper.py --version 2&gt;&amp;1)| sed 's/.* emapper-//' | sed 's/ \\\\/ Expected.*//')</td>\n",
       "      <td>4</td>\n",
       "      <td>metatdenovo.EGGNOG_DOWNLOAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29755</th>\n",
       "      <td>metatdenovo</td>\n",
       "      <td>EGGNOG_DOWNLOAD</td>\n",
       "      <td>emapper.py</td>\n",
       "      <td>./git_repos/metatdenovo/modules/local/eggnog/download.nf</td>\n",
       "      <td>45</td>\n",
       "      <td>eggnog: \\$( echo \\$(emapper.py --version 2&gt;&amp;1)| sed 's/.* emapper-//' )</td>\n",
       "      <td>4</td>\n",
       "      <td>metatdenovo.EGGNOG_DOWNLOAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29756</th>\n",
       "      <td>metatdenovo</td>\n",
       "      <td>EGGNOG_MAPPER</td>\n",
       "      <td>emapper.py</td>\n",
       "      <td>./git_repos/metatdenovo/modules/local/eggnog/mapper.nf</td>\n",
       "      <td>36</td>\n",
       "      <td>emapper.py \\\\</td>\n",
       "      <td>4</td>\n",
       "      <td>metatdenovo.EGGNOG_MAPPER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29757</th>\n",
       "      <td>metatdenovo</td>\n",
       "      <td>EGGNOG_MAPPER</td>\n",
       "      <td>emapper.py</td>\n",
       "      <td>./git_repos/metatdenovo/modules/local/eggnog/mapper.nf</td>\n",
       "      <td>50</td>\n",
       "      <td>eggnog: \\$( echo \\$(emapper.py --version 2&gt;&amp;1)| sed 's/.* emapper-//' | sed 's/ \\\\/ Expected.*//')</td>\n",
       "      <td>4</td>\n",
       "      <td>metatdenovo.EGGNOG_MAPPER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29758</th>\n",
       "      <td>metatdenovo</td>\n",
       "      <td>EGGNOG_MAPPER</td>\n",
       "      <td>emapper.py</td>\n",
       "      <td>./git_repos/metatdenovo/modules/local/eggnog/mapper.nf</td>\n",
       "      <td>62</td>\n",
       "      <td>eggnog: \\$( echo \\$(emapper.py --version 2&gt;&amp;1)| sed 's/.* emapper-//' | sed 's/ \\\\/ Expected.*//')</td>\n",
       "      <td>4</td>\n",
       "      <td>metatdenovo.EGGNOG_MAPPER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29759</th>\n",
       "      <td>metatdenovo</td>\n",
       "      <td>TRANSDECODER_LONGORF</td>\n",
       "      <td>TransDecoder.LongOrfs</td>\n",
       "      <td>./git_repos/metatdenovo/modules/nf-core/transdecoder/longorf/main.nf</td>\n",
       "      <td>29</td>\n",
       "      <td>TransDecoder.LongOrfs \\\\</td>\n",
       "      <td>4</td>\n",
       "      <td>metatdenovo.TRANSDECODER_LONGORF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29760</th>\n",
       "      <td>metatdenovo</td>\n",
       "      <td>TRANSDECODER_LONGORF</td>\n",
       "      <td>TransDecoder.LongOrfs</td>\n",
       "      <td>./git_repos/metatdenovo/modules/nf-core/transdecoder/longorf/main.nf</td>\n",
       "      <td>37</td>\n",
       "      <td>transdecoder: \\$(echo \\$(TransDecoder.LongOrfs --version) | sed -e \"s/TransDecoder.LongOrfs //g\")</td>\n",
       "      <td>4</td>\n",
       "      <td>metatdenovo.TRANSDECODER_LONGORF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29761</th>\n",
       "      <td>metatdenovo</td>\n",
       "      <td>TRANSDECODER_PREDICT</td>\n",
       "      <td>TransDecoder.Predict</td>\n",
       "      <td>./git_repos/metatdenovo/modules/nf-core/transdecoder/predict/main.nf</td>\n",
       "      <td>29</td>\n",
       "      <td>TransDecoder.Predict \\\\</td>\n",
       "      <td>4</td>\n",
       "      <td>metatdenovo.TRANSDECODER_PREDICT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29762</th>\n",
       "      <td>metatdenovo</td>\n",
       "      <td>TRANSDECODER_PREDICT</td>\n",
       "      <td>TransDecoder.Predict</td>\n",
       "      <td>./git_repos/metatdenovo/modules/nf-core/transdecoder/predict/main.nf</td>\n",
       "      <td>37</td>\n",
       "      <td>transdecoder: \\$(echo \\$(TransDecoder.Predict --version) | sed -e \"s/TransDecoder.Predict //g\")</td>\n",
       "      <td>4</td>\n",
       "      <td>metatdenovo.TRANSDECODER_PREDICT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29763</th>\n",
       "      <td>metatdenovo</td>\n",
       "      <td>BBMAP_ALIGN</td>\n",
       "      <td>bbmap.sh</td>\n",
       "      <td>./git_repos/metatdenovo/modules/nf-core/bbmap/align/main.nf</td>\n",
       "      <td>42</td>\n",
       "      <td>bbmap.sh \\\\</td>\n",
       "      <td>4</td>\n",
       "      <td>metatdenovo.BBMAP_ALIGN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29764</th>\n",
       "      <td>metatdenovo</td>\n",
       "      <td>BBMAP_INDEX</td>\n",
       "      <td>bbmap.sh</td>\n",
       "      <td>./git_repos/metatdenovo/modules/nf-core/bbmap/index/main.nf</td>\n",
       "      <td>23</td>\n",
       "      <td>bbmap.sh \\\\</td>\n",
       "      <td>4</td>\n",
       "      <td>metatdenovo.BBMAP_INDEX</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29765</th>\n",
       "      <td>hic</td>\n",
       "      <td>GET_RESTRICTION_FRAGMENTS</td>\n",
       "      <td>fragments.bed</td>\n",
       "      <td>./git_repos/hic/modules/local/hicpro/get_restriction_fragments.nf</td>\n",
       "      <td>20</td>\n",
       "      <td>digest_genome.py -r ${res_site} -o restriction_fragments.bed ${fasta}</td>\n",
       "      <td>0</td>\n",
       "      <td>hic.GET_RESTRICTION_FRAGMENTS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29766</th>\n",
       "      <td>hic</td>\n",
       "      <td>HICPRO2PAIRS</td>\n",
       "      <td>contacts</td>\n",
       "      <td>./git_repos/hic/modules/local/hicpro/hicpro2pairs.nf</td>\n",
       "      <td>22</td>\n",
       "      <td>awk '{OFS=\"\\t\";print \\$1,\\$2,\\$3,\\$5,\\$6,\\$4,\\$7}' $vpairs | bgzip -c &gt; ${prefix}_contacts.pairs.gz</td>\n",
       "      <td>6</td>\n",
       "      <td>hic.HICPRO2PAIRS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29767</th>\n",
       "      <td>hic</td>\n",
       "      <td>HICPRO2PAIRS</td>\n",
       "      <td>contacts</td>\n",
       "      <td>./git_repos/hic/modules/local/hicpro/hicpro2pairs.nf</td>\n",
       "      <td>24</td>\n",
       "      <td>pairix -f ${prefix}_contacts.pairs.gz</td>\n",
       "      <td>6</td>\n",
       "      <td>hic.HICPRO2PAIRS</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>29768 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            pipeline                 process_name                 operator  \\\n",
       "0      diaproteomics        get_software_versions                     echo   \n",
       "1      diaproteomics        get_software_versions                     echo   \n",
       "2      diaproteomics  statistical_post_processing                     echo   \n",
       "3      readsimulator                BOWTIE2_ALIGN                     echo   \n",
       "4      readsimulator                BOWTIE2_ALIGN                     echo   \n",
       "5      readsimulator                BOWTIE2_ALIGN                     echo   \n",
       "6      readsimulator                BOWTIE2_ALIGN                     echo   \n",
       "7      readsimulator                BOWTIE2_ALIGN                     echo   \n",
       "8      readsimulator                BOWTIE2_BUILD                     echo   \n",
       "9      readsimulator                BOWTIE2_BUILD                     echo   \n",
       "10     readsimulator               SAMTOOLS_INDEX                     echo   \n",
       "11     readsimulator               SAMTOOLS_INDEX                     echo   \n",
       "12     readsimulator         INSILICOSEQ_GENERATE                     echo   \n",
       "13     readsimulator         INSILICOSEQ_GENERATE                     echo   \n",
       "14     readsimulator         INSILICOSEQ_GENERATE                     echo   \n",
       "15     readsimulator         INSILICOSEQ_GENERATE                     echo   \n",
       "16     readsimulator           MERGE_SAMPLESHEETS                     echo   \n",
       "17     readsimulator           MERGE_SAMPLESHEETS                     echo   \n",
       "18      callingcards                      GTF2BED                     echo   \n",
       "19      callingcards                  DEMULTIPLEX                     echo   \n",
       "20      callingcards                   COUNT_HOPS                     echo   \n",
       "21      callingcards                     CONCATQC                     echo   \n",
       "22      callingcards                     CONCATQC                     echo   \n",
       "23      callingcards                BOWTIE2_ALIGN                     echo   \n",
       "24      callingcards                BOWTIE2_ALIGN                     echo   \n",
       "...              ...                          ...                      ...   \n",
       "29743        circrna                  PARENT_GENE                    2.bed   \n",
       "29744        circrna                   ANNOTATION                    2.bed   \n",
       "29745        circrna                CIRIQUANT_YML                     $STR   \n",
       "29746        circrna                          DCC                     chr/   \n",
       "29747        circrna                          DCC                     chr/   \n",
       "29748         bacass                         CANU                     canu   \n",
       "29749         bacass                         CANU                     canu   \n",
       "29750         bacass                     NANOPLOT                 NanoStat   \n",
       "29751    demultiplex                   BCLCONVERT              bcl-convert   \n",
       "29752    demultiplex                   BCLCONVERT              bcl-convert   \n",
       "29753    metatdenovo              EGGNOG_DOWNLOAD  download_eggnog_data.py   \n",
       "29754    metatdenovo              EGGNOG_DOWNLOAD               emapper.py   \n",
       "29755    metatdenovo              EGGNOG_DOWNLOAD               emapper.py   \n",
       "29756    metatdenovo                EGGNOG_MAPPER               emapper.py   \n",
       "29757    metatdenovo                EGGNOG_MAPPER               emapper.py   \n",
       "29758    metatdenovo                EGGNOG_MAPPER               emapper.py   \n",
       "29759    metatdenovo         TRANSDECODER_LONGORF    TransDecoder.LongOrfs   \n",
       "29760    metatdenovo         TRANSDECODER_LONGORF    TransDecoder.LongOrfs   \n",
       "29761    metatdenovo         TRANSDECODER_PREDICT     TransDecoder.Predict   \n",
       "29762    metatdenovo         TRANSDECODER_PREDICT     TransDecoder.Predict   \n",
       "29763    metatdenovo                  BBMAP_ALIGN                 bbmap.sh   \n",
       "29764    metatdenovo                  BBMAP_INDEX                 bbmap.sh   \n",
       "29765            hic    GET_RESTRICTION_FRAGMENTS            fragments.bed   \n",
       "29766            hic                 HICPRO2PAIRS                 contacts   \n",
       "29767            hic                 HICPRO2PAIRS                 contacts   \n",
       "\n",
       "                                                                                file_path  \\\n",
       "0                                                       ./git_repos/diaproteomics/main.nf   \n",
       "1                                                       ./git_repos/diaproteomics/main.nf   \n",
       "2                                                       ./git_repos/diaproteomics/main.nf   \n",
       "3                         ./git_repos/readsimulator/modules/nf-core/bowtie2/align/main.nf   \n",
       "4                         ./git_repos/readsimulator/modules/nf-core/bowtie2/align/main.nf   \n",
       "5                         ./git_repos/readsimulator/modules/nf-core/bowtie2/align/main.nf   \n",
       "6                         ./git_repos/readsimulator/modules/nf-core/bowtie2/align/main.nf   \n",
       "7                         ./git_repos/readsimulator/modules/nf-core/bowtie2/align/main.nf   \n",
       "8                         ./git_repos/readsimulator/modules/nf-core/bowtie2/build/main.nf   \n",
       "9                         ./git_repos/readsimulator/modules/nf-core/bowtie2/build/main.nf   \n",
       "10                       ./git_repos/readsimulator/modules/nf-core/samtools/index/main.nf   \n",
       "11                       ./git_repos/readsimulator/modules/nf-core/samtools/index/main.nf   \n",
       "12                   ./git_repos/readsimulator/modules/local/insilicoseq/generate/main.nf   \n",
       "13                   ./git_repos/readsimulator/modules/local/insilicoseq/generate/main.nf   \n",
       "14                   ./git_repos/readsimulator/modules/local/insilicoseq/generate/main.nf   \n",
       "15                   ./git_repos/readsimulator/modules/local/insilicoseq/generate/main.nf   \n",
       "16              ./git_repos/readsimulator/modules/local/custom/merge_samplesheets/main.nf   \n",
       "17              ./git_repos/readsimulator/modules/local/custom/merge_samplesheets/main.nf   \n",
       "18                                 ./git_repos/callingcards/modules/local/gtf2bed/main.nf   \n",
       "19     ./git_repos/callingcards/modules/local/callingcardstools/yeast/demultiplex/main.nf   \n",
       "20      ./git_repos/callingcards/modules/local/callingcardstools/yeast/count_hops/main.nf   \n",
       "21        ./git_repos/callingcards/modules/local/callingcardstools/yeast/concatQC/main.nf   \n",
       "22      ./git_repos/callingcards/modules/local/callingcardstools/mammals/concatQC/main.nf   \n",
       "23                         ./git_repos/callingcards/modules/nf-core/bowtie2/align/main.nf   \n",
       "24                         ./git_repos/callingcards/modules/nf-core/bowtie2/align/main.nf   \n",
       "...                                                                                   ...   \n",
       "29743                    ./git_repos/circrna/modules/local/annotation/parent_gene/main.nf   \n",
       "29744                ./git_repos/circrna/modules/local/annotation/full_annotation/main.nf   \n",
       "29745                             ./git_repos/circrna/modules/local/ciriquant/yml/main.nf   \n",
       "29746                                   ./git_repos/circrna/modules/local/dcc/dcc/main.nf   \n",
       "29747                                   ./git_repos/circrna/modules/local/dcc/dcc/main.nf   \n",
       "29748                                     ./git_repos/bacass/modules/nf-core/canu/main.nf   \n",
       "29749                                     ./git_repos/bacass/modules/nf-core/canu/main.nf   \n",
       "29750                                 ./git_repos/bacass/modules/nf-core/nanoplot/main.nf   \n",
       "29751                          ./git_repos/demultiplex/modules/nf-core/bclconvert/main.nf   \n",
       "29752                          ./git_repos/demultiplex/modules/nf-core/bclconvert/main.nf   \n",
       "29753                            ./git_repos/metatdenovo/modules/local/eggnog/download.nf   \n",
       "29754                            ./git_repos/metatdenovo/modules/local/eggnog/download.nf   \n",
       "29755                            ./git_repos/metatdenovo/modules/local/eggnog/download.nf   \n",
       "29756                              ./git_repos/metatdenovo/modules/local/eggnog/mapper.nf   \n",
       "29757                              ./git_repos/metatdenovo/modules/local/eggnog/mapper.nf   \n",
       "29758                              ./git_repos/metatdenovo/modules/local/eggnog/mapper.nf   \n",
       "29759                ./git_repos/metatdenovo/modules/nf-core/transdecoder/longorf/main.nf   \n",
       "29760                ./git_repos/metatdenovo/modules/nf-core/transdecoder/longorf/main.nf   \n",
       "29761                ./git_repos/metatdenovo/modules/nf-core/transdecoder/predict/main.nf   \n",
       "29762                ./git_repos/metatdenovo/modules/nf-core/transdecoder/predict/main.nf   \n",
       "29763                         ./git_repos/metatdenovo/modules/nf-core/bbmap/align/main.nf   \n",
       "29764                         ./git_repos/metatdenovo/modules/nf-core/bbmap/index/main.nf   \n",
       "29765                   ./git_repos/hic/modules/local/hicpro/get_restriction_fragments.nf   \n",
       "29766                                ./git_repos/hic/modules/local/hicpro/hicpro2pairs.nf   \n",
       "29767                                ./git_repos/hic/modules/local/hicpro/hicpro2pairs.nf   \n",
       "\n",
       "       line_number  \\\n",
       "0              335   \n",
       "1              336   \n",
       "2              971   \n",
       "3               47   \n",
       "4               68   \n",
       "5               69   \n",
       "6               88   \n",
       "7               89   \n",
       "8               27   \n",
       "9               39   \n",
       "10              33   \n",
       "11              45   \n",
       "12              29   \n",
       "13              30   \n",
       "14              53   \n",
       "15              54   \n",
       "16              23   \n",
       "17              26   \n",
       "18              28   \n",
       "19              36   \n",
       "20              39   \n",
       "21              33   \n",
       "22              38   \n",
       "23              47   \n",
       "24              68   \n",
       "...            ...   \n",
       "29743           45   \n",
       "29744           34   \n",
       "29745           38   \n",
       "29746           27   \n",
       "29747           43   \n",
       "29748           35   \n",
       "29749           47   \n",
       "29750           33   \n",
       "29751           59   \n",
       "29752           69   \n",
       "29753           21   \n",
       "29754           28   \n",
       "29755           45   \n",
       "29756           36   \n",
       "29757           50   \n",
       "29758           62   \n",
       "29759           29   \n",
       "29760           37   \n",
       "29761           29   \n",
       "29762           37   \n",
       "29763           42   \n",
       "29764           23   \n",
       "29765           20   \n",
       "29766           22   \n",
       "29767           24   \n",
       "\n",
       "                                                                                                                                                                                                                                                                            line_content  \\\n",
       "0                                                                                                                                                                                                                                       echo $workflow.manifest.version > v_pipeline.txt   \n",
       "1                                                                                                                                                                                                                                       echo $workflow.nextflow.version > v_nextflow.txt   \n",
       "2                                                                                                                                                      msstats.R > msstats.log || echo \"Optional MSstats step failed. Please check logs and re-run or do a manual statistical analysis.\"   \n",
       "3                                                                                                                                                                                                                [ -z \"\\$INDEX\" ] && echo \"Bowtie2 index files not found\" 1>&2 && exit 1   \n",
       "4                                                                                                                                                                                        bowtie2: \\$(echo \\$(bowtie2 --version 2>&1) | sed 's/^.*bowtie2-align-s version //; s/ .*\\$//')   \n",
       "5                                                                                                                                                                                                 samtools: \\$(echo \\$(samtools --version 2>&1) | sed 's/^.*samtools //; s/Using.*\\$//')   \n",
       "6                                                                                                                                                                                        bowtie2: \\$(echo \\$(bowtie2 --version 2>&1) | sed 's/^.*bowtie2-align-s version //; s/ .*\\$//')   \n",
       "7                                                                                                                                                                                                 samtools: \\$(echo \\$(samtools --version 2>&1) | sed 's/^.*samtools //; s/Using.*\\$//')   \n",
       "8                                                                                                                                                                                        bowtie2: \\$(echo \\$(bowtie2 --version 2>&1) | sed 's/^.*bowtie2-align-s version //; s/ .*\\$//')   \n",
       "9                                                                                                                                                                                        bowtie2: \\$(echo \\$(bowtie2 --version 2>&1) | sed 's/^.*bowtie2-align-s version //; s/ .*\\$//')   \n",
       "10                                                                                                                                                                                                samtools: \\$(echo \\$(samtools --version 2>&1) | sed 's/^.*samtools //; s/Using.*\\$//')   \n",
       "11                                                                                                                                                                                                samtools: \\$(echo \\$(samtools --version 2>&1) | sed 's/^.*samtools //; s/Using.*\\$//')   \n",
       "12                                                                                                                                                                                                                                 seed=\\$(echo $seed | sed 's/\\\\[//g' | sed 's/\\\\]//g')   \n",
       "13                                                                                                                                                                                                                             prefix=\\$(echo $prefix | sed 's/\\\\[//g' | sed 's/\\\\]//g')   \n",
       "14                                                                                                                                                                                                                                 seed=\\$(echo $seed | sed 's/\\\\[//g' | sed 's/\\\\]//g')   \n",
       "15                                                                                                                                                                                                                             prefix=\\$(echo $prefix | sed 's/\\\\[//g' | sed 's/\\\\]//g')   \n",
       "16                                                                                                                                                                                                           echo \\\\\"sample\\\\\",\\\\\"fastq_1\\\\\",\\\\\"fastq_2\\\\\" > \"${prefix}_samplesheet.csv\"   \n",
       "17                                                                                                                                                                                                                                                   echo >> \"${prefix}_samplesheet.csv\"   \n",
       "18                                                                                                                                                                                                            perl: \\$(echo \\$(perl --version 2>&1) | sed 's/.*v\\\\(.*\\\\)) built.*/\\\\1/')   \n",
       "19                                                                                                                                                                                                      callingcardstools: \\$(echo \\$(callingcardstools --version 2>&1) | sed 's/.* //')   \n",
       "20                                                                                                                                                                                                      callingcardstools: \\$(echo \\$(callingcardstools --version 2>&1) | sed 's/.* //')   \n",
       "21                                                                                                                                                                                                      callingcardstools: \\$(echo \\$(callingcardstools --version 2>&1) | sed 's/.* //')   \n",
       "22                                                                                                                                                                                                      callingcardstools: \\$(echo \\$(callingcardstools --version 2>&1) | sed 's/.* //')   \n",
       "23                                                                                                                                                                                                               [ -z \"\\$INDEX\" ] && echo \"Bowtie2 index files not found\" 1>&2 && exit 1   \n",
       "24                                                                                                                                                                                       bowtie2: \\$(echo \\$(bowtie2 --version 2>&1) | sed 's/^.*bowtie2-align-s version //; s/ .*\\$//')   \n",
       "...                                                                                                                                                                                                                                                                                  ...   \n",
       "29743                                                                                                                                                                                                      awk -v OFS=\"\\\\t\" '{print \\$4, \\$14}' master_bed12.bed > circrna_host-gene.txt   \n",
       "29744                                                                                                                                                                                                                                              mv master_bed12.bed ${prefix}.bed.tmp   \n",
       "29745  printf \"name: ciriquant\\ntools:\\n  bwa: \\$BWA\\n  hisat2: \\$HISAT2\\n  stringtie: \\$STRINGTIE\\n  samtools: \\$SAMTOOLS\\n\\nreference:\\n  fasta: ${fasta_path}\\n  gtf: ${gtf_path}\\n  bwa_index: ${bwa_path}/\\$BWA_FILE\\n  hisat_index: ${hisat2_path}/${hisat2_prefix}\" >> travis.yml   \n",
       "29746                                                                                                                                                                                                                                                            sed -i 's/^chr//g' $gtf   \n",
       "29747                                                                                                                                                                                                                                                            sed -i 's/^chr//g' $gtf   \n",
       "29748                                                                                                                                                                                                                                                                            canu \\\\   \n",
       "29749                                                                                                                                                                                                        canu: \\$(echo \\$(canu --version 2>&1) | sed 's/^.*canu //; s/Using.*\\$//' )   \n",
       "29750                                                                                                                                                                                                                                                     mv NanoStats.txt ${prefix}.txt   \n",
       "29751                                                                                                                                                                                                                                                                     bcl-convert \\\\   \n",
       "29752                                                                                                                                                                                                            bclconvert: \\$(bcl-convert -V 2>&1 | head -n 1 | sed 's/^.*Version //')   \n",
       "29753                                                                                                                                                                                                                                                         download_eggnog_data.py \\\\   \n",
       "29754                                                                                                                                                                                 eggnog: \\$( echo \\$(emapper.py --version 2>&1)| sed 's/.* emapper-//' | sed 's/ \\\\/ Expected.*//')   \n",
       "29755                                                                                                                                                                                                            eggnog: \\$( echo \\$(emapper.py --version 2>&1)| sed 's/.* emapper-//' )   \n",
       "29756                                                                                                                                                                                                                                                                      emapper.py \\\\   \n",
       "29757                                                                                                                                                                                 eggnog: \\$( echo \\$(emapper.py --version 2>&1)| sed 's/.* emapper-//' | sed 's/ \\\\/ Expected.*//')   \n",
       "29758                                                                                                                                                                                 eggnog: \\$( echo \\$(emapper.py --version 2>&1)| sed 's/.* emapper-//' | sed 's/ \\\\/ Expected.*//')   \n",
       "29759                                                                                                                                                                                                                                                           TransDecoder.LongOrfs \\\\   \n",
       "29760                                                                                                                                                                                  transdecoder: \\$(echo \\$(TransDecoder.LongOrfs --version) | sed -e \"s/TransDecoder.LongOrfs //g\")   \n",
       "29761                                                                                                                                                                                                                                                            TransDecoder.Predict \\\\   \n",
       "29762                                                                                                                                                                                    transdecoder: \\$(echo \\$(TransDecoder.Predict --version) | sed -e \"s/TransDecoder.Predict //g\")   \n",
       "29763                                                                                                                                                                                                                                                                        bbmap.sh \\\\   \n",
       "29764                                                                                                                                                                                                                                                                        bbmap.sh \\\\   \n",
       "29765                                                                                                                                                                                                              digest_genome.py -r ${res_site} -o restriction_fragments.bed ${fasta}   \n",
       "29766                                                                                                                                                                                awk '{OFS=\"\\t\";print \\$1,\\$2,\\$3,\\$5,\\$6,\\$4,\\$7}' $vpairs | bgzip -c > ${prefix}_contacts.pairs.gz   \n",
       "29767                                                                                                                                                                                                                                              pairix -f ${prefix}_contacts.pairs.gz   \n",
       "\n",
       "       Assigned Group                      pipeline_process_name  \n",
       "0                   2        diaproteomics.get_software_versions  \n",
       "1                   2        diaproteomics.get_software_versions  \n",
       "2                   2  diaproteomics.statistical_post_processing  \n",
       "3                   2                readsimulator.BOWTIE2_ALIGN  \n",
       "4                   2                readsimulator.BOWTIE2_ALIGN  \n",
       "5                   2                readsimulator.BOWTIE2_ALIGN  \n",
       "6                   2                readsimulator.BOWTIE2_ALIGN  \n",
       "7                   2                readsimulator.BOWTIE2_ALIGN  \n",
       "8                   2                readsimulator.BOWTIE2_BUILD  \n",
       "9                   2                readsimulator.BOWTIE2_BUILD  \n",
       "10                  2               readsimulator.SAMTOOLS_INDEX  \n",
       "11                  2               readsimulator.SAMTOOLS_INDEX  \n",
       "12                  2         readsimulator.INSILICOSEQ_GENERATE  \n",
       "13                  2         readsimulator.INSILICOSEQ_GENERATE  \n",
       "14                  2         readsimulator.INSILICOSEQ_GENERATE  \n",
       "15                  2         readsimulator.INSILICOSEQ_GENERATE  \n",
       "16                  2           readsimulator.MERGE_SAMPLESHEETS  \n",
       "17                  2           readsimulator.MERGE_SAMPLESHEETS  \n",
       "18                  2                       callingcards.GTF2BED  \n",
       "19                  2                   callingcards.DEMULTIPLEX  \n",
       "20                  2                    callingcards.COUNT_HOPS  \n",
       "21                  2                      callingcards.CONCATQC  \n",
       "22                  2                      callingcards.CONCATQC  \n",
       "23                  2                 callingcards.BOWTIE2_ALIGN  \n",
       "24                  2                 callingcards.BOWTIE2_ALIGN  \n",
       "...               ...                                        ...  \n",
       "29743               0                        circrna.PARENT_GENE  \n",
       "29744               0                         circrna.ANNOTATION  \n",
       "29745               6                      circrna.CIRIQUANT_YML  \n",
       "29746               4                                circrna.DCC  \n",
       "29747               4                                circrna.DCC  \n",
       "29748               6                                bacass.CANU  \n",
       "29749               6                                bacass.CANU  \n",
       "29750               6                            bacass.NANOPLOT  \n",
       "29751               4                     demultiplex.BCLCONVERT  \n",
       "29752               4                     demultiplex.BCLCONVERT  \n",
       "29753               4                metatdenovo.EGGNOG_DOWNLOAD  \n",
       "29754               4                metatdenovo.EGGNOG_DOWNLOAD  \n",
       "29755               4                metatdenovo.EGGNOG_DOWNLOAD  \n",
       "29756               4                  metatdenovo.EGGNOG_MAPPER  \n",
       "29757               4                  metatdenovo.EGGNOG_MAPPER  \n",
       "29758               4                  metatdenovo.EGGNOG_MAPPER  \n",
       "29759               4           metatdenovo.TRANSDECODER_LONGORF  \n",
       "29760               4           metatdenovo.TRANSDECODER_LONGORF  \n",
       "29761               4           metatdenovo.TRANSDECODER_PREDICT  \n",
       "29762               4           metatdenovo.TRANSDECODER_PREDICT  \n",
       "29763               4                    metatdenovo.BBMAP_ALIGN  \n",
       "29764               4                    metatdenovo.BBMAP_INDEX  \n",
       "29765               0              hic.GET_RESTRICTION_FRAGMENTS  \n",
       "29766               6                           hic.HICPRO2PAIRS  \n",
       "29767               6                           hic.HICPRO2PAIRS  \n",
       "\n",
       "[29768 rows x 8 columns]"
      ]
     },
     "execution_count": 222,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 223,
   "id": "1acfa391",
   "metadata": {},
   "outputs": [],
   "source": [
    "#sum_processes = merged_df['process_name'].nunique()\n",
    "#sum_processes = merged_df['file_path'].nunique()\n",
    "\n",
    "#6154 if-conditions\n",
    "\n",
    "\n",
    "grouped_df['Ratio to total processes'] = grouped_df['pipeline_process_name']/sum_processes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 224,
   "id": "863b944e",
   "metadata": {},
   "outputs": [],
   "source": [
    "grouped_df = grouped_df.rename(columns={'file_path': 'Files', 'operator' : 'Total operator occurrences'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 225,
   "id": "998d0a35",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Assigned Group</th>\n",
       "      <th>pipeline</th>\n",
       "      <th>pipeline_process_name</th>\n",
       "      <th>Total operator occurrences</th>\n",
       "      <th>Ratio to total processes</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>86</td>\n",
       "      <td>1308</td>\n",
       "      <td>4767</td>\n",
       "      <td>0.424537</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>73</td>\n",
       "      <td>281</td>\n",
       "      <td>543</td>\n",
       "      <td>0.091204</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>92</td>\n",
       "      <td>1948</td>\n",
       "      <td>5341</td>\n",
       "      <td>0.632262</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>20</td>\n",
       "      <td>27</td>\n",
       "      <td>37</td>\n",
       "      <td>0.008763</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>93</td>\n",
       "      <td>1620</td>\n",
       "      <td>4596</td>\n",
       "      <td>0.525803</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>5</td>\n",
       "      <td>68</td>\n",
       "      <td>265</td>\n",
       "      <td>776</td>\n",
       "      <td>0.086011</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>6</td>\n",
       "      <td>94</td>\n",
       "      <td>2396</td>\n",
       "      <td>13708</td>\n",
       "      <td>0.777670</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Assigned Group  pipeline  pipeline_process_name  \\\n",
       "0               0        86                   1308   \n",
       "1               1        73                    281   \n",
       "2               2        92                   1948   \n",
       "3               3        20                     27   \n",
       "4               4        93                   1620   \n",
       "5               5        68                    265   \n",
       "6               6        94                   2396   \n",
       "\n",
       "   Total operator occurrences  Ratio to total processes  \n",
       "0                        4767                  0.424537  \n",
       "1                         543                  0.091204  \n",
       "2                        5341                  0.632262  \n",
       "3                          37                  0.008763  \n",
       "4                        4596                  0.525803  \n",
       "5                         776                  0.086011  \n",
       "6                       13708                  0.777670  "
      ]
     },
     "execution_count": 225,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "grouped_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6fa8fe10",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import re\n",
    "import csv\n",
    "\n",
    "def find_operators_in_script(directory, output_file_name):\n",
    "    result_list = []\n",
    "\n",
    "    for pipeline in os.listdir(directory):\n",
    "        subdirectory_path = os.path.join(directory, pipeline)\n",
    "        folder_name = os.path.basename(subdirectory_path)\n",
    "        print(pipeline)\n",
    "\n",
    "        for root, dirs, files in os.walk(subdirectory_path):\n",
    "            for filename in files:\n",
    "                file_path = os.path.join(root, filename)\n",
    "                \n",
    "                if filename.endswith('.nf') or filename.endswith('.config'):\n",
    "                    with open(file_path, 'r', encoding='utf-8', errors='ignore') as file:\n",
    "                        lines = file.readlines()\n",
    "                    \n",
    "                    inside_block = False\n",
    "\n",
    "                    for line_num, line in enumerate(lines):\n",
    "                        stripped_line = line.strip()\n",
    "\n",
    "                        if '\"\"\"' in stripped_line:\n",
    "                            inside_block = not inside_block\n",
    "\n",
    "                        if inside_block:\n",
    "                            match = re.match(r'^(\\w+):\\s', stripped_line)\n",
    "                            if match:\n",
    "                                keyword = match.group(1)\n",
    "                                result_list.append([pipeline, keyword, file_path, line_num+1, line.strip()])\n",
    "\n",
    "    if result_list:\n",
    "        write_to_csv(result_list, output_file_name)\n",
    "    else:\n",
    "        print(\"No results found.\")\n",
    "\n",
    "def write_to_csv(result_list, output_file_name):\n",
    "    with open(f'./results/{output_file_name}.csv', 'w', newline='', encoding='utf-8') as csvfile:\n",
    "        writer = csv.writer(csvfile)\n",
    "        writer.writerow(['pipeline', 'keyword', 'file_path', 'line_number', 'line_content'])\n",
    "        writer.writerows(result_list)\n",
    "\n",
    "directory_path = './git_repos'\n",
    "output_file_name = 'keyword_occurrences'\n",
    "\n",
    "find_operators_in_script(directory_path, output_file_name)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "id": "4a3041f4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pipeline</th>\n",
       "      <th>keyword</th>\n",
       "      <th>file_path</th>\n",
       "      <th>line_number</th>\n",
       "      <th>line_content</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>airrflow</td>\n",
       "      <td>fastqc</td>\n",
       "      <td>./git_repos\\airrflow\\modules\\local\\fastqc_post...</td>\n",
       "      <td>28</td>\n",
       "      <td>fastqc: \\$( fastqc --version | sed -e \"s/FastQ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>airrflow</td>\n",
       "      <td>igblastn</td>\n",
       "      <td>./git_repos\\airrflow\\modules\\local\\fetch_datab...</td>\n",
       "      <td>29</td>\n",
       "      <td>igblastn: \\$( igblastn -version | grep -o \"igb...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>airrflow</td>\n",
       "      <td>changeo</td>\n",
       "      <td>./git_repos\\airrflow\\modules\\local\\fetch_datab...</td>\n",
       "      <td>30</td>\n",
       "      <td>changeo: \\$( AssignGenes.py --version | awk -F...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>airrflow</td>\n",
       "      <td>gunzip</td>\n",
       "      <td>./git_repos\\airrflow\\modules\\local\\gunzip.nf</td>\n",
       "      <td>24</td>\n",
       "      <td>gunzip: \\$(echo \\$(gunzip --version 2&gt;&amp;1) | se...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>airrflow</td>\n",
       "      <td>python</td>\n",
       "      <td>./git_repos\\airrflow\\modules\\local\\merge_UMI.nf</td>\n",
       "      <td>26</td>\n",
       "      <td>python: \\$( echo \\$(python --version | grep -o...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3793</th>\n",
       "      <td>viralrecon</td>\n",
       "      <td>tabix</td>\n",
       "      <td>./git_repos\\viralrecon\\modules\\nf-core\\tabix\\t...</td>\n",
       "      <td>39</td>\n",
       "      <td>tabix: \\$(echo \\$(tabix -h 2&gt;&amp;1) | sed 's/^.*V...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3794</th>\n",
       "      <td>viralrecon</td>\n",
       "      <td>unicycler</td>\n",
       "      <td>./git_repos\\viralrecon\\modules\\nf-core\\unicycl...</td>\n",
       "      <td>43</td>\n",
       "      <td>unicycler: \\$(echo \\$(unicycler --version 2&gt;&amp;1...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3795</th>\n",
       "      <td>viralrecon</td>\n",
       "      <td>untar</td>\n",
       "      <td>./git_repos\\viralrecon\\modules\\nf-core\\untar\\m...</td>\n",
       "      <td>48</td>\n",
       "      <td>untar: \\$(echo \\$(tar --version 2&gt;&amp;1) | sed 's...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3796</th>\n",
       "      <td>viralrecon</td>\n",
       "      <td>untar</td>\n",
       "      <td>./git_repos\\viralrecon\\modules\\nf-core\\untar\\m...</td>\n",
       "      <td>60</td>\n",
       "      <td>untar: \\$(echo \\$(tar --version 2&gt;&amp;1) | sed 's...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3797</th>\n",
       "      <td>viralrecon</td>\n",
       "      <td>vcflib</td>\n",
       "      <td>./git_repos\\viralrecon\\modules\\nf-core\\vcflib\\...</td>\n",
       "      <td>32</td>\n",
       "      <td>vcflib: $VERSION</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3798 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        pipeline    keyword  \\\n",
       "0       airrflow     fastqc   \n",
       "1       airrflow   igblastn   \n",
       "2       airrflow    changeo   \n",
       "3       airrflow     gunzip   \n",
       "4       airrflow     python   \n",
       "...          ...        ...   \n",
       "3793  viralrecon      tabix   \n",
       "3794  viralrecon  unicycler   \n",
       "3795  viralrecon      untar   \n",
       "3796  viralrecon      untar   \n",
       "3797  viralrecon     vcflib   \n",
       "\n",
       "                                              file_path  line_number  \\\n",
       "0     ./git_repos\\airrflow\\modules\\local\\fastqc_post...           28   \n",
       "1     ./git_repos\\airrflow\\modules\\local\\fetch_datab...           29   \n",
       "2     ./git_repos\\airrflow\\modules\\local\\fetch_datab...           30   \n",
       "3          ./git_repos\\airrflow\\modules\\local\\gunzip.nf           24   \n",
       "4       ./git_repos\\airrflow\\modules\\local\\merge_UMI.nf           26   \n",
       "...                                                 ...          ...   \n",
       "3793  ./git_repos\\viralrecon\\modules\\nf-core\\tabix\\t...           39   \n",
       "3794  ./git_repos\\viralrecon\\modules\\nf-core\\unicycl...           43   \n",
       "3795  ./git_repos\\viralrecon\\modules\\nf-core\\untar\\m...           48   \n",
       "3796  ./git_repos\\viralrecon\\modules\\nf-core\\untar\\m...           60   \n",
       "3797  ./git_repos\\viralrecon\\modules\\nf-core\\vcflib\\...           32   \n",
       "\n",
       "                                           line_content  \n",
       "0     fastqc: \\$( fastqc --version | sed -e \"s/FastQ...  \n",
       "1     igblastn: \\$( igblastn -version | grep -o \"igb...  \n",
       "2     changeo: \\$( AssignGenes.py --version | awk -F...  \n",
       "3     gunzip: \\$(echo \\$(gunzip --version 2>&1) | se...  \n",
       "4     python: \\$( echo \\$(python --version | grep -o...  \n",
       "...                                                 ...  \n",
       "3793  tabix: \\$(echo \\$(tabix -h 2>&1) | sed 's/^.*V...  \n",
       "3794  unicycler: \\$(echo \\$(unicycler --version 2>&1...  \n",
       "3795  untar: \\$(echo \\$(tar --version 2>&1) | sed 's...  \n",
       "3796  untar: \\$(echo \\$(tar --version 2>&1) | sed 's...  \n",
       "3797                                   vcflib: $VERSION  \n",
       "\n",
       "[3798 rows x 5 columns]"
      ]
     },
     "execution_count": 68,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "df = pd.read_csv('./results/keyword_occurrences.csv')\n",
    "\n",
    "#df.head(60)\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "id": "28fa6d2f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "491"
      ]
     },
     "execution_count": 69,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(df)\n",
    "keyword_counts = df['keyword'].value_counts()\n",
    "#pd.set_option(\"display.max_rows\", 999)\n",
    "\n",
    "len(keyword_counts)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "id": "d717c3bc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pipeline</th>\n",
       "      <th>file_path</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>keyword</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>multiqc</th>\n",
       "      <td>66</td>\n",
       "      <td>130</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>python</th>\n",
       "      <td>56</td>\n",
       "      <td>182</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>fastqc</th>\n",
       "      <td>52</td>\n",
       "      <td>109</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>samtools</th>\n",
       "      <td>45</td>\n",
       "      <td>515</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pigz</th>\n",
       "      <td>29</td>\n",
       "      <td>65</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ITSx</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IsobaricAnalyzer</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ganon</th>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gaas</th>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>yara</th>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>491 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  pipeline  file_path\n",
       "keyword                              \n",
       "multiqc                 66        130\n",
       "python                  56        182\n",
       "fastqc                  52        109\n",
       "samtools                45        515\n",
       "pigz                    29         65\n",
       "...                    ...        ...\n",
       "ITSx                     1          1\n",
       "IsobaricAnalyzer         1          1\n",
       "ganon                    1          6\n",
       "gaas                     1          3\n",
       "yara                     1          3\n",
       "\n",
       "[491 rows x 2 columns]"
      ]
     },
     "execution_count": 70,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "result = df.groupby('keyword').agg({\n",
    "    'pipeline': 'nunique',\n",
    "    'file_path': 'count'  \n",
    "}).sort_values(by='pipeline', ascending=False)\n",
    "result"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
