{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Sturctural analysis\n",
    "\n",
    "Searching for config-files in the git-repos and their subfolders"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import csv\n",
    "import re\n",
    "import pandas as pd\n",
    "import matplotlib.pyplot as plt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_config_details(directory):\n",
    "    config_files_info = []\n",
    "\n",
    "    for pipeline in os.listdir(directory):\n",
    "        subdirectory_path = os.path.join(directory, pipeline)\n",
    "        folder_name = os.path.basename(subdirectory_path)\n",
    "\n",
    "        for root, dirs, files in os.walk(subdirectory_path):\n",
    "            for file in files:\n",
    "                if file.endswith('.config'):\n",
    "                    config_file_name = file\n",
    "                    config_file_path = os.path.join(root, file)\n",
    "                    with open(config_file_path, 'r') as f:\n",
    "                        line_count = sum(1 for line in f)\n",
    "                    config_files_info.append((folder_name, config_file_name, line_count))\n",
    "\n",
    "    return config_files_info"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "def write_config_info_to_csv(config_info, csv_file_path):\n",
    "    with open(csv_file_path, 'w', newline='') as csvfile:\n",
    "        fieldnames = ['pipeline', 'config_file_name', 'line_count']\n",
    "        writer = csv.DictWriter(csvfile, fieldnames=fieldnames)\n",
    "        writer.writeheader()\n",
    "\n",
    "        for folder_name, config_file_name, line_count in config_info:\n",
    "            writer.writerow({'pipeline': folder_name, 'config_file_name': config_file_name, 'line_count': line_count})\n",
    "\n",
    "    print(f\"Results written to: {csv_file_path}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Results written to: ./results/config_file_details.csv\n"
     ]
    }
   ],
   "source": [
    "directory_path = './git_repos/'\n",
    "csv_file_path = './results/config_file_details.csv'\n",
    "\n",
    "config_details = get_config_details(directory_path)\n",
    "write_config_info_to_csv(config_details, csv_file_path)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>line_count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>count</th>\n",
       "      <td>1029.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mean</th>\n",
       "      <td>104.898931</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>std</th>\n",
       "      <td>159.885269</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>min</th>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25%</th>\n",
       "      <td>23.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50%</th>\n",
       "      <td>37.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75%</th>\n",
       "      <td>84.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>max</th>\n",
       "      <td>1228.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        line_count\n",
       "count  1029.000000\n",
       "mean    104.898931\n",
       "std     159.885269\n",
       "min       0.000000\n",
       "25%      23.000000\n",
       "50%      37.000000\n",
       "75%      84.000000\n",
       "max    1228.000000"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.read_csv('./results/config_file_details.csv')\n",
    "df.describe()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Sorted DataFrame based on line_count (descending order):\n",
      "            pipeline         config_file_name  line_count\n",
      "335         fetchngs          nextflow.config           0\n",
      "500      metatdenovo          igenomes.config           0\n",
      "618   phageannotator          nextflow.config           0\n",
      "653   phageannotator          nextflow.config           2\n",
      "658   phageannotator          nextflow.config           2\n",
      "...              ...                      ...         ...\n",
      "36           atacseq           modules.config         976\n",
      "183        cutandrun           modules.config        1047\n",
      "1021      viralrecon  modules_illumina.config        1053\n",
      "823           rnaseq           modules.config        1167\n",
      "398            hicar           modules.config        1228\n",
      "\n",
      "[1029 rows x 3 columns]\n"
     ]
    }
   ],
   "source": [
    "sorted_df = df.sort_values(by='line_count', ascending=True)\n",
    "\n",
    "print(\"Sorted DataFrame based on line_count (descending order):\")\n",
    "print(sorted_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of pipelines with config:\n",
      "94\n",
      "Number of folder_name with nextflow.config:\n",
      "pipeline\n",
      "phageannotator           52\n",
      "fetchngs                 21\n",
      "callingcards              8\n",
      "multiplesequencealign     5\n",
      "cutandrun                 4\n",
      "                         ..\n",
      "gwas                      1\n",
      "genomeskim                1\n",
      "genomeassembler           1\n",
      "genomeannotator           1\n",
      "viralrecon                1\n",
      "Name: count, Length: 94, dtype: int64\n"
     ]
    }
   ],
   "source": [
    "pipeline_nextflow = df[df['config_file_name'] == 'nextflow.config']['pipeline'].value_counts()\n",
    "pipeline_configs = df.groupby('pipeline')['config_file_name'].count().sort_values(ascending=False)\n",
    "\n",
    "print(\"Number of pipelines with config:\")\n",
    "print(len(pipeline_configs))\n",
    "print(\"Number of folder_name with nextflow.config:\")\n",
    "print(pipeline_nextflow)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "- Total: 1029 config-files\n",
    "- Every pipeline has at least 1 nextflow.config\n",
    "- 70 config-files in sarek pipeline\n",
    "- Config details: average of 104 lines, a median of 37 lines, and a 75th percentile of 84 lines. We can see that some configuration files contain a huge amount of lines, the biggest configuration file includes a total of 1228 lines."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>config_file_name</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pipeline</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>airrflow</th>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ampliseq</th>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>atacseq</th>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>bacass</th>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>bactmap</th>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>taxprofiler</th>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>variantcatalogue</th>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>vipr</th>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>viralintegration</th>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>viralrecon</th>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>94 rows Ã— 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  config_file_name\n",
       "pipeline                          \n",
       "airrflow                        14\n",
       "ampliseq                        19\n",
       "atacseq                          7\n",
       "bacass                           9\n",
       "bactmap                          5\n",
       "...                            ...\n",
       "taxprofiler                     18\n",
       "variantcatalogue                 6\n",
       "vipr                             4\n",
       "viralintegration                 7\n",
       "viralrecon                      11\n",
       "\n",
       "[94 rows x 1 columns]"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pipeline_counts = df.groupby('pipeline').agg({'config_file_name' : 'count'})\n",
    "pipeline_counts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAmAAAAG4CAYAAADi7lsPAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjcuMiwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy8pXeV/AAAACXBIWXMAAA9hAAAPYQGoP6dpAAB30klEQVR4nO3dd1gUx/8H8PdSpFexgViiINhRsCNYEyuiscXYY4mxxpKoSTSJGluiiUlU7CXRqLHEEjV2sURUjA2NBRtoAFERURCY3x/8br93cHcc7AEC79fz3MNyOzs7eze397nZ2RlJCCFARERERPnGpKALQERERFTcMAAjIiIiymcMwIiIiIjyGQMwIiIionzGAIyIiIgonzEAIyIiIspnDMCIiIiI8hkDMCIiIqJ8xgCMiIiIKJ8xACPF4uPj8dVXX6Fhw4ZwcnKCqakpJEmCJElYvXo1AODOnTvyc5UqVSrQ8hZ306dPl9+L6dOnF3Rx8t2DBw8wadIk+Pj4wMHBASYmJvLrceTIEQDAkSNH5OcCAwN15qVKI0lS/hS+ABj6WlD+q1Spkvze3Llzx2j5btmyBZ07d4abmxssLCy0vv+BgYFZPjeUM2YFXQAq3O7cuYPmzZvj/v37BV0UomydPXsWbdu2xZMnTwq6KERvpH79+mHdunUFXYxigQEYKTJs2DA5+LKyskLr1q3h5uYGU1NTAIC3t3dBFq/Imz59Or788ksAwLRp04pli5ahhBDo27evHHw5OjqiZcuWKFOmDExMMi4GuLm5FWQRiQrUhg0bNIKvBg0aoHr16rCxsQEAeHh4FFTRiiQGYJRrjx49wv79+wEAFhYW+Oeff/gBpTfW33//jWvXrgEASpUqhatXr8LFxaWAS0X05lizZo28/OWXX+KLL74owNIUfQzAKNfOnz8vL/v7++sNvipVqgQhRH4Ui7Ixffr0YtlSpl5fg4KC9AZfgYGBrK//j69F8aH+GRk8eHABlqR4YCd8yjX1fjTlypUrwJIQZY/1lUg/fkbyFwMwyrXXr1/Ly6o+NERvKtZXIv1SU1PlZX5G8oGgPPHs2TPxww8/iI4dO4qKFSsKGxsbUaJECVGuXDnRsmVLMX36dHH58uVs84mNjRXffPONaN68uShbtqwoUaKEKFmypKhbt66YMGGCuHLlSrZ5REZGCgACgKhYsaL8fFhYmBg8eLDw8PAQVlZWwtHRUfj5+YmZM2eKxMRErXkdPnxYziu7x7Rp07Itgy4pKSli8eLFIiAgQJQqVUpYWlqKypUrix49eoj9+/fL6SpWrCjnGxkZqTUv9TIZIiAgQE5/+PBhg9NER0eLmTNnCj8/P1GmTBlhYmIiHBwcsmx7584d8fPPP4tevXqJGjVqCHt7e2FmZiacnZ1FrVq1xPDhw8WpU6cMLmN2j/79+2tsO23aNK3vkS4pKSli5cqVIigoSFSoUEFYWloKOzs74enpKQYNGqTxfuij7b26f/+++Oyzz0Tt2rWFg4ODsLa2FtWqVRMjR44Ud+7cMShffVatWmXw67Rq1Sp5O/V6HhAQoDP/nNatxMRE8fPPP4uOHTuKChUqCCsrK2FrayuqVq0qBg4cKA4ePGjwsR08eFAMGjRI1KxZUzg4OAgzMzNRsmRJ4eXlJQICAsSUKVPE0aNHRUpKisF5amPIa2HMc0xOVatWTd73zZs3daZ77733NN4vfWn79esnp1u/fr3e/e/du1cMHDhQeHh4CDs7O2FpaSkqVKggunTpIlauXGnQ69+/f/8s9fDJkydi4cKFwt/fX7i6ugpTU1MBQDx58kTezpDzn8qMGTPktGZmZmLt2rVZ8tD3yHzeNuQ8qU7pd9mwYcPk/S1fvlxnupCQEI1y60u7cuVKOd0HH3ygs9zz5s0TrVq1EuXKlRMWFhbCyspKVKhQQdSrV0/06tVLrFy5Ujx48CDb1yAzBmB5YPHixcLJycmgSv3nn3/qzGfFihXCwcFB7/ampqZi7NixIjU1VWc+mU+O6enp4osvvhAmJiY6861cubK4detWlrzyIwC7c+eOqF27tt68hwwZIlJSUt6YAGz79u1a3/PMAdiECROEJEkGvX69e/cWL168yHb/2T2UBGCnT58WVapUyXYfbdq0ETExMXrzyvxebdu2TW/9trKyErt27dKbZ3bepABs06ZNomzZstmWo2PHjuLp06c683n+/Lno3Lmzwce1bNmynLxkWeQmAFNyjskp9S9mfcfq5uZm8OuiXld1fbH+999/olWrVtm+/h4eHiIsLEzvMWQOwEJDQ4W7u7vW/HIagKWnp4vRo0fr/FzlRwBmjO+yDRs2yGn79Omjc1+ZA219abMLtHWd17U93Nzc9L4G2rATvpGNHj0aixYtkv83NTWFn58fPDw8YGlpidjYWFy4cEEeNO/Vq1da85k/fz4mTpwo/29hYYGAgABUqFABT548weHDhxEfH4+0tDQsXLgQd+/exe+//27QgJBffvklvvrqKwBA3bp1UatWLZibm+PChQtyJ8zIyEh06dIF586dg7m5ubytm5sbPvroIwDAtWvXcPDgQQCAl5cXWrVqpbGfBg0aZFuWzGJjY9GyZUvcvn1bfq5atWrw8/ODmZkZLl26hHPnzmHZsmWws7PLcf554eTJk5g+fTpev36NkiVLonnz5nBxcUFMTAzCw8M10t6/fx9CCEiShGrVqqFatWooWbIkzM3N8fjxY4SHh+PWrVsAMm4Jf/bsGXbt2pXlfQ0ODkbNmjVx5swZhIWFAQD8/Py0vuaNGjXK1XEdO3YM7dq1Q1JSkvxcgwYNUKNGDaSkpOD06dNyWf/66y80a9YMoaGhKFWqVLZ5Hzx4EMOGDUNaWhoqVKiAxo0bw97eHpGRkThy5AhSU1Px8uVL9OjRA5cvX0blypVzdQze3t5yfc3utcrLIVMWLFiA8ePHy53Z7ezs0LhxY7i7uyMtLQ1Xr15FWFgYhBDYtWsXAgICcPLkSVhbW2fJq2/fvvjjjz/k/6tWrQofHx84Ozvj9evXiI2NxaVLl4w6MGdOKTnH5FRgYCCWLl0KADh8+DA++OCDLGlu3LiBqKgojed0pY2MjMTdu3cBZLy22oYm+e+//9C0aVO5/gNAlSpV0LBhQ1hYWODq1av4+++/5X23aNECe/fuRdOmTbM9nps3b2Ls2LF49uwZ7Ozs0Lx5c7i6uuLJkyc4duxYtture/36NQYMGIBff/0VQMbQKzt37kSzZs3kNP3798fjx48BAD/99JP8vOpzo1KyZMkc7VvFWN9l6oPAHj58WOf+jh49qvG/vrTqA8hmHmT47NmzePfdd+XLslZWVmjUqBEqVaoECwsLJCQk4NatW7h06ZLGOTJHchyykU6LFy/WiIh79Ogh7t+/rzXtpUuXxOjRo8W+ffuyrDt58qTc3AxAvPPOO+Lhw4caaV69eiUmTpyosb9vv/1W677Uf52WKFFCSJIkqlSpIv7+++8saTdt2iTMzc3l9GvWrNF5vOqtC5lbWfSVQV8LWO/eveV01tbWYvPmzVnSHD9+XLi6ugpJkkSJEiWy/QWo/hoZIqctYGZmZkKSJPH1119nudzw6tUrjf/nzp0rVq1aJWJjY3Xu/9ixY6Jq1apy/uvWrdOZNqeXEw3dJj4+XqPFoEqVKlp/xa9fv15YWVnJ6Tp16qRzv+q/tC0sLISNjY1Yt26dSE9P10h3+fJljX0PHDjQoOPKTk5eK2O2gB04cEBuCTI3NxczZszQevktPDxcVK9eXc7vww8/1JpGtd7W1lbs2bNH535v3bolZsyYIf744w+9x5qdnLaAGfMcY4iHDx/KeZUrV05rGvXLUqVKldKbdsWKFXJaXZel2rVrp3Ge+uWXX7KkCQsLE2+99Zaczt3dXaP1Sp16C5iZmZkAID766CPx/PlzjXQpKSkiLS1N/l9fC9iLFy/EO++8I68vW7as+Oeff7TuXyUn50pDzpPG/i5Tv9x8/fr1LOv//fffLO+zrrS3bt2S11etWjXL+qCgIHl9t27dRHx8vNYyvXr1SuzevVsMGzZM63p9GIAZSXx8vLCzs5PfsOHDh+c6r+bNm8v5NGrUSCQnJ+tMq960bG9vLxISErKkUT85AhAlS5YUUVFROvOcMGGCxgdGF2MHYJcvX9Yo5++//64zvwsXLsgnqoIOwACIGTNmGJS3oSIjI4WlpaUAIBo0aKAzXV4FYF988YWcxtHRUdy9e1dnflu3btV4LY4ePao1nfqXhSRJei+/79q1SyPQeP36tUHHpk9BBGBpaWnCw8NDTpNdf6KHDx+K0qVLy8Fa5h9wixYtkvOaOnWq3ryMJacBmDHPMYby8vLS+2WruixlZWUlvvzyS71p+/btK6/XFlgdOnRI41h37Nihs1yRkZEal96+/PJLrenUAzB9gV9mugKwx48fi8aNG8vrqlSpYtDlXmMHYMb+Lhs+fLicZunSpVnWqwfa8+bN05tWPdAeMmRIlvUlS5YUQMaPxcyBsLHwNgcjCQkJwfPnzwEAFStWxMKFC3OVT0REhEYz808//YQSJUroTD9r1ix5PKOEhAS5qVmfKVOmwNXVVef6QYMGycuqSzb5YdWqVfJy8+bN0bVrV51p69Spg4EDB+ZHsbLl5uaGTz75xKh5VqpUCS1atACQ8R4kJCQYNX99hBAICQmR///8889RoUIFnemDg4PRrl07+f/Fixdnu4+OHTvinXfe0bm+ffv2KFu2LAAgMTFRHkC1sNm5cydu3LgBAGjVqhX69OmjN33ZsmUxbtw4ABmXjzZt2qSxXr0eGHKpt6Dk9zkmu8tTqktNjRs31qh3+tJmzldFdbkTADp16oTOnTvrLFelSpUwZcoU+f8lS5ZkO6aapaUl5s6dqzeNPlFRUfD398epU6cAZJwrQ0ND8dZbb+U6z9zIi++y7N5n1XNWVlYYMWKE3E1FX9rM+aqoPmvW1tawtbXVWW4lGIAZyd69e+XlIUOGwMLCIlf5qFeKOnXqoF69enrT29jYoHfv3lq316V79+5613t5ecHKygoA8PjxYyQmJmabpzGon/jee++9bNNn92WWX7p16wYzs5x3p7x37x62bNmCWbNmYdKkSRg1ahRGjhwpPyIjIwFkBET//POPsYutU0REBB49egQg41b0/v37Z7uNel8aQybmza4OSpKEOnXqyP8XZH8mJfbs2SMv9+rVy6BtWrZsKS+HhoZqrFMPhNesWYMXL14oLGHeyO9zjL4v5n///RfR0dEAgBYtWqB+/fo6v5hv3bolT63m4eGhNYhU30Y9kNRl4MCB8pAODx8+xPXr1/Wmb9u2LZycnLLNV5vr16+jSZMmuHr1KoCMAbKPHj0q/5jJT3nxXab+Pms7z6j6fzVp0gTW1tZyXzdtabMLtFWftSdPnhjUsJEb7IRvJKoOlwDklovcUO+0bUiHTVU6Vcd/9ZGMtXFwcIC7u7veNJIkwcnJCS9fvgQAPHv2LM9+Aaikp6fj4sWL8v+GdOCvX78+JEkq8FG669evn6P0p06dwqefforjx48bXPa4uLjcFC1X1Ougl5eXQZ1v1evqo0ePEB0drbcFpFatWtnmqb7fZ8+eZZv+TaRqhQCA3bt348KFC9luo36smSe5b9++PWxtbZGYmIjw8HBUq1YNAwcORIcOHVC/fn1FndmNpSDOMfq+mNX/b9GiBUxNTeHv7489e/ZkSZtdq0hUVBRiYmLk/5s0aZJt2UqVKgVPT0+5Fff8+fPw8vLSmT6n5xOVs2fP4sMPP5TPFZ06dcKmTZtgaWmZq/yUyovvsjJlysDLywvXrl3Do0ePEBERId88kznQVv39888/s6S9efMmHjx4AEB3oN2zZ0/MmjULAPD+++9j48aN6NmzJ1q2bGm0QWoZgBlBQkKCfCIBoKipNzY2Vl6uWLGiQdtUqlRJXs7ui9rBwcGgPNVP5OoDWOaVhIQEjf2UL18+221sbW3h4OCAp0+f5mHJspeTS0ErV67EBx98kOOgUXV5Oz/kpg6WKVMGlpaW8l29cXFxegMwQ+phftfBvKD6QgCA7du353h79ZHJAcDZ2RmrVq1Cnz59kJKSgqioKMyYMQMzZsyAlZUVGjRogICAAHTu3DnXX+RKFcQ5Rv2L+b///tP4slUFWdbW1vIPu8DAQOzZs0dnWlWazNQ/G1ZWVihdurRB5atUqZIcgGV3js7tpeXevXvLd+z16dMHq1evzlXLvLHk1XdZYGCg/FoeOXJE73uXOTA39H0GgKlTp+Lo0aM4ceIEhBDYuXMndu7cCQCoXLky/P390bp1awQFBcHe3t6g48uMlyCNIPOXo5JfcupN8aoZ6LOjni67L2pDhqkoCJkvQWi7/V4bQ1+jvKS6lJKdiIgIDBs2TA6+ateujR9++AFhYWGIiYnBy5cvITJujIEQQuPSX3p6ep6UXZvc1MHMaQtrPTQ2pS136iOTq7z77rs4e/YsunfvrtGn5uXLlzh69Ci++uor+Pr6wtfXN8fDFhhDQb236lce1FuyVJelmjZtKgd9ui5ZZvfFnB+fDUPPJ5mpB7SRkZEajQIFIa++y3S9z9oC7Xr16snBkXpa9WVdV6ysra1x+PBhLFiwAFWqVNFYFxkZibVr16Jfv34oV64cJk6cmKvXmwGYEWQej0pJfwb14M3Q/h3q6d6UsbFyKvMH1NBxVfKiD0xeBTsLFiyQv1Dbt2+Ps2fPYtSoUfD19UWpUqWyXCrIz1Yvdbmpg5nTFtZ6aGzq9frChQsaAbYhD11932rVqoVNmzYhNjYWu3btwqRJk9C4cWONL+Fz586hRYsW2Lx5c14f5htBW1B1/fr1LJelgIwvZlVLnSqt+lhhnp6eWltw3+TPxm+//SZ3Yj958iTeeeedAjuHAHn3XZa5VUv1g1YVgDVr1kz+HKguN2dOqz5WmK4WMCAjqB07dixu3ryJS5cu4aeffkKvXr00xoZLSkrC/Pnz0aJFixwHYQzAjMDe3l7jV4uq83RuqDc/37t3z6BtVIMGApA/gIWNg4ODxpeH6vq8PomJiQa1MKg3w2trUcgsr/obqQatBYCvv/462/466u9rfspNHYyJidEYVLiw1kNjK1OmjLysuhvSmOzt7dGhQwfMmTMHJ0+eRFxcHFavXi1fyklPT8eIESMKvDUkP2j7Ys7c/0vF1NRUo4O2ECLb/l+A5mfj5cuXBvfNzI9zdK1atXDo0CGNIKxdu3b5dhNVZnn1XVa6dGn5UmJsbCyuXLmC69ev4+HDhwCytmip3ktV2n///Vcj0Da0P1fNmjUxYsQIbNiwAQ8ePEB4eDgGDx4sr//77781BrI1BAMwI2nYsKG8fOjQoVzn4+PjIy+fPHnSoG1OnDghL2d3p8mbysTERKNj9pkzZ7Ld5ty5cwb1pVL/JaUa8VmXlJQU/Pvvv9nmmRvq/YFq1KihN+2zZ880bkrQJS8u96jXwWvXriE+Pj7bbdTrYNmyZfX2/ypO1M8L+/bty/P92dvbo3///jh06JB8J3ZcXJzGzQBFlfoXc1xcHK5cuSIHYLa2tvD19dVIr/qizpwW0B2Aubm5afT7MuQc/fjxY407H/PyHF2rVi0cPHhQDl5OnDhRYEFYXn6XZQ629b136gFZdmlzom7duli+fDmGDh0qP6c+Q4UhGIAZifo4SMuWLUNycnKu8lG/BT08PDzb4QdevnyJjRs3at2+sFH/MBhy2+8vv/xiUL7q09hkdxfaH3/8oXN6KKVUt6ID2V9iXb58uUEdk9UvWxqro7q3t7d823paWhrWr1+f7TbqY7gpuQu4qOnYsaO8/Ouvv2rcQZeXKleurBHk//fff/my34KW+TKk6lJTs2bNsnRIz5zW0C9m9fq9evXqbMu0Zs0auVuDq6srqlWrlu02StSuXVsjCAsNDS2QICwvv8t0vXfaAm0fHx+Ny82GtHTmhPpnPKefMwZgRjJkyBD5mvfdu3cxduzYXOXj5eWF5s2by/+PGjVK7xfr559/Lp/U7e3tDRo/602lPrDqsWPHsG3bNp1p//nnH40vfX3Uh7TQd8JMSEjAp59+alCeuaF+d+yOHTt0prtx4wa+/PJLg/JUH6oh81x3uSVJksavuq+++kpv3rt375bvDgKA4cOHG6UcRUG3bt1QtWpVABlB9/vvv29woJyYmJil74yhl7xSU1PlSzLAmz1oqzGpB0chISE6L0sBGV/Mjo6OWdJWq1ZN72WpYcOGycvbtm3TGAMys/v372PGjBka2+bHTQqqIEx1fggNDUX79u3zNQjLy+8y9cDp6NGjcqDt7++fJdA2MTGRy6GeNnM+6pKTkw1+rdQvr+b0c8YAzEicnJwwZ84c+f8lS5agZ8+eOvsyXblyBWPGjMH+/fuzrJs9ezZMTU0BAMePH0e3bt2y/HJOSUnB1KlT8e2338rPTZs2Lc/H68pLNWvWRI8ePeT/+/btiy1btmRJd+LECbRv3x5paWl6R1ZWUf8gb9y4ET/++GOWNNeuXUPLli1x69atXA+imx31X0off/yx1ktSBw8eRGBgIJ4/f27QnUPql2337dtntP5rY8eOlTuaPn78GK1atdLaevjbb7+hZ8+e8v+dOnXSOOkWd6ampli8eLH8ef7rr7/QvHlzvaO/X7x4EZMnT0aFChWy9CedOHEi/P39sWbNmixDVKjExsZi0KBBckBhb29v8DhMhZ36F+rly5e1Pq9iYmIid9DOLq26Fi1aaFzx6N69u9YbHcLDw9GqVSv5fXJ3d8fo0aMNOQyjqF27Ng4dOiQHYcePH0f79u3zdfDevPouK126NKpXrw4g4/ykquu63jvV8+pp9QXaDx8+hLu7O8aPH6+zO4wQAnv37sUXX3whP9e+fXudZdaG44AZ0YgRI3D58mV5KpZNmzbh999/h5+fHzw9PWFpaYnY2FiEh4fLdzdp+2XWuHFjzJ49W55BfufOnahQoQJatGgBd3d3PHnyBEeOHNH4NRwcHCxPYVKYLVq0CGfOnMGdO3fw4sULdO/eHdWqVUODBg1gZmaGS5cu4ezZswAygpjff/9d7ripfolPnb+/Pzp06IDdu3cDyPgl9tNPP6FRo0aQJAnXr1/H6dOnkZ6ejgEDBiAyMlLjV5KxjBs3DitWrEBsbCyePHmCd955B/Xq1UP16tUhSRLOnz+PK1euAADefvttlC5dGuvWrdObp5+fHypUqIB79+7h0aNH8PLyQtu2beHi4iL/0vbz89MIkgzh5OSEX3/9Fe3atUNSUhKuX7+OevXqoWHDhqhevTpSUlLw999/a3Qs9/DwwIoVK3L4qhR9rVu3xuLFi/Hhhx8iLS0Np0+fRoMGDeDh4QEfHx95QNJHjx7hwoULei9TCiEQGhqK0NBQmJqaolq1aqhevbqcx4MHD3Dy5EmkpKTI28yfPz/XQxsUNqVKlUL16tXlkeCBjABU15hoLVq00Gi9BQy7LLVq1So0bdoUt27dQmJiInr06AEPDw80bNgQJUqUQEREBE6fPi33UbWxscGGDRvkFrf8ogrCWrZsicePH8tB2J49e/JlCJ+8/C4LDAzUeJ8B3d0ftD2f3fv89OlTfPfdd/juu+/g7OwMHx8fuLm5wcLCAjExMbh48aLGDyRPT0+MGTMm23JryJMZJou5hQsXCnt7e42JTbU9JEkS+/bt05nP8uXLs83H1NRUjBkzRqSmpurMx5CJsDPTNcmrOmNPxq1y+/ZtUbNmTb3H/cEHH4iUlBTh6uoqP6drtnohMian9fX11ZvnoEGDxKtXr3I8GbeuNNqcPHlSuLi46C1Hly5dxNOnTzUm6F21apXOPHfv3i0sLCx05pf5vcnJpNSnTp0Sb731VrZ1uXXr1iImJkZvXobUKXWGHr+hCmIybnWHDh3SmJg7u0eNGjWyTGg9cuRIg7e3s7MTISEhhrw0euV0Mm5jnmNyY8SIERqvQ4cOHXSmPX/+fJbX7eHDhwbt59GjR6Jly5bZvg9Vq1YVZ86c0ZtXbuu6oa/hhQsX5MmlAYjmzZuLxMTELOlyUp9zcg401neZuk2bNmlsb29vr3PbtLQ04eTkpJF+w4YNOvN+8OCB3nNq5kdgYKDB9UYdW8DywJgxY/D+++9j9erV2LdvH65evSpH+C4uLvD29kZAQAB69uwJDw8PnfkMHjwYQUFBWLZsGf7880/8+++/iI+Ph52dHdzd3dG6dWsMGjRIbootKipXrozz589j+fLl2LBhA65evYrExESUK1cOfn5+GDJkCNq0aQPgfyOFS5KkdzRiZ2dnnDx5Us7zypUrGnkOHTpUzjMvNW7cGFeuXMHChQuxc+dO3L59GwBQrlw51K9fH++//z46deqUozzbt2+Pc+fO4ccff0RoaCju3r2LxMREo0zR1KhRI0RERGD9+vXYvn273EJjbm6OsmXLolmzZujduzfatm2reF9FXYsWLXDt2jVs27YNu3fvxunTp/Ho0SMkJCTA2tpaHs29SZMmaNeuHerWrZslj0WLFmHEiBE4cOAATp8+jStXruDevXt4/vw5zMzMULJkSdSoUQNt27ZF3759DR6pvShp0aIFfv75Z/l/fS0dderUgZOTk3weqVatmsHzJpYpUwYHDx7Evn37sHHjRoSGhuLRo0d4/fo1SpcuDR8fH3Tp0gXvv/9+gU8RVadOHRw8eBCtWrXC48ePcezYMfmqQH60hOXFd1lgYKDGVHT+/v7y5c7MVP3A1Pve6qsXbm5uePz4MQ4dOoTjx4/j3LlzuHnzJmJjY5GSkgI7OztUrFhRvrrQunVrg8utThLGOEsTFYAbN27A09MTQMaJUzU9BRER0ZuOnfCp0Nq0aZO87OfnV4AlISIiyhkGYFQo3b17F/Pnz5f/7927dwGWhoiIKGcYgNEbp2fPnvj99991Dma7b98+NGvWDE+fPgWQMRTDO++8k48lJCIiUoZ9wOiNU6lSJdy9exd2dnaoV68eKlWqBEtLSzx+/BhnzpzRGPjOxsYGoaGhWjssExERvakYgNEbRxWAZeett97Cpk2bdI7xQ0RE9KZiAEZvnMjISGzduhXHjx/H7du3ERcXh8ePH6NEiRIoVaoU6tevj44dO+K9994r8Nu7iYiIcoMBWB5LT09HdHQ07Ozs8mUOMCIiIsobQgg8f/4crq6uOmdfMRQHYs1j0dHRcHd3L+hiEBERkZHcv38f5cuXV5QHA7A8ZmdnByDjzdI3UjsRERG92RISEuDu7i5/tyvBACyPqS472tvbMwAjIiIqAozRpYjjgBERERHlMwZgRERERPmMARgRERFRPmMARkRERJTPGIARERER5TMGYERERET5jAEYERERUT5jAEZERESUzxiAEREREeUzBmBERERE+SzPpyJKTk5GaGgo4uLiULlyZTRo0CCvd0lERET0RlMUgN29exc//fQTAGDKlClwdHTUWH/69Gm8++67ePjwofxcvXr18Pvvv6NChQpKdk1ERERUaCm6BLlt2zbMnz8fhw4dyhJ8PX/+HF26dMHDhw8hhJAf586dQ4cOHZCamqpk10hISMDGjRsxfvx4BAQEoGrVqnBwcECJEiVQunRpBAYGYu7cuXj8+LFB+e3duxddu3ZF+fLlYWFhgfLly6Nr167Yu3evonIWVulpaYUybyIiosJAEkKI3G7coUMH7N27F19++SU+++wzjXXffvstJk6cCEmSMGrUKLRq1Qr79u3Dzz//DEmSEBISgsGDB+e64AcOHECbNm2yTefi4oL169fj7bff1rpeCIHhw4cjJCREZx5Dhw7FkiVLcjX7eUJCAhwcHPDs2TPY29vnePuCtLVPH8RGRBg1z1Le3uj6yy9GzZOIiCg/GPM7XdElyNu3bwMA6tevn2Xdpk2bIEkSgoODsXDhQgBAp06dEBsbi82bN2PLli2KAjAAcHd3R4sWLVC/fn24u7ujXLlySE9Px4MHD7BlyxZs3boVcXFx6Ny5M8LCwlC7du0seXz22Wdy8OXj44NJkyahSpUquHXrFubOnYvw8HCEhISgVKlSmDFjhqLyFjaxERF4FB5e0MUgIiIqchS1gJUsWRJPnz5FWFgY6tWrJz+fkJCAkiVLIj09HX/88Qc6dOggr9u2bRu6deuGsmXLIjo6OtcFT0tLg6mpqd4027dvR3BwMACga9eu+P333zXW37x5E97e3khNTYWvry+OHTsGKysreX1SUhICAgJw9uxZmJmZ4dq1a6hSpUqOylmYW8CW1qtn9ACsrI8Php0/b9Q8iYiI8oMxv9MV9QF7/vw5gIxgSN2JEyfkACkwMFBjnbu7OwAgPj5eya6zDb4AoEuXLvDy8gIAHDt2LMv6BQsWyH3RFi1apBF8AYC1tTUWLVoEAEhNTZVb8oiIiIiUUBSAOTg4AECWlqwjR44AAOrUqQMbGxut21paWirZtcFU+3/16pXG80II7NixAwDg5eWFRo0aad2+UaNGqFatGoCMFjUFDYZEREREABQGYDVr1gSQcVlRJS0tTe7/1aJFiyzbREVFAQDKlCmjZNcGiYiIwIULFwBAbglTiYyMlMsSEBCgNx/V+gcPHuDOnTtGLycREREVL4o64QcHB+Po0aNYt24dypQpA39/f6xbtw53796FJEno0aNHlm3Onj0LAHk2DlhSUhKioqKwc+dOzJ07V748OmbMGI10EWp392UOzjJTXx8REYHKlSsbscRERERU3CgKwIYNG4alS5ciIiIC8+fPx/z58+V1nTp1gq+vb5Zttm3bBkmSsvQNU2L16tUYOHCgzvUTJkxAnz59NJ67f/++vFy+fHm9+av6rWXejoiIiCg3FF2CtLCwwMGDB9G1a1eYmZlBCAFzc3P07dsX69aty5L+2LFjuHr1KgAYNIaXUnXr1sXp06cxb968LGN4qW4gAABbW1u9+aj3Y0tMTNSbNjk5GQkJCRoPIiIiInWK54IsW7YstmzZguTkZMTHx6NkyZIoUaKE1rTu7u44fPgwAMDPz0/prmVdunSRW9tevnyJW7duYdOmTdi2bRv69OmDhQsXomPHjhrbqHfK11VeFQsLC3n55cuXetN+8803+PLLL3N6CLmSnpYGEwPuBiUiIqI3i9Em47awsEC5cuX0pqlcuXKe9J9ydHTUmArJz88PvXr1wrp169C/f38EBQVhxYoVGDBggJxG/S7MlJQUvfknJyfLy5mHqshs8uTJ+Pjjj+X/ExISNC5hGpOJqWmejFbv0a4dWs6cadQ8iYiI6H+MFoC9ifr27Ytdu3Zh06ZNGDlyJIKCguDk5AQAsLOzk9Nld1nxxYsX8nJ2lystLCw0WszyWl6MVu+SzU0JREREpIyiPmBly5ZFz5498fPPP+PKlSvGKpNRBQUFAcgIov7880/5efWO9w8ePNCbh3rH+7xqzSIiIqLiQ1ELWExMDLZs2YItW7YAyJiaqHnz5ggMDERAQABq1apllEIqUapUKXn57t278nL16tXl5WvXrunNQ329t7e3EUtHRERExZGiFrDRo0ejdu3akCQJQgjExcVh27ZtGDNmDOrWrQsXFxd07doVP/zwA/755x9jlTlHVIOtApqXDytXrgxXV1cAwNGjR/XmoZrGyM3NDZUqVTJ+IYmIiKhYURSALVy4EOHh4XLgNXbsWNSpU0cOyOLj47Fjxw6MGzcO9erVQ8mSJdGlSxd5u/ywefNmeVm9RU6SJPny5LVr13D69Gmt258+fVpuAQsKCsoynAURERFRTikKwFQcHR0RFBSE7777DufPn8fjx4+xfft2jBs3DnXr1pUDsidPnmDnzp0YP3684mEoVq9enWV+x8wWLFiAPXv2AAAqVaqEZs2aaawfO3YszMwyrsKOGjUqyxATL1++xKhRowAAZmZmGDt2rKIyExEREQF5dBekg4MDOnfujM6dOwMA4uPj8f333+OHH35AQkKCUSa0nj59OsaPH49u3bqhWbNmqFKlCmxtbfH8+XNcunQJv/zyC06cOAEgY5yvZcuWycGWiqenJyZMmIDZs2fj7NmzaNq0KT755BNUqVIFt27dwpw5c+SWuokTJ8LDw0NxuYmIiIjyJAATQiA8PBxHjhzBkSNHcPz48SyBl7W1teL9xMfHY9myZVi2bJnONOXLl8fKlSvRunVrretnzpyJmJgYrFy5EuHh4ejVq1eWNIMHD8aMGTMUl5eIiIgIMFIApivgUq0DMgYwbdy4MQIDAxEYGIiGDRsq2ufBgwdx4MABHD58GBEREfjvv//w+PFjWFpaokyZMqhbty46duyIHj166A32TExMsGLFCnTr1g0hISEICwtDXFwcXFxc4Ofnh2HDhqFdu3aKykpERESkTlEAtmDBAhw+fBihoaF49uwZAP0Bl7m5ufIS/78qVaqgSpUqGDZsmFHya9++Pdq3b2+UvIiIiIj0URSAjR8/Xu5gn9cBFxEREVFRYZRLkJIkwd/fH23btkVgYCB8fHw4XAMRERGRDooCsJYtW+L06dNISkrC/v378ddffwEA7O3t4e/vL7eGMSAjIiIi+h9FAdiBAwfw+vVrnDlzBkeOHMHRo0dx8uRJPHv2DLt27cLu3bsBMCAjIiIiUqf4EqS5uTmaNm2Kpk2bYurUqUhNTc02IHNwcIC/vz927Nih+ACIiIiIChujjISvzszMDE2aNMGUKVOwb98+PHnyBCdOnMDkyZNhb28PIQSePn2KXbt2GXvXRERERIVCngzECgBJSUkIDQ2VxwY7d+4cUlNT5bsmiYiIiIorowVgugIuAFlGwFcNV0FERERUHCkKwP766y8cPnxYb8BlY2ODJk2aIDAwEAEBAWjQoEGWORmJiIiIihNFkdDbb7+d5ZKira0tmjZtKgdcfn5+MDU1VVxQIiIioqJCcVOUnZ2dRsBVv359BlxEREREeigKwM6cOYN69erBxMToN1MSERERFVmKAjBfX19jlYOIiIio2GDTFREREVE+M9rtiPHx8Vi1ahUOHDiAy5cvIz4+HgDg7OyMmjVronXr1hg4cCCcnZ2NtUsiIiKiQskoAdjSpUsxYcIEJCUlAdAchiIqKgrR0dHYv38/pk+fjm+//RZDhw41xm6JiIiICiXFAdjs2bMxdepUOehycHCAj48PypYtCyEE/vvvP4SHh+PZs2d48eIFPvzwQzx9+hSTJk1SXHgiIiKiwkhRAHb58mV8/vnnEEKgXLlymDdvHrp37w5zc3ONdKmpqdi8eTMmTpyI6OhofPbZZ+jQoQNq1KihqPBEREREhZGiTvg//vgj0tLSUKpUKZw6dQrvvfdeluALyJigu3fv3jh16hRKly6NtLQ0/Pjjj0p2TURERFRoKQrADh06BEmSMHnyZFSoUCHb9O7u7vjkk08ghMDBgweV7JqIiIio0FIUgEVFRQEAmjRpYvA2TZs2BQBER0cr2TURERFRoaUoAFNNOaSahNsQqrQcPZ+IiIiKK0VRkOqyY04uJ6rSGnLJkoiIiKgoUhSAtWnTBkIIzJ8/H5cuXco2/cWLFzFv3jxIkoS2bdsq2TURERFRoaUoABs7diwsLCyQmJiIZs2aYd68eYiLi8uSLi4uDnPnzkXz5s2RmJgICwsLjB07VsmuiYiIiAotReOAVaxYEUuXLsXAgQORmJiITz/9FJMnT0alSpVQpkwZSJKER48e4c6dOxBCQAgBSZKwdOlSXoIkIiKiYkvxSPj9+vVDyZIlMXz4cERFRUEIgdu3byMyMhKA5rRErq6uCAkJQfv27ZXuloiIiKjQMspckB06dEBkZCS2bdumdzLu4OBgmJkZbf5vIiIiokLJaNGQmZkZunfvju7duxsrSyIiIqIiiYNxEREREeUzBmBERERE+YwBGBEREVE+M6gPmGrKIWOSJClHUxgRERERFRUGBWDqQ0kQERERkTIGBWDTpk3L63IQERERFRsMwIiIiIjymUEB2A8//AAA6Nu3L5ycnPK0QERERERFnUEB2NixYyFJElq3bq0RgLVs2RKSJGHlypWoWLFinhWSiIiIqChRNBL+kSNHIEkSXrx4YazyEBERERV5Bo0DZmlpCQB4+vRpXpaFiIiIqFgwKACrVKkSAGDXrl15WRYiIiKiYsGgS5Dt27fHtWvXMGfOHBw8eBCenp4wNzeX13/22WdwdHTM0Y4lScKKFStytA0RERFRUWBQADZlyhT88ccfuHnzJsLCwnD27Fl5nRACO3bsyNFOhRAMwIiIiKjYMigAc3Z2xtmzZ/Hjjz/i4MGDiIqKQnJyMu7evQtJklCuXDmNFjEiIiIi0s3guyDt7e0xZcoUTJkyRX7OxCSjC9n+/ftRvXp145eOiIiIqAgyqBM+ERERERmPonHADh8+DACoXLmyUQpDREREVBwoagFLTU1FQEAArKyscrztiBEjlOyaiIiIqNBSFIAFBwcjLCwsx9sNHToUS5cuVbJrIiIiokJLUQCWmJiI9u3bIyIiwuBtPvjgAyxfvlzJbomIiIgKNUUBmKenJx4/foy2bdvi3r172aYfMGAAVq1aBQDo06ePkl0TERERFVqKArC//voL5cuXR1RUFNq0aYOYmBit6YQQ6Nu3L9auXQshBPr37481a9Yo2TURERFRoaUoAHN3d8dff/0FFxcX3Lx5E2+//TaePXumkSY9PR19+vTBL7/8AgAYNGgQVq5cCUmSlOyaiIiIqNBSPA5YtWrV8Oeff8LW1hYXL15Ex44d8erVKwAZwVfv3r2xceNGAP/r/8Xgi4iIiIozowzEWr9+fezYsQMWFhY4efIkunXrhpcvX+Ldd9/F5s2bAQDDhg1DSEiIMXZHREREVKgZbST8wMBAbNiwASYmJti7dy8qV64sT9L94YcfYvHixcbaFREREVGhZtSpiIKCguQhJmJiYiCEwMiRI/HTTz8ZczdEREREhZpBUxEZMsSESosWLTBmzBgsXLgQ7777LiZMmKBz+woVKhicLxEREVFRYVAAlpu5HiVJwu+//47ff/9d5/rU1NQc50tERERU2BkUgAkh8rocRERERMWGQQGYavR6IiIiIlLOoACsf//+eV0OIiIiomLDqHdBEhEREVH2FAVgJiYmMDMzw9y5c41VHiIiIqIiT1EAVqJECQgh4O/vb6zyEBERERV5igIwV1dXAICpqalRCkNERERUHCgKwJo3bw4AOH/+vFEKQ0RERFQcKArARo0aBVNTU8yfPx8JCQnGKhMRERFRkaYoAKtfvz4WLVqEu3fvIiAgACdPnjRWuYiIiIiKLIPGAdNl0KBBAIBq1arhn3/+gb+/P9zd3VG7dm04OTnp7RsmSRJWrFihZPdEREREhZKiAGz16tWQJAlARkAlhMC9e/dw//59vdsJIRiAERERUbGlKACrUKGCHIARERERkWEUBWB37twxUjGIiIiIig9ORURERESUzxiAEREREeWzPAnAUlNTERsbi9jYWKSmpubFLgBkDAA7a9YstGvXDu7u7rCwsICtrS08PT0xYMAAHD9+PEf57d27F127dkX58uVhYWGB8uXLo2vXrti7d28eHQEREREVR4r6gKmLiIjAzz//jAMHDuDGjRsQQgDIuDvSw8MDbdq0wfDhw1G9enWj7C8gIADHjh3L8nxKSgpu3LiBGzduYM2aNejbty+WL1+OEiVK6MxLCIHhw4cjJCRE4/moqChs27YN27Ztw9ChQ7FkyRLedEBERESKGaUFbPLkyahduzZ+/vlnXL9+Henp6RBCQAiB9PR0XL9+HT/99BPq1KmDKVOmGGOXiIqKApAxH+WYMWOwZcsWnDlzBqdOncJ3330HNzc3AMC6deswYMAAvXl99tlncvDl4+ODDRs24MyZM9iwYQN8fHwAACEhIfj888+NUnYiIiIq3iShaqrKpVGjRuHnn3+WW7y8vb3RsGFDlC1bFkII/Pfffzhz5gyuXr2asUNJwsiRI/H9998rKnjHjh3Rr18/dOvWTeuAr3FxcWjatCn+/fdfAMCxY8fg7++fJd3Nmzfh7e2N1NRU+Pr64tixY7CyspLXJyUlISAgAGfPnoWZmRmuXbuGKlWqGFzOhIQEODg44NmzZ7C3t8/Fkeq3tF49PAoPN2qeNXv3Rrdff82TvMv6+GAY5w4lIqJCyJjf6YouQZ44cQI//fQTJElC9erVERISgiZNmmhNe+rUKQwfPhyXLl3Cjz/+iJ49e+pMa4hdu3bpXe/i4oJvv/0WnTp1AgBs2bJFawC2YMECuZ/aokWLNIIvALC2tsaiRYvQuHFjpKamYuHChVi0aFGuy01ERESk6BLk0qVLAQCVK1fGiRMn9AZUjRs3xrFjx/DWW28BAJYsWaJk1wYJDAyUl2/dupVlvRACO3bsAAB4eXmhUaNGWvNp1KgRqlWrBgDYvn07FDYaEhERUTGnKAA7fvw4JEnCp59+CgcHh2zTOzg44JNPPoEQIsd3KOZGSkqKvGxikvVQIyMj5b5kAQEBevNSrX/w4AEHoCUiIiJFFAVgjx49AgC5o7oh6tWrBwD477//lOzaIEePHpWXvby8sqyPiIjQu16d+nr17YiIiIhySlEAZmlpCQB48eKFwdskJiYCACwsLJTsOlvp6emYPXu2/H+PHj2ypFGfNLx8+fJ683N3d9e6HREREVFOKQrAKleuDAD4448/DN5m586dACD3BcsrCxYswJkzZwAAwcHB8PX1zZLm+fPn8rKtra3e/GxsbORlVRCpTXJyMhISEjQeREREROoUBWDt27eHEAI//vgjDh48mG36gwcPYtGiRZAkCe3bt1eya72OHj2KTz/9FABQunRpLF68WGu6V69eycv6BmoFNFvsXr58qTPdN998AwcHB/mh3nJGREREBCgMwMaOHQt7e3u8fv0a7dq1w0cffYRz584hPT1dTpOeno5z585hxIgRaNeuHV6/fg17e3uMHTtWadm1unLlCoKDg5GamgoLCwts2rQJZcqU0ZpWdQkV0Oywr01ycrK8nHmoCnWTJ0/Gs2fP5AcvVxIREVFmisYBc3FxwaZNm9C5c2ekpKRgyZIlWLJkCUqUKAFnZ2dIkoTHjx/LwY0QAiVKlMDmzZtRsmRJoxyAusjISLRt2xZPnjyBqakpNmzYoPfuRjs7O3lZ32VFQLOfm77LlRYWFnnev42IiIgKN8VTEbVt2xanT5+Gr6+vPP1QcnIyHj58iOjoaCQnJ8vP+/n54e+//0br1q2NUXYN0dHRaN26NaKjoyFJElauXIng4GC926h3vH/w4IHetOotWbysSEREREoYZTLuunXr4syZMwgLC8OBAwdw+fJlxMfHAwCcnZ1Rs2ZNtG7dGn5+fsbYXRZxcXFo06YNbt++DSBjRPt+/fplu536xODXrl3Tm1Z9vbe3dy5LSkRERGSkAEzFz88vz4IsXZ49e4a3335bnmty9uzZ+OijjwzatnLlynB1dUV0dLTGmGHaHDt2DADg5uaGSpUqKSozERERFW+KL0EWpKSkJHTo0AHn/39y56lTp+KTTz4xeHtJkhAUFAQgo4Xr9OnTWtOdPn1abgELCgqCJEkKS05ERETFmaIAbPHixXLLU35LSUlBcHAwTpw4AQAYM2YMZsyYkeN8xo4dCzOzjIbAUaNGZRli4uXLlxg1ahQAwMzMLM/u3iQiIqLiQ9ElyI8++giSJMHFxQXNmzdHYGAgAgICULNmTWOVT6fevXtj//79AICWLVti8ODBuHz5ss70JUqUgKenZ5bnPT09MWHCBMyePRtnz55F06ZN8cknn6BKlSq4desW5syZg/DwcADAxIkT4eHhkTcHRERERMWG4j5gQgjExsZi69at2Lp1K4CMjvfqAVnt2rUVFzQz1b4A4NChQ9nuo2LFijon0Z45cyZiYmKwcuVKhIeHo1evXlnSDB48OFctbERERESZKQrA4uPjcfToURw5cgRHjhzBpUuXkJ6ejsePH2P79u3Yvn07AMDJyQn+/v5yQFa3bl0jFN14TExMsGLFCnTr1g0hISEICwtDXFwcXFxc4Ofnh2HDhqFdu3YFXUwiIiIqIhQFYI6OjggKCpI7sj99+hTHjh2TA7KLFy8iPT0d8fHx2LFjhzxnpIODA5o3by4HaLkhhFBSdK3at2+fp1MkEREREQFGHobC0dERnTt3RufOnQFkDBGhHpD9888/SE9Px9OnT+VJuYmIiIiKmzwdhsLe3h5ubm5wc3ODq6srrK2tOYQDERERFXtGbQETQiA8PFxu8Tp+/DgSEhLkdUDGRNaNGzdGixYtjLlrIiIiokJDUQCWOeAKDQ3Fs2fP5HXA/wKuwMBABAYGomHDhjA3N1deciIiIqJCSlEAVrJkSQZcRERERDmkKAB7+vSp3KcrKCgI48aNQ+PGjRlwEREREelhlIFYAeCPP/7AkSNHNAZg9fHxYad7IiIiokwUBWDHjx+X+3+dOnUKz549w86dO7Fr1y4AGXdBqgZgDQwMZEBGREREBIUBWNOmTdG0aVNMnToVqampOHPmjByQnTx5Es+ePcOuXbuwe/duAFkDsnr16hnlIIiIiIgKE6MNQ2FmZoYmTZqgSZMmmDJlClJTUxEWFpYlINu9ezd2794NSZKQmppqrN0TERERFRpGHQdMI2MzMzRu3Bh16tRBvXr1UKtWLaxYsQIJCQl5Mo0QERERUWFh9AAsKSkJoaGhcsvXuXPn5JYu9cDL2tra2LsmIiIiKhQUB2CGBlw2Njby+GABAQFo0KCB0l0TERERFUqKO+GfPXtWa8Bla2uLJk2ayAGXn58fzMzy7IonERERUaGhKCI6deqUvGxnZ4emTZvKAZevry9MTU0VF5CIiIioqFEUgLVv314OuOrXrw8TExNjlYuIiIioyFIUgKkGXCUiIiIiw7HJioiIiCifMQAjIiIiymcMwIiIiIjyGQMwIiIionzGAIyIiIgonzEAIyIiIspnDMCIiIiI8hkDMCIiIqJ8ZlAA1rVrV3Tr1g0PHjzI6/IQERERFXkGBWDbt2/H9u3bkZCQoLmxiQnMzMxw9erVPCkcERERUVGUo0uQQgiDniMiIiIi3QwKwOzs7AAA//33X54WhoiIiKg4MCgA8/LyAgB8//33SExMzLJekiTjloqIiIioCDMzJNF7772HsLAw7Nq1C87OzihTpgzMzc3l9W3bttX43xCSJOHWrVs5Ky0RERFREWBQADZq1CicOHECW7ZsQWpqKqKiouR1QgiN/w3FVjMiIiIqrgwKwExMTLBp0yacOnUKBw4cQFRUFJKTk7FmzRpIkoTOnTvD0dExj4tKREREVDQYFICpNG7cGI0bN5b/X7NmDQBg5syZqF69unFLRkRERFREcSR8IiIionyWoxawzNLT041VDiIiIqJigy1gRERERPlMUQuYurS0NGzfvh0HDhzA5cuXER8fDwBwdnZGzZo10bp1a3Tp0gWmpqbG2iURERFRoWSUAGzv3r0YOnRoluEpgIzhJk6ePImQkBCUL18eISEhePvtt42xWyIiIqJCSfElyHXr1qFjx46IioqCEAJCCFSsWBGNGjVCw4YNUbFiRQAZAdn9+/fRoUMH/PLLL4oLTkRERFRYKQrA7t69i6FDhyI9PR3W1taYMWMGHj16hNu3b+PkyZM4deoUbt++jUePHmHmzJmwtbVFeno6hgwZgnv37hnrGIiIiIgKFUUB2Pfff4/k5GTY2tri+PHjmDJlCkqXLp0lXalSpTB58mQcP34ctra2SE5Oxvfff69k10RERESFlqIAbP/+/ZAkCRMnTkTdunWzTV+nTh1MmDABQgjs27dPya6JiIiICi1FAZjqMmLr1q0N3qZNmzYa2xIREREVN4oCsLS0NADI0dASqrQcxJWIiIiKK0UBmJubGwDg5MmTBm+jSuvq6qpk10RERESFlqIArEWLFhBCYPbs2YiOjs42/YMHDzB79mxIkoSWLVsq2TURERFRoaUoABs1ahRMTEwQGxuLhg0bYvPmzfJlSXVpaWnYtGkTGjdujJiYGJiYmGDkyJFKdk1ERERUaCkaCb9mzZr4+uuvMXXqVERHR6NXr15wdHSEj48PypQpA0mS8OjRI4SHh+Pp06fy6Phff/01atasaZQDICIiIipsFE9FNHnyZDg4OGDSpElISkrCkydPcPjwYY00qsDL2toa8+bNw4cffqh0t0RERESFllHmghwxYgR69OiBVatW6Z2Me+DAgXBxcTHGLomIiIgKLaMEYADg4uKCiRMnYuLEicbKkoiIiKhIUjwZNxERERHlDAMwIiIionzGAIyIiIgonzEAIyIiIspnDMCIiIiI8hkDMCIiIqJ8xgCMiIiIKJ8xACMiIiLKZwzAiIiIiPKZopHwv/rqKwBAw4YN8fbbbxulQERERERFnaIAbPr06ZAkCdu2bTNWeYiIiIiKPEWXIEuWLAkAqFChglEKQ0RERFQcKArAqlatCgB49OiRUQpDREREVBwoCsB69uwJIQQ2bdpkrPIQERERFXmKArARI0agTp06WLt2LdasWWOsMhEREREVaYo64T969AjLly/H4MGDMWjQIPzyyy947733ULt2bTg5OcHU1FTv9uw7RkRERMWRogCsUqVKkCQJACCEwMGDB3Hw4EGDtpUkCampqUp2T0RERFQoKQrAgIzAS9syEREREWmnKABbtWqVscpBREREVGwoCsD69+9vrHIQERERFRucC5KIiIgonzEAIyIiIspnRgvA0tPTcejQIcycOROjRo3C4MGD8fDhQ400KSkpSEpKQnJyslH2GRMTg127duGLL75Au3bt4OLiAkmSIEkSBgwYkOP89u7di65du6J8+fKwsLBA+fLl0bVrV+zdu9co5SXAtmxZpKel5Vn+eZk3ERGRsSi+CxIAdu/ejdGjR+POnTsaz48fPx7lypWT/1+xYgVGjhwJW1tbREdHw8bGRtF+y5Qpo2h7FSEEhg8fjpCQEI3no6KisG3bNmzbtg1Dhw7FkiVL5GE3KHcsHR1hYmqKrX36IDYiwqh5l/L2RtdffjFqnkRERHlBcQC2fPlyDBs2TB6CwsXFBXFxcVoDlcGDB+Ozzz7D06dPsW3bNrz//vtKdy9zd3eHt7c39u/fn+NtP/vsMzn48vHxwaRJk1ClShXcunULc+fORXh4OEJCQlCqVCnMmDHDaGUuzmIjIvAoPLygi0FERFQgFF2CvHnzJj766CMAQMuWLXH16lXExMToTF+iRAl069YNQohcBUqZffHFF9i5cycePXqEe/fuYenSpTnO4+bNm5g7dy4AwNfXFydOnECvXr3g5+eHXr16ITQ0FL6+vgCAOXPm4NatW4rLTURERMWbogBs4cKFeP36NWrUqIE9e/bAy8sr2238/f0BABcuXFCyawDAl19+iY4dOyq6FLlgwQJ5RP5FixbByspKY721tTUWLVoEAEhNTcXChQtzvS8iIiIiQGEAdvDgQUiShLFjx6JEiRIGbVOlShUAwL1795Ts2iiEENixYwcAwMvLC40aNdKarlGjRqhWrRoAYPv27Rzxn4iIiBRRFIDdv38fAFC3bl2Dt1F1vE9KSlKya6OIjIxEVFQUACAgIEBvWtX6Bw8eZLnZgIiIiCgnFAVg6hNxGyo2NhYAYG9vr2TXRhGhdhdedpdP1ddHGPnuPSIiIipeFAVgrq6uAIB///3X4G2OHj0KAKhUqZKSXRuFqgUPAMqXL683rbu7u9btiIiIiHJKUQDWvHlzCCHw66+/GpQ+Li4OS5cuhSRJaNmypZJdG8Xz58/lZVtbW71p1ccsS0xM1JkuOTkZCQkJGg8iIiIidYoCsKFDhwIA9uzZg1WrVulN++DBA7Rv3x5xcXEwNTWVty1Ir169kpezu4nAwsJCXn758qXOdN988w0cHBzkh3rLGeWtvBxlnyPsExGRMSkaiNXPzw/Dhw/HkiVL8MEHH2DPnj3o3r27vP7ixYu4dOkS9u/fj40bN+LVq1eQJAnjx49H1apVFRdeKUtLS3k5JSVFb1r16ZMyD1WhbvLkyfj444/l/xMSEhiE5ZO8GmWfI+wTEZGxKR4Jf9GiRXjx4gXWrVuHrVu3YuvWrXLn/D59+sjpVB31BwwYgFmzZindrVHY2dnJy/ouKwLAixcv5GV9lystLCw0Wsso/3GUfSIietMpnozb1NQUa9aswebNm+Hj4wMhhNZH9erV8euvv2LlypVvzHyK6h3vHzx4oDetesd7tmgRERGREkaZjBsAunXrhm7duiE6Ohpnz55FTEwM0tLSULJkSfj4+MgDsL5JqlevLi9fu3ZNb1r19d7e3nlWJiIiIir6jBaAqbi6uqJz587GzjZPVK5cGa6uroiOjpaHx9Dl2LFjAAA3N7c3YggNIiIiKrwUX4IszCRJQlBQEICMFq7Tp09rTXf69Gm5BSwoKOiNuYRKREREhZNRW8Bev36N8+fP4/Lly4iPjwcAODs7o2bNmqhXrx7Mzc2NuTujGDt2LJYtW4bU1FSMGjUKx44d07jL8eXLlxg1ahQAwMzMDGPHji2gkhIREVFRYZQALDExEV9//TVWrFiBJ0+eaE3j5OSEwYMH47PPPtO4+1CJ0NBQ3Lx5U/4/Li5OXr558yZWr16tkX7AgAFZ8vD09MSECRMwe/ZsnD17Fk2bNsUnn3yCKlWq4NatW5gzZw7C//+OuokTJ8LDw8MoZSciIqLiS3EAFhERgXfeeQcPHjzQOydkfHw85s+fj99++w379u1DtWrVlO4ay5cvx5o1a7SuO3HiBE6cOKHxnLYADABmzpyJmJgYrFy5EuHh4ejVq1eWNIMHD8aMGTMUl5mIiIhIUQD29OlTtG7dGg8fPgQA1KxZE/3790eDBg1QpkwZCCEQExODsLAwrFmzBpcuXcK9e/fQunVrXL58GQ4ODkY5CKVMTEywYsUKdOvWDSEhIQgLC0NcXBxcXFzg5+eHYcOGoV27dgVdTCIiIioiFAVgc+bMwcOHDyFJEr766itMmTIlSwf1atWqwd/fH+PGjcM333yDzz77DNHR0ZgzZ47iAVlXr16d5TKjEu3bt0f79u2Nlh8RERGRNorugty+fTskSULPnj0xdepUvXcHSpKEKVOmoGfPnhBCYNu2bUp2TURERFRoKQrA7t69CwDo37+/wduo+mGptiUiIiIqbhQFYKq7GUuXLm3wNqq0+uZTJCIiIirKFAVgtWrVAgDcuHHD4G1UaVXbEhERERU3igKwYcOGQQiBhQsXIj09Pdv06enpWLBgASRJwtChQ5XsmoiIiKjQUhSAde/eHQMHDsTp06fRpUsXPHr0SGfa//77D127dsXff/+NAQMGoGfPnkp2TURERFRoGTQMxdq1a3WuCwgIwOXLl7Fr1y689dZbaNu2Lfz8/FC6dGlIkoT//vsPYWFh2L9/P5KTk+Hn54eAgACsXbsW/fr1M9qBEBERERUWBgVgAwYMyHYCakmS8OrVK+zcuRM7d+7Msl4IAUmScPbsWQwcOBCSJDEAIyIiomLJ4IFY9U0zZGg6Q/MgIiIiKsoMCsAiIyPzuhxERERExYZBAVjFihXzuhxERERExYaiuyCJiIiIKOcYgBERERHlMwZgRERERPnMKAFYREQExo0bB19fXzg7O8Pc3BympqZ6H2ZmBt+ASURERFSkKI6CZs+ejS+++AJpaWkcZoKIiIjIAIoCsM2bN2PKlCkAABMTE/j7+6NOnTpwdHSEiQmvbhIRERFpoygA+/777wEAbm5u2LNnD2rVqmWUQhEREREVZYqaqS5evAhJkvD1118z+CIiIiIykKIAzNzcHABQt25dY5SFiIiIqFhQFIB5enoCAB4/fmyUwhAREREVB4oCsP79+0MIge3btxupOERERERFn6IAbPDgwfD390dISAh27txprDIRERERFWmK7oI0NzfHjh070L9/fwQHB6NXr17o0aMHPD09YW1tne32FSpUULJ7IiIiokJJ8UCsjo6OGD16NE6fPo0NGzZgw4YNBm0nSRJSU1OV7p6IiIio0FE8WurYsWPRtm1bxMXFQQiRowcRERFRcaSoBWz9+vX44YcfAAD29vYIDg5G7dq1ORI+ERERkR6KArBFixYBALy8vHD48GGUKVPGKIUiIiIiKsoUNVNdu3YNkiRh+vTpDL6IiIiIDGSUkfBVA7ISERERUfYUBWBeXl4AgEePHhmlMERERETFgaIAbODAgRBCGDz0BBEREREZYST8jh07Yv369fjxxx+NVSYiIiKiIk3RXZDHjh3D6NGjERsbizFjxuDXX39Fr169DB4Jv3nz5kp2T0RERFQoKQrAAgMDIUmS/P/ff/+Nv//+26BtORI+ERERFVeKpyLiiPZEREREOaMoADt8+LCxykFERERUbCgKwAICAoxVDiIiIqJigxM2EhEREeUzBmBERERE+YwBGBEREVE+U9QHrGXLlrneVpIkHDx4UMnuiYiIiAolRQHYkSNHIEmS3qEo1McJA/43bEXm54mIiIiKC0UBWPPmzbMNpF68eIEbN27g2bNnkCQJnp6eKFeunJLdEhERERVqilvADCGEwO7duzFmzBjEx8dj+fLlaNasmZJdExERERVa+dIJX5IkdOzYEaGhoTA1NUVwcDCioqLyY9dEREREb5x8vQuyXLly+Pjjj/H48WPMnTs3P3dNlGu2ZcsiPS0tz/LPy7yJiOjNpHguyJxSXXrcvXs3vv/++/zePVGOWTo6wsTUFFv79EFsRIRR8y7l7Y2uv/xi1DyJiOjNl+8BWIkSJQAA0dHR+b1rIkViIyLwKDy8oItBRERFQL4PxBoaGgoAsLa2zu9dExEREb0R8jUAO3XqFL766itIkoQGDRrk566JiIiI3hiKLkF+9dVX2aZJT0/HkydPcPbsWfz9999IT0+HJEkYN26ckl0TERERFVqKArDp06fnaER7IQTMzMwwd+5ctGnTRsmuiYiIiAotxZ3w9U1DBGSMAWZnZ4fKlSsjICAAQ4cORfXq1ZXuloiIiKjQUhSApaenG6scRERERMVGvt8FSURERFTcMQAjKkAcZZ+IqHjK94FYieh/OMo+EVHxZNQA7Pnz54iMjMTz58+RZsAv7+bNmxtz90SFFkfZJyIqXowSgC1btgw///wzLl68aPA2kiQhNTXVGLsnIiIiKlQUBWBpaWno1q0bdu7cCSD7ISmIiIiISGEAtmTJEvzxxx8AgDJlymDgwIGoX78+nJ2dYWLC/v1ERERE2igKwNauXQsAqF69Oo4fPw4nJyejFIqIiIioKFPUTBUREQFJkvD5558z+CIiIiIykFGuE1arVs0Y2RAREREVC4oCMA8PDwBAfHy8UQpDRATk3QCyHJiWiN4UivqA9erVC+Hh4di1axdatmxprDIRUTGXF4PTcmBaInqTKArARo8ejV9++QWLFy9GcHAw/P39jVUuIirmODgtERVlii5BWlhYYP/+/ahfvz7atGmDSZMm4cKFC3j16pWxykdERERU5ChqATM1NZWXhRD49ttv8e233xq0LUfCJyIiouJKUQCWeeR7joRPRERElD1FAdi0adOMVQ4iIiKiYoMBGBEREVE+44SNRERERPmMARgRERFRPmMApubevXuYMGECvL29YWNjA2dnZzRo0ADz589HUlJSQReP6I3CUeWJciYvPzOF8fNY3F8PRX3AipLdu3ejT58+ePbsmfxcUlISwsLCEBYWhuXLl2PPnj146623CrCURG+OvBitHgA82rVDy5kzjZon0Zsgrz4zhXWWh+L+ejAAA/DPP/+gR48eSEpKgq2tLSZPnowWLVrg5cuX2LhxI5YtW4br16+jQ4cOCAsLg62tbUEXmeiNkBej1bt4eRk1P6I3CWd40FScXw8GYADGjh2LpKQkmJmZYf/+/WjcuLG8rmXLlvDw8MCkSZNw7do1fPfdd/jiiy8KsLRERERU2BX7PmBhYWE4cuQIAGDw4MEawZfK+PHj4e3tDQBYuHAhXr9+nZ9FJCIioiKm2Adg27dvl5cHDhyoNY2JiQn69esHAHjy5IkcsBERERHlRrEPwI4fPw4AsLGxQf369XWmCwgIkJdDQ0PzvFxERERUdBX7ACzi/+++qFq1KszMdHeJ81LrGBxh5Ds2iIiIqHgp1gHYq1evEBcXBwAoX7683rROTk6wsbEBANy/fz/Py0ZERERFV7G+C/L58+fysiFDS9jY2ODFixdITEzUmSY5ORnJycny/6pxxRISEhSUVE+ZqlaFg5EHnCvh6oqEhATmncf55nXeNlWr5lm9U99HYXlN8uP1IMpOYf2s55XC9nqo8hVCKM9MFGP37t0TAAQA0bdv32zTu7u7CwCiSpUqOtNMmzZNzpMPPvjggw8++Ch6j/v37yuOQYp1C5ilpaW8nJKSkm16VcuWlZWVzjSTJ0/Gxx9/LP+fnp6O+Ph4mJubo0KFCrh//z7s7e0VlJqKooSEBLi7u7N+kFasH6QP60f+EULg+fPncHV1VZxXsQ7A7Ozs5GV9lxVVXrx4AUD/5UoLCwtYWFhoPOfo6Cg3W9rb2/MDQjqxfpA+rB+kD+tH/nBwcDBKPsW6E76lpSVcXFwAAA8ePNCb9smTJ3IA5u7unudlIyIioqKrWAdgAOQR7m/evInU1FSd6a5du5ZlGyIiIqLcKPYBWLNmzQBkXF48d+6cznRHjx6Vl5s2bZrj/VhYWGDatGlZLk8SAawfpB/rB+nD+lE4SUIY417KwuvMmTNo2LAhAGDYsGFYsmRJljTp6emoWbMmIiIi4OjoiJiYGJibm+d3UYmIiKiIKPYtYA0aNIC/vz8AYMWKFTh16lSWNN9++608+v2YMWMYfBEREZEixb4FDADCw8PRtGlTvHz5Era2tpgyZQpatGiBly9fYuPGjQgJCQEAeHp64uzZsxp3TxIRERHlFAOw/7dz5068//77OkfP9fT0xO7du1G1atV8LhkREREVNcX+EqRKp06dcPHiRYwbNw6enp6wtraGo6MjfH19MWfOHISHh+cq+Lp37x4mTJgAb29v2NjYwNnZGQ0aNMD8+fORlJSUB0dCBen8+fOYNWsW2rVrB3d3d1hYWMDW1haenp4YMGAAjh8/nqP89u7di65du6J8+fKwsLBA+fLl0bVrV+zduzePjoAKyqRJkyBJkvw4cuRIttuwfhRtcXFxmDt3Lpo2bYqyZcvCwsICrq6uaNiwISZOnKi1y0xmrCNvMMVj6ZNOu3btEg4ODjqnMqhWrZq4detWQReTjKR58+YGTWHRt29fkZycrDev9PR0MXToUL35DB06VKSnp+fT0VFeunDhgjAzM9N4fw8fPqwzPetH0bdp0yZRsmRJve9xUFCQzu1ZR958DMDyyIULF4S1tbUAIGxtbcXMmTPFyZMnxcGDB8WQIUPkD4CXl5d4/vx5QReXjKBKlSoCgHB1dRVjxowRW7ZsEWfOnBGnTp0S3333nXBzc5Pf9969e+vNa8qUKXJaHx8fsWHDBnHmzBmxYcMG4ePjI6+bOnVqPh0d5ZW0tDTh5+cnAIjSpUsbFICxfhRta9asESYmJnKdmDZtmvjrr7/EuXPnxO7du8UPP/wg2rRpI959912debCOvPkYgOWRwMBAAUCYmZmJkydPZlk/d+5c+QPw5ZdfFkAJydg6dOggfvvtN5Gamqp1fWxsrPD09JTf92PHjmlNd+PGDbk1xNfXVyQlJWmsf/HihfD19ZXr182bN41+LJR/FixYIP8Ymzx5crYBGOtH0Xb16lVhYWEhAAh/f3/x9OlTnWl1taSzjhQODMDywJkzZ+ST6LBhw7SmSUtLE97e3gKAcHJyEikpKflcSioIO3fulOvG6NGjtaYZMWKEnObUqVNa05w6dUpOM3LkyLwsMuWhe/fuCVtbWzngmjZtWrYBGOtH0daqVSsBQLi4uIjY2Nhc5cE6UjiwE34e2L59u7w8cOBArWlMTEzQr18/ABnzTBrS4ZYKv8DAQHn51q1bWdYLIbBjxw4AgJeXFxo1aqQ1n0aNGqFatWoAMuqb4M3MhdKIESOQmJiI/v37a9QNXVg/irZr167h4MGDAICRI0fKcxXnBOtI4cEALA+o7nSzsbFB/fr1daYLCAiQl0NDQ/O8XFTwUlJS5GUTk6wfv8jISERFRQHQrB/aqNY/ePAAd+7cMV4hKV9s2rQJu3btgrOzM+bNm2fQNqwfRdvmzZvl5e7du8vLT548wY0bN/D48eNs82AdKTwYgOUB1aj5VatWhZmZmc50Xl5eWbahok19TlH1919FvR5oW6+O9afwevr0KcaMGQMAmDNnDkqVKmXQdqwfRdvp06cBAA4ODvD29sYvv/yCOnXqwNnZGZ6ennBxccFbb72FL7/8EomJiVrzYB0pPBiAGdmrV68QFxcHAChfvrzetE5OTrCxsQEA3L9/P8/LRgUrPT0ds2fPlv/v0aNHljTq9SC7+uPu7q51O3rzTZo0CY8ePUKTJk0wePBgg7dj/Sjarl69CgCoVKkSRo0ahffffx8XL17USBMZGYnp06ejcePGiI6OzpIH60jhwQDMyJ4/fy4v29raZpteFYDp+jVDRceCBQtw5swZAEBwcDB8fX2zpMlJ/VHVHYD1pzAJDQ3F8uXLYWZmhiVLlkCSJIO3Zf0o2uLj4wFk9AX76aef4OjoiCVLliAmJgavXr1CWFgY2rVrBwC4fPkyunfvjvT0dI08WEcKDwZgRvbq1St5uUSJEtmmt7CwAAC8fPkyz8pEBe/o0aP49NNPAQClS5fG4sWLtabLSf1R1R2A9aewSElJwdChQyGEwLhx41CrVq0cbc/6UbS9ePECAJCcnAxTU1P8+eefGDZsGEqVKgULCwv4+vpi165dchB28uRJbN26VSMP1pHCgwGYkVlaWsrL6h2udUlOTgYAWFlZ5VmZqGBduXIFwcHBSE1NhYWFBTZt2oQyZcpoTZuT+qOqOwDrT2Exa9YsREREoEKFCpg2bVqOt2f9KNrU39/u3btrvYPRxMRE46aNDRs26MyDdeTNxgDMyOzs7ORlQ5p0Vb94DLlcSYVPZGQk2rZtiydPnsDU1BQbNmzQe2dSTuqPqu4ArD+FwbVr1/DNN98AABYtWqRx+cdQrB9Fm/r7q2rl0qZGjRpwc3MDAISFhenMg3Xkzab7Fj3KFUtLS7i4uCAuLg4PHjzQm/bJkyfyB0C9MyQVDdHR0WjdujWio6MhSRJWrlyJ4OBgvduod5rNrv6od5pl/XnzLViwACkpKXjrrbeQlJSEjRs3Zklz+fJlefnQoUN49OgRAKBTp06wsbFh/Sji3N3d5ffckA70UVFRiImJ0XiedaTwYACWB7y9vXH8+HHcvHkTqampOoeiuHbtmsY2VHTExcWhTZs2uH37NoCMFg/VwLv6VK9eXV5Wrx/asP4ULqrLPbdv30bv3r2zTf/111/Ly5GRkbCxsWH9KOJq1Kght2ilpaXpTatan/n7hXWk8OAlyDzQrFkzABnNu+fOndOZTn1MqKZNm+Z5uSh/PHv2DG+//bZ8S/ns2bPx0UcfGbRt5cqV4erqCkCzfmhz7NgxAICbmxsqVaqU+wJTocH6UbQ1b95cXtY2U4Y61Y871aVIFdaRwoMBWB7o0qWLvLxq1SqtadLT07F27VoAgKOjI1q0aJEfRaM8lpSUhA4dOuD8+fMAgKlTp+KTTz4xeHtJkhAUFAQg49epamDGzE6fPi3/eg0KCsrRUAZUMFavXg2RMf+uzod6x/zDhw/Lz6u+HFk/irbOnTvD3NwcALLc3aju6NGj8qj4/v7+GutYRwqRfJ99spjw9/eXZ5o/efJklvVz586VJ0KdNm1a/heQjC45OVm0bdtWfl/HjBmTq3yuX78uzMzMBADh6+srkpKSNNYnJSUJX19fuX79+++/Rig9vQkMmYyb9aNo+/DDD+U6sGHDhizrExISRN26deU0Z86cyZKGdaRwYACWR86fPy+srKwEAGFraytmzZolTp06JQ4dOiSGDh0qf3g8PT1FQkJCQReXjKBr167y+9qyZUtx8eJFcenSJZ2P69ev68zr008/lfPy8fERGzduFGFhYWLjxo3Cx8dHXjd58uR8PELKa4YEYEKwfhRlMTExokKFCnJwNHLkSHHo0CFx9uxZsWrVKuHl5SW/vx9++KHOfFhH3nwMwPLQH3/8Iezt7eWKnvnh6ekpbty4UdDFJCPR9T7relSsWFFnXmlpaWLQoEF6tx88eLBIS0vLvwOkPGdoAMb6UbRdvXpVVK1aVe/7O2jQIJGSkqIzD9aRNx/7gOWhTp064eLFixg3bhw8PT1hbW0NR0dH+Pr6Ys6cOQgPD0fVqlULupj0BjIxMcGKFSuwe/duBAUFwdXVFSVKlICrqyuCgoKwZ88eLF++HCYm/AgXR6wfRZu3tzcuXLiAefPmoWHDhnB2dkaJEiVQvnx59OzZE4cOHcKKFSvk/mLasI68+SQhhCjoQhAREREVJwx9iYiIiPIZAzAiIiKifMYAjIiIiCifMQAjIiIiymcMwIiIiIjyGQMwIiIionzGAIyIiIgonzEAIyIiIspnDMCIiIiI8hkDMCIiIqJ8xgCMiIiIKJ8xACMywJ07dyBJEiRJwurVqwu6OG+stLQ0fP/992jQoAHs7e3l16xLly4FXbQCc+LECQQHB6Ns2bIwMzOTX5OnT58CACpVqgRJkjBgwIACLWdhN336dPm1LSru3buHYcOGoUqVKrC0tJSPb/v27QCAAQMGQJIkVKpUSev2rFtvNrOCLgARFR29e/fG5s2bC7oYb4ydO3ciODgYaWlpBV0UKmTu3buH+vXrIy4urqCLQnmEARgRGcXJkyfl4KtDhw4YO3YsypQpA0mSYG9vX8ClKxjjx49HWloaXF1dMXv2bNSoUQMlSpQAgGL7muTE6tWrMXDgQABAZGSkzpaeomjGjBmIi4uDmZkZZs6ciebNm8PW1hYAULFixQIuHRkDAzAiMooDBw4AAExNTfHrr78W+wDj3r17uHHjBgBgypQp6Nu3r9Z0d+7cycdSFV3Tp0/H9OnTC7oYRqP6PHXp0gWTJk3Smmb16tXsElGIsQ8YERlFVFQUAKBMmTLFPvgC/vd6AICnp2cBloQKI1X9Yd0puhiAEZFRJCcnAwDMzc0LuCRvBtXrAfA1oZxLSUkBwLpTlDEAoxzJfKfRq1evMG/ePNSrVw92dnaws7NDgwYN8OOPPyI1NVVnPqo8srtkEBgYCEmSEBgYmGXdkSNH5HyOHDkCIQRWrFiBZs2aoWTJkrC3t0eDBg2wbt06je1SUlKwZMkSNGrUCM7OzrCzs0PTpk2xadOmHL0WmzdvRuvWrVG6dGlYWVnBy8sLn376KZ48eWLQ9mfOnMGQIUPg6ekJW1tb2NjYwMvLCx999JF86Uqb1atXy8d9584dJCcnY+HChWjUqBFcXFwMel11uXTpEoYOHQoPDw9YW1vDzs4ONWrUwLhx43ReKlOVZc2aNQCAu3fvys8pvSvtxIkT+OCDD1CtWjXY29vD1tYWXl5e6NKlC9auXYuEhASd2+7cuRPvvvsuypcvDwsLC5QsWRKNGzfG7NmzkZiYqHO7zK9veno6QkJC0KRJEzg5OcHGxga1a9fGzJkzkZSUlGV71Z1pLVq0kJ9r0aKFxuuhftnIkDvVXr9+je+//x5+fn6ws7ODo6MjfH19sWDBAqSkpGR7l66+z5G67O4kzPy5PXToELp37w53d3eYm5tn6aN1+fJlzJgxA2+//bb8Ptja2sLDwwP9+/fH6dOnte5H9dlW9f8CgMqVK2epV0eOHDG47Cp37tzBuHHjUKNGDdjZ2cHa2hoeHh4YNmwYLl26pHfbzMcfFhaG3r17y8fm5uaGvn37IiIiQm8+uqjXPZUvv/xS45jV60l2d0Ea6sGDB5g8eTLq1asHJycnWFpaokKFCujZsycOHz6sd9u0tDSsXr0ab7/9NsqWLYsSJUrA0dERHh4eaNWqFWbNmoWrV68qKl+RJYhyYNq0aQKAACAePXok6tSpI/+f+dGpUyeRlpamNR9VmmnTpundX0BAgAAgAgICsqw7fPiwnM/+/ftFp06ddJZl9OjRQggh4uPjRfPmzXWmmzlzptZyREZGymlWrVolBg0apDOPcuXKiStXrug8ptevX4sPP/xQ5/YAhLm5uQgJCdG6/apVq+R0YWFhom7dulm2z+511WbWrFnCxMREZ5ksLCzEmjVrsmyn7zhUj5xKSkoSvXv3zjZfbcf58uVLERwcrHc7V1dXER4ernXf6q/v5cuXRcuWLXXm06BBA5GYmKixff/+/bMt96pVq+T0FStWFABE//79tZbnyZMnokGDBnrLEB4erjVvFX2fI3Xqn29t1F/3KVOmZClLxYoV5bTqn099j08//TTLfgzd9vDhwwaXXQgh1qxZIywsLHTmZ2pqKmbNmqVze/XjX7RokTAzM9Oaj7W1tTh69Kje11ob9bqn66FeT1R1Tf11V5dd3RJCiOXLlwsrKyu9+xw8eLB4/fp1lm2fP38u/P39sy1zt27dcvxaFAcMwChH1E9yTZo0ESVKlBCjR48Wf/31lzh37pz49ddfhbe3t5xmyZIlWvMxNFAwNABr2LChACD69Okjdu/eLc6dOyc2bNggqlWrJqf566+/ROfOnYWZmZn48MMPxf79+8W5c+fEihUrhKurq3wCvnz5cpZ9qQdgfn5+8hffhg0bxNmzZ8WePXtEz5495TTly5cXz54903pM/fr1k9O1a9dOrF+/Xpw5c0aEhYWJZcuWiRo1asjr//jjjyzbq5+ka9euLSRJEv369ZOPe9u2bWLPnj16X9fMfvrpJznPUqVKifnz54tTp06J0NBQMX36dGFjYyMACEmSxO7duzW2vXTpkrh06ZIICgqSgxvVc6pHTqSlpYk2bdrI5fHw8BALFiwQx48fF+fOnRO7du0SU6ZMEVWrVtVaf3r06CFvW6dOHbF27VoRFhYm9u3bJwYOHCgkSRIAhLOzs3jw4IHe17dJkybCxMRE9O/fX+P1bdy4sc4A4sGDB+LSpUti5cqVcpqVK1dqvB5PnjyR02f3Jfn222/L+TRu3Fiuc3/++afo06ePRv3PrwCsdu3aAoCoVauWWLlypThz5ow4evSo+P777+W0f/31l7CxsRE9evQQS5YsEUeOHBHnz58Xe/fuFd9++6183KrXR11iYqK4dOmSmDFjhpxm3759WeqVevCbXdl37dolv/e2trZi2rRp4vjx4+LUqVPi22+/FS4uLvL2P//8s97jb9SokZAkSdSpU0esXLlShIWFiWPHjolx48bJP2IqVKggkpOT9b7emT158kQ+NtW+PvzwQ41jVq+zSgOwFStWyPupWbOmWLRokQgNDRXnz58Xv//+u2jfvr28/uOPP86y/fjx4+X1HTt2FBs2bBAnTpwQ586dE3v37hVz5swR/v7+4t13383R61BcMACjHFE/yZmbm2v8AlV5/PixKFOmjHyi1sbYARgAsXDhwixpHj58KOzt7eXAQpIksW3btizp/vnnH/nEqWotU6cegAEQ7du31/qL8KuvvpLTTJgwIcv6LVu2yOuXLVum9Zhfvnwpt7pUqlQpy34y/0pesWKF1nwMFRMTI6ytreXg6d69e1nSnD9/Xg7C3NzcREpKSpY02X0ZGGrhwoXysQUHB4tXr15pTZeWliaioqI0ntu1a5e8batWrbR+AYaEhMhpevTokWV95td33bp1WdK8evVK1KxZUwAQJUuW1FoX1Ounts+Jir4vya1bt8p5BAUFidTU1Cxp5s+fr1He/AjAVK+vrvdGCCFiY2M1As3MkpOT5UC7YsWKWo9N/b2IjIzMddlTUlKEm5ubHHxpa/28c+eOKFeunAAyWrBiY2OzpMl8DtBWv9SDxq1bt+otsz6GnCOVBGD37t2TP/f9+/fXWoeFEHJLp4mJibh+/brGOnd3dwEg2wDr8ePHetcXV+wDRrk2atQorX1KnJ2d5b4bFy9exLNnz/K8LA0bNsSYMWOyPF+2bFkEBwcDAGJjY9GzZ0+to7LXrl0bzZo1AwAcP35c774sLCywbNkymJllHcVl6tSpqFmzJgBgxYoVGh2xAeCbb74BAAQHB+ODDz7Qmr+lpSV+/PFHABn9VdT7uWTWsmVLDBo0SG95s7Nq1Sq5L9O3334Ld3f3LGl8fHwwefJkABl3Z6lG4ja29PR0zJs3DwDg5uaGtWvXwsLCQmtaExMTuLq6ajz3008/AcjouLxq1Sp5zC11Q4YMQevWrQEAW7duxcOHD3WWp2vXrnj//fezPG9hYYGRI0cCAB4/fpxnfVyWLl0KIKNOLF26FKamplnSfPzxx6hXr16e7F8XExMTLF++XOd7AwAuLi5wdHTUub5EiRLye3337l1cuHDByKX8n23btsl3FU6dOhV169bNkqZixYpyeZKSkrBq1Sqd+VlaWuqsX6NHj5afz+5cUpC+//57JCUlwdXVFUuWLNF6PgMy+qC5ubkhPT0da9eu1Vj36NEjAIC/v7/efTk7Oxun0EUMAzDKtT59+uhcV79+fXk5MjIyz8vSq1cvnetq164tL/fs2VNnujp16gAAbt++rXdfbdu2zfLFr2JiYoL+/fsDAJ48eYLz58/L66KionDu3DkAQI8ePfTuw9vbGy4uLgCAU6dO6Uyn7z0wlGq8IUdHR3Tr1k1nOvWAUbWNsV24cEH+ohwyZIg88KQhUlNTcfToUQBAmzZttAaSKkOGDJG30RfgGlrHs6szuZGamopjx44BAN555x2UKVNGazpJknSOMZZXmjZtmuOO38nJybh37x6uXr2Ky5cv4/LlyxBCyOv/+ecfI5fyf1T1VZIkvT9YunfvDgcHB41ttGnTpg1Kly6tdZ2dnR08PDwA5E29MJYdO3YAADp16gRLS0ud6czMzNC4cWMAWc9F5cqVAwD89ttvWm9IIf0YgFGueXl56Vyn/ovn+fPneV4WfWPlqP8KNyRdduX18/PTu75Bgwby8uXLl+Xls2fPysu9e/fOckdX5odqChLVr0xt1IPL3FKV0cfHR+8t72XKlJG/dNWPy5jCw8Pl5ebNm+do29u3b8tfAg0bNtSbVn29vmMpyDp+69YtvHz5EoBmsKeNr6+v0fevj6H17sWLF/jmm29Qp04d2NjYoGLFiqhRowZq1aqFWrVqwcfHR06bl1PuqN7jSpUq6QycgIxWOVWZclsvgP/Vjfw49+XGs2fPcPPmTQAZrazZnYu2bNkCIOu5SPVj8+TJk6hcuTJGjhyJbdu2ITY2Nn8PqJBiAEa5Zm1trXOdicn/qlZ+zINnaFkMSZeenq53X/pO4AA0Wiri4+Pl5ZiYGL3b6aLvl6WTk1Ou8lSnKqOuFhZ1ZcuW1djG2NS/hFW/rg2lXqbsjkV1HJm3y6wg67j6cCbZ1blSpUoZff/6GFLv7ty5g1q1amHKlCm4ePFitq+RKtjMC8au4/rqBfC/uvGmzgFqrHPR559/jkGDBkGSJMTExOCnn35C165dUaZMGdSqVQvTpk3Df//9Z4wiF0mciogoh7IbZ0j9soo69ZPxL7/8YnArgr4vO219gnLLkPG6dB1bXlAyfpiSbSl7htS7vn37IjIyUh7Pq1evXvD29kapUqXkvmPp6elyXvlRt960Ol5Q1M9FY8eOxeDBgw3aLnOfN3Nzc6xYsQLjx4/Hhg0bcOjQIZw9exYpKSnyZebvvvsO69evR1BQkFGPoShgAEYFQpIkCCGybW168eJFPpXIcNn9olP/dal+mapkyZLysiRJcmf9gubs7IyHDx/qvdSpojr2vOpUq+r3BgDR0dGoVq2awduqlym7Y1Ff/6Z2EFYPvLNrscjuko+hrbvG+rxdu3YNoaGhAIDJkydj5syZWtMZOmixUqr3+E2o428C9XNRUlKS4nNR9erV8fXXX+Prr7/Gy5cvceLECfz6669Yu3YtEhMT0bt3b9y6dSvHrdpFHS9BUoGws7MDoP8EnJ6erndE+IISFhZm8Hr1E5t6f5f9+/cbv2C5pCpjeHg4Xr9+rTNdTEwM7t69q7GNsanfzafqgG6ot956S7409Pfff+tNe+bMGXn5TQmEM6tSpYrcOVq9/6A22a035PMGANevX89BCXW7cuWKvKzvBpnsym2slkzVe3znzh29wezr16/lfohvar0whlKlSsHNzQ1Axs0Gxmz1s7KyQuvWrbFy5Ur5rtKXL19i165dRttHUcEAjApE5cqVAeg/Ae/ZsydfhrDIqf379+scuiA9PV2eksfJyUkjoKhatSqqV68OANi4cSPu3buX94U1gGpIhqdPn+L333/XmW7FihXyiVq1jbHVqVNHvntx+fLleqcMyszMzAwBAQEAgL/++gv379/XmXb58uUAMi6lZTc9T0ExMzOTb0TYt2+fzpZXIUSW6bYyU33e/v33X50dw2NjY412d6v6NGT6+jAuWbJEbz7qd+dlHtIlJ1T1VQiBlStX6ky3ZcsW+ZyTV3X8TdG5c2cAGTevqDrZG1urVq3k5by8yaKwYgBGBUL1Rfn333/jxIkTWdY/fPgQo0ePzu9iGSQ5ORnDhg3T2sF29uzZ8nxygwYNyjJO0meffQYgYw7Nrl276r10lJycjJ9//hmvXr0yYumzGjhwoNxyNH78eK2Byz///INZs2YByBifS9tYasZgYmKCiRMnAsiYn65fv37ypMSZpaenIzo6WuO5jz76CEBGS8agQYO0brty5Uq5BbJbt25v9GWRYcOGAcioL7rq3Hfffacx3Ik2qs9bSkoKFi1alGX969evMXjwYKN1hFcNwwBA/kGS2eLFi7MdT079vbl161auyxMcHCwPHTNr1iytQ17cv38fEyZMAJDRyV59HsqiaOLEifL5afjw4dm2Ru7ZswcXL16U/4+Pj8cff/yht/VMvaVf9SOA/od9wKhADB06FD///DNSU1PRqVMnfPHFF2jWrBlSUlJw4sQJfPvtt0hNTYWHh8cbdxnS19cXO3fuRNOmTTFu3Dh4eHggJiYGa9aswcaNGwEA5cuXx+eff55l2969e2Pfvn1Ys2YNzp07h+rVq2PYsGEICAhAqVKl8OLFC9y6dQvHjx/H1q1bER8fj379+uXp8ZQqVQrz5s3DRx99hOjoaPj6+uLTTz9FkyZNkJaWhgMHDmDevHlITEyEJEkICQnRO1yFUh999BF27tyJv/76C9u2bUOtWrUwYsQI+Pr6wtraGo8ePcLp06exYcMGvPfeexoTj3fo0AHdu3fH5s2bceDAATRs2BDjx4+Ht7c3njx5go0bN8otIM7Ozvjuu+/y7DiMoWvXrmjbti3279+PHTt2wN/fH2PHjkXVqlURGxuL9evXY/369WjQoIF8WVXbZbsOHTqgYsWKuHv3Lj7//HPExcWha9eusLS0xOXLl/HDDz/gwoULaNiwYbaXbw3h4+ODmjVr4vLly1i8eDGePn2KPn36oFy5crh//z7Wr1+PLVu2oGnTplp/gKnnY2lpiVevXuHzzz+HmZkZKlWqJPdpc3Nzg5WVVbblMTc3R0hICDp16oTnz5+jWbNmmDhxIlq1agUzMzOcPHkSs2fPli9Pzp8/X6M/YlFUuXJlLFmyBAMHDkR8fDyaNm2Kvn37omPHjqhQoQJSU1Px4MEDnDlzBlu2bMGtW7ewc+dO+eahhIQEBAUFoVKlSujatSsaNmyIihUrwszMDA8fPsTOnTvlluby5cujU6dOBXm4b6YCGoGfCilDJrwVwrBpWL777juNqT3UH05OTuLo0aMGT0Wkb6oXQ6cz0XdsmSfjHjBggM6yZzcZd2pqqpg0aZIwNTXVmYfqYWNjI5KSknJ1PDk1c+bMXE3GrWKsqYiEEOLFixfi3Xffzfb1yevJuPW9vpnrRGbGmIpIiOwn4/bx8RFnz56V/9+4caPWfI4fPy5PJ5X5YWpqKhYsWJCjybizEx4eLpycnHSWu1atWiI6OjrbPCdNmqQzj5xOxr169WqjTcatj6FTP+ljyL6MMRn3xo0b5ena9D1MTEzEoUOH5O0yT8+m6+Hm5ibOnz+fy1ehaOMlSCow48aNw969e/H222/DyckJFhYWqFy5Mj766CNcuHAhxwNx5qdVq1bh119/RWBgIEqWLAkLCwt4enpi0qRJuHLlitzXSxtTU1PMmTMHV69exfjx4+Hj4wMnJyeYmprCzs4ONWrUQJ8+fbBmzRo8fPjQoF/4xjBlyhSEh4djyJAhqFKlCqysrGBjYwNvb2+MGTMG165dy/PWOBVra2ts3rwZhw4dQt++fVG5cmVYWVnBzs4OXl5e6Nq1K3799Vf5cqU6S0tLbN26FX/88Qe6du0KV1dXlChRAk5OTmjYsCG++eYbXL9+Xet0NG8iR0dHhIaGYuHChahfvz5sbW1hZ2eHunXr4ptvvsHJkyc1hoVQjeSeWbNmzXDu3Dn07dsXrq6uMDc3R7ly5dCtWzccO3YMY8eONWq569atiwsXLmD48OGoWLEizM3N4ezsjAYNGmD+/Pk4c+aMQZd/Z8+ejWXLlsHf3x/Ozs6Khl7p378/rl27hjFjxsDb2xs2NjawsrJClSpVMGTIEISHh8tTbhUXPXv2xJ07dzB79mwEBgaidOnSMDc3h7W1Nd566y106tQJ3333He7cuYMWLVrI21WsWBEXLlzAvHnz0K5dO1SrVg2Ojo4wMzODi4sLAgICMH/+fERERGjcgET/IwlRDAY9ISIqwtavXy9PR3Tz5k1UqVKlgEtERNlhCxgRUSG3YcMGABn9+d56660CLg0RGYIBGBHRGywqKkrv3YkrVqzAnj17AAD9+vXjLABEhQQvQRIRvcFWr16NSZMmoVevXggMDETFihWRnp6OW7du4bfffpOHcihTpgwuX75c5O/eIyoqGIAREb3BVq9ene2YVOXKlcPu3bvZ2ZmoEGEARkT0BouLi8OWLVuwd+9eREREIDY2Fs+fP4ejoyO8vb3RqVMnDB8+XJ5uiIgKBwZgRERERPmMnfCJiIiI8hkDMCIiIqJ8xgCMiIiIKJ8xACMiIiLKZwzAiIiIiPIZAzAiIiKifMYAjIiIiCifMQAjIiIiymcMwIiIiIjy2f8BItfsTffk3D0AAAAASUVORK5CYII=",
      "text/plain": [
       "<Figure size 640x480 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "plt.rc('font', size=20)\n",
    "plt.hist(pipeline_counts, bins=20,color='maroon', edgecolor='white')\n",
    "plt.xlabel('number of configuration files')\n",
    "plt.ylabel('number of workflows')\n",
    "plt.title('configuration files in workflows')\n",
    "plt.tight_layout()\n",
    "plt.savefig(f'./plots/config_file_counts.png', format=\"png\", bbox_inches='tight')\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Task: Proportion of if-conditions in config-files. maybe modify and search for different regular expression-patterns (?)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def count_and_collect_occurrences(directory, pattern, config=None):\n",
    "    result_list = []\n",
    "\n",
    "    for pipeline in os.listdir(directory):\n",
    "        subdirectory_path = os.path.join(directory, pipeline)\n",
    "        folder_name = os.path.basename(subdirectory_path)\n",
    "        print(f\"Processing pipeline: {folder_name}\")\n",
    "\n",
    "        subdirectory_occurrences = 0\n",
    "        lines_with_occurrences = []\n",
    "\n",
    "        for root, _, files in os.walk(subdirectory_path):\n",
    "            for filename in files:\n",
    "                file_path = os.path.join(root, filename)\n",
    "                \n",
    "                if filename.endswith('.config'):\n",
    "                    with open(file_path, 'r', encoding='utf-8', errors='ignore') as file:\n",
    "                        content = file.readlines()\n",
    "\n",
    "                        occurrences = 0\n",
    "\n",
    "                        for line_number, line in enumerate(content, start=1):\n",
    "                            if re.search(pattern, line):\n",
    "                                occurrences += 1\n",
    "                                lines_with_occurrences.append((line_number, line.strip()))\n",
    "\n",
    "                        subdirectory_occurrences += occurrences\n",
    "\n",
    "        result_list.append({\n",
    "            'subdirectory_path': folder_name,\n",
    "            'total_occurrences': subdirectory_occurrences,\n",
    "            'lines_with_occurrences': lines_with_occurrences\n",
    "        })\n",
    "\n",
    "    return result_list\n",
    "\n",
    "def write_results_to_file(result_list, output_file_path, pattern):\n",
    "    with open(output_file_path, 'w', encoding='utf-8') as output_file:\n",
    "        for result in result_list:\n",
    "            output_file.write(f'Total occurrences of \"{pattern.pattern}\" in {result[\"subdirectory_path\"]}: {result[\"total_occurrences\"]}\\n')\n",
    "            \n",
    "            for line_number, line_content in result['lines_with_occurrences']:\n",
    "                output_file.write(f'  - Line {line_number}: {line_content}\\n')\n",
    "\n",
    "def write_results_to_csv(result_list, output_csv_path, pattern):\n",
    "    header = ['pipeline', 'if_occurences']\n",
    "\n",
    "    with open(output_csv_path, 'w', newline='', encoding='utf-8') as csv_file:\n",
    "        writer = csv.writer(csv_file)\n",
    "        writer.writerow(header)\n",
    "\n",
    "        for result in result_list:\n",
    "            writer.writerow([result['subdirectory_path'], result['total_occurrences']])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We just search in the config-files for the if-conditions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "directory_path = './git_repos/'\n",
    "pattern_to_search = re.compile(r'\\bif\\s*\\(')\n",
    "\n",
    "if_in_config = './results/if_condition_in_config.txt'\n",
    "output_csv_path = './results/if_condition_in_config.csv'\n",
    "\n",
    "result_list = count_and_collect_occurrences(directory_path, pattern_to_search, config=True)\n",
    "write_results_to_file(result_list, include_file_path, pattern_to_search)\n",
    "\n",
    "print(f'Results written to: {if_in_config}')\n",
    "\n",
    "result_list = count_and_collect_occurrences(directory_path, pattern_to_search, config=True)\n",
    "write_results_to_csv(result_list, output_csv_path, pattern_to_search)\n",
    "\n",
    "print(f'Results written to: {output_csv_path}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pipeline</th>\n",
       "      <th>if_occurences</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>airrflow</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ampliseq</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>atacseq</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>bacass</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>bactmap</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89</th>\n",
       "      <td>taxprofiler</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>90</th>\n",
       "      <td>variantcatalogue</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91</th>\n",
       "      <td>vipr</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>92</th>\n",
       "      <td>viralintegration</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>93</th>\n",
       "      <td>viralrecon</td>\n",
       "      <td>62</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>94 rows Ã— 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            pipeline  if_occurences\n",
       "0           airrflow              6\n",
       "1           ampliseq              5\n",
       "2            atacseq             30\n",
       "3             bacass             11\n",
       "4            bactmap              5\n",
       "..               ...            ...\n",
       "89       taxprofiler              6\n",
       "90  variantcatalogue              6\n",
       "91              vipr              5\n",
       "92  viralintegration              6\n",
       "93        viralrecon             62\n",
       "\n",
       "[94 rows x 2 columns]"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "df_if_cond = pd.read_csv('./results/if_condition_in_config.csv')\n",
    "df_if_cond"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Pipelines without include occurences"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "def count_and_collect_occurrences(directory, pattern):\n",
    "    result_list = []\n",
    "\n",
    "    for pipeline in os.listdir(directory):\n",
    "        subdirectory_path = os.path.join(directory, pipeline)\n",
    "        folder_name = os.path.basename(subdirectory_path)\n",
    "        print(f\"Processing pipeline: {folder_name}\")\n",
    "\n",
    "        subdirectory_occurrences = 0\n",
    "        lines_with_occurrences = []\n",
    "\n",
    "        for root, _, files in os.walk(subdirectory_path):\n",
    "            for filename in files:\n",
    "                file_path = os.path.join(root, filename)\n",
    "\n",
    "                with open(file_path, 'r', encoding='utf-8', errors='ignore') as file:\n",
    "                    content = file.readlines()\n",
    "\n",
    "                    occurrences = 0\n",
    "\n",
    "                    for line_number, line in enumerate(content, start=1):\n",
    "                        if re.search(pattern, line):\n",
    "                            occurrences += 1\n",
    "                            lines_with_occurrences.append((line_number, line.strip()))\n",
    "\n",
    "                    subdirectory_occurrences += occurrences\n",
    "\n",
    "        result_list.append({\n",
    "            'subdirectory_path': folder_name,\n",
    "            'total_occurrences': subdirectory_occurrences,\n",
    "            'lines_with_occurrences': lines_with_occurrences\n",
    "        })\n",
    "\n",
    "    return result_list\n",
    "\n",
    "def write_results_to_file(result_list, output_file_path, pattern):\n",
    "    with open(output_file_path, 'w', encoding='utf-8') as output_file:\n",
    "        for result in result_list:\n",
    "            output_file.write(f'Total occurrences of \"{pattern}\" in {result[\"subdirectory_path\"]}: {result[\"total_occurrences\"]}\\n')\n",
    "            for line_number, line_content in result['lines_with_occurrences']:\n",
    "                output_file.write(f'  - Line {line_number}: {line_content}\\n')\n",
    "                \n",
    "    print(f'Results written to: {output_file_path}')\n",
    "\n",
    "def write_results_to_csv(result_list, output_csv_path):\n",
    "    header = ['pipeline', 'include_occurrences']\n",
    "\n",
    "    with open(output_csv_path, 'w', newline='', encoding='utf-8') as csv_file:\n",
    "        writer = csv.writer(csv_file)\n",
    "        writer.writerow(header)\n",
    "\n",
    "        for result in result_list:\n",
    "            writer.writerow([result['subdirectory_path'], result['total_occurrences']])\n",
    "            \n",
    "    print(f'Results written to CSV: {output_csv_path}')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Usage of include-statement in subworkflows -> can be found in directory: ./git_repos\\\"pipeline\"\\subworkflows\\local\\"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "directory_path = './git_repos/'\n",
    "pattern_to_search = re.compile(r'\\binclude\\s*{')\n",
    "output_file_path = './results/include_occurrences.txt'\n",
    "output_csv_path = './results/include_occurrences.csv'\n",
    "\n",
    "result_list = count_and_collect_occurrences(directory_path, pattern_to_search)\n",
    "write_results_to_file(result_list, output_file_path, pattern_to_search)\n",
    "write_results_to_csv(result_list, output_csv_path)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total count of includes: 4094\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "df_inc_cond = pd.read_csv('./results/include_occurrences.csv')\n",
    "\n",
    "total_include = df_inc_cond['include_occurrences'].sum()\n",
    "print(\"Total count of includes:\", total_include)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>include_occurrences</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>count</th>\n",
       "      <td>94.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mean</th>\n",
       "      <td>43.553191</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>std</th>\n",
       "      <td>48.114250</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>min</th>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25%</th>\n",
       "      <td>9.250000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50%</th>\n",
       "      <td>32.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75%</th>\n",
       "      <td>62.500000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>max</th>\n",
       "      <td>250.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       include_occurrences\n",
       "count            94.000000\n",
       "mean             43.553191\n",
       "std              48.114250\n",
       "min               0.000000\n",
       "25%               9.250000\n",
       "50%              32.000000\n",
       "75%              62.500000\n",
       "max             250.000000"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_inc_cond.describe()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pipeline</th>\n",
       "      <th>include_occurences</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>airrflow</td>\n",
       "      <td>73</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ampliseq</td>\n",
       "      <td>124</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>atacseq</td>\n",
       "      <td>86</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>bacass</td>\n",
       "      <td>34</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>bactmap</td>\n",
       "      <td>55</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>88</th>\n",
       "      <td>ssds</td>\n",
       "      <td>33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89</th>\n",
       "      <td>taxprofiler</td>\n",
       "      <td>78</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>90</th>\n",
       "      <td>variantcatalogue</td>\n",
       "      <td>45</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>92</th>\n",
       "      <td>viralintegration</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>93</th>\n",
       "      <td>viralrecon</td>\n",
       "      <td>144</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>74 rows Ã— 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            pipeline  include_occurences\n",
       "0           airrflow                  73\n",
       "1           ampliseq                 124\n",
       "2            atacseq                  86\n",
       "3             bacass                  34\n",
       "4            bactmap                  55\n",
       "..               ...                 ...\n",
       "88              ssds                  33\n",
       "89       taxprofiler                  78\n",
       "90  variantcatalogue                  45\n",
       "92  viralintegration                  23\n",
       "93        viralrecon                 144\n",
       "\n",
       "[74 rows x 2 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pipelines_with_include = df_inc_cond[df_inc_cond['include_occurrences'] != 0]\n",
    "#pd.set_option('display.max_rows', None)\n",
    "pipelines_with_include"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Rows (pipelines) where include_occurences is 0: 20\n",
      "           pipeline  include_occurences\n",
      "6           cageseq                   0\n",
      "12          coproid                   0\n",
      "16        crisprvar                   0\n",
      "18  ddamsproteomics                   0\n",
      "19      deepvariant                   0\n",
      "21     denovohybrid                   0\n",
      "23    diaproteomics                   0\n",
      "25       dualrnaseq                   0\n",
      "26            eager                   0\n",
      "28           exoseq                   0\n",
      "40           imcyto                   0\n",
      "42         kmermaid                   0\n",
      "44          lncpipe                   0\n",
      "53         mnaseseq                   0\n",
      "59      neutronstar                   0\n",
      "62             pgdb                   0\n",
      "67    proteomicslfq                   0\n",
      "83          slamseq                   0\n",
      "84        smartseq2                   0\n",
      "91             vipr                   0\n"
     ]
    }
   ],
   "source": [
    "filtered_df = df_inc_cond[df_inc_cond['include_occurrences'] == 0]\n",
    "\n",
    "print(f\"Rows (pipelines) where include_occurences is 0: {len(filtered_df)}\")\n",
    "print(filtered_df)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Parse the repos for wildcards in config or nf files\n",
    "Process wildcards documentation can be found here: https://www.nextflow.io/docs/latest/process.html\n",
    "\n",
    "- During exploration, insights gained led to skipping certain directories\n",
    "\n",
    "- observation: .yml-files contain most of wildcard occurences\n",
    "\n",
    "- to-do: check for raredesease repo"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import re\n",
    "import csv\n",
    "import pandas as pd\n",
    "\n",
    "def find_widlcards_in_processes(directory, pattern_list, output_file_name):\n",
    "    result_list = []\n",
    "\n",
    "    for pipeline in os.listdir(directory):\n",
    "        subdirectory_path = os.path.join(directory, pipeline)\n",
    "        folder_name = os.path.basename(subdirectory_path)\n",
    "\n",
    "        for root, dirs, files in os.walk(subdirectory_path):\n",
    "            for filename in files:\n",
    "                file_path = os.path.join(root, filename)\n",
    "                \n",
    "                if filename.endswith('.config') or filename.endswith('.nf'):\n",
    "                    with open(file_path, 'r', encoding='utf-8', errors='ignore') as file:\n",
    "                        lines = file.readlines()\n",
    "\n",
    "                    for line_num, line in enumerate(lines):\n",
    "                        stripped_line = line.strip()\n",
    "\n",
    "                        if line.strip().startswith('#') or line.strip().startswith('//') or line.strip().startswith('*'):\n",
    "                                continue\n",
    "\n",
    "                        for pattern in pattern_list:\n",
    "                            \n",
    "                            pattern_csv = pattern\n",
    "                            regex_pattern = re.compile(pattern)\n",
    "                            if regex_pattern.search(stripped_line):\n",
    "                                result_list.append([folder_name, pattern_csv, file_path, line_num+1, stripped_line])\n",
    "                                #print(f'Wildcard found {pattern} with content: {stripped_line}')\n",
    "\n",
    "    with open(f'./results/{output_file_name}.csv', 'w', newline='', encoding='utf-8') as csvfile:\n",
    "        writer = csv.writer(csvfile)\n",
    "        writer.writerow(['pipeline', 'wildcard_pattern', 'file_path', 'line_number', 'line_content'])\n",
    "        writer.writerows(result_list)\n",
    "\n",
    "directory_path = './git_repos'\n",
    "\n",
    "pattern_list = pattern_list = [r'(?<!`)\\*\\.', r'\\?\\.', r'\\?\\?\\.']\n",
    "# pattern_list = ['Channel.of', 'fromPath', 'Channel.from']\n",
    "output_file_name = 'wildcard_occurrences'\n",
    "\n",
    "find_widlcards_in_processes(directory_path, pattern_list, output_file_name)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pipeline</th>\n",
       "      <th>wildcard_pattern</th>\n",
       "      <th>file_path</th>\n",
       "      <th>line_number</th>\n",
       "      <th>line_content</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>airrflow</td>\n",
       "      <td>(?&lt;!`)\\*\\.</td>\n",
       "      <td>./git_repos\\airrflow\\conf\\modules.config</td>\n",
       "      <td>53</td>\n",
       "      <td>pattern: \"*.{html,json,log}\"</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>airrflow</td>\n",
       "      <td>(?&lt;!`)\\*\\.</td>\n",
       "      <td>./git_repos\\airrflow\\conf\\modules.config</td>\n",
       "      <td>59</td>\n",
       "      <td>pattern: \"*.fastp.fastq.gz\"</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>airrflow</td>\n",
       "      <td>(?&lt;!`)\\*\\.</td>\n",
       "      <td>./git_repos\\airrflow\\modules\\local\\fastqc_post...</td>\n",
       "      <td>15</td>\n",
       "      <td>tuple val(meta), path(\"*.html\"), emit: html</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>airrflow</td>\n",
       "      <td>(?&lt;!`)\\*\\.</td>\n",
       "      <td>./git_repos\\airrflow\\modules\\local\\fastqc_post...</td>\n",
       "      <td>16</td>\n",
       "      <td>tuple val(meta), path(\"*.zip\") , emit: zip</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>airrflow</td>\n",
       "      <td>(?&lt;!`)\\*\\.</td>\n",
       "      <td>./git_repos\\airrflow\\modules\\local\\parse_logs.nf</td>\n",
       "      <td>26</td>\n",
       "      <td>path \"Table*.tsv\", emit:tables</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4975</th>\n",
       "      <td>viralrecon</td>\n",
       "      <td>(?&lt;!`)\\*\\.</td>\n",
       "      <td>./git_repos\\viralrecon\\modules\\nf-core\\unicycl...</td>\n",
       "      <td>14</td>\n",
       "      <td>tuple val(meta), path('*.scaffolds.fa.gz'), em...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4976</th>\n",
       "      <td>viralrecon</td>\n",
       "      <td>(?&lt;!`)\\*\\.</td>\n",
       "      <td>./git_repos\\viralrecon\\modules\\nf-core\\unicycl...</td>\n",
       "      <td>15</td>\n",
       "      <td>tuple val(meta), path('*.assembly.gfa.gz'), em...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4977</th>\n",
       "      <td>viralrecon</td>\n",
       "      <td>(?&lt;!`)\\*\\.</td>\n",
       "      <td>./git_repos\\viralrecon\\modules\\nf-core\\unicycl...</td>\n",
       "      <td>16</td>\n",
       "      <td>tuple val(meta), path('*.log')            , em...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4978</th>\n",
       "      <td>viralrecon</td>\n",
       "      <td>(?&lt;!`)\\*\\.</td>\n",
       "      <td>./git_repos\\viralrecon\\modules\\nf-core\\vcflib\\...</td>\n",
       "      <td>15</td>\n",
       "      <td>tuple val(meta), path(\"*.gz\"), emit: vcf</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4979</th>\n",
       "      <td>viralrecon</td>\n",
       "      <td>(?&lt;!`)\\*\\.</td>\n",
       "      <td>./git_repos\\viralrecon\\workflows\\nanopore.nf</td>\n",
       "      <td>145</td>\n",
       "      <td>single_barcode_dir = file(\"${params.fastq_dir}...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4980 rows Ã— 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        pipeline wildcard_pattern  \\\n",
       "0       airrflow       (?<!`)\\*\\.   \n",
       "1       airrflow       (?<!`)\\*\\.   \n",
       "2       airrflow       (?<!`)\\*\\.   \n",
       "3       airrflow       (?<!`)\\*\\.   \n",
       "4       airrflow       (?<!`)\\*\\.   \n",
       "...          ...              ...   \n",
       "4975  viralrecon       (?<!`)\\*\\.   \n",
       "4976  viralrecon       (?<!`)\\*\\.   \n",
       "4977  viralrecon       (?<!`)\\*\\.   \n",
       "4978  viralrecon       (?<!`)\\*\\.   \n",
       "4979  viralrecon       (?<!`)\\*\\.   \n",
       "\n",
       "                                              file_path  line_number  \\\n",
       "0              ./git_repos\\airrflow\\conf\\modules.config           53   \n",
       "1              ./git_repos\\airrflow\\conf\\modules.config           59   \n",
       "2     ./git_repos\\airrflow\\modules\\local\\fastqc_post...           15   \n",
       "3     ./git_repos\\airrflow\\modules\\local\\fastqc_post...           16   \n",
       "4      ./git_repos\\airrflow\\modules\\local\\parse_logs.nf           26   \n",
       "...                                                 ...          ...   \n",
       "4975  ./git_repos\\viralrecon\\modules\\nf-core\\unicycl...           14   \n",
       "4976  ./git_repos\\viralrecon\\modules\\nf-core\\unicycl...           15   \n",
       "4977  ./git_repos\\viralrecon\\modules\\nf-core\\unicycl...           16   \n",
       "4978  ./git_repos\\viralrecon\\modules\\nf-core\\vcflib\\...           15   \n",
       "4979       ./git_repos\\viralrecon\\workflows\\nanopore.nf          145   \n",
       "\n",
       "                                           line_content  \n",
       "0                          pattern: \"*.{html,json,log}\"  \n",
       "1                           pattern: \"*.fastp.fastq.gz\"  \n",
       "2           tuple val(meta), path(\"*.html\"), emit: html  \n",
       "3            tuple val(meta), path(\"*.zip\") , emit: zip  \n",
       "4                        path \"Table*.tsv\", emit:tables  \n",
       "...                                                 ...  \n",
       "4975  tuple val(meta), path('*.scaffolds.fa.gz'), em...  \n",
       "4976  tuple val(meta), path('*.assembly.gfa.gz'), em...  \n",
       "4977  tuple val(meta), path('*.log')            , em...  \n",
       "4978           tuple val(meta), path(\"*.gz\"), emit: vcf  \n",
       "4979  single_barcode_dir = file(\"${params.fastq_dir}...  \n",
       "\n",
       "[4980 rows x 5 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "# 4800 in nf and config files\n",
    "wildcards = pd.read_csv('./results/wildcard_occurrences.csv')\n",
    "wildcards"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total wildcard occurences in pipelines: 4980\n"
     ]
    }
   ],
   "source": [
    "print(f'Total wildcard occurences in pipelines: {len(wildcards)}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>file_path</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pipeline</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>airrflow</th>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ampliseq</th>\n",
       "      <td>135</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>atacseq</th>\n",
       "      <td>127</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>bacass</th>\n",
       "      <td>63</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>bactmap</th>\n",
       "      <td>88</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>bamtofastq</th>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cageseq</th>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>callingcards</th>\n",
       "      <td>68</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chipseq</th>\n",
       "      <td>109</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>circdna</th>\n",
       "      <td>78</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>circrna</th>\n",
       "      <td>75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>clipseq</th>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>coproid</th>\n",
       "      <td>42</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>createpanelrefs</th>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>createtaxdb</th>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>crisprseq</th>\n",
       "      <td>57</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cutandrun</th>\n",
       "      <td>116</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ddamsproteomics</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>deepvariant</th>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>demultiplex</th>\n",
       "      <td>45</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>denovohybrid</th>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>detaxizer</th>\n",
       "      <td>35</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>diaproteomics</th>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>differentialabundance</th>\n",
       "      <td>64</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>dualrnaseq</th>\n",
       "      <td>43</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>eager</th>\n",
       "      <td>71</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>epitopeprediction</th>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>exoseq</th>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>fastquorum</th>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>fetchngs</th>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>funcscan</th>\n",
       "      <td>83</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>genomeannotator</th>\n",
       "      <td>41</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>genomeassembler</th>\n",
       "      <td>64</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>genomeskim</th>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gwas</th>\n",
       "      <td>150</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>hgtseq</th>\n",
       "      <td>39</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>hic</th>\n",
       "      <td>31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>hicar</th>\n",
       "      <td>133</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>hlatyping</th>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>imcyto</th>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>isoseq</th>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>kmermaid</th>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>liverctanalysis</th>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>lncpipe</th>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mag</th>\n",
       "      <td>174</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>marsseq</th>\n",
       "      <td>42</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mcmicro</th>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaboigniter</th>\n",
       "      <td>49</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metapep</th>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metatdenovo</th>\n",
       "      <td>133</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>methylseq</th>\n",
       "      <td>64</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mhcquant</th>\n",
       "      <td>67</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mnaseseq</th>\n",
       "      <td>50</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>molkart</th>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>multiplesequencealign</th>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>nanoseq</th>\n",
       "      <td>55</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>nanostring</th>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>nascent</th>\n",
       "      <td>62</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>neutronstar</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pangenome</th>\n",
       "      <td>55</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pathogensurveillance</th>\n",
       "      <td>61</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pgdb</th>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>phageannotator</th>\n",
       "      <td>63</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>phyloplace</th>\n",
       "      <td>21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pixelator</th>\n",
       "      <td>40</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>proteinfold</th>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>proteomicslfq</th>\n",
       "      <td>55</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>quantms</th>\n",
       "      <td>66</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>radseq</th>\n",
       "      <td>88</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rangeland</th>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>raredisease</th>\n",
       "      <td>145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>readsimulator</th>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>riboseq</th>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rnadnavar</th>\n",
       "      <td>240</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rnafusion</th>\n",
       "      <td>64</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rnaseq</th>\n",
       "      <td>169</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rnasplice</th>\n",
       "      <td>129</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rnavar</th>\n",
       "      <td>59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>sammyseq</th>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>sarek</th>\n",
       "      <td>244</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>scflow</th>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>scrnaseq</th>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>slamseq</th>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>smartseq2</th>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>smrnaseq</th>\n",
       "      <td>46</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>spatialtranscriptomics</th>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>spinningjenny</th>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ssds</th>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>taxprofiler</th>\n",
       "      <td>162</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>variantcatalogue</th>\n",
       "      <td>77</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>viralintegration</th>\n",
       "      <td>47</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>viralrecon</th>\n",
       "      <td>160</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                        file_path\n",
       "pipeline                         \n",
       "airrflow                       30\n",
       "ampliseq                      135\n",
       "atacseq                       127\n",
       "bacass                         63\n",
       "bactmap                        88\n",
       "bamtofastq                     23\n",
       "cageseq                        20\n",
       "callingcards                   68\n",
       "chipseq                       109\n",
       "circdna                        78\n",
       "circrna                        75\n",
       "clipseq                        19\n",
       "coproid                        42\n",
       "createpanelrefs                20\n",
       "createtaxdb                     4\n",
       "crisprseq                      57\n",
       "cutandrun                     116\n",
       "ddamsproteomics                 1\n",
       "deepvariant                     5\n",
       "demultiplex                    45\n",
       "denovohybrid                    3\n",
       "detaxizer                      35\n",
       "diaproteomics                   8\n",
       "differentialabundance          64\n",
       "dualrnaseq                     43\n",
       "eager                          71\n",
       "epitopeprediction              14\n",
       "exoseq                          4\n",
       "fastquorum                     12\n",
       "fetchngs                       13\n",
       "funcscan                       83\n",
       "genomeannotator                41\n",
       "genomeassembler                64\n",
       "genomeskim                      3\n",
       "gwas                          150\n",
       "hgtseq                         39\n",
       "hic                            31\n",
       "hicar                         133\n",
       "hlatyping                      11\n",
       "imcyto                          6\n",
       "isoseq                         32\n",
       "kmermaid                       14\n",
       "liverctanalysis                 8\n",
       "lncpipe                        16\n",
       "mag                           174\n",
       "marsseq                        42\n",
       "mcmicro                        10\n",
       "metaboigniter                  49\n",
       "metapep                        10\n",
       "metatdenovo                   133\n",
       "methylseq                      64\n",
       "mhcquant                       67\n",
       "mnaseseq                       50\n",
       "molkart                         4\n",
       "multiplesequencealign          14\n",
       "nanoseq                        55\n",
       "nanostring                      3\n",
       "nascent                        62\n",
       "neutronstar                     2\n",
       "pangenome                      55\n",
       "pathogensurveillance           61\n",
       "pgdb                           13\n",
       "phageannotator                 63\n",
       "phyloplace                     21\n",
       "pixelator                      40\n",
       "proteinfold                     7\n",
       "proteomicslfq                  55\n",
       "quantms                        66\n",
       "radseq                         88\n",
       "rangeland                      19\n",
       "raredisease                   145\n",
       "readsimulator                  25\n",
       "riboseq                        27\n",
       "rnadnavar                     240\n",
       "rnafusion                      64\n",
       "rnaseq                        169\n",
       "rnasplice                     129\n",
       "rnavar                         59\n",
       "sammyseq                       37\n",
       "sarek                         244\n",
       "scflow                         10\n",
       "scrnaseq                       28\n",
       "slamseq                         7\n",
       "smartseq2                       5\n",
       "smrnaseq                       46\n",
       "spatialtranscriptomics          7\n",
       "spinningjenny                   5\n",
       "ssds                           37\n",
       "taxprofiler                   162\n",
       "variantcatalogue               77\n",
       "viralintegration               47\n",
       "viralrecon                    160"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.set_option('display.max_rows', None)\n",
    "pipelines = wildcards.groupby('pipeline').agg({'file_path': 'count'})\n",
    "pipelines"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total pipelines with wildcards: 92\n"
     ]
    }
   ],
   "source": [
    "print(f'Total pipelines with wildcards: {len(pipelines)}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "pd.set_option('display.max_rows', False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "92 pipelines use wildcards in config or processes\n"
     ]
    }
   ],
   "source": [
    "count_pipelines = wildcards['pipeline'].nunique()\n",
    "print(f'{count_pipelines} pipelines use wildcards in config or processes.')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>wildcard_count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>count</th>\n",
       "      <td>92.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mean</th>\n",
       "      <td>54.130435</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>std</th>\n",
       "      <td>52.917148</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>min</th>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25%</th>\n",
       "      <td>13.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50%</th>\n",
       "      <td>41.500000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75%</th>\n",
       "      <td>67.250000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>max</th>\n",
       "      <td>244.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       wildcard_count\n",
       "count       92.000000\n",
       "mean        54.130435\n",
       "std         52.917148\n",
       "min          1.000000\n",
       "25%         13.000000\n",
       "50%         41.500000\n",
       "75%         67.250000\n",
       "max        244.000000"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "wildcards_pipeline = wildcards.groupby('pipeline').size().reset_index(name='wildcard_count')\n",
    "wildcards_pipeline.describe()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# If-conditions inside or outside a process"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import re\n",
    "import csv\n",
    "\n",
    "def find_if_conditions(directory, result_dir):\n",
    "    result_list = []\n",
    "\n",
    "    for pipeline in os.listdir(directory):\n",
    "        subdirectory_path = os.path.join(directory, pipeline)\n",
    "        folder_name = os.path.basename(subdirectory_path)\n",
    "\n",
    "        subdirectory_if_inside = 0\n",
    "        subdirectory_if_outside = 0\n",
    "        \n",
    "        if_patterns = [r'\\bif\\s*\\(', r'\\bif\\s*\\[', r'\\bif\\s*\\{']\n",
    "\n",
    "        for root, dirs, files in os.walk(subdirectory_path):\n",
    "            for filename in files:\n",
    "                file_path = os.path.join(root, filename)\n",
    "                \n",
    "                if filename.endswith(('.nf', '.config')):\n",
    "                    with open(file_path, 'r', encoding='utf-8', errors='ignore') as file:\n",
    "                        content = file.readlines()\n",
    "\n",
    "                        brace_stack = []\n",
    "                        process_flag = False  \n",
    "                        process_name = None\n",
    "\n",
    "                        for i, line in enumerate(content):\n",
    "                            stripped_line = line.strip()\n",
    "                            \n",
    "                            if line.strip().startswith(('#', '//', '*')):\n",
    "                                continue\n",
    "                                \n",
    "                            process_pattern = r'process\\s+(\\w+)\\s*{'\n",
    "\n",
    "                            if re.search(process_pattern, stripped_line):\n",
    "                                process_flag = True\n",
    "                                process_match = re.search(process_pattern, stripped_line)\n",
    "                                if process_match:\n",
    "                                    process_name = process_match.group(1)\n",
    "                                    print(f'Process {process_name} in filepath {file_path}')\n",
    "\n",
    "                            if '{' in stripped_line:\n",
    "                                brace_stack.append('{')\n",
    "\n",
    "                            if '}' in stripped_line:\n",
    "                                if brace_stack:\n",
    "                                    brace_stack.pop()\n",
    "                                    if not brace_stack and process_flag:\n",
    "                                        process_flag = False\n",
    "                                        \n",
    "\n",
    "                            for if_pattern in if_patterns:\n",
    "                                if re.search(if_pattern, stripped_line, re.IGNORECASE):\n",
    "                                    context = \"INSIDE\" if process_flag else \"OUTSIDE\"\n",
    "                                    if process_flag:\n",
    "                                        subdirectory_if_inside += 1\n",
    "                                    else:\n",
    "                                        subdirectory_if_outside += 1\n",
    "                                    result_list.append((folder_name, process_name, context, file_path, i + 1, stripped_line))\n",
    "\n",
    "        with open(f'./results/if_conditions_processes.csv', 'w', newline='') as csv_file:\n",
    "            csv_writer = csv.writer(csv_file)\n",
    "            csv_writer.writerow(['Pipeline', 'Process Name', 'Inside/Outside Process', 'File Path', 'Line Number', 'Line Content'])\n",
    "            csv_writer.writerows(result_list)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Process FASTQC_POSTASSEMBLY in filepath ./git_repos/airrflow\\modules\\local\\fastqc_postassembly.nf\n",
      "Process FETCH_DATABASES in filepath ./git_repos/airrflow\\modules\\local\\fetch_databases.nf\n",
      "Process GUNZIP in filepath ./git_repos/airrflow\\modules\\local\\gunzip.nf\n",
      "Process MERGE_UMI in filepath ./git_repos/airrflow\\modules\\local\\merge_UMI.nf\n",
      "Process PARSE_LOGS in filepath ./git_repos/airrflow\\modules\\local\\parse_logs.nf\n",
      "Process RENAME_FASTQ in filepath ./git_repos/airrflow\\modules\\local\\rename_fastq.nf\n",
      "Process RENAME_FILE in filepath ./git_repos/airrflow\\modules\\local\\rename_file.nf\n",
      "Process SAMPLESHEET_CHECK in filepath ./git_repos/airrflow\\modules\\local\\samplesheet_check.nf\n",
      "Process UNZIP_DB in filepath ./git_repos/airrflow\\modules\\local\\unzip_db.nf\n",
      "Process AIRRFLOW_REPORT in filepath ./git_repos/airrflow\\modules\\local\\airrflow_report\\airrflow_report.nf\n",
      "Process CHANGEO_ASSIGNGENES in filepath ./git_repos/airrflow\\modules\\local\\changeo\\changeo_assigngenes.nf\n",
      "Process CHANGEO_CONVERTDB_FASTA in filepath ./git_repos/airrflow\\modules\\local\\changeo\\changeo_convertdb_fasta.nf\n",
      "Process CHANGEO_CREATEGERMLINES in filepath ./git_repos/airrflow\\modules\\local\\changeo\\changeo_creategermlines.nf\n",
      "Process CHANGEO_MAKEDB in filepath ./git_repos/airrflow\\modules\\local\\changeo\\changeo_makedb.nf\n",
      "Process CHANGEO_PARSEDB_SELECT in filepath ./git_repos/airrflow\\modules\\local\\changeo\\changeo_parsedb_select.nf\n",
      "Process CHANGEO_PARSEDB_SPLIT in filepath ./git_repos/airrflow\\modules\\local\\changeo\\changeo_parsedb_split.nf\n",
      "Process COLLAPSE_DUPLICATES in filepath ./git_repos/airrflow\\modules\\local\\enchantr\\collapse_duplicates.nf\n",
      "Process DEFINE_CLONES in filepath ./git_repos/airrflow\\modules\\local\\enchantr\\define_clones.nf\n",
      "Process DETECT_CONTAMINATION in filepath ./git_repos/airrflow\\modules\\local\\enchantr\\detect_contamination.nf\n",
      "Process DOWSER_LINEAGES in filepath ./git_repos/airrflow\\modules\\local\\enchantr\\dowser_lineages.nf\n",
      "Process FIND_THRESHOLD in filepath ./git_repos/airrflow\\modules\\local\\enchantr\\find_threshold.nf\n",
      "Process REMOVE_CHIMERIC in filepath ./git_repos/airrflow\\modules\\local\\enchantr\\remove_chimeric.nf\n",
      "Process REPORT_FILE_SIZE in filepath ./git_repos/airrflow\\modules\\local\\enchantr\\report_file_size.nf\n",
      "Process SINGLE_CELL_QC in filepath ./git_repos/airrflow\\modules\\local\\enchantr\\single_cell_qc.nf\n",
      "Process VALIDATE_INPUT in filepath ./git_repos/airrflow\\modules\\local\\enchantr\\validate_input.nf\n",
      "Process PRESTO_ASSEMBLEPAIRS in filepath ./git_repos/airrflow\\modules\\local\\presto\\presto_assemblepairs.nf\n",
      "Process PRESTO_BUILDCONSENSUS in filepath ./git_repos/airrflow\\modules\\local\\presto\\presto_buildconsensus.nf\n",
      "Process PRESTO_CLUSTERSETS in filepath ./git_repos/airrflow\\modules\\local\\presto\\presto_clustersets.nf\n",
      "Process PRESTO_COLLAPSESEQ in filepath ./git_repos/airrflow\\modules\\local\\presto\\presto_collapseseq.nf\n",
      "Process PRESTO_FILTERSEQ in filepath ./git_repos/airrflow\\modules\\local\\presto\\presto_filterseq.nf\n",
      "Process PRESTO_FILTERSEQ_POSTASSEMBLY in filepath ./git_repos/airrflow\\modules\\local\\presto\\presto_filterseq_postassembly.nf\n",
      "Process PRESTO_MASKPRIMERS in filepath ./git_repos/airrflow\\modules\\local\\presto\\presto_maskprimers.nf\n",
      "Process PRESTO_MASKPRIMERS_POSTASSEMBLY in filepath ./git_repos/airrflow\\modules\\local\\presto\\presto_maskprimers_postassembly.nf\n",
      "Process PRESTO_PAIRSEQ in filepath ./git_repos/airrflow\\modules\\local\\presto\\presto_pairseq.nf\n",
      "Process PRESTO_PARSEHEADERS in filepath ./git_repos/airrflow\\modules\\local\\presto\\presto_parseheaders.nf\n",
      "Process PRESTO_PARSEHEADERS_METADATA in filepath ./git_repos/airrflow\\modules\\local\\presto\\presto_parseheaders_metadata.nf\n",
      "Process PRESTO_PARSEHEADERS_PRIMERS in filepath ./git_repos/airrflow\\modules\\local\\presto\\presto_parseheaders_primers.nf\n",
      "Process PRESTO_PARSE_CLUSTER in filepath ./git_repos/airrflow\\modules\\local\\presto\\presto_parse_cluster.nf\n",
      "Process PRESTO_POSTCONSENSUS_PAIRSEQ in filepath ./git_repos/airrflow\\modules\\local\\presto\\presto_postconsensus_pairseq.nf\n",
      "Process PRESTO_SPLITSEQ in filepath ./git_repos/airrflow\\modules\\local\\presto\\presto_splitseq.nf\n",
      "Process ADD_META_TO_TAB in filepath ./git_repos/airrflow\\modules\\local\\reveal\\add_meta_to_tab.nf\n",
      "Process FILTER_JUNCTION_MOD3 in filepath ./git_repos/airrflow\\modules\\local\\reveal\\filter_junction_mod3.nf\n",
      "Process FILTER_QUALITY in filepath ./git_repos/airrflow\\modules\\local\\reveal\\filter_quality.nf\n",
      "Process CUSTOM_DUMPSOFTWAREVERSIONS in filepath ./git_repos/airrflow\\modules\\nf-core\\custom\\dumpsoftwareversions\\main.nf\n",
      "Process FASTP in filepath ./git_repos/airrflow\\modules\\nf-core\\fastp\\main.nf\n",
      "Process FASTQC in filepath ./git_repos/airrflow\\modules\\nf-core\\fastqc\\main.nf\n",
      "Process MULTIQC in filepath ./git_repos/airrflow\\modules\\nf-core\\multiqc\\main.nf\n",
      "Process ASSIGNSH in filepath ./git_repos/ampliseq\\modules\\local\\assignsh.nf\n",
      "Process BARRNAP in filepath ./git_repos/ampliseq\\modules\\local\\barrnap.nf\n",
      "Process BARRNAPSUMMARY in filepath ./git_repos/ampliseq\\modules\\local\\barrnapsummary.nf\n",
      "Process COMBINE_TABLE in filepath ./git_repos/ampliseq\\modules\\local\\combine_table.nf\n",
      "Process CUTADAPT_SUMMARY in filepath ./git_repos/ampliseq\\modules\\local\\cutadapt_summary.nf\n",
      "Process CUTADAPT_SUMMARY_MERGE in filepath ./git_repos/ampliseq\\modules\\local\\cutadapt_summary_merge.nf\n",
      "Process DADA2_ADDSPECIES in filepath ./git_repos/ampliseq\\modules\\local\\dada2_addspecies.nf\n",
      "Process DADA2_DENOISING in filepath ./git_repos/ampliseq\\modules\\local\\dada2_denoising.nf\n",
      "Process DADA2_ERR in filepath ./git_repos/ampliseq\\modules\\local\\dada2_err.nf\n",
      "Process DADA2_FILTNTRIM in filepath ./git_repos/ampliseq\\modules\\local\\dada2_filtntrim.nf\n",
      "Process DADA2_MERGE in filepath ./git_repos/ampliseq\\modules\\local\\dada2_merge.nf\n",
      "Process DADA2_QUALITY in filepath ./git_repos/ampliseq\\modules\\local\\dada2_quality.nf\n",
      "Process DADA2_RMCHIMERA in filepath ./git_repos/ampliseq\\modules\\local\\dada2_rmchimera.nf\n",
      "Process DADA2_STATS in filepath ./git_repos/ampliseq\\modules\\local\\dada2_stats.nf\n",
      "Process DADA2_TAXONOMY in filepath ./git_repos/ampliseq\\modules\\local\\dada2_taxonomy.nf\n",
      "Process FILTER_CLUSTERS in filepath ./git_repos/ampliseq\\modules\\local\\filter_clusters.nf\n",
      "Process FILTER_CODONS in filepath ./git_repos/ampliseq\\modules\\local\\filter_codons.nf\n",
      "Process FILTER_LEN_ASV in filepath ./git_repos/ampliseq\\modules\\local\\filter_len_asv.nf\n",
      "Process FILTER_SSU in filepath ./git_repos/ampliseq\\modules\\local\\filter_ssu.nf\n",
      "Process FILTER_STATS in filepath ./git_repos/ampliseq\\modules\\local\\filter_stats.nf\n",
      "Process FORMAT_FASTAINPUT in filepath ./git_repos/ampliseq\\modules\\local\\format_fastainput.nf\n",
      "Process FORMAT_PPLACETAX in filepath ./git_repos/ampliseq\\modules\\local\\format_pplacetax.nf\n",
      "Process FORMAT_TAXONOMY in filepath ./git_repos/ampliseq\\modules\\local\\format_taxonomy.nf\n",
      "Process FORMAT_TAXONOMY_QIIME in filepath ./git_repos/ampliseq\\modules\\local\\format_taxonomy_qiime.nf\n",
      "Process FORMAT_TAXONOMY_SINTAX in filepath ./git_repos/ampliseq\\modules\\local\\format_taxonomy_sintax.nf\n",
      "Process FORMAT_TAXRESULTS in filepath ./git_repos/ampliseq\\modules\\local\\format_taxresults.nf\n",
      "Process FORMAT_TAXRESULTS_KRAKEN2 in filepath ./git_repos/ampliseq\\modules\\local\\format_taxresults_kraken2.nf\n",
      "Process FORMAT_TAXRESULTS_SINTAX in filepath ./git_repos/ampliseq\\modules\\local\\format_taxresults_sintax.nf\n",
      "Process ITSX_CUTASV in filepath ./git_repos/ampliseq\\modules\\local\\itsx_cutasv.nf\n",
      "Process MERGE_STATS in filepath ./git_repos/ampliseq\\modules\\local\\merge_stats.nf\n",
      "Process METADATA_ALL in filepath ./git_repos/ampliseq\\modules\\local\\metadata_all.nf\n",
      "Process METADATA_PAIRWISE in filepath ./git_repos/ampliseq\\modules\\local\\metadata_pairwise.nf\n",
      "Process NOVASEQ_ERR in filepath ./git_repos/ampliseq\\modules\\local\\novaseq_err.nf\n",
      "Process PHYLOSEQ in filepath ./git_repos/ampliseq\\modules\\local\\phyloseq.nf\n",
      "Process PHYLOSEQ_INASV in filepath ./git_repos/ampliseq\\modules\\local\\phyloseq_inasv.nf\n",
      "Process PHYLOSEQ_INTAX in filepath ./git_repos/ampliseq\\modules\\local\\phyloseq_intax.nf\n",
      "Process PICRUST in filepath ./git_repos/ampliseq\\modules\\local\\picrust.nf\n",
      "Process QIIME2_ALPHARAREFACTION in filepath ./git_repos/ampliseq\\modules\\local\\qiime2_alphararefaction.nf\n",
      "Process QIIME2_ANCOM_ASV in filepath ./git_repos/ampliseq\\modules\\local\\qiime2_ancom_asv.nf\n",
      "Process QIIME2_ANCOM_TAX in filepath ./git_repos/ampliseq\\modules\\local\\qiime2_ancom_tax.nf\n",
      "Process QIIME2_BARPLOT in filepath ./git_repos/ampliseq\\modules\\local\\qiime2_barplot.nf\n",
      "Process QIIME2_CLASSIFY in filepath ./git_repos/ampliseq\\modules\\local\\qiime2_classify.nf\n",
      "Process QIIME2_DIVERSITY_ADONIS in filepath ./git_repos/ampliseq\\modules\\local\\qiime2_diversity_adonis.nf\n",
      "Process QIIME2_DIVERSITY_ALPHA in filepath ./git_repos/ampliseq\\modules\\local\\qiime2_diversity_alpha.nf\n",
      "Process QIIME2_DIVERSITY_BETA in filepath ./git_repos/ampliseq\\modules\\local\\qiime2_diversity_beta.nf\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Process QIIME2_DIVERSITY_BETAORD in filepath ./git_repos/ampliseq\\modules\\local\\qiime2_diversity_betaord.nf\n",
      "Process QIIME2_DIVERSITY_CORE in filepath ./git_repos/ampliseq\\modules\\local\\qiime2_diversity_core.nf\n",
      "Process QIIME2_EXPORT_ABSOLUTE in filepath ./git_repos/ampliseq\\modules\\local\\qiime2_export_absolute.nf\n",
      "Process QIIME2_EXPORT_RELASV in filepath ./git_repos/ampliseq\\modules\\local\\qiime2_export_relasv.nf\n",
      "Process QIIME2_EXPORT_RELTAX in filepath ./git_repos/ampliseq\\modules\\local\\qiime2_export_reltax.nf\n",
      "Process QIIME2_EXTRACT in filepath ./git_repos/ampliseq\\modules\\local\\qiime2_extract.nf\n",
      "Process QIIME2_FEATURETABLE_GROUP in filepath ./git_repos/ampliseq\\modules\\local\\qiime2_featuretable_group.nf\n",
      "Process QIIME2_FILTERSAMPLES in filepath ./git_repos/ampliseq\\modules\\local\\qiime2_filtersamples.nf\n",
      "Process QIIME2_FILTERTAXA in filepath ./git_repos/ampliseq\\modules\\local\\qiime2_filtertaxa.nf\n",
      "Process QIIME2_INASV in filepath ./git_repos/ampliseq\\modules\\local\\qiime2_inasv.nf\n",
      "Process QIIME2_INSEQ in filepath ./git_repos/ampliseq\\modules\\local\\qiime2_inseq.nf\n",
      "Process QIIME2_INTAX in filepath ./git_repos/ampliseq\\modules\\local\\qiime2_intax.nf\n",
      "Process QIIME2_INTREE in filepath ./git_repos/ampliseq\\modules\\local\\qiime2_intree.nf\n",
      "Process QIIME2_TRAIN in filepath ./git_repos/ampliseq\\modules\\local\\qiime2_train.nf\n",
      "Process QIIME2_TREE in filepath ./git_repos/ampliseq\\modules\\local\\qiime2_tree.nf\n",
      "Process RENAME_RAW_DATA_FILES in filepath ./git_repos/ampliseq\\modules\\local\\rename_raw_data_files.nf\n",
      "Process SBDIEXPORT in filepath ./git_repos/ampliseq\\modules\\local\\sbdiexport.nf\n",
      "Process SBDIEXPORTREANNOTATE in filepath ./git_repos/ampliseq\\modules\\local\\sbdiexportreannotate.nf\n",
      "Process SUMMARY_REPORT in filepath ./git_repos/ampliseq\\modules\\local\\summary_report.nf\n",
      "Process TRUNCLEN in filepath ./git_repos/ampliseq\\modules\\local\\trunclen.nf\n",
      "Process CUSTOM_DUMPSOFTWAREVERSIONS in filepath ./git_repos/ampliseq\\modules\\nf-core\\custom\\dumpsoftwareversions\\main.nf\n",
      "Process CUTADAPT in filepath ./git_repos/ampliseq\\modules\\nf-core\\cutadapt\\main.nf\n",
      "Process EPANG_PLACE in filepath ./git_repos/ampliseq\\modules\\nf-core\\epang\\place\\main.nf\n",
      "Process EPANG_SPLIT in filepath ./git_repos/ampliseq\\modules\\nf-core\\epang\\split\\main.nf\n",
      "Process FASTQC in filepath ./git_repos/ampliseq\\modules\\nf-core\\fastqc\\main.nf\n",
      "Process GAPPA_EXAMINEASSIGN in filepath ./git_repos/ampliseq\\modules\\nf-core\\gappa\\examineassign\\main.nf\n",
      "Process GAPPA_EXAMINEGRAFT in filepath ./git_repos/ampliseq\\modules\\nf-core\\gappa\\examinegraft\\main.nf\n",
      "Process GAPPA_EXAMINEHEATTREE in filepath ./git_repos/ampliseq\\modules\\nf-core\\gappa\\examineheattree\\main.nf\n",
      "Process HMMER_ESLALIMASK in filepath ./git_repos/ampliseq\\modules\\nf-core\\hmmer\\eslalimask\\main.nf\n",
      "Process HMMER_ESLREFORMAT in filepath ./git_repos/ampliseq\\modules\\nf-core\\hmmer\\eslreformat\\main.nf\n",
      "Process HMMER_HMMALIGN in filepath ./git_repos/ampliseq\\modules\\nf-core\\hmmer\\hmmalign\\main.nf\n",
      "Process HMMER_HMMBUILD in filepath ./git_repos/ampliseq\\modules\\nf-core\\hmmer\\hmmbuild\\main.nf\n",
      "Process KRAKEN2_KRAKEN2 in filepath ./git_repos/ampliseq\\modules\\nf-core\\kraken2\\kraken2\\main.nf\n",
      "Process MAFFT in filepath ./git_repos/ampliseq\\modules\\nf-core\\mafft\\main.nf\n",
      "Process MULTIQC in filepath ./git_repos/ampliseq\\modules\\nf-core\\multiqc\\main.nf\n",
      "Process PIGZ_UNCOMPRESS in filepath ./git_repos/ampliseq\\modules\\nf-core\\pigz\\uncompress\\main.nf\n",
      "Process UNTAR in filepath ./git_repos/ampliseq\\modules\\nf-core\\untar\\main.nf\n",
      "Process VSEARCH_CLUSTER in filepath ./git_repos/ampliseq\\modules\\nf-core\\vsearch\\cluster\\main.nf\n",
      "Process VSEARCH_SINTAX in filepath ./git_repos/ampliseq\\modules\\nf-core\\vsearch\\sintax\\main.nf\n",
      "Process VSEARCH_USEARCHGLOBAL in filepath ./git_repos/ampliseq\\modules\\nf-core\\vsearch\\usearchglobal\\main.nf\n",
      "Process BAMTOOLS_FILTER in filepath ./git_repos/atacseq\\modules\\local\\bamtools_filter.nf\n",
      "Process BAM_REMOVE_ORPHANS in filepath ./git_repos/atacseq\\modules\\local\\bam_remove_orphans.nf\n",
      "Process BEDTOOLS_GENOMECOV in filepath ./git_repos/atacseq\\modules\\local\\bedtools_genomecov.nf\n",
      "Process DESEQ2_QC in filepath ./git_repos/atacseq\\modules\\local\\deseq2_qc.nf\n",
      "Process FRIP_SCORE in filepath ./git_repos/atacseq\\modules\\local\\frip_score.nf\n",
      "Process GENOME_BLACKLIST_REGIONS in filepath ./git_repos/atacseq\\modules\\local\\genome_blacklist_regions.nf\n",
      "Process GET_AUTOSOMES in filepath ./git_repos/atacseq\\modules\\local\\get_autosomes.nf\n",
      "Process GTF2BED in filepath ./git_repos/atacseq\\modules\\local\\gtf2bed.nf\n",
      "Process IGV in filepath ./git_repos/atacseq\\modules\\local\\igv.nf\n",
      "Process MACS2_CONSENSUS in filepath ./git_repos/atacseq\\modules\\local\\macs2_consensus.nf\n",
      "Process MULTIQC in filepath ./git_repos/atacseq\\modules\\local\\multiqc.nf\n",
      "Process MULTIQC_CUSTOM_PEAKS in filepath ./git_repos/atacseq\\modules\\local\\multiqc_custom_peaks.nf\n",
      "Process PLOT_HOMER_ANNOTATEPEAKS in filepath ./git_repos/atacseq\\modules\\local\\plot_homer_annotatepeaks.nf\n",
      "Process PLOT_MACS2_QC in filepath ./git_repos/atacseq\\modules\\local\\plot_macs2_qc.nf\n",
      "Process SAMPLESHEET_CHECK in filepath ./git_repos/atacseq\\modules\\local\\samplesheet_check.nf\n",
      "Process STAR_ALIGN in filepath ./git_repos/atacseq\\modules\\local\\star_align.nf\n",
      "Process STAR_GENOMEGENERATE in filepath ./git_repos/atacseq\\modules\\local\\star_genomegenerate.nf\n",
      "Process TSS_EXTRACT in filepath ./git_repos/atacseq\\modules\\local\\tss_extract.nf\n",
      "Process ATAQV_ATAQV in filepath ./git_repos/atacseq\\modules\\nf-core\\ataqv\\ataqv\\main.nf\n",
      "Process ATAQV_MKARV in filepath ./git_repos/atacseq\\modules\\nf-core\\ataqv\\mkarv\\main.nf\n",
      "Process BOWTIE2_ALIGN in filepath ./git_repos/atacseq\\modules\\nf-core\\bowtie2\\align\\main.nf\n",
      "Process BOWTIE2_BUILD in filepath ./git_repos/atacseq\\modules\\nf-core\\bowtie2\\build\\main.nf\n",
      "Process BWA_INDEX in filepath ./git_repos/atacseq\\modules\\nf-core\\bwa\\index\\main.nf\n",
      "Process BWA_MEM in filepath ./git_repos/atacseq\\modules\\nf-core\\bwa\\mem\\main.nf\n",
      "Process CHROMAP_CHROMAP in filepath ./git_repos/atacseq\\modules\\nf-core\\chromap\\chromap\\main.nf\n",
      "Process CHROMAP_INDEX in filepath ./git_repos/atacseq\\modules\\nf-core\\chromap\\index\\main.nf\n",
      "Process CUSTOM_DUMPSOFTWAREVERSIONS in filepath ./git_repos/atacseq\\modules\\nf-core\\custom\\dumpsoftwareversions\\main.nf\n",
      "Process CUSTOM_GETCHROMSIZES in filepath ./git_repos/atacseq\\modules\\nf-core\\custom\\getchromsizes\\main.nf\n",
      "Process DEEPTOOLS_COMPUTEMATRIX in filepath ./git_repos/atacseq\\modules\\nf-core\\deeptools\\computematrix\\main.nf\n",
      "Process DEEPTOOLS_PLOTFINGERPRINT in filepath ./git_repos/atacseq\\modules\\nf-core\\deeptools\\plotfingerprint\\main.nf\n",
      "Process DEEPTOOLS_PLOTHEATMAP in filepath ./git_repos/atacseq\\modules\\nf-core\\deeptools\\plotheatmap\\main.nf\n",
      "Process DEEPTOOLS_PLOTPROFILE in filepath ./git_repos/atacseq\\modules\\nf-core\\deeptools\\plotprofile\\main.nf\n",
      "Process FASTQC in filepath ./git_repos/atacseq\\modules\\nf-core\\fastqc\\main.nf\n",
      "Process GFFREAD in filepath ./git_repos/atacseq\\modules\\nf-core\\gffread\\main.nf\n",
      "Process GUNZIP in filepath ./git_repos/atacseq\\modules\\nf-core\\gunzip\\main.nf\n",
      "Process HOMER_ANNOTATEPEAKS in filepath ./git_repos/atacseq\\modules\\nf-core\\homer\\annotatepeaks\\main.nf\n",
      "Process KHMER_UNIQUEKMERS in filepath ./git_repos/atacseq\\modules\\nf-core\\khmer\\uniquekmers\\main.nf\n",
      "Process MACS2_CALLPEAK in filepath ./git_repos/atacseq\\modules\\nf-core\\macs2\\callpeak\\main.nf\n",
      "Process PICARD_COLLECTMULTIPLEMETRICS in filepath ./git_repos/atacseq\\modules\\nf-core\\picard\\collectmultiplemetrics\\main.nf\n",
      "Process PICARD_MARKDUPLICATES in filepath ./git_repos/atacseq\\modules\\nf-core\\picard\\markduplicates\\main.nf\n",
      "Process PICARD_MERGESAMFILES in filepath ./git_repos/atacseq\\modules\\nf-core\\picard\\mergesamfiles\\main.nf\n",
      "Process PRESEQ_LCEXTRAP in filepath ./git_repos/atacseq\\modules\\nf-core\\preseq\\lcextrap\\main.nf\n",
      "Process SAMTOOLS_FLAGSTAT in filepath ./git_repos/atacseq\\modules\\nf-core\\samtools\\flagstat\\main.nf\n",
      "Process SAMTOOLS_IDXSTATS in filepath ./git_repos/atacseq\\modules\\nf-core\\samtools\\idxstats\\main.nf\n",
      "Process SAMTOOLS_INDEX in filepath ./git_repos/atacseq\\modules\\nf-core\\samtools\\index\\main.nf\n",
      "Process SAMTOOLS_SORT in filepath ./git_repos/atacseq\\modules\\nf-core\\samtools\\sort\\main.nf\n",
      "Process SAMTOOLS_STATS in filepath ./git_repos/atacseq\\modules\\nf-core\\samtools\\stats\\main.nf\n",
      "Process SUBREAD_FEATURECOUNTS in filepath ./git_repos/atacseq\\modules\\nf-core\\subread\\featurecounts\\main.nf\n",
      "Process TRIMGALORE in filepath ./git_repos/atacseq\\modules\\nf-core\\trimgalore\\main.nf\n",
      "Process UCSC_BEDGRAPHTOBIGWIG in filepath ./git_repos/atacseq\\modules\\nf-core\\ucsc\\bedgraphtobigwig\\main.nf\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Process UMITOOLS_EXTRACT in filepath ./git_repos/atacseq\\modules\\nf-core\\umitools\\extract\\main.nf\n",
      "Process UNTAR in filepath ./git_repos/atacseq\\modules\\nf-core\\untar\\main.nf\n",
      "Process DFAST in filepath ./git_repos/bacass\\modules\\local\\dfast.nf\n",
      "Process KRAKEN2_DB_PREPARATION in filepath ./git_repos/bacass\\modules\\local\\kraken2_db_preparation.nf\n",
      "Process MEDAKA in filepath ./git_repos/bacass\\modules\\local\\medaka.nf\n",
      "Process NANOPOLISH in filepath ./git_repos/bacass\\modules\\local\\nanopolish.nf\n",
      "Process PYCOQC in filepath ./git_repos/bacass\\modules\\local\\pycoqc.nf\n",
      "Process UNICYCLER in filepath ./git_repos/bacass\\modules\\local\\unicycler.nf\n",
      "Process BAKTA_BAKTA in filepath ./git_repos/bacass\\modules\\nf-core\\bakta\\bakta\\main.nf\n",
      "Process BAKTA_BAKTADBDOWNLOAD in filepath ./git_repos/bacass\\modules\\nf-core\\bakta\\baktadbdownload\\main.nf\n",
      "Process CANU in filepath ./git_repos/bacass\\modules\\nf-core\\canu\\main.nf\n",
      "Process CUSTOM_DUMPSOFTWAREVERSIONS in filepath ./git_repos/bacass\\modules\\nf-core\\custom\\dumpsoftwareversions\\main.nf\n",
      "Process FASTP in filepath ./git_repos/bacass\\modules\\nf-core\\fastp\\main.nf\n",
      "Process FASTQC in filepath ./git_repos/bacass\\modules\\nf-core\\fastqc\\main.nf\n",
      "Process GUNZIP in filepath ./git_repos/bacass\\modules\\nf-core\\gunzip\\main.nf\n",
      "Process KRAKEN2_KRAKEN2 in filepath ./git_repos/bacass\\modules\\nf-core\\kraken2\\kraken2\\main.nf\n",
      "Process MINIASM in filepath ./git_repos/bacass\\modules\\nf-core\\miniasm\\main.nf\n",
      "Process MINIMAP2_ALIGN in filepath ./git_repos/bacass\\modules\\nf-core\\minimap2\\align\\main.nf\n",
      "Process MULTIQC in filepath ./git_repos/bacass\\modules\\nf-core\\multiqc\\main.nf\n",
      "Process NANOPLOT in filepath ./git_repos/bacass\\modules\\nf-core\\nanoplot\\main.nf\n",
      "Process PORECHOP_PORECHOP in filepath ./git_repos/bacass\\modules\\nf-core\\porechop\\porechop\\main.nf\n",
      "Process PROKKA in filepath ./git_repos/bacass\\modules\\nf-core\\prokka\\main.nf\n",
      "Process QUAST in filepath ./git_repos/bacass\\modules\\nf-core\\quast\\main.nf\n",
      "Process RACON in filepath ./git_repos/bacass\\modules\\nf-core\\racon\\main.nf\n",
      "Process SAMTOOLS_INDEX in filepath ./git_repos/bacass\\modules\\nf-core\\samtools\\index\\main.nf\n",
      "Process SAMTOOLS_SORT in filepath ./git_repos/bacass\\modules\\nf-core\\samtools\\sort\\main.nf\n",
      "Process UNTAR in filepath ./git_repos/bacass\\modules\\nf-core\\untar\\main.nf\n",
      "Process ALIGNPSEUDOGENOMES in filepath ./git_repos/bactmap\\modules\\local\\alignpseudogenomes.nf\n",
      "Process GET_SOFTWARE_VERSIONS in filepath ./git_repos/bactmap\\modules\\local\\get_software_versions.nf\n",
      "Process MULTIQC in filepath ./git_repos/bactmap\\modules\\local\\multiqc.nf\n",
      "Process SAMPLESHEET_CHECK in filepath ./git_repos/bactmap\\modules\\local\\samplesheet_check.nf\n",
      "Process VCF2PSEUDOGENOME in filepath ./git_repos/bactmap\\modules\\local\\vcf2pseudogenome.nf\n",
      "Process BCFTOOLS_FILTER in filepath ./git_repos/bactmap\\modules\\nf-core\\software\\bcftools\\filter\\main.nf\n",
      "Process BCFTOOLS_MPILEUP in filepath ./git_repos/bactmap\\modules\\nf-core\\software\\bcftools\\mpileup\\main.nf\n",
      "Process BWA_INDEX in filepath ./git_repos/bactmap\\modules\\nf-core\\software\\bwa\\index\\main.nf\n",
      "Process BWA_MEM in filepath ./git_repos/bactmap\\modules\\nf-core\\software\\bwa\\mem\\main.nf\n",
      "Process FASTP in filepath ./git_repos/bactmap\\modules\\nf-core\\software\\fastp\\main.nf\n",
      "Process FASTQC in filepath ./git_repos/bactmap\\modules\\nf-core\\software\\fastqc\\main.nf\n",
      "Process FASTTREE in filepath ./git_repos/bactmap\\modules\\nf-core\\software\\fasttree\\main.nf\n",
      "Process GUBBINS in filepath ./git_repos/bactmap\\modules\\nf-core\\software\\gubbins\\main.nf\n",
      "Process IQTREE in filepath ./git_repos/bactmap\\modules\\nf-core\\software\\iqtree\\main.nf\n",
      "Process MASH_SKETCH in filepath ./git_repos/bactmap\\modules\\nf-core\\software\\mash\\sketch\\main.nf\n",
      "Process MULTIQC in filepath ./git_repos/bactmap\\modules\\nf-core\\software\\multiqc\\main.nf\n",
      "Process RAPIDNJ in filepath ./git_repos/bactmap\\modules\\nf-core\\software\\rapidnj\\main.nf\n",
      "Process RASUSA in filepath ./git_repos/bactmap\\modules\\nf-core\\software\\rasusa\\main.nf\n",
      "Process RAXMLNG in filepath ./git_repos/bactmap\\modules\\nf-core\\software\\raxmlng\\main.nf\n",
      "Process SAMTOOLS_FLAGSTAT in filepath ./git_repos/bactmap\\modules\\nf-core\\software\\samtools\\flagstat\\main.nf\n",
      "Process SAMTOOLS_IDXSTATS in filepath ./git_repos/bactmap\\modules\\nf-core\\software\\samtools\\idxstats\\main.nf\n",
      "Process SAMTOOLS_INDEX in filepath ./git_repos/bactmap\\modules\\nf-core\\software\\samtools\\index\\main.nf\n",
      "Process SAMTOOLS_SORT in filepath ./git_repos/bactmap\\modules\\nf-core\\software\\samtools\\sort\\main.nf\n",
      "Process SAMTOOLS_STATS in filepath ./git_repos/bactmap\\modules\\nf-core\\software\\samtools\\stats\\main.nf\n",
      "Process SNPSITES in filepath ./git_repos/bactmap\\modules\\nf-core\\software\\snpsites\\main.nf\n",
      "Process CHECK_IF_PAIRED_END in filepath ./git_repos/bamtofastq\\modules\\local\\check_paired_end.nf\n",
      "Process CAT_FASTQ in filepath ./git_repos/bamtofastq\\modules\\nf-core\\cat\\fastq\\main.nf\n",
      "Process CUSTOM_DUMPSOFTWAREVERSIONS in filepath ./git_repos/bamtofastq\\modules\\nf-core\\custom\\dumpsoftwareversions\\main.nf\n",
      "Process FASTQC in filepath ./git_repos/bamtofastq\\modules\\nf-core\\fastqc\\main.nf\n",
      "Process MULTIQC in filepath ./git_repos/bamtofastq\\modules\\nf-core\\multiqc\\main.nf\n",
      "Process SAMTOOLS_COLLATEFASTQ in filepath ./git_repos/bamtofastq\\modules\\nf-core\\samtools\\collatefastq\\main.nf\n",
      "Process SAMTOOLS_FAIDX in filepath ./git_repos/bamtofastq\\modules\\nf-core\\samtools\\faidx\\main.nf\n",
      "Process SAMTOOLS_FLAGSTAT in filepath ./git_repos/bamtofastq\\modules\\nf-core\\samtools\\flagstat\\main.nf\n",
      "Process SAMTOOLS_IDXSTATS in filepath ./git_repos/bamtofastq\\modules\\nf-core\\samtools\\idxstats\\main.nf\n",
      "Process SAMTOOLS_INDEX in filepath ./git_repos/bamtofastq\\modules\\nf-core\\samtools\\index\\main.nf\n",
      "Process SAMTOOLS_MERGE in filepath ./git_repos/bamtofastq\\modules\\nf-core\\samtools\\merge\\main.nf\n",
      "Process SAMTOOLS_STATS in filepath ./git_repos/bamtofastq\\modules\\nf-core\\samtools\\stats\\main.nf\n",
      "Process SAMTOOLS_VIEW in filepath ./git_repos/bamtofastq\\modules\\nf-core\\samtools\\view\\main.nf\n",
      "Process get_software_versions in filepath ./git_repos/cageseq\\main.nf\n",
      "Process convert_gtf in filepath ./git_repos/cageseq\\main.nf\n",
      "Process get_chrom_sizes in filepath ./git_repos/cageseq\\main.nf\n",
      "Process fastqc in filepath ./git_repos/cageseq\\main.nf\n",
      "Process make_STAR_index in filepath ./git_repos/cageseq\\main.nf\n",
      "Process make_bowtie_index in filepath ./git_repos/cageseq\\main.nf\n",
      "Process trim_adapters in filepath ./git_repos/cageseq\\main.nf\n",
      "Process trim_5g in filepath ./git_repos/cageseq\\main.nf\n",
      "Process trim_artifacts in filepath ./git_repos/cageseq\\main.nf\n",
      "Process sortmerna in filepath ./git_repos/cageseq\\main.nf\n",
      "Process trimmed_fastqc in filepath ./git_repos/cageseq\\main.nf\n",
      "Process star in filepath ./git_repos/cageseq\\main.nf\n",
      "Process bowtie in filepath ./git_repos/cageseq\\main.nf\n",
      "Process samtools_stats in filepath ./git_repos/cageseq\\main.nf\n",
      "Process get_ctss in filepath ./git_repos/cageseq\\main.nf\n",
      "Process make_bigwig in filepath ./git_repos/cageseq\\main.nf\n",
      "Process cluster_ctss in filepath ./git_repos/cageseq\\main.nf\n",
      "Process generate_counts in filepath ./git_repos/cageseq\\main.nf\n",
      "Process generate_count_matrix in filepath ./git_repos/cageseq\\main.nf\n",
      "Process ctss_qc in filepath ./git_repos/cageseq\\main.nf\n",
      "Process multiqc in filepath ./git_repos/cageseq\\main.nf\n",
      "Process output_documentation in filepath ./git_repos/cageseq\\main.nf\n",
      "Process CONCATQC in filepath ./git_repos/callingcards\\modules\\local\\callingcardstools\\mammals\\concatQC\\main.nf\n",
      "Process COUNT_HOPS in filepath ./git_repos/callingcards\\modules\\local\\callingcardstools\\mammals\\count_hops\\main.nf\n",
      "Process CONCATQC in filepath ./git_repos/callingcards\\modules\\local\\callingcardstools\\yeast\\concatQC\\main.nf\n",
      "Process COUNT_HOPS in filepath ./git_repos/callingcards\\modules\\local\\callingcardstools\\yeast\\count_hops\\main.nf\n",
      "Process DEMULTIPLEX in filepath ./git_repos/callingcards\\modules\\local\\callingcardstools\\yeast\\demultiplex\\main.nf\n",
      "Process CONCATFASTA in filepath ./git_repos/callingcards\\modules\\local\\concatFasta\\main.nf\n",
      "Process CONCATFASTQ in filepath ./git_repos/callingcards\\modules\\local\\concatFastq\\main.nf\n",
      "Process GTF2BED in filepath ./git_repos/callingcards\\modules\\local\\gtf2bed\\main.nf\n",
      "Process BEDTOOLS_MASKFASTA in filepath ./git_repos/callingcards\\modules\\nf-core\\bedtools\\maskfasta\\main.nf\n",
      "Process BOWTIE_ALIGN in filepath ./git_repos/callingcards\\modules\\nf-core\\bowtie\\align\\main.nf\n",
      "Process BOWTIE_BUILD in filepath ./git_repos/callingcards\\modules\\nf-core\\bowtie\\build\\main.nf\n",
      "Process BOWTIE2_ALIGN in filepath ./git_repos/callingcards\\modules\\nf-core\\bowtie2\\align\\main.nf\n",
      "Process BOWTIE2_BUILD in filepath ./git_repos/callingcards\\modules\\nf-core\\bowtie2\\build\\main.nf\n",
      "Process BWA_ALN in filepath ./git_repos/callingcards\\modules\\nf-core\\bwa\\aln\\main.nf\n",
      "Process BWA_INDEX in filepath ./git_repos/callingcards\\modules\\nf-core\\bwa\\index\\main.nf\n",
      "Process BWA_SAMPE in filepath ./git_repos/callingcards\\modules\\nf-core\\bwa\\sampe\\main.nf\n",
      "Process BWA_SAMSE in filepath ./git_repos/callingcards\\modules\\nf-core\\bwa\\samse\\main.nf\n",
      "Process BWAMEM2_INDEX in filepath ./git_repos/callingcards\\modules\\nf-core\\bwamem2\\index\\main.nf\n",
      "Process BWAMEM2_MEM in filepath ./git_repos/callingcards\\modules\\nf-core\\bwamem2\\mem\\main.nf\n",
      "Process CUSTOM_DUMPSOFTWAREVERSIONS in filepath ./git_repos/callingcards\\modules\\nf-core\\custom\\dumpsoftwareversions\\main.nf\n",
      "Process FASTQC in filepath ./git_repos/callingcards\\modules\\nf-core\\fastqc\\main.nf\n",
      "Process MULTIQC in filepath ./git_repos/callingcards\\modules\\nf-core\\multiqc\\main.nf\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Process PICARD_COLLECTMULTIPLEMETRICS in filepath ./git_repos/callingcards\\modules\\nf-core\\picard\\collectmultiplemetrics\\main.nf\n",
      "Process PRESEQ_CCURVE in filepath ./git_repos/callingcards\\modules\\nf-core\\preseq\\ccurve\\main.nf\n",
      "Process RSEQC_BAMSTAT in filepath ./git_repos/callingcards\\modules\\nf-core\\rseqc\\bamstat\\main.nf\n",
      "Process RSEQC_INFEREXPERIMENT in filepath ./git_repos/callingcards\\modules\\nf-core\\rseqc\\inferexperiment\\main.nf\n",
      "Process RSEQC_INNERDISTANCE in filepath ./git_repos/callingcards\\modules\\nf-core\\rseqc\\innerdistance\\main.nf\n",
      "Process RSEQC_JUNCTIONANNOTATION in filepath ./git_repos/callingcards\\modules\\nf-core\\rseqc\\junctionannotation\\main.nf\n",
      "Process RSEQC_JUNCTIONSATURATION in filepath ./git_repos/callingcards\\modules\\nf-core\\rseqc\\junctionsaturation\\main.nf\n",
      "Process RSEQC_READDISTRIBUTION in filepath ./git_repos/callingcards\\modules\\nf-core\\rseqc\\readdistribution\\main.nf\n",
      "Process RSEQC_READDUPLICATION in filepath ./git_repos/callingcards\\modules\\nf-core\\rseqc\\readduplication\\main.nf\n",
      "Process RSEQC_TIN in filepath ./git_repos/callingcards\\modules\\nf-core\\rseqc\\tin\\main.nf\n",
      "Process SAMTOOLS_FAIDX in filepath ./git_repos/callingcards\\modules\\nf-core\\samtools\\faidx\\main.nf\n",
      "Process SAMTOOLS_FLAGSTAT in filepath ./git_repos/callingcards\\modules\\nf-core\\samtools\\flagstat\\main.nf\n",
      "Process SAMTOOLS_IDXSTATS in filepath ./git_repos/callingcards\\modules\\nf-core\\samtools\\idxstats\\main.nf\n",
      "Process SAMTOOLS_INDEX in filepath ./git_repos/callingcards\\modules\\nf-core\\samtools\\index\\main.nf\n",
      "Process SAMTOOLS_MERGE in filepath ./git_repos/callingcards\\modules\\nf-core\\samtools\\merge\\main.nf\n",
      "Process SAMTOOLS_SORT in filepath ./git_repos/callingcards\\modules\\nf-core\\samtools\\sort\\main.nf\n",
      "Process SAMTOOLS_STATS in filepath ./git_repos/callingcards\\modules\\nf-core\\samtools\\stats\\main.nf\n",
      "Process SEQKIT_SPLIT2 in filepath ./git_repos/callingcards\\modules\\nf-core\\seqkit\\split2\\main.nf\n",
      "Process SUBREAD_FEATURECOUNTS in filepath ./git_repos/callingcards\\modules\\nf-core\\subread\\featurecounts\\main.nf\n",
      "Process TRIMMOMATIC in filepath ./git_repos/callingcards\\modules\\nf-core\\trimmomatic\\main.nf\n",
      "Process UMITOOLS_EXTRACT in filepath ./git_repos/callingcards\\modules\\nf-core\\umitools\\extract\\main.nf\n",
      "Process ANNOTATE_BOOLEAN_PEAKS in filepath ./git_repos/chipseq\\modules\\local\\annotate_boolean_peaks.nf\n",
      "Process BAM_FILTER in filepath ./git_repos/chipseq\\modules\\local\\bam_filter.nf\n",
      "Process BAM_REMOVE_ORPHANS in filepath ./git_repos/chipseq\\modules\\local\\bam_remove_orphans.nf\n",
      "Process BEDTOOLS_GENOMECOV in filepath ./git_repos/chipseq\\modules\\local\\bedtools_genomecov.nf\n",
      "Process DESEQ2_QC in filepath ./git_repos/chipseq\\modules\\local\\deseq2_qc.nf\n",
      "Process FRIP_SCORE in filepath ./git_repos/chipseq\\modules\\local\\frip_score.nf\n",
      "Process GENOME_BLACKLIST_REGIONS in filepath ./git_repos/chipseq\\modules\\local\\genome_blacklist_regions.nf\n",
      "Process GTF2BED in filepath ./git_repos/chipseq\\modules\\local\\gtf2bed.nf\n",
      "Process IGV in filepath ./git_repos/chipseq\\modules\\local\\igv.nf\n",
      "Process MACS2_CONSENSUS in filepath ./git_repos/chipseq\\modules\\local\\macs2_consensus.nf\n",
      "Process MULTIQC in filepath ./git_repos/chipseq\\modules\\local\\multiqc.nf\n",
      "Process MULTIQC_CUSTOM_PEAKS in filepath ./git_repos/chipseq\\modules\\local\\multiqc_custom_peaks.nf\n",
      "Process MULTIQC_CUSTOM_PHANTOMPEAKQUALTOOLS in filepath ./git_repos/chipseq\\modules\\local\\multiqc_custom_phantompeakqualtools.nf\n",
      "Process PLOT_HOMER_ANNOTATEPEAKS in filepath ./git_repos/chipseq\\modules\\local\\plot_homer_annotatepeaks.nf\n",
      "Process PLOT_MACS2_QC in filepath ./git_repos/chipseq\\modules\\local\\plot_macs2_qc.nf\n",
      "Process SAMPLESHEET_CHECK in filepath ./git_repos/chipseq\\modules\\local\\samplesheet_check.nf\n",
      "Process STAR_ALIGN in filepath ./git_repos/chipseq\\modules\\local\\star_align.nf\n",
      "Process STAR_GENOMEGENERATE in filepath ./git_repos/chipseq\\modules\\local\\star_genomegenerate.nf\n",
      "Process BOWTIE2_ALIGN in filepath ./git_repos/chipseq\\modules\\nf-core\\modules\\bowtie2\\align\\main.nf\n",
      "Process BOWTIE2_BUILD in filepath ./git_repos/chipseq\\modules\\nf-core\\modules\\bowtie2\\build\\main.nf\n",
      "Process BWA_INDEX in filepath ./git_repos/chipseq\\modules\\nf-core\\modules\\bwa\\index\\main.nf\n",
      "Process BWA_MEM in filepath ./git_repos/chipseq\\modules\\nf-core\\modules\\bwa\\mem\\main.nf\n",
      "Process CHROMAP_CHROMAP in filepath ./git_repos/chipseq\\modules\\nf-core\\modules\\chromap\\chromap\\main.nf\n",
      "Process CHROMAP_INDEX in filepath ./git_repos/chipseq\\modules\\nf-core\\modules\\chromap\\index\\main.nf\n",
      "Process CUSTOM_DUMPSOFTWAREVERSIONS in filepath ./git_repos/chipseq\\modules\\nf-core\\modules\\custom\\dumpsoftwareversions\\main.nf\n",
      "Process CUSTOM_GETCHROMSIZES in filepath ./git_repos/chipseq\\modules\\nf-core\\modules\\custom\\getchromsizes\\main.nf\n",
      "Process DEEPTOOLS_COMPUTEMATRIX in filepath ./git_repos/chipseq\\modules\\nf-core\\modules\\deeptools\\computematrix\\main.nf\n",
      "Process DEEPTOOLS_PLOTFINGERPRINT in filepath ./git_repos/chipseq\\modules\\nf-core\\modules\\deeptools\\plotfingerprint\\main.nf\n",
      "Process DEEPTOOLS_PLOTHEATMAP in filepath ./git_repos/chipseq\\modules\\nf-core\\modules\\deeptools\\plotheatmap\\main.nf\n",
      "Process DEEPTOOLS_PLOTPROFILE in filepath ./git_repos/chipseq\\modules\\nf-core\\modules\\deeptools\\plotprofile\\main.nf\n",
      "Process FASTQC in filepath ./git_repos/chipseq\\modules\\nf-core\\modules\\fastqc\\main.nf\n",
      "Process GFFREAD in filepath ./git_repos/chipseq\\modules\\nf-core\\modules\\gffread\\main.nf\n",
      "Process GUNZIP in filepath ./git_repos/chipseq\\modules\\nf-core\\modules\\gunzip\\main.nf\n",
      "Process HOMER_ANNOTATEPEAKS in filepath ./git_repos/chipseq\\modules\\nf-core\\modules\\homer\\annotatepeaks\\main.nf\n",
      "Process KHMER_UNIQUEKMERS in filepath ./git_repos/chipseq\\modules\\nf-core\\modules\\khmer\\uniquekmers\\main.nf\n",
      "Process MACS2_CALLPEAK in filepath ./git_repos/chipseq\\modules\\nf-core\\modules\\macs2\\callpeak\\main.nf\n",
      "Process PHANTOMPEAKQUALTOOLS in filepath ./git_repos/chipseq\\modules\\nf-core\\modules\\phantompeakqualtools\\main.nf\n",
      "Process PICARD_COLLECTMULTIPLEMETRICS in filepath ./git_repos/chipseq\\modules\\nf-core\\modules\\picard\\collectmultiplemetrics\\main.nf\n",
      "Process PICARD_MARKDUPLICATES in filepath ./git_repos/chipseq\\modules\\nf-core\\modules\\picard\\markduplicates\\main.nf\n",
      "Process PICARD_MERGESAMFILES in filepath ./git_repos/chipseq\\modules\\nf-core\\modules\\picard\\mergesamfiles\\main.nf\n",
      "Process PRESEQ_LCEXTRAP in filepath ./git_repos/chipseq\\modules\\nf-core\\modules\\preseq\\lcextrap\\main.nf\n",
      "Process SAMTOOLS_FLAGSTAT in filepath ./git_repos/chipseq\\modules\\nf-core\\modules\\samtools\\flagstat\\main.nf\n",
      "Process SAMTOOLS_IDXSTATS in filepath ./git_repos/chipseq\\modules\\nf-core\\modules\\samtools\\idxstats\\main.nf\n",
      "Process SAMTOOLS_INDEX in filepath ./git_repos/chipseq\\modules\\nf-core\\modules\\samtools\\index\\main.nf\n",
      "Process SAMTOOLS_SORT in filepath ./git_repos/chipseq\\modules\\nf-core\\modules\\samtools\\sort\\main.nf\n",
      "Process SAMTOOLS_STATS in filepath ./git_repos/chipseq\\modules\\nf-core\\modules\\samtools\\stats\\main.nf\n",
      "Process SUBREAD_FEATURECOUNTS in filepath ./git_repos/chipseq\\modules\\nf-core\\modules\\subread\\featurecounts\\main.nf\n",
      "Process TRIMGALORE in filepath ./git_repos/chipseq\\modules\\nf-core\\modules\\trimgalore\\main.nf\n",
      "Process UCSC_BEDGRAPHTOBIGWIG in filepath ./git_repos/chipseq\\modules\\nf-core\\modules\\ucsc\\bedgraphtobigwig\\main.nf\n",
      "Process UNTAR in filepath ./git_repos/chipseq\\modules\\nf-core\\modules\\untar\\main.nf\n",
      "Process AMPLIFIED_INTERVALS in filepath ./git_repos/circdna\\modules\\local\\amplified_intervals.nf\n",
      "Process CIRCLEFINDER in filepath ./git_repos/circdna\\modules\\local\\circlefinder.nf\n",
      "Process COLLECT_SEEDS in filepath ./git_repos/circdna\\modules\\local\\collect_seeds.nf\n",
      "Process GETCIRCULARREADS in filepath ./git_repos/circdna\\modules\\local\\getcircularreads.nf\n",
      "Process MULTIQC in filepath ./git_repos/circdna\\modules\\local\\multiqc.nf\n",
      "Process SAMBLASTER in filepath ./git_repos/circdna\\modules\\local\\samblaster.nf\n",
      "Process SAMPLESHEET_CHECK in filepath ./git_repos/circdna\\modules\\local\\samplesheet_check.nf\n",
      "Process SUMMARISE_AA in filepath ./git_repos/circdna\\modules\\local\\summarise_aa.nf\n",
      "Process AMPLICONARCHITECT_AMPLICONARCHITECT in filepath ./git_repos/circdna\\modules\\local\\ampliconarchitect\\ampliconarchitect.nf\n",
      "Process AMPLICONCLASSIFIER_AMPLICONCLASSIFIER in filepath ./git_repos/circdna\\modules\\local\\ampliconclassifier\\ampliconclassifier.nf\n",
      "Process AMPLICONCLASSIFIER_AMPLICONSIMILARITY in filepath ./git_repos/circdna\\modules\\local\\ampliconclassifier\\ampliconsimilarity.nf\n",
      "Process AMPLICONCLASSIFIER_MAKEINPUT in filepath ./git_repos/circdna\\modules\\local\\ampliconclassifier\\makeinput.nf\n",
      "Process AMPLICONCLASSIFIER_MAKERESULTSTABLE in filepath ./git_repos/circdna\\modules\\local\\ampliconclassifier\\makeresultstable.nf\n",
      "Process PREPAREAA in filepath ./git_repos/circdna\\modules\\local\\ampliconsuite\\prepareaa.nf\n",
      "Process BEDTOOLS_SORTEDBAM2BED in filepath ./git_repos/circdna\\modules\\local\\bedtools\\sortedbam2bed.nf\n",
      "Process BEDTOOLS_SPLITBAM2BED in filepath ./git_repos/circdna\\modules\\local\\bedtools\\splitbam2bed.nf\n",
      "Process BWA_MEM in filepath ./git_repos/circdna\\modules\\local\\bwa\\mem\\main.nf\n",
      "Process CIRCEXPLORER2_PARSE in filepath ./git_repos/circdna\\modules\\local\\circexplorer2\\parse.nf\n",
      "Process CIRCLEMAP_READEXTRACTOR in filepath ./git_repos/circdna\\modules\\local\\circlemap\\readextractor.nf\n",
      "Process CIRCLEMAP_REALIGN in filepath ./git_repos/circdna\\modules\\local\\circlemap\\realign.nf\n",
      "Process CIRCLEMAP_REPEATS in filepath ./git_repos/circdna\\modules\\local\\circlemap\\repeats.nf\n",
      "Process CNVKIT_SEGMENT in filepath ./git_repos/circdna\\modules\\local\\cnvkit\\segment.nf\n",
      "Process CNVKIT_BATCH in filepath ./git_repos/circdna\\modules\\local\\cnvkit\\batch\\main.nf\n",
      "Process SAMTOOLS_FLAGSTAT in filepath ./git_repos/circdna\\modules\\local\\samtools\\flagstat\\main.nf\n",
      "Process SAMTOOLS_IDXSTATS in filepath ./git_repos/circdna\\modules\\local\\samtools\\idxstats\\main.nf\n",
      "Process SEQTK_SEQ in filepath ./git_repos/circdna\\modules\\local\\seqtk\\seq.nf\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Process UNICYCLER in filepath ./git_repos/circdna\\modules\\local\\unicycler\\main.nf\n",
      "Process BWA_INDEX in filepath ./git_repos/circdna\\modules\\nf-core\\bwa\\index\\main.nf\n",
      "Process CAT_FASTQ in filepath ./git_repos/circdna\\modules\\nf-core\\cat\\fastq\\main.nf\n",
      "Process CUSTOM_DUMPSOFTWAREVERSIONS in filepath ./git_repos/circdna\\modules\\nf-core\\custom\\dumpsoftwareversions\\main.nf\n",
      "Process FASTQC in filepath ./git_repos/circdna\\modules\\nf-core\\fastqc\\main.nf\n",
      "Process MINIMAP2_ALIGN in filepath ./git_repos/circdna\\modules\\nf-core\\minimap2\\align\\main.nf\n",
      "Process MULTIQC in filepath ./git_repos/circdna\\modules\\nf-core\\multiqc\\main.nf\n",
      "Process PICARD_MARKDUPLICATES in filepath ./git_repos/circdna\\modules\\nf-core\\picard\\markduplicates\\main.nf\n",
      "Process SAMTOOLS_FAIDX in filepath ./git_repos/circdna\\modules\\nf-core\\samtools\\faidx\\main.nf\n",
      "Process SAMTOOLS_FLAGSTAT in filepath ./git_repos/circdna\\modules\\nf-core\\samtools\\flagstat\\main.nf\n",
      "Process SAMTOOLS_IDXSTATS in filepath ./git_repos/circdna\\modules\\nf-core\\samtools\\idxstats\\main.nf\n",
      "Process SAMTOOLS_INDEX in filepath ./git_repos/circdna\\modules\\nf-core\\samtools\\index\\main.nf\n",
      "Process SAMTOOLS_SORT in filepath ./git_repos/circdna\\modules\\nf-core\\samtools\\sort\\main.nf\n",
      "Process SAMTOOLS_STATS in filepath ./git_repos/circdna\\modules\\nf-core\\samtools\\stats\\main.nf\n",
      "Process SAMTOOLS_VIEW in filepath ./git_repos/circdna\\modules\\nf-core\\samtools\\view\\main.nf\n",
      "Process TRIMGALORE in filepath ./git_repos/circdna\\modules\\nf-core\\trimgalore\\main.nf\n",
      "Process SAMPLESHEET_CHECK in filepath ./git_repos/circrna\\modules\\local\\samplesheet_check.nf\n",
      "Process ANNOTATION in filepath ./git_repos/circrna\\modules\\local\\annotation\\full_annotation\\main.nf\n",
      "Process PARENT_GENE in filepath ./git_repos/circrna\\modules\\local\\annotation\\parent_gene\\main.nf\n",
      "Process CIRCEXPLORER2_FILTER in filepath ./git_repos/circrna\\modules\\local\\circexplorer2\\filter\\main.nf\n",
      "Process CIRCEXPLORER2_REFERENCE in filepath ./git_repos/circrna\\modules\\local\\circexplorer2\\reference\\main.nf\n",
      "Process CIRCRNA_FINDER_FILTER in filepath ./git_repos/circrna\\modules\\local\\circrna_finder\\filter\\main.nf\n",
      "Process PREPARE_CLR_TEST in filepath ./git_repos/circrna\\modules\\local\\circtest\\prepare\\main.nf\n",
      "Process CIRCTEST in filepath ./git_repos/circrna\\modules\\local\\circtest\\test\\main.nf\n",
      "Process CIRIQUANT in filepath ./git_repos/circrna\\modules\\local\\ciriquant\\ciriquant\\main.nf\n",
      "Process CIRIQUANT_FILTER in filepath ./git_repos/circrna\\modules\\local\\ciriquant\\filter\\main.nf\n",
      "Process CIRIQUANT_YML in filepath ./git_repos/circrna\\modules\\local\\ciriquant\\yml\\main.nf\n",
      "Process COUNTS_COMBINED in filepath ./git_repos/circrna\\modules\\local\\count_matrix\\combined\\main.nf\n",
      "Process MERGE_TOOLS in filepath ./git_repos/circrna\\modules\\local\\count_matrix\\merge_tools\\main.nf\n",
      "Process COUNTS_SINGLE in filepath ./git_repos/circrna\\modules\\local\\count_matrix\\single\\main.nf\n",
      "Process DCC in filepath ./git_repos/circrna\\modules\\local\\dcc\\dcc\\main.nf\n",
      "Process DCC_FILTER in filepath ./git_repos/circrna\\modules\\local\\dcc\\filter\\main.nf\n",
      "Process DESEQ2_DIFFERENTIAL_EXPRESSION in filepath ./git_repos/circrna\\modules\\local\\deseq2\\differential_expression\\main.nf\n",
      "Process FASTA in filepath ./git_repos/circrna\\modules\\local\\fasta\\main.nf\n",
      "Process FIND_CIRC_ANCHORS in filepath ./git_repos/circrna\\modules\\local\\find_circ\\anchors\\main.nf\n",
      "Process FIND_CIRC_FILTER in filepath ./git_repos/circrna\\modules\\local\\find_circ\\filter\\main.nf\n",
      "Process FIND_CIRC in filepath ./git_repos/circrna\\modules\\local\\find_circ\\find_circ\\main.nf\n",
      "Process MAPSPLICE_ALIGN in filepath ./git_repos/circrna\\modules\\local\\mapsplice\\align\\main.nf\n",
      "Process MIRNA_TARGETS in filepath ./git_repos/circrna\\modules\\local\\mirna_targets\\main.nf\n",
      "Process SEGEMEHL_FILTER in filepath ./git_repos/circrna\\modules\\local\\segemehl\\filter\\main.nf\n",
      "Process SJDB in filepath ./git_repos/circrna\\modules\\local\\star\\sjdb\\main.nf\n",
      "Process STRINGTIE_PREPDE in filepath ./git_repos/circrna\\modules\\local\\stringtie\\prepde\\main.nf\n",
      "Process TARGETSCAN_DATABASE in filepath ./git_repos/circrna\\modules\\local\\targetscan\\database\\main.nf\n",
      "Process TARGETSCAN in filepath ./git_repos/circrna\\modules\\local\\targetscan\\predict\\main.nf\n",
      "Process BEDTOOLS_INTERSECT in filepath ./git_repos/circrna\\modules\\nf-core\\bedtools\\intersect\\main.nf\n",
      "Process BEDTOOLS_SORT in filepath ./git_repos/circrna\\modules\\nf-core\\bedtools\\sort\\main.nf\n",
      "Process BOWTIE_ALIGN in filepath ./git_repos/circrna\\modules\\nf-core\\bowtie\\align\\main.nf\n",
      "Process BOWTIE_BUILD in filepath ./git_repos/circrna\\modules\\nf-core\\bowtie\\build\\main.nf\n",
      "Process BOWTIE2_ALIGN in filepath ./git_repos/circrna\\modules\\nf-core\\bowtie2\\align\\main.nf\n",
      "Process BOWTIE2_BUILD in filepath ./git_repos/circrna\\modules\\nf-core\\bowtie2\\build\\main.nf\n",
      "Process BWA_INDEX in filepath ./git_repos/circrna\\modules\\nf-core\\bwa\\index\\main.nf\n",
      "Process CAT_CAT in filepath ./git_repos/circrna\\modules\\nf-core\\cat\\cat\\main.nf\n",
      "Process CAT_FASTQ in filepath ./git_repos/circrna\\modules\\nf-core\\cat\\fastq\\main.nf\n",
      "Process CIRCEXPLORER2_ANNOTATE in filepath ./git_repos/circrna\\modules\\nf-core\\circexplorer2\\annotate\\main.nf\n",
      "Process CIRCEXPLORER2_PARSE in filepath ./git_repos/circrna\\modules\\nf-core\\circexplorer2\\parse\\main.nf\n",
      "Process CSVTK_SPLIT in filepath ./git_repos/circrna\\modules\\nf-core\\csvtk\\split\\main.nf\n",
      "Process CUSTOM_DUMPSOFTWAREVERSIONS in filepath ./git_repos/circrna\\modules\\nf-core\\custom\\dumpsoftwareversions\\main.nf\n",
      "Process FASTQC in filepath ./git_repos/circrna\\modules\\nf-core\\fastqc\\main.nf\n",
      "Process GAWK in filepath ./git_repos/circrna\\modules\\nf-core\\gawk\\main.nf\n",
      "Process HISAT2_ALIGN in filepath ./git_repos/circrna\\modules\\nf-core\\hisat2\\align\\main.nf\n",
      "Process HISAT2_BUILD in filepath ./git_repos/circrna\\modules\\nf-core\\hisat2\\build\\main.nf\n",
      "Process HISAT2_EXTRACTSPLICESITES in filepath ./git_repos/circrna\\modules\\nf-core\\hisat2\\extractsplicesites\\main.nf\n",
      "Process MIRANDA in filepath ./git_repos/circrna\\modules\\nf-core\\miranda\\main.nf\n",
      "Process MULTIQC in filepath ./git_repos/circrna\\modules\\nf-core\\multiqc\\main.nf\n",
      "Process SAMTOOLS_FLAGSTAT in filepath ./git_repos/circrna\\modules\\nf-core\\samtools\\flagstat\\main.nf\n",
      "Process SAMTOOLS_IDXSTATS in filepath ./git_repos/circrna\\modules\\nf-core\\samtools\\idxstats\\main.nf\n",
      "Process SAMTOOLS_INDEX in filepath ./git_repos/circrna\\modules\\nf-core\\samtools\\index\\main.nf\n",
      "Process SAMTOOLS_SORT in filepath ./git_repos/circrna\\modules\\nf-core\\samtools\\sort\\main.nf\n",
      "Process SAMTOOLS_STATS in filepath ./git_repos/circrna\\modules\\nf-core\\samtools\\stats\\main.nf\n",
      "Process SAMTOOLS_VIEW in filepath ./git_repos/circrna\\modules\\nf-core\\samtools\\view\\main.nf\n",
      "Process SEGEMEHL_ALIGN in filepath ./git_repos/circrna\\modules\\nf-core\\segemehl\\align\\main.nf\n",
      "Process SEGEMEHL_INDEX in filepath ./git_repos/circrna\\modules\\nf-core\\segemehl\\index\\main.nf\n",
      "Process STAR_ALIGN in filepath ./git_repos/circrna\\modules\\nf-core\\star\\align\\main.nf\n",
      "Process STAR_GENOMEGENERATE in filepath ./git_repos/circrna\\modules\\nf-core\\star\\genomegenerate\\main.nf\n",
      "Process STRINGTIE_STRINGTIE in filepath ./git_repos/circrna\\modules\\nf-core\\stringtie\\stringtie\\main.nf\n",
      "Process TRIMGALORE in filepath ./git_repos/circrna\\modules\\nf-core\\trimgalore\\main.nf\n",
      "Process get_software_versions in filepath ./git_repos/clipseq\\main.nf\n",
      "Process generate_premap_index in filepath ./git_repos/clipseq\\main.nf\n",
      "Process decompress_fasta in filepath ./git_repos/clipseq\\main.nf\n",
      "Process generate_fai in filepath ./git_repos/clipseq\\main.nf\n",
      "Process decompress_gtf in filepath ./git_repos/clipseq\\main.nf\n",
      "Process generate_star_index in filepath ./git_repos/clipseq\\main.nf\n",
      "Process generate_star_index_no_gtf in filepath ./git_repos/clipseq\\main.nf\n",
      "Process icount_segment in filepath ./git_repos/clipseq\\main.nf\n",
      "Process fastqc in filepath ./git_repos/clipseq\\main.nf\n",
      "Process move_umi in filepath ./git_repos/clipseq\\main.nf\n",
      "Process cutadapt in filepath ./git_repos/clipseq\\main.nf\n",
      "Process premap in filepath ./git_repos/clipseq\\main.nf\n",
      "Process align in filepath ./git_repos/clipseq\\main.nf\n",
      "Process preseq in filepath ./git_repos/clipseq\\main.nf\n",
      "Process dedup in filepath ./git_repos/clipseq\\main.nf\n",
      "Process rseqc in filepath ./git_repos/clipseq\\main.nf\n",
      "Process get_crosslinks in filepath ./git_repos/clipseq\\main.nf\n",
      "Process icount_peak_call in filepath ./git_repos/clipseq\\main.nf\n",
      "Process icount_motif_dreme in filepath ./git_repos/clipseq\\main.nf\n",
      "Process paraclu_peak_call in filepath ./git_repos/clipseq\\main.nf\n",
      "Process paraclu_motif_dreme in filepath ./git_repos/clipseq\\main.nf\n",
      "Process pureclip_peak_call in filepath ./git_repos/clipseq\\main.nf\n",
      "Process pureclip_motif_dreme in filepath ./git_repos/clipseq\\main.nf\n",
      "Process piranha_peak_call in filepath ./git_repos/clipseq\\main.nf\n",
      "Process piranha_motif_dreme in filepath ./git_repos/clipseq\\main.nf\n",
      "Process clipqc in filepath ./git_repos/clipseq\\main.nf\n",
      "Process multiqc in filepath ./git_repos/clipseq\\main.nf\n",
      "Process output_documentation in filepath ./git_repos/clipseq\\main.nf\n",
      "Process decomp_kraken in filepath ./git_repos/coproid\\main.nf\n",
      "Process fastqc in filepath ./git_repos/coproid\\main.nf\n",
      "Process renameGenome1 in filepath ./git_repos/coproid\\main.nf\n",
      "Process renameGenome2 in filepath ./git_repos/coproid\\main.nf\n",
      "Process renameGenome3 in filepath ./git_repos/coproid\\main.nf\n",
      "Process AdapterRemovalCollapse in filepath ./git_repos/coproid\\main.nf\n",
      "Process AdapterRemovalNoCollapse in filepath ./git_repos/coproid\\main.nf\n",
      "Process BowtieIndexGenome1 in filepath ./git_repos/coproid\\main.nf\n",
      "Process AlignToGenome1 in filepath ./git_repos/coproid\\main.nf\n",
      "Process bam2fq in filepath ./git_repos/coproid\\main.nf\n",
      "Process BowtieIndexGenome2 in filepath ./git_repos/coproid\\main.nf\n",
      "Process BowtieIndexGenome3 in filepath ./git_repos/coproid\\main.nf\n",
      "Process AlignToGenome2 in filepath ./git_repos/coproid\\main.nf\n",
      "Process AlignToGenome3 in filepath ./git_repos/coproid\\main.nf\n",
      "Process pmdtoolsgenome1 in filepath ./git_repos/coproid\\main.nf\n",
      "Process pmdtoolsgenome2 in filepath ./git_repos/coproid\\main.nf\n",
      "Process pmdtoolsgenome3 in filepath ./git_repos/coproid\\main.nf\n",
      "Process kraken2 in filepath ./git_repos/coproid\\main.nf\n",
      "Process kraken_parse in filepath ./git_repos/coproid\\main.nf\n",
      "Process kraken_merge in filepath ./git_repos/coproid\\main.nf\n",
      "Process sourcepredict in filepath ./git_repos/coproid\\main.nf\n",
      "Process countBp2genomes in filepath ./git_repos/coproid\\main.nf\n",
      "Process countBp3genomes in filepath ./git_repos/coproid\\main.nf\n",
      "Process damageprofilerGenome1 in filepath ./git_repos/coproid\\main.nf\n",
      "Process damageprofilerGenome2 in filepath ./git_repos/coproid\\main.nf\n",
      "Process damageprofilerGenome3 in filepath ./git_repos/coproid\\main.nf\n",
      "Process concatenateRatios in filepath ./git_repos/coproid\\main.nf\n",
      "Process generate_report_adna_3_genomes in filepath ./git_repos/coproid\\main.nf\n",
      "Process generate_report_adna_2_genomes in filepath ./git_repos/coproid\\main.nf\n",
      "Process generate_report in filepath ./git_repos/coproid\\main.nf\n",
      "Process get_software_versions in filepath ./git_repos/coproid\\main.nf\n",
      "Process multiqc in filepath ./git_repos/coproid\\main.nf\n",
      "Process output_documentation in filepath ./git_repos/coproid\\main.nf\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Process CNVKIT_BATCH in filepath ./git_repos/createpanelrefs\\modules\\nf-core\\cnvkit\\batch\\main.nf\n",
      "Process CUSTOM_DUMPSOFTWAREVERSIONS in filepath ./git_repos/createpanelrefs\\modules\\nf-core\\custom\\dumpsoftwareversions\\main.nf\n",
      "Process GATK4_ANNOTATEINTERVALS in filepath ./git_repos/createpanelrefs\\modules\\nf-core\\gatk4\\annotateintervals\\main.nf\n",
      "Process GATK4_COLLECTREADCOUNTS in filepath ./git_repos/createpanelrefs\\modules\\nf-core\\gatk4\\collectreadcounts\\main.nf\n",
      "Process GATK4_DETERMINEGERMLINECONTIGPLOIDY in filepath ./git_repos/createpanelrefs\\modules\\nf-core\\gatk4\\determinegermlinecontigploidy\\main.nf\n",
      "Process GATK4_FILTERINTERVALS in filepath ./git_repos/createpanelrefs\\modules\\nf-core\\gatk4\\filterintervals\\main.nf\n",
      "Process GATK4_GERMLINECNVCALLER in filepath ./git_repos/createpanelrefs\\modules\\nf-core\\gatk4\\germlinecnvcaller\\main.nf\n",
      "Process GATK4_INTERVALLISTTOOLS in filepath ./git_repos/createpanelrefs\\modules\\nf-core\\gatk4\\intervallisttools\\main.nf\n",
      "Process GATK4_PREPROCESSINTERVALS in filepath ./git_repos/createpanelrefs\\modules\\nf-core\\gatk4\\preprocessintervals\\main.nf\n",
      "Process MULTIQC in filepath ./git_repos/createpanelrefs\\modules\\nf-core\\multiqc\\main.nf\n",
      "Process PICARD_CREATESEQUENCEDICTIONARY in filepath ./git_repos/createpanelrefs\\modules\\nf-core\\picard\\createsequencedictionary\\main.nf\n",
      "Process SAMTOOLS_FAIDX in filepath ./git_repos/createpanelrefs\\modules\\nf-core\\samtools\\faidx\\main.nf\n",
      "Process SAMTOOLS_INDEX in filepath ./git_repos/createpanelrefs\\modules\\nf-core\\samtools\\index\\main.nf\n",
      "Process SAMPLESHEET_CHECK in filepath ./git_repos/createtaxdb\\modules\\local\\samplesheet_check.nf\n",
      "Process CAT_CAT in filepath ./git_repos/createtaxdb\\modules\\nf-core\\cat\\cat\\main.nf\n",
      "Process CUSTOM_DUMPSOFTWAREVERSIONS in filepath ./git_repos/createtaxdb\\modules\\nf-core\\custom\\dumpsoftwareversions\\main.nf\n",
      "Process DIAMOND_MAKEDB in filepath ./git_repos/createtaxdb\\modules\\nf-core\\diamond\\makedb\\main.nf\n",
      "Process KAIJU_MKFMI in filepath ./git_repos/createtaxdb\\modules\\nf-core\\kaiju\\mkfmi\\main.nf\n",
      "Process MALT_BUILD in filepath ./git_repos/createtaxdb\\modules\\nf-core\\malt\\build\\main.nf\n",
      "Process MULTIQC in filepath ./git_repos/createtaxdb\\modules\\nf-core\\multiqc\\main.nf\n",
      "Process ALIGNMENT_SUMMARY in filepath ./git_repos/crisprseq\\modules\\local\\alignment_summary.nf\n",
      "Process CIGAR_PARSER in filepath ./git_repos/crisprseq\\modules\\local\\cigar_parser.nf\n",
      "Process CLUSTERING_SUMMARY in filepath ./git_repos/crisprseq\\modules\\local\\clustering_summary.nf\n",
      "Process CRISPRSEQ_PLOTTER in filepath ./git_repos/crisprseq\\modules\\local\\crisprseq_plotter.nf\n",
      "Process EXTRACT_UMIS in filepath ./git_repos/crisprseq\\modules\\local\\extract_umis.nf\n",
      "Process FIND_ADAPTERS in filepath ./git_repos/crisprseq\\modules\\local\\find_adapters.nf\n",
      "Process ORIENT_REFERENCE in filepath ./git_repos/crisprseq\\modules\\local\\orient_reference.nf\n",
      "Process PREPROCESSING_SUMMARY in filepath ./git_repos/crisprseq\\modules\\local\\preprocessing_summary.nf\n",
      "Process TEMPLATE_REFERENCE in filepath ./git_repos/crisprseq\\modules\\local\\template_reference.nf\n",
      "Process BAGEL2_BF in filepath ./git_repos/crisprseq\\modules\\local\\bagel2\\bf.nf\n",
      "Process BAGEL2_FC in filepath ./git_repos/crisprseq\\modules\\local\\bagel2\\fc.nf\n",
      "Process BAGEL2_GRAPH in filepath ./git_repos/crisprseq\\modules\\local\\bagel2\\graph.nf\n",
      "Process BAGEL2_PR in filepath ./git_repos/crisprseq\\modules\\local\\bagel2\\pr.nf\n",
      "Process MAGECK_GRAPHRRA in filepath ./git_repos/crisprseq\\modules\\local\\mageck\\graphrra.nf\n",
      "Process BOWTIE2_ALIGN in filepath ./git_repos/crisprseq\\modules\\nf-core\\bowtie2\\align\\main.nf\n",
      "Process BOWTIE2_BUILD in filepath ./git_repos/crisprseq\\modules\\nf-core\\bowtie2\\build\\main.nf\n",
      "Process BWA_INDEX in filepath ./git_repos/crisprseq\\modules\\nf-core\\bwa\\index\\main.nf\n",
      "Process BWA_MEM in filepath ./git_repos/crisprseq\\modules\\nf-core\\bwa\\mem\\main.nf\n",
      "Process CAT_FASTQ in filepath ./git_repos/crisprseq\\modules\\nf-core\\cat\\fastq\\main.nf\n",
      "Process CRISPRCLEANR_NORMALIZE in filepath ./git_repos/crisprseq\\modules\\nf-core\\crisprcleanr\\normalize\\main.nf\n",
      "Process CUSTOM_DUMPSOFTWAREVERSIONS in filepath ./git_repos/crisprseq\\modules\\nf-core\\custom\\dumpsoftwareversions\\main.nf\n",
      "Process CUTADAPT in filepath ./git_repos/crisprseq\\modules\\nf-core\\cutadapt\\main.nf\n",
      "Process FASTQC in filepath ./git_repos/crisprseq\\modules\\nf-core\\fastqc\\main.nf\n",
      "Process MAGECK_COUNT in filepath ./git_repos/crisprseq\\modules\\nf-core\\mageck\\count\\main.nf\n",
      "Process MAGECK_MLE in filepath ./git_repos/crisprseq\\modules\\nf-core\\mageck\\mle\\main.nf\n",
      "Process MAGECK_TEST in filepath ./git_repos/crisprseq\\modules\\nf-core\\mageck\\test\\main.nf\n",
      "Process MEDAKA in filepath ./git_repos/crisprseq\\modules\\nf-core\\medaka\\main.nf\n",
      "Process MINIMAP2_ALIGN in filepath ./git_repos/crisprseq\\modules\\nf-core\\minimap2\\align\\main.nf\n",
      "Process MINIMAP2_INDEX in filepath ./git_repos/crisprseq\\modules\\nf-core\\minimap2\\index\\main.nf\n",
      "Process MULTIQC in filepath ./git_repos/crisprseq\\modules\\nf-core\\multiqc\\main.nf\n",
      "Process PEAR in filepath ./git_repos/crisprseq\\modules\\nf-core\\pear\\main.nf\n",
      "Process RACON in filepath ./git_repos/crisprseq\\modules\\nf-core\\racon\\main.nf\n",
      "Process SAMTOOLS_INDEX in filepath ./git_repos/crisprseq\\modules\\nf-core\\samtools\\index\\main.nf\n",
      "Process SEQTK_SEQ in filepath ./git_repos/crisprseq\\modules\\nf-core\\seqtk\\seq\\main.nf\n",
      "Process VSEARCH_CLUSTER in filepath ./git_repos/crisprseq\\modules\\nf-core\\vsearch\\cluster\\main.nf\n",
      "Process VSEARCH_SORT in filepath ./git_repos/crisprseq\\modules\\nf-core\\vsearch\\sort\\main.nf\n",
      "Process clean_samplesheet in filepath ./git_repos/crisprvar\\main.nf\n",
      "Process get_software_versions in filepath ./git_repos/crisprvar\\main.nf\n",
      "Process fastqc in filepath ./git_repos/crisprvar\\main.nf\n",
      "Process trim_galore in filepath ./git_repos/crisprvar\\main.nf\n",
      "Process crispresso in filepath ./git_repos/crisprvar\\main.nf\n",
      "Process multiqc in filepath ./git_repos/crisprvar\\main.nf\n",
      "Process CUSTOM_DUMPSOFTWAREVERSIONS in filepath ./git_repos/cutandrun\\modules\\local\\custom_dumpsoftwareversions.nf\n",
      "Process GTF2BED in filepath ./git_repos/cutandrun\\modules\\local\\gtf2bed.nf\n",
      "Process MULTIQC in filepath ./git_repos/cutandrun\\modules\\local\\multiqc.nf\n",
      "Process PEAK_COUNTS in filepath ./git_repos/cutandrun\\modules\\local\\peak_counts.nf\n",
      "Process PEAK_FRIP in filepath ./git_repos/cutandrun\\modules\\local\\peak_frip.nf\n",
      "Process SAMTOOLS_CUSTOMVIEW in filepath ./git_repos/cutandrun\\modules\\local\\samtools_custom_view.nf\n",
      "Process BEDTOOLS_SORT in filepath ./git_repos/cutandrun\\modules\\local\\for_patch\\bedtools\\sort\\main.nf\n",
      "Process DEEPTOOLS_BAMCOVERAGE in filepath ./git_repos/cutandrun\\modules\\local\\for_patch\\deeptools\\bamcoverage\\main.nf\n",
      "Process SAMTOOLS_VIEW in filepath ./git_repos/cutandrun\\modules\\local\\for_patch\\samtools\\view\\main.nf\n",
      "Process TRIMGALORE in filepath ./git_repos/cutandrun\\modules\\local\\for_patch\\trimgalore\\main.nf\n",
      "Process AWK in filepath ./git_repos/cutandrun\\modules\\local\\linux\\awk.nf\n",
      "Process AWK_SCRIPT in filepath ./git_repos/cutandrun\\modules\\local\\linux\\awk_script.nf\n",
      "Process CUT in filepath ./git_repos/cutandrun\\modules\\local\\linux\\cut.nf\n",
      "Process SORT in filepath ./git_repos/cutandrun\\modules\\local\\linux\\sort.nf\n",
      "Process FIND_UNIQUE_READS in filepath ./git_repos/cutandrun\\modules\\local\\python\\find_unique_reads.nf\n",
      "Process FRAG_LEN_HIST in filepath ./git_repos/cutandrun\\modules\\local\\python\\frag_len_hist.nf\n",
      "Process IGV_SESSION in filepath ./git_repos/cutandrun\\modules\\local\\python\\igv_session.nf\n",
      "Process MERGE_SAMPLE_METADATA in filepath ./git_repos/cutandrun\\modules\\local\\python\\merge_sample_metadata.nf\n",
      "Process CALCULATE_PEAK_REPROD in filepath ./git_repos/cutandrun\\modules\\local\\python\\peak_reprod.nf\n",
      "Process PLOT_CONSENSUS_PEAKS in filepath ./git_repos/cutandrun\\modules\\local\\python\\plot_consensus_peaks.nf\n",
      "Process SAMPLESHEET_CHECK in filepath ./git_repos/cutandrun\\modules\\local\\python\\samplesheet_check.nf\n",
      "Process BEDTOOLS_BAMTOBED in filepath ./git_repos/cutandrun\\modules\\nf-core\\bedtools\\bamtobed\\main.nf\n",
      "Process BEDTOOLS_COMPLEMENT in filepath ./git_repos/cutandrun\\modules\\nf-core\\bedtools\\complement\\main.nf\n",
      "Process BEDTOOLS_GENOMECOV in filepath ./git_repos/cutandrun\\modules\\nf-core\\bedtools\\genomecov\\main.nf\n",
      "Process BEDTOOLS_INTERSECT in filepath ./git_repos/cutandrun\\modules\\nf-core\\bedtools\\intersect\\main.nf\n",
      "Process BEDTOOLS_MERGE in filepath ./git_repos/cutandrun\\modules\\nf-core\\bedtools\\merge\\main.nf\n",
      "Process BOWTIE2_ALIGN in filepath ./git_repos/cutandrun\\modules\\nf-core\\bowtie2\\align\\main.nf\n",
      "Process BOWTIE2_BUILD in filepath ./git_repos/cutandrun\\modules\\nf-core\\bowtie2\\build\\main.nf\n",
      "Process CAT_FASTQ in filepath ./git_repos/cutandrun\\modules\\nf-core\\cat\\fastq\\main.nf\n",
      "Process CUSTOM_GETCHROMSIZES in filepath ./git_repos/cutandrun\\modules\\nf-core\\custom\\getchromsizes\\main.nf\n",
      "Process DEEPTOOLS_COMPUTEMATRIX in filepath ./git_repos/cutandrun\\modules\\nf-core\\deeptools\\computematrix\\main.nf\n",
      "Process DEEPTOOLS_MULTIBAMSUMMARY in filepath ./git_repos/cutandrun\\modules\\nf-core\\deeptools\\multibamsummary\\main.nf\n",
      "Process DEEPTOOLS_PLOTCORRELATION in filepath ./git_repos/cutandrun\\modules\\nf-core\\deeptools\\plotcorrelation\\main.nf\n",
      "Process DEEPTOOLS_PLOTFINGERPRINT in filepath ./git_repos/cutandrun\\modules\\nf-core\\deeptools\\plotfingerprint\\main.nf\n",
      "Process DEEPTOOLS_PLOTHEATMAP in filepath ./git_repos/cutandrun\\modules\\nf-core\\deeptools\\plotheatmap\\main.nf\n",
      "Process DEEPTOOLS_PLOTPCA in filepath ./git_repos/cutandrun\\modules\\nf-core\\deeptools\\plotpca\\main.nf\n",
      "Process FASTQC in filepath ./git_repos/cutandrun\\modules\\nf-core\\fastqc\\main.nf\n",
      "Process GUNZIP in filepath ./git_repos/cutandrun\\modules\\nf-core\\gunzip\\main.nf\n",
      "Process MACS2_CALLPEAK in filepath ./git_repos/cutandrun\\modules\\nf-core\\macs2\\callpeak\\main.nf\n",
      "Process PICARD_MARKDUPLICATES in filepath ./git_repos/cutandrun\\modules\\nf-core\\picard\\markduplicates\\main.nf\n",
      "Process PRESEQ_LCEXTRAP in filepath ./git_repos/cutandrun\\modules\\nf-core\\preseq\\lcextrap\\main.nf\n",
      "Process SAMTOOLS_FAIDX in filepath ./git_repos/cutandrun\\modules\\nf-core\\samtools\\faidx\\main.nf\n",
      "Process SAMTOOLS_FLAGSTAT in filepath ./git_repos/cutandrun\\modules\\nf-core\\samtools\\flagstat\\main.nf\n",
      "Process SAMTOOLS_IDXSTATS in filepath ./git_repos/cutandrun\\modules\\nf-core\\samtools\\idxstats\\main.nf\n",
      "Process SAMTOOLS_INDEX in filepath ./git_repos/cutandrun\\modules\\nf-core\\samtools\\index\\main.nf\n",
      "Process SAMTOOLS_SORT in filepath ./git_repos/cutandrun\\modules\\nf-core\\samtools\\sort\\main.nf\n",
      "Process SAMTOOLS_STATS in filepath ./git_repos/cutandrun\\modules\\nf-core\\samtools\\stats\\main.nf\n",
      "Process SAMTOOLS_VIEW in filepath ./git_repos/cutandrun\\modules\\nf-core\\samtools\\view\\main.nf\n",
      "Process SEACR_CALLPEAK in filepath ./git_repos/cutandrun\\modules\\nf-core\\seacr\\callpeak\\main.nf\n",
      "Process TABIX_BGZIPTABIX in filepath ./git_repos/cutandrun\\modules\\nf-core\\tabix\\bgziptabix\\main.nf\n",
      "Process UCSC_BEDCLIP in filepath ./git_repos/cutandrun\\modules\\nf-core\\ucsc\\bedclip\\main.nf\n",
      "Process UCSC_BEDGRAPHTOBIGWIG in filepath ./git_repos/cutandrun\\modules\\nf-core\\ucsc\\bedgraphtobigwig\\main.nf\n",
      "Process UNTAR in filepath ./git_repos/cutandrun\\modules\\nf-core\\untar\\main.nf\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Process get_software_versions in filepath ./git_repos/ddamsproteomics\\main.nf\n",
      "Process quantifySpectra in filepath ./git_repos/ddamsproteomics\\main.nf\n",
      "Process createSpectraLookup in filepath ./git_repos/ddamsproteomics\\main.nf\n",
      "Process quantLookup in filepath ./git_repos/ddamsproteomics\\main.nf\n",
      "Process countMS2perFile in filepath ./git_repos/ddamsproteomics\\main.nf\n",
      "Process countMS2sPerPlate in filepath ./git_repos/ddamsproteomics\\main.nf\n",
      "Process createTargetDecoyFasta in filepath ./git_repos/ddamsproteomics\\main.nf\n",
      "Process msgfPlus in filepath ./git_repos/ddamsproteomics\\main.nf\n",
      "Process percolator in filepath ./git_repos/ddamsproteomics\\main.nf\n",
      "Process svmToTSV in filepath ./git_repos/ddamsproteomics\\main.nf\n",
      "Process createPSMTable in filepath ./git_repos/ddamsproteomics\\main.nf\n",
      "Process psm2Peptides in filepath ./git_repos/ddamsproteomics\\main.nf\n",
      "Process prepProteinGeneSymbolTable in filepath ./git_repos/ddamsproteomics\\main.nf\n",
      "Process proteinFDR in filepath ./git_repos/ddamsproteomics\\main.nf\n",
      "Process proteinPeptideSetMerge in filepath ./git_repos/ddamsproteomics\\main.nf\n",
      "Process psmQC in filepath ./git_repos/ddamsproteomics\\main.nf\n",
      "Process featQC in filepath ./git_repos/ddamsproteomics\\main.nf\n",
      "Process collectQC in filepath ./git_repos/ddamsproteomics\\main.nf\n",
      "Process output_documentation in filepath ./git_repos/ddamsproteomics\\main.nf\n",
      "Process preprocess_fai in filepath ./git_repos/deepvariant\\main.nf\n",
      "Process preprocess_fastagz in filepath ./git_repos/deepvariant\\main.nf\n",
      "Process preprocess_gzfai in filepath ./git_repos/deepvariant\\main.nf\n",
      "Process preprocess_gzi in filepath ./git_repos/deepvariant\\main.nf\n",
      "Process preprocess_bam in filepath ./git_repos/deepvariant\\main.nf\n",
      "Process make_examples in filepath ./git_repos/deepvariant\\main.nf\n",
      "Process call_variants in filepath ./git_repos/deepvariant\\main.nf\n",
      "Process postprocess_variants in filepath ./git_repos/deepvariant\\main.nf\n",
      "Process get_software_versions in filepath ./git_repos/deepvariant\\main.nf\n",
      "Process CSV2TSV in filepath ./git_repos/demultiplex\\modules\\local\\csv2tsv.nf\n",
      "Process BASES2FASTQ in filepath ./git_repos/demultiplex\\modules\\nf-core\\bases2fastq\\main.nf\n",
      "Process BCL2FASTQ in filepath ./git_repos/demultiplex\\modules\\nf-core\\bcl2fastq\\main.nf\n",
      "Process BCLCONVERT in filepath ./git_repos/demultiplex\\modules\\nf-core\\bclconvert\\main.nf\n",
      "Process CUSTOM_DUMPSOFTWAREVERSIONS in filepath ./git_repos/demultiplex\\modules\\nf-core\\custom\\dumpsoftwareversions\\main.nf\n",
      "Process FALCO in filepath ./git_repos/demultiplex\\modules\\nf-core\\falco\\main.nf\n",
      "Process FASTP in filepath ./git_repos/demultiplex\\modules\\nf-core\\fastp\\main.nf\n",
      "Process FQTK in filepath ./git_repos/demultiplex\\modules\\nf-core\\fqtk\\main.nf\n",
      "Process MD5SUM in filepath ./git_repos/demultiplex\\modules\\nf-core\\md5sum\\main.nf\n",
      "Process MULTIQC in filepath ./git_repos/demultiplex\\modules\\nf-core\\multiqc\\main.nf\n",
      "Process SGDEMUX in filepath ./git_repos/demultiplex\\modules\\nf-core\\sgdemux\\main.nf\n",
      "Process UNTAR in filepath ./git_repos/demultiplex\\modules\\nf-core\\untar\\main.nf\n",
      "Process get_software_versions in filepath ./git_repos/denovohybrid\\main.nf\n",
      "Process porechop in filepath ./git_repos/denovohybrid\\main.nf\n",
      "Process filtlong in filepath ./git_repos/denovohybrid\\main.nf\n",
      "Process nanoplot in filepath ./git_repos/denovohybrid\\main.nf\n",
      "Process fastqc in filepath ./git_repos/denovohybrid\\main.nf\n",
      "Process seqpurge in filepath ./git_repos/denovohybrid\\main.nf\n",
      "Process sample_shortreads in filepath ./git_repos/denovohybrid\\main.nf\n",
      "Process miniasm in filepath ./git_repos/denovohybrid\\main.nf\n",
      "Process unicycler in filepath ./git_repos/denovohybrid\\main.nf\n",
      "Process wtdbg in filepath ./git_repos/denovohybrid\\main.nf\n",
      "Process racon in filepath ./git_repos/denovohybrid\\main.nf\n",
      "Process pilon in filepath ./git_repos/denovohybrid\\main.nf\n",
      "Process draw_assembly_graph in filepath ./git_repos/denovohybrid\\main.nf\n",
      "Process quast in filepath ./git_repos/denovohybrid\\main.nf\n",
      "Process format_final_output in filepath ./git_repos/denovohybrid\\main.nf\n",
      "Process multiqc in filepath ./git_repos/denovohybrid\\main.nf\n",
      "Process output_documentation in filepath ./git_repos/denovohybrid\\main.nf\n",
      "Process FILTER in filepath ./git_repos/detaxizer\\modules\\local\\filter.nf\n",
      "Process FILTER_BLASTN_IDENTCOV in filepath ./git_repos/detaxizer\\modules\\local\\filter_blastn_identcov.nf\n",
      "Process ISOLATE_IDS_FROM_KRAKEN2_TO_BLASTN in filepath ./git_repos/detaxizer\\modules\\local\\isolate_ids_from_kraken2_to_blastn.nf\n",
      "Process KRAKEN2PREPARATION in filepath ./git_repos/detaxizer\\modules\\local\\kraken2preparation.nf\n",
      "Process PARSE_KRAKEN2REPORT in filepath ./git_repos/detaxizer\\modules\\local\\parse_kraken2report.nf\n",
      "Process PREPARE_FASTA4BLASTN in filepath ./git_repos/detaxizer\\modules\\local\\prepare_fasta4blastn.nf\n",
      "Process RENAME_FASTQ_HEADERS_AFTER in filepath ./git_repos/detaxizer\\modules\\local\\rename_fastq_headers_after.nf\n",
      "Process RENAME_FASTQ_HEADERS_PRE in filepath ./git_repos/detaxizer\\modules\\local\\rename_fastq_headers_pre.nf\n",
      "Process SAMPLESHEET_CHECK in filepath ./git_repos/detaxizer\\modules\\local\\samplesheet_check.nf\n",
      "Process SUMMARIZER in filepath ./git_repos/detaxizer\\modules\\local\\summarizer.nf\n",
      "Process SUMMARY_BLASTN in filepath ./git_repos/detaxizer\\modules\\local\\summary_blastn.nf\n",
      "Process SUMMARY_KRAKEN2 in filepath ./git_repos/detaxizer\\modules\\local\\summary_kraken2.nf\n",
      "Process BLAST_BLASTN in filepath ./git_repos/detaxizer\\modules\\nf-core\\blast\\blastn\\main.nf\n",
      "Process BLAST_MAKEBLASTDB in filepath ./git_repos/detaxizer\\modules\\nf-core\\blast\\makeblastdb\\main.nf\n",
      "Process CUSTOM_DUMPSOFTWAREVERSIONS in filepath ./git_repos/detaxizer\\modules\\nf-core\\custom\\dumpsoftwareversions\\main.nf\n",
      "Process FASTP in filepath ./git_repos/detaxizer\\modules\\nf-core\\fastp\\main.nf\n",
      "Process FASTQC in filepath ./git_repos/detaxizer\\modules\\nf-core\\fastqc\\main.nf\n",
      "Process KRAKEN2_KRAKEN2 in filepath ./git_repos/detaxizer\\modules\\nf-core\\kraken2\\kraken2\\main.nf\n",
      "Process MULTIQC in filepath ./git_repos/detaxizer\\modules\\nf-core\\multiqc\\main.nf\n",
      "Process get_software_versions in filepath ./git_repos/diaproteomics\\main.nf\n",
      "Process dda_raw_file_conversion in filepath ./git_repos/diaproteomics\\main.nf\n",
      "Process dda_id_format_conversion in filepath ./git_repos/diaproteomics\\main.nf\n",
      "Process dda_library_generation in filepath ./git_repos/diaproteomics\\main.nf\n",
      "Process assay_generation in filepath ./git_repos/diaproteomics\\main.nf\n",
      "Process library_merging_and_alignment in filepath ./git_repos/diaproteomics\\main.nf\n",
      "Process pseudo_irt_generation in filepath ./git_repos/diaproteomics\\main.nf\n",
      "Process decoy_generation in filepath ./git_repos/diaproteomics\\main.nf\n",
      "Process dia_raw_file_conversion in filepath ./git_repos/diaproteomics\\main.nf\n",
      "Process dia_spectral_library_search in filepath ./git_repos/diaproteomics\\main.nf\n",
      "Process dia_search_output_merging in filepath ./git_repos/diaproteomics\\main.nf\n",
      "Process global_false_discovery_rate_estimation in filepath ./git_repos/diaproteomics\\main.nf\n",
      "Process export_of_scoring_results in filepath ./git_repos/diaproteomics\\main.nf\n",
      "Process chromatogram_indexing in filepath ./git_repos/diaproteomics\\main.nf\n",
      "Process chromatogram_alignment in filepath ./git_repos/diaproteomics\\main.nf\n",
      "Process reformatting in filepath ./git_repos/diaproteomics\\main.nf\n",
      "Process mztab_export in filepath ./git_repos/diaproteomics\\main.nf\n",
      "Process statistical_post_processing in filepath ./git_repos/diaproteomics\\main.nf\n",
      "Process output_visualization in filepath ./git_repos/diaproteomics\\main.nf\n",
      "Process output_documentation in filepath ./git_repos/diaproteomics\\main.nf\n",
      "Process TABULAR_TO_GSEA_CHIP in filepath ./git_repos/differentialabundance\\modules\\local\\tabular_to_gsea_chip.nf\n",
      "Process AFFY_JUSTRMA in filepath ./git_repos/differentialabundance\\modules\\nf-core\\affy\\justrma\\main.nf\n",
      "Process ATLASGENEANNOTATIONMANIPULATION_GTF2FEATUREANNOTATION in filepath ./git_repos/differentialabundance\\modules\\nf-core\\atlasgeneannotationmanipulation\\gtf2featureannotation\\main.nf\n",
      "Process CUSTOM_DUMPSOFTWAREVERSIONS in filepath ./git_repos/differentialabundance\\modules\\nf-core\\custom\\dumpsoftwareversions\\main.nf\n",
      "Process CUSTOM_MATRIXFILTER in filepath ./git_repos/differentialabundance\\modules\\nf-core\\custom\\matrixfilter\\main.nf\n",
      "Process CUSTOM_TABULARTOGSEACLS in filepath ./git_repos/differentialabundance\\modules\\nf-core\\custom\\tabulartogseacls\\main.nf\n",
      "Process CUSTOM_TABULARTOGSEAGCT in filepath ./git_repos/differentialabundance\\modules\\nf-core\\custom\\tabulartogseagct\\main.nf\n",
      "Process DESEQ2_DIFFERENTIAL in filepath ./git_repos/differentialabundance\\modules\\nf-core\\deseq2\\differential\\main.nf\n",
      "Process GEOQUERY_GETGEO in filepath ./git_repos/differentialabundance\\modules\\nf-core\\geoquery\\getgeo\\main.nf\n",
      "Process GSEA_GSEA in filepath ./git_repos/differentialabundance\\modules\\nf-core\\gsea\\gsea\\main.nf\n",
      "Process GUNZIP in filepath ./git_repos/differentialabundance\\modules\\nf-core\\gunzip\\main.nf\n",
      "Process LIMMA_DIFFERENTIAL in filepath ./git_repos/differentialabundance\\modules\\nf-core\\limma\\differential\\main.nf\n",
      "Process PROTEUS_READPROTEINGROUPS in filepath ./git_repos/differentialabundance\\modules\\nf-core\\proteus\\readproteingroups\\main.nf\n",
      "Process RMARKDOWNNOTEBOOK in filepath ./git_repos/differentialabundance\\modules\\nf-core\\rmarkdownnotebook\\main.nf\n",
      "Process SHINYNGS_APP in filepath ./git_repos/differentialabundance\\modules\\nf-core\\shinyngs\\app\\main.nf\n",
      "Process SHINYNGS_STATICDIFFERENTIAL in filepath ./git_repos/differentialabundance\\modules\\nf-core\\shinyngs\\staticdifferential\\main.nf\n",
      "Process SHINYNGS_STATICEXPLORATORY in filepath ./git_repos/differentialabundance\\modules\\nf-core\\shinyngs\\staticexploratory\\main.nf\n",
      "Process SHINYNGS_VALIDATEFOMCOMPONENTS in filepath ./git_repos/differentialabundance\\modules\\nf-core\\shinyngs\\validatefomcomponents\\main.nf\n",
      "Process UNTAR in filepath ./git_repos/differentialabundance\\modules\\nf-core\\untar\\main.nf\n",
      "Process ZIP in filepath ./git_repos/differentialabundance\\modules\\nf-core\\zip\\main.nf\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Process get_software_versions in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process check_replicates in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process uncompress_pathogen_fasta_genome in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process uncompress_host_fasta_genome in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process uncompress_pathogen_gff in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process uncompress_host_gff in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process uncompress_host_gff_trna in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process uncompress_host_gff_trna_file in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process combine_pathogen_host_fasta_genome in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process combine_host_genome_tRNA_gff_files_htseq in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process replace_gene_feature_gff_host_htseq in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process replace_gene_feature_gff_pathogen_htseq in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process replace_attribute_pathogen_gff_htseq in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process combine_pathogen_host_gff_files_htseq in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process extract_annotations_pathogen_htseq in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process extract_annotations_host_htseq in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process extract_reference_names_host_star in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process extract_reference_names_pathogen_star in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process replace_attribute_host_genome_gff_star_salmon in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process replace_attribute_host_tRNA_gff_star_salmon in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process combine_host_genome_tRNA_gff_star_salmon in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process replace_gene_feature_gff_host_salmon in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process replace_attribute_pathogen_gff_star_salmon in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process extract_annotations_pathogen_salmon in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process extract_annotations_host_salmon in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process create_transcriptome_fasta_host in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process create_transcriptome_fasta_host_tRNA in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process combine_host_fasta_transcriptomes in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process create_transcriptome_fasta_pathogen in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process combine_pathogen_host_fasta_transcriptome in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process replace_gene_feature_gff_pathogen_salmon in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process combine_pathogen_host_gff_files_salmon in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process fastqc in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process trimming in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process bbduk in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process fastqc_after_trimming in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process count_total_reads in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process count_total_read_pairs in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process create_decoy_transcriptome_file in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process salmon_index in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process salmon_quantification in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process split_table_salmon_each in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process extract_ambig_uniq_transcripts_genes in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process host_comb_ambig_uniq in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process pathogen_comb_ambig_uniq in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process combine_quantification_tables_salmon in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process split_quantification_tables_salmon in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process combine_annotations_quant_pathogen in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process combine_annotations_quant_host_salmon in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process tximport_host in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process combine_host_quant_gene_level_salmon in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process combine_annotations_quant_gene_level_salmon in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process scatter_plot_pathogen_salmon in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process scatter_plot_host_salmon in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process extract_processed_reads in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process collect_processed_reads in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process salmon_quantification_stats in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process plot_salmon_mapping_stats_host_pathogen in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process RNA_class_statistics_salmon_pathogen in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process RNA_class_statistics_salmon_host in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process plot_RNA_class_salmon_pathogen_each in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process plot_RNA_class_salmon_pathogen_combined in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process plot_RNA_class_salmon_host_each in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process plot_RNA_class_salmon_host_combined in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process STARindex_salmon_alignment in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process ALIGNMENTstar_for_salmon in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process salmon_quantification_alignment_based_mode in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process split_table_salmon_each_salmon_alignment_mode in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process extract_ambig_uniq_transcripts_genes_alignment_based in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process host_comb_ambig_uniq_alignment_based in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process pathogen_comb_ambig_uniq_alignment_based in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process tximport_host_salmon_alignment in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process combine_host_quant_gene_level_salmon_alignment in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process combine_quantification_tables_salmon_alignment_mode in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process split_quantification_tables_salmon_salmon_alignment_mode in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process combine_annotations_quant_pathogen_salmon_alignment_mode in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process combine_annotations_quant_host_salmon_alignment_mode in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process combine_annotations_quant_gene_level_salmon_alignment_mode in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process extract_processed_reads_STAR_for_salmon in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process collect_processed_reads_STAR_for_salmon in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process scatter_plot_pathogen_salmon_alignment_based in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process scatter_plot_host_salmon_alignment_based in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process extract_processed_reads_salmon_alignment_based in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process collect_processed_reads_salmon_alignment_based in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process salmon_quantification_stats_salmon_alignment_based in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process plot_salmon_mapping_stats_host_pathogen_salmon_alignment_based in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process RNA_class_statistics_salmon_pathogen_alignment_based in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process RNA_class_statistics_salmon_host_alignment_based in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process plot_RNA_class_salmon_pathogen_each_alignment_based in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process plot_RNA_class_salmon_host_each_alignment_based in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process plot_RNA_class_salmon_pathogen_combined_alignment_based in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process plot_RNA_class_salmon_host_combined_alignment_based in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process STARindex in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process ALIGNMENTstar in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process remove_crossmapped_reads in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process extract_processed_reads_STAR in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process collect_processed_reads_STAR in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process unique_mapping_stats_STAR in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process collect_stats_STAR_uniquely_mapped in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process count_crossmapped_reads in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process multi_mapping_stats in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process collect_stats_STAR_multi_mapped in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process star_mapping_stats in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process plot_star_mapping_stats in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process HTseq_unique_mapping in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process combine_quantification_tables_htseq_uniquely_mapped in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process htseq_uniquely_mapped_TPM in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process split_quantification_tables_htseq_uniquely_mapped in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process combine_annotations_quant_pathogen_uniquely_mapped_host in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process combine_annotations_quant_host_uniquely_mapped_host in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process scatter_plot_pathogen_htseq in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process scatter_plot_host_htseq in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process htseq_quantification_stats_uniquely_mapped in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process plot_mapping_stats_host_pathogen_htseq_uniquely_mapped in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process RNA_class_statistics_htseq_uniquely_mapped_pathogen in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process RNA_class_statistics_htseq_uniquely_mapped_host in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process plot_RNA_class_htseq_uniquely_mapped_pathogen_each in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process plot_RNA_class_htseq_uniquely_mapped_host_each in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process plot_RNA_class_htseq_uniquely_mapped_pathogen_combined in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process plot_RNA_class_htseq_uniquely_host_combined in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process multiqc in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process output_documentation in filepath ./git_repos/dualrnaseq\\main.nf\n",
      "Process unzip_reference in filepath ./git_repos/eager\\main.nf\n",
      "Process makeBWAIndex in filepath ./git_repos/eager\\main.nf\n",
      "Process makeBT2Index in filepath ./git_repos/eager\\main.nf\n",
      "Process makeFastaIndex in filepath ./git_repos/eager\\main.nf\n",
      "Process makeSeqDict in filepath ./git_repos/eager\\main.nf\n",
      "Process convertBam in filepath ./git_repos/eager\\main.nf\n",
      "Process indexinputbam in filepath ./git_repos/eager\\main.nf\n",
      "Process fastqc in filepath ./git_repos/eager\\main.nf\n",
      "Process fastp in filepath ./git_repos/eager\\main.nf\n",
      "Process adapter_removal in filepath ./git_repos/eager\\main.nf\n",
      "Process post_ar_fastq_trimming in filepath ./git_repos/eager\\main.nf\n",
      "Process lanemerge in filepath ./git_repos/eager\\main.nf\n",
      "Process lanemerge_hostremoval_fastq in filepath ./git_repos/eager\\main.nf\n",
      "Process fastqc_after_clipping in filepath ./git_repos/eager\\main.nf\n",
      "Process bwa in filepath ./git_repos/eager\\main.nf\n",
      "Process bwamem in filepath ./git_repos/eager\\main.nf\n",
      "Process circulargenerator in filepath ./git_repos/eager\\main.nf\n",
      "Process circularmapper in filepath ./git_repos/eager\\main.nf\n",
      "Process bowtie2 in filepath ./git_repos/eager\\main.nf\n",
      "Process hostremoval_input_fastq in filepath ./git_repos/eager\\main.nf\n",
      "Process seqtype_merge in filepath ./git_repos/eager\\main.nf\n",
      "Process samtools_flagstat in filepath ./git_repos/eager\\main.nf\n",
      "Process samtools_filter in filepath ./git_repos/eager\\main.nf\n",
      "Process samtools_flagstat_after_filter in filepath ./git_repos/eager\\main.nf\n",
      "Process endorSpy in filepath ./git_repos/eager\\main.nf\n",
      "Process dedup in filepath ./git_repos/eager\\main.nf\n",
      "Process markduplicates in filepath ./git_repos/eager\\main.nf\n",
      "Process library_merge in filepath ./git_repos/eager\\main.nf\n",
      "Process preseq in filepath ./git_repos/eager\\main.nf\n",
      "Process bedtools in filepath ./git_repos/eager\\main.nf\n",
      "Process damageprofiler in filepath ./git_repos/eager\\main.nf\n",
      "Process mapdamage_calculation in filepath ./git_repos/eager\\main.nf\n",
      "Process mapdamage_rescaling in filepath ./git_repos/eager\\main.nf\n",
      "Process mask_reference_for_pmdtools in filepath ./git_repos/eager\\main.nf\n",
      "Process pmdtools in filepath ./git_repos/eager\\main.nf\n",
      "Process bam_trim in filepath ./git_repos/eager\\main.nf\n",
      "Process additional_library_merge in filepath ./git_repos/eager\\main.nf\n",
      "Process qualimap in filepath ./git_repos/eager\\main.nf\n",
      "Process picard_addorreplacereadgroups in filepath ./git_repos/eager\\main.nf\n",
      "Process genotyping_ug in filepath ./git_repos/eager\\main.nf\n",
      "Process genotyping_hc in filepath ./git_repos/eager\\main.nf\n",
      "Process genotyping_freebayes in filepath ./git_repos/eager\\main.nf\n",
      "Process genotyping_pileupcaller in filepath ./git_repos/eager\\main.nf\n",
      "Process eigenstrat_snp_coverage in filepath ./git_repos/eager\\main.nf\n",
      "Process genotyping_angsd in filepath ./git_repos/eager\\main.nf\n",
      "Process bcftools_stats in filepath ./git_repos/eager\\main.nf\n",
      "Process vcf2genome in filepath ./git_repos/eager\\main.nf\n",
      "Process multivcfanalyzer in filepath ./git_repos/eager\\main.nf\n",
      "Process mtnucratio in filepath ./git_repos/eager\\main.nf\n",
      "Process sexdeterrmine_prep in filepath ./git_repos/eager\\main.nf\n",
      "Process sexdeterrmine in filepath ./git_repos/eager\\main.nf\n",
      "Process nuclear_contamination in filepath ./git_repos/eager\\main.nf\n",
      "Process print_nuclear_contamination in filepath ./git_repos/eager\\main.nf\n",
      "Process metagenomic_complexity_filter in filepath ./git_repos/eager\\main.nf\n",
      "Process malt in filepath ./git_repos/eager\\main.nf\n",
      "Process maltextract in filepath ./git_repos/eager\\main.nf\n",
      "Process decomp_kraken in filepath ./git_repos/eager\\main.nf\n",
      "Process kraken in filepath ./git_repos/eager\\main.nf\n",
      "Process kraken_parse in filepath ./git_repos/eager\\main.nf\n",
      "Process kraken_merge in filepath ./git_repos/eager\\main.nf\n",
      "Process output_documentation in filepath ./git_repos/eager\\main.nf\n",
      "Process get_software_versions in filepath ./git_repos/eager\\main.nf\n",
      "Process multiqc in filepath ./git_repos/eager\\main.nf\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Process CAT_FILES in filepath ./git_repos/epitopeprediction\\modules\\local\\cat_files.nf\n",
      "Process CSVTK_CONCAT in filepath ./git_repos/epitopeprediction\\modules\\local\\csvtk_concat.nf\n",
      "Process CSVTK_SPLIT in filepath ./git_repos/epitopeprediction\\modules\\local\\csvtk_split.nf\n",
      "Process EPYTOPE_CHECK_REQUESTED_MODELS in filepath ./git_repos/epitopeprediction\\modules\\local\\epytope_check_requested_models.nf\n",
      "Process EPYTOPE_GENERATE_PEPTIDES in filepath ./git_repos/epitopeprediction\\modules\\local\\epytope_generate_peptides.nf\n",
      "Process EPYTOPE_PEPTIDE_PREDICTION in filepath ./git_repos/epitopeprediction\\modules\\local\\epytope_peptide_prediction.nf\n",
      "Process EPYTOPE_SHOW_SUPPORTED_MODELS in filepath ./git_repos/epitopeprediction\\modules\\local\\epytope_show_supported_models.nf\n",
      "Process EXTERNAL_TOOLS_IMPORT in filepath ./git_repos/epitopeprediction\\modules\\local\\external_tools_import.nf\n",
      "Process GET_PREDICTION_VERSIONS in filepath ./git_repos/epitopeprediction\\modules\\local\\get_prediction_versions.nf\n",
      "Process MERGE_JSON in filepath ./git_repos/epitopeprediction\\modules\\local\\merge_json.nf\n",
      "Process SAMPLESHEET_CHECK in filepath ./git_repos/epitopeprediction\\modules\\local\\samplesheet_check.nf\n",
      "Process SNPSIFT_SPLIT in filepath ./git_repos/epitopeprediction\\modules\\local\\snpsift_split.nf\n",
      "Process SPLIT_PEPTIDES in filepath ./git_repos/epitopeprediction\\modules\\local\\split_peptides.nf\n",
      "Process VARIANT_SPLIT in filepath ./git_repos/epitopeprediction\\modules\\local\\variant_split.nf\n",
      "Process CUSTOM_DUMPSOFTWAREVERSIONS in filepath ./git_repos/epitopeprediction\\modules\\nf-core\\custom\\dumpsoftwareversions\\main.nf\n",
      "Process GUNZIP in filepath ./git_repos/epitopeprediction\\modules\\nf-core\\gunzip\\main.nf\n",
      "Process MULTIQC in filepath ./git_repos/epitopeprediction\\modules\\nf-core\\multiqc\\main.nf\n",
      "Process genotypegvcfs in filepath ./git_repos/exoseq\\JointDiscoveryGATK.nf\n",
      "Process variantSelect in filepath ./git_repos/exoseq\\JointDiscoveryGATK.nf\n",
      "Process recalSNPs in filepath ./git_repos/exoseq\\JointDiscoveryGATK.nf\n",
      "Process recalIndels in filepath ./git_repos/exoseq\\JointDiscoveryGATK.nf\n",
      "Process combineVariants in filepath ./git_repos/exoseq\\JointDiscoveryGATK.nf\n",
      "Process variantAnnotatesnpEff in filepath ./git_repos/exoseq\\JointDiscoveryGATK.nf\n",
      "Process variantAnnotateGATK in filepath ./git_repos/exoseq\\JointDiscoveryGATK.nf\n",
      "Process variantEvaluate in filepath ./git_repos/exoseq\\JointDiscoveryGATK.nf\n",
      "Process get_software_versions in filepath ./git_repos/exoseq\\JointDiscoveryGATK.nf\n",
      "Process multiqc in filepath ./git_repos/exoseq\\JointDiscoveryGATK.nf\n",
      "Process makeBWAIndex in filepath ./git_repos/exoseq\\PairedSingleSampleWF.nf\n",
      "Process makeFastaIndex in filepath ./git_repos/exoseq\\PairedSingleSampleWF.nf\n",
      "Process fastqc in filepath ./git_repos/exoseq\\PairedSingleSampleWF.nf\n",
      "Process trim_galore in filepath ./git_repos/exoseq\\PairedSingleSampleWF.nf\n",
      "Process bwamem in filepath ./git_repos/exoseq\\PairedSingleSampleWF.nf\n",
      "Process markDuplicates in filepath ./git_repos/exoseq\\PairedSingleSampleWF.nf\n",
      "Process recal_bam_files in filepath ./git_repos/exoseq\\PairedSingleSampleWF.nf\n",
      "Process applyBQSR in filepath ./git_repos/exoseq\\PairedSingleSampleWF.nf\n",
      "Process qualiMap in filepath ./git_repos/exoseq\\PairedSingleSampleWF.nf\n",
      "Process variantCall in filepath ./git_repos/exoseq\\PairedSingleSampleWF.nf\n",
      "Process get_software_versions in filepath ./git_repos/exoseq\\PairedSingleSampleWF.nf\n",
      "Process GenerateMultiQCconfig in filepath ./git_repos/exoseq\\PairedSingleSampleWF.nf\n",
      "Process multiqc in filepath ./git_repos/exoseq\\PairedSingleSampleWF.nf\n",
      "Process SAMPLESHEET_CHECK in filepath ./git_repos/fastquorum\\modules\\local\\samplesheet_check.nf\n",
      "Process ALIGN_BAM in filepath ./git_repos/fastquorum\\modules\\local\\align_bam\\main.nf\n",
      "Process FGBIO_CALLDDUPLEXCONSENSUSREADS in filepath ./git_repos/fastquorum\\modules\\local\\fgbio\\callduplexconsensusreads\\main.nf\n",
      "Process FGBIO_CALLMOLECULARCONSENSUSREADS in filepath ./git_repos/fastquorum\\modules\\local\\fgbio\\callmolecularconsensusreads\\main.nf\n",
      "Process FGBIO_COLLECTDUPLEXSEQMETRICS in filepath ./git_repos/fastquorum\\modules\\local\\fgbio\\collectduplexseqmetrics\\main.nf\n",
      "Process FGBIO_FASTQTOBAM in filepath ./git_repos/fastquorum\\modules\\local\\fgbio\\fastqtobam\\main.nf\n",
      "Process FGBIO_FILTERCONSENSUSREADS in filepath ./git_repos/fastquorum\\modules\\local\\fgbio\\filterconsensusreads\\main.nf\n",
      "Process FGBIO_GROUPREADSBYUMI in filepath ./git_repos/fastquorum\\modules\\local\\fgbio\\groupreadsbyumi\\main.nf\n",
      "Process CUSTOM_DUMPSOFTWAREVERSIONS in filepath ./git_repos/fastquorum\\modules\\nf-core\\modules\\custom\\dumpsoftwareversions\\main.nf\n",
      "Process FASTQC in filepath ./git_repos/fastquorum\\modules\\nf-core\\modules\\fastqc\\main.nf\n",
      "Process MULTIQC in filepath ./git_repos/fastquorum\\modules\\nf-core\\modules\\multiqc\\main.nf\n",
      "Process MULTIQC_MAPPINGS_CONFIG in filepath ./git_repos/fetchngs\\modules\\local\\multiqc_mappings_config\\main.nf\n",
      "Process SRA_FASTQ_FTP in filepath ./git_repos/fetchngs\\modules\\local\\sra_fastq_ftp\\main.nf\n",
      "Process SRA_IDS_TO_RUNINFO in filepath ./git_repos/fetchngs\\modules\\local\\sra_ids_to_runinfo\\main.nf\n",
      "Process SRA_MERGE_SAMPLESHEET in filepath ./git_repos/fetchngs\\modules\\local\\sra_merge_samplesheet\\main.nf\n",
      "Process SRA_RUNINFO_TO_FTP in filepath ./git_repos/fetchngs\\modules\\local\\sra_runinfo_to_ftp\\main.nf\n",
      "Process SRA_TO_SAMPLESHEET in filepath ./git_repos/fetchngs\\modules\\local\\sra_to_samplesheet\\main.nf\n",
      "Process SYNAPSE_GET in filepath ./git_repos/fetchngs\\modules\\local\\synapse_get\\main.nf\n",
      "Process SYNAPSE_LIST in filepath ./git_repos/fetchngs\\modules\\local\\synapse_list\\main.nf\n",
      "Process SYNAPSE_MERGE_SAMPLESHEET in filepath ./git_repos/fetchngs\\modules\\local\\synapse_merge_samplesheet\\main.nf\n",
      "Process SYNAPSE_SHOW in filepath ./git_repos/fetchngs\\modules\\local\\synapse_show\\main.nf\n",
      "Process SYNAPSE_TO_SAMPLESHEET in filepath ./git_repos/fetchngs\\modules\\local\\synapse_to_samplesheet\\main.nf\n",
      "Process CUSTOM_SRATOOLSNCBISETTINGS in filepath ./git_repos/fetchngs\\modules\\nf-core\\custom\\sratoolsncbisettings\\main.nf\n",
      "Process SRATOOLS_FASTERQDUMP in filepath ./git_repos/fetchngs\\modules\\nf-core\\sratools\\fasterqdump\\main.nf\n",
      "Process SRATOOLS_PREFETCH in filepath ./git_repos/fetchngs\\modules\\nf-core\\sratools\\prefetch\\main.nf\n",
      "Process UNTAR in filepath ./git_repos/fetchngs\\modules\\nf-core\\untar\\main.nf\n",
      "Process COMBGC in filepath ./git_repos/funcscan\\modules\\local\\combgc.nf\n",
      "Process DRAMP_DOWNLOAD in filepath ./git_repos/funcscan\\modules\\local\\dramp_download.nf\n",
      "Process ABRICATE_RUN in filepath ./git_repos/funcscan\\modules\\nf-core\\abricate\\run\\main.nf\n",
      "Process AMPCOMBI in filepath ./git_repos/funcscan\\modules\\nf-core\\ampcombi\\main.nf\n",
      "Process AMPIR in filepath ./git_repos/funcscan\\modules\\nf-core\\ampir\\main.nf\n",
      "Process AMPLIFY_PREDICT in filepath ./git_repos/funcscan\\modules\\nf-core\\amplify\\predict\\main.nf\n",
      "Process AMRFINDERPLUS_RUN in filepath ./git_repos/funcscan\\modules\\nf-core\\amrfinderplus\\run\\main.nf\n",
      "Process AMRFINDERPLUS_UPDATE in filepath ./git_repos/funcscan\\modules\\nf-core\\amrfinderplus\\update\\main.nf\n",
      "Process ANTISMASH_ANTISMASHLITE in filepath ./git_repos/funcscan\\modules\\nf-core\\antismash\\antismashlite\\main.nf\n",
      "Process ANTISMASH_ANTISMASHLITEDOWNLOADDATABASES in filepath ./git_repos/funcscan\\modules\\nf-core\\antismash\\antismashlitedownloaddatabases\\main.nf\n",
      "Process BAKTA_BAKTA in filepath ./git_repos/funcscan\\modules\\nf-core\\bakta\\bakta\\main.nf\n",
      "Process BAKTA_BAKTADBDOWNLOAD in filepath ./git_repos/funcscan\\modules\\nf-core\\bakta\\baktadbdownload\\main.nf\n",
      "Process BIOAWK in filepath ./git_repos/funcscan\\modules\\nf-core\\bioawk\\main.nf\n",
      "Process CUSTOM_DUMPSOFTWAREVERSIONS in filepath ./git_repos/funcscan\\modules\\nf-core\\custom\\dumpsoftwareversions\\main.nf\n",
      "Process DEEPARG_DOWNLOADDATA in filepath ./git_repos/funcscan\\modules\\nf-core\\deeparg\\downloaddata\\main.nf\n",
      "Process DEEPARG_PREDICT in filepath ./git_repos/funcscan\\modules\\nf-core\\deeparg\\predict\\main.nf\n",
      "Process DEEPBGC_DOWNLOAD in filepath ./git_repos/funcscan\\modules\\nf-core\\deepbgc\\download\\main.nf\n",
      "Process DEEPBGC_PIPELINE in filepath ./git_repos/funcscan\\modules\\nf-core\\deepbgc\\pipeline\\main.nf\n",
      "Process FARGENE in filepath ./git_repos/funcscan\\modules\\nf-core\\fargene\\main.nf\n",
      "Process GECCO_RUN in filepath ./git_repos/funcscan\\modules\\nf-core\\gecco\\run\\main.nf\n",
      "Process GUNZIP in filepath ./git_repos/funcscan\\modules\\nf-core\\gunzip\\main.nf\n",
      "Process HAMRONIZATION_ABRICATE in filepath ./git_repos/funcscan\\modules\\nf-core\\hamronization\\abricate\\main.nf\n",
      "Process HAMRONIZATION_AMRFINDERPLUS in filepath ./git_repos/funcscan\\modules\\nf-core\\hamronization\\amrfinderplus\\main.nf\n",
      "Process HAMRONIZATION_DEEPARG in filepath ./git_repos/funcscan\\modules\\nf-core\\hamronization\\deeparg\\main.nf\n",
      "Process HAMRONIZATION_FARGENE in filepath ./git_repos/funcscan\\modules\\nf-core\\hamronization\\fargene\\main.nf\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Process HAMRONIZATION_RGI in filepath ./git_repos/funcscan\\modules\\nf-core\\hamronization\\rgi\\main.nf\n",
      "Process HAMRONIZATION_SUMMARIZE in filepath ./git_repos/funcscan\\modules\\nf-core\\hamronization\\summarize\\main.nf\n",
      "Process HMMER_HMMSEARCH in filepath ./git_repos/funcscan\\modules\\nf-core\\hmmer\\hmmsearch\\main.nf\n",
      "Process MACREL_CONTIGS in filepath ./git_repos/funcscan\\modules\\nf-core\\macrel\\contigs\\main.nf\n",
      "Process MULTIQC in filepath ./git_repos/funcscan\\modules\\nf-core\\multiqc\\main.nf\n",
      "Process PRODIGAL in filepath ./git_repos/funcscan\\modules\\nf-core\\prodigal\\main.nf\n",
      "Process PROKKA in filepath ./git_repos/funcscan\\modules\\nf-core\\prokka\\main.nf\n",
      "Process PYRODIGAL in filepath ./git_repos/funcscan\\modules\\nf-core\\pyrodigal\\main.nf\n",
      "Process RGI_MAIN in filepath ./git_repos/funcscan\\modules\\nf-core\\rgi\\main\\main.nf\n",
      "Process TABIX_BGZIP in filepath ./git_repos/funcscan\\modules\\nf-core\\tabix\\bgzip\\main.nf\n",
      "Process UNTAR in filepath ./git_repos/funcscan\\modules\\nf-core\\untar\\main.nf\n",
      "Process FASTASPLITTER in filepath ./git_repos/genomeannotator\\modules\\local\\fastasplitter.nf\n",
      "Process FASTP in filepath ./git_repos/genomeannotator\\modules\\local\\fastp.nf\n",
      "Process GFFREAD in filepath ./git_repos/genomeannotator\\modules\\local\\gffread.nf\n",
      "Process KRAKEN in filepath ./git_repos/genomeannotator\\modules\\local\\kraken.nf\n",
      "Process REPEATMODELER in filepath ./git_repos/genomeannotator\\modules\\local\\repeatmodeler.nf\n",
      "Process SAMPLESHEET_CHECK in filepath ./git_repos/genomeannotator\\modules\\local\\samplesheet_check.nf\n",
      "Process AUGUSTUS_ALIGNTOHINTS in filepath ./git_repos/genomeannotator\\modules\\local\\augustus\\aligntohints.nf\n",
      "Process AUGUSTUS_AUGUSTUSBATCH in filepath ./git_repos/genomeannotator\\modules\\local\\augustus\\augustusbatch.nf\n",
      "Process AUGUSTUS_BAM2HINTS in filepath ./git_repos/genomeannotator\\modules\\local\\augustus\\bam2hints.nf\n",
      "Process AUGUSTUS_FIXJOINGENES in filepath ./git_repos/genomeannotator\\modules\\local\\augustus\\fixjoingenes.nf\n",
      "Process AUGUSTUS_STAGECONFIG in filepath ./git_repos/genomeannotator\\modules\\local\\augustus\\stageconfig.nf\n",
      "Process AUGUSTUS_TRAINING in filepath ./git_repos/genomeannotator\\modules\\local\\augustus\\training.nf\n",
      "Process BUSCO_BUSCO in filepath ./git_repos/genomeannotator\\modules\\local\\busco\\busco.nf\n",
      "Process BUSCO_DOWNLOADDB in filepath ./git_repos/genomeannotator\\modules\\local\\busco\\downloaddb.nf\n",
      "Process CAT_FASTA in filepath ./git_repos/genomeannotator\\modules\\local\\cat\\fasta.nf\n",
      "Process CAT_GFF in filepath ./git_repos/genomeannotator\\modules\\local\\cat\\gff.nf\n",
      "Process DEEPTOOLS_BAMCOVERAGE in filepath ./git_repos/genomeannotator\\modules\\local\\deeptools\\bamcoverage.nf\n",
      "Process EVIDENCEMODELER_EXECUTE in filepath ./git_repos/genomeannotator\\modules\\local\\evidencemodeler\\execute.nf\n",
      "Process EVIDENCEMODELER_MERGE in filepath ./git_repos/genomeannotator\\modules\\local\\evidencemodeler\\merge.nf\n",
      "Process EVIDENCEMODELER_PARTITION in filepath ./git_repos/genomeannotator\\modules\\local\\evidencemodeler\\partition.nf\n",
      "Process EXONERATE_FASTACLEAN in filepath ./git_repos/genomeannotator\\modules\\local\\exonerate\\fastaclean.nf\n",
      "Process GAAS_FASTACLEANER in filepath ./git_repos/genomeannotator\\modules\\local\\gaas\\fastacleaner.nf\n",
      "Process GAAS_FASTAFILTERBYSIZE in filepath ./git_repos/genomeannotator\\modules\\local\\gaas\\fastafilterbysize.nf\n",
      "Process GAAS_FASTASTATISTICS in filepath ./git_repos/genomeannotator\\modules\\local\\gaas\\fastastatistics.nf\n",
      "Process HELPER_ADDEXONS in filepath ./git_repos/genomeannotator\\modules\\local\\helper\\addexons.nf\n",
      "Process HELPER_BAMTOGFF in filepath ./git_repos/genomeannotator\\modules\\local\\helper\\bamtogff.nf\n",
      "Process HELPER_CREATEGFFIDS in filepath ./git_repos/genomeannotator\\modules\\local\\helper\\creategffids.nf\n",
      "Process HELPER_DOWNLOADRFAM in filepath ./git_repos/genomeannotator\\modules\\local\\helper\\downloadrfam.nf\n",
      "Process HELPER_EVM2GFF in filepath ./git_repos/genomeannotator\\modules\\local\\helper\\evm2gff.nf\n",
      "Process HELPER_GFFFILTERMULTI in filepath ./git_repos/genomeannotator\\modules\\local\\helper\\gfffiltermulti.nf\n",
      "Process HELPER_GTF2HINTS in filepath ./git_repos/genomeannotator\\modules\\local\\helper\\gtf2hints.nf\n",
      "Process HELPER_KRAKEN2GFF in filepath ./git_repos/genomeannotator\\modules\\local\\helper\\kraken2gff.nf\n",
      "Process HELPER_MATCH2GMOD in filepath ./git_repos/genomeannotator\\modules\\local\\helper\\match2gmod.nf\n",
      "Process HELPER_MINIMAPTOHINTS in filepath ./git_repos/genomeannotator\\modules\\local\\helper\\minimaptohints.nf\n",
      "Process HELPER_PASA2TRAINING in filepath ./git_repos/genomeannotator\\modules\\local\\helper\\pasa2training.nf\n",
      "Process HELPER_RFAMTOGFF in filepath ./git_repos/genomeannotator\\modules\\local\\helper\\rfamtogff.nf\n",
      "Process HELPER_SPALNTOEVM in filepath ./git_repos/genomeannotator\\modules\\local\\helper\\spalntoevm.nf\n",
      "Process HELPER_SPALNTOGMOD in filepath ./git_repos/genomeannotator\\modules\\local\\helper\\spalntogmod.nf\n",
      "Process HELPER_SPALNTOTRAINING in filepath ./git_repos/genomeannotator\\modules\\local\\helper\\spalntotraining.nf\n",
      "Process INFERNAL_PRESS in filepath ./git_repos/genomeannotator\\modules\\local\\infernal\\press.nf\n",
      "Process INFERNAL_SEARCH in filepath ./git_repos/genomeannotator\\modules\\local\\infernal\\search.nf\n",
      "Process MINIMAP2_ALIGN in filepath ./git_repos/genomeannotator\\modules\\local\\minimap2\\align.nf\n",
      "Process MINIMAP2_INDEX in filepath ./git_repos/genomeannotator\\modules\\local\\minimap2\\index.nf\n",
      "Process PASA_ALIGNASSEMBLE in filepath ./git_repos/genomeannotator\\modules\\local\\pasa\\alignassemble.nf\n",
      "Process PASA_ASMBLSTOTRAINING in filepath ./git_repos/genomeannotator\\modules\\local\\pasa\\asmblstotraining.nf\n",
      "Process PASA_SEQCLEAN in filepath ./git_repos/genomeannotator\\modules\\local\\pasa\\seqclean.nf\n",
      "Process REPEATMASKER_REPEATMASK in filepath ./git_repos/genomeannotator\\modules\\local\\repeatmasker\\repeatmask.nf\n",
      "Process REPEATMASKER_STAGELIB in filepath ./git_repos/genomeannotator\\modules\\local\\repeatmasker\\stagelib.nf\n",
      "Process SAMTOOLS_FAIDX in filepath ./git_repos/genomeannotator\\modules\\local\\samtools\\faidx.nf\n",
      "Process SAMTOOLS_MERGE in filepath ./git_repos/genomeannotator\\modules\\local\\samtools\\merge.nf\n",
      "Process SATSUMA2_SATSUMASYNTENY2 in filepath ./git_repos/genomeannotator\\modules\\local\\satsuma2\\satsumasynteny2.nf\n",
      "Process SPALN_ALIGN in filepath ./git_repos/genomeannotator\\modules\\local\\spaln\\align.nf\n",
      "Process SPALN_MAKEINDEX in filepath ./git_repos/genomeannotator\\modules\\local\\spaln\\makeindex.nf\n",
      "Process SPALN_MERGE in filepath ./git_repos/genomeannotator\\modules\\local\\spaln\\merge.nf\n",
      "Process STAR_ALIGN in filepath ./git_repos/genomeannotator\\modules\\local\\star\\align.nf\n",
      "Process STAR_INDEX in filepath ./git_repos/genomeannotator\\modules\\local\\star\\index.nf\n",
      "Process TRINITY_GENOMEGUIDED in filepath ./git_repos/genomeannotator\\modules\\local\\trinity\\genomeguided.nf\n",
      "Process CAT_FASTQ in filepath ./git_repos/genomeannotator\\modules\\nf-core\\modules\\cat\\fastq\\main.nf\n",
      "Process CUSTOM_DUMPSOFTWAREVERSIONS in filepath ./git_repos/genomeannotator\\modules\\nf-core\\modules\\custom\\dumpsoftwareversions\\main.nf\n",
      "Process FASTQC in filepath ./git_repos/genomeannotator\\modules\\nf-core\\modules\\fastqc\\main.nf\n",
      "Process GUNZIP in filepath ./git_repos/genomeannotator\\modules\\nf-core\\modules\\gunzip\\main.nf\n",
      "Process MULTIQC in filepath ./git_repos/genomeannotator\\modules\\nf-core\\modules\\multiqc\\main.nf\n",
      "Process SAMPLESHEET_CHECK in filepath ./git_repos/genomeassembler\\modules\\local\\samplesheet_check.nf\n",
      "Process MASH_FILTER in filepath ./git_repos/genomeassembler\\modules\\local\\mash\\filter.nf\n",
      "Process BUSCO in filepath ./git_repos/genomeassembler\\modules\\nf-core\\busco\\main.nf\n",
      "Process CUSTOM_DUMPSOFTWAREVERSIONS in filepath ./git_repos/genomeassembler\\modules\\nf-core\\custom\\dumpsoftwareversions\\main.nf\n",
      "Process FASTK_FASTK in filepath ./git_repos/genomeassembler\\modules\\nf-core\\fastk\\fastk\\main.nf\n",
      "Process FASTK_HISTEX in filepath ./git_repos/genomeassembler\\modules\\nf-core\\fastk\\histex\\main.nf\n",
      "Process FASTK_MERGE in filepath ./git_repos/genomeassembler\\modules\\nf-core\\fastk\\merge\\main.nf\n",
      "Process FASTQC in filepath ./git_repos/genomeassembler\\modules\\nf-core\\fastqc\\main.nf\n",
      "Process GENESCOPEFK in filepath ./git_repos/genomeassembler\\modules\\nf-core\\genescopefk\\main.nf\n",
      "Process GENOMESCOPE2 in filepath ./git_repos/genomeassembler\\modules\\nf-core\\genomescope2\\main.nf\n",
      "Process MASH_SCREEN in filepath ./git_repos/genomeassembler\\modules\\nf-core\\mash\\screen\\main.nf\n",
      "Process MASH_SKETCH in filepath ./git_repos/genomeassembler\\modules\\nf-core\\mash\\sketch\\main.nf\n",
      "Process MERQURY in filepath ./git_repos/genomeassembler\\modules\\nf-core\\merqury\\main.nf\n",
      "Process MERQURYFK_KATCOMP in filepath ./git_repos/genomeassembler\\modules\\nf-core\\merquryfk\\katcomp\\main.nf\n",
      "Process MERQURYFK_KATGC in filepath ./git_repos/genomeassembler\\modules\\nf-core\\merquryfk\\katgc\\main.nf\n",
      "Process MERQURYFK_MERQURYFK in filepath ./git_repos/genomeassembler\\modules\\nf-core\\merquryfk\\merquryfk\\main.nf\n",
      "Process MERQURYFK_PLOIDYPLOT in filepath ./git_repos/genomeassembler\\modules\\nf-core\\merquryfk\\ploidyplot\\main.nf\n",
      "Process MERYL_COUNT in filepath ./git_repos/genomeassembler\\modules\\nf-core\\meryl\\count\\main.nf\n",
      "Process MERYL_HISTOGRAM in filepath ./git_repos/genomeassembler\\modules\\nf-core\\meryl\\histogram\\main.nf\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Process MERYL_UNIONSUM in filepath ./git_repos/genomeassembler\\modules\\nf-core\\meryl\\unionsum\\main.nf\n",
      "Process MULTIQC in filepath ./git_repos/genomeassembler\\modules\\nf-core\\multiqc\\main.nf\n",
      "Process NANOPLOT in filepath ./git_repos/genomeassembler\\modules\\nf-core\\nanoplot\\main.nf\n",
      "Process QUAST in filepath ./git_repos/genomeassembler\\modules\\nf-core\\quast\\main.nf\n",
      "Process SAMTOOLS_FASTQ in filepath ./git_repos/genomeassembler\\modules\\nf-core\\samtools\\fastq\\main.nf\n",
      "Process SAMPLESHEET_CHECK in filepath ./git_repos/genomeskim\\modules\\local\\samplesheet_check.nf\n",
      "Process CUSTOM_DUMPSOFTWAREVERSIONS in filepath ./git_repos/genomeskim\\modules\\nf-core\\modules\\custom\\dumpsoftwareversions\\main.nf\n",
      "Process FASTQC in filepath ./git_repos/genomeskim\\modules\\nf-core\\modules\\fastqc\\main.nf\n",
      "Process MULTIQC in filepath ./git_repos/genomeskim\\modules\\nf-core\\modules\\multiqc\\main.nf\n",
      "Process SAMPLESHEET_CHECK in filepath ./git_repos/gwas\\modules\\local\\samplesheet_check.nf\n",
      "Process NO_SAMPLESHEET in filepath ./git_repos/gwas\\modules\\local\\qc_input_validations\\no_samplesheet.nf\n",
      "Process SAMPLESHEET in filepath ./git_repos/gwas\\modules\\local\\qc_input_validations\\samplesheet.nf\n",
      "Process ANALYZE_X in filepath ./git_repos/gwas\\modules\\local\\qc_processes\\analyze_x.nf\n",
      "Process BATCH_PROC in filepath ./git_repos/gwas\\modules\\local\\qc_processes\\batch_proc.nf\n",
      "Process CALCULATE_MAF in filepath ./git_repos/gwas\\modules\\local\\qc_processes\\calculate_maf.nf\n",
      "Process CALCULATE_SAMPLE_HETEROZYGOSITY in filepath ./git_repos/gwas\\modules\\local\\qc_processes\\calculate_sample_heterozygosity.nf\n",
      "Process CALCULATE_SNP_SKEW_STATUS in filepath ./git_repos/gwas\\modules\\local\\qc_processes\\calculate_snp_skew_status.nf\n",
      "Process COMP_PCA in filepath ./git_repos/gwas\\modules\\local\\qc_processes\\comp_pca.nf\n",
      "Process CONVERT_IN_VCF in filepath ./git_repos/gwas\\modules\\local\\qc_processes\\convert_in_vcf.nf\n",
      "Process FIND_HWE_OF_SNPS in filepath ./git_repos/gwas\\modules\\local\\qc_processes\\find_hwe_of_snps.nf\n",
      "Process FIND_RELATED_INDIV in filepath ./git_repos/gwas\\modules\\local\\qc_processes\\find_related_indiv.nf\n",
      "Process FIND_SNP_EXTREME_DIFFERENTIAL_MISSINGNESS in filepath ./git_repos/gwas\\modules\\local\\qc_processes\\find_snp_extreme_differential_missingness.nf\n",
      "Process GET_BAD_INDIVS_MISSING_HET in filepath ./git_repos/gwas\\modules\\local\\qc_processes\\get_bad_indivs_missing_het.nf\n",
      "Process GET_DUPLICATE_MARKERS in filepath ./git_repos/gwas\\modules\\local\\qc_processes\\get_duplicate_markers.nf\n",
      "Process GET_INIT_MAF in filepath ./git_repos/gwas\\modules\\local\\qc_processes\\get_init_maf.nf\n",
      "Process GET_X in filepath ./git_repos/gwas\\modules\\local\\qc_processes\\get_x.nf\n",
      "Process IDENTIFY_INDIV_DISC_SEX_INFO in filepath ./git_repos/gwas\\modules\\local\\qc_processes\\identify_indiv_disc_sex_info.nf\n",
      "Process PRUNE_FOR_IBD in filepath ./git_repos/gwas\\modules\\local\\qc_processes\\prune_for_ibd.nf\n",
      "Process PRUNE_FOR_IBDLD in filepath ./git_repos/gwas\\modules\\local\\qc_processes\\prune_for_ibdld.nf\n",
      "Process REMOVE_DUPLICATE_SNPS in filepath ./git_repos/gwas\\modules\\local\\qc_processes\\remove_duplicate_snps.nf\n",
      "Process REMOVE_QC_INDIVS in filepath ./git_repos/gwas\\modules\\local\\qc_processes\\remove_qc_indivs.nf\n",
      "Process REMOVE_QC_PHASE1 in filepath ./git_repos/gwas\\modules\\local\\qc_processes\\remove_qc_phase1.nf\n",
      "Process REMOVE_SKEW_SNPS in filepath ./git_repos/gwas\\modules\\local\\qc_processes\\remove_skew_snps.nf\n",
      "Process SAMPLE_SHEET in filepath ./git_repos/gwas\\modules\\local\\qc_processes\\sample_sheet.nf\n",
      "Process SHOW_HWE_STATS in filepath ./git_repos/gwas\\modules\\local\\qc_processes\\show_hwe_stats.nf\n",
      "Process SHOW_INIT_MAF in filepath ./git_repos/gwas\\modules\\local\\qc_processes\\show_init_maf.nf\n",
      "Process DRAW_PCA in filepath ./git_repos/gwas\\modules\\local\\qc_report\\draw_pca.nf\n",
      "Process GENERATE_DIFFERENTIAL_MISSINGNESS_PLOT in filepath ./git_repos/gwas\\modules\\local\\qc_report\\generate_differential_missingness_plot.nf\n",
      "Process GENERATE_HWE_PLOT in filepath ./git_repos/gwas\\modules\\local\\qc_report\\generate_hwe_plot.nf\n",
      "Process GENERATE_INDIV_MISSINGNESS_PLOT in filepath ./git_repos/gwas\\modules\\local\\qc_report\\generate_indiv_missingness_plot.nf\n",
      "Process GENERATE_MAF_PLOT in filepath ./git_repos/gwas\\modules\\local\\qc_report\\generate_maf_plot.nf\n",
      "Process GENERATE_MISS_HET_PLOT in filepath ./git_repos/gwas\\modules\\local\\qc_report\\generate_miss_het_plot.nf\n",
      "Process GENERATE_SNP_MISSINGNESS_PLOT in filepath ./git_repos/gwas\\modules\\local\\qc_report\\generate_snp_missingness_plot.nf\n",
      "Process IN_MD5 in filepath ./git_repos/gwas\\modules\\local\\qc_report\\in_md5.nf\n",
      "Process OUT_MD5 in filepath ./git_repos/gwas\\modules\\local\\qc_report\\out_md5.nf\n",
      "Process PRODUCE_REPORTS in filepath ./git_repos/gwas\\modules\\local\\qc_report\\produce_reports.nf\n",
      "Process CUSTOM_DUMPSOFTWAREVERSIONS in filepath ./git_repos/gwas\\modules\\nf-core\\custom\\dumpsoftwareversions\\main.nf\n",
      "Process FASTQC in filepath ./git_repos/gwas\\modules\\nf-core\\fastqc\\main.nf\n",
      "Process BCFTOOLS_MERGE in filepath ./git_repos/gwas\\modules\\nf-core\\modules\\bcftools\\merge\\main.nf\n",
      "Process PLINK_VCF in filepath ./git_repos/gwas\\modules\\nf-core\\modules\\plink\\vcf\\main.nf\n",
      "Process TABIX_BGZIP in filepath ./git_repos/gwas\\modules\\nf-core\\modules\\tabix\\bgzip\\main.nf\n",
      "Process TABIX_TABIX in filepath ./git_repos/gwas\\modules\\nf-core\\modules\\tabix\\tabix\\main.nf\n",
      "Process VCFTOOLS in filepath ./git_repos/gwas\\modules\\nf-core\\modules\\vcftools\\main.nf\n",
      "Process MULTIQC in filepath ./git_repos/gwas\\modules\\nf-core\\multiqc\\main.nf\n",
      "Process BCFTOOLS_MERGE in filepath ./git_repos/gwas\\nf-core\\modules\\bcftools\\merge\\main.nf\n",
      "Process TABIX_BGZIP in filepath ./git_repos/gwas\\nf-core\\modules\\tabix\\bgzip\\main.nf\n",
      "Process TABIX_TABIX in filepath ./git_repos/gwas\\nf-core\\modules\\tabix\\tabix\\main.nf\n",
      "Process VCFTOOLS in filepath ./git_repos/gwas\\nf-core\\modules\\vcftools\\main.nf\n",
      "Process SAMPLESHEET_CHECK in filepath ./git_repos/hgtseq\\modules\\local\\samplesheet_check.nf\n",
      "Process GAWK in filepath ./git_repos/hgtseq\\modules\\local\\gawk\\main.nf\n",
      "Process PARSEOUTPUTS in filepath ./git_repos/hgtseq\\modules\\local\\parseoutputs\\main.nf\n",
      "Process RANALYSIS in filepath ./git_repos/hgtseq\\modules\\local\\ranalysis\\main.nf\n",
      "Process SAMTOOLS_FASTQ in filepath ./git_repos/hgtseq\\modules\\local\\samtools\\fastqlocal\\main.nf\n",
      "Process SAMTOOLS_VIEW_SINGLE in filepath ./git_repos/hgtseq\\modules\\local\\samtools\\viewsingle\\main.nf\n",
      "Process BAMTOOLS_STATS in filepath ./git_repos/hgtseq\\modules\\nf-core\\bamtools\\stats\\main.nf\n",
      "Process BWA_INDEX in filepath ./git_repos/hgtseq\\modules\\nf-core\\bwa\\index\\main.nf\n",
      "Process BWA_MEM in filepath ./git_repos/hgtseq\\modules\\nf-core\\bwa\\mem\\main.nf\n",
      "Process BWAMEM2_INDEX in filepath ./git_repos/hgtseq\\modules\\nf-core\\bwamem2\\index\\main.nf\n",
      "Process BWAMEM2_MEM in filepath ./git_repos/hgtseq\\modules\\nf-core\\bwamem2\\mem\\main.nf\n",
      "Process CUSTOM_DUMPSOFTWAREVERSIONS in filepath ./git_repos/hgtseq\\modules\\nf-core\\custom\\dumpsoftwareversions\\main.nf\n",
      "Process FASTQC in filepath ./git_repos/hgtseq\\modules\\nf-core\\fastqc\\main.nf\n",
      "Process KRAKEN2_KRAKEN2 in filepath ./git_repos/hgtseq\\modules\\nf-core\\kraken2\\kraken2\\main.nf\n",
      "Process KRONA_KTIMPORTTAXONOMY in filepath ./git_repos/hgtseq\\modules\\nf-core\\krona\\ktimporttaxonomy\\main.nf\n",
      "Process KRONA_KTUPDATETAXONOMY in filepath ./git_repos/hgtseq\\modules\\nf-core\\krona\\ktupdatetaxonomy\\main.nf\n",
      "Process MULTIQC in filepath ./git_repos/hgtseq\\modules\\nf-core\\multiqc\\main.nf\n",
      "Process QUALIMAP_BAMQC in filepath ./git_repos/hgtseq\\modules\\nf-core\\qualimap\\bamqc\\main.nf\n",
      "Process SAMTOOLS_FASTQ in filepath ./git_repos/hgtseq\\modules\\nf-core\\samtools\\fastq\\main.nf\n",
      "Process SAMTOOLS_FLAGSTAT in filepath ./git_repos/hgtseq\\modules\\nf-core\\samtools\\flagstat\\main.nf\n",
      "Process SAMTOOLS_IDXSTATS in filepath ./git_repos/hgtseq\\modules\\nf-core\\samtools\\idxstats\\main.nf\n",
      "Process SAMTOOLS_INDEX in filepath ./git_repos/hgtseq\\modules\\nf-core\\samtools\\index\\main.nf\n",
      "Process SAMTOOLS_SORT in filepath ./git_repos/hgtseq\\modules\\nf-core\\samtools\\sort\\main.nf\n",
      "Process SAMTOOLS_STATS in filepath ./git_repos/hgtseq\\modules\\nf-core\\samtools\\stats\\main.nf\n",
      "Process SAMTOOLS_VIEW in filepath ./git_repos/hgtseq\\modules\\nf-core\\samtools\\view\\main.nf\n",
      "Process TRIMGALORE in filepath ./git_repos/hgtseq\\modules\\nf-core\\trimgalore\\main.nf\n",
      "Process UNTAR in filepath ./git_repos/hgtseq\\modules\\nf-core\\untar\\main.nf\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Process MULTIQC in filepath ./git_repos/hic\\modules\\local\\multiqc.nf\n",
      "Process SAMPLESHEET_CHECK in filepath ./git_repos/hic\\modules\\local\\samplesheet_check.nf\n",
      "Process SPLIT_COOLER_DUMP in filepath ./git_repos/hic\\modules\\local\\split_cooler_dump.nf\n",
      "Process COOLTOOLS_EIGSCIS in filepath ./git_repos/hic\\modules\\local\\cooltools\\eigscis.nf\n",
      "Process COOLTOOLS_INSULATION in filepath ./git_repos/hic\\modules\\local\\cooltools\\insulation.nf\n",
      "Process HIC_FIND_TADS in filepath ./git_repos/hic\\modules\\local\\hicexplorer\\hicFindTADs.nf\n",
      "Process HIC_PLOT_DIST_VS_COUNTS in filepath ./git_repos/hic\\modules\\local\\hicexplorer\\hicPlotDistVsCounts.nf\n",
      "Process MERGE_BOWTIE2 in filepath ./git_repos/hic\\modules\\local\\hicpro\\bowtie2_merge.nf\n",
      "Process BUILD_CONTACT_MAPS in filepath ./git_repos/hic\\modules\\local\\hicpro\\build_contact_maps.nf\n",
      "Process COMBINE_MATES in filepath ./git_repos/hic\\modules\\local\\hicpro\\combine_mates.nf\n",
      "Process MAPPING_STATS_DNASE in filepath ./git_repos/hic\\modules\\local\\hicpro\\dnase_mapping_stats.nf\n",
      "Process GET_RESTRICTION_FRAGMENTS in filepath ./git_repos/hic\\modules\\local\\hicpro\\get_restriction_fragments.nf\n",
      "Process GET_VALID_INTERACTION in filepath ./git_repos/hic\\modules\\local\\hicpro\\get_valid_interaction.nf\n",
      "Process GET_VALID_INTERACTION_DNASE in filepath ./git_repos/hic\\modules\\local\\hicpro\\get_valid_interaction_dnase.nf\n",
      "Process HICPRO2PAIRS in filepath ./git_repos/hic\\modules\\local\\hicpro\\hicpro2pairs.nf\n",
      "Process MERGE_STATS in filepath ./git_repos/hic\\modules\\local\\hicpro\\merge_stats.nf\n",
      "Process MERGE_VALID_INTERACTION in filepath ./git_repos/hic\\modules\\local\\hicpro\\merge_valid_interaction.nf\n",
      "Process ICE_NORMALIZATION in filepath ./git_repos/hic\\modules\\local\\hicpro\\run_ice.nf\n",
      "Process TRIM_READS in filepath ./git_repos/hic\\modules\\local\\hicpro\\trim_reads.nf\n",
      "Process BOWTIE2_ALIGN in filepath ./git_repos/hic\\modules\\nf-core\\bowtie2\\align\\main.nf\n",
      "Process BOWTIE2_BUILD in filepath ./git_repos/hic\\modules\\nf-core\\bowtie2\\build\\main.nf\n",
      "Process COOLER_BALANCE in filepath ./git_repos/hic\\modules\\nf-core\\cooler\\balance\\main.nf\n",
      "Process COOLER_CLOAD in filepath ./git_repos/hic\\modules\\nf-core\\cooler\\cload\\main.nf\n",
      "Process COOLER_DUMP in filepath ./git_repos/hic\\modules\\nf-core\\cooler\\dump\\main.nf\n",
      "Process COOLER_MAKEBINS in filepath ./git_repos/hic\\modules\\nf-core\\cooler\\makebins\\main.nf\n",
      "Process COOLER_ZOOMIFY in filepath ./git_repos/hic\\modules\\nf-core\\cooler\\zoomify\\main.nf\n",
      "Process CUSTOM_DUMPSOFTWAREVERSIONS in filepath ./git_repos/hic\\modules\\nf-core\\custom\\dumpsoftwareversions\\main.nf\n",
      "Process CUSTOM_GETCHROMSIZES in filepath ./git_repos/hic\\modules\\nf-core\\custom\\getchromsizes\\main.nf\n",
      "Process FASTQC in filepath ./git_repos/hic\\modules\\nf-core\\fastqc\\main.nf\n",
      "Process CHECKSUMS in filepath ./git_repos/hicar\\modules\\local\\checksums.nf\n",
      "Process GTF2BED in filepath ./git_repos/hicar\\modules\\local\\gtf2bed.nf\n",
      "Process IGV in filepath ./git_repos/hicar\\modules\\local\\igv.nf\n",
      "Process READS_STAT in filepath ./git_repos/hicar\\modules\\local\\reads_stat.nf\n",
      "Process READS_SUMMARY in filepath ./git_repos/hicar\\modules\\local\\reads_summary.nf\n",
      "Process RE_CUTSITE in filepath ./git_repos/hicar\\modules\\local\\re_cut.nf\n",
      "Process ATACQC in filepath ./git_repos/hicar\\modules\\local\\atacreads\\atacqc.nf\n",
      "Process BEDFILES_SORT in filepath ./git_repos/hicar\\modules\\local\\atacreads\\bedsort.nf\n",
      "Process DUMP_READS in filepath ./git_repos/hicar\\modules\\local\\atacreads\\dumpreads.nf\n",
      "Process GET_SCALE in filepath ./git_repos/hicar\\modules\\local\\atacreads\\getscale.nf\n",
      "Process MERGE_PEAK in filepath ./git_repos/hicar\\modules\\local\\atacreads\\mergepeak.nf\n",
      "Process MERGE_READS in filepath ./git_repos/hicar\\modules\\local\\atacreads\\mergereads.nf\n",
      "Process SHIFT_READS in filepath ./git_repos/hicar\\modules\\local\\atacreads\\shiftreads.nf\n",
      "Process BIOC_ATACSEQTFEA in filepath ./git_repos/hicar\\modules\\local\\bioc\\atacseqtfea.nf\n",
      "Process BEDPE2BED in filepath ./git_repos/hicar\\modules\\local\\bioc\\bedpe2bed.nf\n",
      "Process BIOC_CHIPPEAKANNO in filepath ./git_repos/hicar\\modules\\local\\bioc\\chippeakanno.nf\n",
      "Process DIFFHIC in filepath ./git_repos/hicar\\modules\\local\\bioc\\diffhic.nf\n",
      "Process DIFFHICAR in filepath ./git_repos/hicar\\modules\\local\\bioc\\diffhicar.nf\n",
      "Process DIFFSET in filepath ./git_repos/hicar\\modules\\local\\bioc\\diffset.nf\n",
      "Process BIOC_ENRICH in filepath ./git_repos/hicar\\modules\\local\\bioc\\enrich.nf\n",
      "Process ENSEMBL_UCSC_CONVERT in filepath ./git_repos/hicar\\modules\\local\\bioc\\ensembl_ucsc_convert.nf\n",
      "Process BIOC_ENZYMECUT in filepath ./git_repos/hicar\\modules\\local\\bioc\\enzyme_cut.nf\n",
      "Process MERGE_INTERACTIONS in filepath ./git_repos/hicar\\modules\\local\\bioc\\merge_interactions.nf\n",
      "Process PAIR2BAM in filepath ./git_repos/hicar\\modules\\local\\bioc\\pair2bam.nf\n",
      "Process BIOC_PAIRS2HDF5 in filepath ./git_repos/hicar\\modules\\local\\bioc\\pairs2hdf5.nf\n",
      "Process SEQLEVELS_STYLE in filepath ./git_repos/hicar\\modules\\local\\bioc\\seqlevelsstyle.nf\n",
      "Process BIOC_SUBSETLOOPS in filepath ./git_repos/hicar\\modules\\local\\bioc\\subsetloops.nf\n",
      "Process BIOC_TRACKVIEWER in filepath ./git_repos/hicar\\modules\\local\\bioc\\trackviewer.nf\n",
      "Process BEDGRAPH_TRIM in filepath ./git_repos/hicar\\modules\\local\\bioc\\trimbedgraph.nf\n",
      "Process CIRCOS in filepath ./git_repos/hicar\\modules\\local\\circos\\circos.nf\n",
      "Process CIRCOS_PREPARE in filepath ./git_repos/hicar\\modules\\local\\circos\\circos_prepare.nf\n",
      "Process DUMPREADS in filepath ./git_repos/hicar\\modules\\local\\cooler\\dumpreads.nf\n",
      "Process COOLER_LOAD in filepath ./git_repos/hicar\\modules\\local\\cooler\\load\\main.nf\n",
      "Process COOLTOOLS_EIGSCIS in filepath ./git_repos/hicar\\modules\\local\\cooltools\\eigscis.nf\n",
      "Process COOLTOOLS_INSULATION in filepath ./git_repos/hicar\\modules\\local\\cooltools\\insulation.nf\n",
      "Process COOLTOOLS_PILEUP in filepath ./git_repos/hicar\\modules\\local\\cooltools\\pileup.nf\n",
      "Process PLOTNPZ_BY_COOLTOOLS in filepath ./git_repos/hicar\\modules\\local\\cooltools\\plotnpz.nf\n",
      "Process VIRTUAL4C_BY_COOLTOOLS in filepath ./git_repos/hicar\\modules\\local\\cooltools\\virtual4c.nf\n",
      "Process CALL_R1PEAK in filepath ./git_repos/hicar\\modules\\local\\fragmentreads\\call_peak.nf\n",
      "Process R1READS in filepath ./git_repos/hicar\\modules\\local\\fragmentreads\\r1reads.nf\n",
      "Process CHROMSIZES in filepath ./git_repos/hicar\\modules\\local\\genome\\chromsizes.nf\n",
      "Process GENOME_FILTER in filepath ./git_repos/hicar\\modules\\local\\genome\\filter.nf\n",
      "Process HICDCPLUS_CALL_LOOPS in filepath ./git_repos/hicar\\modules\\local\\hicdcplus\\callloops.nf\n",
      "Process HICDCPLUS_FEATURES in filepath ./git_repos/hicar\\modules\\local\\hicdcplus\\features.nf\n",
      "Process HICEXPLORER_CHICAGGREGATESTATISTIC in filepath ./git_repos/hicar\\modules\\local\\hicexplorer\\chicaggregatestatistic.nf\n",
      "Process HICEXPLORER_CHICDIFFERENTIALTEST in filepath ./git_repos/hicar\\modules\\local\\hicexplorer\\chicdifferentialtest.nf\n",
      "Process HICEXPLORER_CHICEXPORTDATA in filepath ./git_repos/hicar\\modules\\local\\hicexplorer\\chicexportdata.nf\n",
      "Process HICEXPLORER_CHICPLOTVIEWPOINT in filepath ./git_repos/hicar\\modules\\local\\hicexplorer\\chicplotviewpoint.nf\n",
      "Process HICEXPLORER_CHICQUALITYCONTROL in filepath ./git_repos/hicar\\modules\\local\\hicexplorer\\chicqualitycontrol.nf\n",
      "Process HICEXPLORER_CHICSIGNIFICANTINTERACTIONS in filepath ./git_repos/hicar\\modules\\local\\hicexplorer\\chicsignificantinteractions.nf\n",
      "Process HICEXPLORER_CHICVIEWPOINT in filepath ./git_repos/hicar\\modules\\local\\hicexplorer\\chicviewpoint.nf\n",
      "Process HICEXPLORER_CHICVIEWPOINTBACKGROUNDMODEL in filepath ./git_repos/hicar\\modules\\local\\hicexplorer\\chicviewpointbackgroundmodel.nf\n",
      "Process HICEXPLORER_HICAGGREGATECONTACTS in filepath ./git_repos/hicar\\modules\\local\\hicexplorer\\hicaggregatecontacts.nf\n",
      "Process HICEXPLORER_HICCORRECTMATRIX in filepath ./git_repos/hicar\\modules\\local\\hicexplorer\\hiccorrectmatrix.nf\n",
      "Process HICEXPLORER_HICFINDTADS in filepath ./git_repos/hicar\\modules\\local\\hicexplorer\\hicfindtads.nf\n",
      "Process HICEXPLORER_HICPCA in filepath ./git_repos/hicar\\modules\\local\\hicexplorer\\hicpca.nf\n",
      "Process HICEXPLORER_HICPLOTMATRIX in filepath ./git_repos/hicar\\modules\\local\\hicexplorer\\hicplotmatrix.nf\n",
      "Process HICEXPLORER_HICPLOTTADS in filepath ./git_repos/hicar\\modules\\local\\hicexplorer\\hicplottads.nf\n",
      "Process HICEXPLORER_HICTRANSFORM in filepath ./git_repos/hicar\\modules\\local\\hicexplorer\\hictransform.nf\n",
      "Process RECENTER_PEAK in filepath ./git_repos/hicar\\modules\\local\\hicexplorer\\recenterpeak.nf\n",
      "Process HICHIPPER_YAML in filepath ./git_repos/hicar\\modules\\local\\hichipper\\create_yaml.nf\n",
      "Process HICHIPPER_HICHIPPER in filepath ./git_repos/hicar\\modules\\local\\hichipper\\hichipper.nf\n",
      "Process ASSIGN_TYPE in filepath ./git_repos/hicar\\modules\\local\\hipeak\\assign_type.nf\n",
      "Process CALL_HIPEAK in filepath ./git_repos/hicar\\modules\\local\\hipeak\\callpeak.nf\n",
      "Process DIFF_HIPEAK in filepath ./git_repos/hicar\\modules\\local\\hipeak\\diff_hipeak.nf\n",
      "Process POST_COUNTS in filepath ./git_repos/hicar\\modules\\local\\hipeak\\post_counts.nf\n",
      "Process PREPARE_COUNTS in filepath ./git_repos/hicar\\modules\\local\\hipeak\\prepare_counts.nf\n",
      "Process DUMP4HOMER in filepath ./git_repos/hicar\\modules\\local\\homer\\dump4homer.nf\n",
      "Process HOMER_FINDHICCOMPARTMENTS in filepath ./git_repos/hicar\\modules\\local\\homer\\find_hic_compartments.nf\n",
      "Process HOMER_FINDMOTIFSGENOME in filepath ./git_repos/hicar\\modules\\local\\homer\\find_motifs_genome.nf\n",
      "Process HOMER_FINDTADSANDLOOPS in filepath ./git_repos/hicar\\modules\\local\\homer\\find_tad_loops.nf\n",
      "Process HOMER_GETHICCORRDIFF in filepath ./git_repos/hicar\\modules\\local\\homer\\get_hic_corr_diff.nf\n",
      "Process HOMER_INSTALL in filepath ./git_repos/hicar\\modules\\local\\homer\\install.nf\n",
      "Process HOMER_RUNHICPCA in filepath ./git_repos/hicar\\modules\\local\\homer\\run_hic_pca.nf\n",
      "Process JUICER_ADDNORM in filepath ./git_repos/hicar\\modules\\local\\juicer\\addnorm.nf\n",
      "Process JUICER_APA in filepath ./git_repos/hicar\\modules\\local\\juicer\\apa.nf\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Process JUICER_EIGENVECTOR in filepath ./git_repos/hicar\\modules\\local\\juicer\\eigenvector.nf\n",
      "Process JUICER_PRE in filepath ./git_repos/hicar\\modules\\local\\juicer\\pre.nf\n",
      "Process MAPS_CALLPEAK in filepath ./git_repos/hicar\\modules\\local\\maps\\callpeak.nf\n",
      "Process MAPS_CUT in filepath ./git_repos/hicar\\modules\\local\\maps\\cut.nf\n",
      "Process MAPS_FEATURE in filepath ./git_repos/hicar\\modules\\local\\maps\\feature.nf\n",
      "Process MAPS_FEND in filepath ./git_repos/hicar\\modules\\local\\maps\\fend.nf\n",
      "Process MAPS_MAPS in filepath ./git_repos/hicar\\modules\\local\\maps\\maps.nf\n",
      "Process MAPS_MERGE in filepath ./git_repos/hicar\\modules\\local\\maps\\merge.nf\n",
      "Process MAPS_RAW2BG2 in filepath ./git_repos/hicar\\modules\\local\\maps\\raw2bg2.nf\n",
      "Process MAPS_REFORMAT in filepath ./git_repos/hicar\\modules\\local\\maps\\reformat.nf\n",
      "Process PAIRSPLOT in filepath ./git_repos/hicar\\modules\\local\\pairix\\pairsplot.nf\n",
      "Process PAIRSQC in filepath ./git_repos/hicar\\modules\\local\\pairix\\pairsqc.nf\n",
      "Process PAIRTOOLS_SAMPLE in filepath ./git_repos/hicar\\modules\\local\\pairtools\\sample.nf\n",
      "Process PAIRTOOLS_STATS in filepath ./git_repos/hicar\\modules\\local\\pairtools\\stats.nf\n",
      "Process PEAKACHU_MODEL in filepath ./git_repos/hicar\\modules\\local\\peakachu\\model.nf\n",
      "Process PEAKACHU_SCORE in filepath ./git_repos/hicar\\modules\\local\\peakachu\\score_genome.nf\n",
      "Process SAMTOOLS_SORT in filepath ./git_repos/hicar\\modules\\local\\samtools\\sort\\main.nf\n",
      "Process BEDTOOLS_GENOMECOV in filepath ./git_repos/hicar\\modules\\nf-core\\bedtools\\genomecov\\main.nf\n",
      "Process BEDTOOLS_SORT in filepath ./git_repos/hicar\\modules\\nf-core\\bedtools\\sort\\main.nf\n",
      "Process BWA_INDEX in filepath ./git_repos/hicar\\modules\\nf-core\\bwa\\index\\main.nf\n",
      "Process BWA_MEM in filepath ./git_repos/hicar\\modules\\nf-core\\bwa\\mem\\main.nf\n",
      "Process CAT_CAT in filepath ./git_repos/hicar\\modules\\nf-core\\cat\\cat\\main.nf\n",
      "Process COOLER_BALANCE in filepath ./git_repos/hicar\\modules\\nf-core\\cooler\\balance\\main.nf\n",
      "Process COOLER_CLOAD in filepath ./git_repos/hicar\\modules\\nf-core\\cooler\\cload\\main.nf\n",
      "Process COOLER_DIGEST in filepath ./git_repos/hicar\\modules\\nf-core\\cooler\\digest\\main.nf\n",
      "Process COOLER_DUMP in filepath ./git_repos/hicar\\modules\\nf-core\\cooler\\dump\\main.nf\n",
      "Process COOLER_MERGE in filepath ./git_repos/hicar\\modules\\nf-core\\cooler\\merge\\main.nf\n",
      "Process COOLER_ZOOMIFY in filepath ./git_repos/hicar\\modules\\nf-core\\cooler\\zoomify\\main.nf\n",
      "Process CUSTOM_DUMPSOFTWAREVERSIONS in filepath ./git_repos/hicar\\modules\\nf-core\\custom\\dumpsoftwareversions\\main.nf\n",
      "Process CUTADAPT in filepath ./git_repos/hicar\\modules\\nf-core\\cutadapt\\main.nf\n",
      "Process FASTQC in filepath ./git_repos/hicar\\modules\\nf-core\\fastqc\\main.nf\n",
      "Process GENMAP_INDEX in filepath ./git_repos/hicar\\modules\\nf-core\\genmap\\index\\main.nf\n",
      "Process GENMAP_MAPPABILITY in filepath ./git_repos/hicar\\modules\\nf-core\\genmap\\mappability\\main.nf\n",
      "Process GFFREAD in filepath ./git_repos/hicar\\modules\\nf-core\\gffread\\main.nf\n",
      "Process GUNZIP in filepath ./git_repos/hicar\\modules\\nf-core\\gunzip\\main.nf\n",
      "Process HOMER_MAKETAGDIRECTORY in filepath ./git_repos/hicar\\modules\\nf-core\\homer\\maketagdirectory\\main.nf\n",
      "Process MACS2_CALLPEAK in filepath ./git_repos/hicar\\modules\\nf-core\\macs2\\callpeak\\main.nf\n",
      "Process MULTIQC in filepath ./git_repos/hicar\\modules\\nf-core\\multiqc\\main.nf\n",
      "Process PAIRIX in filepath ./git_repos/hicar\\modules\\nf-core\\pairix\\main.nf\n",
      "Process PAIRTOOLS_DEDUP in filepath ./git_repos/hicar\\modules\\nf-core\\pairtools\\dedup\\main.nf\n",
      "Process PAIRTOOLS_FLIP in filepath ./git_repos/hicar\\modules\\nf-core\\pairtools\\flip\\main.nf\n",
      "Process PAIRTOOLS_PARSE in filepath ./git_repos/hicar\\modules\\nf-core\\pairtools\\parse\\main.nf\n",
      "Process PAIRTOOLS_RESTRICT in filepath ./git_repos/hicar\\modules\\nf-core\\pairtools\\restrict\\main.nf\n",
      "Process PAIRTOOLS_SELECT in filepath ./git_repos/hicar\\modules\\nf-core\\pairtools\\select\\main.nf\n",
      "Process PAIRTOOLS_SORT in filepath ./git_repos/hicar\\modules\\nf-core\\pairtools\\sort\\main.nf\n",
      "Process SAMTOOLS_FLAGSTAT in filepath ./git_repos/hicar\\modules\\nf-core\\samtools\\flagstat\\main.nf\n",
      "Process SAMTOOLS_IDXSTATS in filepath ./git_repos/hicar\\modules\\nf-core\\samtools\\idxstats\\main.nf\n",
      "Process SAMTOOLS_INDEX in filepath ./git_repos/hicar\\modules\\nf-core\\samtools\\index\\main.nf\n",
      "Process SAMTOOLS_MERGE in filepath ./git_repos/hicar\\modules\\nf-core\\samtools\\merge\\main.nf\n",
      "Process SAMTOOLS_SORT in filepath ./git_repos/hicar\\modules\\nf-core\\samtools\\sort\\main.nf\n",
      "Process SAMTOOLS_STATS in filepath ./git_repos/hicar\\modules\\nf-core\\samtools\\stats\\main.nf\n",
      "Process SAMTOOLS_VIEW in filepath ./git_repos/hicar\\modules\\nf-core\\samtools\\view\\main.nf\n",
      "Process UCSC_BEDCLIP in filepath ./git_repos/hicar\\modules\\nf-core\\ucsc\\bedclip\\main.nf\n",
      "Process UCSC_BEDGRAPHTOBIGWIG in filepath ./git_repos/hicar\\modules\\nf-core\\ucsc\\bedgraphtobigwig\\main.nf\n",
      "Process UCSC_BIGWIGAVERAGEOVERBED in filepath ./git_repos/hicar\\modules\\nf-core\\ucsc\\bigwigaverageoverbed\\main.nf\n",
      "Process UCSC_WIGTOBIGWIG in filepath ./git_repos/hicar\\modules\\nf-core\\ucsc\\wigtobigwig\\main.nf\n",
      "Process CHECK_PAIRED in filepath ./git_repos/hlatyping\\modules\\local\\check_paired.nf\n",
      "Process SAMPLESHEET_CHECK in filepath ./git_repos/hlatyping\\modules\\local\\samplesheet_check.nf\n",
      "Process CUSTOM_DUMPSOFTWAREVERSIONS in filepath ./git_repos/hlatyping\\modules\\nf-core\\custom\\dumpsoftwareversions\\main.nf\n",
      "Process FASTQC in filepath ./git_repos/hlatyping\\modules\\nf-core\\fastqc\\main.nf\n",
      "Process GUNZIP in filepath ./git_repos/hlatyping\\modules\\nf-core\\gunzip\\main.nf\n",
      "Process MULTIQC in filepath ./git_repos/hlatyping\\modules\\nf-core\\multiqc\\main.nf\n",
      "Process OPTITYPE in filepath ./git_repos/hlatyping\\modules\\nf-core\\optitype\\main.nf\n",
      "Process SAMTOOLS_COLLATEFASTQ in filepath ./git_repos/hlatyping\\modules\\nf-core\\samtools\\collatefastq\\main.nf\n",
      "Process SAMTOOLS_VIEW in filepath ./git_repos/hlatyping\\modules\\nf-core\\samtools\\view\\main.nf\n",
      "Process YARA_INDEX in filepath ./git_repos/hlatyping\\modules\\nf-core\\yara\\index\\main.nf\n",
      "Process YARA_MAPPER in filepath ./git_repos/hlatyping\\modules\\nf-core\\yara\\mapper\\main.nf\n",
      "Process IMCTOOLS in filepath ./git_repos/imcyto\\main.nf\n",
      "Process PREPROCESS_FULL_STACK in filepath ./git_repos/imcyto\\main.nf\n",
      "Process PREPROCESS_ILASTIK_STACK in filepath ./git_repos/imcyto\\main.nf\n",
      "Process ILASTIK in filepath ./git_repos/imcyto\\main.nf\n",
      "Process SEGMENTATION in filepath ./git_repos/imcyto\\main.nf\n",
      "Process output_documentation in filepath ./git_repos/imcyto\\main.nf\n",
      "Process get_software_versions in filepath ./git_repos/imcyto\\main.nf\n",
      "Process SAMPLESHEET_CHECK in filepath ./git_repos/isoseq\\modules\\local\\samplesheet_check.nf\n",
      "Process GSTAMA_FILELIST in filepath ./git_repos/isoseq\\modules\\local\\gstama\\filelist\\main.nf\n",
      "Process BAMTOOLS_CONVERT in filepath ./git_repos/isoseq\\modules\\nf-core\\bamtools\\convert\\main.nf\n",
      "Process CUSTOM_DUMPSOFTWAREVERSIONS in filepath ./git_repos/isoseq\\modules\\nf-core\\custom\\dumpsoftwareversions\\main.nf\n",
      "Process GNU_SORT in filepath ./git_repos/isoseq\\modules\\nf-core\\gnu\\sort\\main.nf\n",
      "Process GSTAMA_COLLAPSE in filepath ./git_repos/isoseq\\modules\\nf-core\\gstama\\collapse\\main.nf\n",
      "Process GSTAMA_MERGE in filepath ./git_repos/isoseq\\modules\\nf-core\\gstama\\merge\\main.nf\n",
      "Process GSTAMA_POLYACLEANUP in filepath ./git_repos/isoseq\\modules\\nf-core\\gstama\\polyacleanup\\main.nf\n",
      "Process GUNZIP in filepath ./git_repos/isoseq\\modules\\nf-core\\gunzip\\main.nf\n",
      "Process ISOSEQ3_REFINE in filepath ./git_repos/isoseq\\modules\\nf-core\\isoseq3\\refine\\main.nf\n",
      "Process LIMA in filepath ./git_repos/isoseq\\modules\\nf-core\\lima\\main.nf\n",
      "Process MINIMAP2_ALIGN in filepath ./git_repos/isoseq\\modules\\nf-core\\minimap2\\align\\main.nf\n",
      "Process MULTIQC in filepath ./git_repos/isoseq\\modules\\nf-core\\multiqc\\main.nf\n",
      "Process PBCCS in filepath ./git_repos/isoseq\\modules\\nf-core\\pbccs\\main.nf\n",
      "Process ULTRA_ALIGN in filepath ./git_repos/isoseq\\modules\\nf-core\\ultra\\align\\main.nf\n",
      "Process ULTRA_INDEX in filepath ./git_repos/isoseq\\modules\\nf-core\\ultra\\index\\main.nf\n",
      "Process get_software_versions in filepath ./git_repos/kmermaid\\main.nf\n",
      "Process validate_sketch_value in filepath ./git_repos/kmermaid\\main.nf\n",
      "Process make_protein_index in filepath ./git_repos/kmermaid\\main.nf\n",
      "Process tenx_tgz_extract_bam in filepath ./git_repos/kmermaid\\main.nf\n",
      "Process samtools_fastq_aligned in filepath ./git_repos/kmermaid\\main.nf\n",
      "Process samtools_fastq_unaligned in filepath ./git_repos/kmermaid\\main.nf\n",
      "Process count_umis_per_cell in filepath ./git_repos/kmermaid\\main.nf\n",
      "Process extract_per_cell_fastqs in filepath ./git_repos/kmermaid\\main.nf\n",
      "Process fastp in filepath ./git_repos/kmermaid\\main.nf\n",
      "Process subsample_input in filepath ./git_repos/kmermaid\\main.nf\n",
      "Process sortmerna_index in filepath ./git_repos/kmermaid\\main.nf\n",
      "Process sortmerna in filepath ./git_repos/kmermaid\\main.nf\n",
      "Process translate in filepath ./git_repos/kmermaid\\main.nf\n",
      "Process ska_compute_sketch in filepath ./git_repos/kmermaid\\main.nf\n",
      "Process sourmash_compute_sketch_fastx_nucleotide in filepath ./git_repos/kmermaid\\main.nf\n",
      "Process sourmash_compute_sketch_fastx_peptide in filepath ./git_repos/kmermaid\\main.nf\n",
      "Process sourmash_sig_merge in filepath ./git_repos/kmermaid\\main.nf\n",
      "Process ska_compare_sketches in filepath ./git_repos/kmermaid\\main.nf\n",
      "Process sourmash_compare_sketches in filepath ./git_repos/kmermaid\\main.nf\n",
      "Process multiqc in filepath ./git_repos/kmermaid\\main.nf\n",
      "Process output_documentation in filepath ./git_repos/kmermaid\\main.nf\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Process GET_SOFTWARE_VERSIONS in filepath ./git_repos/liverctanalysis\\modules\\local\\get_software_versions.nf\n",
      "Process SAMPLESHEET_CHECK in filepath ./git_repos/liverctanalysis\\modules\\local\\samplesheet_check.nf\n",
      "Process FASTQC in filepath ./git_repos/liverctanalysis\\modules\\nf-core\\modules\\fastqc\\main.nf\n",
      "Process MULTIQC in filepath ./git_repos/liverctanalysis\\modules\\nf-core\\modules\\multiqc\\main.nf\n",
      "Process combine_public_annotation in filepath ./git_repos/lncpipe\\main.nf\n",
      "Process Make_STARindex in filepath ./git_repos/lncpipe\\main.nf\n",
      "Process Make_bowtie2_index in filepath ./git_repos/lncpipe\\main.nf\n",
      "Process Make_hisat_index in filepath ./git_repos/lncpipe\\main.nf\n",
      "Process Run_fastQC in filepath ./git_repos/lncpipe\\main.nf\n",
      "Process Run_afterQC in filepath ./git_repos/lncpipe\\main.nf\n",
      "Process Run_FastP in filepath ./git_repos/lncpipe\\main.nf\n",
      "Process fastq_star_alignment_For_discovery in filepath ./git_repos/lncpipe\\main.nf\n",
      "Process fastq_tophat_alignment_For_discovery in filepath ./git_repos/lncpipe\\main.nf\n",
      "Process fastq_hisat2_alignment_For_discovery in filepath ./git_repos/lncpipe\\main.nf\n",
      "Process StringTie_assembly in filepath ./git_repos/lncpipe\\main.nf\n",
      "Process StringTie_merge_assembled_gtf in filepath ./git_repos/lncpipe\\main.nf\n",
      "Process Cufflinks_assembly in filepath ./git_repos/lncpipe\\main.nf\n",
      "Process cuffmerge_assembled_gtf in filepath ./git_repos/lncpipe\\main.nf\n",
      "Process Run_fastQC_2 in filepath ./git_repos/lncpipe\\main.nf\n",
      "Process Run_afterQC_2 in filepath ./git_repos/lncpipe\\main.nf\n",
      "Process Run_FastP_2 in filepath ./git_repos/lncpipe\\main.nf\n",
      "Process Merge_assembled_gtf_with_GENCODE in filepath ./git_repos/lncpipe\\main.nf\n",
      "Process Identify_novel_lncRNA_with_criterions in filepath ./git_repos/lncpipe\\main.nf\n",
      "Process Predict_coding_abilities_by_PLEK in filepath ./git_repos/lncpipe\\main.nf\n",
      "Process Predict_coding_abilities_by_CPAT in filepath ./git_repos/lncpipe\\main.nf\n",
      "Process Filter_lncRNA_by_coding_potential_result in filepath ./git_repos/lncpipe\\main.nf\n",
      "Process Summary_renaming_and_classification in filepath ./git_repos/lncpipe\\main.nf\n",
      "Process Rerun_CPAT_to_evaluate_lncRNA in filepath ./git_repos/lncpipe\\main.nf\n",
      "Process Rerun_CPAT_to_evaluate_coding in filepath ./git_repos/lncpipe\\main.nf\n",
      "Process Secondary_basic_statistic in filepath ./git_repos/lncpipe\\main.nf\n",
      "Process Run_htseq_for_quantification in filepath ./git_repos/lncpipe\\main.nf\n",
      "Process Build_kallisto_index_of_GTF_for_quantification in filepath ./git_repos/lncpipe\\main.nf\n",
      "Process Run_kallisto_for_quantification in filepath ./git_repos/lncpipe\\main.nf\n",
      "Process Build_kallisto_index_of_GTF_for_quantification_2 in filepath ./git_repos/lncpipe\\main.nf\n",
      "Process Run_kallisto_for_quantification_2 in filepath ./git_repos/lncpipe\\main.nf\n",
      "Process Get_HTseq_matrix in filepath ./git_repos/lncpipe\\main.nf\n",
      "Process Get_kallisto_matrix in filepath ./git_repos/lncpipe\\main.nf\n",
      "Process Run_LncPipeReporter in filepath ./git_repos/lncpipe\\main.nf\n",
      "Process Run_LncPipeReporter_2 in filepath ./git_repos/lncpipe\\main.nf\n",
      "Process Run_LncPipeReporter_without_Design in filepath ./git_repos/lncpipe\\main.nf\n",
      "Process Run_LncPipeReporter_without_Design_2 in filepath ./git_repos/lncpipe\\main.nf\n",
      "Process ADJUST_MAXBIN2_EXT in filepath ./git_repos/mag\\modules\\local\\adjust_maxbin2_ext.nf\n",
      "Process BIN_SUMMARY in filepath ./git_repos/mag\\modules\\local\\bin_summary.nf\n",
      "Process BOWTIE2_ASSEMBLY_ALIGN in filepath ./git_repos/mag\\modules\\local\\bowtie2_assembly_align.nf\n",
      "Process BOWTIE2_ASSEMBLY_BUILD in filepath ./git_repos/mag\\modules\\local\\bowtie2_assembly_build.nf\n",
      "Process BOWTIE2_REMOVAL_ALIGN in filepath ./git_repos/mag\\modules\\local\\bowtie2_removal_align.nf\n",
      "Process BOWTIE2_REMOVAL_BUILD in filepath ./git_repos/mag\\modules\\local\\bowtie2_removal_build.nf\n",
      "Process BUSCO in filepath ./git_repos/mag\\modules\\local\\busco.nf\n",
      "Process BUSCO_DB_PREPARATION in filepath ./git_repos/mag\\modules\\local\\busco_db_preparation.nf\n",
      "Process BUSCO_SAVE_DOWNLOAD in filepath ./git_repos/mag\\modules\\local\\busco_save_download.nf\n",
      "Process BUSCO_SUMMARY in filepath ./git_repos/mag\\modules\\local\\busco_summary.nf\n",
      "Process CAT in filepath ./git_repos/mag\\modules\\local\\cat.nf\n",
      "Process CAT_DB in filepath ./git_repos/mag\\modules\\local\\cat_db.nf\n",
      "Process CAT_DB_GENERATE in filepath ./git_repos/mag\\modules\\local\\cat_db_generate.nf\n",
      "Process CAT_SUMMARY in filepath ./git_repos/mag\\modules\\local\\cat_summary.nf\n",
      "Process CENTRIFUGE in filepath ./git_repos/mag\\modules\\local\\centrifuge.nf\n",
      "Process CENTRIFUGE_DB_PREPARATION in filepath ./git_repos/mag\\modules\\local\\centrifuge_db_preparation.nf\n",
      "Process COMBINE_TSV in filepath ./git_repos/mag\\modules\\local\\combine_tsv.nf\n",
      "Process CONVERT_DEPTHS in filepath ./git_repos/mag\\modules\\local\\convert_depths.nf\n",
      "Process FILTLONG in filepath ./git_repos/mag\\modules\\local\\filtlong.nf\n",
      "Process GTDBTK_DB_PREPARATION in filepath ./git_repos/mag\\modules\\local\\gtdbtk_db_preparation.nf\n",
      "Process GTDBTK_SUMMARY in filepath ./git_repos/mag\\modules\\local\\gtdbtk_summary.nf\n",
      "Process KRAKEN2 in filepath ./git_repos/mag\\modules\\local\\kraken2.nf\n",
      "Process KRAKEN2_DB_PREPARATION in filepath ./git_repos/mag\\modules\\local\\kraken2_db_preparation.nf\n",
      "Process KRONA in filepath ./git_repos/mag\\modules\\local\\krona.nf\n",
      "Process KRONA_DB in filepath ./git_repos/mag\\modules\\local\\krona_db.nf\n",
      "Process MAG_DEPTHS in filepath ./git_repos/mag\\modules\\local\\mag_depths.nf\n",
      "Process MAG_DEPTHS_PLOT in filepath ./git_repos/mag\\modules\\local\\mag_depths_plot.nf\n",
      "Process MAG_DEPTHS_SUMMARY in filepath ./git_repos/mag\\modules\\local\\mag_depths_summary.nf\n",
      "Process MEGAHIT in filepath ./git_repos/mag\\modules\\local\\megahit.nf\n",
      "Process NANOLYSE in filepath ./git_repos/mag\\modules\\local\\nanolyse.nf\n",
      "Process NANOPLOT in filepath ./git_repos/mag\\modules\\local\\nanoplot.nf\n",
      "Process POOL_PAIRED_READS in filepath ./git_repos/mag\\modules\\local\\pool_paired_reads.nf\n",
      "Process POOL_SINGLE_READS in filepath ./git_repos/mag\\modules\\local\\pool_single_reads.nf\n",
      "Process PORECHOP in filepath ./git_repos/mag\\modules\\local\\porechop.nf\n",
      "Process QUAST in filepath ./git_repos/mag\\modules\\local\\quast.nf\n",
      "Process QUAST_BINS in filepath ./git_repos/mag\\modules\\local\\quast_bins.nf\n",
      "Process QUAST_BINS_SUMMARY in filepath ./git_repos/mag\\modules\\local\\quast_bins_summary.nf\n",
      "Process RENAME_POSTDASTOOL in filepath ./git_repos/mag\\modules\\local\\rename_postdastool.nf\n",
      "Process RENAME_PREDASTOOL in filepath ./git_repos/mag\\modules\\local\\rename_predastool.nf\n",
      "Process SPADES in filepath ./git_repos/mag\\modules\\local\\spades.nf\n",
      "Process SPADESHYBRID in filepath ./git_repos/mag\\modules\\local\\spadeshybrid.nf\n",
      "Process SPLIT_FASTA in filepath ./git_repos/mag\\modules\\local\\split_fasta.nf\n",
      "Process TIARA_CLASSIFY in filepath ./git_repos/mag\\modules\\local\\tiara_classify.nf\n",
      "Process ADAPTERREMOVAL in filepath ./git_repos/mag\\modules\\nf-core\\adapterremoval\\main.nf\n",
      "Process ARIA2 in filepath ./git_repos/mag\\modules\\nf-core\\aria2\\main.nf\n",
      "Process BBMAP_BBNORM in filepath ./git_repos/mag\\modules\\nf-core\\bbmap\\bbnorm\\main.nf\n",
      "Process BCFTOOLS_CONSENSUS in filepath ./git_repos/mag\\modules\\nf-core\\bcftools\\consensus\\main.nf\n",
      "Process BCFTOOLS_INDEX in filepath ./git_repos/mag\\modules\\nf-core\\bcftools\\index\\main.nf\n",
      "Process BCFTOOLS_VIEW in filepath ./git_repos/mag\\modules\\nf-core\\bcftools\\view\\main.nf\n",
      "Process CAT_FASTQ in filepath ./git_repos/mag\\modules\\nf-core\\cat\\fastq\\main.nf\n",
      "Process CHECKM_LINEAGEWF in filepath ./git_repos/mag\\modules\\nf-core\\checkm\\lineagewf\\main.nf\n",
      "Process CHECKM_QA in filepath ./git_repos/mag\\modules\\nf-core\\checkm\\qa\\main.nf\n",
      "Process CONCOCT_CONCOCT in filepath ./git_repos/mag\\modules\\nf-core\\concoct\\concoct\\main.nf\n",
      "Process CONCOCT_CONCOCTCOVERAGETABLE in filepath ./git_repos/mag\\modules\\nf-core\\concoct\\concoctcoveragetable\\main.nf\n",
      "Process CONCOCT_CUTUPFASTA in filepath ./git_repos/mag\\modules\\nf-core\\concoct\\cutupfasta\\main.nf\n",
      "Process CONCOCT_EXTRACTFASTABINS in filepath ./git_repos/mag\\modules\\nf-core\\concoct\\extractfastabins\\main.nf\n",
      "Process CONCOCT_MERGECUTUPCLUSTERING in filepath ./git_repos/mag\\modules\\nf-core\\concoct\\mergecutupclustering\\main.nf\n",
      "Process CUSTOM_DUMPSOFTWAREVERSIONS in filepath ./git_repos/mag\\modules\\nf-core\\custom\\dumpsoftwareversions\\main.nf\n",
      "Process DASTOOL_DASTOOL in filepath ./git_repos/mag\\modules\\nf-core\\dastool\\dastool\\main.nf\n",
      "Process DASTOOL_FASTATOCONTIG2BIN in filepath ./git_repos/mag\\modules\\nf-core\\dastool\\fastatocontig2bin\\main.nf\n",
      "Process FASTP in filepath ./git_repos/mag\\modules\\nf-core\\fastp\\main.nf\n",
      "Process FASTQC in filepath ./git_repos/mag\\modules\\nf-core\\fastqc\\main.nf\n",
      "Process FREEBAYES in filepath ./git_repos/mag\\modules\\nf-core\\freebayes\\main.nf\n",
      "Process GENOMAD_DOWNLOAD in filepath ./git_repos/mag\\modules\\nf-core\\genomad\\download\\main.nf\n",
      "Process GENOMAD_ENDTOEND in filepath ./git_repos/mag\\modules\\nf-core\\genomad\\endtoend\\main.nf\n",
      "Process GTDBTK_CLASSIFYWF in filepath ./git_repos/mag\\modules\\nf-core\\gtdbtk\\classifywf\\main.nf\n",
      "Process GUNC_DOWNLOADDB in filepath ./git_repos/mag\\modules\\nf-core\\gunc\\downloaddb\\main.nf\n",
      "Process GUNC_MERGECHECKM in filepath ./git_repos/mag\\modules\\nf-core\\gunc\\mergecheckm\\main.nf\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Process GUNC_RUN in filepath ./git_repos/mag\\modules\\nf-core\\gunc\\run\\main.nf\n",
      "Process GUNZIP in filepath ./git_repos/mag\\modules\\nf-core\\gunzip\\main.nf\n",
      "Process MAXBIN2 in filepath ./git_repos/mag\\modules\\nf-core\\maxbin2\\main.nf\n",
      "Process METABAT2_JGISUMMARIZEBAMCONTIGDEPTHS in filepath ./git_repos/mag\\modules\\nf-core\\metabat2\\jgisummarizebamcontigdepths\\main.nf\n",
      "Process METABAT2_METABAT2 in filepath ./git_repos/mag\\modules\\nf-core\\metabat2\\metabat2\\main.nf\n",
      "Process METAEUK_EASYPREDICT in filepath ./git_repos/mag\\modules\\nf-core\\metaeuk\\easypredict\\main.nf\n",
      "Process MMSEQS_DATABASES in filepath ./git_repos/mag\\modules\\nf-core\\mmseqs\\databases\\main.nf\n",
      "Process MULTIQC in filepath ./git_repos/mag\\modules\\nf-core\\multiqc\\main.nf\n",
      "Process PRODIGAL in filepath ./git_repos/mag\\modules\\nf-core\\prodigal\\main.nf\n",
      "Process PROKKA in filepath ./git_repos/mag\\modules\\nf-core\\prokka\\main.nf\n",
      "Process PYDAMAGE_ANALYZE in filepath ./git_repos/mag\\modules\\nf-core\\pydamage\\analyze\\main.nf\n",
      "Process PYDAMAGE_FILTER in filepath ./git_repos/mag\\modules\\nf-core\\pydamage\\filter\\main.nf\n",
      "Process SAMTOOLS_FAIDX in filepath ./git_repos/mag\\modules\\nf-core\\samtools\\faidx\\main.nf\n",
      "Process SEQTK_MERGEPE in filepath ./git_repos/mag\\modules\\nf-core\\seqtk\\mergepe\\main.nf\n",
      "Process TIARA_TIARA in filepath ./git_repos/mag\\modules\\nf-core\\tiara\\tiara\\main.nf\n",
      "Process SAMPLESHEET_CHECK in filepath ./git_repos/marsseq\\modules\\local\\samplesheet_check.nf\n",
      "Process CUT_SAM in filepath ./git_repos/marsseq\\modules\\local\\cut\\sam\\main.nf\n",
      "Process DEMULTIPLEX in filepath ./git_repos/marsseq\\modules\\local\\demultiplex\\main.nf\n",
      "Process ERCC_CREATE in filepath ./git_repos/marsseq\\modules\\local\\ercc\\main.nf\n",
      "Process EXTRACT_LABELS in filepath ./git_repos/marsseq\\modules\\local\\extract\\main.nf\n",
      "Process FASTP_SPLIT in filepath ./git_repos/marsseq\\modules\\local\\fastp\\split\\main.nf\n",
      "Process PREPARE in filepath ./git_repos/marsseq\\modules\\local\\prepare\\main.nf\n",
      "Process QC_ALIGNED in filepath ./git_repos/marsseq\\modules\\local\\qc\\align\\main.nf\n",
      "Process QC_BATCH in filepath ./git_repos/marsseq\\modules\\local\\qc\\batch\\main.nf\n",
      "Process QC_REPORT in filepath ./git_repos/marsseq\\modules\\local\\qc\\report\\main.nf\n",
      "Process VELOCITY_CONVERT in filepath ./git_repos/marsseq\\modules\\local\\velocity\\convert\\main.nf\n",
      "Process VELOCITY_WHITELIST in filepath ./git_repos/marsseq\\modules\\local\\velocity\\whitelist\\main.nf\n",
      "Process WGET in filepath ./git_repos/marsseq\\modules\\local\\wget\\main.nf\n",
      "Process BOWTIE2_ALIGN in filepath ./git_repos/marsseq\\modules\\nf-core\\bowtie2\\align\\main.nf\n",
      "Process BOWTIE2_BUILD in filepath ./git_repos/marsseq\\modules\\nf-core\\bowtie2\\build\\main.nf\n",
      "Process CAT_CAT in filepath ./git_repos/marsseq\\modules\\nf-core\\cat\\cat\\main.nf\n",
      "Process CUSTOM_DUMPSOFTWAREVERSIONS in filepath ./git_repos/marsseq\\modules\\nf-core\\custom\\dumpsoftwareversions\\main.nf\n",
      "Process CUTADAPT in filepath ./git_repos/marsseq\\modules\\nf-core\\cutadapt\\main.nf\n",
      "Process FASTQC in filepath ./git_repos/marsseq\\modules\\nf-core\\fastqc\\main.nf\n",
      "Process GUNZIP in filepath ./git_repos/marsseq\\modules\\nf-core\\gunzip\\main.nf\n",
      "Process MULTIQC in filepath ./git_repos/marsseq\\modules\\nf-core\\multiqc\\main.nf\n",
      "Process STAR_ALIGN in filepath ./git_repos/marsseq\\modules\\nf-core\\star\\align\\main.nf\n",
      "Process STAR_GENOMEGENERATE in filepath ./git_repos/marsseq\\modules\\nf-core\\star\\genomegenerate\\main.nf\n",
      "Process SAMPLESHEET_CHECK in filepath ./git_repos/mcmicro\\modules\\local\\samplesheet_check.nf\n",
      "Process ASHLAR in filepath ./git_repos/mcmicro\\modules\\nf-core\\ashlar\\main.nf\n",
      "Process BACKSUB in filepath ./git_repos/mcmicro\\modules\\nf-core\\backsub\\main.nf\n",
      "Process BASICPY in filepath ./git_repos/mcmicro\\modules\\nf-core\\basicpy\\main.nf\n",
      "Process CELLPOSE in filepath ./git_repos/mcmicro\\modules\\nf-core\\cellpose\\main.nf\n",
      "Process CUSTOM_DUMPSOFTWAREVERSIONS in filepath ./git_repos/mcmicro\\modules\\nf-core\\custom\\dumpsoftwareversions\\main.nf\n",
      "Process DEEPCELL_MESMER in filepath ./git_repos/mcmicro\\modules\\nf-core\\deepcell\\mesmer\\main.nf\n",
      "Process FASTQC in filepath ./git_repos/mcmicro\\modules\\nf-core\\fastqc\\main.nf\n",
      "Process MCQUANT in filepath ./git_repos/mcmicro\\modules\\nf-core\\mcquant\\main.nf\n",
      "Process MULTIQC in filepath ./git_repos/mcmicro\\modules\\nf-core\\multiqc\\main.nf\n",
      "Process SCIMAP_MCMICRO in filepath ./git_repos/mcmicro\\modules\\nf-core\\scimap\\mcmicro\\main.nf\n",
      "Process GENERAL_MERGEFILE in filepath ./git_repos/metaboigniter\\modules\\local\\general_mergefile.nf\n",
      "Process MS2QUERY_CHECKMODELFILES in filepath ./git_repos/metaboigniter\\modules\\local\\ms2query_checkmodelfiles.nf\n",
      "Process MS2QUERY_DOWNLOADMODEL in filepath ./git_repos/metaboigniter\\modules\\local\\ms2query_downloadmodel.nf\n",
      "Process MS2QUERY_EXPORT in filepath ./git_repos/metaboigniter\\modules\\local\\ms2query_export.nf\n",
      "Process MS2QUERY_MODELFULLTRAIN in filepath ./git_repos/metaboigniter\\modules\\local\\ms2query_modelfulltrain.nf\n",
      "Process MS2QUERY_MODELTRAIN in filepath ./git_repos/metaboigniter\\modules\\local\\ms2query_modeltrain.nf\n",
      "Process MS2QUERY_SEARCH in filepath ./git_repos/metaboigniter\\modules\\local\\ms2query_search.nf\n",
      "Process OPENMS_FEATUREFINDERMETABO in filepath ./git_repos/metaboigniter\\modules\\local\\openms_featurefindermetabo.nf\n",
      "Process OPENMS_FEATUREFINDERMETABOIDENT in filepath ./git_repos/metaboigniter\\modules\\local\\openms_featurefindermetaboident.nf\n",
      "Process OPENMS_FEATURELINKERUNLABELEDKD in filepath ./git_repos/metaboigniter\\modules\\local\\openms_featurelinkerunlabeledkd.nf\n",
      "Process OPENMS_FILEFILTER in filepath ./git_repos/metaboigniter\\modules\\local\\openms_filefilter.nf\n",
      "Process OPENMS_FILEFILTERMS2 in filepath ./git_repos/metaboigniter\\modules\\local\\openms_filefilter.nf\n",
      "Process OPENMS_GNPSEXPORT in filepath ./git_repos/metaboigniter\\modules\\local\\openms_gnpsexport.nf\n",
      "Process OPENMS_IDMAPPER in filepath ./git_repos/metaboigniter\\modules\\local\\openms_idmapper.nf\n",
      "Process OPENMS_IDMAPPERCONSENSUS in filepath ./git_repos/metaboigniter\\modules\\local\\openms_idmapperconsensus.nf\n",
      "Process OPENMS_MAPALIGNERPOSECLUSTERING in filepath ./git_repos/metaboigniter\\modules\\local\\openms_mapalignerposeclustering.nf\n",
      "Process OPENMS_MAPALIGNERPOSECLUSTERINGMZML in filepath ./git_repos/metaboigniter\\modules\\local\\openms_mapalignerposeclusteringmzml.nf\n",
      "Process OPENMS_MAPRTTRANSFORMER in filepath ./git_repos/metaboigniter\\modules\\local\\openms_maprttransformer.nf\n",
      "Process OPENMS_METABOLITEADDUCTDECHARGER in filepath ./git_repos/metaboigniter\\modules\\local\\openms_metaboliteadductdecharger.nf\n",
      "Process OPENMS_PEAKPICKERHIRES in filepath ./git_repos/metaboigniter\\modules\\local\\openms_peakpickerhires.nf\n",
      "Process OPENMS_SIRIUSPARAMS in filepath ./git_repos/metaboigniter\\modules\\local\\openms_siriusparams.nf\n",
      "Process PYOPENMS_C13DETECTION in filepath ./git_repos/metaboigniter\\modules\\local\\pyopenms_c13detection.nf\n",
      "Process PYOPENMS_CALCULATEBOUNDRIES in filepath ./git_repos/metaboigniter\\modules\\local\\pyopenms_calculateboundries.nf\n",
      "Process PYOPENMS_CONCTSV in filepath ./git_repos/metaboigniter\\modules\\local\\pyopenms_conctsv.nf\n",
      "Process PYOPENMS_EXPORT in filepath ./git_repos/metaboigniter\\modules\\local\\pyopenms_export.nf\n",
      "Process PYOPENMS_EXTRACTFEATUREMZ in filepath ./git_repos/metaboigniter\\modules\\local\\pyopenms_extractfeaturemz.nf\n",
      "Process PYOPENMS_EXTRACTMAPPING in filepath ./git_repos/metaboigniter\\modules\\local\\pyopenms_extractmapping.nf\n",
      "Process PYOPENMS_FILEMERGER in filepath ./git_repos/metaboigniter\\modules\\local\\pyopenms_filemerger.nf\n",
      "Process PYOPENMS_GENERATESEARCHPARAMS in filepath ./git_repos/metaboigniter\\modules\\local\\pyopenms_generatesearchparams.nf\n",
      "Process PYOPENMS_GENERATESEARCHPARAMSUNMAPPED in filepath ./git_repos/metaboigniter\\modules\\local\\pyopenms_generatesearchparamsunmapped.nf\n",
      "Process PYOPENMS_ISOMAPPING in filepath ./git_repos/metaboigniter\\modules\\local\\pyopenms_isomapping.nf\n",
      "Process PYOPENMS_LIBBUILDER in filepath ./git_repos/metaboigniter\\modules\\local\\pyopenms_libbuilder.nf\n",
      "Process PYOPENMS_MERGE in filepath ./git_repos/metaboigniter\\modules\\local\\pyopenms_merge.nf\n",
      "Process PYOPENMS_MSMAPPING in filepath ./git_repos/metaboigniter\\modules\\local\\pyopenms_msmapping.nf\n",
      "Process PYOPENMS_RELOADMAPS in filepath ./git_repos/metaboigniter\\modules\\local\\pyopenms_reloadmaps.nf\n",
      "Process PYOPENMS_REMOVEDUP in filepath ./git_repos/metaboigniter\\modules\\local\\pyopenms_removedup.nf\n",
      "Process PYOPENMS_SPLIT in filepath ./git_repos/metaboigniter\\modules\\local\\pyopenms_split.nf\n",
      "Process PYOPENMS_SPLITCONSENSUS in filepath ./git_repos/metaboigniter\\modules\\local\\pyopenms_splitconsensus.nf\n",
      "Process PYOPENMS_SPLITFEATUREXML in filepath ./git_repos/metaboigniter\\modules\\local\\pyopenms_splitfeaturexml.nf\n",
      "Process PYOPENMS_SPLITMGF in filepath ./git_repos/metaboigniter\\modules\\local\\pyopenms_splitmgf.nf\n",
      "Process SIRIUS_SEARCH in filepath ./git_repos/metaboigniter\\modules\\local\\sirius_search.nf\n",
      "Process CUSTOM_DUMPSOFTWAREVERSIONS in filepath ./git_repos/metaboigniter\\modules\\nf-core\\custom\\dumpsoftwareversions\\main.nf\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Process ASSIGN_NUCL_ENTITY_WEIGHTS in filepath ./git_repos/metapep\\modules\\local\\assign_nucl_entity_weights.nf\n",
      "Process CHECK_SAMPLESHEET_CREATE_TABLES in filepath ./git_repos/metapep\\modules\\local\\check_samplesheet_create_tables.nf\n",
      "Process COLLECT_STATS in filepath ./git_repos/metapep\\modules\\local\\collect_stats.nf\n",
      "Process CREATE_PROTEIN_TSV in filepath ./git_repos/metapep\\modules\\local\\create_protein_tsv.nf\n",
      "Process DOWNLOAD_PROTEINS in filepath ./git_repos/metapep\\modules\\local\\download_proteins.nf\n",
      "Process EPYTOPE_SHOW_SUPPORTED_MODELS in filepath ./git_repos/metapep\\modules\\local\\epytope_show_supported_models.nf\n",
      "Process FINALIZE_MICROBIOME_ENTITIES in filepath ./git_repos/metapep\\modules\\local\\finalize_microbiome_entities.nf\n",
      "Process GENERATE_PEPTIDES in filepath ./git_repos/metapep\\modules\\local\\generate_peptides.nf\n",
      "Process GENERATE_PROTEIN_AND_ENTITY_IDS in filepath ./git_repos/metapep\\modules\\local\\generate_protein_and_entity_ids.nf\n",
      "Process MERGE_PREDICTIONS in filepath ./git_repos/metapep\\modules\\local\\merge_predictions.nf\n",
      "Process MERGE_PREDICTIONS_BUFFER in filepath ./git_repos/metapep\\modules\\local\\merge_predictions_buffer.nf\n",
      "Process PLOT_ENTITY_BINDING_RATIOS in filepath ./git_repos/metapep\\modules\\local\\plot_entity_binding_ratios.nf\n",
      "Process PLOT_SCORE_DISTRIBUTION in filepath ./git_repos/metapep\\modules\\local\\plot_score_distribution.nf\n",
      "Process PREDICT_EPITOPES in filepath ./git_repos/metapep\\modules\\local\\predict_epitopes.nf\n",
      "Process PREPARE_ENTITY_BINDING_RATIOS in filepath ./git_repos/metapep\\modules\\local\\prepare_entity_binding_ratios.nf\n",
      "Process PREPARE_SCORE_DISTRIBUTION in filepath ./git_repos/metapep\\modules\\local\\prepare_score_distribution.nf\n",
      "Process SPLIT_PRED_TASKS in filepath ./git_repos/metapep\\modules\\local\\split_pred_tasks.nf\n",
      "Process UNIFY_MODEL_LENGTHS in filepath ./git_repos/metapep\\modules\\local\\unify_model_lengths.nf\n",
      "Process UNPACK_BIN_ARCHIVES in filepath ./git_repos/metapep\\modules\\local\\unpack_bin_archives.nf\n",
      "Process CUSTOM_DUMPSOFTWAREVERSIONS in filepath ./git_repos/metapep\\modules\\nf-core\\custom\\dumpsoftwareversions\\main.nf\n",
      "Process MULTIQC in filepath ./git_repos/metapep\\modules\\nf-core\\multiqc\\main.nf\n",
      "Process PRODIGAL in filepath ./git_repos/metapep\\modules\\nf-core\\prodigal\\main.nf\n",
      "Process COLLECT_FEATURECOUNTS in filepath ./git_repos/metatdenovo\\modules\\local\\collect_featurecounts.nf\n",
      "Process COLLECT_STATS in filepath ./git_repos/metatdenovo\\modules\\local\\collect_stats.nf\n",
      "Process FORMATSPADES in filepath ./git_repos/metatdenovo\\modules\\local\\formatspades.nf\n",
      "Process FORMAT_PRODIGAL_GFF in filepath ./git_repos/metatdenovo\\modules\\local\\format_prodigal.nf\n",
      "Process FORMAT_TAX in filepath ./git_repos/metatdenovo\\modules\\local\\format_tax.nf\n",
      "Process HMMRANK in filepath ./git_repos/metatdenovo\\modules\\local\\hmmrank.nf\n",
      "Process MERGE_TABLES in filepath ./git_repos/metatdenovo\\modules\\local\\merge_summary_tables.nf\n",
      "Process PROKKAGFF2TSV in filepath ./git_repos/metatdenovo\\modules\\local\\prokkagff2tsv.nf\n",
      "Process SAMPLESHEET_CHECK in filepath ./git_repos/metatdenovo\\modules\\local\\samplesheet_check.nf\n",
      "Process SUM_KOFAMSCAN in filepath ./git_repos/metatdenovo\\modules\\local\\sum_kofamscan.nf\n",
      "Process SUM_TAXONOMY in filepath ./git_repos/metatdenovo\\modules\\local\\sum_taxonomy.nf\n",
      "Process TRANSRATE in filepath ./git_repos/metatdenovo\\modules\\local\\transrate.nf\n",
      "Process UNPIGZ in filepath ./git_repos/metatdenovo\\modules\\local\\unpigz.nf\n",
      "Process WRITESPADESYAML in filepath ./git_repos/metatdenovo\\modules\\local\\writespadesyaml.nf\n",
      "Process CAT_CONTIGS in filepath ./git_repos/metatdenovo\\modules\\local\\cat\\cat_contigs.nf\n",
      "Process CAT_DB in filepath ./git_repos/metatdenovo\\modules\\local\\cat\\cat_db.nf\n",
      "Process CAT_DB_GENERATE in filepath ./git_repos/metatdenovo\\modules\\local\\cat\\cat_db_generate.nf\n",
      "Process CAT_SUMMARY in filepath ./git_repos/metatdenovo\\modules\\local\\cat\\cat_summary.nf\n",
      "Process EGGNOG_DOWNLOAD in filepath ./git_repos/metatdenovo\\modules\\local\\eggnog\\download.nf\n",
      "Process EGGNOG_MAPPER in filepath ./git_repos/metatdenovo\\modules\\local\\eggnog\\mapper.nf\n",
      "Process EGGNOG_SUM in filepath ./git_repos/metatdenovo\\modules\\local\\eggnog\\sum.nf\n",
      "Process EUKULELE_SEARCH in filepath ./git_repos/metatdenovo\\modules\\local\\eukulele\\search.nf\n",
      "Process KOFAMSCAN_DOWNLOAD in filepath ./git_repos/metatdenovo\\modules\\local\\kofamscan\\download.nf\n",
      "Process KOFAMSCAN_SCAN in filepath ./git_repos/metatdenovo\\modules\\local\\kofamscan\\scan.nf\n",
      "Process MEGAHIT_INTERLEAVED in filepath ./git_repos/metatdenovo\\modules\\local\\megahit\\interleaved.nf\n",
      "Process BBMAP_ALIGN in filepath ./git_repos/metatdenovo\\modules\\nf-core\\bbmap\\align\\main.nf\n",
      "Process BBMAP_BBDUK in filepath ./git_repos/metatdenovo\\modules\\nf-core\\bbmap\\bbduk\\main.nf\n",
      "Process BBMAP_BBNORM in filepath ./git_repos/metatdenovo\\modules\\nf-core\\bbmap\\bbnorm\\main.nf\n",
      "Process BBMAP_INDEX in filepath ./git_repos/metatdenovo\\modules\\nf-core\\bbmap\\index\\main.nf\n",
      "Process CAT_CAT in filepath ./git_repos/metatdenovo\\modules\\nf-core\\cat\\cat\\main.nf\n",
      "Process CAT_FASTQ in filepath ./git_repos/metatdenovo\\modules\\nf-core\\cat\\fastq\\main.nf\n",
      "Process CUSTOM_DUMPSOFTWAREVERSIONS in filepath ./git_repos/metatdenovo\\modules\\nf-core\\custom\\dumpsoftwareversions\\main.nf\n",
      "Process FASTQC in filepath ./git_repos/metatdenovo\\modules\\nf-core\\fastqc\\main.nf\n",
      "Process HMMER_HMMSEARCH in filepath ./git_repos/metatdenovo\\modules\\nf-core\\hmmer\\hmmsearch\\main.nf\n",
      "Process MULTIQC in filepath ./git_repos/metatdenovo\\modules\\nf-core\\multiqc\\main.nf\n",
      "Process PRODIGAL in filepath ./git_repos/metatdenovo\\modules\\nf-core\\prodigal\\main.nf\n",
      "Process PROKKA in filepath ./git_repos/metatdenovo\\modules\\nf-core\\prokka\\main.nf\n",
      "Process SAMTOOLS_FLAGSTAT in filepath ./git_repos/metatdenovo\\modules\\nf-core\\samtools\\flagstat\\main.nf\n",
      "Process SAMTOOLS_IDXSTATS in filepath ./git_repos/metatdenovo\\modules\\nf-core\\samtools\\idxstats\\main.nf\n",
      "Process SAMTOOLS_INDEX in filepath ./git_repos/metatdenovo\\modules\\nf-core\\samtools\\index\\main.nf\n",
      "Process SAMTOOLS_SORT in filepath ./git_repos/metatdenovo\\modules\\nf-core\\samtools\\sort\\main.nf\n",
      "Process SAMTOOLS_STATS in filepath ./git_repos/metatdenovo\\modules\\nf-core\\samtools\\stats\\main.nf\n",
      "Process SEQTK_MERGEPE in filepath ./git_repos/metatdenovo\\modules\\nf-core\\seqtk\\mergepe\\main.nf\n",
      "Process SEQTK_SEQ in filepath ./git_repos/metatdenovo\\modules\\nf-core\\seqtk\\seq\\main.nf\n",
      "Process SPADES in filepath ./git_repos/metatdenovo\\modules\\nf-core\\spades\\main.nf\n",
      "Process SUBREAD_FEATURECOUNTS in filepath ./git_repos/metatdenovo\\modules\\nf-core\\subread\\featurecounts\\main.nf\n",
      "Process TRANSDECODER_LONGORF in filepath ./git_repos/metatdenovo\\modules\\nf-core\\transdecoder\\longorf\\main.nf\n",
      "Process TRANSDECODER_PREDICT in filepath ./git_repos/metatdenovo\\modules\\nf-core\\transdecoder\\predict\\main.nf\n",
      "Process TRIMGALORE in filepath ./git_repos/metatdenovo\\modules\\nf-core\\trimgalore\\main.nf\n",
      "Process BISMARK_ALIGN in filepath ./git_repos/methylseq\\modules\\nf-core\\bismark\\align\\main.nf\n",
      "Process BISMARK_COVERAGE2CYTOSINE in filepath ./git_repos/methylseq\\modules\\nf-core\\bismark\\coverage2cytosine\\main.nf\n",
      "Process BISMARK_DEDUPLICATE in filepath ./git_repos/methylseq\\modules\\nf-core\\bismark\\deduplicate\\main.nf\n",
      "Process BISMARK_GENOMEPREPARATION in filepath ./git_repos/methylseq\\modules\\nf-core\\bismark\\genomepreparation\\main.nf\n",
      "Process BISMARK_METHYLATIONEXTRACTOR in filepath ./git_repos/methylseq\\modules\\nf-core\\bismark\\methylationextractor\\main.nf\n",
      "Process BISMARK_REPORT in filepath ./git_repos/methylseq\\modules\\nf-core\\bismark\\report\\main.nf\n",
      "Process BISMARK_SUMMARY in filepath ./git_repos/methylseq\\modules\\nf-core\\bismark\\summary\\main.nf\n",
      "Process BWAMETH_ALIGN in filepath ./git_repos/methylseq\\modules\\nf-core\\bwameth\\align\\main.nf\n",
      "Process BWAMETH_INDEX in filepath ./git_repos/methylseq\\modules\\nf-core\\bwameth\\index\\main.nf\n",
      "Process CAT_FASTQ in filepath ./git_repos/methylseq\\modules\\nf-core\\cat\\fastq\\main.nf\n",
      "Process CUSTOM_DUMPSOFTWAREVERSIONS in filepath ./git_repos/methylseq\\modules\\nf-core\\custom\\dumpsoftwareversions\\main.nf\n",
      "Process FASTQC in filepath ./git_repos/methylseq\\modules\\nf-core\\fastqc\\main.nf\n",
      "Process METHYLDACKEL_EXTRACT in filepath ./git_repos/methylseq\\modules\\nf-core\\methyldackel\\extract\\main.nf\n",
      "Process METHYLDACKEL_MBIAS in filepath ./git_repos/methylseq\\modules\\nf-core\\methyldackel\\mbias\\main.nf\n",
      "Process MULTIQC in filepath ./git_repos/methylseq\\modules\\nf-core\\multiqc\\main.nf\n",
      "Process PICARD_MARKDUPLICATES in filepath ./git_repos/methylseq\\modules\\nf-core\\picard\\markduplicates\\main.nf\n",
      "Process PRESEQ_LCEXTRAP in filepath ./git_repos/methylseq\\modules\\nf-core\\preseq\\lcextrap\\main.nf\n",
      "Process QUALIMAP_BAMQC in filepath ./git_repos/methylseq\\modules\\nf-core\\qualimap\\bamqc\\main.nf\n",
      "Process QUALIMAP_BAMQCCRAM in filepath ./git_repos/methylseq\\modules\\nf-core\\qualimap\\bamqccram\\main.nf\n",
      "Process SAMTOOLS_FAIDX in filepath ./git_repos/methylseq\\modules\\nf-core\\samtools\\faidx\\main.nf\n",
      "Process SAMTOOLS_FLAGSTAT in filepath ./git_repos/methylseq\\modules\\nf-core\\samtools\\flagstat\\main.nf\n",
      "Process SAMTOOLS_INDEX in filepath ./git_repos/methylseq\\modules\\nf-core\\samtools\\index\\main.nf\n",
      "Process SAMTOOLS_SORT in filepath ./git_repos/methylseq\\modules\\nf-core\\samtools\\sort\\main.nf\n",
      "Process SAMTOOLS_STATS in filepath ./git_repos/methylseq\\modules\\nf-core\\samtools\\stats\\main.nf\n",
      "Process TRIMGALORE in filepath ./git_repos/methylseq\\modules\\nf-core\\trimgalore\\main.nf\n",
      "Process UNTAR in filepath ./git_repos/methylseq\\modules\\nf-core\\untar\\main.nf\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Process DEEPLC in filepath ./git_repos/mhcquant\\modules\\local\\deeplc.nf\n",
      "Process GENERATE_PROTEINS_FROM_VCF in filepath ./git_repos/mhcquant\\modules\\local\\generate_proteins_from_vcf.nf\n",
      "Process MHCFLURRY_PREDICTNEOEPITOPESCLASS1 in filepath ./git_repos/mhcquant\\modules\\local\\mhcflurry_predictneoepitopesclass1.nf\n",
      "Process MHCFLURRY_PREDICTPEPTIDESCLASS1 in filepath ./git_repos/mhcquant\\modules\\local\\mhcflurry_predictpeptidesclass1.nf\n",
      "Process MHCFLURRY_PREDICTPSMS in filepath ./git_repos/mhcquant\\modules\\local\\mhcflurry_predictpsms.nf\n",
      "Process MHCNUGGETS_NEOEPITOPESCLASS2POST in filepath ./git_repos/mhcquant\\modules\\local\\mhcnuggets_neoepitopesclass2post.nf\n",
      "Process MHCNUGGETS_NEOEPITOPESCLASS2PRE in filepath ./git_repos/mhcquant\\modules\\local\\mhcnuggets_neoepitopesclass2pre.nf\n",
      "Process MHCNUGGETS_PEPTIDESCLASS2POST in filepath ./git_repos/mhcquant\\modules\\local\\mhcnuggets_peptidesclass2post.nf\n",
      "Process MHCNUGGETS_PEPTIDESCLASS2PRE in filepath ./git_repos/mhcquant\\modules\\local\\mhcnuggets_peptidesclass2pre.nf\n",
      "Process MHCNUGGETS_PREDICTNEOEPITOPESCLASS2 in filepath ./git_repos/mhcquant\\modules\\local\\mhcnuggets_predictneoepitopesclass2.nf\n",
      "Process MHCNUGGETS_PREDICTPEPTIDESCLASS2 in filepath ./git_repos/mhcquant\\modules\\local\\mhcnuggets_predictpeptidesclass2.nf\n",
      "Process MS2PIP in filepath ./git_repos/mhcquant\\modules\\local\\ms2pip.nf\n",
      "Process OPENMS_COMETADAPTER in filepath ./git_repos/mhcquant\\modules\\local\\openms_cometadapter.nf\n",
      "Process OPENMS_DECOYDATABASE in filepath ./git_repos/mhcquant\\modules\\local\\openms_decoydatabase.nf\n",
      "Process OPENMS_FALSEDISCOVERYRATE in filepath ./git_repos/mhcquant\\modules\\local\\openms_falsediscoveryrate.nf\n",
      "Process OPENMS_FEATUREFINDERIDENTIFICATION in filepath ./git_repos/mhcquant\\modules\\local\\openms_featurefinderidentification.nf\n",
      "Process OPENMS_FEATURELINKERUNLABELEDKD in filepath ./git_repos/mhcquant\\modules\\local\\openms_featurelinkerunlabeledkd.nf\n",
      "Process OPENMS_FILEFILTER in filepath ./git_repos/mhcquant\\modules\\local\\openms_filefilter.nf\n",
      "Process OPENMS_IDCONFLICTRESOLVER in filepath ./git_repos/mhcquant\\modules\\local\\openms_idconflictresolver.nf\n",
      "Process OPENMS_IDFILTER in filepath ./git_repos/mhcquant\\modules\\local\\openms_idfilter.nf\n",
      "Process OPENMS_IDMERGER in filepath ./git_repos/mhcquant\\modules\\local\\openms_idmerger.nf\n",
      "Process OPENMS_IDRIPPER in filepath ./git_repos/mhcquant\\modules\\local\\openms_idripper.nf\n",
      "Process OPENMS_IDSCORESWITCHER in filepath ./git_repos/mhcquant\\modules\\local\\openms_idscoreswitcher.nf\n",
      "Process OPENMS_MAPALIGNERIDENTIFICATION in filepath ./git_repos/mhcquant\\modules\\local\\openms_mapaligneridentification.nf\n",
      "Process OPENMS_MAPRTTRANSFORMER in filepath ./git_repos/mhcquant\\modules\\local\\openms_maprttransformer.nf\n",
      "Process OPENMS_MZTABEXPORTER in filepath ./git_repos/mhcquant\\modules\\local\\openms_mztabexporter.nf\n",
      "Process OPENMS_PEAKPICKERHIRES in filepath ./git_repos/mhcquant\\modules\\local\\openms_peakpickerhires.nf\n",
      "Process OPENMS_PEPTIDEINDEXER in filepath ./git_repos/mhcquant\\modules\\local\\openms_peptideindexer.nf\n",
      "Process OPENMS_PERCOLATORADAPTER in filepath ./git_repos/mhcquant\\modules\\local\\openms_percolatoradapter.nf\n",
      "Process OPENMS_PSMFEATUREEXTRACTOR in filepath ./git_repos/mhcquant\\modules\\local\\openms_psmfeatureextractor.nf\n",
      "Process OPENMS_TEXTEXPORTER in filepath ./git_repos/mhcquant\\modules\\local\\openms_textexporter.nf\n",
      "Process PREDICT_POSSIBLE_NEOEPITOPES in filepath ./git_repos/mhcquant\\modules\\local\\predict_possible_neoepitopes.nf\n",
      "Process PYOPENMS_IDFILTER in filepath ./git_repos/mhcquant\\modules\\local\\pyopenms_idfilter.nf\n",
      "Process PYOPENMS_IONANNOTATOR in filepath ./git_repos/mhcquant\\modules\\local\\pyopenms_ionannotator.nf\n",
      "Process RESOLVE_FOUND_NEOEPITOPES in filepath ./git_repos/mhcquant\\modules\\local\\resolve_found_neoepitopes.nf\n",
      "Process SAMPLESHEET_CHECK in filepath ./git_repos/mhcquant\\modules\\local\\samplesheet_check.nf\n",
      "Process TDF2MZML in filepath ./git_repos/mhcquant\\modules\\local\\tdf2mzml.nf\n",
      "Process THERMORAWFILEPARSER in filepath ./git_repos/mhcquant\\modules\\local\\thermorawfileparser.nf\n",
      "Process CUSTOM_DUMPSOFTWAREVERSIONS in filepath ./git_repos/mhcquant\\modules\\nf-core\\custom\\dumpsoftwareversions\\main.nf\n",
      "Process MULTIQC in filepath ./git_repos/mhcquant\\modules\\nf-core\\multiqc\\main.nf\n",
      "Process CheckDesign in filepath ./git_repos/mnaseseq\\main.nf\n",
      "Process BWAIndex in filepath ./git_repos/mnaseseq\\main.nf\n",
      "Process MakeGeneBED in filepath ./git_repos/mnaseseq\\main.nf\n",
      "Process MakeTSSBED in filepath ./git_repos/mnaseseq\\main.nf\n",
      "Process MakeGenomeFilter in filepath ./git_repos/mnaseseq\\main.nf\n",
      "Process FastQC in filepath ./git_repos/mnaseseq\\main.nf\n",
      "Process TrimGalore in filepath ./git_repos/mnaseseq\\main.nf\n",
      "Process BWAMem in filepath ./git_repos/mnaseseq\\main.nf\n",
      "Process SortBAM in filepath ./git_repos/mnaseseq\\main.nf\n",
      "Process MergedLibBAM in filepath ./git_repos/mnaseseq\\main.nf\n",
      "Process MergedLibBAMFilter in filepath ./git_repos/mnaseseq\\main.nf\n",
      "Process MergedLibBAMRemoveOrphan in filepath ./git_repos/mnaseseq\\main.nf\n",
      "Process MergedLibPreseq in filepath ./git_repos/mnaseseq\\main.nf\n",
      "Process MergedLibMetricsPreFilter in filepath ./git_repos/mnaseseq\\main.nf\n",
      "Process MergedLibMetricsPostFilter in filepath ./git_repos/mnaseseq\\main.nf\n",
      "Process MergedLibBigWig in filepath ./git_repos/mnaseseq\\main.nf\n",
      "Process MergedLibPlotFingerprint in filepath ./git_repos/mnaseseq\\main.nf\n",
      "Process MergedLibBAMToBED in filepath ./git_repos/mnaseseq\\main.nf\n",
      "Process MergedLibDANPOS2 in filepath ./git_repos/mnaseseq\\main.nf\n",
      "Process MergedLibPlotProfile in filepath ./git_repos/mnaseseq\\main.nf\n",
      "Process MergedRepBAM in filepath ./git_repos/mnaseseq\\main.nf\n",
      "Process MergedRepNameSortBAM in filepath ./git_repos/mnaseseq\\main.nf\n",
      "Process MergedRepBigWig in filepath ./git_repos/mnaseseq\\main.nf\n",
      "Process MergedRepBAMToBED in filepath ./git_repos/mnaseseq\\main.nf\n",
      "Process MergedRepDANPOS2 in filepath ./git_repos/mnaseseq\\main.nf\n",
      "Process MergedRepPlotProfile in filepath ./git_repos/mnaseseq\\main.nf\n",
      "Process IGV in filepath ./git_repos/mnaseseq\\main.nf\n",
      "Process get_software_versions in filepath ./git_repos/mnaseseq\\main.nf\n",
      "Process MultiQC in filepath ./git_repos/mnaseseq\\main.nf\n",
      "Process output_documentation in filepath ./git_repos/mnaseseq\\main.nf\n",
      "Process SAMPLESHEET_CHECK in filepath ./git_repos/molkart\\modules\\local\\samplesheet_check.nf\n",
      "Process CUSTOM_DUMPSOFTWAREVERSIONS in filepath ./git_repos/molkart\\modules\\nf-core\\custom\\dumpsoftwareversions\\main.nf\n",
      "Process FASTQC in filepath ./git_repos/molkart\\modules\\nf-core\\fastqc\\main.nf\n",
      "Process MULTIQC in filepath ./git_repos/molkart\\modules\\nf-core\\multiqc\\main.nf\n",
      "Process CALCULATE_SEQSTATS in filepath ./git_repos/multiplesequencealign\\modules\\local\\calculate_seqstats.nf\n",
      "Process CREATE_TCOFFEETEMPLATE in filepath ./git_repos/multiplesequencealign\\modules\\local\\create_tcoffee_template.nf\n",
      "Process MULTIQC in filepath ./git_repos/multiplesequencealign\\modules\\local\\multiqc.nf\n",
      "Process PARSE_IRMSD in filepath ./git_repos/multiplesequencealign\\modules\\local\\parse_irmsd.nf\n",
      "Process PARSE_SIM in filepath ./git_repos/multiplesequencealign\\modules\\local\\parse_sim.nf\n",
      "Process PREPARE_MULTIQC in filepath ./git_repos/multiplesequencealign\\modules\\local\\prepare_multiqc.nf\n",
      "Process PREPARE_SHINY in filepath ./git_repos/multiplesequencealign\\modules\\local\\prepare_shiny.nf\n",
      "Process CLUSTALO_ALIGN in filepath ./git_repos/multiplesequencealign\\modules\\nf-core\\clustalo\\align\\main.nf\n",
      "Process CLUSTALO_GUIDETREE in filepath ./git_repos/multiplesequencealign\\modules\\nf-core\\clustalo\\guidetree\\main.nf\n",
      "Process CSVTK_CONCAT in filepath ./git_repos/multiplesequencealign\\modules\\nf-core\\csvtk\\concat\\main.nf\n",
      "Process CSVTK_JOIN in filepath ./git_repos/multiplesequencealign\\modules\\nf-core\\csvtk\\join\\main.nf\n",
      "Process CUSTOM_DUMPSOFTWAREVERSIONS in filepath ./git_repos/multiplesequencealign\\modules\\nf-core\\custom\\dumpsoftwareversions\\main.nf\n",
      "Process FAMSA_ALIGN in filepath ./git_repos/multiplesequencealign\\modules\\nf-core\\famsa\\align\\main.nf\n",
      "Process FAMSA_GUIDETREE in filepath ./git_repos/multiplesequencealign\\modules\\nf-core\\famsa\\guidetree\\main.nf\n",
      "Process FASTQC in filepath ./git_repos/multiplesequencealign\\modules\\nf-core\\fastqc\\main.nf\n",
      "Process KALIGN_ALIGN in filepath ./git_repos/multiplesequencealign\\modules\\nf-core\\kalign\\align\\main.nf\n",
      "Process LEARNMSA_ALIGN in filepath ./git_repos/multiplesequencealign\\modules\\nf-core\\learnmsa\\align\\main.nf\n",
      "Process MAFFT in filepath ./git_repos/multiplesequencealign\\modules\\nf-core\\mafft\\main.nf\n",
      "Process MULTIQC in filepath ./git_repos/multiplesequencealign\\modules\\nf-core\\multiqc\\main.nf\n",
      "Process MUSCLE5_SUPER5 in filepath ./git_repos/multiplesequencealign\\modules\\nf-core\\muscle5\\super5\\main.nf\n",
      "Process TCOFFEE_ALIGN in filepath ./git_repos/multiplesequencealign\\modules\\nf-core\\tcoffee\\align\\main.nf\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Process TCOFFEE_ALNCOMPARE in filepath ./git_repos/multiplesequencealign\\modules\\nf-core\\tcoffee\\alncompare\\main.nf\n",
      "Process TCOFFEE_IRMSD in filepath ./git_repos/multiplesequencealign\\modules\\nf-core\\tcoffee\\irmsd\\main.nf\n",
      "Process TCOFFEE_SEQREFORMAT in filepath ./git_repos/multiplesequencealign\\modules\\nf-core\\tcoffee\\seqreformat\\main.nf\n",
      "Process UNTAR in filepath ./git_repos/multiplesequencealign\\modules\\nf-core\\untar\\main.nf\n",
      "Process ZIP in filepath ./git_repos/multiplesequencealign\\modules\\nf-core\\zip\\main.nf\n",
      "Process BAMBU in filepath ./git_repos/nanoseq\\modules\\local\\bambu.nf\n",
      "Process BAM_RENAME in filepath ./git_repos/nanoseq\\modules\\local\\bam_rename.nf\n",
      "Process BEDTOOLS_BAMBED in filepath ./git_repos/nanoseq\\modules\\local\\bedtools_bamtobed.nf\n",
      "Process BEDTOOLS_GENOMECOV in filepath ./git_repos/nanoseq\\modules\\local\\bedtools_genomecov.nf\n",
      "Process CUTESV in filepath ./git_repos/nanoseq\\modules\\local\\cutesv.nf\n",
      "Process DEEPVARIANT in filepath ./git_repos/nanoseq\\modules\\local\\deepvariant.nf\n",
      "Process DESEQ2 in filepath ./git_repos/nanoseq\\modules\\local\\deseq2.nf\n",
      "Process DEXSEQ in filepath ./git_repos/nanoseq\\modules\\local\\dexseq.nf\n",
      "Process GET_CHROM_SIZES in filepath ./git_repos/nanoseq\\modules\\local\\get_chrom_sizes.nf\n",
      "Process GET_JAFFAL_REF in filepath ./git_repos/nanoseq\\modules\\local\\get_jaffal_ref.nf\n",
      "Process GET_NANOLYSE_FASTA in filepath ./git_repos/nanoseq\\modules\\local\\get_nanolyse_fasta.nf\n",
      "Process GET_TEST_DATA in filepath ./git_repos/nanoseq\\modules\\local\\get_test_data.nf\n",
      "Process GRAPHMAP2_ALIGN in filepath ./git_repos/nanoseq\\modules\\local\\graphmap2_align.nf\n",
      "Process GRAPHMAP2_INDEX in filepath ./git_repos/nanoseq\\modules\\local\\graphmap2_index.nf\n",
      "Process GTF2BED in filepath ./git_repos/nanoseq\\modules\\local\\gtf2bed.nf\n",
      "Process JAFFAL in filepath ./git_repos/nanoseq\\modules\\local\\jaffal.nf\n",
      "Process M6ANET_DATAPREP in filepath ./git_repos/nanoseq\\modules\\local\\m6anet_dataprep.nf\n",
      "Process M6ANET_INFERENCE in filepath ./git_repos/nanoseq\\modules\\local\\m6anet_inference.nf\n",
      "Process MEDAKA_VARIANT in filepath ./git_repos/nanoseq\\modules\\local\\medaka_variant.nf\n",
      "Process MINIMAP2_ALIGN in filepath ./git_repos/nanoseq\\modules\\local\\minimap2_align.nf\n",
      "Process MINIMAP2_INDEX in filepath ./git_repos/nanoseq\\modules\\local\\minimap2_index.nf\n",
      "Process MULTIQC in filepath ./git_repos/nanoseq\\modules\\local\\multiqc.nf\n",
      "Process NANOPOLISH_INDEX_EVENTALIGN in filepath ./git_repos/nanoseq\\modules\\local\\nanopolish_index_eventalign.nf\n",
      "Process PEPPER_MARGIN_DEEPVARIANT in filepath ./git_repos/nanoseq\\modules\\local\\pepper_margin_deepvariant.nf\n",
      "Process QCAT in filepath ./git_repos/nanoseq\\modules\\local\\qcat.nf\n",
      "Process SAMPLESHEET_CHECK in filepath ./git_repos/nanoseq\\modules\\local\\samplesheet_check.nf\n",
      "Process SAMTOOLS_SORT_INDEX in filepath ./git_repos/nanoseq\\modules\\local\\samtools_sort_index.nf\n",
      "Process SAMTOOLS_VIEW_BAM in filepath ./git_repos/nanoseq\\modules\\local\\samtools_view_bam.nf\n",
      "Process SNIFFLES in filepath ./git_repos/nanoseq\\modules\\local\\sniffles.nf\n",
      "Process STRINGTIE2 in filepath ./git_repos/nanoseq\\modules\\local\\stringtie2.nf\n",
      "Process SUBREAD_FEATURECOUNTS in filepath ./git_repos/nanoseq\\modules\\local\\subread_featurecounts.nf\n",
      "Process UCSC_BED12TOBIGBED in filepath ./git_repos/nanoseq\\modules\\local\\ucsc_bed12tobigbed.nf\n",
      "Process UCSC_BEDGRAPHTOBIGWIG in filepath ./git_repos/nanoseq\\modules\\local\\ucsc_bedgraphtobigwig.nf\n",
      "Process XPORE_DATAPREP in filepath ./git_repos/nanoseq\\modules\\local\\xpore_dataprep.nf\n",
      "Process XPORE_DIFFMOD in filepath ./git_repos/nanoseq\\modules\\local\\xpore_diffmod.nf\n",
      "Process BCFTOOLS_SORT in filepath ./git_repos/nanoseq\\modules\\nf-core\\bcftools\\sort\\main.nf\n",
      "Process CUSTOM_DUMPSOFTWAREVERSIONS in filepath ./git_repos/nanoseq\\modules\\nf-core\\custom\\dumpsoftwareversions\\main.nf\n",
      "Process FASTQC in filepath ./git_repos/nanoseq\\modules\\nf-core\\fastqc\\main.nf\n",
      "Process MULTIQC in filepath ./git_repos/nanoseq\\modules\\nf-core\\multiqc\\main.nf\n",
      "Process NANOLYSE in filepath ./git_repos/nanoseq\\modules\\nf-core\\nanolyse\\main.nf\n",
      "Process NANOPLOT in filepath ./git_repos/nanoseq\\modules\\nf-core\\nanoplot\\main.nf\n",
      "Process SAMTOOLS_FAIDX in filepath ./git_repos/nanoseq\\modules\\nf-core\\samtools\\faidx\\main.nf\n",
      "Process SAMTOOLS_FLAGSTAT in filepath ./git_repos/nanoseq\\modules\\nf-core\\samtools\\flagstat\\main.nf\n",
      "Process SAMTOOLS_IDXSTATS in filepath ./git_repos/nanoseq\\modules\\nf-core\\samtools\\idxstats\\main.nf\n",
      "Process SAMTOOLS_INDEX in filepath ./git_repos/nanoseq\\modules\\nf-core\\samtools\\index\\main.nf\n",
      "Process SAMTOOLS_SORT in filepath ./git_repos/nanoseq\\modules\\nf-core\\samtools\\sort\\main.nf\n",
      "Process SAMTOOLS_STATS in filepath ./git_repos/nanoseq\\modules\\nf-core\\samtools\\stats\\main.nf\n",
      "Process STRINGTIE_MERGE in filepath ./git_repos/nanoseq\\modules\\nf-core\\stringtie\\merge\\main.nf\n",
      "Process TABIX_BGZIP in filepath ./git_repos/nanoseq\\modules\\nf-core\\tabix\\bgzip\\main.nf\n",
      "Process TABIX_BGZIPTABIX in filepath ./git_repos/nanoseq\\modules\\nf-core\\tabix\\bgziptabix\\main.nf\n",
      "Process TABIX_TABIX in filepath ./git_repos/nanoseq\\modules\\nf-core\\tabix\\tabix\\main.nf\n",
      "Process UNTAR in filepath ./git_repos/nanoseq\\modules\\nf-core\\untar\\main.nf\n",
      "Process COMPUTE_GENE_SCORES in filepath ./git_repos/nanostring\\modules\\local\\compute_gene_scores.nf\n",
      "Process CREATE_ANNOTATED_TABLES in filepath ./git_repos/nanostring\\modules\\local\\create_annotated_tables.nf\n",
      "Process CREATE_GENE_HEATMAP in filepath ./git_repos/nanostring\\modules\\local\\create_gene_heatmap.nf\n",
      "Process SAMPLESHEET_CHECK in filepath ./git_repos/nanostring\\modules\\local\\samplesheet_check.nf\n",
      "Process NACHO_NORMALIZE in filepath ./git_repos/nanostring\\modules\\local\\nacho\\normalize.nf\n",
      "Process NACHO_QC in filepath ./git_repos/nanostring\\modules\\local\\nacho\\qc.nf\n",
      "Process CUSTOM_DUMPSOFTWAREVERSIONS in filepath ./git_repos/nanostring\\modules\\nf-core\\custom\\dumpsoftwareversions\\main.nf\n",
      "Process MULTIQC in filepath ./git_repos/nanostring\\modules\\nf-core\\multiqc\\main.nf\n",
      "Process BED2SAF in filepath ./git_repos/nascent\\modules\\local\\bed2saf.nf\n",
      "Process GTF2BED in filepath ./git_repos/nascent\\modules\\local\\gtf2bed.nf\n",
      "Process GTF_GENE_FILTER in filepath ./git_repos/nascent\\modules\\local\\gtf_gene_filter.nf\n",
      "Process SAMPLESHEET_CHECK in filepath ./git_repos/nascent\\modules\\local\\samplesheet_check.nf\n",
      "Process GROHMM_PARAMETERTUNING in filepath ./git_repos/nascent\\modules\\local\\grohmm\\parametertuning\\main.nf\n",
      "Process GROHMM_TRANSCRIPTCALLING in filepath ./git_repos/nascent\\modules\\local\\grohmm\\transcriptcalling\\main.nf\n",
      "Process BBMAP_PILEUP in filepath ./git_repos/nascent\\modules\\nf-core\\bbmap\\pileup\\main.nf\n",
      "Process BEDTOOLS_GENOMECOV in filepath ./git_repos/nascent\\modules\\nf-core\\bedtools\\genomecov\\main.nf\n",
      "Process BEDTOOLS_INTERSECT in filepath ./git_repos/nascent\\modules\\nf-core\\bedtools\\intersect\\main.nf\n",
      "Process BEDTOOLS_MERGE in filepath ./git_repos/nascent\\modules\\nf-core\\bedtools\\merge\\main.nf\n",
      "Process BEDTOOLS_SORT in filepath ./git_repos/nascent\\modules\\nf-core\\bedtools\\sort\\main.nf\n",
      "Process BWA_INDEX in filepath ./git_repos/nascent\\modules\\nf-core\\bwa\\index\\main.nf\n",
      "Process BWA_MEM in filepath ./git_repos/nascent\\modules\\nf-core\\bwa\\mem\\main.nf\n",
      "Process BWAMEM2_INDEX in filepath ./git_repos/nascent\\modules\\nf-core\\bwamem2\\index\\main.nf\n",
      "Process BWAMEM2_MEM in filepath ./git_repos/nascent\\modules\\nf-core\\bwamem2\\mem\\main.nf\n",
      "Process CAT_CAT in filepath ./git_repos/nascent\\modules\\nf-core\\cat\\cat\\main.nf\n",
      "Process CAT_FASTQ in filepath ./git_repos/nascent\\modules\\nf-core\\cat\\fastq\\main.nf\n",
      "Process CUSTOM_DUMPSOFTWAREVERSIONS in filepath ./git_repos/nascent\\modules\\nf-core\\custom\\dumpsoftwareversions\\main.nf\n",
      "Process CUSTOM_GETCHROMSIZES in filepath ./git_repos/nascent\\modules\\nf-core\\custom\\getchromsizes\\main.nf\n",
      "Process DEEPTOOLS_BAMCOVERAGE in filepath ./git_repos/nascent\\modules\\nf-core\\deeptools\\bamcoverage\\main.nf\n",
      "Process DRAGMAP_ALIGN in filepath ./git_repos/nascent\\modules\\nf-core\\dragmap\\align\\main.nf\n",
      "Process DRAGMAP_HASHTABLE in filepath ./git_repos/nascent\\modules\\nf-core\\dragmap\\hashtable\\main.nf\n",
      "Process FASTP in filepath ./git_repos/nascent\\modules\\nf-core\\fastp\\main.nf\n",
      "Process FASTQC in filepath ./git_repos/nascent\\modules\\nf-core\\fastqc\\main.nf\n",
      "Process GFFREAD in filepath ./git_repos/nascent\\modules\\nf-core\\gffread\\main.nf\n",
      "Process GUNZIP in filepath ./git_repos/nascent\\modules\\nf-core\\gunzip\\main.nf\n",
      "Process HOMER_FINDPEAKS in filepath ./git_repos/nascent\\modules\\nf-core\\homer\\findpeaks\\main.nf\n",
      "Process HOMER_MAKETAGDIRECTORY in filepath ./git_repos/nascent\\modules\\nf-core\\homer\\maketagdirectory\\main.nf\n",
      "Process HOMER_MAKEUCSCFILE in filepath ./git_repos/nascent\\modules\\nf-core\\homer\\makeucscfile\\main.nf\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Process HOMER_POS2BED in filepath ./git_repos/nascent\\modules\\nf-core\\homer\\pos2bed\\main.nf\n",
      "Process MULTIQC in filepath ./git_repos/nascent\\modules\\nf-core\\multiqc\\main.nf\n",
      "Process PINTS_CALLER in filepath ./git_repos/nascent\\modules\\nf-core\\pints\\caller\\main.nf\n",
      "Process PRESEQ_CCURVE in filepath ./git_repos/nascent\\modules\\nf-core\\preseq\\ccurve\\main.nf\n",
      "Process PRESEQ_LCEXTRAP in filepath ./git_repos/nascent\\modules\\nf-core\\preseq\\lcextrap\\main.nf\n",
      "Process RSEQC_INFEREXPERIMENT in filepath ./git_repos/nascent\\modules\\nf-core\\rseqc\\inferexperiment\\main.nf\n",
      "Process RSEQC_READDISTRIBUTION in filepath ./git_repos/nascent\\modules\\nf-core\\rseqc\\readdistribution\\main.nf\n",
      "Process RSEQC_READDUPLICATION in filepath ./git_repos/nascent\\modules\\nf-core\\rseqc\\readduplication\\main.nf\n",
      "Process SAMTOOLS_FAIDX in filepath ./git_repos/nascent\\modules\\nf-core\\samtools\\faidx\\main.nf\n",
      "Process SAMTOOLS_FLAGSTAT in filepath ./git_repos/nascent\\modules\\nf-core\\samtools\\flagstat\\main.nf\n",
      "Process SAMTOOLS_IDXSTATS in filepath ./git_repos/nascent\\modules\\nf-core\\samtools\\idxstats\\main.nf\n",
      "Process SAMTOOLS_INDEX in filepath ./git_repos/nascent\\modules\\nf-core\\samtools\\index\\main.nf\n",
      "Process SAMTOOLS_SORT in filepath ./git_repos/nascent\\modules\\nf-core\\samtools\\sort\\main.nf\n",
      "Process SAMTOOLS_STATS in filepath ./git_repos/nascent\\modules\\nf-core\\samtools\\stats\\main.nf\n",
      "Process SUBREAD_FEATURECOUNTS in filepath ./git_repos/nascent\\modules\\nf-core\\subread\\featurecounts\\main.nf\n",
      "Process UMITOOLS_DEDUP in filepath ./git_repos/nascent\\modules\\nf-core\\umitools\\dedup\\main.nf\n",
      "Process UNTAR in filepath ./git_repos/nascent\\modules\\nf-core\\untar\\main.nf\n",
      "Process supernova_full in filepath ./git_repos/neutronstar\\main.nf\n",
      "Process supernova in filepath ./git_repos/neutronstar\\main.nf\n",
      "Process supernova_mkoutput in filepath ./git_repos/neutronstar\\main.nf\n",
      "Process quast in filepath ./git_repos/neutronstar\\main.nf\n",
      "Process busco in filepath ./git_repos/neutronstar\\main.nf\n",
      "Process get_supernova_version in filepath ./git_repos/neutronstar\\main.nf\n",
      "Process get_software_versions in filepath ./git_repos/neutronstar\\main.nf\n",
      "Process multiqc in filepath ./git_repos/neutronstar\\main.nf\n",
      "Process output_documentation in filepath ./git_repos/neutronstar\\main.nf\n",
      "Process EXTRACT_COMMUNITIES in filepath ./git_repos/pangenome\\modules\\local\\extract_communities\\main.nf\n",
      "Process MULTIQC_COMMUNITY in filepath ./git_repos/pangenome\\modules\\local\\multiqc_community\\main.nf\n",
      "Process NET2COMMUNITIES in filepath ./git_repos/pangenome\\modules\\local\\net2communities\\main.nf\n",
      "Process PAF2NET in filepath ./git_repos/pangenome\\modules\\local\\paf2net\\main.nf\n",
      "Process SPLIT_APPROX_MAPPINGS_IN_CHUNKS in filepath ./git_repos/pangenome\\modules\\local\\split_approx_mappings_in_chunks\\main.nf\n",
      "Process VG_DECONSTRUCT in filepath ./git_repos/pangenome\\modules\\local\\vg_deconstruct\\main.nf\n",
      "Process CUSTOM_DUMPSOFTWAREVERSIONS in filepath ./git_repos/pangenome\\modules\\nf-core\\custom\\dumpsoftwareversions\\main.nf\n",
      "Process GFAFFIX in filepath ./git_repos/pangenome\\modules\\nf-core\\gfaffix\\main.nf\n",
      "Process MULTIQC in filepath ./git_repos/pangenome\\modules\\nf-core\\multiqc\\main.nf\n",
      "Process ODGI_BUILD in filepath ./git_repos/pangenome\\modules\\nf-core\\odgi\\build\\main.nf\n",
      "Process ODGI_DRAW in filepath ./git_repos/pangenome\\modules\\nf-core\\odgi\\draw\\main.nf\n",
      "Process ODGI_LAYOUT in filepath ./git_repos/pangenome\\modules\\nf-core\\odgi\\layout\\main.nf\n",
      "Process ODGI_SORT in filepath ./git_repos/pangenome\\modules\\nf-core\\odgi\\sort\\main.nf\n",
      "Process ODGI_SQUEEZE in filepath ./git_repos/pangenome\\modules\\nf-core\\odgi\\squeeze\\main.nf\n",
      "Process ODGI_STATS in filepath ./git_repos/pangenome\\modules\\nf-core\\odgi\\stats\\main.nf\n",
      "Process ODGI_UNCHOP in filepath ./git_repos/pangenome\\modules\\nf-core\\odgi\\unchop\\main.nf\n",
      "Process ODGI_VIEW in filepath ./git_repos/pangenome\\modules\\nf-core\\odgi\\view\\main.nf\n",
      "Process ODGI_VIZ in filepath ./git_repos/pangenome\\modules\\nf-core\\odgi\\viz\\main.nf\n",
      "Process SAMTOOLS_FAIDX in filepath ./git_repos/pangenome\\modules\\nf-core\\samtools\\faidx\\main.nf\n",
      "Process SEQWISH_INDUCE in filepath ./git_repos/pangenome\\modules\\nf-core\\seqwish\\induce\\main.nf\n",
      "Process SMOOTHXG in filepath ./git_repos/pangenome\\modules\\nf-core\\smoothxg\\main.nf\n",
      "Process TABIX_BGZIP in filepath ./git_repos/pangenome\\modules\\nf-core\\tabix\\bgzip\\main.nf\n",
      "Process WFMASH in filepath ./git_repos/pangenome\\modules\\nf-core\\wfmash\\main.nf\n",
      "Process ALIGNFEATURESEQUENCES in filepath ./git_repos/pathogensurveillance\\modules\\local\\align_feature_sequences.nf\n",
      "Process ASSIGN_GROUP_REFERENCES in filepath ./git_repos/pathogensurveillance\\modules\\local\\assign_group_references.nf\n",
      "Process BGZIP_MAKE_GZIP in filepath ./git_repos/pathogensurveillance\\modules\\local\\bgzip_make_gzip.nf\n",
      "Process COREGENOMEPHYLOGENYREPORT in filepath ./git_repos/pathogensurveillance\\modules\\local\\core_gene_phylogeny_report.nf\n",
      "Process DOWNLOAD_ASSEMBLIES in filepath ./git_repos/pathogensurveillance\\modules\\local\\downloadassemblies.nf\n",
      "Process FILTER_ASSEMBLY in filepath ./git_repos/pathogensurveillance\\modules\\local\\filter_assembly.nf\n",
      "Process FIND_ASSEMBLIES in filepath ./git_repos/pathogensurveillance\\modules\\local\\findassemblies.nf\n",
      "Process INITIALCLASSIFICATION in filepath ./git_repos/pathogensurveillance\\modules\\local\\initialclassification.nf\n",
      "Process IQTREE2 in filepath ./git_repos/pathogensurveillance\\modules\\local\\iqtree2.nf\n",
      "Process KHMER_TRIMLOWABUND in filepath ./git_repos/pathogensurveillance\\modules\\local\\khmer_trimlowabund.nf\n",
      "Process MAIN_REPORT in filepath ./git_repos/pathogensurveillance\\modules\\local\\mainreport.nf\n",
      "Process MAKEGFFWITHFASTA in filepath ./git_repos/pathogensurveillance\\modules\\local\\makegffwithfasta.nf\n",
      "Process MAKEREGIONFILE in filepath ./git_repos/pathogensurveillance\\modules\\local\\makeregionfile.nf\n",
      "Process MERGE_ASSEMBLIES in filepath ./git_repos/pathogensurveillance\\modules\\local\\mergeassemblies.nf\n",
      "Process PICK_ASSEMBLIES in filepath ./git_repos/pathogensurveillance\\modules\\local\\pickassemblies.nf\n",
      "Process REFORMATPIRATERESULTS in filepath ./git_repos/pathogensurveillance\\modules\\local\\reformat_pirate_results.nf\n",
      "Process RENAMECOREGENEHEADERS in filepath ./git_repos/pathogensurveillance\\modules\\local\\rename_core_gene_headers.nf\n",
      "Process SAMPLESHEET_CHECK in filepath ./git_repos/pathogensurveillance\\modules\\local\\samplesheet_check.nf\n",
      "Process BBMAP_SENDSKETCH in filepath ./git_repos/pathogensurveillance\\modules\\local\\sendsketch.nf\n",
      "Process SOURMASH_COMPARE in filepath ./git_repos/pathogensurveillance\\modules\\local\\sourmash_compare.nf\n",
      "Process SUBSETCOREGENES in filepath ./git_repos/pathogensurveillance\\modules\\local\\subset_core_genes.nf\n",
      "Process VARIANTREPORT in filepath ./git_repos/pathogensurveillance\\modules\\local\\variantreport.nf\n",
      "Process VCFTOSNPALN in filepath ./git_repos/pathogensurveillance\\modules\\local\\vcftosnpaln.nf\n",
      "Process VCFTOTAB in filepath ./git_repos/pathogensurveillance\\modules\\local\\vcftotab.nf\n",
      "Process BAKTA_BAKTA in filepath ./git_repos/pathogensurveillance\\modules\\nf-core\\bakta\\bakta\\main.nf\n",
      "Process BWA_INDEX in filepath ./git_repos/pathogensurveillance\\modules\\nf-core\\bwa\\index\\main.nf\n",
      "Process BWA_MEM in filepath ./git_repos/pathogensurveillance\\modules\\nf-core\\bwa\\mem\\main.nf\n",
      "Process CUSTOM_DUMPSOFTWAREVERSIONS in filepath ./git_repos/pathogensurveillance\\modules\\nf-core\\custom\\dumpsoftwareversions\\main.nf\n",
      "Process FASTP in filepath ./git_repos/pathogensurveillance\\modules\\nf-core\\fastp\\main.nf\n",
      "Process FASTQC in filepath ./git_repos/pathogensurveillance\\modules\\nf-core\\fastqc\\main.nf\n",
      "Process GATK4_VARIANTFILTRATION in filepath ./git_repos/pathogensurveillance\\modules\\nf-core\\gatk4\\variantfiltration\\main.nf\n",
      "Process GRAPHTYPER_GENOTYPE in filepath ./git_repos/pathogensurveillance\\modules\\nf-core\\graphtyper\\genotype\\main.nf\n",
      "Process GRAPHTYPER_VCFCONCATENATE in filepath ./git_repos/pathogensurveillance\\modules\\nf-core\\graphtyper\\vcfconcatenate\\main.nf\n",
      "Process IQTREE in filepath ./git_repos/pathogensurveillance\\modules\\nf-core\\iqtree\\main.nf\n",
      "Process MAFFT in filepath ./git_repos/pathogensurveillance\\modules\\nf-core\\mafft\\main.nf\n",
      "Process MULTIQC in filepath ./git_repos/pathogensurveillance\\modules\\nf-core\\multiqc\\main.nf\n",
      "Process PICARD_ADDORREPLACEREADGROUPS in filepath ./git_repos/pathogensurveillance\\modules\\nf-core\\picard\\addorreplacereadgroups\\main.nf\n",
      "Process PICARD_CREATESEQUENCEDICTIONARY in filepath ./git_repos/pathogensurveillance\\modules\\nf-core\\picard\\createsequencedictionary\\main.nf\n",
      "Process PICARD_MARKDUPLICATES in filepath ./git_repos/pathogensurveillance\\modules\\nf-core\\picard\\markduplicates\\main.nf\n",
      "Process PICARD_SORTSAM in filepath ./git_repos/pathogensurveillance\\modules\\nf-core\\picard\\sortsam\\main.nf\n",
      "Process PIRATE in filepath ./git_repos/pathogensurveillance\\modules\\nf-core\\pirate\\main.nf\n",
      "Process QUAST in filepath ./git_repos/pathogensurveillance\\modules\\nf-core\\quast\\main.nf\n",
      "Process READ2TREE in filepath ./git_repos/pathogensurveillance\\modules\\nf-core\\read2tree\\main.nf\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Process SAMTOOLS_FAIDX in filepath ./git_repos/pathogensurveillance\\modules\\nf-core\\samtools\\faidx\\main.nf\n",
      "Process SAMTOOLS_INDEX in filepath ./git_repos/pathogensurveillance\\modules\\nf-core\\samtools\\index\\main.nf\n",
      "Process SEQTK_SAMPLE in filepath ./git_repos/pathogensurveillance\\modules\\nf-core\\seqtk\\sample\\main.nf\n",
      "Process SOURMASH_SKETCH in filepath ./git_repos/pathogensurveillance\\modules\\nf-core\\sourmash\\sketch\\main.nf\n",
      "Process SPADES in filepath ./git_repos/pathogensurveillance\\modules\\nf-core\\spades\\main.nf\n",
      "Process TABIX_TABIX in filepath ./git_repos/pathogensurveillance\\modules\\nf-core\\tabix\\tabix\\main.nf\n",
      "Process VCFLIB_VCFFILTER in filepath ./git_repos/pathogensurveillance\\modules\\nf-core\\vcflib\\vcffilter\\main.nf\n",
      "Process get_software_versions in filepath ./git_repos/pgdb\\main.nf\n",
      "Process ensembl_fasta_download in filepath ./git_repos/pgdb\\main.nf\n",
      "Process add_reference_proteome in filepath ./git_repos/pgdb\\main.nf\n",
      "Process merge_cdnas in filepath ./git_repos/pgdb\\main.nf\n",
      "Process add_ncrna in filepath ./git_repos/pgdb\\main.nf\n",
      "Process add_pseudogenes in filepath ./git_repos/pgdb\\main.nf\n",
      "Process add_altorfs in filepath ./git_repos/pgdb\\main.nf\n",
      "Process cosmic_download in filepath ./git_repos/pgdb\\main.nf\n",
      "Process cosmic_proteindb in filepath ./git_repos/pgdb\\main.nf\n",
      "Process cosmic_celllines_proteindb in filepath ./git_repos/pgdb\\main.nf\n",
      "Process ensembl_vcf_download in filepath ./git_repos/pgdb\\main.nf\n",
      "Process check_ensembl_vcf in filepath ./git_repos/pgdb\\main.nf\n",
      "Process ensembl_vcf_proteinDB in filepath ./git_repos/pgdb\\main.nf\n",
      "Process gtf_to_fasta in filepath ./git_repos/pgdb\\main.nf\n",
      "Process vcf_proteinDB in filepath ./git_repos/pgdb\\main.nf\n",
      "Process gencode_download in filepath ./git_repos/pgdb\\main.nf\n",
      "Process gnomad_download in filepath ./git_repos/pgdb\\main.nf\n",
      "Process extract_gnomad_vcf in filepath ./git_repos/pgdb\\main.nf\n",
      "Process gnomad_proteindb in filepath ./git_repos/pgdb\\main.nf\n",
      "Process cds_GRCh37_download in filepath ./git_repos/pgdb\\main.nf\n",
      "Process download_all_cbioportal in filepath ./git_repos/pgdb\\main.nf\n",
      "Process cbioportal_proteindb in filepath ./git_repos/pgdb\\main.nf\n",
      "Process merge_proteindbs in filepath ./git_repos/pgdb\\main.nf\n",
      "Process clean_protein_database in filepath ./git_repos/pgdb\\main.nf\n",
      "Process decoy in filepath ./git_repos/pgdb\\main.nf\n",
      "Process output_documentation in filepath ./git_repos/pgdb\\main.nf\n",
      "Process ANICLUSTER_ANICALC in filepath ./git_repos/phageannotator\\modules\\local\\anicluster\\anicalc\\main.nf\n",
      "Process ANICLUSTER_ANICLUST in filepath ./git_repos/phageannotator\\modules\\local\\anicluster\\aniclust\\main.nf\n",
      "Process ANICLUSTER_EXTRACTREPS in filepath ./git_repos/phageannotator\\modules\\local\\anicluster\\extractreps\\main.nf\n",
      "Process APPENDSCREENHITS in filepath ./git_repos/phageannotator\\modules\\local\\appendscreenhits\\main.nf\n",
      "Process COVERM_CONTIG in filepath ./git_repos/phageannotator\\modules\\local\\coverm\\contig\\main.nf\n",
      "Process INSTRAIN_STB in filepath ./git_repos/phageannotator\\modules\\local\\instrain\\stb\\main.nf\n",
      "Process MASH_PASTE in filepath ./git_repos/phageannotator\\modules\\local\\mash\\paste\\main.nf\n",
      "Process PRODIGAL_PRODIGALGV in filepath ./git_repos/phageannotator\\modules\\local\\prodigal\\prodigalgv\\main.nf\n",
      "Process QUALITYFILTERVIRUSES in filepath ./git_repos/phageannotator\\modules\\local\\qualityfilterviruses\\main.nf\n",
      "Process SEQKIT_SEQ in filepath ./git_repos/phageannotator\\modules\\local\\seqkit\\seq\\main.nf\n",
      "Process VIROMEQC_INSTALL in filepath ./git_repos/phageannotator\\modules\\local\\viromeqc\\install\\main.nf\n",
      "Process VIROMEQC_VIROMEQC in filepath ./git_repos/phageannotator\\modules\\local\\viromeqc\\viromeqc\\main.nf\n",
      "Process BACPHLIP in filepath ./git_repos/phageannotator\\modules\\nf-core\\bacphlip\\main.nf\n",
      "Process BLAST_BLASTN in filepath ./git_repos/phageannotator\\modules\\nf-core\\blast\\blastn\\main.nf\n",
      "Process BLAST_MAKEBLASTDB in filepath ./git_repos/phageannotator\\modules\\nf-core\\blast\\makeblastdb\\main.nf\n",
      "Process BOWTIE2_ALIGN in filepath ./git_repos/phageannotator\\modules\\nf-core\\bowtie2\\align\\main.nf\n",
      "Process BOWTIE2_BUILD in filepath ./git_repos/phageannotator\\modules\\nf-core\\bowtie2\\build\\main.nf\n",
      "Process CAT_CAT in filepath ./git_repos/phageannotator\\modules\\nf-core\\cat\\cat\\main.nf\n",
      "Process CHECKV_DOWNLOADDATABASE in filepath ./git_repos/phageannotator\\modules\\nf-core\\checkv\\downloaddatabase\\main.nf\n",
      "Process CHECKV_ENDTOEND in filepath ./git_repos/phageannotator\\modules\\nf-core\\checkv\\endtoend\\main.nf\n",
      "Process CUSTOM_DUMPSOFTWAREVERSIONS in filepath ./git_repos/phageannotator\\modules\\nf-core\\custom\\dumpsoftwareversions\\main.nf\n",
      "Process FASTQC in filepath ./git_repos/phageannotator\\modules\\nf-core\\fastqc\\main.nf\n",
      "Process GENOMAD_DOWNLOAD in filepath ./git_repos/phageannotator\\modules\\nf-core\\genomad\\download\\main.nf\n",
      "Process GENOMAD_ENDTOEND in filepath ./git_repos/phageannotator\\modules\\nf-core\\genomad\\endtoend\\main.nf\n",
      "Process GUNZIP in filepath ./git_repos/phageannotator\\modules\\nf-core\\gunzip\\main.nf\n",
      "Process INSTRAIN_COMPARE in filepath ./git_repos/phageannotator\\modules\\nf-core\\instrain\\compare\\main.nf\n",
      "Process INSTRAIN_PROFILE in filepath ./git_repos/phageannotator\\modules\\nf-core\\instrain\\profile\\main.nf\n",
      "Process IPHOP_DOWNLOAD in filepath ./git_repos/phageannotator\\modules\\nf-core\\iphop\\download\\main.nf\n",
      "Process IPHOP_PREDICT in filepath ./git_repos/phageannotator\\modules\\nf-core\\iphop\\predict\\main.nf\n",
      "Process MASH_SCREEN in filepath ./git_repos/phageannotator\\modules\\nf-core\\mash\\screen\\main.nf\n",
      "Process MASH_SKETCH in filepath ./git_repos/phageannotator\\modules\\nf-core\\mash\\sketch\\main.nf\n",
      "Process MULTIQC in filepath ./git_repos/phageannotator\\modules\\nf-core\\multiqc\\main.nf\n",
      "Process SAMTOOLS_FLAGSTAT in filepath ./git_repos/phageannotator\\modules\\nf-core\\samtools\\flagstat\\main.nf\n",
      "Process SAMTOOLS_IDXSTATS in filepath ./git_repos/phageannotator\\modules\\nf-core\\samtools\\idxstats\\main.nf\n",
      "Process SAMTOOLS_INDEX in filepath ./git_repos/phageannotator\\modules\\nf-core\\samtools\\index\\main.nf\n",
      "Process SAMTOOLS_SORT in filepath ./git_repos/phageannotator\\modules\\nf-core\\samtools\\sort\\main.nf\n",
      "Process SAMTOOLS_STATS in filepath ./git_repos/phageannotator\\modules\\nf-core\\samtools\\stats\\main.nf\n",
      "Process UNTAR in filepath ./git_repos/phageannotator\\modules\\nf-core\\untar\\main.nf\n",
      "Process CUSTOM_DUMPSOFTWAREVERSIONS in filepath ./git_repos/phyloplace\\modules\\nf-core\\custom\\dumpsoftwareversions\\main.nf\n",
      "Process EPANG_PLACE in filepath ./git_repos/phyloplace\\modules\\nf-core\\epang\\place\\main.nf\n",
      "Process EPANG_SPLIT in filepath ./git_repos/phyloplace\\modules\\nf-core\\epang\\split\\main.nf\n",
      "Process GAPPA_EXAMINEASSIGN in filepath ./git_repos/phyloplace\\modules\\nf-core\\gappa\\examineassign\\main.nf\n",
      "Process GAPPA_EXAMINEGRAFT in filepath ./git_repos/phyloplace\\modules\\nf-core\\gappa\\examinegraft\\main.nf\n",
      "Process GAPPA_EXAMINEHEATTREE in filepath ./git_repos/phyloplace\\modules\\nf-core\\gappa\\examineheattree\\main.nf\n",
      "Process HMMER_ESLALIMASK in filepath ./git_repos/phyloplace\\modules\\nf-core\\hmmer\\eslalimask\\main.nf\n",
      "Process HMMER_ESLREFORMAT in filepath ./git_repos/phyloplace\\modules\\nf-core\\hmmer\\eslreformat\\main.nf\n",
      "Process HMMER_HMMALIGN in filepath ./git_repos/phyloplace\\modules\\nf-core\\hmmer\\hmmalign\\main.nf\n",
      "Process HMMER_HMMBUILD in filepath ./git_repos/phyloplace\\modules\\nf-core\\hmmer\\hmmbuild\\main.nf\n",
      "Process MAFFT in filepath ./git_repos/phyloplace\\modules\\nf-core\\mafft\\main.nf\n",
      "Process MULTIQC in filepath ./git_repos/phyloplace\\modules\\nf-core\\multiqc\\main.nf\n",
      "Process RENAME_READS in filepath ./git_repos/pixelator\\modules\\local\\rename_reads.nf\n",
      "Process SAMPLESHEET_CHECK in filepath ./git_repos/pixelator\\modules\\local\\samplesheet_check.nf\n",
      "Process PIXELATOR_COLLECT_METADATA in filepath ./git_repos/pixelator\\modules\\local\\pixelator\\collect_metadata.nf\n",
      "Process PIXELATOR_LIST_OPTIONS in filepath ./git_repos/pixelator\\modules\\local\\pixelator\\list_options.nf\n",
      "Process PIXELATOR_AMPLICON in filepath ./git_repos/pixelator\\modules\\local\\pixelator\\single-cell\\amplicon\\main.nf\n",
      "Process PIXELATOR_ANALYSIS in filepath ./git_repos/pixelator\\modules\\local\\pixelator\\single-cell\\analysis\\main.nf\n",
      "Process PIXELATOR_ANNOTATE in filepath ./git_repos/pixelator\\modules\\local\\pixelator\\single-cell\\annotate\\main.nf\n",
      "Process PIXELATOR_COLLAPSE in filepath ./git_repos/pixelator\\modules\\local\\pixelator\\single-cell\\collapse\\main.nf\n",
      "Process PIXELATOR_DEMUX in filepath ./git_repos/pixelator\\modules\\local\\pixelator\\single-cell\\demux\\main.nf\n",
      "Process PIXELATOR_GRAPH in filepath ./git_repos/pixelator\\modules\\local\\pixelator\\single-cell\\graph\\main.nf\n",
      "Process PIXELATOR_QC in filepath ./git_repos/pixelator\\modules\\local\\pixelator\\single-cell\\qc\\main.nf\n",
      "Process PIXELATOR_REPORT in filepath ./git_repos/pixelator\\modules\\local\\pixelator\\single-cell\\report\\main.nf\n",
      "Process CAT_FASTQ in filepath ./git_repos/pixelator\\modules\\nf-core\\cat\\fastq\\main.nf\n",
      "Process CUSTOM_DUMPSOFTWAREVERSIONS in filepath ./git_repos/pixelator\\modules\\nf-core\\custom\\dumpsoftwareversions\\main.nf\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Process COLABFOLD_BATCH in filepath ./git_repos/proteinfold\\modules\\local\\colabfold_batch.nf\n",
      "Process COMBINE_UNIPROT in filepath ./git_repos/proteinfold\\modules\\local\\combine_uniprot.nf\n",
      "Process DOWNLOAD_PDBMMCIF in filepath ./git_repos/proteinfold\\modules\\local\\download_pdbmmcif.nf\n",
      "Process MMSEQS_COLABFOLDSEARCH in filepath ./git_repos/proteinfold\\modules\\local\\mmseqs_colabfoldsearch.nf\n",
      "Process MULTIFASTA_TO_CSV in filepath ./git_repos/proteinfold\\modules\\local\\multifasta_to_csv.nf\n",
      "Process RUN_ALPHAFOLD2 in filepath ./git_repos/proteinfold\\modules\\local\\run_alphafold2.nf\n",
      "Process RUN_ALPHAFOLD2_MSA in filepath ./git_repos/proteinfold\\modules\\local\\run_alphafold2_msa.nf\n",
      "Process RUN_ALPHAFOLD2_PRED in filepath ./git_repos/proteinfold\\modules\\local\\run_alphafold2_pred.nf\n",
      "Process SAMPLESHEET_CHECK in filepath ./git_repos/proteinfold\\modules\\local\\samplesheet_check.nf\n",
      "Process ARIA2 in filepath ./git_repos/proteinfold\\modules\\nf-core\\aria2\\main.nf\n",
      "Process CUSTOM_DUMPSOFTWAREVERSIONS in filepath ./git_repos/proteinfold\\modules\\nf-core\\custom\\dumpsoftwareversions\\main.nf\n",
      "Process GUNZIP in filepath ./git_repos/proteinfold\\modules\\nf-core\\gunzip\\main.nf\n",
      "Process MMSEQS_CREATEINDEX in filepath ./git_repos/proteinfold\\modules\\nf-core\\mmseqs\\createindex\\main.nf\n",
      "Process MMSEQS_TSV2EXPROFILEDB in filepath ./git_repos/proteinfold\\modules\\nf-core\\mmseqs\\tsv2exprofiledb\\main.nf\n",
      "Process MULTIQC in filepath ./git_repos/proteinfold\\modules\\nf-core\\multiqc\\main.nf\n",
      "Process UNTAR in filepath ./git_repos/proteinfold\\modules\\nf-core\\untar\\main.nf\n",
      "Process sdrf_parsing in filepath ./git_repos/proteomicslfq\\main.nf\n",
      "Process raw_file_conversion in filepath ./git_repos/proteomicslfq\\main.nf\n",
      "Process mzml_indexing in filepath ./git_repos/proteomicslfq\\main.nf\n",
      "Process generate_decoy_database in filepath ./git_repos/proteomicslfq\\main.nf\n",
      "Process openms_peakpicker in filepath ./git_repos/proteomicslfq\\main.nf\n",
      "Process search_engine_msgf in filepath ./git_repos/proteomicslfq\\main.nf\n",
      "Process search_engine_comet in filepath ./git_repos/proteomicslfq\\main.nf\n",
      "Process index_peptides in filepath ./git_repos/proteomicslfq\\main.nf\n",
      "Process extract_percolator_features in filepath ./git_repos/proteomicslfq\\main.nf\n",
      "Process percolator in filepath ./git_repos/proteomicslfq\\main.nf\n",
      "Process fdr_idpep in filepath ./git_repos/proteomicslfq\\main.nf\n",
      "Process idpep in filepath ./git_repos/proteomicslfq\\main.nf\n",
      "Process idscoreswitcher_to_qval in filepath ./git_repos/proteomicslfq\\main.nf\n",
      "Process consensusid in filepath ./git_repos/proteomicslfq\\main.nf\n",
      "Process fdr_consensusid in filepath ./git_repos/proteomicslfq\\main.nf\n",
      "Process idfilter in filepath ./git_repos/proteomicslfq\\main.nf\n",
      "Process idscoreswitcher_for_luciphor in filepath ./git_repos/proteomicslfq\\main.nf\n",
      "Process luciphor in filepath ./git_repos/proteomicslfq\\main.nf\n",
      "Process proteomicslfq in filepath ./git_repos/proteomicslfq\\main.nf\n",
      "Process msstats in filepath ./git_repos/proteomicslfq\\main.nf\n",
      "Process ptxqc in filepath ./git_repos/proteomicslfq\\main.nf\n",
      "Process get_software_versions in filepath ./git_repos/proteomicslfq\\main.nf\n",
      "Process output_documentation in filepath ./git_repos/proteomicslfq\\main.nf\n",
      "Process PREPROCESS_EXPDESIGN in filepath ./git_repos/quantms\\modules\\local\\preprocess_expdesign.nf\n",
      "Process SAMPLESHEET_CHECK in filepath ./git_repos/quantms\\modules\\local\\samplesheet_check.nf\n",
      "Process ASSEMBLE_EMPIRICAL_LIBRARY in filepath ./git_repos/quantms\\modules\\local\\assemble_empirical_library\\main.nf\n",
      "Process DECOMPRESS in filepath ./git_repos/quantms\\modules\\local\\decompress_dotd\\main.nf\n",
      "Process DIANNCONVERT in filepath ./git_repos/quantms\\modules\\local\\diannconvert\\main.nf\n",
      "Process DIANNSUMMARY in filepath ./git_repos/quantms\\modules\\local\\diannsummary\\main.nf\n",
      "Process DIANN_PRELIMINARY_ANALYSIS in filepath ./git_repos/quantms\\modules\\local\\diann_preliminary_analysis\\main.nf\n",
      "Process GENERATE_DIANN_CFG in filepath ./git_repos/quantms\\modules\\local\\generate_diann_cfg\\main.nf\n",
      "Process INDIVIDUAL_FINAL_ANALYSIS in filepath ./git_repos/quantms\\modules\\local\\individual_final_analysis\\main.nf\n",
      "Process MSSTATS in filepath ./git_repos/quantms\\modules\\local\\msstats\\main.nf\n",
      "Process MSSTATSTMT in filepath ./git_repos/quantms\\modules\\local\\msstatstmt\\main.nf\n",
      "Process MZMLSTATISTICS in filepath ./git_repos/quantms\\modules\\local\\mzmlstatistics\\main.nf\n",
      "Process CONSENSUSID in filepath ./git_repos/quantms\\modules\\local\\openms\\consensusid\\main.nf\n",
      "Process DECOYDATABASE in filepath ./git_repos/quantms\\modules\\local\\openms\\decoydatabase\\main.nf\n",
      "Process EPIFANY in filepath ./git_repos/quantms\\modules\\local\\openms\\epifany\\main.nf\n",
      "Process EXTRACTPSMFEATURES in filepath ./git_repos/quantms\\modules\\local\\openms\\extractpsmfeatures\\main.nf\n",
      "Process FALSEDISCOVERYRATE in filepath ./git_repos/quantms\\modules\\local\\openms\\falsediscoveryrate\\main.nf\n",
      "Process FILEMERGE in filepath ./git_repos/quantms\\modules\\local\\openms\\filemerge\\main.nf\n",
      "Process IDCONFLICTRESOLVER in filepath ./git_repos/quantms\\modules\\local\\openms\\idconflictresolver\\main.nf\n",
      "Process IDFILTER in filepath ./git_repos/quantms\\modules\\local\\openms\\idfilter\\main.nf\n",
      "Process IDMAPPER in filepath ./git_repos/quantms\\modules\\local\\openms\\idmapper\\main.nf\n",
      "Process IDPEP in filepath ./git_repos/quantms\\modules\\local\\openms\\idpep\\main.nf\n",
      "Process IDSCORESWITCHER in filepath ./git_repos/quantms\\modules\\local\\openms\\idscoreswitcher\\main.nf\n",
      "Process INDEXPEPTIDES in filepath ./git_repos/quantms\\modules\\local\\openms\\indexpeptides\\main.nf\n",
      "Process ISOBARICANALYZER in filepath ./git_repos/quantms\\modules\\local\\openms\\isobaricanalyzer\\main.nf\n",
      "Process MSSTATSCONVERTER in filepath ./git_repos/quantms\\modules\\local\\openms\\msstatsconverter\\main.nf\n",
      "Process MZMLINDEXING in filepath ./git_repos/quantms\\modules\\local\\openms\\mzmlindexing\\main.nf\n",
      "Process OPENMSPEAKPICKER in filepath ./git_repos/quantms\\modules\\local\\openms\\openmspeakpicker\\main.nf\n",
      "Process PROTEININFERENCE in filepath ./git_repos/quantms\\modules\\local\\openms\\proteininference\\main.nf\n",
      "Process PROTEINQUANTIFIER in filepath ./git_repos/quantms\\modules\\local\\openms\\proteinquantifier\\main.nf\n",
      "Process PROTEOMICSLFQ in filepath ./git_repos/quantms\\modules\\local\\openms\\proteomicslfq\\main.nf\n",
      "Process LUCIPHORADAPTER in filepath ./git_repos/quantms\\modules\\local\\openms\\thirdparty\\luciphoradapter\\main.nf\n",
      "Process PERCOLATOR in filepath ./git_repos/quantms\\modules\\local\\openms\\thirdparty\\percolator\\main.nf\n",
      "Process SEARCHENGINECOMET in filepath ./git_repos/quantms\\modules\\local\\openms\\thirdparty\\searchenginecomet\\main.nf\n",
      "Process SEARCHENGINEMSGF in filepath ./git_repos/quantms\\modules\\local\\openms\\thirdparty\\searchenginemsgf\\main.nf\n",
      "Process SEARCHENGINESAGE in filepath ./git_repos/quantms\\modules\\local\\openms\\thirdparty\\searchenginesage\\main.nf\n",
      "Process PMULTIQC in filepath ./git_repos/quantms\\modules\\local\\pmultiqc\\main.nf\n",
      "Process SDRFPARSING in filepath ./git_repos/quantms\\modules\\local\\sdrfparsing\\main.nf\n",
      "Process SILICOLIBRARYGENERATION in filepath ./git_repos/quantms\\modules\\local\\silicolibrarygeneration\\main.nf\n",
      "Process TDF2MZML in filepath ./git_repos/quantms\\modules\\local\\tdf2mzml\\main.nf\n",
      "Process THERMORAWFILEPARSER in filepath ./git_repos/quantms\\modules\\local\\thermorawfileparser\\main.nf\n",
      "Process CUSTOM_DUMPSOFTWAREVERSIONS in filepath ./git_repos/quantms\\modules\\nf-core\\custom\\dumpsoftwareversions\\main.nf\n",
      "Process MULTIQC in filepath ./git_repos/quantms\\modules\\nf-core\\multiqc\\main.nf\n",
      "Process CDHIT_TO_RBDIV in filepath ./git_repos/radseq\\modules\\local\\cdhit_to_rbdiv.nf\n",
      "Process COMBINE_UNIQUE_READS in filepath ./git_repos/radseq\\modules\\local\\combine_uniq_forward_reads.nf\n",
      "Process CREATE_INTERVALS in filepath ./git_repos/radseq\\modules\\local\\create_intervals.nf\n",
      "Process PREPARE_FORWARD_READS in filepath ./git_repos/radseq\\modules\\local\\prepare_forward_reads.nf\n",
      "Process RBMERGE2FASTA in filepath ./git_repos/radseq\\modules\\local\\rbmerge2fasta.nf\n",
      "Process SAMPLESHEET_CHECK in filepath ./git_repos/radseq\\modules\\local\\samplesheet_check.nf\n",
      "Process BEDOPS_MERGE_BED in filepath ./git_repos/radseq\\modules\\local\\bedops\\merge\\main.nf\n",
      "Process RAINBOW_DIV in filepath ./git_repos/radseq\\modules\\local\\rainbow\\div\\rainbow_div.nf\n",
      "Process RAINBOW_MERGE in filepath ./git_repos/radseq\\modules\\local\\rainbow\\merge\\rainbow_merge.nf\n",
      "Process BCFTOOLS_CONCAT in filepath ./git_repos/radseq\\modules\\nf-core\\bcftools\\concat\\main.nf\n",
      "Process BCFTOOLS_SORT in filepath ./git_repos/radseq\\modules\\nf-core\\bcftools\\sort\\main.nf\n",
      "Process BCFTOOLS_STATS in filepath ./git_repos/radseq\\modules\\nf-core\\bcftools\\stats\\main.nf\n",
      "Process BEDTOOLS_BAMTOBED in filepath ./git_repos/radseq\\modules\\nf-core\\bedtools\\bamtobed\\main.nf\n",
      "Process BEDTOOLS_COVERAGE in filepath ./git_repos/radseq\\modules\\nf-core\\bedtools\\coverage\\main.nf\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Process BEDTOOLS_INTERSECT in filepath ./git_repos/radseq\\modules\\nf-core\\bedtools\\intersect\\main.nf\n",
      "Process BEDTOOLS_MAKEWINDOWS in filepath ./git_repos/radseq\\modules\\nf-core\\bedtools\\makewindows\\main.nf\n",
      "Process BEDTOOLS_MERGE_COV in filepath ./git_repos/radseq\\modules\\nf-core\\bedtools\\merge\\main.nf\n",
      "Process BEDTOOLS_SORT in filepath ./git_repos/radseq\\modules\\nf-core\\bedtools\\sort\\main.nf\n",
      "Process BWA_INDEX in filepath ./git_repos/radseq\\modules\\nf-core\\bwa\\index\\main.nf\n",
      "Process BWA_MEM in filepath ./git_repos/radseq\\modules\\nf-core\\bwa\\mem\\main.nf\n",
      "Process BWAMEM2_INDEX in filepath ./git_repos/radseq\\modules\\nf-core\\bwamem2\\index\\main.nf\n",
      "Process BWAMEM2_MEM in filepath ./git_repos/radseq\\modules\\nf-core\\bwamem2\\mem\\main.nf\n",
      "Process CDHIT in filepath ./git_repos/radseq\\modules\\nf-core\\cdhit\\cdhit\\main.nf\n",
      "Process CUSTOM_DUMPSOFTWAREVERSIONS in filepath ./git_repos/radseq\\modules\\nf-core\\custom\\dumpsoftwareversions\\main.nf\n",
      "Process FASTP in filepath ./git_repos/radseq\\modules\\nf-core\\fastp\\main.nf\n",
      "Process FASTQC in filepath ./git_repos/radseq\\modules\\nf-core\\fastqc\\main.nf\n",
      "Process FREEBAYES in filepath ./git_repos/radseq\\modules\\nf-core\\freebayes\\main.nf\n",
      "Process MINIMAP2_ALIGN in filepath ./git_repos/radseq\\modules\\nf-core\\minimap2\\align\\main.nf\n",
      "Process MINIMAP2_INDEX in filepath ./git_repos/radseq\\modules\\nf-core\\minimap2\\index\\main.nf\n",
      "Process MULTIQC in filepath ./git_repos/radseq\\modules\\nf-core\\multiqc\\main.nf\n",
      "Process SAMTOOLS_FAIDX in filepath ./git_repos/radseq\\modules\\nf-core\\samtools\\faidx\\main.nf\n",
      "Process SAMTOOLS_FLAGSTAT in filepath ./git_repos/radseq\\modules\\nf-core\\samtools\\flagstat\\main.nf\n",
      "Process SAMTOOLS_IDXSTATS in filepath ./git_repos/radseq\\modules\\nf-core\\samtools\\idxstats\\main.nf\n",
      "Process SAMTOOLS_INDEX in filepath ./git_repos/radseq\\modules\\nf-core\\samtools\\index\\main.nf\n",
      "Process SAMTOOLS_MERGE in filepath ./git_repos/radseq\\modules\\nf-core\\samtools\\merge\\main.nf\n",
      "Process SAMTOOLS_STATS in filepath ./git_repos/radseq\\modules\\nf-core\\samtools\\stats\\main.nf\n",
      "Process SEQTK_SEQ in filepath ./git_repos/radseq\\modules\\nf-core\\seqtk\\seq\\main.nf\n",
      "Process TABIX_TABIX in filepath ./git_repos/radseq\\modules\\nf-core\\tabix\\tabix\\main.nf\n",
      "Process UMITOOLS_DEDUP in filepath ./git_repos/radseq\\modules\\nf-core\\umitools\\dedup\\main.nf\n",
      "Process CHECK_RESULTS in filepath ./git_repos/rangeland\\modules\\local\\check_results.nf\n",
      "Process FORCE_GENERATE_ANALYSIS_MASK in filepath ./git_repos/rangeland\\modules\\local\\force-generate_analysis_mask.nf\n",
      "Process FORCE_GENERATE_TILE_ALLOW_LIST in filepath ./git_repos/rangeland\\modules\\local\\force-generate_tile_allow_list.nf\n",
      "Process FORCE_HIGHER_LEVEL in filepath ./git_repos/rangeland\\modules\\local\\force-higher_level.nf\n",
      "Process FORCE_MOSAIC in filepath ./git_repos/rangeland\\modules\\local\\force-mosaic.nf\n",
      "Process FORCE_PREPROCESS in filepath ./git_repos/rangeland\\modules\\local\\force-preprocess.nf\n",
      "Process FORCE_PYRAMID in filepath ./git_repos/rangeland\\modules\\local\\force-pyramid.nf\n",
      "Process HIGHER_LEVEL_CONFIG in filepath ./git_repos/rangeland\\modules\\local\\higher_level_force_config.nf\n",
      "Process MERGE in filepath ./git_repos/rangeland\\modules\\local\\merge.nf\n",
      "Process PREPROCESS_CONFIG in filepath ./git_repos/rangeland\\modules\\local\\preprocess_force_config.nf\n",
      "Process SAMPLESHEET_CHECK in filepath ./git_repos/rangeland\\modules\\local\\samplesheet_check.nf\n",
      "Process CUSTOM_DUMPSOFTWAREVERSIONS in filepath ./git_repos/rangeland\\modules\\nf-core\\custom\\dumpsoftwareversions\\main.nf\n",
      "Process MULTIQC in filepath ./git_repos/rangeland\\modules\\nf-core\\multiqc\\main.nf\n",
      "Process UNTAR in filepath ./git_repos/rangeland\\modules\\nf-core\\untar\\main.nf\n",
      "Process ADD_MOST_SEVERE_CSQ in filepath ./git_repos/raredisease\\modules\\local\\add_most_severe_consequence.nf\n",
      "Process ADD_MOST_SEVERE_PLI in filepath ./git_repos/raredisease\\modules\\local\\add_most_severe_pli.nf\n",
      "Process BUILD_BED in filepath ./git_repos/raredisease\\modules\\local\\create_bed_from_fai.nf\n",
      "Process FILTER_VEP in filepath ./git_repos/raredisease\\modules\\local\\filter_vep.nf\n",
      "Process GET_CHROM_SIZES in filepath ./git_repos/raredisease\\modules\\local\\get_chrom_sizes.nf\n",
      "Process MT_DELETION in filepath ./git_repos/raredisease\\modules\\local\\mt_deletion_script.nf\n",
      "Process SAMPLESHEET_CHECK in filepath ./git_repos/raredisease\\modules\\local\\samplesheet_check.nf\n",
      "Process ENSEMBLVEP in filepath ./git_repos/raredisease\\modules\\local\\ensemblvep\\main.nf\n",
      "Process GATK4_COLLECTREADCOUNTS in filepath ./git_repos/raredisease\\modules\\local\\gatk4\\collectreadcounts\\main.nf\n",
      "Process GATK4_DENOISEREADCOUNTS in filepath ./git_repos/raredisease\\modules\\local\\gatk4\\denoisereadcounts\\main.nf\n",
      "Process GENS in filepath ./git_repos/raredisease\\modules\\local\\gens\\main.nf\n",
      "Process SENTIEON_BQSR in filepath ./git_repos/raredisease\\modules\\local\\sentieon\\bqsr.nf\n",
      "Process SENTIEON_BWAMEM in filepath ./git_repos/raredisease\\modules\\local\\sentieon\\bwamem.nf\n",
      "Process SENTIEON_BWAINDEX in filepath ./git_repos/raredisease\\modules\\local\\sentieon\\bwamemindex.nf\n",
      "Process SENTIEON_DATAMETRICS in filepath ./git_repos/raredisease\\modules\\local\\sentieon\\datametrics.nf\n",
      "Process SENTIEON_DEDUP in filepath ./git_repos/raredisease\\modules\\local\\sentieon\\dedup.nf\n",
      "Process SENTIEON_DNAMODELAPPLY in filepath ./git_repos/raredisease\\modules\\local\\sentieon\\dnamodelapply.nf\n",
      "Process SENTIEON_DNASCOPE in filepath ./git_repos/raredisease\\modules\\local\\sentieon\\dnascope.nf\n",
      "Process SENTIEON_LOCUSCOLLECTOR in filepath ./git_repos/raredisease\\modules\\local\\sentieon\\locuscollector.nf\n",
      "Process SENTIEON_READWRITER in filepath ./git_repos/raredisease\\modules\\local\\sentieon\\readwriter.nf\n",
      "Process SENTIEON_TNSCOPE in filepath ./git_repos/raredisease\\modules\\local\\sentieon\\tnscope.nf\n",
      "Process SENTIEON_WGSMETRICSALGO in filepath ./git_repos/raredisease\\modules\\local\\sentieon\\wgsmetricsalgo.nf\n",
      "Process BCFTOOLS_ANNOTATE in filepath ./git_repos/raredisease\\modules\\nf-core\\bcftools\\annotate\\main.nf\n",
      "Process BCFTOOLS_CONCAT in filepath ./git_repos/raredisease\\modules\\nf-core\\bcftools\\concat\\main.nf\n",
      "Process BCFTOOLS_FILTER in filepath ./git_repos/raredisease\\modules\\nf-core\\bcftools\\filter\\main.nf\n",
      "Process BCFTOOLS_MERGE in filepath ./git_repos/raredisease\\modules\\nf-core\\bcftools\\merge\\main.nf\n",
      "Process BCFTOOLS_NORM in filepath ./git_repos/raredisease\\modules\\nf-core\\bcftools\\norm\\main.nf\n",
      "Process BCFTOOLS_REHEADER in filepath ./git_repos/raredisease\\modules\\nf-core\\bcftools\\reheader\\main.nf\n",
      "Process BCFTOOLS_ROH in filepath ./git_repos/raredisease\\modules\\nf-core\\bcftools\\roh\\main.nf\n",
      "Process BCFTOOLS_VIEW in filepath ./git_repos/raredisease\\modules\\nf-core\\bcftools\\view\\main.nf\n",
      "Process BWA_INDEX in filepath ./git_repos/raredisease\\modules\\nf-core\\bwa\\index\\main.nf\n",
      "Process BWAMEM2_INDEX in filepath ./git_repos/raredisease\\modules\\nf-core\\bwamem2\\index\\main.nf\n",
      "Process BWAMEM2_MEM in filepath ./git_repos/raredisease\\modules\\nf-core\\bwamem2\\mem\\main.nf\n",
      "Process CADD in filepath ./git_repos/raredisease\\modules\\nf-core\\cadd\\main.nf\n",
      "Process CAT_CAT in filepath ./git_repos/raredisease\\modules\\nf-core\\cat\\cat\\main.nf\n",
      "Process CHROMOGRAPH in filepath ./git_repos/raredisease\\modules\\nf-core\\chromograph\\main.nf\n",
      "Process CUSTOM_DUMPSOFTWAREVERSIONS in filepath ./git_repos/raredisease\\modules\\nf-core\\custom\\dumpsoftwareversions\\main.nf\n",
      "Process DEEPVARIANT in filepath ./git_repos/raredisease\\modules\\nf-core\\deepvariant\\main.nf\n",
      "Process EKLIPSE in filepath ./git_repos/raredisease\\modules\\nf-core\\eklipse\\main.nf\n",
      "Process EXPANSIONHUNTER in filepath ./git_repos/raredisease\\modules\\nf-core\\expansionhunter\\main.nf\n",
      "Process FASTQC in filepath ./git_repos/raredisease\\modules\\nf-core\\fastqc\\main.nf\n",
      "Process GATK4_BEDTOINTERVALLIST in filepath ./git_repos/raredisease\\modules\\nf-core\\gatk4\\bedtointervallist\\main.nf\n",
      "Process GATK4_COLLECTREADCOUNTS in filepath ./git_repos/raredisease\\modules\\nf-core\\gatk4\\collectreadcounts\\main.nf\n",
      "Process GATK4_CREATESEQUENCEDICTIONARY in filepath ./git_repos/raredisease\\modules\\nf-core\\gatk4\\createsequencedictionary\\main.nf\n",
      "Process GATK4_DETERMINEGERMLINECONTIGPLOIDY in filepath ./git_repos/raredisease\\modules\\nf-core\\gatk4\\determinegermlinecontigploidy\\main.nf\n",
      "Process GATK4_FILTERMUTECTCALLS in filepath ./git_repos/raredisease\\modules\\nf-core\\gatk4\\filtermutectcalls\\main.nf\n",
      "Process GATK4_GERMLINECNVCALLER in filepath ./git_repos/raredisease\\modules\\nf-core\\gatk4\\germlinecnvcaller\\main.nf\n",
      "Process GATK4_INTERVALLISTTOOLS in filepath ./git_repos/raredisease\\modules\\nf-core\\gatk4\\intervallisttools\\main.nf\n",
      "Process GATK4_MERGEBAMALIGNMENT in filepath ./git_repos/raredisease\\modules\\nf-core\\gatk4\\mergebamalignment\\main.nf\n",
      "Process GATK4_MERGEVCFS in filepath ./git_repos/raredisease\\modules\\nf-core\\gatk4\\mergevcfs\\main.nf\n",
      "Process GATK4_MUTECT2 in filepath ./git_repos/raredisease\\modules\\nf-core\\gatk4\\mutect2\\main.nf\n",
      "Process GATK4_POSTPROCESSGERMLINECNVCALLS in filepath ./git_repos/raredisease\\modules\\nf-core\\gatk4\\postprocessgermlinecnvcalls\\main.nf\n",
      "Process GATK4_PREPROCESSINTERVALS in filepath ./git_repos/raredisease\\modules\\nf-core\\gatk4\\preprocessintervals\\main.nf\n",
      "Process GATK4_PRINTREADS in filepath ./git_repos/raredisease\\modules\\nf-core\\gatk4\\printreads\\main.nf\n",
      "Process GATK4_REVERTSAM in filepath ./git_repos/raredisease\\modules\\nf-core\\gatk4\\revertsam\\main.nf\n",
      "Process GATK4_SAMTOFASTQ in filepath ./git_repos/raredisease\\modules\\nf-core\\gatk4\\samtofastq\\main.nf\n",
      "Process GATK4_SELECTVARIANTS in filepath ./git_repos/raredisease\\modules\\nf-core\\gatk4\\selectvariants\\main.nf\n",
      "Process GATK4_SHIFTFASTA in filepath ./git_repos/raredisease\\modules\\nf-core\\gatk4\\shiftfasta\\main.nf\n",
      "Process GATK4_SPLITINTERVALS in filepath ./git_repos/raredisease\\modules\\nf-core\\gatk4\\splitintervals\\main.nf\n",
      "Process GATK4_VARIANTFILTRATION in filepath ./git_repos/raredisease\\modules\\nf-core\\gatk4\\variantfiltration\\main.nf\n",
      "Process GENMOD_ANNOTATE in filepath ./git_repos/raredisease\\modules\\nf-core\\genmod\\annotate\\main.nf\n",
      "Process GENMOD_COMPOUND in filepath ./git_repos/raredisease\\modules\\nf-core\\genmod\\compound\\main.nf\n",
      "Process GENMOD_MODELS in filepath ./git_repos/raredisease\\modules\\nf-core\\genmod\\models\\main.nf\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Process GENMOD_SCORE in filepath ./git_repos/raredisease\\modules\\nf-core\\genmod\\score\\main.nf\n",
      "Process GLNEXUS in filepath ./git_repos/raredisease\\modules\\nf-core\\glnexus\\main.nf\n",
      "Process HAPLOCHECK in filepath ./git_repos/raredisease\\modules\\nf-core\\haplocheck\\main.nf\n",
      "Process HAPLOGREP2_CLASSIFY in filepath ./git_repos/raredisease\\modules\\nf-core\\haplogrep2\\classify\\main.nf\n",
      "Process HMTNOTE_ANNOTATE in filepath ./git_repos/raredisease\\modules\\nf-core\\hmtnote\\annotate\\main.nf\n",
      "Process MANTA_GERMLINE in filepath ./git_repos/raredisease\\modules\\nf-core\\manta\\germline\\main.nf\n",
      "Process MOSDEPTH in filepath ./git_repos/raredisease\\modules\\nf-core\\mosdepth\\main.nf\n",
      "Process MULTIQC in filepath ./git_repos/raredisease\\modules\\nf-core\\multiqc\\main.nf\n",
      "Process PEDDY in filepath ./git_repos/raredisease\\modules\\nf-core\\peddy\\main.nf\n",
      "Process PICARD_ADDORREPLACEREADGROUPS in filepath ./git_repos/raredisease\\modules\\nf-core\\picard\\addorreplacereadgroups\\main.nf\n",
      "Process PICARD_COLLECTHSMETRICS in filepath ./git_repos/raredisease\\modules\\nf-core\\picard\\collecthsmetrics\\main.nf\n",
      "Process PICARD_COLLECTMULTIPLEMETRICS in filepath ./git_repos/raredisease\\modules\\nf-core\\picard\\collectmultiplemetrics\\main.nf\n",
      "Process PICARD_COLLECTWGSMETRICS in filepath ./git_repos/raredisease\\modules\\nf-core\\picard\\collectwgsmetrics\\main.nf\n",
      "Process PICARD_LIFTOVERVCF in filepath ./git_repos/raredisease\\modules\\nf-core\\picard\\liftovervcf\\main.nf\n",
      "Process PICARD_MARKDUPLICATES in filepath ./git_repos/raredisease\\modules\\nf-core\\picard\\markduplicates\\main.nf\n",
      "Process PICARD_RENAMESAMPLEINVCF in filepath ./git_repos/raredisease\\modules\\nf-core\\picard\\renamesampleinvcf\\main.nf\n",
      "Process PICARD_SORTVCF in filepath ./git_repos/raredisease\\modules\\nf-core\\picard\\sortvcf\\main.nf\n",
      "Process QUALIMAP_BAMQC in filepath ./git_repos/raredisease\\modules\\nf-core\\qualimap\\bamqc\\main.nf\n",
      "Process RHOCALL_ANNOTATE in filepath ./git_repos/raredisease\\modules\\nf-core\\rhocall\\annotate\\main.nf\n",
      "Process SAMTOOLS_FAIDX in filepath ./git_repos/raredisease\\modules\\nf-core\\samtools\\faidx\\main.nf\n",
      "Process SAMTOOLS_INDEX in filepath ./git_repos/raredisease\\modules\\nf-core\\samtools\\index\\main.nf\n",
      "Process SAMTOOLS_MERGE in filepath ./git_repos/raredisease\\modules\\nf-core\\samtools\\merge\\main.nf\n",
      "Process SAMTOOLS_SORT in filepath ./git_repos/raredisease\\modules\\nf-core\\samtools\\sort\\main.nf\n",
      "Process SAMTOOLS_STATS in filepath ./git_repos/raredisease\\modules\\nf-core\\samtools\\stats\\main.nf\n",
      "Process SAMTOOLS_VIEW in filepath ./git_repos/raredisease\\modules\\nf-core\\samtools\\view\\main.nf\n",
      "Process SMNCOPYNUMBERCALLER in filepath ./git_repos/raredisease\\modules\\nf-core\\smncopynumbercaller\\main.nf\n",
      "Process STRANGER in filepath ./git_repos/raredisease\\modules\\nf-core\\stranger\\main.nf\n",
      "Process SVDB_MERGE in filepath ./git_repos/raredisease\\modules\\nf-core\\svdb\\merge\\main.nf\n",
      "Process SVDB_QUERY in filepath ./git_repos/raredisease\\modules\\nf-core\\svdb\\query\\main.nf\n",
      "Process TABIX_BGZIPTABIX in filepath ./git_repos/raredisease\\modules\\nf-core\\tabix\\bgziptabix\\main.nf\n",
      "Process TABIX_TABIX in filepath ./git_repos/raredisease\\modules\\nf-core\\tabix\\tabix\\main.nf\n",
      "Process TIDDIT_COV in filepath ./git_repos/raredisease\\modules\\nf-core\\tiddit\\cov\\main.nf\n",
      "Process TIDDIT_SV in filepath ./git_repos/raredisease\\modules\\nf-core\\tiddit\\sv\\main.nf\n",
      "Process UCSC_WIGTOBIGWIG in filepath ./git_repos/raredisease\\modules\\nf-core\\ucsc\\wigtobigwig\\main.nf\n",
      "Process UNTAR in filepath ./git_repos/raredisease\\modules\\nf-core\\untar\\main.nf\n",
      "Process UPD in filepath ./git_repos/raredisease\\modules\\nf-core\\upd\\main.nf\n",
      "Process VCFANNO in filepath ./git_repos/raredisease\\modules\\nf-core\\vcfanno\\main.nf\n",
      "Process SAMPLESHEET_CHECK in filepath ./git_repos/readsimulator\\modules\\local\\samplesheet_check.nf\n",
      "Process CRABS_DBIMPORT in filepath ./git_repos/readsimulator\\modules\\local\\crabs\\dbimport\\main.nf\n",
      "Process CRABS_INSILICOPCR in filepath ./git_repos/readsimulator\\modules\\local\\crabs\\insilicopcr\\main.nf\n",
      "Process CREATE_SAMPLESHEET in filepath ./git_repos/readsimulator\\modules\\local\\custom\\create_samplesheet\\main.nf\n",
      "Process MERGE_FASTAS in filepath ./git_repos/readsimulator\\modules\\local\\custom\\merge_fastas\\main.nf\n",
      "Process MERGE_SAMPLESHEETS in filepath ./git_repos/readsimulator\\modules\\local\\custom\\merge_samplesheets\\main.nf\n",
      "Process INSILICOSEQ_GENERATE in filepath ./git_repos/readsimulator\\modules\\local\\insilicoseq\\generate\\main.nf\n",
      "Process JAPSA_CAPSIM in filepath ./git_repos/readsimulator\\modules\\local\\japsa\\capsim\\main.nf\n",
      "Process UNCOMPRESS_FASTA in filepath ./git_repos/readsimulator\\modules\\local\\uncompress_fasta\\main.nf\n",
      "Process UNZIP in filepath ./git_repos/readsimulator\\modules\\local\\unzip\\main.nf\n",
      "Process WGSIM in filepath ./git_repos/readsimulator\\modules\\local\\wgsim\\main.nf\n",
      "Process ART_ILLUMINA in filepath ./git_repos/readsimulator\\modules\\nf-core\\art\\illumina\\main.nf\n",
      "Process BEDTOOLS_GETFASTA in filepath ./git_repos/readsimulator\\modules\\nf-core\\bedtools\\getfasta\\main.nf\n",
      "Process BOWTIE2_ALIGN in filepath ./git_repos/readsimulator\\modules\\nf-core\\bowtie2\\align\\main.nf\n",
      "Process BOWTIE2_BUILD in filepath ./git_repos/readsimulator\\modules\\nf-core\\bowtie2\\build\\main.nf\n",
      "Process CUSTOM_DUMPSOFTWAREVERSIONS in filepath ./git_repos/readsimulator\\modules\\nf-core\\custom\\dumpsoftwareversions\\main.nf\n",
      "Process FASTQC in filepath ./git_repos/readsimulator\\modules\\nf-core\\fastqc\\main.nf\n",
      "Process MULTIQC in filepath ./git_repos/readsimulator\\modules\\nf-core\\multiqc\\main.nf\n",
      "Process NCBIGENOMEDOWNLOAD in filepath ./git_repos/readsimulator\\modules\\nf-core\\ncbigenomedownload\\main.nf\n",
      "Process SAMTOOLS_INDEX in filepath ./git_repos/readsimulator\\modules\\nf-core\\samtools\\index\\main.nf\n",
      "Process SAMPLESHEET_CHECK in filepath ./git_repos/riboseq\\modules\\local\\samplesheet_check.nf\n",
      "Process BEDTOOLS_BAMTOBED in filepath ./git_repos/riboseq\\modules\\nf-core\\bedtools\\bamtobed\\main.nf\n",
      "Process BEDTOOLS_GENOMECOV in filepath ./git_repos/riboseq\\modules\\nf-core\\bedtools\\genomecov\\main.nf\n",
      "Process BOWTIE_ALIGN in filepath ./git_repos/riboseq\\modules\\nf-core\\bowtie\\align\\main.nf\n",
      "Process BOWTIE_BUILD in filepath ./git_repos/riboseq\\modules\\nf-core\\bowtie\\build\\main.nf\n",
      "Process CUSTOM_DUMPSOFTWAREVERSIONS in filepath ./git_repos/riboseq\\modules\\nf-core\\custom\\dumpsoftwareversions\\main.nf\n",
      "Process CUSTOM_GETCHROMSIZES in filepath ./git_repos/riboseq\\modules\\nf-core\\custom\\getchromsizes\\main.nf\n",
      "Process CUTADAPT in filepath ./git_repos/riboseq\\modules\\nf-core\\cutadapt\\main.nf\n",
      "Process FASTQC in filepath ./git_repos/riboseq\\modules\\nf-core\\fastqc\\main.nf\n",
      "Process HISAT2_ALIGN in filepath ./git_repos/riboseq\\modules\\nf-core\\hisat2\\align\\main.nf\n",
      "Process HISAT2_BUILD in filepath ./git_repos/riboseq\\modules\\nf-core\\hisat2\\build\\main.nf\n",
      "Process HISAT2_EXTRACTSPLICESITES in filepath ./git_repos/riboseq\\modules\\nf-core\\hisat2\\extractsplicesites\\main.nf\n",
      "Process MULTIQC in filepath ./git_repos/riboseq\\modules\\nf-core\\multiqc\\main.nf\n",
      "Process RSEM_PREPAREREFERENCE in filepath ./git_repos/riboseq\\modules\\nf-core\\rsem\\preparereference\\main.nf\n",
      "Process SAMTOOLS_INDEX in filepath ./git_repos/riboseq\\modules\\nf-core\\samtools\\index\\main.nf\n",
      "Process SAMTOOLS_SORT in filepath ./git_repos/riboseq\\modules\\nf-core\\samtools\\sort\\main.nf\n",
      "Process SUBREAD_FEATURECOUNTS in filepath ./git_repos/riboseq\\modules\\nf-core\\subread\\featurecounts\\main.nf\n",
      "Process UMITOOLS_DEDUP in filepath ./git_repos/riboseq\\modules\\nf-core\\umitools\\dedup\\main.nf\n",
      "Process UMITOOLS_EXTRACT in filepath ./git_repos/riboseq\\modules\\nf-core\\umitools\\extract\\main.nf\n",
      "Process GTF2BED in filepath ./git_repos/rnadnavar\\modules\\local\\gtf2bed.nf\n",
      "Process BUILD_INTERVALS in filepath ./git_repos/rnadnavar\\modules\\local\\build_intervals\\main.nf\n",
      "Process RUN_CONSENSUS in filepath ./git_repos/rnadnavar\\modules\\local\\consensus\\main.nf\n",
      "Process CREATE_INTERVALS_BED in filepath ./git_repos/rnadnavar\\modules\\local\\create_intervals_bed\\main.nf\n",
      "Process SAMTOOLS_EXTRACT_READ_IDS in filepath ./git_repos/rnadnavar\\modules\\local\\extract_reads_id\\main.nf\n",
      "Process MAF2BED in filepath ./git_repos/rnadnavar\\modules\\local\\maf2bed\\main.nf\n",
      "Process MAF_FILTERING in filepath ./git_repos/rnadnavar\\modules\\local\\maf_filtering\\main.nf\n",
      "Process RNA_FILTERING in filepath ./git_repos/rnadnavar\\modules\\local\\maf_rna_filtering\\main.nf\n",
      "Process SAGE in filepath ./git_repos/rnadnavar\\modules\\local\\sage\\main.nf\n",
      "Process VCF2MAF in filepath ./git_repos/rnadnavar\\modules\\local\\vcf2maf\\vcf2maf\\main.nf\n",
      "Process VCFFILTER in filepath ./git_repos/rnadnavar\\modules\\local\\vcflib\\filter\\main.nf\n",
      "Process VT_DECOMPOSE in filepath ./git_repos/rnadnavar\\modules\\local\\vt\\decompose\\main.nf\n",
      "Process VT_NORMALISE in filepath ./git_repos/rnadnavar\\modules\\local\\vt\\normalise\\main.nf\n",
      "Process BCFTOOLS_SORT in filepath ./git_repos/rnadnavar\\modules\\nf-core\\bcftools\\sort\\main.nf\n",
      "Process BCFTOOLS_STATS in filepath ./git_repos/rnadnavar\\modules\\nf-core\\bcftools\\stats\\main.nf\n",
      "Process BWA_INDEX in filepath ./git_repos/rnadnavar\\modules\\nf-core\\bwa\\index\\main.nf\n",
      "Process BWA_MEM in filepath ./git_repos/rnadnavar\\modules\\nf-core\\bwa\\mem\\main.nf\n",
      "Process BWAMEM2_INDEX in filepath ./git_repos/rnadnavar\\modules\\nf-core\\bwamem2\\index\\main.nf\n",
      "Process BWAMEM2_MEM in filepath ./git_repos/rnadnavar\\modules\\nf-core\\bwamem2\\mem\\main.nf\n",
      "Process CAT_CAT in filepath ./git_repos/rnadnavar\\modules\\nf-core\\cat\\cat\\main.nf\n",
      "Process CAT_FASTQ in filepath ./git_repos/rnadnavar\\modules\\nf-core\\cat\\fastq\\main.nf\n",
      "Process CUSTOM_DUMPSOFTWAREVERSIONS in filepath ./git_repos/rnadnavar\\modules\\nf-core\\custom\\dumpsoftwareversions\\main.nf\n",
      "Process DRAGMAP_ALIGN in filepath ./git_repos/rnadnavar\\modules\\nf-core\\dragmap\\align\\main.nf\n",
      "Process DRAGMAP_HASHTABLE in filepath ./git_repos/rnadnavar\\modules\\nf-core\\dragmap\\hashtable\\main.nf\n",
      "Process ENSEMBLVEP_DOWNLOAD in filepath ./git_repos/rnadnavar\\modules\\nf-core\\ensemblvep\\download\\main.nf\n",
      "Process ENSEMBLVEP_VEP in filepath ./git_repos/rnadnavar\\modules\\nf-core\\ensemblvep\\vep\\main.nf\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Process FASTP in filepath ./git_repos/rnadnavar\\modules\\nf-core\\fastp\\main.nf\n",
      "Process FASTQC in filepath ./git_repos/rnadnavar\\modules\\nf-core\\fastqc\\main.nf\n",
      "Process FREEBAYES in filepath ./git_repos/rnadnavar\\modules\\nf-core\\freebayes\\main.nf\n",
      "Process GATK4_APPLYBQSR in filepath ./git_repos/rnadnavar\\modules\\nf-core\\gatk4\\applybqsr\\main.nf\n",
      "Process GATK4_APPLYBQSR_SPARK in filepath ./git_repos/rnadnavar\\modules\\nf-core\\gatk4\\applybqsrspark\\main.nf\n",
      "Process GATK4_APPLYVQSR in filepath ./git_repos/rnadnavar\\modules\\nf-core\\gatk4\\applyvqsr\\main.nf\n",
      "Process GATK4_BASERECALIBRATOR in filepath ./git_repos/rnadnavar\\modules\\nf-core\\gatk4\\baserecalibrator\\main.nf\n",
      "Process GATK4_BASERECALIBRATOR_SPARK in filepath ./git_repos/rnadnavar\\modules\\nf-core\\gatk4\\baserecalibratorspark\\main.nf\n",
      "Process GATK4_BEDTOINTERVALLIST in filepath ./git_repos/rnadnavar\\modules\\nf-core\\gatk4\\bedtointervallist\\main.nf\n",
      "Process GATK4_CALCULATECONTAMINATION in filepath ./git_repos/rnadnavar\\modules\\nf-core\\gatk4\\calculatecontamination\\main.nf\n",
      "Process GATK4_CREATESEQUENCEDICTIONARY in filepath ./git_repos/rnadnavar\\modules\\nf-core\\gatk4\\createsequencedictionary\\main.nf\n",
      "Process GATK4_ESTIMATELIBRARYCOMPLEXITY in filepath ./git_repos/rnadnavar\\modules\\nf-core\\gatk4\\estimatelibrarycomplexity\\main.nf\n",
      "Process GATK4_FILTERMUTECTCALLS in filepath ./git_repos/rnadnavar\\modules\\nf-core\\gatk4\\filtermutectcalls\\main.nf\n",
      "Process GATK4_FILTERVARIANTTRANCHES in filepath ./git_repos/rnadnavar\\modules\\nf-core\\gatk4\\filtervarianttranches\\main.nf\n",
      "Process GATK4_GATHERBQSRREPORTS in filepath ./git_repos/rnadnavar\\modules\\nf-core\\gatk4\\gatherbqsrreports\\main.nf\n",
      "Process GATK4_GATHERPILEUPSUMMARIES in filepath ./git_repos/rnadnavar\\modules\\nf-core\\gatk4\\gatherpileupsummaries\\main.nf\n",
      "Process GATK4_GENOMICSDBIMPORT in filepath ./git_repos/rnadnavar\\modules\\nf-core\\gatk4\\genomicsdbimport\\main.nf\n",
      "Process GATK4_GENOTYPEGVCFS in filepath ./git_repos/rnadnavar\\modules\\nf-core\\gatk4\\genotypegvcfs\\main.nf\n",
      "Process GATK4_GETPILEUPSUMMARIES in filepath ./git_repos/rnadnavar\\modules\\nf-core\\gatk4\\getpileupsummaries\\main.nf\n",
      "Process GATK4_INDEXFEATUREFILE in filepath ./git_repos/rnadnavar\\modules\\nf-core\\gatk4\\indexfeaturefile\\main.nf\n",
      "Process GATK4_INTERVALLISTTOBED in filepath ./git_repos/rnadnavar\\modules\\nf-core\\gatk4\\intervallisttobed\\main.nf\n",
      "Process GATK4_INTERVALLISTTOOLS in filepath ./git_repos/rnadnavar\\modules\\nf-core\\gatk4\\intervallisttools\\main.nf\n",
      "Process GATK4_LEARNREADORIENTATIONMODEL in filepath ./git_repos/rnadnavar\\modules\\nf-core\\gatk4\\learnreadorientationmodel\\main.nf\n",
      "Process GATK4_MARKDUPLICATES in filepath ./git_repos/rnadnavar\\modules\\nf-core\\gatk4\\markduplicates\\main.nf\n",
      "Process GATK4_MARKDUPLICATES_SPARK in filepath ./git_repos/rnadnavar\\modules\\nf-core\\gatk4\\markduplicatesspark\\main.nf\n",
      "Process GATK4_MERGEMUTECTSTATS in filepath ./git_repos/rnadnavar\\modules\\nf-core\\gatk4\\mergemutectstats\\main.nf\n",
      "Process GATK4_MERGEVCFS in filepath ./git_repos/rnadnavar\\modules\\nf-core\\gatk4\\mergevcfs\\main.nf\n",
      "Process GATK4_MUTECT2 in filepath ./git_repos/rnadnavar\\modules\\nf-core\\gatk4\\mutect2\\main.nf\n",
      "Process GATK4_SPLITNCIGARREADS in filepath ./git_repos/rnadnavar\\modules\\nf-core\\gatk4\\splitncigarreads\\main.nf\n",
      "Process GATK4_VARIANTFILTRATION in filepath ./git_repos/rnadnavar\\modules\\nf-core\\gatk4\\variantfiltration\\main.nf\n",
      "Process GATK4_VARIANTRECALIBRATOR in filepath ./git_repos/rnadnavar\\modules\\nf-core\\gatk4\\variantrecalibrator\\main.nf\n",
      "Process GFFREAD in filepath ./git_repos/rnadnavar\\modules\\nf-core\\gffread\\main.nf\n",
      "Process GUNZIP in filepath ./git_repos/rnadnavar\\modules\\nf-core\\gunzip\\main.nf\n",
      "Process HISAT2_ALIGN in filepath ./git_repos/rnadnavar\\modules\\nf-core\\hisat2\\align\\main.nf\n",
      "Process HISAT2_BUILD in filepath ./git_repos/rnadnavar\\modules\\nf-core\\hisat2\\build\\main.nf\n",
      "Process HISAT2_EXTRACTSPLICESITES in filepath ./git_repos/rnadnavar\\modules\\nf-core\\hisat2\\extractsplicesites\\main.nf\n",
      "Process MANTA_SOMATIC in filepath ./git_repos/rnadnavar\\modules\\nf-core\\manta\\somatic\\main.nf\n",
      "Process MOSDEPTH in filepath ./git_repos/rnadnavar\\modules\\nf-core\\mosdepth\\main.nf\n",
      "Process MULTIQC in filepath ./git_repos/rnadnavar\\modules\\nf-core\\multiqc\\main.nf\n",
      "Process PICARD_FILTERSAMREADS in filepath ./git_repos/rnadnavar\\modules\\nf-core\\picard\\filtersamreads\\main.nf\n",
      "Process SAMBLASTER in filepath ./git_repos/rnadnavar\\modules\\nf-core\\samblaster\\main.nf\n",
      "Process SAMTOOLS_BAM2FQ in filepath ./git_repos/rnadnavar\\modules\\nf-core\\samtools\\bam2fq\\main.nf\n",
      "Process SAMTOOLS_COLLATEFASTQ in filepath ./git_repos/rnadnavar\\modules\\nf-core\\samtools\\collatefastq\\main.nf\n",
      "Process SAMTOOLS_CONVERT in filepath ./git_repos/rnadnavar\\modules\\nf-core\\samtools\\convert\\main.nf\n",
      "Process SAMTOOLS_FAIDX in filepath ./git_repos/rnadnavar\\modules\\nf-core\\samtools\\faidx\\main.nf\n",
      "Process SAMTOOLS_FLAGSTAT in filepath ./git_repos/rnadnavar\\modules\\nf-core\\samtools\\flagstat\\main.nf\n",
      "Process SAMTOOLS_IDXSTATS in filepath ./git_repos/rnadnavar\\modules\\nf-core\\samtools\\idxstats\\main.nf\n",
      "Process SAMTOOLS_INDEX in filepath ./git_repos/rnadnavar\\modules\\nf-core\\samtools\\index\\main.nf\n",
      "Process SAMTOOLS_MERGE in filepath ./git_repos/rnadnavar\\modules\\nf-core\\samtools\\merge\\main.nf\n",
      "Process SAMTOOLS_MPILEUP in filepath ./git_repos/rnadnavar\\modules\\nf-core\\samtools\\mpileup\\main.nf\n",
      "Process SAMTOOLS_SORT in filepath ./git_repos/rnadnavar\\modules\\nf-core\\samtools\\sort\\main.nf\n",
      "Process SAMTOOLS_STATS in filepath ./git_repos/rnadnavar\\modules\\nf-core\\samtools\\stats\\main.nf\n",
      "Process SAMTOOLS_VIEW in filepath ./git_repos/rnadnavar\\modules\\nf-core\\samtools\\view\\main.nf\n",
      "Process STAR_ALIGN in filepath ./git_repos/rnadnavar\\modules\\nf-core\\star\\align\\main.nf\n",
      "Process STAR_GENOMEGENERATE in filepath ./git_repos/rnadnavar\\modules\\nf-core\\star\\genomegenerate\\main.nf\n",
      "Process STRELKA_SOMATIC in filepath ./git_repos/rnadnavar\\modules\\nf-core\\strelka\\somatic\\main.nf\n",
      "Process TABIX_BGZIPTABIX in filepath ./git_repos/rnadnavar\\modules\\nf-core\\tabix\\bgziptabix\\main.nf\n",
      "Process TABIX_TABIX in filepath ./git_repos/rnadnavar\\modules\\nf-core\\tabix\\tabix\\main.nf\n",
      "Process UNTAR in filepath ./git_repos/rnadnavar\\modules\\nf-core\\untar\\main.nf\n",
      "Process UNZIP in filepath ./git_repos/rnadnavar\\modules\\nf-core\\unzip\\main.nf\n",
      "Process VCFTOOLS in filepath ./git_repos/rnadnavar\\modules\\nf-core\\vcftools\\main.nf\n",
      "Process SAMPLESHEET_CHECK in filepath ./git_repos/rnafusion\\modules\\local\\samplesheet_check.nf\n",
      "Process ARRIBA_DOWNLOAD in filepath ./git_repos/rnafusion\\modules\\local\\arriba\\download\\main.nf\n",
      "Process ARRIBA_VISUALISATION in filepath ./git_repos/rnafusion\\modules\\local\\arriba\\visualisation\\main.nf\n",
      "Process CONVERT2BED in filepath ./git_repos/rnafusion\\modules\\local\\convert2bed\\main.nf\n",
      "Process ENSEMBL_DOWNLOAD in filepath ./git_repos/rnafusion\\modules\\local\\ensembl\\main.nf\n",
      "Process FUSIONCATCHER in filepath ./git_repos/rnafusion\\modules\\local\\fusioncatcher\\detect\\main.nf\n",
      "Process FUSIONCATCHER_DOWNLOAD in filepath ./git_repos/rnafusion\\modules\\local\\fusioncatcher\\download\\main.nf\n",
      "Process FUSIONINSPECTOR in filepath ./git_repos/rnafusion\\modules\\local\\fusioninspector\\main.nf\n",
      "Process FUSIONREPORT in filepath ./git_repos/rnafusion\\modules\\local\\fusionreport\\detect\\main.nf\n",
      "Process FUSIONREPORT_DOWNLOAD in filepath ./git_repos/rnafusion\\modules\\local\\fusionreport\\download\\main.nf\n",
      "Process HGNC_DOWNLOAD in filepath ./git_repos/rnafusion\\modules\\local\\hgnc\\main.nf\n",
      "Process PICARD_COLLECTRNASEQMETRICS in filepath ./git_repos/rnafusion\\modules\\local\\picard\\collectrnaseqmetrics\\main.nf\n",
      "Process RRNA_TRANSCRIPTS in filepath ./git_repos/rnafusion\\modules\\local\\rrnatranscripts\\main.nf\n",
      "Process STARFUSION_BUILD in filepath ./git_repos/rnafusion\\modules\\local\\starfusion\\build\\main.nf\n",
      "Process STARFUSION in filepath ./git_repos/rnafusion\\modules\\local\\starfusion\\detect\\main.nf\n",
      "Process STARFUSION_DOWNLOAD in filepath ./git_repos/rnafusion\\modules\\local\\starfusion\\download\\main.nf\n",
      "Process GTF_TO_REFFLAT in filepath ./git_repos/rnafusion\\modules\\local\\uscs\\custom_gtftogenepred\\main.nf\n",
      "Process VCF_COLLECT in filepath ./git_repos/rnafusion\\modules\\local\\vcf_collect\\main.nf\n",
      "Process AGAT_CONVERTSPGFF2TSV in filepath ./git_repos/rnafusion\\modules\\nf-core\\agat\\convertspgff2tsv\\main.nf\n",
      "Process ARRIBA in filepath ./git_repos/rnafusion\\modules\\nf-core\\arriba\\main.nf\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Process CAT_CAT in filepath ./git_repos/rnafusion\\modules\\nf-core\\cat\\cat\\main.nf\n",
      "Process CAT_FASTQ in filepath ./git_repos/rnafusion\\modules\\nf-core\\cat\\fastq\\main.nf\n",
      "Process CUSTOM_DUMPSOFTWAREVERSIONS in filepath ./git_repos/rnafusion\\modules\\nf-core\\custom\\dumpsoftwareversions\\main.nf\n",
      "Process FASTP in filepath ./git_repos/rnafusion\\modules\\nf-core\\fastp\\main.nf\n",
      "Process FASTQC in filepath ./git_repos/rnafusion\\modules\\nf-core\\fastqc\\main.nf\n",
      "Process GATK4_BEDTOINTERVALLIST in filepath ./git_repos/rnafusion\\modules\\nf-core\\gatk4\\bedtointervallist\\main.nf\n",
      "Process GATK4_CREATESEQUENCEDICTIONARY in filepath ./git_repos/rnafusion\\modules\\nf-core\\gatk4\\createsequencedictionary\\main.nf\n",
      "Process GATK4_MARKDUPLICATES in filepath ./git_repos/rnafusion\\modules\\nf-core\\gatk4\\markduplicates\\main.nf\n",
      "Process MULTIQC in filepath ./git_repos/rnafusion\\modules\\nf-core\\multiqc\\main.nf\n",
      "Process PICARD_COLLECTINSERTSIZEMETRICS in filepath ./git_repos/rnafusion\\modules\\nf-core\\picard\\collectinsertsizemetrics\\main.nf\n",
      "Process PICARD_COLLECTWGSMETRICS in filepath ./git_repos/rnafusion\\modules\\nf-core\\picard\\collectwgsmetrics\\main.nf\n",
      "Process SAMTOOLS_FAIDX in filepath ./git_repos/rnafusion\\modules\\nf-core\\samtools\\faidx\\main.nf\n",
      "Process SAMTOOLS_INDEX in filepath ./git_repos/rnafusion\\modules\\nf-core\\samtools\\index\\main.nf\n",
      "Process SAMTOOLS_SORT in filepath ./git_repos/rnafusion\\modules\\nf-core\\samtools\\sort\\main.nf\n",
      "Process SAMTOOLS_VIEW in filepath ./git_repos/rnafusion\\modules\\nf-core\\samtools\\view\\main.nf\n",
      "Process STAR_ALIGN in filepath ./git_repos/rnafusion\\modules\\nf-core\\star\\align\\main.nf\n",
      "Process STAR_GENOMEGENERATE in filepath ./git_repos/rnafusion\\modules\\nf-core\\star\\genomegenerate\\main.nf\n",
      "Process STRINGTIE_MERGE in filepath ./git_repos/rnafusion\\modules\\nf-core\\stringtie\\merge\\main.nf\n",
      "Process STRINGTIE_STRINGTIE in filepath ./git_repos/rnafusion\\modules\\nf-core\\stringtie\\stringtie\\main.nf\n",
      "Process BEDTOOLS_GENOMECOV in filepath ./git_repos/rnaseq\\modules\\local\\bedtools_genomecov\\main.nf\n",
      "Process CAT_ADDITIONAL_FASTA in filepath ./git_repos/rnaseq\\modules\\local\\cat_additional_fasta\\main.nf\n",
      "Process DESEQ2_QC in filepath ./git_repos/rnaseq\\modules\\local\\deseq2_qc\\main.nf\n",
      "Process DUPRADAR in filepath ./git_repos/rnaseq\\modules\\local\\dupradar\\main.nf\n",
      "Process GTF2BED in filepath ./git_repos/rnaseq\\modules\\local\\gtf2bed\\main.nf\n",
      "Process GTF_FILTER in filepath ./git_repos/rnaseq\\modules\\local\\gtf_filter\\main.nf\n",
      "Process MULTIQC in filepath ./git_repos/rnaseq\\modules\\local\\multiqc\\main.nf\n",
      "Process MULTIQC_CUSTOM_BIOTYPE in filepath ./git_repos/rnaseq\\modules\\local\\multiqc_custom_biotype\\main.nf\n",
      "Process PREPROCESS_TRANSCRIPTS_FASTA_GENCODE in filepath ./git_repos/rnaseq\\modules\\local\\preprocess_transcripts_fasta_gencode\\main.nf\n",
      "Process RSEM_MERGE_COUNTS in filepath ./git_repos/rnaseq\\modules\\local\\rsem_merge_counts\\main.nf\n",
      "Process STAR_ALIGN_IGENOMES in filepath ./git_repos/rnaseq\\modules\\local\\star_align_igenomes\\main.nf\n",
      "Process STAR_GENOMEGENERATE_IGENOMES in filepath ./git_repos/rnaseq\\modules\\local\\star_genomegenerate_igenomes\\main.nf\n",
      "Process SUMMARIZEDEXPERIMENT in filepath ./git_repos/rnaseq\\modules\\local\\summarizedexperiment\\main.nf\n",
      "Process TX2GENE in filepath ./git_repos/rnaseq\\modules\\local\\tx2gene\\main.nf\n",
      "Process TXIMPORT in filepath ./git_repos/rnaseq\\modules\\local\\tximport\\main.nf\n",
      "Process UMITOOLS_PREPAREFORRSEM in filepath ./git_repos/rnaseq\\modules\\local\\umitools_prepareforrsem\\main.nf\n",
      "Process BBMAP_BBSPLIT in filepath ./git_repos/rnaseq\\modules\\nf-core\\bbmap\\bbsplit\\main.nf\n",
      "Process CAT_FASTQ in filepath ./git_repos/rnaseq\\modules\\nf-core\\cat\\fastq\\main.nf\n",
      "Process CUSTOM_DUMPSOFTWAREVERSIONS in filepath ./git_repos/rnaseq\\modules\\nf-core\\custom\\dumpsoftwareversions\\main.nf\n",
      "Process CUSTOM_GETCHROMSIZES in filepath ./git_repos/rnaseq\\modules\\nf-core\\custom\\getchromsizes\\main.nf\n",
      "Process FASTP in filepath ./git_repos/rnaseq\\modules\\nf-core\\fastp\\main.nf\n",
      "Process FASTQC in filepath ./git_repos/rnaseq\\modules\\nf-core\\fastqc\\main.nf\n",
      "Process FQ_SUBSAMPLE in filepath ./git_repos/rnaseq\\modules\\nf-core\\fq\\subsample\\main.nf\n",
      "Process GFFREAD in filepath ./git_repos/rnaseq\\modules\\nf-core\\gffread\\main.nf\n",
      "Process GUNZIP in filepath ./git_repos/rnaseq\\modules\\nf-core\\gunzip\\main.nf\n",
      "Process HISAT2_ALIGN in filepath ./git_repos/rnaseq\\modules\\nf-core\\hisat2\\align\\main.nf\n",
      "Process HISAT2_BUILD in filepath ./git_repos/rnaseq\\modules\\nf-core\\hisat2\\build\\main.nf\n",
      "Process HISAT2_EXTRACTSPLICESITES in filepath ./git_repos/rnaseq\\modules\\nf-core\\hisat2\\extractsplicesites\\main.nf\n",
      "Process KALLISTO_INDEX in filepath ./git_repos/rnaseq\\modules\\nf-core\\kallisto\\index\\main.nf\n",
      "Process KALLISTO_QUANT in filepath ./git_repos/rnaseq\\modules\\nf-core\\kallisto\\quant\\main.nf\n",
      "Process PICARD_MARKDUPLICATES in filepath ./git_repos/rnaseq\\modules\\nf-core\\picard\\markduplicates\\main.nf\n",
      "Process PRESEQ_LCEXTRAP in filepath ./git_repos/rnaseq\\modules\\nf-core\\preseq\\lcextrap\\main.nf\n",
      "Process QUALIMAP_RNASEQ in filepath ./git_repos/rnaseq\\modules\\nf-core\\qualimap\\rnaseq\\main.nf\n",
      "Process RSEM_CALCULATEEXPRESSION in filepath ./git_repos/rnaseq\\modules\\nf-core\\rsem\\calculateexpression\\main.nf\n",
      "Process RSEM_PREPAREREFERENCE in filepath ./git_repos/rnaseq\\modules\\nf-core\\rsem\\preparereference\\main.nf\n",
      "Process RSEQC_BAMSTAT in filepath ./git_repos/rnaseq\\modules\\nf-core\\rseqc\\bamstat\\main.nf\n",
      "Process RSEQC_INFEREXPERIMENT in filepath ./git_repos/rnaseq\\modules\\nf-core\\rseqc\\inferexperiment\\main.nf\n",
      "Process RSEQC_INNERDISTANCE in filepath ./git_repos/rnaseq\\modules\\nf-core\\rseqc\\innerdistance\\main.nf\n",
      "Process RSEQC_JUNCTIONANNOTATION in filepath ./git_repos/rnaseq\\modules\\nf-core\\rseqc\\junctionannotation\\main.nf\n",
      "Process RSEQC_JUNCTIONSATURATION in filepath ./git_repos/rnaseq\\modules\\nf-core\\rseqc\\junctionsaturation\\main.nf\n",
      "Process RSEQC_READDISTRIBUTION in filepath ./git_repos/rnaseq\\modules\\nf-core\\rseqc\\readdistribution\\main.nf\n",
      "Process RSEQC_READDUPLICATION in filepath ./git_repos/rnaseq\\modules\\nf-core\\rseqc\\readduplication\\main.nf\n",
      "Process RSEQC_TIN in filepath ./git_repos/rnaseq\\modules\\nf-core\\rseqc\\tin\\main.nf\n",
      "Process SALMON_INDEX in filepath ./git_repos/rnaseq\\modules\\nf-core\\salmon\\index\\main.nf\n",
      "Process SALMON_QUANT in filepath ./git_repos/rnaseq\\modules\\nf-core\\salmon\\quant\\main.nf\n",
      "Process SAMTOOLS_FLAGSTAT in filepath ./git_repos/rnaseq\\modules\\nf-core\\samtools\\flagstat\\main.nf\n",
      "Process SAMTOOLS_IDXSTATS in filepath ./git_repos/rnaseq\\modules\\nf-core\\samtools\\idxstats\\main.nf\n",
      "Process SAMTOOLS_INDEX in filepath ./git_repos/rnaseq\\modules\\nf-core\\samtools\\index\\main.nf\n",
      "Process SAMTOOLS_SORT in filepath ./git_repos/rnaseq\\modules\\nf-core\\samtools\\sort\\main.nf\n",
      "Process SAMTOOLS_STATS in filepath ./git_repos/rnaseq\\modules\\nf-core\\samtools\\stats\\main.nf\n",
      "Process SORTMERNA in filepath ./git_repos/rnaseq\\modules\\nf-core\\sortmerna\\main.nf\n",
      "Process STAR_ALIGN in filepath ./git_repos/rnaseq\\modules\\nf-core\\star\\align\\main.nf\n",
      "Process STAR_GENOMEGENERATE in filepath ./git_repos/rnaseq\\modules\\nf-core\\star\\genomegenerate\\main.nf\n",
      "Process STRINGTIE_STRINGTIE in filepath ./git_repos/rnaseq\\modules\\nf-core\\stringtie\\stringtie\\main.nf\n",
      "Process SUBREAD_FEATURECOUNTS in filepath ./git_repos/rnaseq\\modules\\nf-core\\subread\\featurecounts\\main.nf\n",
      "Process TRIMGALORE in filepath ./git_repos/rnaseq\\modules\\nf-core\\trimgalore\\main.nf\n",
      "Process UCSC_BEDCLIP in filepath ./git_repos/rnaseq\\modules\\nf-core\\ucsc\\bedclip\\main.nf\n",
      "Process UCSC_BEDGRAPHTOBIGWIG in filepath ./git_repos/rnaseq\\modules\\nf-core\\ucsc\\bedgraphtobigwig\\main.nf\n",
      "Process UMITOOLS_DEDUP in filepath ./git_repos/rnaseq\\modules\\nf-core\\umitools\\dedup\\main.nf\n",
      "Process UMITOOLS_EXTRACT in filepath ./git_repos/rnaseq\\modules\\nf-core\\umitools\\extract\\main.nf\n",
      "Process UNTAR in filepath ./git_repos/rnaseq\\modules\\nf-core\\untar\\main.nf\n",
      "Process BEDTOOLS_GENOMECOV in filepath ./git_repos/rnasplice\\modules\\local\\bedtools_genomecov.nf\n",
      "Process CONTRASTSHEET_CHECK in filepath ./git_repos/rnasplice\\modules\\local\\contrastsheet_check.nf\n",
      "Process CREATE_BAMLIST in filepath ./git_repos/rnasplice\\modules\\local\\create_bamlist.nf\n",
      "Process CREATE_BAMLIST_SINGLE in filepath ./git_repos/rnasplice\\modules\\local\\create_bamlist_single.nf\n",
      "Process DEXSEQ_ANNOTATION in filepath ./git_repos/rnasplice\\modules\\local\\dexseq_annotation.nf\n",
      "Process DEXSEQ_COUNT in filepath ./git_repos/rnasplice\\modules\\local\\dexseq_count.nf\n",
      "Process DEXSEQ_DTU in filepath ./git_repos/rnasplice\\modules\\local\\dexseq_dtu.nf\n",
      "Process DEXSEQ_EXON in filepath ./git_repos/rnasplice\\modules\\local\\dexseq_exon.nf\n",
      "Process DRIMSEQ_FILTER in filepath ./git_repos/rnasplice\\modules\\local\\drimseq_filter.nf\n",
      "Process EDGER_EXON in filepath ./git_repos/rnasplice\\modules\\local\\edger_exon.nf\n",
      "Process SUBREAD_FLATTENGTF in filepath ./git_repos/rnasplice\\modules\\local\\flattengtf.nf\n",
      "Process GFFREAD_TX2GENE in filepath ./git_repos/rnasplice\\modules\\local\\gffread_tx2gene.nf\n",
      "Process GTF_2_GFF3 in filepath ./git_repos/rnasplice\\modules\\local\\gtf_2_gff3.nf\n",
      "Process GTF_GENE_FILTER in filepath ./git_repos/rnasplice\\modules\\local\\gtf_gene_filter.nf\n",
      "Process MISO_INDEX in filepath ./git_repos/rnasplice\\modules\\local\\miso_index.nf\n",
      "Process MISO_RUN in filepath ./git_repos/rnasplice\\modules\\local\\miso_run.nf\n",
      "Process MISO_SASHIMI in filepath ./git_repos/rnasplice\\modules\\local\\miso_sashimi.nf\n",
      "Process MISO_SETTINGS in filepath ./git_repos/rnasplice\\modules\\local\\miso_settings.nf\n",
      "Process PREPROCESS_TRANSCRIPTS_FASTA_GENCODE in filepath ./git_repos/rnasplice\\modules\\local\\preprocess_transcripts_fasta_gencode.nf\n",
      "Process RMATS_POST in filepath ./git_repos/rnasplice\\modules\\local\\rmats_post.nf\n",
      "Process RMATS_POST_SINGLE in filepath ./git_repos/rnasplice\\modules\\local\\rmats_post_single.nf\n",
      "Process RMATS_PREP in filepath ./git_repos/rnasplice\\modules\\local\\rmats_prep.nf\n",
      "Process RMATS_PREP_SINGLE in filepath ./git_repos/rnasplice\\modules\\local\\rmats_prep_single.nf\n",
      "Process SAMPLESHEET_CHECK in filepath ./git_repos/rnasplice\\modules\\local\\samplesheet_check.nf\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Process STAGER in filepath ./git_repos/rnasplice\\modules\\local\\stager.nf\n",
      "Process STAR_ALIGN_IGENOMES in filepath ./git_repos/rnasplice\\modules\\local\\star_align_igenomes.nf\n",
      "Process STAR_GENOMEGENERATE_IGENOMES in filepath ./git_repos/rnasplice\\modules\\local\\star_genomegenerate_igenomes.nf\n",
      "Process CLUSTEREVENTS in filepath ./git_repos/rnasplice\\modules\\local\\suppa_clusterevents.nf\n",
      "Process CLUSTERGROUPS in filepath ./git_repos/rnasplice\\modules\\local\\suppa_clustergroups.nf\n",
      "Process DIFFSPLICE in filepath ./git_repos/rnasplice\\modules\\local\\suppa_diffsplice.nf\n",
      "Process GENERATE_EVENTS in filepath ./git_repos/rnasplice\\modules\\local\\suppa_generateevents.nf\n",
      "Process PSIPEREVENT in filepath ./git_repos/rnasplice\\modules\\local\\suppa_psiperevent.nf\n",
      "Process PSIPERISOFORM in filepath ./git_repos/rnasplice\\modules\\local\\suppa_psiperisoform.nf\n",
      "Process SPLIT_FILES in filepath ./git_repos/rnasplice\\modules\\local\\suppa_split_files.nf\n",
      "Process TXIMPORT in filepath ./git_repos/rnasplice\\modules\\local\\tximport.nf\n",
      "Process CAT_FASTQ in filepath ./git_repos/rnasplice\\modules\\nf-core\\cat\\fastq\\main.nf\n",
      "Process CUSTOM_DUMPSOFTWAREVERSIONS in filepath ./git_repos/rnasplice\\modules\\nf-core\\custom\\dumpsoftwareversions\\main.nf\n",
      "Process CUSTOM_GETCHROMSIZES in filepath ./git_repos/rnasplice\\modules\\nf-core\\custom\\getchromsizes\\main.nf\n",
      "Process FASTQC in filepath ./git_repos/rnasplice\\modules\\nf-core\\fastqc\\main.nf\n",
      "Process GFFREAD in filepath ./git_repos/rnasplice\\modules\\nf-core\\gffread\\main.nf\n",
      "Process GUNZIP in filepath ./git_repos/rnasplice\\modules\\nf-core\\gunzip\\main.nf\n",
      "Process MULTIQC in filepath ./git_repos/rnasplice\\modules\\nf-core\\multiqc\\main.nf\n",
      "Process RSEM_PREPAREREFERENCE in filepath ./git_repos/rnasplice\\modules\\nf-core\\rsem\\preparereference\\main.nf\n",
      "Process SALMON_INDEX in filepath ./git_repos/rnasplice\\modules\\nf-core\\salmon\\index\\main.nf\n",
      "Process SALMON_QUANT in filepath ./git_repos/rnasplice\\modules\\nf-core\\salmon\\quant\\main.nf\n",
      "Process SAMTOOLS_FLAGSTAT in filepath ./git_repos/rnasplice\\modules\\nf-core\\samtools\\flagstat\\main.nf\n",
      "Process SAMTOOLS_IDXSTATS in filepath ./git_repos/rnasplice\\modules\\nf-core\\samtools\\idxstats\\main.nf\n",
      "Process SAMTOOLS_INDEX in filepath ./git_repos/rnasplice\\modules\\nf-core\\samtools\\index\\main.nf\n",
      "Process SAMTOOLS_SORT in filepath ./git_repos/rnasplice\\modules\\nf-core\\samtools\\sort\\main.nf\n",
      "Process SAMTOOLS_STATS in filepath ./git_repos/rnasplice\\modules\\nf-core\\samtools\\stats\\main.nf\n",
      "Process STAR_ALIGN in filepath ./git_repos/rnasplice\\modules\\nf-core\\star\\align\\main.nf\n",
      "Process STAR_GENOMEGENERATE in filepath ./git_repos/rnasplice\\modules\\nf-core\\star\\genomegenerate\\main.nf\n",
      "Process SUBREAD_FEATURECOUNTS in filepath ./git_repos/rnasplice\\modules\\nf-core\\subread\\featurecounts\\main.nf\n",
      "Process TRIMGALORE in filepath ./git_repos/rnasplice\\modules\\nf-core\\trimgalore\\main.nf\n",
      "Process UCSC_BEDCLIP in filepath ./git_repos/rnasplice\\modules\\nf-core\\ucsc\\bedclip\\main.nf\n",
      "Process UCSC_BEDGRAPHTOBIGWIG in filepath ./git_repos/rnasplice\\modules\\nf-core\\ucsc\\bedgraphtobigwig\\main.nf\n",
      "Process UMITOOLS_EXTRACT in filepath ./git_repos/rnasplice\\modules\\nf-core\\umitools\\extract\\main.nf\n",
      "Process UNTAR in filepath ./git_repos/rnasplice\\modules\\nf-core\\untar\\main.nf\n",
      "Process GTF2BED in filepath ./git_repos/rnavar\\modules\\local\\gtf2bed.nf\n",
      "Process SAMPLESHEET_CHECK in filepath ./git_repos/rnavar\\modules\\local\\samplesheet_check.nf\n",
      "Process CAT_FASTQ in filepath ./git_repos/rnavar\\modules\\nf-core\\modules\\cat\\fastq\\main.nf\n",
      "Process CUSTOM_DUMPSOFTWAREVERSIONS in filepath ./git_repos/rnavar\\modules\\nf-core\\modules\\custom\\dumpsoftwareversions\\main.nf\n",
      "Process ENSEMBLVEP in filepath ./git_repos/rnavar\\modules\\nf-core\\modules\\ensemblvep\\main.nf\n",
      "Process FASTQC in filepath ./git_repos/rnavar\\modules\\nf-core\\modules\\fastqc\\main.nf\n",
      "Process GATK4_APPLYBQSR in filepath ./git_repos/rnavar\\modules\\nf-core\\modules\\gatk4\\applybqsr\\main.nf\n",
      "Process GATK4_BASERECALIBRATOR in filepath ./git_repos/rnavar\\modules\\nf-core\\modules\\gatk4\\baserecalibrator\\main.nf\n",
      "Process GATK4_BEDTOINTERVALLIST in filepath ./git_repos/rnavar\\modules\\nf-core\\modules\\gatk4\\bedtointervallist\\main.nf\n",
      "Process GATK4_CREATESEQUENCEDICTIONARY in filepath ./git_repos/rnavar\\modules\\nf-core\\modules\\gatk4\\createsequencedictionary\\main.nf\n",
      "Process GATK4_HAPLOTYPECALLER in filepath ./git_repos/rnavar\\modules\\nf-core\\modules\\gatk4\\haplotypecaller\\main.nf\n",
      "Process GATK4_INDEXFEATUREFILE in filepath ./git_repos/rnavar\\modules\\nf-core\\modules\\gatk4\\indexfeaturefile\\main.nf\n",
      "Process GATK4_INTERVALLISTTOOLS in filepath ./git_repos/rnavar\\modules\\nf-core\\modules\\gatk4\\intervallisttools\\main.nf\n",
      "Process GATK4_MARKDUPLICATES in filepath ./git_repos/rnavar\\modules\\nf-core\\modules\\gatk4\\markduplicates\\main.nf\n",
      "Process GATK4_MERGEVCFS in filepath ./git_repos/rnavar\\modules\\nf-core\\modules\\gatk4\\mergevcfs\\main.nf\n",
      "Process GATK4_SPLITNCIGARREADS in filepath ./git_repos/rnavar\\modules\\nf-core\\modules\\gatk4\\splitncigarreads\\main.nf\n",
      "Process GATK4_VARIANTFILTRATION in filepath ./git_repos/rnavar\\modules\\nf-core\\modules\\gatk4\\variantfiltration\\main.nf\n",
      "Process GFFREAD in filepath ./git_repos/rnavar\\modules\\nf-core\\modules\\gffread\\main.nf\n",
      "Process GUNZIP in filepath ./git_repos/rnavar\\modules\\nf-core\\modules\\gunzip\\main.nf\n",
      "Process MULTIQC in filepath ./git_repos/rnavar\\modules\\nf-core\\modules\\multiqc\\main.nf\n",
      "Process SAMTOOLS_FAIDX in filepath ./git_repos/rnavar\\modules\\nf-core\\modules\\samtools\\faidx\\main.nf\n",
      "Process SAMTOOLS_FLAGSTAT in filepath ./git_repos/rnavar\\modules\\nf-core\\modules\\samtools\\flagstat\\main.nf\n",
      "Process SAMTOOLS_IDXSTATS in filepath ./git_repos/rnavar\\modules\\nf-core\\modules\\samtools\\idxstats\\main.nf\n",
      "Process SAMTOOLS_INDEX in filepath ./git_repos/rnavar\\modules\\nf-core\\modules\\samtools\\index\\main.nf\n",
      "Process SAMTOOLS_MERGE in filepath ./git_repos/rnavar\\modules\\nf-core\\modules\\samtools\\merge\\main.nf\n",
      "Process SAMTOOLS_SORT in filepath ./git_repos/rnavar\\modules\\nf-core\\modules\\samtools\\sort\\main.nf\n",
      "Process SAMTOOLS_STATS in filepath ./git_repos/rnavar\\modules\\nf-core\\modules\\samtools\\stats\\main.nf\n",
      "Process SNPEFF in filepath ./git_repos/rnavar\\modules\\nf-core\\modules\\snpeff\\main.nf\n",
      "Process STAR_ALIGN in filepath ./git_repos/rnavar\\modules\\nf-core\\modules\\star\\align\\main.nf\n",
      "Process STAR_GENOMEGENERATE in filepath ./git_repos/rnavar\\modules\\nf-core\\modules\\star\\genomegenerate\\main.nf\n",
      "Process TABIX_BGZIPTABIX in filepath ./git_repos/rnavar\\modules\\nf-core\\modules\\tabix\\bgziptabix\\main.nf\n",
      "Process TABIX_TABIX in filepath ./git_repos/rnavar\\modules\\nf-core\\modules\\tabix\\tabix\\main.nf\n",
      "Process UNTAR in filepath ./git_repos/rnavar\\modules\\nf-core\\modules\\untar\\main.nf\n",
      "Process CUT_SIZES_GENOME in filepath ./git_repos/sammyseq\\modules\\local\\chromosomes_size.nf\n",
      "Process SAMPLESHEET_CHECK in filepath ./git_repos/sammyseq\\modules\\local\\samplesheet_check.nf\n",
      "Process RTWOSAMPLESMLE in filepath ./git_repos/sammyseq\\modules\\local\\rtwosamplesmle\\main.nf\n",
      "Process BWA_ALN in filepath ./git_repos/sammyseq\\modules\\nf-core\\bwa\\aln\\main.nf\n",
      "Process BWA_INDEX in filepath ./git_repos/sammyseq\\modules\\nf-core\\bwa\\index\\main.nf\n",
      "Process BWA_SAMPE in filepath ./git_repos/sammyseq\\modules\\nf-core\\bwa\\sampe\\main.nf\n",
      "Process BWA_SAMSE in filepath ./git_repos/sammyseq\\modules\\nf-core\\bwa\\samse\\main.nf\n",
      "Process CAT_FASTQ in filepath ./git_repos/sammyseq\\modules\\nf-core\\cat\\fastq\\main.nf\n",
      "Process CUSTOM_DUMPSOFTWAREVERSIONS in filepath ./git_repos/sammyseq\\modules\\nf-core\\custom\\dumpsoftwareversions\\main.nf\n",
      "Process CUSTOM_DUMPSOFTWAREVERSIONS in filepath ./git_repos/sammyseq\\modules\\nf-core\\custom\\dumpsoftwareversions\\dumpsoftwareversions\\main.nf\n",
      "Process DEEPTOOLS_BAMCOVERAGE in filepath ./git_repos/sammyseq\\modules\\nf-core\\deeptools\\bamcoverage\\main.nf\n",
      "Process FASTQC in filepath ./git_repos/sammyseq\\modules\\nf-core\\fastqc\\main.nf\n",
      "Process GUNZIP in filepath ./git_repos/sammyseq\\modules\\nf-core\\gunzip\\main.nf\n",
      "Process MULTIQC in filepath ./git_repos/sammyseq\\modules\\nf-core\\multiqc\\main.nf\n",
      "Process PICARD_MARKDUPLICATES in filepath ./git_repos/sammyseq\\modules\\nf-core\\picard\\markduplicates\\main.nf\n",
      "Process SAMTOOLS_FAIDX in filepath ./git_repos/sammyseq\\modules\\nf-core\\samtools\\faidx\\main.nf\n",
      "Process SAMTOOLS_FLAGSTAT in filepath ./git_repos/sammyseq\\modules\\nf-core\\samtools\\flagstat\\main.nf\n",
      "Process SAMTOOLS_IDXSTATS in filepath ./git_repos/sammyseq\\modules\\nf-core\\samtools\\idxstats\\main.nf\n",
      "Process SAMTOOLS_INDEX in filepath ./git_repos/sammyseq\\modules\\nf-core\\samtools\\index\\main.nf\n",
      "Process SAMTOOLS_STATS in filepath ./git_repos/sammyseq\\modules\\nf-core\\samtools\\stats\\main.nf\n",
      "Process TRIMMOMATIC in filepath ./git_repos/sammyseq\\modules\\nf-core\\trimmomatic\\main.nf\n",
      "Process UNTAR in filepath ./git_repos/sammyseq\\modules\\nf-core\\untar\\main.nf\n",
      "Process CUT_SIZES_GENOME in filepath ./git_repos/sammyseq\\subworkflows\\local\\faidx_cut_chr_size.nf\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Process ADD_INFO_TO_VCF in filepath ./git_repos/sarek\\modules\\local\\add_info_to_vcf\\main.nf\n",
      "Process BUILD_INTERVALS in filepath ./git_repos/sarek\\modules\\local\\build_intervals\\main.nf\n",
      "Process CREATE_INTERVALS_BED in filepath ./git_repos/sarek\\modules\\local\\create_intervals_bed\\main.nf\n",
      "Process ASCAT in filepath ./git_repos/sarek\\modules\\nf-core\\ascat\\main.nf\n",
      "Process BCFTOOLS_ANNOTATE in filepath ./git_repos/sarek\\modules\\nf-core\\bcftools\\annotate\\main.nf\n",
      "Process BCFTOOLS_CONCAT in filepath ./git_repos/sarek\\modules\\nf-core\\bcftools\\concat\\main.nf\n",
      "Process BCFTOOLS_MPILEUP in filepath ./git_repos/sarek\\modules\\nf-core\\bcftools\\mpileup\\main.nf\n",
      "Process BCFTOOLS_SORT in filepath ./git_repos/sarek\\modules\\nf-core\\bcftools\\sort\\main.nf\n",
      "Process BCFTOOLS_STATS in filepath ./git_repos/sarek\\modules\\nf-core\\bcftools\\stats\\main.nf\n",
      "Process BWA_INDEX in filepath ./git_repos/sarek\\modules\\nf-core\\bwa\\index\\main.nf\n",
      "Process BWA_MEM in filepath ./git_repos/sarek\\modules\\nf-core\\bwa\\mem\\main.nf\n",
      "Process BWAMEM2_INDEX in filepath ./git_repos/sarek\\modules\\nf-core\\bwamem2\\index\\main.nf\n",
      "Process BWAMEM2_MEM in filepath ./git_repos/sarek\\modules\\nf-core\\bwamem2\\mem\\main.nf\n",
      "Process CAT_CAT in filepath ./git_repos/sarek\\modules\\nf-core\\cat\\cat\\main.nf\n",
      "Process CAT_FASTQ in filepath ./git_repos/sarek\\modules\\nf-core\\cat\\fastq\\main.nf\n",
      "Process CNVKIT_ANTITARGET in filepath ./git_repos/sarek\\modules\\nf-core\\cnvkit\\antitarget\\main.nf\n",
      "Process CNVKIT_BATCH in filepath ./git_repos/sarek\\modules\\nf-core\\cnvkit\\batch\\main.nf\n",
      "Process CNVKIT_GENEMETRICS in filepath ./git_repos/sarek\\modules\\nf-core\\cnvkit\\genemetrics\\main.nf\n",
      "Process CNVKIT_REFERENCE in filepath ./git_repos/sarek\\modules\\nf-core\\cnvkit\\reference\\main.nf\n",
      "Process CONTROLFREEC_ASSESSSIGNIFICANCE in filepath ./git_repos/sarek\\modules\\nf-core\\controlfreec\\assesssignificance\\main.nf\n",
      "Process CONTROLFREEC_FREEC in filepath ./git_repos/sarek\\modules\\nf-core\\controlfreec\\freec\\main.nf\n",
      "Process CONTROLFREEC_FREEC2BED in filepath ./git_repos/sarek\\modules\\nf-core\\controlfreec\\freec2bed\\main.nf\n",
      "Process CONTROLFREEC_FREEC2CIRCOS in filepath ./git_repos/sarek\\modules\\nf-core\\controlfreec\\freec2circos\\main.nf\n",
      "Process CONTROLFREEC_MAKEGRAPH in filepath ./git_repos/sarek\\modules\\nf-core\\controlfreec\\makegraph\\main.nf\n",
      "Process CUSTOM_DUMPSOFTWAREVERSIONS in filepath ./git_repos/sarek\\modules\\nf-core\\custom\\dumpsoftwareversions\\main.nf\n",
      "Process DEEPVARIANT in filepath ./git_repos/sarek\\modules\\nf-core\\deepvariant\\main.nf\n",
      "Process DRAGMAP_ALIGN in filepath ./git_repos/sarek\\modules\\nf-core\\dragmap\\align\\main.nf\n",
      "Process DRAGMAP_HASHTABLE in filepath ./git_repos/sarek\\modules\\nf-core\\dragmap\\hashtable\\main.nf\n",
      "Process ENSEMBLVEP_DOWNLOAD in filepath ./git_repos/sarek\\modules\\nf-core\\ensemblvep\\download\\main.nf\n",
      "Process ENSEMBLVEP_VEP in filepath ./git_repos/sarek\\modules\\nf-core\\ensemblvep\\vep\\main.nf\n",
      "Process FASTP in filepath ./git_repos/sarek\\modules\\nf-core\\fastp\\main.nf\n",
      "Process FASTQC in filepath ./git_repos/sarek\\modules\\nf-core\\fastqc\\main.nf\n",
      "Process FGBIO_CALLMOLECULARCONSENSUSREADS in filepath ./git_repos/sarek\\modules\\nf-core\\fgbio\\callmolecularconsensusreads\\main.nf\n",
      "Process FGBIO_FASTQTOBAM in filepath ./git_repos/sarek\\modules\\nf-core\\fgbio\\fastqtobam\\main.nf\n",
      "Process FGBIO_GROUPREADSBYUMI in filepath ./git_repos/sarek\\modules\\nf-core\\fgbio\\groupreadsbyumi\\main.nf\n",
      "Process FREEBAYES in filepath ./git_repos/sarek\\modules\\nf-core\\freebayes\\main.nf\n",
      "Process GATK4_APPLYBQSR in filepath ./git_repos/sarek\\modules\\nf-core\\gatk4\\applybqsr\\main.nf\n",
      "Process GATK4_APPLYVQSR in filepath ./git_repos/sarek\\modules\\nf-core\\gatk4\\applyvqsr\\main.nf\n",
      "Process GATK4_BASERECALIBRATOR in filepath ./git_repos/sarek\\modules\\nf-core\\gatk4\\baserecalibrator\\main.nf\n",
      "Process GATK4_CALCULATECONTAMINATION in filepath ./git_repos/sarek\\modules\\nf-core\\gatk4\\calculatecontamination\\main.nf\n",
      "Process GATK4_CNNSCOREVARIANTS in filepath ./git_repos/sarek\\modules\\nf-core\\gatk4\\cnnscorevariants\\main.nf\n",
      "Process GATK4_CREATESEQUENCEDICTIONARY in filepath ./git_repos/sarek\\modules\\nf-core\\gatk4\\createsequencedictionary\\main.nf\n",
      "Process GATK4_ESTIMATELIBRARYCOMPLEXITY in filepath ./git_repos/sarek\\modules\\nf-core\\gatk4\\estimatelibrarycomplexity\\main.nf\n",
      "Process GATK4_FILTERMUTECTCALLS in filepath ./git_repos/sarek\\modules\\nf-core\\gatk4\\filtermutectcalls\\main.nf\n",
      "Process GATK4_FILTERVARIANTTRANCHES in filepath ./git_repos/sarek\\modules\\nf-core\\gatk4\\filtervarianttranches\\main.nf\n",
      "Process GATK4_GATHERBQSRREPORTS in filepath ./git_repos/sarek\\modules\\nf-core\\gatk4\\gatherbqsrreports\\main.nf\n",
      "Process GATK4_GATHERPILEUPSUMMARIES in filepath ./git_repos/sarek\\modules\\nf-core\\gatk4\\gatherpileupsummaries\\main.nf\n",
      "Process GATK4_GENOMICSDBIMPORT in filepath ./git_repos/sarek\\modules\\nf-core\\gatk4\\genomicsdbimport\\main.nf\n",
      "Process GATK4_GENOTYPEGVCFS in filepath ./git_repos/sarek\\modules\\nf-core\\gatk4\\genotypegvcfs\\main.nf\n",
      "Process GATK4_GETPILEUPSUMMARIES in filepath ./git_repos/sarek\\modules\\nf-core\\gatk4\\getpileupsummaries\\main.nf\n",
      "Process GATK4_HAPLOTYPECALLER in filepath ./git_repos/sarek\\modules\\nf-core\\gatk4\\haplotypecaller\\main.nf\n",
      "Process GATK4_INTERVALLISTTOBED in filepath ./git_repos/sarek\\modules\\nf-core\\gatk4\\intervallisttobed\\main.nf\n",
      "Process GATK4_LEARNREADORIENTATIONMODEL in filepath ./git_repos/sarek\\modules\\nf-core\\gatk4\\learnreadorientationmodel\\main.nf\n",
      "Process GATK4_MARKDUPLICATES in filepath ./git_repos/sarek\\modules\\nf-core\\gatk4\\markduplicates\\main.nf\n",
      "Process GATK4_MERGEMUTECTSTATS in filepath ./git_repos/sarek\\modules\\nf-core\\gatk4\\mergemutectstats\\main.nf\n",
      "Process GATK4_MERGEVCFS in filepath ./git_repos/sarek\\modules\\nf-core\\gatk4\\mergevcfs\\main.nf\n",
      "Process GATK4_MUTECT2 in filepath ./git_repos/sarek\\modules\\nf-core\\gatk4\\mutect2\\main.nf\n",
      "Process GATK4_VARIANTRECALIBRATOR in filepath ./git_repos/sarek\\modules\\nf-core\\gatk4\\variantrecalibrator\\main.nf\n",
      "Process GATK4SPARK_APPLYBQSR in filepath ./git_repos/sarek\\modules\\nf-core\\gatk4spark\\applybqsr\\main.nf\n",
      "Process GATK4SPARK_BASERECALIBRATOR in filepath ./git_repos/sarek\\modules\\nf-core\\gatk4spark\\baserecalibrator\\main.nf\n",
      "Process GATK4SPARK_MARKDUPLICATES in filepath ./git_repos/sarek\\modules\\nf-core\\gatk4spark\\markduplicates\\main.nf\n",
      "Process MANTA_GERMLINE in filepath ./git_repos/sarek\\modules\\nf-core\\manta\\germline\\main.nf\n",
      "Process MANTA_SOMATIC in filepath ./git_repos/sarek\\modules\\nf-core\\manta\\somatic\\main.nf\n",
      "Process MANTA_TUMORONLY in filepath ./git_repos/sarek\\modules\\nf-core\\manta\\tumoronly\\main.nf\n",
      "Process MOSDEPTH in filepath ./git_repos/sarek\\modules\\nf-core\\mosdepth\\main.nf\n",
      "Process MSISENSORPRO_MSISOMATIC in filepath ./git_repos/sarek\\modules\\nf-core\\msisensorpro\\msisomatic\\main.nf\n",
      "Process MSISENSORPRO_SCAN in filepath ./git_repos/sarek\\modules\\nf-core\\msisensorpro\\scan\\main.nf\n",
      "Process MULTIQC in filepath ./git_repos/sarek\\modules\\nf-core\\multiqc\\main.nf\n",
      "Process NGSCHECKMATE_NCM in filepath ./git_repos/sarek\\modules\\nf-core\\ngscheckmate\\ncm\\main.nf\n",
      "Process SAMBLASTER in filepath ./git_repos/sarek\\modules\\nf-core\\samblaster\\main.nf\n",
      "Process SAMTOOLS_BAM2FQ in filepath ./git_repos/sarek\\modules\\nf-core\\samtools\\bam2fq\\main.nf\n",
      "Process SAMTOOLS_COLLATEFASTQ in filepath ./git_repos/sarek\\modules\\nf-core\\samtools\\collatefastq\\main.nf\n",
      "Process SAMTOOLS_CONVERT in filepath ./git_repos/sarek\\modules\\nf-core\\samtools\\convert\\main.nf\n",
      "Process SAMTOOLS_FAIDX in filepath ./git_repos/sarek\\modules\\nf-core\\samtools\\faidx\\main.nf\n",
      "Process SAMTOOLS_INDEX in filepath ./git_repos/sarek\\modules\\nf-core\\samtools\\index\\main.nf\n",
      "Process SAMTOOLS_MERGE in filepath ./git_repos/sarek\\modules\\nf-core\\samtools\\merge\\main.nf\n",
      "Process SAMTOOLS_MPILEUP in filepath ./git_repos/sarek\\modules\\nf-core\\samtools\\mpileup\\main.nf\n",
      "Process SAMTOOLS_STATS in filepath ./git_repos/sarek\\modules\\nf-core\\samtools\\stats\\main.nf\n",
      "Process SAMTOOLS_VIEW in filepath ./git_repos/sarek\\modules\\nf-core\\samtools\\view\\main.nf\n",
      "Process SENTIEON_APPLYVARCAL in filepath ./git_repos/sarek\\modules\\nf-core\\sentieon\\applyvarcal\\main.nf\n",
      "Process SENTIEON_BWAMEM in filepath ./git_repos/sarek\\modules\\nf-core\\sentieon\\bwamem\\main.nf\n",
      "Process SENTIEON_DEDUP in filepath ./git_repos/sarek\\modules\\nf-core\\sentieon\\dedup\\main.nf\n",
      "Process SENTIEON_DNAMODELAPPLY in filepath ./git_repos/sarek\\modules\\nf-core\\sentieon\\dnamodelapply\\main.nf\n",
      "Process SENTIEON_DNASCOPE in filepath ./git_repos/sarek\\modules\\nf-core\\sentieon\\dnascope\\main.nf\n",
      "Process SENTIEON_GVCFTYPER in filepath ./git_repos/sarek\\modules\\nf-core\\sentieon\\gvcftyper\\main.nf\n",
      "Process SENTIEON_HAPLOTYPER in filepath ./git_repos/sarek\\modules\\nf-core\\sentieon\\haplotyper\\main.nf\n",
      "Process SENTIEON_VARCAL in filepath ./git_repos/sarek\\modules\\nf-core\\sentieon\\varcal\\main.nf\n",
      "Process SNPEFF_DOWNLOAD in filepath ./git_repos/sarek\\modules\\nf-core\\snpeff\\download\\main.nf\n",
      "Process SNPEFF_SNPEFF in filepath ./git_repos/sarek\\modules\\nf-core\\snpeff\\snpeff\\main.nf\n",
      "Process STRELKA_GERMLINE in filepath ./git_repos/sarek\\modules\\nf-core\\strelka\\germline\\main.nf\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Process STRELKA_SOMATIC in filepath ./git_repos/sarek\\modules\\nf-core\\strelka\\somatic\\main.nf\n",
      "Process SVDB_MERGE in filepath ./git_repos/sarek\\modules\\nf-core\\svdb\\merge\\main.nf\n",
      "Process TABIX_BGZIPTABIX in filepath ./git_repos/sarek\\modules\\nf-core\\tabix\\bgziptabix\\main.nf\n",
      "Process TABIX_TABIX in filepath ./git_repos/sarek\\modules\\nf-core\\tabix\\tabix\\main.nf\n",
      "Process TIDDIT_SV in filepath ./git_repos/sarek\\modules\\nf-core\\tiddit\\sv\\main.nf\n",
      "Process UNTAR in filepath ./git_repos/sarek\\modules\\nf-core\\untar\\main.nf\n",
      "Process UNZIP in filepath ./git_repos/sarek\\modules\\nf-core\\unzip\\main.nf\n",
      "Process VCFTOOLS in filepath ./git_repos/sarek\\modules\\nf-core\\vcftools\\main.nf\n",
      "Process GET_SOFTWARE_VERSIONS in filepath ./git_repos/scflow\\modules\\local\\get_software_versions.nf\n",
      "Process SAMPLESHEET_CHECK in filepath ./git_repos/scflow\\modules\\local\\samplesheet_check.nf\n",
      "Process SCFLOW_CHECKINPUTS in filepath ./git_repos/scflow\\modules\\local\\process\\scflow\\checkinputs.nf\n",
      "Process SCFLOW_CLUSTER in filepath ./git_repos/scflow\\modules\\local\\process\\scflow\\cluster.nf\n",
      "Process SCFLOW_DGE in filepath ./git_repos/scflow\\modules\\local\\process\\scflow\\dge.nf\n",
      "Process SCFLOW_DIRICHLET in filepath ./git_repos/scflow\\modules\\local\\process\\scflow\\dirichlet.nf\n",
      "Process SCFLOW_FINALIZE in filepath ./git_repos/scflow\\modules\\local\\process\\scflow\\finalize.nf\n",
      "Process SCFLOW_INTEGRATE in filepath ./git_repos/scflow\\modules\\local\\process\\scflow\\integrate.nf\n",
      "Process SCFLOW_IPA in filepath ./git_repos/scflow\\modules\\local\\process\\scflow\\ipa.nf\n",
      "Process SCFLOW_MAPCELLTYPES in filepath ./git_repos/scflow\\modules\\local\\process\\scflow\\mapcelltypes.nf\n",
      "Process SCFLOW_MERGE in filepath ./git_repos/scflow\\modules\\local\\process\\scflow\\merge.nf\n",
      "Process SCFLOW_MERGEQCTABLES in filepath ./git_repos/scflow\\modules\\local\\process\\scflow\\mergeqctables.nf\n",
      "Process SCFLOW_PLOTREDDIMGENES in filepath ./git_repos/scflow\\modules\\local\\process\\scflow\\plotreddimgenes.nf\n",
      "Process SCFLOW_QC in filepath ./git_repos/scflow\\modules\\local\\process\\scflow\\qc.nf\n",
      "Process SCFLOW_REDUCEDIMS in filepath ./git_repos/scflow\\modules\\local\\process\\scflow\\reducedims.nf\n",
      "Process SCFLOW_REPORTINTEGRATED in filepath ./git_repos/scflow\\modules\\local\\process\\scflow\\reportintegrated.nf\n",
      "Process ALEVINQC in filepath ./git_repos/scrnaseq\\modules\\local\\alevinqc.nf\n",
      "Process CONCAT_H5AD in filepath ./git_repos/scrnaseq\\modules\\local\\concat_h5ad.nf\n",
      "Process GENE_MAP in filepath ./git_repos/scrnaseq\\modules\\local\\gene_map.nf\n",
      "Process GFFREAD_TRANSCRIPTOME in filepath ./git_repos/scrnaseq\\modules\\local\\gffread_transcriptome.nf\n",
      "Process GTF_GENE_FILTER in filepath ./git_repos/scrnaseq\\modules\\local\\gtf_gene_filter.nf\n",
      "Process MTX_TO_H5AD in filepath ./git_repos/scrnaseq\\modules\\local\\mtx_to_h5ad.nf\n",
      "Process MTX_TO_SEURAT in filepath ./git_repos/scrnaseq\\modules\\local\\mtx_to_seurat.nf\n",
      "Process SAMPLESHEET_CHECK in filepath ./git_repos/scrnaseq\\modules\\local\\samplesheet_check.nf\n",
      "Process SIMPLEAF_INDEX in filepath ./git_repos/scrnaseq\\modules\\local\\simpleaf_index.nf\n",
      "Process SIMPLEAF_QUANT in filepath ./git_repos/scrnaseq\\modules\\local\\simpleaf_quant.nf\n",
      "Process STAR_ALIGN in filepath ./git_repos/scrnaseq\\modules\\local\\star_align.nf\n",
      "Process CELLRANGER_COUNT in filepath ./git_repos/scrnaseq\\modules\\nf-core\\cellranger\\count\\main.nf\n",
      "Process CELLRANGER_MKGTF in filepath ./git_repos/scrnaseq\\modules\\nf-core\\cellranger\\mkgtf\\main.nf\n",
      "Process CELLRANGER_MKREF in filepath ./git_repos/scrnaseq\\modules\\nf-core\\cellranger\\mkref\\main.nf\n",
      "Process CELLRANGERARC_COUNT in filepath ./git_repos/scrnaseq\\modules\\nf-core\\cellrangerarc\\count\\main.nf\n",
      "Process CELLRANGERARC_MKGTF in filepath ./git_repos/scrnaseq\\modules\\nf-core\\cellrangerarc\\mkgtf\\main.nf\n",
      "Process CELLRANGERARC_MKREF in filepath ./git_repos/scrnaseq\\modules\\nf-core\\cellrangerarc\\mkref\\main.nf\n",
      "Process CUSTOM_DUMPSOFTWAREVERSIONS in filepath ./git_repos/scrnaseq\\modules\\nf-core\\custom\\dumpsoftwareversions\\main.nf\n",
      "Process FASTQC in filepath ./git_repos/scrnaseq\\modules\\nf-core\\fastqc\\main.nf\n",
      "Process GFFREAD in filepath ./git_repos/scrnaseq\\modules\\nf-core\\gffread\\main.nf\n",
      "Process GUNZIP in filepath ./git_repos/scrnaseq\\modules\\nf-core\\gunzip\\main.nf\n",
      "Process KALLISTOBUSTOOLS_COUNT in filepath ./git_repos/scrnaseq\\modules\\nf-core\\kallistobustools\\count\\main.nf\n",
      "Process KALLISTOBUSTOOLS_REF in filepath ./git_repos/scrnaseq\\modules\\nf-core\\kallistobustools\\ref\\main.nf\n",
      "Process MULTIQC in filepath ./git_repos/scrnaseq\\modules\\nf-core\\multiqc\\main.nf\n",
      "Process STAR_GENOMEGENERATE in filepath ./git_repos/scrnaseq\\modules\\nf-core\\star\\genomegenerate\\main.nf\n",
      "Process UNIVERSC in filepath ./git_repos/scrnaseq\\modules\\nf-core\\universc\\main.nf\n",
      "Process UNZIP in filepath ./git_repos/scrnaseq\\modules\\nf-core\\unzip\\main.nf\n",
      "Process gunzipFasta in filepath ./git_repos/slamseq\\main.nf\n",
      "Process gtf2bed in filepath ./git_repos/slamseq\\main.nf\n",
      "Process get_software_versions in filepath ./git_repos/slamseq\\main.nf\n",
      "Process checkDesign in filepath ./git_repos/slamseq\\main.nf\n",
      "Process trim in filepath ./git_repos/slamseq\\main.nf\n",
      "Process map in filepath ./git_repos/slamseq\\main.nf\n",
      "Process filter in filepath ./git_repos/slamseq\\main.nf\n",
      "Process snp in filepath ./git_repos/slamseq\\main.nf\n",
      "Process count in filepath ./git_repos/slamseq\\main.nf\n",
      "Process collapse in filepath ./git_repos/slamseq\\main.nf\n",
      "Process rates in filepath ./git_repos/slamseq\\main.nf\n",
      "Process utrrates in filepath ./git_repos/slamseq\\main.nf\n",
      "Process tcperreadpos in filepath ./git_repos/slamseq\\main.nf\n",
      "Process tcperutrpos in filepath ./git_repos/slamseq\\main.nf\n",
      "Process summary in filepath ./git_repos/slamseq\\main.nf\n",
      "Process deseq2 in filepath ./git_repos/slamseq\\main.nf\n",
      "Process multiqc in filepath ./git_repos/slamseq\\main.nf\n",
      "Process output_documentation in filepath ./git_repos/slamseq\\main.nf\n",
      "Process makeSTARindex in filepath ./git_repos/smartseq2\\main.nf\n",
      "Process make_rsem_reference in filepath ./git_repos/smartseq2\\main.nf\n",
      "Process fastqc in filepath ./git_repos/smartseq2\\main.nf\n",
      "Process STAR in filepath ./git_repos/smartseq2\\main.nf\n",
      "Process featureCounts in filepath ./git_repos/smartseq2\\main.nf\n",
      "Process summarize_FC in filepath ./git_repos/smartseq2\\main.nf\n",
      "Process make_matrices_fc in filepath ./git_repos/smartseq2\\main.nf\n",
      "Process rsem in filepath ./git_repos/smartseq2\\main.nf\n",
      "Process summmarize_TPM in filepath ./git_repos/smartseq2\\main.nf\n",
      "Process make_matrices_tpm in filepath ./git_repos/smartseq2\\main.nf\n",
      "Process TraCeR in filepath ./git_repos/smartseq2\\main.nf\n",
      "Process TCR_summary in filepath ./git_repos/smartseq2\\main.nf\n",
      "Process BraCeR in filepath ./git_repos/smartseq2\\main.nf\n",
      "Process BCR_summary in filepath ./git_repos/smartseq2\\main.nf\n",
      "Process get_software_versions in filepath ./git_repos/smartseq2\\main.nf\n",
      "Process multiqc in filepath ./git_repos/smartseq2\\main.nf\n",
      "Process BLAT_MIRNA in filepath ./git_repos/smrnaseq\\modules\\local\\blat_mirna.nf\n",
      "Process INDEX_CONTAMINANTS in filepath ./git_repos/smrnaseq\\modules\\local\\bowtie_contaminants.nf\n",
      "Process INDEX_GENOME in filepath ./git_repos/smrnaseq\\modules\\local\\bowtie_genome.nf\n",
      "Process BOWTIE_MAP_CONTAMINANTS in filepath ./git_repos/smrnaseq\\modules\\local\\bowtie_map_contaminants.nf\n",
      "Process BOWTIE_MAP_SEQ in filepath ./git_repos/smrnaseq\\modules\\local\\bowtie_map_mirna.nf\n",
      "Process INDEX_MIRNA in filepath ./git_repos/smrnaseq\\modules\\local\\bowtie_mirna.nf\n",
      "Process TABLE_MERGE in filepath ./git_repos/smrnaseq\\modules\\local\\datatable_merge.nf\n",
      "Process EDGER_QC in filepath ./git_repos/smrnaseq\\modules\\local\\edger_qc.nf\n",
      "Process FILTER_STATS in filepath ./git_repos/smrnaseq\\modules\\local\\filter_stats.nf\n",
      "Process FORMAT_FASTA_MIRNA in filepath ./git_repos/smrnaseq\\modules\\local\\format_fasta_mirna.nf\n",
      "Process MIRDEEP2_MAPPER in filepath ./git_repos/smrnaseq\\modules\\local\\mirdeep2_mapper.nf\n",
      "Process MIRDEEP2_PIGZ in filepath ./git_repos/smrnaseq\\modules\\local\\mirdeep2_prepare.nf\n",
      "Process MIRDEEP2_RUN in filepath ./git_repos/smrnaseq\\modules\\local\\mirdeep2_run.nf\n",
      "Process MIRTOP_QUANT in filepath ./git_repos/smrnaseq\\modules\\local\\mirtop_quant.nf\n",
      "Process MIRTRACE_RUN in filepath ./git_repos/smrnaseq\\modules\\local\\mirtrace.nf\n",
      "Process PARSE_FASTA_MIRNA in filepath ./git_repos/smrnaseq\\modules\\local\\parse_fasta_mirna.nf\n",
      "Process SAMPLESHEET_CHECK in filepath ./git_repos/smrnaseq\\modules\\local\\samplesheet_check.nf\n",
      "Process SEQCLUSTER_SEQUENCES in filepath ./git_repos/smrnaseq\\modules\\local\\seqcluster_collapse.nf\n",
      "Process CAT_FASTQ in filepath ./git_repos/smrnaseq\\modules\\nf-core\\cat\\fastq\\main.nf\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Process CUSTOM_DUMPSOFTWAREVERSIONS in filepath ./git_repos/smrnaseq\\modules\\nf-core\\custom\\dumpsoftwareversions\\main.nf\n",
      "Process FASTP in filepath ./git_repos/smrnaseq\\modules\\nf-core\\fastp\\main.nf\n",
      "Process FASTQC in filepath ./git_repos/smrnaseq\\modules\\nf-core\\fastqc\\main.nf\n",
      "Process MULTIQC in filepath ./git_repos/smrnaseq\\modules\\nf-core\\multiqc\\main.nf\n",
      "Process SAMTOOLS_FLAGSTAT in filepath ./git_repos/smrnaseq\\modules\\nf-core\\samtools\\flagstat\\main.nf\n",
      "Process SAMTOOLS_IDXSTATS in filepath ./git_repos/smrnaseq\\modules\\nf-core\\samtools\\idxstats\\main.nf\n",
      "Process SAMTOOLS_INDEX in filepath ./git_repos/smrnaseq\\modules\\nf-core\\samtools\\index\\main.nf\n",
      "Process SAMTOOLS_SORT in filepath ./git_repos/smrnaseq\\modules\\nf-core\\samtools\\sort\\main.nf\n",
      "Process SAMTOOLS_STATS in filepath ./git_repos/smrnaseq\\modules\\nf-core\\samtools\\stats\\main.nf\n",
      "Process ST_CLUSTERING in filepath ./git_repos/spatialtranscriptomics\\modules\\local\\st_clustering.nf\n",
      "Process ST_QUALITY_CONTROLS in filepath ./git_repos/spatialtranscriptomics\\modules\\local\\st_quality_controls.nf\n",
      "Process ST_READ_DATA in filepath ./git_repos/spatialtranscriptomics\\modules\\local\\st_read_data.nf\n",
      "Process ST_SPATIAL_DE in filepath ./git_repos/spatialtranscriptomics\\modules\\local\\st_spatial_de.nf\n",
      "Process CUSTOM_DUMPSOFTWAREVERSIONS in filepath ./git_repos/spatialtranscriptomics\\modules\\nf-core\\custom\\dumpsoftwareversions\\main.nf\n",
      "Process FASTQC in filepath ./git_repos/spatialtranscriptomics\\modules\\nf-core\\fastqc\\main.nf\n",
      "Process MULTIQC in filepath ./git_repos/spatialtranscriptomics\\modules\\nf-core\\multiqc\\main.nf\n",
      "Process SPACERANGER_COUNT in filepath ./git_repos/spatialtranscriptomics\\modules\\nf-core\\spaceranger\\count\\main.nf\n",
      "Process UNTAR in filepath ./git_repos/spatialtranscriptomics\\modules\\nf-core\\untar\\main.nf\n",
      "Process JOIN_FILES in filepath ./git_repos/spinningjenny\\modules\\local\\joinfiles.nf\n",
      "Process RUN_MODEL in filepath ./git_repos/spinningjenny\\modules\\local\\model.nf\n",
      "Process MAKE_PLOT in filepath ./git_repos/spinningjenny\\modules\\local\\rplot.nf\n",
      "Process SAMPLESHEET_CHECK in filepath ./git_repos/spinningjenny\\modules\\local\\samplesheet_check.nf\n",
      "Process XML_MOD in filepath ./git_repos/spinningjenny\\modules\\local\\xmlmod.nf\n",
      "Process CUSTOM_DUMPSOFTWAREVERSIONS in filepath ./git_repos/spinningjenny\\modules\\nf-core\\custom\\dumpsoftwareversions\\main.nf\n",
      "Process FASTQC in filepath ./git_repos/spinningjenny\\modules\\nf-core\\fastqc\\main.nf\n",
      "Process MULTIQC in filepath ./git_repos/spinningjenny\\modules\\nf-core\\multiqc\\main.nf\n",
      "Process SAMPLESHEET_CHECK in filepath ./git_repos/ssds\\modules\\local\\samplesheet_check.nf\n",
      "Process CALCULATESPOT in filepath ./git_repos/ssds\\modules\\local\\modules\\calculatespot\\main.nf\n",
      "Process GENERATE_SSDS_COVERAGE in filepath ./git_repos/ssds\\modules\\local\\modules\\generate_ssds_coverage\\main.nf\n",
      "Process MULTIQC in filepath ./git_repos/ssds\\modules\\local\\modules\\multiqc_ssds\\main.nf\n",
      "Process PARSESSDS in filepath ./git_repos/ssds\\modules\\local\\modules\\parse_ssds_bam\\main.nf\n",
      "Process BEDTOOLS_BAMTOBED in filepath ./git_repos/ssds\\modules\\nf-core\\modules\\bedtools\\bamtobed\\main.nf\n",
      "Process BEDTOOLS_MAKEWINDOWS in filepath ./git_repos/ssds\\modules\\nf-core\\modules\\bedtools\\makewindows\\main.nf\n",
      "Process BEDTOOLS_SORT in filepath ./git_repos/ssds\\modules\\nf-core\\modules\\bedtools\\sort\\main.nf\n",
      "Process BWA_INDEX in filepath ./git_repos/ssds\\modules\\nf-core\\modules\\bwa\\index\\main.nf\n",
      "Process BWA_MEM in filepath ./git_repos/ssds\\modules\\nf-core\\modules\\bwa\\mem\\main.nf\n",
      "Process CUSTOM_DUMPSOFTWAREVERSIONS in filepath ./git_repos/ssds\\modules\\nf-core\\modules\\custom\\dumpsoftwareversions\\main.nf\n",
      "Process DEEPTOOLS_PLOTFINGERPRINT in filepath ./git_repos/ssds\\modules\\nf-core\\modules\\deeptools\\plotfingerprint\\main.nf\n",
      "Process FASTQC in filepath ./git_repos/ssds\\modules\\nf-core\\modules\\fastqc\\main.nf\n",
      "Process MULTIQC in filepath ./git_repos/ssds\\modules\\nf-core\\modules\\multiqc\\main.nf\n",
      "Process PICARD_COLLECTMULTIPLEMETRICS in filepath ./git_repos/ssds\\modules\\nf-core\\modules\\picard\\collectmultiplemetrics\\main.nf\n",
      "Process PICARD_MARKDUPLICATES in filepath ./git_repos/ssds\\modules\\nf-core\\modules\\picard\\markduplicates\\main.nf\n",
      "Process PICARD_SORTSAM in filepath ./git_repos/ssds\\modules\\nf-core\\modules\\picard\\sortsam\\main.nf\n",
      "Process SAMTOOLS_IDXSTATS in filepath ./git_repos/ssds\\modules\\nf-core\\modules\\samtools\\idxstats\\main.nf\n",
      "Process SAMTOOLS_INDEX in filepath ./git_repos/ssds\\modules\\nf-core\\modules\\samtools\\index\\main.nf\n",
      "Process SAMTOOLS_VIEW in filepath ./git_repos/ssds\\modules\\nf-core\\modules\\samtools\\view\\main.nf\n",
      "Process TRIMGALORE in filepath ./git_repos/ssds\\modules\\nf-core\\modules\\trimgalore\\main.nf\n",
      "Process UCSC_BEDGRAPHTOBIGWIG in filepath ./git_repos/ssds\\modules\\nf-core\\modules\\ucsc\\bedgraphtobigwig\\main.nf\n",
      "Process KRAKEN2_STANDARD_REPORT in filepath ./git_repos/taxprofiler\\modules\\local\\kraken2_standard_report.nf\n",
      "Process KRONA_CLEANUP in filepath ./git_repos/taxprofiler\\modules\\local\\krona_cleanup.nf\n",
      "Process SAMPLESHEET_CHECK in filepath ./git_repos/taxprofiler\\modules\\local\\samplesheet_check.nf\n",
      "Process ADAPTERREMOVAL in filepath ./git_repos/taxprofiler\\modules\\nf-core\\adapterremoval\\main.nf\n",
      "Process BBMAP_BBDUK in filepath ./git_repos/taxprofiler\\modules\\nf-core\\bbmap\\bbduk\\main.nf\n",
      "Process BOWTIE2_ALIGN in filepath ./git_repos/taxprofiler\\modules\\nf-core\\bowtie2\\align\\main.nf\n",
      "Process BOWTIE2_BUILD in filepath ./git_repos/taxprofiler\\modules\\nf-core\\bowtie2\\build\\main.nf\n",
      "Process BRACKEN_BRACKEN in filepath ./git_repos/taxprofiler\\modules\\nf-core\\bracken\\bracken\\main.nf\n",
      "Process BRACKEN_COMBINEBRACKENOUTPUTS in filepath ./git_repos/taxprofiler\\modules\\nf-core\\bracken\\combinebrackenoutputs\\main.nf\n",
      "Process CAT_FASTQ in filepath ./git_repos/taxprofiler\\modules\\nf-core\\cat\\fastq\\main.nf\n",
      "Process CENTRIFUGE_CENTRIFUGE in filepath ./git_repos/taxprofiler\\modules\\nf-core\\centrifuge\\centrifuge\\main.nf\n",
      "Process CENTRIFUGE_KREPORT in filepath ./git_repos/taxprofiler\\modules\\nf-core\\centrifuge\\kreport\\main.nf\n",
      "Process CUSTOM_DUMPSOFTWAREVERSIONS in filepath ./git_repos/taxprofiler\\modules\\nf-core\\custom\\dumpsoftwareversions\\main.nf\n",
      "Process DIAMOND_BLASTX in filepath ./git_repos/taxprofiler\\modules\\nf-core\\diamond\\blastx\\main.nf\n",
      "Process FALCO in filepath ./git_repos/taxprofiler\\modules\\nf-core\\falco\\main.nf\n",
      "Process FASTP in filepath ./git_repos/taxprofiler\\modules\\nf-core\\fastp\\main.nf\n",
      "Process FASTQC in filepath ./git_repos/taxprofiler\\modules\\nf-core\\fastqc\\main.nf\n",
      "Process FILTLONG in filepath ./git_repos/taxprofiler\\modules\\nf-core\\filtlong\\main.nf\n",
      "Process GANON_CLASSIFY in filepath ./git_repos/taxprofiler\\modules\\nf-core\\ganon\\classify\\main.nf\n",
      "Process GANON_REPORT in filepath ./git_repos/taxprofiler\\modules\\nf-core\\ganon\\report\\main.nf\n",
      "Process GANON_TABLE in filepath ./git_repos/taxprofiler\\modules\\nf-core\\ganon\\table\\main.nf\n",
      "Process GUNZIP in filepath ./git_repos/taxprofiler\\modules\\nf-core\\gunzip\\main.nf\n",
      "Process KAIJU_KAIJU in filepath ./git_repos/taxprofiler\\modules\\nf-core\\kaiju\\kaiju\\main.nf\n",
      "Process KAIJU_KAIJU2KRONA in filepath ./git_repos/taxprofiler\\modules\\nf-core\\kaiju\\kaiju2krona\\main.nf\n",
      "Process KAIJU_KAIJU2TABLE in filepath ./git_repos/taxprofiler\\modules\\nf-core\\kaiju\\kaiju2table\\main.nf\n",
      "Process KMCP_PROFILE in filepath ./git_repos/taxprofiler\\modules\\nf-core\\kmcp\\profile\\main.nf\n",
      "Process KMCP_SEARCH in filepath ./git_repos/taxprofiler\\modules\\nf-core\\kmcp\\search\\main.nf\n",
      "Process KRAKEN2_KRAKEN2 in filepath ./git_repos/taxprofiler\\modules\\nf-core\\kraken2\\kraken2\\main.nf\n",
      "Process KRAKENTOOLS_COMBINEKREPORTS in filepath ./git_repos/taxprofiler\\modules\\nf-core\\krakentools\\combinekreports\\main.nf\n",
      "Process KRAKENTOOLS_KREPORT2KRONA in filepath ./git_repos/taxprofiler\\modules\\nf-core\\krakentools\\kreport2krona\\main.nf\n",
      "Process KRAKENUNIQ_PRELOADEDKRAKENUNIQ in filepath ./git_repos/taxprofiler\\modules\\nf-core\\krakenuniq\\preloadedkrakenuniq\\main.nf\n",
      "Process KRONA_KTIMPORTTAXONOMY in filepath ./git_repos/taxprofiler\\modules\\nf-core\\krona\\ktimporttaxonomy\\main.nf\n",
      "Process KRONA_KTIMPORTTEXT in filepath ./git_repos/taxprofiler\\modules\\nf-core\\krona\\ktimporttext\\main.nf\n",
      "Process MALT_RUN in filepath ./git_repos/taxprofiler\\modules\\nf-core\\malt\\run\\main.nf\n",
      "Process MEGAN_RMA2INFO in filepath ./git_repos/taxprofiler\\modules\\nf-core\\megan\\rma2info\\main.nf\n",
      "Process METAPHLAN_MERGEMETAPHLANTABLES in filepath ./git_repos/taxprofiler\\modules\\nf-core\\metaphlan\\mergemetaphlantables\\main.nf\n",
      "Process METAPHLAN_METAPHLAN in filepath ./git_repos/taxprofiler\\modules\\nf-core\\metaphlan\\metaphlan\\main.nf\n",
      "Process MINIMAP2_ALIGN in filepath ./git_repos/taxprofiler\\modules\\nf-core\\minimap2\\align\\main.nf\n",
      "Process MINIMAP2_INDEX in filepath ./git_repos/taxprofiler\\modules\\nf-core\\minimap2\\index\\main.nf\n",
      "Process MOTUS_MERGE in filepath ./git_repos/taxprofiler\\modules\\nf-core\\motus\\merge\\main.nf\n",
      "Process MOTUS_PROFILE in filepath ./git_repos/taxprofiler\\modules\\nf-core\\motus\\profile\\main.nf\n",
      "Process MULTIQC in filepath ./git_repos/taxprofiler\\modules\\nf-core\\multiqc\\main.nf\n",
      "Process PORECHOP_PORECHOP in filepath ./git_repos/taxprofiler\\modules\\nf-core\\porechop\\porechop\\main.nf\n",
      "Process PRINSEQPLUSPLUS in filepath ./git_repos/taxprofiler\\modules\\nf-core\\prinseqplusplus\\main.nf\n",
      "Process SAMTOOLS_FASTQ in filepath ./git_repos/taxprofiler\\modules\\nf-core\\samtools\\fastq\\main.nf\n",
      "Process SAMTOOLS_INDEX in filepath ./git_repos/taxprofiler\\modules\\nf-core\\samtools\\index\\main.nf\n",
      "Process SAMTOOLS_STATS in filepath ./git_repos/taxprofiler\\modules\\nf-core\\samtools\\stats\\main.nf\n",
      "Process SAMTOOLS_VIEW in filepath ./git_repos/taxprofiler\\modules\\nf-core\\samtools\\view\\main.nf\n",
      "Process TAXPASTA_MERGE in filepath ./git_repos/taxprofiler\\modules\\nf-core\\taxpasta\\merge\\main.nf\n",
      "Process TAXPASTA_STANDARDISE in filepath ./git_repos/taxprofiler\\modules\\nf-core\\taxpasta\\standardise\\main.nf\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Process UNTAR in filepath ./git_repos/taxprofiler\\modules\\nf-core\\untar\\main.nf\n",
      "Process Hail_sample_QC in filepath ./git_repos/variantcatalogue\\modules\\local\\Hail_sample_QC.nf\n",
      "Process Hail_variant_MT_QC in filepath ./git_repos/variantcatalogue\\modules\\local\\Hail_variant_MT_QC.nf\n",
      "Process Hail_variant_QC in filepath ./git_repos/variantcatalogue\\modules\\local\\Hail_variant_QC.nf\n",
      "Process MT_Liftover in filepath ./git_repos/variantcatalogue\\modules\\local\\MT_Liftover.nf\n",
      "Process MT_Step1_input_tsv in filepath ./git_repos/variantcatalogue\\modules\\local\\MT_Step1_input_tsv.nf\n",
      "Process MT_Step2_participant_data in filepath ./git_repos/variantcatalogue\\modules\\local\\MT_Step2_participant_data.nf\n",
      "Process MT_Step3_metadata in filepath ./git_repos/variantcatalogue\\modules\\local\\MT_Step3_metadata.nf\n",
      "Process MT_Step3_metadata_sample in filepath ./git_repos/variantcatalogue\\modules\\local\\MT_Step3_metadata_sample.nf\n",
      "Process Picard_CollectAlignmentSummaryMetrics in filepath ./git_repos/variantcatalogue\\modules\\local\\Picard_CollectAlignmentSummaryMetrics.nf\n",
      "Process Picard_QualityScoreDistribution in filepath ./git_repos/variantcatalogue\\modules\\local\\Picard_QualityScoreDistribution.nf\n",
      "Process SAMPLESHEET_CHECK in filepath ./git_repos/variantcatalogue\\modules\\local\\samplesheet_check.nf\n",
      "Process shift_back in filepath ./git_repos/variantcatalogue\\modules\\local\\shift_back.nf\n",
      "Process BCFTOOLS_NORM in filepath ./git_repos/variantcatalogue\\modules\\local\\bcftools\\norm\\main.nf\n",
      "Process BWA_MEM in filepath ./git_repos/variantcatalogue\\modules\\local\\bwa\\mem\\main.nf\n",
      "Process PICARD_COLLECTHSMETRICS in filepath ./git_repos/variantcatalogue\\modules\\local\\collecthsmetrics\\main.nf\n",
      "Process DEEPVARIANT in filepath ./git_repos/variantcatalogue\\modules\\local\\deepvariant\\main.nf\n",
      "Process GATK4_FILTERMUTECTCALLS in filepath ./git_repos/variantcatalogue\\modules\\local\\filtermutectcalls\\main.nf\n",
      "Process GATK4_LEFTALIGNANDTRIMVARIANTS in filepath ./git_repos/variantcatalogue\\modules\\local\\gatk4\\leftalignandtrimvariants\\main.nf\n",
      "Process GATK4_MERGEMUTECTSTATS in filepath ./git_repos/variantcatalogue\\modules\\local\\gatk4\\mergemutectstats\\main.nf\n",
      "Process GATK4_PRINTREADS in filepath ./git_repos/variantcatalogue\\modules\\local\\gatk4\\printreads\\main.nf\n",
      "Process GATK4_SAMTOFASTQ in filepath ./git_repos/variantcatalogue\\modules\\local\\gatk4\\samtofastq\\main.nf\n",
      "Process HAPLOCHECK in filepath ./git_repos/variantcatalogue\\modules\\local\\haplocheck\\main.nf\n",
      "Process GATK4_VARIANTFILTRATION in filepath ./git_repos/variantcatalogue\\modules\\local\\variantfiltration\\main.nf\n",
      "Process BCFTOOLS_ANNOTATE in filepath ./git_repos/variantcatalogue\\modules\\nf-core\\bcftools\\annotate\\main.nf\n",
      "Process BCFTOOLS_NORM in filepath ./git_repos/variantcatalogue\\modules\\nf-core\\bcftools\\norm\\main.nf\n",
      "Process BWA_INDEX in filepath ./git_repos/variantcatalogue\\modules\\nf-core\\bwa\\index\\main.nf\n",
      "Process BWA_MEM in filepath ./git_repos/variantcatalogue\\modules\\nf-core\\bwa\\mem\\main.nf\n",
      "Process CUSTOM_DUMPSOFTWAREVERSIONS in filepath ./git_repos/variantcatalogue\\modules\\nf-core\\custom\\dumpsoftwareversions\\main.nf\n",
      "Process DEEPVARIANT in filepath ./git_repos/variantcatalogue\\modules\\nf-core\\deepvariant\\main.nf\n",
      "Process ENSEMBLVEP_VEP in filepath ./git_repos/variantcatalogue\\modules\\nf-core\\ensemblvep\\vep\\main.nf\n",
      "Process FASTQC in filepath ./git_repos/variantcatalogue\\modules\\nf-core\\fastqc\\main.nf\n",
      "Process GATK4_FILTERMUTECTCALLS in filepath ./git_repos/variantcatalogue\\modules\\nf-core\\gatk4\\filtermutectcalls\\main.nf\n",
      "Process GATK4_LEFTALIGNANDTRIMVARIANTS in filepath ./git_repos/variantcatalogue\\modules\\nf-core\\gatk4\\leftalignandtrimvariants\\main.nf\n",
      "Process GATK4_MARKDUPLICATES in filepath ./git_repos/variantcatalogue\\modules\\nf-core\\gatk4\\markduplicates\\main.nf\n",
      "Process GATK4_MERGEMUTECTSTATS in filepath ./git_repos/variantcatalogue\\modules\\nf-core\\gatk4\\mergemutectstats\\main.nf\n",
      "Process GATK4_MERGEVCFS in filepath ./git_repos/variantcatalogue\\modules\\nf-core\\gatk4\\mergevcfs\\main.nf\n",
      "Process GATK4_MUTECT2 in filepath ./git_repos/variantcatalogue\\modules\\nf-core\\gatk4\\mutect2\\main.nf\n",
      "Process GATK4_PRINTREADS in filepath ./git_repos/variantcatalogue\\modules\\nf-core\\gatk4\\printreads\\main.nf\n",
      "Process GATK4_SAMTOFASTQ in filepath ./git_repos/variantcatalogue\\modules\\nf-core\\gatk4\\samtofastq\\main.nf\n",
      "Process GATK4_VARIANTFILTRATION in filepath ./git_repos/variantcatalogue\\modules\\nf-core\\gatk4\\variantfiltration\\main.nf\n",
      "Process GLNEXUS in filepath ./git_repos/variantcatalogue\\modules\\nf-core\\glnexus\\main.nf\n",
      "Process HAPLOCHECK in filepath ./git_repos/variantcatalogue\\modules\\nf-core\\haplocheck\\main.nf\n",
      "Process MOSDEPTH in filepath ./git_repos/variantcatalogue\\modules\\nf-core\\mosdepth\\main.nf\n",
      "Process MULTIQC in filepath ./git_repos/variantcatalogue\\modules\\nf-core\\multiqc\\main.nf\n",
      "Process PICARD_COLLECTHSMETRICS in filepath ./git_repos/variantcatalogue\\modules\\nf-core\\picard\\collecthsmetrics\\main.nf\n",
      "Process PICARD_COLLECTWGSMETRICS in filepath ./git_repos/variantcatalogue\\modules\\nf-core\\picard\\collectwgsmetrics\\main.nf\n",
      "Process SAMTOOLS_INDEX in filepath ./git_repos/variantcatalogue\\modules\\nf-core\\samtools\\index\\main.nf\n",
      "Process TABIX_TABIX in filepath ./git_repos/variantcatalogue\\modules\\nf-core\\tabix\\tabix\\main.nf\n",
      "Process TRIMMOMATIC in filepath ./git_repos/variantcatalogue\\modules\\nf-core\\trimmomatic\\main.nf\n",
      "Process trim_and_combine in filepath ./git_repos/vipr\\main.nf\n",
      "Process decont in filepath ./git_repos/vipr\\main.nf\n",
      "Process kraken in filepath ./git_repos/vipr\\main.nf\n",
      "Process tadpole in filepath ./git_repos/vipr\\main.nf\n",
      "Process gap_fill_assembly in filepath ./git_repos/vipr\\main.nf\n",
      "Process polish_assembly in filepath ./git_repos/vipr\\main.nf\n",
      "Process final_mapping in filepath ./git_repos/vipr\\main.nf\n",
      "Process var_calling in filepath ./git_repos/vipr\\main.nf\n",
      "Process genomecov in filepath ./git_repos/vipr\\main.nf\n",
      "Process vipr_tools in filepath ./git_repos/vipr\\main.nf\n",
      "Process ABRIDGED_TSV in filepath ./git_repos/viralintegration\\modules\\local\\abridged_tsv.nf\n",
      "Process CAT_FASTA in filepath ./git_repos/viralintegration\\modules\\local\\cat_fasta.nf\n",
      "Process CHIMERIC_CONTIG_EVIDENCE_ANALYZER in filepath ./git_repos/viralintegration\\modules\\local\\chimeric_contig_evidence_analyzer.nf\n",
      "Process EXTRACT_CHIMERIC_GENOMIC_TARGETS in filepath ./git_repos/viralintegration\\modules\\local\\extract_chimeric_genomic_targets.nf\n",
      "Process INSERTION_SITE_CANDIDATES in filepath ./git_repos/viralintegration\\modules\\local\\insertion_site_candidates.nf\n",
      "Process POLYA_STRIPPER in filepath ./git_repos/viralintegration\\modules\\local\\polyA_stripper.nf\n",
      "Process REMOVE_DUPLICATES in filepath ./git_repos/viralintegration\\modules\\local\\remove_duplicates.nf\n",
      "Process SAMPLESHEET_CHECK in filepath ./git_repos/viralintegration\\modules\\local\\samplesheet_check.nf\n",
      "Process STAR_ALIGN_VALIDATE in filepath ./git_repos/viralintegration\\modules\\local\\star_align_validate.nf\n",
      "Process SUMMARY_REPORT in filepath ./git_repos/viralintegration\\modules\\local\\summary_report.nf\n",
      "Process VIRUS_REPORT in filepath ./git_repos/viralintegration\\modules\\local\\virus_report.nf\n",
      "Process CUSTOM_DUMPSOFTWAREVERSIONS in filepath ./git_repos/viralintegration\\modules\\nf-core\\custom\\dumpsoftwareversions\\main.nf\n",
      "Process FASTQC in filepath ./git_repos/viralintegration\\modules\\nf-core\\fastqc\\main.nf\n",
      "Process MULTIQC in filepath ./git_repos/viralintegration\\modules\\nf-core\\multiqc\\main.nf\n",
      "Process SAMTOOLS_INDEX in filepath ./git_repos/viralintegration\\modules\\nf-core\\samtools\\index\\main.nf\n",
      "Process SAMTOOLS_SORT in filepath ./git_repos/viralintegration\\modules\\nf-core\\samtools\\sort\\main.nf\n",
      "Process STAR_ALIGN in filepath ./git_repos/viralintegration\\modules\\nf-core\\star\\align\\main.nf\n",
      "Process STAR_GENOMEGENERATE in filepath ./git_repos/viralintegration\\modules\\nf-core\\star\\genomegenerate\\main.nf\n",
      "Process TRIMMOMATIC in filepath ./git_repos/viralintegration\\modules\\nf-core\\trimmomatic\\main.nf\n",
      "Process ASCIIGENOME in filepath ./git_repos/viralrecon\\modules\\local\\asciigenome.nf\n",
      "Process COLLAPSE_PRIMERS in filepath ./git_repos/viralrecon\\modules\\local\\collapse_primers.nf\n",
      "Process CUTADAPT in filepath ./git_repos/viralrecon\\modules\\local\\cutadapt.nf\n",
      "Process FILTER_BLASTN in filepath ./git_repos/viralrecon\\modules\\local\\filter_blastn.nf\n",
      "Process IVAR_VARIANTS_TO_VCF in filepath ./git_repos/viralrecon\\modules\\local\\ivar_variants_to_vcf.nf\n",
      "Process KRAKEN2_BUILD in filepath ./git_repos/viralrecon\\modules\\local\\kraken2_build.nf\n",
      "Process MAKE_BED_MASK in filepath ./git_repos/viralrecon\\modules\\local\\make_bed_mask.nf\n",
      "Process MAKE_VARIANTS_LONG_TABLE in filepath ./git_repos/viralrecon\\modules\\local\\make_variants_long_table.nf\n",
      "Process MULTIQC in filepath ./git_repos/viralrecon\\modules\\local\\multiqc_illumina.nf\n",
      "Process MULTIQC in filepath ./git_repos/viralrecon\\modules\\local\\multiqc_nanopore.nf\n",
      "Process PLOT_BASE_DENSITY in filepath ./git_repos/viralrecon\\modules\\local\\plot_base_density.nf\n",
      "Process PLOT_MOSDEPTH_REGIONS in filepath ./git_repos/viralrecon\\modules\\local\\plot_mosdepth_regions.nf\n",
      "Process RENAME_FASTA_HEADER in filepath ./git_repos/viralrecon\\modules\\local\\rename_fasta_header.nf\n",
      "Process SAMPLESHEET_CHECK in filepath ./git_repos/viralrecon\\modules\\local\\samplesheet_check.nf\n",
      "Process SNPEFF_ANN in filepath ./git_repos/viralrecon\\modules\\local\\snpeff_ann.nf\n",
      "Process SNPEFF_BUILD in filepath ./git_repos/viralrecon\\modules\\local\\snpeff_build.nf\n",
      "Process SNPSIFT_EXTRACTFIELDS in filepath ./git_repos/viralrecon\\modules\\local\\snpsift_extractfields.nf\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Process ABACAS in filepath ./git_repos/viralrecon\\modules\\nf-core\\abacas\\main.nf\n",
      "Process ARTIC_GUPPYPLEX in filepath ./git_repos/viralrecon\\modules\\nf-core\\artic\\guppyplex\\main.nf\n",
      "Process ARTIC_MINION in filepath ./git_repos/viralrecon\\modules\\nf-core\\artic\\minion\\main.nf\n",
      "Process BANDAGE_IMAGE in filepath ./git_repos/viralrecon\\modules\\nf-core\\bandage\\image\\main.nf\n",
      "Process BCFTOOLS_CONSENSUS in filepath ./git_repos/viralrecon\\modules\\nf-core\\bcftools\\consensus\\main.nf\n",
      "Process BCFTOOLS_FILTER in filepath ./git_repos/viralrecon\\modules\\nf-core\\bcftools\\filter\\main.nf\n",
      "Process BCFTOOLS_MPILEUP in filepath ./git_repos/viralrecon\\modules\\nf-core\\bcftools\\mpileup\\main.nf\n",
      "Process BCFTOOLS_NORM in filepath ./git_repos/viralrecon\\modules\\nf-core\\bcftools\\norm\\main.nf\n",
      "Process BCFTOOLS_QUERY in filepath ./git_repos/viralrecon\\modules\\nf-core\\bcftools\\query\\main.nf\n",
      "Process BCFTOOLS_SORT in filepath ./git_repos/viralrecon\\modules\\nf-core\\bcftools\\sort\\main.nf\n",
      "Process BCFTOOLS_STATS in filepath ./git_repos/viralrecon\\modules\\nf-core\\bcftools\\stats\\main.nf\n",
      "Process BEDTOOLS_GETFASTA in filepath ./git_repos/viralrecon\\modules\\nf-core\\bedtools\\getfasta\\main.nf\n",
      "Process BEDTOOLS_MASKFASTA in filepath ./git_repos/viralrecon\\modules\\nf-core\\bedtools\\maskfasta\\main.nf\n",
      "Process BEDTOOLS_MERGE in filepath ./git_repos/viralrecon\\modules\\nf-core\\bedtools\\merge\\main.nf\n",
      "Process BLAST_BLASTN in filepath ./git_repos/viralrecon\\modules\\nf-core\\blast\\blastn\\main.nf\n",
      "Process BLAST_MAKEBLASTDB in filepath ./git_repos/viralrecon\\modules\\nf-core\\blast\\makeblastdb\\main.nf\n",
      "Process BOWTIE2_ALIGN in filepath ./git_repos/viralrecon\\modules\\nf-core\\bowtie2\\align\\main.nf\n",
      "Process BOWTIE2_BUILD in filepath ./git_repos/viralrecon\\modules\\nf-core\\bowtie2\\build\\main.nf\n",
      "Process CAT_FASTQ in filepath ./git_repos/viralrecon\\modules\\nf-core\\cat\\fastq\\main.nf\n",
      "Process CUSTOM_DUMPSOFTWAREVERSIONS in filepath ./git_repos/viralrecon\\modules\\nf-core\\custom\\dumpsoftwareversions\\main.nf\n",
      "Process CUSTOM_GETCHROMSIZES in filepath ./git_repos/viralrecon\\modules\\nf-core\\custom\\getchromsizes\\main.nf\n",
      "Process FASTP in filepath ./git_repos/viralrecon\\modules\\nf-core\\fastp\\main.nf\n",
      "Process FASTQC in filepath ./git_repos/viralrecon\\modules\\nf-core\\fastqc\\main.nf\n",
      "Process GUNZIP in filepath ./git_repos/viralrecon\\modules\\nf-core\\gunzip\\main.nf\n",
      "Process IVAR_CONSENSUS in filepath ./git_repos/viralrecon\\modules\\nf-core\\ivar\\consensus\\main.nf\n",
      "Process IVAR_TRIM in filepath ./git_repos/viralrecon\\modules\\nf-core\\ivar\\trim\\main.nf\n",
      "Process IVAR_VARIANTS in filepath ./git_repos/viralrecon\\modules\\nf-core\\ivar\\variants\\main.nf\n",
      "Process KRAKEN2_KRAKEN2 in filepath ./git_repos/viralrecon\\modules\\nf-core\\kraken2\\kraken2\\main.nf\n",
      "Process MINIA in filepath ./git_repos/viralrecon\\modules\\nf-core\\minia\\main.nf\n",
      "Process MOSDEPTH in filepath ./git_repos/viralrecon\\modules\\nf-core\\mosdepth\\main.nf\n",
      "Process NANOPLOT in filepath ./git_repos/viralrecon\\modules\\nf-core\\nanoplot\\main.nf\n",
      "Process NEXTCLADE_DATASETGET in filepath ./git_repos/viralrecon\\modules\\nf-core\\nextclade\\datasetget\\main.nf\n",
      "Process NEXTCLADE_RUN in filepath ./git_repos/viralrecon\\modules\\nf-core\\nextclade\\run\\main.nf\n",
      "Process PANGOLIN in filepath ./git_repos/viralrecon\\modules\\nf-core\\pangolin\\main.nf\n",
      "Process PICARD_COLLECTMULTIPLEMETRICS in filepath ./git_repos/viralrecon\\modules\\nf-core\\picard\\collectmultiplemetrics\\main.nf\n",
      "Process PICARD_MARKDUPLICATES in filepath ./git_repos/viralrecon\\modules\\nf-core\\picard\\markduplicates\\main.nf\n",
      "Process PLASMIDID in filepath ./git_repos/viralrecon\\modules\\nf-core\\plasmidid\\main.nf\n",
      "Process PYCOQC in filepath ./git_repos/viralrecon\\modules\\nf-core\\pycoqc\\main.nf\n",
      "Process QUAST in filepath ./git_repos/viralrecon\\modules\\nf-core\\quast\\main.nf\n",
      "Process SAMTOOLS_FLAGSTAT in filepath ./git_repos/viralrecon\\modules\\nf-core\\samtools\\flagstat\\main.nf\n",
      "Process SAMTOOLS_IDXSTATS in filepath ./git_repos/viralrecon\\modules\\nf-core\\samtools\\idxstats\\main.nf\n",
      "Process SAMTOOLS_INDEX in filepath ./git_repos/viralrecon\\modules\\nf-core\\samtools\\index\\main.nf\n",
      "Process SAMTOOLS_SORT in filepath ./git_repos/viralrecon\\modules\\nf-core\\samtools\\sort\\main.nf\n",
      "Process SAMTOOLS_STATS in filepath ./git_repos/viralrecon\\modules\\nf-core\\samtools\\stats\\main.nf\n",
      "Process SAMTOOLS_VIEW in filepath ./git_repos/viralrecon\\modules\\nf-core\\samtools\\view\\main.nf\n",
      "Process SPADES in filepath ./git_repos/viralrecon\\modules\\nf-core\\spades\\main.nf\n",
      "Process TABIX_BGZIP in filepath ./git_repos/viralrecon\\modules\\nf-core\\tabix\\bgzip\\main.nf\n",
      "Process TABIX_TABIX in filepath ./git_repos/viralrecon\\modules\\nf-core\\tabix\\tabix\\main.nf\n",
      "Process UNICYCLER in filepath ./git_repos/viralrecon\\modules\\nf-core\\unicycler\\main.nf\n",
      "Process UNTAR in filepath ./git_repos/viralrecon\\modules\\nf-core\\untar\\main.nf\n",
      "Process VCFLIB_VCFUNIQ in filepath ./git_repos/viralrecon\\modules\\nf-core\\vcflib\\vcfuniq\\main.nf\n"
     ]
    }
   ],
   "source": [
    "directory = './git_repos/'\n",
    "result_dir = 'reviewed_results'\n",
    "find_if_conditions(directory, result_dir)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1459"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "df = pd.read_csv('./reviewed_results/if_conditions_processes.csv')\n",
    "df['File Path'].nunique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Inside/Outside Process\n",
       "OUTSIDE    4902\n",
       "INSIDE     2125\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df['Inside/Outside Process'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Pipeline</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Inside/Outside Process</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>INSIDE</th>\n",
       "      <td>89</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OUTSIDE</th>\n",
       "      <td>94</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                        Pipeline\n",
       "Inside/Outside Process          \n",
       "INSIDE                        89\n",
       "OUTSIDE                       94"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.groupby('Inside/Outside Process').agg({'Pipeline' : 'nunique'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
