{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import networkx as nx\n",
    "import matplotlib.pyplot as plt\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import csv\n",
    "\n",
    "directory_path = './dags'\n",
    "directory_plots = './plots'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Count all the nodes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_all_nodes(dot_file):\n",
    "    G = nx.nx_agraph.read_dot(dot_file)\n",
    "    all_nodes = G.nodes()\n",
    "    \n",
    "    return all_nodes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Count all processes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We count all the process node here, short explanation:\n",
    "exploration shows, that the processes have no shape in the dot-file, we count the amount of processes in a workflow, in the dot-file they have. Examples:\n",
    "\n",
    "process node: v271 [label=\"NFCORE_AMPLISEQ:AMPLISEQ:QIIME2_BARPLOT\"];\n",
    "\n",
    "channel operator node: v173 [shape=circle,label=\"\",fixedsize=true,width=0.1,xlabel=\"collect\"];\n",
    "\n",
    "channel node: v330 [shape=point];"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_processes(dot_file):\n",
    "    G = nx.nx_agraph.read_dot(dot_file)\n",
    "    \n",
    "    # identify nodes without 'shape' attribute, assuming they represent processes\n",
    "    processes = [node for node in G.nodes() if 'shape' not in G.nodes[node]]\n",
    "    \n",
    "    return processes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Count all channels with operators"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_channel_operators(dot_file):\n",
    "    G = nx.nx_agraph.read_dot(dot_file)\n",
    "    \n",
    "    # count nodes with 'shape' == 'circle' attribute as channel operator\n",
    "    count_operators = [node for node, attrs in G.nodes(data=True) if 'shape' in attrs and attrs['shape'] == 'circle']\n",
    "\n",
    "    return count_operators"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_channels(dot_file):\n",
    "    G = nx.nx_agraph.read_dot(dot_file)\n",
    "    \n",
    "    # count nodes with 'shape' == 'point' attribute as channel\n",
    "    count_operators = [node for node, attrs in G.nodes(data=True) if 'shape' in attrs and attrs['shape'] == 'point']\n",
    "\n",
    "    return count_operators"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "data = []\n",
    "\n",
    "for filename in os.listdir(directory_path):\n",
    "    if filename.endswith(\".dot\"):\n",
    "        dot_file_path = os.path.join(directory_path, filename)\n",
    "        pipeline_name = filename.split('_')[0]\n",
    "        \n",
    "        operator_count = len(get_channel_operators(dot_file_path))\n",
    "        process_count = len(get_processes(dot_file_path))\n",
    "        channel_count = len(get_channels(dot_file_path))\n",
    "        node_count = len(get_all_nodes(dot_file_path))\n",
    "\n",
    "        data.append([pipeline_name, operator_count, process_count, channel_count, node_count])\n",
    "\n",
    "        print(f'Pipeline: {pipeline_name}, Number of operator nodes: {operator_count}')\n",
    "        print(f'Pipeline: {pipeline_name}, Number of process nodes: {process_count}')\n",
    "        print(f'Pipeline: {pipeline_name}, Number of channel nodes: {channel_count}')\n",
    "        print(f'Pipeline: {pipeline_name}, Number of all nodes: {node_count}')\n",
    "        print('-------------------------------------------------')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "csv_nodes_path = './results/nodes_in_pipelines.csv'\n",
    "header = ['pipeline', 'operators', 'processes', 'channels', 'total_nodes']\n",
    "\n",
    "with open(csv_nodes_path, 'w', newline='') as csv_file:\n",
    "    writer = csv.writer(csv_file)\n",
    "    writer.writerow(header)\n",
    "    writer.writerows(data)\n",
    "\n",
    "print(f'Data has been written to {csv_nodes_path}')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We read the csv-file next and derive our averages there"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "csv_nodes_path = './results/nodes_in_pipelines.csv'\n",
    "df = pd.read_csv(csv_nodes_path)\n",
    "\n",
    "sum_processes = df['operators'].sum()\n",
    "\n",
    "print(df.describe())\n",
    "print(f'Sum of processes in DAGs {sum_processes}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAnsAAAG4CAYAAAA9oCEQAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjcuMiwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy8pXeV/AAAACXBIWXMAAA9hAAAPYQGoP6dpAAB1+klEQVR4nO3dd3gUVfs38O+k90ZogUiHhF6SUEJIQIr0EhSQR0IHCwiKBVRAFJUmKKgYRYogSu+PNJNQEiBAHgQJYOgkhDRIhZBy3j94d367ye5mk9lNIHw/17XXtbtz5px7ZnZn7z1TjiSEECAiIiKiSsmsogMgIiIiItNhskdERERUiTHZIyIiIqrEmOwRERERVWJM9oiIiIgqMSZ7RERERJUYkz0iIiKiSozJHhEREVElxmSPiIiIqBJjskdajR49GpIkQZIkrFmzpqLDqZTS0tIwb948tG/fHq6urjA3N+c6p+ca9ztPp/DwcHm7BAUFGa1eQ/aBN27ckN+rW7eu0dp+3lhUdACmFhQUhIiICPm1p6cn/v33X1hbW5c474cffogFCxYAAEJCQrjzIaO5ceMGunTpgtu3b1d0KERE5Y77wPJV6ZO9om7fvo0ff/wRU6dOrehQ6Dk2adIkeSdna2uL7t27o1atWjA3NwcAeHt7V2R4REQmxX1g+Xrukj0A+PLLLzF+/HjY2dlVdCj0HEpMTMSBAwcAANbW1jh37hwaNWpUwVEREZUP7gPL33N5zl5iYiJWrFhR0WHQc+rs2bPy84CAAO7kiP6/NWvWQAgBIQRGjx5d0eGQiXAfWP6eq2Svffv28vMFCxYgIyOjAqOh59X9+/fl5zVr1qzASIiIyh/3geXvuUr2XnvtNTRp0gTAk6uAli5dWsER0fMoLy9Pfm5m9lx9BYmIuA+sAM/VWjY3N8fcuXPl119//TXS0tIU1VmWS9JV5SVJKlWZM2fOYMKECWjcuDHs7e3h5OSEgIAArFu3DkKIYnUcPnwYwcHBaNCgAWxsbFCjRg0MHjwYR48eLfVyAk8S5IULF8LPzw9Vq1aFra0t6tevj/Hjx+PMmTOlri86OhrTp09H69atUbVqVVhZWaFGjRoIDAzEggULNP796VK3bl15Pd24cQMAcPXqVXz00Udo06YNqlatCjMzM7Ru3brU8RWVlZWFb7/9Fr169ULt2rVhY2MDV1dXNG/eHG+99RZOnjypc171z8mYMWPk99euXauxrSVJ0viMloa221Yo3WZr1qyR61QdVisoKMDvv/+OgQMHon79+rC1tYUkSdixY0ex+fPy8rB69WoMGjQIderUga2tLZycnNCkSROMGzcOBw8eLPVyPnr0CL/88gteeeUVNGjQAE5OTrCyskK1atUQEBCADz/8UO+2UHf48GFMnjwZzZo1g5ubG6ytreHh4YFevXphxYoVePjwoUH13L59G59++im6dOmC6tWrw9raGo6OjqhXrx78/PwwZswYbNy4ESkpKTrryMvLw/r16zFkyBDUr18fDg4OcjwtW7ZE3759sWTJEly4cMGgmGJjYzFr1iz4+fmhevXqsLKyQtWqVdG+fXvMnj0bCQkJBtWTlZWFlStXom/fvnjhhRdgZ2cHGxsb1K5dG61bt8bgwYPx/fffIy4uzqD69DHk1itz584t9l3Jz8/HunXr5JP8ra2tUbNmTQwaNAh79uxRHBcAnDhxQm43ICBAZ7mcnBxYWVkZVPbRo0ewsbGBJEmwt7fXSIKKUrL/Uaftt+XcuXN4++230bx5c7i5uUGSJAwaNMig+opKTU1Fhw4d5DaaNGmCmzdvlss+UOXEiRN466230KxZM7i6usqf15deegkrVqxAdna2znnz8/Ph6OgISZJgbW2NnJwcnWX9/f3lmEsqGxgYKJc9fvy41jLR0dF466230LZtW7i6usLS0hJubm5o3LgxOnfujHfeeQf79+83eL+kQVRygYGBAoAAIH744QdRWFgoWrRoIb/3wQcf6Jz3gw8+kMuFhIRoLRMWFiaXCQwMNCgmVXl9q79omfnz5wtzc3ON99Ufo0aNEoWFhUIIIbKyskT//v11lpUkSXz99dd6YwwJCZHLr169WkRFRYlatWrprNPMzEzMnDnToOVPS0sTwcHBOutSPVxcXMTmzZv11lWnTh25/PXr18WPP/4obGxsitXVqlUrg2LTZffu3aJGjRolxvzqq6+K7OzsYvOrf05KesyZM6dMMZpim61evVrjOxAfHy8CAgK01rd9+3aNeU+cOCEaNGhQ4vL26NFDJCUlGbSMW7du1btM6o8ffvhBZz23bt0SQUFBJdbh4eEhjhw5ojemlStXCltbW4Ni8vf311rH5cuXhbe3t8GfkX///VdnPI8ePRKTJ0/Wu78AIGxtbcXy5cv1LltkZKTB6xuAyMvL01tfSYp+hrWZM2eOxnflzp07olOnTnrjGjNmjCgoKFAUW15ennBwcBAAhKWlpdbvuRBCHDhwQKNtfWX/+usvuVz37t11tq10/6NOvbwQT9ants/KwIED5XkM/Z27ffu2xue4Xbt28ne7rPvA69evy+/XqVNH77JlZWWJYcOGlVh/zZo1xb59+3TW06tXL7nswYMHtZbJzs4WlpaWGvXqKvvw4UNhbW0tAAg7Ozvx+PFjjel5eXli4sSJBq+fjz76SO960Oa5uxpXkiR89tln8r+W5cuXY/r06ahevXrFBqbHDz/8gI8++ggA0K5dO7Ro0QJCCBw9ehTXrl0DAKxbtw5NmjTBhx9+iODgYOzfvx9WVlbo3Lkz6tevj4yMDBw+fBipqakQQuCdd96Br68vOnfuXGL7N2/exDvvvIP79+/D3t4e3bp1Q/Xq1ZGYmIiwsDBkZ2ejsLAQX375JfLz87Fw4UKddSUmJqJbt26IjY2V3/P29kbr1q3h6OiIpKQkHDt2DCkpKXjw4AFeeeUV/Prrrxg5cmSJcW7evBnvv/8+AKBWrVro1KkTnJ2dkZCQoKgH948//sDIkSNRUFAA4EkPcefOndGwYUNkZWXh6NGjci/Jb7/9huvXr+Ovv/6CjY2NXEetWrXw5ptvAgAuXbqEw4cPAwC8vLzw4osvarTn5+dX5lhVjLnNVHJzczFgwACcOXMGFhYW6NSpExo2bIhHjx5pnHANAEeOHEHv3r01/un6+fmhWbNmePz4MU6cOIGrV68CAA4ePIjOnTvj2LFjqFq1qs72lyxZgvfee0/uxZYkCa1atULTpk3h4OCAtLQ0nD9/HpcvXwbwpNdEm9jYWLz44ou4e/euXE/r1q3RrFkz2NnZIT4+HkeOHEFmZiYSEhLQo0cP/Pe//0XXrl2L1bVjxw5MnjxZfu3k5ISOHTuidu3asLCwQHp6Oq5cuYILFy7g8ePHWuPJzMxE9+7d5dtQmJmZoU2bNvD29oaDgwNycnIQHx+Pc+fO6e0ZBIDs7Gz06tVLo+egXr168PHxgaurK+7fv4/IyEjEx8fj4cOHmDJlCjIyMjBr1qxidd2+fRu9evVCZmYmAMDS0hK+vr5o2LAh7OzskJ2djRs3buDcuXMVdv5zVlYWXnrpJVy4cAF2dnYICAiAp6cnMjMzERYWhqSkJADA6tWr0aRJE3zwwQdlbsvCwgL+/v7Yv38/8vLycPz4cfTo0aNYufDwcI3XhpbVdWTIGPsfXRYtWoRPP/0UANCgQQP4+fnBzs4ON27cgKWlZYnzq7t06RJ69uwpf467du2KnTt3wtHREYDp94E5OTno1q0bTp06Jb/n4eGBgIAAODg4IC4uDseOHUNBQQHu3r2LAQMGYOPGjRg6dGixuoKCgrB//34AQFhYGLp3716szPHjx4v1xOoqGxUVhdzcXABAp06diq3bGTNmIDQ0VH5dq1Yt+Pn5wd3dHUIIpKam4uLFi/K+rUxKnR4+Y4r27Kn4+vrK77/99tta531aevasrKxErVq1xPHjxzXK5OfnixkzZsjlXFxcxNy5cwUAERQUJG7evKlR/sGDB6Jr165y+aCgIJ3tq//DtrKykv81PnjwoFidw4cP14j3r7/+0lpnQUGBRvtt27YV0dHRxco9fPhQzJ07V0iSJAAIe3t7ce3aNa11qvfsWVhYCCsrKxEaGir3cqo8evRI57LqExcXJ/+bByB8fHzElStXii3XkiVLhJmZmVxuypQpOuss2ltmLKbYZuqxWlhYyJ/z69evFyurWsdpaWkavUENGjTQup3Xr1+v0SPWv39/ncu2d+9e+fMAQHTr1k3ExsZqLXvt2jXxySefiDVr1hSblp2drdHz0L17d3H58uVi5dLT08XkyZM1egKKrkchhGjVqpVc5q233tLZq5KZmSk2bdqk9UjC0qVL5TqaNm0qLl26pLWOwsJCcerUKfH666+LW7duaS0zatQojfWurachPz9ffP/993JPg7m5uYiMjCxW7u2335brCggIEPHx8VrbzMvLE+Hh4WLkyJEiPz9faxlDlbZnT7UMISEhIjU1VaNcdna2GDFihFzWwcFBZGVlKYrvyy+/lOvT1Suu6mWsWrVqiWXVe8mPHTtWbLop9j/q33sLCwvh7OxcrFdeCM19Zkm/cydPnhRVqlSRywwePFjvPrc0+0BDe/Zef/11uZy5ubn4+uuvi/XmXrlyRbRr104u5+TkpPW3JSoqSi7TsWNHre3NmjVLLqPa1rrKfvLJJ3LZzz//XGNacnKyvG81NzcXa9asKfb7pZKQkCC+/fZb8fPPP+tcD7o8t8nen3/+qbHDuH37drF5n5Zkz9bWVly8eFFruYKCgmKHf5o1ayYePnyotfytW7fkD5YkSSIhIUFrOfWdLgDRp08fnYdBCgoKNLq9fX19tZZbt26dXKZ169YiMzNT5/ILIcTs2bPl8pMnT9ZaRj3ZAyDWr1+vt87SUv/xrF+/vrh//77Osl9//bVc1szMTGeCWh7JnrG2mXqsAESLFi1ETk6O3jjUt5uLi0uxPx3qtm3bplF/REREsTJ5eXmibt26cpl+/fqV+XDhvHnz5HpeeumlEutR3/5fffWVxrTMzEx5mqenp84ddEnUT2nQdRjIEEeOHJHrqV27trh7967e8r/88ovGuihK/UdR32FjYyptsgdAjBgxQmd9Dx8+FJ6ennLZ33//XVF8J06c0JsEqB/aGzt2rPDy8tJZNicnR++hPSFMs/9RX3eSJGn9zhWl73fuwIEDGgnp+PHjS0z6jZ3sxcXFaSS73377rc760tLSNPYnY8aMKVYmLy9PODo6CuDJYXhtfxJUSX39+vXlP4a6yqon9UU7bXbv3i1PGzlypN51ocRzm+wJobkBJk6cWGzepyXZmz59ut761H9cAYhdu3bpLd+lSxe57O7du7WWUd/pSpIk4uLi9NZ5+fJljZ6X//3vf8XKtG7dWp5+9OhRvfUJ8WRH7eLiIgAIZ2dnrYmLerLn5+dXYp2lcf/+fXlnDEBs3bpVb/mCggLRrFkzufyHH36otVx5JHvG2mZFkz1957kI8aT3Sf3coiVLlpQYd+/eveXyw4cPLzb9999/l6fb29uL5OTkEuvU5vHjx6JatWryj+GNGzdKnCc+Pl5eRy1atCg2TRVX69atyxSTEEL06NFD7zYw1KBBg+R6fv311xLLFxYWysmIJEnF1mujRo3k+rT1appCaZM9KyurEpPa999/Xy7/7rvvKopPPQmwsLAo9od1//79clvr1q2Te5u0lT106JBctkePHsXaMtX+R/37/Morrxi03Lp+537//Xf5KAKg/xx4dcZO9tR/q1u2bFniH68//vhDLm9tba318/3SSy/JZf7880+NaUWTevX6ipbNycmR15G2pH7Dhg3yvNOmTdMbtxLP1dW4RX3++efy89WrV8vnvz1tgoOD9U5v3ry5/NzW1ha9e/c2uPz169dLbN/f3x8NGjTQW6Zx48bo2LGj/DosLExj+t27d/G///0PwJPzEQw5V9DGxkauMz09vcSrEIcPH15inaURGRkpn2dRpUoVDBgwQG95MzMzjB07Vn5ddB2UJ2Nss6JcXV3Rq1cvvWViY2ORmJgI4Mn6CAkJKTHW8ePHy8+Lnu8EAH/++af8fMSIEXB3dy+xTm1Onz4tn8PVsWNH1KlTp8R5PDw84OXlBQC4cOECHjx4IE9TXd0MAOfPn8eRI0fKFNcLL7wgP//+++/LVEd+fr58ZbOFhUWJ+wzgybmKqvMQhRCIjIzUGdd3331XprhMrXPnzqhRo4beMm3atJGfq67YLyvVeXvAk3V+7Ngxjenqn9+uXbvK5+GVVFbb+Xrlsf9Rss/8/vvv8eqrr+Lx48eQJAmLFy/GV199Veb6lPjrr7/k52PGjNF7pwsAGDx4MNzc3AA8ORc5KiqqWBn1bVJ0XR47dkw+X099O2srGxkZKZ+v6+/vX+x8PfXv2datW+X9p7E9dxdoqOvSpQt69OiBgwcPIi8vD59++inWrl1b0WEV06xZM73TXV1d5eeNGzeGhYX+zape3pATqzt06FBiGeDJD6jqByMmJkZjmvqXSQiBt956y6A6VSfxA09OGG/ZsqXOsu3atTOoTkOpL0P79u1LXK8A5B8C1fxCiBJ3PKZgjG1WVOvWrUu8J5Z6HV5eXqhSpUqJMaivs8TERCQkJMDDw0N+78SJE/JzbRdJGEr9M5iSkmLwZ1CV4AkhEB8fDxcXFwBPLlgYPHgwfvvtNxQUFKB79+4YOnQohg4disDAQIOWHQCGDRuGVatWAQBCQ0MRHR2NkJAQ9OrVS040S/L333/Lt5OwsbHBe++9Z9B80dHR8vOiA9IPGzZMPon+o48+woEDBzBy5Ej06NEDdevWNah+U2vRokWJZdS3Q3p6uuI2g4KC5D8gYWFheOmll+RpqgSuYcOGqF27drEkQL2selKgLdkrj/1PWfeZn376qXx7FAsLC/z8888G/bEzBSGE3JEAaK4DXSwtLeHn5ydvx7Nnz2psG0B/slc0Ua9WrRqaNm2KixcvFitb0nZu37496tSpg5s3b+L27dto1qwZQkJC0L9/f3To0EH+Q6nUc53sAU9691T/iNevX48PP/zwqRuA2dnZWe909Z1ASWWLltd3XycV9X8e+nh6esrPk5OTNaap39MrISGhTD0FJd13T9+VnGWhvgyG9AIB0PgRfPz4MTIzM+Hk5GTUuAxhjG1WlCHrtyzrrHr16rCxsZGvnk1JSdFI9u7duyc/r1+/vkF1aqP+Gbx8+XKZrmwr+hlcunQpzp49i0uXLiEvLw8bN27Exo0bIUkSvLy8EBAQgJ49e6Jv3746r47s0aMHpk+fLt/kPSYmRv6hd3d3h7+/P4KCghAcHKyxvXQtW1ZWllG+X+PGjcOBAwewZcsWAEBERAQiIiIA/N9Vjt26dcPgwYON/t0zlCH7O/WeFEP2dyVR/8FW/9HPzs7G6dOnNcqoJwHqZXNycuRE297eHr6+vsXaKY/9T1m22/Hjx+XPgaWlJbZs2VJir6Mppaena2zXsqwrbVe5+/j4wNHREZmZmThz5gwyMzPlK4uLJvXAk21+8eJFnWVVZYqytLTE+vXr0bdvX2RkZMgDPixduhRWVlZo27YtunTpgr59+yIgIKDMnQfP9WFc4Mkl3qoPamFhIebMmVPBERVXmo1ril4kOzs7g8rZ29vLz1W3a1Axxj/q/Px8vdON9Q9IJSsrS36uvmz6FC1XdD2UF2Nss6IMWb9lWWclxaH+2sHBweA6izLFZ7BatWqIjo7G3LlzNRJUIQRiY2MRGhqKoUOHwsPDA1999ZV8+4yivv76a+zatQudOnXSeD8lJQU7d+7E9OnTUadOHQQHB+PmzZvlsmxmZmbYtGkT1qxZU6xHPSEhAX/88QcmTZoEDw8PjB8/XvEN6suiInrNVUkA8ORG96qjI+q34lDvgVb9wBctq+/QHlA++5+y7DOLJs///vtvqeswJvX1BJRtXWlbT6pb3ABPbiSvGoxAPanXtp3Vy+bk5Mi3gtGV1ANPTkc4d+4cxo4dqxGX6jZVCxcuRGBgILy8vLB9+3aDlq+o5z7ZA4DPPvtM3mls2bJFo0vY2AoLC01Wt6nouyu4OvW7kqt2hirqH+BBgwbJg52X5lHeA6OrJxb67riurmi5ouuhvBhjm5VFWdZZSXGovy66Yy8N9c/gtGnTyvQZ1PbP3MHBAXPmzMHt27cRHR2NJUuWYNCgQRrnFt6/fx8zZ85EcHCw1tFuAKB///44fvw47ty5g/Xr12PSpElo2rSpPF0IgW3btqFdu3a4cuWKzmVr3bp1mZZN26gFkiQhJCQE586dQ1xcHFatWoWQkBCNHtb8/HysWrUKfn5+JfYOVwa6kgBdh+tUCYF6WUPur/e07n/8/PzwxRdfyK9nzJiBr7/+2ujtGKroH8CyrCtd60nbodyi5+upl1XlEaqy6n8AdCX1KnXr1sWqVauQnJyMQ4cOYfbs2ejatatGQn7lyhUMGTKkTOubyR6Ali1b4uWXXwbwZIf6ySefGDyv+sYrqecJMM4/8PJ269atUpcrehK9+k2rK/qfoKHUD3EYug7Ue12srKwqLNkzxjYri7Kss6SkJI0bIOv77BhyQZEupv4MmpmZwcfHB++88w62b9+Oe/fu4dixYxrDTu3cuRNbt27VW0+tWrUwcuRIrFy5Ev/88w9u376Nzz77TE7oUlNT8c4772jMo75s165dM8mfygYNGmDs2LFYs2YNrl69iitXrmDGjBnyaSFXr16Vb9Bb2WlLAlQJXJMmTTR6ebUlASWdxwU83fufmTNnaiR87777boWNNe/s7KzxO1yWdaVr36dvOxedXrVqVfnPWWm2c1G2trZ48cUX8emnn+Kvv/5Camoqtm7dilatWsllZs6cifj4eIPqU2Gy9//NmzcP5ubmAIA9e/YYPM6g+pcpNTW1xPLnz58vW4AVSNuVStqon0jftm1bjWnqFwz8888/uHPnjnGCMyH1q/hOnTql8xCcOvWRC9q0aVMhh5kA42yzslBfZ5cuXTLo0J76OqtRo4bGDyWg+dlRv+qutNq3by8/j4iIkK90NBUzMzP4+/tj27ZtGlcx79q1q1T11K5dGx9//DF++ukn+b0DBw5oxN+6dWtYW1sDeHLRlaHbX4lGjRph0aJFmDdvnvxeaZftWVX0vD1t5+upuLu7yxfZFS2r79De077/KZrwvfPOO1i2bJnJ2tNFNQKOStGryrXJz8/XGGlD176vXbt28m98TEwM0tPTNZL6mjVrapRX9fQVLQsYnuwVZWtriyFDhiAsLEy+8vzx48fyCB+GYrL3/zVp0gT/+c9/5Ncff/yxQfPVrVtX/kLFxcWVeJhp06ZNZQ+ygkRGRmpcFavNlStXNH5gil41Wa9ePY0LXypip1BanTp1kn9Ak5OTsXfvXr3lhRAaV3N369bNpPHpY4xtVhbe3t7yDqmgoADr168vcZ7Vq1frjUH9VkK///57iUOG6eLv7y9fSZuVlaWRPJmSJEno27ev/Fr9gpPS6Nevn/w8Ly9PI5G2tbXV+LyVZy+LelxlXbZnTbt27eQLH2JiYrBnzx75yI62z7Dqh15VVv3Qnq6rbJ+F/c/MmTMxf/58+fX06dMrZN+uvqxr167VeaqEyq5du+TOGfVbfBWlfsi+sLAQ+/bt03q+nopqOxctqy+pN5Srq6vGOb2l/a4x2VMzZ84cuTv40KFD8hVH+jg5OaFJkyYAnvxb2LBhg86yMTEx5fYDY0xCCEydOlXnoaHCwkJMnTpV/oL5+PhodDmrqI9L+c033+DQoUMGx2Cqew/p4+LigmHDhsmv33vvPb0nPH/33Xc4d+4cgCe9OhMmTDB5jLoYa5uVliRJmDhxovx63rx5eg837N27F7t375Zfq48zqzJkyBD5CrusrCyMGTPGoFMmirK2tsa0adPk17NmzSpVT3vRnWtmZqbO8W6LUj+0VPQKSEOTV/U6zMzM5PuEqah/v7Zu3Yo1a9YYVC+g/ftVlrgq6qrc8lY0CVC/Z6u2HhxVYlC0rL4/WM/K/mfWrFnFEr5vvvmmXNpWmTBhgnxbqLNnz2qMM1tUenq6PIY68OTenfqu6lbfRl988UWJSb2q8+eLL76Qk/rOnTvrTOoNOSKooui7Zrz7Mz+d9I2goY36eJjqD313+VYfgsnFxUXr6BD79u0T7u7uGndE17f6DSmjUtpRPNTvQD9nzhytZbSNszpy5Eit46yqjz8J6B7yKT8/X3Tr1k2j3q+++krneJWZmZnit99+E127dhVDhw7VWkZ9BA1t47UqdfXqVY2hgDp27CiuXr2qUaagoEB8++23wtzcXC73tIyNq3SblSXWomPjNmnSRMTExBQr9/vvvwt7e3u5nL6xcffs2WPw2LjXr18Xn3zyiVi7dm2xaZmZmRqjDDg5OYnQ0FCRm5urta6UlBTx008/ibZt2xYbfSEsLEzUqFFDzJ49W1y4cEHr/Pn5+WL9+vUa3/sNGzZolGnYsKEYPny42Lt3r844Ll68KNq2bSvXoW3EBSGKj6IyY8YMkZKSorXso0ePxI4dO8SgQYO0Dpdna2srJkyYIMLCwnQOfxUVFaUx9NSECRO0ljNUaUfQ0LX/UleWUY4MsXDhwmK/E97e3lrLpqSkaHx+VY+oqCi9bZhi/1Oa3xYVQ9bh559/rlH3smXLdNZn6rFxLSwsxIoVK4qNuhQXFyf8/Pw0vv+6hpVTOXXqlNac4N69e1rLt2jRoljZL7/8Umf9c+bMEa1atRLfffedzuFLMzIyNEaCMTc31zk+ti5M9oq4c+eOsLGxKVWy9+DBA+Hh4aGxk+3cubMYP368+M9//iMPSaTagT1ryd7cuXOFq6urAJ4MJj5gwAAxYcIEMWDAAI0dEQDxzjvv6G07JSVFtGnTRmMeOzs70a1bNzF69GgxceJE8fLLL4tWrVrJw9EAEMHBwVrrM3WyJ8STpER9R2phYSG6du0qxo8fL0aMGCFq166tsTwdOnTQOTaxEOWT7Blrm5U11oiICGFnZ6fxnejQoYMYO3as+M9//qMxFBcA0ahRI5GUlKS3zq+++kpjHkmSROvWrcXIkSPFpEmTxMsvvyyaNGkiT1+6dKnWeq5evSrq1aunUZezs7N46aWXxLhx48T48ePFkCFDhLe3t8Z4m9qSPfU6qlevLl566SUxevRoMX78eNGvXz9Rs2ZNjTIBAQHFfoDUP8O2traiffv2YsSIEWLSpEnilVde0RijVlXm3LlzWpft0aNHomfPnhrlrayshL+/vxg1apSYNGmSGD58uPD19dXYz7Vr165YXep1ODo6ioCAADFy5EgxceJEMWTIENG0aVONMlWrVhXx8fF6t2FJnqVkT1sS8MYbb+gs37JlS42yDg4OBo3xbOz9T2l+W1QMXYdFE75vvvlGazlTJHvZ2dnC19dXo/3atWuL4cOHi/Hjx4uuXbsWW4+bN28ucdnz8/OFk5OTRr1NmzbVWX7q1KnFPhf6knr1z7MkSaJhw4Zi0KBBYvz48SIkJET06NFD408xAPHRRx+VGHdRTPa0mDZtWrGNVdIH8syZM8Ld3b3YfOo73O+++04IUfqxcUti6mRv9erVIjIyUiOhLfowMzMT7733nkGDwefk5IjJkycLCwsLnfUV/XH74osvtNZVHsmeEE8Gq65evXqJsY4YMUJkZ2frras8kj1jbTMlsUZFRYn69euXuM66d+9eYqKn8vvvvxu0HQCI0NBQnfWkpqaKl19+WWtvi7aHi4uLWLNmjUYdJ06cMPgzDEAMHTpUZGRkFIulefPmBtdRr169YgOpF5Wfny8++eQTjWRb38PS0lK8+eabxeop+qdA36NVq1Y6e1pL41lK9rQlAfqSh6JJQK9evQxuy5j7H/XyhirNOvzss8802vj222+LlTFFsifEk577V155pcT1VLNmzRLH+VbXp08fjfm1fV9Utm3bplG2pKR+8eLFBn/PrKysxKeffmpw3Oqe+xE0tJk5cyZ++umnUt0nrG3btrh06RK+/vpr7N69G9evX0dhYSFq166NHj164I033tC4Z9azpmPHjjh37hxCQ0Oxbds23LhxA1lZWahZsya6du2KN954Az4+PgbVZWtrix9++AEffPAB1q9fj7/++gtXrlxBamoqCgsL4ezsjPr166NVq1Z48cUX8dJLL1XIKBTq+vXrh7i4OPzyyy/Ys2cP/vnnH6SkpMDW1hYeHh7o2rUrRo0apXHFZ0Uz5jYriw4dOiA2Nhbr16/Hjh078L///Q9JSUmwtLREjRo10LlzZ4wYMQI9e/Y0uM5hw4ahX79+WLduHf773//i3LlzSE5ORkFBAVxdXdGkSRN07twZQ4cO1biasSg3Nzds2rQJFy5cwMaNGxEeHo7r168jNTUVZmZmcHFxQcOGDdG2bVt0794dPXr0KDYCRvv27ZGUlIRDhw7h2LFjiImJwdWrV5GamoqCggI4OTmhQYMG6NChA/7zn//Az89Payz/+9//cOLECYSFheHUqVO4fPkyEhISkJOTAzs7O9SoUQOtW7fGgAED8Morr8gn7etibm6OefPmYcqUKVi3bh0OHTqEixcvIiUlBXl5eXByckKdOnXQokULdO3aFX369NF6/k9qaiqOHDmCiIgIREdH499//8W9e/fw6NEj2NnZoXbt2mjXrh2Cg4MxYMCAEofSq2zMzc0REBAgXzghSRICAwN1lu/atSu+/fZb+XVprs58lvY/qosbVbcwmzp1KiRJMnh4QiUcHBzwxx9/YNq0afj1118RHh6OhIQEPHz4EO7u7mjevDn69etX7ObFJenatSv27dsnv9a37QIDAyFJknw+tL6LcIAnt60JDg7GwYMHERkZifPnz+PGjRvIyMiQ90Xe3t7o1q0bRo0aZfAIIUVJQhURET3TRo8eLV+Nt3r16nK/CTURET2dnq+/YkRERETPGSZ7RERERJUYkz0iIiKiSozJHhEREVElxmSPiIiIqBJjskdERERUifHWKyZWWFiIhIQEODo6ymPmERER0bNHCIHMzEx4eHg8U/eW5E2VTSwhIQGenp4VHQYREREZye3bt1G7du2KDsNgTPZMzNHREcCTD0ZFjwJBREREZZeRkQFPT0/5t/1ZwWTPxFSHbp2cnJjsERERVQLP2mlZz84BZyIiIiIqNSZ7RERERJUYkz0iIiKiSozJHhEREVElxmSPiIiIqBJjskdERERUiTHZIyIiIqrEmOwRERERVWJM9oiIiIgqMSZ7RERERJWYyYdLy83NxbFjx5CSkoJ69erBz8/P1E0SERER0f+nKNm7efMmvvvuOwDArFmz4OLiojH9xIkTGDp0KO7evSu/17ZtW2zduhUvvPCCkqaJiIiIyACKDuNu374dixcvxl9//VUs0cvMzMSgQYNw9+5dCCHkx5kzZ9C3b1/k5+craZpMrLCggHUTERFVAop69g4ePAhJkjBo0KBi00JDQ5GUlARJkjB16lS8+OKL2L9/P77//ntcvHgRa9euxbhx45Q0TyZkZm6ObSNHIjk21qj1NurdG93mzzdJ3VW9vTFkwwaj1klERPSsU5TsXbt2DQDQrl27YtM2bdoESZIwePBgLFu2DADQv39/JCcnY/PmzdiyZQuTvadccmwsEmNijFqnu5eXyeomIiKi4hQdxk1KSgIAVK9eXeP9jIwMnD17FgAwZswYjWnDhw8HAJw7d05J00RERERkAEXJXmZmJgCgoMh5UsePH0dBQQHMzc0RFBSkMc3T0xMAkJaWpqRpIiIiIjKAomTP2dkZAJCQkKDxfnh4OACgVatWsLe31zqvjY2NkqaJiIiIyACKkr3mzZsDeHJVrkpBQYF8vl7Xrl2LzRMfHw+g+KFfIiIiIjI+RRdoDB48GBEREfj1119RvXp1BAQE4Ndff8XNmzchSRJeeeWVYvOcPn0aAHifPSIiIqJyoKhnb9KkSfD29oYQAosXL8bAgQOxZcsWAE+uvPXx8Sk2z/bt2yFJUrFz+YiIiIjI+BQle9bW1jh8+DCGDBkCCwsLCCFgaWmJ1157Db/++mux8keOHMHFixcBAD169FDSNIAnVwPv2bMHs2fPRu/eveHu7g5JkiBJEkaPHm1QHWvWrJHnKemxZs0axTETERERlSfFY+PWqFEDW7ZsQW5uLtLS0lClShVYWVlpLevp6YmwsDAAgK+vr9Kmed4fERERUQkUJ3sq1tbWqFmzpt4y9erVQ7169YzVpAZPT094e3vjwIEDZa5j//798PDw0Dm9du3aZa6biIiIqCIYLdmrCLNnz4avry98fX1RvXp13LhxQ1Ey2bhxY9StW9d4ARIRERFVMEXJXo0aNRAYGCg/mjVrZqy4DPLpp5+Wa3tEREREzxpFyV5SUhK2bNkiX4FbpUoVdOnSBUFBQQgMDESLFi2MEiQRERERlY2iZG/q1KmIiIjA+fPnUVhYiJSUFGzfvl2+ybKrq6tG8teqVSujBE1EREREhlGU7C1btgwA8ODBA0RERCAiIgLh4eH4+++/UVhYiLS0NOzcuRM7d+4EALi4uCAgIEBO/tq0aaN4AYxp9OjRiI2Nxf379+Hk5ISGDRuie/fueP3111GrVq2KDo+IiIio1IxygYaLiwsGDhyIgQMHAgDS09M1kr9z586hsLAQ9+/fx+7du7F7925IkoT8/HxjNG80ERER8vPU1FSkpqbi5MmTWLJkCZYtW4ZJkyZVYHREREREpWeSq3GdnZ0xYMAADBgwAACQlpaGb775Bt9++y0yMjIghDBFs2VWv359DBkyBB07doSnpycA4Nq1a9i6dSu2bNmCR48eYfLkyZAkCRMnTtRbV25uLnJzc+XXGRkZJo2diIiISB+TJHtCCMTExCA8PBzh4eE4evRosSTPzs7OFE2X2uDBgxESEgJJkjTe9/X1xbBhw7Bnzx4MGTIEeXl5mD59OgYMGIAaNWrorO/LL7/kVcJERET01FA0XJqKEAJnz57F119/jQEDBsDNzQ2+vr547733sGfPHqSnp8PGxgbdunXDvHnzcOTIEaSlpRmjacWcnZ2LJXrq+vXrhzlz5gAAcnJysGrVKr31zZw5E+np6fLj9u3bRo2XiIiIqDQU9ewtXboUYWFhOHbsGNLT0wFA7r2ztbVFx44dERQUhKCgILRv3x6WlpbKI64AEyZMwCeffAIhBCIiIvDRRx/pLGttbQ1ra+tyjI6IiIhIN0XJ3rvvvgtJkiCEqFTJXVHVqlWDu7s7kpOTER8fX9HhEBERERnMKOfsSZKEgIAA9OzZE0FBQWjTpo3eQ6PPoqftohIiIiIiQyhK9rp164YTJ04gJycHBw4cwMGDBwEATk5O8v30KkPyl5SUhNTUVACAh4dHBUdDREREZDhFyd6hQ4eQl5eHU6dOITw8HBEREYiMjER6ejr27NmDvXv3Anj2k7/Q0FC5Zy8wMLCCoyEiIiIynOLDuJaWlvD394e/vz8++ugj5Ofnl5j8OTs7IyAgQB5Zo6LcuHED9+/f1zuSx549e/DZZ58BAGxsbDBmzJjyCo+IiIhIMaPfZ8/CwgKdOnVCp06dMGvWLOTn5yM6Ohp79uzB999/j/T0dDx48AB79uxR3NaxY8cQFxcnv05JSZGfx8XFYc2aNRrlR48erfH6xo0b6Nq1Kzp27Ij+/fujdevWqFatGoQQuHbtGrZs2YItW7bIvXqLFy/msGlERET0TDHJTZWBJ/ekO3bsmHxj5TNnziA/P1++etcYfv75Z6xdu1brtOPHj+P48eMa7xVN9lSioqIQFRWlsx07OzssXbq0xNEziIiIiJ42Rkv2dCV3AIqNnKG6RUtFa9euHdavX4+oqCicPn0ad+/eRUpKCvLz8+Hq6opmzZrhxRdfxPjx41GtWrWKDpeIiIio1BQlewcPHkRYWJje5M7e3h6dOnVCUFAQAgMD4efnBwsL4+SYa9asKXaotjQcHR0xcuRIjBw50ijxEBERET1tFGVdvXr1KnZY1sHBAf7+/nJy5+vrC3Nzc8WBEhEREVHpKe5ic3R01Eju2rVrx+SOiIiI6CmhKNk7deoU2rZtCzMzM2PFQ0RERERGpCjZ8/HxMVYcRERERGQC7JIjIiIiqsSMduuVtLQ0rF69GocOHcKFCxeQlpYGAHBzc0Pz5s3RvXt3jBkzBm5ubsZqkoiIiIhKYJRk78cff8SMGTOQk5MDQPPWK/Hx8UhISMCBAwcwd+5cLFmyhDcnJiIiIionipO9r776Ch999JGc4Dk7O6NNmzaoUaMGhBC4d+8eYmJikJ6ejuzsbLz++ut48OAB3n//fcXBP+8KCwpgxiufiYiISA9Fyd6FCxfwySefQAiBmjVrYtGiRXj55ZdhaWmpUS4/Px+bN2/Ge++9h4SEBHz88cfo27cvmjVrpij4552ZuTm2jRyJ5NhYo9bbqHdvdJs/36h1EhERUcVQlOytWLECBQUFqFq1KqKiovDCCy9ob8TCAiNGjEDnzp3h6+uL5ORkrFixAj/88IOS5glAcmwsEmNijFqnu5eXUesjIiKiiqPoaty//voLkiRh5syZOhM9dZ6envjggw8ghMDhw4eVNE1EREREBlCU7MXHxwMAOnXqZPA8/v7+AICEhAQlTRMRERGRARQle6ph0fLz8w2eR1WWo24QERERmZ6ijEt16LY0h2RVZQ057EtEREREyihK9nr06AEhBBYvXozz58+XWP7vv//GokWLIEkSevbsqaRpIiIiIjKAomRv2rRpsLa2RlZWFjp37oxFixYhJSWlWLmUlBQsXLgQXbp0QVZWFqytrTFt2jQlTRMRERGRARTdeqVOnTr48ccfMWbMGGRlZeHDDz/EzJkzUbduXVSvXh2SJCExMRE3btyAEAJCCEiShB9//JGHcYmIiIjKgeIRNEaNGoUqVapg8uTJiI+PhxAC165dw/Xr1wFoDp3m4eGB0NBQ9OnTR2mzRERERGQAo4yN27dvX1y/fh3bt2/HoUOHcOHCBaSlpQEA3Nzc0Lx5c3Tv3h2DBw+GhYVRmiQiIiIiAxgt87KwsMDLL7+Ml19+2VhVEhEREZFCvNkdERERUSXGZI+IiIioEmOyR0RERFSJGXTOnmpYNGOSJKlUw6wRERERUekZlOyp3z6FiIiIiJ4dBiV7c+bMMXUcRERERGQCTPaIiIiIKjGDkr1vv/0WAPDaa6/B1dXVpAERERERkfEYlOxNmzYNkiShe/fuGslet27dIEkSfvnlF9SpU8dkQRIRERFR2SgaQSM8PBySJCE7O9tY8RARERGRERl0nz0bGxsAwIMHD0wZCxEREREZmUHJXt26dQEAe/bsMWUsRERERGRkBh3G7dOnDy5duoQFCxbg8OHDaNy4MSwtLeXpH3/8MVxcXErVsCRJWLVqVanmISIiIqLSMSjZmzVrFnbt2oW4uDhER0fj9OnT8jQhBHbu3FmqRoUQTPaIiIiIyoFByZ6bmxtOnz6NFStW4PDhw4iPj0dubi5u3rwJSZJQs2ZNjZ4+IiIiIno6GHw1rpOTE2bNmoVZs2bJ75mZPTnl78CBA2jatKnxoyMiIiIiRQy6QIOIiIiInk2K7rMXFhYGAKhXr55RgiEiIiIi41LUs5efn4/AwEDY2tqWet433nhDSdNEREREZABFyd7gwYMRHR1d6vkmTpyIH3/8UUnTRERERGQARcleVlYW+vTpg9jYWIPnGT9+PH7++WclzRIRERGRgRQle40bN0Zqaip69uyJW7dulVh+9OjRWL16NQBg5MiRSpomIiIiIgMoSvYOHjyI2rVrIz4+Hj169EBSUpLWckIIvPbaa1i3bh2EEAgJCcHatWuVNE1EREREBlCU7Hl6euLgwYNwd3dHXFwcevXqhfT0dI0yhYWFGDlyJDZs2AAAGDt2LH755RdIkqSkaSIiIiIygOL77DVp0gT//e9/4eDggL///hv9+vXDo0ePADxJ9EaMGIHff/8dwP+dr8dEj4iIiKh8GOWmyu3atcPOnTthbW2NyMhIBAcH4+HDhxg6dCg2b94MAJg0aRJCQ0ON0RwRERERGchoI2gEBQVh48aNMDMzw59//ol69eph586dAIDXX38dP/zwg7GaIiIiIiIDGXW4tIEDB8q3VUlKSoIQAm+99Ra+++47YzZDRERERAYyaLg0Q26rotK1a1e8/fbbWLZsGYYOHYoZM2bonP+FF14wuF4iIiIiKj2Dkr2yjH0rSRK2bt2KrVu36pyen59f6nqJiIiIyHAGJXtCCFPHQUREREQmYFCypxr1goiIiIieLQYleyEhIaaOg4iIiIhMwKhX4xIRERHR00VRsmdmZgYLCwssXLjQWPEQERERkREpSvasrKwghEBAQICx4iEiIiIiI1KU7Hl4eAAAzM3NjRIMERERERmXomSvS5cuAICzZ88aJRgiIiIiMi5Fyd6UKVNgbm6OxYsXIyMjw1gxEREREZGRKEr22rVrh+XLl+PmzZsIDAxEZGSkseIiIiIiIiMw6D57uowdOxYA0KRJE5w7dw4BAQHw9PREy5Yt4erqqvdcPkmSsGrVKiXNExEREVEJFCV7a9asgSRJAJ4kb0II3Lp1C7dv39Y7nxCCyR4RERFROVCU7L3wwgtyskdERERETx9Fyd6NGzeMFAYRERERmQKHSyMiIiKqxJjsEREREVViig7j6pKfn4/79+8DAFxdXWFhYZJmiIiIiKgERuvZi42NxZQpU+Dt7Q0bGxvUqFEDNWrUgI2NDby9vTF16lRcvHjRWM0RERERkQGMkuzNnDkTLVu2xPfff4/Lly+jsLAQQggIIVBYWIjLly/ju+++Q6tWrTBr1ixjNElEREREBlB8fHXKlCn4/vvvIYQAAHh7e6N9+/aoUaMGhBC4d+8eTp06hYsXL6KgoAALFixAdnY2vvnmG8XBExEREZF+ipK948eP47vvvoMkSWjatClCQ0PRqVMnrWWjoqIwefJknD9/HitWrMCwYcN0liUiIiIi41B0GPfHH38EANSrVw/Hjx/Xm7x17NgRR44cQf369QEAK1euVNI0ERERERlAUbJ39OhRSJKEDz/8EM7OziWWd3Z2xgcffAAhBI4ePaqkaQBAUlIS9uzZg9mzZ6N3795wd3eHJEmQJAmjR48udX1//vknhgwZgtq1a8Pa2hq1a9fGkCFD8OeffyqOlYiIiKgiKDqMm5iYCABo06aNwfO0bdsWAHDv3j0lTQMAqlevrrgO4MlYvZMnT0ZoaKjG+/Hx8di+fTu2b9+OiRMnYuXKlRwejoiIiJ4pinr2bGxsAADZ2dkGz5OVlQUAsLa2VtJ0MZ6enujZs2eZ5v3444/lRK9NmzbYuHEjTp06hY0bN8qJbGhoKD755BOjxUtERERUHhQle/Xq1QMA7Nq1y+B5du/eDQDyuXtKzJ49G7t370ZiYiJu3boln0NYGnFxcVi4cCEAwMfHB8ePH8fw4cPh6+uL4cOH49ixY/Dx8QEALFiwAFevXlUcNxEREVF5UZTs9enTB0IIrFixAocPHy6x/OHDh7F8+XJIkoQ+ffooaRoA8Omnn6Jfv36KDucuXboU+fn5AIDly5fD1tZWY7qdnR2WL18O4MnIIMuWLStzW0RERETlTVGyN23aNDg5OSEvLw+9e/fGm2++iTNnzqCwsFAuU1hYiDNnzuCNN95A7969kZeXBycnJ0ybNk1p7IoJIbBz504AgJeXFzp06KC1XIcOHdCkSRMAwI4dO+R7ChIRERE97RQle+7u7ti0aRMsLS2Rn5+PlStXws/PD/b29qhVqxZq164Ne3t7+Pn54ccff0R+fj6srKywefNmVKlSxVjLUGbXr19HfHw8ACAwMFBvWdX0O3fu4MaNG6YOjYiIiMgoFA+X1rNnT5w4cQI+Pj7yEGm5ubm4e/cuEhISkJubK7/v6+uLkydPonv37saIXbHY2Fj5uZeXl96y6tPV5yMiIiJ6mikeLg0AWrdujVOnTiE6OhqHDh3ChQsXkJaWBgBwc3ND8+bN0b17d/j6+hqjOaO5ffu2/Lx27dp6y3p6emqdj4iIiOhpZpRkT8XX1/epS+j0yczMlJ87ODjoLWtvby8/V90+Rpvc3Fzk5ubKrzMyMhRESERERKSM4sO4z7JHjx7Jz62srPSWVb8v4MOHD3WW+/LLL+Hs7Cw/1HsEiYiIiMqbomTvhx9+wMWLF40VS7lT3RQaAB4/fqy3rHpvXdHbs6ibOXMm0tPT5QcP+RIREVFFUnQY980334QkSXB3d0eXLl0QFBSEwMBANG/e3FjxmZSjo6P8XN+hWUBzlBB9h3ytra2NPjoIERERUVkpPmdPCIHk5GRs27YN27ZtA/Dkogz15K9ly5aKAzUF9Ysy7ty5o7eseg8dD80SERHRs0JRspeWloaIiAiEh4cjPDwc58+fR2FhIVJTU7Fjxw7s2LEDAODq6oqAgAA5+WvdurURQleuadOm8vNLly7pLas+3dvb22QxERERERmTomTPxcUFAwcOxMCBAwEADx48wJEjR+Tk7++//0ZhYSHS0tKwc+dOeQxdZ2dndOnSRU4GK0q9evXg4eGBhIQERERE6C175MgRAECtWrVQt27dcoiOiIiISDmjXo3r4uKCAQMG4Ouvv8bZs2eRmpqKnTt3Yvr06WjTpg0kSYIQAg8ePMDu3buN2XSZSJIkJ6qXLl3CiRMntJY7ceKE3LM3cOBASJJUbjESERERKWHSW684OTmhVq1aqFWrFjw8PGBnZ/fUJUrTpk2DhcWTDs4pU6YUu63Kw4cPMWXKFACAhYXFUzGmLxEREZGhjHpTZSEEYmJi5MO4R48elW8qLIQA8OS2JR07dkTXrl0Vt3fs2DHExcXJr1NSUuTncXFxWLNmjUb50aNHF6ujcePGmDFjBr766iucPn0a/v7++OCDD9CgQQNcvXoVCxYsQExMDADgvffeQ6NGjRTHTURERFReFCV7RZO7Y8eOIT09XZ4G/F9yFxQUhKCgILRv3x6WlpbKIwfw888/Y+3atVqnHT9+HMePH9d4T1uyBwDz589HUlISfvnlF8TExGD48OHFyowbNw6ff/654piJiIiIypOiZK9KlSrlmtyZipmZGVatWoXg4GCEhoYiOjoaKSkpcHd3h6+vLyZNmoTevXtXdJhEREREpaYo2Xvw4IF8Dt7AgQMxffp0dOzYsdySuzVr1hQ7VKtEnz590KdPH6PVR0RERFTRjHJTZQDYtWsXwsPDNW6mrLoCl4iIiIgqhqJk7+jRo/L5elFRUUhPT8fu3buxZ88eAE+uxlXdTDkoKIjJHxEREVE5U5Ts+fv7w9/fHx999BHy8/Nx6tQpOfmLjIxEeno69uzZg7179wIonvy1bdvWKAtBRERERNoZ7dYrFhYW6NSpEzp16oRZs2YhPz8f0dHRxZK/vXv3Yu/evZAkCfn5+cZqnoiIiIi0MOp99jQqtrBAx44d0apVK7Rt2xYtWrTAqlWrkJGRIZ/nR0RERESmZfRkLycnB8eOHZN79M6cOSP34KkneXZ2dsZumoiIiIiKUJzsGZrc2dvby/ffCwwMhJ+fn9KmiYiIiKgEii/QOH36tNbkzsHBAZ06dZKTO19fX3kMWiIiIiIqH4qyr6ioKPm5o6Mj/P395eTOx8cH5ubmigMkIiIiorJTlOz16dNHTu7atWsHMzMzY8VFREREREagKNlT3TyZiIiIiJ5O7IojIiIiqsSY7BERERFVYkz2iIiIiCoxJntERERElRiTPSIiIqJKjMkeERERUSXGZI+IiIioEmOyR0RERFSJGZTsDRkyBMHBwbhz546p4yEiIiIiIzIo2duxYwd27NiBjIwMzZnNzGBhYYGLFy+aJDgiIiIiUqZUh3GFEAa9R0RERERPB4OSPUdHRwDAvXv3TBoMERERERmXQcmel5cXAOCbb75BVlZWsemSJBk3KiIiIiIyCgtDCr366quIjo7Gnj174ObmhurVq8PS0lKe3rNnT43XhpAkCVevXi1dtERERERUKgYle1OmTMHx48exZcsW5OfnIz4+Xp4mhNB4bSj2BhIRERGZnkHJnpmZGTZt2oSoqCgcOnQI8fHxyM3Nxdq1ayFJEgYMGAAXFxcTh0pEREREpWVQsqfSsWNHdOzYUX69du1aAMD8+fPRtGlT40ZGRERERIpxBA0iIiKiSqxUPXtFFRYWGisOIiIiIjIB9uwRERERVWKKevbUFRQUYMeOHTh06BAuXLiAtLQ0AICbmxuaN2+O7t27Y9CgQTA3NzdWk0RERERUAqMke3/++ScmTpxY7JYswJNbrERGRiI0NBS1a9dGaGgoevXqZYxmiYiIiKgEig/j/vrrr+jXrx/i4+MhhIAQAnXq1EGHDh3Qvn171KlTB8CT5O/27dvo27cvNmzYoDhwIiIiIiqZomTv5s2bmDhxIgoLC2FnZ4fPP/8ciYmJuHbtGiIjIxEVFYVr164hMTER8+fPh4ODAwoLCzFhwgTcunXLWMtARERERDooSva++eYb5ObmwsHBAUePHsWsWbNQrVq1YuWqVq2KmTNn4ujRo3BwcEBubi6++eYbJU0TERERkQEUJXsHDhyAJEl477330Lp16xLLt2rVCjNmzIAQAvv371fSNBEREREZQFGypzoU2717d4Pn6dGjh8a8RERERGQ6ipK9goICACjV7VRUZXlDZiIiIiLTU5Ts1apVCwAQGRlp8Dyqsh4eHkqaJiIiIiIDKEr2unbtCiEEvvrqKyQkJJRY/s6dO/jqq68gSRK6deumpGkiIiIiMoCiZG/KlCkwMzNDcnIy2rdvj82bN8uHdtUVFBRg06ZN6NixI5KSkmBmZoa33npLSdNEREREZABFI2g0b94cn332GT766CMkJCRg+PDhcHFxQZs2bVC9enVIkoTExETExMTgwYMH8qgan332GZo3b26UBSAiIiIi3RQPlzZz5kw4Ozvj/fffR05ODu7fv4+wsDCNMqokz87ODosWLcLrr7+utFkiIiIiMoBRxsZ944038Morr2D16tU4dOgQLly4gLS0NACAm5sbmjdvju7du2PMmDFwd3c3RpNEREREZACjJHsA4O7ujvfeew/vvfeesaokIiIiIoUUXaBBRERERE83JntERERElRiTPSIiIqJKjMkeERERUSXGZI+IiIioEmOyR0RERFSJMdkjIiIiqsSY7BERERFVYkz2iIiIiCoxRSNozJs3DwDQvn179OrVyygBEREREZHxKEr25s6dC0mSsH37dmPFQ0RERERGpOgwbpUqVQAAL7zwglGCISIiIiLjUpTsNWzYEACQmJholGCIiIiIyLgUJXvDhg2DEAKbNm0yVjxEREREZESKkr033ngDrVq1wrp167B27VpjxURERERERqLoAo3ExET8/PPPGDduHMaOHYsNGzbg1VdfRcuWLeHq6gpzc3O98/NcPyIiIiLTUpTs1a1bF5IkAQCEEDh8+DAOHz5s0LySJCE/P19J80QaHGrUQGFBAcxK+JNRFqaq19R1ExERKUr2gCdJnrbnROXNxsUFZubm2DZyJJJjY41Wb6PevdFt/nyj1wsAVb29MWTDBqPWSUREpE5Rsrd69WpjxUFkNMmxsUiMiTFafe5eXiapl4iIqDwoSvZCQkKMFQcRERERmQDHxiUiIiKqxJjsEREREVViii/QUCksLER4eDiioqKQmJiInJwcfP7556hZs6Zc5vHjx8jPz4e5uTmsra2N1TQRERER6WCUZG/v3r2YOnUqbty4ofH+u+++q5HsrVq1Cm+99RYcHByQkJAAe3t7YzRPRERERDooPoz7888/Y8CAAbh+/TqEEKhSpYrOW7CMGzcOLi4uyMrKwvbt25U2TUREREQlUJTsxcXF4c033wQAdOvWDRcvXkRSUpLO8lZWVggODoYQAgcOHFDStNFIkmTQIygoqKJDJSIiIio1RcnesmXLkJeXh2bNmmHfvn3w+v/3I9MnICAAAPC///1PSdNEREREZABF5+wdPnwYkiRh2rRpsLKyMmieBg0aAABu3bqlpGmje/311/HGG2/onM7zC4mIiOhZpCjZu337NgCgdevWBs+jSppycnKUNG101apVQ/PmzSs6DCIiIiKjUnQYV5IkAKUbEzc5ORkA4OTkpKRpIiIiIjKAomTPw8MDAHDlyhWD54mIiAAA1K1bV0nTRERERGQARclely5dIITAb7/9ZlD5lJQU/Pjjj5AkCd26dVPSNBEREREZQFGyN3HiRADAvn37sHr1ar1l79y5gz59+iAlJQXm5ubyvE+LzZs3o0mTJrC1tYWjoyMaNWqEkJAQhIWFVXRoRERERGWmKNnz9fXF5MmTIYTA+PHj8fLLL2PTpk3y9L///ht//PEHxo0bhyZNmuDMmTOQJAnvvvsuGjZsqDh4Y7p48SKuXLmCR48eISsrC3FxcVi3bh26deuGwYMHIz09vaJDJCIiIio1xcOlLV++HNnZ2fj111+xbds2bNu2Tb5wY+TIkXI51UUco0ePxhdffKG0WaOxs7PDgAED8OKLL8LLywsODg5ITk5GREQEVq5cidTUVOzYsQMDBw7EwYMHYWlpqbe+3Nxc5Obmyq8zMjJMvQhEREREOikeLs3c3Bxr167F5s2b0aZNGwghtD6aNm2K3377Db/88oucDD4N4uPjsXHjRowfPx6dO3dG69at0aNHD3z++ef4559/0KZNGwBPLiz54YcfSqzvyy+/hLOzs/zw9PQ09SIQERER6aS4Z08lODgYwcHBSEhIwOnTp5GUlISCggJUqVIFbdq0kW+m/LRxcXHROa169erYsmULvL298fjxYyxfvhxTp07VW9/MmTPxzjvvyK8zMjKY8BEREVGFMVqyp+Lh4YEBAwYYu9oKU79+ffTo0QN79+5FXFwcEhIS5FvOaGNtbQ1ra+tyjJCIiIhIN8WHcZ8HTZs2lZ/Hx8dXYCREREREpWPUnr28vDycPXsWFy5cQFpaGgDAzc0NzZs3R9u2bUu8uOFpVZoRQoiIiIieJkZJ9rKysvDZZ59h1apVuH//vtYyrq6uGDduHD7++GM4Ojoao9lyc/HiRfm5vkO4RERERE8bxYdxY2Nj0axZMyxevBhpaWk6r8ZNS0vD4sWL0aJFC1y+fNkYsZeLa9eu4eDBgwCenL9Xq1atCo6IiIiIyHCKkr0HDx6ge/fuuH37NoQQaN68ORYtWoSIiAhcunQJsbGxiIiIkJM8IQRu3bqF7t27PxU3Kd69ezfy8/N1Tr937x6GDh2KvLw8AMCbb75ZXqERERERGYWiw7gLFizA3bt3IUkS5s2bh1mzZhW7h16TJk0QEBCA6dOn48svv8THH3+MhIQELFiwoMJvrjxlyhTk5eUhODgYHTt2RN26dWFra4uUlBSEh4fLN1UGgM6dOzPZIyIiomeOomRvx44dkCQJw4YNw0cffaS3rCRJmDVrFs6fP48//vgD27dvr/BkDwASEhKwfPlyLF++XGeZ4OBg/Pzzz7ylChERET1zFCV7N2/eBACEhIQYPM/o0aPxxx9/yPNWpLVr1yIiIgJRUVG4du0aUlJSkJGRAQcHB3h6eqJTp04ICQlBx44dKzpUIiIiojJRlOw5OjoiNzcX1apVM3geVVkHBwclTRtFYGAgAgMDKzoMIiIiIpNRdIFGixYtAAD//vuvwfOoyqrmJSIiIiLTUZTsTZo0CUIILFu2DIWFhSWWLywsxNKlSyFJEiZOnKikaSIiIiIygKJk7+WXX8aYMWNw4sQJDBo0CImJiTrL3rt3D0OGDMHJkycxevRoDBs2TEnTRERERGQAg87ZW7dunc5pgYGBuHDhAvbs2YP69eujZ8+e8PX1RbVq1SBJEu7du4fo6GgcOHAAubm58PX1RWBgINatW4dRo0YZbUGIiIiIqDiDkr3Ro0cXu39eUZIk4dGjR9i9ezd2795dbLoQApIk4fTp0xgzZgwkSWKyR0RERGRiBl+NK4RQXM7QOoiIiIjIOAxK9q5fv27qOIiIiIjIBAxK9urUqWPqOIieSw41aqCwoABm5uYmqf9ZrZuIiIxH0U2ViUgZGxcXmJmbY9vIkUiOjTVq3Y1690a3+fNNUndVb28M2bDBqHUSEZFpMNkjegokx8YiMSbGqHW6e3mZrG4iInp2KLrPHhERERE93YyS7MXGxmL69Onw8fGBm5sbLC0tYW5urvdhYcFORSIiIiJTU5xxffXVV5g9ezYKCgp4axUiIiKip4yiZG/z5s2YNWsWAMDMzAwBAQFo1aoVXFxcYGbGI8REREREFU1RsvfNN98AAGrVqoV9+/ahRYsWRgmKiIiIiIxDUffb33//DUmS8NlnnzHRIyIiInoKKUr2LC0tAQCtW7c2RixEREREZGSKkr3GjRsDAFJTU40SDBEREREZl6JkLyQkBEII7Nixw0jhEBEREZExKUr2xo0bh4CAAISGhmL37t3GiomIiIiIjETR1biWlpbYuXMnQkJCMHjwYAwfPhyvvPIKGjduDDs7uxLnf+GFF5Q0T0REREQlUHxTZRcXF0ydOhUnTpzAxo0bsXHjRoPmkyQJ+fn5SpsnIiIiIj0U3/l42rRp6NmzJ1JSUiCEKNWDiIiIiExLUc/e+vXr8e233wIAnJycMHjwYLRs2ZIjaBARERE9JRQle8uXLwcAeHl5ISwsDNWrVzdKUERERERkHIq63y5dugRJkjB37lwmekRERERPIaOMoKG6uTIRkTEUFhQ8U/USET3NFB3G9fLyQlRUFBITE40VDxERzMzNsW3kSCTHxhqtzqre3hiyYYPR6iMielYoSvbGjBmDyMhIbNy4ES+99JKxYiIiQnJsLBJjYio6DCKiZ57iETT69euH9evXY8WKFcaKiYiIiIiMRFHP3pEjRzB16lQkJyfj7bffxm+//Ybhw4cbPIJGly5dlDRPRERERCVQlOwFBQVBkiT59cmTJ3Hy5EmD5uUIGkRERESmp3i4NI6EQURERPT0UpTshYWFGSsOIiIiIjIBRcleYGCgseIgIiIiIhPgALZERERElRiTPSIiIqJKjMkeERERUSWm6Jy9bt26lXleSZJw+PBhJc0TERERUQkUJXvh4eGQJEnv7VfU78MH/N+tWoq+T0RERETGpyjZ69KlS4lJW3Z2Nv7991+kp6dDkiQ0btwYNWvWVNIsERERERlIcc+eIYQQ2Lt3L95++22kpaXh559/RufOnZU0TUREREQGKJcLNCRJQr9+/XDs2DGYm5tj8ODBiI+PL4+miYiIiJ5r5Xo1bs2aNfHOO+8gNTUVCxcuLM+miYiIiJ5L5X7rFdXh271795Z300RkJA41aqCwoKCiwyAiIgMoOmevLKysrAAACQkJ5d00ERmJjYsLzMzNsW3kSCTHxhq17ka9e6Pb/PlGrZOI6HlW7snesWPHAAB2dnbl3TQRGVlybCwSY2KMWqe7l5dR6yMiet6V62HcqKgozJs3D5Ikwc/PrzybJiIiInouKerZmzdvXollCgsLcf/+fZw+fRonT55EYWEhJEnC9OnTlTRNRERERAZQlOzNnTu3VCNhCCFgYWGBhQsXokePHkqaJiIiIiIDKD5nT99QacCTe+w5OjqiXr16CAwMxMSJE9G0aVOlzRIRERGRARQle4WFhcaKg4iIiIhMoNzvs0dERERE5YfJHhEREVElxmSPiIiIqBIz6k2VMzMzcf36dWRmZqLAgKGUunTpYszmiYiIiKgIoyR7P/30E77//nv8/fffBs8jSRLy8/ON0TwRERER6aAo2SsoKEBwcDB2794NoOTbsBARERFR+VKU7K1cuRK7du0CAFSvXh1jxoxBu3bt4ObmBjMzng5IREREVNEUJXvr1q0DADRt2hRHjx6Fq6urUYIiIiIiIuNQ1P0WGxsLSZLwySefMNEjIiIiegoZ5VhrkyZNjFENERERERmZomSvUaNGAIC0tDSjBENERERExqUo2Rs+fDiEENizZ4+x4iEiMgmHGjVQaMD9P8vqWa2bqKLwO1N+FF2gMXXqVGzYsAE//PADBg8ejICAAGPFRURkVDYuLjAzN8e2kSORHBtr1Lob9e6NbvPnm6Tuqt7eGLJhg1HrJHoamOr7yO9McYqSPWtraxw4cABDhgxBjx49MHXqVLz66qvw8vKCjY2NsWIkIjKa5NhYJMbEGLVOdy8vk9VNVJnxO1M+FCV75ubm8nMhBJYsWYIlS5YYNC9H0CAiIiIyPUXJXtERMziCBhEREdHTRVGyN2fOHGPFQUREREQmwGSPiIiIqBLjALZERERElRiTPSIiIqJKjMmemlu3bmHGjBnw9vaGvb093Nzc4Ofnh8WLFyMnJ6eiwyMiIiIqNUXn7FUme/fuxciRI5Geni6/l5OTg+joaERHR+Pnn3/Gvn37UL9+/QqMkoiIiKh02LMH4Ny5c3jllVeQnp4OBwcHzJ8/H5GRkTh8+DAmTJgAALh8+TL69u2LrKysCo6WiIiIyHDs2QMwbdo05OTkwMLCAgcOHEDHjh3lad26dUOjRo3w/vvv49KlS/j6668xe/bsCoyWiIiIyHDPfc9edHQ0wsPDAQDjxo3TSPRU3n33XXh7ewMAli1bhry8vPIMkYiIiKjMnvtkb8eOHfLzMWPGaC1jZmaGUaNGAQDu378vJ4dERERET7vnPtk7evQoAMDe3h7t2rXTWS4wMFB+fuzYMZPHRURERGQMz32yFxsbCwBo2LAhLCx0n8Lo5eVVbB4iIiKip91znew9evQIKSkpAIDatWvrLevq6gp7e3sAwO3bt00eGxEREZExPNdX42ZmZsrPHRwcSixvb2+P7Oxsvbdfyc3NRW5urvxadd++jIwMBZHqialhQzgXFBi1TisPD2RkZLBuE9fLusu37mcxZuDJd9xU+w+iivasfWdU9QohTFK/yYjn2K1btwQAAUC89tprJZb39PQUAESDBg10lpkzZ45cJx988MEHH3zwUfket2/fNmY6YnLPdc+ejY2N/Pzx48clllf12Nna2uosM3PmTLzzzjvy68LCQqSlpaFKlSqQJElBtJoyMjLg6emJ27dvw8nJyWj10rOB2//5xW3//OK2r3hCCGRmZsLDw6OiQymV5zrZc3R0lJ8bMjJGdnY2AP2HfK2trWFtba3xnouLS9kCNICTkxO/9M8xbv/nF7f984vbvmI5OztXdAil9lxfoGFjYwN3d3cAwJ07d/SWvX//vpzseXp6mjw2IiIiImN4rpM9APLIGHFxccjPz9dZ7tKlS8XmISIiInraPffJXufOnQE8OUR75swZneUiIiLk5/7+/iaPqyTW1taYM2dOsUPG9Hzg9n9+cds/v7jtqawkIZ6164eN69SpU2jfvj0AYNKkSVi5cmWxMoWFhWjevDliY2Ph4uKCpKQkWFpalneoRERERKX23Pfs+fn5ISAgAACwatUqREVFFSuzZMkSedSMt99+m4keERERPTOe+549AIiJiYG/vz8ePnwIBwcHzJo1C127dsXDhw/x+++/IzQ0FADQuHFjnD59WuMqXiIiIqKnGZO9/2/37t34z3/+o/Ou240bN8bevXvRsGHDco6MiIiIqOye+8O4Kv3798fff/+N6dOno3HjxrCzs4OLiwt8fHywYMECxMTEPBWJ3q1btzBjxgx4e3vD3t4ebm5u8PPzw+LFi5GTk1PR4VEpSZJk0CMoKKjEuv78808MGTIEtWvXhrW1NWrXro0hQ4bgzz//NP2CUDFJSUnYs2cPZs+ejd69e8Pd3V3enqNHjy51fcbYvjk5OVi0aBH8/Pzg5uYGBwcHeHt7Y8aMGbh161apYyLtjLHt16xZY/D+Yc2aNSXWx23/nKvA0TuolPbs2SOcnZ11Dt/SpEkTcfXq1YoOk0pB17Ys+ggMDNRZR2FhoZg4caLe+SdOnCgKCwvLb8FI7/YICQkxuB5jbd+4uDjRpEkTnXU4OzuLvXv3KlxqEsI423716tUG7x9Wr16tty5ue3quR9B4lpw7dw6vvPIKcnJy4ODggJkzZ2qcV/jTTz/h8uXL6Nu3L6Kjo/WO8kFPn9dffx1vvPGGzun29vY6p3388cfyeaVt2rTB+++/jwYNGuDq1atYuHAhYmJiEBoaiqpVq+Lzzz83euxUMk9PT3h7e+PAgQOlntcY2zcrKwv9+vXD5cuXAQATJkzA8OHDYWtri7CwMHz55ZdIT0/Hyy+/jKioKLRs2bLsC0salGx7lf379+sdnqt27do6p3HbEwD27D0rgoKCBABhYWEhIiMji01fuHCh/C/t008/rYAIqSxU22zOnDllmv/ff/8VFhYWAoDw8fEROTk5GtOzs7OFj4+P/NmJi4szQtRkiNmzZ4vdu3eLxMREIYQQ169fL3XvjrG275w5c+S2Fy5cWGx6ZGSk3E7Xrl1Lt6BUjDG2vXrP3vXr18scC7c9CSEEk71nwKlTp+Qv66RJk7SWKSgoEN7e3gKAcHV1FY8fPy7nKKkslCZ7b7zxhlxHVFSU1jJRUVFymbfeektBtKREWX7wjbF9Hz9+LFxcXAQA4e3tLQoKCrTWM2nSJLme06dPG7xcVLKKSva47UmFF2g8A3bs2CE/HzNmjNYyZmZmGDVqFIAn4/iGh4eXQ2RUkYQQ2LlzJwDAy8sLHTp00FquQ4cOaNKkCYAnnyXBC/CfCcbavuHh4Xjw4AEAICQkBGZm2nf76hcObNu2TWH09DTgticVJnvPgKNHjwJ4ct5Wu3btdJYLDAyUnx87dszkcVHFun79OuLj4wFobnttVNPv3LmDGzdumDo0MgJjbV/V/qOkenx8fORzQ7n/qBy47UmFyd4zQDV6R8OGDWFhofuaGi8vr2Lz0LNh8+bNaNKkCWxtbeHo6IhGjRohJCQEYWFhOudR38bq214bfjaePcbavobWY2FhgQYNGmitgyrW6NGjUb16dVhZWcHd3R0dOnTAxx9/LP8Z0IXbnlSY7D3lHj16hJSUFAD6r7gCAFdXV/nf2e3bt00eGxnPxYsXceXKFTx69AhZWVmIi4vDunXr0K1bNwwePBjp6enF5lHfxiV9Njw9PbXOR08vY21f1Wt7e3u4uLgYVE9ycjJyc3NLEy6ZUEREBJKSkpCXl4fU1FScPHkS8+fPR8OGDfHjjz/qnI/bnlR465WnXGZmpvzckNup2NvbIzs7G1lZWaYMi4zEzs4OAwYMwIsvvggvLy84ODggOTkZERERWLlyJVJTU7Fjxw4MHDgQBw8e1BiXuTSfDfVbt/Cz8Www1vZV1WPo/kO9Hmtra4NiJdOoX78+hgwZgo4dO8rJ2LVr17B161Zs2bIFjx49wuTJkyFJEiZOnFhsfm57UmGy95R79OiR/NzKyqrE8qov6MOHD00WExlPfHy81n/cPXr0wJQpU9C7d2/ExMQgIiICP/zwA6ZOnSqXKc1nQ33Hzc/Gs8FY21dVT2n2H9rqofI1ePBghISEQJIkjfd9fX0xbNgw7NmzB0OGDEFeXh6mT5+OAQMGoEaNGhplue1JhYdxn3I2Njby88ePH5dYXtX9bmtra7KYyHj0HVqpXr06tmzZIu+oly9frjG9NJ8N9cMy/Gw8G4y1fVX1lGb/oa0eKl/Ozs7FEj11/fr1w5w5cwA8GQpt1apVxcpw25MKk72nnKOjo/zckMNv2dnZAAzrtqenX/369dGjRw8AQFxcHBISEuRppflsqD4XAD8bzwpjbV9VPaXZf2irh54+EyZMkBPCiIiIYtO57UmFyd5TzsbGBu7u7gCe3FZBn/v378tfWPUTtunZ1rRpU/m5+tV36iftl/TZUD9pn5+NZ4Oxtq+qnuzsbPmeayXVU7VqVZ6z9QyoVq2a/Pug7cpcbntSYbL3DPD29gbwpGcnPz9fZ7lLly4Vm4eefbpugqyeBKpve2342Xj2GGv7GlpPfn4+rl69qrUOenrpu0k6tz2pMNl7BnTu3BnAk39nZ86c0VlOvRvf39/f5HFR+bh48aL8XH0w9Hr16smvtR3CUXfkyBEAQK1atVC3bl3jB0lGZ6ztq9p/lFTP6dOn5SMD3H88G5KSkpCamgpAc9+gwm1PKkz2ngGDBg2Sn69evVprmcLCQqxbtw7Ak5P+u3btWh6hkYldu3YNBw8eBPDk/L1atWrJ0yRJwsCBAwE8+dd+4sQJrXWcOHFC/lc/cOBAvSd909PDWNs3KCgIzs7OAIC1a9fq7Alas2aN/Hzw4MFKw6dyEBoaKm9PbSNkcNuTrMJG5aVSCQgIEACEhYWFiIyMLDZ94cKF8kDWc+bMKf8AqdR27dol8vLydE5PTEwUbdq0kbfrkiVLipW5fPmysLCwEACEj4+PyMnJ0Ziek5MjfHx85M/OlStXjL4cZJjr16/L2zIkJMSgeYy1fT/55BO57YULFxabHhkZKbcTGBhY2kWjEpR221+/fl2cPXtWb5ndu3cLKysrAUDY2NiIO3fuaC3HbU9CCCEJwVHRnwUxMTHw9/fHw4cP4eDggFmzZqFr1654+PAhfv/9d4SGhgIAGjdujNOnT2tcyUdPp7p16yIvLw/BwcHo2LEj6tatC1tbW6SkpCA8PFy+qTLw5HDMoUOHtJ44PXPmTHz11VcAgDZt2uCDDz5AgwYNcPXqVSxYsAAxMTFyuS+++KL8FvA5d+zYMcTFxcmvU1JS8N577wF4cqhs/PjxGuXVB6NXZ4ztm5mZCR8fH1y5cgUAMHHiRAwfPhy2trYICwvDF198gaysLNja2iIyMhKtW7dWsujPPaXbPjw8HF27dkXHjh3Rv39/tG7dGtWqVYMQAteuXcOWLVuwZcsWuaduxYoVePPNN7XGwm1PANiz9yzZtWuXcHJykv+lFX00btxY/PvvvxUdJhmoTp06Orel+iM4OFjcv39fZz0FBQVi7NixeusYN26cKCgoKL+FIxESEmLQ9lU9dDHW9v33339Fo0aNdNbh5OQkdu/ebezV8FxSuu3DwsIMms/Ozk78+OOPJcbDbU/s2XvG3Lx5E9988w327t2LO3fuwMrKCg0bNsTLL7+Mt956C3Z2dhUdIhkoIiICERERiIqKwrVr15CSkoKMjAw4ODjA09MTnTp1QkhICDp27GhQffv27UNoaCiio6ORkpICd3d3+Pr6YtKkSejdu7eJl4aKGj16NNauXWtw+ZJ2xcbYvtnZ2fjuu++wefNmxMXF4fHjx/D09ESfPn3w9ttvo06dOgbHS7op3faZmZnYtWsXoqKicPr0ady9excpKSnIz8+Hq6srmjVrhhdffBHjx49HtWrVDGqD2/75xmSPiIiIqBLj1bhERERElRiTPSIiIqJKjMkeERERUSXGZI+IiIioEmOyR0RERFSJMdkjIiIiqsSY7BERERFVYkz2iIiIiCoxJntERERElRiTPSIiIqJKjMkeERERUSXGZI+IcOPGDUiSBEmSsGbNmooO56lVUFCAb775Bn5+fnBycpLX2aBBgyo6NCIinSwqOgAiomfFiBEjsHnz5ooOg4ioVJjsEREZIDIyUk70+vbti2nTpqF69eqQJAlOTk4VHB0RkW5M9oiIDHDo0CEAgLm5OX777TcmeET0zOA5e0REBoiPjwcAVK9enYkeET1TmOwRERkgNzcXAGBpaVnBkRARlQ6TPSI1c+fOla+wBIBHjx5h0aJFaNu2LRwdHeHo6Ag/Pz+sWLEC+fn5OutR1TF37ly97QUFBUGSJAQFBRWbFh4eLtcTHh4OIQRWrVqFzp07o0qVKnBycoKfnx9+/fVXjfkeP36MlStXokOHDnBzc4OjoyP8/f2xadOmUq2LzZs3o3v37qhWrRpsbW3h5eWFDz/8EPfv3zdo/lOnTmHChAlo3LgxHBwcYG9vDy8vL7z55pv4999/dc63Zs0aeblv3LiB3NxcLFu2DB06dIC7u7tB61WX8+fPY+LEiWjUqBHs7Ozg6OiIZs2aYfr06bhx44bWeVSxrF27FgBw8+ZN+T31z4qhin7GHjx4gDlz5qBZs2ZwcHCAm5sbgoKCsGHDBr311K1bF5IkYfTo0QCAM2fOYPTo0ahXrx6sra21xlWW5S8qOTkZ8+bNg7+/P6pVqwZra2t4enrC398f8+bNw+XLl3XOm5+fj1WrVqFPnz7w8PCAtbU13N3d0aVLFyxbtgyPHj3S2/aZM2cwbtw4NG7cGPb29rCxsYGnpyfatWuHN998E7t27YIQoth8jx49wrfffougoCC4u7vD0tISbm5u8PLyQp8+fbB06VK9yy+EwJYtWxAcHAxPT0/Y2NjA1dUVfn5++Oyzz/DgwQO9cV+5cgVTpkxB8+bN4eDgACsrK3h4eKB169YYO3Ys/vjjD/nPBJFJCCKSzZkzRwAQAERiYqJo1aqV/Lroo3///qKgoEBrPaoyc+bM0dteYGCgACACAwOLTQsLC5PrOXDggOjfv7/OWKZOnSqEECItLU106dJFZ7n58+drjeP69etymdWrV4uxY8fqrKNmzZrin3/+0blMeXl54vXXX9c5PwBhaWkpQkNDtc6/evVquVx0dLRo3bp1sflLWq/afPHFF8LMzExnTNbW1mLt2rXF5tO3HKpHaah/xq5duyYaNGigs96hQ4eKvLw8rfXUqVNHABAhISHihx9+EBYWFnrjKuvyq1u/fr2wt7fXuy7q1Kmjdd64uDjRtGlTvfM2atRIXLlyRev8X3/9td74VY/MzEyN+RISEkpsF4B49913tbablJQk/P399c5bvXp1ceLECa3zb9q0SVhZWZXY/vnz5/WueyIlmOwRqVH/Ie7UqZOwsrISU6dOFQcPHhRnzpwRv/32m/D29pbLrFy5Ums9xk722rdvLwCIkSNHir1794ozZ86IjRs3iiZNmshlDh48KAYMGCAsLCzE66+/Lg4cOCDOnDkjVq1aJTw8PAQAYW5uLi5cuFCsLfVkz9fXVwAQfn5+YuPGjeL06dNi3759YtiwYXKZ2rVri/T0dK3LNGrUKLlc7969xfr168WpU6dEdHS0+Omnn0SzZs3k6bt27So2v3qy17JlSyFJkhg1apS83Nu3bxf79u3Tu16L+u677+Q6q1atKhYvXiyioqLEsWPHxNy5c+UERpIksXfvXo15z58/L86fPy8GDhwoAAgPDw/5PdWjNNQ/Y76+vsLMzExMnjxZHDp0SERHR4tVq1aJxo0by2WmTJmitR5Vste0aVNhbm4u6tatK1asWCEv15dffmmU5VdZu3atXIeNjY2YMmWK2Ldvnzh79qw4cuSIWLFihejVq5eoX79+sXkTEhJE9erVBQDh6Ogo3n33XfHf//5XnD17VoSFhYmZM2cKOzs7AUDUr19fPHjwQGP+c+fOyYlevXr1xJIlS8Thw4dFTEyMOHr0qPjll1/Ea6+9JhwcHIole8HBwXLc//nPf8S2bdvEiRMnRHR0tNizZ4/49NNPRZs2bbQme1lZWfL33crKSkyaNEns3LlTnD17Vhw9elTMnz9fVKlSRQAQrq6u4saNGxrzJyYmyuu2WrVqYt68eeLAgQPi7NmzIjIyUqxfv15MnDhRuLu7M9kjk2KyR6RG/YfY0tJShIWFFSuTmpoq/3C1bNlSaz3GTvYAiGXLlhUrc/fuXeHk5CT/iEuSJLZv316snPqPpaoXUJ16sgdA9OnTR2uP0rx58+QyM2bMKDZ9y5Yt8vSffvpJ6zI/fPhQdOvWTQAQdevWLdaOerIHQKxatUprPYZKSkqSEwkPDw9x69atYmXOnj0r/yjXqlVLPH78uFiZkJAQvT1XhlL/jAEQv/32W7EyGRkZcq+ymZmZ+Pvvv4uVUSV7AESLFi3E/fv3tbZnjOWPj4+X66hWrZrexOT27dvF3uvXr58AIDw9PcXVq1e1zqcew8cff6wx7ZNPPhEAhL29vUhMTNTZ9oMHDzR62x8+fCgsLS319typpKamFnvvrbfeEgCEs7OziI6O1jrfjRs3RM2aNeVkUt2qVasM6rl7+PChyMnJ0RsfkRJM9ojUqP8Qv/POOzrLffjhh3K5or0QQpimZ08XVRICQAwfPlxnOdXh3TZt2hSbpp7sWVtbi/j4eK11FBQUiObNm8s9GY8ePdKY3q5dOwFADB48WGccQghx8eJFjR5JderJXrdu3fTWY4gFCxbI9W3cuFFnuc8//1wut2nTpmLTTZHs9evXT2e5kydPyuXeeOONYtPVk70jR47orMcYy6/+edf2Z0Kf8+fPy/Pu3LlTb9n3339fTkrVTZgwQednV5/4+HiD2y4qOTlZ2NjYCADim2++0Vv2+++/l/8gZmdny+/Pnz9f/q4QVSReoEGkw8iRI3VOa9eunfz8+vXrJo9l+PDhOqe1bNlSfj5s2DCd5Vq1agUAuHbtmt62evbsCQ8PD63TzMzMEBISAgC4f/8+zp49K0+Lj4/HmTNnAACvvPKK3ja8vb3h7u4OAIiKitJZTt82MJTq/nguLi4IDg7WWW78+PHF5jG1MWPG6Jzm5+eHZs2alRiPp6cnAgICdE43xvLv3bsXAFCvXj0MHDhQZx3a7Ny5EwBgZ2eHvn376i3bpUsXAEBCQgJu374tv1+zZk0AwMWLF3Hq1CmD265SpQqsrKwAAL/++qvei6qK2r9/v3zBSEmfZ1XceXl58ndAPe779+/L64GoIjDZI9LBy8tL5zQ3Nzf5eWZmpsljady4sc5pLi4upSpXUry+vr56p/v5+cnPL1y4ID8/ffq0/HzEiBHFrlgt+khJSQEAJCYm6mxLPZEtK1WMbdq00XvblOrVq6Nu3boa85iaoev633//xePHj7WWKWkdKV3+vLw8+XVAQECprz5WfS5ycnJgYWGh9zPRr18/eT71z8WIESNgaWmJ3Nxc+Pv7o3///li5ciX++ecfrVffqlhbW8t/gLZs2YKGDRvi/fffx759+5Cenm5Q3MCTpE1f3M2bN9ca94ABA+Tv3eDBg9GtWzcsXboUZ86cQUFBgQFrj8g4mOwR6WBnZ6dzmpnZ/311ymOnbWgshpQrLCzU21a1atX0Tq9evbr8PC0tTX6elJSkdz5dcnJydE5zdXUtU53qVDGqx61LjRo1NOYxNUPXtRBC5y1vSlpHSpc/LS1NTqhUPVWlYYzPhZeXFzZu3AhXV1fk5+djz549eP3119G8eXNUq1YNr732Go4ePaq1nhUrVqB///4Antw2Z9GiRejbty+qVKkCPz8/LF68GBkZGSaJu0qVKti1axdq1aoFIQTCwsLwzjvvwMfHB25ubggODsaePXvK1A5RaXC4NCLSUFLPja6eFPWkd8OGDQb3yulLVszNzQ2qwxCG9Ejp6yUyhbKua3WGriNjLH9pe/WA//tc1KtXD7t27TJ4vnr16mm8Dg4ORvfu3fHHH39g//79OHr0KJKTk5GSkoL169dj/fr1CAkJwS+//KLxB8jJyQm7du3CqVOnsGnTJoSFheHcuXMoKChAdHQ0oqOjsWjRIuzYsQMdO3YsFreVlZXGodmS1K5dW+N1QEAA4uLisHXrVuzbtw9HjhzBnTt3kJGRgW3btmHbtm3o1asXtm3bpvfPGpESTPaITECSJAghSuxFy87OLqeIDHfv3j2909V7PNQPZ1epUkV+XvTQVkVyc3PD3bt39R4uVlEtu/pymdK9e/fg6empc7pqXUuSVOZeTqXL7+bmBjMzMxQWFiIhIaHU7as+F/fu3YOXlxcsLMr+s+Ps7IyJEydi4sSJAJ6cw7dr1y4sX74cCQkJWLt2Ldq0aYO333672Lx+fn7yYfHMzEyEh4dj9erV2L59O5KSkhAcHIyrV6/C1tZWI+7Hjx+jSpUqZerVVLGxscHIkSPlc1CvXbuGvXv3YsWKFbhy5Qr279+Pjz76CEuXLi1zG0T68DAukQk4OjoCgN7RJgoLC/WOJFFRoqOjDZ6untC1adNGfn7gwAHjB1ZGqhhjYmKQl5ens1xSUhJu3rypMY+pGbquGzVqJF9oUFpKl9/S0lJ+ffTo0VL3fqo+Fzk5OTh+/Hip5i1J06ZN8eGHH+LEiROwt7cHAINGinF0dET//v2xbds2TJ06FQBw9+5dHDt2rFjcgPE/z/Xr18eUKVMQHR0t9wSWdoQbotJgskdkAqpDUOoneRdlyEniFeHAgQO4e/eu1mmFhYXysGGurq5o27atPK1hw4Zo2rQpAOD333/HrVu3TB+sAbp37w7gybBkW7du1Vlu1apVciKjmsfUVOtSm9OnT8sXRiiJxxjLrzrn7fr166W+qlT96t2FCxeWal5DeXp6yhcnqS78MdSLL74oP1eft3fv3vIFLUuXLi3VlbyGcnJyki/SKW3cRKXBZI/IBAIDAwEAJ0+e1NqbcffuXblH4WmTm5uLSZMmab3w5KuvvsL58+cBAGPHjoW1tbXG9I8//hjAk7FIhwwZguTkZL3tfP/99yWOh6rUmDFj5HOh3n33XY1beqicO3cOX3zxBQCgVq1aGDRokEljUtm1a5fWHp2srCz5UKWZmRkmTZpU5jaMsfxvvfWW3HM2adIkvVcr37lzR+O1r68vevbsCeDJH5w5c+bojffGjRvYuHGjxns7duzQO/7s7du3cenSJQCa5/pdu3YNERERettT77VTn7dWrVryrXHOnTuHSZMm6U34kpKS8PPPP2u8t3//fp1/nAAgPT1dvpVM0XMUiYyJ5+wRmcDEiRPx/fffIz8/H/3798fs2bPRuXNnPH78GMePH8eSJUuQn5+PRo0aPXWHcn18fLB79274+/tj+vTpaNSoEZKSkrB27Vr8/vvvAJ6chP7JJ58Um3fEiBHYv38/1q5dizNnzqBp06aYNGkSAgMDUbVqVWRnZ+Pq1as4evQotm3bhrS0NIwaNcqky1O1alUsWrQIb775JhISEuDj44MPP/wQnTp1QkFBAQ4dOoRFixYhKysLkiQhNDRU7y1KjMnHxwevvvoqIiIiMHToUDg5OeHvv//GggULcPnyZQDAm2++qegWNMZY/ho1auCHH37AqFGjkJSUBD8/P0yYMAG9e/dGjRo1kJWVhQsXLmDXrl24fPkyrl69qjH/6tWr4ePjg7t372LevHnYv38/xo4dixYtWsDGxgapqan4+++/8eeff+Kvv/7CoEGDMGLECHn+ZcuWYeTIkejbty+6desGb29vODs74/79+zh9+jSWL1+Ohw8fAgBef/11eb5bt26ha9euaNq0KQYPHgwfHx/UqlULwJME8Y8//pCT7TZt2qB9+/YacS9ZsgSRkZG4cOECfvnlF5w4cQITJ05Eu3bt4ODggAcPHuCff/7BoUOHsG/fPrRo0ULjfoUbN25E//790aNHD/Ts2RPNmzeHm5sbMjMzceHCBaxYsQLx8fHF4iYyuoq5lzPR00l9dAN91Ee30DakmhBPBm5XlSn6cHV1FREREQaPoKGrDSE0R5y4fv16mZZNfQSN1atXi9GjR+uMvWbNmuKff/7R2U5+fr54//33hbm5uc46VA97e/tiw0QZujylNX/+fHnIOG0Pa2trsXbtWp3zm2IEjWvXrol69erpjCk4OFjrsHVC/N8IGiEhIQa1q3T5hRBizZo1wtbWVu821bV+bty4IY+7XNJjzJgxGvOqvif6Hubm5uKLL77QmK/okIO6Ht7e3jo/a6mpqeKll14yqJ6uXbtqzKs+uo2+x5tvvqkxzBuRsbFnj8hEpk+fjqZNm2Lp0qU4deoUcnJy4OHhgT59+uD999/HCy+8UNEh6rR69Wr07NkToaGhOH/+PLKyslCnTh0MGjQIH374YYm3S1mwYAHGjRuH0NBQ/PXXX7hx4wYyMjJgZ2eHF154Aa1bt0bPnj0xePBg+epHU5s1axb69euHFStW4K+//kJCQgLMzMzwwgsvoGfPnpg2bZp8U+HyUq9ePZw5cwaLFy/G9u3bcfPmTVhaWqJVq1aYOHGiUUYQUTHG8oeEhKBnz5747rvv8Oeff+Lq1avIyclBjRo18MILL6B3794aPXLq6tSpg5MnT2Lnzp34448/cPLkSdy7dw95eXlwcXFBo0aN0LFjRwwYMKDYiCCbNm3CoUOHcPDgQfzvf/9DYmIiUlJSYGNjg7p166JLly6YPHkyWrRooTFfQEAAoqKicPDgQYSHh+PWrVu4d+8eHj16BDc3N7Rq1QrBwcEYPXq0zgtg3Nzc8N///hd//fUX1q9fj2PHjuHu3bt49OgRnJyc0KBBA/j5+aFv377y4WqVZcuWYcCAATh48CBOnz6Nu3fvIjk5Gebm5vD09ESnTp0wfvx4+Pv7l7D1iJSRhCjnG0sRET3H5s6di08//RRA+d/Xj4ieT7xAg4iIiKgSY7JHREREVIkx2SMiIiKqxJjsEREREVViTPaIiIiIKjFejUtERERUibFnj4iIiKgSY7JHREREVIkx2SMiIiKqxJjsEREREVViTPaIiIiIKjEme0RERESVGJM9IiIiokqMyR4RERFRJcZkj4iIiKgS+3+Jg2VKvyBoMwAAAABJRU5ErkJggg==",
      "text/plain": [
       "<Figure size 640x480 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "import csv\n",
    "import matplotlib.pyplot as plt\n",
    "\n",
    "counts = []\n",
    "with open('./results/nodes_in_pipelines.csv', newline='') as csvfile:\n",
    "    reader = csv.reader(csvfile)\n",
    "    next(reader)\n",
    "    for row in reader:\n",
    "        try:\n",
    "            count = int(row[2])  # csv-file: pipeline, operators, processes, channels, total_nodes\n",
    "            counts.append(count)\n",
    "        except ValueError:\n",
    "            print(f\"Skipping row with non-numeric value in the specified column: {row}\")\n",
    "\n",
    "plt.rc('font', size=20)\n",
    "plt.hist(counts, bins=20,color='maroon', edgecolor='white')\n",
    "plt.title('Number of processes in workflows')\n",
    "plt.xlabel('number of processes')\n",
    "plt.ylabel('number of workflows')\n",
    "plt.tight_layout()\n",
    "plt.savefig(f'{directory_plots}/number_of_processes.png', format=\"png\", bbox_inches='tight')\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAmsAAAG4CAYAAAAaDqD4AAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjcuMiwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy8pXeV/AAAACXBIWXMAAA9hAAAPYQGoP6dpAABwY0lEQVR4nO3dd3gUxf8H8Pem90JCAoHQCQk99BaSIEVAalBBlKqIShVQigqKIFVQQBDpClFAihTpnQAJEJoEAaUJhBbSISG5+f3B7/Z7l9xdLuxdciTv1/Pcw5KdnZmt97nZ3RlJCCFARERERBbJqrArQERERET6MVgjIiIismAM1oiIiIgsGIM1IiIiIgvGYI2IiIjIgjFYIyIiIrJgDNaIiIiILBiDNSIiIiILxmCNiIiIyIIxWCtm+vXrB0mSIEkSVqxYUdjVKZISEhLw1VdfoXHjxvD09IS1tTW3ORUbFSpUkI/369evF3Z16P9NmjRJ3i+TJk0yWb5///03PvroI9SoUQOurq5yGZr7f8WKFfLf+vXrZ7Kyi5OXNlgLCwvTOijKlSuHjIwMo5YdO3YsDxwyi+vXr6Nu3bqYOHEioqOjkZiYCJVKVdjVIiIyua1bt6Ju3br44YcfcPHiRaSmphZ2lYosm8KugKncunULP/74I4YNG1bYVaFi7P3338etW7cAAI6OjmjdujXKlCkDa2trAEBQUFBhVo9ewPXr11GxYkUAQPny5dlaRAQgLS0Nffv2xdOnTwEApUuXRosWLVCyZElIkgQAcHNzK8wqFilFJlgDgG+++QbvvvsunJycCrsqVAzFx8dj165dAAB7e3ucPXsWVatWLeRaERGZ3pYtW5CQkAAAqFGjBmJiYuDo6FjItSq6XtrboLrEx8dj/vz5hV0NKqZOnz4tT4eEhDBQo2Lp+vXrEEJACIEKFSoUdnXITDSvd7169WKgZmZFIlhr3LixPD19+nQkJycXYm2ouHr8+LE8Xbp06UKsCRGRefF6V7CKRLD2zjvvoFq1agCev4k3Z86cQq4RFUfPnj2Tp62sisSpRUSkE693BatIbGFra2utV5G//fZb+V76izpw4ID8xmhYWJhRy2i+nZqfNKdOncJ7772HgIAAODs7w83NDSEhIVi1ahWEELny2Lt3LyIiIlC5cmU4ODigVKlS6NatGw4fPpzv9QSeB7gzZsxAo0aNULJkSTg6OqJSpUp49913cerUqXznFxMTg5EjR6Ju3booWbIk7OzsUKpUKYSGhmL69Olav8j00fX6/z///IMJEyYgODgYJUuWhJWVFerWrZvv+uWUmpqK77//Hu3atUPZsmXh4OAAT09P1KxZE0OGDMGJEyf0Lqt5nPTv31/++8qVK7X2talel3/48CGmTZuG0NBQlC5dGvb29vD29kZwcDDGjBmDixcv5pnH9evX5Tpp3qbav38/3nrrLVSuXBmOjo7w9vZGixYt8P3338sPERsrLS0NCxcuRKdOnVC+fHk4OTnB1dUVVatWxYABA7Bv374889D1un92djZ+/fVXdOnSBZUqVYKjoyMkScKmTZu0ln3y5Ak2bdqEYcOGoUWLFvD19YWdnR1cXFxQoUIFdO/eHcuWLUNmZmae5atfLgCAGzdu5NqveZ3zQgisW7cOvXr1QuXKleHi4gIXFxdUrlwZb731FtavX6/zPM9J8w34AwcOAADu3r2LqVOnolGjRihVqhSsra3h4eGRa9lbt27hyy+/RMuWLeHr6wt7e3u4urqiYsWKaNSoEfr374/IyEg8fPgwz3rkxZiuO3StS0JCAqZPn46GDRvC29tbvg4NHDgQFy5cUFwvAJg2bZpc7ueff6433e7du7X2raG0+/fvl9O1adPGYPk3b97EF198gSZNmsjHpK+vL5o0aYKJEyfKLycZou+7afv27ejVqxeqVq0KFxcXSJKEuXPn5pmfLqdPn4avr69cTteuXfH06VOt7j9Wrlwpp+/fv3+u80G9X1+E0nPm+PHjcj1CQkL0pktPT4ednZ1RaZ8+fQoHBwdIkgRnZ2etYFWz3ps2bcJbb72FatWqwc3NTd7HNWrUQJs2bfD1118jJiYm/70EiJdUaGioACAAiIULFwqVSiVq1aol/+3TTz/Vu+ynn34qp+vbt6/ONPv375fThIaGGlUndXpDmzVnmilTpghra2utv2t++vTpI1QqlRBCiNTUVNGpUye9aSVJEt9++63BOvbt21dOv3z5cnHs2DFRpkwZvXlaWVmJcePGGbX+CQkJIiIiQm9e6o+Hh4dYt26dwbzKly8vp7927Zr48ccfhYODQ6686tSpY1Td9NmyZYsoVapUnnV+6623RFpaWq7lNY+TvD4TJ05UVNelS5cKd3d3g2VYW1uLESNGiKysLL35XLt2TU5fvnx5kZmZKQYPHmww34CAAPHXX38ZVc+1a9catU1fe+01kZiYqDef5cuXa52nt2/fFiEhITrz2rhxo7zc8ePHhYuLi1H7pEKFCuL06dN5lm/MR5fLly+L4ODgPJetX7++uHr1qsHtqnnN279/v9i0aZPw9PTMlZe7u7vWcosWLRKOjo5GrUPz5s0N1sEYOc9dY9blyJEjBq9D1tbWYvHixYrrduzYMaPWdfz48UZvly+++EJO9/XXX+tNN2XKFJ3XMM2Pg4OD+OabbwyuQ87vpsTERNGtWzed+c2ZM0debuLEiUZdi/bv3y/c3NzktP3795evJ5p55PXZv3+/nGfOc9kQU5wzz549k68Btra2Oq/dQgixa9curTwNpd23b5+crnXr1rnmx8fHi6ZNmxq9fXbv3m1wO+RUZN4GlSQJkydPRteuXQEA8+bNw8iRI+Hr61u4FTNg4cKFmDBhAgCgfv36qFWrFoQQOHz4MP79918AwKpVq1CtWjWMHTsWERER2LlzJ+zs7NCiRQtUqlQJycnJ2Lt3Lx49egQhBD7++GM0bNgQLVq0yLP8Gzdu4OOPP8bjx4/h7OyMVq1awdfXF/Hx8di/fz/S0tKgUqnwzTffICsrCzNmzNCbV3x8PFq1aoW4uDj5b0FBQahbty5cXV1x//59HDlyBA8fPkRiYiLeeOMN/Pzzz+jdu3ee9Vy3bh0++eQTAECZMmXQrFkzuLu7486dO4paUH/77Tf07t0b2dnZAJ630LZo0QJVqlRBamoqDh8+jDt37gAA1qxZg2vXrmHfvn1wcHCQ8yhTpgw++ugjAMClS5ewd+9eAEBgYCBeeeUVrfIaNWr0wnWdNWsWxowZI//f3t4eoaGhKFeuHB4/foz9+/cjISEB2dnZmDt3Lm7cuIHff//dYIuP2qeffopFixYBAGrWrIng4GBIkoTTp0/LLRqXL19Gq1atEBUVhUqVKunNa86cORg1apT8q9fV1RVNmzaFv78/srOzcfHiRcTExEAIga1btyI0NBRRUVF5vsGdkZGBzp0749SpU7CxsUGzZs1QpUoVPH36VOtBZ+D5szTq/p58fHxQo0YNlC1bFs7OzkhPT8fVq1cRHR2NrKwsXL9+HaGhoTh9+jSqVKmilU9QUBA++ugjpKSkYNWqVfL69OnTJ89tCgBxcXEIDQ3FgwcP5L/VqlULdevWhSRJiI2Nxfnz5wE8b11v3rw5Dh06hICAgDzzjoqKwqRJk/Ds2TN4eXmhZcuW8Pb2xv379xEbGyun27RpEwYPHiz/383NDU2bNkXZsmVhY2ODpKQkXL58GRcuXDDYymhOFy5cwLhx45CamgofHx+EhITAy8sLt2/fxr59+/DkyRNkZ2dj8ODBqFmzJpo2bfrCZTVo0AAuLi5ITU1FdHQ00tLS4OzsnCtdzlYhY9PquwszZMgQLFiwQP6/+npbqlQp+XqbmpqKp0+fYty4cbh3755Rj/MIIfD2229j69atkCQJDRs2RFBQEIQQuHDhglHnv6YNGzbgrbfekvssHT16NGbOnCnPb9SokXy927t3Ly5dugQAeOWVVxAYGKiVV5kyZfJVNmC6c8bGxgbNmzfHzp078ezZMxw9elRnq2fO/Wxs2pz7OTs7Gx07dtS6E1WzZk3UrFkTHh4eePr0KeLj43H27FncvXvX2M2hLV+hnQXJ2bKm1rBhQ/nvw4cP17mspbSs2dnZiTJlyoijR49qpcnKyhKjR4+W03l4eIhJkyYJACIsLEzcuHFDK31iYqIIDw+X04eFhektX7Nlzc7OTgDPW41ytnAkJiaKnj17atV33759OvPMzs7WKr9evXoiJiYmV7onT56ISZMmCUmSBADh7Ows/v33X515av46t7GxEXZ2dmLx4sVyK6Pa06dP9a6rIVevXtVqfWnQoIG4fPlyrvWaPXu2sLKyktMNHTpUb575+fWYH1FRUVqtr6+++qq4e/euVpqnT5+KMWPGaO2v2bNn68xPs2XN1tZWABBeXl7izz//zJV2+/btWq03YWFhufaB2p49e+RtZWtrK77++muRmpqaK11sbKyoXr26nOcHH3ygMz/N7WljYyOfi7paazSPg+PHj4vx48eL8+fP68xXCCHu3bsn3nnnHTn/V155RW/anC2RxsjIyBB16tSRlytZsqTYuXNnrnQ7d+4U3t7eWudOZmamzjw1r3k2NjZCkiQxefLkXOk1t4VmHYYMGaK31SAlJUWsXbvW4B0JY+W3Zc3e3l5YW1uL2bNni2fPnmmlu3nzpqhZs6acNjw8XHH92rVrJ+ena5+kpaXJ50XJkiUNpn3y5Imwt7cXAISTk5POfffbb79pnZd9+vQRSUlJWmmSkpLE22+/rZVu/fr1Ouuv+d2kPi9q1aolzp07lyut5rGQV8vaTz/9pHWdmT59us7y1XLepTHEmGujqc+Zb775Rk6j7+5Qs2bNcu1nfWk1W/WPHDmiNW/jxo3yvNKlS4vjx4/r3RYXLlwQn376qThx4oTeNLoUuWBtx44dWheBW7du5VrWUoI1R0dHcfHiRZ3psrOzRVBQkFb6GjVqiCdPnuhMf/PmTfnElSRJ3LlzR2c6zRMMgOjQoYPIzs7WWwfNC1vDhg11plu1apWcpm7duiIlJUXv+guhfdtg8ODBOtNoXvABiF9++cVgnvnVp08fOe9KlSqJx48f60377bffymmtrKz0BpjmCtZatmwp59ukSRORkZGhN+2wYcPktG5ubiI5OTlXGs3gQ71OOX8waDp06JAcYAMQ27Zty5UmOztbVK1a1ej9dffuXeHj4yMHdrrO05y3IWvVqiXS09MN5ptf7du3l/PXdy6+SLC2bNkyrYD45MmTetNGR0fL5y4AsXLlSp3pNK95gOFbbkI8D8DUaf39/fUG2aaW32ANgPjxxx/15nf+/Hn5+DN0bTPWtGnTDH4xa94amzlzpsG0e/fulee3adMm1/zs7GxRsWJFOU1ERITe/aBSqUSXLl3ktJUrV9Z5bc756EWpUqXEgwcP8lxvQ8Ha1KlT5XnW1tZi6dKleeZn6mDN1OfM8ePH5flNmzbNNV8zKB8wYIAIDAzUmzY9Pd1gUD5q1Ci5rJ9++sngtnhRReIFA03t2rWTHxLMyMjA5MmTC7lG+g0ePFhvj/ZWVlZ4/fXXtf72zTffaN2C0+Tv749mzZoBAIQQRr0YIEkSvv/+e71v8lhZWeH777+Xm9JjYmJw9uzZXOm+/fZbeXrevHlwcXExWO64cePkh6AjIyPzfNCyUaNGRt0uNVZiYiJ+++03+f8zZ87U+VC22vDhw1GjRg0AgEqlwuLFi01Wl7zExcXh0KFD8v8XLFgAOzs7vemnTp0Kb29vAEBycjLWrFmTZxlvv/22fOzoEhISgrfeekv+/08//ZQrzZYtW3DlyhUAz2+J5LW/SpUqhZEjRwJ4futh7dq1edZz+vTpJu/LSXO4uT179pgs3x9//FGeHjx4MOrXr683bcOGDfHee+/J/1+4cGGe+ZcpUwaffvqpwTSaXRh5eXnl+5ZYQalVqxYGDRqkd37NmjXRsGFDAMZf2wzRvIW1f//+XPM1//bmm2/K1+i80uq6Bbpr1y5cu3YNAGBnZ4d58+bp3Q+SJGHBggWwtbUF8PyFqt27d+e5Pl988YV8zueXEAIjR47E+PHjATx/vGL9+vUYMGDAC+WnhKnPmfr168PV1RXA8++unENhHTlyRH5JICwsDOHh4XrTRkVFybeGmzdvLu8jNc1zrWTJkvpXUoEiF6wBwNdffy1PL1++XH7+y9JEREQYnF+zZk152tHREe3btzc6vfoCYUjz5s1RuXJlg2kCAgK0nhHJecG6e/cuzpw5A+D5F4gxz8o5ODjIeSYlJeX5plfPnj3zzDM/NE88Ly8vdO7c2WB6KysrrYuXrou2uWiWVadOHdSrV89gemdnZ/Tq1Uvn8voY8wxW37595ekDBw7kehNr+/bt8rSx+6tVq1by9JEjRwym9fT0RLt27YzKV1N6ejr27duH7777Dp999hmGDx+OIUOGyJ/IyEg5rfo4Vio1NRUnT56U/2/MF9+7774rT8fExCAtLc1g+oiICNjYGH7kWP1mNwCcP39eK+i3JDl/lOoSHBwsTysd7kvzS/zkyZO5vpjVzyZVrlwZ/v7+chBmKC2gO1jTfOu5ffv2efZHVqZMGbz66qvy//M6fyVJwptvvmkwjT5ZWVno06eP/Maom5sbduzYIT/3XZDMcc6on1sDnq9rzmuM5r4LDw+X919eaXXt53LlysnTP/74I7KysvKsf34VmRcMNLVs2RJt2rTB7t278ezZM3z55ZdarxlbCnVrjT6enp7ydEBAQJ4XZ830xnQM3KRJkzzTAEDTpk0RFRUFAFoPLwPAsWPH5GkhBIYMGWJUnv/88488fevWLdSuXVtvWkO/sF6E5jo0btw4z+0KQD7p1csLIQqkpUKzrpp1MKR58+aYN28eAOR6+D4nSZK0OpXWp3HjxpAkCUIIJCYmao2XCWgfB9u2bTMq8ElKSpKn8+qyoG7duvnqyykhIQFffPEFVq1ahZSUFKOWMUW3FQBw9uxZ+aUVFxcXg8e2Wt26deHs7Iy0tDRkZ2fj7NmzBls7jTknbG1t0a1bN6xZswbZ2dlo3bo1evTogR49eiA0NBReXl7Gr5QZ1apVK880mnXVPG5ehPpLfMeOHcjKysLhw4flH8JpaWly0KBuaQkPD8fChQtzpX3y5Amio6MBPP+RpG790/Si5++WLVsA5H3+VqhQASVKlDAqX03p6eno0qWL/CPLx8cHO3bs0AqKC5K5zpmwsDDs2LEDwPPAVzMQVgdgVapUQdmyZXO1uOoLmnUFaz169MDEiROhUqnw559/onr16ujfvz/at2+P2rVrm6QfuiIZrAHPW9fUTci//PILxo4da3GDaLu7uxucrxlE5JU2Z3pdfcDkpPlrwBB/f395WvMtHQDy25Lqac03noyVV79rpm5W1lyH8uXLG7WMZn9kmZmZSElJKZBBipXWNa8AxNPTM8/b1sDzX93u7u5ITEyU66UZrGkeBzn7PDOGKY+BGzduoGXLlrh582a+6mBsUJcXzX3m7+9v1IXaysoK/v7+8tt1ee03Y7fHnDlzcPr0aVy6dAnPnj1DZGQkIiMjIUkSAgMDERISgrZt26Jjx456H7EwN2OubZq3nYy5tuVF80v8wIEDcgB29OhRrVtjmv/qSqt+g7ZZs2a5bo0B5j9/X/TaOGfOHLn1x9/fH3v37i3U4fHMdc7k3HdqmkG5Oo2Pjw+qV6+OixcvaqVNT09HTEwMAP1BeWBgIGbPno2PP/4YQghcuXIF48ePx/jx4+U3sMPCwtCtWze5A//8KpK3QYHnzzmpb2+pVCpMnDixkGuUW35aZszRimPsgPear6vn/EJT+isXQJ5NxqZ+TknzVoauV/F1yZnOVF/seVFa17zqaewxkFe+So8DUx4DvXv3lgM1Nzc3jBo1Cjt37sS1a9eQmpqK7OxseexKzV/M+e6kUo8X2Wc50+a134zdHj4+PoiJicGkSZPg5+cn/10Igbi4OCxevBg9evSAn58fpk2bJrduFKTCeJZO3WoGaLea5Lw1BjwPiKpXr54rrea0Zn6azH3+vui1UTOwTEhIePHuJEzEXOdMgwYN5Fvep06dku84aQblmvtOHbjlTKsOynU9r6Y2YsQIHDp0CO3atdMKNpOTk7Fz506MGzdO7tJJ3f1IfhTZYA0AJk+eLF8I1q9fb7JnUnQx1YW+IKWnpxuVTvNZAPWBr6Z5snTt2lX+EszPR/Mh74Kg2ZKU17NB+tLl3A7morSuedXT2GMgr3w1j4MzZ87k+xhQ+hySWlRUFI4ePSrX8cSJE5g1axbatm2LChUqwNnZWetCao6g+0X2Wc60pjy+XFxc5N7xY2JiMHv2bHTt2lXrofTHjx9j3LhxiIiIMGo0hZed5nNrp0+flr+Y1QFYQECAVnCr/kLXTGtM/2rmPn9f1IgRI9CpUye5vA4dOrzwCDimYK5zRt13JvC8LzT1Ouq7ranez5ppjdnPai1atMCOHTtw7949rF+/HsOHD0e9evW0rjn79u1D48aN5euUsYp0sFa7dm354VUhhMEhQ3LSjJ6NeVjQFC1MBc3Y20Sa6XK+daTZ6bD6bUBLp3nrwNhtcOPGDXnazs6uwII1pXXN6y2xx48fGxWwJCcnax3jlnocqDslBp6/6Zmzo86cNLeVqWjus1u3bhkV/KhUKq3n9l707T5DrKys0KBBA3z88cfYuHEj7t27hyNHjmg9UL5582b8/vvvJi/b0uT8Ej906BDS0tLkN01ztpSpv6Q102reGmvQoIHOcsx9/r4oOzs7rF+/Xitga9++faEFbOY8Z3S9/asOwKpVq6YVlKuHQdNMm9fzarp4e3sjIiICc+fOxalTp3Dv3j18//33ch2fPHmC999/36i81Ip0sAYAX331FaytrQEAW7duNTjOoybNL+NHjx7lmf5FmjULm+ZD4YYcP35cns75NqLmSwp//fUX/vvvP9NUzow0H6KNjo426taP5q8gdQ//BUGzruqXPPKiWde83h4VQhh1Tpw4cUK+gHp4eGg9rwZA6yWFnTt3GlVPc9B8di6vF3gAGPWGZH73dZ06deRrTmpqqlHXhrNnz8qtBNbW1qhTp06+ynwRVlZWaN68OTZs2KD1pu0ff/xh9rItQc7nmXJ25ZAzrfo4OHDggNZtNEO3xsx9/iqhDthee+01AP9rYcvrzWxzMOc5k3M/63peTc3b21u+buRMq+95NWN4e3tj6NChWufWX3/9la+eKop8sFatWjW8/fbb8v8/++wzo5ZTD0YMAFevXs31ynZOxvQTZWmioqK03srU5fLly1pBXc5fnBUrVtR6ceNFBw4uSM2aNYO9vT2A5w+2btu2zWB6IYTW28SaXU6Ym2ZZsbGxOvu50/TkyRP8+uuvOpfX5+eff84zzYoVK+RpzS8uNfUFH3g+NNf9+/fzzNMcNG835HWL986dO0YFJpoP3hvzcLuLi4tWS4vmttNn+fLl8nSjRo3y9dyOUpIkoWPHjvL/7927V2BlF6acLS6Gbndpfonv37/f6NYWzfNv+/bteZ4X8fHx8osPOZc3Bzs7O/z+++/y+Zuamor27dsXeMBmznOmfv368stgsbGx2Lp1q3y3TNezhur9qU6rGZQb03OAIU2bNtV6ezc/51qRD9YAYOLEifIvnz179uDgwYN5LuPm5ia/tZGVlYXVq1frTRsbG6uzo1BLJ4TAsGHD9D5vp1KpMGzYMLlFpUGDBjp/vWh2zvndd9/lq3PR+Pj4fNZaOQ8PD62+icaMGWPwVuCCBQvkIMnKykqrM0ZzCwwMRMuWLeX/Dx061GDA8Pnnn8tfCG5ublqd2erzyy+/GPzVf/jwYa3+yDT7N1KLiIiQx9ZMT0/H22+/bfRbe6mpqfl6TsUQzXFLN2/erDdddnY2Bg0aJPe3Z4iHh4ccBN6/f9+o9dK8xbFgwQKcO3dOb9rY2Fh5bFYAWmN5KpGSkmL0eJ+at+jM1amnpdH8Ej9z5ox8vAQFBaFUqVK50qu/2DXTav5dl7Zt28qt0BkZGRgxYoTetOrrsXqfVa5cGa1bt87fSr0AfQFbfp+pUspc54zmLW+VSqXVD6uuQFu9P3OmNbSfje32R3PcYiB/51qxCNYqVqyIgQMHyv/XvK1niOYX3dixY3X+2vjzzz/Rtm1bi+0d3BA7Ozts374dffr0yfXMXVJSEt5++22tW1rffPONznzefvtt+RdgVlYWOnbsiOnTp+v9Ak5NTUVkZCRatWqFoUOHmmht8mfixInyQ62XL19Gu3btcjVJq1QqzJs3T+sC+9FHH+W6BWhu06ZNk28RHD58GBEREbl+oWdmZmLChAmYPXu2/DfNddTH1tYWKpUKnTt31nn7cseOHejSpYscsLds2RIdOnTIlc7a2hoLFy6U67l79260bNlSfq5Hl3PnzmHcuHEoV66cUZ04G6Njx47yuXjw4EGMHj0aT5480UoTHx+PiIgIbNu2zagWLHt7e3mg6KysLGzcuDHPZXr37i3/sMnMzES7du10dnC6d+9evPrqq3IAWK9ePa1OjZU4deoUypcvj4kTJ+Kvv/7SmSY7OxurV6+W++UDoHP/FkU5v8Tj4uIA6G8pU/9dM23OFqGcrKysMG3aNPn/kZGReO+993LdqUlJScHAgQOxbt06+W/Tp083Sf9cxlAHbOoW1tTUVLz66qsFGrCZ85zR3KfqTtiDgoK0nrVVCw0Nla8hmh22G2pBfeONN9CxY0esW7dO7/fezZs30bNnTzkYr1q1qvwD1xhFtp+1nD777DOsWLECT58+NXqZYcOGYdGiRbhz5w4SExPRsmVLNG/eHIGBgXj69ClOnjwp9/GyfPly9O/f31zVN4vx48fju+++w+rVq7F582a0atUKvr6+uHfvHvbt26d1Qfn444/1/sqztrbG2rVr0aZNG8TGxiIzMxNjx47FV199hSZNmqBcuXKws7PD48ePcfnyZVy8eFE+0fIaxcFcKlWqhCVLlqB3797Izs7GsWPHUK1aNYSEhKBy5cpIS0vD4cOHtZ7Ba9KkCWbMmFHgdW3atCmmTZuGMWPGAHg+tFO5cuUQHh4Of39/PH78GAcOHND6ddetWzd5OCdD/Pz80L17d8yZMwevvvoqateuLT9nc/r0aa1nR3x8fLBs2TK9P0xat26NhQsX4oMPPkB2djaOHz+ORo0aoWrVqggODoanpyeePHmC+Ph4nDlzxiy3SgMDA/HOO+9g1apVAIDZs2djzZo1aNiwIXx8fHD9+nUcOnQImZmZcHV1xcyZM41qyYqIiMCUKVMAPP9xsnLlSlSpUkXrWaVZs2bJ03Z2doiMjERoaCgePHiA+Ph4tGrVCnXq1EHdunUBPG+h0byt7ePjg8jISL3PP72I+Ph4fPXVV/jqq6/g6+uL4OBglCpVCjY2NoiPj8epU6e0um0ICQkx+YghliwsLExr9A1AfwuK+ktc8+F3Y26NvfHGGzh06JDcB+WSJUvw22+/ITw8HL6+vrh//z727dun1bo/YsSIAr822tnZYcOGDejevTu2bdsmt7Dt2LHDYAfNpizfXOeMoRa0nLy8vFCrVi2tlr28gnKVSoXt27dj+/btsLW1Rc2aNREQEAB3d3ekpKTgxo0bOH78uHwXy9raGt9//73BOudi+uFGC4a+gdwNGTFihNYAuDBiwO1Tp04Jb2/vXMupP3Z2dmLBggVCiPwP5J6X/A4mb2igXrWcg+9GRUUJPz8/vetnZWUlxowZY9Qg0Onp6WLw4MFaA+wa+jg6OoqpU6fqzMuYwaBNYcuWLcLX1zfPuvbq1UukpaUZzMtcA7mrLVmyRLi5uRmsp7W1tRg+fLjIysrSm0/OgckzMzPFe++9ZzDfKlWqiPPnzxtVz3379mkN6p7Xp0aNGuL27du58nnR7ZmWlibatm1rsMyyZcuKI0eOGH2OJSUlierVqxvMU5e///5bBAcH57kN6tWrJ65evWpwvTSvefv3789zOxw/ftzocxGA6NGjh0hOTs4z37zkdyB3Y9bFmGvbi4iOjtbaBpIkifv37+tNX7t2ba3033zzjdFlTZ48WR4MXN/HwcFB7zVRLb/fC2rGbsOnT5+Kjh07ymldXV3F0aNHc6Uz9UDuaqY8Z9SysrJyXTvXrVunN/2wYcO00rZr185g/q+99prR55mPj4/YtGmTUfXWVGxa1oDnA4j/9NNP+Xo+pl69erh06RK+/fZbbNmyBdeuXYNKpULZsmXRpk0bfPjhh3KHiS+jpk2b4uzZs1i8eDE2bNiA69evIzU1FaVLl0Z4eDg+/PBDg78oNDk6OmLhwoX49NNP8csvv2Dfvn24fPkyHj16BJVKBXd3d1SqVAl16tTBK6+8gldffbVARgEw5LXXXsPVq1exbNkybN26FX/99RcePnwIR0dH+Pn5ITw8HH369DFqSCZzGzhwILp06YKffvoJf/75Jy5fvoyEhAS4urrC398frVu3xoABA/J9PNra2mLx4sV4/fXXsXTpUkRHR+Pu3btwcnJCYGAgXn/9dQwePNjoHu7Dw8Nx6dIlbNy4Edu2bcPx48cRHx+P5ORkODk5wdfXF4GBgWjWrBnat28v/2o2FScnJ/z5559Ys2YNVq5cidjYWCQnJ8Pb2xuVKlVCREQE+vXrB09PT62Hyg1xc3NDdHQ0Fi5ciC1btiAuLg6JiYl5Pr8WEBCAkydPYv369fj9998RHR0ttyj6+PigcePG6NGjByIiIkz+KEXjxo1x//597NmzB0eOHEFsbCz++ecfPHr0CNnZ2XBzc0PlypXRpEkTvP3222jUqJFJy38Z1KtXD+7u7vJjINWrVzf4HFF4eLhWi4uxXTkAz+/uvPPOO1iyZIncSXNiYiI8PDxQqVIltGvXDu+++67RI8uYi729PX7//Xf5UYGUlBS8+uqr2Llzp9Y40eZijnPG2toaISEh8stkkiQhNDRUb/rw8HCtlq+89vMff/yB2NhY7N27FydOnEBcXBz+++8/pKWlwd7eHiVLlkTt2rXRoUMHvPXWWy/0vScJUQx6QCQimea4nuXLlzdZh7RERGQexeIFAyIiIqKXFYM1IiIiIgvGYI2IiIjIgjFYIyIiIrJgDNaIiIiILBiDNSIiIiILxq47zEylUuHOnTtwdXV9KYekIiIisnRCCKSkpMDPz6/AhukqSMWqU9zCcOfOHfj7+xd2NYiIiIq8W7duoWzZsoVdDZNjsGZmrq6uAJ4fQIXdWz8REVFRlJycDH9/f/k7t6hhsGZm6lufbm5uDNaIiIjMqKg+blT0buwSERERFSEM1oiIiIgsGIM1IiIiIgvGYI2IiIjIgjFYIyIiIrJgDNaIiIiILBiDNSIiIiILxmCNiIiIyIIxWCMiIiKyYAzWiIiIiCyY2YebysjIwJEjR/Dw4UNUrFgRjRo1MneRREREREWGomDtxo0bWLBgAQBg/Pjx8PDw0Jp//Phx9OjRA3fv3pX/Vq9ePfz+++8oV66ckqKJiIiIigVFt0E3btyIWbNmYd++fbkCtZSUFHTt2hV3796FEEL+nDp1Ch07dkRWVpaSoos9VXb2S5Pvy1RXIiIiS6OoZW337t2QJAldu3bNNW/x4sW4f/8+JEnCsGHD8Morr2Dnzp344YcfcPHiRaxcuRIDBw5UUnyxZmVtjQ29e+NBXJzJ8qzavj1aTZli0nzNkScAlAwKQvfVq02WHxERkaVSFKz9+++/AID69evnmrd27VpIkoRu3bph7ty5AIBOnTrhwYMHWLduHdavX89gTaEHcXGIj401WX7egYEmz9cceRIRERUnim6D3r9/HwDg6+ur9ffk5GScPn0aANC/f3+teT179gQAnD17VknRRERERMWComAtJSUFAJCd49mho0ePIjs7G9bW1ggLC9Oa5+/vDwBISEhQUjQRERFRsaAoWHN3dwcA3LlzR+vvBw4cAADUqVMHzs7OOpd1cHBQUjQRERFRsaAoWKtZsyaA52+FqmVnZ8vPq4WHh+da5vbt2wBy3zolIiIiotwUvWDQrVs3HDx4ED///DN8fX0REhKCn3/+GTdu3IAkSXjjjTdyLXPy5EkAYD9rREREREZQ1LL2/vvvIygoCEIIzJo1C126dMH69esBPH/zs0GDBrmW2bhxIyRJyvUsGxERERHlpihYs7e3x969e9G9e3fY2NhACAFbW1u88847+Pnnn3OlP3ToEC5evAgAaNOmjZKiiYiIiIoFxWODlipVCuvXr0dGRgYSEhLg5eUFOzs7nWn9/f2xf/9+AEDDhg2VFk1ERERU5JlsIHd7e3uULl3aYJqKFSuiYsWKpiqSiIiIqMhTdBuUiIiIiMxLUctaqVKlEBoaKn9q1KhhqnoRERERERQGa/fv38f69evlN0C9vLzQsmVLhIWFITQ0FLVq1TJJJYmIiIiKK0XB2rBhw3Dw4EGcP38eKpUKDx8+xMaNG+VOcj09PbWCtzp16pik0kRERETFhaJgbe7cuQCAxMREHDx4EAcPHsSBAwdw7tw5qFQqJCQkYPPmzdi8eTMAwMPDAyEhIXLwFhwcrHgFiIiIiIoyk7wN6uHhgS5duqBLly4AgKSkJK3g7ezZs1CpVHj8+DG2bNmCLVu2QJIkZGVlmaJ4IiIioiLLLG+Duru7o3Pnzpg9ezZOnTqF+/fv4/PPP5cHfhdCQAhhjqKJiIiIihST9bOmSQiB2NhYHDhwAAcOHMDhw4eRnJysFaA5OTmZo2giIiKiIsUkwZq+4Ew9DwAcHR3RtGlThIWFISwsDI0bNzZF0URERERFmqJgbc6cOdi/fz+OHDmCpKQkAIaDM1tbW+U1JiIiIipGFAVro0aNgiRJEEIwOCMiIiIyA5PcBpUkCSEhIWjbti3CwsIQHBwMSZJMkTURERFRsaYoWGvVqhWOHz+O9PR07Nq1C7t37wYAuLm5yf2pMXgjIiIienGKgrU9e/bg2bNniI6OxoEDB3Dw4EFERUUhKSkJW7duxbZt2wAweCMiIiJ6UYpvg9ra2qJ58+Zo3rw5JkyYgKysrDyDN3d3d4SEhMgjGxARERGRbibvFNfGxgbNmjXD+PHjsXPnTjx+/BhHjx7FuHHj4ObmBiEEEhMTsXXrVlMXTURERFTkmKVTXABIT0/HkSNH5L7XTp06haysLPntUSIiIiLKm8mCNX3BGYBcIxeou/ggIiIiIsMUBWu7d+/G/v37DQZnzs7OaNasGcLCwhAaGopGjRrBxsZsDXpERERERYqiqKldu3a5bmu6uLigefPmcnDWsGFDWFtbK64oERERUXGkuInL1dVVKzirX78+gzMiIiIiE1EUrEVHR6NevXqwsjL5S6VEREREBIXBWoMGDUxVDyIiIiLSgU1iRERERBbMZK9lJiQkYPny5dizZw8uXLiAhIQEAECJEiVQs2ZNtG7dGv3790eJEiVMVSQRERFRkWeSYO3HH3/E6NGjkZ6eDkC7647bt2/jzp072LVrFyZNmoTZs2dj0KBBpiiWiIiIqMhTHKxNmzYNEyZMkAM0d3d3BAcHo1SpUhBC4N69e4iNjUVSUhLS0tLwwQcfIDExEZ988oniyhMREREVdYqCtQsXLuDzzz+HEAKlS5fGzJkz8frrr8PW1lYrXVZWFtatW4cxY8bgzp07+Oyzz9CxY0fUqFFDUeWJiIiIijpFLxjMnz8f2dnZKFmyJI4dO4a33norV6AGPB/cvVevXjh27Bh8fHyQnZ2N+fPnKymaiIiIqFhQFKzt27cPkiRh3LhxKFeuXJ7p/f398emnn0IIgb179yopmoiIiKhYUBSs3b59GwDQrFkzo5dp3rw5AODOnTtKiiYiIiIqFhQFa+phpdQDuBtDnZajHhARERHlTVHEpL71mZ9bmuq0xtw2JSIiIiruFAVrbdq0gRACs2bNwvnz5/NMf+7cOcycOROSJKFt27ZKiiYiIiIqFhQFayNGjIC9vT1SU1PRokULzJw5Ew8fPsyV7uHDh5gxYwZatmyJ1NRU2NvbY8SIEUqKJiIiIioWFPWzVr58efz444/o378/UlNTMXbsWIwbNw4VKlSAr68vJElCfHw8rl+/DiEEhBCQJAk//vgjb4MSERERGUHxCAZ9+vSBl5cXBg8ejNu3b0MIgX///RfXrl0DoD30lJ+fHxYvXowOHTooLZaIiIioWDDJ2KAdO3bEtWvXsHHjRoMDuXfr1g02NiYbO56IiIioyDNZ5GRjY4PXX38dr7/+uqmyJCIiIir22NkZERERkQVjsEZERERkwRisEREREVkwo55ZUw8rZUqSJOVrmCoiIiKi4sioYE2z+w0iIiIiKjhGBWsTJ040dz2IiIiISAcGa0REREQWzKhg7fvvvwcAvPPOO/D09DRrhYiIiIjof4wK1kaMGAFJktC6dWutYK1Vq1aQJAnLli1D+fLlzVZJIiIiouJK0QgGBw4cgCRJSEtLM1V9iIiIiEiDUf2sOTg4AAASExPNWRciIiIiysGoYK1ChQoAgK1bt5qzLkRERESUg1G3QTt06IBLly5h+vTp2Lt3LwICAmBrayvP/+yzz+Dh4ZGvgiVJwtKlS/O1DBEREVFxY1SwNn78ePzxxx+4evUqYmJicPLkSXmeEAKbN2/OV6FCCAZrREREREYwKlgrUaIETp48ifnz52Pv3r24ffs2MjIycOPGDUiShNKlS2u1tBERERGRaRj9NqibmxvGjx+P8ePHy3+zsnr+yNuuXbtQvXp109eOiIiIqJgz6gUDIiIiIiocivpZ279/PwCgYsWKJqkMEREREWlT1LKWlZWF0NBQODo65nvZDz/8UEnRRERERMWComCtW7duiImJyfdygwYNwo8//qikaCIiIqJiQVGwlpqaig4dOiAuLs7oZd59910sWbJESbFERERExYaiYC0gIACPHj1C27ZtcfPmzTzT9+vXD8uXLwcA9O7dW0nRRERERMWComBt9+7dKFu2LG7fvo02bdrg/v37OtMJIfDOO+9g1apVEEKgb9++WLlypZKiiYiIiIoFRcGav78/du/eDW9vb1y9ehXt2rVDUlKSVhqVSoXevXtj9erVAIABAwZg2bJlkCRJSdFERERExYLiftaqVauGP//8Ey4uLjh37hxee+01PH36FMDzQK1Xr1749ddfAfzveTUGakRERETGMUmnuPXr18fmzZthb2+PqKgoRERE4MmTJ+jRowfWrVsHAHj//fexePFiUxQnu3//PrZu3YovvvgC7du3h7e3NyRJgiRJ6NevX77z27FjB7p3746yZcvC3t4eZcuWRffu3bFjxw6T1puIiIjIWIo6xdUUFhaGyMhI9OjRAzt27EDFihXx4MEDAMAHH3yABQsWmKooma+vr0nyEUJg8ODBuYLJ27dvY+PGjdi4cSMGDRqERYsWsVWQiIiICpRJh5vq0qWL3C3H/fv3IYTAkCFDzBKo5eTv74+2bdu+0LKfffaZHKgFBwcjMjIS0dHRiIyMRHBwMABg8eLF+Pzzz01WXyIiIiJjGNWyZky3HGrh4eEYPnw45s6dix49emD06NF6ly9XrpzR+eryxRdfoGHDhmjYsCF8fX1x/fr1fA99dfXqVcyYMQMA0KBBAxw6dEgekaFhw4bo3LkzQkNDcfLkSUyfPh39+/dH5cqVFdWbiIiIyFhGBWsvMvanJEn4/fff8fvvv+udn5WVle98NX355ZeKlgeAOXPmyPWYN29erqGznJycMG/ePDRt2hRZWVmYO3cu5s2bp7hcIiIiImMYdRtUCGGWT2ETQmDz5s0AgMDAQDRp0kRnuiZNmqBatWoAgE2bNllE3YmIiKh4MKplTT3qQFFz7do13L59GwAQGhpqMG1oaCj+/vtv/Pfffy90u5WIiIjoRRgVrPXt29fc9SgUmmOaBgYGGkyrOT8uLo7BGhERERUIk74N+rK5deuWPF22bFmDaf39/XUuR0RERGROioI1Kysr2NjYyG9TvmxSUlLkaRcXF4NpnZ2d5enU1FS96TIyMpCcnKz1ISIiInpRioI1Ozs7CCEQEhJiqvoUKPWwWMDzdTHE3t5enn7y5InedN988w3c3d3lj2aLHBEREVF+KQrW/Pz8AADW1tYmqUxBc3BwkKczMzMNps3IyJCnc3bvoWncuHFISkqSP7xlSkREREooCtZatmwJADh9+rRJKlPQXF1d5WlDtzYBIC0tTZ42dMvU3t4ebm5uWh8iIiKiF6UoWBs6dCisra0xa9asl/LZLM2XCv777z+DaTVbyHhrk4iIiAqKomCtfv36mDdvHm7cuIHQ0FBERUWZql4Fonr16vL0pUuXDKbVnB8UFGS2OhERERFpMqqfNX0GDBgAAKhWrRrOnj2LkJAQ+Pv7o3bt2vD09DT4LJskSVi6dKmS4hWrWLEi/Pz8cOfOHRw8eNBg2kOHDgEAypQpgwoVKhRA7YiIiIgUBmsrVqyAJEkAngdfQgjcvHkzz4fqhRAWEaxJkoQuXbpg4cKFuHTpEo4fP65zyKnjx4/LLWtdunSR15mIiIjI3BQFa+XKlXvpA5cRI0bgp59+QlZWFoYOHYpDhw5pve355MkTDB06FABgY2ODESNGFFJNiYiIqDhSFKxdv37dRNV4MUeOHMHVq1fl/z98+FCevnr1KlasWKGVvl+/frnyCAgIwOjRozFt2jScPHkSzZs3x6efforKlSvjn3/+wfTp0xEbGwsAGDNmDKpWrWqWdSEiIiLSRVGwVtiWLFmClStX6px39OhRHD16VOtvuoI1AJgyZQru37+PZcuWITY2Fj179syVZuDAgfj6668V15mIiIgoP4r12KBqVlZWWLp0KbZt24YuXbrAz88PdnZ28PPzQ5cuXbB9+3YsWbIEVlbcXERERFSwzNKylpWVhcePHwMAPD09YWNjnga8FStW5LrVqUSHDh3QoUMHk+VHREREpJTJmori4uIwdOhQBAUFwcHBAaVKlUKpUqXg4OCAoKAgDBs2DBcvXjRVcURERETFgkmCtXHjxqF27dr44Ycf8Pfff0OlUkEIASEEVCoV/v77byxYsAB16tTB+PHjTVEkERERUbGg+P7k0KFD8cMPP0AIAeB57/6NGzdGqVKlIITAvXv3EB0djYsXLyI7OxvTp09HWloavvvuO8WVJyIiIirqFAVrR48exYIFCyBJEqpXr47FixejWbNmOtMeO3YMgwcPxvnz5zF//ny8+eabetMSERER0XOKboP++OOPAJ4P23T06FGDwVfTpk1x6NAhVKpUCQCwaNEiJUUTERERFQuKgrXDhw9DkiSMHTsW7u7ueaZ3d3fHp59+CiEEDh8+rKRoIiIiomJBUbAWHx8PAAgODjZ6mXr16gEA7t27p6RoIiIiomJBUbDm4OAAAEhLSzN6mdTUVACAvb29kqKJiIiIigVFwVrFihUBAH/88YfRy2zZsgUA5GfXiIiIiEg/RcFahw4dIITA/PnzsXfv3jzT7927F/PmzYMkSRwpgBRxKVUKquxsk+drjjyJiIiUUNR1x4gRIzB//nykpKSgffv2eO+99zBgwAAEBwfL42iqVCrExsZi6dKlWLJkCbKysuDu7o4RI0aYov5UTDl4eMDK2hobevfGg7g4k+RZMigI3VevNkleREREpqIoWPP29sbatWvRuXNnZGZmYtGiRVi0aBHs7OxQokQJSJKER48eITMzEwAghICdnR3WrVsHLy8vk6wAFW8P4uIQHxtb2NUgIiIyG8XDTbVt2xbHjx9HgwYN5CGmMjIycPfuXdy5cwcZGRny3xs2bIgTJ06gdevWpqg7ERERUZGneLgpAKhbty6io6MRExODPXv24MKFC0hISAAAlChRAjVr1kTr1q3RsGFDUxRHREREVGyYJFhTa9iwIQMyIiIiIhNSfBuUiIiIiMxHUbC2cOFCXLx40VR1ISIiIqIcFN0G/eijjyBJEry9vdGyZUuEhYUhNDQUNWvWNFX9iIiIiIo1xc+sCSHw4MEDbNiwARs2bADw/KUCzeCtdu3aiitKREREVBwpCtYSEhJw8OBBHDhwAAcOHMD58+ehUqnw6NEjbNq0CZs2bQIAeHp6IiQkRA7e6tata4KqExERERV9ioI1Dw8PdOnSBV26dAEAJCYm4tChQ3Lwdu7cOahUKiQkJGDz5s3yGKLu7u5o2bKlHMwRERERkW4m7brDw8MDnTt3RufOnQEASUlJWsHb2bNnoVKpkJiYKA/oTkRERET6mbXrDjc3N5QpUwZlypSBn58fnJycIEmSOYskIiIiKlJM2rImhEBsbKzcknb48GEkJyfL8wDA0dERTZs2RXh4uCmLJiIiIiqSFAVrOYOzI0eOICkpSZ4H/C84CwsLQ1hYGBo3bgxbW1vlNSciIiIqBhQFa15eXgzOiIiIiMxIUbCWmJgoP4PWpUsXjBw5Ek2bNmVwRkRERGQiJukUFwD++OMPHDhwQKsz3ODgYL5QQERERKSAomDt8OHD8vNqx44dQ1JSErZs2YKtW7cCeP42qLoz3LCwMAZvRERERPmkKFhr3rw5mjdvjgkTJiArKwvR0dFy8BYVFYWkpCRs3boV27ZtA5A7eKtXr55JVoKIiIioqDJZ1x02NjZo1qwZmjVrhvHjxyMrKwsxMTG5grdt27Zh27ZtkCQJWVlZpiqeiIiIqEgyaT9rWhnb2KBp06aoU6cO6tWrh1q1amHp0qVITk6Wn3MjIiIiIsNMHqylp6fjyJEjcovaqVOn5BY0zSDNycnJ1EUTERERFTmKgzVjgzNnZ2e5/7XQ0FA0atRIadFERERERZ7iFwxOnjypMzhzcXFBs2bN5OCsYcOGsLEx211XIiIioiJJUfR07NgxedrV1RXNmzeXg7MGDRrA2tpacQWJiIiIijNFwVqHDh3k4Kx+/fqwsrIyVb2IiIiICAqDNXXnt0RERERkHmwKIyIiIrJgDNaIiIiILBiDNSIiIiILxmCNiIiIyIIxWCMiIiKyYAzWiIiIiCwYgzUiIiIiC8ZgjYiIiMiCGRWsde/eHREREfjvv//MXR8iIiIi0mBUsLZp0yZs2rQJycnJ2gtbWcHGxgYXL140S+WIiIiIirt83QYVQhj1NyIiIiIyDaOCNVdXVwDAvXv3zFoZIiIiItJmVLAWGBgIAPjuu++Qmpqaa74kSaatFREREREBAGyMSfTWW28hJiYGW7duRYkSJeDr6wtbW1t5ftu2bbX+bwxJkvDPP//kr7ZERERExYxRwdrQoUNx9OhRrF+/HllZWbh9+7Y8Twih9X9jsTWOiIiIKG9GBWtWVlZYu3Ytjh07hj179uD27dvIyMjAypUrIUkSOnfuDA8PDzNXlYiIiKj4MSpYU2vatCmaNm0q/3/lypUAgClTpqB69eqmrRkRERERcQQDIiIiIkuWr5a1nFQqlanqQUREREQ6sGWNiIiIyIIpalnTlJ2djU2bNmHPnj24cOECEhISAAAlSpRAzZo10bp1a3Tt2hXW1tamKpKIiIioyDNJsLZjxw4MGjQoV5cewPMuOqKiorB48WKULVsWixcvRrt27UxRLBEREVGRp/g26M8//4zXXnsNt2/fhhACQgiUL18eTZo0QePGjVG+fHkAz4O3W7duoWPHjli9erXiihMREREVB4qCtRs3bmDQoEFQqVRwcnLC119/jfj4ePz777+IiorCsWPH8O+//yI+Ph5TpkyBi4sLVCoV3nvvPdy8edNU60BERERUZCkK1r777jtkZGTAxcUFhw8fxvjx4+Hj45MrXcmSJTFu3DgcPnwYLi4uyMjIwHfffaekaCIiIqJiQVGwtmvXLkiShDFjxqBu3bp5pq9Tpw5Gjx4NIQR27typpGgiIiKiYkFRsKa+ldm6dWujl2nTpo3WskRERESkn6JgLTs7GwDy1R2HOi071CUiIiLKm6JgrUyZMgCAqKgoo5dRp/Xz81NSNBEREVGxoChYCw8PhxAC06ZNw507d/JM/99//2HatGmQJAmtWrVSUjQRERFRsaAoWBs6dCisrKzw4MEDNG7cGOvWrZNvjWrKzs7G2rVr0bRpU9y/fx9WVlYYMmSIkqKJiIiIigVFIxjUrFkTkydPxoQJE3Dnzh307NkTHh4eCA4Ohq+vLyRJQnx8PGJjY5GYmCiPajB58mTUrFnTJCtAREREVJQpHm5q3LhxcHd3xyeffIL09HQ8fvwY+/fv10qjDtKcnJwwc+ZMfPDBB0qLJSIiIioWTDI26Icffog33ngDy5cvNziQe//+/eHt7W2KIomIiIiKBZMEawDg7e2NMWPGYMyYMabKkoiIiKjYUzyQOxEVHpWOF3osMU8iInpxJmtZI6KCZ2VtjQ29e+NBXJxJ8isZFITuq1ebJC8iIjINBmtEL7kHcXGIj40t7GoQEZGZ8DYoERERkQVjsEZERERkwRisEREREVkwBmtEREREFozBGhEREZEFY7BGREREZMEUdd3x1VdfAQAaN26Mdu3amaRCRERERPQ/ioK1SZMmQZIkbNy40VT1ISIiIiINim6Denl5AQDKlStnksoQERERkTZFwVqVKlUAAPHx8SapTGGQJMmoT1hYWGFXlYiIiIohRcHam2++CSEE1q5da6r6EBEREZEGRc+sffjhh1i5ciVWrVqF0NBQ9O3b11T1KnAffPABPvzwQ73znZ2dC7A2RERERM8pCtbi4+OxZMkSDBw4EAMGDMDq1avx1ltvoXbt2vD09IS1tbXB5S3pWTcfHx/UrFmzsKtBREREpEVRsFahQgVIkgQAEEJg79692Lt3r1HLSpKErKwsJcUTERERFXmKgjXgeZCma5qIiIiIlFMUrC1fvtxU9SAiIiIiHRQFay/zCwU5rVu3DpGRkbh58yZsbGxQqlQpNGvWDP369UN4eHhhV4+IiIiKKY4N+v8uXryIy5cv4+nTp0hNTcXVq1exatUqtGrVCt26dUNSUlJhV5GIiIiKIcXPrL3snJyc0LlzZ7zyyisIDAyEi4sLHjx4gIMHD2LRokV49OgRNm3ahC5dumD37t2wtbU1mF9GRgYyMjLk/ycnJ5t7FYiIiKgIM1mwplKpcODAARw7dgzx8fFIT0/H119/jdKlS8tpMjMzkZWVBWtra9jb25uqaEVu374NDw+PXH9v06YNhg4divbt2yM2NhYHDx7EwoULMWzYMIP5ffPNN/jyyy/NVFsiIiIqbkxyG3Tbtm2oWrUq2rRpgy+++AI//PADVqxYgcePH2ulW7p0KVxdXeHj44O0tDRTFK2YrkBNzdfXF+vXr4ednR0AYN68eXnmN27cOCQlJcmfW7dumaqqREREVAwpDtaWLFmCzp0749q1axBCwMvLS28XHgMHDoSHhwdSU1OxceNGpUUXiEqVKqFNmzYAgKtXr+LOnTsG09vb28PNzU3rQ0RERPSiFAVrV69exUcffQQAaNWqFS5evIj79+/rTW9nZ4eIiAgIIbBr1y4lRReo6tWry9O3b98uxJoQERFRcaMoWJs7dy6ePXuGGjVqYPv27QgMDMxzmZCQEADAmTNnlBRdoNjZLxERERUWRcHa3r17IUkSRowYIT/XlZfKlSsDAG7evKmk6AJ18eJFedrPz68Qa0JERETFjaJgTf3wfN26dY1extnZGQCQnp6upOgC8++//2L37t0Anj+/VqZMmUKuERERERUnioI1zUHcjfXgwQMAsIgH77ds2WJwMPl79+6hR48eePbsGQDIz+cRERERFRRF/az5+fnh6tWruHz5MurXr2/UMgcPHgQAVKhQQUnRJjF06FA8e/YMERERaNq0KSpUqABHR0c8fPgQBw4ckDvFBYAWLVowWCMiIqICpyhYa9myJa5cuYI1a9agV69eeaZ/+PAhfvzxR0iShFatWikp2mTu3LmDefPmGexDLSIiAkuWLLGYjnyJiIio+FB0G3TQoEEAgO3bt2P58uUG0/7333/o0KEDHj58CGtra3nZwrRy5Up8+eWXePXVVxEQEIASJUrAxsYGHh4eqFWrFt5//31ERUVh/fr1BjvPJSIiIjIXRS1rDRs2xODBg7Fo0SK8++672L59O15//XV5/rlz53D+/Hns2rULv/76K54+fQpJkjBq1ChUqVJFceWVCg0NRWhoaGFXg4iIiEgvxWODzps3D2lpafj555+xYcMGbNiwQX7xoHfv3nI69UsI/fr1w9SpU5UWS0RERFQsKB5uytraGitXrsS6desQHBwMIYTOT/Xq1bFmzRosW7ZMDuaIiIiIyDDFLWtqERERiIiIwJ07d3Dy5Encv38f2dnZ8PLyQnBwsNwZLhEREREZz2TBmpqfnx86d+5s6myJiIiIiiXFt0GJiIiIyHxM2rL27NkznD59GhcuXEBCQgIAoESJEqhZsybq1asHW1tbUxZHREREVOSZJFhLTU3F5MmTsXTpUjx+/FhnGk9PTwwcOBCfffYZXF1dTVEsERERUZGn+DZoXFwcatSogVmzZiEhIUHv26AJCQmYNWsWatWqhb///tsUdSciIiIq8hS1rCUmJqJ169a4e/cuAKBmzZro27cvGjVqBF9fXwghcP/+fcTExGDlypU4f/48bt68idatW+PChQtwd3c3yUoQERERFVWKWtamT5+Ou3fvQpIkTJ48GWfPnsWoUaMQEhKCgIAAVKtWDSEhIfj4449x5swZfP311wCej8c5ffp0k6wAEZGpqbKzX6p8iahoU9SytmnTJkiShDfffBMTJkwwmFaSJIwfPx7nz5/Hb7/9ho0bN3IkAyKySFbW1tjQuzcexMWZLM+SQUHovnq1yfIjouJDUbB248YNAEDfvn2NXqZfv3747bff5GWJiCzRg7g4xMfGFnY1iIiU3QZVv9Xp4+Nj9DLqtC4uLkqKJiIiIioWFAVrtWrVAgBcuXLF6GXUadXLEhEREZF+ioK1999/H0IIzJ07FyqVKs/0KpUKc+bMgSRJGDRokJKiiYiIiIoFRcHa66+/jv79++P48ePo2rUr4uPj9aa9d+8eunfvjhMnTqBfv3548803lRRNREREVCwY9YLBqlWr9M4LDQ3FhQsXsHXrVlSqVAlt27ZFw4YN4ePjA0mScO/ePcTExGDXrl3IyMhAw4YNERoailWrVqFPnz4mWxEiIiKiosioYK1fv36QJMlgGkmS8PTpU2zZsgVbtmzJNV8IAUmScPLkSfTv3x+SJDFYIyIiIsqD0V13CCEUpzM2DyIiIiJ6zqhg7dq1a+auBxERERHpYFSwVr58eXPXg4iIiIh0UPQ2KBERERGZF4M1IiIiIgvGYI2IiIjIgpkkWIuLi8PIkSPRoEEDlChRAra2trC2tjb4sbFRNIY8ERERUbGgOGKaNm0avvjiC2RnZ7NrDiIiIiITUxSsrVu3DuPHjwcAWFlZISQkBHXq1IGHhwesrHiHlYiIiEgpRcHad999BwAoU6YMtm/fjlq1apmkUkRERET0nKLmr3PnzkGSJEyePJmBGhEREZEZKArWbG1tAQB169Y1RV2IiIiIKAdFwVpAQAAA4NGjRyapDBERERFpUxSs9e3bF0IIbNq0yUTVISo8LqVKQZWdbZa8zZUvEREVfYpeMBg4cCAiIyOxePFitG3bFp06dTJVvYgKnIOHB6ysrbGhd288iIszWb5V27dHqylTzJYvEREVbYqCNVtbW2zevBl9+/ZFt27d0LNnT7zxxhsICAiAk5NTnsuXK1dOSfFEZvEgLg7xsbEmy887MNCs+RIRUdGmuFNcDw8PDBs2DMePH0dkZCQiIyONWk6SJGRlZSktnoiIiKhIU9xz7YgRI9C2bVs8fPgQQoh8fYiIiIjIMEUta7/88gu+//57AICbmxu6deuG2rVrcwQDIiIiIhNRFKzNmzcPABAYGIj9+/fD19fXJJUiIiIioucUNX9dunQJkiRh0qRJDNSIiIiIzMAkIxioO8clIiIiItNSFKwF/n/XAfHx8SapDBERERFpUxSs9e/fH0IIo7vrICIiIqL8URSsDRw4EK+99hp++eUXzJ8/31R1IiIiIqL/p+ht0EOHDmHYsGF48OABhg8fjjVr1qBnz55Gj2DQsmVLJcUTERERFXmKgrWwsDBIkiT//8SJEzhx4oRRy3IEAyIiIqK8KR5uiiMREBEREZmPomBt//79pqoHEREREemgKFgLDQ01VT2IiIiISAcO4ElERERkwRisEREREVkwBmtEREREFkzRM2utWrV64WUlScLevXuVFE9ERERU5CkK1g4cOABJkgx236HZDxvwv64+cv6diAqfS6lSUGVnw8ra2uR5myvfl4W5tm1x365ExYGiYK1ly5Z5Bl1paWm4cuUKkpKSIEkSAgICULp0aSXFEpGZOHh4wMraGht698aDuDiT5VsyKAjdV682WX4vI3NsW25XouJBccuaMYQQ2LZtG4YPH46EhAQsWbIELVq0UFI0EZnRg7g4xMfGFnY1iiRuWyLKrwJ5wUCSJLz22ms4cuQIrK2t0a1bN9y+fbsgiiYiIiJ6qRXo26ClS5fGxx9/jEePHmHGjBkFWTQRERHRS6nAu+5Q3/7ctm1bQRdNRERE9NIp8GDNzs4OAHDnzp2CLpqIiIjopVPgwdqRI0cAAE5OTgVdNBEREdFLp0CDtWPHjuGrr76CJElo1KhRQRZNRERE9FJS1HXHV199lWcalUqFx48f4+TJkzhx4gRUKhUkScLIkSOVFE1ERERULCgK1iZNmpSvkQiEELCxscGMGTPQpk0bJUUTERERFQuKgjUABoeaAp73sebq6oqKFSsiNDQUgwYNQvXq1ZUWS0RERFQsKArWVCqVqepBRERERDoU+NugRERERGQ8BmtEREREFozBGhEREZEFU/yCgaaUlBRcu3YNKSkpyM7OzjN9y5YtTVk8ERERUZFjkmDtp59+wg8//IBz584ZvYwkScjKyjJF8URERERFlqJgLTs7GxEREdiyZQuAvLvxICIiIqL8URSsLVq0CH/88QcAwNfXF/3790f9+vVRokQJWFnxcTgiIiIipRQFa6tWrQIAVK9eHYcPH4anp6dJKkVEREREzylq/oqLi4MkSfj8888ZqBGRXi6lSkFlxEtH+WWOPInoOXOdXzxv888kLxhUq1bNFNkQURHl4OEBK2trbOjdGw/i4kySZ9X27dFqyhST5qmZL1FxZ+pzFgBKBgWh++rVJsuvuFAUrFWtWhVnzpxBQkKCqepDREXYg7g4xMfGmiQv78BAk+epmS8Rmf78ohej6DZoz549IYTA1q1bTVUfIiIiItKgKFgbNmwYateujYULF+Lw4cOmqhMRERER/T9FwZq9vT127dqF+vXro02bNvjkk09w5swZPH361FT1IyIiIirWFD2zZm1tLU8LITB79mzMnj3bqGU5ggERERFR3hQFazlHLOAIBkRERESmpShYmzhxoqnqQUREREQ6MFgjIiIismAcwJOIiIjIgjFYIyIiIrJgDNY03Lx5E6NHj0ZQUBCcnZ1RokQJNGrUCLNmzUJ6enphV4+IiIiKIZOMDVoUbNu2Db1790ZSUpL8t/T0dMTExCAmJgZLlizB9u3bUalSpUKsJRERERU3bFkDcPbsWbzxxhtISkqCi4sLpkyZgqioKOzduxfvvfceAODvv/9Gx44dkZqaWsi1JSIiouKELWsARowYgfT0dNjY2GDXrl1o2rSpPK9Vq1aoWrUqPvnkE1y6dAnffvstvvjii0KsLRERERUnxb5lLSYmBgcOHAAADBw4UCtQUxs1ahSCgoIAAHPnzsWzZ88KsopERERUjBX7YG3Tpk3ydP/+/XWmsbKyQp8+fQAAjx8/loM7IiIiInMr9sHa4cOHAQDOzs6oX7++3nShoaHy9JEjR8xeLyIiIiKAwRri4uIAAFWqVIGNjf5H+AIDA3MtQ0RERGRuxTpYe/r0KR4+fAgAKFu2rMG0np6ecHZ2BgDcunXL7HUjIiIiAor526ApKSnytIuLS57pnZ2dkZaWZrD7joyMDGRkZMj/V/fblpycrKCmeupTpQrcs7NNlp+dnx+Sk5NNmq858jRXvi9TXc2VL+v6cuXrXKWKWa4tRGqmPg/Mdcyq8xRCmDxviyCKsZs3bwoAAoB455138kzv7+8vAIjKlSvrTTNx4kQ5T3744Ycffvjhp+A+t27dMmWYYDGKdcuag4ODPJ2ZmZlnenWLmaOjo94048aNw8cffyz/X6VSISEhAV5eXpAkSUFt/yc5ORn+/v64desW3NzcTJInkRqPLzI3HmNkakIIpKSkwM/Pr7CrYhbFOlhzdXWVp40ZmSAtLQ2A4Vum9vb2sLe31/qbh4fHi1UwD25ubrzQkdnw+CJz4zFGpuTu7l7YVTCbYv2CgYODA7y9vQEA//33n8G0jx8/loM1f39/s9eNiIiICCjmwRoAeWSCq1evIisrS2+6S5cu5VqGiIiIyNyKfbDWokULAM9vcZ46dUpvuoMHD8rTzZs3N3u9DLG3t8fEiRNz3W4lMgUeX2RuPMaI8kcSoqi+52qc6OhoNG7cGADw/vvvY9GiRbnSqFQq1KxZE3FxcfDw8MD9+/dha2tb0FUlIiKiYqjYt6w1atQIISEhAIClS5fi2LFjudLMnj1bHrVg+PDhDNSIiIiowBT7ljUAiI2NRfPmzfHkyRO4uLhg/PjxCA8Px5MnT/Drr79i8eLFAICAgACcPHlS6y1SIiIiInNisPb/tmzZgrfffltvz8oBAQHYtm0bqlSpUsA1IyIiouKs2N8GVevUqRPOnTuHkSNHIiAgAE5OTvDw8ECDBg0wffp0xMbGFnqgdvPmTYwePRpBQUFwdnZGiRIl0KhRI8yaNQvp6emFWjcqeKdPn8bUqVPRvn17+Pv7w97eHi4uLggICEC/fv1w+PDhfOW3Y8cOdO/eHWXLloW9vT3Kli2L7t27Y8eOHUbnkZ6ejpkzZ6JRo0YoUaIEXFxcEBQUhNGjR+PmzZv5XUWyUJ988gkkSZI/Bw4cyHMZHl9EChTi6AmUD1u3bhXu7u56h9ioVq2a+Oeffwq7mlRAWrZsadTQK++8847IyMgwmJdKpRKDBg0ymM+gQYOESqUymM/Vq1dFtWrV9Obh7u4utm3bZsrNQIXgzJkzwsbGRmvf7t+/X296Hl9EyjFYewmcOXNGODk5CQDCxcVFTJkyRURFRYm9e/eK9957T75YBQYGipSUlMKuLhWAypUrCwDCz89PDB8+XKxfv15ER0eLY8eOiW+//VaUKVNGPi569eplMK/x48fLaYODg0VkZKSIjo4WkZGRIjg4WJ43YcIEvXmkpKSIwMBAOe17770n9u7dK6KiosSUKVOEi4uLACCcnJzE2bNnTb05qIBkZ2eLhg0bCgDCx8fHqGCNxxeRcgzWXgJhYWECgLCxsRFRUVG55s+YMUO+iH355ZeFUEMqaB07dhS//fabyMrK0jn/wYMHIiAgQD4uDh06pDPdlStX5FaSBg0aiPT0dK35aWlpokGDBvLxd/XqVZ35TJw4US5rxowZueZHRUXJ5YSHh+dzbclSzJkzR/5hOG7cuDyDNR5fRKbBYM3CRUdHyxep999/X2ea7OxsERQUJAAIT09PkZmZWcC1JEu0ZcsW+dgZNmyYzjQffvihnObYsWM60xw7dkxOM2TIkFzzMzMzhYeHhwAggoKCRHZ2ts583n//fTmfkydPvviKUaG4efOm3IK1f/9+rQBKX7DG44vINPiCgYXbtGmTPN2/f3+daaysrNCnTx8Az8cwNeZhXyr6wsLC5Ol//vkn13whBDZv3gwACAwMRJMmTXTm06RJE1SrVg3A8+NR5HiB/MCBA0hMTAQA9O3bF1ZWui8r/fr1k6c3bNhg7GqQhfjwww+RmpqKvn37ah1b+vD4IjIdBmsWTv1Gn7OzM+rXr683XWhoqDx95MgRs9eLLF9mZqY8resL7tq1a7h9+zYA7eNHF/X8//77D9evX9eap/nWqaF8GjRoAGdnZwA8Rl82a9euxdatW1GiRAnMnDnTqGV4fBGZDoM1C6ceOaFKlSqwsbHRmy4wMDDXMlS8aY5nq3l8qGkeJ7rmazJ0fBmbj42NDSpXrqwzD7JciYmJGD58OABg+vTpKFmypFHL8fgiMh0Gaxbs6dOnePjwIQCgbNmyBtN6enrKvypv3bpl9rqRZVOpVJg2bZr8/zfeeCNXGs3jJK/jy9/fX+dymv93dnaGh4eHUfk8ePAAGRkZBtOSZfjkk08QHx+PZs2aYeDAgUYvx+OLyHQYrFmwlJQUedrFxSXP9OpgLTU11Wx1opfDnDlzEB0dDQDo1q0bGjRokCtNfo4v9bEF5D6+1Pnk5xjVlQ9ZniNHjmDJkiWwsbHBokWLIEmS0cvy+CIyHQZrFuzp06fytJ2dXZ7p7e3tAQBPnjwxW53I8h08eBBjx44FAPj4+GDhwoU60+Xn+FIfW0Du40udT36OUV35kGXJzMzEoEGDIITAyJEjUatWrXwtz+OLyHQYrFkwBwcHeVrzYXF91M3+jo6OZqsTWba//voL3bp1Q1ZWFuzt7bF27Vr4+vrqTJuf40vzllLO40udT36OUV35kGWZOnUq4uLiUK5cOUycODHfy/P4IjIdBmsWzNXVVZ42pkk/LS0NgHG3C6jouXbtGtq2bYvHjx/D2toakZGRBt+ey8/xpT62gNzHlzqf/ByjuvIhy3Hp0iV88803AIB58+Zp3V40Fo8vItPR/3ohFToHBwd4e3vj4cOH+O+//wymffz4sXyh0nxYl4qHO3fuoHXr1rhz5w4kScKyZcvQrVs3g8toPvSd1/Gl+dB3zuOrbNmyOHHiBNLS0pCYmGjwIXB1PiVLltS6ZUWWZc6cOcjMzESlSpWQnp6OX3/9NVeaCxcuyNP79u1DfHw8AKBTp05wdnbm8UVkQgzWLFxQUBAOHz6Mq1evIisrS2/3HZcuXdJahoqPhw8fok2bNvj3338BPG8JUXeSbEj16tXlac3jRxdDx1f16tXx+++/y+n0dX6alZUld87LY9SyqW8n/vvvv+jVq1ee6SdPnixPX7t2Dc7Ozjy+iEyIt0EtXIsWLQA8b94/deqU3nSafWo1b97c7PUiy5CUlIR27drh4sWLAIBp06bho48+MmrZihUrws/PD4D28aPLoUOHAABlypRBhQoVtOapj9G88jl58qTc+stjtOjj8UVkOgzWLFzXrl3l6eXLl+tMo1KpsGrVKgCAh4cHwsPDC6JqVMjS09PRsWNHnD59GgAwYcIEfPrpp0YvL0kSunTpAuB5i8Xx48d1pjt+/Ljc8tGlS5dc3TeEhYXB3d0dALBy5cpcwwWprVixQp7O6xYtFa4VK1ZAPB87Wu9H86WD/fv3y39XB1s8vohMqMBHI6V8CwkJEQCEjY2NiIqKyjV/xowZ8gDGEydOLPgKUoHLyMgQbdu2lff78OHDXyifv//+W9jY2AgAokGDBiI9PV1rfnp6umjQoIF8/F2+fFlnPp9//rlclxkzZuSaHxUVJZcTGhr6QnUly2LMQO48vohMg8HaS+D06dPC0dFRABAuLi5i6tSp4tixY2Lfvn1i0KBB8kUsICBAJCcnF3Z1qQB0795d3u+tWrUS586dE+fPn9f7+fvvv/XmNXbsWDmv4OBg8euvv4qYmBjx66+/iuDgYHneuHHj9OaRnJwsAgIC5LSDBg0S+/btE8eOHRNTp04VLi4uAoBwdHQUsbGxZtgiVNCMCdaE4PFFZAoM1l4Sf/zxh3Bzc5MvVjk/AQEB4sqVK4VdTSog+o4DfZ/y5cvrzSs7O1sMGDDA4PIDBw4U2dnZBut05coVUbVqVb15uLm5iS1btph4S1BhMTZY4/FFpByfWXtJdOrUCefOncPIkSMREBAAJycneHh4oEGDBpg+fTpiY2NRpUqVwq4mvYSsrKywdOlSbNu2DV26dIGfnx/s7Ozg5+eHLl26YPv27ViyZAmsrAxfLqpUqYLY2FhMnz4dDRo0gIeHB5ycnFCtWjWMHDkS586dw2uvvVZAa0WWgscXkXKSEHqe1iQiIiKiQseWNSIiIiILxmCNiIiIyIIxWCMiIiKyYAzWiIiIiCwYgzUiIiIiC8ZgjYiIiMiCMVgjIiIismAM1oiIiIgsGIM1IiIiIgvGYI2IiIjIgjFYIyIiIrJgDNaIiqjr169DkiRIkoQVK1YUdnUsVnZ2Nr777js0atQIbm5u8jbr2rVrYVeNiAgAYFPYFSAiKky9evXCunXrCrsaRER6sWWNiIqtqKgoOVDr2LEjdu/ejXPnzuH8+fP4/vvvC7l2xceBAwfkFs0DBw4UdnWILA5b1oio2NqzZw8AwNraGmvWrIGbm1sh14iIKDe2rBFRsXX79m0AgK+vLwM1IrJYDNaIqNjKyMgAANja2hZyTYiI9GOwRkXepEmT5OdhAODp06eYOXMm6tWrB1dXV7i6uqJRo0aYP38+srKy9OajzmPSpEkGywsLC4MkSQgLC8s1L+ezOUIILF26FC1atICXlxfc3NzQqFEj/Pzzz1rLZWZmYtGiRWjSpAlKlCgBV1dXNG/eHGvXrs3Xtli3bh1at24NHx8fODo6IjAwEGPHjsXjx4+NWj46OhrvvfceAgIC4OLiAmdnZwQGBuKjjz7ClStX9C63YsUKeb2vX7+OjIwMzJ07F02aNIG3t7dR21Wf8+fPY9CgQahatSqcnJzg6uqKGjVqYOTIkbh+/brOZdR1WblyJQDgxo0b8t80j5X8yszMxA8//IDw8HCULFkSdnZ2KFWqFDp06IBffvkFKpVK77L9+vWDJEmoUKECgOetfh9//DECAgLg5OSEkiVLokOHDvjzzz+Nqkt6ejrmzp2L8PBw+Pr6ws7ODj4+Pmjbti2WL1+O7OxsvctWqFABkiShX79+AIBTp06hX79+qFixIuzt7XNtn3///RezZ89Gp06dUKFCBTg6OsLR0RHly5fHm2++iR07dugsR/3Gcnh4uPy38PDwXPtC19vMptzWd+/exaeffooaNWrA1dU117Nz2dnZWLFiBdq1a4dSpUrBzs4OHh4eqFq1Kl555RVMnToVFy9e1FsekWKCqIibOHGiACAAiPj4eFGnTh35/zk/nTp1EtnZ2TrzUaeZOHGiwfJCQ0MFABEaGppr3v79++V8du3aJTp16qS3LsOGDRNCCJGQkCBatmypN92UKVN01uPatWtymuXLl4sBAwbozaN06dLir7/+0rtOz549Ex988IHe5QEIW1tbsXjxYp3LL1++XE4XExMj6tatm2v5vLarLlOnThVWVlZ662Rvby9WrlyZazlD66H+5Nf169dFUFCQwTxbtGghHj16pHP5vn37CgCifPnyIiYmRvj4+OjNZ/jw4QbrEh0dLcqUKWOwLo0aNRLx8fE6ly9fvrwAIPr27SsWLlwobGxs9G6ff//916jt+fbbb4tnz55plaN5jBr6LF++3Gzb+tixY8Lb2zvX8vv37xdCCJGSkiJCQkLyrGNERITBfUKkBIM1KvI0g7VmzZoJOzs7MWzYMLF7925x6tQpsWbNGq0L/6JFi3TmY+pgrXHjxgKA6N27t9i2bZs4deqUiIyMFNWqVZPT7N69W3Tu3FnY2NiIDz74QOzatUucOnVKLF26VPj5+QkAwtraWly4cCFXWZpfhA0bNpS/oCMjI8XJkyfF9u3bxZtvvimnKVu2rEhKStK5Tn369JHTtW/fXvzyyy8iOjpaxMTEiJ9++knUqFFDnv/HH3/kWl4zWKtdu7aQJEn06dNHXu+NGzeK7du3G9yuOS1YsEDOs2TJkmLWrFni2LFj4siRI2LSpEnC2dlZABCSJIlt27ZpLXv+/Hlx/vx50aVLFwFA+Pn5yX9Tf/IjJSVFVKpUSa5P165dxR9//CFOnjwp1q1bJx8TAETTpk1FVlZWrjzUAUTJkiVFhQoVhL29vRg7dqw4dOiQOHHihPj+++9F6dKl5Xxmz56tsy7nzp2T193Hx0dMnDhR7NmzR8TGxoqdO3eKjz76SA6+GjduLDIzM3PloQ7WqlevLqytrUWFChXE/Pnz5e37zTffyGmvXLki7OzsRKdOncT3338v9uzZI06fPi327NkjfvjhB61j44svvtAqJzMzU5w/f14sW7ZMTrNs2bJc++Lx48dm2dZeXl7Cz89PuLi4iAkTJogDBw6I6OhosXTpUnHp0iUhhBCjRo2S83vttddEZGSkOHr0qDh16pTYsWOHmD59uggJCRE9evQw6lghehEM1qjI0wzWbG1t5V/Mmh49eiR8fX3lYEIXUwdrAMTcuXNzpbl7965wc3OTv7glSRIbN27Mle7s2bNyq5K6FU5TzlaLDh065GrZEEKIr776Sk4zevToXPPXr18vz//pp590rvOTJ09Eq1atBABRoUKFXOVoBmsAxNKlS3XmY6z79+8LJycnOdC6efNmrjSnT5+Wg5YyZcroDEo0W1iUGD16tLxun332Wa75KpVK9O7dW07zww8/6K2L+jg9ePBgrjS3b98WZcuWFQCEk5OTuHfvXq5yateuLQCIOnXqiAcPHuis759//ikfO0uWLMk1Xx2sARC1atXSCpZySk1NFXfu3NE7X6VSiX79+gkAwtnZWSQmJuZKo3le6Do/NZl6W7u4uIgzZ87oLc/f318AyDMY09eKR2QKDNaoyNMM1j7++GO96caOHSun0/WFYo6WNX00v0x69uypN5369mhwcHCueZrBmr29vbh9+7bOPLKzs0XNmjUFAOHp6SmePn2qNb9+/foCgOjWrZveegghxMWLF7VaBDVpBmutWrUymI8xpk+fLucXGRmpN93XX38tp1u7dm2u+aYI1p4+fSo8PDzklihdLTlCCJGUlCS8vLzkdPrqAkAMGTJEb3m//fabnG7GjBla87Zs2SLPO3v2rMF6v/HGGwKAaN68ea55msHaoUOHDOZjjEePHglra2sBQKxfvz7XfGODNXNs66+++spg3W1tbQUA8d133xleSSIz4gsGVKz07t1b77z69evL09euXTN7XXr27Kl3Xu3ateXpN998U2+6OnXqAHj+gLchbdu2hZ+fn855VlZW6Nu3LwDg8ePHOH36tDzv9u3bOHXqFADgjTfeMFhGUFAQvL29AQDHjh3Tm87QPjCWun80Dw8PRERE6E337rvv5lrG1E6dOoXExEQAzx9ct7a21pnOzc1N3oYXL17E3bt39ebZv39/vfO6desGDw8PALnXafPmzQCAatWqaR1DurRs2RIAEBMTo/dlA39/f4SEhBjMJ6dnz57hv//+Q1xcHC5cuIALFy7gzp078PLyAgCcPXs2X/lpMse2zut4LF26NADgt99+Q3p6+gvUmkg5BmtUrAQGBuqdV6JECXk6JSXF7HUJCAjQO0/9ZWxsurzq27BhQ4PzGzVqJE9fuHBBnj558qQ83atXr1xv6eX8PHz4EAAQHx+vt6y8gghjqOsYHBxssNsNX19f+Y0/zfUyJc18GzdubDCt5nx99bGzszO4jWxtbREcHKwzD/X++vvvv/PcV0OGDAHw/K3KhIQEnWUZu6+ePXuGBQsWoEmTJnBxcYG/vz+qV6+OWrVqyZ/79+8DgHyMvAhTb2sXFxdUqlTJYD7qHzJRUVGoWLEihgwZgo0bN+LBgwfGVptIMQZrVKw4OTnpnWdl9b/TwVC3BgVdF2PSGeqmAAB8fHwMzvf19ZWnNb+41V+w+WWoBcLT0/OF8tSkrqNmvfUpVaqU1jKmpplvXvVR18VQfUqUKAEbG8ODy6jLyZmHqfeXMfsqISEBTZs2xZAhQ3DixAlkZmYaTP/kyZMXqqO6LDVTbGvNH0X6fP755xgwYAAkScL9+/exYMECdO/eHb6+vqhVqxYmTpyIe/fuGbcCRC+Iw00RFQN59RsmhND5d82gdfXq1Ua3tBj6ktd36+pFGNMfmr51M4cX3c75ycNQPur91bx5cyxatCjPfNT03SI3Zl8NHz5cvlXetWtXDBgwALVr14aPjw8cHBzk9SlXrhxu3bplsv1him1tzPrZ2tpi6dKlGDVqFCIjI7Fv3z6cPHkSmZmZ8m3eb7/9Fr/88gu6dOlidP2J8oPBGpGRJEmCECLPVqy0tLQCqpHx8vrlr9kio3k7WP2cEfB8/WvWrGn6yr2AEiVK4O7duwZvt6qp111zvUxdF7X4+HiDt60194O++jx69AjZ2dkGAwn1/sqZh5eXF+7du4cHDx4UyL5KTk7Gb7/9BgB46623sHr1ar1pje142RBTb+v8qF69OiZPnozJkyfjyZMnOHr0KNasWYNVq1YhNTUVvXr1wj///CM/40ZkSrwNSmQkV1dXAIa/dFQqlcGe/AtLTEyM0fM1v+TVz0YBwK5du0xfsRekrmNsbCyePXumN939+/dx48YNrWXMVRcAOHHihMG00dHROpfTlJmZafAh/KysLJw5c0ZnHur9dfnyZXm9zenKlSvy9jf0wszff/+N1NRUvfONHTHC1Nv6RTk6OqJ169ZYtmwZZs6cCeD57d2tW7eatBwiNQZrREaqWLEiAO2H7nPavn07kpKSCqpKRtu1a5feN+JUKpU87JKnpyfq1asnz6tSpQqqV68OAPj1119x8+ZN81fWCK1btwYAJCYm4vfff9ebbunSpfLtMPUypla/fn352aeVK1fqfd4xJSVFHh6sevXqBltg1PtDl40bN8o/GHKuU+fOneXpGTNmGFV/JTSHZzP0nGJet2QdHBzkafV4rbqYY1sr9corr8jTSl6eIDKEwRqRkUJDQwE8/0V/9OjRXPPv3r2LYcOGFXS1jJKRkYH3339f55fbtGnTcP78eQDAgAEDYG9vrzX/s88+A/B8TNXu3bsbfAsuIyMDP/zwA54+fWrC2ufWv39/+cWLUaNG4datW7nSnD17FlOnTgUAlClTBl27djVLXezt7eUuQv766y98+eWXudIIITBkyBD5y1z9JqY+CxcuxJEjR3L9PT4+HqNHjwbw/MUT9ZuKahEREQgKCpLzWLp0qcFyLly4gC1bthhMY0iVKlXkVrFVq1bpTLN161bMmzfPYD6awdQ///yjN505trUhCQkJ+OOPPww+/6bZ4qz+QUdkcoXUvxtRgdHsFNeQvDrmvHDhgjxMj6enp5gzZ46IiYkRR48eFTNmzBC+vr7Cy8tLVK1a1ahOcQ11/qnZiey1a9deaN00O8Vt0KCB3BHvr7/+Kk6dOiX+/PNP0bNnTzlN2bJldXYGLIR2J6Le3t5iwoQJYteuXSI2NlYcOXJErFy5Urz77ruiRIkSAoBISUl5ofXJD83hpnx8fMS3334rjh8/Lo4ePSq+/PJL4eLiIgDdw03lXC+lIxgkJydrDYHUrVs3sWXLFnHq1Cmxfv16ERYWJs8zZrip8uXLCwcHBzFu3Dhx+PBhER0dLebPny8PMQYYHm5Kve4ARLt27cTKlSvF8ePH5f0+depU0axZMwFAjBo1KlcemmOD5qVjx45aZW3YsEEezmzgwIHC2tpaVK1aVZQsWdJgnuqRGSpWrCg2bdok4uLixJUrV8SVK1dEcnKyWbZ1XvtdfQ5VqFBBfPzxx+K3334Tx48fFydPnhRbtmwRgwYNkkeCKFu2rEhNTc1zexG9CAZrVOSZKlgTQohvv/1WTpPz4+npKQ4ePGj0CAYFGawtX75cHvJH1yevgdyzsrLEJ598IvdCb+jj7Ows0tPTX2h98mvKlCkvNJC7mqmCNSGeb+/AwECD26Z58+ZGD+Sua3Bx9UfX8GKazp49K/9oyOvz5Zdf5lo+P8HazZs3Rbly5fTmX65cOfHXX3/lmecPP/ygN4+cA7mbclsbYuxA82XKlBGnT5/Oc1sRvSjeBiXKh5EjR2LHjh1o164dPD09YW9vj4oVK+Kjjz7CmTNn5F7hLdHy5cuxZs0ahIWFwcvLC/b29ggICMAnn3yCv/76S342TRdra2tMnz4dFy9exKhRoxAcHAxPT09YW1vD1dUVNWrUQO/evbFy5UrcvXsXjo6OBbJO48ePR2xsLN577z1UrlwZjo6OcHZ2RlBQEIYPH45Lly6hT58+BVKXChUq4OzZs5g/fz5CQ0Ph5eUFW1tb+Pr64tVXX8XPP/+MQ4cOGfVmYoMGDXD69GkMGzYMlStXhoODA7y8vPDqq69i+/bt+O677wwuX7t2bVy8eBErV65E165d4e/vDwcHB9jZ2aF06dIICwvDZ599hlOnTuGLL75QtN7+/v44ffo0xowZg4CAANjb28Pd3R116tTBxIkTcebMGYPHltoHH3yA33//HW3btoWPj4/BvuZMua0NKV++PM6cOYOZM2eiffv2qFatGjw8PGBjYwNvb2+EhoZi1qxZiIuL03oZh8jUJCEKsBMiIiLSqV+/fli5ciXKly+P69evF3Z1iMiCsGWNiIiIyIIxWCMiIiKyYAzWiIiIiCwYgzUiIiIiC8ZgjYiIiMiC8W1QIiIiIgvGljUiIiIiC8ZgjYiIiMiCMVgjIiIismAM1oiIiIgsGIM1IiIiIgvGYI2IiIjIgjFYIyIiIrJgDNaIiIiILBiDNSIiIiIL9n9mUtcOl7tLcwAAAABJRU5ErkJggg==",
      "text/plain": [
       "<Figure size 640x480 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "counts = []\n",
    "with open('./results/nodes_in_pipelines.csv', newline='') as csvfile:\n",
    "    reader = csv.reader(csvfile)\n",
    "    next(reader)\n",
    "    for row in reader:\n",
    "        try:\n",
    "            count = int(row[1])  # csv-file: pipeline, operators, processes, channels, total_nodes\n",
    "            counts.append(count)\n",
    "        except ValueError:\n",
    "            print(f\"Skipping row with non-numeric value in the specified column: {row}\")\n",
    "\n",
    "plt.rc('font', size=20)\n",
    "plt.hist(counts, bins=20,color='maroon', edgecolor='white')\n",
    "plt.title('Number of operators in workflows')\n",
    "plt.xlabel('number of operators')\n",
    "plt.ylabel('number of workflows')\n",
    "plt.tight_layout()\n",
    "plt.savefig(f'{directory_plots}/number_of_operators.png', format=\"png\", bbox_inches='tight')\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Longest path in a workflow (with all nodes and processes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import networkx as nx\n",
    "import matplotlib.pyplot as plt\n",
    "import numpy as np\n",
    "import csv\n",
    "\n",
    "directory_path = './dags'\n",
    "directory_plots = './plots'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We are looking for the longest path, considering all nodes here"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def find_longest_path(dot_file):\n",
    "    G = nx.nx_agraph.read_dot(dot_file)\n",
    "\n",
    "    longest_path_nodes = nx.dag_longest_path(G)\n",
    "\n",
    "    return longest_path_nodes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Here you can change the label, to get all nodes or just the processes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import csv\n",
    "import networkx as nx\n",
    "\n",
    "def longest_path_processes(dot_file):\n",
    "    g = nx.nx_agraph.read_dot(dot_file)\n",
    "    \n",
    "    # we want to delete the operator and channel nodes, but keep the edges \n",
    "    nodes_to_delete = [node for node, attr in g.nodes(data=True) if 'shape' in attr]\n",
    "\n",
    "    node_labels = {}\n",
    "\n",
    "    for node, attr in g.nodes(data=True):\n",
    "        node_labels[node] = attr.get('label', '')  # get the label of the node, otherwise empty string\n",
    "\n",
    "    for node in nodes_to_delete:\n",
    "        # connect in and out nodes with parent and child\n",
    "        for in_src, _ in g.in_edges(node):\n",
    "            for _, out_dst in g.out_edges(node):\n",
    "                g.add_edge(in_src, out_dst)\n",
    "\n",
    "        g.remove_node(node)\n",
    "        \n",
    "    # find longest path in modified graph\n",
    "    longest_path = nx.dag_longest_path(g)\n",
    "    \n",
    "    longest_path_labels = [node_labels[node] for node in longest_path]\n",
    "    \n",
    "    return len(longest_path), longest_path_labels\n",
    "\n",
    "def longest_path_to_csv(directory):\n",
    "    csv_data = []\n",
    "    \n",
    "    for filename in os.listdir(directory):\n",
    "        if filename.endswith('.dot'):\n",
    "            dot_file = os.path.join(directory, filename)\n",
    "            pipeline = filename.split('.')[0]\n",
    "            length, longest_path_labels = longest_path_processes(dot_file)\n",
    "            # to not consider empty pipelines\n",
    "            if len(longest_path_labels) != 0:\n",
    "                length = len(longest_path_labels) -1\n",
    "            csv_data.append((pipeline, length, longest_path_labels))\n",
    "    \n",
    "    with open('./results/longest_paths_label.csv', mode='w', newline='') as file:\n",
    "        writer = csv.writer(file)\n",
    "        writer.writerow(['pipeline', \"longest_path_length\", \"longest_path_label\"])\n",
    "        writer.writerows(csv_data)\n",
    "\n",
    "directory_path = \"./dags\"\n",
    "longest_path_to_csv(directory_path)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "\n",
    "pd.set_option('display.max_colwidth', None)\n",
    "df_longest_paths = pd.read_csv('./results/longest_paths_label.csv')\n",
    "df_longest_paths"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_longest_paths.describe()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We plot out results for the processes next"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAlgAAAG4CAYAAACZypnDAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjcuMiwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy8pXeV/AAAACXBIWXMAAA9hAAAPYQGoP6dpAAB5wklEQVR4nO3dd1gUx/8H8PfSO6ggiqLYxY6KHcEeS6yxxR4TW6LRaEw0xUSjsUejJsZuNJqoscTeUVRUUFSMmNijIgKigKAgML8/+N7+7uDuOLwFDny/nucejtvZmdmbvb3Pzc7OSkIIASIiIiJSjFlBV4CIiIioqGGARURERKQwBlhERERECmOARURERKQwBlhERERECmOARURERKQwBlhERERECmOARURERKQwBlhERERECmOARUSUzwIDAyFJEiRJQkBAQEFX57WptkGSpIKuCpHJYYBF+SIgIEA+EH/zzTcFXR0iekN4eXlpBILqDzs7O5QuXRrVqlVDq1atMHHiRGzatAnR0dGKlN2vXz+N8hYuXGh0nq9evcKBAwfw6aefomnTpqhQoQIcHR1hbW2NkiVLokaNGujXrx/mzZuH69evK7AV9NoEUT7w9/cXAAQAMW3atIKuDilk7dq1crsOGTKkoKtTIO7cuSO/B+XLlzdonePHj8vr+Pv752n98pJqG0z5q6R8+fIa9TTkYWlpKXr37i3OnDnz2uU+e/ZM2NjYaORbp06d184vLS1NrFixQnh5eeVqW+rUqSM2btwo0tPTX7tsej0WeRq9ERERmYg2bdqgevXq8v/p6el49uwZnj59iitXruDRo0cAMnuJtm7dij///BMTJkzAzJkzYW1tnauytmzZgpcvX2q8duXKFVy6dAn16tXLVV7Pnj1Dnz59cPjwYY3XnZ2d4evrCzc3Nzg6OuLJkyeIiorChQsX5LKvXLmCgQMH4vz581i8eHGuyiXjMMAiIqLXIoQo6CrkysCBAzF06FCdy+/cuYPVq1dj+fLlePLkCTIyMrBgwQJERERg9+7dMDMzfFTN+vXr5ec2NjZywLN+/fpcBVjPnj1D06ZNNU73tWvXDl9++SWaN28Oc3PzbOu8fPkSQUFB+Pnnn7Fr1y5kZGQgKSnJ4DJJGRyDRUREBKBChQr47rvv8Pfff2tcfLBv3z5MnjzZ4Hxu3bqF06dPAwDMzMwwZ84cedmmTZuQlpZmUD5CCAwePFgOrszNzfHTTz/h0KFDaNmypdbgCsgM6Nq1a4ft27fj2rVraNmypcF1J+UwwCIiIlLj7u6OAwcOoH79+vJrixYtwq1btwxaX733KiAgAKNGjULx4sUBANHR0di/f79B+WzcuBG7d++W/58/fz5Gjx5t0Loq1apVw/Hjx/HRRx/laj0yHgMsKrQOHjyI9957D1WrVoWTkxNsbW1Rvnx59OjRA2vXrsWrV69yzGPo0KHyFT7r1q0DACQlJeGnn35CixYt4O7uDmtra3h6eqJ///7yr1JDXb16FaNHj0aVKlVgZ2cHV1dXNGzYEN9//z1iY2MBAOvWrZProO/0hbqQkBBMmDAB9erVg5ubG6ysrFCqVCn4+/tjzpw5ePr0qUH5xMbGYv78+Wjbti08PDxgY2MDOzs7lC9fHg0aNED//v2xdu1aPHz4UGM91fs2bNgw+bX169drvVLLmGkI1K8Au3v3LgDgn3/+wSeffIKaNWvC2dkZTk5OqF27NqZOnYrIyEiD8n3x4gV27tyJcePGye1sZWUFBwcHeHl5oWfPnlizZg1SU1N15qFqtwoVKsiv3bt3T+cVa4Y4duwY+vXrh4oVK8LGxgYlSpRAy5YtsXTpUoP2ZwC4fv06Jk+ejCZNmsDV1RVWVlZwdnZG5cqV0aRJE4wZMwY7d+5EQkKCQfnpY8j2aWvDBw8e4KuvvkLdunXh4uICe3t7VK9eHWPHjsW9e/eMrpcSrK2tsWnTJvm0YHp6OmbPnp3jekIIbNiwQf5/0KBBsLKyQp8+feTX1AMwffmol9esWTN8/PHHudkEmZmZWY6nJe/fv49vv/0WLVu2lI97jo6OqFChAho1aoRhw4Zh8+bN8nGLDFCwY+zpTaHkVYSPHz8Wbdq0yfHqmSpVqoiQkBC9eQ0ZMkROv3btWnHt2jXh7e2tN9+vv/7aoHrOmTNHWFpa6syndOnS4vTp07m6Ei8uLk706tUrx213cXERW7du1ZvXzp07RbFixQy6EqlMmTI637ecHsZcJad+BdidO3fEypUrhbW1tc6ynJ2dc9zus2fPCgcHB4Pq7uXlJS5evKg1H/V2M+ShLutVhCkpKWLEiBF6169fv76IiYnRu23Tpk0TFhYWBtVnwIABuWsMLXRtn7qsbbhjxw7h7Oyss162trZiz549RtdNW/lr167N9fpvv/22xucqp6vx1NvW1tZWJCQkCCGEOH36tPy6lZWVePLkid58AgMDNd6XLVu25Lruhlq+fLmwtbU1aL9p3rx5ntWjqOEgdypUHj9+jObNm2t01VeqVAmNGzeGtbU1rl27hnPnzgEAbty4gVatWuHAgQNo3rx5jnlHRkaibdu2iIyMhIuLC/z8/FCqVCnExsbi2LFjiI+PBwBMnz4dNWrUQN++fXXmtXDhQnz22Wfy/zY2NmjVqhXKli2L2NhYHD9+HI8ePUKXLl0wfvx4g7Y9KioKrVu3RkREhPyat7c36tWrB0dHR0RHR+PUqVOIjY2VrzrasGEDBgwYkC2v0NBQvPPOO/JYEFtbWzRp0gReXl6wtrZGQkICbt26hfDwcCQnJ2dbv23btnBwcMD169dx9OhRAED16tXRpk2bbGmrVKli0Pbl5K+//pJ/wZcuXRotW7aEg4MDbty4gdOnTyM9PR3x8fHo378/rKys0LVrV635PH36FM+fPwcAlCxZEjVr1kTZsmVhb2+P5ORk3Lx5E+fPn0daWhru3r0Lf39/XLx4EZUrV9bIx9vbGx9++CESExPx66+/AgAcHR0xePDgXG/byJEjsW7dOpiZmaFx48aoXr06MjIycPbsWfzzzz8AgIsXL2Lw4MHYt2+f1jwWLVqEb7/9Vv7f1dUVTZo0QenSpSFJEuLi4nD9+nVEREQgPT0913VUwtGjRzFy5Eikp6ejXLlyaNq0KZycnHDnzh0EBgYiLS0NL168QJ8+fXD16lWN3sGC0rt3b/k03bNnzxAeHo66devqTK/eO9W9e3c4OjoCyOyBqly5Mm7evInU1FT8/vvvGDNmjM58jh8/Lj+3sbFBt27djN0UrXbu3IlRo0bJ/zs5OaFp06YoW7YsLCwsEB8fj3///RdXr17V26NLWhR0hEdvBqV6sDp27CjnY2dnJ3777bdsaUJCQkTFihXldJ6enuLp06da81PviVH1jEyePFkkJSVppHvy5Ilo3bq1nLZixYoiIyNDa55Xr14VVlZWctrOnTuL6OhojTRJSUli1KhRGuVCTw9Wenq6aNWqlUZvhrbeuRcvXohvvvlGSJIkAAh7e3tx+/btbOm6desm59WrVy8RFxentdyXL1+KvXv3ipEjR2pdntfzYKn3PlhZWQkzMzMxb968bL0I169fF3Xr1pXTurq6ZnvPVc6ePSumTp0qwsPDdZb7+PFjMWjQIDm/Nm3a6Exr7DxYqvb39fUVERERGukyMjLEokWLNHoQTpw4kS2/V69eiRIlSshpvv/+e5Gamqq17CdPnog1a9aIOXPmGFRXfdTrpYt6G1pbWwt7e3uxYcOGbJ+fq1evijJlyshphw0bZnT9spb/Oj1Y//zzj8Z2/vLLLzrTJiUlCUdHRznt3r17NZZ/88038rLGjRvrLVf9eNOkSZNc19tQ6p+bjz76KNuxTyUxMVFs2bJFfPbZZ3lWl6KGARblCyUCrGPHjmkc6Hbt2qUz7Z07dzROQ3z77bda02U91TVlyhSdeUZFRQl7e3s57dmzZ7Wme+edd+Q0DRs2FCkpKTrz7N27t0b5uoKUX3/9VU5Tr149kZiYqDNPIYT4+uuv5fSjRo3Ktlz1ZWxtbZ1jXvrkZ4AFQMyePVtn2ujoaFG6dGk57aeffmp0+eoB/bVr17SmMTbAAjJPZ+trB/V9Slt7hoeHF8gpnNwGWJIkif379+tMu2fPHjmtg4ODePXqldF1NDbAysjIEGZmZnIe06dP15l2w4YNcrqSJUtmq/+tW7c03rOsAbW6SpUqKR5sZpWYmKjxQ1TXj0Z6PRzkToXGL7/8Ij9/++23dZ4CAjIH1k6dOlX+f/ny5TnO2ePm5oavv/5a53J3d3d07txZ/j8kJCRbmri4OOzatUv+f+7cubCystKZ54IFCwyaW0f9FhtLliyBg4OD3vRTpkyBi4sLAGDz5s3IyMjQWK4a4GxnZ5djXqaiYsWKmDhxos7lbm5umD59uvz/2rVrDb4cXhf1iw6OHDliVF76zJ49W287vPfee/Jzbfud+oB1Nzc3ZSunoC5duuCtt97SubxTp04oVaoUAOD58+cmcasXSZLk03wA9F5Aon56sH///rCw0ByFU7FiRY3hCqpTy9rExcXJz1WfZX3OnTuHjz76SO/jxo0bGuuo7zclSpTgPSUVxjFYVGioj0lQ/8LRZdiwYZgyZQoyMjLw6NEj/PPPPxqzOGf19ttvw8bGRm+ePj4+2LJlCwDIV0SpO3PmjHy1V+nSpXO8gs7T0xMtW7ZEYGCgzjSPHj3CpUuXAABlypRBixYt9OYJZI7ZaNq0Kfbv34/4+HhcvXoVderUkZeXK1cOt27dwtOnT7Fp0ya8++67OeZZ0N59991sX1hZ9evXDx999BFSUlIQGxuLq1ev6r16Kjk5GWfPnkV4eDhiYmKQmJioMT5J/epJVRsozcbGBl26dNGbxsfHR36ubb8rV66c/PzYsWOIiIiAt7e3YnVUSu/evfUulyQJdevWRVRUFIDMba1Vq1Z+VE0vBwcHeQxmYmKi1jQPHjzAsWPH5P8HDhyoNd2gQYPkq5E3bNiA7777TuuPLPVy7O3tc6xjREQEli1bpjfNO++8ozEm0s3NDba2tnjx4gXCw8Nx8uRJzpmlIAZYVCg8fPhQ4waszZo1y3EdNzc3VK1aVf4VfPHiRb0BVu3atXPMs0SJEvJz1QFXnfqXsK+vr0G/CH19ffUGWMHBwfJzIYTB89moXwhw//59jQCrb9++mDVrFoDML4Lff/8dffv2RevWrVG6dGmD8s9vTZo0yTGNg4MDatWqhQsXLgAAwsLCtAZYcXFx+Prrr/Hrr7/q/MLMKq8uT69WrZreXk4g5/2ubNmyaNasGc6cOYOEhAQ0bNgQAwYMQI8ePdCiRQuNHpiCpMRnrCCo7yNOTk5a02zYsEHuKa5evToaNmyoNV2fPn3w8ccfIyUlRQ7K2rZtmy2do6Oj3FuWV7OwW1paokePHti0aRPS09PRtm1bvPPOO3jnnXfg7++v0RaUewywqFCIiYmRn9va2qJkyZIGrefl5SUHWDl9QTo7O+eYn6Wlpfxc27xE6mWULVvWoDqWKVNG73L1uZ0iIyNz/JWqTdbTGl988QVOnDiB06dPQwiB3bt3y1dKVahQAX5+fmjbti26deum8wslv6n30ujj6ekpB1jq+43KvXv30LJlS/z333+5Kt/QQCy3crvf6TrtuWbNGrRq1QqPHj1CcnIyVq5ciZUrV8Lc3By1a9dGy5Yt0bFjR7Rt2zbHnsC8osRnLL9lZGRotL1qwtCs1E/3DRo0SGd+xYoVQ5cuXfDnn38CyDytqC3AKl68uPy5ffbsWY71HDp0qNZ59HL6kffDDz/g4sWLuH79Ol69eoXNmzdj8+bNkCQJ1atXh5+fH9q3b4/OnTvn2MNPmjgGiwoF1WX1gGHd5drS5vQFqcT4A/V62tnZGbROTtujxK/4rF/KdnZ2OH78OH744QdUqlRJY9mdO3fw66+/YvDgwShdujQ+/fRTvHjxwug6GOt13k9tbT5gwAA5uHJycsLEiRNx8OBB3LlzB8+fP0d6ejpE5gVAGqels45jU4pS416qVauGy5cvY8KECRpBQHp6Oi5duoQff/wRHTt2RPny5bFy5UpFysytwjjG599//9UYv6kaI6bu3Llz8g85SZK0To2iTj0A2759u9b9tHz58vJz9alZlFayZEmEhITgm2++gYeHh/y6EAIRERFYsWIF3nnnHXh4eGD27NkFNsVHYcQAiwoF9QHAuekuV0+bH6dJ1L/ctc0fpU1O26OeZ/fu3eUv/9w8tP2ytbS0xPjx43Hz5k2Eh4dj2bJl6Nevn0aPWnJyMubPn49WrVoVeJD1Ou9n1jY/c+aMPP7F0dER586dw/z589G+fXt4eXnB3t5eYzxMXvVa5RU3NzcsXLgQUVFRCAoKwsyZM9GxY0eNXsjIyEiMGDEC48aNK8CaFh6qefVUtJ2qVh/cLoTQmL1e26N79+5y+uTkZGzbti1bnn5+fvLzS5cu5ekcVA4ODpg2bRru37+PkJAQLFiwAN27d4erq6uc5unTp5gyZQp69epV6G7yXVAYYFGhoH5l1IsXLwweD6N+2w31g0VeUS/jwYMHBq2T9TY0Wbm7u8vPs14FpJRatWphzJgx2Lx5Mx48eICwsDAMHz5cXn7u3LnXOjWpJENP6amny9rmqklRgcxTKvrG5AEwmdu25JalpSVatGiBqVOnYt++fYiNjcWBAwfg7+8vp1myZInWKxJJk+qiFiBzfFiNGjU0lqempuKPP/4wqgxtVxO2atVKfv7y5Uv89ddfRpVhCDMzMzRs2BCffPIJduzYgcePH+PUqVMaAeGuXbvk05ukHwMsKhTKlCmjMe7qzJkzOa7z5MkTeRZsABo3bs0r6gOqQ0NDDfqll9OXnPov5r///tvgwM0Y9erVw6pVqzBixAj5NW0H+Pw85aM+2F+X58+f4+rVq/L/WdtcfTxbzZo1c8zv5MmTOaYpDKe9LC0t0aFDBxw6dEhjoLn6jYQpu4iICI0bM/fp0ydbe+/evVueUsHS0hKNGzc26NGgQQM5jxMnTmS7OrRly5YaV4IuWrQo33uOzMzM0Lx5c2zfvh0dOnSQX8+PYK8oYIBFhYb6LzrVjZn1Wb9+vTxuxsPDA9WqVcurqsmaNWsmD9KNjIzUe3UgkHl1X1BQkN40FSpUyHagzS/q0wc8fvw423L1Qa95PSB58+bNOc5rtXnzZvlUiqura7ZL/NVP/+V0yjEyMtKgL5L8fA+MZWVlhXbt2sn/a2tTypSSkoIBAwbIQY2FhQUmT56cLZ366cEuXbrg7NmzBj1CQ0PlKRNElhtEA5mBu/rttk6fPo2lS5fmxabmSJIkjTkAud8YhgEWFRojR46Un+/YsQMHDhzQmfb+/fv47rvvNNbNj56GEiVKaEyAOnnyZL1jJyZNmmTQoFH1A+3ixYtzNemlak4hlZSUFI3B+Pqon27TNoGl+mXcOZ3qNNbt27exYMECnctjY2Mxbdo0+f+hQ4dqnehRRX1C2KzS09MxYsQIpKSk5FgvFxcXOXCLjo4ukCDr6dOnBg/Cz6lNKbMd33rrLYSFhcmvffbZZ/Dy8tJIFxMTo9HDldPg9qzU02s7TTh48GB06tRJ/v+TTz5R9AKFxMREg8d2cb/JPQZYVGi0atUKHTt2lP/v3bs3tm7dmi1dWFgY2rRpI1/i7Onpma8DeqdNmyb3YoWGhqJnz57ZpgtITk7G6NGjsWXLFlhbW+eY58CBA9G6dWsAmVcEdu7cGXPmzNE5QP758+fYvHkzWrdujbFjx2ose/ToETw9PTFx4kScP39e6/pCCBw4cEBjZnv1A72K+ummc+fO5Xrqg9ywsrLC1KlTsWDBgmzBxL///ot27drh0aNHADIDv08//TRbHp07d5YD7RMnTmDSpEnZBu9HRUWhV69e2Lt3r0FXrFpbW6Nq1aoAMttmx44dr7V9xti1axeqVKmCefPm4c6dO1rTvHz5EosWLdIYP6OtTd9kd+/exddff40aNWpo9D736NEDM2bMyJb+t99+k3tVnZycNHp5DKEeYN28eVO+AENFkiRs3LhRY/8aMWIEOnXqhKCgIL1BdWhoKPr166e3/AsXLqB8+fKYNm0a/v77b61p0tPT8dtvv2HJkiXya9xvDMN5sCjfLV++HDt37jQ4/fTp0+VeobVr16J58+a4desWnj9/jj59+qBKlSpo3LgxrKysEBERgbNnz8rd+vb29ti8ebNBt5pQSu3atTFz5kz5dMLevXtRvnx5tGrVCmXKlMGTJ09w7NgxPHv2DC4uLpgwYYLc86Lrtjnm5ubYsmUL2rVrh7CwMKSmpuLzzz/H9OnT0aRJE5QrVw5WVlZ4+vQp/v33X1y7dk3uSenVq1e2/J49e4aFCxdi4cKFKF68OHx8fFCmTBlYW1sjOjoaV65c0fiirlq1Kj7++ONs+bi7u6N58+Y4ffo0Xr58ibp16+Ktt95C6dKl5W2pVKkSRo8ebdybiszbDo0fPx6TJk3CwoUL0bJlSzg4OODGjRs4deqU3BNobm6OlStXap0rrXr16hg0aJDcW7BgwQJs2rQJvr6+KFmyJO7evYuTJ08iNTUVjo6OmDdvHkaNGpVj3Xr16oWZM2cCyAyG169fj8qVK2vM6TR//nyj3wN9bt++jcmTJ2Py5MkoV64c6tSpg5IlS0IIgaioKJw9e1ZjPrQBAwYYNGFvUbJx40aEhobK/6enpyM+Ph5Pnz7FlStXNMboAZmfx8mTJ2P69Olae8DVTw/26tUr1/NEVa5cGY0aNZJ/6Kxfv17jVjpA5rxZwcHB6N27tzxT/P79+7F//364uLjA19cXbm5ucHJywosXL/Do0SOEh4fLPzZUfHx8sk3JAmT+oJg+fTqmT58Od3d3+Pj4oFSpUrCwsEBUVBQuXLigkZefn1+OgRv9T77c8ZDeeOo3e87tI+sNWqOiojTuNK/rUblyZXH+/Hm99VK/2bMhN4LNzc2NZ82aJSwsLHTWr3Tp0uL06dNixYoV8mvjxo3Tm2dycrIYNWqU3nzVH7a2tmLWrFkaeTx48EBYW1sb/P4HBASIR48e6azThQsXhJOTk871/f39c3xfdVG/Ue+dO3fEL7/8IqysrHSW5eTkJP744w+9eSYlJYn27dvr3eayZcuKU6dOadyUWd92xMfHixo1aujNU52h+arTlZcQQmzdulVIkmRQe5qZmYkxY8aI1NRUg8p93TqpZG3DnOT2M5mTrDcMN+RhZWUl+vbtK86dO6cz38uXL2usc+TIkdeq348//ijn4ezsLF68eKE13atXr8TPP/8sypUrl6tt8fHxEWvXrhXp6enZ8jx79qzBxxIA4p133hEJCQmvtZ1vIvZgUaHj7u6Oo0eP4uDBg/j9999x6tQpREVF4dWrVyhZsiR8fHzQvXt3DBw4UKMHIb9NmTIFXbp0wdKlS3HkyBFERkbCzs4OXl5e6NWrF0aMGAFXV1eNQe459bTZ2tri559/xmeffYaNGzfi2LFj+Pfff/HkyRNkZGTA2dkZFStWRN26ddGmTRu89dZb2WZiV+9FCwoKwoULF3Dz5k3ExMTIPTfly5eHr68v+vbtq3WWaXX169fHlStXsGTJEhw7dgy3b9+WJ+xU2ogRI9CiRQssX74cR44cwYMHDyCEQPny5dGlSxeMHTs2x5nx7ezssH//fmzatAnr169HWFgYEhIS4OrqiooVK6JXr14YOnQoihUrluNFCipOTk44f/48fv75Z+zevRsRERF49uxZvo3Heuedd/Do0SMcOnQIp0+fxuXLl3H79m15BnBnZ2dUrVoVLVq0wODBg7NNNfCmsra2hrOzM5ydnVGmTBnUr18fDRo0QPv27XOc1kW996p06dIaF+HkRt++ffHJJ58gLS0N8fHx2Llzp9YeIgsLC4waNQrDhw/H4cOHcfToUZw+fRqPHj3CkydP8OrVKzg7O6NEiRKoXbs2fH190bFjR733cmzcuDGio6Nx5MgRnDp1CmFhYbh16xaePHmC9PR0ODk5oVKlSmjSpAkGDhyIRo0avdY2vqkkIThjGFFBGjBgADZt2gQg8yo4dr//Py8vL3kuqjt37mQbZExEZKo4yJ2oACUlJWHfvn3y/76+vgVYGyIiUgoDLKIC9OWXX8qncXx9fbUOQiUiosKHARZRHti2bRs+/fRT3Lx5U+vy2NhYjBkzRmPSUG3TChARUeHEQe5EeeD58+eYP38+5s+fj6pVq6J27dooUaIEUlJScOfOHZw7d05jEssBAwagd+/eBVhjIiJSEgMsojz277//4t9//9W6zNzcHB9++CEWLlyYz7UiIqK8xACLKA+8++67cHNzw4EDB3Dx4kU8fvwYsbGxSE5ORrFixeDl5YWAgAC89957+XKPRCIiyl+cpiGPZWRkIDIyEo6OjvlyLzwiIiLKG0IIJCYmwsPDQ+edN1TYg5XHIiMj4enpWdDVICIiIoXcv38fZcuW1ZuGAVYec3R0BJDZGFln1CYiIqLCIyEhAZ6envJ3uz4MsPKY6rSgk5MTAywiIqIiwJAhP5wHi4iIiEhhDLCIiIiIFMYAi4iIiEhhDLCIiIiIFMYAi4iIiEhhDLCIiIiIFMYAi4iIiEhhDLCIiIiIFMYAi4iIiEhhDLCIiIiIFJbnt8pJSUnBqVOnEBsbiwoVKqBRo0Z5XSQRERFRgTIqwLp37x6WLVsGAJg6dSpcXFw0lp89exbvvPMOHj16JL9Wv359/PnnnyhXrpwxRRMRERGZLKNOEe7YsQPz58/HsWPHsgVXiYmJ6N69Ox49egQhhPy4cOECOnfujLS0NGOKJspXGenphTJvIiIqGEb1YB0+fBiSJKF79+7Zlq1YsQLR0dGQJAnjxo1DmzZtcPDgQfz000+4du0a1q9fj+HDhxtTPFG+MTM3x/YBAxATEaFovm7e3uj522+K5klERAXPqADr9u3bAIAGDRpkW7ZlyxZIkoQePXpg0aJFAIC3334bMTEx2Lp1K7Zt28YAiwqVmIgIRIWFFXQ1iIioEDDqFGF0dDQAwN3dXeP1hIQEXLx4EQAwbNgwjWX9+vUDAFy+fNmYoomIiIhMllEBVmJiIgAgPcsYktOnTyM9PR3m5uYICAjQWObp6QkAiIuLM6ZoIiIiIpNlVIDl7OwMAIiMjNR4PTAwEABQt25d2Nvba13XxsbGmKKJiIiITJZRAVatWrUAZF5NqJKeni6Pv2rVqlW2dR4+fAgg+2lFIiIioqLCqEHuPXr0wIkTJ7Bhwwa4u7vDz88PGzZswL179yBJEvr06ZNtndDQUADgPFhERERUZBnVgzVy5Eh4e3tDCIH58+ejW7du2LZtG4DMKwYbNmyYbZ0dO3ZAkqRsY7OIiIiIigqjAixra2scPXoUPXv2hIWFBYQQsLS0xKBBg7Bhw4Zs6U+ePIlr164BANq1a2dM0UREREQmy+h7EZYqVQrbtm1DSkoK4uLiUKJECVhZWWlN6+npiePHjwMAfH19jS2aiIiIyCQpdrNna2trlC5dWm+aChUqoEKFCkoVSURERGSSjDpFSERERETZGdWDVapUKfj7+8uPmjVrKlUvIiIiokLLqAArOjoa27Ztk68cLFGiBFq2bImAgAD4+/ujdu3ailSSiIiIqDAxKsAaN24cTpw4gfDwcGRkZCA2NhY7duyQJx4tVqyYRsBVt25dRSpNREREZMqMCrAWLVoEAHj27BlOnDiBEydOIDAwEFeuXEFGRgbi4uKwa9cu7Nq1CwDg4uICPz8/OeDy8fExegOIiIiITI0khBBKZxofH68RcF2+fBkZGRmZBUqS/DctLU3pok1OQkICnJ2dER8fDycnp4KuDhnhl/r1ERUWpmiepXx8MPLiRUXzJCKivJGb7/Q8uYrQ2dkZXbt2xYIFC3DhwgVER0fjq6++km8OLYRAHsR1RERERCZBsXmw1AkhEBYWhsDAQAQGBiIoKAgJCQkaQZWdnV1eFE1ERERU4BQJsHQFVKplAGBra4umTZsiICAAAQEBaNy4sRJFExEREZkcowKsH374AcePH8epU6cQHx8PQH9AZWlpaXyNiYiIiEycUQHWxIkTIUkShBAMqIiIiIj+R5FThJIkwc/PD+3bt0dAQAB8fHzkqwWJiIiI3jRGBVitW7fG2bNnkZycjEOHDuHw4cMAACcnJ3m+KwZcRERE9KYxKsA6cuQIXr16hfPnzyMwMBAnTpzAmTNnEB8fjz179mDv3r0AGHARERHRm0XxiUbT0tKyBVxJSUmZhf0vqHJ2doafn588w3tRxolGiw5ONEpE9GYr0IlGLSws0KxZM0ydOhUHDx7E06dPcfr0aUyZMgVOTk4QQuDZs2fYs2eP0kUTERERmYQ8mWgUAJKTk3Hq1Cl5bqwLFy4gLS1NvuqQiIiIqKhSLMDSFVAByDaDu2o6ByIiIqKiyKgA6/Dhwzh+/LjegMre3h7NmjVDQEAA/P390ahRI1hY5FnHGREREVGBMyrS6dChQ7ZTfg4ODmjevLkcUPn6+sLc3NzoihIREREVFkZ3JTk6OmoEVA0aNGBARURERG80owKs8+fPo379+jAzU/xiRCIiIqJCy6gAq2HDhkrVg4iIiKjIYNcTERERkcIUu5wvLi4Oa9euxZEjR3D16lXExcUBAIoXL45atWqhbdu2GDZsGIoXL65UkUREREQmSZEA65dffsGkSZOQnJwMQHOahocPHyIyMhKHDh3CN998gwULFmDEiBFKFEtERERkkowOsGbPno0vvvhCDqqcnZ3h4+ODUqVKQQiBx48fIywsDPHx8UhKSsLo0aPx7NkzTJ482ejKExEREZkiowKsq1ev4quvvoIQAqVLl8a8efPQu3dvWFpaaqRLS0vD1q1b8emnnyIyMhJffvklOnfujJo1axpVeSIiIiJTZNQg96VLlyI9PR1ubm4IDg7Gu+++my24AjJvAN2/f38EBwejZMmSSE9Px9KlS40pmoiIiMhkGRVgHTt2DJIkYcqUKShXrlyO6T09PfHZZ59BCIGjR48aUzQRERGRyTIqwHr48CEAoFmzZgav07x5cwBAZGSkMUUTERERmSyjAizVLXFUN3k2hCotZ38nIiKiosqoKEd1WjA3p/tUaQ05pUhERERUGBkVYLVr1w5CCMyfPx/h4eE5pr9y5QrmzZsHSZLQvn17Y4omIiIiMllGBVjjx4+HtbU1nj9/jhYtWmDevHmIjY3Nli42NhZz585Fy5Yt8fz5c1hbW2P8+PHGFE1ERERksoyaB6t8+fL45ZdfMGzYMDx//hyff/45pkyZAi8vL7i7u0OSJERFReHu3bsQQkAIAUmS8Msvv/AUIRERERVZRs/kPnjwYJQoUQKjRo3Cw4cPIYTA7du3cefOHQCat83x8PDAihUr0KlTJ2OLJSIiIjJZityLsHPnzrhz5w527Nih92bPPXr0gIWFYveXJiIiIjJJikU7FhYW6N27N3r37q1UlkRERESFEiejIiIiIlIYAywiIiIihRXqACs6Ohp79uzB119/jY4dO8LV1RWSJEGSJAwdOtSgPNatWyevk9Nj3bp1ebo9REREVDQYNAZLdUscJUmSlKtb7Gjj7u6uUG2IiIiIlGNQgKU+1YKp8vT0hLe3Nw4dOvTaeRw8eBAeHh46l5ctW/a18yYiIqI3h0EB1rRp0/K6Hq/l66+/hq+vL3x9feHu7o67d++iQoUKr51f1apV4eXlpVwFiYiI6I1UqAOsb7/9tqCrQERERJSNQQHWjz/+CAAYNGgQihUrlqcVIiIiIirsDAqwxo8fD0mS0LZtW40Aq3Xr1pAkCWvWrEH58uXzrJJEREREhYlRM7kHBgZCkiQkJSUpVZ8CNXToUERERODp06dwcnJC5cqV0bZtW4wePRplypQp6OoRERFRIWHQPFg2NjYAgGfPnuVlXQrciRMnEB0djVevXuHJkyc4d+4cZs6cicqVK+OXX34p6OoRERFRIWFQD5aXlxf++ecf7NmzB82aNcvrOuW7ihUromfPnmjatCk8PT0BALdv38aff/6Jbdu24eXLlxg1ahQkScKIESP05pWSkoKUlBT5/4SEhDytOxEREZkegwKsTp064fr165gzZw6OHj2KqlWrwtLSUl7+5ZdfwsXFJVcFS5KE1atX52qdvNCjRw8MGTIEkiRpvO7r64u+fftiz5496NmzJ169eoUJEyaga9euKFWqlM78vv/+e17dSERE9IaThAGziMbFxaFJkya4efNm5kr/C0ZUq2YNTnIihIAkSUhPT89tffVSnwdryJAhit3aZubMmfjyyy8BAN999x2++OILnWm19WB5enoiPj4eTk5OitSHCsYv9esjKixM0TxL+fhg5MWLiuZJRER5IyEhAc7OzgZ9pxvUg1W8eHGEhoZi6dKlOHr0KB4+fIiUlBTcu3cPkiShdOnSGj1aRc0HH3yAr776CkIInDhxQm+AZW1tDWtr63ysHREREZkag68idHJywtSpUzF16lT5NTOzzDHyhw4dQo0aNZSvnYkoWbIkXF1dERMTg4cPHxZ0dYiIiMjEGXQVIRWO+zESERGRaTBqHqzjx48DgFH3/ysMoqOj8eTJEwDQezNoIiIiIsDIHqy0tDT4+/vD1tY21+uOGTPGmKLz1YoVK+QeLH9//wKuDREREZk6owKsHj16ICQkJNfrjRgxwiQm7rx79y7CcrgqbM+ePZgxYwaAzAlXhw0blh9VIyIiokLMqFOEz58/R6dOnXDy5El4e3sbtM7777+PNWvW5HpqB21OnTolTx0BALGxsfLzmzdvZpumYejQoRr/3717F61atULTpk3x9ttvo169eihZsiSEELh9+za2bduGbdu2yb1X8+fP5y1ziIiIKEdGBVhVq1bFv//+i/bt2+P06dMoV66c3vRDhw7Fhg0bAAADBgwwpmgAwKpVq7B+/Xqty06fPo3Tp09nK1+b4OBgBAcH6yzHzs4OP/zwQ46zuBMREREBRgZYhw8fRvPmzfHgwQO0a9cOQUFBKFmyZLZ0QggMHjwYv/32G4DMSUDXrFljTNGKaNCgATZu3Ijg4GCEhobi0aNHiI2NRVpaGooVK4aaNWuiTZs2eP/997VuFxEREZE2Bs3krs8///wDPz8/PHnyBHXq1EFgYCCcnZ3l5RkZGRg4cCB+//13AMB7772HlStXKnKKsDDIzayvZNo4kzsR0ZstN9/pRs+DVa1aNezfvx8ODg64cuUKunTpgpcvXwLIDK769+8vB1fvv/8+Vq1a9cYEV0RERPRmUmSi0QYNGmDXrl2wtrbGmTNn0KtXL7x48QLvvPMOtm7dCgAYOXIkVqxYoURxRERERCZNsZncAwICsHnzZpiZmeHAgQOoUKECdu3aBQAYPXo0fv75Z6WKIiIiIjJpit4qp1u3bli1ahWAzNnPhRD46KOPsGzZMiWLISIiIjJpBl1F+N9//xmcYatWrfDxxx9j0aJFeOeddzBp0iSd6+c0rQMRERFRYWRQgPU69xqUJAl//vkn/vzzT53L09LScp0vERERkakzKMAyciYHIiIiojeKQQHW2rVr87oeREREREWGQQHWkCFD8roeREREREWGolcREhEREZGRAZaZmRksLCwwd+5cpepDREREVOgZFWBZWVlBCAE/Pz+l6kNERERU6BkVYHl4eAAAzM3NFakMERERUVFgVIDVsmVLAMDFixcVqQwRERFRUWBUgDV27FiYm5tj/vz5SEhIUKpORERERIWaUQFWgwYNsGTJEty7dw/+/v44c+aMUvUiIiIiKrQMmgdLl/feew8AUK1aNVy+fBl+fn7w9PREnTp1UKxYMb1jsyRJwurVq40pnoiIiMgkGRVgrVu3DpIkAcgMmIQQ+O+//3D//n296wkhGGARERFRkWVUgFWuXDk5wCIiIiKiTEYFWHfv3lWoGkRERERFB2+VQ0RERKQwBlhERERECjPqFKEuaWlpePr0KQCgWLFisLDIk2KIiIiITJJiPVgREREYO3YsvL29YWNjg1KlSqFUqVKwsbGBt7c3xo0bh2vXrilVHBEREZHJUiTAmjJlCurUqYOffvoJ//zzDzIyMiCEgBACGRkZ+Oeff7Bs2TLUrVsXU6dOVaJIIiIiIpNl9Lm7sWPH4qeffoIQAgDg7e2Nxo0bo1SpUhBC4PHjxzh//jyuXbuG9PR0zJkzB0lJSVi8eLHRlSciIiIyRUYFWKdPn8ayZcsgSRJq1KiBFStWoFmzZlrTBgcHY9SoUQgPD8fSpUvRt29fnWmJiIiICjOjThH+8ssvAIAKFSrg9OnTegOmpk2b4uTJk6hYsSIAYPny5cYUTURERGSyjAqwgoKCIEkSPv/8czg7O+eY3tnZGZ999hmEEAgKCjKmaCIiIiKTZVSAFRUVBQDw8fExeJ369esDAB4/fmxM0UREREQmy6gAy8bGBgCQlJRk8DrPnz8HAFhbWxtTNBEREZHJMirAqlChAgDgr7/+Mnid3bt3A4A8FouIiIioqDEqwOrUqROEEFi6dCmOHj2aY/qjR49iyZIlkCQJnTp1MqZoIiIiIpNlVIA1fvx4ODk54dWrV+jYsSM+/PBDXLhwARkZGXKajIwMXLhwAWPGjEHHjh3x6tUrODk5Yfz48cbWnYiIiMgkGTUPlqurK7Zs2YKuXbsiNTUVy5cvx/Lly2FlZYXixYtDkiQ8efIEqampAAAhBKysrLB161aUKFFCkQ0gIiIiMjVG3yqnffv2OHv2LBo2bCjfHiclJQWPHj1CZGQkUlJS5Nd9fX1x7tw5tG3bVom6ExEREZkko2+VAwD16tXD+fPnERISgiNHjuDq1auIi4sDABQvXhy1atVC27Zt4evrq0RxlA8y0tNhZm5e6PImIiIyBYoEWCq+vr4MoooIM3NzbB8wADEREYrm6+btjZ6//aZonkRERKZG0QCLipaYiAhEhYUVdDWIiIgKHaPGYP3888+4du2aUnUhIiIiKhKM6sH68MMPIUkSXF1d0bJlSwQEBMDf3x+1atVSqn5EREREhY7RpwiFEIiJicH27duxfft2AJkD29UDrjp16hhdUSIiIqLCwqgAKy4uDidOnEBgYCACAwMRHh6OjIwMPHnyBDt37sTOnTsBAMWKFYOfn58ccNWrV0+BqhMRERGZJqMCLBcXF3Tr1g3dunUDADx79gwnT56UA64rV64gIyMDcXFx2LVrl3zPQmdnZ7Rs2VIOwIiIiIiKEkWvInRxcUHXrl3RtWtXAEB8fLxGwHX58mVkZGTg2bNn8k2fiYiIiIoao2dy18fJyQllypRBmTJl4OHhATs7O0iSlJdFEhERERU4RXuwhBAICwuTe6yCgoKQkJAgLwMAW1tbNG3aFK1atVKyaCIiIiKTYVSAlTWgOnXqFOLj4+VlwP8HVAEBAQgICEDjxo1haWlpfM2JiIiITJRRAVaJEiUYUBERERFlYVSA9ezZM3lMVbdu3TBhwgQ0bdqUARURERG90RSZaBQA/vrrLwQGBmpMMOrj48NB7URERPTGMSrACgoKksdfBQcHIz4+Hrt378aePXsAZF5FqJpgNCAggAEXERERvRGMCrCaN2+O5s2b44svvkBaWhrOnz8vB1xnzpxBfHw89uzZg7179wLIHnDVr19fkY0gIiIiMiWKTdNgYWGBZs2aoVmzZpg6dSrS0tIQEhKSLeDau3cv9u7dC0mSkJaWplTxRERERCZD0XmwNDK2sEDTpk1Rt25d1K9fH7Vr18bq1auRkJAgj9siIiIiKooUD7CSk5Nx6tQpuefqwoULck+VemBlZ2endNFEREREJsHoAMvQgMre3l6eH8vf3x+NGjUytmgiIiIik2T0IPfQ0FCtAZWDgwOaNWsmB1S+vr6wsFC2wyw6Ohrnz5/H+fPnERISgpCQEDx58gQAMGTIEKxbty5X+R04cAArVqzA+fPnERMTAzc3NzRq1AgjRozAW2+9pWjdiYiIqOgyKuIJDg6Wnzs6OqJ58+ZyQNWwYUOYm5sbXUF93N3dFclHCIFRo0ZhxYoVGq8/fPgQO3bswI4dOzBixAgsX76c00wQERFRjowKsDp16iQHVA0aNICZmZlS9co1T09PeHt749ChQ7le98svv5SDKx8fH0yePBmVKlXCrVu3MHfuXISFhWHFihVwc3PDd999p3TViYiIqIgxKsBSTShaUL7++mv4+vrC19cX7u7uuHv3LipUqJCrPG7evIm5c+cCABo2bIiTJ0/C1tYWAODr64uuXbvC398foaGhmDNnDoYNG4ZKlSopvi1ERERUdBRcl5MCvv32W3Tp0sWoU4U//PCDPIZsyZIlcnClYmdnhyVLlgAA0tLSsGjRotcui4iIiN4MhTrAMpYQArt27QIAVK9eHU2aNNGarkmTJqhWrRoAYOfOnZzHi4iIiPR6owOsO3fu4OHDhwAAf39/vWlVyx88eIC7d+/mddWIiIioEHujA6yIiAj5efXq1fWmVV+uvh4RERFRVm90gHX//n35edmyZfWm9fT01LoeERERUVZ5di/CwiAxMVF+7uDgoDetvb29/Pz58+c606WkpCAlJUX+PyEhwYgaEhERUWH0RvdgvXz5Un5uZWWlN621tbX8/MWLFzrTff/993B2dpYf6j1fRFk5lCqFjPT0PMs/L/MmIiLd3ugeLBsbG/l5amqq3rTqvVJZp3JQN2XKFHzyySfy/wkJCQyySCcbFxeYmZtj+4ABiFF4bJ+btzd6/vabonkSEZFhDAqwevbsCUmSsHjx4hzHKhUmjo6O8nN9p/0AICkpSX6u73SitbW1Rm8XkSFiIiIQFRZW0NUgIiKFGHSKcOfOndi5c2e28URmZmawsLDAtWvX8qRyeU09WHzw4IHetOoD29kjRURERPrkagyWtgk2C/OkmzVq1JCfX79+XW9a9eXe3t55ViciIiIq/AwKsFSn0h4/fpynlclvFSpUgIeHBwDgxIkTetOePHkSAFCmTBl4eXnlddWIiIioEDMowFJNsrl48WKtY5UkSVK2VvlEkiR069YNQGYP1dmzZ7WmO3v2rNyD1a1bt0K7vURERJQ/DBrk/u677yIkJAR79uxB8eLF4e7uDktLS3l5+/btNf43hCRJuHXrVu5qmwfGjx+PlStXIi0tDWPHjsXJkyc1rhJ88eIFxo4dCwCwsLDA+PHjC6imREREVFgYFGCNHTsWp0+fxrZt25CWlibfvw/IHIOl/r+hlOgFOnXqFG7evCn/HxsbKz+/efMm1q1bp5F+6NCh2fKoWrUqJk2ahNmzZyM0NBTNmzfHZ599hkqVKuHWrVuYM2cOwv53ddenn36KKlWqGF1vIiIiKtoMCrDMzMywZcsWBAcH48iRI3j48CFSUlKwfv16SJKErl27wsXFJY+rmt2qVauwfv16rctOnz6N06dPa7ymLcACgJkzZyI6Ohpr1qxBWFgY+vXrly3N8OHD8d133xldZyIiIir6cjXRaNOmTdG0aVP5f1VwM3PmTI0r8gobMzMzrF69Gr169cKKFSsQEhKC2NhYuLq6wtfXFyNHjkTHjh0LuppERERUSBTqmdzXrVuX7TSgMTp16oROnToplh8RERG9mYwKsDIyMpSqBxEREVGR8Ubf7JmIiIgoLyh2ijA9PR07d+7EkSNHcPXqVcTFxQEAihcvjlq1aqFt27bo3r07zM3NlSqSiIiIyCQpEmAdOHAAI0aMyDZ9A5A5HcOZM2ewYsUKlC1bFitWrECHDh2UKJaIiIjIJBl9inDDhg3o0qULHj58CCEEhBAoX748mjRpgsaNG6N8+fIAMgOu+/fvo3Pnzvjtt9+MrjgRERGRqTIqwLp37x5GjBiBjIwM2NnZ4bvvvkNUVBRu376NM2fOIDg4GLdv30ZUVBRmzpwJBwcHZGRk4IMPPsB///2n1DYQERERmRSjAqzFixcjJSUFDg4OCAoKwtSpU1GyZMls6dzc3DBlyhQEBQXBwcEBKSkpWLx4sTFFExEREZksowKsQ4cOQZIkfPrpp6hXr16O6evWrYtJkyZBCIGDBw8aUzQRERGRyTIqwFKd5mvbtq3B67Rr105jXSIiIqKixqgAKz09HQByNfWCKi0nKSUiIqKiyqgAq0yZMgCAM2fOGLyOKq2Hh4cxRRMRERGZLKMCrFatWkEIgdmzZyMyMjLH9A8ePMDs2bMhSRJat25tTNFEREREJsuoAGvs2LEwMzNDTEwMGjdujK1bt8qnDdWlp6djy5YtaNq0KaKjo2FmZoaPPvrImKKJiIiITJZRM7nXqlULM2bMwBdffIHIyEj069cPLi4u8PHxgbu7OyRJQlRUFMLCwvDs2TN5dvcZM2agVq1aimwAERERkakx+lY5U6ZMgbOzMyZPnozk5GQ8ffoUx48f10ijCqzs7Owwb948jB492thiiYiIiEyWIvciHDNmDPr06YO1a9fqvdnzsGHD4OrqqkSRRERERCZLkQALAFxdXfHpp5/i008/VSpLIiIiokLJ6Js9ExEREZEmBlhERERECmOARURERKQwBlhUpGRomYeNiIgovyk2yJ3IFJiZm2P7gAGIiYhQLM8qHTui9cyZiuVHRERFHwMsKnJiIiIQFRamWH6u1asrlhcREb0ZeIqQiIiISGEMsIiIiIgUxgCLiIiISGFGjcGaPn06AKBx48bo0KGDIhUiIiIiKuyMCrC++eYbSJKEHTt2KFUfIiIiokLPqFOEJUqUAACUK1dOkcoQERERFQVGBViVK1cGAERFRSlSGSIiIqKiwKgAq2/fvhBCYMuWLUrVh4iIiKjQMyrAGjNmDOrWrYtff/0V69evV6pORERERIWaUYPco6KisGrVKgwfPhzvvfcefvvtN7z77ruoU6cOihUrBnNzc73rc+wWERERFUVGBVheXl6QJAkAIITA0aNHcfToUYPWlSQJaWlpxhRPREREZJKMvhehEELrcyIiIqI3lVEB1tq1a5WqBxEREVGRYVSANWTIEKXqQURERFRk8F6ERERERApjgEVERESkMKMHuatkZGQgMDAQwcHBiIqKQnJyMr777juULl1aTpOamoq0tDSYm5vD2tpaqaKJiIiITIoiAdbevXsxbtw43L17V+P1iRMnagRYq1evxkcffQQHBwdERkbC3t5eieKJiIiITIrRpwhXrVqFrl274s6dOxBCoESJEjqnaxg+fDhcXFzw/Plz7Nixw9iiiYiIiEySUQHWzZs38eGHHwIAWrdujWvXriE6OlpneisrK/Tq1QtCCBw6dMiYoomIiIhMllEB1qJFi/Dq1SvUrFkT+/btQ/Xq1XNcx8/PDwBw6dIlY4omIiIiMllGBVhHjx6FJEkYP348rKysDFqnUqVKAID//vvPmKKJiIiITJZRAdb9+/cBAPXq1TN4HdXA9uTkZGOKJiIiIjJZRgVY6jd6NlRMTAwAwMnJyZiiiYiIiEyWUQGWh4cHAODff/81eJ0TJ04AALy8vIwpmoiIiMhkGRVgtWzZEkIIbNq0yaD0sbGx+OWXXyBJElq3bm1M0UREREQmy6gAa8SIEQCAffv2Ye3atXrTPnjwAJ06dUJsbCzMzc3ldYmIiIiKGqMCLF9fX4waNQpCCLz//vvo3bs3tmzZIi+/cuUK/vjjDwwfPhzVqlXDhQsXIEkSJk6ciMqVKxtdeSIiIiJTZPStcpYsWYKkpCRs2LAB27dvx/bt2+XB7wMGDJDTqQbCDx06FLNmzTK2WCqkHEqVQkZ6OszMzQu6KkRERHnG6ADL3Nwc69evR9euXfH999/j4sWLWtPVqFEDX375Jfr162dskVSI2bi4wMzcHNsHDEBMRISieVfp2BGtZ85UNE8iIqLXocjNngGgV69e6NWrFyIjIxEaGoro6Gikp6ejRIkS8PHxkScYJQKAmIgIRIWFKZqnqwF3EiAiIsoPigVYKh4eHujatavS2RIREREVGkYNciciIiKi7BTtwXr16hUuXryIq1evIi4uDgBQvHhx1KpVC/Xr14elpaWSxRERERGZJEUCrOfPn2PGjBlYvXo1nj59qjVNsWLFMHz4cHz55ZdwdHRUolhFqK54zIm/vz8CAwPztjJERERUJBh9ijAiIgI1a9bE/PnzERcXByGE1kdcXBzmz5+P2rVr459//lGi7kREREQmyagerGfPnqFt27Z49OgRAKBWrVoYMmQIGjVqBHd3dwghEB0djZCQEKxfvx7h4eH477//0LZtW1y9ehXOzs6KbIQSRo8ejTFjxuhcbm9vn4+1ISIiosLMqABrzpw5ePToESRJwvTp0zF16tRsp9yqVasGPz8/TJgwAd9//z2+/PJLREZGYs6cOSY14WjJkiVRq1atgq4GERERFQFGnSLcuXMnJElC37598cUXX+gdzyRJEqZOnYq+fftCCIEdO3YYUzQRERGRyTIqwLp37x4AYMiQIQavM3ToUI11iYiIiIoaowIs1dWAJUuWNHgdVVoHBwdjiiYiIiIyWUYFWLVr1wYA3Lhxw+B1VGlV65qKrVu3olq1arC1tYWjoyOqVKmCIUOG4Pjx4wVdNSIiIipkjAqwRo4cCSEEFi1ahIyMjBzTZ2Rk4IcffoAkSRgxYoQxRSvu2rVr+Pfff/Hy5Us8f/4cN2/exK+//orWrVujR48eiI+PL+gqEhERUSFhVIDVu3dvDBs2DGfPnkX37t0RFRWlM+3jx4/Rs2dPnDt3DkOHDkXfvn2NKVoxdnZ26NevH1auXImgoCCEhYXh0KFD+OKLL1CiRAkAmYP5u3XrhlevXuWYX0pKChISEjQeRERE9GYxaJqGX3/9Vecyf39/XL16FXv27EHFihXRvn17+Pr6omTJkpAkCY8fP0ZISAgOHTqElJQU+Pr6wt/fH7/++isGDx6s2Ia8rocPH8LFxSXb6+3atcPYsWPRsWNHhIWF4cSJE/j5558xbtw4vfl9//33+Pbbb/OotkSmIyM9HWbm5oUubyKi/GBQgDV06NAcbykjSRJevnyJ3bt3Y/fu3dmWCyEgSRJCQ0MxbNgwSJJkEgGWtuBKxd3dHdu2bYO3tzdSU1OxZMmSHAOsKVOm4JNPPpH/T0hIgKenp1LVJTIZZubm2D5gAGIiIhTN183bGz1/+03RPImI8pvBE40KIYxOZ2gepqRixYpo164d9u7di5s3byIyMhIeHh4601tbW8Pa2jofa0hUcGIiIhAVFlbQ1SAiMjkGBVh37tzJ63qYtBo1amDv3r0AMk8p6guwiIiIiAwKsMqXL5/X9TBphbHnjYiIiAqOUVcRvimuXbsmP2fvFREREeWEAVYObt++jcOHDwPIHI9VpkyZAq4RERERmbo3OsDavXs30tLSdC5//Pgx3nnnHXn+qw8//DC/qkZERESFmMFXEeoTERGBFStWICgoCLdv30ZiYmKOM7tLkqQ3uMkPY8eOxatXr9CrVy80bdoUXl5esLW1RWxsLAIDA7F8+XI8efIEANCiRQsGWERERGQQowOs2bNn4+uvv0Z6enqhHAweGRmJJUuWYMmSJTrT9OrVC6tWreL0C0RERGQQowKsrVu3YurUqQAAMzMz+Pn5oW7dunBxcYGZmemffVy/fj1OnDiB4OBg3L59G7GxsUhISICDgwM8PT3RrFkzDBkyBE2bNi3oqhIREVEhYlSAtXjxYgBAmTJlsG/fPtSuXVuRSuUXf39/+Pv7F3Q1iIiIqIgxqpvpypUrkCQJM2bMKHTBFREREVFeMSrAsrS0BADUq1dPiboQERERFQlGBVhVq1YFAPlKOyIiIiIyMsAaMmQIhBDYuXOnQtUhIiIiKvyMCrCGDx8OPz8/rFixArt371aqTkRERESFmlFXEVpaWmLXrl0YMmQIevTogX79+qFPnz6oWrUq7Ozscly/XLlyxhRPREREZJKMnmjUxcUF48aNw9mzZ7F582Zs3rzZoPVMYSZ3IiIiorxg9Gyg48ePR/v27REbGwshRK4eREREREWRUT1YGzduxI8//ggAcHJyQo8ePVCnTp1CM5M7ERERUV4wKsBS3b+vevXqOH78ONzd3RWpFBEREVFhZlQ30/Xr1yFJEr755hsGV0RERET/o8hM7qoJR4mIiIjIyACrevXqAICoqChFKkNERERUFBgVYA0bNgxCCIOnZiAiIiJ6Exg9k3uXLl2wceNGLF26VKk6ERERERVqRl1FePLkSYwbNw4xMTH4+OOPsWnTJvTr18/gmdxbtmxpTPFEREREJsmoACsgIACSJMn/nzt3DufOnTNoXc7kTkREREWV0bfK4YzsRERERJqMCrCOHz+uVD2IiIiIigyjAix/f3+l6kGvISM9HWbm5gVdDaJCIy8/M/w8EpE6o08RUsExMzfH9gEDEBMRoWi+VTp2ROuZMxXNk8gU5NVnxs3bGz1/+03RPImocGOAVcjFREQgKixM0Txd/zeBLFFRlBefGSKirIyaB4uIiIiIsjOqB6t169avva4kSTh69KgxxRMRERGZJKMCrMDAQEiSpHeqBvV5soD/n9Yh6+tERERERYVRAVbLli1zDJSSkpJw48YNxMfHQ5IkVK1aFaVLlzamWCIiIiKTZnQPliGEENi7dy8+/vhjxMXFYdWqVWjRooUxRRMRERGZrHwZ5C5JErp06YJTp07B3NwcPXr0wMOHD/OjaCIiIqJ8l69XEZYuXRqffPIJnjx5grlz5+Zn0URERET5Jt+naVCdGty7d29+F01ERESUL/I9wLKysgIAREZG5nfRRERERPki3wOsU6dOAQDs7Ozyu2giIiKifJGvAVZwcDCmT58OSZLQqFGj/CyaiIiIKN8YNU3D9OnTc0yTkZGBp0+fIjQ0FOfOnUNGRgYkScKECROMKZqIiIjIZBkVYH3zzTe5mpFdCAELCwvMnTsX7dq1M6ZoIiIiIpNlVIAFQO9tcoDMObAcHR1RoUIF+Pv7Y8SIEahRo4axxRJRDhxKlUJGejrMzM0Luiq5UljrXVjl5XvNdqQ3mVEBVkZGhlL1ICKF2bi4wMzcHNsHDEBMRISieVfp2BGtZ85UNE+VvKp3Xta5MMurfcTN2xs9f/tN0TyJChOje7CIyLTFREQgKixM0Txdq1dXND9tlK53ftS5sMqLfYToTZfv0zQQERERFXUMsIiIiIgUpugpwsTERNy5cweJiYlIT0/PMX3Lli2VLJ6IiIjIJCgSYK1cuRI//fQTrly5YvA6kiQhLS1NieKJiIiITIpRAVZ6ejp69eqF3bt3A8h5ygYiIiKiN4FRAdby5cvx119/AQDc3d0xbNgwNGjQAMWLF4eZGYd3ERER0ZvJqADr119/BQDUqFEDQUFBKFasmCKVIiIiIirMjOpmioiIgCRJ+OqrrxhcEREREf2PIufxqlWrpkQ2REREREWCUQFWlSpVAABxcXGKVIaIiIioKDAqwOrXrx+EENizZ49S9SEiIiIq9IwKsMaNG4c6derg559/RlBQkFJ1IiIiIirUjAqwrK2tcejQITRo0ADt2rXD5MmTcenSJbx8+VKp+hEREREVOkZN02Bubi4/F0JgwYIFWLBggUHrciZ3IioqHEqVQkZ6OszUjolKysu8iQpKUf/MGBVgZZ25nTO5E9GbyMbFBWbm5tg+YABiIiIUzdvN2xs9f/tN0TyJTEFR/8wYFWBNmzZNqXoQERV6MRERiAoLK+hqEBUaRfkzwwCLiIiISGG8YSARERGRwhhgERERESmMAZaa//77D5MmTYK3tzfs7e1RvHhxNGrUCPPnz0dycnJBV4+IiIgKCaPGYBUle/fuxYABAxAfHy+/lpycjJCQEISEhGDVqlXYt28fKlasWIC1JCIiosKAPVgALl++jD59+iA+Ph4ODg6YOXMmzpw5g6NHj+KDDz4AAPzzzz/o3Lkznj9/XsC1JSIiIlPHHiwA48ePR3JyMiwsLHDo0CE0bdpUXta6dWtUqVIFkydPxvXr17Fw4UJ8/fXXBVhbIiIiMnVvfA9WSEgIAgMDAQDDhw/XCK5UJk6cCG9vbwDAokWL8OrVq/ysIhERERUyb3yAtXPnTvn5sGHDtKYxMzPD4MGDAQBPnz6VAzIiIiIibd74ACsoKAgAYG9vjwYNGuhM5+/vLz8/depUnteLiIiICq83PsCK+N89kCpXrgwLC91D0qpXr55tHSIiIiJt3ugA6+XLl4iNjQUAlC1bVm/aYsWKwd7eHgBw//79PK8bERERFV5v9FWEiYmJ8nMHB4cc09vb2yMpKUnvVA0pKSlISUmR/1fNq5WQkGBETfXUqXJlOKenK5qnlYcHEhISmHce58u88zfvwlhnIPMznlfHD/UyCmO9qfArbPueKl8hRM6JxRvsv//+EwAEADFo0KAc03t6egoAolKlSjrTTJs2Tc6TDz744IMPPvgoeo/79+/nGDO80T1YNjY28vPU1NQc06t6pmxtbXWmmTJlCj755BP5/4yMDMTFxaFEiRKQJMmI2mpKSEiAp6cn7t+/DycnJ8XyJeOwXUwP28T0sE1ME9slZ0IIJCYmwsPDI8e0b3SA5ejoKD83ZIb2pKQkAPpPJ1pbW8Pa2lrjNRcXl9eroAGcnJz4QTBBbBfTwzYxPWwT08R20c/Z2dmgdG/0IHcbGxu4uroCAB48eKA37dOnT+UAy9PTM8/rRkRERIXXGx1gAZBnaL958ybS0tJ0prt+/Xq2dYiIiIi0eeMDrBYtWgDIPP134cIFnelOnDghP2/evHme1ysn1tbWmDZtWrbTkVSw2C6mh21ietgmpontoixJCEOuNSy6zp8/j8aNGwMARo4cieXLl2dLk5GRgVq1aiEiIgIuLi6Ijo6GpaVlfleViIiICok3vgerUaNG8PPzAwCsXr0awcHB2dIsWLBAnr39448/ZnBFREREer3xPVgAEBYWhubNm+PFixdwcHDA1KlT0apVK7x48QK///47VqxYAQCoWrUqQkNDNa4+JCIiIsqKAdb/7N69GwMHDtQ5+2vVqlWxd+9eVK5cOZ9rRkRERIXNG3+KUOXtt9/GlStXMGHCBFStWhV2dnZwcXFBw4YNMWfOHISFhZlEcPXff/9h0qRJ8Pb2hr29PYoXL45GjRph/vz5SE5OLujqFRnR0dHYs2cPvv76a3Ts2BGurq6QJAmSJGHo0KG5zu/AgQPo2bMnypYtC2tra5QtWxY9e/bEgQMHlK98EXXx4kXMmjULHTt2hKenJ6ytreHg4ICqVati6NChCAoKylV+bBPjJSQk4Pfff8fEiRPh7++PypUrw9nZGVZWVihZsiQCAgIwd+5cPHnyxKD82CZ5a/LkyfJxTJIkBAYG5rgO28QIht1UhkzBnj17hLOzs86p+6tVqyZu3bpV0NUsEnS9xwDEkCFDDM4nIyNDjBgxQm9+I0aMEBkZGXm3MUVAy5YtDbp9xaBBg0RKSorevNgmyjl8+LBB7eLq6ioOHDigMx+2Sd67dOmSsLCw0HhPjx8/rjM928R4DLAKiUuXLgk7OzsBQDg4OIiZM2eKM2fOiKNHj4oPPvhA3uGrV68uEhMTC7q6hZ76QcTT01O0b9/+tQKsqVOnyuv5+PiIzZs3i/Pnz4vNmzcLHx8fedkXX3yRdxtTBFSqVEkAEB4eHuLjjz8W27ZtE+fPnxfBwcFi4cKFokyZMvJ72b9/f715sU2Uc/jwYeHp6SkGDx4sFi9eLLZv3y6Cg4PF6dOnxR9//CF69+4tzM3NBQBhZWUlLl++rDUftkneSk9PF76+vgKAKFmypEEBFtvEeAywComAgAABQFhYWIgzZ85kWz537lx5h//2228LoIZFy9dffy12794toqKihBBC3LlzJ9cB1o0bN+RfjA0bNhTJyckay5OSkkTDhg3ldr1586bSm1FkdO7cWfzxxx8iLS1N6/KYmBhRtWpVuY1OnjypNR3bRFm62kPdjh075Hbp2bNntuVsk7z3ww8/yD/Ap0yZkmOAxTZRBgOsQuD8+fPyB2LkyJFa06Snpwtvb28BQBQrVkykpqbmcy2LttcJsMaMGSOvExwcrDVNcHCwnOajjz5SsMZvnt27d8vv5bhx47SmYZsUjOrVq8unCrNim+St//77Tzg4OMgB1bRp03IMsNgmyuAg90Jg586d8vNhw4ZpTWNmZobBgwcDyLxvoiGDFynvCCGwa9cuAED16tXRpEkTremaNGmCatWqAchsZ8GLel9bQECA/PzWrVvZlrNNCo69vT0A4OXLlxqvs03y3pgxY/D8+XMMGTJE4zOiC9tEOQywCgHV1VH29vZo0KCBznT+/v7y81OnTuV5vUi3O3fu4OHDhwA020Ub1fIHDx7g7t27eV21Iis1NVV+bmaW/dDGNikYERERuHTpEoDML2x1bJO8tWXLFuzZswfFixfHvHnzDFqHbaIcBliFgGoW+cqVK8PCwkJnOvWDl2odKhjq73/WL5Ws2G7KUL9fqLb3nG2Sf5KTk3Hjxg0sXLgQrVq1Qnp6OoDMO2GoY5vknWfPnsnv95w5c+Dm5mbQemwT5ej+tiaT8PLlS8TGxgIAypYtqzdtsWLFYG9vj6SkJNy/fz8/qkc6qL//ObWbp6en1vXIcBkZGZg9e7b8f58+fbKlYZvkrXXr1ukcwgAAkyZNwoABAzReY5vkncmTJyMqKgrNmjXD8OHDDV6PbaIcBlgmLjExUX7u4OCQY3pVgPX8+fO8rBblIDftphqfAoDt9pp++OEHnD9/HgDQo0cPNGzYMFsatknBqFevHpYvX47GjRtnW8Y2yRunTp3CqlWrYGFhgeXLl0OSJIPXZZsoh6cITZz6oFArK6sc01tbWwMAXrx4kWd1opzlpt1UbQaw3V7HiRMn8PnnnwMASpYsiZ9//llrOrZJ3urevTvCw8MRHh6O8+fPY/PmzejRowcuXbqEAQMGYM+ePdnWYZsoLzU1FSNGjIAQAhMmTEDt2rVztT7bRDkMsEycjY2N/Fx9EK8uKSkpAABbW9s8qxPlLDftpmozgO2WW3///Td69OiBtLQ0WFtbY8uWLXB3d9ealm2St1xcXFCrVi3UqlULvr6+6NevH7Zv345ff/0Vt2/fRrdu3bBu3TqNddgmyps1axYiIiJQrlw5TJs2Ldfrs02UwwDLxDk6OsrPDemCTUpKAmDY6UTKO7lpN1WbAWy33Lhz5w7at2+Pp0+fwtzcHJs3b9Z71RPbpGAMGjQIvXv3RkZGBj766CM8ffpUXsY2Udb169fx/fffAwCWLFmicQrPUGwT5XAMlomzsbGBq6srYmNj8eDBA71pnz59Ku/w6oMPKf+pDw7Nqd3UB4ey3QwTGRmJtm3bIjIyEpIkYc2aNejRo4feddgmBadbt27YsmULkpKSsH//frz77rsA2CZK++GHH5CamoqKFSsiOTkZv//+e7Y0V69elZ8fO3YMUVFRAIC3334b9vb2bBMFMcAqBLy9vREUFISbN28iLS1N51QN169f11iHCk6NGjXk5+rtog3bLXdiY2PRrl073L59G0DmL3XVJLv6sE0KjvoUAffu3ZOfs02UpTpld/v2bfTv3z/H9DNmzJCf37lzB/b29mwTBfEUYSHQokULAJndsRcuXNCZTn0eoObNm+d5vUi3ChUqwMPDA4Bmu2hz8uRJAECZMmXg5eWV11Ur1OLj49GhQwdcu3YNADB79mx8+OGHBq3LNik4qokrAc1TSWwT08M2UQ4DrEKge/fu8vO1a9dqTZORkYFff/0VQOZg01atWuVH1UgHSZLQrVs3AJm/8s6ePas13dmzZ+Vfgd26dcvV5dRvmuTkZHTu3BkXL14EAHzxxRf47LPPDF6fbVJwtm7dKj9Xv6qNbaKsdevWQWTeY1jnQ33g+/Hjx+XXVQES20RB+XzvQ3pNfn5+8p3Lz5w5k2353Llz5RtvTps2Lf8rWMS9zs2e//nnH713pE9OTta4I/2///6bBzUvGlJSUkT79u3lNvj4449fKx+2ibLWrl0rXrx4oTfNwoUL5Xbz8vISr1690ljONslfhtzsmW2iDAZYhcTFixeFra2tACAcHBzErFmzRHBwsDh27JgYMWKE/IGpWrWqSEhIKOjqFnpBQUFi7dq18mPevHnye9y8eXONZWvXrtWZz+effy6v5+PjI37//XcREhIifv/9d+Hj4yMvmzJlSv5tXCHUs2dP+b1q3bq1uHLliggPD9f5+Oeff3TmxTZRTvny5UXx4sXFBx98INavXy9OnTolLl26JIKCgsRPP/0kmjdvLr+fVlZW4vDhw1rzYZvkH0MCLCHYJkpggFWI/PXXX8LJyUnesbM+qlatKm7cuFHQ1SwShgwZovN91vbQJT09Xbz33nt61x0+fLhIT0/Px60rfHLTFgBE+fLldebFNlFO+fLlDWqPsmXLikOHDunMh22SfwwNsNgmxmOAVcjcvXtXTJgwQVStWlXY2dkJFxcX0bBhQzFnzhyRlJRU0NUrMpQKsFT27t0runXrJjw8PISVlZXw8PAQ3bp1E/v27cuHrSn8lAywVNgmxrt586ZYvny56Nu3r6hTp45wd3cXFhYWwsHBQVSqVEn06tVLrF271uBjE9sk7xkaYKmwTV6fJIQQICIiIiLF8CpCIiIiIoUxwCIiIiJSGAMsIiIiIoUxwCIiIiJSGAMsIiIiIoUxwCIiIiJSGAMsIiIiIoUxwCIiIiJSGAMsIiIiIoUxwCIiIiJSGAMsIiIiIoUxwMoH69atgyRJkCQJd+/eLejqmJRvvvlGfm8Ki/T0dCxevBiNGjWCk5OTXP/u3bsbnAf3CcpvAQEBkCQJAQEBBV0VnYYOHQpJkuDl5VXQVSkylDrWPHnyBJMmTYK3tzdsbW3lPBctWgSgcB7L85pFQVeAqLDp378/tm7dWtDVICLKF/Hx8WjatClu3LhR0FUpVNiDRYoLDAyUf8kEBgYWdHUUdebMGTm46ty5Mw4fPowrV64gPDwcP/74YwHXjnTx8vKCJEkYOnRoQVdFUew1oPywbNkyObiaPHkygoKCEB4ejvDwcAwaNKiAa2e62INFlAtHjhwBAJibm2PTpk1wcnIq4BoRFR3r1q3DunXrCroalIXquNewYUPMmTOngGtTeLAHiygXHj58CABwd3dncEVEbwTVca9q1aoFXJPChQEWUS6kpKQAACwtLQu4JkRE+YPHvdfDAMuECCGwbds29OrVC56enrCxsUGxYsXQqFEjzJgxA8+ePdO5btarb549e4avv/4aNWvWhL29PVxcXNCyZUv89ttvBtXlr7/+QocOHeDq6go7OztUrVoVn376KaKiogBoH9Ny9+5dSJKEVq1aya+1atVKHiOieug7BfDy5UvMmzcP9evXh6OjIxwdHdGoUSMsXboUaWlpBtU9J+Hh4RgxYgSqVKkCOzs7ODo6ombNmpgwYYLOq2xUdV+/fj0A4N69e9m2Ky/ExMTgyy+/hI+PD1xcXGBjYwMvLy8MGjQIp06d0rtu1ja6fv06PvjgA3h5ecHa2hru7u7o0aMHzp49m2M9Xr16hcWLF8PX1xeOjo5wcXFBw4YN8cMPPyA1NVVu+5zaFwAOHz6MgQMHokKFCrC1tYWTkxPq1q2LyZMn49GjR3rXjYyMxOeff4769evD2dkZVlZWKFWqFGrXro3+/ftj3bp1SEhIkNOrrpy7d+8eAGD9+vXZ2i23V9VlHWOYkZGBlStXolmzZihevDjs7e1Rt25dzJo1Cy9evNCb19mzZ/Hll18iICAApUqVgpWVFZycnFCjRg2MHj0a165d07qe6sqwb7/9Vn4t63bldNXYw4cP8cknn6By5cqwtbVFiRIl0KFDB+zfv19vndPT07Fu3Tp06NBBrrOLiwuqVKmCNm3aYNasWTrrnZOcriJUbdc333wDAAgJCUH//v1RtmxZWFtbo0yZMhg0aBAiIiJeq3wgc2ylJElo2rSp1uWnTp2S6+Hi4oL09PRsaZ4+fQozMzNIkoRly5ZpzUfJz/aFCxcwdOhQVKhQAdbW1rk+Ht2/fx/Vq1eHJElwcHDA4cOHNfZzXZ+f17ki9e7du5gwYQJq1qwJR0dH2NnZoUqVKhg5ciTCw8O1rhMSEiKXefDgQa1p2rZtK6dZsGCB1jTjx4+HJEkoWbIkhBAay16+fIkff/wRAQEBcHV1haWlJYoXL47q1aujU6dO+OGHH3J/FaagPLd27VoBQAAQd+7c0ZomOjpaNG/eXE6n7eHu7i7Onj2rdf0hQ4YIAKJ8+fIiIiJCeHl56cznww8/1FnXjIwMMXLkSJ3rlipVSly8eFGUL19eABBDhgyR171z547e+qsea9euldeZNm2a/HpUVJSoW7euzvXefvttkZ6e/jpNIJs1a5YwMzPTWYa1tbVYv359tvUM2a7cMGSfOHjwoHByctJb5ocffqjzPVFvoz///FPY2dlpzcPc3Fz8/vvvOuv69OlT0ahRI511aNSokQgLC9PavuqeP38uevTooXd7HBwcxO7du7Wuf/LkyRzfDwAa6/v7++eY3t/fX+e2a3P8+HF53YMHD4q33npLZ97e3t4iMjJSaz7q+4Cuh7m5uVi2bNlrrZt131K9F/7+/iIoKEiUKFFC53rz5s3TWufExETh5+eXY7m9evXK1Xuqon4c00aV/7Rp08SSJUuEhYWF1vLt7OzEiRMnXqsOc+fOFQCEhYWFSExMzLZ8xowZGmWFhIRkS7Njxw55eXh4eLblSn62f/75Z63vg0pOx5rr168LT09PAUAUK1ZMBAcHCyE093NDPjvqx3Jd1q9fL6ytrfXu77Nmzcq2Xlpamvx+ffbZZ9mWp6SkaBzfOnfurLV81fdL1v0zMjJS1KhRI8ftnThxos5t04YBVj7IaQd//vy58Pb2FgCElZWVGDlypNi1a5e4ePGiCAoKEjNnzpQPhsWKFRN3797NlofqwOTm5iaqVKkiHB0dxZdffikCAwNFaGioWLlypShbtqxcjwMHDmit66xZs+Q0ZcuWFUuXLhXnzp0TJ0+eFF988YWwtbUVFStWFG5ubvIHXCU1NVWEh4eLNWvWyHmsWbNGhIeHazyePn0qr6P+oWzWrJmwsrIS48aNE4cPHxYXLlwQmzZtkt8bAGL58uWv3Q7Lli2T83FzcxPz588XwcHB4tSpU+Kbb74R9vb2AoCQJEns3btXY11V3bt16yYACA8Pj2zblRs57RNhYWHCyspKABCWlpZi/Pjx4vjx4+L8+fPil19+ERUqVJDXnzx5stYyVAdhHx8fYWNjIypUqCCWLl0qzp49K4KDg8U333wjbGxsBADh5OQkoqOjtebToUMHuaymTZuKzZs3i9DQULF//34xYMAAAUA0btxYTqMtwEpLSxOtWrWS39/+/fuLrVu3itDQUBEcHCwWL14sypUrJ38GQkNDNdZ/+fKl8PDwEACEo6OjmDx5sti/f7+4cOGCOHv2rPjjjz/E+PHjhaenp0aAdfv2bREeHi6v261bt2ztdvv2bcMbTmh+8fj6+goAon379mLHjh0iNDRU7NixQ7Rr105O4+PjI169epUtn5UrV4pixYqJIUOGiDVr1oigoCBx8eJFsWfPHjF9+nTh6uoqv19Hjx7VWPfp06ciPDxcjB49WuOLPOsjNTVVXkcVYFWtWlW4urqKkiVLitmzZ4tTp06J8+fPi4ULFwoXFxc5uLh69Wq2Ok+cOFEur0uXLmLz5s3i9OnT4sKFC+LAgQNizpw5ws/PT7zzzju5ek9VDA2wmjRpIiRJEnXr1hVr1qwRISEh4uTJk2LChAnyD6hy5cqJlJSUXNfh3Llzcjn79+/PtrxNmzY5BqMff/yxACBcXV1FRkaGxjIlP9s1atQQ5ubmwsvLSyxdulQ+nn3//fdyWn3HmtDQUHk/K126tMZx7Pnz5/J+pOvzo/7ZySnA2rNnj5AkSQCZP6SmTZsmgoKCRHBwsFiwYIFcDwDip59+yrZ+x44d5WNNVkFBQRpt4uzsLNLS0jTSxMXFyfvGkiVLNJb16tVLXnfgwIFi+/bt4uzZsyIkJETs2bNHfPvtt8LHx4cBlinK6cv0o48+kncKbb+GhBDi7t27onTp0vIOkJXqwARAuLi4aD043rhxQ/5C7dq1a7blkZGR8vKKFSuKx48fZ0tz+vRp+eCQNcBSUf8COn78uNbtUVH/UFpaWmpN/+TJE+Hu7i4AiDp16ujNT5fo6Gj5F46Hh4f477//sqW5ePGiHGSVKVNG48tJJacvAEPltE+ovrjNzc3FwYMHsy2Pi4uTf3GZmZlpbW/VQRiAaNCggXj27Fm2NBs3bpTTLFy4MNvy7du3y8u7deuW7aAlhBDz58/XOLhpC7BUaSwtLcW+ffu0vidxcXGiZs2aAoBo0aKFxrKjR4/K+evq4RJCiFevXon4+Phsr2vrcX1dWX/ZjxgxQmu64cOHy2mWLl2abfmDBw9EUlKSznKePXsm6tSpo/X9UDGk10BFvTevfPny4sGDB9nSBAUFyV+C48aNy7Zc1dORUwD15MmTHOujjaEBFgDRqVMnrQHUd999J6fZvn17ruuQlpYmHB0dBZC9tyQ1NVU+jnTt2lUA2ntLVD0lPXv2zLZM6c927dq1NX60ZqXrWHP8+HF5OytVqqT3h4Yhnx99+2JqaqooU6aMHFyFhYVlS6P+HWdnZydiYmI0ls+ZM0cO/rP2LKp6Fdu1aydsbW0FkL1nUb1X8cqVK/LrL168EJaWlgLIuYcqt/s1A6x8oO/LNCYmRg5qFi9erDefn376Sf6SynpgVg+wfvzxR5159OvXTwCZPWFZff/993Ieu3bt0pnHhAkT8iTA+uSTT3Sm+/zzz+V02gKFnKg+nADE5s2bdaZTPzhv2bIl2/L8CLDUf0GPHDlSZx6nTp2S040ZMybbcvWD8OXLl7XmkZGRIf867dGjR7blqt4rGxsbERUVpTOP+vXr6wywUlNT5QPnhAkTdG6PEELs27dPzufGjRvy67/99pv8urYAKid5FWC5u7vrDJISExPlnt4aNWq8Vlk7d+6Uy4qNjc22/HUDrL/++ktnuiZNmgggs+ctK9UXUU7HqtdlaIBlY2Oj9QegEEIkJCTIPwJz2t90UZ32zdpbovrMOTk5yftB1t4S9Z6SrMfivPhsnzx5Uu+2aDvW7Ny5U/7eqV27tnj06JHePIwNsP744w95mXrvWlbqP/jmzp2rsezs2bPysqw9i23bthVAZm9i69at5efqxo8fLwCIEiVKaPQqPnz40KDvvdfBQe4F7ODBg3j58iUAoE+fPnrTtmzZEkDmgOMLFy5oTSNJEt59912deTRo0ABA5iDMrIPmjx49CgAoUaIEOnfurDOPwYMH663n6xowYIDOZap6A8CdO3dynbdqHhcXFxf06tVLZ7r3338/2zr5Tb3c4cOH60zXvHlzeHt7Z1snq9q1a6NOnTpal0mSBB8fHwDA7du3NZalpaXh5MmTAIC33noL7u7uOvPQN9ng+fPn5cHrhu7jABAcHCw/L126tPx87dq1evPIT3369IGdnZ3WZQ4ODvL2Xrt2LccB/ElJSbh79y7+/vtvXL16FVevXtW4auvy5cuK1NnFxUXv51v1Wcu6PwD/3w5//PEHkpOTFanP62jXrh1KliypdZmjoyOqVKkCQPs2GEI1ePvChQt4/vy5/Lpq4mQ/Pz80a9YMtra2iI+PR1hYmJzm5MmTyMjIAAD4+/tr5Kv0Z9vT0xN+fn6GbdT/rF+/Hr169cLLly/RtGlTnDhxAqVKlcpVHrml2gZJkvDee+/pTNe7d284OztrrKPSoEEDODg4AIDGBNavXr3CmTNnAGS2m6rtsk5yrfq/ZcuWGhcBlChRAlZWVgCADRs2KHYxFcCrCAtcaGio/Lx06dJarwRSPWrVqiWnVV3Nl5WrqytKlCihs7zixYvLzxMTEzWWXb16FQBQr149mJub68yjdu3asLa21r9hr6F69eo6l+mrtyFU2+bj46P3UmN3d3f5CibVOvlNVa6VlZUc/OjSuHFjAMCNGzeQmpqqNY2+9xX4//c26/t669Yt+So49QBXm4YNG+pcpr6PN23aVO8+rjqAApr7eIsWLVCxYkUAmVcCNWrUCN9//z3OnDmjc7vzg6+vr97ljRo1kp9r259iY2MxdepUVKtWDY6OjqhQoQJq1aqF2rVro3bt2hqBUGxsrCJ1rlKlCszMdB/6de0PADBkyBAAmXc0qFChAj766CPs2LEDMTExitTNUK+7TxtKFRilpaVpXNGn+pIOCAiAlZWVfKWh+pe56nnx4sVRu3ZtjXyV/mzr+uGky+LFizFs2DCkp6ejQ4cOOHLkCIoVK5arPF6Haru9vLx0BsaA5vuS9fNiYWGB5s2bA9B8v8+fP4/k5GQ4OTnBx8dHDrCCgoLkKzyfPXuGK1euAPj/4FnF2toaffv2BQBs27YNlStXxuTJk7Fv3z7Ex8e/3gb/DwOsAhYdHf1a6+n69ajr17SK+oE16+XFT58+BQC9HwAgcxbzvPhQ6qu7vnobIi4uDgB09sKoU/2aU62T31TlFi9eHBYW+m+2oKqrEEJuv6wM3Sd07Q9AzvuEm5ubzmVK7OOWlpbYvXu3/Ks+JCQEU6dORfPmzeHi4oKOHTti06ZNr7VvGCOn90V9f8u6P124cAHVq1fH999/j3///TfbZeNZ5TTlg6EM3R9UvTDqvvrqK7z33nuQJAnR0dFYtmwZevbsCXd3d9SuXRvTpk3D48ePFamnPq+7TxuqYcOG2XpLsvaUqP/VFmBl7SkBlP9s5/Y4vGjRIggh4Obmhj///DPH91EpSh1/tfUsqvcqmpubo3HjxrC1tUVCQoLcs3jixAmdvYoAsHTpUrz99tsAMqfgmTdvHjp37owSJUqgUaNGmD9/vsb0L4birXIKmOoAYGVlpfO0nzZly5bNqyoVaYbMD5PTF11+KUx11Uf9Sy4wMFBvD6u6rMFLjRo1EB4ejt27d2P37t04ceKE3Mt24MABHDhwAAsXLsS+fftyDHyUklMb6Wqf1NRU9OnTB0+ePIGlpSXGjh2Lbt26oWrVqihWrJjcQ3z79m1UqlRJb175ydLSEqtXr8bEiROxefNmHDt2DKGhoUhNTZVPay5cuBAbN25Et27dCrq6r83CwgLNmjXDoUOH5C/wkJAQjZ4SANl6SxITE3X2lKhT6rOt70yDNr169cKff/6JmJgYDBw4EFu3bs0x0FOSsdudtWfxrbfewokTJwD8//ut6lk8duwYAgMD0bBhQzlNsWLFtPb6OTk54a+//sL58+exZcsWHD9+HJcvX0Z6ejpCQkIQEhKCefPmYefOnTrnR9OGAVYBU33ZpKamokSJEhpjTfJbsWLFEBUVlWOPQ3p6us5fVKaqePHiePTokc5Tq+pUv8DVT0vmJ1W5T548QVpamt4DoKqukiQp3quonl9O+4S+U0TqAZWVlZXGqe7cMjc3R/fu3dG9e3cAwKNHj7B//3789NNPuHDhAi5cuICRI0dix44dr11GbuTUW6P+vqnvT8eOHZPHBy1btgwffPCB1vVN9XNWo0YNzJgxAzNmzMCLFy9w+vRpbNq0Cb/++iueP3+O/v3749atWwV6PDNWQEAADh06JPeWZO0pAZCttyQyMlJvT0lBf7bnz5+P0qVLY+nSpdi5cyf69++PzZs353mQpdpuY4+/vr6+sLe3R1JSEgIDA9GmTZtsvYqq56oAa9KkSXp7FdU1atRIPq2fmJiIwMBArF27Fjt27EB0dDR69eqFW7duwdbW1qDt5inCAqZ+Hv7QoUMFWBOgZs2aAIBLly7p7VoPDw+Xb52gTV7Nam4M1Zd6WFgYXr16pTNddHS0PGuxMYGAMVTlpqamagye1eb8+fMAMsfVqAZqKqVSpUqwsbEBoDmOSht9y/NyHy9dujTee+89BAcHo379+gCAPXv2ZDudllf7ZEhIiMHL1fenv//+W37er18/nevn9L6bwmfN1tYWbdu2xZo1azBv3jwAmacz9+zZU8A1M07W3hL18VcqWcdhqdLo6ikxhc/2kiVLMHr0aACZY44GDhyY56fWVdt99+5dvT/WXr16Jb8v2o6/qp5FIPP9DgkJQVJSkkavIqDZs/jkyRP5ApHczDzv6OiIt99+G9u3b8e4ceMAZP6gy2mWfXUMsApYx44d5UHXP/zwg6JXMORWmzZtAGT+utq7d6/OdL/++qvefFRfygD0BmL5qW3btgAyBzv++eefOtOtXr1a7qJWrZPf1MtdvXq1znTBwcHyLUnyoq4WFhbyVX0HDx7U2VsjhMCGDRt05tOiRQv51+jy5ctfayxDTiwtLTW+ELNeIavaJ5XeH7du3apzbFRSUhK2bNkCILPHR703R/1zrms8ZUZGBlasWKG3fFP7rKmOIYByg/ILiqq3BMi8vZO2nhL1/9UDLD8/P60XEpjKZ3vZsmUYMWIEgMwrQgcPHqx1zJ1SVNsghMCaNWt0ptu2bZs8sFzXdquPw1IF8eq9ioBmz+LixYv19ioa4nX3awZYBaxMmTIYNmwYgMzLsEeOHKk3yIqOjsaqVavypC5DhgyRx35MmDBB62mf4OBgnffWUlH/Irl165aylXxNw4YNkwd0Tpw4Effv38+W5vLly5g1axaAzHZRnYbKb40aNZKvTlu1ahUOHz6cLU18fDxGjhwJIHNAr+oXqdJUZbx8+RIjR47U+kt34cKFuHjxos48bGxsMGnSJACZpwj69euHpKQknekTExOxdOlSjdeCgoJw8+ZNneukpqbK4ywcHByyDbpX7ZNK749RUVGYOHGi1mWffPKJ/Gs9a/uophEAIN/fMqspU6bofV+B/P2sxcXF4a+//tI7Rka9h7JChQp5Wp+8ZmlpKfdOrV69WmtPCfD/X/gnTpzIsafEVD7bkiRh+fLl8lQRmzZtwtChQ/MsyOrRowc8PDwAALNmzdI65cj9+/fl44SdnZ38vZiV+g+pn376CUD291u9Z/HHH38EkDk9Sd26dbPld/v2bfnYocvr7tccg2UCFixYgDNnzuDq1atYs2YNzp49ixEjRsjzfjx79gx///03jhw5gn379qF27doa8zUpxcPDA9OmTcPUqVNx+/ZtNGjQAJ9//jl8fX2RkpKCgwcPYsGCBfDw8EBSUhJiYmK0nqIoV64cypYtiwcPHmD+/PkoU6YMqlWrJp/nd3d3h6Ojo+L118fNzQ3z5s3Dhx9+iMjISDRs2BCff/45mjVrhvT0dBw5cgTz5s3D8+fPIUkSVqxYUaB3jl+xYgUaN26M1NRUdO7cGWPHjsXbb78NBwcHhIWFYfbs2fIYnkmTJuXZ6cyePXuiffv2OHToEHbt2gU/Pz+MHz8elStXRkxMDDZu3IiNGzeiUaNG8ikNbfvE5MmTcfToURw9ehT79+9HjRo1MGrUKDRt2hQuLi5ITEzEP//8g8DAQOzcuRM2Njb46KOP5PWPHj2KGTNmwM/PD507d0adOnXg5uaGFy9e4N9//8Xy5cvlYOT999/PNqakWbNmOH78OEJCQjB79mx07NhR7p2wtbVFmTJlXuv9adiwIX7++WfcuXMHo0aNgqenJ+7fv4+ff/5Zvimtj48PRo0apbFehw4dULJkSURHR+OLL77AvXv30LVrV7i6uuLmzZtYuXIljh49iubNm+P06dM6y1edLgEyfxR98cUX8nQvQOZl8UqNr0lISEC3bt3g5eWFnj17onHjxihfvjwsLCzw6NEj7N69W/7xV7ZsWfmqrMIsICAAR44ckXtVsvaUAP/fW6I+X5a+nhJT+WxLkoSVK1ciIyMDa9euxYYNG2BhYYHVq1crfurZ0tISK1aswNtvv43ExES0aNECn376Kdq0aQMLCwucOXMGs2fPln+QzJ8/H66urlrzatSoEezs7JCcnCy3i7aAVjUOS73ttPUq/vfff2jVqhVq1KiBHj16oGHDhvLx4P79+/jjjz/knmgfHx95+gyDKDptKWllyI19nzx5ovemseqPVq1aZVvf0BnGc6pLTjd7dnV1FSEhIfItM0aNGqW1HNWs89oeum72rE9uZofXZ+bMma91s2eV/LpVjhDK3hBWn5y2KaebPfv4+IjQ0FD5f103jk5OThaDBw82aB+vUKGCxrrq+4m+R8+ePcWLFy+ylf3gwQNRvHhxresYe7Pn9u3b66xP9erVxcOHD7Xmc+DAAXk2bW2PgIAAcfXqVa2fG3V9+vTRmYeumz3ro+szaejN3MuUKSMuXryYm7dUlpubPetj6LbmJOs97nTdBFs1eziQObN7Tjelz6/PthA5H2vS09M1Ppfvv/9+tvsnGjuTu8q6dete62bPWanfD1LbfQeFyLw5vHreCxYs0JqXITe1BjJv3K7rWK0LTxGaiOLFi2P//v04evQohg0bhipVqsDBwQEWFhYoXrw4fH198eGHH2Lfvn1au5WVouo63rVrF9q3b4/ixYvDxsYGlStXxrhx4xAWFoaGDRvK42hUs+5mNXr0aPz5559o3749SpYsma+XAuszdepUhIWF4YMPPkClSpVga2sLe3t7eHt74+OPP8b169fzbKb63Grfvj1u3ryJqVOnol69enBycoK1tTXKlSuHAQMGICgoCEuXLtU7aaQSXFxccOrUKSxatEjuVXV0dES9evXkyT7Vf9Xr2idsbW2xfv16hIaGYvTo0ahZsyacnZ1hYWEBFxcX1KtXD8OHD8e2bdsQERGhsa5q4r8JEyagSZMmKFeuHGxsbGBjYwMvLy/07dsXe/fuxZ9//qkxLkmlTJkyOH/+PIYPH47KlStrTfM6rKys5KsYmzRpAhcXF9jZ2aF27dr47rvvcPHiRfnUSFYdOnRAaGgoBg4cCA8PD1haWsLNzQ3+/v5YsWIFjh49Kvey6bNx40bMnTsXjRo1grOzc57tD+XLl8elS5cwb948dOzYEdWqVYOLiwssLCzg6uoKf39/zJ8/HxERETlOollYqHpLVHSd+mvVqpX8XFdPiTpT+WwDmach165dK99JY9WqVRg9enSeTAsyZMgQXL9+HR9//DG8vb1hb28PW1tbVKpUCR988AHCwsIwZcqUHPNRb4cWLVpona6icePGGm2nq1fRz88PwcHBmD59Olq3bo3KlSvD0dERlpaWcHd3R/v27fHLL7/g0qVL8iTUhpJEXryLVKQ9ePAAnp6eADI/jPpu+UBvho0bN8q3y7l586Y8d1NRFBgYKH+hHj9+PFdXJhHRm4M9WJRrmzdvlp83adKkAGtCpkK1T7i5ucm3tCEiepMxwCINSUlJem9KGxYWhhkzZgDIvD+dau4sKroePnyo9zYtq1evxr59+wBk3gjcFOZmIiIqaKYxMIZMRkxMDLy9vdG9e3e89dZbqFatGqytrREZGYkDBw5g9erVePHiBSRJwsKFCwu6upQPDh8+jMmTJ6Nfv34ICAhA+fLlkZGRgVu3buGPP/7Azp07AWReHfr5558XbGWJiEwEAyzK5uXLl/j999/x+++/a11uZWWFlStXypNQUtEXExODJUuWYMmSJVqXly5dGnv37tV5aTUR0ZuGARZpKFOmDP744w/s378foaGhiI6OxtOnT2FnZwcvLy+0bdsWY8eORfny5Qu6qpRPunTpgp9//hkHDhxAREQEYmJikJiYCBcXF3h7e+Ptt9/GqFGj8n1uMyIiU8arCImIiIgUxkHuRERERApjgEVERESkMAZYRERERApjgEVERESkMAZYRERERApjgEVERESkMAZYRERERApjgEVERESkMAZYRERERAr7P0Lp6AYuuP0VAAAAAElFTkSuQmCC",
      "text/plain": [
       "<Figure size 640x480 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "import csv\n",
    "import matplotlib.pyplot as plt\n",
    "directory_plots = 'plots'\n",
    "\n",
    "df_longest_paths = pd.read_csv('./results/longest_paths_label.csv')\n",
    "\n",
    "longest_path_lengths = df_longest_paths['longest_path_length']\n",
    "\n",
    "plt.rc('font', size=20)\n",
    "plt.hist(longest_path_lengths, bins=20, color='maroon', edgecolor='white')\n",
    "plt.title('Longest paths in DAGs')\n",
    "plt.xlabel('length of longest paths in workflows')\n",
    "plt.ylabel('number of workflows')\n",
    "plt.tight_layout()\n",
    "\n",
    "plt.savefig(f'{directory_plots}/longest_path.png', format=\"png\", bbox_inches='tight')\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Find shortest path"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Taking the first node as source and last node as target of longest path, calculate the shortest path between the source and target. As some dot-files have no process inside, e.g. gwas.dot exception handling is required\n",
    "\n",
    "2nd solution idea: calculate shortest path, considering all nodes with no out-degree"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import csv\n",
    "import networkx as nx\n",
    "\n",
    "def shortest_path_processes(dot_file):\n",
    "    \n",
    "    # first code is similar to calculating the longest path, so we have just the DAG with out processes\n",
    "    G = nx.nx_agraph.read_dot(dot_file)\n",
    "    \n",
    "    # we want to delete the operator and channel nodes, but keep the edges \n",
    "    nodes_to_delete = [node for node, attr in G.nodes(data=True) if 'shape' in attr]\n",
    "\n",
    "    node_labels = {}\n",
    "\n",
    "    for node, attr in G.nodes(data=True):\n",
    "        node_labels[node] = attr.get('label', '')  # get the label of the node, otherwise empty string\n",
    "\n",
    "    for node in nodes_to_delete:\n",
    "        # connect in and out nodes with parent and child\n",
    "        for in_src, _ in G.in_edges(node):\n",
    "            for _, out_dst in G.out_edges(node):\n",
    "                G.add_edge(in_src, out_dst)\n",
    "\n",
    "        G.remove_node(node)\n",
    "    \n",
    "    if len(G.nodes()) == 0:\n",
    "        return []\n",
    "    \n",
    "    longest_path = nx.dag_longest_path(G)\n",
    "    \n",
    "    # if longest path empty, return empty list\n",
    "    if len(longest_path) == 0:\n",
    "        return []\n",
    "    \n",
    "    source_node = longest_path[0]\n",
    "    \n",
    "    # get all nodes with no outgoing edges\n",
    "    target_nodes = [node for node in G.nodes() if G.out_degree(node) == 0]\n",
    "\n",
    "    shortest_shortest_path_length = float('inf')\n",
    "    shortest_shortest_path = None\n",
    "\n",
    "    # calculate shortest path for each target_node and take the shortest one\n",
    "    for target_node in target_nodes:\n",
    "        try:\n",
    "            shortest_path = nx.shortest_path(G, source=source_node, target=target_node)\n",
    "            shortest_path_length = nx.shortest_path_length(G, source=source_node, target=target_node)\n",
    "            if shortest_path_length < shortest_shortest_path_length:\n",
    "                shortest_shortest_path_length = shortest_path_length\n",
    "                shortest_shortest_path = shortest_path\n",
    "        except nx.NetworkXNoPath:\n",
    "            pass\n",
    "        \n",
    "    shortest_path_labels = [node_labels[node] for node in shortest_shortest_path]\n",
    "    \n",
    "    return shortest_path_labels\n",
    "\n",
    "def shortest_path_to_csv(directory):\n",
    "    csv_data = []\n",
    "    \n",
    "    for filename in os.listdir(directory):\n",
    "        if filename.endswith('.dot'):\n",
    "            dot_file = os.path.join(directory, filename)\n",
    "            pipeline = filename.split('.')[0]\n",
    "            shortest_path_labels = shortest_path_processes(dot_file)\n",
    "            \n",
    "            if len(shortest_path_labels) != 0:\n",
    "                length = len(shortest_path_labels) -1\n",
    "            csv_data.append((pipeline, length, shortest_path_labels))\n",
    "            \n",
    "    with open('./results/shortest_path_label.csv', mode='w', newline='') as file:\n",
    "        writer = csv.writer(file)\n",
    "        writer.writerow(['pipeline', \"shortest_path_length\", \"shortest_path_label\"])\n",
    "        writer.writerows(csv_data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "directory_path = './dags'\n",
    "shortest_path_to_csv(directory_path)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "\n",
    "df_shortest_paths = pd.read_csv('./results/shortest_path_label.csv')\n",
    "\n",
    "pd.set_option('display.max_colwidth', None)\n",
    "pd.set_option('display.max_rows', None)\n",
    "\n",
    "df_shortest_paths"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAlgAAAG4CAYAAACZypnDAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjcuMiwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy8pXeV/AAAACXBIWXMAAA9hAAAPYQGoP6dpAAB2KUlEQVR4nO3deXhM1/8H8PfNHlllJYTYYxfEHglFbUXEWrW3aNXWlpZWaZVS1NpW1VpbS2tfaok11oiUpIIiFBFBNhESSc7vj/zmfmeSyWSSuVm9X88zT+7knnvOuTN37nzm3HPPkYQQAkRERESkGKOirgARERFRacMAi4iIiEhhDLCIiIiIFMYAi4iIiEhhDLCIiIiIFMYAi4iIiEhhDLCIiIiIFMYAi4iIiEhhDLCIiIiIFMYAi0oEPz8/SJIESZJw/Pjxoq4OkaKOHz8uH99+fn5FXZ18U+2DJElFXRWiImdS1BWg0u3kyZP47bffcOHCBdy9excJCQkwNjaGjY0N3N3dUbNmTTRp0gQ+Pj7w9vaGkRFjfiIqOB4eHrh7967WdZaWlrCzs4OtrS3c3NzQuHFjNGnSBB06dICLi4vBZQ8YMAC///67/HzhwoX46KOPDMrz1atXCAwMRGBgIIKCghAdHY0nT54gNTUVdnZ2cHJyQoMGDdCkSRO89dZb8PT0NHQ3SF+CqABcu3ZNtGrVSgDQ+1G5cuUc8/P19ZXTHTt2rND2ozAcO3ZM3jdfX9+iro5B1q5dK+/L0KFDi7o6RSIyMlKvY1pdaTkG1D/PxVXlypXzdF4CIExNTUXfvn3FmTNn8l1ufHy8sLCw0Mi3QYMG+c4vLS1NrFy5Unh4eORpXxo0aCA2btwo0tPT81026YctWKS4K1euoF27doiNjZX/5+zsDG9vb5QrVw5GRkaIjY3F1atXcePGDWRkZAAA4uPji6jGRPQ6euONNzRadNLT0xEfH4+4uDhcuXIFDx8+BJDZSrRt2zb8+eefmDRpEmbPng1zc/M8lbV161a8fPlS439XrlzB33//jUaNGuUpr/j4ePTr1w+HDx/W+L+dnR28vb3h7OwMGxsbPH36FNHR0QgJCZHLvnLlCt555x1cuHABS5YsyVO5lDcMsEhRr169woABA+TgytXVFcuXL4e/vz+MjY2zpY+NjcXu3buxceNGhISEFHZ1iUhBQoiirkKevPPOOxg2bFiO6yMjI7F69WqsWLECT58+RUZGBhYuXIiIiAjs2bMnT10a1q9fLy9bWFjIAc/69evzFGDFx8ejZcuWuHbtmvy/jh074osvvkDr1q21nmdfvnyJU6dO4aeffsKuXbuQkZGB58+f610m5Q87vJCidu3ahYiICACZJ5Fjx46hT58+Wj/0AODg4IBhw4bhyJEjuHjxYmFWlYhIpypVquCbb77BP//8o3Hzwf79+zFlyhS987l16xZOnz4NADAyMsK8efPkdZs3b0ZaWppe+QghMGTIEDm4MjY2xo8//ohDhw6hbdu2OZ5nLSws0LFjR2zfvh1Xr15F27Zt9a475R8DLFLUoUOH5OUePXqgdu3aem9brVq1gqgSEZFBXF1d8ddff6Fx48by/xYvXoxbt27ptb1665Wfnx/GjBkDBwcHAEBMTAwOHDigVz4bN27Enj175OcLFizA+++/r9e2KrVq1cKxY8fw4Ycf5mk7yjsGWKSo+/fvy8tVqlQp0LJiY2Mxb948eHt7w8nJCZaWlqhatSpGjhyJ8PDwPOX15MkTzJ07F76+vihfvjzMzc3h5OQELy8vTJ48GVevXs01jzt37si3qHt4eMj/DwoKwrvvvgtPT0/Y2dlBkiRMnDgRM2fOhCRJaNeunZz2xIkTGre6a8tPm+DgYEyaNAmNGjWCs7MzzMzMUK5cOfj6+mLevHmIi4vT+3VYsGABOnToADc3N1hYWKBMmTKoXLkymjRpgoEDB2Lt2rV48OCBxnbDhg2DJEkYPny4/L/169dr3RdDhiHw8PCQ87lz5w4A4Pr16/joo49Qt25d+Q6w+vXrY9q0aYiKitIr3xcvXmDnzp0YP3482rRpA1dXV5iZmcHa2hoeHh7o3bs31qxZg9TU1BzzWLduHSRJ0jju7969q/U10HcYg6NHj2LAgAGoWrUqLCws4OjoiLZt22L58uV49eqVXnlcu3YNU6ZMQYsWLeDk5AQzMzPY2dmhevXqaNGiBT744APs3LkTiYmJeuWniz77p+09vH//PqZPn46GDRvC3t4eVlZW8PT0xLhx43K846+wmZubY/PmzfJlwfT0dMydOzfX7YQQ2LBhg/x88ODBMDMzQ79+/eT/qQdguvJRL69Vq1aYMGFCXnZBZmRklOtlyXv37uGrr75C27Zt4erqCnNzc9jY2KBKlSpo1qwZhg8fji1btuDJkyf5qsNroWj72FNp07VrV/lulX79+imWb9a7CIOCgkSFChVyvFPG2NhYrFy5Uq+8V69eLezs7HTeeWNsbCwmTpwo0tLScswn691jKSkpYsyYMVrzmzBhgpgxY4bBd1jGxsaKgICAXLe3t7cX27Zt0/k67Ny5U5QtW1av+lSoUEFj26FDh+q9L4bcJad+B1hkZKT45ZdfhLm5eY5l2dnZ5brf586dE9bW1nrV3cPDQ1y6dElrPup3UOrzUJf1LsKUlBQxatQonds3btxYPH78WOe+zZgxQ5iYmOhVn0GDBuXtzdAip/1Tl/U93LFjh87Pn6Wlpdi7d6/BddNW/tq1a/O8/VtvvaXxucrtbjz199bS0lIkJiYKIYQ4ffq0/H8zMzPx9OlTnfkcP35c43XZunVrnuuurxUrVghLS0u9jpvWrVsXWD1KOnZyJ0VVr15dXt67dy8iIiLydJlQH+Hh4Zg6dSqSkpLg4uICHx8fODo64sGDBzh69ChevHiB9PR0jBkzBvXq1UPLli1zzGvBggWYPHmy/Nzc3By+vr6oVKkS4uLicOzYMcTGxiI9PR2LFy/G3bt38eeff+rVAjFp0iSsWLECAFC/fn00bNgQpqamuHHjBoyMjNCsWTOMHTsWDx48wM6dOwEAbm5u8Pf3z5aXo6Njtv9FR0ejffv2cp83AKhduzYaNWoEGxsbxMTEICgoCE+ePJHvOtqwYQMGDRqULa+LFy+iT58+cl8QS0tLtGjRAh4eHjA3N0diYiJu3bqFsLAwJCcnZ9u+Q4cOsLa2xrVr1xAYGAgA8PT0xBtvvJEtbY0aNXJ97fSxe/du+Rd8+fLl0bZtW1hbW+Pff//F6dOnkZ6ejoSEBAwcOBBmZmbo0aOH1nzi4uKQlJQEAHBxcUHdunVRsWJFWFlZITk5GTdv3sSFCxeQlpaGO3fuwNfXF5cuXdI41oHM137s2LF49uwZfv31VwCAjY0NhgwZkud9Gz16NNatWwcjIyM0b94cnp6eyMjIwLlz53D9+nUAwKVLlzBkyBDs379fax6LFy/GV199JT93cnJCixYtUL58eUiShNjYWFy7dg0RERFIT0/Pcx2VEBgYiNGjRyM9PR2VKlVCy5YtYWtri8jISBw/fhxpaWl48eIF+vXrh/Dw8AJvFddH37595ct08fHxCAsLQ8OGDXNMr9461atXL9jY2ADIbIGqXr06bt68idTUVPz222/44IMPcszn2LFj8rKFhQV69uxp6K5otXPnTowZM0Z+bmtri5YtW6JixYowMTFBQkICbty4gfDwcJ0tugQdPzOI8kH91xoA4ezsLL7//nsRFRVlUL7qLVjm5ubC2NhYLFy4ULx69Uoj3X///Sfq1asnp23Xrl2OeZ45c0YYGxvLaTt37iwePnyokebly5di8uTJGvu0cOFCrfmpt2Cp8nV3dxcnT57Mlvbly5fycn7GQEpPTxft2rXTaM0IDg7Olu7Fixdi5syZQpIkAUBYWVmJ27dvZ0vXs2dPOa+AgAARGxurtdyXL1+Kffv2idGjR2tdX9DjYKm3PpiZmQkjIyMxf/78bK0I165dEw0bNpTTOjk5iZiYGK15njt3TkybNk2EhYXlWO6jR4/E4MGD5fzeeOONHNMaOg6WqkXO29tbREREaKTLyMgQixcv1jgeT5w4kS2/V69eCUdHRznNt99+K1JTU7WW/fTpU7FmzRoxb948veqqi3q9cqL+HpqbmwsrKyuxYcMGkZGRoZEuPDxco5V6+PDhBtcva/n5acG6fv26xn7+/PPPOaZ9/vy5sLGxkdPu27dPY/3MmTPldc2bN9dZbvv27eW0LVq0yHO99aX+ufnwww/F8+fPtaZ79uyZ2Lp1q/j0008LrC4lHQMsUlyvXr2yNSNLkiRq1aolBg8eLJYsWSLOnz+fLTjSRT3Ayu2kFhYWJgcUkiTlGNy1bdtW44SVkpKSY57jx4+X09ra2srN/OrUv1gBiDJlyojr16/num/5CbB+/fVXeZtGjRqJZ8+e6Uz/5ZdfyunHjBmTbb3qy9jc3DzXvHQpzAALgJg7d26OaWNiYkT58uXltJMnTza4/C5dusj5Xb16VWsaQwMsAKJGjRo634c+ffrofD/DwsKK5BJOXgMsSZLEgQMHcky7d+9eOa21tXWezhn6lJ+fACsjI0MYGRnJeXz99dc5pt2wYYOczsXFJVv9b926pfGaZQ2o1VWrVk3xYDOrZ8+eyWW4u7tnC3opb9jJnRS3ceNG9O3bV+N/Qghcv34dGzZswIQJE9C8eXOULVsWb7/9Nk6cOJGn/OvXr49Ro0bluL5evXrw9vaWy9U2vlZERAROnjwpP//hhx9gZmaWY55z5syBk5MTACAxMRGbN2/OtZ4ffvghatasmWu6/Pj+++/l5WXLlsHa2lpn+qlTp8Le3h4AsGXLFnlwVxVVB+cyZcrkmldxUbVqVXz88cc5rnd2dsbXX38tP1+7dq3et8PnRH3MpCNHjhiUly5z587V+T6MGDFCXg4ODs62Xr3DurOzs7KVU1D37t3RuXPnHNd37doV5cqVAwAkJSVpjP1UVCRJki/zAdB5A4n65cGBAwfCxESzV07VqlXRunVr+bnq0rI26gM3qz7Lupw/fx4ffvihzse///6rsY36cePo6Mg5JQ3EPlikOCsrK2zduhWHDx/G999/jyNHjmj9YktKSsKWLVuwZcsW9OjRA+vWrUPZsmVzzT9r8KaNl5cXLly4AADynUrq1PszNGzYUOP2a22srKwwcOBALFu2TN5+9OjROrcZMGBArvXMj4cPH+Lvv/8GAFSoUAFt2rTJdRsLCwu0bNkSBw4cQEJCAsLDw9GgQQN5faVKlXDr1i3ExcVh8+bNePvttwuk7kp6++23s31hZTVgwAB8+OGHSElJwZMnTxAeHq7z7qnk5GScO3cOYWFhePz4MZ49e6bRP0n97knVe6A0CwsLdO/eXWcaLy8veVnb8V2pUiV5+ejRowXSF1IJuX2WJUlCw4YNER0dDSBzX+vVq1cYVdPJ2toaCQkJAIBnz55pTXP//n0cPXpUfv7OO+9oTTd48GB5jKwNGzbgm2++0TqAqXo5VlZWudYxIiICP/zwg840ffr00egT6ezsDEtLS7x48QJhYWE4efIkx8wyAAMsKjAdO3ZEx44d8fTpU5w4cQJnzpzBpUuXcOnSJfnkpLJ79274+Pjg7NmzGr8Otalfv36uZat3Cs9aFgCEhobKy+q/IHVp3bq1HGBdunRJZ1pTU1O96pkfZ8+elZeFEHqPZ6M+Zs+9e/c0Aqz+/ftjzpw5ADK/CH777Tf0798f7du3R/ny5RWqubJatGiRaxpra2vUq1dPbsUMDQ3VGmDFxsbiyy+/xK+//prjF2ZWBXV7eq1atXS2pgK5H98VK1ZEq1atcObMGSQmJqJp06YYNGgQ/P390aZNm1w/Y4VFic9yUVA/RmxtbbWm2bBhg9xS7OnpiaZNm2pN169fP0yYMAEpKSlyUNahQ4ds6WxsbOTWsoIahd3U1BT+/v7YvHkz0tPT0aFDB/Tp0wd9+vSBr6+v1pttKGcMsKjAOTo6onfv3ujduzcAICMjAyEhIdi4cSN++eUXvHjxAgDwzz//4PPPP8fSpUt15mdnZ5drmaampvKytvGCHj9+LC9XrlxZr/1QH4sqty/XsmXL5tq6kl/qYztFRUXl+itVm6yXNT7//HOcOHECp0+fhhACe/bske+UqlKlCnx8fNChQwf07Nkzxy+UwqbeSqOLu7u7HGCpv+8qd+/eRdu2bfHff//lqXx9A7G8yuvxndNlzzVr1qBdu3Z4+PAhkpOT8csvv+CXX36BsbEx6tevj7Zt26JLly7o0KFDgR2ruVHis1zYMjIyNN571YChWalf7hs8eHCO+ZUtWxbdu3fHn3/+CSDzsqK2AMvBwUH+3Oozb+uwYcO0TgOU22W/RYsW4dKlS7h27RpevXolX2WQJAmenp7w8fFBp06d0K1bN1hYWORaj9cZ+2BRoTMyMoK3tzeWLFmCkJAQuY8FAI2AKydK9AtQ3ZYP6NfcnjVdbl+ulpaW+auYHpT4FZ/1S7lMmTI4duwYFi1alG1E/cjISPz6668YMmQIypcvj8mTJ+f6HhWGMmXK6JUut/dt0KBBcnBla2uLjz/+GAcPHkRkZCSSkpKQnp4OkXlDkMal5az92JSiVL+XWrVq4fLly5g0aZJGEJCeno6///4bS5cuRZcuXVC5cmX88ssvipSZVyWxj8+NGzc05lxUP3+pnD9/Xu4vJkmS1qFR1KkHYNu3b9d6nKr/EFQfmkVpLi4uCA4OxsyZM+Hm5ib/XwiBiIgIrFy5En369IGbmxvmzp1bZEN8lAQMsKhI1a5dGwsXLpSfv3z5UmunXaWpdyDWt7ldPV1RXmJRDxh69eolf/nn5aHtl62pqSkmTpyImzdvIiwsDD/88AMGDBiAChUqyGmSk5OxYMECtGvXrsiDLG3jcWmj6307c+aM3P/FxsYG58+fx4IFC9CpUyd4eHjAyspKoz9MQbVaFRRnZ2d8//33iI6OxqlTpzB79mx06dJFoxUyKioKo0aNwvjx44uwpiXH+fPnNZ5ru1St3rldCKExer22R69eveT0ycnJ+OOPP7Ll6ePjIy///fffBToGlbW1NWbMmIF79+4hODgYCxcuRK9eveQbfYDMVvCpU6ciICCgxE3yXVgYYFGR69Kli8bzhw8fFniZ6ndW6XtpSH3KDvUTTWFzdXWVl7PeBaSUevXq4YMPPsCWLVtw//59hIaGYuTIkfL68+fP5+vSpJL0fd/U02V931SDogKZl1Q8PT115lVcpm3JK1NTU7Rp0wbTpk3D/v378eTJE/z111/w9fWV0yxbtqxQftyUdFu3bpWXHR0dUadOHY31qamp+P333w0qQ9vdhOpTar18+RK7d+82qAx9GBkZoWnTpvjoo4+wY8cOPHr0CEFBQRoB4a5du+TLm6SJARYVuazX8c3NzQu8TPW7sM6cOaPXNqqWDgC53nWYF3m9TKL+i/mff/7RmP+xoDRq1AirVq3SGB5D2wm+MC/5qHf2z0lSUpLGvJRZ3zf1/mx169bNNT/1oT1yUhIue5mamuLNN9/EoUOHNDqaq08kTNlFRERoTMzcr1+/bO/3nj175CEVTE1N0bx5c70eTZo0kfM4ceJEtrtD27Ztq3En6OLFiwu95cjIyAitW7fG9u3b8eabb8r/L4xgryRigEVFLuvt7vp2XjZE+/bt5eXQ0FBcvnxZZ/oXL17gt99+07q9odQDTH068VapUiXbibawqA8f8OjRo2zr87ovhtiyZUuu41pt2bJFvpTi5OSU7RZ/9ct/uV1yjIqK0uuLpDBfA0OZmZmhY8eO8nNt7yllSklJwaBBg+SgxsTEBFOmTMmWTv3yYPfu3XHu3Dm9HhcvXpSHTBBZJogGMgP3Tz/9VH5++vRpLF++vCB2NVeSJKFbt27ycx432jHAIkWpxr3SV1paGr788kv5uaura66zvCvB09NTY3yXcePG6fwynD59OmJiYgBkdoRWcpwo9Vuf1cdZ0kX9RLtkyZI8veaqMYVUUlJSNDr966J+uU3bAJb52Zf8un37tkb/vayePHmCGTNmyM+HDRumdaBHlV27duWYV3p6OkaNGoWUlJRc62Vvby8HbjExMUUSZMXFxendCT+395Qy38fOnTtrDO/y6aefatxZDGTeparewpVb5/as1NNru0w4ZMgQdO3aVX7+0UcfKXqDwrNnz/Tu28XjRg+FNWQ8vR769+8vz423dOlSnXMQhoWFiQ4dOmhMFfH9999rTas+Vc6xY8dyrceMGTPk9DNmzNCaJutchG+99ZZ49OiRRpqUlBQxbdo0jTrqMxehvlOkCJE5r6CVlZW87fnz53PdJi0tTWNuMjMzMzF37lyRlJSkNf2zZ8/E5s2bRbt27USfPn2y1dve3l589NFHOZadkZEhDhw4IBwcHOQyZ8+enS1ddHS0vN7CwkLcvXtXj1dAf9rmIlywYEG2uQivX78uGjVqJKd1dHTM9t4KIURERIQ8rRIA8fHHH4vk5GSNNA8fPpTnalR/n3RNa+Tp6Smn+/3333Pdr/xMl6R+TGa1du1aUbVqVfHdd99pnXtSiMx5KhctWqSx/6dPn9ar7PzUSUX9PYyMjMw1z6FDhxo0tY2u8nPLLzIyUkyfPl1jXkcAwt/fX+s0MosWLdKYUuvFixd5qtu///6rUU5QUFC2NLGxsaJmzZoa6bp06SJOnjyZ7XOgLjg4WD4/53QuPXbsmChXrpz48ssvRXh4uNZ80tLSxMaNG+U5MwGITZs25Wk/XxccB4sKhGpA0fHjx8PDwwP169eHk5MTTE1NERcXhytXruD69esa2/j7+2PcuHGFVseWLVti7ty5mDx5MoDMvhOVKlVCu3bt4O7ujri4OBw/flxjzCt/f39MmjRJ0XoYGRmhV69e2LRpE4DMzqydO3dGpUqVYGxsDCBzDJxp06bJ2xgbG2Pr1q3o2LEjQkNDkZqais8++wxff/01WrRogUqVKsHMzAxxcXG4ceMGrl69KrekBAQEZKtDfHw8vv/+e3z//fdwcHCAl5cXKlSoAHNzc8TExODKlSuIjIyU09esWRMTJkzIlo+rqytat26N06dP4+XLl2jYsCE6d+6M8uXLy6061apVw/vvv2/w6/bdd99h4sSJ+OSTT/D999+jbdu2sLa2xr///ougoCD59nFjY2P88ssvcHFxyZaHp6cnBg8eLLcWLFy4EJs3b4a3tzdcXFxw584dnDx5EqmpqbCxscH8+fMxZsyYXOsWEBCA2bNnA8gcuHX9+vWoXr26xphOCxYsMPg10OX27duYMmUKpkyZgkqVKqFBgwZwcXGBEALR0dE4d+6cxnhogwYNQqtWrQq0TsXNxo0bcfHiRfl5eno6EhIS5HOUeh89IPOzOmXKFHz99dda+9qpXx4MCAjI8zhR1atXR7NmzeRZKNavX59tIOSyZcvi7Nmz6Nu3rzxS/IEDB3DgwAHY29vD29sbzs7OsLW1xYsXL/Dw4UOEhYVlu3nIy8sr25AsQGYL99dff42vv/4arq6u8PLyQrly5WBiYoLo6GiEhIRo5OXj41Ngs1aUeEUd4VHpsnLlSlGlShWNX0m5PSwtLcXXX3+tcyLXgmjBUlm1apWwtbXVWUdjY2MxYcIEkZaWlmM++W3BEkKI//77T7i5ueVYfk75JScnizFjxggTExO9X+s5c+Zo5HH//n2NX6O5Pfz8/MTDhw9z3JeQkBCdr6e+LTTaZG39+Pnnn4WZmVmOZdna2ubagvT8+XPRqVMnnftcsWJFERQUpHdLU0JCgqhTp47OPNUp3YK1bds2jZYpXQ8jIyPxwQcfiNTUVL3KzW+dVIpTC5a+DzMzM9G/f3+drcuXL1/W2ObIkSP5qt/SpUvlPOzs7HJsBXv16pX46aefRKVKlfK0L15eXmLt2rVaW7vOnTun97kEgOjTp4/Wie8pE1uwSFHvvfce3nvvPYSHh+PEiRM4d+4crl27hrt37yIhIQFCCNjY2KBcuXJo0KAB2rdvjz59+ug1B2FBGTlyJHr27IlffvkFBw4cwI0bNxAbGwsbGxu4u7ujQ4cOGDFiRLbbsZXk7u6Oy5cvY9myZTh06BCuX7+OZ8+e5dqJ29LSEj/99BM+/fRTbNy4EUePHsWNGzfw9OlTZGRkwM7ODlWrVkXDhg3xxhtvoHPnztlGYq9QoQKePn2Ko0eP4tSpUwgJCcHNmzfx+PFjueWmcuXK8Pb2Rv/+/bWOMq2ucePGuHLlCpYtW4ajR4/i9u3b8oCdShs1ahTatGmDFStW4MiRI7h//z6EEKhcuTK6d++OcePGaYzjpU2ZMmVw4MABbN68GevXr0doaCgSExPh5OSEqlWrIiAgAMOGDUPZsmVx/Phxvepla2uLCxcu4KeffsKePXsQERGB+Pj4QuuP1adPHzx8+BCHDh3C6dOncfnyZdy+fVseAdzOzg41a9ZEmzZtMGTIkAI9tksSc3Nz2NnZwc7ODhUqVEDjxo3RpEkTdOrUKdehWdRbr8qXL68xrEJe9O/fHx999BHS0tKQkJCAnTt3am0hMjExwZgxYzBy5EgcPnwYgYGBOH36NB4+fIinT5/i1atXsLOzg6OjI+rXrw9vb2906dJF51yOzZs3R0xMDI4cOYKgoCCEhobi1q1bePr0KdLT02Fra4tq1aqhRYsWeOedd9CsWbN87ePrQhKCI4QRUcng4eEhj0UVGRmZrZMxEVFxwbsIiYiIiBTGAIuIiIhIYQywiIiIiBTGAIuIiIhIYQywiIiIiBTGAIuIiIhIYRymoYBlZGQgKioKNjY2Wkf+JSIiopJBCIFnz57Bzc1NY7J4bTjQaAGLioqCu7t7UVeDiIiIFHLv3j1UrFhRZxoGWAXMxsYGQOabkXUEbSIiIio5EhMT4e7uLn+368IAq4CpLgva2toywCIiIioF9Onyw07uRERERApjgEVERESkMAZYRERERApjgEVERESkMAZYRERERApjgEVERESkMAZYRERERApjgEVERESkMAZYRERERApjgEVERESksAKfKiclJQVBQUF48uQJqlSpgmbNmhV0kURERERFyqAA6+7du/jhhx8AANOmTYO9vb3G+nPnzqFPnz54+PCh/L/GjRvjzz//RKVKlQwpmoiIiKjYMugS4Y4dO7BgwQIcPXo0W3D17Nkz9OrVCw8fPoQQQn6EhISgW7duSEtLM6RoIq0y0tNLVL5ERFQ6GdSCdfjwYUiShF69emVbt3LlSsTExECSJIwfPx5vvPEGDh48iB9//BFXr17F+vXrMXLkSEOKJ8rGyNgY2wcNwuOICMXydK5dG703bVIsPyIiKv0MCrBu374NAGjSpEm2dVu3boUkSfD398fixYsBAG+99RYeP36Mbdu24Y8//mCARQXicUQEokNDi7oaRET0GjPoEmFMTAwAwNXVVeP/iYmJuHTpEgBg+PDhGusGDBgAALh8+bIhRRMREREVWwYFWM+ePQMApGfpn3L69Gmkp6fD2NgYfn5+Guvc3d0BALGxsYYUTURERFRsGRRg2dnZAQCioqI0/n/8+HEAQMOGDWFlZaV1WwsLC0OKJiIiIiq2DAqw6tWrByDzbkKV9PR0uf9Vu3btsm3z4MEDANkvKxIRERGVFgZ1cvf398eJEyewYcMGuLq6wsfHBxs2bMDdu3chSRL69euXbZuLFy8CAMfBIiIiolLLoBas0aNHo3bt2hBCYMGCBejZsyf++OMPAJl3DDZt2jTbNjt27IAkSdn6ZhERERGVFgYFWObm5ggMDETv3r1hYmICIQRMTU0xePBgbNiwIVv6kydP4urVqwCAjh07GlI0ERERUbFl8FyE5cqVwx9//IGUlBTExsbC0dERZmZmWtO6u7vj2LFjAABvb29DiyYiIiIqlhSb7Nnc3Bzly5fXmaZKlSqoUqWKUkUSERERFUsGXSIkIiIiouwMasEqV64cfH195UfdunWVqhcRERFRiWVQgBUTE4M//vhDvnPQ0dERbdu2hZ+fH3x9fVG/fn1FKklERERUkhgUYI0fPx4nTpxAWFgYMjIy8OTJE+zYsUMeeLRs2bIaAVfDhg0VqTQRERFRcWZQgLV48WIAQHx8PE6cOIETJ07g+PHjuHLlCjIyMhAbG4tdu3Zh165dAAB7e3v4+PjIAZeXl5fBO0BERERU3EhCCKF0pgkJCRoB1+XLl5GRkZFZoCTJf9PS0pQuuthJTEyEnZ0dEhISYGtrW9TVeS383LgxokNDFcuvnJcXRl+6pFh+RERUMuXlO71A7iK0s7NDjx49sHDhQoSEhCAmJgbTp0+XJ4cWQqAA4joiIiKiYkGxcbDUCSEQGhqK48eP4/jx4zh16hQSExM1gqoyZcoURNFERERERU6RACungEq1DgAsLS3RsmVL+Pn5wc/PD82bN1eiaCIiIqJix6AAa9GiRTh27BiCgoKQkJAAQHdAZWpqaniNiYiIiIo5gwKsjz/+GJIkQQjBgIqIiIjo/ylyiVCSJPj4+KBTp07w8/ODl5eXfLcgERER0evGoACrffv2OHfuHJKTk3Ho0CEcPnwYAGBrayuPd8WAi4iIiF43BgVYR44cwatXr3DhwgUcP34cJ06cwJkzZ5CQkIC9e/di3759ABhwERER0etF8YFG09LSsgVcz58/zyzs/4MqOzs7+Pj4yCO8l2YcaLTwcaBRIiIqCEU60KiJiQlatWqFadOm4eDBg4iLi8Pp06cxdepU2NraQgiB+Ph47N27V+miiYiIiIqFAhloFACSk5MRFBQkj40VEhKCtLQ0+a5DIiIiotJKsQArp4AKQLYR3FXDORARERGVRgYFWIcPH8axY8d0BlRWVlZo1aoV/Pz84Ovri2bNmsHEpMAazoiIiIiKnEGRzptvvpntkp+1tTVat24tB1Te3t4wNjY2uKJEREREJYXBTUk2NjYaAVWTJk0YUBEREdFrzaAA68KFC2jcuDGMjBS/GZGIiIioxDIowGratKlS9SAiIiIqNdj0RERERKQwxW7ni42Nxdq1a3HkyBGEh4cjNjYWAODg4IB69eqhQ4cOGD58OBwcHJQqkoiIiKhYUiTA+vnnn/HJJ58gOTkZgOYwDQ8ePEBUVBQOHTqEmTNnYuHChRg1apQSxRIREREVSwYHWHPnzsXnn38uB1V2dnbw8vJCuXLlIITAo0ePEBoaioSEBDx//hzvv/8+4uPjMWXKFIMrT0RERFQcGRRghYeHY/r06RBCoHz58pg/fz769u0LU1NTjXRpaWnYtm0bJk+ejKioKHzxxRfo1q0b6tata1DliYiIiIojgzq5L1++HOnp6XB2dsbZs2fx9ttvZwuugMwJoAcOHIizZ8/CxcUF6enpWL58uSFFExERERVbBgVYR48ehSRJmDp1KipVqpRrend3d3z66acQQiAwMNCQoomIiIiKLYMCrAcPHgAAWrVqpfc2rVu3BgBERUUZUjQRERFRsWVQgKWaEkc1ybM+VGk5+jsRERGVVgZFOarLgnm53KdKq88lRSIiIqKSyKAAq2PHjhBCYMGCBQgLC8s1/ZUrVzB//nxIkoROnToZUjQRERFRsWVQgDVx4kSYm5sjKSkJbdq0wfz58/HkyZNs6Z48eYLvvvsObdu2RVJSEszNzTFx4kRDiiYiIiIqtgwaB6ty5cr4+eefMXz4cCQlJeGzzz7D1KlT4eHhAVdXV0iShOjoaNy5cwdCCAghIEkSfv75Z14iJCIiolLL4JHchwwZAkdHR4wZMwYPHjyAEAK3b99GZGQkAM1pc9zc3LBy5Up07drV0GKJiIiIii1F5iLs1q0bIiMjsWPHDp2TPfv7+8PERLH5pYmIiIiKJcWiHRMTE/Tt2xd9+/ZVKksiIiKiEomDUREREREpjAEWERERkcIYYBEREREpTK8+WKopcZQkSVKeptghIiIiKin0CrDUh1ogIiIiIt30CrBmzJhR0PUgIiIiKjUYYBEREREpTK8Aa+nSpQCAwYMHo2zZsgVaISIiIqKSTq8Aa+LEiZAkCR06dNAIsNq3bw9JkrBmzRpUrly5wCpJREREVJIYNJL78ePHIUkSnj9/rlR9iIiIiEo8vcbBsrCwAADEx8cXZF2IiIiISgW9AiwPDw8AwN69ewuyLkRERESlgl6XCLt27Ypr165h3rx5CAwMRM2aNWFqaiqv/+KLL2Bvb5+ngiVJwurVq/O0DREREVFJIAk9RhGNjY1FixYtcPPmzcyNJAnA/wYgVT3XlxACkiQhPT09r/XVy5QpUzB//nz5+bFjx+Dn56dzm7/++gsrV67EhQsX8PjxYzg7O6NZs2YYNWoUOnfunO+6JCYmws7ODgkJCbC1tc13PqS/nxs3RnRoqGL5lfPywuhLlxTLj4iISqa8fKfr1YLl4OCAixcvYvny5QgMDMSDBw+QkpKCu3fvQpIklC9fXqNFqyhdvnwZixYt0ju9EAJjxozBypUrNf7/4MED7NixAzt27MCoUaOwYsWKPAeSRERE9HrS+y5CW1tbTJs2DdOmTZP/Z2SU2YXr0KFDqFOnjvK1y6OMjAy89957SEtLg4uLC2JiYnLd5osvvpCDKy8vL0yZMgXVqlXDrVu38N133yE0NBQrV66Es7Mzvvnmm4LeBSIiIioF9OrkXlIsXboUwcHB8PT0xMiRI3NNf/PmTXz33XcAgKZNm+L06dMYMGAAvL29MWDAAAQFBaFp06YAgHnz5uHWrVsFWn8iIiIqHQwKsI4dO4ajR4+iSpUqStUn3+7du4fp06cDAH766SeYmZnlus2iRYuQlpYGAFi2bBksLS011pcpUwbLli0DAKSlpWHx4sXKVpqIiIhKJYMCrLS0NPj6+mYLTPTxwQcfGFK01vySkpIwdOjQXDu0A5l9r3bt2gUA8PT0RIsWLbSma9GiBWrVqgUA2LlzJ/S4J4CIiIhecwYFWP7+/ggODs7zdqNGjcLPP/9sSNEatm7dir1798LBwUHj7kFdIiMj8eDBAwCAr6+vzrSq9ffv38edO3cMqisRERGVfgYFWElJSejatSsiIiL03ubdd9/FqlWrDClWQ3x8PCZMmAAgs5+Us7OzXtup19nT01NnWvX1edlXIiIiej0ZFGDVrFkTT58+RadOnfDff//lmn7YsGFYu3YtAGDQoEGGFC2bMmUKoqOj0apVK706tqvcu3dPXq5YsaLOtO7u7lq3IyIiItLGoADr8OHDqFixIh48eICOHTvmOCyCEAKDBw/Gr7/+CiEEhg4divXr1xtSNAAgKCgIq1atgomJSZ7HqXr27Jm8bG1trTOtlZWVvJyUlKQzbUpKChITEzUeRERE9HoxKMByd3fH4cOH4eTkhJs3b+LNN99EQkKCRpqMjAwMGjQImzZtAgCMGDECa9asMXjQztTUVIwaNQpCCEyaNAn169fP0/YvX76Ul3O749Dc3FxefvHihc603377Lezs7OSHeusXERERvR4MHgerVq1aOHDgAKytrXHlyhV0795dDl4yMjIwcOBA/PbbbwD+1/9KiRHR58yZg4iICFSqVAkzZszI8/YWFhbycmpqqs60KSkp8nJud0xOnToVCQkJ8oOXFImIiF4/igw02qRJE+zatQvm5uY4c+YMAgIC8OLFC/Tp0wfbtm0DAIwePTrbdDT5de3aNXz77bcAMsevUr+Epy8bGxt5ObfLfs+fP5eXc7ucaG5uDltbW40HERERvV70nionN35+ftiyZQv69OmDv/76C1WqVMHjx48BAO+//z5++OEHpYrCokWLkJqaiqpVqyI5OVluIVMXHh4uLx89ehTR0dEAgLfeegtWVlYaHdvv37+vszz1Vihe8iMiIqLcKBZgAUDPnj2xatUqjBgxQu7w/uGHH2Lp0qVKFiNfsrt9+zYGDhyYa/pZs2bJy5GRkbCystKYO/HatWs6t1dfX7t27bxWl4iIiF4zegVY+gzBoNKuXTtMmDABixcvRp8+ffDJJ5/kuH2lSpX0zldpVapUgZubG6KionDixAmdaU+ePAkAqFChAjw8PAqhdkRERFSS6RVg5WeuQUmS8Oeff+LPP//Mcb1qHsC8WrduHdatW6czzcyZM/HVV18ByJwzMev0OZIkoWfPnvjpp59w7do1nDt3Tut0OefOnZNbsHr27KlIB30iIiIq3fTq5C6EKJBHUZs4cSJMTDJjzHHjxmUbguHFixcYN24cAMDExAQTJ04s7CoSERFRCaRXC5Zq9PXSpmbNmvjkk08wd+5cXLx4Ea1bt8ann36KatWq4datW5g3bx5CQ0MBAJMnT0aNGjWKuMZERERUEugVYA0dOrSg61FkZs+ejZiYGKxZswahoaEYMGBAtjQjR47EN998UwS1IyIiopJIkXGwSjIjIyOsXr0a+/btQ8+ePeHm5gYzMzO4ubmhZ8+e2L9/P1atWgUjo9f+pSIiIiI9ScKAzlBGRkYwMjLCnDlzMGXKFCXrVWokJibCzs4OCQkJHHS0kPzcuDGi///SrhLKeXlh9KVLiuVHREQlU16+0w1qljEzM4MQAj4+PoZkQ0RERFSqGBRgubm5AQCMjY0VqQwRERFRaWBQgNW2bVsAwCVePiEiIiKSGRRgjRs3DsbGxliwYAESExOVqhMRERFRiWZQgNWkSRMsW7YMd+/eha+vL86cOaNUvYiIiIhKLIMmex4xYgQAoFatWrh8+TJ8fHzg7u6OBg0aoGzZsjr7ZkmShNWrVxtSPBEREVGxZFCAtW7dOnluPkmSIITAf//9h3v37uncTgjBAIuIiIhKLYMCrEqVKnHyYyIiIqIsDAqw7ty5o1A1iIiIiEoPzv9CREREpDAGWEREREQKM+gSYU7S0tIQFxcHAChbtixMTAqkGCIiIqJiSbEWrIiICIwbNw61a9eGhYUFypUrh3LlysHCwgK1a9fG+PHjcfXqVaWKIyIiIiq2FAmwpk6digYNGuDHH3/E9evXkZGRASEEhBDIyMjA9evX8cMPP6Bhw4aYNm2aEkUSERERFVsGX7sbN24cfvzxRwghAAC1a9dG8+bNUa5cOQgh8OjRI1y4cAFXr15Feno65s2bh+fPn2PJkiUGV56IiIioODIowDp9+jR++OEHSJKEOnXqYOXKlWjVqpXWtGfPnsWYMWMQFhaG5cuXo3///jmmJSIiIirJDLpE+PPPPwMAqlSpgtOnT+sMmFq2bImTJ0+iatWqAIAVK1YYUjQRERFRsWVQgHXq1ClIkoTPPvsMdnZ2uaa3s7PDp59+CiEETp06ZUjRRERERMWWQQFWdHQ0AMDLy0vvbRo3bgwAePTokSFFExERERVbBgVYFhYWAIDnz5/rvU1SUhIAwNzc3JCiiYiIiIotgwKsKlWqAAB2796t9zZ79uwBALkvFhEREVFpY1CA1bVrVwghsHz5cgQGBuaaPjAwEMuWLYMkSejatashRRMREREVWwYFWBMnToStrS1evXqFLl26YOzYsQgJCUFGRoacJiMjAyEhIfjggw/QpUsXvHr1Cra2tpg4caKhdSciIiIqlgwaB8vJyQlbt25Fjx49kJqaihUrVmDFihUwMzODg4MDJEnC06dPkZqaCgAQQsDMzAzbtm2Do6OjIjtAREREVNwYPFVOp06dcO7cOTRt2lSeHiclJQUPHz5EVFQUUlJS5P97e3vj/Pnz6NChgxJ1JyIiIiqWDJ4qBwAaNWqECxcuIDg4GEeOHEF4eDhiY2MBAA4ODqhXrx46dOgAb29vJYojIiIiKtYUCbBUvL29GUQRERHRa8/gS4REREREpMmgAOunn37C1atXlaoLERERUalg0CXCsWPHQpIkODk5oW3btvDz84Ovry/q1aunVP2IiIiIShyD+2AJIfD48WNs374d27dvB5DZsV094GrQoIHBFSUiIiIqKQwKsGJjY3HixAkcP34cx48fR1hYGDIyMvD06VPs3LkTO3fuBACULVsWPj4+csDVqFEjBapOREREVDxJQgihVGbx8fE4efKkHHBduXJFY1R3SZIAAHZ2dmjbtq0cgJVmiYmJsLOzQ0JCAmxtbYu6Oq+Fnxs3RnRoqGL5lfPywuhLlxTLj4iISqa8fKcrOkyDvb09evTogR49egAAEhISNAKuy5cvIyMjA/Hx8fKkz0RERESlTYEO02Bra4sKFSqgQoUKcHNzQ5kyZeRWLCIiIqLSStEWLCEEQkND5RarU6dOITExUV4HAJaWlmjZsiXatWunZNFERERExYZBAVbWgCooKAgJCQnyOuB/AZWfnx/8/PzQvHlzmJqaGl5zIiIiomLKoADL0dGRARURERFRFgYFWPHx8XKfqp49e2LSpElo2bIlAyoiIiJ6rSky0CgA7N69G8ePH9cYYNTLy4ud2omIiOi1Y1CAderUKbn/1dmzZ5GQkIA9e/Zg7969ADLvIlQNMOrn58eAi4iIiF4Lig00mpaWhgsXLsgB15kzZ5CcnJxZyP8HVVkDrsaNGytRdLHGgUYLHwcaJSKigpCX73RFR3JXl5aWhuDgYI2A6/nz53KwJUkS0tLSCqLoYoUBVuFjgEVERAWhyEZy18jYxAQtW7ZEw4YN0bhxY9SvXx+rV69GYmIiCiimIyIiIioWFA+wkpOTERQUJLdchYSEyC1V6oFVmTJllC6aiIiIqFgwOMDSN6CysrKSx8fy9fVFs2bNDC2aiIiIqFgyKMBq3bo1Ll68qDWgsra2RqtWreSAytvbGyYmBXZFkoiIiKjYMCjiOXv2rLxsY2OD1q1bywFV06ZNYWxsbHAFiYiIiEoagwKsrl27ygFVkyZNYGRkpFS9iIiIiEosgwIs1YCiRERERPQ/bHIiIiIiUhgDLCIiIiKFMcAiIiIiUhgDLCIiIiKFMcAiIiIiUhgDLCIiIiKFMcAiIiIiUhgDLCIiIiKF6RVg9e7dGwEBAbh//35B14eIiIioxNMrwNq5cyd27tyJxMREzY2NjGBiYoKrV68WSOWIiIiISqI8XSIUQuj1PyIiIqLXmV4Blo2NDQDg0aNHBVoZIiIiotJArwDL09MTALBkyRIkJSVlWy9JkrK1IiIiIirBTPRJ9PbbbyM4OBh79+6Fg4MDXF1dYWpqKq/v1KmTxnN9SJKEW7du5a22RERERCWAXgHWuHHjcPr0afzxxx9IS0vDgwcP5HVCCI3n+mKrFxEREZVWegVYRkZG2Lp1K86ePYsjR47gwYMHSElJwfr16yFJEnr06AF7e/sCrioRERFRyaBXgKXSsmVLtGzZUn6+fv16AMDs2bNRp04dZWtGREREVEJxJHciIiIiheWpBSurjIwMpepBREREVGqwBYuIiIhIYQa1YKlLT0/Hzp07ceTIEYSHhyM2NhYA4ODggHr16qFDhw7o1asXjI2NlSqSiIiIqFhSJMD666+/MGrUqGzDNwCZwzGcOXMGK1euRMWKFbFy5Uq8+eabShRLREREVCwZfIlww4YN6N69Ox48eAAhBIQQqFy5Mlq0aIHmzZujcuXKADIDrnv37qFbt27YtGmTwRUnIiIiKq4MCrDu3r2LUaNGISMjA2XKlME333yD6Oho3L59G2fOnMHZs2dx+/ZtREdHY/bs2bC2tkZGRgbee+89/Pfff0rtAxEREVGxYlCAtWTJEqSkpMDa2hqnTp3CtGnT4OLiki2ds7Mzpk6dilOnTsHa2hopKSlYsmSJIUUTERERFVsGBViHDh2CJEmYPHkyGjVqlGv6hg0b4pNPPoEQAgcPHjSkaCIiIqJiy6AAS3WZr0OHDnpv07FjR41tiYiIiEobgwKs9PR0AMjT0AuqtByklIiIiEorgwKsChUqAADOnDmj9zaqtG5uboYUTURERFRsGRRgtWvXDkIIzJ07F1FRUbmmv3//PubOnQtJktC+fXtDiiYiIiIqtgwKsMaNGwcjIyM8fvwYzZs3x7Zt2+TLhurS09OxdetWtGzZEjExMTAyMsKHH35oSNFERERExZZBI7nXq1cPs2bNwueff46oqCgMGDAA9vb28PLygqurKyRJQnR0NEJDQxEfHy+P7j5r1izUq1dPkR0gIiIiKm4Mnipn6tSpsLOzw5QpU5CcnIy4uDgcO3ZMI40qsCpTpgzmz5+P999/39BiiYiIiIotReYi/OCDD9CvXz+sXbtW52TPw4cPh5OTkxJFEhERERVbigRYAODk5ITJkydj8uTJSmVJREREVCIZPNkzEREREWligEVERESkMAZYRERERApjgEVERESkMAZYRERERApjgEVERESksBIdYF26dAlz5sxBly5d4O7uDnNzc1hbW6NmzZoYNmwYTp06laf8/vrrL/Tu3RsVK1aEubk5KlasiN69e+Ovv/4qoD0gIiKi0kgSqmHWSxhfX1+cPHky13SDBw/GqlWrYGZmlmMaIQTGjBmDlStX5phm1KhRWLFiBSRJylM9ExMTYWdnh4SEBNja2uZpW8qfnxs3RnRoqGL5lfPywuhLlxTLj4iISqa8fKeX2BasBw8eAADc3NwwYcIE/PHHH7hw4QLOnj2L77//HhUqVAAAbNiwAcOGDdOZ1xdffCEHV15eXtiyZQsuXLiALVu2wMvLCwCwcuVKTJ8+veB2iIiIiEoNg1qwvv76awBA8+bN8eabbypWKX10794dQ4YMQUBAAIyNjbOtf/LkCVq3bo0bN24AAE6ePAkfH59s6W7evInatWsjLS0NTZs2xcmTJ2FpaSmvT05Ohq+vLy5evAgTExNcu3YN1apV07uebMEqfGzBIiKiglBoLVgzZ87EV199hZSUFEOyyZe9e/eiX79+WoMrIHPqnoULF8rP//jjD63pFi1ahLS0NADAsmXLNIIrIHOC6mXLlgEA0tLSsHjxYgVqT0RERKWZQQGWo6MjAKBSpUqKVEZpfn5+8vKtW7eyrRdCYNeuXQAAT09PtGjRQms+LVq0QK1atQAAO3fuRAnttkZERESFxKAAq3r16gCA6OhoRSqjtNTUVHnZyCj7rkZGRsp9uXx9fXXmpVp///593LlzR7lKEhERUaljUIDVv39/CCGwdetWpeqjqBMnTsjLnp6e2dZHREToXK9Ofb36dkRERERZGRRgffDBB2jYsCF+/fVXrF+/Xqk6KSIjIwNz586Vn/fr1y9bmnv37snLFStW1Jmfu7u71u2IiIiIsjIxZOPo6GisWrUKI0eOxIgRI7Bp0ya8/fbbaNCgAcqWLZtjB3SVguy7tWjRIly4cAEA4O/vj6ZNm2ZL8+zZM3nZ2tpaZ35WVlbyclJSUo7pUlJSNDr9JyYm6l1nIiIiKh0MCrA8PDzkgTeFEAgMDERgYKBe20qSJN+9p7QTJ07gs88+AwC4uLjgp59+0pru5cuX8rKugUgBwNzcXF5+8eJFjum+/fZbfPXVV3mpLhEREZUyBg80KoSQ76pTLev7KAj//PMP/P39kZaWBnNzc2zduhWurq5a01pYWMjL6h3itVFvlco6lIO6qVOnIiEhQX7wciIREdHrx6AWrLVr1ypVD0VERkaiU6dOiIuLg7GxMbZs2aLz7kAbGxt5WddlPwB4/vy5vKzrcqK5ublGaxcRERG9fgwKsIYOHapUPQwWFRWFDh06ICoqCpIkYc2aNfD399e5jXrH9vv37+tMq94Spd7hnYiIiCirEjsXobonT56gY8eOuH37NoDMEdmHDBmS63Z16tSRl69du6Yzrfr62rVr57OmRERE9Doo8QFWQkIC3nzzTVy9ehUAMHfuXIwdO1avbatUqQI3NzcAmmNmaXPy5EkAQIUKFeDh4ZH/ChMREVGpp1iAlZGRgaNHj2L27NkYN24cRo4ciYcPH2qkSU1NRXJysmJzFyYnJ6Nbt2649P8T8X7++ef49NNP9d5ekiT07NkTQGYL1blz57SmO3funNyC1bNnT/nOSSIiIiJtFAmw9u3bhxo1aqBjx4748ssv8eOPP2LdunWIi4vTSLd69WrY2NjAxcVFo9N4fqSmpsLf3x+nT58GAEyYMAHffPNNnvOZOHEiTEwyu6KNGzcu2xAML168wLhx4wAAJiYmmDhxokH1JiIiotLP4ABr1apV6NGjByIjIyGEgKOjY45DMIwcORL29vZISkrCjh07DCp34MCBOHToEACgffv2GDlyJMLDw3N83LhxQ2s+NWvWxCeffAIAuHjxIlq3bo3ff/8dFy9exO+//47WrVvj4sWLAIDJkyejRo0aBtWbiIiISj+D7iK8efOm3N+pffv2WL58OTw9PbVOrAxkDuYZEBCAVatW4dChQ3jnnXfyXfb27dvl5aNHj6JBgwY601euXDnHSZpnz56NmJgYrFmzBqGhoRgwYEC2NCNHjsxXCxkRERG9fgxqwVq8eDFevXqFunXrYv/+/blOmAwAPj4+AIC///7bkKIVZWRkhNWrV2Pfvn3o2bMn3NzcYGZmBjc3N/Ts2RP79+/HqlWrcgwciYiIiNQZ1IIVGBgISZIwceLEXKeaUalWrRoA4L///jOk6AIZCb5r167o2rWr4vkSERHR68WgJhnV4JuNGjXSexvVpMnJycmGFE1ERERUbBkUYKlP9Kyvx48fAwBsbW0NKZqIiIio2DIowFIN0pnTHXraqAb05GCdREREVFoZFGC1bdsWQghs3rxZr/RPnjzBzz//DEmS0L59e0OKJiIiIiq2DAqwRo0aBQDYv38/1q5dqzPt/fv30bVrVzx58gTGxsbytkRERESljUEBlre3N8aMGQMhBN5991307dsXW7dulddfuXIFv//+O0aOHIlatWohJCQEkiTh448/RvXq1Q2uPBEREVFxZNAwDQCwbNkyPH/+HBs2bMD27duxfft2ufP7oEGD5HSqjvDDhg3DnDlzDC2WiIiIqNgyeORMY2NjrF+/Htu2bYOXlxeEEFofderUwebNm7FmzRpOlkxERESlmsEtWCoBAQEICAhAVFQULl68iJiYGKSnp8PR0RFeXl7yAKNEREREpZ1iAZaKm5sbevTooXS2RERERCUGJ9cjIiIiUpiiLVivXr3CpUuXEB4ejtjYWACAg4MD6tWrh8aNG8PU1FTJ4oiIiIiKJUUCrKSkJMyaNQurV69GXFyc1jRly5bFyJEj8cUXX8DGxkaJYomIiIiKJYMvEUZERKBu3bpYsGABYmNjc7yLMDY2FgsWLED9+vVx/fp1JepOREREVCwZ1IIVHx+PDh064OHDhwCAevXqYejQoWjWrBlcXV0hhEBMTAyCg4Oxfv16hIWF4b///kOHDh0QHh4OOzs7RXaCiIiIqDgxqAVr3rx5ePjwISRJwqxZs3D58mV8/PHH8PHxQc2aNVGrVi34+Pjgo48+wt9//41vvvkGABAVFYV58+YpsgNERERExY1BAdbOnTshSRL69++Pzz//XOcAopIkYdq0aejfvz+EENixY4chRRMREREVWwYFWHfv3gUADB06VO9thg0bprEtERERUWljUICluhvQxcVF721Uaa2trQ0pmoiIiKjYMijAql+/PgDg33//1XsbVVrVtkRERESljUEB1ujRoyGEwOLFi5GRkZFr+oyMDCxatAiSJGHUqFGGFE1ERERUbBkUYPXt2xfDhw/HuXPn0KtXL0RHR+eY9tGjR+jduzfOnz+PYcOGoX///oYUTURERFRs6TUO1q+//prjOl9fX4SHh2Pv3r2oWrUqOnXqBG9vb7i4uECSJDx69AjBwcE4dOgQUlJS4O3tDV9fX/z6668YMmSIYjvyOspIT4eRsXGJy5uIiKi0k4QQIrdERkZGOodgUBFC5Jgu6zpJkpCWlpaHqpZMiYmJsLOzQ0JCAmxtbRXPf/ugQXgcEaFons61a6P3pk2K5lmYfm7cGNGhoYrlV87LC6MvXVIsPyIiKpny8p2u90juesRhuabTNw/S3+OICEWDCSIiIjKcXgFWZGRkQdeDiIiIqNTQK8CqXLlyQdeDiIiIqNQw6C5CIiIiIsqOARYRERGRwhhgERERESlMkQArIiICkyZNQtOmTeHg4ABTU1MYGxvrfJiY6H0DIxEREVGJYnCUM3fuXHz55ZdIT0/nMAxEREREMDDA2rZtG6ZNmwYgczBSHx8fNGzYEPb29jAy4tVHIiIiej0ZFGAtWbIEAFChQgXs378f9evXV6RSRERERCWZQc1MV65cgSRJmDVrFoMrIiIiov9nUIBlamoKAGjUqJESdSEiIiIqFQwKsGrWrAkAePr0qSKVISIiIioNDAqwhg4dCiEEdu7cqVB1iIiIiEo+gwKskSNHwsfHBytXrsSePXuUqhMRERFRiWbQXYSmpqbYtWsXhg4dCn9/fwwYMAD9+vVDzZo1UaZMmVy3r1SpkiHFExERERVLBg80am9vj/Hjx+PcuXPYsmULtmzZotd2kiQhLS3N0OKJiIiIih2DRwOdOHEiOnXqhCdPnkAIkacHERERUWlkUAvWxo0bsXTpUgCAra0t/P390aBBA47kTkRERK81gwKsZcuWAQA8PT1x7NgxuLq6KlIpIiIiopLMoGama9euQZIkzJw5k8EVERER0f9TZCR31YCjRERERGRggOXp6QkAiI6OVqQyRERERKWBQQHW8OHDIYTQe2gGIiIioteBwSO5d+/eHRs3bsTy5cuVqhMRERFRiWbQXYQnT57E+PHj8fjxY0yYMAGbN2/GgAED9B7JvW3btoYUT0RERFQsGRRg+fn5QZIk+fn58+dx/vx5vbblSO5ERERUWhk8VQ5HZCciIiLSZFCAdezYMaXqQURERFRqGBRg+fr6KlUPIiIiolKDEwYSERERKYwBFhEREZHCGGARERERKcygPljt27fP97aSJCEwMNCQ4omIiIiKJYMCrOPHj0OSJJ1DNaiPkwX8b1iHrP8nIiIiKi0MCrDatm2ba6D0/Plz/Pvvv0hISIAkSahZsybKly9vSLFERERExZrBLVj6EEJg3759mDBhAmJjY7Fq1Sq0adPGkKKJiIiIiq1C6eQuSRK6d++OoKAgGBsbw9/fHw8ePCiMoomIiIgKXaHeRVi+fHl89NFHePr0Kb777rvCLJqIiIio0BT6MA2qS4P79u0r7KKJiIiICkWhB1hmZmYAgKioqMIumoiIiKhQFHqAFRQUBAAoU6ZMYRdNREREVCgKNcA6e/Ysvv76a0iShGbNmhVm0URERESFxqBhGr7++utc02RkZCAuLg4XL17E+fPnkZGRAUmSMGnSJEOKJiIiIiq2DAqwZs6cmacR2YUQMDExwXfffYeOHTsaUjQRERFRsWVQgAVA5zQ5QOYYWDY2NqhSpQp8fX0xatQo1KlTx9BiiYiIiIotgwKsjIwMpepBREREVGoU+l2ERERERKUdAywiIiIihTHAIiIiIlKYwZ3c1T179gyRkZF49uwZ0tPTc03ftm1bJYsnIiIiKhYUCbB++eUX/Pjjj7hy5Yre20iShLS0NCWKJyIiIipWDAqw0tPTERAQgD179gDIfcgGIiIioteBQQHWihUrsHv3bgCAq6srhg8fjiZNmsDBwQFGRuzeRURERK8ngwKsX3/9FQBQp04dnDp1CmXLllWkUkSkjIz0dBgZG5e4vImISjqDAqyIiAhIkoTp06czuCIqhoyMjbF90CA8johQNF/n2rXRe9MmRfMkIipNFOnkXqtWLSWyIaIC8DgiAtGhoUVdDSKi14pBHaVq1KgBAIiNjVWkMkRERESlgUEB1oABAyCEwN69e5WqDxEREVGJZ1CANX78eDRo0AA//fQTTp06pVSdiIiIiEo0gwIsc3NzHDp0CE2aNEHHjh0xZcoU/P3333j58qVS9SMiIiIqcQzq5G6sdou2EAILFy7EwoUL9dqWI7kTERFRaWVQgJV15HaO5E5ERERkYIA1Y8YMpepBREREVGowwCIiIiJSGCcMJCIiIlIYAywiIiIihTHAUvPff//hk08+Qe3atWFlZQUHBwc0a9YMCxYsQHJyclFXj4iIiEoIReYiLA327duHQYMGISEhQf5fcnIygoODERwcjFWrVmH//v2oWrVqEdaSiIiISgK2YAG4fPky+vXrh4SEBFhbW2P27Nk4c+YMAgMD8d577wEArl+/jm7duiEpKamIa0tERETFHVuwAEycOBHJyckwMTHBoUOH0LJlS3ld+/btUaNGDUyZMgXXrl3D999/jy+//LIIa0tERETF3WvfghUcHIzjx48DAEaOHKkRXKl8/PHHqF27NgBg8eLFePXqVWFWkYhKgIz09BKZd0EqqHqX1NeDNJX2z8xr34K1c+dOeXn48OFa0xgZGWHIkCGYOnUq4uLicPz4cXTs2LGQakhEJYGRsTG2DxqExxERiubrXLs2em/apGiehaUgXpOS/HqQptL+mXntA6xTp04BAKysrNCkSZMc0/n6+srLQUFBDLCIKJvHERGIDg0t6moUK3xNSJfSfHy89pcII/4/cq5evTpMTHKONz09PbNtQ0RERKTNax1gvXz5Ek+ePAEAVKxYUWfasmXLwsrKCgBw7969Aq8bERERlVyv9SXCZ8+eycvW1ta5preyssLz5891DtWQkpKClJQU+blqXK3ExEQDaqqjTtWrw07hznxW1asXWH0Lg9KvCV8P7XmW5NekoPC1zo6fR9KlpH1mVPkKIXJPLF5j//33nwAgAIjBgwfnmt7d3V0AENWqVcsxzYwZM+Q8+eCDDz744IOP0ve4d+9erjHDa92CZWFhIS+npqbmml7VMmVpaZljmqlTp+Kjjz6Sn2dkZCA2NhaOjo6QJMmA2pIhEhMT4e7ujnv37sHW1raoq0OlDI8vKkg8vooPIQSePXsGNze3XNO+1gGWjY2NvKzPCO3Pnz8HoPtyorm5OczNzTX+Z29vn78KkuJsbW15gqICw+OLChKPr+LBzs5Or3SvdSd3CwsLODk5AQDu37+vM21cXJwcYLm7uxd43YiIiKjkeq0DLADyCO03b95EWlpajumuXbuWbRsiIiIibV77AKtNmzYAMi//hYSE5JjuxIkT8nLr1q0LvF6kLHNzc8yYMSPb5VsiJfD4ooLE46tkkoTQ517D0uvChQto3rw5AGD06NFYsWJFtjQZGRmoV68eIiIiYG9vj5iYGJiamhZ2VYmIiKiEeO1bsJo1awYfHx8AwOrVq3H27NlsaRYuXCiP3j5hwgQGV0RERKTTa9+CBQChoaFo3bo1Xrx4AWtra0ybNg3t2rXDixcv8Ntvv2HlypUAgJo1a+LixYsadx8SERERZcUA6//t2bMH77zzTo6jv9asWRP79u1D9erVC7lmREREVNK89pcIVd566y1cuXIFkyZNQs2aNVGmTBnY29ujadOmmDdvHkJDQxlclRCXLl3CnDlz0KVLF7i7u8Pc3BzW1taoWbMmhg0bhlOnThV1FamUmjJlCiRJkh/Hjx8v6ipRCfbkyRN89913aN26NcqVKwdzc3O4ubmhefPmmDx5stYuLVR8sAWLShVfX1+cPHky13SDBw/GqlWrYGZmVgi1otfB5cuX0bRpU43hXo4dOwY/P7+iqxSVWNu2bcP777+Pp0+f5pimZ8+e2LlzZ+FVivLktR7JnUqfBw8eAADc3NzQt29f+Pj4oFKlSkhPT8fZs2excOFCPHjwABs2bEBaWho2b95cxDWm0iAjIwPvvfce0tLS4OLigpiYmKKuEpVgv/76K4YPH46MjAy4uLjg/fffR5s2beDg4IDo6GjcunULe/bs4Q1XxRxbsKhU6d69O4YMGYKAgAAYGxtnW//kyRO0bt0aN27cAACcPHlSvouUKL8WL16MSZMmwdPTE/7+/vj2228BsAWL8i4iIgJeXl5ISUmBj48P9uzZk+PULKmpqWyFL8bYB4tKlb1796Jfv35agysAcHJywsKFC+Xnf/zxR2FVjUqpe/fuYfr06QCAn376iV94ZJBx48YhJSUFTk5O2L59u85573isFW8MsOi1o96icOvWraKrCJUKH3zwAZKSkjB06FC2VpFBrl27hsDAQADAhx9+KM+VSyUTAyx67aSmpsrLRkb8CFD+bd26FXv37oWDgwPmz59f1NWhEm7btm3yct++feXluLg4/Pvvvzo7vFPxw28Xeu2ozyvp6elZhDWhkiw+Ph4TJkwAAMybNw/Ozs5FXCMq6c6dOwcAsLOzQ+3atbFp0yY0bNgQDg4OqFmzJpycnFC1alV89dVXSEpKKuLaUm4YYNFrJSMjA3PnzpWf9+vXrwhrQyXZlClTEB0djVatWmHkyJFFXR0qBa5evQoA8PDwwLhx4/DOO+/gypUrGmkiIyMxc+ZMtGzZElFRUUVRTdITAyx6rSxatAgXLlwAAPj7+6Np06ZFXCMqiYKCgrBq1SqYmJhgxYoVkCSpqKtEpUBsbCyAzL5YP/zwA+zt7bFixQrExMTg5cuXCA4ORpcuXQAA4eHh6Nu3LzIyMoqyyqQDAyx6bZw4cQKfffYZAMDFxQU//fRTEdeISqLU1FSMGjUKQghMmjQJ9evXL+oqUSnx/PlzAEBKSgqMjY1x4MABjB49Gs7OzjA3N0fTpk2xd+9eOcg6c+YMtm/fXpRVJh0YYNFr4Z9//oG/vz/S0tJgbm6OrVu3wtXVtairRSXQnDlzEBERgUqVKmHGjBlFXR0qRSwsLOTlvn37okWLFtnSGBkZadxQsWXLlkKpG+UdAywq9SIjI9GpUyfExcXB2NgYW7Zsga+vb1FXi0qga9euyYOILlu2DFZWVkVcIypNbGxs5GVVK5U2devWRYUKFQAAwcHBBV4vyh9OlUOlWlRUFDp06ICoqChIkoQ1a9bA39+/qKtFJdSiRYuQmpqKqlWrIjk5Gb/99lu2NOHh4fLy0aNHER0dDSBzQnkGZKSLu7u7fLxUrFgx17QPHjzgtEzFGAMsKrWePHmCjh074vbt2wAyWxyGDBlSxLWikiwlJQUAcPv2bQwcODDX9LNmzZKXIyMjGWCRTnXr1pVbpNLT03WmVa03MeHXeHHFS4RUKiUkJODNN9+Ub3ueO3cuxo4dW8S1IiLKWdu2beXl3GaZUP1wVF0qpOKHARaVOsnJyejWrRsuXboEAPj888/x6aefFnGtqDRYt24dhBA6H+od348dOyb/38PDo+gqTiVCjx49YGpqCgA67w48ceKEPKo7J6svvhhgUamSmpoKf39/nD59GgAwYcIEfPPNN0VcKyKi3Dk6OuLdd98FABw+fFhrH79nz55h4sSJ8vPRo0cXVvUoj3jxlkqVgQMH4tChQwCA9u3bY+TIkRqdjrMyMzNDzZo1C6t6REQ6ffXVV9i3bx/+++8/DB48GKdPn0bv3r1ha2uLsLAwzJs3D9euXQMAvP/++/D29i7iGlNOJCGEKOpKECklryNqV65cGXfu3CmYytBraebMmfjqq68AZF4i9PPzK9oKUYkTERGBHj164ObNmzmmGTFiBFasWCFfUqTih5cIiYiIipHatWvj77//xvz589G8eXM4ODjAzMwMFStWRP/+/XH06FGsXr2awVUxxxYsIiIiIoWxBYuIiIhIYQywiIiIiBTGAIuIiIhIYQywiIiIiBTGAIuIiIhIYQywiIiIiBTGAIuIiIhIYQywiIiIiBTGAIuIiIhIYQywiIiIiBTGAIuIiIhIYQywFLZu3TpIkgRJknDnzp2irk6xMnPmTPm1KSnS09OxZMkSNGvWDLa2tnL9e/XqVSjlq8qbOXNmoZRHr5/jx4/Lx9nx48eLujpa3blzR67junXriro6pYaHhwckScKwYcMMymffvn1488034eTkBGNjY0iSBHt7e3n963oeMynqChAVZwMHDsS2bduKuhpERMXSjz/+iLFjxxZ1NYoltmCRQUrCr9/8OnPmjBxcdevWDYcPH8aVK1cQFhaGpUuXFnHtijelfhkXtNLc4vy6thpQ4Xnx4gWmTZsGAPD09MQff/yB0NBQhIWF4ezZs0Vcu6LHFiyiHBw5cgQAYGxsjM2bN8PW1raIa0T0evLw8IAQoqirQVkEBwcjISEBALBgwQJ069atiGtUvLAFiygHDx48AAC4uroyuCIiykJ1jgSAmjVrFmFNiicGWEQ5SElJAQCYmpoWcU2IiIof1TkS4HlSGwZYRUQIgT/++AMBAQFwd3eHhYUFypYti2bNmmHWrFmIj4/Pcdthw4ZBkiR4eHgAAOLj4/Hll1+ibt26sLKygr29Pdq2bYtNmzbpVZfdu3fLd4CUKVMGNWvWxOTJkxEdHQ1Ae38a1V097dq1k//Xrl07ud+HPnf8vHz5EvPnz0fjxo1hY2MDGxsbNGvWDMuXL0daWppedc9NWFgYRo0ahRo1aqBMmTKwsbFB3bp1MWnSpBz73Kjqvn79egDA3bt3s+1XXkVFReGzzz5D48aNYWdnBzMzM5QrVw7169fHwIEDsW7dOiQmJuaaT3BwMAYOHIiKFSvC3NwcFSpUwODBgxEREZHrthkZGdi4cSO6du2KcuXKwczMDM7OzmjXrh1+/PFHpKam5rht1jtAExISMGvWLHh5ecHe3l5+r/38/CBJEu7evQsAWL9+fbbXzs/PT2sZcXFx+Oabb9CyZUs4OTnB3Nwcbm5u6NmzJ7Zv357r/u3YsQO9evWSXxsbGxtUrVoVPj4+mD59Oi5cuCCnVfUdHD58uPy/KlWqZKtrXvsVqvZftY/Xr1/HqFGjUKVKFVhYWKB8+fLo27dvrv1T4uLisHbtWrzzzjuoU6cOrK2t5WPmzTffxMqVK3N8v1SfV5Wvvvoq237l1jdu69ateOONN+Ds7AxLS0vUqlULU6ZMQWxsrM7tbty4gXHjxqFevXpynd3c3NCoUSOMGDECv//+u8aXsr5yu4sw6/FZEOeW4OBguYyDBw9qTdOhQwc5zcKFC7WmmThxIiRJgouLS46XPffs2YM+ffrIx7KjoyNatmyJuXPnIikpKcc6Zu1TmJKSgsWLF6NFixZwcnLKV5+87777Ts6zR48eePnypXyc6/r85PWzk9/zU7du3SBJElq2bKl1fVBQkFwne3t7pKenZ0sTFxcHIyMjSJKEH374Idv6kJAQjBw5EjVr1oSVlRUsLCzg7u6OJk2aYOzYsdi9e7f291KQotauXSsACAAiMjJSa5qYmBjRunVrOZ22h6urqzh37pzW7YcOHSoAiMqVK4uIiAjh4eGRYz5jx47Nsa4ZGRli9OjROW5brlw5cenSJVG5cmUBQAwdOlTeNjIyUmf9VY+1a9fK28yYMUP+f3R0tGjYsGGO27311lsiPT09P2+BbM6cOcLIyCjHMszNzcX69euzbafPfuXFyZMnha2tba557tmzJ8e6zJgxQyxbtkyYmJho3bZMmTLixIkTOdbh6dOnuR5ztWvXFnfu3NG6vfp7d+PGDa3H3Nq1a4Wvr2+u++nr65st/3379gl7e3ud23Xr1k08e/Ys27ZpaWmib9++uZbbpEkTeZtjx47p9T4fO3Ys9zdYjWr/fX19xf79+4WVlZXWfI2MjMTChQtzzEf1mdP18PLyEg8fPszXtuqfZfXX4siRI+Ltt9/Ocbvq1atrLVMIIbZu3SrMzMxyLTssLCxPr6kQmucb9XOKSmGcW9LS0uTP8aeffpptfUpKiihTpozG8aqNqm4BAQHZ1r148UL4+/vrfP3c3NxEaGio1rzVv3+Cg4NFo0aNsm0/Y8YMOb22c7u6KVOmyNsNHjxYvHr1Sggh9Pqcq392tJWtzpDz03fffScACBMTE63nh1mzZmnkExwcnC3Njh07cjw+v//+e53fI6qHtrIZYCkstwArKSlJ1K5dWwAQZmZmYvTo0WLXrl3i0qVL4tSpU2L27NnC0dFRABBly5bVekCpAixnZ2dRo0YNYWNjI7744gtx/PhxcfHiRfHLL7+IihUryvX466+/tNZ1zpw5cpqKFSuK5cuXi/Pnz4uTJ0+Kzz//XFhaWoqqVasKZ2fnbB/C1NRUERYWJtasWSPnsWbNGhEWFqbxiIuLk7dRPwm2atVKmJmZifHjx4vDhw+LkJAQsXnzZvm1ASBWrFiR7/fhhx9+kPNxdnYWCxYsEGfPnhVBQUFi5syZ8hefJEli3759Gtuq6t6zZ0/5hJZ1v/T18uVL4ebmJgAIGxsbMWXKFHHgwAEREhIizp07J37//XcxceJE4e7urjPAatGihZAkSTRs2FCsWbNGBAcHi5MnT4pJkybJH/5KlSqJlJSUbHmkpaWJli1bynn5+vqKbdu2iYsXL4rdu3eLXr16yeuqVaum9USh/t41aNBAmJqainHjxonDhw+Lixcvii1btogzZ86I27dvi7CwMHmfe/bsme21u337tkbehw4dEsbGxgKA8PDwEPPmzRPHjx8Xly5dEnv27BHvvPOOXHbv3r2z1W3ZsmXy+jZt2oh169aJU6dOidDQUBEYGCiWLFkiOnfuLJo1ayZvk5SUJMLCwsQ333wjb3vw4MFsdU1KStL7vRbif188NWrUEPb29sLOzk7MmTNHnDlzRpw5c0bMnj1bI9j+888/teZTsWJF0bx5czFr1iyxd+9eERwcLE6fPi02btwoOnfurPFeZnX9+nURFhYmp3n//fez7df9+/fl9OoBVqtWrQQA0atXL7F9+3YREhIi9u/fL7p16yanGTBgQLYyo6Oj5c+Ui4uL+Prrr8WhQ4fEpUuXxJkzZ8TGjRvFqFGjhJOTU4EHWAV5bunSpYsAIJo3b55t3alTpzS+cO3s7ERaWppGmtjYWPnzumzZsmx59OvXT96+YcOG4tdffxXBwcHi4MGDYvjw4UKSJAFAODg4aLyHKurfPw0aNBCSJIkhQ4aIffv2iZCQELFjxw6xf/9+OX1OAVZ6erp499135bzGjx8vMjIy5PWqz7muz4/6Z0dXgGXo+en8+fPy+gMHDmTL/4033tB4X+bPn58tzYQJEwQA4eTkpLGfly9flt+vKlWqiIULF4rAwEARGhoqTp06JdasWSMGDx4srK2tGWAVhtwCrA8//FD+8GmLpIUQ4s6dO6J8+fICgHjnnXeyrVcFWACEvb29CA8Pz5bm33//FRYWFgKA6NGjR7b1UVFR8vqqVauKR48eZUtz+vRpjV+k2n7lqJ+cc/u1r34SNDU11Zr+6dOnwtXVVT5B5EdMTIz8S9LNzU38999/2dJcunRJ/kKoUKGCSE1NzZZGvaUwvwIDA+V91hZAqbx69UokJCRk+7/6iaFr165aAyj1k9z27duzrV++fLm8fsiQIRonEJVp06bJaaZMmZJtvfp7Z2RkJA4dOqRzv3P7ZaySlJQkv9+dOnUSz58/15pu5cqVcvlHjhzRWOfj4yN/6al+YWvz9OnTbP/Tp8U5L9R/2dvZ2YmrV69mSxMeHi4HWW5ublrf0xs3bugsR/2HTdbXQ0XXl5q6rK1533zzTbY0GRkZolOnTgLIbCmIiYnRWL969Wp5e10B1IsXL0RycrLO+miTlwCrIM8t8+bNk1+DrF+oqpaSjh07CktLSwFkby1Rbym5cuWKxrq9e/fK69544w2tx4X656Bfv37Z1qsfzwDE6tWrde6Pts9pSkqKCAgIkPOYOXNmjtvr+/nRdSwaen5KS0sTNjY2Asjespiamip/F/To0UMA2lsWVa2KWX/ATZ8+XQAQVlZWIjo6Osf9i4+P19oqygBLYboOuMePH8tBzZIlS3Tm8+OPP8oni6xfOuoB1tKlS3PMY8CAAQLIbAnL6ttvv5Xz2LVrV455TJo0qUACrI8++ijHdJ999pmcLj4+Xmee2qhOggDEli1bckynHphs3bo123olAqxNmzbJZWgLoHKj2tbCwkJrECyEEImJiXIgPGnSpGzrVb/cnZycRGJiotY80tLShKenp3y8vHz5UmO9+ns3YsSIXOutb4Clan3StX8qzZo1EwDEoEGDNP5fo0aNHPc9NwUZYC1YsCDHdOrHqLZjTx9eXl4CgPjwww+1rs9PgNWkSROtX3BCCPHXX3/leM6YPXt2jucaJeQlwCrIc8u5c+fk7bO2lnTo0EEAmS0k7du3l5fVTZw4UQAQjo6O2V5nVeuYqamp1h+FWcsxMTERUVFRGuvUj+f27dvnuj9ZP6fPnj2T85ckSWsrW07l5TfAUuL8pGrVzdqyGBQUJAAIW1tb+TjP2rKo3qqY9fv0vffeE0Dm5fj8YCf3QnTw4EG8fPkSANCvXz+dadu2bQsAePXqFUJCQrSmkSQJb7/9do55NGnSBEBmB76sneYDAwMBAI6OjjrHLhkyZIjOeubXoEGDclynqjcAREZG5jlv1fhV9vb2CAgIyDHdu+++m20bpZUvX15eXrt2bb7z6dixI1xcXLSus7GxQY0aNQAAt2/f1lgXFRUld4Dv168fbGxstOZhbGwsd1iNi4vDpUuXcqyLrvcur3bt2gUA8PX1zXH/VFSfiawdxFWv8Z49e/DkyRPF6mYISZIwdOjQHNcPHz5c7pCd27EnhEB0dDRu3LiB8PBw+eHm5gYAuHz5smL1fvvtt3O8iUP9c5n1OFO9B3FxcfJ7WlQK8tzSpEkTWFtbA4BGJ+5Xr17hzJkzADJvdFDd5JC1o7fqedu2bTVe57S0NJw4cQJA5mfd3d09xzq899578ja6OpLn9XMaGxuLDh064MiRIzAxMcGGDRvw4Ycf5imPvFLq/KR6vUNCQjRuAlC9Pj4+PmjVqhUsLS2RkJCA0NBQOc3JkyeRkZEBIPM8pE51XF+9elXjJhl9McAqRBcvXpSXy5cvn+3OHvVHvXr15LSqu/mycnJygqOjY47lOTg4yMvPnj3TWBceHg4AaNSoEYyNjXPMo379+jA3N9e9Y/ng6emZ4zpd9daHat+8vLx03jrs6uoq34mp2kZpbdq0QdWqVQFk3j3UrFkzfPvttzhz5ozOu/ay0vV6Af97zXJ6nwGgefPmOvNQX6/r9WjQoIHOfPJC9Zk4ePCgzs+DJElYsGABgOyfB1Ugc/PmTVSvXh0jRozAli1bcP/+fcXqmVdVqlSBk5NTjuudnZ1zPfb27duH7t27w87ODuXLl0etWrVQv359+bFv3z4AUDSozO/nskePHvLcc/7+/mjfvj0WLVqEkJAQrXdtFaSCPLeYmJigdevWADSDpwsXLiA5ORm2trbw8vKSv/BPnTol7398fDyuXLkC4H8Bgcrt27eRnJwMoGg+pw8fPkTbtm1x/vx5WFpaYseOHYr+kMqJUucnVWCUlpaGoKAg+f+q98jPzw9mZmbynYbq751q2cHBAfXr19fId+DAgTA1NUVKSgpat26Nt956CytWrMA///yj18C3DLAKUUxMTL62U33wsipTpozO7YyM/vf2Zj3JxcXFAUCurQbGxsYoW7asPtXME11111VvfahuJXd1dc01bbly5TS2UZqpqSn27NmD2rVrA8i81XvatGlo3bo17O3t0aVLF2zevDnX/dT3vc6aj/p+5fZ6qF6LrNtlpdTx8OrVK53DkeQk6+dhxIgRmDZtGkxMTJCQkIC1a9fi7bffhru7O6pXr45PPvkkW4tLQcvtcwX87/3I+loLIfDuu++ie/fu2LdvX66BwIsXL/Jf0Szy+7l0dHTE7t27UaFCBQghcOzYMXz00Udo2rQpHBwcEBAQgL179ypWT10K8twCaG8tUW8pMTY2RvPmzWFpaYnExES5teTEiRM5tpQU9ef00KFD+OeffwBkDuvRvXt3vbc1hFL73bRp02wti1lbFdX/aguwsrYqApnB+pYtW1C2bFmkpaVh7969eP/991GvXj24uLhg8ODBOHXqVI51ZoBViFQfaDMzM4SFhen96NmzZxHXvGTSZ7wqfX6FGKpOnToICwvDjh07MGLECFSrVg1A5hfjX3/9hUGDBqF58+b5DsD1ldvroe9roavFMy/Uv+D69euXp89EVrNnz8bNmzcxe/ZstG/fXv6SvXXrFhYuXAhPT0+sWLFCkXrrw5Bjb82aNVi9ejWAzBbmdevWISIiAomJiUhLS4PI7DuLwYMH68ynsPn4+ODmzZvYuHEj3n77bVSsWBEAkJiYiO3bt+Ott95C586dc/zBWFJoay1RXd5TfYFray1RpSlbtqzO1qX8jLOnTV4+p61bt5YDnJkzZ+LkyZOK1CEvDDk/mZiYoFWrVgD+93oHBwdrtCoCyNayqKtVUSUgIACRkZH4+eef0bt3bzg7OwPIbDneuHEj2rZti2HDhsnBszoGWIVIdTkvNTUVjo6OqFevnl4PVdO7klS/bnL7Uk9PT5dbu0oK1WWAnC6tqnv06JHGNgXF2NgYvXr1wurVq3Hz5k1ERUVh9erVcp+QkJAQjB49WvFy1fcrt9dD9Vpk3a6gWFhYyIFQfHy83p8H9cvn6ipXroxp06YhMDAQ8fHxCAoKwoQJE2BhYYFXr17hgw8+0Oh7UZDUX8ucqD57WV/rX375BQBQrVo1nDlzBkOHDoWnpydsbGw0vjSL4+fSwsICgwYNwqZNm3Dv3j3cunULS5culadROXjwID7//PMirqVhvL29YWVlBSDzy1xbS4n6suoLX1dLSV4+p+rrlfqcVq9eHUePHoWLiwuSk5PRrVs3jUttBUXJ81PWlsWsrYoAsrUs6up/pc7Ozg6jRo3Cn3/+iZiYGPzzzz/49ttv5X6Q69evx7Jly7JtxwCrEKmiaCCzSbYo1a1bFwDw999/62wqDwsL0znyslK/tpSk+gIODQ3Fq1evckwXExMjjzie05d2QSlfvjxGjBiBs2fPonHjxgCAvXv3Knq5B9Dcr/Pnz+tMq96J09DXQ9/jQvWZOH36tKItG6ampmjdujUWL16MzZs3A4A8e0J+6plXkZGRePr0aY7rHz9+LM8kkPW1Vl2q6dmzJywtLbVuL4TQeSNCcVG1alWMGzcOwcHBcovW1q1bi7hWhsnaWhIcHIznz59rtJQAmq0lT58+lW9G0NZSUrVqVfnHRmF+TtXVqVMHgYGBcHJyQlJSErp27ZrrjAOGUvL8lLVlUb3/lUrWlkVVmtxaFbOqU6cOPvvsM5w7d04OtrUd1wywClGXLl3kTteLFi1SbDqY/HjjjTcAAE+fPpU7y2rz66+/6szHwsJCXs7PFBgFoUOHDgAyW0X+/PPPHNOtXr1abnZWbVPYTE1NNU4M+emTpIubm5vc/2vbtm059udJT0+XpyApW7asHPTll+q4yO2Y6NGjBwDg+fPnWqeoUILqWAeydwgvqONXCKHzs7Nu3bocjz3VeUFXwLl7925ERUXprIO+70FhsLW1hbe3NwBlO+UXFfXWElXfMvWWEkCztWTJkiU6W0pMTEzk/x8+fBj37t3LsexVq1YByGwVz+myVn7Vq1cPgYGBcHR0xLNnz9C5c+dcAx9DKHl+Um9ZPHz4sNZWRfXn6gGWj4+PRv88fbm7u8uts9qOawZYhahChQryraaXL1/G6NGjdQZZMTEx8odJaUOHDpXvDpw0aRIeP36cLc3Zs2dz/dJTH4bg1q1bylYyn4YPHy7/Gvz444+1nqwuX76MOXPmAMh8X3r16lUgdTl16hRu3ryZ4/rU1FS5b4a1tbV8fV9JY8eOBZDZajJu3DitfRm++uorXL16FUDmbeCG3jmqOi5yOybGjBkj3203ffp0HDhwQGf606dPZ+sfsnHjRp2fI/XW4ipVqmitpz51zatZs2bh+vXr2f4fERGB2bNny+Vn7WOpGnJjz549Wi8D3rp1Cx988EGu5ev7Hijh4MGDePjwYY7rExIS5BaIrO9BSaT+o+jHH38EkP2LXL21ZOnSpQAyh45p2LCh1jxVn9NXr15hxIgRWu8yXrNmjXw8BwQEaBy/SmnQoAGOHDkCBwcHJCYm4s0339S4A15pSp2fTE1N5dd79erVWlsVgf+9TydOnNDZqggAO3fu1Pmj9969e7h27RoA7ce1SY5bUoFYuHAhzpw5g/DwcKxZswbnzp3DqFGj5PFV4uPj8c8//+DIkSPYv38/6tevrzFek1Lc3NwwY8YMTJs2Dbdv30aTJk3w2WefwdvbGykpKTh48CAWLlwINzc3PH/+HI8fP9Z6OaVSpUqoWLEi7t+/jwULFqBChQqoVasWTEwyDy1XV9ccxzYpKM7Ozpg/fz7Gjh2LqKgoNG3aFJ999hlatWqF9PR0HDlyBPPnz0dSUhIkScLKlSsLbCb4wMBAzJo1Cz4+PujWrRsaNGgAZ2dnvHjxAjdu3MCKFSvkSz3vvvuu/LopacyYMdi0aRPOnj2L9evX4+7duxg7diyqVq2Khw8fYs2aNfJkytWqVcP06dMNLrNVq1Y4duwYgoODMXfuXHTp0kX+dWlpaYkKFSoAyGzZ2LJlC7p06YKUlBR0794dAQEBCAgIkG8GePjwIUJCQrBjxw5cuXIFy5Ytk8fEAoDBgwfjk08+Qe/evdGqVStUq1YNFhYWePToEQ4fPoyffvoJQGYA+84772jU08vLCxYWFnj58iWmT58OExMTeHh4yL9mK1SokONlOl1q1KiBmJgYtGjRAp9++qnGr+a5c+ciISEBALBs2TKYmZlpbDtkyBBMnjwZDx48QKtWrTBlyhTUrVsXL1++xNGjR7F48WKkpKSgcePGOi8TtmrVCpGRkdi9ezd+/vlntG7dWm7VsrW11etOR31t2bIFb731Fjp27IhOnTqhXr16cHBwwLNnzxAeHo7ly5fjwYMHAID3339fsXKLSrNmzVCmTBkkJyfL76W2L2k/Pz8cPXpUTqOrpaRbt27o27cvtm3bhiNHjqB58+b4+OOPUbt2bcTFxeG3337DmjVrAGT2Qfr+++8LZueQeXPFkSNH8MYbbyAuLg6dOnVCYGBgtmBFCUqen/z8/HDkyBGN1ztrZ39Vy6L6eFk59b9avHgxBg0ahG7duqF9+/aoXbs27OzsEBcXh4sXL2LZsmVytw6tx3W+hielHOkzsu3Tp0815hPT9WjXrl227fUdYTy3uuQ22bOTk5MIDg4W7u7uAoAYM2aM1nJUo85re+Q02bMueRkdXpfZs2fna7JnFSVGclffZ12P3r17ixcvXmTbXrU+t9G41ScZ1kbJyZ71cf/+feHg4KC1HG11DAwMFOXKldPrtcr6numzjb29vTh48KDWuqpPaJv1Ychkz3v37tWY/Ff9YWRklONI76mpqfK0NNoelpaWYuvWrbken6GhocLc3FxrHjlN9pzb/uZ0PKrPLqHrMXbs2HxNtJyXkdx1UercIoTmHHfa5h0UInOid/X91zXBtxDKTvasz8wEuc24cPHiRXkSdgcHB/H333/nq7zczmOGnp9Uss4HqW3eQSGEPNK+6r3L6ZjUZ1JrY2NjMWfOHK3b8xJhEXBwcMCBAwcQGBiI4cOHo0aNGrC2toaJiQkcHBzg7e2NsWPHYv/+/Th8+HCB1UOSJKxYsQK7du1Cp06d4ODgAAsLC1SvXh3jx49HaGgomjZtisTERACZd1Jo8/777+PPP/9Ep06d4OLiUiCtMPkxbdo0hIaG4r333kO1atVgaWkJKysr1K5dGxMmTMC1a9cKbKR6lSlTpmD//v2YNGkSWrRogUqVKsHCwgIWFhbw8PBA//79sW/fPvz5558a/YGU5uDggJMnT2LDhg3o3LkzXF1dYWpqCkdHR/j5+WH58uX4+++/UblyZUXKq1ChAi5cuICRI0eievXque5b+/btcevWLSxfvhydO3dG+fLlYWZmBgsLC7i7u6NTp06YPXu21vfs2rVrWLZsGXr16oU6derA0dERJiYmKFu2LFq0aIGZM2fi+vXr6NSpk9ay586di19++QU+Pj5wcHBQbBiKbt264eLFixg+fDgqV64MMzMzuLi4ICAgAEFBQfj444+1bmdqaop9+/Zh6dKlaNq0KcqUKQNLS0tUr14dY8aMwaVLl9C3b99cy2/UqBHOnj2LgQMHolKlSgUyYLDK4sWL8eeff2LMmDFo2rQpKlSoADMzM1haWqJmzZoYNmwYgoKCsHz58nz1dSmO1Fus2rRpo/W4ad68uca4XLruVAMy+81t374du3fvRu/eveHm5gYzMzOULVsWzZs3x7fffovr16+jUaNGSu2GTk2aNMGhQ4dgZ2cnj/SubZgUQyl1flK1LKrkdOmvXbt28rKuVsWtW7di06ZNGDZsGBo1aoRy5crBxMQE1tbWqFevnnxn8tSpU7VuL/1/dEmk1f379+VpG1atWoWRI0cWcY2Iii8/Pz+cOHECvr6+OqcxIaLSr3T8lKACs2XLFnm5RYsWRVgTIiKikoMB1mvs+fPnOu/8CQ0NxaxZswBkNhWrxs4iIiIi3YpHZxkqEo8fP0bt2rXRq1cvdO7cGbVq1YK5uTmioqLw119/YfXq1Xjx4gUkSSrQO1aIiIhKGwZYr7mXL1/it99+w2+//aZ1vZmZGX755ReN2+KJiIhINwZYr7EKFSrg999/x4EDB3Dx4kXExMQgLi4OZcqUgYeHBzp06IBx48YpdmcZERHR64J3ERIREREpjJ3ciYiIiBTGAIuIiIhIYQywiIiIiBTGAIuIiIhIYQywiIiIiBTGAIuIiIhIYQywiIiIiBTGAIuIiIhIYQywiIiIiBT2f56lZ7QWNDTsAAAAAElFTkSuQmCC",
      "text/plain": [
       "<Figure size 640x480 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "import csv\n",
    "import matplotlib.pyplot as plt\n",
    "\n",
    "directory_plots = './plots'\n",
    "\n",
    "df_longest_paths = pd.read_csv('./results/shortest_path_label.csv')\n",
    "\n",
    "shortest_path_lengths = df_longest_paths['shortest_path_length']\n",
    "\n",
    "plt.rc('font', size=20)\n",
    "plt.hist(shortest_path_lengths, bins=20, color='maroon', edgecolor='white')\n",
    "plt.title('Shortest paths in DAGs')\n",
    "plt.xlabel('length of shortest paths in workflows')\n",
    "plt.ylabel('number of workflows')\n",
    "plt.tight_layout()\n",
    "\n",
    "plt.savefig(f'{directory_plots}/shortest_path.png', format=\"png\", bbox_inches='tight')\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Measure the in- and out- degree of process nodes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import networkx as nx\n",
    "import os\n",
    "import csv\n",
    "import pandas as pd\n",
    "\n",
    "directory_path = './dags'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_processes_degree(dot_file):\n",
    "    G = nx.nx_agraph.read_dot(dot_file)\n",
    "    \n",
    "    processes = [node for node in G.nodes() if 'shape' not in G.nodes[node]]\n",
    "    \n",
    "    process_data = []\n",
    "    for node in processes:\n",
    "        in_degree = G.in_degree(node)\n",
    "        out_degree = G.out_degree(node)\n",
    "        \n",
    "        # e.g.: if its 2 in and 0 out -> we take the 2\n",
    "        ratio = in_degree / out_degree if out_degree != 0 else in_degree\n",
    "        \n",
    "        process_data.append((node, in_degree, out_degree, ratio))\n",
    "    \n",
    "    return process_data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The average in and out degree ratio "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def average_ratio_over_all_nodes(dot_file):\n",
    "    \n",
    "    process_data = get_processes_degree(dot_file)\n",
    "    total_ratio = sum(data[3] for data in process_data)\n",
    "    \n",
    "    if process_data:\n",
    "        average_ratio = total_ratio / len(process_data)\n",
    "    else:\n",
    "        average_ratio = 0\n",
    "    \n",
    "    return average_ratio"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "output_file = './results/in_out_degree_ratio_processes.csv'\n",
    "\n",
    "with open(output_file, 'w', newline='') as csvfile:\n",
    "    fieldnames = ['pipeline', 'average_ratio']\n",
    "    writer = csv.DictWriter(csvfile, fieldnames=fieldnames)\n",
    "    writer.writeheader()\n",
    "    \n",
    "    for filename in os.listdir(directory_path):\n",
    "        if filename.endswith(\".dot\"):\n",
    "            dot_file_path = os.path.join(directory_path, filename)\n",
    "            \n",
    "            average_ratio = average_ratio_over_all_nodes(dot_file_path)\n",
    "            if average_ratio != 0:\n",
    "                writer.writerow({'pipeline': filename, 'average_ratio': average_ratio})\n",
    "    \n",
    "    print(f'Results written to: {output_file}')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Average in-out ratio over all pipelines:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "       average_ratio\n",
      "count      73.000000\n",
      "mean        0.842224\n",
      "std         0.466091\n",
      "min         0.354167\n",
      "25%         0.538690\n",
      "50%         0.653472\n",
      "75%         0.936609\n",
      "max         2.238095\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "csv_in_out = './results/in_out_degree_ratio_processes.csv'\n",
    "df = pd.read_csv(csv_in_out)\n",
    "\n",
    "print(df.describe()) # mean 0.842224 -> might be intresting in writing"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAlgAAAG4CAYAAACZypnDAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjcuMiwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy8pXeV/AAAACXBIWXMAAA9hAAAPYQGoP6dpAABmTklEQVR4nO3deXwNV/8H8M9k3xcJIYTYxb4lBJFQ0lpqX2tJYm+LKtUWrWqL0mqrpS1KUS19UEuDttbEFntsj6jaSZBEIosQWc7vD0/md6/ce3OTmeyf9+t1X8adM+ecmTtz871nzpwjCSEEiIiIiEg1JsVdASIiIqKyhgEWERERkcoYYBERERGpjAEWERERkcoYYBERERGpjAEWERERkcoYYBERERGpjAEWERERkcoYYBERERGpjAEWlVvBwcGQJAmSJGHNmjXFXR1FAgIC5H0JCwsr7urkW2mvP5Ufnp6e8rl68+bN4q4OlWAMsMqhIUOGyF8QkiThq6++Ku4qERERlSkMsMqZpKQkbN++Xeu9tWvXFlNtiIiIyiYGWOXMxo0b8fTpU633zp8/j7NnzxZPhYiIiMogs+KuABUtzdYqKysrOdhau3YtmjdvXky1IqXYb4mIqGRhC1Y5cu3aNRw5cgQAYGJigoULF8rr1q9fj8zMzOKqGhERUZnCAKsc0Wy9CggIwIQJE1ChQgUAQGxsLP7888/iqhoREVGZwgCrnBBCYN26dfL/R4wYAQsLCwwaNEh+T19n99jYWJibm0OSJJiamiI6OtrocmvXri0/rRgaGqo33cWLF/H666+jbt26sLGxgaurK1q3bo3PPvsM8fHxAIA1a9bIeQUHBxtdB7Wkpqbi22+/xcsvv4xq1arBysoKzs7OaNy4MSZOnIjjx4/nK7+MjAwsW7YMAQEBqFSpEqytrVGrVi0MHjwYe/bskdMZ81h4QYY5+PPPPzF+/Hg0btwYLi4uMDc3h5OTE1q2bInx48fjjz/+MNiqefr0aXz22Wfo2bMnatWqBTs7O1hYWMDNzQ3t2rXDrFmzcPv27fwcEtVoPiWrWd9x48ahfv36sLOzg7OzM1q3bo158+YhKSkpzzx1HeN79+5h/vz58PHxQeXKlWFqagonJyed29++fRuzZ89G27Zt4ebmJh+rtm3b4qOPPsKdO3fyvZ/Xr1/HnDlz0LFjR1StWhVWVlawsbFBrVq10KdPHyxZsgSxsbF55vP48WP88MMPePXVV1GjRg3Y2NjA3t4edevWxahRo7B//36j67R//36MHj0aTZo0gZOTE8zNzeHq6govLy8EBARg1qxZOHjwIDIyMvTmcefOHXz88cfo2LEj3NzcYGlpCXt7e9SsWRM+Pj4ICQnBhg0b5O8GNWRlZWHlypXo3Lmzwesxv9Q8tgBw69YtvPPOO2jYsKF8Hjdt2lTregsLC5PP1YCAAJ356Euza9cuDB06FHXr1oWdnR0kScLixYt15hEVFYWZM2fCx8dHPqcrVqyINm3aYPbs2YiJicnXvmVkZGDdunUYNGgQatWqBXt7e9ja2qJmzZoYOnQotm7dCiFEvvIsVoLKhQMHDggAAoCwtrYWycnJQgghjhw5Ir9vYWEhHj58qHP7V155RU73xRdfGFWmZt4uLi7i2bNnOtMtXLhQmJuby2lffFWpUkUcOXJErF69Wn4vKCioQMdBU1BQkJzf6tWrDaYNDQ0VlStX1lvHnNdrr70mHj9+nGfZN2/eFE2bNjWY19ixY8WzZ89EjRo15Pdu3LihMz9/f385zYEDBwyWffHiRdG6des89wWAGDx4sM48vL29jdre3NxcLFy4MM/jkZ/6G0OzDkII8fHHHwsTExOD51h4eHi+6rht2zbh7OycKy9HR8dc286bN09YWVkZPFZWVlbis88+M2r/nj59Kt58801hZmZm1GeQc73rsnHjRqPO7Z49e4pHjx7pzSclJUX06tXLqPMCgPjxxx915rNs2TJhbW1tVB7t27c36njl5e7du6JVq1YGyxozZoxIT0836nrModaxzfHzzz8LW1tbvfk4ODiIbdu2aX3f+/v768zrxTSPHj0Sffv21Znv119/rbXt06dPxYQJE4SpqanB/bK2thZLliwx6jM4cOCAqF27dp7Hqm3btuLu3btG5Vnc2Mm9nNBsnerTpw/s7e0BAO3atUOdOnVw9epVPHv2DL/99hveeOONXNsPHz4cf/31FwDg119/xTvvvJNnmb/++qu8PHjwYJibm+dK89VXX+G9996T/29lZYVOnTqhWrVqiI+Px4EDB3Dv3j307NkTU6ZMMXp/1fSf//wHw4YNQ1ZWFgDA1NQUHTp0QJ06dZCamopDhw7Jv9TWr1+PGzduYP/+/bCystKZX1xcHDp37ozr16/L79WvXx/e3t4wMzPDhQsXcPr0afz444/y56SWsLAw9OrVCykpKfJ71atXh4+PDypUqIDHjx/jn3/+wblz55CRkZHridMcOb+ULS0t0ahRI9SpUweOjo4QQuDevXs4fvw44uPjkZGRIX++7777rqr7Yqxvv/0WH330EYDnLapt2rSBpaUl/vvf/+LEiRMAnrdEde/eHfv374ePj0+eeR49ehRz5sxBRkYGXFxc0LFjR7i6uiI2NhaRkZFaaSdOnIjvvvtO/r+trS06d+6MypUr4/79+zhw4ABSU1Px9OlTzJgxAw8ePMDXX3+tt+zU1FQEBgYiIiJCfs/GxgYdOnRAtWrVIIRAdHQ0Tp8+jYcPHyIjI0M+d1/09ddfY9q0aXKrgL29PXx9feHh4YGsrCxcunQJJ0+ehBACO3bsgL+/P44ePQobG5tceY0YMQJ//PGH/P86deqgRYsWqFChAjIyMhAXF4cLFy4YHJxz27ZtmDBhgvx/BwcH+Pr6olq1ajAzM0NSUhKuXLmCixcv4tmzZ3rzyY+EhAS89NJL+Oeff+T36tatCx8fH5ibm+Ps2bM4e/YsVq5cCTs7O6PzVfPYAsCmTZsQHByM7OxsAICZmRk6duyIWrVqITk5GeHh4Xjw4AEGDRqE+fPn5+sYCCEwfPhw7NixA5IkwdvbG15eXhBC4OLFi1otwY8fP8bLL78s9+cFgJo1a6J169ZwdnZGYmIijh49iujoaDx58gSTJk1CcnIyZs6cqbf8TZs2YdiwYXKrppWVFdq2bQtPT0+YmpriypUriIiIQGZmJo4dOwZfX1+cPHkSbm5u+drPIld8sR0VlcePHwt7e3v5F8DOnTu11s+ZM0de16ZNG7152NnZyekuXrxosMxnz54JV1dXOf3Ro0dzpbl48aKwsLCQ0/To0UPExsbmKnfChAkCgLC0tJTTFlUL1tWrV7X2u3Xr1uLKlStaabKyssSXX36p1UIyadIkveUOHTpUTmdjYyM2bdqUK82hQ4eEu7u7kCRJ6xgpacG6ffu21mdSs2ZN8ddff+lMm5CQIJYtWybeeecdnetff/11sXPnTpGWlqZzfWZmpli9erX8a9vc3Fxcv35dZ1pj658fOXkBz1tmraysxLp163KlO3bsmFaLRP369cWTJ0/yrKOZmZmQJEl8+umnuVpmnz59Ki//5z//0arLyJEjRVJSklb6pKQkMXz4cK10mzdv1rtvgwcPltOZmpqKjz/+WKSmpuZKl5WVJfbv3y969+6ts3Vk79698jlrbm4u5s6dqzOfyMhI0bBhQ7nM119/XWeanPV2dnZi165deut/7do1MXfuXPHHH3/kWtesWTM5n4kTJ+ptDU5JSREbN24U7733nt5yjKX5PWBlZSV++eWXXGn27NkjXzuare36rkc1j60QQty/f1+rtbRNmza5rqeMjAzx8ccf5/quNKYFK6cltEmTJuL8+fO50mqe0yNHjpS3q127ttizZ0+u9JmZmeL777+X62Fqaqrzb4AQz/8O2NjYyHlOmTJF552Ua9euiQ4dOsjpunXrpjO/koQBVjmwbt06+aSsVKmSyMjI0Fp/7do1rS/3qKgonflo/hGYMWOGwTJDQ0O1LkJdBgwYoBW4pKen681v4MCBWnUsqgBL88ukVq1aIjExUW9+X331lZzWxMREZ0Bx8eJFrf34/fff9eZ39uzZXLeAlARYw4YNk9PUqFFD3L9/X2/Zavntt9/kMt9991296QozwAIgfvvtN71p//nnH60v+O+++y7POgIQc+fONViHrKwsUbNmTTl9//79RXZ2ts602dnZonfv3lrXTFZWVq50e/bs0arDhg0bDNbBUN3q1q0r56MrqNB07949UalSJTlguHPnjtb6JUuWyHnNmjWrQHVKSUmR8/Dw8NB7rNR0+fJlreNp6DgcOXIk121mXdej2sdWCCHeeecdOb/q1asb/B6aPn26Vh2NCbAAiMqVK4u4uDiDdT148KCcvlq1auLevXsG0//0009y+ldeeUVnms6dOxt9TaWmpgovLy85/bFjxwymL24MsMqBLl26yCfkW2+9pTNN+/bt8wye/vrrL60/0Ia+ADV/Zc+ePTvX+ocPH2r9Ety/f7/Bfbh9+7bWl1tRBFiJiYlavwQNBUNCPP9ibdSokZz+/fffz5Vm2rRp8vqOHTvmWcexY8eqEmDdvXtXK1j7888/8yxbDZmZmXILYMuWLfWmK8wAy5jjPGvWLDl98+bN86xj1apVc/1QedGff/4pp7ewsBAxMTEG09+9e1frmtDVuqjZF1Jf/zhjbNu2Tc7npZdeMmqbzz77TN7myy+/1Fo3b948ed3ixYsLVKfo6Og8PwO1aQYjbdu2zTP9iBEj8rwe1T62WVlZomLFivL6n376yWBeqampwsnJKd8B1vfff59nPfv06SOn19Ui/KLs7GzRoEEDAUBIkpQrgDt79qycX7169URmZmaeeW7YsEHextCdgpKATxGWcXfv3tV6SmX48OE6040YMUJeXrdunXyfX1OXLl1QuXJlAM+fZNG8B68pJSVFqy+GrjKPHj0q32+vUqWK3iddcnh4eKBjx44G06jt6NGjSE9PBwC4uLigV69eBtObmJhg1KhR8v8PHDiQK43mE36vvfZannUYNmyYkbU1bO/evfITgXXr1sUrr7yiSr4AcOHCBaxduxYff/wxpk2bhokTJ8qvt956S+6/ceHCBZ3nVWEbOXJknmmCgoLk5XPnziExMdFg+v79+8PMzHAXVs3rrlu3bqhSpYrB9FWrVtX6XF48f9LT07XOn0mTJhnMz5Bdu3bJy0OGDDFqm86dO8vLhw8f1lpXvXp1eXnt2rV4/PhxvutUsWJFWFtbA3h+rhw8eDDfeeSX5jHW/A7Ux5hzSe1je+nSJcTFxQEAzM3NMWDAAIN52draok+fPkaVm0OSJAwePNhgmszMTPlpSjMzM/Tv39+ofDt16gQAEELg6NGjWus1j9XAgQNhamqaZ56GjlVJw07uZZxmsNSgQQO0bt1aZ7pBgwbhrbfeQnp6uhyUdenSRSuNqakphgwZIj+y+8svv6BDhw658tqyZQuePHkCAPDx8UHdunVzpdGcmsfb21urE6U+3t7eRTpiuWZn5TZt2uT5BxUA2rdvr7W9EELet+zsbJw/f15eb0xn6latWkGSJMWPJh87dkxeziuYNdbatWsxf/58XLlyxaj0GRkZSEpKgrOzsyrlG6tt27Z5pqlbty5cXFzw8OFDCCFw9uxZ+Q+DLq1atcozT83zR/O8MKR9+/bycCZnzpzRWnf27Fn5oQMbGxu0adPGqDx10ewgv3PnTqOmytIcyuLFISW6d+8OOzs7pKamIjIyEvXr10dISAh69OiBVq1a6XzA5UXm5ubo27cv1q9fj6ysLHTp0gUDBgzAgAED4O/vDxcXF+N30AhCCK3r0Zjj6ePjk+f1qPax1dzey8vLqAdfvL29sWbNmjzT5fD09JTHRNTn/PnzcuBsZWWF6dOnG5X3yZMn5eUX903zWEVERGDixIl55qd57AsytElRYoBVxv3888/ysqFfaM7OzujZsyd+//13AM//eL4YYAHPW6NyAqxNmzZhyZIlub48NZ8e1Ndipjl+TbVq1fLeETz/hV+Ucn41AkCNGjWM2sbT01NefvbsGVJSUuDg4AAASE5O1hr7x5j9trOzg6OjIx49emRcpfV48OCBvFyrVi1FeQkhMHr0aKxevTrf26akpBR5gKXZumKIh4cHHj58CED7s9elYsWKeean9Px5cYwnzc/Qw8PDqIBfH83xibZt25bv7V9s4atQoQJWr16NYcOG4dmzZ4iOjsbcuXMxd+5cWFtbw8fHB/7+/ujVq5fB4PTrr7/GmTNncPnyZWRkZGDDhg3YsGEDJElCgwYN4Ofnh8DAQPTo0UPvU7rGSkpK0noS0ZjzxMHBIc/rUe1jWxTflcacz5r7lZqaqvVkrLFe3DfNPPfv35/vMcHyamkubrxFWIYdP34cly9fBvC8qTav202aAdiWLVu0HuXP0apVK3h5eQF4/njzi6O/379/X75IzMzM9DY7p6amysv6Hkt+ka2trVHp1KJZR2PLfjGd5jHUzA8o2v3WrEd+HjXX5ccff9QKrnr27Il169bh4sWLSExMRHp6OsTz/p0QQmgFF8Vxi7Agx1nXua8p51aWIUrPnxfroOZnaMzAqoboGoB2wIABOHXqFAYOHAgLCwv5/SdPniA8PByffPIJWrdujdatW+u9/VepUiWcPHkSc+bMgbu7u/y+EAJRUVFYsWIFBgwYAHd3dyxYsEDv8BPGKKzrUe1jWxTflcacz0r3C8i9b0rzVPL5FwUGWGWY5thXQgitEcF1vTTv26elpWHz5s0689UM1DRbqwBgw4YN8kkfGBiISpUq6cxD8wsgLS3NqP0pSL8OJTT/iBlb9ovpNJvzX/zSK8r91qzHi39Y8mvRokXy8rx58xAaGorhw4ejUaNGcHJy0vrjCuQdrBS2ghxnNcYfU3r+vFgHNT9DzXPx7NmzWgGxMS99Y1k1adIEGzduRFxcHHbs2IF3330Xvr6+Wq3cp0+fRqdOnbBp0yadedjZ2ckj2588eRJffvkl+vTpA1dXVzlNYmIiZsyYgf79+xf49vmLQapa16Pax7akfFdq1qN58+b53i8hBObMmaM3z23bthUoz5KMAVYZ9ezZM/znP/9RlIfm7UVNw4cPl/sVhYaGIjk5WV73yy+/aKXTR/PL8u7du0bVJz9T9KhBs9nc2Clfbt26JS9bWFho/VF0dHTU+kNjzH6npqaq8stRc0C+GzduFDifO3fu4N9//wXw/LZyXoOHJicnF3szvrGfnWZ/Ds3zs6CUnj8v1kHzM7xz546iydk188r5PNXk4OCAHj16YOHChTh69Cji4+OxZs0a+RZodnY23njjDbmvpi4mJiZo3bo1pk6diq1bt+LBgwc4fPiw1g/B7du3y90a8svR0VHrx4Axn1FycnKe16Pax7akfFdq7tf169dVaY0u7POwuDHAKqNCQ0ORkJAA4Hnn0TZt2hj10uwfER4ervOXao0aNeROu0+ePMGWLVsAAJcvX5Y75trZ2aF3795669e8eXN5+dSpU0b9EtHsLFkUWrRoIS+fOHHCqOZozScrW7RoodV538TEBE2aNNHKMy+nT59W5VeaZkdvXU83Gkuzz0T9+vXz7Ad0+PDhYv+VqdmRVp9///1X7n8lSZLWZ19Qmnm8+PSUPprnT8uWLbXWNW/eXO53lJaWlu+5LzVpduj++++/C5yPsRwcHBAUFIT9+/fD0tISwPO+RcZ8NjlMTEzQvn17bNmyBS+//LL8vuYTy/khSRKaNm0q/1/zQRB9Tpw4kef5rPax1fyujIqKMqr1sjC+K5s3by5/dsnJyfn67PQp6vOwqDHAKqM0bw/27NkTx44dM+p16tQp+ak/8cIE0Zo0W6dybhNqtl7169fPYH+Bdu3aya05MTExeT4deOfOHRw6dMjwTqusXbt28hdKXFwcdu7caTC9EELruGs+TpxD8wm+9evX51mHF2/BFlTXrl3lYOjff/8t8JeZicn/f2UYc7vihx9+KFA5atJ3DmvSfOKqWbNmqnTE1/z8d+3aleeky/fv35eno3pxe+D5tESaTzYuXbq0wHXr2bOnvLx+/XqjJoRWQ82aNdGoUSP5/5od940lSRJ69OihKI8cmsdT8/tLH83rWx+1j22jRo3k1tCMjAy9t1ZzPH78uECd6/NibW2tdU4ams7JWJrHat++fbhw4YLiPEsUJYNoUckUGxurNaikoWk3dNGcOqdOnTo60yQkJMhTuJiYmIjo6GitUat3796dZzn9+/eX0+c1kvugQYO0BsUrjpHc69WrZ3DSXM3RrPWN5H7hwgWt/diyZYve/EriSO5xcXFCkiR5H69evao3reYo7mrUPz9eLNfQSO5XrlzRGsl96dKlqtTxxZHchw4dqjdtdna21mwFhT2Se2ZmpqhTp46cT9euXfVOxv6ilJSUXNO+5DUCeI6MjAxRpUoVuVzNaVaSk5MNfgdo0hzZfNiwYUZto0tUVJTW8VRjJHe1j60Q2gMUF8ZI7vrSvCgsLEwrb33fm7roG/U9ICBAzq9p06a5ppLSJz09XSQkJBhdfnFggFUGff311/IJ6+DgoHduNX3+/fdfrYvo8OHDOtNpjurbr18/eblKlSpGjch7/vx5rZGrS+JchNeuXdOai9DX11dcu3ZNK01WVpb49ttvtWaWNzTCsGawaGtrq3MuwsOHDxfKXIQuLi5yOkNzESYmJorly5eL6dOn51rXtm1bOY8WLVqIy5cv5zoeS5cuFebm5sLU1FRYWVkVa4BlaC7C48ePa81FWLduXaPmIjS2ji/ORThmzBiRkpKilSY5OVmEhIRopTP0o0gzEMuZi1DXnH05cxH26dNH51yEe/bs0Tpn27ZtK06cOKG33HPnzon3339fODs7iwsXLmitCw4OFh06dBBr1qzR+0cvNjZWayR0BwcHrbksDxw4ICpXrixmz56td67TzMxM8csvv2h9F/z6669662wMzTrpm4tw7969+ZqLUM1jK4QQMTExWqOz65uL8JNPPsn1XalmgCWE9vemJEninXfeEfHx8TrTPn36VGzbtk306dNHeHt760xz4cIFre/Y+vXr65zfMMe///4r5s6dK9zd3UVoaKjR9S4ODLDKoObNm8sna0hISIHy8PHxkfMYO3aszjSbN2/O1VIAQEydOtXocj7//HOtba2trUX37t3F2LFjRb9+/eQvFScnJ3kiUyX7pcmYAEuI5y0xml+WZmZmolOnTmLMmDFi6NCholq1alr70LZtW4NB7YMHD4Snp6fWNvXr1xcjRowQISEhonXr1lrHUjMAuHXrls48jf3jv3fvXq0vM+B5a9bAgQPF+PHjxbBhw4S3t7f8R6R3794689D8JW9ubi4CAgLEqFGjxKBBg7RaKObNm6dV/+IIsL755ht5uXbt2mL48OFi1KhRok2bNlrpbGxsREREhN48C1rHN998U6sce3t70atXLzF27FjRu3dvrYnYgeeT3RqSlJSkdX3mBOqBgYFi9OjRYtSoUeLll1/WCqb1tXisWLFC69zOCTIHDRokxo8fL0aOHCkCAwPlufJyXi8GAZrXkqmpqWjYsKEYMGCAGDt2rBg+fLgICAjQ+qEAQKxYsUIrjxenbnFzcxOvvPKKCA4OFmPGjBE9e/bUOrcACD8/P50tffkRHx+v1eKUcwxGjBghgoODRYsWLbQ+G2POZzWPbY7ffvtNbj3O+R7q3LmzGDNmjBg0aJBwc3OTr8cvvvhCTtepUyed+RU0wHr69KkIDAzUqrOFhYVo3769GDlypBg/frwYMmSI8Pb21vpx1apVK715hoaGarUiA8/no+zXr58YN26cCA4OFj169Mj1XcsAi4rUuXPntE7AvXv3Fiifb7/9Vs7D0dFRZ8Dw9OlTrV9VOa8zZ87kq6z58+fnuhWm+apSpYo4cuSIWLFihfze5MmTC7RfmowNsIR4/gWQ8wVm6DV06FCdrQkvun79umjcuLHBvMaMGSOePXsm3N3d5ff0tQ7k54//2bNnRbNmzfLcF0D/7ZcffvjB4GdmYmIiZs+eLbKzs4s9wBJCiI8++kjrj9OLr8qVK+dZrpI6fvrpp1qtCrpeVlZWYv78+Ubll5aWJsaOHZvrD7i+fA3d2t6/f7/W5MR5vRo1aiSio6O18pg4caLR29vb2+cKroQQ4tixYwbPqRdfAwYMMLhf+XH79m2tQErXKyQkRKSnpxsdYKl1bDWtWbMmVyCi+XJwcBDbtm0Tu3fvlt/r1auXzrwKGmAJ8bwl8cMPPzRYF82Xubm5ePPNNw3mefbsWdGqVSujj5Wnp6eIjIzMV72LGgOsMmbq1KnyCVilSpUC/7p78OCB1pedvr4eL05G7OXlVaDyzp8/L8aNGydq1aolrKysRIUKFUTLli3FvHnz5P4dCxYskMvRNYF0fuUnwBLief+Ib775RnTt2lW4u7sLCwsL4ejoKLy8vMQbb7yR75ndnz17Jr7//nvh5+cnXFxchKWlpfD09BQDBw7U6sNmbW0tgOfN8fpuveb3j392drbYsmWLGDlypKhbt65wcHAQpqamwtnZWbRq1Uq8/vrrYteuXQbPn/Pnz4uQkBDh6ekpH4uGDRuKiRMnagXZJSHAEuL5H/CQkBBRp04dYWNjIxwdHUXLli3Fp59+arBPi1p1vHnzpvjggw+Et7e3cHV1FWZmZsLV1VX4+PiIDz/8UG/rpCFRUVFixowZwsfHR1SqVEmYmZkJGxsbUbt2bdG3b1/x/fffi4cPH+aZT1ZWlti8ebMICQkRXl5ewtnZWZiamgp7e3tRp04d0bNnTzF//nyDf9AuXbokvv32W/Haa6+JZs2aCWdnZ2FmZiasrKxE1apVRWBgoFi0aJF48OCB3jwSEhLExo0bxeTJk4Wfn59wd3cXlpaWwszMTFSoUEF4e3uLSZMmiePHj+f7WOUlIyNDrFixQvj7+2tdj/379xe7du2S0+UnwBJCnWOr6caNG+Ltt98W9evXl8/jxo0bixkzZsjnkGb/x5EjR+rMR0mAlSM2NlYsWrRIvPLKK6J69erCxsZGmJubCxcXF9GyZUsRFBQk1qxZk6vrhyF///23eP3110XTpk3l68TW1lZ4enqKwMBAMXv2bHHkyBGRnZ1doDoXJUmIEj5SF9H/DBs2TH7ybsOGDUZPpFqa/fvvv6hXrx6A58Mi5IzMT4ZpDo/Brzgqb2bNmoX58+cDAD777DO8//77xVyj8onDNFCp8PjxY62Z1729vYuxNkVn48aN8nJ52WciKjghhNYsHPzeKD4MsKhU+OCDD+QJVr29vVG7du3irVARuHXrlta0NEOHDi3G2hBRafDtt9/iypUrAIAqVarA39+/mGtUfjHAomK1efNmTJ8+HVevXtW5Pj4+Hm+88QYWL14svzd9+vQiql3hGTx4MH7//Xekp6frXP/333+jQ4cOclDZpEkTvPLKK0VYQyIqSY4ePYpx48bh3LlzOtenpqZizpw5mDp1qvze22+/nedsC1R42AeLitWaNWsQEhICAKhXrx6aNGkCFxcXpKen48aNGzh+/LhWEDJs2DCjRlwu6Tw9PXHr1i3Y29ujZcuW8PT0hJWVFR4+fIgTJ05ozYtma2uLw4cPa02ZQYaxDxaVNWFhYfLI8zVq1EDz5s1RqVIlZGVl4c6dOzh69KjWJM8dO3bE/v37YWpqWlxVLvcY2lKJceXKFblp+0WmpqZ488038dVXXxVxrQpXSkoKwsPDER4ernN9rVq1sHHjRgZXRCS7deuW1sTgLxoyZAhWrVrF4KqYsQWLitWzZ8+wZ88e/PXXXzhz5gwePHiA+Ph4pKWlwdnZGZ6enggICMCoUaNQv3794q6uam7cuIEtW7bg0KFDuH79OuLj4/Hw4UNYWFigYsWKaNWqFXr27InXXntNnrORjMcWLCprsrOzER4ejl27duHEiRO4d+8e4uPjkZKSAkdHR1SrVg0dO3bEiBEj2LG9hGCAVciys7MRExMDe3t7rS99IiIiKl2EEEhJSYG7uztMTAx3Y+ctwkIWExMDDw+P4q4GERERqeTOnTuoVq2awTQMsAqZvb09gOcfhoODQzHXhoiIiAoqOTkZHh4e8t92QxhgFbKc24IODg4MsIiIiMoAY7r8cBwsIiIiIpUxwCIiIiJSWakOsGJjY7Fjxw7Mnj0b3bp1g6urKyRJgiRJCA4ONiqPNWvWyNvk9VqzZk2h7g8RERGVDaW6D5abm1txV4GIiIgol1IdYGny8PCAl5cXdu/eXeA8/v77b7i7u+tdn9cjmURERERAKQ+wZs+eDW9vb3h7e8PNzQ03b95EzZo1C5xfvXr14OnpqV4FiYiIqFwq1QHWxx9/XNxVICIiIsqlVHdyJyIiIiqJGGARERERqYwBlobg4GC4ubnBwsICrq6uaNu2LT744ANER0cXd9WIiIioFCn0ACs9PR379u3Df/7zH5w4caKwi1MkPDwcsbGxyMjIwMOHD3H8+HHMmzcPderUwfLly4u7ekRERFRKKOrkfuvWLXz33XcAgJkzZ8LJyUlr/bFjxzBgwADcu3dPfq9ly5b4/fffUb16dSVFq6pWrVro168ffH194eHhAQC4fv06fv/9d2zevBlPnz7FhAkTIEkSxo0bZzCv9PR0pKeny/9PTk4u1LoTERFRySMJIURBN168eDGmTp2Kli1b4tSpU1rrUlJSULduXcTFxeHFIho1aoTIyEiYman7EKPmMA1BQUFGjbyelJQEBwcHvRM37tixA/369UNGRgZsbGxw7do1VK5cWW9+c+bM0fl0Y045pUV2VhZMTE1LXd5ERESFJTk5GY6Ojkb9TVcU4ezZsweSJKFPnz651q1YsQKxsbGQJAmTJ0/GSy+9hL///hvff/89Ll26hLVr12L06NFKileFo6OjwfU9e/bERx99hA8++ABpaWlYtWoVZs2apTf9jBkzMHXqVPn/ycnJcqtYaWJiaootw4YhLipK1Xwrenmh36+/qponERFRSaMowLp+/ToAoFWrVrnWbdy4EZIkoW/fvli8eDEA4NVXX0VcXBw2bdqEzZs3l4gAyxhjx47Fhx9+CCEEwsPDDQZYlpaWsLS0LMLaFZ64qCjcj4ws7moQERGVOoo6ucfGxgLIPSdgcnIyzpw5AwAICQnRWjdkyBAAwLlz55QUXaQqVaoEV1dXAOAThURERJQnRQFWSkoKACArK0vr/SNHjiArKwumpqYICAjQWpdzuywhIUFJ0UVOQVc1IiIiKmcUBVg5/ZdiYmK03g8LCwMANGvWDLa2tjq3tbKyUlJ0kYqNjcXDhw8BwOBk0ERERESAwgCrcePGAICtW7fK72VlZcn9rzp16pRrm5xbbC/eVizJVqxYIbdg+fv7F3NtiIiIqKRT1Mm9b9++CA8Px7p16+Dm5gY/Pz+sW7cOt27dgiRJGDRoUK5tcoZzKAnjYN28eROJiYlo0aKF3jQ7duzAp59+CuB5q9uLfcqIiIiIXqQowBo/fjyWL1+OqKgoLFq0CIsWLZLXvfrqq2jdunWubbZu3QpJknL1zSqIw4cP4+rVq/L/4+Pj5eWrV6/mGgcrODhY6/83b95Ep06d4Ovri1dffRXNmzdHpUqVIITA9evXsXnzZmzevFluvVq0aBGqVq2quN5ERERUtikKsCwtLbFv3z5MnDgRoaGhyMjIgIWFBQYPHoylS5fmSn/w4EFcunQJkiSha9euSooGAKxcuRJr167Vue7IkSM4cuSI1nsvBlg5IiIiEBERobccGxsbfP3113mO4k5EREQEKAywAKBy5crYvHkz0tPTkZCQABcXF1hYWOhM6+HhgQMHDgAAvL29lRatWKtWrfDLL78gIiICp06dwr179xAfH4/MzEw4OzujUaNGeOmllzBmzBhUqlSpuKtLREREpYSiqXIob/kZVr+kWd6ypeoDjVZu0QLj/zdGGhERUWmSn7/pip4iJCIiIqLcFN0irFy5Mvz9/eVXo0aN1KoXERERUamlKMCKjY2Vn7QDABcXF3Ts2BEBAQHw9/dHkyZNVKkkERERUWmiKMCaPHkywsPDceHCBWRnZyM+Ph5bt26VBx51dnbWCriaNWumSqWJiIiISjJFAdbixYsBAI8ePUJ4eDjCw8MRFhaG8+fPIzs7GwkJCdi+fTu2b98OAHBycoKfn58ccBka4JOIiIiotCqUpwiTkpK0Aq5z584hOzv7eYGSJP+bmZmpdtElDp8i1ManCImIqLQq9qcIHR0d0atXL3z55Zc4ffo0YmNj8eGHH8qTQwshwNEhiIiIqKxSPNCoLkIIREZGIiwsDGFhYTh06BCSk5O1giobG5vCKJqIiIio2KkSYOkLqHLWAYC1tTV8fX0REBCAgIAAtGnTRo2iiYiIiEocRQHW119/jQMHDuDw4cNISkoCYDigMjc3V15jIiIiohJOUYA1bdo0SJIEIQQDKiIiIqL/UeUWoSRJ8PPzQ2BgIAICAtCiRQv5aUEiIiKi8kZRgNW5c2ccO3YMaWlp2L17N/bs2QMAcHBwkMe7YsBFRERE5Y2iAGvv3r3IyMjAiRMnEBYWhvDwcBw9ehRJSUnYsWMHdu7cCYABFxEREZUvqg80mpmZmSvgevz48fPC/hdUOTo6ws/PTx7hvSzjQKPaONAoERGVVsU60KiZmRnatWuHmTNn4u+//0ZiYiKOHDmCGTNmwMHBAUIIPHr0CDt27FC7aCIiIqISoVAGGgWAtLQ0HD58WB4b6/Tp08jMzJSfOiQiIiIqq1QLsPQFVAByjeCeM5wDERERUVmkKMDas2cPDhw4YDCgsrW1Rbt27RAQEAB/f3/4+PjAzKzQGs6IiIiIip2iSOfll1/OdcvPzs4O7du3lwMqb29vmJqaKq4oERERUWmhuCnJ3t5eK6Bq1aoVAyoiIiIq1xQFWCdOnEDLli1hYqL6w4hEREREpZaiAKt169Zq1YOIiIiozGDTExEREZHKVHucLyEhAatXr8bevXtx8eJFJCQkAAAqVKiAxo0bo0uXLggJCUGFChXUKpKIiIioRFIlwFq+fDneeecdpKWlAdAepiE6OhoxMTHYvXs35syZgy+//BLjxo1To1giIiKiEklxgLVgwQLMmjVLDqocHR3RokULVK5cGUIIPHjwAJGRkUhKSsLjx4/x+uuv49GjR3j33XcVV56IiIioJFIUYF28eBEffvghhBCoUqUKvvjiCwwcOBDm5uZa6TIzM7Fp0yZMnz4dMTEx+OCDD9CjRw80atRIUeWJiIiISiJFndyXLl2KrKwsVKxYEREREXjttddyBVfA8wmghw4dioiICFSqVAlZWVlYunSpkqKJiIiISixFAdb+/fshSRJmzJiB6tWr55new8MD7733HoQQ2Ldvn5KiiYiIiEosRQFWdHQ0AKBdu3ZGb9O+fXsAQExMjJKiiYiIiEosRQFWzpQ4OZM8GyMnLUd/JyIiorJKUZSTc1swP7f7ctIac0uRiIiIqDRSFGB17doVQggsWrQIFy5cyDP9+fPn8cUXX0CSJAQGBiopmoiIiKjEUhRgTZkyBZaWlkhNTUWHDh3wxRdfID4+Ple6+Ph4fP755+jYsSNSU1NhaWmJKVOmKCmaiIiIqMRSNA5WjRo1sHz5coSEhCA1NRXvv/8+ZsyYAU9PT7i5uUGSJNy/fx83b96EEAJCCEiShOXLl/MWIREREZVZikdyHzlyJFxcXDBhwgRER0dDCIHr16/jxo0bALSnzXF3d8eKFSvQvXt3pcUSERERlViqzEXYo0cP3LhxA1u3bjU42XPfvn1hZqba/NJEREREJZJq0Y6ZmRkGDhyIgQMHqpUlERERUanEwaiIiIiIVMYAi4iIiEhlDLCIiIiIVGZUH6ycKXHUJElSvqbYISIiIiotjAqwNIdaICIiIiLDjAqwPvroo8KuBxEREVGZwQCLiIiISGVGBVjffvstAGDEiBFwdnYu1AoRERERlXZGBVhTpkyBJEno0qWLVoDVuXNnSJKEn376CTVq1Ci0ShIRERGVJopGcg8LC4MkSXj8+LFa9aEyzq5yZWRnZcGkEJ5MBVCoeRMRERnLqADLysoK6enpePToUSFXh8o6KycnmJiaYsuwYYiLilI174peXuj366+q5klERFQQRgVYnp6e+Oeff7Bjxw60a9eusOtE5UBcVBTuR0YWdzWIiIgKhVEBVvfu3XH58mUsXLgQ+/btQ7169WBubi6v/+CDD+Dk5JSvgiVJwqpVq/K1DREREVFpYFSANXPmTPzxxx+4evUqTp48iVOnTsnrhBDYvn17vgoVQjDAIiIiojLLqACrQoUKOHXqFJYuXYp9+/YhOjoa6enpuHXrFiRJQpUqVbRatIiIiIjKM6OfInRwcMDMmTMxc+ZM+T0Tk+dzRe/evRsNGzZUv3ZEREREpZBJcVeAiIiIqKxRNA7WgQMHAAA1a9ZUpTJEREREZYGiFqzMzEz4+/vD2to639u+8cYbSoomIiIiKrEUBVh9+/bFyZMn873duHHjsHz5ciVFExEREZVYigKs1NRUdO/eHVH5GJF7zJgxWLlypZJiiYiIiEo0RQFWvXr18PDhQwQGBuL27dt5pg8ODsbq1asBAMOGDVNSNBEREVGJpSjA2rNnD6pVq4bo6Gh07doVsbGxOtMJITBixAj8/PPPEEIgKCgIa9euVVI0ERERUYmlKMDy8PDAnj174OrqiqtXr+Lll19GUlKSVprs7GwMGzYMv/5vEt5Ro0bhp59+giRJSoomIiIiKrEUj4NVv359/Pnnn7Czs8P58+fRs2dPPH36FMDz4Gro0KH47bffAPx//ysGV0RERFSWqTLQaKtWrbB9+3ZYWlri6NGj6N+/P548eYIBAwZg06ZNAIDx48djxYoVahRHREREVKKpNpJ7QEAANmzYABMTE/z111+oWbOmPAn066+/jh9++EGtooiIiIhKNFWnyundu7c8BENsbCyEEJg4cSK+++47NYshIiIiKtGMmirHmCEYcnTq1AlvvfUWFi9ejAEDBuCdd97Ru3316tWNzpeIiIiotDAqwCrIXIOSJOH333/H77//rnd9ZmZmvvMl0seucmVkZ2XBxNRU9bwLK18iIiqbjAqwhBCFXQ8ixaycnGBiaootw4YhLh+zC+SlopcX+v1vmBEiIiJjGBVg5Yy+TlQaxEVF4X5kZHFXg4iIyjGjAqygoKDCrgcRERFRmaHqU4REREREpDDAMjExgZmZGT7//HO16kNERERU6ikKsCwsLCCEgJ+fn1r1ISIiIir1FAVY7u7uAABTPr5OREREJFMUYHXs2BEAcObMGVUqQ0RERFQWKAqwJk2aBFNTUyxatAjJyclq1YmIiIioVFMUYLVq1QpLlizBrVu34O/vj6NHj6pVL6PExsZix44dmD17Nrp16wZXV1dIkgRJkhAcHJzv/P766y/069cP1apVg6WlJapVq4Z+/frhr7/+Ur/yREREVGYZNQ6WPqNGjQIA1K9fH+fOnYOfnx88PDzQtGlTODs7G+ybJUkSVq1apaR4uLm5Kdo+hxACEyZMwIoVK7Tej46OxtatW7F161aMGzcOy5YtgyRJqpRJREREZZeiAGvNmjVywCFJEoQQuH37Nu7cuWNwOyGEKgGWJg8PD3h5eWH37t353vaDDz6Qg6sWLVrg3XffRe3atXHt2jV8/vnniIyMxIoVK1CxYkXMnTtXtToTERFR2aQowKpevXqxtujMnj0b3t7e8Pb2hpubG27evJnviamvXr0qj+PVunVrHDx4ENbW1gAAb29v9OrVC/7+/jh16hQWLlyIkJAQ1K5dW/V9ISIiorJDUYB18+ZNlapRMB9//LHiPL7++mtkZmYCAJYsWSIHVzlsbGywZMkS+Pr6IjMzE4sXL8aSJUsUl0tERERlV7meKkcIge3btwMAGjRogLZt2+pM17ZtW9SvXx8AsG3bNgghiqyOREREVPqU6wDrxo0biI6OBgD4+/sbTJuz/u7du8XeckdEREQlW6EEWJmZmYiLi0NcXJx8+60kioqKkpcbNGhgMK3mes3tiIiIiF6kWoAVFRWFSZMmwcvLC1ZWVqhcuTIqV64MKysreHl5YfLkybh06ZJaxalC82nHatWqGUzr4eGhczsiIiKiFynq5J5jxowZWLRoEbKzs3P1TxJC4J9//sGVK1fwww8/YPr06Zg/f74axSqWkpIiL9vZ2RlMa2trKy+npqbqTZeeno709HT5/xzhnoiIqPxRHGBNmjQJ33//vRxYeXl5oU2bNqhcuTKEEHjw4AFOnDiBS5cuISsrCwsXLsTjx4/xzTffKK68Uk+fPpWXLSwsDKa1tLSUl588eaI33WeffabK041ERERUeikKsI4cOYLvvvsOkiShYcOGWLFiBdq1a6czbUREBCZMmIALFy5g6dKlGDx4sN60RcXKykpefvbsmcG0mq1SLw7loGnGjBmYOnWq/P/k5GSt24tERERU9inqg7V8+XIAQM2aNXHkyBGDAZOvry8OHjyIWrVqAQCWLVumpGhV2Nvby8uGbvsBwOPHj+VlQ7cTLS0t4eDgoPUiIiKi8kVRgHXo0CFIkoT3338fjo6OeaZ3dHTEe++9ByEEDh06pKRoVWh2bL97967BtJod29kiRURERIYoCrDu378P4Pn8fcZq2bIlAODBgwdKilZFw4YN5eXLly8bTKu53svLq9DqRERERKWfogArpw+T5u2zvOTcitPsNF5catasCXd3dwBAeHi4wbQHDx4EAFStWhWenp6FXTUiIiIqxRQFWDkTK//xxx9GbxMaGgoAcl+s4iRJEnr37g3geQvVsWPHdKY7duyY3ILVu3fvYp3gmoiIiEo+RQFW9+7dIYTA0qVLsW/fvjzT79u3D0uWLIEkSejevbuSolUzZcoUmJk9f5hy0qRJuYZgePLkCSZNmgQAMDMzw5QpU4q6ikRERFTKKBqmYcqUKVi6dClSUlLQrVs3jB07FqNGjUKLFi1gYvI8dsvOzkZkZCRWrVqFlStXIjMzE46OjqoEKocPH8bVq1fl/8fHx8vLV69exZo1a7TSBwcH58qjXr16eOedd7BgwQKcOnUK7du3x3vvvYfatWvj2rVrWLhwISIjIwEA06dPR926dRXXm4iIiMo2RQGWq6srNm7ciF69euHZs2dYtmwZli1bBgsLC1SoUAGSJOHhw4fyGFNCCFhYWGDTpk1wcXFRXPmVK1di7dq1OtcdOXIER44c0XpPV4AFAPPmzUNsbCx++uknREZGYsiQIbnSjB49GnPnzlVcZyIiIir7FM9FGBgYiGPHjqF169YQQkAIgfT0dNy7dw8xMTFIT0+X3/f29sbx48fRpUsXNequGhMTE6xatQo7d+5E79694e7uDgsLC7i7u6N3797YtWsXVq5cKbfKERERERmiylyEzZs3x4kTJ3Dy5Ens3bsXFy9eREJCAgCgQoUKaNy4Mbp06QJvb281ipOtWbMm121AJbp3715i+oYRERFR6aVKgJXD29tb9SCKiIiIqLThPS8iIiIilSkKsH744QdcunRJrboQERERlQmKbhG++eabkCQJrq6u6NixIwICAuDv74/GjRurVT8iIiKiUkdxHywhBOLi4rBlyxZs2bIFwPOO7ZoBV9OmTRVXlIiIiKi0UBRgJSQkIDw8HGFhYQgLC8OFCxeQnZ2Nhw8fYtu2bdi2bRsAwNnZGX5+fnLA1bx5cxWqTkRERFQyKQqwnJyc0Lt3b3k+v0ePHuHgwYNywHX+/HlkZ2cjISEB27dvl+csdHR0RMeOHeUAjIiIiKgsUXWYBicnJ/Tq1Qu9evUCACQlJWkFXOfOnUN2djYePXokT/pMREREVNYU6jANDg4OqFq1KqpWrQp3d3fY2NhAkqTCLJKIiIio2KnagiWEQGRkpNxidejQISQnJ8vrAMDa2hq+vr7o1KmTmkUTERERlRiKAqwXA6rDhw8jKSlJXgf8f0AVEBCAgIAAtGnTBubm5sprTkRERFRCKQqwXFxcGFARERERvUBRgPXo0SO5T1Xv3r3x9ttvw9fXlwEVERERlWuqDDQKAH/88QfCwsK0Bhht0aIFO7UTERFRuaMowDp06JDc/yoiIgJJSUkIDQ3Fjh07ADx/ijBngNGAgAAGXERERFQuKAqw2rdvj/bt22PWrFnIzMzEiRMn5IDr6NGjSEpKwo4dO7Bz504AuQOuli1bqrITRERERCWJasM0mJmZoV27dmjXrh1mzpyJzMxMnDx5MlfAtXPnTuzcuROSJCEzM1Ot4omIiIhKDFXHwdLK2MwMvr6+aNasGVq2bIkmTZpg1apVSE5OlvttEREREZVFqgdYaWlpOHz4sNxydfr0abmlSjOwsrGxUbtoIiIiohJBcYBlbEBla2srj4/l7+8PHx8fpUUTERERlUiKO7mfOnVKZ0BlZ2eHdu3ayQGVt7c3zMwK7Y4kERERUYmhKOKJiIiQl+3t7dG+fXs5oGrdujVMTU0VV5CIiIiotFEUYHXv3l0OqFq1agUTExO16kVERERUaikKsHIGFCUiIiKi/8cmJyIiIiKVMcAiIiIiUhkDLCIiIiKVMcAiIiIiUhkDLCIiIiKVMcAiIiIiUhkDLCIiIiKVMcAiIiIiUplRAVa/fv3Qv39/3L17t7DrQ0RERFTqGRVgbdu2Ddu2bUNycrL2xiYmMDMzw6VLlwqlckRERESlUb5uEQohjHqPiIiIqDwzKsCyt7cHADx48KBQK0NERERUFhgVYDVo0AAA8M033yA1NTXXekmS1K0VERERUSlmZkyi1157DSdPnsSOHTtQoUIFuLm5wdzcXF4fGBio9X9jSJKEa9eu5a+2RERERKWAUQHWpEmTcOTIEWzevBmZmZmIjo6W1wkhtP5vLLZ6ERERUVllVIBlYmKCjRs3IiIiAnv37kV0dDTS09Oxdu1aSJKEXr16wcnJqZCrSkRERFQ6GBVg5fD19YWvr6/8/7Vr1wIA5s2bh4YNG6pbMyIiIqJSiiO5ExEREaksXy1YL8rOzlarHkRERERlBluwiIiIiFSmqAVLU1ZWFrZt24a9e/fi4sWLSEhIAABUqFABjRs3RpcuXdCnTx+YmpqqVSQRERFRiaRKgPXXX39h3LhxuYZvAJ4Px3D06FGsWLEC1apVw4oVK/Dyyy+rUSwRERFRiaT4FuG6devQs2dPREdHQwgBIQRq1KiBtm3bok2bNqhRowaA5wHXnTt30KNHD/z666+KK05ERERUUikKsG7duoVx48YhOzsbNjY2mDt3Lu7fv4/r16/j6NGjiIiIwPXr13H//n3MmzcPdnZ2yM7OxtixY3H79m219oGIiIioRFEUYH3zzTdIT0+HnZ0dDh06hJkzZ6JSpUq50lWsWBEzZszAoUOHYGdnh/T0dHzzzTdKiiYqMnaVKyM7K6vQ8i/MvImIqHgo6oO1e/duSJKE6dOno3nz5nmmb9asGd555x3MmTMHf//9N7788kslxRMVCSsnJ5iYmmLLsGGIi4pSNe+KXl7ox1vmRERljqIAK+c2X5cuXYzepmvXrpgzZw5vEVKpExcVhfuRkcVdDSIiKgUU3SLM+t+tjfwMvZCTloOUEhERUVmlKMCqWrUqAODo0aNGb5OT1t3dXUnRRERERCWWogCrU6dOEEJgwYIFiImJyTP93bt3sWDBAkiShM6dOyspmoiIiKjEUhRgTZo0CSYmJoiLi0ObNm2wadMm+bahpqysLGzcuBG+vr6IjY2FiYkJJk6cqKRoIiIiohJLUSf3xo0b49NPP8WsWbMQExODIUOGwMnJCS1atICbmxskScL9+/cRGRmJR48eyaO7f/rpp2jcuLEqO0BERERU0iieKmfGjBlwdHTEu+++i7S0NCQmJuLAgQNaaXICKxsbG3zxxRd4/fXXlRZLREREVGKpMhfhG2+8gUGDBmH16tUGJ3sOCQmBq6urGkUSERERlViqBFgA4OrqiunTp2P69OlqZUlERERUKime7JmIiIiItDHAIiIiIlIZAywiIiIilTHAIiIiIlIZAywiIiIilTHAIiIiIlIZAywiIiIilTHAIiIiIlIZAywiIiIilSkayf2TTz4BALRp0wYvv/yyKhUi42VnZcHE1LS4q0FEREQvUBRgzZkzB5IkYevWrWrVh/LBxNQUW4YNQ1xUlKr51u3WDZ3nzVM1TyIiovJEUYDl4uKChIQEVK9eXa36UD7FRUXhfmSkqnm6Nmigan5ERETljaI+WHXq1AEA3L9/X5XKEBEREZUFigKswYMHQwiBjRs3qlUfIiIiolJPUYD1xhtvoFmzZvj555+xdu1atepEREREVKop6oN1//59rFy5EqNHj8aoUaPw66+/4rXXXkPTpk3h7OwM0zyecGPfLSIiIiqLFAVYnp6ekCQJACCEwL59+7Bv3z6jtpUkCZmZmUqKJyIiIiqRFAVYwPPAStcyERERUXmlKMBavXq1WvUgIiIiKjMUBVhBQUFq1aPY5NzizIu/vz/CwsIKtzJERERUJnAuQiIiIiKVKe6DVVa8/vrreOONN/Sut7W1LcLaEBERUWmmWoCVnZ2NsLAwRERE4P79+0hLS8PcuXNRpUoVOc2zZ8+QmZkJU1NTWFpaqlW0KipVqoTGjRsXdzWIiIioDFAlwNq5cycmT56Mmzdvar0/bdo0rQBr1apVmDhxIuzs7BATE8NWISIiIiqTFPfBWrlyJXr16oUbN25ACAEXFxe9wzWMHj0aTk5OSE1NxdatW5UWTURERFQiKQqwrl69ijfffBMA0LlzZ1y6dAmxsbF601tYWKB///4QQmD37t1KiiYiIiIqsRQFWIsXL0ZGRgYaNWqEXbt2oUGDBnlu4+fnBwA4e/askqJVt2nTJtSvXx/W1tawt7dH3bp1ERQUhAMHDhR31YiIiKiUURRg7du3D5IkYcqUKbCwsDBqm9q1awMAbt++raRo1V26dAlXrlzB06dPkZqaiqtXr+Lnn39G586d0bdvXyQlJRV3FYmIiKiUUNTJ/c6dOwCA5s2bG71NTsf2tLQ0JUWrxsbGBr169cJLL72EBg0awM7ODnFxcQgPD8eyZcvw8OFDbNu2Db1798aePXtgbm5uML/09HSkp6fL/09OTi7sXSAiIqISRlGApTnRs7Hi4uIAAA4ODkqKVk10dDScnJxyvd+1a1dMmjQJ3bp1Q2RkJMLDw/HDDz9g8uTJBvP77LPP8PHHHxdSbYmIiKg0UHSL0N3dHQBw5coVo7cJDw8HAHh6eiopWjW6gqscbm5u2Lx5s3z7c8mSJXnmN2PGDCQlJcmvnFY+IiIiKj8UBVgdO3aEEALr1683Kn18fDyWL18OSZLQuXNnJUUXmVq1aqFr164Anj81GRMTYzC9paUlHBwctF5ERERUvigKsMaNGwcA2LVrF1avXm0w7d27d9G9e3fEx8fD1NRU3rY0aNiwobwcHR1djDUhIiKi0kBRgOXt7Y0JEyZACIExY8Zg4MCB2Lhxo7z+/Pnz+M9//oPRo0ejfv36OH36NCRJwrRp01CnTh3FlS8q+eljRkRERKR4qpwlS5bg8ePHWLduHbZs2YItW7bInd+HDRsmp8sJUoKDgzF//nylxRapS5cuycs5/c6IiIiI9FE8VY6pqSnWrl2LTZs2oUWLFhBC6Hw1bNgQ69evx08//SQHYKXB9evXsWfPHgDP+2NVrVq1mGtEREREJZ0qkz0DQP/+/dG/f3/ExMTg1KlTiI2NRVZWFlxcXNCiRQt5gNGSJDQ0FN26dYOZme7D8ODBAwwYMAAZGRkAIE8LRERERGSIagFWDnd3d/Tq1UvtbAvFpEmTkJGRgf79+8PX1xeenp6wtrZGfHw8wsLC5IFGAaBDhw4MsIiIiMgoqgdYpU1MTAyWLFlicIyr/v37Y+XKlbC0tCzCmhEREVFppWqAlZGRgTNnzuDixYtISEgAAFSoUAGNGzdGy5Yt85xmpqitXbsW4eHhiIiIwPXr1xEfH4/k5GTY2dnBw8MD7dq1Q1BQEHx9fYu7qkRERFSKqBJgpaam4tNPP8WqVauQmJioM42zszNGjx6NDz74APb29moUq5i/vz/8/f2LuxpERERUxih+ijAqKgqNGjXCokWLkJCQoPcpwoSEBCxatAhNmjTBP//8o0bdiYiIiEokRS1Yjx49QpcuXXDv3j0AQOPGjREUFAQfHx+4ublBCIHY2FicPHkSa9euxYULF3D79m106dIFFy9ehKOjoyo7QURERFSSKGrBWrhwIe7duwdJkvDpp5/i3LlzmDZtGvz8/FCvXj3Ur18ffn5+mDp1Ks6ePYu5c+cCeN6xfOHCharsABEREVFJoyjA2rZtGyRJwuDBgzFr1iyDA4hKkoSZM2di8ODBEEJg69atSoomIiIiKrEUBVi3bt0CAAQFBRm9TXBwsNa2RERERGWNogAr52nASpUqGb1NTlo7OzslRRMRERGVWIoCrCZNmgAA/v33X6O3yUmbsy0RERFRWaMowBo/fjyEEFi8eDGys7PzTJ+dnY2vv/4akiRh3LhxSoomIiIiKrEUBVgDBw5ESEgIjh07hj59+uD+/ft60z548AD9+vXD8ePHERwcjMGDByspmoiIiKjEMmocrJ9//lnvOn9/f1y8eBE7duxArVq1EBgYCG9vb1SqVAmSJOHBgwc4efIkdu/ejfT0dHh7e8Pf3x8///wzRo4cqdqOEBEREZUURgVYwcHBBodgAJ4Pw/D06VOEhoYiNDQ013ohBCRJwqlTpxASEgJJkhhgERERUZlk9EjuQgjF6YzNg4iIiKg0MyrAunHjRmHXg4iIiKjMMCrAqlGjRmHXg4iIiKjMUPQUIRERERHlxgCLiIiISGUMsIiIiIhUpkqAFRUVhbfffhutW7dGhQoVYG5uDlNTU4MvMzOjH2AkIiIiKlUURzkLFizA7NmzkZWVxWEYiIiIiKAwwNq0aRNmzpwJADAxMYGfnx+aNWsGJycnmJjw7iMRERGVT4oCrG+++QYAULVqVezatQtNmjRRpVJEREREpZmiZqbz589DkiR8+umnDK6IiIiI/kdRgGVubg4AaN68uRp1ISIiIioTFAVY9erVAwA8fPhQlcoQERERlQWKAqygoCAIIbBt2zaVqkNERERU+ikKsEaPHg0/Pz+sWLECoaGhatWJiIiIqFRT9BShubk5tm/fjqCgIPTt2xdDhgzBoEGDUK9ePdjY2OS5ffXq1ZUUT0RERFQiKR5o1MnJCZMnT8axY8ewYcMGbNiwwajtJElCZmam0uKJiIiIShzFo4FOmTIFgYGBiI+PhxAiXy+i8s6ucmVkZ2UVWv6FmTeVDTz/tPF4kFoUtWD98ssv+PbbbwEADg4O6Nu3L5o2bcqR3ImMZOXkBBNTU2wZNgxxUVGq5l3Rywv9fv1V1Typ7OH5p43Hg9SiKMBasmQJAKBBgwY4cOAA3NzcVKkUUXkTFxWF+5GRxV0NKqd4/mnj8SA1KGpmunz5MiRJwpw5cxhcEREREf2PKiO55ww4SkREREQKA6wGDRoAAO7fv69KZYiIiIjKAkUBVkhICIQQRg/NQERERFQeKB7JvWfPnvjll1+wdOlStepEREREVKopeorw4MGDmDx5MuLi4vDWW29h/fr1GDJkiNEjuXfs2FFJ8UREREQlkqIAKyAgAJIkyf8/fvw4jh8/btS2HMmdiIiIyirFU+VwRHYiIiIibYoCrAMHDqhVDyIiIqIyQ1GA5e/vr1Y9iIiIiMoMThhIREREpDIGWEREREQqY4BFREREpDJFfbA6d+5c4G0lScK+ffuUFE9ERERUIikKsMLCwiBJksGhGjTHyQL+f1iHF98nIiIiKisUBVgdO3bMM1B6/Pgx/v33XyQlJUGSJNSrVw9VqlRRUiwRERFRiaa4BcsYQgjs3LkTb731FhISErBy5Up06NBBSdFEREREJVaRdHKXJAk9e/bE4cOHYWpqir59+yI6OrooiiYqt+wqV0Z2Vlah5V9a86bSj+cHlQaKp8rJjypVqmDq1Kl477338Pnnn+Obb74pyuKJyhUrJyeYmJpiy7BhiIuKUjXvut26ofO8eYWSd0UvL/T79VdV86SypbDPayI1FGmABUC+Nbhz504GWERFIC4qCvcjI1XN07VBg0LLm8gYhXleE6mhyMfBsrCwAADExMQUddFERERERaLIA6zDhw8DAGxsbIq6aCIiIqIiUaQBVkREBD755BNIkgQfH5+iLJqIiIioyCjqg/XJJ5/kmSY7OxuJiYk4deoUjh8/juzsbEiShLfffltJ0UREREQllqIAa86cOfkakV0IATMzM3z++efo2rWrkqKJiIiISizFTxEamiYHeD4Glr29PWrWrAl/f3+MGzcODRs2VFosERERUYmlKMDKzs5Wqx5EREREZUaRP0VIREREVNYxwCIiIiJSGQMsIiIiIpWpOlVOSkoKbty4gZSUFGQZMRlnx44d1SyeiIiIqERQJcD68ccf8f333+P8+fNGbyNJEjIzM9UonoiIiKhEURRgZWVloX///ggNDQWQ95ANREREROWBogBr2bJl+OOPPwAAbm5uCAkJQatWrVChQgWYmLB7FxEREZVPigKsn3/+GQDQsGFDHDp0CM7OzqpUioiIiKg0U9TMFBUVBUmS8OGHHzK4IiIiIvofVe7j1a9fX41siIiIiMoERQFW3bp1AQAJCQmqVIaIyK5yZWQbMcxLQRRWvqU5byoahXleA6Xz/CuNdc4PRX2whgwZgsjISOzYsQOdO3dWq05EVI5ZOTnBxNQUW4YNQ1xUlGr51u3WDZ3nzVM937KQNxW+wjqvAaCilxf6/fqrqnlqKm3XY2EfD2MpCrAmT56MX3/9FT/88AP69u0LPz8/tepFROVcXFQU7kdGqpafa4MGhZJvWcibik5hfI5FoTRdjyWFoluElpaW2L17N1q1aoWuXbvi3XffxdmzZ/H06VO16kdERERU6ihqwTI1NZWXhRD48ssv8eWXXxq1LUdyJyIiorJKUYD14sjtHMmdiIiISGGA9dFHH6lVDyIiIqIygwEWERERkco4YSARERGRyhhgEREREamMAZaG27dv45133oGXlxdsbW1RoUIF+Pj4YNGiRUhLSyvu6hEREVEpoagPVlmyc+dODBs2DElJSfJ7aWlpOHnyJE6ePImVK1di165dqFWrVjHWkoiIiEoDtmABOHfuHAYNGoSkpCTY2dlh3rx5OHr0KPbt24exY8cCAP755x/06NEDqampxVxbIiIiKunYggVgypQpSEtLg5mZGXbv3g1fX195XefOnVG3bl28++67uHz5Mr766ivMnj27GGtLREREJV25b8E6efIkwsLCAACjR4/WCq5yTJs2DV5eXgCAxYsXIyMjoyirSERERKVMuQ+wtm3bJi+HhIToTGNiYoKRI0cCABITE+WAjIiIiEiXch9gHTp0CABga2uLVq1a6U3n7+8vLx8+fLjQ60VERESlV7kPsKKiogAAderUgZmZ/i5pDRo0yLUNERERkS7lOsB6+vQp4uPjAQDVqlUzmNbZ2Rm2trYAgDt37hR63YiIiKj0KtdPEaakpMjLdnZ2eaa3tbXF48ePDQ7VkJ6ejvT0dPn/OeNqJScnK6ipgTrVqQPHrCxV87Rwd0dycjLzLuR8mXfR5l0a61ya87atU6fQvvdy8i9Nx6M0H+ucMkrL9ViYxyMnXyFE3olFOXb79m0BQAAQI0aMyDO9h4eHACBq166tN81HH30k58kXX3zxxRdffJW91507d/KMGcp1C5aVlZW8/OzZszzT57RMWVtb600zY8YMTJ06Vf5/dnY2EhIS4OLiAkmSFNS2ZEtOToaHhwfu3LkDBweH4q4OFQJ+xmUfP+Oyj5+xMkIIpKSkwN3dPc+05TrAsre3l5eNGaH98ePHAAzfTrS0tISlpaXWe05OTgWrYCnk4ODAi7aM42dc9vEzLvv4GReco6OjUenKdSd3KysruLq6AgDu3r1rMG1iYqIcYHl4eBR63YiIiKj0KtcBFgB5hParV68iMzNTb7rLly/n2oaIiIhIl3IfYHXo0AHA89t/p0+f1psuPDxcXm7fvn2h16u0sbS0xEcffZTr9iiVHfyMyz5+xmUfP+OiIwlhzLOGZdeJEyfQpk0bAMD48eOxbNmyXGmys7PRuHFjREVFwcnJCbGxsTA3Ny/qqhIREVEpUe5bsHx8fODn5wcAWLVqFSIiInKl+fLLL+XR29966y0GV0RERGRQuW/BAoDIyEi0b98eT548gZ2dHWbOnIlOnTrhyZMn+O2337BixQoAQL169XDq1Cmtpw+JiIiIXsQA639CQ0MxfPhwvaO/1qtXDzt37kSdOnWKuGZERERU2pT7W4Q5Xn31VZw/fx5vv/026tWrBxsbGzg5OaF169ZYuHAhIiMjy3Rwdfv2bbzzzjvw8vKCra0tKlSoAB8fHyxatAhpaWmK8p4zZw4kSTLqFRYWps4OkSw2NhY7duzA7Nmz0a1bN7i6usrHOzg4uFDK/O233/Dyyy+jSpUqsLKygqenJ0aMGIFjx44VSnnlXVF9xryWi8+ZM2cwf/58dOvWDR4eHrC0tISdnR3q1auH4OBgHDp0SPUyeR0rZNykMlSW7dixQzg6OuqdEqB+/fri2rVrBc4/P9MHHThwQL0dIyGEMHi8g4KCVC3ryZMnomfPnnrLMzExEZ988omqZVLRfca8lotHx44djTrmI0aMEOnp6YrL43WsjnI9kjsB586dw6BBg5CWlgY7OzvMmDFDq//Zjz/+iH/++Qc9evTAyZMnjZoU25ALFy4YXF+zZk1F+ZNhHh4e8PLywu7duwsl/9GjR2PHjh0AgE6dOuGtt96Cu7s7Lly4gPnz5+PatWuYPXs2qlSpgjFjxhRKHcq7wv6Mc/BaLjrR0dEAAHd3dwwcOBB+fn6oXr06srKyEBERgS+//BLR0dFYt24dMjMzsX79ekXl8TpWSXFHeFS8AgICBABhZmYmjh49mmv9559/Lv9q+fjjjwtUhuavXip6s2fPFqGhoeL+/ftCCCFu3LhRKK0bYWFhcr6vvvqqyMzM1FofFxcnqlevLgAIZ2dnkZiYqFrZ5V1Rfca8lotHjx49xH/+859c11SOuLg4Ua9ePfmzOXjwYIHL4nWsHl4l5diJEyfkC2n8+PE602RlZQkvLy/5Ynr27Fm+y+GXcslSWH98u3fvLgAIU1NTvTPNb9iwQS570aJFqpVN2hhglT+hoaHyZzN58uQC58PrWD3s5F6Obdu2TV4OCQnRmcbExAQjR44E8Hw+RnZcJV1SU1Oxb98+AEDXrl1RrVo1nen69esnTzC7ZcuWIqsfUVkXEBAgL1+7dq1AefA6VhcDrHIs56kTW1tbtGrVSm86f39/efnw4cOFXi8qfU6cOIH09HQA2ufLiywsLNC2bVt5m4yMjCKpH1FZ9+zZM3nZxKRgf9p5HauLAVY5ljM6fZ06dWBmpv95hwYNGuTapqC6du0KFxcXWFhYoFKlSggICMCCBQuQmJioKF8qXprnheb5okvO+szMTPz777+FWi8qPLyWSxbN+XLzugb14XWsLgZY5dTTp08RHx8PAHqbgXM4OzvD1tYWAHDnzh1F5e7duxcJCQnIyMhAXFwcwsPDMWPGDNSqVQvbt29XlDcVH83zIq/zycPDQ+d2VLrwWi45srOzsWDBAvn/gwYNKlA+vI7VxWEayqmUlBR52ZihF2xtbfH48WOkpqYWqLwmTZqgT58+8PHxgbu7OzIyMvDPP//g119/xe7du/Ho0SP0798foaGh6NatW4HKoOKTn/MpJ1gHUODziYoPr+WS5+uvv8aJEycAAH379kXr1q0LlA+vY3UxwCqnnj59Ki9bWFjkmd7S0hIA8OTJk3yXNWXKFMyZMyfX+23atMHIkSOxfPlyTJgwAVlZWRgzZgyuXr0Ka2vrfJdDxSc/51POuQQU7Hyi4sNrueQJDw/H+++/DwCoVKkSfvjhhwLnxetYXbxFWE5ZWVnJy5qdI/XJ6fhYkC9LJycng+vHjx8vD1YXExPDp1JKofycTznnElCw84mKD6/lkuW///0v+vbti8zMTFhaWmLjxo1wc3MrcH68jtXFAKucsre3l5eNad59/PgxAONuJxbE+PHj5WXNzppUOuTnfMo5l4DCO5+o+PBaLho3btxAYGAgEhMTYWpqig0bNhh88s8YvI7VxQCrnLKysoKrqysA4O7duwbTJiYmyheTZsdGNTVs2FBezpkWgkoPzQ6xeZ1Pmh1iC+t8ouLDa7nwxcTEoEuXLoiJiYEkSfjpp5/Qt29fxfnyOlYXA6xyzMvLCwBw9epVZGZm6k13+fLlXNuoTQhRKPlS0dD8o6p5vuiSs97MzAx16tQp1HpR0eO1XLji4+PRtWtXXL9+HQCwZMkSeTBopXgdq4sBVjnWoUMHAM+bek+fPq03nWYzf/v27QulLpcuXZKX3d3dC6UMKjze3t5yp1hDt4WePXuGY8eO5dqGyg5ey4UnKSkJL7/8snyMFyxYgDfffFO1/Hkdq4sBVjnWp08feXn16tU602RnZ+Pnn38G8LyDa6dOnQqlLsuXL5eXlfYjoKJnb2+Pl156CcDz8ZH03V7YsmULkpOTAUCVWxpU8vBaLhxpaWno0aMHzpw5AwCYNWsW3nvvPVXL4HWssuKeDJGKl5+fnwAgzMzMxNGjR3Ot//zzz+VJPT/66KNc61evXm1w/fnz58W///5rsA7Lli2T86hcubJITU0t6O6QEQoyEXBen7MQQuzbt09O06tXL5GZmam1Pi4uTlSvXl0AEE5OTiIhIUHhnpA+hfEZ81ouPunp6SIwMFA+tm+99VaB8uF1XLQ4DlY5980336B9+/Z48uQJAgMDMXPmTHTq1AlPnjzBb7/9hhUrVgAA6tWrh2nTpuU7/9OnT2PMmDHo1KkTunXrhiZNmsDFxQWZmZm4fPkyfvnlF+zZswcAYGpqiuXLl2sNYEfKHT58GFevXpX/nzOCP/C8/92aNWu00gcHBxeonM6dO2PIkCH47bff8Mcff6Br166YMmUK3N3dceHCBcybNw+3b98G8PzWhrOzc4HKodyK4jPmtVx8hg4dit27dwN4fp2NHj0aFy9e1JvewsIC9erVK1BZvI5VVNwRHhW/P/74Qzg4OMi/Wl581atXT+8v17x+EWmuN/RycXER27ZtK+Q9LZ+CgoKM+gxyXroY88tXCCHS0tJE9+7d9eZtYmJicHsqmKL4jHktF5/8fLYARI0aNXTmw+u4aLEFi/Dqq6/i/Pnz+Oabb7Bz507cvXsXFhYWqFOnDgYOHIiJEyfCxsamQHl3794dq1atQkREBCIjI/HgwQM8fPgQQghUqFABzZo1wyuvvILg4GA4ODiovGdU1KytrbFz506sX78ea9aswblz5/Do0SO4ubnBz88PEydOhK+vb3FXkwqA13L5wetYHZIQfKaWiIiISE18ipCIiIhIZQywiIiIiFTGAIuIiIhIZQywiIiIiFTGAIuIiIhIZQywiIiIiFTGAIuIiIhIZQywiIiIiFTGAIuIiIhIZQywiIiIiFTGAIuIiIhIZQywiKhUWrNmDSRJgiRJuHnzZnFXh4hICwMsIiIiIpUxwCIiIiJSGQMsIiIiIpUxwCIiIiJSGQMsIiIiIpUxwCKiEikxMRHvv/8+GjRoAGtra1SqVAldunTBpk2bjM4jMzMTq1atQvfu3eHu7g5LS0u4urqiY8eOWLx4MZ4+fZpnHufPn8eIESNQtWpVWFlZoXr16hg+fDjOnDkDAAgODoYkSfD09My17c2bN+UnHdesWQMA2LJli1wfMzMzBAQE5Nru7t27mDFjBlq2bAlnZ2e53MGDB+PAgQNG7XtiYiLmzp0LX19fuLq6wtLSEu7u7ujduze2bNliVB5EpIAgIiph/vvf/4oqVaoIADpfo0aNEqtXr5b/f+PGjVx5XL16VTRs2FBvHgBE3bp1xZUrV/TWY82aNcLc3Fzntubm5mLNmjUiKChIABA1atTItf2NGzfk9D/99JMYMWJErnz8/f21tlm5cqWwtrY2WO/Ro0eLjIwMvfXeuXOncHJyMphHjx49REpKirEfCRHlEwMsIipRHj16JKpVqyYHAoMHDxa7du0Sp06dEuvXrxetW7cWAIS3t7feACsmJka4ubkJAMLe3l5MmzZN/Pnnn+LMmTPiwIEDYsaMGcLGxkYAELVq1RKPHj3KVY9Dhw4JExMTAUBYW1uLmTNnioMHD4rjx4+L7777TlSrVk1YWFiIFi1aGBVgNW3aVAAQfn5+Yv369eLUqVNi7969YuXKlXL6VatWyekbN24slixZIg4fPizOnDkjfv/9d9G9e3d5/dSpU3Uev927dwtTU1MBQHh6eoqFCxeKsLAwcebMGREaGiqGDx8u59GvXz9FnxUR6ccAi4hKlKlTp8oBwPz583Otf/bsmQgMDNRqjXkxwOrZs6cAIDw8PMS1a9d0lnPmzBlha2srAIgPPvgg1/pmzZoJAMLCwkIcOXIk1/oHDx6IWrVqyXXIK8ACIEaOHCmys7N11uf27dty0BcUFKS3hWrmzJkCgDAxMRH//POP1rrU1FQ5sAwMDBSPHz/WmceKFSvkOu3du1dnGiJShgEWEZUYT58+Fc7OznKLT1ZWls50d+7c0bp1pxlgXbhwQX5/+/btBst79913BQDh7u6u9X5ERIScx9tvv613++3btxsdYDk5OYnk5GS9eU2bNk2uy5MnT/Smy8jIEFWrVhUAxKxZs7TWLVmyRAAQVlZW4sGDB3rzEEIIHx8fAUAMGzbMYDoiKhh2cieiEuP06dNITEwEAAQFBcHERPdXVLVq1RAYGKhz3fbt2wEANjY26NGjh8HyOnbsCACIiYnBnTt35Pf37dsnLwcFBendvkePHnBxcTFYRo5XX30V9vb2etfn1PvVV1+FlZWV3nRmZmbw9fUFAEREROjMw9/fH5UqVTJYn5x9fzEPIlKHWXFXgIgox4ULF+Rlb29vg2l9fHywc+fOXO+fOnUKAJCWlgYzM+O/4u7fvw8PDw8AwMWLFwEAlpaWaNy4sd5tTE1N0bx5c62ATJ+mTZvqXZeUlISrV68CAJYvX47ly5cbXWdNOfv+999/Q5KkAuVBROpgCxYRlRg5rVcA8myBcXNz0/l+bGxsgcpOS0vLVY8KFSrA1NTU4HYVK1Y0Kn9nZ2e969Soc0ZGBh49eqQoDyJSD1uwiKjEEELIy3m1wGim1ZSVlQUAqFmzJv744w+jy65Zs6bRaQvCUKCWU2cAmDJlCkaPHm1UnhYWFjrzGDRoED788MMC1JKI1MIAi4hKjAoVKsjLDx48QL169fSm1dfqk9Mn6sGDB2jQoEG+bhPmyGltSkhIQFZWlsHgKC4uLt/5v0izH1daWprB25L6WFlZwcbGBmlpaXj06FGB8iAi9fAWIRGVGE2aNJGXT548aTCtvvUtWrQA8DxQOXLkSIHq0ahRIwBAenq6Vr+wF2VlZeHs2bMFKkNTxYoVUbVqVQDA3r179bbO5SVn348cOcJbf0TFjAEWEZUYrVq1kluP1q1bpzfQiI6Oxu7du3Wu6927t7z8+eefF6geL730krz8888/6023c+dOPHz4sEBlvKhXr14AgOvXr2Pz5s2K8nj8+DG+++47VepFRAXDAIuISgxLS0uEhIQAAM6ePYsvvvgiV5rMzEyMHTsWz54905mHt7e3PITDrl278NFHHxks8+bNm9iwYYPWe76+vvJTf9999x2OHj2aa7u4uDi8/fbbee+UkaZPnw5LS0sAwIQJE+QnAvXZtWsXzp8/r/XehAkT4OrqCgD48MMP8eeffxrM48iRIzh48KCCWhORPpIoaFs0EVEhSEpKQuPGjXH37l0AwNChQzFy5EhUqlQJV65cwVdffYWTJ0/C29tbvk1448YNrcmWY2Ji0Lp1a9y7dw8A0KZNG4waNQpNmjSBlZUVHj58iPPnz+Ovv/7C/v370adPn1ytRocPH4a/vz+ys7NhbW2NqVOn4pVXXoGlpSVOnTqFzz77DPfv30ejRo1w9uxZeHp64saNG1p53Lx5U+48v3r1agQHBxvc9zVr1sgBpoWFBUaMGIGePXuievXqyMzMxN27d3HixAls3rwZ165dQ2hoKHr27KmVx969e9GtWzdkZmbCxMQE/fv3R//+/VG7dm0AwL1793D69Gls3boV58+fx5IlSzBx4sR8fEJEZJTiHOWUiEiXixcvisqVK+udqDgkJCTPyZ5v3rypNV+hoVdISIjOehia7NnMzEz8+OOP8gTODRo0yLW95kjuq1evNmrff/vtN+Hg4JBnnU1MTMT+/ft15rFv3z6Dx0/ztXbtWqPqRUT5w1uERFTiNGrUCP/973/x7rvvom7durC0tISrqys6deqE9evX46effsozjxo1auD48ePYunUrhgwZgpo1a8LGxgbm5uaoWLEi2rVrh2nTpiE8PByrVq3SmUdQUBBOnTqFYcOGwd3dHRYWFqhatSoGDRqEw4cPY8yYMUhOTgYAODo6qrLvgwcPxs2bN7FgwQIEBASgUqVKMDc3h42NDWrVqoVXX30VX331FW7evIlOnTrpzKNz5864du0ali5dildeeQVVqlSBhYUFrKys4OHhgcDAQMybNw+XL1/GyJEjVak3EWnjLUIiIgXq1KmDa9euYfjw4Vi3bl1xV4eISgi2YBERFdDJkydx7do1AEDbtm2LuTZEVJIwwCIi0iNnfkBdHj58iLFjxwJ4/vTj4MGDi6paRFQKcCR3IiI9unbtipo1a6Jv375o2rQpHB0dkZiYiCNHjuD777+Xn1L84IMP5OERiIgA9sEiItLL09MTt27dMpjmjTfewJIlS2BiwhsCRPT/GGAREekRHh6O0NBQhIeH4969e4iPj4eZmRkqV66MDh06YNy4cWjXrl1xV5OISiAGWEREREQqY5s2ERERkcoYYBERERGpjAEWERERkcoYYBERERGpjAEWERERkcoYYBERERGpjAEWERERkcoYYBERERGpjAEWERERkcr+D1yHYfmDXrtTAAAAAElFTkSuQmCC",
      "text/plain": [
       "<Figure size 640x480 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "import csv\n",
    "import matplotlib.pyplot as plt\n",
    "\n",
    "counts = []\n",
    "with open('./results/in_out_degree_ratio_processes.csv', newline='') as csvfile:\n",
    "    reader = csv.reader(csvfile)\n",
    "    next(reader)\n",
    "    for row in reader:\n",
    "        try:\n",
    "            count = float(row[1])\n",
    "            counts.append(count)\n",
    "        except ValueError:\n",
    "            print(f\"Skipping row with non-numeric value in the specified column: {row}\")\n",
    "\n",
    "plt.rc('font', size=20)\n",
    "plt.hist(counts, bins=20,color='maroon', edgecolor='white')\n",
    "plt.title('Avg logical process degree')\n",
    "plt.xlabel('degree')\n",
    "plt.ylabel('number of workflows')\n",
    "plt.tight_layout()\n",
    "plt.savefig(f'./plots/avg_process_degree.png', format=\"png\", bbox_inches='tight')\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import csv\n",
    "import networkx as nx\n",
    "\n",
    "def get_processes_degree(dot_file):\n",
    "    G = nx.nx_agraph.read_dot(dot_file)\n",
    "    \n",
    "    processes = [node for node in G.nodes() if 'shape' not in G.nodes[node]]\n",
    "    \n",
    "    process_data = []\n",
    "    for node in processes:\n",
    "        in_degree = G.in_degree(node)\n",
    "        out_degree = G.out_degree(node)\n",
    "        \n",
    "        # e.g.: if its 2 in and 0 out -> we take the 2\n",
    "        ratio = in_degree / out_degree if out_degree != 0 else in_degree\n",
    "        \n",
    "        process_data.append((node, in_degree, out_degree, ratio))\n",
    "    \n",
    "    return process_data\n",
    "\n",
    "directory_path = './dags'\n",
    "output_file = './results/processes_degrees.csv'\n",
    "\n",
    "with open(output_file, 'w', newline='') as csvfile:\n",
    "    writer = csv.writer(csvfile)\n",
    "    writer.writerow(['pipeline', 'node', 'in_degree', 'out_degree', 'in_out_degree_ratio'])\n",
    "    \n",
    "    for filename in os.listdir(directory_path):\n",
    "        if filename.endswith(\".dot\"):\n",
    "            dot_file_path = os.path.join(directory_path, filename)\n",
    "            process_data = get_processes_degree(dot_file_path)\n",
    "            \n",
    "            file_name = filename.split('_')[0]\n",
    "            \n",
    "            for node, in_degree, out_degree, ratio in process_data:\n",
    "                writer.writerow([file_name, node, in_degree, out_degree, ratio])\n",
    "\n",
    "print(f'Results written to: {output_file}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>in_degree</th>\n",
       "      <th>out_degree</th>\n",
       "      <th>in_out_degree_ratio</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>count</th>\n",
       "      <td>2211.000000</td>\n",
       "      <td>2211.000000</td>\n",
       "      <td>2211.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mean</th>\n",
       "      <td>2.058797</td>\n",
       "      <td>3.440072</td>\n",
       "      <td>0.827782</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>std</th>\n",
       "      <td>2.447305</td>\n",
       "      <td>4.013259</td>\n",
       "      <td>1.224450</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>min</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25%</th>\n",
       "      <td>1.000000</td>\n",
       "      <td>2.000000</td>\n",
       "      <td>0.333333</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50%</th>\n",
       "      <td>1.000000</td>\n",
       "      <td>3.000000</td>\n",
       "      <td>0.500000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75%</th>\n",
       "      <td>2.000000</td>\n",
       "      <td>4.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>max</th>\n",
       "      <td>49.000000</td>\n",
       "      <td>63.000000</td>\n",
       "      <td>29.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         in_degree   out_degree  in_out_degree_ratio\n",
       "count  2211.000000  2211.000000          2211.000000\n",
       "mean      2.058797     3.440072             0.827782\n",
       "std       2.447305     4.013259             1.224450\n",
       "min       0.000000     0.000000             0.000000\n",
       "25%       1.000000     2.000000             0.333333\n",
       "50%       1.000000     3.000000             0.500000\n",
       "75%       2.000000     4.000000             1.000000\n",
       "max      49.000000    63.000000            29.000000"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_degree = pd.read_csv('./results/processes_degrees.csv')\n",
    "#pd.set_option('display.max_rows', None) # max pipelines: amliseq 37, viralrecon, rnaseq 37\n",
    "df_degree.describe()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv(output_file)\n",
    "\n",
    "max_ratios = {}\n",
    "for _, row in df.iterrows():\n",
    "    pipeline = row['pipeline']\n",
    "    ratio = row['in_degree']\n",
    "    \n",
    "    if pipeline not in max_ratios or ratio > max_ratios[pipeline]:\n",
    "        max_ratios[pipeline] = ratio\n",
    "\n",
    "max_ratios_df = pd.DataFrame(list(max_ratios.items()), columns=['pipeline', 'in_degree'])\n",
    "\n",
    "print(max_ratios_df)\n",
    "\n",
    "max_ratios_df.to_csv('./results/max_in_degree_processes.csv', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAlwAAAG4CAYAAACQITm5AAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjcuMiwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy8pXeV/AAAACXBIWXMAAA9hAAAPYQGoP6dpAABu7ElEQVR4nO3dd1gU1/s28HvoAgIiNopd7AUVO4LGEktENFai2DUmlkRjoiaaHjUxGo3R2MvXEjV2jbGDBSMotogaVFTAhiJdFDjvH/yYd1d2l4XZBYH7c117OeycOfPs2WV5PHPmHEkIIUBERERERmNS2AEQERERFXdMuIiIiIiMjAkXERERkZEx4SIiIiIyMiZcREREREbGhIuIiIjIyJhwERERERkZEy4iIiIiI2PCRURERGRkTLiIAKxduxaSJEGSJAwbNqyww1Hkyy+/lF/Ll19+Wdjh5FlRj59KjmHDhsmf1bVr1xZ2OPSGY8JVjPj4+Mi//NmPPXv25KmOqVOn5qiDf/SIiIiUYcJVzK1bt07vshkZGdi0aZMRoyEiIiqZmHAVc/v27UNcXJxeZQ8fPowHDx4YOSIiIqKShwlXMVWvXj0AwMuXL7Flyxa9jlm/fn2O46no+fLLLyGEgBCCl4OJiN4QTLiKqUGDBsHc3ByAeiKlTUJCAnbt2gUAaNKkCRo2bGjM8IiIiEoUJlzFlJOTE7p16wYAOHv2LP777z+d5bdt24bU1FQAQEBAgNHjIyIiKkmYcBVjQ4cOlbdz6+XK3m9mZobBgwfn6Tznz5/HDz/8gJ49e6J69eqwtbWFhYUFKlSogDZt2mDmzJm4d++ezjri4uLg5uYm3xk5ceLEXM87Z84cubyDgwPu3r2bp7iVEEJg27ZtGDRoEGrUqAFbW1vY2tqiRo0aGDx4MLZv3w4hRJ7qPHDgAPr27QtXV1dYWVnBxcUFb731FtauXYtXr14B0O829PxMq3D58mV89tlnaNmyJSpWrAgLCwvY2tqidu3aGDBgAFatWoX4+Hitx9+9exdLly7FoEGD0KBBA9jb28Pc3Bxly5ZFo0aN8P777+Ps2bN5ag9DUb1798SJEwCAqKgozJo1C02aNIGjoyNsbGxQp04dTJo0CTdv3sy1Tk1tnJqailWrVqFLly6oXLkyLCwsIEkSLl68mOP4pKQkLFq0CF27dpXf7zJlyqBBgwb48MMP8c8//+T5dSYkJGDx4sV45513ULVqVdja2sLS0hLOzs5466238NVXX+Hff//NtR4hBHbu3ImAgAC4u7vD3t4eVlZWcHNzQ+/evbFu3Tqkp6frFdP169cxbdo0tGrVCk5OTrCwsIC9vT1q1qyJVq1aYfz48di1axcSEhK01pGUlIRly5ahR48eqFy5MqytrWFlZQVXV1c0adIEfn5++O233xAREaF3W+nj8OHD6N+/P9zc3LT+PuaVIdsWAJKTk/Hjjz+iVatWKFu2LGxsbODu7o4RI0bg3LlzcjnVu8610VTm0qVLmDRpEho0aABHR0dIkoTevXtrPP7p06eYP38+OnfuLLeZg4MD6tWrhw8++AChoaF6v65sR48exbhx41C/fn04OjrKn+euXbvi119/lTsJigxBxYa3t7cAIACIpUuXirS0NOHo6CgAiKpVq4rMzEyNx925c0dIkiQAiB49egghhBgwYIBc1+zZs7We09PTUy6n62Fubi7mzp2rM/4TJ04IExMT+Zj9+/drLRsSEiLMzc3lshs3bsy9gXRYs2aNXFdAQIDOsjdv3hQeHh65vuZmzZqJiIiIXM/94sUL0b9/f511tW7dWkRHR4uAgAD5uTVr1misb/bs2Xq9d0IIERcXJwYMGCC//7oeFSpU0FjH1KlT9ToegBg0aJBITk7WGVNe4teH6u/F8ePHxZ49e4SDg4PWGC0tLcXixYvzFOO1a9dE/fr1NdYXFhamduzevXtFxYoVc22rwYMH59pW2ZYuXSrKlCmj13vw119/aa3n0qVLokmTJrnWUbt2bfHvv//m2kZmZmZ6xeTv76+xjjNnzggXFxe96gAgXr16pVd76fLy5UsxZMgQnedp1aqV3r+P2QzZtkIIceHCBVG1alWt9UiSJGbNmiWEEGrPa/N6mdmzZwtTU9Mc9fr6+uY49tdffxX29vY6X5ckSWLEiBEiLS0t19d279494ePjk2tbOTs7i6CgoFzre1OYgYotCwsL9O/fH8uWLUNkZCSCgoLg7e2do9z69evl3hjVXjF9ZPdcWVpaon79+qhZsybs7e0hhMCDBw/wzz//IDY2Fq9evcKnn34KAJg2bZrGury9vfHpp5/ihx9+AAAMHz4cly9fRoUKFdTKJScnY/DgwfL/Mt97770898rlV3h4OLy9vfHkyRP5uYYNG6JJkyaQJAlhYWG4cuUKgKyev7Zt2yIoKAju7u4a6xNCoH///mrzpZUrVw4+Pj6wt7dHZGQkAgMDERwcjD59+qBmzZoGey0xMTHo2LEjbty4IT/n4OCAtm3bolKlSnj16hXu3r2L8+fPIzExES9evNBYz/379yGEgCRJqF27NmrXro2yZcvC3NwcT58+RVhYGG7dugUA2Lx5M+Lj47Fv3z6d/9s2ltDQUMycORMvX76Eo6MjfHx84OjoiHv37uHEiRN4+fIl0tLSMGHCBJiYmGD8+PG51vn06VO8/fbbuHfvHqysrODl5YUqVaogMTExR6/eH3/8AX9/f2RkZAAATE1N0a5dO9SsWRNJSUk4efIkYmJiAACbNm3CnTt3cOzYMVhZWWk9/8SJE7F48WL5Z1NTU3h6eqJWrVqwsrLCkydPcPHiRURGRgKA1vcxKCgI77zzjtzbZGZmhubNm6N27dowNzdHZGQkTp06hRcvXuDGjRto06YNgoODUbdu3Rx1LVy4EF999ZX8s5OTE1q1aoVKlSpBkiQ8e/YM169fR3h4uNwWr7t//z66du2KxMREAIC5uTk8PT1Rs2ZNWFtbIzk5GZGRkbh06ZLOHrK88vf3x7Zt2+Sfy5Ytiw4dOsDBwQG3b99GUFAQzp49Cz8/P71/Hw3ZtgBw8+ZNdO7cGU+fPpWf8/DwQOPGjZGRkYHz58/j2rVr+Prrr+Hk5JTnNvjxxx/l969GjRpo0aIFrK2tERkZKY8NzvbRRx9h4cKF8s9ly5ZFq1at4OzsjBcvXiAsLAxXr16FEAKrV69GTEwM9u/fDxMTzRfYwsPD8dZbb8l3zEuShCZNmqB+/fqwtrZGdHQ0goKCkJiYiJiYGHTu3Bl//fUXOnTokOfXWeAKNd0jg3q9h0uIrP8hZj83YsQIjcfVrFlTABAODg4iNTVVCKF/D9f7778v9u/fL1JSUjTuT09PF2vWrBE2NjYCyOrpun37ttb6Xr58qdZr1q1btxxlhg8fLu+vVq2aiI+P11qfvvTp4UpLSxONGzeWy5UrV078/fffOcr9/fffwsnJSS7XtGlT8fLlS411/v7772r/Y5s5c2aOsvfu3RNt27aVe1+yyyrp4Xr16pVcJwBRqlQp8euvv2qMMy0tTezZs0f07t1bY13z5s0Ta9asEU+ePNG4XwghgoKC5M8ZALFhwwatZY3Zw2VhYSEAiI8//li8ePFCrVx0dLTo2LGjXNbS0lKEh4fnGmN2L867776bow0yMjLkNo2IiBC2trbycc2bNxc3b97MUX7+/PlqPb0TJkzQ+tqWLl2q9vnp37+/uH//vsayV65cERMnTtT4mX3w4IGoUKGCXM+gQYNEVFRUjnIPHz4Ufn5+crmGDRuK9PR0tTKvXr0SZcuWlcv88MMPWj//T58+FatXr9bY+z1p0iS5Di8vLxEdHa2xjlevXokTJ04If3//HLHkler3gLbPSUREhGjWrJna50nX76Mh21YIITIzM0W7du3kcuXLlxfHjx/PUW7Xrl2idOnSat8Zuv7kq5YxMzMT9vb2YufOnTnKqbbHqlWr5GNsbW3lKyuvO3bsmFpPpbarHcnJyaJu3bpyuU6dOokbN27kKBcfHy/GjRsnl6tUqZJ4/vy51tf2pmDCVYxoSriEEMLd3V0AEHZ2djkSo9OnT8vHjBkzRn5e34RLX1u2bJHrmzZtms6y//33n9ofpkWLFsn7tm7dKj9vamoqTp8+rTg2IfRLuFavXi2XMTc3F6GhoVrrO3funNrllHXr1uUo8+rVK1GpUiW9/rAmJCSIGjVqqH0pKkm4VqxYofblWhDd8nfu3BFWVlYCgGjRooXWcsZMuACIcePGaS2bkpIiGjVqJJft169frjECEF26dBEZGRk64xg6dKhcvnr16iIuLk5r2Z9//lkua2JiovE/Kc+ePROlS5fW63XlZsSIEXI9o0aN0lk2PT1ddOjQQS6/ZcsWtf1XrlyR97Vt2zbfMWUnNQDEf//9l+969JWeni5cXV31aofY2Ngclzq1/T4asm2FEGLfvn1q34Hnzp3TWt/+/fvVYtQ34ZIkSQQGBuqMNSEhQb40b2pqmmv5a9euyb//ZcuW1Xi5/Ouvv5ZjePvtt3O9RKz6OzVnzhydZd8ETLiKEW0J1zfffCM/v2nTJrVjxo4dK+87deqU/LyhE6709HQ5iWratGmu5VWTGysrK3HlyhVx7949tXEqhogrmz4JV8uWLfVKjrK9//77cvlWrVrl2L937161/x3m1lO3ceNGgyVcderUkct8+umnub4WQ+nWrZv8ha7t9Roz4SpdunSu7Xz48GG1xFpTz93rCde1a9d01hkXF6fW0/Dnn3/qLJ+RkaE2Juyzzz7LUWbOnDny/ipVquToidHX48eP5djs7e1FYmJirscEBwfL537nnXfU9qn+J05br6g+atWqJddTEL0XqsmJtbW1ePbsmc7yqr072n4fDd22QgjRt29fef/QoUNzra9z5855Trj69++fa70LFy6Uy48cOTLX8kKo/715/Xfg5cuXonz58vJ/MiIjI3OtLzo6Wh4/2rBhQ71iKEwcw1UCDBkyBLNmzYIQAuvXr8egQYMAAGlpafjjjz8AZF2nb9u2raLzXLlyBRcuXEBkZCQSEhKQlpamtj97zM6VK1eQmZmp9Ro+kDV+6+DBg9i6dStevHiBQYMGwdHRUZ41v3Xr1vjiiy8UxZsXSUlJanfZjBgxItdjRo0ahaVLlwIAQkJCkJycDBsbG3l/9h1zANCzZ0/Y2dnprM/Pzw+lSpVSfGfO3bt3cf36dfnnDz/8UFF9qu7du4dz587h5s2beP78OVJTU9Xu1rxz5w4AQAiBS5cuwcvLy2Dn1oevr2+u7fzWW2/B1dUVUVFRePXqFU6dOqX1ziwAaNSokdaxNtnOnDkj/z6ULVsWvXr10lnexMQEI0aMwJQpUwAAx48fz1Hm4MGD8vbo0aNhaWmps05tjhw5IsfWs2dP2Nra5npMy5YtYW1tjZSUFJw6dUptX+XKleXtY8eOITw8PNf20aRy5crydDZLlizBjBkz8lxHXqi2cY8ePVCmTBmd5fv374/x48fn+J5TZei2BYDAwEB5W5+xq/7+/jh8+HCu5VQNHDgw1zIHDhzIU3kA6NixI37//XcAwKlTp9CnTx95X2hoKB4/fgwg6/u9SpUqudbn7OyMOnXqIDw8HFevXsXz58/h4OCgVyyFgQlXCVClShW0b98egYGBOHz4MB4+fIiKFStiz549eP78OYCspCy/1q1bh++//16v2+kB4NWrV4iPj8/1C+3333/H2bNnce/ePVy9elV+3s7ODhs3boSpqWm+Y86rS5cuyYN7bW1t0ahRo1yPadKkCWxsbJCcnIyMjAxcunQJbdq0kferThfQokWLXOsrVaoUGjRogJCQkLy/ABWqA7lr1aoFV1dXRfUBQHBwMD777DOcPHlS7+kwYmNjFZ83r1q1apVrGUmS0LJlS0RFRQEAwsLCdCZczZo1y7XOsLAwebtly5YwM8v9q1f1P0BhYWHyjQnZVKeOUDJgODg4WN6+efOm3gl4dixxcXFq/5lwdXVFmzZtcObMGSQkJKB58+bw9/eHn58f2rVrh9KlS+tV/4ABA3D06FEAwMyZM3Ho0CH4+/ujc+fOqFq1ah5eoX5Ufx9btmyZa3lbW1s0aNAA58+f11rG0G0bFRWl9nvj6emZa136lHmdPp9p1de2fv16eeJsXbJ/p4CsmyK01RcbG6t3W2X/DRNCIDo6mgkXFb6hQ4ciMDAQGRkZ2LhxI6ZMmSLPvSVJUr4SLiEERo4ciTVr1uT52MTExFwTLgcHB2zYsAEdOnRAZmam/PySJUtQrVq1PJ9TCdW7Et3c3HT2zmUzMTGBm5ub3Jv0eoKh+rO+SY+Li4vihOvRo0fydvXq1RXVBQCrV6/GqFGj8jzvWPbdZwVJtfdFFzc3N3lb9b3XpFy5crnWp1qHPv9zB6CWVLx8+RKJiYly71xCQoJaT6eS9zH7rkggqyc2P5+vuLg4td7b1atXo0OHDnjw4AFSUlKwYsUKrFixAqampmjYsCHat2+Pbt26oVOnTlqTz5EjR+LQoUPYvn07gKyenezeHWdnZ3h5eaFjx47w8/PT6z3Ijep7lJfPia6Ey9Btq/qdYW1tDUdHx1yPd3FxyfM5c2vPpKQktd/fDRs25Pkcr6/xq9pWN27cULt7Or91vmk48WkJ0a9fP1hbWwPI+t/I48eP5UsS7dq1y9cX9ooVK9SSrZ49e2LDhg24evUq4uLikJaWJq/pJ4RQ+0OjmkDp4ujoqHYbsrW1NTp37pznWJVKSkqSt1X/sORGtezrCYZqndnvTV7qyy/VOPS5xKFLeHg4xo4dKydbjRo1wqJFixASEoLHjx/LlxSzH6qrGOj7GTCk/LRzbolhqVKlcq0vP5+f18upxvF6TEreR10T2urr9ck6a9eujUuXLuGjjz5SSwoyMjJw8eJFLFq0CN26dUOVKlWwYsUKjXWamJhg69atWLt2bY4e5ZiYGPzxxx8YO3YsnJ2dMWrUKDx79kzRazDG76Oh27agvjNy+0wb4zNjjDrfNEy4SojSpUvLl0UuX76MTz/9VP5w5nXurWw//fSTvP3dd99h7969eO+991C/fn04ODjAwsJCrXxeezTS0tIwePBgtTESKSkpGD58eL7iVUL1D1pycrLex6mWff1SiuoXYUpKSp7ryy/VOFS/wPNjwYIF8ueoe/fuCA0NxYQJE9C8eXOUK1cux/xRhdGrpSo/7azvJTBd8vP5eb2cahyvx6TkfVT9HC5cuFAtQdb3oekSX7ly5fDzzz/j4cOHOHnyJL777jt069ZNbQxdTEwMxowZo3VlCUmSEBAQgEuXLiEiIgKrVq1CQECA2n8Q09PTsWrVKrRo0SLX3khdVN8jQ/0+GrptC+s743WvJ3HPnz/P8+tSHcP6ep2TJ0/OV1v5+PgY/LUaEhOuEkQ1scpeFsbKygr9+vXLc13379+XB7SWKVNG62Sm2RISEvLc3Ttt2jR5ElFXV1f5F/Kvv/5Sm+ixIKh2sWdP9JmbzMxMtXEKr09AqPqz6tgGXaKjo/Uqp4vqRLLZg9jzK3uMDQB88803OSZFfF1BLr+kSW5LTGkql5+JI1+n+vnRNwbVtrKwsFBLsuzs7NR6IZS8j6qfh9zWXM0Pc3NztGvXDjNmzMCBAwcQGxuLgwcPqk3CvHjx4lwvt9WoUQMjRozA2rVrcevWLdy8eRNTp06VL0neunVLbbLVvMrPe/T6OKTXGbptVT+LKSkpen2nGuI743UODg5qN2kY4rUZ+3P4JmDCVYJ06tQJzs7Oas/5+vrC3t4+z3WpXm+vXbt2roOAT506lacxPn///becVJmYmGDjxo1qsxlPmzZNbSC9sTVu3FgepJ+UlCQngrpcunRJ/t+lqakpGjdurLa/SZMm8rbqumfapKamGuQ1qw4cv3nzpt7Jniaqn4P69evrLBsfH4/Lly/n+1yGoDowVxshhNqA9KZNmyo+r4eHh7x97tw5rbOrqzp9+rTa8a/PzK86sPvYsWP5jk21nr///jvf9ejL3NwcXbt2xaFDh9CwYUP5+b179+apnlq1auHHH3/E119/LT+numJDXqn+Puqz7mdSUlKuv4+Gbls3NzeULVtW/lmf7w2lYz61Ub3RxxCvTbWtAgMDdd79WVQx4SpBTE1Nc9xGnN/LiaqDxvXp2s6eHkEfT548wbBhw+QE7dNPP0X79u0xatQo+TbiFy9e5LjcaEy2trZo3ry5/LO2haNVqY5va9GiRY5ueNXu73379uW6PMnOnTsNslhrlSpV1G7TX7JkSb7rysvnYOXKlfle9NdQ9uzZk+tYkSNHjsi9Aubm5oqnSwGANm3ayD0CT548wf79+3WWF0Jg3bp18s8dO3bMUaZbt27y9ooVK/L9u9C1a1f5P0wRERHYt29fvurJKwsLC7XxmKo3c+RFz549FdcBqN/peeDAgVzHhP3xxx+5trkx2la1Z3DTpk25lt+4caPic2qi2u7Lli3TumSUvtq2bSvfYZiUlKR1bF9RxoSrhJk5c6Z8t0xISAi6du2ar3qqVasm/4/76tWr8lp5mvzxxx95+qIZMWIEHj58CABo3ry52mWCFStWyHfdXLlyRV6fsSCMHTtW3l6yZInO3pqwsDAsW7ZM/nncuHE5ynTr1g0VK1YEkPUFo2tescTERMyaNSs/YWv08ccfy9vz58/HyZMn81WP6lia3bt3ay3333//KbrcYyiJiYn47LPPtO5PTU3F1KlT5Z99fX0Ncgecg4MDBgwYIP/8ySef6BzPtmTJEly6dAlAVlI7evToHGVGjx4tjzu6e/cuJk+enK/YXFxc8N5778k/jxs3Tu/LUJmZmTnGTcXFxel9Q4TqpbvX21nfaUN01ZEXXbt2lb9bUlJSdH63PH36FLNnz861TkO3LQC1MawbN27U2YN14MCBPM/Bpa+xY8fKCVJUVBTGjx+fpylhXu/ltbS0VPsMz5gxQ68rCdmUJNsFJv9zptKbRttM8/mhz0zzrVq1kst4eHiI69evq+3PyMgQv/76qzA3Nxempqbysg4AxJ07dzTWuWTJErmMjY2NxnW0jhw5Is8uLEmSOHjwoKLXKkT+1lKsWLGiOHbsmMb4smdMBnSvpfjbb7+pzfCsbS3F7LXTDLmWYps2bdRm1l6yZEme11L87LPP5DrKlCmj8b04cuSIcHZ2lt9TQ8SfF5rWUpwyZUqOmdljYmLEW2+9pVb233//NViMt27dUluyqnXr1uLWrVtqZTIyMsSiRYuEqampXE7Xqgaqvy+A7rUUr169qnUtxZiYGLVlpipVqiS2bdumdbmiqKgosXDhQlGrVi2xePFitX1r1qwR1atXF/PmzdO6bmpqaqpYsGCB/HsMIMcyXaVKlRKjR48Wx48f17pGYnBwsKhatapcx+jRozWW09frs8dPnTo1x9qAt27dytNaioZsWyGyPiOtW7eW6zPGWor6en3tyR49emhdfzQzM1OcOXNGfPDBB8La2lrjrPuJiYlqKyzY2dmJ5cuXa1yfUYisJZZWrFghmjZtKqZMmaJ33IWFCVcxUtAJ15EjR9QW2TU3Nxc+Pj5ixIgRon///mpfMt99952oUqWKzoTr2rVrolSpUnKZ5cuXa41vypQpaomProWT9aFPwpUdY7ly5dS+ZBo3biwCAgJEQECAWkKW/WWoKWnMlpmZKXr06KF2TLly5US/fv3EqFGjRKdOnYS5ubkAIFq2bCkGDx4sl9O0PqMQ+icD9+/fV1s+BchawLxnz55i9OjRYtiwYaJDhw7Czs5OAFlLk7zu0aNHOdqjadOm4r333hNDhgxR+/Ls2rWrGDJkSKEmXD/++KP8R9LR0VG8++67YvTo0aJr1645/jD98ssvWuvMb4xbtmxRS6bMzMxEhw4dxKhRo8SgQYPU1vIDspaEyl5QXhvVJaSArHXtWrVqJYYOHSrGjBkj/Pz81JISTQsSC5G1/qfqouvZn9933nlHjBo1SowYMUL4+vqKGjVqqCVKmhIu1ToqV64sevbsKUaMGCGGDx8uunXrprZEFwDh7++fIx7V/aVLlxZeXl7C399fjBkzRvTp00fUq1cvx++NtgWu86JPnz5q9To5Ocm/j2+99Za8Rqqnp6fa76O2z7Mh2zZbeHi4cHR0zPF7N2zYMDF06FC1tlm0aJG8LUmS1hjzk3AJIcQXX3yhdqwkSaJhw4Zi0KBBYuzYseK9994THTp0kNddzH5oW+bo1q1bolq1ampl7e3txdtvvy1GjhwpRo0aJfr06SPq1q2r9veHCRcVqIJOuIQQYunSpWqLNL/+MDExEbNmzRKZmZk6E660tDTRpEkTeX9ua7C9Xr5Xr16KXq++CZcQQty4cUN4eHhofc2qX4ARERG5njs1NVVtfTRNj9atW4vo6Gi1L/gdO3ZorC8vycDTp0+Fn59frq8FgHBxcdFYx5kzZ3L8MXn90bt3b/H8+XMREBBQqAnX8ePHxe7du4W9vb3WWC0sLMTChQt11qkkxr1794oKFSrk2t6DBg3SuMCvJgsXLpQTY10PSZI09nBli4yMVOvly+1RoUKFHL2a27ZtU0sadD1MTEzE+PHjNfasqvYG5vZo3Lix1p6VvEpLS1P7PdP0aNGihYiKitLr82zItlUVGhoqKleurPO9/uKLL8TLly/l5+zs7LTWp3psXv3xxx9yL7Y+jxYtWuhc+/Pp06eiX79+en+OHBwcxNq1a/Mcd0HjTPOkyLhx49C2bVssWLAAx48fR0xMDEqVKgUXFxd07NgRI0aMULtDS5vp06fLS2tUqlQp1wGTFhYW2LRpE5o1a4bU1FTs2bMHy5Yt0zhWytDc3d0RGhqK7du3488//8S5c+fkNcDKly+Pli1b4t1330Xfvn1z3FmmiZWVFbZv3479+/dj1apV+OeffxAbG4uyZcuibt268Pf3x5AhQ2Bubq42kNcQS1g4Ojpix44dCAkJwaZNm3DixAlERUUhLi4OpUqVgqurK5o0aYK3334b7777rsY6WrdujX///RcLFy7E3r17cfv2bQBZ72OzZs3w3nvv4Z133lEcq6H06tULly9fxrJly7Bv3z7cv38faWlpcHV1RdeuXfHhhx+idu3aRjt/z549ERERgdWrV2Pfvn34999/ERsbi1KlSsHZ2RkdOnTA0KFD9VpeJtukSZPw3nvvYe3atfj7779x7do1eQyUk5MT6tatC29vbwwYMAC1atXSWk+VKlVw5MgRBAcHY9u2bQgKCsL9+/cRFxcHMzMzlC1bFrVq1ULz5s3RpUsX+Pj45LhD+d1338WDBw9w6NAhnD59GpcuXcLt27flJVjs7e3h7u6Odu3aYejQoahXr57GWJ4+fYqgoCAEBgYiJCQE//33Hx49eoQXL17A2toarq6uaNasGfr27YtevXrptfqDPiwsLLBx40YMHToUK1asQHBwsPz7WLt2bQwePBgBAQE55hnMjSHaVlWzZs3w77//4rfffsP27dvx33//IS0tDS4uLmjXrh3GjRuHli1bqo1tMtayN/3794evry+2bNmCv//+GyEhIXjy5AmSkpJgY2MDFxcX1K1bF15eXujevTvc3d111ufo6IitW7fi6tWr2Lx5M06cOIE7d+7g6dOnMDExgYODA2rWrImmTZuiU6dO6Ny5c445/95EkhB5XI+DiAqNi4uLPBXDgwcP5EH3pJ2Pj4+8JMzx48ff+MkRiQzp8OHD6NKlC4CsGwNUFz2ngsW7FImKiNOnT8vJlqurK5MtIsrV1q1b5e38LGRNhsOEi6gIePXqldpUDoMGDSrEaIioKAgNDVWbz23gwIGFGA0x4SIqZLNmzcKiRYvw9OlTjfvDw8PRqVMneVZpa2trjB8/viBDJKI3TPZs/ZpWLcjMzMTmzZvRuXNnebLh7t2757oaBBkXB80TFbJ79+7hm2++wZQpU9C4cWPUrl0bdnZ2SExMxNWrV3H58mW1CQUXLlyocbFgIio5Dh06hEOHDsHR0RHNmjWDq6srzM3N8fjxYwQHB6sNlq9QoQKWL19eiNESwISL6I2Rnp6O8+fP4/z58xr329nZYdGiRQgICCjgyIjoTfXs2TOds8k3adIEf/75pzyLPhUeJlxEhWzhwoXw8fHBsWPHcO3aNTx58gRPnjyBEAKOjo6oX78+OnXqhFGjRsHR0bGwwyWiN8DVq1exa9cunD59Gnfv3kVsbCzi4uJgbW2N8uXLo1WrVujduzf8/Pz0mp6GjI/TQhhZZmYmYmJiULp0aX7oiYiIijAhBBITE+Hs7Jznud/Yw2VkMTExcHNzK+wwiIiIyEDu378PV1fXPB3DhMvISpcuDSDrzbGzsyvkaIiIiCi/EhIS4ObmJv9tzwsmXEaWfRnRzs6OCRcREVExkJ8hQpyHi4iIiMjImHARERERGRkTLiIiIiIjY8JFREREZGRMuIiIiIiMjAkXERERkZEx4SIiIiIyMiZcREREREbGhIuIiIjIyJhwERERERmZ0Zf2SUtLw6lTpxAbG4tq1aqhRYsWxj4lERER0RtFUcJ19+5dLFmyBAAwY8YMODg4qO0/e/Ys3n33XTx48EB+rmnTpvjzzz9RuXJlJacmIiIiKjIUXVLcuXMnfvrpJxw7dixHspWYmIjevXvjwYMHEELIj/Pnz6NHjx5IT09XcmoyssyMjCJZNxER0ZtIUQ/X4cOHIUkSevfunWPf8uXL8fjxY0iShIkTJ+Ktt97C33//jd9++w3Xrl3DunXrMHLkSCWnJyMyMTXFDn9/PAkPN2i95erWRZ+NGw1aJxER0ZtOUcJ1+/ZtAECzZs1y7Nu6dSskSYKfnx8WLlwIAHjnnXfw5MkTbNu2Ddu3b2fC9YZ7Eh6Oh2FhhR0GERFRkafokuLjx48BABUqVFB7PiEhARcuXAAADB8+XG3fwIEDAQCXLl1ScmoiIiKiIkNRwpWYmAgAyHhtTM7p06eRkZEBU1NT+Pj4qO1zc3MDADx79kzJqYmIiIiKDEUJl729PQAgJiZG7fkTJ04AABo3bgwbGxuNx1pZWSk5NREREVGRoSjhatCgAYCsuxWzZWRkyOO3OnTokOOY6OhoADkvQxIREREVV4oGzfv5+SEwMBAbNmxAhQoV4OXlhQ0bNuDu3buQJAn9+/fPcUxoaCgAcB4uIiIiKjEU9XCNHTsWdevWhRACP/30E3x9fbF9+3YAWXckNm/ePMcxO3fuhCRJOcZ2ERERERVXihIuS0tLHD16FH369IGZmRmEEDA3N8eQIUOwYcOGHOWDgoJw7do1AEDnzp2VnJqIiIioyFC8lmLFihWxfft2pKWl4dmzZyhbtiwsLCw0lnVzc8Px48cBAJ6enkpPTURERFQkGGzxaktLS1SqVElnmWrVqqFatWqGOiURERFRkaDokiIRERER5U5RD1fFihXh7e0tP+rXr2+ouIiIiIiKDUUJ1+PHj7F9+3b5zsSyZcuiffv28PHxgbe3Nxo2bGiQIImIiIiKMkUJ18SJExEYGIgrV64gMzMTsbGx2LlzpzwRapkyZdQSsMaNGxskaCIiIqKiRFHCtXDhQgDA8+fPERgYiMDAQJw4cQKXL19GZmYmnj17ht27d2P37t0AAAcHB3h5eckJmIeHh+IXQERERPSmk4QQwtCVxsfHqyVgly5dQmZmZtYJJUn+Nz093dCnfuMkJCTA3t4e8fHxsLOzK+xw8uT3pk3xMCzMoHVW9PDA2AsXDFonERFRQVDyN90odyna29ujV69emD9/Ps6fP4/Hjx/jiy++kBe7FkLACHkeERER0RvJYPNwqRJCICwsDCdOnMCJEydw8uRJJCQkqCVZ1tbWxjg1ERER0RvHIAmXtgQrex8AlCpVCq1bt4aPjw98fHzQsmVLQ5yaiIiI6I2nKOFasGABjh8/jlOnTiE+Ph6A7gTL3NxcecRERERERYyihGvKlCmQJAlCCCZYRERERFoY5JKiJEnw8vJCly5d4OPjAw8PD/luRCIiIqKSTlHC1bFjR5w9exYpKSk4dOgQDh8+DACws7OT59tiAkZEREQlnaKE68iRI3j16hXOnTuHEydOIDAwEGfOnEF8fDz27duH/fv3A2ACRkRERCWbwSc+TU9Pz5GAJScnZ53s/5Ise3t7eHl5yTPQ50dCQgIOHDiAkJAQhIaGIjo6Gk+ePEFqaiocHBxQr149dO/eHSNHjkTZsmVzre/gwYNYvnw5zp07hydPnqBcuXJo0aIFxowZg7fffltRnJz49P/jxKdERFRUKfmbbpSZ5lWlp6cjJCQE+/btw2+//SbfzShJEjIyMvJd75EjR9C5c+dcyzk5OeF///sfunbtqnG/EALjxo3D8uXLtdYxZswYLFu2LF+9cky41DHhIiKiokrJ33SjTHwKACkpKTh16pQ8N9f58+eRnp4u39VoCG5ubujQoQOaNWsGNzc3VKpUCZmZmYiKisL27duxY8cOxMbGolevXggJCUGjRo1y1PH555/LyZaHhwemTZuGGjVq4NatW5g3bx7CwsKwfPlylCtXDt9++61B4iYiIqKSxWA9XNoSLAA5ZpjPnj5i5syZ+T5fRkYGTE1NdZbZtWsX/Pz8AAB9+vTBn3/+qbY/IiICdevWRXp6Opo3b46goCCUKlVK7TV5e3sjNDQUZmZmuH79OmrUqJGnONnDpY49XEREVFQVWg/X4cOHcfz4cZ0Jlo2NDdq0aQMfHx94e3ujRYsWMDNT3rGWW7IFAL1790adOnVw/fp1BAUF5di/YMECOebFixerJVtAVnK4ePFitG7dGunp6Vi4cCEWL16sOHYiIiIqWRRlPl27ds1xidDW1hZt27aVEyxPT0+9kiNjsbGxAQC8ePFC7XkhhDxov06dOmjVqpXG41u1aoXatWvjxo0b2LVrFxYtWsQ7LImIiChPFHc1lS5dWi3BatasWaEmWKrCw8Nx8eJFAFlJlao7d+4gOjoaAODt7a2zHm9vb9y4cQNRUVGIjIxEtWrVjBIvERERFU+KEq5z586hadOmMDExMVQ8iqWkpCA6Ohp79+7FvHnz5DshJ02apFYuPDxc3n49GXud6v7w8HAmXERERJQnihKu5s2bGyoORdauXYvhw4dr3T916lT4+/urPXf//n1529XVVWf9bm5uGo8jIiIi0ofRpoV4EzRp0gTLli1Dy5Ytc+xLTEyUt21tbXXWkz0ODACSkpJ0lk1LS0NaWpr8c0JCgr7hEhERUTFlsGuBz549w/z589GtWze4ubnBxsYGNjY2cHNzQ7du3TB//nw8e/bMUKdT07t3b1y5cgVXrlzBuXPnsHnzZvj5+eHixYvw9/fHvn37chyjOojewsJCZ/2Wlpbydmpqqs6yP/zwA+zt7eWHau8YERERlUwGSbh+//13VKlSBdOmTcOhQ4cQHR2N1NRUpKamIjo6GocOHcK0adNQpUoVnTO655eDgwMaNGiABg0awNPTEwMHDsSOHTuwfv163L59G76+vli7dq3aMVZWVvL2y5cvddav2mP1+tQRr5s+fTri4+PlBy9BEhERkeJLinPmzMHMmTPlqSHs7e3h4eGBihUrQgiBR48eISwsDPHx8UhOTsb777+P58+fY9q0aYqDz82QIUOwb98+bN26FR9++CF8fX1RpkwZAFl3V2bL7TJh9lqQQO6XHy0tLdV6xIiIiIgU9XBdvXoVX3zxBYQQqFSpEv73v//h8ePHOHbsGDZt2oTNmzfj2LFjePLkCTZu3AhnZ2cIIfD555/j33//NdRr0MnX1xdAVtL0119/yc+rDpSPiorSWYdqLxUvERIREVFeKUq4fv31V2RkZKBcuXIIDg7G4MGDYW5unqOcmZkZBg0ahODgYJQvXx4ZGRn49ddflZxab+XKlZO37969K2/Xq1dP3r5+/brOOlT3161b14DRERERUUmgKOE6duwYJEnC9OnTUbly5VzLu7m54dNPP4UQAkePHlVyar1lT24KqF8OrFatGpydnQEAgYGBOuvIXhbIxcUFVatWNXyQREREVKwpSriyk5k2bdrofUzbtm0BADExMUpOrbdt27bJ2w0bNpS3JUmSLzdev34dZ8+e1Xj82bNn5R4uX19fLutDREREeaYo4cpewid7AWh9ZJdVOjv92rVrc6yP+LoFCxbgwIEDAICqVauiXbt2avsnT54sL6Q9YcKEHFM+pKamYsKECQCyLotOnjxZUcxERERUMinKerIvI+bl8mB2WX0uQery5ZdfwsXFBWPGjMH69etx+vRpXLp0CadOncLSpUvRrl07fPzxxwCy5tlasWKFnFxlc3d3x9SpUwEAoaGhaNu2Lf744w+Ehobijz/+QNu2bREaGgoA+OSTT1CrVi1FMRMREVHJpGhaiM6dO+PatWv46aef0Lt3b7VLdppcvnwZP/74IyRJQpcuXZScGkDWZKsrVqzAihUrtJZxdXXF6tWr0alTJ437v/vuOzx+/BirV69GWFgYBg4cmKPMyJEj8e233yqOl4iIiEomRT1ckydPhqWlJZKSktCuXTv8+OOPiI2NzVEuNjYW8+bNQ/v27ZGUlARLS0vFl+eOHj2KZcuWYcCAAWjUqBEqVKgAMzMz2NraokaNGujbty/WrFmDGzduoHPnzlrrMTExwapVq7B//374+vrC2dkZFhYWcHZ2hq+vLw4cOICVK1e+UQt0ExERUdEiiewZS/Np/fr1agtHS5KEqlWrokKFCpAkCQ8fPkRkZCSEEBBCQJIkrF27FkOGDFEcfFGQkJAAe3t7xMfHw87OrrDDyZPfmzbFw7Awg9ZZ0cMDYy9cMGidREREBUHJ33TFM80PHToUZcuWxbhx4xAdHQ0hBG7fvo07d+4AAFTzOWdnZyxfvhzdu3dXeloiIiKiIkNxwgUAPXr0wJ07d7Bz504cOXIEV69elReqdnR0RIMGDdCpUyf4+fnlGLhOREREVNwZLPsxMzNDv3790K9fP0NVSURERFQscCQ4ERERkZEx4SIiIiIyMiZcREREREam1xiu7CV8DEmSpDwtCURERERUVOmVcCmcqouIiIioRNMr4Zo9e7ax4yAiIiIqtphwERERERmZXgnXokWLAABDhgxBmTJljBoQERERUXGjV8I1efJkSJKETp06qSVcHTt2hCRJWL16NapUqWK0IImIiIiKMkUzzZ84cQKSJCE5OdlQ8RAREREVO3rNw2VlZQUAeP78uTFjISIiIiqW9Eq4qlatCgDYt2+fMWMhIiIiKpb0uqTYvXt3XL9+HXPnzsXRo0fh7u4Oc3Nzef/nn38OBweHPJ1YkiSsWrUqT8cQERERFUWS0GNW02fPnqFVq1aIiIjIOkiSAPz/CVGzf9aXEAKSJCEjIyOv8RY5CQkJsLe3R3x8POzs7Ao7nDz5vWlTPAwLM2idFT08MPbCBYPWSUREVBCU/E3Xq4fL0dERoaGh+PXXX3H06FFER0cjLS0Nd+/ehSRJqFSpklqPFxERERH9f3rfpWhnZ4cZM2ZgxowZ8nMmJllDwA4dOoR69eoZPjoiIiKiYkCvQfNERERElH+K5uE6fvw4AKBatWoGCYaIiIioOFLUw5Weng5vb2+UKlUqz8eOHz9eyamJiIiIigxFCZefnx9CQkLyfNyYMWPw+++/Kzk1ERERUZGhKOFKSkpC9+7dER4ervcxo0aNwsqVK5WcloiIiKhIUZRwubu74+nTp+jSpQvu3buXa/lhw4ZhzZo1AAB/f38lpyYiIiIqMhQlXIcPH4arqyuio6PRuXNnPH78WGM5IQSGDBmC9evXQwiBgIAArFu3TsmpiYiIiIoMRQmXm5sbDh8+DCcnJ0RERKBr166Ij49XK5OZmQl/f39s3LgRADBixAisXr06z7PTExERERVViufhql27Nv766y/Y2tri8uXL6NmzJ168eAEgK9kaNGgQtmzZAuD/j99iskVEREQliUEmPm3WrBl2794NS0tLnDlzBn379kVqaireffddbNu2DQAwduxYLF++3BCnIyIiIipSDDbTvI+PDzZv3gwTExMcPHgQ1apVw+7duwEA77//PpYuXWqoUxEREREVKQZd2sfX11ee8uHx48cQQuDDDz/EkiVLDHkaIiIioiJFr6V99JnyIVuHDh0wadIkLFy4EO+++y6mTp2q9fjKlSvrXS8RERFRUaVXwpWftRIlScKff/6JP//8U+v+9PT0PNdLREREVNTolXAJIYwdBxEREVGxpVfClT07PBERERHlnV4JV0BAgLHjICIiIiq2DHqXIhERERHlpCjhMjExgZmZGebNm2eoeIiIiIiKHUUJl4WFBYQQ8PLyMlQ8RERERMWOooTL2dkZAGBqamqQYIiIiIiKI0UJV/v27QEAFy5cMEgwRERERMWRooRrwoQJMDU1xU8//YSEhARDxURERERUrChKuJo1a4bFixfj7t278Pb2xpkzZwwVFxEREVGxodc8XNqMGDECAFC7dm1cunQJXl5ecHNzQ6NGjVCmTBmdY7skScKqVauUnJ6IiIioSFCUcK1duxaSJAHISqCEELh37x7u37+v8zghBBMuIiIiKjEUJVyVK1eWEy4iIiIi0kxRwhUZGWmgMIiIiIiKLy7tQ0RERGRkTLiIiIiIjEzRJUVt0tPTERcXBwAoU6YMzMyMchoiIiKiIsFgPVzh4eGYMGEC6tatCysrK1SsWBEVK1aElZUV6tati4kTJ+LatWuGOh0RERFRkWGQhGv69Olo1KgRfvvtN9y4cQOZmZkQQkAIgczMTNy4cQNLlixB48aNMWPGDEOckoiIiKjIUHytb8KECfjtt98ghAAA1K1bFy1btkTFihUhhMCjR49w7tw5XLt2DRkZGZg7dy6Sk5Pxyy+/KA6eiIiIqChQlHCdPn0aS5YsgSRJqFevHpYvX442bdpoLBscHIxx48bhypUr+PXXXzFgwACtZYmIiIiKE0WXFH///XcAQLVq1XD69GmdCVTr1q0RFBSE6tWrAwCWLVum5NRERERERYaihOvkyZOQJAmfffYZ7O3tcy1vb2+PTz/9FEIInDx5UsmpiYiIiIoMRQnXw4cPAQAeHh56H9O0aVMAwKNHj5ScmoiIiKjIUJRwWVlZAQCSk5P1PiYpKQkAYGlpqeTUREREREWGooSrWrVqAIA9e/bofczevXsBQB7LRURERFTcKUq4unfvDiEEfv31Vxw9ejTX8kePHsXixYshSRK6d++u5NRERERERYaihGvy5Mmws7PDq1ev0K1bN3zwwQc4f/48MjMz5TKZmZk4f/48xo8fj27duuHVq1ews7PD5MmTlcZOREREVCQomofLyckJW7duRa9evfDy5UssW7YMy5Ytg4WFBRwdHSFJEp4+fYqXL18CAIQQsLCwwLZt21C2bFmDvAAiIiKiN53ipX26dOmCs2fPonnz5vJyPmlpaXjw4AFiYmKQlpYmP+/p6Yl//vkHnTp1MkTsREREREWC4qV9AKBJkyY4d+4cQkJCcOTIEVy9ehXPnj0DADg6OqJBgwbo1KkTPD09DXE6IiIioiLFIAlXNk9PTyZVRERERK9RfEmRiIiIiHRTlHAtXboU165dM1QseXbhwgV8//336NatG9zc3GBpaQlbW1u4u7tj2LBheV4+6ODBg+jTpw9cXV1haWkJV1dX9OnTBwcPHjTSKyAiIqKSQBJCiPwebGJiAkmS4OTkhPbt28PHxwfe3t5o0KCBIWPUyNvbG0FBQbmWGzJkCFauXAkLCwutZYQQGDduHJYvX661zJgxY7Bs2TJIkpSnOBMSEmBvb4/4+HjY2dnl6djC9nvTpngYFmbQOit6eGDshQsGrZOIiKggKPmbrngMlxACT548wY4dO7Bjxw4AWQPlVROwRo0aKT1NDtHR0QAAZ2dn9OvXD15eXqhcuTIyMjIQHByM+fPnIzo6Ghs2bEB6ejo2bdqkta7PP/9cTrY8PDwwbdo01KhRA7du3cK8efMQFhaG5cuXo1y5cvj2228N/lqIiIioeFPUw/X8+XMEBgbixIkTOHHiBK5cuSJPeqraE1SmTBl4eXnJCViTJk0UB96zZ08MHToUffv2hampaY79sbGxaNu2LW7evAkACAoKgpeXV45yERERqFu3LtLT09G8eXMEBQWhVKlS8v6UlBR4e3sjNDQUZmZmuH79OmrUqKF3nOzhUsceLiIiKqqU/E1XNIbLwcEBvr6+WLBgAcLCwhAbG4tdu3Zh8uTJaNy4MSRJghACz549w+7du/Hxxx+jWbNmcHR0RO/evZWcGvv27UP//v01JltA1qSs8+fPl3/evn27xnILFixAeno6AGDx4sVqyRYAWFtbY/HixQCA9PR0LFy4UFHcREREVPIo6uHKTXx8PIKCguQesEuXLqn1gGVkZBjr1ACApKQklC5dGgDQo0cP7Nu3T22/EAJubm6Ijo5GnTp1EB4errWuOnXq4MaNG3B1dcW9e/f0HsvFHi517OEiIqKiqtB6uHJjZ2cHFxcXuLi4wNnZGdbW1nkedK5E9pJCQNYA/9fduXNHHgvm7e2ts67s/VFRUYiMjDRckERERFTsGXTiUyEEwsLC5B6tkydPIiEhQd4HAKVKlULr1q3RoUMHQ55ao8DAQHm7Tp06Ofar9mhp2q9KdX94eDiqVatmgAiJiIioJFCUcL2eYJ06dQrx8fHyPuD/J1g+Pj7w8fFBy5YtYW5urjzyXGRmZmLOnDnyz/37989R5v79+/K2q6urzvrc3Nw0HkdERESUG0UJV9myZd+YBOt1CxYswLlz5wAAfn5+aN68eY4yiYmJ8ratra3O+mxsbOTtpKQkreXS0tKQlpYm/5zdw0dEREQll6KE6/nz5/KYLF9fX3z00Udo3bp1oSRYqgIDA/HZZ58BAMqXL4+lS5dqLPfixQt5W9fEqABgaWkpb6empmot98MPP+Crr77KS7hERERUzCkeNC+EgBACe/bsga+vL959910sWLAAFy5cgBFvgNTq33//hZ+fH9LT02FpaYmtW7eiQoUKGstaWVnJ26oD7DVR7bV6feoIVdOnT0d8fLz84OVHIiIiUtTDdfLkSXn8VnBwMOLj47F37155+gU7Ozt5wlMfHx94eHgY9S7FO3fuoEuXLoiLi4OpqSk2b96s8+7D7CkjAN2XCQEgOTlZ3tZ1+dHS0lKtN4yIiIhIUcLVtm1btG3bFjNnzkR6ejrOnTsnJ2BnzpxBfHw89u3bh/379wPImYA1bdrUIC8CAGJiYtCpUyfExMRAkiSsXr0afn5+Oo9RHSgfFRWls6xqT5XqAHoiIiKi3BhsWggzMzO0adMGbdq0wYwZM5Ceno6QkJAcCdj+/fuxf/9+SJIkz/CuVGxsLDp37ozbt28DyJoxfujQobkeV69ePXn7+vXrOsuq7q9bt24+IyUiIqKSyKDzcKlVbGaG1q1bo3HjxmjatCkaNmyIVatWISEhwaBju+Lj49G1a1dcu3YNADBnzhx88MEHeh1brVo1ODs7IyYmRm3OLk2CgoIAAC4uLqhataqimImIiKhkMXjClZKSglOnTsk9W+fPn5d7slQTLWtra4Ocq0ePHrjwf0vFzJw5E59++qnex0uSBF9fXyxduhTXr1/H2bNn0apVqxzlzp49K/dw+fr6Fuhs+URERFT0KU649E2wbGxs5Pm5vL290aJFC0XnffnyJfz8/HD69GkAwKRJk/Dtt9/muZ7JkydjxYoVSE9Px4QJExAUFKR2F2JqaiomTJgAIKvXbvLkyYriJiIiopJH8aD50NBQjQmWra0t2rRpIydYnp6eMDMzXIfaoEGDcOjQIQBAx44dMXLkSFy9elVreQsLC7i7u+d43t3dHVOnTsWcOXMQGhqKtm3b4tNPP0WNGjVw69YtzJ07F2H/t4DzJ598glq1ahnsNRAREVHJIAkFA6pUF4QuXbo02rZtKydYzZs3h6mpqUGC1CSvl/WqVKmiddHpzMxMjB49GqtXr9Z6/MiRI7F8+XKNi2DromRl8cL2e9OmePh/yaahVPTwwNj/uwRMRERUlCj5m66oy6l79+5ygtWsWbM8JyNvChMTE6xatQp9+/bF8uXLERISgtjYWDg5OcHT0xNjx45Ft27dCjtMIiIiKqIUJVzZE5wWBmPMYt+9e3d0797d4PUSERFRyVY0u6SIiIiIihAmXERERERGxoSLiIiIyMiYcBEREREZGRMuIiIiIiNjwkVERERkZEy4iIiIiIyMCRcRERGRkemVcPXp0wd9+/ZFVFSUseMhIiIiKnb0Srh27dqFXbt2ISEhQf1gExOYmZnh2rVrRgmOiIiIqDjI0yVFTcvpGGOJHSIiIqLiRK+Eq3Tp0gCAR48eGTUYIiIiouJIr4SrTp06AIBffvkFSUlJOfZLkmTYqIiIiIiKETN9Cg0ePBghISHYt28fHB0dUaFCBZibm8v7u3TpovazPiRJwq1bt/IWLREREVERpFfCNWHCBJw+fRrbt29Heno6oqOj5X1CCLWf9cVeMSIiIiop9Eq4TExMsHXrVgQHB+PIkSOIjo5GWloa1q1bB0mS0KtXLzg4OBg5VCIiIqKiSa+EK1vr1q3RunVr+ed169YBAL777jvUq1fPsJERERERFROcaZ6IiIjIyPLUw/W6zMxMQ8VBREREVGyxh4uIiIjIyBT1cKnKyMjArl27cOTIEVy9ehXPnj0DADg6OqJBgwbo1KkTevfuDVNTU0OdkoiIiKhIMEjCdfDgQYwZMybHdBFA1vQPZ86cwfLly+Hq6orly5eja9euhjgtERERUZGg+JLihg0b0LNnT0RHR0MIASEEqlSpglatWqFly5aoUqUKgKwE7P79++jRowc2btyoOHAiIiKiokJRwnX37l2MGTMGmZmZsLa2xrfffouHDx/i9u3bOHPmDIKDg3H79m08fPgQ3333HWxtbZGZmYnRo0fj3r17hnoNVITYVqyIzIwMo9VvzLqJiIjyS9ElxV9++QVpaWmwtbVFUFAQmjRporFcuXLlMH36dHTv3h1eXl5ITk7GL7/8gvnz5ys5PRVBVg4OMDE1xQ5/fzwJDzdo3eXq1kUf9p4SEdEbSFHCdejQIUiShE8++URrsqWqcePGmDp1Kr788kv8/fffTLhKsCfh4XgYFlbYYRARERUIRZcUsy8LdurUSe9jOnfurHYsERERUXGnKOHK+L/xMnmZ6iG7LCdNJSIiopJCUcLl4uICADhz5ozex2SXdXZ2VnJqIiIioiJDUcLVoUMHCCEwZ84cxMTE5Fo+KioKc+bMgSRJ6Nixo5JTExERERUZihKuCRMmwMTEBE+ePEHLli2xbds2+TKjqoyMDGzduhWtW7fG48ePYWJigg8//FDJqYmIiIiKDEV3KTZo0ADffPMNZs6ciZiYGAwcOBAODg7w8PBAhQoVIEkSHj58iLCwMDx//lyeff6bb75BgwYNDPICiIiIiN50ipf2mT59Ouzt7TFt2jSkpKQgLi4Ox48fVyuTnWhZW1vjxx9/xPvvv6/0tERERERFhkHWUhw/fjz69++PNWvW6Fy8evjw4XBycjLEKYmIiIiKDIMkXADg5OSETz75BJ988omhqiQiIiIqFhQvXk1EREREujHhIiIiIjIyJlxERERERsaEi4iIiMjImHARERERGRkTLiIiIiIjY8JFREREZGRMuIiIiIiMjAkXERERkZEpmmn+66+/BgC0bNkSXbt2NUhARERERMWNooTryy+/hCRJ2Llzp6HiISIiIip2FF1SLFu2LACgcuXKBgmGiIiIqDhSlHDVrFkTAPDw4UODBENERERUHClKuAYMGAAhBLZu3WqoeIiIiIiKHUUJ1/jx49G4cWOsX78e69atM1RMRERERMWKokHzDx8+xMqVKzFy5EiMGDECGzduxODBg9GoUSOUKVMGpqamOo/n2C8iIiIqCRQlXFWrVoUkSQAAIQSOHj2Ko0eP6nWsJElIT09XcnoiIiKiIkFRwgVkJVqatomIiIgoi6KEa82aNYaKg4iIiKjYUpRwBQQEGCoOIiIiomKLaykSERERGRkTLiIiIiIjUzxoPltmZiZOnDiB4OBgPHz4ECkpKfj2229RqVIluczLly+Rnp4OU1NTWFpaGurURERERG80gyRc+/fvx8SJExEZGan2/JQpU9QSrlWrVuHDDz+Era0tYmJiYGNjY4jTExEREb3RFF9SXLlyJXr16oU7d+5ACIGyZctqnR5i5MiRcHBwQFJSEnbu3Kn01ERERERFgqKEKyIiAh988AEAoGPHjrh27RoeP36stbyFhQX69u0LIQQOHTqk5NRERERERYaihGvhwoV49eoV6tevjwMHDqBOnTq5HuPl5QUAuHjxopJTExERERUZihKuo0ePQpIkTJ48GRYWFnodU6NGDQDAvXv3lJyaiIiIqMhQlHDdv38fANCkSRO9j8keKJ+SkqLk1ERERERFhqKES3Xhan09efIEAGBnZ6fk1ERERERFhqKEy9nZGQBw8+ZNvY8JDAwEAFStWlXJqYmIiIiKDEUJV/v27SGEwKZNm/QqHxsbi99//x2SJKFjx45KTg0AePz4Mfbt24dZs2ahW7ducHJygiRJkCQJw4YNy3N9Bw8eRJ8+feDq6gpLS0u4urqiT58+OHjwoOJYiYiIqORSNPHpmDFjsGrVKhw4cABr1qzB8OHDtZaNiopCnz59EBsbCzMzM4wZM0bJqQEAFSpUUFwHkHVJdNy4cVi+fLna89HR0di5cyd27tyJMWPGYNmyZfJlVCIiIiJ9Kerh8vT0xLhx4yCEwKhRo9CvXz9s3bpV3n/58mX88ccfGDlyJGrXro3z589DkiRMmTIFNWvWVBy8Kjc3N3Tp0iVfx37++edysuXh4YHNmzfj3Llz2Lx5Mzw8PAAAy5cvxxdffGGweImIiKjkULy0z+LFi5GcnIwNGzZgx44d2LFjh9wL5O/vL5fLHlg/bNgwfP/990pPCwCYNWsWPD094enpiQoVKiAyMhLVqlXLUx0RERGYN28eAKB58+YICgpCqVKlAGQllL169YK3tzdCQ0Mxd+5cDB8+XJ7agoiIiEgfipf2MTU1xbp167Bt2zZ4eHhACKHxUa9ePWzatAmrV6822GW5r776Cj179lR0aXHBggVIT08HkJU8Zidb2aytrbF48WIAQHp6OhYuXJjvcxEREVHJZJDFqwGgb9++6Nu3L2JiYhAaGorHjx8jIyMDZcuWhYeHxxvZKySEwO7duwEAderUQatWrTSWa9WqFWrXro0bN25g165dWLRoEcdyERERkd4MlnBlc3Z2Rq9evQxdrVHcuXMH0dHRAABvb2+dZb29vXHjxg1ERUXl69IlERERlVyKLykWZeHh4fJ2butAqu5XPY6IiIgoNwbt4Xr16hUuXLiAq1ev4tmzZwAAR0dHNGjQAE2bNoW5ubkhT6dY9tJEAODq6qqzrJubm8bjiIiIiHJjkIQrKSkJ33zzDVatWoW4uDiNZcqUKYORI0fi888/R+nSpQ1xWsUSExPlbVtbW51ls9eABLJerzZpaWlIS0uTf05ISFAQIRERERUHii8phoeHo379+vjpp5/w7NkzrXcpPnv2DD/99BMaNmyIGzduGCJ2xV68eCFvW1hY6CxraWkpb6empmot98MPP8De3l5+qPaMERERUcmkKOF6/vw5OnXqhPv370MIgQYNGuDHH39EYGAgrl+/jvDwcAQGBsqJlhAC9+7dQ6dOnRAfH2+o15BvVlZW8vbLly91llXttXp96ghV06dPR3x8vPzg5UciIiJSlHDNnTsXDx48gCRJ+Oabb3Dp0iVMmTIFXl5ecHd3R+3ateHl5YWPP/4YFy9exLfffgsAiImJwdy5cw3yApRQvbSp6zIhACQnJ8vbui4/Wlpaws7OTu1BREREJZuihGvXrl2QJAkDBgzAzJkzdc5NJUkSZsyYgQEDBkAIgZ07dyo5tUGoDpSPiorSWVa1p4qXCYmIiCgvFCVcd+/eBQAEBATofcywYcPUji1M9erVk7evX7+us6zq/rp16xotJiIiIip+FCVc2Zfkypcvr/cx2WVzuyuwIFSrVg3Ozs4AgMDAQJ1lg4KCAAAuLi6oWrWqsUMjIiKiYkRRwtWwYUMAwH///af3Mdlls48tTJIkwdfXF0BWD9bZs2c1ljt79qzcw+Xr68tlfYiIiChPFCVcY8eOhRACCxcuRGZmZq7lMzMzsWDBAkiShDFjxig5tcFMnjwZZmZZ05FNmDAhx5QPqampmDBhAgDAzMwMkydPLugQiYiIqIhTNPFpv379cPDgQaxZswa9e/fG8uXLUbFiRY1lHz16hLFjx+Kff/7B8OHDMWDAACWnBgCcOnUKERER8s+xsbHydkREBNauXatWPnv8mCp3d3dMnToVc+bMQWhoKNq2bYtPP/0UNWrUwK1btzB37lyEhYUBAD755BPUqlVLcdxERERUsuiVcK1fv17rPm9vb1y9ehX79u1D9erV0aVLF3h6eqJ8+fKQJAmPHj1CSEgIDh06hLS0NHh6esLb2xvr16/H0KFDFQW/cuVKrFu3TuO+06dP4/Tp02rPaUq4AOC7777D48ePsXr1aoSFhWHgwIE5yowcOVKe1oKIiIgoL/RKuIYNG5bruCVJkvDixQvs3bsXe/fuzbFfCAFJkhAaGorhw4dDkiTFCZehmJiYYNWqVejbty+WL1+OkJAQxMbGwsnJCZ6enhg7diy6detW2GESERFREaX3JUUhhOJy+tahr7Vr1+a4bKhE9+7d0b17d4PVR0RERATomXDduXPH2HEQERERFVt6JVxVqlQxdhxERERExZaiaSGIiIiIKHdMuIiIiIiMjAkXERERkZEZJOEKDw/HRx99hObNm8PR0RHm5uYwNTXV+cie3Z2IiIiouFOc9cyZMwezZs1CRkaGwad9ICIiIioOFCVc27Ztw4wZMwBkTR7q5eWFxo0bw8HBASYmvFpJREREBChMuH755RcAgIuLCw4cOICGDRsaJCgiIiKi4kRRN9Tly5chSRK++eYbJltEREREWihKuMzNzQEATZo0MUQsRERERMWSooTL3d0dAPD06VODBENERERUHClKuAICAiCEwK5duwwUDhEREVHxoyjhGjlyJLy8vLB8+XLs3bvXUDERERERFSuK7lI0NzfH7t27ERAQAD8/PwwcOBD9+/eHu7s7rK2tcz2+cuXKSk5PREREVCQonvjUwcEBEydOxNmzZ7F582Zs3rxZr+MkSUJ6errS0xMRERG98RTPTjp58mR06dIFsbGxEELk6UFERERUEijq4frf//6HRYsWAQDs7Ozg5+eHRo0acaZ5IiIiIhWKEq7FixcDAOrUqYPjx4+jQoUKBgmKiIiIqDhR1A11/fp1SJKEL7/8kskWERERkRYGmWk+ewJUIiIiIspJUcJVp04dAMDDhw8NEgwRERFRcaQo4Ro+fDiEEHpPBUFERERUEimeab5nz5743//+h19//dVQMZGeMjMyCjsEIiIi0oOiuxSDgoIwceJEPHnyBJMmTcKmTZswcOBAvWeab9++vZLTl3gmpqbY4e+PJ+HhBq23Vrdu6Pjddwatk4iIqCRTlHD5+PhAkiT553/++Qf//POPXsdypnnDeBIejodhYQat0+n/xuYRERGRYShe2oczxhMRERHppijhOn78uKHiICIiIiq2FCVc3t7ehoqDiIiIqNjigodERERERsaEi4iIiMjImHARERERGZmiMVwdO3bM97GSJOHo0aNKTk9ERERUJChKuE6cOAFJknRODaE6Txfw/6eReP15IiIiouJKUcLVvn37XBOn5ORk/Pfff4iPj4ckSXB3d0elSpWUnJaIiIioSFHcw6UPIQT279+PSZMm4dmzZ1i5ciXatWun5NRERERERUaBDJqXJAk9e/bEqVOnYGpqCj8/P0RHRxfEqYmIiIgKXYHepVipUiV8/PHHePr0KebNm1eQpyYiIiIqNAU+LUT2pcT9+/cX9KmJiIiICkWBJ1wWFhYAgJiYmII+NREREVGhKPCE69SpUwAAa2vrgj41Ub5lZmSwbiIiyjdFdynmVXBwML7++mtIkoQWLVoU5KmJFDExNcUOf388CQ83aL21unVDx+++M0rd5erWRZ+NGw1aJxER5Y+ihOvrr7/OtUxmZibi4uIQGhqKf/75B5mZmZAkCR999JGSUxMVuCfh4XgYFmbQOp3q1DFa3URE9OZQlHB9+eWXeZoxXggBMzMzzJs3D507d1ZyaiIiIqIiQ/ElRV3L+gBZc3CVLl0a1apVg7e3N8aMGYN69eopPS0RERFRkaEo4crMzDRUHERERETFVoHfpUhERERU0jDhIiIiIjIyJlxERERERmbQebgSExNx584dJCYmIkOPCRfbt29vyNMTERERvZEMknCtWLECv/32Gy5fvqz3MZIkIT093RCnJyIiInqjKUq4MjIy0LdvX+zduxdA7lNEEBEREZVEihKuZcuWYc+ePQCAChUqYPjw4WjWrBkcHR1hYsLhYURERESAwoRr/fr1AIB69erh5MmTKFOmjEGCIiIiIipOFHVDhYeHQ5IkfPHFF0y2iIiIiLQwyHW/2rVrG6IaIiIiomJJUcJVq1YtAMCzZ88MEgwRERFRcaQo4Ro4cCCEENi3b5+h4iEiIiIqdhQlXBMnTkSjRo2wdOlSnDx50lAxERERFYpMPSbtfhPrpjeforsULS0tcejQIfTp0wedO3fGxIkTMXjwYNSpUwdWVlaGipGIiKhAmJiaYoe/P56Ehxu03nJ166LPxo0GrZOKFkUJl6mpqbwthMD8+fMxf/58vY7lTPNERPQmehIejodhYYUdBhUzihKu12eW50zzRERERDkpSrhmz55tqDiIiIiIii0mXERERERGxgUPiYiIiIyMCRcRERGRkTHhUnHv3j1MnToVdevWhY2NDRwdHdGiRQv89NNPSElJKezwiIiIqIhSNIarONm/fz/8/f0RHx8vP5eSkoKQkBCEhIRg5cqVOHDgAKpXr16IUZIuthUrIjMjAyYq05VQ0WSs99GYnw/WXbB1U9FX0j57TLgAXLp0Cf3790dKSgpsbW0xffp0dOjQAampqdiyZQtWrFiBGzduoEePHggJCYGtrW1hh0waWDk4GGXSwlrduqHjd98ZrD7KnTHfR2NMasm6c+JEn5SbkjbJLBMuAJMnT0ZKSgrMzMxw6NAhtG7dWt7XsWNH1KpVC9OmTcP169fx888/Y9asWYUYLeXG0JMWOtWpY7C6SH/Geh+NMakl6ybKn5L02SvxY7hCQkJw4sQJAMDIkSPVkq1sU6ZMQd26dQEACxcuxKtXrwoyRCIiIiriSnzCtWvXLnl7+PDhGsuYmJhg6NChAIC4uDg5QSMiIiLSR4lPuE6ePAkAsLGxQbNmzbSW8/b2lrdPnTpl9LiIiIio+CjxCVf4/w3Wq1mzJszMtA9pq6MyjifcwAP8iIiIqHgr0QnXixcvEBsbCwBwdXXVWbZMmTKwsbEBANy/f9/osREREVHxUaLvUkxMTJS39ZnqwcbGBsnJyUhKStJaJi0tDWlpafLP2fN6JSQkKIhUR0w1a8I+I8OgdVo4OyMhIYF1G7leY9dtU7Om0T53BYHvY9Guuyh//tgeBaeotXV2vUKIvB8sSrB79+4JAAKAGDJkSK7l3dzcBABRo0YNrWVmz54t18kHH3zwwQcffBS/x/379/Occ5ToHi4rKyt5++XLl7mWz+65KlWqlNYy06dPx8cffyz/nJmZiWfPnqFs2bKQJEmvuBISEuDm5ob79+/Dzs5Or2PIMNj2hYdtX3jY9oWHbV948tP2QggkJibC2dk5z+cr0QlX6dKl5W1dlwmzJScnA9B9+dHS0hKWlpZqzzk4OOQrPjs7O/4CFhK2feFh2xcetn3hYdsXnry2vb29fb7OU6IHzVtZWcHJyQkAEBUVpbNsXFycnHC5ubkZPTYiIiIqPkp0wgVAnkE+IiIC6enpWstdv349xzFERERE+ijxCVe7du0AZF0uPH/+vNZygYGB8nbbtm2NGpOlpSVmz56d49IkGR/bvvCw7QsP277wsO0LT0G3vSREfu5tLD7OnTuHli1bAgDGjh2LZcuW5SiTmZmJBg0aIDw8HA4ODnj8+DHMzc0LOlQiIiIqokp8D1eLFi3g5eUFAFi1ahWCg4NzlJk/f748u/ykSZOYbBEREVGelPgeLgAICwtD27ZtkZqaCltbW8yYMQMdOnRAamoqtmzZguXLlwMA3N3dERoaqnZ3IxEREVFumHD9n7179+K9997TOjutu7s79u/fj5o1axZwZERERFTUlfhLitneeecdXL58GR999BHc3d1hbW0NBwcHNG/eHHPnzkVYWJjRk6179+5h6tSpqFu3LmxsbODo6IgWLVrgp59+QkpKilHPXdw8fvwY+/btw6xZs9CtWzc4OTlBkiRIkoRhw4blub6DBw+iT58+cHV1haWlJVxdXdGnTx8cPHjQ8MEXcRcuXMD333+Pbt26wc3NDZaWlrC1tYW7uzuGDRuGkydP5qk+tr1+EhISsGXLFkyZMgXe3t6oWbMm7O3tYWFhgfLly8PHxwfz5s3D06dP9aqP7W4Y06ZNk797JEnCiRMncj2GbZ83qu2r6+Hj45NrXUZt+zzPTU9GsW/fPmFvb691GYHatWuLW7duFXaYRYa2dgQgAgIC9K4nMzNTjBkzRmd9Y8aMEZmZmcZ7MUVI+/bt9VoWY8iQISItLU1nXWz7vDl8+LBebe/k5CQOHjyotR62u+FcvHhRmJmZqbXd8ePHtZZn2+ePPp97AMLb21trHQXR9ky43gAXL14U1tbWAoCwtbUV3333nThz5ow4evSoGD16tPxm16lTRyQmJhZ2uEWC6i+Jm5ub6NKlS74SrhkzZsjHeXh4iM2bN4tz586JzZs3Cw8PD3nfzJkzjfdiipAaNWoIAMLZ2VlMmjRJbN++XZw7d04EBweLn3/+Wbi4uMhtNmjQIJ11se3z5vDhw8LNzU0MHTpU/PLLL2LHjh0iODhYnD59Wvzxxx+iX79+wtTUVAAQFhYW4tKlSxrrYbsbRkZGhvD09BQARPny5fVKuNj2+ZPdLu+//764cuWK1sft27e11lEQbc+E6w3g4+MjAAgzMzNx5syZHPvnzZsnv9lfffVVIURY9MyaNUvs3btXPHz4UAghxJ07d/KccP3333/y/06bN28uUlJS1PYnJyeL5s2by+9dRESEoV9GkdOjRw/xxx9/iPT0dI37nzx5Itzd3eX3IigoSGM5tn3eaWtzVTt37pTbvk+fPjn2s90NZ8GCBfJ/lKdPn55rwsW2z7/stp09e3a+ji+otmfCVcjOnTsnf1jGjh2rsUxGRoaoW7euACDKlCkjXr58WcBRFn35SbjGjx8vHxMcHKyxTHBwsFzmww8/NGDExdfevXvlNps4caLGMmx746lTp458afF1bHfDuHfvnrC1tZUTrNmzZ+eacLHt809pwlVQbc9B84Vs165d8vbw4cM1ljExMcHQoUMBZK3pqM+gS1JGCIHdu3cDAOrUqYNWrVppLNeqVSvUrl0bQNZ7KXjTb65UB67eunUrx362vXHZ2NgAAF68eKH2PNvdcMaPH4+kpCQEBAToNVCbbV94CrLtmXAVsuw7tmxsbNCsWTOt5by9veXtU6dOGT2uku7OnTuIjo4GoN72mmTvj4qKQmRkpLFDK/Jevnwpb5uY5PwKYtsbT3h4OC5evAgg64+LKra7YWzduhX79u2Do6MjfvzxR72OYdsXnoJseyZchSx7BvuaNWvCzMxMaznVL8fsY8h4VNv49T9Mr+N7kzeq65Jqalu2vWGlpKTgv//+w88//4wOHTogIyMDQNaqGarY7so9f/5cbte5c+eiXLlyeh3HtjeMbdu2oXbt2ihVqhRKly6NWrVqISAgAMePH9d6TEG2vfa/8GR0L168QGxsLADA1dVVZ9kyZcrAxsYGycnJuH//fkGEV6KptnFu742bm5vG4yinzMxMzJkzR/65f//+Ocqw7ZVbu3at1iEKADB16lT4+/urPcd2V27atGl4+PAh2rRpg5EjR+p9HNveMK5du6b2c0REBCIiIrB+/Xr07t0ba9euhb29vVqZgmx7JlyFKDExUd62tbXNtXx2wpWUlGTMsAh5e2+yx8QA4HuTiwULFuDcuXMAAD8/PzRv3jxHGba98TRp0gTLli1Dy5Ytc+xjuytz6tQprFy5EmZmZli2bBkkSdL7WLa9MtbW1ujVqxfeeust1KlTB7a2tnjy5AkCAwOxbNkyPH36FLt27YKvry8OHz6sth5yQbY9E65CpDpo1cLCItfylpaWAIDU1FSjxURZ8vLeZL8vAN8bXQIDA/HZZ58BAMqXL4+lS5dqLMe2V653795yMpuamopbt25h69at2LlzJ/z9/bFw4UL07NlT7Ri2e/69fPkSY8aMgRACH330ERo2bJin49n2ykRHR8PBwSHH8507d8aECRPQrVs3hIWFITAwEEuXLsXEiRPlMgXZ9hzDVYisrKzkbdWBxNqkpaUBAEqVKmW0mChLXt6b7PcF4Hujzb///gs/Pz+kp6fD0tISW7duRYUKFTSWZdsr5+DggAYNGqBBgwbw9PTEwIEDsWPHDqxfvx63b9+Gr68v1q5dq3YM2z3/vv/+e4SHh6Ny5cqYPXt2no9n2yujKdnKVqFCBWzfvl1OphYvXqy2vyDbnglXISpdurS8rU/3ZHJyMgD9Lj+SMnl5b7LfF4DvjSZ37txBly5dEBcXB1NTU2zevFnn3UBse+MZMmQI+vXrh8zMTHz44YeIi4uT97Hd8+f69ev44YcfAGT9MVe97KQvtr1xVa9eHZ07dwaQNa4rJiZG3leQbc9LioXIysoKTk5OiI2NRVRUlM6ycXFx8putOnCPjEN18GRu743q4Em+N+piYmLQqVMnxMTEQJIkrF69Gn5+fjqPYdsbl6+vL7Zu3Yrk5GT89ddfGDx4MAC2e34tWLAAL1++RPXq1ZGSkoItW7bkKHP16lV5+9ixY3j48CEA4J133oGNjQ3bvgDUq1cP+/fvB5B1CdLZ2RlAwX7umXAVsrp16+LkyZOIiIhAenq61qkhrl+/rnYMGVe9evXkbdW214TvjWaxsbHo3Lkzbt++DSDrf//ZE/jqwrY3LtWpCu7evStvs93zJ/sy0+3btzFo0KBcy3/zzTfy9p07d2BjY8O2LwDaJiotyLbnJcVC1q5dOwBZXZXnz5/XWk517qK2bdsaPa6Srlq1avL/gFTbXpOgoCAAgIuLC6pWrWrs0IqE+Ph4dO3aVb5Ne86cOfjggw/0OpZtb1zZkzwC6pdF2O6Fh21vfKpTRmS3NVCwbc+Eq5D17t1b3l6zZo3GMpmZmVi/fj2ArMGBHTp0KIjQSjRJkuDr6wsg6381Z8+e1Vju7Nmz8v96fH1983QreHGVkpKCHj164MKFCwCAmTNn4tNPP9X7eLa9cW3btk3eVr2bju2eP2vXroXIWpdY60N1IP3x48fl57P/aLPtjev27ds4fPgwgKzxXC4uLvK+Am37fK3ASAbl5eUlr0J+5syZHPvnzZuneHHOki4/i1ffuHFD5wryKSkpaivI37x50wiRFy1paWmiS5cucltPmjQpX/Ww7fNuzZo1IjU1VWeZn3/+WX5vqlatKl69eqW2n+1uHPosXs22z589e/bk+ByrevjwofDw8JDbf/78+TnKFFTbM+F6A1y4cEGUKlVKABC2trbi+++/F8HBweLYsWNizJgx8gfF3d1dJCQkFHa4RcLJkyfFmjVr5MePP/4ot2Pbtm3V9q1Zs0ZrPZ999pl8nIeHh9iyZYsICQkRW7ZsUfslnj59esG9uDdYnz595Dbp2LGjuHz5srhy5YrWx40bN7TWxbbPmypVqghHR0cxevRosW7dOnHq1Clx8eJFcfLkSfHbb7+Jtm3bym1mYWEhDh8+rLEetrvh6ZNwCcG2z48qVaoIZ2dnMWHCBLFp0yZx5swZERYWJg4fPixmzpwpypYtK7dbu3btxIsXLzTWUxBtz4TrDbFnzx5hZ2cnv6mvP9zd3cV///1X2GEWGQEBAVrbUtNDm4yMDDFixAidx44cOVJkZGQU4Kt7c+WlzQGIKlWqaK2LbZ83VapU0avNXV1dxaFDh7TWw3Y3PH0TLrZ93un7ue/bt6+Ii4vTWk9BtD0TrjdIZGSk+Oijj4S7u7uwtrYWDg4Oonnz5mLu3LkiOTm5sMMrUgyVcGXbv3+/8PX1Fc7OzsLCwkI4OzsLX19fceDAgQJ4NUWHIROubGx7/URERIhly5aJAQMGiEaNGokKFSoIMzMzYWtrK2rUqCH69u0r1qxZo/d3CdvdcPRNuLKx7fV34sQJ8dVXX4m3335buLu7C0dHR2FmZiYcHBxEw4YNxdixYzUO1dHGmG0vCaHlXkkiIiIiMgjepUhERERkZEy4iIiIiIyMCRcRERGRkTHhIiIiIjIyJlxERERERsaEi4iIiMjImHARERERGRkTLiIiIiIjY8JFREREZGRMuIiIiIiMjAkXERERkZEx4SKiImnt2rWQJAmSJCEyMrKwwyEi0okJFxEREZGRMeEiIiIiMjImXERERERGxoSLiIiIyMiYcBEREREZGRMuInojxcXF4bPPPkOdOnVQqlQplC9fHp06dcK2bdv0riM9PR2rVq1C9+7d4ezsDEtLSzg5OaF9+/ZYuHAhXrx4kWsdly9fxpAhQ+Di4gIrKytUrlwZ7733Hi5cuAAAGDZsGCRJQtWqVXMcGxkZKd9JuXbtWgDAjh075HjMzMzg4+OT47ioqChMnz4dTZs2RZkyZeTzDhgwAMePH9frtcfFxeHbb79F69at4eTkBEtLSzg7O8PX1xc7duzQqw4iMiBBRPSG+ffff0WlSpUEAI2PESNGiDVr1sg/37lzJ0cdERERol69elrrACBq1aolbt68qTWOtWvXCnNzc43Hmpubi7Vr14qAgAABQFSpUiXH8Xfu3JHLr169WgwZMiRHPd7e3mrHrFy5UpQqVUpn3CNHjhSvXr3SGvf+/fuFg4ODzjp69OghEhMT9X1LiEghJlxE9EZ5/vy5cHV1lRODAQMGiAMHDojQ0FCxadMm0bx5cwFAeHp6ak24YmJiRIUKFQQAUbp0aTFlyhTx119/iQsXLojjx4+L6dOnC2trawFAVK9eXTx//jxHHCdPnhQmJiYCgChVqpSYMWOGCAoKEv/8849YsmSJcHV1FRYWFsLDw0OvhKtRo0YCgPDy8hKbNm0SoaGh4siRI2LlypVy+VWrVsnlGzRoIBYvXixOnTolLly4IP7880/RvXt3ef/HH3+ssf0OHTokTE1NBQBRtWpVMXfuXHHixAlx4cIFsXfvXvHee+/JdfTp00fRe0VE+mPCRURvlI8//lhOCL7//vsc+1++fCm6dOmi1lvzesLVs2dPAUC4ubmJW7duaTzPhQsXhI2NjQAgPv/88xz7GzduLAAICwsLcfr06Rz7Hz16JKpXry7HkFvCBUAMHTpUZGZmaozn3r17chIYEBCgtQdrxowZAoAwMTERN27cUNuXlJQkJ5pdunQRycnJGutYvny5HNORI0c0liEiw2LCRURvjBcvXogyZcrIPUIZGRkay92/f1/tUp9qwnXlyhX5+d27d+s837Rp0wQA4ezsrPZ8cHCwXMdHH32k9fjdu3frnXA5ODiIhIQErXVNmTJFjiU1NVVruVevXgkXFxcBQMycOVNt3+LFiwUAYWVlJR49eqS1DiGEaNGihQAg/P39dZYjIsPgoHkiemOcP38ecXFxAICAgACYmGj+inJ1dUWXLl007tu9ezcAwNraGj169NB5vvbt2wMAYmJicP/+ffn5o0ePytsBAQFaj+/RowfKli2r8xzZ3nnnHZQuXVrr/uy433nnHVhZWWktZ2ZmhtatWwMAgoODNdbh7e2N8uXL64wn+7W/XgcRGYdZYQdARJTtypUr8ranp6fOsi1atMD+/ftzPB8aGgoASElJgZmZ/l9xDx8+hJubGwDg6tWrAABLS0s0aNBA6zGmpqZo0qSJWoKmTaNGjbTui4+PR0REBADg999/x++//653zKqyX/vff/8NSZLyVQcRGQd7uIjojZHduwUg1x6aChUqaHz+8ePH+Tp3SkpKjjgcHR1hamqq87hy5crpVX+ZMmW07jNEzK9evcLz588V1UFExsMeLiJ6Ywgh5O3cemhUy6rKyMgAAFSrVg179uzR+9zVqlXTu2x+6ErcsmMGgMmTJ2PkyJF61WlhYaGxjv79++OLL77IR5REZCxMuIjojeHo6ChvP3r0CO7u7lrLausVyh5T9ejRI9SpUydPlxWzZfdGPXv2DBkZGTqTpSdPnuS5/tepjgNLSUnReRlTGysrK1hbWyMlJQXPnz/PVx1EZDy8pEhEb4yGDRvK2yEhITrLatvv4eEBICtxOX36dL7iqF+/PgAgLS1NbVzZ6zIyMnDx4sV8nUNVuXLl4OLiAgA4cuSI1t673GS/9tOnT/NSIdEbhgkXEb0xmjVrJvcubdiwQWviER0djUOHDmnc5+vrK2/PmzcvX3G89dZb8vb69eu1ltu/fz+ePn2ar3O8rlevXgCA27dvY/v27YrqSE5OxpIlSwwSFxEZBhMuInpjWFpaYvjw4QCAixcv4scff8xRJj09HaNHj8bLly811uHp6SlPGXHgwAHMnj1b5zkjIyOxefNmtedat24t31W4ZMkSnDlzJsdxT548wUcffZT7i9LTJ598AktLSwDAuHHj5DsOtTlw4AAuX76s9ty4cePg5OQEAPjiiy/w119/6azj9OnTCAoKUhA1EelLEvntuyYiMoL4+Hg0aNAAUVFRAIBBgwZh6NChKF++PG7evImff/4ZISEh8PT0lC8r3rlzR23x6JiYGDRv3hwPHjwAALRs2RIjRoxAw4YNYWVlhadPn+Ly5cs4ePAgjh07ht69e+foVTp16hS8vb2RmZmJUqVK4eOPP8bbb78NS0tLhIaG4ocffsDDhw9Rv359XLx4EVWrVsWdO3fU6oiMjJQH469ZswbDhg3T+drXrl0rJ5wWFhYYMmQIevbsicqVKyM9PR1RUVE4d+4ctm/fjlu3bmHv3r3o2bOnWh1HjhxBt27dkJ6eDhMTE/Tt2xd9+/ZFjRo1AAAPHjzA+fPnsXPnTly+fBmLFy/Ghx9+mId3iIjypTBnXSUi0uTq1auiYsWKWhdeHj58eK6LV0dGRqqtt6jrMXz4cI1x6Fq82szMTKxYsUJekLpOnTo5jledaX7NmjV6vfYtW7YIOzu7XGM2MTERx44d01jH0aNHdbaf6mPdunV6xUVEyvCSIhG9cerXr49///0X06ZNQ61atWBpaQknJyd06NABmzZtwurVq3Oto0qVKvjnn3+wc+dODBw4ENWqVYO1tTXMzc1Rrlw5tGnTBlOmTEFgYCBWrVqlsY6AgACEhobC398fzs7OsLCwgIuLC/r3749Tp05h1KhRSEhIAADY29sb5LUPGDAAkZGRmDNnDnx8fFC+fHmYm5vD2toa1atXxzvvvIOff/4ZkZGR6NChg8Y6OnbsiFu3buHXX3/F22+/jUqVKsHCwgJWVlZwc3NDly5d8N133+H69esYOnSoQeImIt14SZGISIGaNWvi1q1beO+997Bhw4bCDoeI3lDs4SIiyqeQkBDcunULANCqVatCjoaI3mRMuIiItMhe31CTp0+fYvTo0QCy7q4cMGBAQYVFREUQZ5onItKic+fOqFatGvz8/NCoUSPY29sjLi4Op0+fxm+//SbfBfn555/L0zEQEWnCMVxERFpUrVoVd+/e1Vlm/PjxWLx4MUxMeMGAiLRjwkVEpEVgYCD27t2LwMBAPHjwALGxsTAzM0PFihXRrl07jBkzBm3atCnsMImoCGDCRURERGRk7AMnIiIiMjImXERERERGxoSLiIiIyMiYcBEREREZGRMuIiIiIiNjwkVERERkZEy4iIiIiIyMCRcRERGRkTHhIiIiIjKy/wdKQHdJ5+c4ugAAAABJRU5ErkJggg==",
      "text/plain": [
       "<Figure size 640x480 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "import csv\n",
    "import matplotlib.pyplot as plt\n",
    "\n",
    "max_degree = []\n",
    "\n",
    "with open('./results/max_in_degree_processes.csv', newline='') as csvfile:\n",
    "    reader = csv.reader(csvfile)\n",
    "    next(reader)\n",
    "    for row in reader:\n",
    "        try:\n",
    "            degree = float(row[1])\n",
    "            max_degree.append(degree)\n",
    "        except ValueError:\n",
    "            print(f\"Skipping row with non-numeric value in the specified column: {row}\")\n",
    "\n",
    "plt.rc('font', size=20)\n",
    "plt.hist(max_degree, bins=20, color='maroon', edgecolor='white')\n",
    "plt.title('Max logical process degree')\n",
    "plt.xlabel('degree')\n",
    "plt.ylabel('number of workflows')\n",
    "plt.tight_layout()\n",
    "plt.savefig(f'{directory_plots}/max_process_degree.png', format=\"png\", bbox_inches='tight')\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAj0AAAG4CAYAAACuBFb3AAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjcuMiwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy8pXeV/AAAACXBIWXMAAA9hAAAPYQGoP6dpAABhDklEQVR4nO3deXxM1/8/8NfNLrKLLYTYxb4ksUsoaS21rw0iFK2i9hataovSarWlqkptLS2KNmhrD7Hvy0dU7SQiiZBFyDbn90e+ub8ZyUwmuXeymNfz8ZiHMffcc973zp2Zd8699xxJCCFARERE9JKzKOoAiIiIiAoDkx4iIiIyC0x6iIiIyCww6SEiIiKzwKSHiIiIzAKTHiIiIjILTHqIiIjILDDpISIiIrPApIeIiIjMApMeeikMHz4ckiRBkiSsWbOmqMNRJCAgQN6WgwcPFnU4+VbS4yfz4eXlJR+rt2/fLupwqBAw6SnhBg0aJH9oJUnCV199VdQhERERFUtMekqwhIQE/PHHHzqvrV27toiiISIiKt6Y9JRgmzZtwvPnz3Veu3jxIs6fP180ARERERVjVkUdABWcdq+OnZ2dnACtXbsWTZo0KaKoSCleB0NEZBrs6Smhbty4gSNHjgAALCwssHDhQnnZhg0bkJGRUVShERERFUtMekoo7V6egIAAvPXWW3BzcwMAxMTE4K+//iqq0IiIiIolJj0lkBAC69evl/8/dOhQ2NjYYMCAAfJr+i5ojomJgbW1NSRJgqWlJSIjI41ut0aNGvJdYqGhoXrLXb58GW+//TZq1aoFe3t7uLu7w8fHB5999hni4uIAAGvWrJHrGj58uNExqCU5ORnffvstXn31VVSuXBl2dnZwdXVFgwYNMG7cOJw4cSJf9aWnp2P58uUICAhAuXLlUKpUKVSvXh0DBw7Enj175HLG3CJbkFu+//rrL4wZMwYNGjRAmTJlYG1tDRcXFzRr1gxjxozBn3/+abD378yZM/jss8/QvXt3VK9eHQ4ODrCxsUH58uXRunVrzJo1C3fv3s3PLlGN9t2J2vGOHj0aderUgYODA1xdXeHj44N58+YhISEhzzpz28cPHjzA/Pnz4efnhwoVKsDS0hIuLi65rn/37l3Mnj0bLVu2RPny5eV91bJlS3z00Ue4d+9evrfz5s2bmDNnDtq3b49KlSrBzs4O9vb2qF69Onr16oUlS5YgJiYmz3qePn2K77//Hq+//jqqVq0Ke3t7ODo6olatWhgxYgT2799vdEz79+/HyJEj0bBhQ7i4uMDa2hru7u7w9vZGQEAAZs2ahUOHDiE9PV1vHffu3cPHH3+M9u3bo3z58rC1tYWjoyOqVasGPz8/hISEYOPGjfJ3gxoyMzOxcuVKdOzY0eDnMb/U3LcAcOfOHUydOhX16tWTj+NGjRrpfN4OHjwoH6sBAQG51qOvzK5duzB48GDUqlULDg4OkCQJX3/9da51REREYObMmfDz85OP6bJly6JFixaYPXs2oqKi8rVt6enpWL9+PQYMGIDq1avD0dERpUuXRrVq1TB48GBs27YNQoh81akKQSXOgQMHBAABQJQqVUokJiYKIYQ4cuSI/LqNjY149OhRruu/9tprcrkvvvjCqDa16y5TpoxIS0vLtdzChQuFtbW1XPbFR8WKFcWRI0fE6tWr5deCg4MLtB+0BQcHy/WtXr3aYNnQ0FBRoUIFvTFmP9544w3x9OnTPNu+ffu2aNSokcG6Ro0aJdLS0kTVqlXl127dupVrff7+/nKZAwcOGGz78uXLwsfHJ89tASAGDhyYax2+vr5GrW9tbS0WLlyY5/7IT/zG0I5BCCE+/vhjYWFhYfAYCwsLy1eM27dvF66urjnqcnZ2zrHuvHnzhJ2dncF9ZWdnJz777DOjtu/58+finXfeEVZWVka9B9mf99xs2rTJqGO7e/fu4smTJ3rrSUpKEj169DDquAAgfvzxx1zrWb58uShVqpRRdbRp08ao/ZWX+/fvi+bNmxts68033xSpqalGfR6zqbVvs61bt06ULl1abz1OTk5i+/btOt/3/v7+udb1YpknT56I3r1751rv4sWLddZ9/vy5eOutt4SlpaXB7SpVqpRYsmSJUe/BgQMHRI0aNfLcVy1bthT37983qk618ELmEki7F6dXr15wdHQEALRu3Ro1a9bE9evXkZaWhl9//RVjx47Nsf6QIUPw999/AwB++eUXTJ06Nc82f/nlF/n5wIEDYW1tnaPMV199hffee0/+v52dHTp06IDKlSsjLi4OBw4cwIMHD9C9e3dMnDjR6O1V02+//YagoCBkZmYCACwtLdG2bVvUrFkTycnJOHz4sPwXzYYNG3Dr1i3s378fdnZ2udYXGxuLjh074ubNm/JrderUga+vL6ysrHDp0iWcOXMGP/74o/w+qeXgwYPo0aMHkpKS5NeqVKkCPz8/uLm54enTp/j3339x4cIFpKen57jTL1v2X5S2traoX78+atasCWdnZwgh8ODBA5w4cQJxcXFIT0+X39/p06erui3G+vbbb/HRRx8ByOp5bNGiBWxtbfG///0PJ0+eBJDVY9O1a1fs378ffn5+edZ59OhRzJkzB+np6ShTpgzat28Pd3d3xMTE4Ny5czplx40bh++++07+f+nSpdGxY0dUqFAB0dHROHDgAJKTk/H8+XPMmDEDDx8+xOLFi/W2nZycjMDAQBw7dkx+zd7eHm3btkXlypUhhEBkZCTOnDmDR48eIT09XT52X7R48WJMmTJF/uvZ0dERrVq1gqenJzIzM3HlyhWcOnUKQgjs2LED/v7+OHr0KOzt7XPUNXToUPz555/y/2vWrImmTZvCzc0N6enpiI2NxaVLlwwO6Ld9+3a89dZb8v+dnJzQqlUrVK5cGVZWVkhISMC1a9dw+fJlpKWl6a0nP+Lj4/HKK6/g33//lV+rVasW/Pz8YG1tjfPnz+P8+fNYuXIlHBwcjK5XzX0LAJs3b8bw4cOh0WgAAFZWVmjfvj2qV6+OxMREhIWF4eHDhxgwYADmz5+fr30ghMCQIUOwY8cOSJIEX19feHt7QwiBy5cv6/SYPn36FK+++qp8fSgAVKtWDT4+PnB1dcXjx49x9OhRREZG4tmzZxg/fjwSExMxc+ZMve1v3rwZQUFBcu+fnZ0dWrZsCS8vL1haWuLatWs4duwYMjIycPz4cbRq1QqnTp1C+fLl87WdBVaoKRYp9vTpU+Ho6Chnyjt37tRZPmfOHHlZixYt9Nbh4OAgl7t8+bLBNtPS0oS7u7tc/ujRoznKXL58WdjY2MhlunXrJmJiYnK0+9ZbbwkAwtbWVi5bWD09169f19luHx8fce3aNZ0ymZmZ4ssvv9TpSRg/frzedgcPHiyXs7e3F5s3b85R5vDhw8LDw0NIkqSzj5T09Ny9e1fnPalWrZr4+++/cy0bHx8vli9fLqZOnZrr8rffflvs3LlTpKSk5Lo8IyNDrF69Wv6r1NraWty8eTPXssbGnx/ZdQFZPZh2dnZi/fr1OcodP35c5y/3OnXqiGfPnuUZo5WVlZAkSXz66ac5ejCfP38uP//tt990Yhk2bJhISEjQKZ+QkCCGDBmiU27Lli16t23gwIFyOUtLS/Hxxx+L5OTkHOUyMzPF/v37Rc+ePXPtRdi7d698zFpbW4u5c+fmWs+5c+dEvXr15DbffvvtXMtkL3dwcBC7du3SG/+NGzfE3LlzxZ9//pljWePGjeV6xo0bp7fXNCkpSWzatEm89957etsxlvb3gJ2dnfj5559zlNmzZ4/82dHuldb3eVRz3wohRHR0tE6vYosWLXJ8ntLT08XHH3+c47vSmJ6e7B7Dhg0biosXL+Yoq31MDxs2TF6vRo0aYs+ePTnKZ2RkiGXLlslxWFpa5vobIETW74C9vb1c58SJE3M943Djxg3Rtm1buVyXLl1yrc8UmPSUMOvXr5cPlHLlyon09HSd5Tdu3ND5wo2IiMi1Hu0v5hkzZhhsMzQ0VOeDkZt+/frpJBOpqal66+vfv79OjIWV9Gh/wKtXry4eP36st76vvvpKLmthYZHrj/zly5d1tuP333/XW9/58+dznL5QkvQEBQXJZapWrSqio6P1tq2WX3/9VW5z+vTpesuZMukBIH799Ve9Zf/991+dL93vvvsuzxgBiLlz5xqMITMzU1SrVk0u37dvX6HRaHItq9FoRM+ePXU+M5mZmTnK7dmzRyeGjRs3GozBUGy1atWS68nth17bgwcPRLly5eQf8Xv37uksX7JkiVzXrFmzChRTUlKSXIenp6fefaWmq1ev6uxPQ/vhyJEjOU6R5vZ5VHvfCiHE1KlT5fqqVKli8Hto2rRpOjEak/QAEBUqVBCxsbEGYz106JBcvnLlyuLBgwcGy//0009y+ddeey3XMh07djT6M5WcnCy8vb3l8sePHzdYXi1MekqYTp06yQfJu+++m2uZNm3a5JnQ/P333zo/moa+lLT/Gp09e3aO5Y8ePdL5i2n//v0Gt+Hu3bs6XziFkfQ8fvxY5y8mQwmKEFlfdvXr15fLv//++znKTJkyRV7evn37PGMcNWqUKknP/fv3dRKov/76K8+21ZCRkSH3lDVr1kxvOVMmPcbs51mzZsnlmzRpkmeMlSpVyvHHw4v++usvubyNjY2IiooyWP7+/fs6n4nceuG0r63Td72VMbZv3y7X88orrxi1zmeffSav8+WXX+osmzdvnrzs66+/LlBMkZGReb4HatNOEFq2bJln+aFDh+b5eVR732ZmZoqyZcvKy3/66SeDdSUnJwsXF5d8Jz3Lli3LM85evXrJ5XPrOX2RRqMRdevWFQCEJEk5kqrz58/L9dWuXVtkZGTkWefGjRvldQz1qKuJd2+VIPfv39e5O2DIkCG5lhs6dKj8fP369fJ5Y22dOnVChQoVAGTdQaB9TldbUlKSzrn93No8evSofP62YsWKeu8wyObp6Yn27dsbLKO2o0ePIjU1FQBQpkwZ9OjRw2B5CwsLjBgxQv7/gQMHcpTRvrPqjTfeyDOGoKAgI6M1bO/evfKdWLVq1cJrr72mSr0AcOnSJaxduxYff/wxpkyZgnHjxsmPd999V74e4NKlS7keV6Y2bNiwPMsEBwfLzy9cuIDHjx8bLN+3b19YWRm+vFH7c9elSxdUrFjRYPlKlSrpvC8vHj+pqak6x8/48eMN1mfIrl275OeDBg0yap2OHTvKz8PDw3WWValSRX6+du1aPH36NN8xlS1bFqVKlQKQdawcOnQo33Xkl/Y+1v4O1MeYY0ntfXvlyhXExsYCAKytrdGvXz+DdZUuXRq9evUyqt1skiRh4MCBBstkZGTId7FZWVmhb9++RtXboUMHAIAQAkePHtVZrr2v+vfvD0tLyzzrNLSvTIUXMpcg2glM3bp14ePjk2u5AQMG4N1330VqaqqcKHXq1EmnjKWlJQYNGiTfvvjzzz+jbdu2OeraunUrnj17BgDw8/NDrVq1cpTRnvbC19dX50I5fXx9fQt15GHtC1JbtGiR548cALRp00ZnfSGEvG0ajQYXL16UlxtzwWzz5s0hSZLi2zSPHz8uP88rwTTW2rVrMX/+fFy7ds2o8unp6UhISICrq6sq7RurZcuWeZapVasWypQpg0ePHkEIgfPnz8tf1rlp3rx5nnVqHz/ax4Uhbdq0kYd2OHv2rM6y8+fPyxeW29vbo0WLFkbVmRvti6B37txp1DQ02rf1v3h7fdeuXeHg4IDk5GScO3cOderUQUhICLp164bmzZvnehPDi6ytrdG7d29s2LABmZmZ6NSpE/r164d+/frB398fZcqUMX4DjSCE0Pk8GrM//fz88vw8qr1vtdf39vY26uYGX19frFmzJs9y2by8vOQx2/S5ePGinMza2dlh2rRpRtV96tQp+fmL26a9r44dO4Zx48blWZ/2vi/IMA8FwaSnBFm3bp383NBfMq6urujevTt+//13AFk/aC8mPUBWr0120rN582YsWbIkxxea9l1b+nqWtMfXqFy5ct4bgqy/hAtT9l9XAFC1alWj1vHy8pKfp6WlISkpCU5OTgCAxMREnbFJjNluBwcHODs748mTJ8YFrcfDhw/l59WrV1dUlxACI0eOxOrVq/O9blJSUqEnPdq9EIZ4enri0aNHAHTf+9yULVs2z/qUHj8vjkGj/R56enoalYTroz1+yvbt2/O9/os9YW5ubli9ejWCgoKQlpaGyMhIzJ07F3PnzkWpUqXg5+cHf39/9OjRw2DCuHjxYpw9exZXr15Feno6Nm7ciI0bN0KSJNStWxft2rVDYGAgunXrpvfuSGMlJCTo3AFmzHHi5OSU5+dR7X1bGN+VxhzP2tuVnJysc0eisV7cNu069+/fn+8xi/LqkVULT2+VECdOnMDVq1cBZHUz5nWqRDsp2rp1q85tzdmaN28Ob29vAFm3er44inN0dLR84FpZWentMk1OTpaf67tF80WlS5c2qpxatGM0tu0Xy2nvQ+36gMLdbu048nPbbW5+/PFHnYSne/fuWL9+PS5fvozHjx8jNTUVIuvaPwghdH7wi+L0VkH2c27Hvrbs0zCGKD1+XoxBzffQmMEYDclt0Mp+/frh9OnT6N+/P2xsbOTXnz17hrCwMHzyySfw8fGBj4+P3lNX5cqVw6lTpzBnzhx4eHjIrwshEBERgRUrVqBfv37w8PDAggUL9N6KbwxTfR7V3reF8V1pzPGsdLuAnNumtE4l739+MOkpIbTH5hFC6Izsm9tD+zxwSkoKtmzZkmu92smTdq8OAGzcuFE+EAMDA1GuXLlc69D+UKakpBi1PQW5TkAJ7R8WY9t+sZx2V/SLX0SFud3acbz4ZZ9fixYtkp/PmzcPoaGhGDJkCOrXrw8XFxedHzwg7wTC1Aqyn9UYH0np8fNiDGq+h9rH4vnz53WSVGMe+sbaadiwITZt2oTY2Fjs2LED06dPR6tWrXR6g8+cOYMOHTpg8+bNudbh4OAgj1B96tQpfPnll+jVqxfc3d3lMo8fP8aMGTPQt2/fAp/6fTFxVOvzqPa+LS7fldpxNGnSJN/bJYTAnDlz9Na5ffv2AtVZGJj0lABpaWn47bffFNWhfWpM25AhQ+TrVEJDQ5GYmCgv+/nnn3XK6aP9BXb//n2j4snP9Bdq0O7yNXY6hTt37sjPbWxsdH6onJ2ddb78jdnu5ORkVf7C0h7E69atWwWu5969e/jvv/8AZJ0SzWvAwcTExELrgtbH2PdO+/oA7eOzoJQePy/GoP0e3rt3T9EEwdp1Zb+fanJyckK3bt2wcOFCHD16FHFxcVizZo18+k6j0WDs2LHytX+5sbCwgI+PDyZPnoxt27bh4cOHCA8P1/nj7I8//pBPyeeXs7OzToJuzHuUmJiY5+dR7X1bXL4rtbfr5s2bqvTamvo4VAuTnhIgNDQU8fHxALIuEGzRooVRD+3z7WFhYbn+RVe1alX5wsxnz55h69atAICrV6/KF186ODigZ8+eeuNr0qSJ/Pz06dNGZezaF8QVhqZNm8rPT548aVRXqvYdbU2bNtW5QNvCwgINGzbUqTMvZ86cUeWvGe2LeXO7q8xY2ufg69Spk+d1JeHh4UUzV44W7Ysl9fnvv//k63kkSdJ57wtKu44X71rRR/v4adasmc6yJk2ayNexpKSk5HuuN23aF+3+888/Ba7HWE5OTggODsb+/ftha2sLIOtaFWPem2wWFhZo06YNtm7dildffVV+XftO0fyQJAmNGjWS/699sb8+J0+ezPN4Vnvfan9XRkREGNXLZ4rvyiZNmsjvXWJiYr7eO30K+zgsKCY9JYD2qa3u3bvj+PHjRj1Onz4t320lXpikVJt2L072KS7tXp4+ffoYPP/cunVrudcjKioqz7uy7t27h8OHDxveaJW1bt1a/pDHxsZi586dBssLIXT2u/atldm075zasGFDnjG8ePqwoDp37iwnKP/991+Bv2AsLP7/x9+Yrvbvv/++QO2oSd8xrE37TpfGjRurcrG19vu/a9euPCf+jI6Olqd6eXF9IGvKD+07ypYuXVrg2Lp37y4/37Bhg1GTkqqhWrVqqF+/vvx/7YuzjSVJErp166aojmza+1P7+0sffZMya1N739avX1/uNUxPT9d7WjDb06dPC3QBdV5KlSqlc0wamirFWNr7at++fbh06ZLiOk1CySA/ZHoxMTE6A9EZGtI+N9rTUtSsWTPXMvHx8fL0CBYWFiIyMlJn9Nndu3fn2U7fvn3l8nmNyDxgwACdgbSKYkTm2rVrG5y4UXtUWn0jMl+6dElnO7Zu3aq3vuI4InNsbKyQJEnexuvXr+stqz0asxrx58eL7RoakfnatWs6IzIvXbpUlRhfHJF58ODBestqNBqdUcdNPSJzRkaGqFmzplxP586d9U4I/KKkpKQcUyrkNZJvtvT0dFGxYkW5Xe0pDBITEw1+B2jTHqE4KCjIqHVyExERobM/1RiRWe19K4TuoKamGJFZX5kXHTx4UKfuvCZq1qZv9OaAgAC5vkaNGuWYpkWf1NRUER8fb3T7SjDpKeYWL14sH0ROTk565xLS57///tM5sMPDw3Mtpz06Z58+feTnFStWNGpkzYsXL+qMQFsc5966ceOGztxbrVq1Ejdu3NApk5mZKb799ludGYcNjRSqncCVLl0617m3wsPDTTL3VpkyZeRyhubeevz4sfjhhx/EtGnTcixr2bKlXEfTpk3F1atXc+yPpUuXCmtra2Fpaakzu3hRJD2G5t46ceKEztxbtWrVMmruLWNjfHHurTfffFMkJSXplElMTBQhISE65Qz9oaKdHGXPvZXbHFXZc2/16tUr17m39uzZo3PMtmzZUpw8eVJvuxcuXBDvv/++cHV1FZcuXdJZNnz4cNG2bVuxZs0avT9EMTExOiMaOzk56czdduDAAVGhQgUxe/ZsvXP7ZWRkiJ9//lnnu+CXX37RG7MxtGPSN/fW3r178zX3lpr7VgghoqKidEZZ1jf31ieffJLju1LNpEcI3e9NSZLE1KlTRVxcXK5lnz9/LrZv3y569eolfH19cy1z6dIlne/YOnXq5DqfV7b//vtPzJ07V3h4eIjQ0FCj41aCSU8x16RJE/kACgkJKVAdfn5+ch2jRo3KtcyWLVty/EUNQEyePNnodj7//HOddUuVKiW6du0qRo0aJfr06SN/0F1cXOTJ9JRslzZjkh4hsnostL/ArKysRIcOHcSbb74pBg8eLCpXrqyzDS1btjSYaD58+FB4eXnprFOnTh0xdOhQERISInx8fHT2pfaP8p07d3Kt09gf5L179+p8wQBZvT79+/cXY8aMEUFBQcLX11f+Yu/Zs2eudWj/xWttbS0CAgLEiBEjxIABA3T+kp83b55O/EWR9HzzzTfy8xo1aoghQ4aIESNGiBYtWuiUs7e3F8eOHdNbZ0FjfOedd3TacXR0FD169BCjRo0SPXv21JkMGMiacNGQhIQEnc9ndvIcGBgoRo4cKUaMGCFeffVVnQRXX8/AihUrdI7t7MRvwIABYsyYMWLYsGEiMDBQnhsq+/HiD7P2Z8nS0lLUq1dP9OvXT4waNUoMGTJEBAQE6CTvAMSKFSt06nhxWoTy5cuL1157TQwfPly8+eabonv37jrHFgDRrl27XHvE8iMuLk6nZyZ7HwwdOlQMHz5cNG3aVOe9MeZ4VnPfZvv111/lXtbs76GOHTuKN998UwwYMECUL19e/jx+8cUXcrkOHTrkWl9Bk57nz5+LwMBAnZhtbGxEmzZtxLBhw8SYMWPEoEGDhK+vr84fPM2bN9dbZ2hoqE5vK5A1/1qfPn3E6NGjxfDhw0W3bt1yfNcy6SFx4cIFnYNi7969Barn22+/letwdnbO9Uf8+fPnOn99ZD/Onj2br7bmz5+f4zSO9qNixYriyJEjYsWKFfJrEyZMKNB2aTM26REi60OZ/aVi6DF48GC9M0Nru3nzpmjQoIHBut58802RlpYmPDw85Nf0/RWdnx/k8+fP68xmbeih79TB999/b/A9s7CwELNnzxYajabIkx4hhPjoo490fjBefFSoUCHPdpXE+Omnn+r89Z3bw87OTsyfP9+o+lJSUsSoUaNy/Kjqq9fQadn9+/frTJCZ16N+/foiMjJSp45x48YZvb6jo2OOhEeIrBnvDR1TLz769etncLvy4+7duzrJTW6PkJAQkZqaanTSo9a+1bZmzZocyYH2w8nJSWzfvl3s3r1bfq1Hjx651lXQpEeIrB63Dz/80GAs2g9ra2vxzjvvGKzz/Pnzonnz5kbvKy8vL3Hu3Ll8xV1QTHqKscmTJ8sHRcWKFQv8V9DDhw91voD0XTvw4oSY3t7eBWrv4sWLYvTo0aJ69erCzs5OuLm5iWbNmol58+bJ1wssWLBAbie3SUzzKz9JjxBZ59u/+eYb0blzZ+Hh4SFsbGyEs7Oz8Pb2FmPHjs33jL9paWli2bJlol27dqJMmTLC1tZWeHl5if79++tcE1WqVCkBZHUl6zttmN8fZI1GI7Zu3SqGDRsmatWqJZycnISlpaVwdXUVzZs3F2+//bbYtWuXwePn4sWLIiQkRHh5ecn7ol69emLcuHE6iW9xSHqEyPpRDQkJETVr1hT29vbC2dlZNGvWTHz66acGr5FQK8bbt2+LDz74QPj6+gp3d3dhZWUl3N3dhZ+fn/jwww/19uIZEhERIWbMmCH8/PxEuXLlhJWVlbC3txc1atQQvXv3FsuWLROPHj3Ks57MzEyxZcsWERISIry9vYWrq6uwtLQUjo6OombNmqJ79+5i/vz5Bn9krly5Ir799lvxxhtviMaNGwtXV1dhZWUl7OzsRKVKlURgYKBYtGiRePjwod464uPjxaZNm8SECRNEu3bthIeHh7C1tRVWVlbCzc1N+Pr6ivHjx4sTJ07ke1/lJT09XaxYsUL4+/vrfB779u0rdu3aJZfLT9IjhDr7VtutW7fEpEmTRJ06deTjuEGDBmLGjBnyMaR9Pd2wYcNyrUdJ0pMtJiZGLFq0SLz22muiSpUqwt7eXlhbW4syZcqIZs2aieDgYLFmzZocly0Y8s8//4i3335bNGrUSP6clC5dWnh5eYnAwEAxe/ZsceTIEYMTXqtNEqKI70ElsxQUFCTf8bRx40ajJ/Mryf777z/Url0bQNYt4tkjbJNh2kMF8OuKzM2sWbMwf/58AMBnn32G999/v4gjKtl4yzoVuqdPn+rMyOvr61uE0RSeTZs2yc/NZZuJqOCEEDqj6fN7QzkmPVToPvjgA3mSP19fX9SoUaNoAyoEd+7c0ZnyYfDgwUUYDRGVBN9++y2uXbsGAKhYsSL8/f2LOKKSj0kPqWbLli2YNm0arl+/nuvyuLg4jB07Vp7ZHQCmTZtWSNGZzsCBA/H7778jNTU11+X//PMP2rZtKyd6DRs2xGuvvVaIERJRcXL06FGMHj0aFy5cyHV5cnIy5syZg8mTJ8uvTZo0Kc9R0ylvvKaHVLNmzRqEhIQAAGrXro2GDRuiTJkySE1Nxa1bt3DixAmdxCAoKMiokVOLOy8vL9y5cweOjo5o1qwZvLy8YGdnh0ePHuHkyZM68wCVLl0a4eHhOsPRk2G8podeNgcPHpRHkK5atSqaNGmCcuXKITMzE/fu3cPRo0d1Jhpt37499u/fD0tLy6IK+aXBtJFM4tq1a3K37IssLS3xzjvv4KuvvirkqEwrKSkJYWFhCAsLy3V59erVsWnTJiY8RCS7c+eOzuS0Lxo0aBBWrVrFhEcl7Okh1aSlpWHPnj34+++/cfbsWTx8+BBxcXFISUmBq6srvLy8EBAQgBEjRqBOnTpFHa5qbt26ha1bt+Lw4cO4efMm4uLi8OjRI9jY2KBs2bJo3rw5unfvjjfeeENnZnYyDnt66GWj0WgQFhaGXbt24eTJk3jw4AHi4uKQlJQEZ2dnVK5cGe3bt8fQoUN58bLKmPSYmEajQVRUFBwdHXW+vImIiEgdQggkJSXBw8NDZzLlF/H0lolFRUXB09OzqMMgIiJ66d27dw+VK1fWu5xJj4k5OjoCyHojnJycijgaIiKil09iYiI8PT3l31x9mPSYWPYpLScnJyY9REREJpTXZSQcp4eIiIjMApMeIiIiMgtMeoiIiMgsMOkhIiIis8Ckh4iIiMwCkx4iIiIyC0x6iIiIyCww6SEiIiKzwKSHiIiIzAKTHiIiIjILJp+GIjU1FeHh4YiLi0O1atXg5+dn6iaJiIiIclCU9Ny5cwffffcdAGDmzJlwcXHRWX78+HH069cPDx48kF9r1qwZfv/9d1SpUkVJ00RERET5ouj01rZt27Bo0SLs378/R8KTlJSEXr164cGDBxBCyI8zZ86gW7duyMjIUNI0ETSZmSWiTiIiKh4U9fTs2bMHkiShV69eOZatWLECMTExkCQJEyZMwCuvvIJ//vkHy5Ytw5UrV7B27VqMHDlSSfNk5iwsLbE1KAixERGq1FfW2xt9fvlFlbqIiKj4UZT03Lx5EwDQvHnzHMs2bdoESZLQu3dvfP311wCA119/HbGxsdi8eTO2bNnCpIcUi42IQPS5c0UdBhERlQCKTm/FxMQAAMqXL6/zemJiIs6ePQsACAkJ0Vk2aNAgAMCFCxeUNE1ERESUL4qSnqSkJABA5gvXQRw5cgSZmZmwtLREQECAzjJPT08AQHx8vJKmiYiIiPJFUdLj7OwMAIiKitJ5/eDBgwCAxo0bo3Tp0rmua2dnp6RpIiIionxRlPQ0aNAAQNZdXNkyMzPl63k6dOiQY53IyEgAOU+JEREREZmSoguZe/fujbCwMKxfvx7ly5dHu3btsH79ety5cweSJGHAgAE51jl9+jQAcJweIiIiKlSKenrGjBkDb29vCCGwaNEi9OzZE1u2bAGQdaeWj49PjnW2bdsGSZJyXOtDREREZEqKkh5bW1vs27cPffr0gZWVFYQQsLa2xtChQ7F+/foc5Q8dOoQrV64AADp37qykaSIiIqJ8UTz3VoUKFbBlyxakpqYiPj4eZcqUgY2NTa5lPT09ceDAAQCAr6+v0qaJiIiIjKbahKO2traoWLGiwTLVqlVDtWrV1GqSiIiIyGiKTm8RERERlRSKenoqVKgAf39/+VG/fn214iIiIiJSlaKkJyYmBlu2bJHv2CpTpgzat2+PgIAA+Pv7o2HDhqoESURERKSUoqRnwoQJCAsLw6VLl6DRaBAXF4dt27bJgxW6urrqJEGNGzdWJWgiIiKi/FKU9GTPnv7kyROEhYUhLCwMBw8exMWLF6HRaBAfH48//vgDf/zxBwDAxcUF7dq1k5Ogpk2bKt4AIiIiImNIQgihdqUJCQk6SdCFCxeg0WiyGpQk+d+MjAy1my52EhMT4ezsjISEBDg5ORV1OC+dH5o1Q/S5c6rUVaFpU4w5e1aVuoiIqPAY+1trkru3nJ2d0aNHD3z55Zc4c+YMYmJi8OGHH8oTlAohYIJci4iIiEgv1cbp0SaEwLlz53Dw4EEcPHgQhw8fRmJiok6iY29vb4qmiYiIiHKlStKjL8nJXgYApUqVQqtWrRAQEICAgAC0aNFCjaaJiIiIjKIo6Vm8eDEOHDiA8PBwJCQkADCc5FhbWyuPmIiIiKgAFCU9U6ZMgSRJEEIwySEiIqJiTZXTW5IkoV27dggMDERAQACaNm0q36VFREREVBwoSno6duyI48ePIyUlBbt378aePXsAAE5OTvJ4PEyCiIiIqDhQlPTs3bsX6enpOHnyJA4ePIiwsDAcPXoUCQkJ2LFjB3bu3AmASRAREREVPdUHJ8zIyMiRBD19+jSrsf9LdJydndGuXTt5pOaXGQcnNC0OTkhEREU2OKGVlRVat26NmTNn4p9//sHjx49x5MgRzJgxA05OThBC4MmTJ9ixY4faTRMRERHpZZLBCQEgJSUF4eHh8tg9Z86cQUZGhny3FxEREVFhUi3p0ZfkAMgxEnP2re1EREREhUVR0rNnzx4cOHDAYJJTunRptG7dWp5Z3c/PD1ZWJutgIiIiIsqVouzj1VdfzXG6ysHBAW3atJGTHF9fX1haWioOlIiIiEgJxV0ujo6OOklO8+bNmeQQERFRsaMo6Tl58iSaNWsGCwvVbwIjIiIiUpWipMfHx0etOIiIiIhMil00REREZBZUu40qPj4eq1evxt69e3H58mXEx8cDANzc3NCgQQN06tQJISEhcHNzU6tJIiIiIqOpkvT88MMPmDp1KlJSUgDo3rIeGRmJqKgo7N69G3PmzMGXX36J0aNHq9EsERERkdEUJz0LFizArFmz5ETH2dkZTZs2RYUKFSCEwMOHD3Hu3DkkJCTg6dOnePvtt/HkyRNMnz5dcfBERERExlKU9Fy+fBkffvghhBCoWLEivvjiC/Tv3x/W1tY65TIyMrB582ZMmzYNUVFR+OCDD9CtWzfUr19fUfBERERExlJ0IfPSpUuRmZmJsmXL4tixY3jjjTdyJDxA1iSkgwcPxrFjx1CuXDlkZmZi6dKlSpomIiIiyhdFSc/+/fshSRJmzJiBKlWq5Fne09MT7733HoQQ2Ldvn5KmiYiIiPJFUdITGRkJAGjdurXR67Rp0wYAEBUVpaRpIiIionxRlPRkTzeRPdGoMbLLchRnIiIiKkyKMo/sU1r5OVWVXdaY02FEREREalGU9HTu3BlCCCxatAiXLl3Ks/zFixfxxRdfQJIkBAYGKmmaiIiIKF8UJT0TJ06Era0tkpOT0bZtW3zxxReIi4vLUS4uLg6ff/452rdvj+TkZNja2mLixIlKmiYiIiLKF0Xj9FStWhU//PADQkJCkJycjPfffx8zZsyAl5cXypcvD0mSEB0djdu3b0MIASEEJEnCDz/8wNNbREREVKgUj8g8bNgwlClTBm+99RYiIyMhhMDNmzdx69YtALpTUnh4eGDFihXo2rWr0maJiIiI8kWVube6deuGW7duYdu2bQYnHO3duzesrFSb45SIiIjIaKplIFZWVujfvz/69++vVpVEREREquFgOURERGQWmPQQERGRWWDSQ0RERGbBqGt6sqebUJMkSfmavoKIiIhICaOSHu3bzomIiIhKIqOSno8++sjUcRARERGZFJMeIiIiMgtGJT3ffvstAGDo0KFwdXU1aUBEREREpmDU3VsTJ07EpEmT8ODBA53XO3bsiFdeeQV37twxSXB5iYmJwY4dOzB79mx06dIF7u7ukCQJkiRh+PDh+a7v77//Rp8+fVC5cmXY2tqicuXK6NOnD/7++2/1gyciIqJCpWhE5oMHD0KSJDx9+lStePKlfPnyqtQjhMBbb72FFStW6LweGRmJbdu2Ydu2bRg9ejSWL18OSZJUaZOIiIgKl1E9PXZ2dgCAJ0+emDIWRTw9PREYGFigdT/44AM54WnatCk2btyIkydPYuPGjWjatCkAYMWKFfjwww9Vi5eIiIgKl1FJj5eXFwBgx44dpowl32bPno3Q0FBER0fj7t27+OGHH/Jdx/Xr1/H5558DAHx8fHDkyBEMGjQIvr6+GDRoEMLDw+Hj4wMAWLhwIW7cuKHqNhAREVHhMOr0VteuXXH16lUsXLgQ+/btQ+3atWFtbS0v/+CDD+Di4pKvhiVJwqpVq/K1zos+/vhjResDwOLFi+VBEpcsWYJSpUrpLLe3t8eSJUvQqlUrZGRk4Ouvv8aSJUsUt0tERESFSxJGjDwYHx+Pli1b4vr161kr/d91Ldmr5vc6FyEEJElCZmZmfuM16Pbt26hWrRoAIDg4GGvWrMkzDk9PT0RGRqJu3bqIiIjQW7Zu3br4999/UblyZdy9e9fobU5MTISzszMSEhLg5ORk9LaQcX5o1gzR586pUleFpk0x5uxZVeoiIqLCY+xvrVE9PW5ubjh9+jSWLl2Kffv2ITIyEqmpqbhz5w4kSULFihV1en5Kilu3biEyMhIA4O/vb7Csv78//v33X9y/f18nuSIiIqKSwei7t5ycnDBz5kzMnDlTfs3CIuuSoN27d6NevXrqR2di2j07devWNVhWe3lERASTHiIiohLGrGdZv3fvnvy8cuXKBst6enrmuh4RERGVDIrG6Tlw4AAAlNhej6SkJPm5g4ODwbKlS5eWnycnJ+stl5qaitTUVPn/iYmJCiIkIiIitSjq6cnIyIC/v3+OO56MMXbsWCVNq+L58+fycxsbG4NlbW1t5efPnj3TW+6zzz6Ds7Oz/NDuITJXGpUvWKf/zxT7lu8XEb2sFPX09O7dG/v27YOvr2++1hs9ejRWrVqFZcuWKWlesexBFwEgLS3NYFnt3htDSd6MGTMwefJk+f+JiYlmn/hYWFpia1AQYg3cHZdftbp0Qcd581Srr6RSe9+W9fZGn19+UaUuIqLiRlHSk5ycjK5du+LQoUPw9vY2ap0333wTP/30U7GYzsHR0VF+buiUFQCdqTYMnQqztbXV6RWiLLEREardWg4A7nlceG5O1N63REQvK0Wnt2rXro1Hjx4hMDAQd+/ezbP88OHDsXr1agBAUFCQkqZVoX3x8v379w2W1b542dx7boiIiEoiRUnPnj17ULlyZURGRqJz586IiYnJtZwQAkOHDsW6desghEBwcDDWrl2rpGlVaN9mf/XqVYNltZcb26tFRERExYeipMfT0xN79uyBu7s7rl+/jldffRUJCQk6ZTQaDYKCgvDL/10nMGLEiGJzeqtatWrw8PAAAISFhRkse+jQIQBApUqV5LnIiIiIqORQPE5PnTp18Ndff8HBwQEXL15E9+7d5buiNBoNBg8ejF9//RVA1vU8K1euLBYJD5A1fUbPnj0BZPXkHD9+PNdyx48fl3t6evbsWWziJyIiIuOpMjhh8+bN8ccff8DW1hZHjx5F37598ezZM/Tr1w+bN28GAIwZMwYrVqxQozlVTZw4EVZWWddzjx8/Psft6M+ePcP48eMBAFZWVpg4cWJhh0hEREQqUHT3lraAgABs3LgR/fr1w99//41q1aohNjYWAPD222/ju+++U6spWXh4uDwJKgDExcXJz69fv55jwtHhw4fnqKN27dqYOnUqFixYgNOnT6NNmzZ47733UKNGDdy4cQMLFy7Euf+7M2batGmoVauW6ttBREREpqda0gNknfpZuXIlRowYIV/UPG7cOHz77bdqNiNbuXKl3guijxw5giNHjui8llvSAwDz5s1DTEwMfvrpJ5w7dw6DBg3KUWbkyJGYO3eu4piJiIioaBiV9BhzO3q2Dh064N1338XXX3+Nfv36YerUqXrXr1KlitH1mpKFhQVWrVqFvn37YsWKFTh16hTi4uLg7u4OX19fjBkzBl26dCnqMImIiEgBo5KegsytJUkSfv/9d/z+++96l2dkZOS7Xm1r1qzJcQpLia5du6Jr166q1UdERETFh1FJjxDC1HEQERERmZRRSU/2KMpEREREJZVRSU9wcLCp4yAiIiIyKVXG6SEiIiIq7hQlPRYWFrCyssLnn3+uVjxEREREJqEo6bGxsYEQAu3atVMrHiIiIiKTUJT0ZE/WaWlpqUowRERERKaiKOlp3749AODs2bOqBENERERkKoqSnvHjx8PS0hKLFi1CYmKiWjERERERqU5R0tO8eXMsWbIEd+7cgb+/P44ePapWXERERESqUjTh6IgRIwAAderUwYULF9CuXTt4enqiUaNGcHV1NXitjyRJWLVqlZLmiYiIiIymKOlZs2YNJEkCkJXECCFw9+5d3Lt3z+B6QggmPVTsOFSoAE1mJixMcGG+qeolIiLjKUp6qlSpIic9RCWdnYsLLCwtsTUoCLEREarVW6tLF3ScN89k9RIRkXEUJT23b99WKQyi4iM2IgLR586pVp973bomrZeIiIzDaSiIiIjILDDpISIiIrOg6PSWPhkZGXj8+DEAwNXVFVZWJmmGiIiIyGiq9fRERERg/Pjx8Pb2hp2dHSpUqIAKFSrAzs4O3t7emDBhAq5cuaJWc0RERET5okrSM2PGDDRq1AjLli3Dv//+C41GAyEEhBDQaDT4999/8d1336Fx48aYOXOmGk0SERER5Yvi807jx4/HsmXLIIQAAHh7e6NFixaoUKEChBB4+PAhTp48iStXriAzMxMLFy7E06dP8c033ygOnoiIiMhYipKeI0eO4LvvvoMkSahXrx5WrFiB1q1b51r22LFjeOutt3Dp0iUsXboUAwcO1FuWiIiISG2KTm/98MMPAIBq1arhyJEjBpOYVq1a4dChQ6hevToAYPny5UqaJiIiIsoXRUnP4cOHIUkS3n//fTg7O+dZ3tnZGe+99x6EEDh8+LCSpomIiIjyRVHSEx0dDQBo2rSp0es0a9YMAPDw4UMlTRMRERHli6Kkx87ODgDw9OlTo9dJTk4GANja2ippmoiIiChfFCU91apVAwD8+eefRq8TGhoKAPK1PURERESFQVHS07VrVwghsHTpUuzbty/P8vv27cOSJUsgSRK6du2qpGkiIiKifFGU9EycOBFOTk5IT09Hly5d8M477+DMmTPQaDRyGY1GgzNnzmDs2LHo0qUL0tPT4eTkhIkTJyqNnYiIiMhoisbpcXd3x6ZNm9CjRw+kpaVh+fLlWL58OWxsbODm5gZJkvDo0SOkpaUBAIQQsLGxwebNm1GmTBlVNoCIiIjIGIqnoQgMDMTx48fh4+MjTz2RmpqKBw8eICoqCqmpqfLrvr6+OHHiBDp16qRG7ERERERGU2X68yZNmuDkyZM4deoU9u7di8uXLyM+Ph4A4ObmhgYNGqBTp07w9fVVozkiIiKifFMl6cnm6+vLxIaIiIiKJVVmWSciIiIq7hQlPd9//z2uXLmiVixEREREJqPo9NY777wDSZLg7u6O9u3bIyAgAP7+/mjQoIFa8RERERGpQvE1PUIIxMbGYuvWrdi6dSuArIuXtZOgRo0aKQ6UiIiISAlFSU98fDzCwsJw8OBBHDx4EJcuXYJGo8GjR4+wfft2bN++HQDg6uqKdu3ayUlQkyZNVAidiIiIyHiKkh4XFxf07NkTPXv2BAA8efIEhw4dkpOgixcvQqPRID4+Hn/88Yc8R5ezszPat28vJ0VEREREpqbqLesuLi7o0aMHevToAQBISEjQSYIuXLgAjUaDJ0+eyBOPEhERERUGk96y7uTkhEqVKqFSpUrw8PCAvb09JEkyZZNEREREuVK1p0cIgXPnzsk9O4cPH0ZiYqK8DABKlSqFVq1aoUOHDmo2TURERGSQoqTnxSQnPDwcCQkJ8jLg/yc5AQEBCAgIQIsWLWBtba08ciIiIqJ8UJT0lClThkkOERERlQiKkp4nT57I1+j07NkTkyZNQqtWrZjkEBERUbGjyuCEAPDnn3/i4MGDOoMSNm3alBcuExERUbGgKOk5fPiwfD3PsWPHkJCQgNDQUOzYsQNA1t1b2YMSBgQEMAkiIiKiIqMo6WnTpg3atGmDWbNmISMjAydPnpSToKNHjyIhIQE7duzAzp07AeRMgpo1a6bKRhARERHlRbVb1q2srNC6dWu0bt0aM2fOREZGBk6dOpUjCdq5cyd27twJSZKQkZGhVvNEREREBqk6To9OxVZWaNWqFRo3boxmzZqhYcOGWLVqFRITE+XrgIiIiIgKi+pJT0pKCsLDw+UenjNnzsg9OtrJjr29vdpNExEREemlOOkxNskpXbq0PH6Pv78//Pz8lDZNREREZDTFFzKfPn061yTHwcEBrVu3lpMcX19fWFmZ7GwaERERkUGKspBjx47Jzx0dHdGmTRs5yfHx8YGlpaXiAImIiIjUoCjp6dq1q5zkNG/eHBYWJp20nYiIiKjAFCU92YMQEhERERV37JohIiIis8Ckh4iIiMwCkx4iIiIyC0x6iIiIyCww6SEiIiKzwKSHiIiIzAKTHiIiIjILTHqIiIjILBiV9PTp0wd9+/bF/fv3TR0PERERkUkYlfRs374d27dvR2Jiou7KFhawsrLClStXTBIcERERkVrydXpLexZ1Q68RERERFTdGJT2Ojo4AgIcPH5o0GCIiIiJTMSrpqVu3LgDgm2++QXJyco7lkiSpGxURERGRyoyaZf2NN97AqVOnsGPHDri5uaF8+fKwtraWlwcGBur83xiSJOHGjRv5i5aIiIiogIxKesaPH48jR45gy5YtyMjIQGRkpLxMCKHzf2Oxd4iIiIgKk1FJj4WFBTZt2oRjx45h7969iIyMRGpqKtauXQtJktCjRw+4uLiYOFQiIiKigjMq6cnWqlUrtGrVSv7/2rVrAQDz5s1DvXr11I2MiIiISEUckZmIiIjMQr56el6k0WjUioOIiIjIpNjTQ0RERGZBUU+PtszMTGzfvh179+7F5cuXER8fDwBwc3NDgwYN0KlTJ/Tq1QuWlpZqNUlERERkNFWSnr///hujR4/OcSs7kHVr+tGjR7FixQpUrlwZK1aswKuvvqpGs0RERERGU3x6a/369ejevTsiIyMhhIAQAlWrVkXLli3RokULVK1aFUBWEnTv3j1069YNv/zyi+LAiYiIiPJDUdJz584djB49GhqNBvb29pg7dy6io6Nx8+ZNHD16FMeOHcPNmzcRHR2NefPmwcHBARqNBqNGjcLdu3fV2gYiIiKiPClKer755hukpqbCwcEBhw8fxsyZM1GuXLkc5cqWLYsZM2bg8OHDcHBwQGpqKr755hslTRMRERHli6KkZ/fu3ZAkCdOmTUOTJk3yLN+4cWNMnToVQgj8888/SpomIiIiyhdFSU/2KapOnToZvU7nzp111iUiIiIqDIqSnszMTADI123o2WU5sCEREREVJkVJT6VKlQAAR48eNXqd7LIeHh5KmiYiIiLKF0VJT4cOHSCEwIIFCxAVFZVn+fv372PBggWQJAkdO3ZU0jQRERFRvihKesaPHw8LCwvExsaiRYsW2Lx5s3zKS1tmZiY2bdqEVq1aISYmBhYWFhg3bpySpomIiIjyRdGIzA0aNMCnn36KWbNmISoqCoMGDYKLiwuaNm2K8uXLQ5IkREdH49y5c3jy5Ik8SvOnn36KBg0aqLIBRERERMZQPA3FjBkz4OzsjOnTpyMlJQWPHz/GgQMHdMpkJzv29vb44osv8PbbbyttloiIiChfVJl7a+zYsRgwYABWr15tcMLRkJAQuLu7q9EkERERUb6oNsu6u7s7pk2bhmnTpqlVJREREZFqFE84SkRERFQSMOkhIiIis8Ckh4iIiMwCkx4iIiIyC2af9EiSZNQjICCgqEMlIiIiBcw+6SEiIiLzoNot6yXd22+/jbFjx+pdXrp06UKMhoiIiNTGpOf/lCtXjlNjEBERvcR4eouIiIjMgqKenk8++QQA0KJFC7z66quqBERERERkCoqSnjlz5kCSJGzbtk2teIiIiIhMQtHprTJlygAAqlSpokowRWnz5s2oU6cOSpUqBUdHR9SqVQvBwcE5ZownIiKikklR0lOzZk0AQHR0tCrBFKUrV67g2rVreP78OZKTk3H9+nWsW7cOHTt2RO/evZGQkFDUIRIREZECipKegQMHQgiBTZs2qRVPobO3t8egQYPw448/4vDhwzh37hx2796NWbNmyT1Z27dvR8+ePZGenp5nfampqUhMTNR5EBERUdFTlPSMHTsWjRs3xrp167B27Vq1YipUkZGR2LhxI9588020bdsWTZo0QefOnTF37lz873//Q9OmTQEAYWFh+P777/Os77PPPoOzs7P88PT0NPUmqEqTmVnUIVARcqhQwWTHAI8tIipqii5kjo6OxsqVKzFy5EiMGDECv/zyC9544w00atQIrq6usLS0NLh+cbgWyMXFRe+y8uXLY8uWLfD29kZaWhqWLFmCCRMmGKxvxowZmDx5svz/xMTEEpX4WFhaYmtQEGIjIlSpr1aXLug4b54qdZHp2bm4qH4MAEBZb2/0+eUX1eojIioIRUmPl5cXJEkCAAghsG/fPuzbt8+odSVJQkZGhpLmC0X16tXRuXNn7Ny5E9evX0dUVBQ8PDz0lre1tYWtrW0hRqi+2IgIRJ87p0pd7nXrqlIPFS41jwEiouJC8eCEQggIIXSeG/soKerVqyc/j4yMLMJIiIiIqKAU9fSsXr1arTiKtZKUoBEREVHuFCU9wcHBasVRrF25ckV+bujUFhERERVfnHsrDzdv3sSePXsAZF3fU6lSpSKOiIiIiArCrJOe0NBQgxdTP3z4EP369ZPH53nnnXcKKzQiIiJSmaLTW9o0Gg0OHjyIY8eOITo6GikpKZg7dy4qVqwol0lLS0NGRgYsLS2LxR1O48ePR3p6Ovr27YtWrVrBy8sLpUqVQlxcHA4ePIjly5fj0aNHAIC2bdsy6SEiIirBVEl6du7ciQkTJuD27ds6r0+ZMkUn6Vm1ahXGjRsHBwcHREVFoXTp0mo0r0hUVBSWLFmCJUuW6C3Tt29frFy5slgkakRERFQwik9vrVy5Ej169MCtW7cghECZMmX03u00cuRIuLi4IDk5uVjMzL527Vp8/PHHeO2111C7dm24ubnBysoKLi4uaNiwIcaMGYOjR49iy5YtBgcxJCIiouJPUU/P9evX5VM+HTt2xNKlS1G3bl1YWOSeS9nY2Mi9Jrt378aQIUOUNK+Yv78//P39izQGIiIiKhyKenq+/vprpKeno379+ti1axfqGjH6brt27QAA58+fV9I0ERERUb4oSnr27dsHSZIwceJE2NjYGLVOjRo1AAB3795V0jQRERFRvihKeu7duwcAaNKkidHrZF+8nJKSoqRpIiIionxRlPRoTzZqrNjYWACAk5OTkqaJiIiI8kVR0pM9JcO1a9eMXicsLAxA1gztRERERIVFUdLTvn17CCGwYcMGo8rHxcXhhx9+gCRJ6Nixo5KmiYiIiPJFUdIzevRoAMCuXbvynHH9/v376Nq1K+Li4mBpaSmvS0RERFQYFCU9vr6+eOuttyCEwJtvvon+/ftj06ZN8vKLFy/it99+w8iRI1GnTh2cOXMGkiRhypQpqFmzpuLgiYiIiIyleBqKJUuW4OnTp1i/fj22bt2KrVu3yhc4BwUFyeWyL3YePnw45s+fr7RZIiIionxRPA2FpaUl1q5di82bN6Np06YQQuT6qFevHjZs2ICffvpJToqIiIiICotqs6z37dsXffv2RVRUFE6fPo2YmBhkZmaiTJkyaNq0qTwoIREREVFRUC3pyebh4YEePXqoXS29QJOZCQtLy6IOg4iIqMRQPemhwmFhaYmtQUGIjYhQrc5aXbqg47x5qtVHRERUnKia9KSnp+Ps2bO4fPky4uPjAQBubm5o0KABmjVrBmtrazWbM3uxERGIPndOtfrcjZgwloiIqKRSJelJTk7Gp59+ilWrVuHx48e5lnF1dcXIkSPxwQcfwNHRUY1miYiIiIym+O6tiIgI1K9fH4sWLUJ8fLzeu7fi4+OxaNEiNGzYEP/++68asRMREREZTVFPz5MnT9CpUyc8ePAAANCgQQMEBwfDz88P5cuXhxACMTExOHXqFNauXYtLly7h7t276NSpEy5fvgxnZ2dVNoKIiIgoL4p6ehYuXIgHDx5AkiR8+umnuHDhAqZMmYJ27dqhdu3aqFOnDtq1a4fJkyfj/PnzmDt3LgAgKioKCxcuVGUDiIiIiIyhKOnZvn07JEnCwIEDMWvWLIODDkqShJkzZ2LgwIEQQmDbtm1KmiYiIiLKF0VJz507dwAAwcHBRq8zfPhwnXWJiIiICoOipCf7Lqxy5coZvU52WQcHByVNExEREeWLoqSnYcOGAID//vvP6HWyy2avS0RERFQYFCU9Y8aMgRACX3/9NTQaTZ7lNRoNFi9eDEmSMHr0aCVNExEREeWLoqSnf//+CAkJwfHjx9GrVy9ER0frLfvw4UP06dMHJ06cwPDhwzFw4EAlTRMRERHli1Hj9Kxbt07vMn9/f1y+fBk7duxA9erVERgYCF9fX5QrVw6SJOHhw4c4deoUdu/ejdTUVPj6+sLf3x/r1q3DsGHDVNsQIiIiIkOMSnqGDx9u8HZ0IOuW9OfPnyM0NBShoaE5lgshIEkSTp8+jZCQEEiSxKSHiIiICo3RIzILIRSXM7YOIiIiIrUZlfTcunXL1HEQERERmZRRSU/VqlVNHQcRERGRSSmeZZ2IiIioJGDSQ0RERGaBSQ8RERGZBVWSnoiICEyaNAk+Pj5wc3ODtbU1LC0tDT6srIy+cYyIiIhIMcWZx4IFCzB79mxkZmbylnQiIiIqthQlPZs3b8bMmTMBABYWFmjXrh0aN24MFxcXWFjwzBkRmZYmMxMWlpYlpl4iKlqKkp5vvvkGAFCpUiXs2rWLM6cTUaGysLTE1qAgxEZEqFZnWW9v9PnlF9XqI6LiQ1HSc/HiRUiShE8//ZQJDxEVidiICESfO1fUYRBRCaDoHJS1tTUAoEmTJmrEQkRERGQyipKe2rVrAwAePXqkSjBEREREpqIo6QkODoYQAtu3b1cpHCIiIiLTUJT0jBw5Eu3atcOKFSsQGhqqVkxEREREqlN0IbO1tTX++OMPBAcHo3fv3hg0aBAGDBiA2rVrw97ePs/1q1SpoqR5IiIiIqMpHpzQxcUFEyZMwPHjx7Fx40Zs3LjRqPUkSUJGRobS5omIiIiMongEwYkTJyIwMBBxcXEQQuTrQURERFRYFPX0/Pzzz/j2228BAE5OTujduzcaNWrEEZmJiIio2FGU9CxZsgQAULduXRw4cADly5dXJSgiIiIitSnqjrl69SokScKcOXOY8BAREVGxpsqIzNmDFBIREREVV4qSnrp16wIAoqOjVQmGiIiIyFQUJT0hISEQQhh9mzoRERFRUVE8InP37t3x888/Y+nSpWrFRERERKQ6RXdvHTp0CBMmTEBsbCzeffddbNiwAYMGDTJ6ROb27dsraZ6IiIjIaIqSnoCAAEiSJP//xIkTOHHihFHrckRmIiIiKkyKp6HgyMpERERUEihKeg4cOKBWHEREREQmpSjp8ff3VysOIiIiIpPiBFlERERkFpj0EBERkVlg0kNERERmQdE1PR07dizwupIkYd++fUqaJyIiIjKaoqTn4MGDkCTJ4G3r2uP4AP//FvcXXyciIiIyJUVJT/v27fNMXp4+fYr//vsPCQkJkCQJtWvXRsWKFZU0S0RERJRvint6jCGEwM6dO/Huu+8iPj4eK1euRNu2bZU0TURERJQvhXIhsyRJ6N69O8LDw2FpaYnevXsjMjKyMJomIiIiAlDId29VrFgRkydPxqNHj/D5558XZtNERERk5gr9lvXs01o7d+4s7KaJiIjIjBV60mNjYwMAiIqKKuymiYiIyIwVetITHh4OALC3ty/spomIiMiMFWrSc+zYMXzyySeQJAl+fn6F2TQRERGZOUW3rH/yySd5ltFoNHj8+DFOnz6NEydOQKPRQJIkTJo0SUnTRERERPmiKOmZM2dOvkZWFkLAysoKn3/+OTp37qykaSIiIqJ8UZT0ADA4BQWQNUaPo6MjqlWrBn9/f4wePRr16tVT2iwRERFRvihKejQajVpxEBEREZlUod+9RURERFQUmPQQERGRWWDSQ0RERGZB8YXM2pKSknDr1i0kJSUhMzMzz/Lt27dXs3kiIiIivVRJen788UcsW7YMFy9eNHodSZKQkZGhRvNEREREeVKU9GRmZqJv374IDQ0FkPft60RERERFRVHSs3z5cvz5558AgPLlyyMkJATNmzeHm5sbLCx4uRAREREVH4qSnnXr1gEA6tWrh8OHD8PV1VWVoIjo5eJQoQI0mZmwsLQs6lCKlCn2QUnar6aKtSTVy/eraPeBoqQnIiICkiThww8/ZMJDRHrZubjAwtISW4OCEBsRoUqdtbp0Qcd581Spq7CovQ/Kenujzy+/qFJXYVB7+wHT7gO+XyXr/TKGKhcy16lTR41qiOglFxsRgehz51Spy71uXVXqKWxq7oOSqKRtf0mLV20v2/YruvCmVq1aAID4+HhVgiEiIiIyFUVJz6BBgyCEwI4dO9SKh4iIiMgkFCU9EyZMQKNGjfD999/j8OHDasVEREREpDpFSY+trS12796N5s2bo3Pnzpg+fTrOnz+P58+fqxUfERERkSoUXchsqXXLmRACX375Jb788kuj1uWIzERERFSYFCU9L47AzBGZiYiIqLhSlPR89NFHasVBREREZFJMeoiIiMgscIIsIiIiMgtMeoiIiMgsMOnRcvfuXUydOhXe3t4oXbo03Nzc4Ofnh0WLFiElJaWowyMiIiIFVJl762Wwc+dOBAUFISEhQX4tJSUFp06dwqlTp7By5Urs2rUL1atXL8IoiYiIqKDY0wPgwoULGDBgABISEuDg4IB58+bh6NGj2LdvH0aNGgUA+Pfff9GtWzckJycXcbRERERUEOzpATBx4kSkpKTAysoKu3fvRqtWreRlHTt2RK1atTB9+nRcvXoVX331FWbPnl2E0RIREVFBmH1Pz6lTp3Dw4EEAwMiRI3USnmxTpkyBt7c3AODrr79Genp6YYZIREREKjD7pGf79u3y85CQkFzLWFhYYNiwYQCAx48fy0kSERERlRxmn/Rkzw5funRpNG/eXG85f39/+Xl4eLjJ4yIiIiJ1mX3SExERAQCoWbMmrKz0X+JUt27dHOsQERFRyWHWSc/z588RFxcHAKhcubLBsq6urihdujQA4N69eyaPjYiIiNRl1ndvJSUlyc8dHBzyLF+6dGk8ffrU4G3rqampSE1Nlf+fPe5PYmKigkj1xFOzJpwzM1Wrz8bDA4mJiarWa4o6TVVvSYrVVPUy1qzPlSk+r9r1qxWvqWM1Bb5ffL9MsQ+y6xRCGC4ozNjdu3cFAAFADB06NM/ynp6eAoCoUaOG3jIfffSRXCcffPDBBx988FF4j3v37hn8HTfrnh47Ozv5eVpaWp7ls3twSpUqpbfMjBkzMHnyZPn/Go0G8fHxKFOmDCRJUhDt/5eYmAhPT0/cu3cPTk5OqtRJRYPv5cuD7+XLhe9nySKEQFJSEjw8PAyWM+ukx9HRUX5uzEjLT58+BWD4VJitrS1sbW11XnNxcSlYgHlwcnLih/Elwffy5cH38uXC97PkcHZ2zrOMWV/IbGdnB3d3dwDA/fv3DZZ9/PixnPR4enqaPDYiIiJSl1knPQDkkZavX7+OjIwMveWuXr2aYx0iIiIqOcw+6Wnbti2ArFNXZ86c0VsuLCxMft6mTRuTx2WIra0tPvrooxyn0ajk4Xv58uB7+XLh+/lykoTI6/6ul9vJkyfRokULAMCYMWOwfPnyHGU0Gg0aNGiAiIgIuLi4ICYmBtbW1oUdKhERESlg9j09fn5+aNeuHQBg1apVOHbsWI4yX375pTwK87vvvsuEh4iIqAQy+54eADh37hzatGmDZ8+ewcHBATNnzkSHDh3w7Nkz/Prrr1ixYgUAoHbt2jh9+rTOXV9ERERUMjDp+T+hoaEYMmSI3pEia9eujZ07d6JmzZqFHBkRERGpwexPb2V7/fXXcfHiRUyaNAm1a9eGvb09XFxc4OPjg4ULF+LcuXNFkvDExMRgx44dmD17Nrp06QJ3d3dIkgRJkjB8+PBCj4cK7uzZs5g/fz66dOkCT09P2NrawsHBAbVr18bw4cNx+PDhog6RjJSYmIhff/0VU6ZMgb+/P2rWrAlnZ2fY2NigXLlyCAgIwOeff45Hjx4VdaikwPTp0+XvW0mScPDgwaIOiRRiT08xZ2gU5+DgYKxZs6bwgqEC8/f3x6FDh/IsN3ToUKxcuRI2NjaFEBUV1N69e9G5c+c8y7m7u+Pnn3/Gq6++WghRkZouXLgAHx8fnaFMDhw4gICAgKILihQz6xGZSxpPT094e3tj9+7dRR0K5VNkZCQAwMPDA/3790e7du1QpUoVZGZm4tixY/jyyy8RGRmJ9evXIyMjAxs2bCjiiCkvnp6e6NChA5o3bw5PT09UrFgRGo0G9+/fx5YtW7B161bExcWhR48eOHXqFBo1alTUIZORNBoNRo0ahYyMDJQrVw4xMTFFHRKpxdjJOalozJ49W4SGhoro6GghhBC3bt2SJ1YLDg4u2uDIaN26dRO//fabyMjIyHV5bGysqF27tvzeHjp0qJAjpPzQ9z5q27Ztm/x+9unTpxCiIrUsXrxYABB169YVM2bMkN/HAwcOFHVopBCv6SnmPv74Y3Tv3h3ly5cv6lBIgR07dmDAgAGwtLTMdbm7uzu+/PJL+f9btmwprNCoAPS9j9p69eqFunXrAoBRpzapeLh37x4+/PBDAMD333/PU80vGSY9RMWE9rUCN27cKLpASDWlS5cGADx//ryIIyFjjR07FsnJyQgODub1Oy8hJj1ExURaWpr83MKCH82SLiIiAufPnwcAuceHirdNmzZhx44dcHNzwxdffFHU4ZAJ8JuVqJjQnt+NP5IlU0pKCv777z989dVX6NChAzIzMwFkjeROxduTJ0/k92nhwoUoW7ZsEUdEpsC7t4iKAY1GgwULFsj/HzBgQBFGQ/mxZs0ahISE6F0+depUBAUFFWJEVBDTp09HdHQ0WrdujZEjRxZ1OGQiTHqIioHFixfj5MmTAIDevXvDx8eniCMipZo0aYLly5fLExpT8RUeHo6VK1fCysoKy5cvNzg+GpVsPL1FVMTCwsLw/vvvAwDKlSuH77//vogjovzo1asXLl26hEuXLuHkyZPYuHEjevfujfPnzyMoKAg7duwo6hDJgLS0NIwePRpCCEyaNAkNGzYs6pDIhJj0EBWh//3vf+jduzcyMjJga2uLTZs2cXiCEsbFxQUNGjRAgwYN4Ovri0GDBmHr1q1Yt24dbt68iZ49e3Lk9GJs/vz5iIiIQJUqVfDRRx8VdThkYkx6iIrIrVu3EBgYiMePH8PS0hIbN26Ev79/UYdFKhk6dCj69+8PjUaDcePG4fHjx0UdEr3g6tWr+OyzzwAAS5YskYcYoJcXr+khKgJRUVHo1KkToqKiIEkSfvrpJ/Tu3buowyKV9ezZE5s2bcLTp0/x119/4Y033ijqkEjL4sWLkZaWhurVqyMlJQW//vprjjKXL1+Wn+/fvx/R0dEAsiapZpJU8jDpISpkcXFx6Ny5M27evAkg6y/MYcOGFXFUZAratz3fuXOnCCOh3KSmpgIAbt68icGDB+dZ/tNPP5Wf37p1i0lPCcTTW0SFKCEhAa+++iquXLkCAFiwYAHeeeedIo6KTCV7olkAcHBwKMJIiAhg0kNUaFJSUtCtWzecPXsWADBr1iy89957RRwVmdLmzZvl57wrqPhZs2YNhBAGH9oXNx84cEB+3cvLq+gCpwJj0kNUCNLS0tC7d28cOXIEQNYIvXPnzi3iqKig1qxZk+d8WosXL8auXbsAAF5eXmjbtm1hhEZEBvCanmIuPDwc169fl/8fFxcnP79+/XqOW2GHDx9eSJFRfgwePBi7d+8GAHTs2BEjR47UuUDyRTY2Nqhdu3ZhhUf5NGfOHEyZMgV9+/ZF27ZtUaNGDTg4OCApKQmXLl3CL7/8Iie4NjY2+PHHH2Flxa9boqImCSFEUQdB+g0fPhxr1641ujzfzuIpvyO8Vq1aFbdv3zZNMKSYl5eXURcmV65cGT/99BM6d+5cCFGRKcyZMwcff/wxgKzTW5x5vWTjnx5ERPm0b98+7N27FwcOHEBERAQePnyIR48ewc7ODuXLl0eTJk3QvXt3DBgwAPb29kUdLhH9H/b0EBERkVnghcxERERkFpj0EBERkVlg0kNERERmgUkPERERmQUmPURERGQWmPQQERGRWWDSQ0RERGaBSQ8RERGZBSY9REREZBaY9BAREZFZYNJDREREZoFJDxEVuTVr1kCSJEiSxNnlichkmPQQERGRWWDSQ0RERGaBSQ8RERGZBSY9REREZBaY9BAREZFZYNJDRCb3+PFjvP/++6hbty5KlSqFcuXKoVOnTti8ebPRdWRkZGDVqlXo2rUrPDw8YGtrC3d3d7Rv3x5ff/01nj9/nmcdFy9exNChQ1GpUiXY2dmhSpUqGDJkCM6ePQsAGD58OCRJgpeXV451b9++Ld9htmbNGgDA1q1b5XisrKwQEBCQY7379+9jxowZaNasGVxdXeV2Bw4ciAMHDhi17Y8fP8bcuXPRqlUruLu7w9bWFh4eHujZsye2bt1qVB1EBEAQEZnQ//73P1GxYkUBINfHiBEjxOrVq+X/37p1K0cd169fF/Xq1dNbBwBRq1Ytce3aNb1xrFmzRlhbW+e6rrW1tVizZo0IDg4WAETVqlVzrH/r1i25/E8//SSGDh2aox5/f3+ddVauXClKlSplMO6RI0eK9PR0vXHv3LlTuLi4GKyjW7duIikpydi3hMhsMekhIpN58uSJqFy5svzjPHDgQLFr1y5x+vRpsWHDBuHj4yMACF9fX71JT1RUlChfvrwAIBwdHcWUKVPEX3/9Jc6ePSsOHDggZsyYIezt7QUAUb16dfHkyZMccRw+fFhYWFgIAKJUqVJi5syZ4tChQ+LEiRPiu+++E5UrVxY2NjaiadOmRiU9jRo1EgBEu3btxIYNG8Tp06fF3r17xcqVK+Xyq1atkss3aNBALFmyRISHh4uzZ8+K33//XXTt2lVePnny5Fz33+7du4WlpaUAILy8vMTChQvFwYMHxdmzZ0VoaKgYMmSIXEefPn0UvVdE5oBJDxGZzOTJk+Uf5fnz5+dYnpaWJgIDA3V6LV5Merp37y4ACE9PT3Hjxo1c2zl79qwoXbq0ACA++OCDHMsbN24sAAgbGxtx5MiRHMsfPnwoqlevLseQV9IDQAwbNkxoNJpc47l7966ciAUHB+vtyZk5c6YAICwsLMS///6rsyw5OVlO9gIDA8XTp09zrWPFihVyTHv37s21DBFlYdJDRCbx/Plz4erqKveMZGZm5lru3r17OqedtJOeS5cuya//8ccfBtubPn26ACA8PDx0Xj927Jhcx6RJk/Su/8cffxid9Li4uIjExES9dU2ZMkWO5dmzZ3rLpaeni0qVKgkAYtasWTrLlixZIgAIOzs78fDhQ711CCGEn5+fACCCgoIMliMyd7yQmYhM4syZM3j8+DEAIDg4GBYWuX/dVK5cGYGBgbku++OPPwAA9vb26Natm8H22rdvDwCIiorCvXv35Nf37dsnPw8ODta7frdu3VCmTBmDbWR7/fXX4ejoqHd5dtyvv/467Ozs9JazsrJCq1atAADHjh3LtQ5/f3+UK1fOYDzZ2/5iHUSky6qoAyCil9OlS5fk576+vgbL+vn5YefOnTleP336NAAgJSUFVlbGf11FR0fD09MTAHD58mUAgK2tLRo0aKB3HUtLSzRp0kQnSdKnUaNGepclJCTg+vXrAIAffvgBP/zwg9Exa8ve9n/++QeSJBWoDiLSxZ4eIjKJ7F4eAHn2VJQvXz7X12NiYgrUdkpKSo443NzcYGlpaXC9smXLGlW/q6ur3mVqxJyeno4nT54oqoOIcmJPDxGZhBBCfp5XT4V2WW2ZmZkAgGrVquHPP/80uu1q1aoZXbYgDCVP2TEDwMSJEzFy5Eij6rSxscm1jgEDBuDDDz8sQJRE9CImPURkEm5ubvLzhw8fonbt2nrL6usdyb7G5uHDh6hbt26+TnFly+6ViY+PR2ZmpsGEJTY2Nt/1v0j7uqCUlBSDp9T0sbOzg729PVJSUvDkyZMC1UFEOfH0FhGZRMOGDeXnp06dMlhW3/KmTZsCyEoejhw5UqA46tevDwBITU3Vuc7oRZmZmTh//nyB2tBWtmxZVKpUCQCwd+9evb1Yecne9iNHjvC0FZFKmPQQkUk0b95c7mVZv3693h//yMhI7N69O9dlPXv2lJ9//vnnBYrjlVdekZ+vW7dOb7mdO3fi0aNHBWrjRT169AAA3Lx5E1u2bFFUx9OnT/Hdd9+pEheRuWPSQ0QmYWtri5CQEADA+fPn8cUXX+Qok5GRgVGjRiEtLS3XOnx9feXb2Xft2oWPPvrIYJu3b9/Gxo0bdV5r1aqVfLfVd999h6NHj+ZYLzY2FpMmTcp7o4w0bdo02NraAgDeeust+U4sfXbt2oWLFy/qvPbWW2/B3d0dAPDhhx/ir7/+MljHkSNHcOjQIQVRE738JFHQvlciojwkJCSgQYMGuH//PgBg8ODBGDZsGMqVK4dr167hq6++wqlTp+Dr6yuf4rp165bOhJ9RUVHw8fHBgwcPAAAtWrTAiBEj0LBhQ9jZ2eHRo0e4ePEi/v77b+zfvx+9evXK0bsSHh4Of39/aDQalCpVCpMnT8Zrr70GW1tbnD59Gp999hmio6NRv359nD9/Hl5eXrh165ZOHbdv35YvkF69ejWGDx9ucNvXrFkjJ302NjYYOnQounfvjipVqiAjIwP379/HyZMnsWXLFty4cQOhoaHo3r27Th179+5Fly5dkJGRAQsLC/Tt2xd9+/ZFjRo1AAAPHjzAmTNnsG3bNly8eBFLlizBuHHj8vEOEZmZohwZkYhefpcvXxYVKlTQO1lmSEhInhOO3r59W2d+LkOPkJCQXOMwNOGolZWV+PHHH+VJROvWrZtjfe0RmVevXm3Utv/666/Cyckpz5gtLCzE/v37c61j3759Bvef9mPt2rVGxUVkrnh6i4hMqn79+vjf//6H6dOno1atWrC1tYW7uzs6dOiADRs24KeffsqzjqpVq+LEiRPYtm0bBg0ahGrVqsHe3h7W1tYoW7YsWrdujSlTpiAsLAyrVq3KtY7g4GCcPn0aQUFB8PDwgI2NDSpVqoQBAwYgPDwcb775JhITEwEAzs7Oqmz7wIEDcfv2bSxYsAABAQEoV64crK2tYW9vj+rVq+P111/HV199hdu3b6NDhw651tGxY0fcuHEDS5cuxWuvvYaKFSvCxsYGdnZ28PT0RGBgIObNm4erV69i2LBhqsRN9LLi6S0iov9Ts2ZN3LhxA0OGDMH69euLOhwiUhl7eoiIkHXb/I0bNwAALVu2LOJoiMgUmPQQkVnIng8rN48ePcKoUaMAZN11NnDgwMIKi4gKEUdkJiKz0LlzZ1SrVg29e/dGo0aN4OzsjMePH+PIkSNYtmyZfHfYBx98IN8qTkQvF17TQ0RmwcvLC3fu3DFYZuzYsViyZAksLNgJTvQyYtJDRGYhLCwMoaGhCAsLw4MHDxAXFwcrKytUqFABbdu2xejRo9G6deuiDpOITIhJDxEREZkF9uESERGRWWDSQ0RERGaBSQ8RERGZBSY9REREZBaY9BAREZFZYNJDREREZoFJDxEREZkFJj1ERERkFpj0EBERkVn4f82vOICTuWEPAAAAAElFTkSuQmCC",
      "text/plain": [
       "<Figure size 640x480 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "import csv\n",
    "import matplotlib.pyplot as plt\n",
    "import pandas as pd\n",
    "\n",
    "in_degrees = {}\n",
    "\n",
    "with open('./results/processes_degrees.csv', newline='') as csvfile:\n",
    "    reader = csv.reader(csvfile)\n",
    "    next(reader)\n",
    "    for row in reader:\n",
    "        try:\n",
    "            pipeline = row[0]\n",
    "            in_degree = float(row[2]) # in degree\n",
    "            \n",
    "            if pipeline not in in_degrees:\n",
    "                in_degrees[pipeline] = []\n",
    "            in_degrees[pipeline].append(in_degree)\n",
    "        except ValueError:\n",
    "            print(f\"Skipping row with non-numeric value in the specified column: {row}\")\n",
    "\n",
    "avg_in_degrees = {pipeline: sum(degrees) / len(degrees) for pipeline, degrees in in_degrees.items()}\n",
    "\n",
    "# Plot the histogram\n",
    "plt.rc('font', size=20)\n",
    "plt.hist(avg_in_degrees.values(), bins=20, color='maroon', edgecolor='white')\n",
    "plt.title('Avg logical process degree')\n",
    "plt.xlabel('degree')\n",
    "plt.ylabel('number of workflows')\n",
    "plt.tight_layout()\n",
    "plt.savefig(f'{directory_plots}/avg_in_degree.png', format=\"png\", bbox_inches='tight')\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Usage of Queue channel vs. Value channel keywords in DAGs"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We want to check the xlabel of each value/channel node to compare it to the dags. Just run the cell below, all imports and parths are assigned again.\n",
    "\n",
    "Before we checked like that: \n",
    "queue_channels = ['Channel.of', 'Channel.fromPath', 'Channel.empty', 'map', 'flatMap', 'ifEmpty', 'join']\n",
    "value_channels = ['first', 'collect', 'reduce']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import csv\n",
    "import networkx as nx\n",
    "\n",
    "def extract_xlabels_from_dot_file(dot_file, channel_list):\n",
    "    G = nx.nx_agraph.read_dot(dot_file)\n",
    "\n",
    "    matching_nodes = []\n",
    "    for node, data in G.nodes(data=True):\n",
    "        if 'xlabel' in data and data['xlabel'] in channel_list:\n",
    "            matching_nodes.append({\n",
    "                'pipeline': os.path.basename(dot_file).split('_')[0],\n",
    "                'node': node,\n",
    "                'operator_label': data['xlabel'],\n",
    "                'operator': 'queue channel' if data['xlabel'] in queue_channels else 'value channel'\n",
    "            })\n",
    "    return matching_nodes\n",
    "\n",
    "def count_nodes_by_labels(dot_file, channel_list):\n",
    "    matching_nodes = extract_xlabels_from_dot_file(dot_file, channel_list)\n",
    "    return len(matching_nodes), matching_nodes\n",
    "\n",
    "def iterate_over_dot_files(directory_path, queue_channels, value_channels):\n",
    "    all_queue_nodes = []\n",
    "    all_value_nodes = []\n",
    "    results_list = []\n",
    "\n",
    "    for filename in os.listdir(directory_path):\n",
    "        if filename.endswith(\".dot\"):\n",
    "            dot_file_path = os.path.join(directory_path, filename)\n",
    "            pipeline_name = os.path.basename(filename).split('_')[0]\n",
    "            \n",
    "            queue_count, queue_matching_nodes = count_nodes_by_labels(dot_file_path, queue_channels)\n",
    "            all_queue_nodes.extend(queue_matching_nodes)\n",
    "\n",
    "            value_count, value_matching_nodes = count_nodes_by_labels(dot_file_path, value_channels)\n",
    "            all_value_nodes.extend(value_matching_nodes)\n",
    "\n",
    "    results_list.extend(all_queue_nodes)\n",
    "    results_list.extend(all_value_nodes)\n",
    "\n",
    "    return results_list\n",
    "\n",
    "directory_path = './dags'\n",
    "queue_channels = ['Channel.of', 'Channel.fromPath', 'Channel.empty', 'map', 'flatMap', 'ifEmpty', 'join']\n",
    "value_channels = ['first', 'collect', 'reduce']\n",
    "\n",
    "results_list = iterate_over_dot_files(directory_path, queue_channels, value_channels)\n",
    "\n",
    "csv_operator_path = './results/operator_channels_counts_with_label.csv'\n",
    "header = ['pipeline', 'node', 'operator_label', 'operator']\n",
    "\n",
    "with open(csv_operator_path, 'w', newline='') as csv_file:\n",
    "    writer = csv.DictWriter(csv_file, fieldnames=header)\n",
    "    writer.writeheader()\n",
    "    writer.writerows(results_list)\n",
    "\n",
    "print(f'Results written to: {csv_operator_path}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "\n",
    "df_operators = pd.read_csv('./results/operator_channels_counts_with_label.csv')\n",
    "label_counts = df_operators['operator_label'].value_counts()\n",
    "\n",
    "print('Distinct Labels and Their Sum of Counts:')\n",
    "print(label_counts)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "\n",
    "df_operators = pd.read_csv('./results/operator_channels_counts_with_label.csv')\n",
    "\n",
    "\n",
    "queue_operators = df_operators[df_operators['operator'] == 'queue channel']\n",
    "value_operators = df_operators[df_operators['operator'] == 'value channel']\n",
    "\n",
    "\n",
    "print(f'Sum of queue operator nodes in DAGs {len(queue_operators)}')\n",
    "print(f'Sum of value operator nodes in DAGs {len(value_operators)}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "different_pipelines = df_operators.groupby('operator')['pipeline'].nunique()\n",
    "print(different_pipelines)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "different_pipelines = df_operators.groupby('operator_label')['pipeline'].nunique()\n",
    "print(different_pipelines)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# We focus on all nextflow channel operators here: (https://www.nextflow.io/docs/latest/operator.html)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import csv\n",
    "import networkx as nx\n",
    "import pandas as pd\n",
    "\n",
    "def extract_xlabels_from_dot_file(dot_file, channel_list):\n",
    "    G = nx.nx_agraph.read_dot(dot_file)\n",
    "\n",
    "    matching_nodes = []\n",
    "    for node, data in G.nodes(data=True):\n",
    "        if 'xlabel' in data and data['xlabel'] in channel_list:\n",
    "            matching_nodes.append({\n",
    "                'pipeline': os.path.basename(dot_file).split('_')[0],\n",
    "                'node': node,\n",
    "                'operator_label': data['xlabel'],\n",
    "                'operator': 'queue channel' if data['xlabel'] in queue_channels else 'value channel'\n",
    "            })\n",
    "    return matching_nodes\n",
    "\n",
    "def count_nodes_by_labels(dot_file, channel_list):\n",
    "    matching_nodes = extract_xlabels_from_dot_file(dot_file, channel_list)\n",
    "    return len(matching_nodes), matching_nodes\n",
    "\n",
    "def iterate_over_dot_files(directory_path, queue_channels, value_channels):\n",
    "    all_queue_nodes = []\n",
    "    all_value_nodes = []\n",
    "    results_list = []\n",
    "\n",
    "    for filename in os.listdir(directory_path):\n",
    "        if filename.endswith(\".dot\"):\n",
    "            dot_file_path = os.path.join(directory_path, filename)\n",
    "            pipeline_name = os.path.basename(filename).split('_')[0]\n",
    "            \n",
    "            queue_count, queue_matching_nodes = count_nodes_by_labels(dot_file_path, queue_channels)\n",
    "            all_queue_nodes.extend(queue_matching_nodes)\n",
    "\n",
    "            value_count, value_matching_nodes = count_nodes_by_labels(dot_file_path, value_channels)\n",
    "            all_value_nodes.extend(value_matching_nodes)\n",
    "\n",
    "    results_list.extend(all_queue_nodes)\n",
    "    results_list.extend(all_value_nodes)\n",
    "\n",
    "    return results_list\n",
    "\n",
    "directory_path = './dags'\n",
    "\n",
    "df = pd.read_csv('./data/nextflow_operators.csv')\n",
    "queue_channels = df[df['operator_type'] == 'queue']['operator'].tolist()\n",
    "value_channels = df[df['operator_type'] == 'value']['operator'].tolist()\n",
    "\n",
    "results_list = iterate_over_dot_files(directory_path, queue_channels, value_channels)\n",
    "\n",
    "csv_operator_path = './results/nextflow_operators_dags.csv'\n",
    "header = ['pipeline', 'node', 'operator_label', 'operator']\n",
    "\n",
    "with open(csv_operator_path, 'w', newline='') as csv_file:\n",
    "    writer = csv.DictWriter(csv_file, fieldnames=header)\n",
    "    writer.writeheader()\n",
    "    writer.writerows(results_list)\n",
    "\n",
    "print(f'Results written to: {csv_operator_path}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "\n",
    "df_nextflow_op = pd.read_csv('./results/nextflow_operators_dags.csv')\n",
    "\n",
    "grouped = df.groupby('operator_type')['operator'].count()\n",
    "print(grouped)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "operator_counts = df_nextflow_op['operator_label'].value_counts()\n",
    "\n",
    "print('Distinct Labels and Their Sum of Counts:')\n",
    "print(operator_counts)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "\n",
    "df_operators = pd.read_csv('./results/nextflow_operators_dags.csv')\n",
    "\n",
    "\n",
    "queue_operators = df_operators[df_operators['operator'] == 'queue channel']\n",
    "value_operators = df_operators[df_operators['operator'] == 'value channel']\n",
    "\n",
    "\n",
    "print(f'Sum of queue operator nodes in DAGs {len(queue_operators)}')\n",
    "print(f'Sum of value operator nodes in DAGs {len(value_operators)}')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Nextflow operators in processes (in GitHub repositories)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import re\n",
    "import csv\n",
    "import pandas as pd\n",
    "\n",
    "def find_operators_in_processes(directory, pattern_list, output_file_name):\n",
    "    result_list = []\n",
    "\n",
    "    for pipeline in os.listdir(directory):\n",
    "        subdirectory_path = os.path.join(directory, pipeline)\n",
    "        folder_name = os.path.basename(subdirectory_path)\n",
    "\n",
    "        for root, dirs, files in os.walk(subdirectory_path):\n",
    "            for filename in files:\n",
    "                file_path = os.path.join(root, filename)\n",
    "                \n",
    "                if filename.endswith('.nf') or filename.endswith('.config'):\n",
    "                    with open(file_path, 'r', encoding='utf-8', errors='ignore') as file:\n",
    "                        lines = file.readlines()\n",
    "                    \n",
    "                    process_flag = False\n",
    "                    process_name = None\n",
    "                    brace_stack = []\n",
    "\n",
    "                    process_pattern = r'workflow\\s+(\\w+)\\s*{'\n",
    "\n",
    "                    for line_num, line in enumerate(lines):\n",
    "                        stripped_line = line.strip()\n",
    "                        \n",
    "                        if line.strip().startswith('#') or line.strip().startswith('//') or line.strip().startswith('*'):\n",
    "                                continue\n",
    "\n",
    "                        if re.search(process_pattern, stripped_line):\n",
    "                            process_flag = True\n",
    "                            process_match = re.search(process_pattern, stripped_line)\n",
    "                            if process_match:\n",
    "                                process_name = process_match.group(1)\n",
    "                            print(f'Process {process_name} in filepath {file_path}')\n",
    "                            brace_stack.append('{')\n",
    "\n",
    "                        if '{' in stripped_line:\n",
    "                            brace_stack.append('{')\n",
    "\n",
    "                        if '}' in stripped_line:\n",
    "                            if brace_stack:\n",
    "                                brace_stack.pop()\n",
    "                                if not brace_stack and process_flag:\n",
    "                                    process_flag = False\n",
    "\n",
    "                        if process_flag:\n",
    "                            pattern_found = False\n",
    "                            for pattern in pattern_list:\n",
    "                                regex_pattern = re.compile(r'\\.' + re.escape(pattern) + r'(?!\\w)' , re.IGNORECASE)\n",
    "                                if regex_pattern.search(stripped_line):\n",
    "                                    result_list.append([folder_name, process_name, pattern, file_path, line_num+1, stripped_line])\n",
    "                                    print(f'Nextflow operator {pattern} with content: {stripped_line}')\n",
    "                                    pattern_found = True\n",
    "                                    break\n",
    "                            if not pattern_found:\n",
    "                                result_list.append([folder_name, process_name, 'none', file_path, 'none', 'none'])\n",
    "\n",
    "                    if brace_stack:\n",
    "                        print(\"Error: Unclosed process block in file:\", file_path)\n",
    "\n",
    "    with open(f'./results/{output_file_name}.csv', 'w', newline='', encoding='utf-8') as csvfile:\n",
    "        writer = csv.writer(csvfile)\n",
    "        writer.writerow(['pipeline', 'workflow', 'operator', 'file_path', 'line_number', 'line_content'])\n",
    "        writer.writerows(result_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Process NFCORE_AIRRFLOW in filepath ./git_repos\\airrflow\\main.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\airrflow\\main.nf\n",
      "Process ASSEMBLED_INPUT_CHECK in filepath ./git_repos\\airrflow\\subworkflows\\local\\assembled_input_check.nf\n",
      "Nextflow operator splitCsv with content: .splitCsv(header: true, sep:'\\t')\n",
      "Nextflow operator map with content: .map { get_meta(it) }\n",
      "Nextflow operator branch with content: .branch { it ->\n",
      "Nextflow operator set with content: .set{ ch_metadata }\n",
      "Error: Unclosed process block in file: ./git_repos\\airrflow\\subworkflows\\local\\assembled_input_check.nf\n",
      "Process BULK_QC_AND_FILTER in filepath ./git_repos\\airrflow\\subworkflows\\local\\bulk_qc_and_filter.nf\n",
      "Nextflow operator collect with content: ch_imgt.collect()\n",
      "Nextflow operator mix with content: ch_logs = ch_logs.mix(CHANGEO_CREATEGERMLINES.out.logs)\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(CHANGEO_CREATEGERMLINES.out.versions.ifEmpty(null))\n",
      "Nextflow operator collect with content: ch_imgt.collect()\n",
      "Nextflow operator mix with content: ch_logs = ch_logs.mix(REMOVE_CHIMERIC.out.logs)\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(REMOVE_CHIMERIC.out.versions.ifEmpty(null))\n",
      "Nextflow operator map with content: .map{ it -> [ it[1] ] }\n",
      "Nextflow operator collect with content: .collect()\n",
      "Nextflow operator mix with content: ch_logs = ch_logs.mix(DETECT_CONTAMINATION.out.logs)\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(DETECT_CONTAMINATION.out.versions.ifEmpty(null))\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(COLLAPSE_DUPLICATES.out.versions.ifEmpty(null))\n",
      "Nextflow operator mix with content: ch_logs = ch_logs.mix(COLLAPSE_DUPLICATES.out.logs)\n",
      "Error: Unclosed process block in file: ./git_repos\\airrflow\\subworkflows\\local\\bulk_qc_and_filter.nf\n",
      "Process CLONAL_ANALYSIS in filepath ./git_repos\\airrflow\\subworkflows\\local\\clonal_analysis.nf\n",
      "Nextflow operator map with content: ch_find_threshold = ch_repertoire.map{ it -> it[1] }\n",
      "Nextflow operator collect with content: .collect()\n",
      "Nextflow operator flatten with content: .flatten()\n",
      "Nextflow operator map with content: .map{ it -> it.getName().toString() }\n",
      "Nextflow operator dump with content: .dump(tag: 'ch_find_threshold_samplesheet')\n",
      "Nextflow operator collectFile with content: .collectFile(name: 'find_threshold_samplesheet.txt', newLine: true)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(FIND_CLONAL_THRESHOLD.out.versions)\n",
      "Nextflow operator splitText with content: .splitText( limit:1 ) { it.trim().toString() }\n",
      "Nextflow operator dump with content: .dump(tag: 'clone_threshold')\n",
      "Nextflow operator filter with content: .filter { it != 'NA'}\n",
      "Nextflow operator filter with content: .filter { it != 'NaN' }\n",
      "Nextflow operator ifEmpty with content: .ifEmpty { error \"Automatic clone_threshold is 'NA'. Consider setting --clonal_threshold manually.\"}\n",
      "Nextflow operator map with content: ch_find_threshold = ch_repertoire.map{ it -> it[1] }\n",
      "Nextflow operator collect with content: .collect()\n",
      "Nextflow operator flatten with content: .flatten()\n",
      "Nextflow operator map with content: .map{ it -> it.getName().toString() }\n",
      "Nextflow operator dump with content: .dump(tag: 'ch_find_threshold_samplesheet')\n",
      "Nextflow operator collectFile with content: .collectFile(name: 'find_threshold_samplesheet.txt', newLine: true)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(REPORT_THRESHOLD.out.versions)\n",
      "Nextflow operator map with content: ch_repertoire.map{ it -> [ it[0].\"${params.cloneby}\",\n",
      "Nextflow operator groupTuple with content: .groupTuple()\n",
      "Nextflow operator map with content: .map{ get_meta_tabs(it) }\n",
      "Nextflow operator set with content: .set{ ch_define_clones }\n",
      "Nextflow operator collect with content: clone_threshold.collect(),\n",
      "Nextflow operator collect with content: ch_imgt.collect(),\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(DEFINE_CLONES_COMPUTE.out.versions)\n",
      "Nextflow operator collect with content: .collect()\n",
      "Nextflow operator map with content: .map { it -> [ [id:'all_reps'], it ] }\n",
      "Nextflow operator set with content: .set{ch_all_repertoires_cloned}\n",
      "Nextflow operator map with content: ch_all_repertoires_cloned_samplesheet = ch_all_repertoires_cloned.map{ it -> it[1] }\n",
      "Nextflow operator collect with content: .collect()\n",
      "Nextflow operator flatten with content: .flatten()\n",
      "Nextflow operator map with content: .map{ it -> it.getName().toString() }\n",
      "Nextflow operator dump with content: .dump(tag: 'ch_all_repertoires_cloned_samplesheet')\n",
      "Nextflow operator collectFile with content: .collectFile(name: 'all_repertoires_cloned_samplesheet.txt', newLine: true)\n",
      "Nextflow operator collect with content: clone_threshold.collect(),\n",
      "Nextflow operator collect with content: ch_imgt.collect(),\n",
      "Nextflow operator flatten with content: .flatten()\n",
      "Nextflow operator map with content: .map { it -> [ [id: \"${it.baseName}\".replaceFirst(\"__clone-pass\", \"\")], it ] }\n",
      "Nextflow operator set with content: .set{ch_repertoires_cloned}\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(DOWSER_LINEAGES.out.versions)\n",
      "Process FASTQ_INPUT_CHECK in filepath ./git_repos\\airrflow\\subworkflows\\local\\fastq_input_check.nf\n",
      "Nextflow operator splitCsv with content: .splitCsv ( header:true, sep:'\\t' )\n",
      "Nextflow operator map with content: .map { create_fastq_channels(it) }\n",
      "Nextflow operator set with content: .set { ch_reads }\n",
      "Error: Unclosed process block in file: ./git_repos\\airrflow\\subworkflows\\local\\fastq_input_check.nf\n",
      "Process PRESTO_SANS_UMI in filepath ./git_repos\\airrflow\\subworkflows\\local\\presto_sans_umi.nf\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(FASTP.out.versions.ifEmpty([]))\n",
      "Nextflow operator map with content: ch_gunzip = FASTP.out.reads.map{ meta,reads -> [meta, reads[0], reads[1]] }\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(GUNZIP_SANS_UMI.out.versions.ifEmpty(null))\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(PRESTO_ASSEMBLEPAIRS_SANS_UMI.out.versions.ifEmpty(null))\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(PRESTO_FILTERSEQ_POSTASSEMBLY_SANS_UMI.out.versions.ifEmpty(null))\n",
      "Nextflow operator collect with content: ch_cprimers.collect(),\n",
      "Nextflow operator collect with content: ch_vprimers.collect()\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(PRESTO_MASKPRIMERS_POSTASSEMBLY_SANS_UMI.out.versions.ifEmpty(null))\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(FASTQC_POSTASSEMBLY_SANS_UMI.out.versions.ifEmpty(null))\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(PRESTO_PARSEHEADERS_PRIMERS_SANS_UMI.out.versions.ifEmpty(null))\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(PRESTO_PARSEHEADERS_METADATA_SANS_UMI.out.versions.ifEmpty(null))\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(PRESTO_COLLAPSESEQ_SANS_UMI.out.versions.ifEmpty(null))\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(PRESTO_SPLITSEQ_SANS_UMI.out.versions.ifEmpty(null))\n",
      "Nextflow operator collect with content: fastp_reads_json = FASTP.out.json.collect{ meta,json -> json }\n",
      "Nextflow operator collect with content: fastp_reads_html = FASTP.out.html.collect{ meta,html -> html }\n",
      "Nextflow operator collect with content: presto_assemblepairs_logs = PRESTO_ASSEMBLEPAIRS_SANS_UMI.out.logs.collect()\n",
      "Nextflow operator collect with content: presto_maskprimers_logs = PRESTO_MASKPRIMERS_POSTASSEMBLY_SANS_UMI.out.logs.collect()\n",
      "Nextflow operator collect with content: presto_collapseseq_logs = PRESTO_COLLAPSESEQ_SANS_UMI.out.logs.collect()\n",
      "Nextflow operator collect with content: presto_splitseq_logs = PRESTO_SPLITSEQ_SANS_UMI.out.logs.collect()\n",
      "Error: Unclosed process block in file: ./git_repos\\airrflow\\subworkflows\\local\\presto_sans_umi.nf\n",
      "Process PRESTO_UMI in filepath ./git_repos\\airrflow\\subworkflows\\local\\presto_umi.nf\n",
      "Nextflow operator map with content: ch_reads.map{ meta, reads -> [meta, [reads[0], reads[1]]] }\n",
      "Nextflow operator set with content: .set{ ch_reads_R1_R2 }\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(FASTP.out.versions.ifEmpty([]))\n",
      "Nextflow operator map with content: .map{ meta, reads -> [meta.id, meta, reads[0], reads[1]] }\n",
      "Nextflow operator map with content: .map{ meta, reads -> [meta.id, meta, reads[2]] }\n",
      "Nextflow operator join with content: ch_meta_R1_R2_index = ch_meta_R1_R2.join( ch_meta_index )\n",
      "Nextflow operator map with content: .map{ id, meta1, R1, R2, meta2, index -> [ meta1, R1, R2, index ] }\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(MERGE_UMI.out.versions.ifEmpty(null))\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(FASTP.out.versions.ifEmpty([]))\n",
      "Nextflow operator map with content: ch_rename_fastq_umi = FASTP.out.reads.map{ meta,reads -> [meta, reads[0], reads[1]] }\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(GUNZIP_UMI.out.versions.ifEmpty(null))\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(PRESTO_FILTERSEQ_UMI.out.versions.ifEmpty(null))\n",
      "Nextflow operator collect with content: ch_cprimers.collect(),\n",
      "Nextflow operator collect with content: ch_vprimers.collect()\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(PRESTO_MASKPRIMERS_UMI.out.versions.ifEmpty(null))\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(PRESTO_PAIRSEQ_UMI.out.versions.ifEmpty(null))\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(PRESTO_CLUSTERSETS_UMI.out.versions.ifEmpty(null))\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(PRESTO_PARSE_CLUSTER_UMI.out.versions.ifEmpty(null))\n",
      "Nextflow operator collect with content: ch_clustersets_logs = PRESTO_CLUSTERSETS_UMI.out.logs.collect()\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(PRESTO_BUILDCONSENSUS_UMI.out.versions.ifEmpty(null))\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(PRESTO_POSTCONSENSUS_PAIRSEQ_UMI.out.versions.ifEmpty(null))\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(PRESTO_ASSEMBLEPAIRS_UMI.out.versions.ifEmpty(null))\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(FASTQC_POSTASSEMBLY_UMI.out.versions.ifEmpty(null))\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(PRESTO_PARSEHEADERS_COLLAPSE_UMI.out.versions.ifEmpty(null))\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(PRESTO_PARSEHEADERS_PRIMERS_UMI.out.versions.ifEmpty(null))\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(PRESTO_PARSEHEADERS_METADATA_UMI.out.versions.ifEmpty(null))\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(PRESTO_COLLAPSESEQ_UMI.out.versions.ifEmpty(null))\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(PRESTO_SPLITSEQ_UMI.out.versions.ifEmpty(null))\n",
      "Nextflow operator collect with content: fastp_reads_json = FASTP.out.json.collect{ meta,json -> json }\n",
      "Nextflow operator collect with content: fastp_reads_html = FASTP.out.html.collect{ meta,html -> html }\n",
      "Nextflow operator collect with content: presto_maskprimers_logs = PRESTO_MASKPRIMERS_UMI.out.logs.collect()\n",
      "Nextflow operator collect with content: presto_pairseq_logs = PRESTO_PAIRSEQ_UMI.out.logs.collect()\n",
      "Nextflow operator collect with content: presto_buildconsensus_logs = PRESTO_BUILDCONSENSUS_UMI.out.logs.collect()\n",
      "Nextflow operator collect with content: presto_postconsensus_pairseq_logs = PRESTO_POSTCONSENSUS_PAIRSEQ_UMI.out.logs.collect()\n",
      "Nextflow operator collect with content: presto_assemblepairs_logs = PRESTO_ASSEMBLEPAIRS_UMI.out.logs.collect()\n",
      "Nextflow operator collect with content: presto_collapseseq_logs = PRESTO_COLLAPSESEQ_UMI.out.logs.collect()\n",
      "Nextflow operator collect with content: presto_splitseq_logs = PRESTO_SPLITSEQ_UMI.out.logs.collect()\n",
      "Error: Unclosed process block in file: ./git_repos\\airrflow\\subworkflows\\local\\presto_umi.nf\n",
      "Process REPERTOIRE_ANALYSIS_REPORTING in filepath ./git_repos\\airrflow\\subworkflows\\local\\repertoire_analysis_reporting.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(PARSE_LOGS.out.versions)\n",
      "Nextflow operator mix with content: ch_logs = ch_vdj_annotation_logs.mix(ch_bulk_qc_and_filter_logs,\n",
      "Nextflow operator collect with content: ch_logs_tabs =  ch_logs.collect()\n",
      "Nextflow operator flatten with content: .flatten()\n",
      "Nextflow operator map with content: .map{ it -> it.getName().toString() }\n",
      "Nextflow operator dump with content: .dump(tag: 'ch_logs_tabs')\n",
      "Nextflow operator collectFile with content: .collectFile(name: 'all_logs_tabs.txt', newLine: true)\n",
      "Nextflow operator collect with content: ch_logs.collect().ifEmpty([]),\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(REPORT_FILE_SIZE.out.versions)\n",
      "Nextflow operator collect with content: ch_parsed_logs.collect().ifEmpty([]),\n",
      "Nextflow operator ifEmpty with content: REPORT_FILE_SIZE.out.table.ifEmpty([]),\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(AIRRFLOW_REPORT.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\airrflow\\subworkflows\\local\\repertoire_analysis_reporting.nf\n",
      "Process SEQUENCE_ASSEMBLY in filepath ./git_repos\\airrflow\\subworkflows\\local\\sequence_assembly.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(FASTQ_INPUT_CHECK.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ch_presto_software)\n",
      "Error: Unclosed process block in file: ./git_repos\\airrflow\\subworkflows\\local\\sequence_assembly.nf\n",
      "Process SINGLE_CELL_QC_AND_FILTERING in filepath ./git_repos\\airrflow\\subworkflows\\local\\single_cell_qc_and_filtering.nf\n",
      "Nextflow operator map with content: .map{ it -> [   it[0].id,\n",
      "Nextflow operator set with content: .set{ch_onlymeta}\n",
      "Nextflow operator map with content: .map { it -> it[1]}\n",
      "Nextflow operator collect with content: .collect()\n",
      "Nextflow operator set with content: .set{ch_repertoire_allsc}\n",
      "Nextflow operator flatten with content: .flatten()\n",
      "Nextflow operator map with content: .map { it -> [ \"${it.baseName}\".replaceFirst(\"__scqc-pass\", \"\"), it ] }\n",
      "Nextflow operator set with content: .set{ch_repertoire_after_scqc_with_sampleid}\n",
      "Nextflow operator mix with content: ch_logs = ch_logs.mix(SINGLE_CELL_QC.out.logs)\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(SINGLE_CELL_QC.out.versions.ifEmpty(null))\n",
      "Nextflow operator join with content: ch_repertoire_after_scqc_withmeta = ch_onlymeta.join(ch_repertoire_after_scqc_with_sampleid)\n",
      "Nextflow operator map with content: .map{ it -> [ it[1], it[2] ]}\n",
      "Error: Unclosed process block in file: ./git_repos\\airrflow\\subworkflows\\local\\single_cell_qc_and_filtering.nf\n",
      "Process VDJ_ANNOTATION in filepath ./git_repos\\airrflow\\subworkflows\\local\\vdj_annotation.nf\n",
      "Nextflow operator ifEmpty with content: .ifEmpty{ error \"IGBLAST DB not found: ${params.igblast_base}\" }\n",
      "Nextflow operator set with content: .set { ch_igblast_zipped }\n",
      "Nextflow operator collect with content: UNZIP_IGBLAST( ch_igblast_zipped.collect() )\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(UNZIP_IGBLAST.out.versions.ifEmpty(null))\n",
      "Nextflow operator ifEmpty with content: .ifEmpty { error \"IGBLAST DB not found: ${params.igblast_base}\" }\n",
      "Nextflow operator set with content: .set { ch_igblast }\n",
      "Nextflow operator ifEmpty with content: .ifEmpty{ error \"IMGTDB not found: ${params.imgtdb_base}\" }\n",
      "Nextflow operator set with content: .set { ch_imgt_zipped }\n",
      "Nextflow operator collect with content: UNZIP_IMGT( ch_imgt_zipped.collect() )\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(UNZIP_IMGT.out.versions.ifEmpty(null))\n",
      "Nextflow operator ifEmpty with content: .ifEmpty { error \"IMGT DB not found: ${params.imgtdb_base}\" }\n",
      "Nextflow operator set with content: .set { ch_imgt }\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(FETCH_DATABASES.out.versions.ifEmpty(null))\n",
      "Nextflow operator collect with content: ch_igblast.collect()\n",
      "Nextflow operator mix with content: ch_logs = ch_logs.mix(CHANGEO_ASSIGNGENES.out.logs)\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(CHANGEO_ASSIGNGENES.out.versions.ifEmpty(null))\n",
      "Nextflow operator collect with content: ch_imgt.collect()\n",
      "Nextflow operator mix with content: ch_logs = ch_logs.mix(CHANGEO_MAKEDB.out.logs)\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(CHANGEO_MAKEDB.out.versions.ifEmpty(null))\n",
      "Nextflow operator mix with content: ch_logs = ch_logs.mix(FILTER_QUALITY.out.logs)\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(FILTER_QUALITY.out.versions.ifEmpty(null))\n",
      "Nextflow operator mix with content: ch_logs = ch_logs.mix(CHANGEO_PARSEDB_SPLIT.out.logs)\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(CHANGEO_PARSEDB_SPLIT.out.versions.ifEmpty(null))\n",
      "Nextflow operator mix with content: ch_logs = ch_logs.mix(FILTER_JUNCTION_MOD3.out.logs)\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(FILTER_JUNCTION_MOD3.out.versions.ifEmpty(null))\n",
      "Nextflow operator ifEmpty with content: ch_repertoire = FILTER_JUNCTION_MOD3.out.tab.ifEmpty(null)\n",
      "Nextflow operator ifEmpty with content: ch_repertoire = FILTER_QUALITY.out.tab.ifEmpty(null)\n",
      "Nextflow operator mix with content: ch_logs = ch_logs.mix(ADD_META_TO_TAB.out.logs)\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(ADD_META_TO_TAB.out.versions.ifEmpty(null))\n",
      "Error: Unclosed process block in file: ./git_repos\\airrflow\\subworkflows\\local\\vdj_annotation.nf\n",
      "Process AIRRFLOW in filepath ./git_repos\\airrflow\\workflows\\airrflow.nf\n",
      "Nextflow operator mix with content: ch_versions                 = ch_versions.mix(SEQUENCE_ASSEMBLY.out.versions)\n",
      "Nextflow operator collect with content: ch_validated_samplesheet    = SEQUENCE_ASSEMBLY.out.samplesheet.collect()\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix( ASSEMBLED_INPUT_CHECK.out.versions.ifEmpty([]) )\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(CHANGEO_CONVERTDB_FASTA_FROM_AIRR.out.versions.ifEmpty([]))\n",
      "Nextflow operator mix with content: ch_reassign_logs = ch_reassign_logs.mix(CHANGEO_CONVERTDB_FASTA_FROM_AIRR.out.logs)\n",
      "Nextflow operator mix with content: ch_fasta = ASSEMBLED_INPUT_CHECK.out.ch_fasta.mix(ch_fasta_from_tsv)\n",
      "Nextflow operator collect with content: ch_validated_samplesheet = ASSEMBLED_INPUT_CHECK.out.validated_input.collect()\n",
      "Nextflow operator collect with content: ch_validated_samplesheet.collect()\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix( VDJ_ANNOTATION.out.versions.ifEmpty([]))\n",
      "Nextflow operator branch with content: .branch { it ->\n",
      "Nextflow operator dump with content: .dump(tag: 'bulk')\n",
      "Nextflow operator collect with content: VDJ_ANNOTATION.out.imgt.collect()\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix( BULK_QC_AND_FILTER.out.versions.ifEmpty([]))\n",
      "Nextflow operator dump with content: .dump(tag: 'single')\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix( SINGLE_CELL_QC_AND_FILTERING.out.versions.ifEmpty([]) )\n",
      "Nextflow operator mix with content: .mix(SINGLE_CELL_QC_AND_FILTERING.out.repertoires)\n",
      "Nextflow operator dump with content: .dump(tag: 'sc bulk mix')\n",
      "Nextflow operator collect with content: VDJ_ANNOTATION.out.imgt.collect(),\n",
      "Nextflow operator collect with content: ch_report_logo_img.collect().ifEmpty([])\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix( CLONAL_ANALYSIS.out.versions.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_presto_filterseq_logs.collect().ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_presto_maskprimers_logs.collect().ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_presto_pairseq_logs.collect().ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_presto_clustersets_logs.collect().ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_presto_buildconsensus_logs.collect().ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_presto_postconsensus_pairseq_logs.collect().ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_presto_assemblepairs_logs.collect().ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_presto_collapseseq_logs.collect().ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_presto_splitseq_logs.collect().ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_reassign_logs.collect().ifEmpty([]),\n",
      "Nextflow operator collect with content: VDJ_ANNOTATION.out.changeo_makedb_logs.collect().ifEmpty([]),\n",
      "Nextflow operator collect with content: VDJ_ANNOTATION.out.logs.collect().ifEmpty([]),\n",
      "Nextflow operator collect with content: BULK_QC_AND_FILTER.out.logs.collect().ifEmpty([]),\n",
      "Nextflow operator collect with content: SINGLE_CELL_QC_AND_FILTERING.out.logs.collect().ifEmpty([]),\n",
      "Nextflow operator collect with content: CLONAL_ANALYSIS.out.logs.collect().ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_input.collect(),\n",
      "Nextflow operator collect with content: ch_report_rmd.collect(),\n",
      "Nextflow operator collect with content: ch_report_css.collect(),\n",
      "Nextflow operator collect with content: ch_report_logo.collect(),\n",
      "Nextflow operator collect with content: ch_validated_samplesheet.collect()\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix( REPERTOIRE_ANALYSIS_REPORTING.out.versions )\n",
      "Nextflow operator dump with content: ch_versions.dump(tag: \"channel_versions\")\n",
      "Nextflow operator collectFile with content: ch_versions.unique().collectFile(name: 'collated_versions.yml')\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_workflow_summary.collectFile(name: 'workflow_summary_mqc.yaml'))\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_methods_description.collectFile(name: 'methods_description_mqc.yaml'))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(CUSTOM_DUMPSOFTWAREVERSIONS.out.mqc_yml.collect())\n",
      "Nextflow operator ifEmpty with content: ch_multiqc_files = ch_multiqc_files.mix(ch_fastp_html.ifEmpty([]))\n",
      "Nextflow operator ifEmpty with content: ch_multiqc_files = ch_multiqc_files.mix(ch_fastp_json.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(ch_fastqc_postassembly_mqc.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files.collect(),\n",
      "Nextflow operator collect with content: ch_multiqc_config.collect(),\n",
      "Nextflow operator collect with content: ch_multiqc_custom_config.collect().ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_report_logo.collect().ifEmpty([])\n",
      "Nextflow operator toList with content: multiqc_report = MULTIQC.out.report.toList()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Error: Unclosed process block in file: ./git_repos\\airrflow\\workflows\\airrflow.nf\n",
      "Process NFCORE_AMPLISEQ in filepath ./git_repos\\ampliseq\\main.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\ampliseq\\main.nf\n",
      "Process CUTADAPT_WORKFLOW in filepath ./git_repos\\ampliseq\\subworkflows\\local\\cutadapt_workflow.nf\n",
      "Nextflow operator set with content: CUTADAPT_BASIC ( ch_file ).reads.set { ch_trimmed_reads }\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator groupTuple with content: .groupTuple(by: 0 )\n",
      "Nextflow operator set with content: .set { ch_cutadapt_logs }\n",
      "Nextflow operator set with content: CUTADAPT_READTHROUGH ( ch_trimmed_reads ).reads.set { ch_trimmed_reads }\n",
      "Nextflow operator set with content: CUTADAPT_DOUBLEPRIMER ( ch_trimmed_reads ).reads.set { ch_trimmed_reads }\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator groupTuple with content: .groupTuple(by: 0 )\n",
      "Nextflow operator set with content: .set { ch_cutadapt_doubleprimer_logs }\n",
      "Nextflow operator combine with content: ch_summaries = CUTADAPT_SUMMARY_STD.out.tsv.combine( CUTADAPT_SUMMARY_DOUBLEPRIMER.out.tsv )\n",
      "Nextflow operator branch with content: .branch {\n",
      "Nextflow operator countFastq with content: failed: it[0].single_end ? it[1].countFastq() < params.min_read_counts : it[1][0].countFastq() < params.min_read_counts || it[1][1].countFastq() < params.min_read_counts\n",
      "Nextflow operator set with content: .set { ch_trimmed_reads_result }\n",
      "Nextflow operator set with content: ch_trimmed_reads_result.passed.set { ch_trimmed_reads_passed }\n",
      "Nextflow operator map with content: .map { meta, reads -> [ meta.id ] }\n",
      "Nextflow operator collect with content: .collect()\n",
      "Nextflow operator subscribe with content: .subscribe {\n",
      "Nextflow operator join with content: samples = it.join(\"\\n\")\n",
      "Error: Unclosed process block in file: ./git_repos\\ampliseq\\subworkflows\\local\\cutadapt_workflow.nf\n",
      "Process DADA2_PREPROCESSING in filepath ./git_repos\\ampliseq\\subworkflows\\local\\dada2_preprocessing.nf\n",
      "Nextflow operator map with content: .map { meta, reads -> [ reads ] }\n",
      "Nextflow operator collect with content: .collect()\n",
      "Nextflow operator map with content: .map { reads -> [ \"single_end\", reads ] }\n",
      "Nextflow operator set with content: .set { ch_all_trimmed_reads }\n",
      "Nextflow operator map with content: .map { meta, reads -> [ reads[0] ] }\n",
      "Nextflow operator collect with content: .collect()\n",
      "Nextflow operator map with content: .map { reads -> [ \"FW\", reads ] }\n",
      "Nextflow operator set with content: .set { ch_all_trimmed_fw }\n",
      "Nextflow operator map with content: .map { meta, reads -> [ reads[1] ] }\n",
      "Nextflow operator collect with content: .collect()\n",
      "Nextflow operator map with content: .map { reads -> [ \"RV\", reads ] }\n",
      "Nextflow operator set with content: .set { ch_all_trimmed_rv }\n",
      "Nextflow operator mix with content: .mix ( ch_all_trimmed_rv )\n",
      "Nextflow operator set with content: .set { ch_all_trimmed_reads }\n",
      "Nextflow operator dump with content: DADA2_QUALITY1 ( ch_all_trimmed_reads.dump(tag: 'into_dada2_quality') )\n",
      "Nextflow operator mix with content: ch_versions_dada2_preprocessing = ch_versions_dada2_preprocessing.mix(DADA2_QUALITY1.out.versions)\n",
      "Nextflow operator subscribe with content: DADA2_QUALITY1.out.warning.subscribe { if ( it.baseName.toString().startsWith(\"WARNING\") ) log.warn it.baseName.toString().replace(\"WARNING \",\"DADA2_QUALITY1: \") }\n",
      "Nextflow operator toSortedList with content: .toSortedList()\n",
      "Nextflow operator set with content: .set { ch_trunc }\n",
      "Nextflow operator first with content: ch_versions_dada2_preprocessing = ch_versions_dada2_preprocessing.mix(TRUNCLEN.out.versions.first())\n",
      "Nextflow operator subscribe with content: ch_trunc.subscribe {\n",
      "Nextflow operator toInteger with content: if ( \"${it[0][1]}\".toInteger() + \"${it[1][1]}\".toInteger() <= 10 ) { log.warn \"`--trunclenf` was set to ${it[0][1]} and `--trunclenr` to ${it[1][1]}, this is too low! Please either change `--trunc_qmin` (and `--trunc_rmin`), or set `--trunclenf` and `--trunclenr`.\" }\n",
      "Nextflow operator toInteger with content: else if ( \"${it[0][1]}\".toInteger() <= 10 ) { log.warn \"`--trunclenf` was set to ${it[0][1]}, this is too low! Please either change `--trunc_qmin` (and `--trunc_rmin`), or set `--trunclenf` and `--trunclenr`.\" }\n",
      "Nextflow operator toInteger with content: else if ( \"${it[1][1]}\".toInteger() <= 10 ) { log.warn \"`--trunclenr` was set to ${it[1][1]}, this is too low! Please either change `--trunc_qmin` (and `--trunc_rmin`), or set `--trunclenf` and `--trunclenr`.\" }\n",
      "Nextflow operator toSortedList with content: .toSortedList()\n",
      "Nextflow operator set with content: .set { ch_trunc }\n",
      "Nextflow operator combine with content: ch_trimmed_reads.combine(ch_trunc).set { ch_trimmed_reads }\n",
      "Nextflow operator dump with content: DADA2_FILTNTRIM ( ch_trimmed_reads.dump(tag: 'into_filtntrim')  )\n",
      "Nextflow operator first with content: ch_versions_dada2_preprocessing = ch_versions_dada2_preprocessing.mix(DADA2_FILTNTRIM.out.versions.first())\n",
      "Nextflow operator branch with content: .branch {\n",
      "Nextflow operator countFastq with content: failed: it[0].single_end ? it[1].countFastq() < params.min_read_counts : it[1][0].countFastq() < params.min_read_counts || it[1][1].countFastq() < params.min_read_counts\n",
      "Nextflow operator set with content: .set { ch_dada2_filtntrim_results }\n",
      "Nextflow operator set with content: ch_dada2_filtntrim_results.passed.set { ch_dada2_filtntrim_results_passed }\n",
      "Nextflow operator map with content: .map { meta, reads, logs, args -> [ meta.id ] }\n",
      "Nextflow operator collect with content: .collect()\n",
      "Nextflow operator subscribe with content: .subscribe {\n",
      "Nextflow operator join with content: samples = it.join(\"\\n\")\n",
      "Nextflow operator map with content: .map{ meta, reads, logs, args -> [meta, reads] }\n",
      "Nextflow operator set with content: .set{ ch_dada2_filtntrim_reads_passed }\n",
      "Nextflow operator map with content: .map{ meta, reads, logs, args -> [meta, logs] }\n",
      "Nextflow operator set with content: .set{ ch_dada2_filtntrim_logs_passed }\n",
      "Nextflow operator map with content: .map{ meta, reads, logs, args -> args }\n",
      "Nextflow operator set with content: .set{ ch_dada2_filtntrim_args_passed }\n",
      "Nextflow operator map with content: .map { meta, reads -> [ reads ] }\n",
      "Nextflow operator collect with content: .collect()\n",
      "Nextflow operator map with content: .map { reads -> [ \"single_end\", reads ] }\n",
      "Nextflow operator set with content: .set { ch_all_preprocessed_reads }\n",
      "Nextflow operator map with content: .map { meta, reads -> [ reads[0] ] }\n",
      "Nextflow operator collect with content: .collect()\n",
      "Nextflow operator map with content: .map { reads -> [ \"FW\", reads ] }\n",
      "Nextflow operator set with content: .set { ch_all_preprocessed_fw }\n",
      "Nextflow operator map with content: .map { meta, reads -> [ reads[1] ] }\n",
      "Nextflow operator collect with content: .collect()\n",
      "Nextflow operator map with content: .map { reads -> [ \"RV\", reads ] }\n",
      "Nextflow operator set with content: .set { ch_all_preprocessed_rv }\n",
      "Nextflow operator mix with content: .mix ( ch_all_preprocessed_rv )\n",
      "Nextflow operator set with content: .set { ch_all_preprocessed_reads }\n",
      "Nextflow operator dump with content: DADA2_QUALITY2 ( ch_all_preprocessed_reads.dump(tag: 'into_dada2_quality2') )\n",
      "Nextflow operator subscribe with content: DADA2_QUALITY2.out.warning.subscribe { if ( it.baseName.toString().startsWith(\"WARNING\") ) log.warn it.baseName.toString().replace(\"WARNING \",\"DADA2_QUALITY2: \") }\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator groupTuple with content: .groupTuple(by: 0 )\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator flatten with content: meta.id = ids.flatten().sort()\n",
      "Nextflow operator flatten with content: [ meta, reads.flatten().sort() ] }\n",
      "Nextflow operator set with content: .set { ch_filt_reads }\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator groupTuple with content: .groupTuple(by: 0 )\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator flatten with content: meta.id = ids.flatten().sort()\n",
      "Nextflow operator flatten with content: [ meta, reads.flatten().sort() ] }\n",
      "Nextflow operator set with content: .set { ch_filt_logs }\n",
      "Nextflow operator collect with content: qc_svg              = ch_DADA2_QUALITY1_SVG.collect()\n",
      "Nextflow operator collect with content: qc_svg_preprocessed = ch_DADA2_QUALITY2_SVG.collect()\n",
      "Error: Unclosed process block in file: ./git_repos\\ampliseq\\subworkflows\\local\\dada2_preprocessing.nf\n",
      "Process DADA2_TAXONOMY_WF in filepath ./git_repos\\ampliseq\\subworkflows\\local\\dada2_taxonomy_wf.nf\n",
      "Nextflow operator map with content: ch_shinfo = Channel.fromList(params.dada_ref_databases[params.dada_ref_taxonomy][\"shfile\"]).map { file(it) }\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator set with content: .set { ch_assigntax }\n",
      "Nextflow operator map with content: .map { meta, db -> db }\n",
      "Nextflow operator set with content: .set { ch_assigntax }\n",
      "Nextflow operator mix with content: ch_versions_dada_taxonomy = ch_versions_dada_taxonomy.mix(DADA2_TAXONOMY.out.versions)\n",
      "Nextflow operator ifEmpty with content: ch_versions_dada_taxonomy = ch_versions_dada_taxonomy.mix( FORMAT_TAXRESULTS_STD.out.versions.ifEmpty(null) )\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator set with content: .set { ch_fasta_map }\n",
      "Nextflow operator ifEmpty with content: ch_versions_dada_taxonomy = ch_versions_dada_taxonomy.mix(VSEARCH_USEARCHGLOBAL.out.versions.ifEmpty(null))\n",
      "Nextflow operator collect with content: ASSIGNSH( ch_dada2_tax1, ch_shinfo.collect(), VSEARCH_USEARCHGLOBAL.out.txt, ASV_SH_name + \".${val_dada_ref_taxonomy}.tsv\")\n",
      "Nextflow operator ifEmpty with content: ch_versions_dada_taxonomy = ch_versions_dada_taxonomy.mix(ASSIGNSH.out.versions.ifEmpty(null))\n",
      "Error: Unclosed process block in file: ./git_repos\\ampliseq\\subworkflows\\local\\dada2_taxonomy_wf.nf\n",
      "Process KRAKEN2_TAXONOMY_WF in filepath ./git_repos\\ampliseq\\subworkflows\\local\\kraken2_taxonomy_wf.nf\n",
      "Nextflow operator branch with content: .branch {\n",
      "Nextflow operator set with content: }.set { ch_kraken2_ref_taxonomy }\n",
      "Nextflow operator subscribe with content: ch_kraken2_ref_taxonomy.failed.subscribe { error \"$it is neither a directory nor a file that ends in '.tar.gz' or '.tgz'. Please review input.\" }\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator map with content: ch_kraken2db = UNTAR.out.untar.map{ it[1] }\n",
      "Nextflow operator mix with content: ch_kraken2db = ch_kraken2db.mix(ch_kraken2_ref_taxonomy.dir)\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator set with content: .set { ch_fasta_kraken2 }\n",
      "Nextflow operator mix with content: ch_versions_kraken2_taxonomy = ch_versions_kraken2_taxonomy.mix(KRAKEN2_KRAKEN2.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\ampliseq\\subworkflows\\local\\kraken2_taxonomy_wf.nf\n",
      "Process PARSE_INPUT in filepath ./git_repos\\ampliseq\\subworkflows\\local\\parse_input.nf\n",
      "Nextflow operator ifEmpty with content: .ifEmpty { error(\"${error_message}\") }\n",
      "Nextflow operator map with content: .map { read ->\n",
      "Nextflow operator take with content: meta.id           = read.baseName.toString().indexOf(\"_\") != -1 ? read.baseName.toString().take(read.baseName.toString().indexOf(\"_\")) : read.baseName\n",
      "Nextflow operator take with content: meta.run          = multiple_sequencing_runs ? read.take(read.findLastIndexOf{\"/\"})[-1] : \"1\"\n",
      "Nextflow operator set with content: .set { ch_reads }\n",
      "Nextflow operator ifEmpty with content: .ifEmpty { error(\"${error_message}\") }\n",
      "Nextflow operator map with content: .map { name, reads ->\n",
      "Nextflow operator take with content: meta.id           = name.toString().indexOf(\"_\") != -1 ? name.toString().take(name.toString().indexOf(\"_\")) : name\n",
      "Nextflow operator take with content: meta.run          = multiple_sequencing_runs ? reads[0].take(reads[0].findLastIndexOf{\"/\"})[-1] : \"1\"\n",
      "Nextflow operator set with content: .set { ch_reads }\n",
      "Nextflow operator flatMap with content: .flatMap { meta, reads -> [ meta.run ] }\n",
      "Nextflow operator unique with content: .unique()\n",
      "Nextflow operator set with content: .set { ch_folders }\n",
      "Nextflow operator collect with content: .collect()\n",
      "Nextflow operator subscribe with content: .subscribe {\n",
      "Nextflow operator count with content: .count()\n",
      "Nextflow operator subscribe with content: .subscribe { if ( it == 1 ) error(\"Found only one folder with read data but \\\"--multiple_sequencing_runs\\\" was specified. Please review data input.\") }\n",
      "Nextflow operator map with content: .map { meta, reads -> [ meta.id ] }\n",
      "Nextflow operator toList with content: .toList()\n",
      "Nextflow operator subscribe with content: .subscribe {\n",
      "Nextflow operator unique with content: if( it.size() != it.unique().size() ) {\n",
      "Nextflow operator take with content: ids = it.take(10);\n",
      "Nextflow operator map with content: .map { meta, reads -> meta.id }\n",
      "Nextflow operator subscribe with content: .subscribe { if ( \"$it\".contains(\".\") ) error(\"Please review data input, sampleIDs may not contain dots, but \\\"$it\\\" does.\") }\n",
      "Nextflow operator map with content: .map { meta, reads -> meta.id }\n",
      "Nextflow operator subscribe with content: .subscribe { if ( params.metadata && \"$it\"[0].isNumber() ) error(\"Please review data input, sampleIDs may not start with a number, but \\\"$it\\\" does. The pipeline unintentionally modifies such strings and the metadata will not match any more.\") }\n",
      "Error: Unclosed process block in file: ./git_repos\\ampliseq\\subworkflows\\local\\parse_input.nf\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Process PHYLOSEQ_WORKFLOW in filepath ./git_repos\\ampliseq\\subworkflows\\local\\phyloseq_workflow.nf\n",
      "Nextflow operator first with content: ch_phyloseq_inmeta = ch_meta.first() // The .first() is to make sure it's a value channel\n",
      "Nextflow operator first with content: ch_phyloseq_intree = ch_tree.map { it = it[1] }.first()\n",
      "Nextflow operator combine with content: PHYLOSEQ ( ch_tax.combine(ch_phyloseq_inasv), ch_phyloseq_inmeta, ch_phyloseq_intree )\n",
      "Error: Unclosed process block in file: ./git_repos\\ampliseq\\subworkflows\\local\\phyloseq_workflow.nf\n",
      "Process QIIME2_ANCOM in filepath ./git_repos\\ampliseq\\subworkflows\\local\\qiime2_ancom.nf\n",
      "Nextflow operator combine with content: .combine( ch_asv )\n",
      "Nextflow operator combine with content: .combine( ch_metacolumn_all )\n",
      "Nextflow operator set with content: .set{ ch_for_filtersamples }\n",
      "Nextflow operator combine with content: .combine( QIIME2_FILTERSAMPLES_ANCOM.out.qza )\n",
      "Nextflow operator combine with content: .combine( ch_tax )\n",
      "Nextflow operator combine with content: .combine( ch_taxlevel )\n",
      "Nextflow operator set with content: .set{ ch_for_ancom_tax }\n",
      "Nextflow operator subscribe with content: QIIME2_ANCOM_TAX.out.ancom.subscribe { if ( it.baseName[0].toString().startsWith(\"WARNING\") ) log.warn it.baseName[0].toString().replace(\"WARNING \",\"QIIME2_ANCOM_TAX: \") }\n",
      "Nextflow operator combine with content: QIIME2_ANCOM_ASV ( ch_metadata.combine( QIIME2_FILTERSAMPLES_ANCOM.out.qza.flatten() ) )\n",
      "Nextflow operator mix with content: ancom = QIIME2_ANCOM_ASV.out.ancom.mix(QIIME2_ANCOM_TAX.out.ancom)\n",
      "Error: Unclosed process block in file: ./git_repos\\ampliseq\\subworkflows\\local\\qiime2_ancom.nf\n",
      "Process QIIME2_BARPLOTAVG in filepath ./git_repos\\ampliseq\\subworkflows\\local\\qiime2_barplotavg.nf\n",
      "Nextflow operator combine with content: .combine(ch_metadata)\n",
      "Nextflow operator combine with content: .combine(ch_metadata_category_barplot)\n",
      "Error: Unclosed process block in file: ./git_repos\\ampliseq\\subworkflows\\local\\qiime2_barplotavg.nf\n",
      "Process QIIME2_DIVERSITY in filepath ./git_repos\\ampliseq\\subworkflows\\local\\qiime2_diversity.nf\n",
      "Nextflow operator subscribe with content: QIIME2_DIVERSITY_CORE.out.depth.subscribe { if ( it.baseName.toString().startsWith(\"WARNING\") ) log.warn it.baseName.toString().replace(\"WARNING \",\"QIIME2_DIVERSITY_CORE: \") }\n",
      "Nextflow operator combine with content: .combine( QIIME2_DIVERSITY_CORE.out.vector.flatten() )\n",
      "Nextflow operator set with content: .set{ ch_to_diversity_alpha }\n",
      "Nextflow operator combine with content: .combine( QIIME2_DIVERSITY_CORE.out.distance.flatten() )\n",
      "Nextflow operator combine with content: .combine( ch_metacolumn_pairwise )\n",
      "Nextflow operator set with content: .set{ ch_to_diversity_beta }\n",
      "Nextflow operator combine with content: .combine( QIIME2_DIVERSITY_CORE.out.distance.flatten() )\n",
      "Nextflow operator combine with content: .combine( ch_qiime_adonis_formula )\n",
      "Nextflow operator set with content: .set{ ch_to_diversity_beta }\n",
      "Nextflow operator combine with content: .combine( QIIME2_DIVERSITY_CORE.out.pcoa.flatten() )\n",
      "Nextflow operator set with content: .set{ ch_to_diversity_betaord }\n",
      "Error: Unclosed process block in file: ./git_repos\\ampliseq\\subworkflows\\local\\qiime2_diversity.nf\n",
      "Process QIIME2_EXPORT in filepath ./git_repos\\ampliseq\\subworkflows\\local\\qiime2_export.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\ampliseq\\subworkflows\\local\\qiime2_export.nf\n",
      "Process QIIME2_PREPTAX in filepath ./git_repos\\ampliseq\\subworkflows\\local\\qiime2_preptax.nf\n",
      "Nextflow operator flatten with content: ch_qiime_ref_taxonomy.flatten()\n",
      "Nextflow operator branch with content: .branch {\n",
      "Nextflow operator set with content: }.set { ch_qiime_ref_tax_branched }\n",
      "Nextflow operator subscribe with content: ch_qiime_ref_tax_branched.failed.subscribe { error \"$it is neither a compressed (ends with `.gz`) or decompressed sequence (ends with `.fna`) or taxonomy file (ends with `.tax`). Please review input.\" }\n",
      "Nextflow operator mix with content: ch_qiime2_preptax_versions = ch_qiime2_preptax_versions.mix(PIGZ_UNCOMPRESS.out.versions)\n",
      "Nextflow operator mix with content: ch_qiime_db_files = ch_qiime_db_files.mix(ch_qiime_ref_tax_branched.decompressed)\n",
      "Nextflow operator filter with content: ch_ref_database_fna = ch_qiime_db_files.filter {\n",
      "Nextflow operator filter with content: ch_ref_database_tax = ch_qiime_db_files.filter {\n",
      "Nextflow operator combine with content: ch_ref_database = ch_ref_database_fna.combine(ch_ref_database_tax)\n",
      "Nextflow operator flatten with content: ch_qiime_ref_taxonomy.flatten()\n",
      "Nextflow operator branch with content: .branch {\n",
      "Nextflow operator set with content: }.set { ch_qiime_ref_tax_branched }\n",
      "Nextflow operator subscribe with content: ch_qiime_ref_tax_branched.failed.subscribe { error \"$it is neither a directory nor a file that ends in '.tar.gz' or '.tgz'. Please review input.\" }\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator mix with content: ch_qiime2_preptax_versions = ch_qiime2_preptax_versions.mix(UNTAR.out.versions)\n",
      "Nextflow operator map with content: ch_qiime_db_dir = UNTAR.out.untar.map{ it[1] }\n",
      "Nextflow operator mix with content: ch_qiime_db_dir = ch_qiime_db_dir.mix(ch_qiime_ref_tax_branched.dir)\n",
      "Nextflow operator map with content: ch_ref_database_fna = ch_qiime_db_dir.map{ dir ->\n",
      "Nextflow operator map with content: ch_ref_database_tax = ch_qiime_db_dir.map{ dir ->\n",
      "Nextflow operator combine with content: ch_ref_database = ch_ref_database_fna.combine(ch_ref_database_tax)\n",
      "Nextflow operator mix with content: ch_qiime2_preptax_versions = ch_qiime2_preptax_versions.mix(FORMAT_TAXONOMY_QIIME.out.versions)\n",
      "Nextflow operator combine with content: ch_ref_database = FORMAT_TAXONOMY_QIIME.out.fasta.combine(FORMAT_TAXONOMY_QIIME.out.tax)\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator set with content: .set { ch_ref_database }\n",
      "Nextflow operator mix with content: ch_qiime2_preptax_versions = ch_qiime2_preptax_versions.mix(QIIME2_EXTRACT.out.versions)\n",
      "Nextflow operator mix with content: ch_qiime2_preptax_versions = ch_qiime2_preptax_versions.mix(QIIME2_TRAIN.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\ampliseq\\subworkflows\\local\\qiime2_preptax.nf\n",
      "Process QIIME2_TAXONOMY in filepath ./git_repos\\ampliseq\\subworkflows\\local\\qiime2_taxonomy.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\ampliseq\\subworkflows\\local\\qiime2_taxonomy.nf\n",
      "Process SINTAX_TAXONOMY_WF in filepath ./git_repos\\ampliseq\\subworkflows\\local\\sintax_taxonomy_wf.nf\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator set with content: .set { ch_fasta_sintax }\n",
      "Nextflow operator mix with content: ch_versions_sintax_taxonomy = ch_versions_sintax_taxonomy.mix(VSEARCH_SINTAX.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\ampliseq\\subworkflows\\local\\sintax_taxonomy_wf.nf\n",
      "Process FASTA_NEWICK_EPANG_GAPPA in filepath ./git_repos\\ampliseq\\subworkflows\\nf-core\\fasta_newick_epang_gappa\\main.nf\n",
      "Nextflow operator filter with content: ch_hmmer_data = ch_pp_data.filter { it.data.alignmethod == 'hmmer' }\n",
      "Nextflow operator filter with content: ch_mafft_data = ch_pp_data.filter { it.data.alignmethod == 'mafft' }\n",
      "Nextflow operator filter with content: .filter { ! it.data.hmmfile }\n",
      "Nextflow operator map with content: .map { [ it.meta, it.data.refseqfile ] },\n",
      "Nextflow operator map with content: .mix(HMMER_HMMBUILD.out.hmm.map { [ it[0], it[1] ] })\n",
      "Nextflow operator mix with content: .mix(\n",
      "Nextflow operator filter with content: .filter { it.data.hmmfile }\n",
      "Nextflow operator map with content: .map { [ it.meta, it.data.hmmfile ] }\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(HMMER_HMMBUILD.out.versions.first())\n",
      "Nextflow operator filter with content: .filter { ! it.data.hmmfile }\n",
      "Nextflow operator map with content: .map { [ it.meta, it.data.refseqfile ] }\n",
      "Nextflow operator map with content: .mix(HMMER_UNALIGNREF.out.seqreformated.map { [ it[0], it[1] ] })\n",
      "Nextflow operator mix with content: .mix(\n",
      "Nextflow operator filter with content: .filter { it.data.hmmfile }\n",
      "Nextflow operator map with content: .map { [ it.meta, it.data.refseqfile ] }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(HMMER_UNALIGNREF.out.versions)\n",
      "Nextflow operator mix with content: .mix(ch_hmmer_unaligned)\n",
      "Nextflow operator groupTuple with content: .groupTuple(size: 2, sort: { a, b -> a =~ /\\.hmm/ ? 1 : -1 })\n",
      "Nextflow operator map with content: ch_hmmer_alignref.map { [ it[0], it[1][0] ] },\n",
      "Nextflow operator map with content: ch_hmmer_alignref.map { it[1][1] }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(HMMER_HMMALIGNREF.out.versions)\n",
      "Nextflow operator map with content: .mix(ch_hmmer_data.map { [ it.meta, it.data.queryseqfile ] })\n",
      "Nextflow operator mix with content: .mix(ch_hmm)\n",
      "Nextflow operator groupTuple with content: .groupTuple(size: 2, sort: { a, b -> a =~ /\\.hmm/ ? 1 : -1 })\n",
      "Nextflow operator map with content: ch_hmmer_alignquery.map { [ it[0], it[1][0] ] },\n",
      "Nextflow operator map with content: ch_hmmer_alignquery.map { it[1][1] }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(HMMER_HMMALIGNQUERY.out.versions)\n",
      "Nextflow operator map with content: HMMER_MASKREF ( HMMER_HMMALIGNREF.out.sthlm.map { [ it[0], it[1], [], [], [], [], [], [] ] }, [] )\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(HMMER_MASKREF.out.versions)\n",
      "Nextflow operator map with content: HMMER_MASKQUERY ( HMMER_HMMALIGNQUERY.out.sthlm.map { [ it[0], it[1], [], [], [], [], [], [] ] }, [] )\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(HMMER_MASKQUERY.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(HMMER_AFAFORMATREF.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(HMMER_AFAFORMATQUERY.out.versions)\n",
      "Nextflow operator map with content: ch_mafft_data.map { [ it.meta, it.data.refseqfile ] },\n",
      "Nextflow operator map with content: ch_mafft_data.map { [ it.data.queryseqfile ] }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(MAFFT.out.versions)\n",
      "Nextflow operator map with content: ch_mafft_data.map { [ it.meta, it.data.refseqfile ] }\n",
      "Nextflow operator join with content: .join(MAFFT.out.fas)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(EPANG_SPLIT.out.versions)\n",
      "Nextflow operator map with content: ch_epang_query = ch_pp_data.map { [ it.meta, it.data.model, it.data.refphylogeny ] }\n",
      "Nextflow operator join with content: .join ( HMMER_AFAFORMATQUERY.out.seqreformated )\n",
      "Nextflow operator join with content: .join ( HMMER_AFAFORMATREF.out.seqreformated )\n",
      "Nextflow operator mix with content: .mix(\n",
      "Nextflow operator map with content: ch_pp_data.map { [ it.meta, it.data.model, it.data.refphylogeny ] }\n",
      "Nextflow operator join with content: .join(EPANG_SPLIT.out.query.map { [ it[0], it[1] ] } )\n",
      "Nextflow operator join with content: .join(EPANG_SPLIT.out.reference.map { [ it[0], it[1] ] } )\n",
      "Nextflow operator map with content: .map { [ [ id:it[0].id, model:it[1] ], it[3], it[4], it[2] ] }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(EPANG_PLACE.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GAPPA_GRAFT.out.versions)\n",
      "Nextflow operator map with content: .map { [ [ id:it[0].id ], it[1] ] }\n",
      "Nextflow operator join with content: .join( ch_pp_data.map { [ it.meta, it.data.taxonomy ] } )\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GAPPA_ASSIGN.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GAPPA_HEATTREE.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\ampliseq\\subworkflows\\nf-core\\fasta_newick_epang_gappa\\main.nf\n",
      "Process AMPLISEQ in filepath ./git_repos\\ampliseq\\workflows\\ampliseq.nf\n",
      "Nextflow operator map with content: .map{ meta, readfw, readrv ->\n",
      "Nextflow operator dump with content: ch_input_reads.dump(tag:'ch_input_reads')\n",
      "Nextflow operator branch with content: .branch {\n",
      "Nextflow operator countFastq with content: failed: it[0].single_end ? it[1].countFastq() < params.min_read_counts : it[1][0].countFastq() < params.min_read_counts || it[1][1].countFastq() < params.min_read_counts\n",
      "Nextflow operator set with content: .set { ch_reads_result }\n",
      "Nextflow operator set with content: ch_reads_result.passed.set { ch_reads }\n",
      "Nextflow operator map with content: .map { meta, reads -> [ meta.id ] }\n",
      "Nextflow operator collect with content: .collect()\n",
      "Nextflow operator subscribe with content: .subscribe {\n",
      "Nextflow operator join with content: samples = it.join(\"\\n\")\n",
      "Nextflow operator dump with content: ch_reads.dump(tag: 'ch_reads')\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(RENAME_RAW_DATA_FILES.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(FASTQC.out.versions.first())\n",
      "Nextflow operator set with content: ).reads.set { ch_trimmed_reads }\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(CUTADAPT_WORKFLOW.out.versions.first())\n",
      "Nextflow operator set with content: ).reads.set { ch_filt_reads }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(DADA2_PREPROCESSING.out.versions)\n",
      "Nextflow operator join with content: .join( ch_errormodel )\n",
      "Nextflow operator set with content: .set { ch_derep_errormodel }\n",
      "Nextflow operator dump with content: DADA2_DENOISING ( ch_derep_errormodel.dump(tag: 'into_denoising')  )\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(DADA2_DENOISING.out.versions.first())\n",
      "Nextflow operator join with content: .join( DADA2_DENOISING.out.denoised )\n",
      "Nextflow operator join with content: .join( DADA2_DENOISING.out.mergers )\n",
      "Nextflow operator join with content: .join( DADA2_RMCHIMERA.out.rds )\n",
      "Nextflow operator set with content: .set { ch_track_numbers }\n",
      "Nextflow operator collect with content: DADA2_STATS.out.stats.map { meta, stats -> stats }.collect(),\n",
      "Nextflow operator collect with content: DADA2_RMCHIMERA.out.rds.map { meta, rds -> rds }.collect() )\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(VSEARCH_CLUSTER.out.versions.ifEmpty(null))\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(FILTER_CLUSTERS.out.versions.ifEmpty(null))\n",
      "Nextflow operator collect with content: BARRNAPSUMMARY ( BARRNAP.out.gff.collect() )\n",
      "Nextflow operator subscribe with content: BARRNAPSUMMARY.out.warning.subscribe {\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(BARRNAP.out.versions.ifEmpty(null))\n",
      "Nextflow operator ifEmpty with content: FILTER_SSU ( ch_unfiltered_fasta, ch_dada2_asv.ifEmpty( [] ), BARRNAPSUMMARY.out.summary )\n",
      "Nextflow operator collect with content: BARRNAPSUMMARY ( BARRNAP.out.gff.collect() )\n",
      "Nextflow operator subscribe with content: BARRNAPSUMMARY.out.warning.subscribe { if ( it.baseName.toString().startsWith(\"WARNING\") ) log.warn \"Barrnap could not identify any rRNA in the ASV sequences. We recommended to use the --skip_barrnap option for these sequences.\" }\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(BARRNAP.out.versions.ifEmpty(null))\n",
      "Nextflow operator ifEmpty with content: FILTER_LEN_ASV ( ch_dada2_fasta, ch_dada2_asv.ifEmpty( [] ) )\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(FILTER_LEN_ASV.out.versions.ifEmpty(null))\n",
      "Nextflow operator countLines with content: ch_dada2_fasta.subscribe { if (it.countLines() == 0) error(\"ASV length filtering activated by '--min_len_asv' or '--max_len_asv' removed all ASVs, please adjust settings.\") }\n",
      "Nextflow operator ifEmpty with content: FILTER_CODONS ( ch_dada2_fasta, ch_dada2_asv.ifEmpty( [] ) )\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(FILTER_CODONS.out.versions.ifEmpty(null))\n",
      "Nextflow operator countLines with content: ch_dada2_fasta.subscribe { if (it.countLines() == 0) error(\"ASV codon filtering activated by '--filter_codons' removed all ASVs, please adjust settings.\") }\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(ITSX_CUTASV.out.versions.ifEmpty(null))\n",
      "Nextflow operator collect with content: FORMAT_TAXONOMY ( ch_dada_ref_taxonomy.collect(), val_dada_ref_taxonomy )\n",
      "Nextflow operator set with content: ).tax.set { ch_dada2_tax }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(DADA2_TAXONOMY_WF.out.versions)\n",
      "Nextflow operator map with content: ch_tax_for_phyloseq = ch_tax_for_phyloseq.mix ( ch_dada2_tax.map { it = [ \"dada2\", file(it) ] } )\n",
      "Nextflow operator set with content: ).qiime2_tsv.set { ch_kraken2_tax }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(KRAKEN2_TAXONOMY_WF.out.versions)\n",
      "Nextflow operator map with content: ch_tax_for_phyloseq = ch_tax_for_phyloseq.mix ( ch_kraken2_tax.map { it = [ \"kraken2\", file(it) ] } )\n",
      "Nextflow operator collect with content: ch_sintax_ref_taxonomy.collect(),\n",
      "Nextflow operator set with content: ).tax.set { ch_sintax_tax }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SINTAX_TAXONOMY_WF.out.versions)\n",
      "Nextflow operator map with content: ch_tax_for_phyloseq = ch_tax_for_phyloseq.mix ( ch_sintax_tax.map { it = [ \"sintax\", file(it) ] } )\n",
      "Nextflow operator map with content: ch_pp_data = ch_fasta.map { it ->\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix( FASTA_NEWICK_EPANG_GAPPA.out.versions )\n",
      "Nextflow operator map with content: ch_tax_for_phyloseq = ch_tax_for_phyloseq.mix ( PHYLOSEQ_INTAX_PPLACE ( ch_pplace_tax ).tsv.map { it = [ \"pplace\", file(it) ] } )\n",
      "Nextflow operator collect with content: ch_qiime_ref_taxonomy.collect(),\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix( QIIME2_TAXONOMY.out.versions.ifEmpty(null) ) //usually a .first() is here, dont know why this leads here to a warning\n",
      "Nextflow operator map with content: ch_tax_for_phyloseq = ch_tax_for_phyloseq.mix ( PHYLOSEQ_INTAX_QIIME2 ( ch_qiime2_tax ).tsv.map { it = [ \"qiime2\", file(it) ] } )\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix( FILTER_STATS.out.versions.ifEmpty(null) )\n",
      "Nextflow operator set with content: METADATA_PAIRWISE ( ch_metadata ).category.set { ch_metacolumn_pairwise }\n",
      "Nextflow operator flatten with content: ch_metacolumn_pairwise = ch_metacolumn_pairwise.splitCsv().flatten()\n",
      "Nextflow operator join with content: ch_metacolumn_pairwise = ch_metacolumn_all.join(ch_metacolumn_pairwise)\n",
      "Nextflow operator set with content: METADATA_ALL ( ch_metadata ).category.set { ch_metacolumn_all }\n",
      "Nextflow operator branch with content: ch_metacolumn_all.branch { passed: it != \"\" }.set { result }\n",
      "Nextflow operator flatten with content: ch_metacolumn_all = ch_metacolumn_all.splitCsv().flatten()\n",
      "Nextflow operator set with content: METADATA_PAIRWISE ( ch_metadata ).category.set { ch_metacolumn_pairwise }\n",
      "Nextflow operator flatten with content: ch_metacolumn_pairwise = ch_metacolumn_pairwise.splitCsv().flatten()\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(PICRUST.out.versions.ifEmpty(null))\n",
      "Nextflow operator ifEmpty with content: SBDIEXPORTREANNOTATE ( ch_sintax_tax, \"sintax\", db_version, params.cut_its, ch_barrnapsummary.ifEmpty([]) )\n",
      "Nextflow operator ifEmpty with content: SBDIEXPORTREANNOTATE ( ch_dada2_tax, \"dada2\", db_version, params.cut_its, ch_barrnapsummary.ifEmpty([]) )\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SBDIEXPORT.out.versions.first())\n",
      "Nextflow operator ifEmpty with content: ch_metadata.ifEmpty([]),\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(PHYLOSEQ_WORKFLOW.out.versions.first())\n",
      "Nextflow operator collectFile with content: ch_versions.unique().collectFile(name: 'collated_versions.yml')\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_workflow_summary.collectFile(name: 'workflow_summary_mqc.yaml'))\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_methods_description.collectFile(name: 'methods_description_mqc.yaml'))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(CUSTOM_DUMPSOFTWAREVERSIONS.out.mqc_yml.collect())\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(FASTQC.out.zip.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(CUTADAPT_WORKFLOW.out.logs.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files.collect(),\n",
      "Nextflow operator toList with content: ch_multiqc_config.toList(),\n",
      "Nextflow operator toList with content: ch_multiqc_custom_config.toList(),\n",
      "Nextflow operator toList with content: ch_multiqc_logo.toList()\n",
      "Nextflow operator toList with content: multiqc_report = MULTIQC.out.report.toList()\n",
      "Nextflow operator ifEmpty with content: ch_metadata.ifEmpty( [] ),\n",
      "Nextflow operator ifEmpty with content: ch_input_fasta.ifEmpty( [] ), // fasta input\n",
      "Nextflow operator collect with content: !params.input_fasta && !params.skip_fastqc && !params.skip_multiqc ? MULTIQC.out.plots : [], //.collect().flatten().collectFile(name: \"mqc_fastqc_per_sequence_quality_scores_plot_1.svg\")\n",
      "Nextflow operator collect with content: !params.skip_cutadapt ? CUTADAPT_WORKFLOW.out.summary.collect().ifEmpty( [] ) : [],\n",
      "Nextflow operator first with content: DADA2_PREPROCESSING.out.args.first().ifEmpty( [] ),\n",
      "Nextflow operator ifEmpty with content: !params.skip_dada_quality ? DADA2_PREPROCESSING.out.qc_svg.ifEmpty( [] ) : [],\n",
      "Nextflow operator ifEmpty with content: !params.skip_dada_quality ? DADA2_PREPROCESSING.out.qc_svg_preprocessed.ifEmpty( [] ) : [],\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator groupTuple with content: .groupTuple(by: 0 )\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator flatten with content: meta.run = runs.flatten()\n",
      "Nextflow operator flatten with content: [ meta, svgs.flatten() ]\n",
      "Nextflow operator ifEmpty with content: }.ifEmpty( [[],[]] ),\n",
      "Nextflow operator ifEmpty with content: DADA2_MERGE.out.asv.ifEmpty( [] ),\n",
      "Nextflow operator ifEmpty with content: ch_unfiltered_fasta.ifEmpty( [] ), // this is identical to DADA2_MERGE.out.fasta if !params.input_fasta\n",
      "Nextflow operator ifEmpty with content: DADA2_MERGE.out.dada2asv.ifEmpty( [] ),\n",
      "Nextflow operator ifEmpty with content: DADA2_MERGE.out.dada2stats.ifEmpty( [] ),\n",
      "Nextflow operator ifEmpty with content: params.vsearch_cluster ? FILTER_CLUSTERS.out.asv.ifEmpty( [] ) : [],\n",
      "Nextflow operator ifEmpty with content: !params.skip_barrnap ? BARRNAPSUMMARY.out.summary.ifEmpty( [] ) : [],\n",
      "Nextflow operator ifEmpty with content: params.filter_ssu ? FILTER_SSU.out.stats.ifEmpty( [] ) : [],\n",
      "Nextflow operator ifEmpty with content: params.filter_ssu ? FILTER_SSU.out.fasta.ifEmpty( [] ) : [],\n",
      "Nextflow operator ifEmpty with content: params.min_len_asv || params.max_len_asv ? FILTER_LEN_ASV.out.stats.ifEmpty( [] ) : [],\n",
      "Nextflow operator ifEmpty with content: params.min_len_asv || params.max_len_asv ? FILTER_LEN_ASV.out.len_orig.ifEmpty( [] ) : [],\n",
      "Nextflow operator ifEmpty with content: params.filter_codons ? FILTER_CODONS.out.fasta.ifEmpty( [] ) : [],\n",
      "Nextflow operator ifEmpty with content: params.filter_codons ? FILTER_CODONS.out.stats.ifEmpty( [] ) : [],\n",
      "Nextflow operator ifEmpty with content: params.cut_its != \"none\" ? ITSX_CUTASV.out.summary.ifEmpty( [] ) : [],\n",
      "Nextflow operator ifEmpty with content: !params.skip_taxonomy && params.dada_ref_taxonomy && !params.skip_dada_taxonomy ? ch_dada2_tax.ifEmpty( [] ) : [],\n",
      "Nextflow operator ifEmpty with content: !params.skip_taxonomy && params.dada_ref_taxonomy && !params.skip_dada_taxonomy ? DADA2_TAXONOMY_WF.out.cut_tax.ifEmpty( [[],[]] ) : [[],[]],\n",
      "Nextflow operator ifEmpty with content: !params.skip_taxonomy && params.sintax_ref_taxonomy ? ch_sintax_tax.ifEmpty( [] ) : [],\n",
      "Nextflow operator ifEmpty with content: !params.skip_taxonomy && ( params.kraken2_ref_taxonomy || params.kraken2_ref_tax_custom ) ? KRAKEN2_TAXONOMY_WF.out.tax_tsv.ifEmpty( [] ) : [],\n",
      "Nextflow operator ifEmpty with content: !params.skip_taxonomy && params.pplace_tree ? ch_pplace_tax.ifEmpty( [] ) : [],\n",
      "Nextflow operator ifEmpty with content: !params.skip_taxonomy && params.pplace_tree ? FASTA_NEWICK_EPANG_GAPPA.out.heattree.ifEmpty( [[],[]] ) : [[],[]],\n",
      "Nextflow operator ifEmpty with content: !params.skip_taxonomy && ( params.qiime_ref_taxonomy || params.qiime_ref_tax_custom || params.classifier ) && run_qiime2_taxonomy ? QIIME2_TAXONOMY.out.tsv.ifEmpty( [] ) : [],\n",
      "Nextflow operator countLines with content: run_qiime2 && ( params.exclude_taxa != \"none\" || params.min_frequency != 1 || params.min_samples != 1 ) ? ch_dada2_asv.countLines()+\",\"+QIIME2_FILTERTAXA.out.tsv.countLines() : \"\",\n",
      "Nextflow operator ifEmpty with content: run_qiime2 && ( params.exclude_taxa != \"none\" || params.min_frequency != 1 || params.min_samples != 1 ) ? FILTER_STATS.out.tsv.ifEmpty( [] ) : [],\n",
      "Nextflow operator ifEmpty with content: run_qiime2 && !params.skip_barplot ? QIIME2_BARPLOT.out.folder.ifEmpty( [] ) : [],\n",
      "Nextflow operator ifEmpty with content: run_qiime2 && !params.skip_abundance_tables ? QIIME2_EXPORT.out.abs_tsv.ifEmpty( [] ) : [],\n",
      "Nextflow operator ifEmpty with content: run_qiime2 && !params.skip_diversity_indices && params.metadata ? QIIME2_DIVERSITY.out.depth.ifEmpty( [] ) : [],\n",
      "Nextflow operator collect with content: run_qiime2 && !params.skip_diversity_indices && params.metadata ? QIIME2_DIVERSITY.out.alpha.collect().ifEmpty( [] ) : [],\n",
      "Nextflow operator collect with content: run_qiime2 && !params.skip_diversity_indices && params.metadata ? QIIME2_DIVERSITY.out.beta.collect().ifEmpty( [] ) : [],\n",
      "Nextflow operator collect with content: run_qiime2 && !params.skip_diversity_indices && params.metadata ? QIIME2_DIVERSITY.out.adonis.collect().ifEmpty( [] ) : [],\n",
      "Nextflow operator collect with content: run_qiime2 && !params.skip_ancom && params.metadata ? QIIME2_ANCOM.out.ancom.collect().ifEmpty( [] ) : [],\n",
      "Nextflow operator ifEmpty with content: params.picrust ? PICRUST.out.pathways.ifEmpty( [] ) : [],\n",
      "Nextflow operator collect with content: params.sbdiexport ? SBDIEXPORT.out.sbditables.mix(SBDIEXPORTREANNOTATE.out.sbdiannottables).collect().ifEmpty( [] ) : [],\n",
      "Nextflow operator collect with content: !params.skip_taxonomy ? PHYLOSEQ_WORKFLOW.out.rds.map{info,rds -> [rds]}.collect().ifEmpty( [] ) : []\n",
      "Nextflow operator mix with content: ch_versions    = ch_versions.mix(SUMMARY_REPORT.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\ampliseq\\workflows\\ampliseq.nf\n",
      "Process NFCORE_ATACSEQ in filepath ./git_repos\\atacseq\\main.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\atacseq\\main.nf\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Process ALIGN_STAR in filepath ./git_repos\\atacseq\\subworkflows\\local\\align_star.nf\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(STAR_ALIGN.out.versions.first())\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BAM_SORT_STATS_SAMTOOLS.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\atacseq\\subworkflows\\local\\align_star.nf\n",
      "Process BAM_BEDGRAPH_BIGWIG_BEDTOOLS_UCSC in filepath ./git_repos\\atacseq\\subworkflows\\local\\bam_bedgraph_bigwig_bedtools_ucsc.nf\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(BEDTOOLS_GENOMECOV.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(UCSC_BEDGRAPHTOBIGWIG.out.versions.first())\n",
      "Error: Unclosed process block in file: ./git_repos\\atacseq\\subworkflows\\local\\bam_bedgraph_bigwig_bedtools_ucsc.nf\n",
      "Process BAM_FILTER_BAMTOOLS in filepath ./git_repos\\atacseq\\subworkflows\\local\\bam_filter_bamtools.nf\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(BAMTOOLS_FILTER.out.versions.first())\n",
      "Nextflow operator branch with content: .branch {\n",
      "Nextflow operator set with content: .set { ch_bam }\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SAMTOOLS_INDEX.out.versions.first())\n",
      "Nextflow operator join with content: ch_bam.single_end.join(SAMTOOLS_INDEX.out.bai),\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(BAM_STATS_SAMTOOLS.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SAMTOOLS_SORT.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(BAM_REMOVE_ORPHANS.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(BAM_SORT_STATS_SAMTOOLS.out.versions.first())\n",
      "Nextflow operator mix with content: bam      = BAM_SORT_STATS_SAMTOOLS.out.bam.mix(ch_bam.single_end)                    // channel: [ val(meta), [ bam ] ]\n",
      "Nextflow operator mix with content: bai      = BAM_SORT_STATS_SAMTOOLS.out.bai.mix(SAMTOOLS_INDEX.out.bai)               // channel: [ val(meta), [ bai ] ]\n",
      "Nextflow operator mix with content: stats    = BAM_SORT_STATS_SAMTOOLS.out.stats.mix(BAM_STATS_SAMTOOLS.out.stats)       // channel: [ val(meta), [ stats ] ]\n",
      "Nextflow operator mix with content: flagstat = BAM_SORT_STATS_SAMTOOLS.out.flagstat.mix(BAM_STATS_SAMTOOLS.out.flagstat) // channel: [ val(meta), [ flagstat ] ]\n",
      "Nextflow operator mix with content: idxstats = BAM_SORT_STATS_SAMTOOLS.out.idxstats.mix(BAM_STATS_SAMTOOLS.out.idxstats) // channel: [ val(meta), [ idxstats ] ]\n",
      "Error: Unclosed process block in file: ./git_repos\\atacseq\\subworkflows\\local\\bam_filter_bamtools.nf\n",
      "Process BAM_PEAKS_CALL_QC_ANNOTATE_MACS2_HOMER in filepath ./git_repos\\atacseq\\subworkflows\\local\\bam_peaks_call_qc_annotate_macs2_homer.nf\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(MACS2_CALLPEAK.out.versions.first())\n",
      "Nextflow operator filter with content: .filter {\n",
      "Nextflow operator set with content: .set { ch_macs2_peaks }\n",
      "Nextflow operator join with content: .join(ch_macs2_peaks, by: [0])\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator set with content: .set { ch_bam_peaks }\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(FRIP_SCORE.out.versions.first())\n",
      "Nextflow operator join with content: .join(FRIP_SCORE.out.txt, by: [0])\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator set with content: .set { ch_bam_peak_frip }\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(MULTIQC_CUSTOM_PEAKS.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(HOMER_ANNOTATEPEAKS.out.versions.first())\n",
      "Nextflow operator collect with content: ch_macs2_peaks.collect{it[1]},\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(PLOT_MACS2_QC.out.versions)\n",
      "Nextflow operator collect with content: HOMER_ANNOTATEPEAKS.out.txt.collect{it[1]},\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(PLOT_HOMER_ANNOTATEPEAKS.out.versions)\n",
      "Nextflow operator count with content: peak_count_multiqc           = MULTIQC_CUSTOM_PEAKS.out.count   // channel: [ val(meta), [ counts ] ]\n",
      "Error: Unclosed process block in file: ./git_repos\\atacseq\\subworkflows\\local\\bam_peaks_call_qc_annotate_macs2_homer.nf\n",
      "Process BED_CONSENSUS_QUANTIFY_QC_BEDTOOLS_FEATURECOUNTS_DESEQ2 in filepath ./git_repos\\atacseq\\subworkflows\\local\\bed_consensus_quantify_qc_bedtools_featurecounts_deseq2.nf\n",
      "Nextflow operator collect with content: .collect { it[1] }\n",
      "Nextflow operator filter with content: .filter { it.size() > 1 }\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator set with content: .set { ch_consensus_peaks }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(MACS2_CONSENSUS.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(HOMER_ANNOTATEPEAKS.out.versions)\n",
      "Nextflow operator join with content: .join(ch_peaks)\n",
      "Nextflow operator collect with content: .collect { it[1] }\n",
      "Nextflow operator filter with content: .filter { it.size() > 1 }\n",
      "Nextflow operator map with content: .map { [ it ] }\n",
      "Nextflow operator concat with content: .concat(MACS2_CONSENSUS.out.saf)\n",
      "Nextflow operator collect with content: .collect()\n",
      "Nextflow operator filter with content: .filter { it.size() == 3 }\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator set with content: .set { ch_bam_saf }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SUBREAD_FEATURECOUNTS.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(DESEQ2_QC.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\atacseq\\subworkflows\\local\\bed_consensus_quantify_qc_bedtools_featurecounts_deseq2.nf\n",
      "Process BIGWIG_PLOT_DEEPTOOLS in filepath ./git_repos\\atacseq\\subworkflows\\local\\bigwig_plot_deeptools.nf\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(DEEPTOOLS_COMPUTEMATRIX_SCALE_REGIONS.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(DEEPTOOLS_COMPUTEMATRIX_REFERENCE_POINT.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(DEEPTOOLS_PLOTPROFILE.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(DEEPTOOLS_PLOTHEATMAP.out.versions.first())\n",
      "Error: Unclosed process block in file: ./git_repos\\atacseq\\subworkflows\\local\\bigwig_plot_deeptools.nf\n",
      "Process INPUT_CHECK in filepath ./git_repos\\atacseq\\subworkflows\\local\\input_check.nf\n",
      "Nextflow operator splitCsv with content: .splitCsv ( header:true, sep:',' )\n",
      "Nextflow operator map with content: .map { create_fastq_channel(it, seq_center) }\n",
      "Nextflow operator set with content: .set { reads }\n",
      "Error: Unclosed process block in file: ./git_repos\\atacseq\\subworkflows\\local\\input_check.nf\n",
      "Process PREPARE_GENOME in filepath ./git_repos\\atacseq\\subworkflows\\local\\prepare_genome.nf\n",
      "Nextflow operator map with content: ch_fasta    = GUNZIP_FASTA ( [ [:], params.fasta ] ).gunzip.map{ it[1] }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GUNZIP_FASTA.out.versions)\n",
      "Nextflow operator map with content: ch_gtf      = GUNZIP_GTF ( [ [:], params.gtf ] ).gunzip.map{ it[1] }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GUNZIP_GTF.out.versions)\n",
      "Nextflow operator map with content: ch_gff      = GUNZIP_GFF ( [ [:], params.gff ] ).gunzip.map{ it[1] }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GUNZIP_GFF.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GFFREAD.out.versions)\n",
      "Nextflow operator map with content: ch_blacklist = GUNZIP_BLACKLIST ( [ [:], params.blacklist ] ).gunzip.map{ it[1] }\n",
      "Nextflow operator mix with content: ch_versions  = ch_versions.mix(GUNZIP_BLACKLIST.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GTF2BED.out.versions)\n",
      "Nextflow operator map with content: ch_gene_bed = GUNZIP_GENE_BED ( [ [:], params.gene_bed ] ).gunzip.map{ it[1] }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GUNZIP_GENE_BED.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(TSS_EXTRACT.out.versions)\n",
      "Nextflow operator map with content: ch_tss_bed = GUNZIP_TSS_BED ( [ [:], params.tss_bed ] ).gunzip.map{ it[1] }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GUNZIP_TSS_BED.out.versions)\n",
      "Nextflow operator map with content: CUSTOM_GETCHROMSIZES ( ch_fasta.map { [ [:], it ] } )\n",
      "Nextflow operator map with content: ch_chrom_sizes = CUSTOM_GETCHROMSIZES.out.sizes.map { it[1] }\n",
      "Nextflow operator map with content: ch_fai         = CUSTOM_GETCHROMSIZES.out.fai.map{ it[1] }\n",
      "Nextflow operator mix with content: ch_versions    = ch_versions.mix(CUSTOM_GETCHROMSIZES.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GET_AUTOSOMES.out.versions)\n",
      "Nextflow operator ifEmpty with content: ch_blacklist.ifEmpty([]),\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GENOME_BLACKLIST_REGIONS.out.versions)\n",
      "Nextflow operator mix with content: ch_versions  = ch_versions.mix(UNTAR_BWA_INDEX.out.versions)\n",
      "Nextflow operator map with content: ch_bwa_index = BWA_INDEX ( ch_fasta.map { [ [:], it ] } ).index\n",
      "Nextflow operator mix with content: ch_versions  = ch_versions.mix(BWA_INDEX.out.versions)\n",
      "Nextflow operator mix with content: ch_versions  = ch_versions.mix(UNTAR_BOWTIE2_INDEX.out.versions)\n",
      "Nextflow operator map with content: ch_bowtie2_index = BOWTIE2_BUILD ( ch_fasta.map { [ [:], it ] } ).index\n",
      "Nextflow operator mix with content: ch_versions      = ch_versions.mix(BOWTIE2_BUILD.out.versions)\n",
      "Nextflow operator mix with content: ch_versions  = ch_versions.mix(UNTAR.out.versions)\n",
      "Nextflow operator map with content: ch_chromap_index = CHROMAP_INDEX ( ch_fasta.map { [ [:], it ] } ).index\n",
      "Nextflow operator mix with content: ch_versions  = ch_versions.mix(CHROMAP_INDEX.out.versions)\n",
      "Nextflow operator map with content: ch_star_index = UNTAR_STAR_INDEX ( [ [:], params.star_index ] ).untar.map{ it[1] }\n",
      "Nextflow operator mix with content: ch_versions   = ch_versions.mix(UNTAR_STAR_INDEX.out.versions)\n",
      "Nextflow operator mix with content: ch_versions   = ch_versions.mix(STAR_GENOMEGENERATE.out.versions)\n",
      "Nextflow operator map with content: ch_macs_gsize = KHMER_UNIQUEKMERS.out.kmers.map { it.text.trim() }\n",
      "Nextflow operator mix with content: ch_versions   = ch_versions.mix(KHMER_UNIQUEKMERS.out.versions)\n",
      "Nextflow operator ifEmpty with content: versions      = ch_versions.ifEmpty(null)     // channel: [ versions.yml ]\n",
      "Error: Unclosed process block in file: ./git_repos\\atacseq\\subworkflows\\local\\prepare_genome.nf\n",
      "Process BAM_MARKDUPLICATES_PICARD in filepath ./git_repos\\atacseq\\subworkflows\\nf-core\\bam_markduplicates_picard\\main.nf\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(PICARD_MARKDUPLICATES.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SAMTOOLS_INDEX.out.versions.first())\n",
      "Nextflow operator join with content: .join(SAMTOOLS_INDEX.out.bai, by: [0], remainder: true)\n",
      "Nextflow operator join with content: .join(SAMTOOLS_INDEX.out.csi, by: [0], remainder: true)\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BAM_STATS_SAMTOOLS.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\atacseq\\subworkflows\\nf-core\\bam_markduplicates_picard\\main.nf\n",
      "Process BAM_SORT_STATS_SAMTOOLS in filepath ./git_repos\\atacseq\\subworkflows\\nf-core\\bam_sort_stats_samtools\\main.nf\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SAMTOOLS_SORT.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SAMTOOLS_INDEX.out.versions.first())\n",
      "Nextflow operator join with content: .join(SAMTOOLS_INDEX.out.bai, by: [0], remainder: true)\n",
      "Nextflow operator join with content: .join(SAMTOOLS_INDEX.out.csi, by: [0], remainder: true)\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator set with content: .set { ch_bam_bai }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BAM_STATS_SAMTOOLS.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\atacseq\\subworkflows\\nf-core\\bam_sort_stats_samtools\\main.nf\n",
      "Process BAM_STATS_SAMTOOLS in filepath ./git_repos\\atacseq\\subworkflows\\nf-core\\bam_stats_samtools\\main.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_STATS.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_FLAGSTAT.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_IDXSTATS.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\atacseq\\subworkflows\\nf-core\\bam_stats_samtools\\main.nf\n",
      "Process FASTQ_ALIGN_BOWTIE2 in filepath ./git_repos\\atacseq\\subworkflows\\nf-core\\fastq_align_bowtie2\\main.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BOWTIE2_ALIGN.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BAM_SORT_STATS_SAMTOOLS.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\atacseq\\subworkflows\\nf-core\\fastq_align_bowtie2\\main.nf\n",
      "Process FASTQ_ALIGN_BWA in filepath ./git_repos\\atacseq\\subworkflows\\nf-core\\fastq_align_bwa\\main.nf\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(BWA_MEM.out.versions.first())\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BAM_SORT_STATS_SAMTOOLS.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\atacseq\\subworkflows\\nf-core\\fastq_align_bwa\\main.nf\n",
      "Process FASTQ_ALIGN_CHROMAP in filepath ./git_repos\\atacseq\\subworkflows\\nf-core\\fastq_align_chromap\\main.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(CHROMAP_CHROMAP.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BAM_SORT_STATS_SAMTOOLS.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\atacseq\\subworkflows\\nf-core\\fastq_align_chromap\\main.nf\n",
      "Process FASTQ_FASTQC_UMITOOLS_TRIMGALORE in filepath ./git_repos\\atacseq\\subworkflows\\nf-core\\fastq_fastqc_umitools_trimgalore\\main.nf\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(FASTQC.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(UMITOOLS_EXTRACT.out.versions.first())\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator set with content: .set { umi_reads }\n",
      "Nextflow operator first with content: ch_versions   = ch_versions.mix(TRIMGALORE.out.versions.first())\n",
      "Nextflow operator join with content: .join(trim_log, remainder: true)\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator set with content: .set { ch_num_trimmed_reads }\n",
      "Nextflow operator filter with content: .filter { meta, reads, num_reads -> num_reads >= min_trimmed_reads.toFloat() }\n",
      "Nextflow operator map with content: .map { meta, reads, num_reads -> [ meta, reads ] }\n",
      "Nextflow operator set with content: .set { trim_reads }\n",
      "Nextflow operator map with content: .map { meta, reads, num_reads -> [ meta, num_reads ] }\n",
      "Nextflow operator set with content: .set { trim_read_count }\n",
      "Nextflow operator ifEmpty with content: versions = ch_versions.ifEmpty(null) // channel: [ versions.yml ]\n",
      "Error: Unclosed process block in file: ./git_repos\\atacseq\\subworkflows\\nf-core\\fastq_fastqc_umitools_trimgalore\\main.nf\n",
      "Process ATACSEQ in filepath ./git_repos\\atacseq\\workflows\\atacseq.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(PREPARE_GENOME.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(INPUT_CHECK.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(FASTQ_FASTQC_UMITOOLS_TRIMGALORE.out.versions)\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(FASTQ_ALIGN_BWA.out.versions)\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(FASTQ_ALIGN_BOWTIE2.out.versions)\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(FASTQ_ALIGN_CHROMAP.out.versions)\n",
      "Nextflow operator map with content: .map {\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ALIGN_STAR.out.versions)\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator groupTuple with content: .groupTuple(by: [0])\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator flatten with content: [ meta, bam.flatten() ]\n",
      "Nextflow operator set with content: .set { ch_sort_bam }\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(PICARD_MERGESAMFILES_LIBRARY.out.versions.first())\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(MERGED_LIBRARY_MARKDUPLICATES_PICARD.out.versions)\n",
      "Nextflow operator join with content: MERGED_LIBRARY_MARKDUPLICATES_PICARD.out.bam.join(MERGED_LIBRARY_MARKDUPLICATES_PICARD.out.bai, by: [0]),\n",
      "Nextflow operator first with content: PREPARE_GENOME.out.filtered_bed.first(),\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(MERGED_LIBRARY_FILTER_BAM.out.versions)\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(MERGED_LIBRARY_PRESEQ_LCEXTRAP.out.versions.first())\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(MERGED_LIBRARY_PICARD_COLLECTMULTIPLEMETRICS.out.versions.first())\n",
      "Nextflow operator join with content: MERGED_LIBRARY_FILTER_BAM.out.bam.join(MERGED_LIBRARY_FILTER_BAM.out.flagstat, by: [0]),\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(MERGED_LIBRARY_BAM_TO_BIGWIG.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(MERGED_LIBRARY_BIGWIG_PLOT_DEEPTOOLS.out.versions)\n",
      "Nextflow operator join with content: .join(MERGED_LIBRARY_FILTER_BAM.out.bai, by: [0])\n",
      "Nextflow operator set with content: .set { ch_bam_bai }\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator set with content: .set { ch_control_bam_bai }\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator combine with content: .combine(ch_control_bam_bai, by: 0)\n",
      "Nextflow operator map with content: .map { it -> [ it[1] , it[2] + it[4], it[3] + it[5] ] }\n",
      "Nextflow operator set with content: .set { ch_bam_bai }\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(MERGED_LIBRARY_DEEPTOOLS_PLOTFINGERPRINT.out.versions.first())\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator set with content: .set { ch_bam_library }\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator set with content: .set { ch_bam_library }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(MERGED_LIBRARY_CALL_ANNOTATE_PEAKS.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(MERGED_LIBRARY_CONSENSUS_PEAKS.out.versions)\n",
      "Nextflow operator join with content: .join(MERGED_LIBRARY_MARKDUPLICATES_PICARD.out.bai, by: [0])\n",
      "Nextflow operator join with content: .join(MERGED_LIBRARY_CALL_ANNOTATE_PEAKS.out.peaks, by: [0])\n",
      "Nextflow operator set with content: .set { ch_bam_peaks }\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(MERGED_LIBRARY_ATAQV_ATAQV.out.versions.first())\n",
      "Nextflow operator collect with content: MERGED_LIBRARY_ATAQV_ATAQV.out.json.collect{it[1]}\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(MERGED_LIBRARY_ATAQV_MKARV.out.versions)\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator groupTuple with content: .groupTuple()\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator set with content: .set { ch_merged_library_replicate_bam }\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(PICARD_MERGESAMFILES_REPLICATE.out.versions.first())\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(MERGED_REPLICATE_MARKDUPLICATES_PICARD.out.versions)\n",
      "Nextflow operator join with content: MERGED_REPLICATE_MARKDUPLICATES_PICARD.out.bam.join(MERGED_REPLICATE_MARKDUPLICATES_PICARD.out.flagstat, by: [0]),\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(MERGED_REPLICATE_BAM_TO_BIGWIG.out.versions)\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator set with content: .set { ch_bam_merged_control }\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator combine with content: .combine( ch_bam_merged_control, by: 0)\n",
      "Nextflow operator map with content: .map { it -> [ it[1] , it[2], it[3] ] }\n",
      "Nextflow operator set with content: .set { ch_bam_replicate }\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator set with content: .set { ch_bam_replicate }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(MERGED_REPLICATE_CALL_ANNOTATE_PEAKS.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(MERGED_REPLICATE_CONSENSUS_PEAKS.out.versions)\n",
      "Nextflow operator collect with content: MERGED_LIBRARY_BAM_TO_BIGWIG.out.bigwig.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: MERGED_LIBRARY_CALL_ANNOTATE_PEAKS.out.peaks.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_macs2_consensus_library_bed.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_ucsc_bedgraphtobigwig_replicate_bigwig.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_macs2_replicate_peaks.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_macs2_consensus_replicate_bed.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator join with content: ].join('') },\n",
      "Nextflow operator join with content: ].join('') },\n",
      "Process NFCORE_BACASS in filepath ./git_repos\\bacass\\main.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\bacass\\main.nf\n",
      "Process BAKTA_DBDOWNLOAD_RUN in filepath ./git_repos\\bacass\\subworkflows\\local\\bakta_dbdownload_run.nf\n",
      "Nextflow operator map with content: ch_baktadb_tar  = Channel.from(ch_path_baktadb).map{ db -> [ [id: 'baktadb'], db ]}\n",
      "Nextflow operator map with content: ch_baktadb      = UNTAR.out.untar.map{ meta, db -> db }\n",
      "Nextflow operator mix with content: ch_versions     = ch_versions.mix(UNTAR.out.versions)\n",
      "Nextflow operator map with content: ch_baktadb      = Channel.from(ch_path_baktadb).map{ db -> db }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BAKTA_BAKTADBDOWNLOAD.out.versions)\n",
      "Nextflow operator mix with content: ch_versions             = ch_versions.mix(BAKTA_BAKTA.out.versions)\n",
      "Nextflow operator ifEmpty with content: versions                = ch_versions.ifEmpty(null) // channel: [ path(versions.yml) ]\n",
      "Error: Unclosed process block in file: ./git_repos\\bacass\\subworkflows\\local\\bakta_dbdownload_run.nf\n",
      "Process FASTQ_TRIM_FASTP_FASTQC in filepath ./git_repos\\bacass\\subworkflows\\nf-core\\fastq_trim_fastp_fastqc\\main.nf\n",
      "Nextflow operator first with content: ch_versions     = ch_versions.mix(FASTQC_RAW.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions       = ch_versions.mix(FASTP.out.versions.first())\n",
      "Nextflow operator join with content: .join(ch_trim_json)\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator set with content: .set { ch_trim_reads }\n",
      "Nextflow operator first with content: ch_versions      = ch_versions.mix(FASTQC_TRIM.out.versions.first())\n",
      "Nextflow operator ifEmpty with content: versions = ch_versions.ifEmpty(null) // channel: [ path(versions.yml) ]\n",
      "Error: Unclosed process block in file: ./git_repos\\bacass\\subworkflows\\nf-core\\fastq_trim_fastp_fastqc\\main.nf\n",
      "Process BACASS in filepath ./git_repos\\bacass\\workflows\\bacass.nf\n",
      "Nextflow operator multiMap with content: .multiMap (criteria)\n",
      "Nextflow operator set with content: .set { ch_input }\n",
      "Nextflow operator filter with content: .filter{ it != null }\n",
      "Nextflow operator set with content: .set { ch_shortreads }\n",
      "Nextflow operator filter with content: .filter{ it != null }\n",
      "Nextflow operator set with content: .set { ch_longreads }\n",
      "Nextflow operator filter with content: .filter{ it != null }\n",
      "Nextflow operator set with content: .set { ch_fast5 }\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(FASTQ_TRIM_FASTP_FASTQC.out.versions.ifEmpty(null))\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(NANOPLOT.out.versions.ifEmpty(null))\n",
      "Nextflow operator dump with content: ch_fast5.dump(tag: 'fast5')\n",
      "Nextflow operator ifEmpty with content: ch_versions       = ch_versions.mix(PYCOQC.out.versions.ifEmpty(null))\n",
      "Nextflow operator dump with content: ch_longreads.dump(tag: 'longreads')\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix( PORECHOP_PORECHOP.out.versions.ifEmpty(null) )\n",
      "Nextflow operator dump with content: .dump(tag: 'fastp')\n",
      "Nextflow operator join with content: .join(PORECHOP_PORECHOP.out.reads)\n",
      "Nextflow operator dump with content: .dump(tag: 'ch_for_assembly')\n",
      "Nextflow operator set with content: .set { ch_for_assembly }\n",
      "Nextflow operator dump with content: .dump(tag: 'fastp')\n",
      "Nextflow operator map with content: .map{ meta,reads -> tuple(meta,reads,[]) }\n",
      "Nextflow operator dump with content: .dump(tag: 'ch_for_assembly')\n",
      "Nextflow operator set with content: .set { ch_for_assembly }\n",
      "Nextflow operator dump with content: .dump(tag: 'porechop')\n",
      "Nextflow operator map with content: .map{ meta,lr -> tuple(meta,[],lr) }\n",
      "Nextflow operator dump with content: .dump(tag: 'ch_for_assembly')\n",
      "Nextflow operator set with content: .set { ch_for_assembly }\n",
      "Nextflow operator dump with content: ch_assembly = ch_assembly.mix( UNICYCLER.out.scaffolds.dump(tag: 'unicycler') )\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix( UNICYCLER.out.versions.ifEmpty(null) )\n",
      "Nextflow operator map with content: ch_for_assembly.map { meta, reads, lr -> tuple( meta, lr ) },\n",
      "Nextflow operator map with content: ch_for_assembly.map { meta, reads, lr -> meta.genome_size }\n",
      "Nextflow operator dump with content: ch_assembly = ch_assembly.mix( CANU.out.assembly.dump(tag: 'canu') )\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(CANU.out.versions.ifEmpty(null))\n",
      "Nextflow operator map with content: ch_for_assembly.map{ meta,sr,lr -> tuple(meta,lr) },\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(MINIMAP2_ALIGN.out.versions.ifEmpty(null))\n",
      "Nextflow operator join with content: .join(MINIMAP2_ALIGN.out.paf)\n",
      "Nextflow operator map with content: .map { meta, sr, lr, paf-> tuple(meta, lr, paf) }\n",
      "Nextflow operator set with content: .set { ch_for_miniasm }\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(MINIASM.out.versions.ifEmpty(null))\n",
      "Nextflow operator map with content: ch_for_assembly.map{ meta,sr,lr -> tuple(meta,lr) },\n",
      "Nextflow operator map with content: MINIASM.out.assembly.map { meta, assembly -> assembly },\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(MINIMAP2_CONSENSUS.out.versions.ifEmpty(null))\n",
      "Nextflow operator join with content: .join(MINIASM.out.assembly)\n",
      "Nextflow operator join with content: .join(MINIMAP2_CONSENSUS.out.paf)\n",
      "Nextflow operator map with content: .map { meta, sr, lr, assembly, paf -> tuple(meta, lr, assembly, paf) }\n",
      "Nextflow operator set with content: .set{ ch_for_racon }\n",
      "Nextflow operator dump with content: ch_assembly = ch_assembly.mix( RACON.out.improved_assembly.dump(tag: 'miniasm') )\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(RACON.out.versions.ifEmpty(null))\n",
      "Nextflow operator join with content: .join( ch_assembly )\n",
      "Nextflow operator set with content: .set { ch_for_polish }\n",
      "Nextflow operator map with content: ch_for_polish.map { meta, sr, lr, fasta -> tuple(meta, lr)  },\n",
      "Nextflow operator map with content: ch_for_polish.map { meta, sr, lr, fasta -> fasta  },\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(MINIMAP2_POLISH.out.versions.ifEmpty(null))\n",
      "Nextflow operator dump with content: MINIMAP2_POLISH.out.bam.dump(tag: 'samtools_sort')\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(SAMTOOLS_INDEX.out.versions.ifEmpty(null))\n",
      "Nextflow operator join with content: .join( MINIMAP2_POLISH.out.bam )    // tuple val(meta), file(bam)\n",
      "Nextflow operator join with content: .join( SAMTOOLS_INDEX.out.bai )     // tuple  val(meta), file(bai)\n",
      "Nextflow operator join with content: .join( ch_fast5 )             // tuple val(meta), file(fast5)\n",
      "Nextflow operator set with content: .set { ch_for_nanopolish }          // tuple val(meta), val(reads), file(longreads), file(assembly), file(bam), file(bai), file(fast5)\n",
      "Nextflow operator dump with content: ch_for_nanopolish.dump(tag: 'into_nanopolish')\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(NANOPOLISH.out.versions.ifEmpty(null))\n",
      "Nextflow operator join with content: .join( ch_assembly )\n",
      "Nextflow operator map with content: .map { meta, sr, lr, assembly -> tuple(meta, lr, assembly) }\n",
      "Nextflow operator set with content: .set { ch_for_medaka }\n",
      "Nextflow operator dump with content: MEDAKA ( ch_for_medaka.dump(tag: 'into_medaka') )\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(MEDAKA.out.versions.ifEmpty(null))\n",
      "Nextflow operator dump with content: ch_for_kraken2_short.dump(tag: 'kraken2_short'),\n",
      "Nextflow operator dump with content: KRAKEN2_DB_PREPARATION.out.db.map { info, db -> db }.dump(tag: 'kraken2_db_preparation'),\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(KRAKEN2.out.versions.ifEmpty(null))\n",
      "Nextflow operator map with content: .map { meta, reads ->\n",
      "Nextflow operator dump with content: .dump(tag: 'kraken2_long'),\n",
      "Nextflow operator dump with content: KRAKEN2_DB_PREPARATION.out.db.map { info, db -> db }.dump(tag: 'kraken2_db_preparation'),\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(KRAKEN2_LONG.out.versions.ifEmpty(null))\n",
      "Nextflow operator collect with content: .collect{ it[1] }\n",
      "Nextflow operator map with content: .map { consensus_collect -> tuple([id: \"report\"], consensus_collect) }\n",
      "Nextflow operator set with content: .set { ch_to_quast }\n",
      "Nextflow operator ifEmpty with content: ch_versions      = ch_versions.mix(QUAST.out.versions.ifEmpty(null))\n",
      "Nextflow operator ifEmpty with content: ch_versions     = ch_versions.mix(GUNZIP.out.versions.ifEmpty(null))\n",
      "Nextflow operator collect with content: ch_prokka_txt_multiqc   = PROKKA.out.txt.collect()\n",
      "Nextflow operator ifEmpty with content: ch_versions             = ch_versions.mix(PROKKA.out.versions.ifEmpty(null))\n",
      "Nextflow operator ifEmpty with content: ch_versions     = ch_versions.mix(GUNZIP.out.versions.ifEmpty(null))\n",
      "Nextflow operator collect with content: ch_bakta_txt_multiqc    = BAKTA_DBDOWNLOAD_RUN.out.bakta_txt_multiqc.collect()\n",
      "Nextflow operator mix with content: ch_versions             = ch_versions.mix(BAKTA_DBDOWNLOAD_RUN.out.versions)\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(DFAST.out.versions.ifEmpty(null))\n",
      "Nextflow operator collectFile with content: ch_versions.unique().collectFile(name: 'collated_versions.yml')\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_workflow_summary.collectFile(name: 'workflow_summary_mqc.yaml'))\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_methods_description.collectFile(name: 'methods_description_mqc.yaml'))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(CUSTOM_DUMPSOFTWAREVERSIONS.out.mqc_yml.collect())\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(ch_fastqc_raw_multiqc.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(ch_fastqc_trim_multiqc.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(ch_trim_json_multiqc.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(ch_kraken_short_multiqc.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(ch_kraken_long_multiqc.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(ch_quast_multiqc.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(ch_prokka_txt_multiqc.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(ch_bakta_txt_multiqc.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(ch_nanoplot_txt_multiqc.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(ch_porechop_log_multiqc.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(ch_pycoqc_multiqc.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files.collect(),\n",
      "Nextflow operator collect with content: ch_multiqc_custom_config.collect().ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_multiqc_logo.collect().ifEmpty([])\n",
      "Nextflow operator toList with content: multiqc_report = MULTIQC.out.report.toList()\n",
      "Error: Unclosed process block in file: ./git_repos\\bacass\\workflows\\bacass.nf\n",
      "Process NFCORE_BACTMAP in filepath ./git_repos\\bactmap\\main.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\bactmap\\main.nf\n",
      "Process BAM_SORT_SAMTOOLS in filepath ./git_repos\\bactmap\\subworkflows\\bam_sort_samtools.nf\n",
      "Nextflow operator join with content: BAM_STATS_SAMTOOLS ( SAMTOOLS_SORT.out.bam.join(SAMTOOLS_INDEX.out.bai, by: [0]) )\n",
      "Error: Unclosed process block in file: ./git_repos\\bactmap\\subworkflows\\bam_sort_samtools.nf\n",
      "Process BAM_STATS_SAMTOOLS in filepath ./git_repos\\bactmap\\subworkflows\\bam_stats_samtools.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\bactmap\\subworkflows\\bam_stats_samtools.nf\n",
      "Process CREATE_PHYLOGENY in filepath ./git_repos\\bactmap\\subworkflows\\create_phylogeny.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\bactmap\\subworkflows\\create_phylogeny.nf\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Process INPUT_CHECK in filepath ./git_repos\\bactmap\\subworkflows\\input_check.nf\n",
      "Nextflow operator splitCsv with content: .splitCsv ( header:true, sep:',' )\n",
      "Nextflow operator map with content: .map { create_fastq_channels(it) }\n",
      "Nextflow operator set with content: .set { sample_info }\n",
      "Error: Unclosed process block in file: ./git_repos\\bactmap\\subworkflows\\input_check.nf\n",
      "Process SUB_SAMPLING in filepath ./git_repos\\bactmap\\subworkflows\\sub_sampling.nf\n",
      "Nextflow operator combine with content: reads_and_genome_size = reads.combine([params.genome_size])\n",
      "Nextflow operator map with content: genome_size = MASH_SKETCH.out.stats.map { meta, file -> [meta, WorkflowBactmap.find_genome_size(file.text)]}\n",
      "Nextflow operator combine with content: reads_and_genome_size = reads.combine(genome_size, by: 0)\n",
      "Nextflow operator first with content: version = params.genome_size ? RASUSA.out.version : MASH_SKETCH.out.version.mix(RASUSA.out.version.first().ifEmpty(null)) //    path: *.version.txt\n",
      "Error: Unclosed process block in file: ./git_repos\\bactmap\\subworkflows\\sub_sampling.nf\n",
      "Process VARIANTS_BCFTOOLS in filepath ./git_repos\\bactmap\\subworkflows\\variants_bcftools.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\bactmap\\subworkflows\\variants_bcftools.nf\n",
      "Process BACTMAP in filepath ./git_repos\\bactmap\\workflows\\bactmap.nf\n",
      "Nextflow operator first with content: ch_software_versions = ch_software_versions.mix(FASTP.out.version.first().ifEmpty(null))\n",
      "Nextflow operator first with content: ch_software_versions = ch_software_versions.mix(BWA_MEM.out.version.first().ifEmpty(null))\n",
      "Nextflow operator first with content: ch_software_versions = ch_software_versions.mix(BAM_SORT_SAMTOOLS.out.samtools_version.first().ifEmpty(null))\n",
      "Nextflow operator first with content: ch_software_versions = ch_software_versions.mix(VARIANTS_BCFTOOLS.out.bcftools_version.first().ifEmpty(null))\n",
      "Nextflow operator collect with content: VCF2PSEUDOGENOME.out.pseudogenome.map { pseudogenome -> pseudogenome[1] }.collect(),\n",
      "Nextflow operator branch with content: .branch {\n",
      "Nextflow operator toInteger with content: ALIGNMENT_NUM_PASS: aligned_pseudogenomes[0].toInteger() >= 4\n",
      "Nextflow operator toInteger with content: ALIGNMENT_NUM_FAIL: aligned_pseudogenomes[0].toInteger() < 4\n",
      "Nextflow operator set with content: .set { aligned_pseudogenomes_branch }\n",
      "Nextflow operator view with content: aligned_pseudogenomes_branch.ALIGNMENT_NUM_FAIL.view { \"Insufficient (${it[0]}) genomes after filtering to continue. Check results/pseudogenomes/low_quality_pseudogenomes.tsv for details\"}\n",
      "Nextflow operator map with content: .map{ it[1] }\n",
      "Nextflow operator set with content: .set { aligned_pseudogenomes }\n",
      "Nextflow operator ifEmpty with content: ch_software_versions = ch_software_versions.mix(SNPSITES.out.version.ifEmpty(null))\n",
      "Nextflow operator ifEmpty with content: ch_software_versions = ch_software_versions.mix(GUBBINS.out.version.ifEmpty(null))\n",
      "Nextflow operator ifEmpty with content: ch_software_versions = ch_software_versions.mix(CREATE_PHYLOGENY.out.rapidnj_version.ifEmpty(null))\n",
      "Nextflow operator ifEmpty with content: ch_software_versions = ch_software_versions.mix(CREATE_PHYLOGENY.out.fasttree_version.ifEmpty(null))\n",
      "Nextflow operator ifEmpty with content: ch_software_versions = ch_software_versions.mix(CREATE_PHYLOGENY.out.iqtree_version.ifEmpty(null))\n",
      "Nextflow operator ifEmpty with content: ch_software_versions = ch_software_versions.mix(CREATE_PHYLOGENY.out.raxmlng_version.ifEmpty(null))\n",
      "Nextflow operator collect with content: ch_software_versions.map { it }.collect()\n",
      "Nextflow operator collect with content: ch_multiqc_custom_config.collect().ifEmpty([]),\n",
      "Nextflow operator collect with content: GET_SOFTWARE_VERSIONS.out.yaml.collect(),\n",
      "Nextflow operator collectFile with content: ch_workflow_summary.collectFile(name: 'workflow_summary_mqc.yaml'),\n",
      "Nextflow operator collect with content: FASTP.out.json.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: BAM_SORT_SAMTOOLS.out.stats.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: VARIANTS_BCFTOOLS.out.stats.collect{it[1]}.ifEmpty([])\n",
      "Nextflow operator toList with content: multiqc_report = MULTIQC.out.report.toList()\n",
      "Error: Unclosed process block in file: ./git_repos\\bactmap\\workflows\\bactmap.nf\n",
      "Process NFCORE_BAMTOFASTQ in filepath ./git_repos\\bamtofastq\\main.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\bamtofastq\\main.nf\n",
      "Process ALIGNMENT_TO_FASTQ in filepath ./git_repos\\bamtofastq\\subworkflows\\local\\alignment_to_fastq.nf\n",
      "Nextflow operator join with content: .join(SAMTOOLS_VIEW_UNMAP_MAP.out.bam, remainder: true)\n",
      "Nextflow operator join with content: .join(SAMTOOLS_VIEW_MAP_UNMAP.out.bam, remainder: true)\n",
      "Nextflow operator map with content: .map{ meta, unmap_unmap, unmap_map, map_unmap ->\n",
      "Nextflow operator join with content: .join(SAMTOOLS_VIEW_UNMAP_MAP.out.cram, remainder: true)\n",
      "Nextflow operator join with content: .join(SAMTOOLS_VIEW_MAP_UNMAP.out.cram, remainder: true)\n",
      "Nextflow operator map with content: .map{ meta, unmap_unmap, unmap_map, map_unmap ->\n",
      "Nextflow operator mix with content: ch_unmapped_bam_cram = Channel.empty().mix(all_unmapped_bam,all_unmapped_cram)\n",
      "Nextflow operator map with content: COLLATE_FASTQ_UNMAP(SAMTOOLS_MERGE_UNMAP.out.cram.mix(SAMTOOLS_MERGE_UNMAP.out.bam), fasta.map{ it ->\n",
      "Nextflow operator map with content: COLLATE_FASTQ_MAP(SAMTOOLS_VIEW_MAP_MAP.out.cram.mix(SAMTOOLS_VIEW_MAP_MAP.out.bam), fasta.map{ it ->\n",
      "Nextflow operator join with content: .join(COLLATE_FASTQ_UNMAP.out.fastq)\n",
      "Nextflow operator map with content: .map{ meta, mapped_reads, unmapped_reads ->\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(CAT_FASTQ.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(COLLATE_FASTQ_MAP.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(COLLATE_FASTQ_UNMAP.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_MERGE_UNMAP.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_VIEW_MAP_MAP.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_VIEW_MAP_UNMAP.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_VIEW_UNMAP_MAP.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_VIEW_UNMAP_UNMAP.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\bamtofastq\\subworkflows\\local\\alignment_to_fastq.nf\n",
      "Process PREPARE_INDICES in filepath ./git_repos\\bamtofastq\\subworkflows\\local\\prepare_indices.nf\n",
      "Nextflow operator branch with content: input.branch{\n",
      "Nextflow operator set with content: }.set{samtools_input}\n",
      "Nextflow operator map with content: input_to_index = samtools_input.to_index.map{ it -> [it[0], it[1]] }\n",
      "Nextflow operator mix with content: ch_versions    = ch_versions.mix(SAMTOOLS_INDEX.out.versions)\n",
      "Nextflow operator mix with content: ch_index_files = Channel.empty().mix(SAMTOOLS_INDEX.out.bai, SAMTOOLS_INDEX.out.crai)\n",
      "Nextflow operator join with content: ch_new = input_to_index.join(ch_index_files)\n",
      "Nextflow operator mix with content: ch_out = samtools_input.is_indexed.mix(ch_new)\n",
      "Nextflow operator map with content: SAMTOOLS_FAIDX(fasta.map{ it -> [[id:it[0].baseName], it] })\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_FAIDX.out.versions)\n",
      "Nextflow operator map with content: fasta_fai   = SAMTOOLS_FAIDX.out.fai.map{ meta, fai -> [fai] }\n",
      "Error: Unclosed process block in file: ./git_repos\\bamtofastq\\subworkflows\\local\\prepare_indices.nf\n",
      "Process PRE_CONVERSION_QC in filepath ./git_repos\\bamtofastq\\subworkflows\\local\\pre_conversion_qc.nf\n",
      "Nextflow operator branch with content: input.branch{\n",
      "Nextflow operator set with content: }.set{fastqc_input}\n",
      "Nextflow operator map with content: .map{ it ->\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_IDXSTATS.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_FLAGSTAT.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_STATS.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(FASTQC_PRE_CONVERSION.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\bamtofastq\\subworkflows\\local\\pre_conversion_qc.nf\n",
      "Process BAMTOFASTQ in filepath ./git_repos\\bamtofastq\\workflows\\bamtofastq.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(PREPARE_INDICES.out.versions)\n",
      "Nextflow operator collect with content: fasta_fai = params.fasta ? params.fasta_fai ? Channel.fromPath(params.fasta_fai).collect() : PREPARE_INDICES.out.fasta_fai : []\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(PRE_CONVERSION_QC.out.versions)\n",
      "Nextflow operator join with content: ch_paired_end = ch_input.join(CHECK_IF_PAIRED_END.out.paired_end)\n",
      "Nextflow operator join with content: ch_single_end = ch_input.join(CHECK_IF_PAIRED_END.out.single_end)\n",
      "Nextflow operator map with content: ch_input_new = ch_single_end.map{ meta, bam, bai, txt ->\n",
      "Nextflow operator map with content: .mix(ch_paired_end.map{ meta, bam, bai, txt ->\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(CHECK_IF_PAIRED_END.out.versions)\n",
      "Nextflow operator mix with content: samtools_chr_out = Channel.empty().mix( SAMTOOLS_CHR.out.bam,\n",
      "Nextflow operator join with content: ch_input_chr = samtools_chr_out.join(Channel.empty().mix( SAMTOOLS_CHR_INDEX.out.bai,\n",
      "Nextflow operator map with content: ch_input_new = ch_input_chr.map{ it ->\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_CHR.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_CHR_INDEX.out.versions)\n",
      "Nextflow operator branch with content: ch_input_new.branch{\n",
      "Nextflow operator set with content: }.set{conversion_input}\n",
      "Nextflow operator map with content: conversion_input.ch_single.map{ it -> [ it[0], it[1] ]}, // meta, bam/cram\n",
      "Nextflow operator map with content: fasta.map{ it ->                                         // meta, fasta\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_COLLATEFASTQ_SINGLE_END.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ALIGNMENT_TO_FASTQ.out.versions)\n",
      "Nextflow operator mix with content: ch_reads_post_qc = Channel.empty().mix(SAMTOOLS_COLLATEFASTQ_SINGLE_END.out.fastq_singleton, ALIGNMENT_TO_FASTQ.out.reads)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(FASTQC_POST_CONVERSION.out.versions)\n",
      "Nextflow operator collectFile with content: ch_versions.unique().collectFile(name: 'collated_versions.yml')\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_workflow_summary.collectFile(name: 'workflow_summary_mqc.yaml'))\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_methods_description.collectFile(name: 'methods_description_mqc.yaml'))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(CUSTOM_DUMPSOFTWAREVERSIONS.out.mqc_yml.collect())\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(PRE_CONVERSION_QC.out.flagstat.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(PRE_CONVERSION_QC.out.idxstats.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(PRE_CONVERSION_QC.out.stats.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(PRE_CONVERSION_QC.out.zip.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(FASTQC_POST_CONVERSION.out.zip.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files.collect(),\n",
      "Nextflow operator toList with content: ch_multiqc_config.toList(),\n",
      "Nextflow operator toList with content: ch_multiqc_custom_config.toList(),\n",
      "Nextflow operator toList with content: ch_multiqc_logo.toList()\n",
      "Nextflow operator toList with content: multiqc_report = MULTIQC.out.report.toList()\n",
      "Error: Unclosed process block in file: ./git_repos\\bamtofastq\\workflows\\bamtofastq.nf\n",
      "Process NFCORE_CALLINGCARDS_MAMMALS in filepath ./git_repos\\callingcards\\main.nf\n",
      "Process NFCORE_CALLINGCARDS_YEAST in filepath ./git_repos\\callingcards\\main.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\callingcards\\main.nf\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Process ALIGN in filepath ./git_repos\\callingcards\\subworkflows\\local\\align.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BWAMEM2_MEM.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_INDEX.out.versions)\n",
      "Nextflow operator join with content: .join(SAMTOOLS_INDEX.out.bai)\n",
      "Nextflow operator set with content: .set{ ch_tmp }\n",
      "Nextflow operator mix with content: ch_bam = ch_bam.mix(ch_tmp)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(FASTQ_ALIGN_BWAALN.out.versions)\n",
      "Nextflow operator join with content: .join(FASTQ_ALIGN_BWAALN.out.bai)\n",
      "Nextflow operator set with content: .set{ ch_tmp }\n",
      "Nextflow operator mix with content: ch_bam = ch_bam.mix(ch_tmp)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BOWTIE2_ALIGN.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_INDEX.out.versions)\n",
      "Nextflow operator join with content: .join(SAMTOOLS_INDEX.out.bai)\n",
      "Nextflow operator set with content: .set{ ch_tmp }\n",
      "Nextflow operator mix with content: ch_bam = ch_bam.mix(ch_tmp)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BOWTIE_ALIGN.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_SORT.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_INDEX.out.versions)\n",
      "Nextflow operator join with content: .join(SAMTOOLS_INDEX.out.bai)\n",
      "Nextflow operator set with content: .set{ ch_tmp }\n",
      "Nextflow operator mix with content: ch_bam = ch_bam.mix(ch_tmp)\n",
      "Error: Unclosed process block in file: ./git_repos\\callingcards\\subworkflows\\local\\align.nf\n",
      "Process PREPARE_GENOME in filepath ./git_repos\\callingcards\\subworkflows\\local\\prepare_genome.nf\n",
      "Nextflow operator map with content: fasta.map{meta, fasta -> fasta}\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BEDTOOLS_MASKFASTA.out.versions)\n",
      "Nextflow operator map with content: .map{meta, fasta -> [[id: 'masked_' + meta.id], fasta]}\n",
      "Nextflow operator map with content: .map{meta, fasta -> [[id: 'concat_' + meta.id], fasta]}\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GTF2BED.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BWAMEM2_INDEX.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BWA_INDEX.out.versions)\n",
      "Nextflow operator collect with content: ch_bwa_index = Channel.of([[id: 'input_genome_index'],params.bwa_index]).collect()\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BOWTIE2_BUILD.out.versions)\n",
      "Nextflow operator collect with content: ch_bowtie2_index = Channel.of([[id: 'input_genome_index'],params.bowtie2_index]).collect()\n",
      "Nextflow operator map with content: BOWTIE_BUILD ( ch_fasta.map{meta, fasta -> fasta}  )\n",
      "Nextflow operator concat with content: ch_versions = ch_versions.concat(BOWTIE_BUILD.out.versions)\n",
      "Nextflow operator collect with content: ch_bowtie_index = Channel.of([params.bowtie_index]).collect()\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_FAIDX.out.versions)\n",
      "Nextflow operator collect with content: ch_fasta_index = Channel.of([\"\",params.fasta_index]).collect()\n",
      "Error: Unclosed process block in file: ./git_repos\\callingcards\\subworkflows\\local\\prepare_genome.nf\n",
      "Process PREPARE_READS in filepath ./git_repos\\callingcards\\subworkflows\\local\\mammals\\prepare_reads.nf\n",
      "Nextflow operator set with content: FASTQC ( reads ).html.set { fastqc_html }\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(FASTQC.out.versions.first())\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SEQKIT_SPLIT2.out.versions)\n",
      "Nextflow operator transpose with content: .transpose()\n",
      "Nextflow operator map with content: .map{ meta, read1 ->\n",
      "Nextflow operator set with content: .set{ ch_split_reads }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(UMITOOLS_EXTRACT.out.versions)\n",
      "Nextflow operator mix with content: ch_reads = ch_reads.mix(TRIMMOMATIC.out.trimmed_reads)\n",
      "Nextflow operator mix with content: trimmomatic_log = trimmomatic_log.mix(TRIMMOMATIC.out.log)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(TRIMMOMATIC.out.versions)\n",
      "Nextflow operator mix with content: ch_reads = ch_reads.mix(UMITOOLS_EXTRACT.out.reads)\n",
      "Error: Unclosed process block in file: ./git_repos\\callingcards\\subworkflows\\local\\mammals\\prepare_reads.nf\n",
      "Process PROCESS_ALIGNMENTS in filepath ./git_repos\\callingcards\\subworkflows\\local\\mammals\\process_alignments.nf\n",
      "Nextflow operator map with content: fasta.map{meta, fasta -> fasta},\n",
      "Nextflow operator map with content: fai.map{meta,fai -> fai} )\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(COUNT_HOPS.out.versions)\n",
      "Nextflow operator map with content: .map{meta, pkl ->\n",
      "Nextflow operator groupTuple with content: .groupTuple()\n",
      "Nextflow operator set with content: .set{ ch_grouped_pkl }\n",
      "Nextflow operator map with content: .map{meta, pkl ->\n",
      "Nextflow operator groupTuple with content: .groupTuple()\n",
      "Nextflow operator set with content: .set{ ch_grouped_barcode_qc_pkl }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(CONCATQC.out.versions)\n",
      "Nextflow operator map with content: .map{meta, bam ->\n",
      "Nextflow operator groupTuple with content: .groupTuple()\n",
      "Nextflow operator mix with content: .mix(COUNT_HOPS.out.failing_bam\n",
      "Nextflow operator map with content: .map{meta, bam ->\n",
      "Nextflow operator groupTuple with content: .groupTuple())\n",
      "Nextflow operator set with content: .set{ ch_bam }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_MERGE.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_SORT.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_INDEX.out.versions)\n",
      "Nextflow operator join with content: .join(SAMTOOLS_INDEX.out.bai)\n",
      "Nextflow operator set with content: .set{ ch_bam_bai }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BAM_STATS_SAMTOOLS.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(PICARD_COLLECTMULTIPLEMETRICS.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\callingcards\\subworkflows\\local\\mammals\\process_alignments.nf\n",
      "Process PREPARE_READS in filepath ./git_repos\\callingcards\\subworkflows\\local\\yeast\\prepare_reads.nf\n",
      "Nextflow operator set with content: FASTQCRAW.out.html.set{ raw_fastqc_html }\n",
      "Nextflow operator set with content: FASTQCRAW.out.zip.set{ raw_fastqc_zip }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SEQKIT_SPLIT2.out.versions)\n",
      "Nextflow operator transpose with content: .transpose()\n",
      "Nextflow operator map with content: .map{ meta, read ->\n",
      "Nextflow operator groupTuple with content: .groupTuple()\n",
      "Nextflow operator map with content: .map{meta, reads ->\n",
      "Nextflow operator combine with content: .combine(barcode_details.map{meta, barcode_json -> [meta.id, barcode_json]}, by:0)\n",
      "Nextflow operator map with content: .map{id, meta, read, barcode_details ->\n",
      "Nextflow operator set with content: .set{ split_reads_with_barcode_details }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(DEMULTIPLEX.out.versions)\n",
      "Nextflow operator transpose with content: .transpose()\n",
      "Nextflow operator map with content: .map{meta, read ->\n",
      "Nextflow operator groupTuple with content: .groupTuple()\n",
      "Nextflow operator map with content: .map{ meta, reads ->\n",
      "Nextflow operator set with content: .set{ demux_reads }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(CONCATFASTQ.out.versions)\n",
      "Nextflow operator map with content: .map{meta, pickle ->\n",
      "Nextflow operator groupTuple with content: .groupTuple()\n",
      "Nextflow operator join with content: .join(barcode_details, by:0)\n",
      "Nextflow operator map with content: .map{meta, pickle, barcode_details ->\n",
      "Nextflow operator set with content: .set{ ch_demux_qc }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(CONCATQC.out.versions)\n",
      "Nextflow operator map with content: .map{\n",
      "Nextflow operator groupTuple with content: .groupTuple()\n",
      "Nextflow operator map with content: .map{meta, reads ->\n",
      "Nextflow operator filter with content: .filter{meta, reads ->\n",
      "Nextflow operator set with content: .set{ demux_concat_reads }\n",
      "Nextflow operator map with content: .map{ meta, reads ->\n",
      "Nextflow operator set with content: .set{ trimmomatic_input }\n",
      "Nextflow operator mix with content: trimmomatic_log = trimmomatic_log.mix(TRIMMOMATIC.out.log)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(TRIMMOMATIC.out.versions)\n",
      "Nextflow operator mix with content: output_reads = output_reads.mix(TRIMMOMATIC.out.trimmed_reads)\n",
      "Nextflow operator set with content: FASTQCDEMUX.out.html.set{ demux_fastqc_html }\n",
      "Nextflow operator set with content: FASTQCDEMUX.out.zip.set { demux_fastqc_zip  }\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(FASTQCDEMUX.out.versions.first())\n",
      "Nextflow operator toInteger with content: return matcher[0][1].toInteger()\n",
      "Error: Unclosed process block in file: ./git_repos\\callingcards\\subworkflows\\local\\yeast\\prepare_reads.nf\n",
      "Process PROCESS_ALIGNMENTS in filepath ./git_repos\\callingcards\\subworkflows\\local\\yeast\\process_alignments.nf\n",
      "Nextflow operator map with content: fasta.map{meta, fasta -> fasta},\n",
      "Nextflow operator map with content: fai.map{meta,fai -> fai}\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(COUNT_HOPS.out.versions)\n",
      "Nextflow operator map with content: .map{meta, bam, bai ->\n",
      "Nextflow operator set with content: .set{ ch_passing_bam }\n",
      "Nextflow operator map with content: .map{meta, bam, bai ->\n",
      "Nextflow operator set with content: .set{ ch_failing_bam }\n",
      "Nextflow operator mix with content: .mix(ch_failing_bam)\n",
      "Nextflow operator map with content: .map{meta_join, meta, bam, bai ->\n",
      "Nextflow operator set with content: .set{ ch_parse_bam_out }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BAM_STATS_SAMTOOLS.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(PICARD_COLLECTMULTIPLEMETRICS.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BAM_RSEQC.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\callingcards\\subworkflows\\local\\yeast\\process_alignments.nf\n",
      "Process BAM_RSEQC in filepath ./git_repos\\callingcards\\subworkflows\\nf-core\\bam_rseqc\\main.nf\n",
      "Nextflow operator map with content: .map { [ it[0], it[1] ] }\n",
      "Nextflow operator set with content: .set { ch_bam }\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(RSEQC_BAMSTAT.out.versions.first())\n",
      "Nextflow operator mix with content: ch_innerdistance          = ch_innerdistance_distance.mix(ch_innerdistance_freq, ch_innerdistance_mean, ch_innerdistance_pdf, ch_innerdistance_rscript)\n",
      "Nextflow operator first with content: ch_versions               = ch_versions.mix(RSEQC_INNERDISTANCE.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions        = ch_versions.mix(RSEQC_INFEREXPERIMENT.out.versions.first())\n",
      "Nextflow operator mix with content: ch_junctionannotation             = ch_junctionannotation_bed.mix(ch_junctionannotation_interact_bed, ch_junctionannotation_xls, ch_junctionannotation_pdf, ch_junctionannotation_events_pdf, ch_junctionannotation_rscript, ch_junctionannotation_log)\n",
      "Nextflow operator first with content: ch_versions                        = ch_versions.mix(RSEQC_JUNCTIONANNOTATION.out.versions.first())\n",
      "Nextflow operator mix with content: ch_junctionsaturation         = ch_junctionsaturation_pdf.mix(ch_junctionsaturation_rscript)\n",
      "Nextflow operator first with content: ch_versions                   = ch_versions.mix(RSEQC_JUNCTIONSATURATION.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions         = ch_versions.mix(RSEQC_READDISTRIBUTION.out.versions.first())\n",
      "Nextflow operator mix with content: ch_readduplication         = ch_readduplication_seq_xls.mix(ch_readduplication_pos_xls, ch_readduplication_pdf, ch_readduplication_rscript)\n",
      "Nextflow operator first with content: ch_versions                = ch_versions.mix(RSEQC_READDUPLICATION.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(RSEQC_TIN.out.versions.first())\n",
      "Error: Unclosed process block in file: ./git_repos\\callingcards\\subworkflows\\nf-core\\bam_rseqc\\main.nf\n",
      "Process BAM_STATS_SAMTOOLS in filepath ./git_repos\\callingcards\\subworkflows\\nf-core\\bam_stats_samtools\\main.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_STATS.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_FLAGSTAT.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_IDXSTATS.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\callingcards\\subworkflows\\nf-core\\bam_stats_samtools\\main.nf\n",
      "Process FASTQ_ALIGN_BWAALN in filepath ./git_repos\\callingcards\\subworkflows\\nf-core\\fastq_align_bwaaln\\main.nf\n",
      "Nextflow operator combine with content: .combine(ch_index)\n",
      "Nextflow operator map with content: .map{\n",
      "Nextflow operator multiMap with content: .multiMap {\n",
      "Nextflow operator map with content: ch_reads_newid = ch_prepped_input.map{ meta, reads, meta_index, index -> [ meta, reads ] }\n",
      "Nextflow operator map with content: ch_index_newid = ch_prepped_input.map{ meta, reads, meta_index, index -> [ meta, index ] }\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix( BWA_ALN.out.versions.first() )\n",
      "Nextflow operator join with content: .join ( BWA_ALN.out.sai )\n",
      "Nextflow operator branch with content: .branch {\n",
      "Nextflow operator join with content: .join ( ch_index_newid )\n",
      "Nextflow operator multiMap with content: .multiMap {\n",
      "Nextflow operator join with content: .join ( ch_index_newid )\n",
      "Nextflow operator multiMap with content: .multiMap {\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix( BWA_SAMPE.out.versions.first() )\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix( BWA_SAMSE.out.versions.first() )\n",
      "Nextflow operator mix with content: ch_bam_for_index = BWA_SAMPE.out.bam.mix( BWA_SAMSE.out.bam )\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SAMTOOLS_INDEX.out.versions.first())\n",
      "Nextflow operator map with content: ch_bam_for_emit = ch_bam_for_index.map{ meta, bam -> [meta - meta.subMap('key_read_ref'), bam] }\n",
      "Nextflow operator map with content: ch_bai_for_emit = SAMTOOLS_INDEX.out.bai.map{ meta, bai -> [meta - meta.subMap('key_read_ref'), bai] }\n",
      "Nextflow operator map with content: ch_csi_for_emit = SAMTOOLS_INDEX.out.csi.map{ meta, csi -> [meta - meta.subMap('key_read_ref'), csi] }\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Error: Unclosed process block in file: ./git_repos\\callingcards\\subworkflows\\nf-core\\fastq_align_bwaaln\\main.nf\n",
      "Process CALLINGCARDS_MAMMALS in filepath ./git_repos\\callingcards\\workflows\\callingcards_mammals.nf\n",
      "Nextflow operator multiMap with content: .multiMap{ sample, fastq_1, fastq_2, barcode_details ->\n",
      "Nextflow operator set with content: .set{ ch_input }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(PREPARE_GENOME.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(PREPARE_READS.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ALIGN.out.versions)\n",
      "Nextflow operator map with content: .map{meta, bam, bai -> [meta.id, meta, bam, bai] }\n",
      "Nextflow operator combine with content: .combine(ch_input.barcode_details\n",
      "Nextflow operator map with content: .map{meta, barcode_details ->\n",
      "Nextflow operator map with content: .map{id, meta, bam, bai, barcode_details ->\n",
      "Nextflow operator set with content: .set{ ch_aln_with_details }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(PROCESS_ALIGNMENTS.out.versions)\n",
      "Nextflow operator collectFile with content: ch_versions.unique().collectFile(name: 'collated_versions.yml')\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_workflow_summary.collectFile(name: 'workflow_summary_mqc.yaml'))\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_methods_description.collectFile(name: 'methods_description_mqc.yaml'))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(CUSTOM_DUMPSOFTWAREVERSIONS.out.mqc_yml.collect())\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(PREPARE_READS.out.fastqc_zip.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(PREPARE_READS.out.fastqc_zip.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(PREPARE_READS.out.trimmomatic_log.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(PROCESS_ALIGNMENTS.out.samtools_stats.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(PROCESS_ALIGNMENTS.out.samtools_flagstat.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(PROCESS_ALIGNMENTS.out.samtools_idxstats.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(PROCESS_ALIGNMENTS.out.picard_qc.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(PROCESS_ALIGNMENTS.out.rseqc_bamstat.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(PROCESS_ALIGNMENTS.out.rseqc_inferexperiment.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(PROCESS_ALIGNMENTS.out.rseqc_innerdistance.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(PROCESS_ALIGNMENTS.out.rseqc_readdistribution.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(PROCESS_ALIGNMENTS.out.rseqc_readduplication.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(PROCESS_ALIGNMENTS.out.rseqc_tin.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files.collect(),\n",
      "Nextflow operator toList with content: ch_multiqc_config.toList(),\n",
      "Nextflow operator toList with content: ch_multiqc_custom_config.toList(),\n",
      "Nextflow operator toList with content: ch_multiqc_logo.toList()\n",
      "Nextflow operator toList with content: multiqc_report = MULTIQC.out.report.toList()\n",
      "Error: Unclosed process block in file: ./git_repos\\callingcards\\workflows\\callingcards_mammals.nf\n",
      "Process CALLINGCARDS_YEAST in filepath ./git_repos\\callingcards\\workflows\\callingcards_yeast.nf\n",
      "Nextflow operator multiMap with content: .multiMap{ sample, fastq_1, fastq_2, barcode_details ->\n",
      "Nextflow operator set with content: .set{ ch_input }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(PREPARE_GENOME.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(PREPARE_READS.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ALIGN.out.versions)\n",
      "Nextflow operator map with content: .map{ meta, barcode_details ->\n",
      "Nextflow operator set with content: .set{ ch_barcode_details_join }\n",
      "Nextflow operator map with content: .map{ meta, bam, bai ->\n",
      "Nextflow operator combine with content: .combine(ch_barcode_details_join, by:0)\n",
      "Nextflow operator map with content: .map{ key, meta, bam, bai, barcode_details ->\n",
      "Nextflow operator set with content: .set{ ch_aln_with_details }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(PROCESS_ALIGNMENTS.out.versions)\n",
      "Nextflow operator collectFile with content: ch_versions.unique().collectFile(name: 'collated_versions.yml')\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_workflow_summary.collectFile(name: 'workflow_summary_mqc.yaml'))\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_methods_description.collectFile(name: 'methods_description_mqc.yaml'))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(CUSTOM_DUMPSOFTWAREVERSIONS.out.mqc_yml.collect())\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(PREPARE_READS.out.raw_fastqc_zip.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(PREPARE_READS.out.demux_fastqc_zip.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(PREPARE_READS.out.trimmomatic_log.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(PROCESS_ALIGNMENTS.out.samtools_stats.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(PROCESS_ALIGNMENTS.out.samtools_flagstat.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(PROCESS_ALIGNMENTS.out.samtools_idxstats.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(PROCESS_ALIGNMENTS.out.picard_qc.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(PROCESS_ALIGNMENTS.out.rseqc_bamstat.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(PROCESS_ALIGNMENTS.out.rseqc_inferexperiment.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(PROCESS_ALIGNMENTS.out.rseqc_innerdistance.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(PROCESS_ALIGNMENTS.out.rseqc_readdistribution.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(PROCESS_ALIGNMENTS.out.rseqc_readduplication.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(PROCESS_ALIGNMENTS.out.rseqc_tin.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files.collect(),\n",
      "Nextflow operator toList with content: ch_multiqc_config.toList(),\n",
      "Nextflow operator toList with content: ch_multiqc_custom_config.toList(),\n",
      "Nextflow operator toList with content: ch_multiqc_logo.toList()\n",
      "Nextflow operator toList with content: multiqc_report = MULTIQC.out.report.toList()\n",
      "Error: Unclosed process block in file: ./git_repos\\callingcards\\workflows\\callingcards_yeast.nf\n",
      "Process NFCORE_CHIPSEQ in filepath ./git_repos\\chipseq\\main.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\chipseq\\main.nf\n",
      "Process FILTER_BAM_BAMTOOLS in filepath ./git_repos\\chipseq\\subworkflows\\local\\filter_bam_bamtools.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BAM_FILTER.out.versions,\n",
      "Error: Unclosed process block in file: ./git_repos\\chipseq\\subworkflows\\local\\filter_bam_bamtools.nf\n",
      "Process INPUT_CHECK in filepath ./git_repos\\chipseq\\subworkflows\\local\\input_check.nf\n",
      "Nextflow operator splitCsv with content: .splitCsv ( header:true, sep:',' )\n",
      "Nextflow operator map with content: .map { create_fastq_channel(it, seq_center) }\n",
      "Nextflow operator set with content: .set { reads }\n",
      "Nextflow operator join with content: def read_group = \"\\'@RG\\\\tID:${meta.id}\\\\tSM:${meta.id.split('_')[0..-2].join('_')}\\\\tPL:ILLUMINA\\\\tLB:${meta.id}\\\\tPU:1\\'\"\n",
      "Nextflow operator join with content: read_group = \"\\'@RG\\\\tID:${meta.id}\\\\tSM:${meta.id.split('_')[0..-2].join('_')}\\\\tPL:ILLUMINA\\\\tLB:${meta.id}\\\\tPU:1\\\\tCN:${seq_center}\\'\"\n",
      "Error: Unclosed process block in file: ./git_repos\\chipseq\\subworkflows\\local\\input_check.nf\n",
      "Process PREPARE_GENOME in filepath ./git_repos\\chipseq\\subworkflows\\local\\prepare_genome.nf\n",
      "Nextflow operator map with content: ch_fasta    = GUNZIP_FASTA ( [ [:], params.fasta ] ).gunzip.map{ it[1] }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GUNZIP_FASTA.out.versions)\n",
      "Nextflow operator map with content: ch_gtf      = GUNZIP_GTF ( [ [:], params.gtf ] ).gunzip.map{ it[1] }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GUNZIP_GTF.out.versions)\n",
      "Nextflow operator map with content: ch_gff      = GUNZIP_GFF ( [ [:], params.gff ] ).gunzip.map{ it[1] }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GUNZIP_GFF.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GFFREAD.out.versions)\n",
      "Nextflow operator map with content: ch_blacklist = GUNZIP_BLACKLIST ( [ [:], params.blacklist ] ).gunzip.map{ it[1] }\n",
      "Nextflow operator mix with content: ch_versions  = ch_versions.mix(GUNZIP_BLACKLIST.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GTF2BED.out.versions)\n",
      "Nextflow operator map with content: ch_gene_bed = GUNZIP_GENE_BED ( [ [:], params.gene_bed ] ).gunzip.map{ it[1] }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GUNZIP_GENE_BED.out.versions)\n",
      "Nextflow operator mix with content: ch_versions    = ch_versions.mix(CUSTOM_GETCHROMSIZES.out.versions)\n",
      "Nextflow operator ifEmpty with content: ch_blacklist.ifEmpty([])\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GENOME_BLACKLIST_REGIONS.out.versions)\n",
      "Nextflow operator map with content: ch_bwa_index = UNTAR_BWA_INDEX ( [ [:], params.bwa_index ] ).untar.map{ it[1] }\n",
      "Nextflow operator mix with content: ch_versions  = ch_versions.mix(UNTAR_BWA_INDEX.out.versions)\n",
      "Nextflow operator mix with content: ch_versions  = ch_versions.mix(BWA_INDEX.out.versions)\n",
      "Nextflow operator map with content: ch_bowtie2_index = UNTAR_BOWTIE2_INDEX ( [ [:], params.bowtie2_index ] ).untar.map{ it[1] }\n",
      "Nextflow operator mix with content: ch_versions  = ch_versions.mix(UNTAR_BOWTIE2_INDEX.out.versions)\n",
      "Nextflow operator mix with content: ch_versions      = ch_versions.mix(BOWTIE2_BUILD.out.versions)\n",
      "Nextflow operator map with content: ch_chromap_index = UNTAR_CHROMAP_INDEX ( [ [:], params.chromap_index ] ).untar.map{ it[1] }\n",
      "Nextflow operator mix with content: ch_versions  = ch_versions.mix(UNTAR.out.versions)\n",
      "Nextflow operator mix with content: ch_versions  = ch_versions.mix(CHROMAP_INDEX.out.versions)\n",
      "Nextflow operator map with content: ch_star_index = UNTAR_STAR_INDEX ( [ [:], params.star_index ] ).untar.map{ it[1] }\n",
      "Nextflow operator mix with content: ch_versions   = ch_versions.mix(UNTAR_STAR_INDEX.out.versions)\n",
      "Nextflow operator mix with content: ch_versions   = ch_versions.mix(STAR_GENOMEGENERATE.out.versions)\n",
      "Nextflow operator ifEmpty with content: versions    = ch_versions.ifEmpty(null) // channel: [ versions.yml ]\n",
      "Error: Unclosed process block in file: ./git_repos\\chipseq\\subworkflows\\local\\prepare_genome.nf\n",
      "Process ALIGN_BOWTIE2 in filepath ./git_repos\\chipseq\\subworkflows\\nf-core\\align_bowtie2.nf\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(BOWTIE2_ALIGN.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(BAM_SORT_SAMTOOLS.out.versions.first())\n",
      "Error: Unclosed process block in file: ./git_repos\\chipseq\\subworkflows\\nf-core\\align_bowtie2.nf\n",
      "Process ALIGN_BWA_MEM in filepath ./git_repos\\chipseq\\subworkflows\\nf-core\\align_bwa_mem.nf\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(BWA_MEM.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(BAM_SORT_SAMTOOLS.out.versions.first())\n",
      "Error: Unclosed process block in file: ./git_repos\\chipseq\\subworkflows\\nf-core\\align_bwa_mem.nf\n",
      "Process ALIGN_CHROMAP in filepath ./git_repos\\chipseq\\subworkflows\\nf-core\\align_chromap.nf\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(CHROMAP_CHROMAP.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(BAM_SORT_SAMTOOLS.out.versions.first())\n",
      "Error: Unclosed process block in file: ./git_repos\\chipseq\\subworkflows\\nf-core\\align_chromap.nf\n",
      "Process ALIGN_STAR in filepath ./git_repos\\chipseq\\subworkflows\\nf-core\\align_star.nf\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(STAR_ALIGN.out.versions.first())\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BAM_SORT_SAMTOOLS.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\chipseq\\subworkflows\\nf-core\\align_star.nf\n",
      "Process BAM_SORT_SAMTOOLS in filepath ./git_repos\\chipseq\\subworkflows\\nf-core\\bam_sort_samtools.nf\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SAMTOOLS_SORT.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SAMTOOLS_INDEX.out.versions.first())\n",
      "Nextflow operator join with content: BAM_STATS_SAMTOOLS(SAMTOOLS_SORT.out.bam.join(SAMTOOLS_INDEX.out.bai, by: [0]))\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BAM_STATS_SAMTOOLS.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\chipseq\\subworkflows\\nf-core\\bam_sort_samtools.nf\n",
      "Process BAM_STATS_SAMTOOLS in filepath ./git_repos\\chipseq\\subworkflows\\nf-core\\bam_stats_samtools.nf\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SAMTOOLS_STATS.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SAMTOOLS_FLAGSTAT.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SAMTOOLS_IDXSTATS.out.versions.first())\n",
      "Error: Unclosed process block in file: ./git_repos\\chipseq\\subworkflows\\nf-core\\bam_stats_samtools.nf\n",
      "Process FASTQC_TRIMGALORE in filepath ./git_repos\\chipseq\\subworkflows\\nf-core\\fastqc_trimgalore.nf\n",
      "Nextflow operator set with content: FASTQC ( reads ).html.set { fastqc_html }\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(FASTQC.out.versions.first())\n",
      "Nextflow operator set with content: TRIMGALORE ( reads ).reads.set { trim_reads }\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(TRIMGALORE.out.versions.first())\n",
      "Nextflow operator ifEmpty with content: versions = ch_versions.ifEmpty(null) // channel: [ versions.yml ]\n",
      "Error: Unclosed process block in file: ./git_repos\\chipseq\\subworkflows\\nf-core\\fastqc_trimgalore.nf\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Process MARK_DUPLICATES_PICARD in filepath ./git_repos\\chipseq\\subworkflows\\nf-core\\mark_duplicates_picard.nf\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(PICARD_MARKDUPLICATES.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SAMTOOLS_INDEX.out.versions.first())\n",
      "Nextflow operator join with content: BAM_STATS_SAMTOOLS(PICARD_MARKDUPLICATES.out.bam.join(SAMTOOLS_INDEX.out.bai, by: [0]))\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BAM_STATS_SAMTOOLS.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\chipseq\\subworkflows\\nf-core\\mark_duplicates_picard.nf\n",
      "Process CHIPSEQ in filepath ./git_repos\\chipseq\\workflows\\chipseq.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(PREPARE_GENOME.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(INPUT_CHECK.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(FASTQC_TRIMGALORE.out.versions)\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(ALIGN_BWA_MEM.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(ALIGN_BOWTIE2.out.versions.first())\n",
      "Nextflow operator branch with content: .branch {\n",
      "Nextflow operator set with content: .set { ch_genome_bam_chromap }\n",
      "Nextflow operator collect with content: .collect()\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(ALIGN_CHROMAP.out.versions.first())\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ALIGN_STAR.out.versions)\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator join with content: meta_clone.id = meta_clone.id.split('_')[0..-2].join('_')\n",
      "Nextflow operator groupTuple with content: .groupTuple(by: [0])\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator flatten with content: [ it[0], it[1].flatten() ]\n",
      "Nextflow operator set with content: .set { ch_sort_bam }\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(PICARD_MERGESAMFILES.out.versions.first().ifEmpty(null))\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(MARK_DUPLICATES_PICARD.out.versions)\n",
      "Nextflow operator join with content: MARK_DUPLICATES_PICARD.out.bam.join(MARK_DUPLICATES_PICARD.out.bai, by: [0]),\n",
      "Nextflow operator first with content: PREPARE_GENOME.out.filtered_bed.first(),\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(FILTER_BAM_BAMTOOLS.out.versions.first().ifEmpty(null))\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(PRESEQ_LCEXTRAP.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(PICARD_COLLECTMULTIPLEMETRICS.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(PHANTOMPEAKQUALTOOLS.out.versions.first())\n",
      "Nextflow operator join with content: PHANTOMPEAKQUALTOOLS.out.spp.join(PHANTOMPEAKQUALTOOLS.out.rdata, by: [0]),\n",
      "Nextflow operator join with content: FILTER_BAM_BAMTOOLS.out.bam.join(FILTER_BAM_BAMTOOLS.out.flagstat, by: [0])\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(BEDTOOLS_GENOMECOV.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(UCSC_BEDGRAPHTOBIGWIG.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(DEEPTOOLS_COMPUTEMATRIX.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(DEEPTOOLS_PLOTPROFILE.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(DEEPTOOLS_PLOTHEATMAP.out.versions.first())\n",
      "Nextflow operator join with content: .join(FILTER_BAM_BAMTOOLS.out.bai, by: [0])\n",
      "Nextflow operator set with content: .set { ch_genome_bam_bai }\n",
      "Nextflow operator combine with content: .combine(ch_genome_bam_bai)\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator set with content: .set { ch_ip_control_bam_bai }\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(DEEPTOOLS_PLOTFINGERPRINT.out.versions.first())\n",
      "Nextflow operator map with content: ch_macs_gsize = KHMER_UNIQUEKMERS.out.kmers.map { it.text.trim() }\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator set with content: .set { ch_ip_control_bam }\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(MACS2_CALLPEAK.out.versions.first())\n",
      "Nextflow operator filter with content: .filter { meta, peaks -> peaks.size() > 0 }\n",
      "Nextflow operator set with content: .set { ch_macs2_peaks }\n",
      "Nextflow operator join with content: .join(ch_macs2_peaks, by: [0])\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator set with content: .set { ch_ip_bam_peaks }\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(FRIP_SCORE.out.versions.first())\n",
      "Nextflow operator join with content: .join(FRIP_SCORE.out.txt, by: [0])\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator set with content: .set { ch_ip_peaks_frip }\n",
      "Nextflow operator count with content: ch_custompeaks_count_multiqc = MULTIQC_CUSTOM_PEAKS.out.count\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(HOMER_ANNOTATEPEAKS_MACS2.out.versions.first())\n",
      "Nextflow operator collect with content: ch_macs2_peaks.collect{it[1]}\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(PLOT_MACS2_QC.out.versions)\n",
      "Nextflow operator collect with content: HOMER_ANNOTATEPEAKS_MACS2.out.txt.collect{it[1]},\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(PLOT_HOMER_ANNOTATEPEAKS.out.versions)\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator join with content: [ meta.antibody, meta.id.split('_')[0..-2].join('_'), peak ]\n",
      "Nextflow operator groupTuple with content: .groupTuple()\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator max with content: meta_new.replicates_exist = groups.max { groups.value }.value > 1\n",
      "Nextflow operator set with content: .set { ch_antibody_peaks }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(MACS2_CONSENSUS.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(HOMER_ANNOTATEPEAKS_CONSENSUS.out.versions)\n",
      "Nextflow operator join with content: MACS2_CONSENSUS.out.boolean_txt.join(HOMER_ANNOTATEPEAKS_CONSENSUS.out.txt, by: [0]),\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ANNOTATE_BOOLEAN_PEAKS.out.versions)\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator groupTuple with content: .groupTuple()\n",
      "Nextflow operator set with content: .set { ch_antibody_bams }\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator join with content: .join(ch_antibody_bams)\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator flatten with content: [ meta, bams.flatten().sort(), saf ]\n",
      "Nextflow operator set with content: .set { ch_saf_bams }\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SUBREAD_FEATURECOUNTS.out.versions.first())\n",
      "Nextflow operator collect with content: UCSC_BEDGRAPHTOBIGWIG.out.bigwig.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_macs2_peaks.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_macs2_consensus_bed_lib.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_macs2_consensus_txt_lib.collect{it[1]}.ifEmpty([])\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(IGV.out.versions)\n",
      "Nextflow operator collectFile with content: ch_versions.unique().collectFile(name: 'collated_versions.yml')\n",
      "Nextflow operator collect with content: ch_multiqc_custom_config.collect().ifEmpty([]),\n",
      "Nextflow operator collect with content: CUSTOM_DUMPSOFTWAREVERSIONS.out.mqc_yml.collect(),\n",
      "Nextflow operator collectFile with content: ch_workflow_summary.collectFile(name: 'workflow_summary_mqc.yaml'),\n",
      "Nextflow operator collect with content: FASTQC_TRIMGALORE.out.fastqc_zip.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: FASTQC_TRIMGALORE.out.trim_zip.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: FASTQC_TRIMGALORE.out.trim_log.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_samtools_stats.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_samtools_flagstat.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_samtools_idxstats.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: MARK_DUPLICATES_PICARD.out.stats.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: MARK_DUPLICATES_PICARD.out.flagstat.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: MARK_DUPLICATES_PICARD.out.idxstats.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: MARK_DUPLICATES_PICARD.out.metrics.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: FILTER_BAM_BAMTOOLS.out.stats.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: FILTER_BAM_BAMTOOLS.out.flagstat.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: FILTER_BAM_BAMTOOLS.out.idxstats.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_picardcollectmultiplemetrics_multiqc.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_preseq_multiqc.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_deeptoolsplotprofile_multiqc.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_deeptoolsplotfingerprint_multiqc.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: PHANTOMPEAKQUALTOOLS.out.spp.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: MULTIQC_CUSTOM_PHANTOMPEAKQUALTOOLS.out.nsc.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: MULTIQC_CUSTOM_PHANTOMPEAKQUALTOOLS.out.rsc.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: MULTIQC_CUSTOM_PHANTOMPEAKQUALTOOLS.out.correlation.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_custompeaks_frip_multiqc.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_custompeaks_count_multiqc.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_plothomerannotatepeaks_multiqc.collect().ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_subreadfeaturecounts_multiqc.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_deseq2_pca_multiqc.collect().ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_deseq2_clustering_multiqc.collect().ifEmpty([])\n",
      "Nextflow operator toList with content: multiqc_report = MULTIQC.out.report.toList()\n",
      "Error: Unclosed process block in file: ./git_repos\\chipseq\\workflows\\chipseq.nf\n",
      "Process NFCORE_CIRCDNA in filepath ./git_repos\\circdna\\main.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\circdna\\main.nf\n",
      "Process INPUT_CHECK in filepath ./git_repos\\circdna\\subworkflows\\local\\input_check.nf\n",
      "Nextflow operator splitCsv with content: .splitCsv ( header:true, sep:',' )\n",
      "Nextflow operator map with content: .map { create_fastq_channels(it) }\n",
      "Nextflow operator set with content: .set { reads }\n",
      "Nextflow operator splitCsv with content: .splitCsv ( header:true, sep:',' )\n",
      "Nextflow operator map with content: .map { create_bam_channels(it) }\n",
      "Nextflow operator set with content: .set { reads }\n",
      "Error: Unclosed process block in file: ./git_repos\\circdna\\subworkflows\\local\\input_check.nf\n",
      "Process BAM_MARKDUPLICATES_PICARD in filepath ./git_repos\\circdna\\subworkflows\\nf-core\\bam_markduplicates_picard\\main.nf\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(PICARD_MARKDUPLICATES.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SAMTOOLS_INDEX.out.versions.first())\n",
      "Nextflow operator join with content: .join(SAMTOOLS_INDEX.out.bai, by: [0], remainder: true)\n",
      "Nextflow operator join with content: .join(SAMTOOLS_INDEX.out.csi, by: [0], remainder: true)\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BAM_STATS_SAMTOOLS.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\circdna\\subworkflows\\nf-core\\bam_markduplicates_picard\\main.nf\n",
      "Process BAM_STATS_SAMTOOLS in filepath ./git_repos\\circdna\\subworkflows\\nf-core\\bam_stats_samtools\\main.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_STATS.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_FLAGSTAT.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_IDXSTATS.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\circdna\\subworkflows\\nf-core\\bam_stats_samtools\\main.nf\n",
      "Process CIRCDNA in filepath ./git_repos\\circdna\\workflows\\circdna.nf\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator join with content: meta.id = meta.id.split('_')[0..-2].join('_')\n",
      "Nextflow operator groupTuple with content: .groupTuple(by: [0])\n",
      "Nextflow operator branch with content: .branch {\n",
      "Nextflow operator flatten with content: return [ meta, fastq.flatten() ]\n",
      "Nextflow operator flatten with content: return [ meta, fastq.flatten() ]\n",
      "Nextflow operator set with content: .set { ch_fastq }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(INPUT_CHECK.out.versions)\n",
      "Nextflow operator mix with content: .mix(ch_fastq.single)\n",
      "Nextflow operator set with content: .set { ch_cat_fastq }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(CAT_FASTQ.out.versions)\n",
      "Nextflow operator mix with content: ch_versions         = ch_versions.mix(FASTQC.out.versions)\n",
      "Nextflow operator mix with content: ch_versions                 = ch_versions.mix(TRIMGALORE.out.versions)\n",
      "Nextflow operator collect with content: ch_bwa_index = BWA_INDEX.out.index.map{ meta, index -> [\"bwa_index\", index] }.collect()\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BWA_INDEX.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BWA_MEM.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_INDEX_BAM.out.versions)\n",
      "Nextflow operator mix with content: ch_versions         = ch_versions.mix(SAMTOOLS_SORT_BAM.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_INDEX_BAM.out.versions)\n",
      "Nextflow operator collect with content: ch_fasta = ch_fasta_meta.map{ meta, index -> [index] }.collect()\n",
      "Nextflow operator join with content: ch_bam_sorted.join(ch_bam_sorted_bai).\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BAM_STATS_SAMTOOLS.out.versions)\n",
      "Nextflow operator collect with content: SAMTOOLS_FAIDX.out.fai.collect()\n",
      "Nextflow operator join with content: ch_bam_sorted.join(ch_bam_sorted_bai),\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_VIEW_FILTER.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_SORT_FILTERED.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_INDEX_FILTERED.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BAM_MARKDUPLICATES_PICARD.out.versions)\n",
      "Nextflow operator join with content: ch_bam_sorted.join(ch_bam_sorted_bai),\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(CNVKIT_BATCH.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(CNVKIT_SEGMENT.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(COLLECT_SEEDS.out.versions)\n",
      "Nextflow operator join with content: ch_aa_seeds.join(ch_bam_sorted).join(ch_bam_sorted_bai)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(AMPLIFIED_INTERVALS.out.versions)\n",
      "Nextflow operator join with content: ch_bam_sorted.join(ch_bam_sorted_bai).\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(AMPLICONARCHITECT_AMPLICONARCHITECT.out.versions)\n",
      "Nextflow operator collect with content: ch_aa_graphs.flatten().collect().ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_aa_cycles.flatten().collect().ifEmpty([])\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(AMPLICONCLASSIFIER_AMPLICONCLASSIFIER.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(AMPLICONCLASSIFIER_AMPLICONSIMILARITY.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(AMPLICONCLASSIFIER_MAKERESULTSTABLE.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_SORT_QNAME_CF.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMBLASTER.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BEDTOOLS_SPLITBAM2BED.out.versions)\n",
      "Nextflow operator join with content: ch_full_bam_sorted.join(ch_full_bam_sorted_bai)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BEDTOOLS_SORTEDBAM2BED.out.versions)\n",
      "Nextflow operator join with content: ch_b2b_split.join(ch_b2b_sorted)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_SORT_QNAME_CM.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(CIRCLEMAP_READEXTRACTOR.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_SORT_RE.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_INDEX_RE.out.versions)\n",
      "Nextflow operator join with content: ch_re_sorted_bam.join(ch_re_sorted_bai)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(CIRCLEMAP_REPEATS.out.versions)\n",
      "Nextflow operator join with content: ch_re_sorted_bam.join(ch_re_sorted_bai).\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(CIRCLEMAP_REALIGN.out.versions)\n",
      "Nextflow operator join with content: ch_bam_sorted.join(ch_bam_sorted_bai)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(CIRCEXPLORER2_PARSE.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(UNICYCLER.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SEQTK_SEQ.out.versions)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Nextflow operator map with content: .map { meta, fastq -> [ meta + [single_end: true], fastq ] }\n",
      "Nextflow operator set with content: .set { ch_circular_fastq }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(MINIMAP2_ALIGN.out.versions)\n",
      "Nextflow operator collectFile with content: ch_versions.unique().collectFile(name: 'collated_versions.yml')\n",
      "Nextflow operator collect with content: ch_multiqc_custom_config.collect().ifEmpty([]),\n",
      "Nextflow operator collect with content: CUSTOM_DUMPSOFTWAREVERSIONS.out.mqc_yml.collect(),\n",
      "Nextflow operator collectFile with content: ch_workflow_summary.collectFile(name: 'workflow_summary_mqc.yaml'),\n",
      "Nextflow operator collect with content: ch_fastqc_multiqc.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_trimgalore_multiqc.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_trimgalore_multiqc_log.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_samtools_stats.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_samtools_flagstat.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_samtools_idxstats.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_markduplicates_flagstat.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_markduplicates_stats.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_markduplicates_idxstats.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_markduplicates_multiqc.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator toList with content: multiqc_report       = MULTIQC.out.report.toList()\n",
      "Error: Unclosed process block in file: ./git_repos\\circdna\\workflows\\circdna.nf\n",
      "Process NFCORE_CIRCRNA in filepath ./git_repos\\circrna\\main.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\circrna\\main.nf\n",
      "Process CIRCRNA_DISCOVERY in filepath ./git_repos\\circrna\\subworkflows\\local\\circrna_discovery.nf\n",
      "Nextflow operator map with content: segemehl_filter = SEGEMEHL_ALIGN.out.results.map{ meta, results ->  [ meta + [tool: \"segemehl\"], results ] }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SEGEMEHL_ALIGN.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SEGEMEHL_FILTER.out.versions)\n",
      "Nextflow operator collect with content: sjdb = STAR_1ST_PASS.out.tab.map{ meta, tab -> return tab }.collect().map{[[id: \"star_sjdb\"], it]}\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(STAR_1ST_PASS.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(STAR_2ND_PASS.out.versions)\n",
      "Nextflow operator map with content: circexplorer2_filter = CIRCEXPLORER2_ANN.out.txt.map{ meta, txt -> [ meta + [tool: \"circexplorer2\"], txt ] }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(CIRCEXPLORER2_REF.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(CIRCEXPLORER2_PAR.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(CIRCEXPLORER2_ANN.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(CIRCEXPLORER2_FLT.out.versions)\n",
      "Nextflow operator join with content: circrna_finder_stage = STAR_2ND_PASS.out.sam.join( STAR_2ND_PASS.out.junction).join(STAR_2ND_PASS.out.tab)\n",
      "Nextflow operator map with content: circrna_finder_filter = circrna_finder_stage.map{ meta, sam, junction, tab -> [ meta + [tool: \"circrna_finder\"], sam, junction, tab ] }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(CIRCRNA_FINDER_FILTER.out.versions)\n",
      "Nextflow operator collect with content: FIND_CIRC_ALIGN( reads, bowtie2_index.collect(), false, true )\n",
      "Nextflow operator join with content: SAMTOOLS_VIEW( FIND_CIRC_ALIGN.out.aligned.join( SAMTOOLS_INDEX.out.bai ), fasta_tuple, [] )\n",
      "Nextflow operator collect with content: FIND_CIRC( FIND_CIRC_ANCHORS.out.anchors, bowtie2_index.collect(), fasta )\n",
      "Nextflow operator map with content: find_circ_filter = FIND_CIRC.out.bed.map{ meta, bed -> [ meta + [tool: \"find_circ\"], bed ] }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(FIND_CIRC_ALIGN.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_INDEX.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_VIEW.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(FIND_CIRC_ANCHORS.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(FIND_CIRC.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(FIND_CIRC_FILTER.out.versions)\n",
      "Nextflow operator map with content: CIRIQUANT_YML( gtf, fasta, bwa_index.map{ meta, index -> return index }, hisat2_index.map{ meta, index -> return index } )\n",
      "Nextflow operator collect with content: CIRIQUANT( reads, CIRIQUANT_YML.out.yml.collect() )\n",
      "Nextflow operator map with content: CIRIQUANT_FILTER( CIRIQUANT.out.gtf.map{ meta, gtf -> [ meta + [tool: \"ciriquant\"], gtf ] }, bsj_reads )\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(CIRIQUANT.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(CIRIQUANT_FILTER.out.versions)\n",
      "Nextflow operator filter with content: mate1 = reads.filter{ meta, reads -> !meta.single_end }.map{ meta, reads -> return [ [id: meta.id, single_end: true], reads[0] ] }\n",
      "Nextflow operator collect with content: DCC_MATE1_SJDB( DCC_MATE1_1ST_PASS.out.tab.map{ meta, tab -> return tab }.collect().map{[[id: \"mate1_sjdb\"], it]}, bsj_reads )\n",
      "Nextflow operator filter with content: mate2 = reads.filter{ meta, reads -> !meta.single_end }.map{ meta, reads -> return [ [id: meta.id, single_end: true], reads[1] ] }\n",
      "Nextflow operator collect with content: DCC_MATE2_SJDB( DCC_MATE2_1ST_PASS.out.tab.map{ meta, tab -> return tab }.collect().map{[[id: \"mate2_sjdb\"], it]}, bsj_reads )\n",
      "Nextflow operator map with content: dcc_stage = STAR_2ND_PASS.out.junction.map{ meta, junction -> return [ meta.id, meta, junction]}\n",
      "Nextflow operator join with content: .join(\n",
      "Nextflow operator map with content: DCC_MATE1_2ND_PASS.out.junction.map{ meta, junction -> return [ meta.id, junction] },\n",
      "Nextflow operator join with content: .join(\n",
      "Nextflow operator map with content: DCC_MATE2_2ND_PASS.out.junction.map{ meta, junction -> return [ meta.id, junction] },\n",
      "Nextflow operator map with content: .map{ id, meta, junction, mate1, mate2 -> return [ meta, junction, mate1, mate2 ]}\n",
      "Nextflow operator map with content: dcc = dcc_stage.map{ it ->  [ it[0], it[1], it[2] ?: [], it[3] ?: [] ] }\n",
      "Nextflow operator map with content: DCC_FILTER( DCC.out.txt.map{ meta, txt -> [ meta + [tool: \"dcc\"], txt ] }, bsj_reads )\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(DCC_MATE1_1ST_PASS.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(DCC_MATE1_SJDB.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(DCC_MATE1_2ND_PASS.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(DCC_MATE2_1ST_PASS.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(DCC_MATE2_SJDB.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(DCC_MATE2_2ND_PASS.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(DCC.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(DCC_FILTER.out.versions)\n",
      "Nextflow operator collect with content: MAPSPLICE_ALIGN( reads, bowtie_index.collect(), chromosomes, ch_gtf )\n",
      "Nextflow operator map with content: mapsplice_filter = MAPSPLICE_ANNOTATE.out.txt.map{ meta, txt -> [ meta + [tool: \"mapsplice\"], txt ] }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(MAPSPLICE_REFERENCE.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(MAPSPLICE_ALIGN.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(MAPSPLICE_PARSE.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(MAPSPLICE_ANNOTATE.out.versions)\n",
      "Nextflow operator mix with content: circrna_filtered = CIRCEXPLORER2_FLT.out.results.mix(SEGEMEHL_FILTER.out.results,\n",
      "Nextflow operator collect with content: ANNOTATION( circrna_filtered, gtf, ch_biotypes.collect(), exon_boundary )\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ANNOTATION.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(FASTA.out.versions)\n",
      "Nextflow operator mix with content: ch_matrix = CIRCEXPLORER2_FLT.out.matrix.mix(SEGEMEHL_FILTER.out.matrix,\n",
      "Nextflow operator collect with content: tools_selected = params.tool.split(',').collect{it.trim().toLowerCase()}\n",
      "Nextflow operator groupTuple with content: MERGE_TOOLS( ch_matrix.map{ meta, bed -> [ [id: meta.id], bed ] }.groupTuple(), tool_filter, duplicates_fun )\n",
      "Nextflow operator collect with content: COUNTS_COMBINED( MERGE_TOOLS.out.merged.map{ meta, bed -> return [ bed ] }.collect() )\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(MERGE_TOOLS.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(COUNTS_COMBINED.out.versions)\n",
      "Nextflow operator groupTuple with content: COUNTS_SINGLE( ch_matrix.map{ meta, bed -> [ [tool: meta.tool], bed ] }.groupTuple() )\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(COUNTS_SINGLE.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\circrna\\subworkflows\\local\\circrna_discovery.nf\n",
      "Process DIFFERENTIAL_EXPRESSION in filepath ./git_repos\\circrna\\subworkflows\\local\\differential_expression.nf\n",
      "Nextflow operator map with content: ch_fasta.map{ it -> [ [id: it.simpleName], [it] ]\n",
      "Nextflow operator set with content: }.set{ fasta_tuple }\n",
      "Nextflow operator map with content: ch_gtf.map{ it -> [ [id: it.simpleName], [it] ]\n",
      "Nextflow operator set with content: }.set{ gtf_tuple }\n",
      "Nextflow operator collect with content: STRINGTIE_PREPDE( STRINGTIE_STRINGTIE.out.transcript_gtf.map{ meta, gtf -> return [ gtf ] }.collect() )\n",
      "Nextflow operator map with content: qc_reports = qc_reports.mix(HISAT2_ALIGN.out.summary.map{ meta, log -> log})\n",
      "Nextflow operator map with content: qc_reports = qc_reports.mix(BAM_SORT_STATS_SAMTOOLS.out.stats.map{ meta, stats -> stats})\n",
      "Nextflow operator map with content: qc_reports = qc_reports.mix(BAM_SORT_STATS_SAMTOOLS.out.flagstat.map{ meta, flagstat -> flagstat})\n",
      "Nextflow operator map with content: qc_reports = qc_reports.mix(BAM_SORT_STATS_SAMTOOLS.out.idxstats.map{ meta, idxstats -> idxstats})\n",
      "Nextflow operator collect with content: PARENT_GENE( clr_matrix, gtf, ch_biotypes.collect(), exon_boundary )\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(HISAT2_ALIGN.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BAM_SORT_STATS_SAMTOOLS.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(STRINGTIE_STRINGTIE.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(STRINGTIE_PREPDE.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(DESEQ2_DIFFERENTIAL_EXPRESSION.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(PARENT_GENE.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(PREPARE_CLR_TEST.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(CIRCTEST.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\circrna\\subworkflows\\local\\differential_expression.nf\n",
      "Process INPUT_CHECK in filepath ./git_repos\\circrna\\subworkflows\\local\\input_check.nf\n",
      "Nextflow operator splitCsv with content: .splitCsv ( header:true, sep:',' )\n",
      "Nextflow operator map with content: .map { create_fastq_channel(it) }\n",
      "Nextflow operator set with content: .set { reads }\n",
      "Nextflow operator splitCsv with content: .splitCsv(header: true, sep: ',')\n",
      "Nextflow operator map with content: .map{ row ->\n",
      "Nextflow operator toList with content: .toList()\n",
      "Error: Unclosed process block in file: ./git_repos\\circrna\\subworkflows\\local\\input_check.nf\n",
      "Process MIRNA_PREDICTION in filepath ./git_repos\\circrna\\subworkflows\\local\\mirna_prediction.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(TARGETSCAN_DATABASE.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(TARGETSCAN.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(MIRANDA.out.versions)\n",
      "Nextflow operator join with content: consolidate_targets = TARGETSCAN.out.txt.join(MIRANDA.out.txt).join(circrna_bed12)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(MIRNA_TARGETS.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\circrna\\subworkflows\\local\\mirna_prediction.nf\n",
      "Process PREPARE_GENOME in filepath ./git_repos\\circrna\\subworkflows\\local\\prepare_genome.nf\n",
      "Nextflow operator splitFasta with content: ch_fasta.splitFasta( record: [id:true] )\n",
      "Nextflow operator map with content: .map{ record -> record.id.toString() }\n",
      "Nextflow operator set with content: .set{ ID }\n",
      "Nextflow operator splitFasta with content: ch_fasta.splitFasta( file: true )\n",
      "Nextflow operator map with content: .merge( ID ).map{ file, id -> file.copyTo(directory + \"/${id}.fa\") }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BOWTIE_BUILD.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BOWTIE2_BUILD.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BWA_INDEX.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(HISAT2_EXTRACTSPLICESITES.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(HISAT2_BUILD.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SEGEMEHL_INDEX.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(STAR_GENOMEGENERATE.out.versions)\n",
      "Nextflow operator collect with content: hisat2       = HISAT2_BUILD.out.index.collect()\n",
      "Nextflow operator collect with content: star         = STAR_GENOMEGENERATE.out.index.collect()\n",
      "Nextflow operator collect with content: splice_sites = HISAT2_EXTRACTSPLICESITES.out.txt.collect()\n",
      "Error: Unclosed process block in file: ./git_repos\\circrna\\subworkflows\\local\\prepare_genome.nf\n",
      "Process FASTQC_TRIMGALORE in filepath ./git_repos\\circrna\\subworkflows\\nf-core\\fastqc_trimgalore.nf\n",
      "Nextflow operator set with content: FASTQC ( reads ).html.set { fastqc_html }\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(FASTQC.out.versions.first())\n",
      "Nextflow operator set with content: TRIMGALORE ( reads ).reads.set { trim_reads }\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(TRIMGALORE.out.versions.first())\n",
      "Error: Unclosed process block in file: ./git_repos\\circrna\\subworkflows\\nf-core\\fastqc_trimgalore.nf\n",
      "Process BAM_SORT_STATS_SAMTOOLS in filepath ./git_repos\\circrna\\subworkflows\\nf-core\\bam_sort_stats_samtools\\main.nf\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SAMTOOLS_SORT.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SAMTOOLS_INDEX.out.versions.first())\n",
      "Nextflow operator join with content: .join(SAMTOOLS_INDEX.out.bai, by: [0], remainder: true)\n",
      "Nextflow operator join with content: .join(SAMTOOLS_INDEX.out.csi, by: [0], remainder: true)\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator set with content: .set { ch_bam_bai }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BAM_STATS_SAMTOOLS.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\circrna\\subworkflows\\nf-core\\bam_sort_stats_samtools\\main.nf\n",
      "Process BAM_STATS_SAMTOOLS in filepath ./git_repos\\circrna\\subworkflows\\nf-core\\bam_stats_samtools\\main.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_STATS.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_FLAGSTAT.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_IDXSTATS.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\circrna\\subworkflows\\nf-core\\bam_stats_samtools\\main.nf\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Process CIRCRNA in filepath ./git_repos\\circrna\\workflows\\circrna.nf\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator join with content: meta.id = meta.id.split('_')[0..-2].join('_')\n",
      "Nextflow operator groupTuple with content: .groupTuple(by: [0])\n",
      "Nextflow operator branch with content: .branch {\n",
      "Nextflow operator flatten with content: return [ meta, fastq.flatten() ]\n",
      "Nextflow operator flatten with content: return [ meta, fastq.flatten() ]\n",
      "Nextflow operator set with content: .set { ch_fastq }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(INPUT_CHECK.out.versions)\n",
      "Nextflow operator mix with content: .mix(ch_fastq.single)\n",
      "Nextflow operator set with content: .set { ch_cat_fastq }\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(CAT_FASTQ.out.versions.first().ifEmpty(null))\n",
      "Nextflow operator map with content: bowtie2_index  = params.fasta ? params.bowtie2 ? Channel.fromPath(params.bowtie2).map{ it -> [[id:'bowtie2'], it] } : PREPARE_GENOME.out.bowtie2 : []\n",
      "Nextflow operator map with content: bwa_index      = params.fasta ? params.bwa ? Channel.fromPath(params.bwa).map{ it -> [[id:'bwa'], it] } : PREPARE_GENOME.out.bwa : []\n",
      "Nextflow operator map with content: hisat2_index   = params.fasta ? params.hisat2 && ( params.tool.contains('ciriquant') || params.module.contains('differential_expression') ) ? Channel.fromPath(params.hisat2).map{ [[id:\"hisat2\"], it]} : PREPARE_GENOME.out.hisat2 : []\n",
      "Nextflow operator map with content: star_index     = params.fasta ? params.star ? Channel.fromPath(params.star).map{[[id:'star'], it]}: PREPARE_GENOME.out.star : []\n",
      "Nextflow operator mix with content: ch_versions    = ch_versions.mix(PREPARE_GENOME.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(FASTQC_TRIMGALORE.out.versions)\n",
      "Nextflow operator collect with content: ch_reports  = ch_reports.mix(FASTQC_TRIMGALORE.out.trim_zip.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_reports  = ch_reports.mix(FASTQC_TRIMGALORE.out.trim_log.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(CIRCRNA_DISCOVERY.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(MIRNA_PREDICTION.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(DIFFERENTIAL_EXPRESSION.out.versions)\n",
      "Nextflow operator mix with content: ch_reports  = ch_reports.mix(DIFFERENTIAL_EXPRESSION.out.reports)\n",
      "Nextflow operator collectFile with content: ch_versions.unique().collectFile(name: 'collated_versions.yml')\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_workflow_summary.collectFile(name: 'workflow_summary_mqc.yaml'))\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_methods_description.collectFile(name: 'methods_description_mqc.yaml'))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(CUSTOM_DUMPSOFTWAREVERSIONS.out.mqc_yml.collect())\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(FASTQC_TRIMGALORE.out.fastqc_zip.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(ch_reports.collect().ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files.collect(),\n",
      "Nextflow operator toList with content: ch_multiqc_config.toList(),\n",
      "Nextflow operator toList with content: ch_multiqc_custom_config.toList(),\n",
      "Nextflow operator toList with content: ch_multiqc_logo.toList()\n",
      "Nextflow operator toList with content: multiqc_report = MULTIQC.out.report.toList()\n",
      "Error: Unclosed process block in file: ./git_repos\\circrna\\workflows\\circrna.nf\n",
      "Process NFCORE_CREATEPANELREFS in filepath ./git_repos\\createpanelrefs\\main.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\createpanelrefs\\main.nf\n",
      "Process GERMLINECNVCALLER_COHORT in filepath ./git_repos\\createpanelrefs\\subworkflows\\local\\germlinecnvcaller_cohort.nf\n",
      "Nextflow operator mix with content: .mix(PICARD_CREATESEQUENCEDICTIONARY.out.reference_dict)\n",
      "Nextflow operator collect with content: .collect()\n",
      "Nextflow operator set with content: .set { ch_dict }\n",
      "Nextflow operator mix with content: .mix(SAMTOOLS_FAIDX.out.fai)\n",
      "Nextflow operator collect with content: .collect()\n",
      "Nextflow operator set with content: .set { ch_fai }\n",
      "Nextflow operator branch with content: .branch { meta, alignment, index ->\n",
      "Nextflow operator set with content: .set { ch_for_mix }\n",
      "Nextflow operator mix with content: .mix(SAMTOOLS_INDEX.out.crai)\n",
      "Nextflow operator set with content: .set { ch_index }\n",
      "Nextflow operator join with content: .join(ch_index)\n",
      "Nextflow operator mix with content: .mix(ch_for_mix.alignment_with_index)\n",
      "Nextflow operator combine with content: .combine(GATK4_PREPROCESSINTERVALS.out.interval_list.map{it -> it[1]})\n",
      "Nextflow operator set with content: .set {ch_readcounts_in}\n",
      "Nextflow operator mix with content: .mix(GATK4_COLLECTREADCOUNTS.out.hdf5)\n",
      "Nextflow operator collect with content: .collect { it[1] }\n",
      "Nextflow operator map with content: .map {tsv -> [[id:'cohort'],tsv]}\n",
      "Nextflow operator set with content: .set { ch_readcounts_out }\n",
      "Nextflow operator map with content: .map {meta, it -> it}\n",
      "Nextflow operator flatten with content: .flatten()\n",
      "Nextflow operator set with content: .set { ch_intervallist_out }\n",
      "Nextflow operator combine with content: .combine(GATK4_FILTERINTERVALS.out.interval_list)\n",
      "Nextflow operator map with content: .map{ meta, counts, meta2, il -> [meta, counts, il, []] }\n",
      "Nextflow operator set with content: .set {ch_contigploidy_in}\n",
      "Nextflow operator combine with content: .combine(ch_intervallist_out)\n",
      "Nextflow operator combine with content: .combine(GATK4_DETERMINEGERMLINECONTIGPLOIDY.out.calls)\n",
      "Nextflow operator map with content: .map{ meta, counts, il, meta2, calls -> [meta + [id:il.baseName],  counts, il, calls, []] }\n",
      "Nextflow operator set with content: .set {ch_cnvcaller_in}\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_FAIDX.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(PICARD_CREATESEQUENCEDICTIONARY.out.versions)\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SAMTOOLS_INDEX.out.versions.first())\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GATK4_PREPROCESSINTERVALS.out.versions)\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(GATK4_COLLECTREADCOUNTS.out.versions.first())\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GATK4_ANNOTATEINTERVALS.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GATK4_FILTERINTERVALS.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GATK4_INTERVALLISTTOOLS.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GATK4_DETERMINEGERMLINECONTIGPLOIDY.out.versions)\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(GATK4_GERMLINECNVCALLER.out.versions.first())\n",
      "Error: Unclosed process block in file: ./git_repos\\createpanelrefs\\subworkflows\\local\\germlinecnvcaller_cohort.nf\n",
      "Process CREATEPANELREFS in filepath ./git_repos\\createpanelrefs\\workflows\\createpanelrefs.nf\n",
      "Nextflow operator map with content: .map{ meta, align, index ->\n",
      "Nextflow operator groupTuple with content: .groupTuple()\n",
      "Nextflow operator branch with content: .branch{\n",
      "Nextflow operator map with content: .map {meta, bam -> [ meta, [], bam ]}\n",
      "Nextflow operator set with content: .set { ch_cnvkit_input }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(CNVKIT_BATCH.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GERMLINECNVCALLER_COHORT.out.versions)\n",
      "Nextflow operator collectFile with content: ch_versions.unique().collectFile(name: 'collated_versions.yml')\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_workflow_summary.collectFile(name: 'workflow_summary_mqc.yaml'))\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_methods_description.collectFile(name: 'methods_description_mqc.yaml'))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(CUSTOM_DUMPSOFTWAREVERSIONS.out.mqc_yml.collect())\n",
      "Nextflow operator collect with content: ch_multiqc_files.collect(),\n",
      "Nextflow operator toList with content: ch_multiqc_config.toList(),\n",
      "Nextflow operator toList with content: ch_multiqc_custom_config.toList(),\n",
      "Nextflow operator toList with content: ch_multiqc_logo.toList()\n",
      "Nextflow operator toList with content: multiqc_report = MULTIQC.out.report.toList()\n",
      "Error: Unclosed process block in file: ./git_repos\\createpanelrefs\\workflows\\createpanelrefs.nf\n",
      "Process NFCORE_CREATETAXDB in filepath ./git_repos\\createtaxdb\\main.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\createtaxdb\\main.nf\n",
      "Process INPUT_CHECK in filepath ./git_repos\\createtaxdb\\subworkflows\\local\\input_check.nf\n",
      "Nextflow operator splitCsv with content: .splitCsv ( header:true, sep:',' )\n",
      "Nextflow operator map with content: .map { create_fastq_channel(it) }\n",
      "Nextflow operator set with content: .set { reads }\n",
      "Error: Unclosed process block in file: ./git_repos\\createtaxdb\\subworkflows\\local\\input_check.nf\n",
      "Process CREATETAXDB in filepath ./git_repos\\createtaxdb\\workflows\\createtaxdb.nf\n",
      "Nextflow operator map with content: .map{meta, fasta_dna, fasta_aa  -> [[id: params.dbname], fasta_aa]}\n",
      "Nextflow operator filter with content: .filter{meta, fasta_aa ->\n",
      "Nextflow operator groupTuple with content: .groupTuple()\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(CAT_CAT_AA.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(KAIJU_MKFMI.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(DIAMOND_MAKEDB.out.versions.first())\n",
      "Nextflow operator collectFile with content: ch_versions.unique().collectFile(name: 'collated_versions.yml')\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_workflow_summary.collectFile(name: 'workflow_summary_mqc.yaml'))\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_methods_description.collectFile(name: 'methods_description_mqc.yaml'))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(CUSTOM_DUMPSOFTWAREVERSIONS.out.mqc_yml.collect())\n",
      "Nextflow operator collect with content: ch_multiqc_files.collect(),\n",
      "Nextflow operator toList with content: ch_multiqc_config.toList(),\n",
      "Nextflow operator toList with content: ch_multiqc_custom_config.toList(),\n",
      "Nextflow operator toList with content: ch_multiqc_logo.toList()\n",
      "Nextflow operator toList with content: multiqc_report = MULTIQC.out.report.toList()\n",
      "Error: Unclosed process block in file: ./git_repos\\createtaxdb\\workflows\\createtaxdb.nf\n",
      "Process NFCORE_CRISPRSEQ in filepath ./git_repos\\crisprseq\\main.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\crisprseq\\main.nf\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Process CRISPRSEQ_SCREENING in filepath ./git_repos\\crisprseq\\workflows\\crisprseq_screening.nf\n",
      "Nextflow operator map with content: .map{ meta, fastq_1, fastq_2, x, y, z ->\n",
      "Nextflow operator set with content: .set { ch_input }\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(FASTQC.out.versions.first())\n",
      "Nextflow operator combine with content: ch_input_cutadapt = ch_input.combine(Channel.value([[]]))\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(CUTADAPT.out.versions)\n",
      "Nextflow operator map with content: .map{ meta, fastq  ->\n",
      "Nextflow operator set with content: .set { ch_input }\n",
      "Nextflow operator map with content: .map { meta, fastqs  ->\n",
      "Nextflow operator reduce with content: .reduce { a, b ->\n",
      "Nextflow operator map with content: .map { condition, fastqs_1, single_end, fastqs_2 ->\n",
      "Nextflow operator last with content: .last()\n",
      "Nextflow operator set with content: .set { joined }\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(MAGECK_COUNT.out.versions.first())\n",
      "Nextflow operator count with content: MAGECK_COUNT.out.count.map {\n",
      "Nextflow operator set with content: }.set { ch_counts }\n",
      "Nextflow operator set with content: .set { ch_counts }\n",
      "Nextflow operator collect with content: ch_crispr_normalize.concat(ch_counts,ch_crisprcleanr).collect(),\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(CRISPRCLEANR_NORMALIZE.out.versions)\n",
      "Nextflow operator map with content: CRISPRCLEANR_NORMALIZE.out.norm_count_file.map {\n",
      "Nextflow operator set with content: }.set { ch_counts }\n",
      "Nextflow operator splitCsv with content: .splitCsv(header:true, sep:';' )\n",
      "Nextflow operator set with content: .set { ch_contrasts }\n",
      "Nextflow operator combine with content: counts = ch_contrasts.combine(ch_counts)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(MAGECK_TEST.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(MAGECK_GRAPHRRA.out.versions)\n",
      "Nextflow operator splitCsv with content: .splitCsv(header:true, sep:';' )\n",
      "Nextflow operator set with content: .set { ch_bagel }\n",
      "Nextflow operator combine with content: counts = ch_bagel.combine(ch_counts)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BAGEL2_FC.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BAGEL2_BF.out.versions)\n",
      "Nextflow operator combine with content: ch_bagel_pr = BAGEL2_BF.out.bf.combine(ch_bagel_reference_essentials)\n",
      "Nextflow operator combine with content: .combine(ch_bagel_reference_nonessentials)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BAGEL2_PR.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BAGEL2_GRAPH.out.versions)\n",
      "Nextflow operator combine with content: ch_mle = ch_counts.combine(ch_design)\n",
      "Nextflow operator map with content: ch_mle.map {\n",
      "Nextflow operator set with content: }.set { ch_designed_mle }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(MAGECK_MLE.out.versions)\n",
      "Nextflow operator collectFile with content: ch_versions.unique{ it.text }.collectFile(name: 'collated_versions.yml')\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_workflow_summary.collectFile(name: 'workflow_summary_mqc.yaml'))\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_methods_description.collectFile(name: 'methods_description_mqc.yaml'))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(CUSTOM_DUMPSOFTWAREVERSIONS.out.mqc_yml.collect())\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(FASTQC.out.zip.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files.collect(),\n",
      "Nextflow operator collect with content: ch_multiqc_config.collect().ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_multiqc_custom_config.collect().ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_multiqc_logo.collect().ifEmpty([])\n",
      "Nextflow operator toList with content: multiqc_report = MULTIQC.out.report.toList()\n",
      "Nextflow operator mix with content: ch_versions    = ch_versions.mix(MULTIQC.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\crisprseq\\workflows\\crisprseq_screening.nf\n",
      "Process CRISPRSEQ_TARGETED in filepath ./git_repos\\crisprseq\\workflows\\crisprseq_targeted.nf\n",
      "Nextflow operator multiMap with content: .multiMap { meta, fastq_1, fastq_2, reference, protospacer, template ->\n",
      "Nextflow operator set with content: .set { ch_input }\n",
      "Nextflow operator groupTuple with content: .groupTuple()\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator branch with content: .branch {\n",
      "Nextflow operator flatten with content: return [ meta, fastqs.flatten() ]\n",
      "Nextflow operator flatten with content: return [ meta, fastqs.flatten() ]\n",
      "Nextflow operator set with content: .set { ch_fastq }\n",
      "Nextflow operator tap with content: .tap{ meta_reference }\n",
      "Nextflow operator filter with content: .filter{ meta, sequence -> sequence instanceof String }\n",
      "Nextflow operator collectFile with content: .collectFile() { meta, reference ->\n",
      "Nextflow operator map with content: .map{ new_file ->\n",
      "Nextflow operator join with content: .join(meta_reference\n",
      "Nextflow operator map with content: .map{ meta, reference ->\n",
      "Nextflow operator map with content: .map{ metaid, new_file, meta ->\n",
      "Nextflow operator set with content: .set{ ch_seq_reference }\n",
      "Nextflow operator tap with content: .tap{ meta_template }\n",
      "Nextflow operator filter with content: .filter{ meta, sequence -> sequence instanceof String }\n",
      "Nextflow operator collectFile with content: .collectFile() { meta, template ->\n",
      "Nextflow operator map with content: .map{ new_file ->\n",
      "Nextflow operator join with content: .join(meta_template\n",
      "Nextflow operator map with content: .map{ meta, template ->\n",
      "Nextflow operator map with content: .map{ metaid, new_file, meta ->\n",
      "Nextflow operator set with content: .set{ ch_seq_template }\n",
      "Nextflow operator join with content: .join(ch_input.protospacer)\n",
      "Nextflow operator set with content: .set{ reference_protospacer }\n",
      "Nextflow operator combine with content: .combine(ch_protospacer)\n",
      "Nextflow operator set with content: .set{ reference_protospacer }\n",
      "Nextflow operator combine with content: .combine(ch_reference)\n",
      "Nextflow operator set with content: .set{ reference_protospacer }\n",
      "Nextflow operator combine with content: .combine(ch_reference)\n",
      "Nextflow operator combine with content: .combine(ch_protospacer)\n",
      "Nextflow operator set with content: .set{ reference_protospacer }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ORIENT_REFERENCE.out.versions)\n",
      "Nextflow operator groupTuple with content: .groupTuple(by: [0])\n",
      "Nextflow operator mix with content: .mix(ch_fastq.single)\n",
      "Nextflow operator branch with content: .branch {\n",
      "Nextflow operator set with content: .set { ch_cat_fastq }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(CAT_FASTQ.out.versions)\n",
      "Nextflow operator mix with content: .mix( ch_cat_fastq.single )\n",
      "Nextflow operator set with content: .set { ch_pear_fastq }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(PEAR.out.versions)\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(FASTQC.out.versions.first())\n",
      "Nextflow operator join with content: .join(ch_pear_fastq)\n",
      "Nextflow operator groupTuple with content: .groupTuple(by: [0])\n",
      "Nextflow operator branch with content: .branch {\n",
      "Nextflow operator set with content: .set { ch_adapter_seqs }\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(FIND_ADAPTERS.out.versions.first())\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(CUTADAPT.out.versions)\n",
      "Nextflow operator mix with content: .mix(CUTADAPT.out.reads)\n",
      "Nextflow operator groupTuple with content: .groupTuple(by: [0])\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator flatten with content: return [ meta, fastq.flatten() ]\n",
      "Nextflow operator set with content: .set{ ch_trimmed }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SEQTK_SEQ_MASK.out.versions)\n",
      "Nextflow operator mix with content: .mix(ch_cat_fastq.single)\n",
      "Nextflow operator join with content: .join(PEAR.out.assembled, remainder: true)\n",
      "Nextflow operator join with content: .join(SEQTK_SEQ_MASK.out.fastx)\n",
      "Nextflow operator join with content: .join(CUTADAPT.out.log)\n",
      "Nextflow operator map with content: .map { meta, reads, assembled, masked, trimmed ->\n",
      "Nextflow operator set with content: .set { ch_preprocessing_summary_data }\n",
      "Nextflow operator mix with content: .mix(ch_cat_fastq.single)\n",
      "Nextflow operator join with content: .join(PEAR.out.assembled, remainder: true)\n",
      "Nextflow operator join with content: .join(SEQTK_SEQ_MASK.out.fastx)\n",
      "Nextflow operator map with content: .map { meta, reads, assembled, masked ->\n",
      "Nextflow operator set with content: .set { ch_preprocessing_summary_data }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(PREPROCESSING_SUMMARY.out.versions)\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(EXTRACT_UMIS.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(VSEARCH_CLUSTER.out.versions.first())\n",
      "Nextflow operator transpose with content: .transpose()\n",
      "Nextflow operator collectFile with content: .collectFile( storeDir:params.outdir ) {\n",
      "Nextflow operator countFasta with content: [ \"${it[0].id}_ubs.txt\", \"${it[1].countFasta()}\\t${it[1].baseName}\\n\" ]\n",
      "Nextflow operator transpose with content: .transpose()\n",
      "Nextflow operator branch with content: .branch{\n",
      "Nextflow operator countFasta with content: single: cluster.countFasta() >= params.umi_bin_size && cluster.countFasta() == 1\n",
      "Nextflow operator countFasta with content: cluster: cluster.countFasta() >= params.umi_bin_size && cluster.countFasta() > 1\n",
      "Nextflow operator set with content: .set{ ch_umi_bysize }\n",
      "Nextflow operator tap with content: .tap{ meta_channel }\n",
      "Nextflow operator map with content: .map{ meta, cluster ->\n",
      "Nextflow operator collectFile with content: .collectFile() { meta, name, fasta ->\n",
      "Nextflow operator map with content: .map{ new_file ->\n",
      "Nextflow operator join with content: .join(meta_channel\n",
      "Nextflow operator map with content: .map { meta, original_file ->\n",
      "Nextflow operator map with content: .map{ file_name, new_file, meta ->\n",
      "Nextflow operator set with content: .set{ ch_single_clusters_consensus }\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(VSEARCH_SORT.out.versions.first())\n",
      "Nextflow operator tap with content: .tap{ meta_channel_2 } // [[id:sample_id, ...], sample_top.fasta]\n",
      "Nextflow operator map with content: .map{ meta, fasta ->\n",
      "Nextflow operator collectFile with content: .collectFile(cache:true,sort:true) { meta, name, fasta ->\n",
      "Nextflow operator map with content: .map{ new_file ->\n",
      "Nextflow operator join with content: .join(meta_channel_2\n",
      "Nextflow operator map with content: .map { meta, original_file ->\n",
      "Nextflow operator map with content: .map{ file_name, new_file, meta ->\n",
      "Nextflow operator set with content: .set{ ch_top_clusters_sequence } // [[id:sample_id, ..., cluster_id:sample], sample_top.fasta]\n",
      "Nextflow operator tap with content: .tap{ meta_channel_3 } // [[id:sample_id, ...], sample]\n",
      "Nextflow operator map with content: .map{ meta, cluster ->\n",
      "Nextflow operator collectFile with content: .collectFile(cache:true,sort:true) { meta, name, fasta ->\n",
      "Nextflow operator map with content: .map{ new_file ->\n",
      "Nextflow operator join with content: .join(meta_channel_3\n",
      "Nextflow operator map with content: .map { meta, original_file ->\n",
      "Nextflow operator map with content: .map{ file_name, new_file, meta ->\n",
      "Nextflow operator set with content: .set{ ch_clusters_sequence }\n",
      "Nextflow operator join with content: .join(ch_top_clusters_sequence),\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(MINIMAP2_ALIGN_UMI_1.out.versions.first())\n",
      "Nextflow operator countLines with content: .filter{ it[1].countLines() > 0 }\n",
      "Nextflow operator set with content: .set{ ch_minimap_1 }\n",
      "Nextflow operator join with content: .join(ch_top_clusters_sequence)\n",
      "Nextflow operator join with content: .join(ch_minimap_1)\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(RACON_1.out.versions.first())\n",
      "Nextflow operator join with content: .join(RACON_1.out.improved_assembly),\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(MINIMAP2_ALIGN_UMI_2.out.versions.first())\n",
      "Nextflow operator countLines with content: .filter{ it[1].countLines() > 0 }\n",
      "Nextflow operator set with content: .set{ ch_minimap_2 }\n",
      "Nextflow operator join with content: .join(RACON_1.out.improved_assembly)\n",
      "Nextflow operator join with content: .join(ch_minimap_2)\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(RACON_2.out.versions.first())\n",
      "Nextflow operator join with content: .join(RACON_2.out.improved_assembly)\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(MEDAKA.out.versions.first())\n",
      "Nextflow operator tap with content: .tap{ meta_channel_4 }\n",
      "Nextflow operator map with content: .map{ meta, file ->\n",
      "Nextflow operator collectFile with content: .collectFile() { meta, file ->\n",
      "Nextflow operator map with content: .map{ new_file ->\n",
      "Nextflow operator join with content: .join(meta_channel_4\n",
      "Nextflow operator map with content: .map{ meta, consensus ->\n",
      "Nextflow operator map with content: .map{ name, file, meta ->\n",
      "Nextflow operator set with content: .set{ ch_umi_consensus }\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SEQTK_SEQ_FATOFQ.out.versions.first())\n",
      "Nextflow operator join with content: .join(PREPROCESSING_SUMMARY.out.summary)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(CLUSTERING_SUMMARY.out.versions)\n",
      "Nextflow operator join with content: .join(ORIENT_REFERENCE.out.reference),\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(MINIMAP2_ALIGN_ORIGINAL.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BWA_INDEX.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BWA_MEM.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BOWTIE2_BUILD.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BOWTIE2_ALIGN.out.versions)\n",
      "Nextflow operator join with content: .join(CLUSTERING_SUMMARY.out.summary)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ALIGNMENT_SUMMARY.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_INDEX.out.versions)\n",
      "Nextflow operator join with content: .join(ch_seq_template)\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(TEMPLATE_REFERENCE.out.versions.first())\n",
      "Nextflow operator join with content: .join(ORIENT_REFERENCE.out.reference),\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator set with content: .set { ch_template_bam }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(MINIMAP2_ALIGN_TEMPLATE.out.versions)\n",
      "Nextflow operator join with content: .join(SAMTOOLS_INDEX.out.bai)\n",
      "Nextflow operator join with content: .join(ORIENT_REFERENCE.out.reference)\n",
      "Nextflow operator join with content: .join(ch_input.protospacer)\n",
      "Nextflow operator join with content: .join(ch_seq_template, remainder: true)\n",
      "Nextflow operator join with content: .join(ch_template_bam, remainder: true)\n",
      "Nextflow operator join with content: .join(TEMPLATE_REFERENCE.out.fasta, remainder: true)\n",
      "Nextflow operator join with content: .join(ALIGNMENT_SUMMARY.out.summary)\n",
      "Nextflow operator map with content: .map { meta, reads, index, reference, protospacer, template, template_bam, reference_template, summary ->\n",
      "Nextflow operator set with content: .set { ch_to_parse_cigar }\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(CIGAR_PARSER.out.versions.first())\n",
      "Nextflow operator join with content: .join(ORIENT_REFERENCE.out.reference)\n",
      "Nextflow operator join with content: .join(ch_input.protospacer)\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(CRISPRSEQ_PLOTTER.out.versions.first())\n",
      "Nextflow operator collectFile with content: ch_versions.unique{ it.text }.collectFile(name: 'collated_versions.yml')\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_workflow_summary.collectFile(name: 'workflow_summary_mqc.yaml'))\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_methods_description.collectFile(name: 'methods_description_mqc.yaml'))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(CIGAR_PARSER.out.processing.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(CIGAR_PARSER.out.edition.collect{it[2]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(CIGAR_PARSER.out.qcindels.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(CUSTOM_DUMPSOFTWAREVERSIONS.out.mqc_yml.collect())\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(FASTQC.out.zip.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(CUTADAPT.out.log.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files.collect(),\n",
      "Nextflow operator toList with content: ch_multiqc_config.toList(),\n",
      "Nextflow operator toList with content: ch_multiqc_custom_config.toList(),\n",
      "Nextflow operator toList with content: ch_multiqc_logo.toList()\n",
      "Nextflow operator toList with content: multiqc_report = MULTIQC.out.report.toList()\n",
      "Error: Unclosed process block in file: ./git_repos\\crisprseq\\workflows\\crisprseq_targeted.nf\n",
      "Process NFCORE_CUTANDRUN in filepath ./git_repos\\cutandrun\\main.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\cutandrun\\main.nf\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Process ALIGN_BOWTIE2 in filepath ./git_repos\\cutandrun\\subworkflows\\local\\align_bowtie2.nf\n",
      "Nextflow operator map with content: ch_index = index.map { [[id:it.baseName], it] }\n",
      "Nextflow operator collect with content: ch_index.collect{ it[1] },\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BOWTIE2_TARGET_ALIGN.out.versions)\n",
      "Nextflow operator map with content: ch_spikein_index = spikein_index.map { [[id:it.baseName], it] }\n",
      "Nextflow operator collect with content: ch_spikein_index.collect{ it[1] },\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BOWTIE2_SPIKEIN_ALIGN.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BAM_SORT_STATS_SAMTOOLS.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BAM_SORT_STATS_SAMTOOLS_SPIKEIN.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\cutandrun\\subworkflows\\local\\align_bowtie2.nf\n",
      "Process CONSENSUS_PEAKS in filepath ./git_repos\\cutandrun\\subworkflows\\local\\consensus_peaks.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SORT.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BEDTOOLS_MERGE.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(AWK.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\cutandrun\\subworkflows\\local\\consensus_peaks.nf\n",
      "Process DEDUPLICATE_LINEAR in filepath ./git_repos\\cutandrun\\subworkflows\\local\\deduplicate_linear.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix( SAMTOOLS_SORT.out.versions )\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix( BEDTOOLS_BAMTOBED.out.versions )\n",
      "Nextflow operator mix with content: ch_versions          = ch_versions.mix( FIND_UNIQUE_READS.out.versions )\n",
      "Nextflow operator join with content: .join( bai )\n",
      "Nextflow operator join with content: .join( ch_linear_duplicates )\n",
      "Nextflow operator set with content: .set { ch_bam_bai_txt }\n",
      "Nextflow operator map with content: .map { row -> row[0,1,2]}\n",
      "Nextflow operator set with content: .set { ch_bam_bai }\n",
      "Nextflow operator map with content: .map { row -> row[3]}\n",
      "Nextflow operator set with content: .set { ch_txt}\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix( SAMTOOLS_VIEW.out.versions )\n",
      "Nextflow operator branch with content: bam.branch { it ->\n",
      "Nextflow operator set with content: .set { ch_split }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix( SAMTOOLS_SORT.out.versions )\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix( BEDTOOLS_BAMTOBED.out.versions )\n",
      "Nextflow operator mix with content: ch_versions          = ch_versions.mix( FIND_UNIQUE_READS.out.versions )\n",
      "Nextflow operator join with content: .join( bai )\n",
      "Nextflow operator join with content: .join( ch_linear_duplicates )\n",
      "Nextflow operator set with content: .set { ch_bam_bai_txt }\n",
      "Nextflow operator map with content: .map { row -> row[0,1,2]}\n",
      "Nextflow operator set with content: .set { ch_bam_bai }\n",
      "Nextflow operator map with content: .map { row -> row[3]}\n",
      "Nextflow operator set with content: .set { ch_txt}\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix( SAMTOOLS_VIEW.out.versions )\n",
      "Nextflow operator toSortedList with content: .toSortedList { row -> row[0].id }\n",
      "Nextflow operator flatMap with content: .flatMap()\n",
      "Nextflow operator toSortedList with content: .toSortedList { row -> row[0].id }\n",
      "Nextflow operator flatMap with content: .flatMap()\n",
      "Nextflow operator concat with content: ch_bam = ch_sorted_targets.concat( ch_sorted_controls )\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix( BAM_SORT_STATS_SAMTOOLS.out.versions )\n",
      "Error: Unclosed process block in file: ./git_repos\\cutandrun\\subworkflows\\local\\deduplicate_linear.nf\n",
      "Process DEEPTOOLS_QC in filepath ./git_repos\\cutandrun\\subworkflows\\local\\deeptools_qc.nf\n",
      "Nextflow operator join with content: .join( bai )\n",
      "Nextflow operator set with content: .set { ch_bam_bai }\n",
      "Nextflow operator map with content: ch_bam_bai.map { row -> [row[0].id] }\n",
      "Nextflow operator collect with content: .collect()\n",
      "Nextflow operator map with content: .map { row -> [row] }\n",
      "Nextflow operator set with content: .set { ch_ids }\n",
      "Nextflow operator map with content: ch_bam_bai.map { row -> [row[1]] }\n",
      "Nextflow operator collect with content: .collect()\n",
      "Nextflow operator map with content: .map { row -> [row] }\n",
      "Nextflow operator collect with content: .combine( ch_bam_bai.map { row -> [row[2]] }.collect().map { row -> [row] } )\n",
      "Nextflow operator combine with content: .combine( ch_ids )\n",
      "Nextflow operator map with content: .map { row -> [[id: 'all_target_bams'], row[0], row[1], row[2], row[1].size()] }\n",
      "Nextflow operator filter with content: .filter { row -> row[4] > 1 }\n",
      "Nextflow operator map with content: .map { row -> [row[0], row[1], row[2], row[3]] }\n",
      "Nextflow operator set with content: .set { ch_bam_bai_all }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(DEEPTOOLS_MULTIBAMSUMMARY.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(DEEPTOOLS_PLOTCORRELATION.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(DEEPTOOLS_PLOTPCA.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(DEEPTOOLS_PLOTFINGERPRINT.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\cutandrun\\subworkflows\\local\\deeptools_qc.nf\n",
      "Process EXTRACT_FRAGMENTS in filepath ./git_repos\\cutandrun\\subworkflows\\local\\extract_fragments.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_SORT.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BEDTOOLS_BAMTOBED.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(AWK.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\cutandrun\\subworkflows\\local\\extract_fragments.nf\n",
      "Process EXTRACT_METADATA_AWK in filepath ./git_repos\\cutandrun\\subworkflows\\local\\extract_metadata_awk.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(AWK_SCRIPT.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(AWK.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\cutandrun\\subworkflows\\local\\extract_metadata_awk.nf\n",
      "Process FASTQC_TRIMGALORE in filepath ./git_repos\\cutandrun\\subworkflows\\local\\fastqc_trimgalore.nf\n",
      "Nextflow operator set with content: FASTQC ( reads ).html.set { fastqc_html }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(FASTQC.out.versions)\n",
      "Nextflow operator mix with content: ch_versions     = ch_versions.mix(TRIMGALORE.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\cutandrun\\subworkflows\\local\\fastqc_trimgalore.nf\n",
      "Process INPUT_CHECK in filepath ./git_repos\\cutandrun\\subworkflows\\local\\input_check.nf\n",
      "Nextflow operator splitCsv with content: .splitCsv ( header:true, sep:\",\" )\n",
      "Nextflow operator map with content: .map { get_samplesheet_paths(it) }\n",
      "Nextflow operator set with content: .set { reads }\n",
      "Nextflow operator toInteger with content: meta.replicate     = row.replicate.toInteger()\n",
      "Error: Unclosed process block in file: ./git_repos\\cutandrun\\subworkflows\\local\\input_check.nf\n",
      "Process MARK_DUPLICATES_PICARD in filepath ./git_repos\\cutandrun\\subworkflows\\local\\mark_duplicates_picard.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix( PICARD_MARKDUPLICATES.out.versions )\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix( BAM_SORT_STATS_SAMTOOLS.out.versions )\n",
      "Nextflow operator branch with content: bam.branch { it ->\n",
      "Nextflow operator set with content: .set { ch_split_bam }\n",
      "Nextflow operator branch with content: bai.branch { it ->\n",
      "Nextflow operator set with content: .set { ch_split_bai }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix( PICARD_MARKDUPLICATES.out.versions )\n",
      "Nextflow operator mix with content: ch_bam      = BAM_SORT_STATS_SAMTOOLS.out.bam.mix(ch_split_bam.target)\n",
      "Nextflow operator mix with content: ch_bai      = BAM_SORT_STATS_SAMTOOLS.out.bai.mix(ch_split_bai.target)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix( BAM_SORT_STATS_SAMTOOLS.out.versions )\n",
      "Error: Unclosed process block in file: ./git_repos\\cutandrun\\subworkflows\\local\\mark_duplicates_picard.nf\n",
      "Process PEAK_QC in filepath ./git_repos\\cutandrun\\subworkflows\\local\\peak_qc.nf\n",
      "Nextflow operator flatten with content: .map { row -> [row[0].id, row ].flatten()}\n",
      "Nextflow operator flatten with content: .join ( fragments_bed.map { row -> [row[0].id, row ].flatten()} )\n",
      "Nextflow operator flatten with content: .join ( flagstat.map { row -> [row[0].id, row ].flatten()} )\n",
      "Nextflow operator map with content: .map { row -> [ row[1], row[2], row[4], row[6] ]}\n",
      "Nextflow operator set with content: .set { ch_frip }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(PEAK_FRIP.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(PRIMARY_PEAK_COUNTS.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(CONSENSUS_PEAK_COUNTS.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(CUT_CALC_REPROD.out.versions)\n",
      "Nextflow operator map with content: .map { row -> [ row[0].group, row[1] ] }\n",
      "Nextflow operator groupTuple with content: .groupTuple(by: [0])\n",
      "Nextflow operator flatten with content: .map { row -> [ [id: row[0]], row[1].flatten() ] }\n",
      "Nextflow operator map with content: .map { row -> [ row[0], row[1], row[1].size() ] }\n",
      "Nextflow operator filter with content: .filter { row -> row[2] > 1 }\n",
      "Nextflow operator map with content: .map { row -> [ row[0], row[1] ] }\n",
      "Nextflow operator set with content: .set { ch_peak_bed_group }\n",
      "Nextflow operator flatMap with content: ch_peak_bed_group.flatMap{\n",
      "Nextflow operator collect with content: def files_copy = row[1].collect()\n",
      "Nextflow operator set with content: .set { ch_beds_intersect }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BEDTOOLS_INTERSECT.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(CALCULATE_PEAK_REPROD.out.versions)\n",
      "Nextflow operator toSortedList with content: .toSortedList { row -> row[0].id }\n",
      "Nextflow operator map with content: .map { list ->\n",
      "Nextflow operator set with content: .set { ch_merged_bed_sorted }\n",
      "Nextflow operator ifEmpty with content: ch_merged_bed_sorted.ifEmpty([])\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(PLOT_CONSENSUS_PEAKS.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\cutandrun\\subworkflows\\local\\peak_qc.nf\n",
      "Process PREPARE_GENOME in filepath ./git_repos\\cutandrun\\subworkflows\\local\\prepare_genome.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GUNZIP_FASTA.out.versions)\n",
      "Nextflow operator map with content: ch_fasta = Channel.from( file(params.fasta) ).map { row -> [[id:\"fasta\"], row] }\n",
      "Nextflow operator mix with content: ch_versions      = ch_versions.mix(GUNZIP_SPIKEIN_FASTA.out.versions)\n",
      "Nextflow operator map with content: ch_spikein_fasta = Channel.from( file(params.spikein_fasta) ).map { row -> [[id:\"spikein_fasta\"], row] }\n",
      "Nextflow operator map with content: ch_gtf      = GUNZIP_GTF ( [ [:], params.gtf ] ).gunzip.map { it[1] }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GUNZIP_GTF.out.versions)\n",
      "Nextflow operator map with content: ch_gene_bed = GUNZIP_BED ( [ [:], params.gene_bed ] ).gunzip.map { it[1] }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GUNZIP_BED.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GTF2BED.out.versions)\n",
      "Nextflow operator map with content: ch_tabix = ch_gene_bed.map {\n",
      "Nextflow operator map with content: ch_tabix = ch_tabix.map {\n",
      "Nextflow operator mix with content: ch_versions       = ch_versions.mix(TABIX_BGZIPTABIX.out.versions)\n",
      "Nextflow operator mix with content: ch_versions    = ch_versions.mix(SAMTOOLS_FAIDX.out.versions)\n",
      "Nextflow operator map with content: ch_chrom_sizes = TARGET_CHROMSIZES ( ch_fasta ).sizes.map{ it[1] }\n",
      "Nextflow operator mix with content: ch_versions    = ch_versions.mix(TARGET_CHROMSIZES.out.versions)\n",
      "Nextflow operator map with content: ch_spikein_chrom_sizes = SPIKEIN_CHROMSIZES ( ch_spikein_fasta ).sizes.map{ it[1] }\n",
      "Nextflow operator map with content: ch_bt2_index = UNTAR_INDEX_TARGET ( [ [], params.bowtie2 ] ).untar.map{ row -> [ [id:\"target_index\"], row[1] ] }\n",
      "Nextflow operator mix with content: ch_versions  = ch_versions.mix(UNTAR_INDEX_TARGET.out.versions)\n",
      "Nextflow operator map with content: ch_bt2_index = BOWTIE2_BUILD_TARGET ( ch_fasta ).index.map{ row -> [ [id:\"target_index\"], row[1] ] }\n",
      "Nextflow operator mix with content: ch_versions  = ch_versions.mix(BOWTIE2_BUILD_TARGET.out.versions)\n",
      "Nextflow operator map with content: ch_bt2_spikein_index = UNTAR_INDEX_SPIKEIN ( [ [], params.spikein_bowtie2 ] ).untar.map{ row -> [ [id:\"spikein_index\"], row[1] ] }\n",
      "Nextflow operator mix with content: ch_versions          = ch_versions.mix(UNTAR_INDEX_SPIKEIN.out.versions)\n",
      "Nextflow operator map with content: ch_bt2_spikein_index = BOWTIE2_BUILD_SPIKEIN ( ch_spikein_fasta ).index.map{ row -> [ [id:\"spikein_index\"], row[1] ] }\n",
      "Nextflow operator mix with content: ch_versions          = ch_versions.mix(BOWTIE2_BUILD_SPIKEIN.out.versions)\n",
      "Nextflow operator map with content: ch_chrom_sizes.map {row -> [ [id: \"chromsizes\"], row ]}\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BLACKLIST_AWK.out.versions)\n",
      "Nextflow operator map with content: .map {row -> [ [id: \"blacklist\"], row ]}\n",
      "Nextflow operator combine with content: .combine( BLACKLIST_AWK.out.file )\n",
      "Nextflow operator map with content: .map {row -> [ row[0], row[1], row[3] ]}\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BLACKLIST_BEDTOOLS_INTERSECT.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BLACKLIST_BEDTOOLS_SORT.out.versions)\n",
      "Nextflow operator mix with content: ch_versions               = ch_versions.mix(BLACKLIST_BEDTOOLS_COMPLEMENT.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\cutandrun\\subworkflows\\local\\prepare_genome.nf\n",
      "Process PREPARE_PEAKCALLING in filepath ./git_repos\\cutandrun\\subworkflows\\local\\prepare_peakcalling.nf\n",
      "Nextflow operator splitCsv with content: metadata.splitCsv ( header:true, sep:\",\" )\n",
      "Nextflow operator map with content: .map { row -> [ row[0].id, row[1] ]}\n",
      "Nextflow operator set with content: .set { ch_metadata }\n",
      "Nextflow operator map with content: ch_bam.map { row -> [ row[0].id, row[0], row[1] ]}\n",
      "Nextflow operator join with content: .join ( ch_metadata )\n",
      "Nextflow operator map with content: .map { row ->\n",
      "Nextflow operator toInteger with content: def denominator = row[3].find{ it.key == \"bt2_total_aligned\" }?.value.toInteger()\n",
      "Nextflow operator set with content: .set { ch_bam_scale_factor }\n",
      "Nextflow operator map with content: ch_bam.map { row ->\n",
      "Nextflow operator set with content: .set { ch_bam_scale_factor }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BEDTOOLS_GENOMECOV.out.versions)\n",
      "Nextflow operator map with content: .map { [it[0].id, it[0].group, it[2]] }\n",
      "Nextflow operator toSortedList with content: .toSortedList( { a, b -> a[0] <=> b[0] } )\n",
      "Nextflow operator map with content: .map { list ->\n",
      "Nextflow operator flatten with content: .map { row -> [row[0].id, row ].flatten()}\n",
      "Nextflow operator flatten with content: .join ( ch_bai.map { row -> [row[0].id, row ].flatten()} )\n",
      "Nextflow operator map with content: .map { row -> [row[1], row[2], row[4]] }\n",
      "Nextflow operator set with content: .set { ch_bam_bai }\n",
      "Nextflow operator branch with content: ch_bam_bai.branch { it ->\n",
      "Nextflow operator set with content: .set { ch_bam_bai_split }\n",
      "Nextflow operator map with content: .map { row ->\n",
      "Nextflow operator set with content: .set { ch_bam_bai_split_target }\n",
      "Nextflow operator map with content: .map { row ->\n",
      "Nextflow operator set with content: .set { ch_bam_bai_split_igg }\n",
      "Nextflow operator mix with content: .mix(ch_bam_bai_split_igg)\n",
      "Nextflow operator set with content: .set { ch_bam_bai_scale_factor }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(DEEPTOOLS_BAMCOVERAGE.out.versions)\n",
      "Nextflow operator map with content: .map { [it[0].id, it[0].group, it[3]] }\n",
      "Nextflow operator toSortedList with content: .toSortedList( { a, b -> a[0] <=> b[0] } )\n",
      "Nextflow operator map with content: .map { list ->\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BEDTOOLS_SORT.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(UCSC_BEDCLIP.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(UCSC_BEDGRAPHTOBIGWIG.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\cutandrun\\subworkflows\\local\\prepare_peakcalling.nf\n",
      "Process SAMTOOLS_VIEW_SORT_STATS in filepath ./git_repos\\cutandrun\\subworkflows\\local\\samtools_view_sort_stats.nf\n",
      "Nextflow operator map with content: SAMTOOLS_VIEW ( bam.map{ row -> [ row[0], row[1], [], ] }, [], regions )\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SAMTOOLS_VIEW.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SAMTOOLS_SORT.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SAMTOOLS_INDEX.out.versions.first())\n",
      "Nextflow operator map with content: ch_bam_sample_id = SAMTOOLS_SORT.out.bam.map  { row -> [row[0].id, row] }\n",
      "Nextflow operator map with content: ch_bai_sample_id = SAMTOOLS_INDEX.out.bai.map { row -> [row[0].id, row] }\n",
      "Nextflow operator join with content: ch_bam_bai = ch_bam_sample_id.join(ch_bai_sample_id, by: [0]).map {row -> [row[1][0], row[1][1], row[2][1]]}\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(BAM_STATS_SAMTOOLS.out.versions.first())\n",
      "Error: Unclosed process block in file: ./git_repos\\cutandrun\\subworkflows\\local\\samtools_view_sort_stats.nf\n",
      "Process BAM_SORT_STATS_SAMTOOLS in filepath ./git_repos\\cutandrun\\subworkflows\\nf-core\\bam_sort_stats_samtools\\main.nf\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SAMTOOLS_SORT.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SAMTOOLS_INDEX.out.versions.first())\n",
      "Nextflow operator join with content: .join(SAMTOOLS_INDEX.out.bai, by: [0], remainder: true)\n",
      "Nextflow operator join with content: .join(SAMTOOLS_INDEX.out.csi, by: [0], remainder: true)\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator set with content: .set { ch_bam_bai }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BAM_STATS_SAMTOOLS.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\cutandrun\\subworkflows\\nf-core\\bam_sort_stats_samtools\\main.nf\n",
      "Process BAM_STATS_SAMTOOLS in filepath ./git_repos\\cutandrun\\subworkflows\\nf-core\\bam_stats_samtools\\main.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_STATS.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_FLAGSTAT.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_IDXSTATS.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\cutandrun\\subworkflows\\nf-core\\bam_stats_samtools\\main.nf\n",
      "Process CUTANDRUN in filepath ./git_repos\\cutandrun\\workflows\\cutandrun.nf\n",
      "Nextflow operator mix with content: ch_software_versions = ch_software_versions.mix(PREPARE_GENOME.out.versions)\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator join with content: meta.id = meta.id.split(\"_\")[0..-2].join(\"_\")\n",
      "Nextflow operator groupTuple with content: .groupTuple(by: [0])\n",
      "Nextflow operator branch with content: .branch {\n",
      "Nextflow operator flatten with content: return [ meta, fastq.flatten() ]\n",
      "Nextflow operator flatten with content: return [ meta, fastq.flatten() ]\n",
      "Nextflow operator set with content: .set { ch_fastq }\n",
      "Nextflow operator mix with content: ch_software_versions = ch_software_versions.mix(CAT_FASTQ.out.versions)\n",
      "Nextflow operator mix with content: .mix(ch_fastq.single)\n",
      "Nextflow operator set with content: .set { ch_cat_fastq }\n",
      "Nextflow operator mix with content: ch_software_versions = ch_software_versions.mix(FASTQC_TRIMGALORE.out.versions)\n",
      "Nextflow operator mix with content: ch_software_versions          = ch_software_versions.mix(ALIGN_BOWTIE2.out.versions)\n",
      "Nextflow operator mix with content: ch_software_versions   = ch_software_versions.mix(EXTRACT_BT2_TARGET_META.out.versions)\n",
      "Nextflow operator collect with content: PREPARE_GENOME.out.allowed_regions.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator mix with content: ch_software_versions = ch_software_versions.mix(FILTER_READS.out.versions)\n",
      "Nextflow operator mix with content: ch_software_versions = ch_software_versions.mix(PRESEQ_LCEXTRAP.out.versions)\n",
      "Nextflow operator collect with content: PREPARE_GENOME.out.fasta.collect(),\n",
      "Nextflow operator collect with content: PREPARE_GENOME.out.fasta_index.collect()\n",
      "Nextflow operator mix with content: ch_software_versions      = ch_software_versions.mix(MARK_DUPLICATES_PICARD.out.versions)\n",
      "Nextflow operator collect with content: PREPARE_GENOME.out.fasta.collect(),\n",
      "Nextflow operator collect with content: PREPARE_GENOME.out.fasta_index.collect()\n",
      "Nextflow operator mix with content: ch_software_versions = ch_software_versions.mix(DEDUPLICATE_PICARD.out.versions)\n",
      "Nextflow operator collect with content: ch_dummy_file.collect(),\n",
      "Nextflow operator mix with content: ch_software_versions          = ch_software_versions.mix(EXTRACT_PICARD_DUP_META.out.versions)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Nextflow operator collect with content: PREPARE_GENOME.out.fasta.collect(),\n",
      "Nextflow operator collect with content: PREPARE_GENOME.out.fasta_index.collect(),\n",
      "Nextflow operator mix with content: ch_software_versions      = ch_software_versions.mix(DEDUPLICATE_LINEAR.out.versions)\n",
      "Nextflow operator collect with content: PREPARE_GENOME.out.chrom_sizes.collect(),\n",
      "Nextflow operator mix with content: ch_software_versions = ch_software_versions.mix(PREPARE_PEAKCALLING.out.versions)\n",
      "Nextflow operator filter with content: ch_bedgraph.filter { it -> it[0].is_control == false }\n",
      "Nextflow operator set with content: .set { ch_bedgraph_target }\n",
      "Nextflow operator filter with content: ch_bedgraph.filter { it -> it[0].is_control == true }\n",
      "Nextflow operator set with content: .set { ch_bedgraph_control }\n",
      "Nextflow operator filter with content: ch_samtools_bam.filter { it -> it[0].is_control == false }\n",
      "Nextflow operator set with content: .set { ch_bam_target }\n",
      "Nextflow operator filter with content: ch_samtools_bam.filter { it -> it[0].is_control == true }\n",
      "Nextflow operator set with content: .set { ch_bam_control }\n",
      "Nextflow operator map with content: ch_bedgraph_control.map{ row -> [row[0].control_group + \"_\" + row[0].replicate, row] }\n",
      "Nextflow operator cross with content: .cross( ch_bedgraph_target.map{ row -> [row[0].control_group, row] } )\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator set with content: .set { ch_bedgraph_paired }\n",
      "Nextflow operator mix with content: ch_software_versions = ch_software_versions.mix(SEACR_CALLPEAK_IGG.out.versions)\n",
      "Nextflow operator map with content: ch_bam_control.map{ row -> [row[0].control_group + \"_\" + row[0].replicate, row] }\n",
      "Nextflow operator cross with content: .cross( ch_bam_target.map{ row -> [row[0].control_group, row] } )\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator set with content: .set { ch_bam_paired }\n",
      "Nextflow operator mix with content: ch_software_versions = ch_software_versions.mix(MACS2_CALLPEAK_IGG.out.versions)\n",
      "Nextflow operator map with content: ch_bedgraph_target.map{ row-> [ row[0], row[1], [] ] }\n",
      "Nextflow operator set with content: .set { ch_bedgraph_target_fctrl }\n",
      "Nextflow operator mix with content: ch_software_versions = ch_software_versions.mix(SEACR_CALLPEAK_NOIGG.out.versions)\n",
      "Nextflow operator map with content: ch_bam_target.map{ row-> [ row[0], row[1], [] ] }\n",
      "Nextflow operator set with content: .set { ch_samtools_bam_target_fctrl }\n",
      "Nextflow operator mix with content: ch_software_versions = ch_software_versions.mix(MACS2_CALLPEAK_NOIGG.out.versions)\n",
      "Nextflow operator mix with content: ch_software_versions = ch_software_versions.mix(PEAK_TO_BED.out.versions)\n",
      "Nextflow operator mix with content: ch_software_versions = ch_software_versions.mix(AWK_EXTRACT_SUMMITS.out.versions)\n",
      "Nextflow operator mix with content: ch_software_versions = ch_software_versions.mix(AWK_NAME_PEAK_BED.out.versions)\n",
      "Nextflow operator map with content: .map { row -> [ 1, row[1] ] }\n",
      "Nextflow operator groupTuple with content: .groupTuple(by: [0])\n",
      "Nextflow operator map with content: .map { row ->\n",
      "Nextflow operator flatten with content: [ new_meta, row[1].flatten() ]\n",
      "Nextflow operator map with content: .map { row ->\n",
      "Nextflow operator filter with content: .filter { row -> row[2] > 1 }\n",
      "Nextflow operator map with content: .map { row ->\n",
      "Nextflow operator set with content: .set { ch_peaks_bed_all }\n",
      "Nextflow operator mix with content: ch_software_versions      = ch_software_versions.mix(CONSENSUS_PEAKS_ALL.out.versions)\n",
      "Nextflow operator map with content: .map { row -> [ row[0].group, row[1] ] }\n",
      "Nextflow operator groupTuple with content: .groupTuple(by: [0])\n",
      "Nextflow operator flatten with content: .map { row -> [ [id: row[0]], row[1].flatten() ] }\n",
      "Nextflow operator set with content: .set { ch_peaks_bed_group }\n",
      "Nextflow operator mix with content: ch_software_versions      = ch_software_versions.mix(CONSENSUS_PEAKS.out.versions)\n",
      "Nextflow operator map with content: PREPARE_GENOME.out.fasta.map {it[1]},\n",
      "Nextflow operator map with content: PREPARE_GENOME.out.fasta_index.map {it[1]},\n",
      "Nextflow operator collect with content: ch_peaks_primary.collect{it[1]}.filter{ it -> it.size() > 1}.ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_peaks_secondary.collect{it[1]}.filter{ it -> it.size() > 1}.ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_bigwig.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator filter with content: ch_bigwig.filter { it[0].is_control == false }\n",
      "Nextflow operator set with content: .set { ch_bigwig_no_igg }\n",
      "Nextflow operator collect with content: PREPARE_GENOME.out.bed.collect()\n",
      "Nextflow operator mix with content: ch_software_versions = ch_software_versions.mix(DEEPTOOLS_COMPUTEMATRIX_GENE.out.versions)\n",
      "Nextflow operator mix with content: ch_software_versions = ch_software_versions.mix(DEEPTOOLS_PLOTHEATMAP_GENE.out.versions)\n",
      "Nextflow operator flatten with content: .map { row -> [row[0].id, row ].flatten()}\n",
      "Nextflow operator set with content: .set { ch_peaks_summits_id }\n",
      "Nextflow operator flatten with content: .map { row -> [row[0].id, row ].flatten()}\n",
      "Nextflow operator join with content: .join ( ch_peaks_summits_id )\n",
      "Nextflow operator filter with content: .filter ( it -> it[-1].size() > 1)\n",
      "Nextflow operator set with content: .set { ch_dt_bigwig_summits }\n",
      "Nextflow operator map with content: .map { row -> row[1,2] }\n",
      "Nextflow operator set with content: .set { ch_ordered_bigwig }\n",
      "Nextflow operator map with content: .map { row -> row[-1] }\n",
      "Nextflow operator set with content: .set { ch_ordered_peaks_max }\n",
      "Nextflow operator mix with content: ch_software_versions = ch_software_versions.mix(DEEPTOOLS_COMPUTEMATRIX_PEAKS.out.versions)\n",
      "Nextflow operator mix with content: ch_software_versions = ch_software_versions.mix(DEEPTOOLS_PLOTHEATMAP_PEAKS.out.versions)\n",
      "Nextflow operator map with content: ch_bigwig_no_igg.map{it[1]}.toSortedList().map{ [[id:'all_genes'], it]},\n",
      "Nextflow operator toSortedList with content: PREPARE_GENOME.out.bed.toSortedList()\n",
      "Nextflow operator mix with content: ch_software_versions = ch_software_versions.mix(DEEPTOOLS_QC.out.versions)\n",
      "Nextflow operator filter with content: ch_samtools_bai.filter { it -> it[0].is_control == false }\n",
      "Nextflow operator set with content: .set { ch_bai_target }\n",
      "Nextflow operator filter with content: ch_samtools_flagstat.filter { it -> it[0].is_control == false }\n",
      "Nextflow operator set with content: .set { ch_flagstat_target }\n",
      "Nextflow operator mix with content: ch_software_versions           = ch_software_versions.mix(PEAK_QC.out.versions)\n",
      "Nextflow operator flatten with content: ch_bam_target.map { row -> [row[0].id, row ].flatten()}\n",
      "Nextflow operator flatten with content: .join ( ch_bai_target.map { row -> [row[0].id, row ].flatten()} )\n",
      "Nextflow operator map with content: .map { row -> [row[1], row[2], row[4]] }\n",
      "Nextflow operator set with content: .set { ch_bam_bai }\n",
      "Nextflow operator mix with content: ch_software_versions = ch_software_versions.mix(SAMTOOLS_CUSTOMVIEW.out.versions)\n",
      "Nextflow operator toSortedList with content: .toSortedList { row -> row[0].id }\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator mix with content: ch_software_versions = ch_software_versions.mix(FRAG_LEN_HIST.out.versions)\n",
      "Nextflow operator collectFile with content: ch_software_versions.unique().collectFile()\n",
      "Nextflow operator collect with content: ch_multiqc_custom_config.collect().ifEmpty([]),\n",
      "Nextflow operator collect with content: CUSTOM_DUMPSOFTWAREVERSIONS.out.mqc_yml.collect(),\n",
      "Nextflow operator collect with content: CUSTOM_DUMPSOFTWAREVERSIONS.out.mqc_unique_yml.collect(),\n",
      "Nextflow operator collectFile with content: ch_workflow_summary.collectFile(name: \"workflow_summary_mqc.yml\"),\n",
      "Nextflow operator collect with content: FASTQC_TRIMGALORE.out.fastqc_zip.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: FASTQC_TRIMGALORE.out.trim_zip.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: FASTQC_TRIMGALORE.out.trim_log.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_bowtie2_log.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_bowtie2_spikein_log.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_samtools_stats.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_samtools_flagstat.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_samtools_idxstats.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_markduplicates_metrics.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_preseq_output.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_dt_corrmatrix.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_dt_pcadata.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_dt_fpmatrix.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_peakqc_count_mqc.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_peakqc_frip_mqc.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_peakqc_count_consensus_mqc.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_peakqc_reprod_perc_mqc.collect().ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_frag_len_hist_mqc.collect().ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_linear_duplication_mqc.collect{it[1]}.ifEmpty([])\n",
      "Nextflow operator toList with content: multiqc_report = MULTIQC.out.report.toList()\n",
      "Error: Unclosed process block in file: ./git_repos\\cutandrun\\workflows\\cutandrun.nf\n",
      "Process NFCORE_DEMULTIPLEX in filepath ./git_repos\\demultiplex\\main.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\demultiplex\\main.nf\n",
      "Process BASES_DEMULTIPLEX in filepath ./git_repos\\demultiplex\\subworkflows\\local\\bases_demultiplex\\main.nf\n",
      "Nextflow operator map with content: ch_reads.transpose().map{\n",
      "Nextflow operator groupTuple with content: .groupTuple(by: [0])\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator flatten with content: return [ meta, fastq.flatten() ]\n",
      "Nextflow operator join with content: rg.ID = [fcid,lane].join(\".\")\n",
      "Nextflow operator join with content: rg.PU = [fcid, lane, index].findAll().join(\".\")\n",
      "Error: Unclosed process block in file: ./git_repos\\demultiplex\\subworkflows\\local\\bases_demultiplex\\main.nf\n",
      "Process FQTK_DEMULTIPLEX in filepath ./git_repos\\demultiplex\\subworkflows\\local\\fqtk_demultiplex\\main.nf\n",
      "Nextflow operator map with content: ch_reads.transpose().map{\n",
      "Nextflow operator groupTuple with content: .groupTuple(by: [0])\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator flatten with content: return [ meta, fastq.flatten() ]\n",
      "Nextflow operator join with content: rg.ID = [fcid,lane].join(\".\")\n",
      "Nextflow operator join with content: rg.PU = [fcid, lane, index].findAll().join(\".\")\n",
      "Error: Unclosed process block in file: ./git_repos\\demultiplex\\subworkflows\\local\\fqtk_demultiplex\\main.nf\n",
      "Process SINGULAR_DEMULTIPLEX in filepath ./git_repos\\demultiplex\\subworkflows\\local\\singular_demultiplex\\main.nf\n",
      "Nextflow operator map with content: ch_reads.transpose().map{\n",
      "Nextflow operator groupTuple with content: .groupTuple(by: [0])\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator flatten with content: return [ meta, fastq.flatten() ]\n",
      "Nextflow operator join with content: rg.ID = [fcid,lane].join(\".\")\n",
      "Nextflow operator join with content: rg.PU = [fcid, lane, index].findAll().join(\".\")\n",
      "Error: Unclosed process block in file: ./git_repos\\demultiplex\\subworkflows\\local\\singular_demultiplex\\main.nf\n",
      "Process BCL_DEMULTIPLEX in filepath ./git_repos\\demultiplex\\subworkflows\\nf-core\\bcl_demultiplex\\main.nf\n",
      "Nextflow operator branch with content: .branch { meta, samplesheet, run ->\n",
      "Nextflow operator set with content: }.set { ch_flowcells }\n",
      "Nextflow operator multiMap with content: .multiMap { meta, samplesheet, run ->\n",
      "Nextflow operator set with content: }.set { ch_flowcells_tar }\n",
      "Nextflow operator join with content: ch_flowcells_tar_merged = ch_flowcells_tar.samplesheets.join( ch_flowcells_tar.run_dirs )\n",
      "Nextflow operator mix with content: ch_flowcells = ch_flowcells.dir.mix(ch_flowcells_tar_merged)\n",
      "Nextflow operator mix with content: ch_fastq    = ch_fastq.mix(BCLCONVERT.out.fastq)\n",
      "Nextflow operator mix with content: ch_interop  = ch_interop.mix(BCLCONVERT.out.interop)\n",
      "Nextflow operator mix with content: ch_reports  = ch_reports.mix(BCLCONVERT.out.reports)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BCLCONVERT.out.versions)\n",
      "Nextflow operator mix with content: ch_fastq    = ch_fastq.mix(BCL2FASTQ.out.fastq)\n",
      "Nextflow operator mix with content: ch_interop  = ch_interop.mix(BCL2FASTQ.out.interop)\n",
      "Nextflow operator mix with content: ch_reports  = ch_reports.mix(BCL2FASTQ.out.reports)\n",
      "Nextflow operator mix with content: ch_stats    = ch_stats.mix(BCL2FASTQ.out.stats)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BCL2FASTQ.out.versions)\n",
      "Nextflow operator map with content: ch_reads.transpose().map{\n",
      "Nextflow operator groupTuple with content: .groupTuple(by: [0])\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator flatten with content: return [ meta, fastq.flatten() ]\n",
      "Nextflow operator join with content: rg.ID = [fcid,lane].join(\".\")\n",
      "Nextflow operator join with content: rg.PU = [fcid, lane, index].findAll().join(\".\")\n",
      "Error: Unclosed process block in file: ./git_repos\\demultiplex\\subworkflows\\nf-core\\bcl_demultiplex\\main.nf\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Process DEMULTIPLEX in filepath ./git_repos\\demultiplex\\workflows\\demultiplex.nf\n",
      "Nextflow operator dump with content: ch_inputs.dump(tag: 'DEMULTIPLEX::inputs',{FormattingService.prettyFormat(it)})\n",
      "Nextflow operator branch with content: .branch { meta, samplesheet, run, manifest ->\n",
      "Nextflow operator multiMap with content: .multiMap { meta, samplesheet, run, manifest ->\n",
      "Nextflow operator dump with content: ch_inputs.dump(tag: 'DEMULTIPLEX::inputs',{FormattingService.prettyFormat(it)})\n",
      "Nextflow operator branch with content: .branch { meta, samplesheet, run ->\n",
      "Nextflow operator multiMap with content: .multiMap { meta, samplesheet, run ->\n",
      "Nextflow operator join with content: if (demultiplexer in ['bclconvert', 'bcl2fastq']) ch_flowcells_tar_merged = ch_flowcells_tar.samplesheets.join(ch_flowcells_tar.run_dirs, failOnMismatch:true, failOnDuplicate:true)\n",
      "Nextflow operator join with content: ch_flowcells_tar_merged = ch_flowcells_tar.samplesheets.join( UNTAR ( ch_flowcells_tar.run_dirs ).untar, failOnMismatch:true, failOnDuplicate:true )\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(UNTAR.out.versions)\n",
      "Nextflow operator mix with content: ch_flowcells = ch_flowcells.dir.mix(ch_flowcells_tar_merged)\n",
      "Nextflow operator mix with content: ch_raw_fastq = ch_raw_fastq.mix(BASES_DEMULTIPLEX.out.fastq)\n",
      "Nextflow operator map with content: ch_multiqc_files = ch_multiqc_files.mix(BASES_DEMULTIPLEX.out.metrics.map { meta, metrics -> return metrics} )\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BASES_DEMULTIPLEX.out.versions)\n",
      "Nextflow operator mix with content: ch_raw_fastq = ch_raw_fastq.mix( BCL_DEMULTIPLEX.out.fastq )\n",
      "Nextflow operator map with content: ch_multiqc_files = ch_multiqc_files.mix( BCL_DEMULTIPLEX.out.reports.map { meta, report -> return report} )\n",
      "Nextflow operator map with content: ch_multiqc_files = ch_multiqc_files.mix( BCL_DEMULTIPLEX.out.stats.map   { meta, stats  -> return stats } )\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BCL_DEMULTIPLEX.out.versions)\n",
      "Nextflow operator map with content: fastq_read_structure = ch_flowcells.map{it[2]}\n",
      "Nextflow operator splitCsv with content: .splitCsv(header:true)\n",
      "Nextflow operator map with content: .map{[it.fastq, it.read_structure]}\n",
      "Nextflow operator collect with content: fastqs_with_paths = fastq_read_structure.combine(UNTAR.out.untar.collect{it[1]}).toList()\n",
      "Nextflow operator merge with content: ch_input = ch_flowcells.merge( fastqs_with_paths ) { a,b -> tuple(a[0], a[1], b)}\n",
      "Nextflow operator mix with content: ch_raw_fastq = ch_raw_fastq.mix(FQTK_DEMULTIPLEX.out.fastq)\n",
      "Nextflow operator map with content: ch_multiqc_files = ch_multiqc_files.mix(FQTK_DEMULTIPLEX.out.metrics.map { meta, metrics -> return metrics} )\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(FQTK_DEMULTIPLEX.out.versions)\n",
      "Nextflow operator mix with content: ch_raw_fastq = ch_raw_fastq.mix(SINGULAR_DEMULTIPLEX.out.fastq)\n",
      "Nextflow operator map with content: ch_multiqc_files = ch_multiqc_files.mix(SINGULAR_DEMULTIPLEX.out.metrics.map { meta, metrics -> return metrics} )\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SINGULAR_DEMULTIPLEX.out.versions)\n",
      "Nextflow operator dump with content: ch_raw_fastq.dump(tag: \"DEMULTIPLEX::Demultiplexed Fastq\",{FormattingService.prettyFormat(it)})\n",
      "Nextflow operator map with content: ch_multiqc_files = ch_multiqc_files.mix( FASTP.out.json.map { meta, json -> return json} )\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(FASTP.out.versions)\n",
      "Nextflow operator map with content: ch_multiqc_files = ch_multiqc_files.mix( FALCO.out.txt.map { meta, txt -> return txt} )\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(FALCO.out.versions)\n",
      "Nextflow operator transpose with content: MD5SUM(ch_fastq_to_qc.transpose())\n",
      "Nextflow operator collectFile with content: ch_versions.unique().collectFile(name: 'collated_versions.yml')\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_workflow_summary.collectFile(name: 'workflow_summary_mqc.yaml'))\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_methods_description.collectFile(name: 'methods_description_mqc.yaml'))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(CUSTOM_DUMPSOFTWAREVERSIONS.out.mqc_yml.collect())\n",
      "Nextflow operator collect with content: ch_multiqc_files.collect().dump(tag: \"DEMULTIPLEX::MultiQC files\",{FormattingService.prettyFormat(it)})\n",
      "Nextflow operator collect with content: ch_multiqc_files.collect(),\n",
      "Nextflow operator toList with content: ch_multiqc_config.toList(),\n",
      "Nextflow operator toList with content: ch_multiqc_custom_config.toList(),\n",
      "Nextflow operator toList with content: ch_multiqc_logo.toList()\n",
      "Nextflow operator toList with content: multiqc_report = MULTIQC.out.report.toList()\n",
      "Nextflow operator collect with content: def all_columns = input_schema.columns.keySet().collect()\n",
      "Nextflow operator first with content: Channel.value(input_csv).splitCsv(strip:true).first().map({ row ->\n",
      "Nextflow operator map with content: Channel.value(input_csv).splitCsv(header:true, strip:true).map({ row ->\n",
      "Nextflow operator toInteger with content: meta.lane = row.lane.toInteger()\n",
      "Error: Unclosed process block in file: ./git_repos\\demultiplex\\workflows\\demultiplex.nf\n",
      "Process NFCORE_DETAXIZER in filepath ./git_repos\\detaxizer\\main.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\detaxizer\\main.nf\n",
      "Process INPUT_CHECK in filepath ./git_repos\\detaxizer\\subworkflows\\local\\input_check.nf\n",
      "Nextflow operator splitCsv with content: .splitCsv ( header:true, sep:',' )\n",
      "Nextflow operator map with content: .map { create_fastq_channel(it) }\n",
      "Nextflow operator set with content: .set { reads }\n",
      "Error: Unclosed process block in file: ./git_repos\\detaxizer\\subworkflows\\local\\input_check.nf\n",
      "Process DETAXIZER in filepath ./git_repos\\detaxizer\\workflows\\detaxizer.nf\n",
      "Nextflow operator map with content: ch_input.map {\n",
      "Nextflow operator branch with content: ch_input.branch {\n",
      "Nextflow operator set with content: }.set {\n",
      "Nextflow operator map with content: ch_short.shortReads.map{\n",
      "Nextflow operator set with content: }.set{\n",
      "Nextflow operator branch with content: ch_input.branch {\n",
      "Nextflow operator set with content: }.set {\n",
      "Nextflow operator map with content: ch_long.longReads.map {\n",
      "Nextflow operator set with content: }.set {\n",
      "Nextflow operator mix with content: ch_combined_short_long = ch_short.mix(ch_long)\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(FASTQC.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(FASTP.out.versions.first())\n",
      "Nextflow operator map with content: ch_kraken2_db = Channel.fromPath(params.kraken2db).map {\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(KRAKEN2PREPARATION.out.versions)\n",
      "Nextflow operator first with content: KRAKEN2PREPARATION.out.db.first(),\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(KRAKEN2_KRAKEN2.out.versions.first())\n",
      "Nextflow operator take with content: KRAKEN2_KRAKEN2.out.report.take(1)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(PARSE_KRAKEN2REPORT.out.versions)\n",
      "Nextflow operator map with content: ch_parsed_kraken2_report = PARSE_KRAKEN2REPORT.out.to_filter.map {meta, path -> path}\n",
      "Nextflow operator combine with content: KRAKEN2_KRAKEN2.out.classified_reads_assignment.combine(ch_parsed_kraken2_report).set{ ch_combined }\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(ISOLATE_IDS_FROM_KRAKEN2_TO_BLASTN.out.versions.first())\n",
      "Nextflow operator join with content: ch_prepare_summary_kraken2 = KRAKEN2_KRAKEN2.out.classified_reads_assignment.join(ISOLATE_IDS_FROM_KRAKEN2_TO_BLASTN.out.classified).map {\n",
      "Nextflow operator map with content: ch_combined_kraken2 = ch_prepare_summary_kraken2.map {\n",
      "Nextflow operator groupTuple with content: .groupTuple(by: [0])\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator flatten with content: path = path.flatten()\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(ch_kraken2_summary.versions.first())\n",
      "Nextflow operator map with content: ch_kraken2_summary = ch_kraken2_summary.summary.map {\n",
      "Nextflow operator join with content: .join(\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(PREPARE_FASTA4BLASTN.out.versions.first())\n",
      "Nextflow operator map with content: ch_reference_fasta_with_meta = ch_reference_fasta.map {\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BLAST_MAKEBLASTDB.out.versions)\n",
      "Nextflow operator flatMap with content: .flatMap { meta, fastaList ->\n",
      "Nextflow operator first with content: ch_blastn_db = BLAST_MAKEBLASTDB.out.db.first()\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(BLAST_BLASTN.out.versions.first())\n",
      "Nextflow operator map with content: ch_combined_blast = BLAST_BLASTN.out.txt.map {\n",
      "Nextflow operator groupTuple with content: ch_combined_blast = ch_combined_blast.groupTuple(\n",
      "Nextflow operator map with content: ).map {\n",
      "Nextflow operator flatten with content: meta, paths -> [ meta, paths.flatten() ]\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(FILTER_BLASTN_IDENTCOV.out.versions.first())\n",
      "Nextflow operator map with content: ch_filtered_combined = FILTER_BLASTN_IDENTCOV.out.classified.map {\n",
      "Nextflow operator groupTuple with content: .groupTuple ()\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator flatten with content: paths = paths.flatten()\n",
      "Nextflow operator join with content: ch_blastn_combined = ch_combined_blast.join(ch_filtered_combined, remainder: true).map{\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(ch_blastn_summary.versions.first())\n",
      "Nextflow operator map with content: ch_blastn_summary = ch_blastn_summary.summary.map {\n",
      "Nextflow operator join with content: .join(ISOLATE_IDS_FROM_KRAKEN2_TO_BLASTN.out.classified_ids, by:[0])\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(FILTER.out.versions.first())\n",
      "Nextflow operator map with content: ch_blastn2filter = FILTER_BLASTN_IDENTCOV.out.classified_ids.map {\n",
      "Nextflow operator groupTuple with content: .groupTuple(by:[0])\n",
      "Nextflow operator map with content: ch_combined_short_long_id = RENAME_FASTQ_HEADERS_PRE.out.fastq.map {\n",
      "Nextflow operator join with content: ch_blastnfilter = ch_combined_short_long_id.join(\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(FILTER.out.versions.first())\n",
      "Nextflow operator join with content: .join(ISOLATE_IDS_FROM_KRAKEN2_TO_BLASTN.out.classified_ids, by:[0])\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(FILTER.out.versions.first())\n",
      "Nextflow operator map with content: ch_blastn2filter = FILTER_BLASTN_IDENTCOV.out.classified_ids.map {\n",
      "Nextflow operator groupTuple with content: .groupTuple(by:[0])\n",
      "Nextflow operator map with content: ch_combined_short_long_id = FASTP.out.reads.map {\n",
      "Nextflow operator join with content: ch_blastnfilter = ch_combined_short_long_id.join(\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(FILTER.out.versions.first())\n",
      "Nextflow operator map with content: ch_headers = RENAME_FASTQ_HEADERS_PRE.out.headers.map {\n",
      "Nextflow operator map with content: ch_filtered2rename = FILTER.out.filtered.map {\n",
      "Nextflow operator join with content: ch_rename_filtered = ch_filtered2rename.join(ch_headers, by:[0])\n",
      "Nextflow operator collect with content: ch_summary = ch_kraken2_summary.mix(ch_blastn_summary).collect().map {\n",
      "Nextflow operator map with content: ch_summary = ch_kraken2_summary.map {\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ch_summary.versions)\n",
      "Nextflow operator collectFile with content: ch_versions.unique().collectFile(name: 'collated_versions.yml')\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_workflow_summary.collectFile(name: 'workflow_summary_mqc.yaml'))\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_methods_description.collectFile(name: 'methods_description_mqc.yaml'))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(CUSTOM_DUMPSOFTWAREVERSIONS.out.mqc_yml.collect())\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(FASTQC.out.zip.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files.collect(),\n",
      "Nextflow operator toList with content: ch_multiqc_config.toList(),\n",
      "Nextflow operator toList with content: ch_multiqc_custom_config.toList(),\n",
      "Nextflow operator toList with content: ch_multiqc_logo.toList()\n",
      "Nextflow operator toList with content: multiqc_report = MULTIQC.out.report.toList()\n",
      "Error: Unclosed process block in file: ./git_repos\\detaxizer\\workflows\\detaxizer.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\diaproteomics\\main.nf\n",
      "Process NFCORE_DIFFERENTIALABUNDANCE in filepath ./git_repos\\differentialabundance\\main.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\differentialabundance\\main.nf\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Process DIFFERENTIALABUNDANCE in filepath ./git_repos\\differentialabundance\\workflows\\differentialabundance.nf\n",
      "Nextflow operator join with content: .join(UNTAR.out.untar)\n",
      "Nextflow operator mix with content: .mix(AFFY_JUSTRMA_RAW.out.versions)\n",
      "Nextflow operator splitCsv with content: .splitCsv ( header:true, sep:(params.contrasts.endsWith('tsv') ? '\\t' : ','))\n",
      "Nextflow operator first with content: .map{ it.tail().first() }\n",
      "Nextflow operator map with content: .map{\n",
      "Nextflow operator unique with content: .unique()   // uniquify to keep each contrast variable only once (in case it exists in multiple lines for blocking etc.)\n",
      "Nextflow operator combine with content: ch_contrast_variables.combine(proteus_in)\n",
      "Nextflow operator first with content: .first()\n",
      "Nextflow operator map with content: .map{ meta, matrix -> tuple(exp_meta, matrix) }\n",
      "Nextflow operator first with content: .first()\n",
      "Nextflow operator map with content: .map{ meta, matrix -> tuple(exp_meta, matrix) }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(PROTEUS.out.versions)\n",
      "Nextflow operator mix with content: .mix(GEOQUERY_GETGEO.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GUNZIP_GTF.out.versions)\n",
      "Nextflow operator map with content: .map{\n",
      "Nextflow operator mix with content: .mix(GTF_TO_TABLE.out.versions)\n",
      "Nextflow operator map with content: .map{ meta, matrix ->\n",
      "Nextflow operator join with content: .join(ch_in_norm)\n",
      "Nextflow operator map with content: .map{tuple(it[0], [it[1], it[2]])}\n",
      "Nextflow operator join with content: ch_input.join(ch_matrices_for_validation),\n",
      "Nextflow operator transpose with content: .transpose()\n",
      "Nextflow operator branch with content: .branch {\n",
      "Nextflow operator map with content: .map{it[1]}\n",
      "Nextflow operator splitCsv with content: .splitCsv ( header:true, sep:'\\t' )\n",
      "Nextflow operator map with content: .map{\n",
      "Nextflow operator join with content: it.id = it.values().join('_')\n",
      "Nextflow operator join with content: .join(CUSTOM_MATRIXFILTER.out.filtered)     // -> meta, samplesheet, filtered matrix\n",
      "Nextflow operator first with content: .first()\n",
      "Nextflow operator mix with content: .mix(LIMMA_DIFFERENTIAL.out.versions)\n",
      "Nextflow operator map with content: .map{ it.tail() }\n",
      "Nextflow operator first with content: .first()\n",
      "Nextflow operator first with content: ch_contrasts.first(),\n",
      "Nextflow operator mix with content: .mix(DESEQ2_DIFFERENTIAL.out.versions)\n",
      "Nextflow operator join with content: ch_processed_matrices = ch_processed_matrices.join(DESEQ2_NORM.out.rlog_counts)\n",
      "Nextflow operator join with content: ch_processed_matrices = ch_processed_matrices.join(DESEQ2_NORM.out.vst_counts)\n",
      "Nextflow operator map with content: .map{ it.tail() }\n",
      "Nextflow operator map with content: .map{it[0]} // revert back to contrasts meta map\n",
      "Nextflow operator combine with content: .combine( VALIDATOR.out.sample_meta.map { it[1] } )\n",
      "Nextflow operator map with content: VALIDATOR.out.feature_meta.map{ it[1] },\n",
      "Nextflow operator map with content: .map{ it.tail() }\n",
      "Nextflow operator combine with content: .combine(CUSTOM_TABULARTOGSEACLS.out.cls)\n",
      "Nextflow operator map with content: .map{ tuple(it[1], it[0], it[2]) }\n",
      "Nextflow operator combine with content: .combine(ch_gene_sets)\n",
      "Nextflow operator map with content: ch_gsea_inputs.map{ tuple(it[0].reference, it[0].target) }, // *\n",
      "Nextflow operator first with content: TABULAR_TO_GSEA_CHIP.out.chip.first()\n",
      "Nextflow operator join with content: .join(GSEA_GSEA.out.report_tsvs_target)\n",
      "Nextflow operator mix with content: .mix(TABULAR_TO_GSEA_CHIP.out.versions)\n",
      "Nextflow operator mix with content: .mix(GSEA_GSEA.out.versions)\n",
      "Nextflow operator map with content: .map{\n",
      "Nextflow operator unique with content: .unique()\n",
      "Nextflow operator combine with content: ch_mat = ch_raw.combine(ch_processed_matrices)\n",
      "Nextflow operator join with content: .join(VALIDATOR.out.feature_meta)                       // meta, samples, features\n",
      "Nextflow operator join with content: .join(ch_mat)                                           // meta, samples, features, raw, norm (or just norm)\n",
      "Nextflow operator map with content: .map{\n",
      "Nextflow operator first with content: .first()\n",
      "Nextflow operator combine with content: .combine(ch_all_matrices.map{ it.tail() })\n",
      "Nextflow operator mix with content: .mix(VALIDATOR.out.versions)\n",
      "Nextflow operator mix with content: .mix(PLOT_EXPLORATORY.out.versions)\n",
      "Nextflow operator mix with content: .mix(PLOT_DIFFERENTIAL.out.versions)\n",
      "Nextflow operator collectFile with content: ch_versions.unique().collectFile(name: 'collated_versions.yml')\n",
      "Nextflow operator map with content: .map{ tuple(exp_meta, it) }\n",
      "Nextflow operator map with content: .map{ it.tail() }\n",
      "Nextflow operator flatten with content: .map{it.flatten()}\n",
      "Nextflow operator combine with content: .combine(VALIDATOR.out.contrasts.map{it.tail()})\n",
      "Nextflow operator combine with content: .combine(CUSTOM_DUMPSOFTWAREVERSIONS.out.yml)\n",
      "Nextflow operator combine with content: .combine(ch_logo_file)\n",
      "Nextflow operator combine with content: .combine(ch_css_file)\n",
      "Nextflow operator combine with content: .combine(ch_citations_file)\n",
      "Nextflow operator combine with content: .combine(ch_differential.map{it[1]}.toList())\n",
      "Nextflow operator combine with content: .combine(ch_model.map{it[1]}.toList())\n",
      "Nextflow operator combine with content: .combine(ch_gsea_results\n",
      "Nextflow operator flatMap with content: .map{it.tail()}.flatMap().toList()\n",
      "Nextflow operator first with content: ch_app_differential = ch_differential.first().map{it[0].keySet().join(',')}\n",
      "Nextflow operator concat with content: .concat(\n",
      "Nextflow operator join with content: ch_differential.map{it[0].values().join(',')}\n",
      "Nextflow operator collectFile with content: .collectFile(name: 'contrasts.csv', newLine: true, sort: false)\n",
      "Nextflow operator map with content: .map{\n",
      "Nextflow operator collect with content: .combine(ch_differential.map{it[1]}.collect().map{[it]})\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SHINYNGS_APP.out.versions)\n",
      "Nextflow operator collect with content: params.exploratory_assay_names.split(',').collect { \"${it}_matrix\".toString() } +\n",
      "Nextflow operator map with content: .map{\n",
      "Nextflow operator collect with content: [report_file_names, it.collect{ f -> f.name}].transpose().collectEntries()\n",
      "Nextflow operator combine with content: .combine(ch_report_input_files)\n",
      "Nextflow operator map with content: .map{[it[0], it[1..-1]]}\n",
      "Error: Unclosed process block in file: ./git_repos\\differentialabundance\\workflows\\differentialabundance.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\dualrnaseq\\main.nf\n",
      "Process NFCORE_EPITOPEPREDICTION in filepath ./git_repos\\epitopeprediction\\main.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\epitopeprediction\\main.nf\n",
      "Process INPUT_CHECK in filepath ./git_repos\\epitopeprediction\\subworkflows\\local\\input_check.nf\n",
      "Nextflow operator splitCsv with content: .splitCsv ( header:true, sep:',' )\n",
      "Nextflow operator map with content: .map { get_samplesheet_paths(it) }\n",
      "Nextflow operator set with content: .set { reads }\n",
      "Nextflow operator join with content: allele_string = file(alleles).readLines().join(';')\n",
      "Process EPITOPEPREDICTION in filepath ./git_repos\\epitopeprediction\\workflows\\epitopeprediction.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(INPUT_CHECK.out.versions)\n",
      "Nextflow operator branch with content: .branch {\n",
      "Nextflow operator set with content: .set { ch_samples_from_sheet }\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(GUNZIP_VCF.out.versions.ifEmpty(null))\n",
      "Nextflow operator mix with content: .mix(ch_samples_from_sheet.variant_uncompressed)\n",
      "Nextflow operator mix with content: .mix(ch_samples_from_sheet.peptide)\n",
      "Nextflow operator mix with content: .mix(ch_variants_uncompressed)\n",
      "Nextflow operator branch with content: .branch {\n",
      "Nextflow operator filter with content: .filter( ~/(?i)^net.*/ )\n",
      "Nextflow operator map with content: .map { it -> \"${it.split('-')[0]}: ${it.split('-')[1]}\"}\n",
      "Nextflow operator collect with content: .collect()\n",
      "Nextflow operator ifEmpty with content: GET_PREDICTION_VERSIONS(ch_external_versions.ifEmpty(\"\"))\n",
      "Nextflow operator ifEmpty with content: ch_prediction_tool_versions = GET_PREDICTION_VERSIONS.out.versions.ifEmpty(null)\n",
      "Nextflow operator mix with content: .mix(ch_samples_uncompressed.variant, ch_samples_uncompressed.peptide)\n",
      "Nextflow operator combine with content: .combine(ch_prediction_tool_versions)\n",
      "Nextflow operator first with content: .first()\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(SHOW_SUPPORTED_MODELS.out.versions.ifEmpty(null))\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(EPYTOPE_CHECK_REQUESTED_MODELS.out.versions.ifEmpty(null))\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(EPYTOPE_CHECK_REQUESTED_MODELS_PROTEIN.out.versions.ifEmpty(null))\n",
      "Nextflow operator map with content: .map { meta_data, input_file -> tuple( meta_data, input_file ) },\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(EPYTOPE_CHECK_REQUESTED_MODELS_PEP.out.versions.ifEmpty(null))\n",
      "Nextflow operator mix with content: .mix( EPYTOPE_CHECK_REQUESTED_MODELS_PEP.out.log )\n",
      "Nextflow operator mix with content: .mix (EPYTOPE_CHECK_REQUESTED_MODELS_PROTEIN.out.log )\n",
      "Nextflow operator subscribe with content: .subscribe {\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(EXTERNAL_TOOLS_IMPORT.out.versions.ifEmpty(null))\n",
      "Nextflow operator branch with content: .branch {\n",
      "Nextflow operator set with content: .set { ch_variants }\n",
      "Nextflow operator set with content: .set { ch_split_variants }\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix( VARIANT_SPLIT.out.versions.ifEmpty(null) )\n",
      "Nextflow operator set with content: .set { ch_split_variants }\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix( SNPSIFT_SPLIT.out.versions.ifEmpty(null) )\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix( CSVTK_SPLIT.out.versions.ifEmpty(null) )\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(EPYTOPE_GENERATE_PEPTIDES.out.versions.ifEmpty(null))\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(SPLIT_PEPTIDES_PROTEIN.out.versions.ifEmpty(null))\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix( SPLIT_PEPTIDES_PEPTIDES.out.versions.ifEmpty(null) )\n",
      "Nextflow operator combine with content: .combine( ch_prediction_tool_versions )\n",
      "Nextflow operator transpose with content: .transpose(),\n",
      "Nextflow operator collect with content: EXTERNAL_TOOLS_IMPORT.out.nonfree_tools.collect().ifEmpty([])\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix( EPYTOPE_PEPTIDE_PREDICTION_PROTEIN.out.versions.ifEmpty(null) )\n",
      "Nextflow operator combine with content: .combine( ch_prediction_tool_versions )\n",
      "Nextflow operator transpose with content: .transpose(),\n",
      "Nextflow operator collect with content: EXTERNAL_TOOLS_IMPORT.out.nonfree_tools.collect().ifEmpty([])\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix( EPYTOPE_PEPTIDE_PREDICTION_PEP.out.versions.ifEmpty(null) )\n",
      "Nextflow operator mix with content: .mix( ch_split_variants.splitted )\n",
      "Nextflow operator combine with content: .combine( ch_prediction_tool_versions )\n",
      "Nextflow operator transpose with content: .transpose(),\n",
      "Nextflow operator collect with content: EXTERNAL_TOOLS_IMPORT.out.nonfree_tools.collect().ifEmpty([])\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix( EPYTOPE_PEPTIDE_PREDICTION_VAR.out.versions.ifEmpty(null) )\n",
      "Nextflow operator mix with content: .mix( EPYTOPE_PEPTIDE_PREDICTION_VAR.out.predicted, EPYTOPE_PEPTIDE_PREDICTION_PROTEIN.out.predicted )\n",
      "Nextflow operator groupTuple with content: .groupTuple()\n",
      "Nextflow operator flatMap with content: .flatMap { meta_data, predicted -> [[[ sample:meta_data.sample, alleles:meta_data.alleles, files:predicted.size() ], predicted ]] }\n",
      "Nextflow operator branch with content: .branch {\n",
      "Nextflow operator set with content: .set { ch_predicted_peptides }\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix( CAT_TSV.out.versions.ifEmpty(null) )\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix( CSVTK_CONCAT.out.versions.ifEmpty(null) )\n",
      "Nextflow operator mix with content: .mix( EPYTOPE_PEPTIDE_PREDICTION_VAR.out.fasta, EPYTOPE_PEPTIDE_PREDICTION_PROTEIN.out.fasta )\n",
      "Nextflow operator groupTuple with content: .groupTuple()\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix( CAT_FASTA.out.versions.ifEmpty(null) )\n",
      "Nextflow operator mix with content: .mix( EPYTOPE_PEPTIDE_PREDICTION_VAR.out.json )\n",
      "Nextflow operator mix with content: .mix( EPYTOPE_PEPTIDE_PREDICTION_PROTEIN.out.json )\n",
      "Nextflow operator groupTuple with content: .groupTuple()\n",
      "Nextflow operator flatMap with content: .flatMap { meta, json -> [[[ sample:meta.sample, alleles:meta.alleles, files:json.size() ], json ]] }\n",
      "Nextflow operator branch with content: .branch {\n",
      "Nextflow operator set with content: .set { ch_json_reports }\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix( MERGE_JSON_SINGLE.out.versions.ifEmpty(null) )\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix( MERGE_JSON_MULTI.out.versions.ifEmpty(null) )\n",
      "Nextflow operator collectFile with content: ch_versions.unique().collectFile(name: 'collated_versions.yml')\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_workflow_summary.collectFile(name: 'workflow_summary_mqc.yaml'))\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_methods_description.collectFile(name: 'methods_description_mqc.yaml'))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(CUSTOM_DUMPSOFTWAREVERSIONS.out.mqc_yml.collect())\n",
      "Nextflow operator collect with content: ch_multiqc_files.collect(),\n",
      "Nextflow operator toList with content: ch_multiqc_config.toList(),\n",
      "Nextflow operator toList with content: ch_multiqc_custom_config.toList(),\n",
      "Nextflow operator toList with content: ch_multiqc_logo.toList()\n",
      "Nextflow operator toList with content: multiqc_report = MULTIQC.out.report.toList()\n",
      "Error: Unclosed process block in file: ./git_repos\\epitopeprediction\\workflows\\epitopeprediction.nf\n",
      "Process NFCORE_FASTQUORUM in filepath ./git_repos\\fastquorum\\main.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\fastquorum\\main.nf\n",
      "Process INPUT_CHECK in filepath ./git_repos\\fastquorum\\subworkflows\\local\\input_check.nf\n",
      "Nextflow operator splitCsv with content: .splitCsv ( header:true, sep:',' )\n",
      "Nextflow operator map with content: .map { create_fastq_channel(it) }\n",
      "Nextflow operator set with content: .set { reads }\n",
      "Error: Unclosed process block in file: ./git_repos\\fastquorum\\subworkflows\\local\\input_check.nf\n",
      "Process FASTQUORUM in filepath ./git_repos\\fastquorum\\workflows\\fastquorum.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(INPUT_CHECK.out.versions)\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(FASTQC.out.versions.first())\n",
      "Nextflow operator collectFile with content: ch_versions.unique().collectFile(name: 'collated_versions.yml')\n",
      "Nextflow operator set with content: CALLDDUPLEXCONSENSUSREADS.out.bam.set { ch_consensus_bam }\n",
      "Nextflow operator set with content: CALLMOLECULARCONSENSUSREADS.out.bam.set { ch_consensus_bam }\n",
      "Nextflow operator mix with content: ch_multiqc_files = ch_multiqc_files.mix(Channel.from(ch_multiqc_config))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(ch_multiqc_custom_config.collect().ifEmpty([]))\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_workflow_summary.collectFile(name: 'workflow_summary_mqc.yaml'))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(CUSTOM_DUMPSOFTWAREVERSIONS.out.mqc_yml.collect())\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(FASTQC.out.zip.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(GROUPREADSBYUMI.out.histogram.map{it[1]}.collect())\n",
      "Nextflow operator collect with content: ch_multiqc_files.collect()\n",
      "Nextflow operator toList with content: multiqc_report = MULTIQC.out.report.toList()\n",
      "Nextflow operator mix with content: ch_versions    = ch_versions.mix(MULTIQC.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\fastquorum\\workflows\\fastquorum.nf\n",
      "Process NFCORE_FETCHNGS in filepath ./git_repos\\fetchngs\\main.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\fetchngs\\main.nf\n",
      "Process PIPELINE_INITIALISATION in filepath ./git_repos\\fetchngs\\subworkflows\\local\\nf_core_fetchngs_utils\\main.nf\n",
      "Nextflow operator splitCsv with content: .splitCsv(header:false, sep:'', strip:true)\n",
      "Nextflow operator map with content: .map { it[0] }\n",
      "Nextflow operator unique with content: .unique()\n",
      "Nextflow operator set with content: .set { ch_ids }\n",
      "Process PIPELINE_COMPLETION in filepath ./git_repos\\fetchngs\\subworkflows\\local\\nf_core_fetchngs_utils\\main.nf\n",
      "Nextflow operator mix with content: versions = versions.mix(pipeline_version_info)\n",
      "Nextflow operator collectFile with content: versions.collectFile(name: 'fetchngs_mqc_versions.yml', storeDir: \"${params.outdir}/pipeline_info\")\n",
      "Nextflow operator join with content: error(\"Mixture of ids provided via --input: ${no_match_ids.join(', ')}\\nPlease provide either SRA / ENA / GEO / DDBJ or Synapse ids!\")\n",
      "Nextflow operator join with content: error(\"Mixture of ids provided via --input: ${no_match_ids.join(', ')}\\nPlease provide either SRA / ENA / GEO / DDBJ or Synapse ids!\")\n",
      "Nextflow operator collect with content: def ena_metadata_fields = params.ena_metadata_fields ? params.ena_metadata_fields.split(',').collect{ it.trim().toLowerCase() } : valid_ena_metadata_fields\n",
      "Nextflow operator join with content: error(\"Invalid option: '${params.ena_metadata_fields}'. Minimally required fields for '--ena_metadata_fields': '${valid_ena_metadata_fields.join(',')}'\")\n",
      "Nextflow operator min with content: int c=Math.min(groupCount, matcher.groupCount())\n",
      "Error: Unclosed process block in file: ./git_repos\\fetchngs\\subworkflows\\local\\nf_core_fetchngs_utils\\main.nf\n",
      "Process FASTQ_DOWNLOAD_PREFETCH_FASTERQDUMP_SRATOOLS in filepath ./git_repos\\fetchngs\\subworkflows\\nf-core\\fastq_download_prefetch_fasterqdump_sratools\\main.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(CUSTOM_SRATOOLSNCBISETTINGS.out.versions)\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SRATOOLS_PREFETCH.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SRATOOLS_FASTERQDUMP.out.versions.first())\n",
      "Error: Unclosed process block in file: ./git_repos\\fetchngs\\subworkflows\\nf-core\\fastq_download_prefetch_fasterqdump_sratools\\main.nf\n",
      "Process NEXTFLOW_PIPELINE_UTILS in filepath ./git_repos\\fetchngs\\subworkflows\\nf-core\\nextflowpipelineutils\\main.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\fetchngs\\subworkflows\\nf-core\\nextflowpipelineutils\\main.nf\n",
      "Process NFCORE_PIPELINE_UTILS in filepath ./git_repos\\fetchngs\\subworkflows\\nf-core\\nfcore_pipeline_utils\\main.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\fetchngs\\subworkflows\\nf-core\\nfcore_pipeline_utils\\main.nf\n",
      "Process NF_VALIDATION_PLUGIN_UTILS in filepath ./git_repos\\fetchngs\\subworkflows\\nf-core\\nfvalidation_plugin_utils\\main.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\fetchngs\\subworkflows\\nf-core\\nfvalidation_plugin_utils\\main.nf\n",
      "Process SRA in filepath ./git_repos\\fetchngs\\workflows\\sra\\main.nf\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SRA_IDS_TO_RUNINFO.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SRA_RUNINFO_TO_FTP.out.versions.first())\n",
      "Nextflow operator splitCsv with content: .splitCsv(header:true, sep:'\\t')\n",
      "Nextflow operator map with content: .map{ meta ->\n",
      "Nextflow operator unique with content: .unique()\n",
      "Nextflow operator set with content: .set { ch_sra_metadata }\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator branch with content: .branch {\n",
      "Nextflow operator set with content: .set { ch_sra_reads }\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SRA_FASTQ_FTP.out.versions.first())\n",
      "Nextflow operator map with content: ch_sra_reads.sra.map { meta, reads -> [ meta, meta.run_accession ] },\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(FASTQ_DOWNLOAD_PREFETCH_FASTERQDUMP_SRATOOLS.out.versions.first())\n",
      "Nextflow operator mix with content: .mix(SRA_FASTQ_FTP.out.fastq, FASTQ_DOWNLOAD_PREFETCH_FASTERQDUMP_SRATOOLS.out.reads)\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator flatten with content: def reads = fastq instanceof List ? fastq.flatten() : [ fastq ]\n",
      "Nextflow operator set with content: .set { ch_sra_metadata }\n",
      "Nextflow operator collect with content: SRA_TO_SAMPLESHEET.out.samplesheet.collect{it[1]},\n",
      "Nextflow operator collect with content: SRA_TO_SAMPLESHEET.out.mappings.collect{it[1]}\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SRA_MERGE_SAMPLESHEET.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(MULTIQC_MAPPINGS_CONFIG.out.versions)\n",
      "Nextflow operator unique with content: versions        = ch_versions.unique()\n",
      "Error: Unclosed process block in file: ./git_repos\\fetchngs\\workflows\\sra\\main.nf\n",
      "Process SYNAPSE in filepath ./git_repos\\fetchngs\\workflows\\synapse\\main.nf\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SYNAPSE_LIST.out.versions.first())\n",
      "Nextflow operator splitCsv with content: .splitCsv(header:false, sep:' ')\n",
      "Nextflow operator map with content: .map { it[0] }\n",
      "Nextflow operator unique with content: .unique()\n",
      "Nextflow operator set with content: .set { ch_samples }\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SYNAPSE_SHOW.out.versions.first())\n",
      "Nextflow operator map with content: .map { it -> WorkflowMain.synapseShowToMap(it) }\n",
      "Nextflow operator set with content: .set { ch_samples_meta }\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SYNAPSE_GET.out.versions.first())\n",
      "Nextflow operator map with content: .map { meta, fastq -> [ WorkflowMain.synapseSampleNameFromFastQ( fastq , \"*{1,2}*\"), fastq ] }\n",
      "Nextflow operator groupTuple with content: .groupTuple(sort: { it -> it.baseName })\n",
      "Nextflow operator set with content: .set { ch_fastq }\n",
      "Nextflow operator map with content: .map { meta, fastq -> [ WorkflowMain.synapseSampleNameFromFastQ( fastq , \"*{1,2}*\"), meta.id ] }\n",
      "Nextflow operator groupTuple with content: .groupTuple()\n",
      "Nextflow operator join with content: .join(ch_fastq)\n",
      "Nextflow operator map with content: .map { id, synids, fastq ->\n",
      "Nextflow operator join with content: def meta = [ id:id, synapse_ids:synids.join(';') ]\n",
      "Nextflow operator set with content: .set { ch_fastq }\n",
      "Nextflow operator collect with content: SYNAPSE_TO_SAMPLESHEET.out.samplesheet.collect{ it[1] }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SYNAPSE_MERGE_SAMPLESHEET.out.versions)\n",
      "Nextflow operator unique with content: versions    = ch_versions.unique()\n",
      "Error: Unclosed process block in file: ./git_repos\\fetchngs\\workflows\\synapse\\main.nf\n",
      "Process NFCORE_FUNCSCAN in filepath ./git_repos\\funcscan\\main.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\funcscan\\main.nf\n",
      "Process AMP in filepath ./git_repos\\funcscan\\subworkflows\\local\\amp.nf\n",
      "Nextflow operator mix with content: ch_versions                 = ch_versions.mix(AMPLIFY_PREDICT.out.versions)\n",
      "Nextflow operator mix with content: ch_ampresults_for_ampcombi  = ch_ampresults_for_ampcombi.mix(AMPLIFY_PREDICT.out.tsv)\n",
      "Nextflow operator mix with content: ch_versions                 = ch_versions.mix(MACREL_CONTIGS.out.versions)\n",
      "Nextflow operator mix with content: ch_versions                 = ch_versions.mix(GUNZIP_MACREL_PRED.out.versions)\n",
      "Nextflow operator mix with content: ch_versions                 = ch_versions.mix(GUNZIP_MACREL_ORFS.out.versions)\n",
      "Nextflow operator mix with content: ch_ampresults_for_ampcombi  = ch_ampresults_for_ampcombi.mix(GUNZIP_MACREL_PRED.out.gunzip)\n",
      "Nextflow operator mix with content: ch_macrel_faa               = ch_macrel_faa.mix(GUNZIP_MACREL_ORFS.out.gunzip)\n",
      "Nextflow operator mix with content: ch_faa_for_ampcombi         = ch_faa_for_ampcombi.mix(ch_macrel_faa)\n",
      "Nextflow operator mix with content: ch_versions                 = ch_versions.mix(AMPIR.out.versions)\n",
      "Nextflow operator mix with content: ch_ampresults_for_ampcombi  = ch_ampresults_for_ampcombi.mix(AMPIR.out.amps_tsv)\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator combine with content: ch_in_for_amp_hmmsearch = ch_faa_for_amp_hmmsearch.combine(ch_amp_hmm_models_meta)\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(AMP_HMMER_HMMSEARCH.out.versions)\n",
      "Nextflow operator groupTuple with content: .groupTuple()\n",
      "Nextflow operator join with content: .join( ch_faa_for_ampcombi )\n",
      "Nextflow operator multiMap with content: .multiMap{\n",
      "Nextflow operator mix with content: ch_ampcombi_summaries = ch_ampcombi_summaries.mix(AMPCOMBI.out.csv)\n",
      "Nextflow operator multiMap with content: .multiMap{\n",
      "Nextflow operator collectFile with content: .combine(ch_ampcombi_summaries_out.summary.collectFile(name: 'ampcombi_complete_summary.csv', keepHeader:true))\n",
      "Error: Unclosed process block in file: ./git_repos\\funcscan\\subworkflows\\local\\amp.nf\n",
      "Process ARG in filepath ./git_repos\\funcscan\\subworkflows\\local\\arg.nf\n",
      "Nextflow operator first with content: .first()\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(AMRFINDERPLUS_UPDATE.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(AMRFINDERPLUS_RUN.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(HAMRONIZATION_AMRFINDERPLUS.out.versions)\n",
      "Nextflow operator mix with content: ch_input_to_hamronization_summarize = ch_input_to_hamronization_summarize.mix(HAMRONIZATION_AMRFINDERPLUS.out.json)\n",
      "Nextflow operator combine with content: .combine(ch_fargene_classes)\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator multiMap with content: .multiMap {\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(FARGENE.out.versions)\n",
      "Nextflow operator transpose with content: HAMRONIZATION_FARGENE ( FARGENE.out.hmm.transpose(), 'json', '0.1', '0.1' )\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(HAMRONIZATION_FARGENE.out.versions)\n",
      "Nextflow operator mix with content: ch_input_to_hamronization_summarize = ch_input_to_hamronization_summarize.mix(HAMRONIZATION_FARGENE.out.json)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(RGI_MAIN.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(HAMRONIZATION_RGI.out.versions)\n",
      "Nextflow operator mix with content: ch_input_to_hamronization_summarize = ch_input_to_hamronization_summarize.mix(HAMRONIZATION_RGI.out.json)\n",
      "Nextflow operator first with content: .first()\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(DEEPARG_DOWNLOADDATA.out.versions)\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator set with content: .set { ch_input_for_deeparg }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(DEEPARG_PREDICT.out.versions)\n",
      "Nextflow operator mix with content: HAMRONIZATION_DEEPARG ( DEEPARG_PREDICT.out.arg.mix(DEEPARG_PREDICT.out.potential_arg), 'json', '1.0.2', params.arg_deeparg_data_version )\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(HAMRONIZATION_DEEPARG.out.versions)\n",
      "Nextflow operator mix with content: ch_input_to_hamronization_summarize = ch_input_to_hamronization_summarize.mix(HAMRONIZATION_DEEPARG.out.json)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ABRICATE_RUN.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(HAMRONIZATION_ABRICATE.out.versions)\n",
      "Nextflow operator mix with content: ch_input_to_hamronization_summarize = ch_input_to_hamronization_summarize.mix(HAMRONIZATION_ABRICATE.out.json)\n",
      "Nextflow operator map with content: .map{\n",
      "Nextflow operator collect with content: .collect()\n",
      "Nextflow operator set with content: .set { ch_input_for_hamronization_summarize }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(HAMRONIZATION_SUMMARIZE.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\funcscan\\subworkflows\\local\\arg.nf\n",
      "Process BGC in filepath ./git_repos\\funcscan\\subworkflows\\local\\bgc.nf\n",
      "Nextflow operator first with content: .first()\n",
      "Nextflow operator first with content: .first()\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(UNTAR_CSS.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(UNTAR_DETECTION.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(UNTAR_MODULES.out.versions)\n",
      "Nextflow operator map with content: ANTISMASH_ANTISMASHLITEDOWNLOADDATABASES ( UNTAR_CSS.out.untar.map{ it[1] }, UNTAR_DETECTION.out.untar.map{ it[1] }, UNTAR_MODULES.out.untar.map{ it[1] } )\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ANTISMASH_ANTISMASHLITEDOWNLOADDATABASES.out.versions)\n",
      "Nextflow operator join with content: ch_antismash_input = fna.join(gff, by: 0)\n",
      "Nextflow operator filter with content: .filter {\n",
      "Nextflow operator multiMap with content: .multiMap {\n",
      "Nextflow operator filter with content: ch_antismash_input = gbk.filter {\n",
      "Nextflow operator filter with content: ch_antismash_input = gbk.filter {\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ANTISMASH_ANTISMASHLITE.out.versions)\n",
      "Nextflow operator mix with content: .mix(ANTISMASH_ANTISMASHLITE.out.gbk_input)\n",
      "Nextflow operator groupTuple with content: .groupTuple()\n",
      "Nextflow operator map with content: .map{\n",
      "Nextflow operator flatten with content: [meta, files.flatten()]\n",
      "Nextflow operator mix with content: ch_bgcresults_for_combgc = ch_bgcresults_for_combgc.mix(ch_antismashresults_for_combgc)\n",
      "Nextflow operator first with content: .first()\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(DEEPBGC_DOWNLOAD.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(DEEPBGC_PIPELINE.out.versions)\n",
      "Nextflow operator mix with content: ch_bgcresults_for_combgc = ch_bgcresults_for_combgc.mix(DEEPBGC_PIPELINE.out.bgc_tsv)\n",
      "Nextflow operator groupTuple with content: ch_gecco_input = fna.groupTuple()\n",
      "Nextflow operator multiMap with content: .multiMap {\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GECCO_RUN.out.versions)\n",
      "Nextflow operator mix with content: .mix(GECCO_RUN.out.clusters)\n",
      "Nextflow operator groupTuple with content: .groupTuple()\n",
      "Nextflow operator map with content: .map{\n",
      "Nextflow operator flatten with content: [meta, files.flatten()]\n",
      "Nextflow operator mix with content: ch_bgcresults_for_combgc = ch_bgcresults_for_combgc.mix(ch_geccoresults_for_combgc)\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator combine with content: ch_in_for_bgc_hmmsearch = ch_faa_for_bgc_hmmsearch.combine(ch_bgc_hmm_models_meta)\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BGC_HMMER_HMMSEARCH.out.versions)\n",
      "Nextflow operator collectFile with content: ch_combgc_summaries = COMBGC.out.tsv.map{ it[1] }.collectFile(name: 'combgc_complete_summary.tsv', storeDir: \"${params.outdir}/reports/combgc\", keepHeader:true)\n",
      "Error: Unclosed process block in file: ./git_repos\\funcscan\\subworkflows\\local\\bgc.nf\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Process FUNCSCAN in filepath ./git_repos\\funcscan\\workflows\\funcscan.nf\n",
      "Nextflow operator branch with content: .branch {\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GUNZIP_FASTA_PREP.out.versions)\n",
      "Nextflow operator mix with content: .mix(fasta_prep.uncompressed)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BIOAWK.out.versions)\n",
      "Nextflow operator join with content: .join( BIOAWK.out.longest )\n",
      "Nextflow operator map with content: .map{\n",
      "Nextflow operator mix with content: ch_versions              = ch_versions.mix(PRODIGAL_GFF.out.versions)\n",
      "Nextflow operator mix with content: ch_versions              = ch_versions.mix(PRODIGAL_GBK.out.versions)\n",
      "Nextflow operator mix with content: ch_versions              = ch_versions.mix(PYRODIGAL.out.versions)\n",
      "Nextflow operator mix with content: ch_versions              = ch_versions.mix(PROKKA.out.versions)\n",
      "Nextflow operator first with content: .first()\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix( BAKTA_BAKTADBDOWNLOAD.out.versions )\n",
      "Nextflow operator mix with content: ch_versions              = ch_versions.mix(BAKTA_BAKTA.out.versions)\n",
      "Nextflow operator filter with content: .filter {\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(AMP.out.versions)\n",
      "Nextflow operator filter with content: .filter {\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ARG.out.versions)\n",
      "Nextflow operator filter with content: .filter {\n",
      "Nextflow operator filter with content: .filter {\n",
      "Nextflow operator filter with content: .filter {\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BGC.out.versions)\n",
      "Nextflow operator collectFile with content: ch_versions.unique().collectFile(name: 'collated_versions.yml')\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_workflow_summary.collectFile(name: 'workflow_summary_mqc.yaml'))\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_methods_description.collectFile(name: 'methods_description_mqc.yaml'))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(CUSTOM_DUMPSOFTWAREVERSIONS.out.mqc_yml.collect())\n",
      "Nextflow operator collect with content: if(params.annotation_tool=='prokka'){ch_multiqc_files = ch_multiqc_files.mix( PROKKA.out.txt.collect{it[1]}.ifEmpty([])) }\n",
      "Nextflow operator collect with content: ch_multiqc_files.collect(),\n",
      "Nextflow operator toList with content: ch_multiqc_config.toList(),\n",
      "Nextflow operator toList with content: ch_multiqc_custom_config.toList(),\n",
      "Nextflow operator toList with content: ch_multiqc_logo.toList()\n",
      "Nextflow operator toList with content: multiqc_report = MULTIQC.out.report.toList()\n",
      "Error: Unclosed process block in file: ./git_repos\\funcscan\\workflows\\funcscan.nf\n",
      "Process NFCORE_GENOMEANNOTATOR in filepath ./git_repos\\genomeannotator\\main.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\genomeannotator\\main.nf\n",
      "Process ASSEMBLY_PREPROCESS in filepath ./git_repos\\genomeannotator\\subworkflows\\local\\assembly_preprocess.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\genomeannotator\\subworkflows\\local\\assembly_preprocess.nf\n",
      "Process AUGUSTUS_PIPELINE in filepath ./git_repos\\genomeannotator\\subworkflows\\local\\augustus_pipeline.nf\n",
      "Nextflow operator collect with content: hints.collect(),\n",
      "Nextflow operator collect with content: aug_config_folder.collect().map{ it[0].toString() },\n",
      "Nextflow operator collect with content: aug_extrinsic_cfg.collect(),\n",
      "Nextflow operator multiMap with content: .multiMap { m,gff ->\n",
      "Nextflow operator set with content: }.set { ch_gffs }\n",
      "Nextflow operator collectFile with content: ch_gffs.gffs.collectFile { mkey, file -> [ \"${mkey}.augustus.gff\", file ] }\n",
      "Nextflow operator map with content: .map { file -> [ file.simpleName, file ] }\n",
      "Nextflow operator set with content: .set { ch_merged_gffs }\n",
      "Nextflow operator join with content: ch_gffs.metadata.join(\n",
      "Nextflow operator map with content: .map { k,m,f -> tuple(m,f) }\n",
      "Nextflow operator set with content: .set { ch_genome_gff }\n",
      "Nextflow operator join with content: AUGUSTUS_CREATEGFFIDS.out.gff.join(genome)\n",
      "Error: Unclosed process block in file: ./git_repos\\genomeannotator\\subworkflows\\local\\augustus_pipeline.nf\n",
      "Process BUSCO_QC in filepath ./git_repos\\genomeannotator\\subworkflows\\local\\busco_qc.nf\n",
      "Nextflow operator collect with content: ch_lineage_dir.collect()\n",
      "Error: Unclosed process block in file: ./git_repos\\genomeannotator\\subworkflows\\local\\busco_qc.nf\n",
      "Process EVM in filepath ./git_repos\\genomeannotator\\subworkflows\\local\\evm.nf\n",
      "Nextflow operator splitText with content: EVIDENCEMODELER_PARTITION.out.commands.splitText(by: params.nevm, file: true)\n",
      "Nextflow operator groupTuple with content: EVIDENCEMODELER_EXECUTE.out.log.groupTuple(by: [0]),\n",
      "Nextflow operator collect with content: genome.collect()\n",
      "Nextflow operator join with content: HELPER_EVM2GFF.out.gff.join(genome)\n",
      "Error: Unclosed process block in file: ./git_repos\\genomeannotator\\subworkflows\\local\\evm.nf\n",
      "Process FASTA_PREPROCESS in filepath ./git_repos\\genomeannotator\\subworkflows\\local\\fasta_preprocess.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\genomeannotator\\subworkflows\\local\\fasta_preprocess.nf\n",
      "Process GENOME_ALIGN in filepath ./git_repos\\genomeannotator\\subworkflows\\local\\genome_align.nf\n",
      "Nextflow operator splitCsv with content: .splitCsv ( header:true, sep:',' )\n",
      "Nextflow operator map with content: .map { create_target_channel(it) }\n",
      "Nextflow operator set with content: .set { targets }\n",
      "Nextflow operator map with content: targets.map { m,f,g ->\n",
      "Nextflow operator join with content: .join(\n",
      "Nextflow operator map with content: targets.map { m,f,g ->\n",
      "Nextflow operator set with content: .set { targets_clean }\n",
      "Nextflow operator flatMap with content: FASTASPLITTER.out.chunks.flatMap{ row ->\n",
      "Nextflow operator collect with content: row[1..-1].collect { [row[0].clone(), it]  }\n",
      "Nextflow operator set with content: }.set {genome_chunks}\n",
      "Nextflow operator combine with content: genome_chunks.combine(targets_clean)\n",
      "Nextflow operator groupTuple with content: grouped_chains = SATSUMA2_SATSUMASYNTENY2.out.chain.groupTuple(by: [0,1,2,3])\n",
      "Error: Unclosed process block in file: ./git_repos\\genomeannotator\\subworkflows\\local\\genome_align.nf\n",
      "Process INPUT_CHECK in filepath ./git_repos\\genomeannotator\\subworkflows\\local\\input_check.nf\n",
      "Nextflow operator splitCsv with content: .splitCsv ( header:true, sep:',' )\n",
      "Nextflow operator map with content: .map { create_fastq_channel(it) }\n",
      "Nextflow operator set with content: .set { reads }\n",
      "Error: Unclosed process block in file: ./git_repos\\genomeannotator\\subworkflows\\local\\input_check.nf\n",
      "Process MINIMAP_ALIGN_TRANSCRIPTS in filepath ./git_repos\\genomeannotator\\subworkflows\\local\\minimap_align_transcripts.nf\n",
      "Nextflow operator splitFasta with content: EXONERATE_FASTACLEAN.out.fasta.splitFasta(by: 100000, file: true),\n",
      "Nextflow operator collect with content: genome.collect(),\n",
      "Nextflow operator groupTuple with content: .groupTuple()\n",
      "Nextflow operator branch with content: .branch {\n",
      "Nextflow operator set with content: .set { ch_bams }\n",
      "Nextflow operator mix with content: SAMTOOLS_MERGE.out.bam.mix(ch_bams.single)\n",
      "Error: Unclosed process block in file: ./git_repos\\genomeannotator\\subworkflows\\local\\minimap_align_transcripts.nf\n",
      "Process NCRNA in filepath ./git_repos\\genomeannotator\\subworkflows\\local\\ncrna.nf\n",
      "Nextflow operator map with content: GUNZIP_RFAM_CM.out.gunzip.map {m,f -> f}\n",
      "Nextflow operator collect with content: INFERNAL_PRESS.out.cm.collect()\n",
      "Nextflow operator multiMap with content: .multiMap { m,t ->\n",
      "Nextflow operator set with content: }.set { ch_rfam_tbls }\n",
      "Nextflow operator collectFile with content: ch_rfam_tbls.tbl.collectFile { mkey, file -> [ \"${mkey}.rfam.gff\", file ] }\n",
      "Nextflow operator map with content: .map { file -> [ file.simpleName, file ] }\n",
      "Nextflow operator set with content: .set { ch_merged_tbls }\n",
      "Nextflow operator join with content: ch_rfam_tbls.metadata.join(\n",
      "Nextflow operator map with content: .map { k,m,f -> tuple(m,f) }\n",
      "Nextflow operator set with content: .set { ch_rfam_gff }\n",
      "Nextflow operator map with content: GUNZIP_RFAM_FAMILY.out.gunzip.map{m,f -> f}\n",
      "Error: Unclosed process block in file: ./git_repos\\genomeannotator\\subworkflows\\local\\ncrna.nf\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Process PASA_PIPELINE in filepath ./git_repos\\genomeannotator\\subworkflows\\local\\pasa_pipeline.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\genomeannotator\\subworkflows\\local\\pasa_pipeline.nf\n",
      "Process REPEATMASKER in filepath ./git_repos\\genomeannotator\\subworkflows\\local\\repeatmasker.nf\n",
      "Nextflow operator map with content: GUNZIP.out.gunzip.map {m,g -> g}\n",
      "Nextflow operator collect with content: REPEATMASKER_STAGELIB.out.library.collect().map{it[0].toString()},\n",
      "Nextflow operator collect with content: rm_lib.collect(),\n",
      "Nextflow operator mix with content: versions = REPEATMASKER_STAGELIB.out.versions.mix(REPEATMASKER_REPEATMASK.out.versions,FASTASPLITTER.out.versions,REPEATMASKER_CAT_FASTA.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\genomeannotator\\subworkflows\\local\\repeatmasker.nf\n",
      "Process RNASEQ_ALIGN in filepath ./git_repos\\genomeannotator\\subworkflows\\local\\rnaseq_align.nf\n",
      "Nextflow operator splitCsv with content: .splitCsv ( header:true, sep:',' )\n",
      "Nextflow operator map with content: .map { create_fastq_channel(it) }\n",
      "Nextflow operator set with content: .set { reads }\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator join with content: meta.id = (meta.id.contains(\"SRR\")) ? meta.id : meta.id.split('_')[0..-2].join('_')\n",
      "Nextflow operator groupTuple with content: .groupTuple(by: [0])\n",
      "Nextflow operator branch with content: .branch {\n",
      "Nextflow operator flatten with content: return [ meta, fastq.flatten() ]\n",
      "Nextflow operator flatten with content: return [ meta, fastq.flatten() ]\n",
      "Nextflow operator set with content: .set { ch_fastq }\n",
      "Nextflow operator mix with content: .mix( ch_fastq.single )\n",
      "Nextflow operator set with content: .set { ch_cat_fastq }\n",
      "Nextflow operator collect with content: STAR_INDEX.out.star_index.collect(),\n",
      "Nextflow operator collect with content: Channel.from(params.dummy_gff).collect(),\n",
      "Nextflow operator collectFile with content: junctions = STAR_ALIGN_PASS_ONE.out.junctions.collectFile(name: 'all_juncs.gtf')\n",
      "Nextflow operator collect with content: STAR_INDEX.out.star_index.collect(),\n",
      "Nextflow operator collect with content: junctions.collect(),\n",
      "Nextflow operator mix with content: versions = STAR_INDEX.out.versions.mix(STAR_ALIGN_PASS_ONE.out.versions,FASTP.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\genomeannotator\\subworkflows\\local\\rnaseq_align.nf\n",
      "Process SPALN_ALIGN_PROTEIN in filepath ./git_repos\\genomeannotator\\subworkflows\\local\\spaln_align_protein.nf\n",
      "Nextflow operator splitFasta with content: ch_fasta_chunks = GAAS_FASTAFILTERBYSIZE.out.fasta.splitFasta(by: params.nproteins, file: true, elem: [1])\n",
      "Nextflow operator flatten with content: SPALN_ALIGN.out.align.groupTuple().map { m,files -> tuple(m,files.flatten()) },\n",
      "Error: Unclosed process block in file: ./git_repos\\genomeannotator\\subworkflows\\local\\spaln_align_protein.nf\n",
      "Process GENOMEANNOTATOR in filepath ./git_repos\\genomeannotator\\workflows\\genomeannotator.nf\n",
      "Nextflow operator mix with content: ch_transcripts = ch_transcripts.mix(TRANSCRIPT_PREPROCESS.out.fasta)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ASSEMBLY_PREPROCESS.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GENOME_ALIGN.out.versions)\n",
      "Nextflow operator mix with content: ch_hints = ch_hints.mix(GENOME_ALIGN.out.hints)\n",
      "Nextflow operator mix with content: ch_genes_gff = ch_genes_gff.mix(GENOME_ALIGN.out.gff)\n",
      "Nextflow operator map with content: ch_repeats = REPEATMODELER.out.fasta.map {m,fasta -> fasta}\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(REPEATMASKER.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(REPEATMASKER.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SPALN_ALIGN_PROTEIN.out.versions)\n",
      "Nextflow operator mix with content: ch_hints = ch_hints.mix(SPALN_ALIGN_PROTEIN.out.hints)\n",
      "Nextflow operator mix with content: ch_proteins_gff = ch_proteins_gff.mix(SPALN_ALIGN_PROTEIN.out.evm)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SPALN_ALIGN_MODELS.out.versions)\n",
      "Nextflow operator mix with content: ch_hints = ch_hints.mix(SPALN_ALIGN_MODELS.out.hints)\n",
      "Nextflow operator mix with content: ch_genes_gff = ch_genes_gff.mix(SPALN_ALIGN_MODELS.out.gff)\n",
      "Nextflow operator collect with content: ASSEMBLY_PREPROCESS.out.fasta.collect(),\n",
      "Nextflow operator map with content: RNASEQ_ALIGN.out.bam.map{ meta, bam ->\n",
      "Nextflow operator groupTuple with content: }.groupTuple(by:[0])\n",
      "Nextflow operator set with content: .set{bam_mapped}\n",
      "Nextflow operator mix with content: ch_hints = ch_hints.mix(AUGUSTUS_BAM2HINTS.out.gff)\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(RNASEQ_ALIGN.out.versions.first(),AUGUSTUS_BAM2HINTS.out.versions,SAMTOOLS_MERGE.out.versions)\n",
      "Nextflow operator mix with content: ch_transcripts = ch_transcripts.mix(TRINITY_GENOMEGUIDED.out.fasta)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(TRINITY_GENOMEGUIDED.out.versions)\n",
      "Nextflow operator collect with content: ASSEMBLY_PREPROCESS.out.fasta.collect(),\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(MINIMAP_ALIGN_TRANSCRIPTS.out.versions)\n",
      "Nextflow operator mix with content: ch_transcripts_gff = ch_transcripts_gff.mix(MINIMAP_ALIGN_TRANSCRIPTS.out.gff)\n",
      "Nextflow operator mix with content: ch_hints = ch_hints.mix(MINIMAP_ALIGN_TRANSCRIPTS.out.hints)\n",
      "Nextflow operator collectFile with content: ch_transcripts.map { m,t -> t }.collectFile(name: \"transcripts.merged.fa\").map { it ->\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(PASA_PIPELINE.out.versions)\n",
      "Nextflow operator mix with content: ch_genes_gff = ch_genes_gff.mix(PASA_PIPELINE.out.gff)\n",
      "Nextflow operator collectFile with content: all_hints = ch_hints.unique().collectFile(name: 'hints.gff')\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(AUGUSTUS_PIPELINE.out.versions)\n",
      "Nextflow operator mix with content: ch_genes_gff = ch_genes_gff.mix(AUGUSTUS_PIPELINE.out.gff)\n",
      "Nextflow operator mix with content: ch_proteins_fa = ch_proteins_fa.mix(AUGUSTUS_PIPELINE.out.proteins)\n",
      "Nextflow operator collectFile with content: ch_genes_gff.map{m,g -> g}.collectFile(name: 'genes.gff3'),\n",
      "Nextflow operator collectFile with content: ch_proteins_gff.map{m,p -> p}.mix(ch_empty_gff).collectFile(name: 'proteins.gff3'),\n",
      "Nextflow operator collectFile with content: ch_transcripts_gff.map{m,t ->t}.mix(ch_empty_gff).collectFile(name: 'transcripts.gff3'),\n",
      "Nextflow operator mix with content: ch_proteins_fa = ch_proteins_fa.mix(EVM.out.proteins)\n",
      "Nextflow operator collectFile with content: ch_versions.unique().collectFile(name: 'collated_versions.yml')\n",
      "Nextflow operator mix with content: ch_multiqc_files = ch_multiqc_files.mix(Channel.from(ch_multiqc_config))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(ch_multiqc_custom_config.collect().ifEmpty([]))\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_workflow_summary.collectFile(name: 'workflow_summary_mqc.yaml'))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(CUSTOM_DUMPSOFTWAREVERSIONS.out.mqc_yml.collect())\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(ch_busco_qc.collect().ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files.collect()\n",
      "Nextflow operator toList with content: multiqc_report = MULTIQC.out.report.toList()\n",
      "Nextflow operator mix with content: ch_versions    = ch_versions.mix(MULTIQC.out.versions)\n",
      "Process NFCORE_GENOMEASSEMBLER in filepath ./git_repos\\genomeassembler\\main.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\genomeassembler\\main.nf\n",
      "Process ASSEMBLY_COMPARISON in filepath ./git_repos\\genomeassembler\\subworkflows\\local\\assembly_comparison.nf\n",
      "Nextflow operator map with content: assembly_ch.map { sample, assembly -> assembly.pri_asm }\n",
      "Nextflow operator collect with content: .collect(),\n",
      "Error: Unclosed process block in file: ./git_repos\\genomeassembler\\subworkflows\\local\\assembly_comparison.nf\n",
      "Process BUILD_FASTK_DATABASES in filepath ./git_repos\\genomeassembler\\subworkflows\\local\\build_fastk_database.nf\n",
      "Nextflow operator groupTuple with content: fkdb_ch = FASTK_FASTK.out.hist.groupTuple()\n",
      "Nextflow operator groupTuple with content: .join( FASTK_FASTK.out.ktab.groupTuple(), remainder: true )\n",
      "Nextflow operator groupTuple with content: .join( FASTK_FASTK.out.prof.groupTuple(), remainder: true )\n",
      "Nextflow operator flatten with content: .map { meta, hist, ktab, prof -> [meta, hist, ktab ? ktab.flatten() : [] , prof ? prof.flatten() : [] ] }\n",
      "Nextflow operator branch with content: .branch { meta, hist, ktab, prof ->\n",
      "Nextflow operator multiMap with content: fk_single = fkdb_ch.single_hist.multiMap { meta, hist, ktab, prof ->\n",
      "Nextflow operator first with content: versions_ch = FASTK_FASTK.out.versions.first().mix( FASTK_MERGE.out.versions.first() )\n",
      "Nextflow operator mix with content: histogram = fk_single.hist.mix( FASTK_MERGE.out.hist )\n",
      "Nextflow operator mix with content: ktab      = fk_single.ktab.mix( FASTK_MERGE.out.ktab )\n",
      "Nextflow operator mix with content: prof      = fk_single.prof.mix( FASTK_MERGE.out.prof )\n",
      "Error: Unclosed process block in file: ./git_repos\\genomeassembler\\subworkflows\\local\\build_fastk_database.nf\n",
      "Process BUILD_MERYL_DATABASES in filepath ./git_repos\\genomeassembler\\subworkflows\\local\\build_meryl_database.nf\n",
      "Nextflow operator groupTuple with content: MERYL_UNIONSUM ( MERYL_COUNT.out.meryl_db.groupTuple()\n",
      "Nextflow operator flatten with content: .map { meta, meryldbs -> [ meta, meryldbs.flatten() ] } // Meryl count can process pairs of files so need to flatten\n",
      "Nextflow operator first with content: versions_ch = MERYL_COUNT.out.versions.first().mix(\n",
      "Nextflow operator first with content: MERYL_UNIONSUM.out.versions.first(),\n",
      "Nextflow operator first with content: MERYL_HISTOGRAM.out.versions.first()\n",
      "Error: Unclosed process block in file: ./git_repos\\genomeassembler\\subworkflows\\local\\build_meryl_database.nf\n",
      "Process CONTAMINATION_SCREEN in filepath ./git_repos\\genomeassembler\\subworkflows\\local\\contamination_screen.nf\n",
      "Nextflow operator transpose with content: MASH_SCREEN ( ch_reads.transpose(), mash_screen_db )\n",
      "Nextflow operator groupTuple with content: MASH_FILTER ( MASH_SCREEN.out.screen.groupTuple() )\n",
      "Nextflow operator first with content: versions = MASH_SCREEN.out.versions.first()\n",
      "Error: Unclosed process block in file: ./git_repos\\genomeassembler\\subworkflows\\local\\contamination_screen.nf\n",
      "Process EVALUATE_GENE_SPACE in filepath ./git_repos\\genomeassembler\\subworkflows\\local\\evaluate_gene_space.nf\n",
      "Nextflow operator map with content: assembly_ch.map { meta, assembly -> meta.busco_lineages ? [ meta, assembly ] : [ meta + [ busco_lineages: params.busco_lineages ], assembly ] }\n",
      "Nextflow operator flatMap with content: .flatMap { meta, assembly ->\n",
      "Nextflow operator collect with content: meta.busco_lineages.collect { [ meta, it, assembly.pri_asm ] } :\n",
      "Nextflow operator multiMap with content: .multiMap { meta, busco_lineage, assembly ->\n",
      "Nextflow operator first with content: versions_ch = BUSCO.out.versions.first()\n",
      "Error: Unclosed process block in file: ./git_repos\\genomeassembler\\subworkflows\\local\\evaluate_gene_space.nf\n",
      "Process FASTK_GENOME_PROPERTIES in filepath ./git_repos\\genomeassembler\\subworkflows\\local\\fastk_genome_properties.nf\n",
      "Nextflow operator map with content: FASTK_HISTEX ( fastk_hist_ktab.map { meta, hist, ktab -> [ meta, hist ] } )\n",
      "Nextflow operator first with content: versions_ch = FASTK_HISTEX.out.versions.first().mix(\n",
      "Nextflow operator first with content: GENESCOPEFK.out.versions.first(),\n",
      "Nextflow operator first with content: MERQURYFK_PLOIDYPLOT.out.versions.first(),\n",
      "Nextflow operator first with content: MERQURYFK_KATGC.out.versions.first()\n",
      "Error: Unclosed process block in file: ./git_repos\\genomeassembler\\subworkflows\\local\\fastk_genome_properties.nf\n",
      "Process FASTK_KMER_COMPLETENESS in filepath ./git_repos\\genomeassembler\\subworkflows\\local\\fastk_kmer_completeness.nf\n",
      "Nextflow operator map with content: fastk_db.map{ meta, hist, ktab -> [ meta.findAll{ !(it in ['single_end']) }, hist, ktab ] }\n",
      "Nextflow operator combine with content: .combine( assembly_ch.map { meta, assembly ->\n",
      "Nextflow operator map with content: .map { meta, hist, ktab, assembly, buildid -> [ meta + [ build: buildid ], hist, ktab, assembly, [] ] }\n",
      "Nextflow operator first with content: versions_ch = MERQURYFK_MERQURYFK.out.versions.first()\n",
      "Error: Unclosed process block in file: ./git_repos\\genomeassembler\\subworkflows\\local\\fastk_kmer_completeness.nf\n",
      "Process HIC_DATA_PROPERTIES in filepath ./git_repos\\genomeassembler\\subworkflows\\local\\hic_data_properties.nf\n",
      "Nextflow operator map with content: logs_ch = logs_ch.mix( FASTQC.out.zip.map{ meta, zip -> zip } )\n",
      "Nextflow operator first with content: versions_ch = versions_ch.mix( FASTQC.out.versions.first() )\n",
      "Error: Unclosed process block in file: ./git_repos\\genomeassembler\\subworkflows\\local\\hic_data_properties.nf\n",
      "Process HIFI_DATA_PROPERTIES in filepath ./git_repos\\genomeassembler\\subworkflows\\local\\hifi_data_properties.nf\n",
      "Nextflow operator map with content: logs_ch = logs_ch.mix( NANOPLOT.out.log.map{ meta, log -> log } )\n",
      "Nextflow operator first with content: versions_ch = versions_ch.mix( NANOPLOT.out.versions.first() )\n",
      "Error: Unclosed process block in file: ./git_repos\\genomeassembler\\subworkflows\\local\\hifi_data_properties.nf\n",
      "Process ILLUMINA_DATA_PROPERTIES in filepath ./git_repos\\genomeassembler\\subworkflows\\local\\illumina_data_properties.nf\n",
      "Nextflow operator map with content: logs_ch = logs_ch.mix( FASTQC.out.zip.map{ meta, zip -> zip } )\n",
      "Nextflow operator first with content: versions_ch = versions_ch.mix( FASTQC.out.versions.first() )\n",
      "Error: Unclosed process block in file: ./git_repos\\genomeassembler\\subworkflows\\local\\illumina_data_properties.nf\n",
      "Process INPUT_CHECK in filepath ./git_repos\\genomeassembler\\subworkflows\\local\\input_check.nf\n",
      "Nextflow operator splitCsv with content: .splitCsv ( header:true, sep:',' )\n",
      "Nextflow operator map with content: .map { create_fastq_channel(it) }\n",
      "Nextflow operator set with content: .set { reads }\n",
      "Error: Unclosed process block in file: ./git_repos\\genomeassembler\\subworkflows\\local\\input_check.nf\n",
      "Process MERYL_GENOME_PROPERTIES in filepath ./git_repos\\genomeassembler\\subworkflows\\local\\meryl_genome_properties.nf\n",
      "Nextflow operator first with content: versions_ch = GENOMESCOPE2.out.versions.first()\n",
      "Error: Unclosed process block in file: ./git_repos\\genomeassembler\\subworkflows\\local\\meryl_genome_properties.nf\n",
      "Process MERYL_KMER_COMPLETENESS in filepath ./git_repos\\genomeassembler\\subworkflows\\local\\meryl_kmer_completeness.nf\n",
      "Nextflow operator map with content: meryl_db.map{ meta, meryldb -> [ meta.findAll{ !(it in ['single_end']) }, meryldb ] }\n",
      "Nextflow operator combine with content: .combine( assembly_ch.map { meta, assembly ->\n",
      "Nextflow operator map with content: .map { meta, meryldb, assembly, buildid -> [ meta + [ build: buildid ], meryldb, assembly ] }\n",
      "Nextflow operator first with content: versions_ch = MERQURY.out.versions.first()\n",
      "Error: Unclosed process block in file: ./git_repos\\genomeassembler\\subworkflows\\local\\meryl_kmer_completeness.nf\n",
      "Process ONT_DATA_PROPERTIES in filepath ./git_repos\\genomeassembler\\subworkflows\\local\\ont_data_properties.nf\n",
      "Nextflow operator map with content: logs_ch = logs_ch.mix( NANOPLOT.out.log.map{ meta, log -> log } )\n",
      "Nextflow operator first with content: versions_ch = versions_ch.mix( NANOPLOT.out.versions.first() )\n",
      "Error: Unclosed process block in file: ./git_repos\\genomeassembler\\subworkflows\\local\\ont_data_properties.nf\n",
      "Process PREPARE_INPUT in filepath ./git_repos\\genomeassembler\\subworkflows\\local\\prepare_input.nf\n",
      "Nextflow operator branch with content: ch_input.branch { file ->\n",
      "Nextflow operator set with content: }.set { input }\n",
      "Nextflow operator splitCsv with content: input.csv_ch.splitCsv( header: ['sample_id', 'datatype', 'read1', 'read2'], skip: 1, sep: ',' )\n",
      "Nextflow operator set with content: .set { csv_records }\n",
      "Nextflow operator splitCsv with content: input.tsv_ch.splitCsv( header: ['sample_id', 'datatype', 'read1', 'read2'], skip: 1, sep: '\\t' )\n",
      "Nextflow operator mix with content: .mix( csv_records )\n",
      "Nextflow operator branch with content: .branch { record -> def seqs = record.read2 ? [ file( record.read1, checkIfExists: true ), file( record.read2, checkIfExists: true ) ] : file( record.read1, checkIfExists: true )\n",
      "Nextflow operator set with content: }.set { tsv_input }\n",
      "Nextflow operator map with content: .map { file -> readYAML( file ) }\n",
      "Nextflow operator multiMap with content: .multiMap { file ->\n",
      "Nextflow operator set with content: .set { ymlfile }\n",
      "Nextflow operator flatten with content: .flatten()\n",
      "Nextflow operator dump with content: .dump( tag: 'YAML Samples' )\n",
      "Nextflow operator multiMap with content: .multiMap { data ->\n",
      "Nextflow operator collect with content: hic_ch      : ( data.hic      ? [ data.metadata + [ single_end: false ], data.hic.collect { [ file( it.read1, checkIfExists: true ), file( it.read2, checkIfExists: true ) ] } ] : [] )\n",
      "Nextflow operator collect with content: hifi_ch     : ( data.hifi     ? [ data.metadata + [ single_end: true ], data.hifi.collect { file( it.reads, checkIfExists: true ) } ] : [] )\n",
      "Nextflow operator collect with content: ont_ch      : ( data.ont      ? [ data.metadata + [ single_end: true ], data.ont.collect { file( it.reads, checkIfExists: true ) } ] : [] )\n",
      "Nextflow operator collect with content: illumina_ch : ( data.illumina ? [ data.metadata, data.illumina.collect { it.reads ? file( it.reads, checkIfExists: true ) : [ file( it.read1, checkIfExists: true ), file( it.read2, checkIfExists: true ) ] } ] : [] )\n",
      "Nextflow operator collect with content: rnaseq_ch   : ( data.rnaseq   ? [ data.metadata, data.rnaseq.collect { it.reads ? file( it.reads, checkIfExists: true ) : [ file( it.read1, checkIfExists: true ), file( it.read2, checkIfExists: true ) ] } ] : [] )\n",
      "Nextflow operator collect with content: isoseq_ch   : ( data.isoseq   ? [ data.metadata + [ single_end: true ], data.isoseq.collect { file( it.reads, checkIfExists: true ) } ] : [] )\n",
      "Nextflow operator set with content: .set{ yml_input }\n",
      "Nextflow operator filter with content: .filter { !it.isEmpty() }\n",
      "Nextflow operator transpose with content: .transpose()     // Data is [ sample, [ id:'assemblerX_build1', pri_asm: '/path/to/primary_asm', alt_asm: '/path/to/alternate_asm' ]]\n",
      "Nextflow operator map with content: .map { sample, assembly ->\n",
      "Nextflow operator set with content: .set { assembly_ch }\n",
      "Nextflow operator filter with content: .filter { !it.isEmpty() }\n",
      "Nextflow operator mix with content: .mix( tsv_input.hifi_ch )\n",
      "Nextflow operator transpose with content: .transpose()   // Transform to [ [ id: 'sample_name', single_end: true ], file('/path/to/read')  ]\n",
      "Nextflow operator branch with content: .branch { meta, filename ->\n",
      "Nextflow operator set with content: }.set { hifi }\n",
      "Nextflow operator mix with content: hifi.fastx_ch.mix( SAMTOOLS_FASTQ.out.fastq )\n",
      "Nextflow operator set with content: .set { hifi_fastx_ch }\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SAMTOOLS_FASTQ.out.versions.first())\n",
      "Nextflow operator filter with content: yml_input.hic_ch.filter { !it.isEmpty() }\n",
      "Nextflow operator transpose with content: .transpose()\n",
      "Nextflow operator mix with content: .mix( tsv_input.hic_ch )\n",
      "Nextflow operator set with content: .set { hic_fastx_ch }\n",
      "Nextflow operator filter with content: yml_input.ont_ch.filter { !it.isEmpty() }\n",
      "Nextflow operator transpose with content: .transpose()\n",
      "Nextflow operator mix with content: .mix( tsv_input.ont_ch )\n",
      "Nextflow operator set with content: .set { ont_fastx_ch }\n",
      "Nextflow operator filter with content: yml_input.illumina_ch.filter { !it.isEmpty() }\n",
      "Nextflow operator transpose with content: .transpose()\n",
      "Nextflow operator map with content: .map { meta, reads -> [ meta + [ single_end: reads instanceof Path ], reads ] }\n",
      "Nextflow operator mix with content: .mix( tsv_input.illumina_ch )\n",
      "Nextflow operator set with content: .set { illumina_fastx_ch }\n",
      "Nextflow operator filter with content: yml_input.rnaseq_ch.filter { !it.isEmpty() }\n",
      "Nextflow operator transpose with content: .transpose()\n",
      "Nextflow operator map with content: .map { meta, reads -> [ meta + [ single_end: reads instanceof Path ], reads ] }\n",
      "Nextflow operator mix with content: .mix( tsv_input.rnaseq_ch )\n",
      "Nextflow operator set with content: .set { rnaseq_fastx_ch }\n",
      "Nextflow operator filter with content: yml_input.isoseq_ch.filter { !it.isEmpty() }\n",
      "Nextflow operator transpose with content: .transpose()\n",
      "Nextflow operator mix with content: .mix( tsv_input.isoseq_ch )\n",
      "Nextflow operator set with content: .set { isoseq_fastx_ch }\n",
      "Nextflow operator dump with content: assemblies = assembly_ch.dump( tag: 'Input: Assemblies' )\n",
      "Nextflow operator dump with content: hic        = hic_fastx_ch.dump( tag: 'Input: Hi-C' )\n",
      "Nextflow operator dump with content: hifi       = hifi_fastx_ch.dump( tag: 'Input: PacBio HiFi' )\n",
      "Nextflow operator dump with content: ont        = ont_fastx_ch.dump( tag: 'Input: ONT' )\n",
      "Nextflow operator dump with content: illumina   = illumina_fastx_ch.dump( tag: 'Input: Illumina' )\n",
      "Nextflow operator dump with content: rnaseq     = rnaseq_fastx_ch.dump( tag: 'Input: Illumina RnaSeq' )\n",
      "Nextflow operator dump with content: isoseq     = isoseq_fastx_ch.dump( tag: 'Input: PacBio IsoSeq' )\n",
      "Error: Unclosed process block in file: ./git_repos\\genomeassembler\\subworkflows\\local\\prepare_input.nf\n",
      "Process GENOMEASSEMBLER in filepath ./git_repos\\genomeassembler\\workflows\\genomeassembler.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix ( PREPARE_INPUT.out.versions )\n",
      "Process NFCORE_GENOMESKIM in filepath ./git_repos\\genomeskim\\main.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\genomeskim\\main.nf\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Process INPUT_CHECK in filepath ./git_repos\\genomeskim\\subworkflows\\local\\input_check.nf\n",
      "Nextflow operator splitCsv with content: .splitCsv ( header:true, sep:',' )\n",
      "Nextflow operator map with content: .map { create_fastq_channel(it) }\n",
      "Nextflow operator set with content: .set { reads }\n",
      "Error: Unclosed process block in file: ./git_repos\\genomeskim\\subworkflows\\local\\input_check.nf\n",
      "Process GENOMESKIM in filepath ./git_repos\\genomeskim\\workflows\\genomeskim.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(INPUT_CHECK.out.versions)\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(FASTQC.out.versions.first())\n",
      "Nextflow operator collectFile with content: ch_versions.unique().collectFile(name: 'collated_versions.yml')\n",
      "Nextflow operator mix with content: ch_multiqc_files = ch_multiqc_files.mix(Channel.from(ch_multiqc_config))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(ch_multiqc_custom_config.collect().ifEmpty([]))\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_workflow_summary.collectFile(name: 'workflow_summary_mqc.yaml'))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(CUSTOM_DUMPSOFTWAREVERSIONS.out.mqc_yml.collect())\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(FASTQC.out.zip.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files.collect()\n",
      "Nextflow operator toList with content: multiqc_report = MULTIQC.out.report.toList()\n",
      "Nextflow operator mix with content: ch_versions    = ch_versions.mix(MULTIQC.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\genomeskim\\workflows\\genomeskim.nf\n",
      "Process NFCORE_GWAS in filepath ./git_repos\\gwas\\main.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\gwas\\main.nf\n",
      "Process INPUT_CHECK in filepath ./git_repos\\gwas\\subworkflows\\local\\input_check.nf\n",
      "Nextflow operator splitCsv with content: .splitCsv ( header:true, sep:',' )\n",
      "Nextflow operator map with content: .map { create_fastq_channel(it) }\n",
      "Nextflow operator set with content: .set { reads }\n",
      "Error: Unclosed process block in file: ./git_repos\\gwas\\subworkflows\\local\\input_check.nf\n",
      "Process QC_INPUT_VALIDATION in filepath ./git_repos\\gwas\\subworkflows\\local\\qc.nf\n",
      "Nextflow operator ifEmpty with content: .ifEmpty { error \"No matching plink files\" }        \\\n",
      "Nextflow operator multiMap with content: .multiMap {  it ->\n",
      "Nextflow operator set with content: }.set {checked_input}\n",
      "Process QC_PROCESSES in filepath ./git_repos\\gwas\\subworkflows\\local\\qc.nf\n",
      "Process QC_WF in filepath ./git_repos\\gwas\\subworkflows\\local\\qc.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\gwas\\subworkflows\\local\\qc.nf\n",
      "Process GWAS in filepath ./git_repos\\gwas\\workflows\\gwas.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(INPUT_CHECK.out.versions)\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(FASTQC.out.versions.first())\n",
      "Nextflow operator collectFile with content: ch_versions.unique().collectFile(name: 'collated_versions.yml')\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_workflow_summary.collectFile(name: 'workflow_summary_mqc.yaml'))\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_methods_description.collectFile(name: 'methods_description_mqc.yaml'))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(CUSTOM_DUMPSOFTWAREVERSIONS.out.mqc_yml.collect())\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(FASTQC.out.zip.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files.collect(),\n",
      "Nextflow operator toList with content: ch_multiqc_config.toList(),\n",
      "Nextflow operator toList with content: ch_multiqc_custom_config.toList(),\n",
      "Nextflow operator toList with content: ch_multiqc_logo.toList()\n",
      "Nextflow operator toList with content: multiqc_report = MULTIQC.out.report.toList()\n",
      "Nextflow operator mix with content: ch_versions    = ch_versions.mix(MULTIQC.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\gwas\\workflows\\gwas.nf\n",
      "Process NFCORE_HGTSEQ in filepath ./git_repos\\hgtseq\\main.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\hgtseq\\main.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\hgtseq\\conf\\test_full.config\n",
      "Process BAM_QC in filepath ./git_repos\\hgtseq\\subworkflows\\local\\bam_qc\\main.nf\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SAMTOOLS_STATS.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SAMTOOLS_FLAGSTAT.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SAMTOOLS_IDXSTATS.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(QUALIMAP_BAMQC.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(BAMTOOLS_STATS.out.versions.first())\n",
      "Error: Unclosed process block in file: ./git_repos\\hgtseq\\subworkflows\\local\\bam_qc\\main.nf\n",
      "Process CLASSIFY_UNMAPPED in filepath ./git_repos\\hgtseq\\subworkflows\\local\\classify_unmapped\\main.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_VIEW_SINGLE.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_VIEW_BOTH.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(PARSEOUTPUTS.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_FASTQ_SINGLE.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_FASTQ_BOTH.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(KRAKEN2_SINGLE.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(KRAKEN2_BOTH.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\hgtseq\\subworkflows\\local\\classify_unmapped\\main.nf\n",
      "Process PREPARE_READS in filepath ./git_repos\\hgtseq\\subworkflows\\local\\prepare_reads\\main.nf\n",
      "Nextflow operator map with content: fasta_meta = Channel.value(file(fasta)).map{ it -> [[id:it[0].baseName], it] }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(TRIMGALORE.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BWAMEM1_INDEX.out.versions)\n",
      "Nextflow operator collect with content: bwaindex      = params.fasta ? params.bwaindex      ? Channel.fromPath(params.bwaindex).collect().map{ it -> [[id:it[0].baseName], it] } : BWAMEM1_INDEX.out.index : []\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BWAMEM1_MEM.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BWAMEM2_INDEX.out.versions)\n",
      "Nextflow operator collect with content: bwamem2index  = params.fasta ? params.bwamem2index  ? Channel.fromPath(params.bwamem2index).collect()  : BWAMEM2_INDEX.out.index : []\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BWAMEM2_MEM.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\hgtseq\\subworkflows\\local\\prepare_reads\\main.nf\n",
      "Process READS_QC in filepath ./git_repos\\hgtseq\\subworkflows\\local\\reads_qc\\main.nf\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(FASTQC_UNTRIMMED.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(FASTQC_TRIMMED.out.versions.first())\n",
      "Error: Unclosed process block in file: ./git_repos\\hgtseq\\subworkflows\\local\\reads_qc\\main.nf\n",
      "Process REPORTING in filepath ./git_repos\\hgtseq\\subworkflows\\local\\reporting\\main.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GAWK_SINGLE.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GAWK_BOTH.out.versions)\n",
      "Nextflow operator combine with content: single_input = fakemeta.combine(GAWK_SINGLE.out.collated_reads)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(KRONA_KTIMPORTTAXONOMY_SINGLE.out.versions)\n",
      "Nextflow operator combine with content: both_input = fakemeta.combine(GAWK_BOTH.out.collated_reads)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(KRONA_KTIMPORTTAXONOMY_BOTH.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\hgtseq\\subworkflows\\local\\reporting\\main.nf\n",
      "Process SORTBAM in filepath ./git_repos\\hgtseq\\subworkflows\\local\\sortbam\\main.nf\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SAMTOOLS_SORT.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SAMTOOLS_INDEX.out.versions.first())\n",
      "Nextflow operator join with content: .join(SAMTOOLS_INDEX.out.bai, by: [0], remainder: true)\n",
      "Nextflow operator set with content: .set { bam_bai }\n",
      "Error: Unclosed process block in file: ./git_repos\\hgtseq\\subworkflows\\local\\sortbam\\main.nf\n",
      "Process HGTSEQ in filepath ./git_repos\\hgtseq\\workflows\\hgtseq.nf\n",
      "Nextflow operator splitCsv with content: .splitCsv ( header:true, sep:',' )\n",
      "Nextflow operator map with content: .map { create_input_channel(it) }\n",
      "Nextflow operator map with content: ch_krakendb = UNTAR_KRAKEN.out.untar.map{ it[1] }\n",
      "Nextflow operator map with content: ch_kronadb = UNTAR_KRONA.out.untar.map{ it[1] }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(PREPARE_READS.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(READS_QC.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BAM_QC.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(CLASSIFY_UNMAPPED.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BAM_QC.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(CLASSIFY_UNMAPPED.out.versions)\n",
      "Nextflow operator collect with content: CLASSIFY_UNMAPPED.out.classified_single.collect{ it[1] },\n",
      "Nextflow operator collect with content: CLASSIFY_UNMAPPED.out.classified_both.collect{ it[1] },\n",
      "Nextflow operator collect with content: CLASSIFY_UNMAPPED.out.candidate_integrations.collect{ it[1] },\n",
      "Nextflow operator collect with content: CLASSIFY_UNMAPPED.out.classified_single.collect{ it[0].id }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(REPORTING.out.versions)\n",
      "Nextflow operator collectFile with content: ch_versions.unique().collectFile(name: 'collated_versions.yml')\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_workflow_summary.collectFile(name: 'workflow_summary_mqc.yaml'))\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_methods_description.collectFile(name: 'methods_description_mqc.yaml'))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(CUSTOM_DUMPSOFTWAREVERSIONS.out.mqc_yml.collect())\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(READS_QC.out.fastqc_untrimmed.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(READS_QC.out.fastqc_trimmed.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(BAM_QC.out.stats.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(BAM_QC.out.flagstat.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(BAM_QC.out.idxstats.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(BAM_QC.out.qualimap.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(BAM_QC.out.bamstats.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(CLASSIFY_UNMAPPED.out.report_single.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(CLASSIFY_UNMAPPED.out.report_both.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files.collect(),\n",
      "Nextflow operator toList with content: ch_multiqc_config.toList(),\n",
      "Nextflow operator toList with content: ch_multiqc_custom_config.toList(),\n",
      "Nextflow operator toList with content: ch_multiqc_logo.toList()\n",
      "Nextflow operator toList with content: multiqc_report = MULTIQC.out.report.toList()\n",
      "Error: Unclosed process block in file: ./git_repos\\hgtseq\\workflows\\hgtseq.nf\n",
      "Process NFCORE_HIC in filepath ./git_repos\\hic\\main.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\hic\\main.nf\n",
      "Process COMPARTMENTS in filepath ./git_repos\\hic\\subworkflows\\local\\compartments.nf\n",
      "Nextflow operator collect with content: fasta.map{it -> it[1]}.collect(),\n",
      "Nextflow operator collect with content: chrsize.map{it -> it[1]}.collect()\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(COOLTOOLS_EIGSCIS.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\hic\\subworkflows\\local\\compartments.nf\n",
      "Process COOLER in filepath ./git_repos\\hic\\subworkflows\\local\\cooler.nf\n",
      "Nextflow operator combine with content: chromsize.combine(cool_bins)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(COOLER_MAKEBINS.out.versions)\n",
      "Nextflow operator combine with content: pairs.combine(cool_bins),\n",
      "Nextflow operator collect with content: chromsize.map{it -> it[1]}.collect()\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(COOLER_CLOAD.out.versions)\n",
      "Nextflow operator map with content: .map{ it -> addResolution(it) }\n",
      "Nextflow operator set with content: .set{ ch_cool }\n",
      "Nextflow operator map with content: ch_cool.map{[it[0], it[1], \"\"]}\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(COOLER_BALANCE.out.versions)\n",
      "Nextflow operator min with content: ch_res_zoomify = cool_bins.min()\n",
      "Nextflow operator flatten with content: ch_res_zoomify = Channel.from(params.res_zoomify).splitCsv().flatten().unique().toInteger()\n",
      "Nextflow operator combine with content: .combine(ch_res_zoomify)\n",
      "Nextflow operator filter with content: .filter{ it[2] == it[3] }\n",
      "Nextflow operator map with content: .map{ it->[it[0], it[1]] }\n",
      "Nextflow operator set with content: .set{ ch_cool_zoomify }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(COOLER_ZOOMIFY.out.versions)\n",
      "Nextflow operator map with content: COOLER_BALANCE.out.cool.map{[it[0], it[1], \"\"]}\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(COOLER_DUMP.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SPLIT_COOLER_DUMP.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\hic\\subworkflows\\local\\cooler.nf\n",
      "Process HICPRO in filepath ./git_repos\\hic\\subworkflows\\local\\hicpro.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(HICPRO_MAPPING.out.versions)\n",
      "Nextflow operator collect with content: fragments.collect()\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GET_VALID_INTERACTION.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GET_VALID_INTERACTION_DNASE.out.versions)\n",
      "Nextflow operator groupTuple with content: ch_valid_pairs = ch_valid_pairs.map{ it -> removeChunks(it)}.groupTuple()\n",
      "Nextflow operator groupTuple with content: ch_hicpro_stats = HICPRO_MAPPING.out.mapstats.map{it->removeChunks(it)}.groupTuple()\n",
      "Nextflow operator concat with content: .concat(HICPRO_MAPPING.out.pairstats.map{it->removeChunks(it)}.groupTuple(),\n",
      "Nextflow operator groupTuple with content: ch_valid_stats.map{it->removeChunks(it)}.groupTuple())\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(MERGE_VALID_INTERACTION.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(MERGE_STATS.out.versions)\n",
      "Nextflow operator collect with content: chrsize.collect()\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(HICPRO2PAIRS.out.versions)\n",
      "Nextflow operator combine with content: MERGE_VALID_INTERACTION.out.valid_pairs.combine(map_res),\n",
      "Nextflow operator collect with content: chrsize.collect()\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ICE_NORMALIZATION.out.versions)\n",
      "Nextflow operator concat with content: mqc = MERGE_VALID_INTERACTION.out.mqc.concat(MERGE_STATS.out.mqc)\n",
      "Error: Unclosed process block in file: ./git_repos\\hic\\subworkflows\\local\\hicpro.nf\n",
      "Process HICPRO_MAPPING in filepath ./git_repos\\hic\\subworkflows\\local\\hicpro_mapping.nf\n",
      "Nextflow operator map with content: ch_reads_r1 = reads.map{ it -> pairToSingle(it,\"R1\") }\n",
      "Nextflow operator map with content: ch_reads_r2 = reads.map{ it -> pairToSingle(it,\"R2\") }\n",
      "Nextflow operator concat with content: ch_reads = ch_reads_r1.concat(ch_reads_r2)\n",
      "Nextflow operator collect with content: index.collect(),\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BOWTIE2_ALIGN.out.versions)\n",
      "Nextflow operator collect with content: ligation_site.collect()\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(TRIM_READS.out.versions)\n",
      "Nextflow operator collect with content: index.collect(),\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BOWTIE2_ALIGN_TRIMMED.out.versions)\n",
      "Nextflow operator combine with content: .combine(BOWTIE2_ALIGN_TRIMMED.out.bam, by:[0])\n",
      "Nextflow operator set with content: .set { ch_bowtie2_align}\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(MERGE_BOWTIE2.out.versions)\n",
      "Nextflow operator map with content: .map { singleToPair(it) }\n",
      "Nextflow operator groupTuple with content: .groupTuple()\n",
      "Nextflow operator set with content: .set {ch_bams}\n",
      "Nextflow operator map with content: .map { singleToPair(it) }\n",
      "Nextflow operator groupTuple with content: .groupTuple()\n",
      "Nextflow operator set with content: .set {ch_bams}\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(COMBINE_MATES.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\hic\\subworkflows\\local\\hicpro_mapping.nf\n",
      "Process INPUT_CHECK in filepath ./git_repos\\hic\\subworkflows\\local\\input_check.nf\n",
      "Nextflow operator splitCsv with content: .splitCsv ( header:true, sep:',' )\n",
      "Nextflow operator map with content: .map { create_fastq_channels(it) }\n",
      "Nextflow operator splitFastq with content: .splitFastq( by: params.fastq_chunks_size, pe:true, file: true, compress:true)\n",
      "Nextflow operator map with content: .map { it -> [it[0], [it[1], it[2]]]}\n",
      "Nextflow operator groupTuple with content: .groupTuple(by: [0])\n",
      "Nextflow operator flatMap with content: .flatMap { it -> setMetaChunk(it) }\n",
      "Nextflow operator collate with content: .collate(2)\n",
      "Nextflow operator set with content: .set { reads }\n",
      "Nextflow operator splitCsv with content: .splitCsv ( header:true, sep:',' )\n",
      "Nextflow operator map with content: .map { create_fastq_channels(it) }\n",
      "Nextflow operator map with content: .map { it -> [it[0], [it[1], it[2]]]}\n",
      "Nextflow operator groupTuple with content: .groupTuple(by: [0])\n",
      "Nextflow operator flatMap with content: .flatMap { it -> setMetaChunk(it) }\n",
      "Nextflow operator collate with content: .collate(2)\n",
      "Nextflow operator set with content: .set { reads }\n",
      "Error: Unclosed process block in file: ./git_repos\\hic\\subworkflows\\local\\input_check.nf\n",
      "Process PREPARE_GENOME in filepath ./git_repos\\hic\\subworkflows\\local\\prepare_genome.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BOWTIE2_BUILD.out.versions)\n",
      "Nextflow operator map with content: .map { it -> [[:], it]}\n",
      "Nextflow operator ifEmpty with content: .ifEmpty { exit 1, \"Genome index: Provided index not found: ${params.bwt2_index}\" }\n",
      "Nextflow operator set with content: .set { ch_index }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(CUSTOM_GETCHROMSIZES.out.versions)\n",
      "Nextflow operator map with content: .map { it -> [[:], it]}\n",
      "Nextflow operator set with content: .set {ch_chromsize}\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GET_RESTRICTION_FRAGMENTS.out.versions)\n",
      "Nextflow operator map with content: .map{ it -> [[:], it] }\n",
      "Nextflow operator set with content: .set {ch_resfrag}\n",
      "Error: Unclosed process block in file: ./git_repos\\hic\\subworkflows\\local\\prepare_genome.nf\n",
      "Process TADS in filepath ./git_repos\\hic\\subworkflows\\local\\tads.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(COOLTOOLS_INSULATION.out.versions)\n",
      "Nextflow operator mix with content: ch_tads = ch_tads.mix(COOLTOOLS_INSULATION.out.tsv)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(HIC_FIND_TADS.out.versions)\n",
      "Nextflow operator mix with content: ch_tads = ch_tads.mix(HIC_FIND_TADS.out.results)\n",
      "Error: Unclosed process block in file: ./git_repos\\hic\\subworkflows\\local\\tads.nf\n",
      "Process HIC in filepath ./git_repos\\hic\\workflows\\hic.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(PREPARE_GENOME.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(FASTQC.out.versions)\n",
      "Nextflow operator view with content: INPUT_CHECK.out.reads.view()\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(HICPRO.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(COOLER.out.versions)\n",
      "Nextflow operator combine with content: .combine(ch_ddecay_res)\n",
      "Nextflow operator filter with content: .filter{ it[0].resolution == it[2] }\n",
      "Nextflow operator map with content: .map { it -> [it[0], it[1]]}\n",
      "Nextflow operator set with content: .set{ ch_distdecay }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(HIC_PLOT_DIST_VS_COUNTS.out.versions)\n",
      "Nextflow operator combine with content: .combine(ch_comp_res)\n",
      "Nextflow operator filter with content: .filter{ it[0].resolution == it[2] }\n",
      "Nextflow operator map with content: .map { it -> [it[0], it[1], it[2]]}\n",
      "Nextflow operator set with content: .set{ ch_cool_compartments }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(COMPARTMENTS.out.versions)\n",
      "Nextflow operator combine with content: .combine(ch_tads_res)\n",
      "Nextflow operator filter with content: .filter{ it[0].resolution == it[2] }\n",
      "Nextflow operator map with content: .map { it -> [it[0], it[1]]}\n",
      "Nextflow operator set with content: .set{ ch_cool_tads }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(TADS.out.versions)\n",
      "Nextflow operator collectFile with content: ch_versions.unique().collectFile(name: 'collated_versions.yml')\n",
      "Nextflow operator mix with content: ch_multiqc_files = ch_multiqc_files.mix(ch_multiqc_config)\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(ch_multiqc_custom_config.collect().ifEmpty([]))\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_workflow_summary.collectFile(name: 'workflow_summary_mqc.yaml'))\n",
      "Nextflow operator map with content: ch_multiqc_files = ch_multiqc_files.mix(FASTQC.out.zip.map{it->it[1]})\n",
      "Nextflow operator mix with content: ch_multiqc_files = ch_multiqc_files.mix(HICPRO.out.mqc)\n",
      "Nextflow operator collect with content: ch_multiqc_custom_config.collect().ifEmpty([]),\n",
      "Nextflow operator collectFile with content: ch_workflow_summary.collectFile(name: 'workflow_summary_mqc.yaml'),\n",
      "Nextflow operator map with content: FASTQC.out.zip.map{it->it[1]},\n",
      "Nextflow operator collect with content: HICPRO.out.mqc.collect()\n",
      "Nextflow operator toList with content: multiqc_report = MULTIQC.out.report.toList()\n",
      "Error: Unclosed process block in file: ./git_repos\\hic\\workflows\\hic.nf\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Process NFCORE_HICAR in filepath ./git_repos\\hicar\\main.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\hicar\\main.nf\n",
      "Process APA in filepath ./git_repos\\hicar\\subworkflows\\local\\apa.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\hicar\\subworkflows\\local\\apa.nf\n",
      "Process BAM_STAT in filepath ./git_repos\\hicar\\subworkflows\\local\\bam_stats.nf\n",
      "Nextflow operator join with content: ch_bam_bai = SAMTOOLS_SORT.out.bam.join(SAMTOOLS_INDEX.out.bai, by: [0])\n",
      "Error: Unclosed process block in file: ./git_repos\\hicar\\subworkflows\\local\\bam_stats.nf\n",
      "Process ATAC_PEAK in filepath ./git_repos\\hicar\\subworkflows\\local\\callatacpeak.nf\n",
      "Nextflow operator mix with content: ch_version = ch_version.mix(SHIFT_READS.out.versions)\n",
      "Nextflow operator map with content: .map{meta, bed -> [meta.group, bed]}\n",
      "Nextflow operator groupTuple with content: .groupTuple()\n",
      "Nextflow operator map with content: .map{it -> [[id:it[0]], it[1]]} // id is group\n",
      "Nextflow operator set with content: .set{read4merge}\n",
      "Nextflow operator mix with content: ch_version = ch_version.mix(MERGE_READS.out.versions)\n",
      "Nextflow operator map with content: BEDTOOLS_GENOMECOV_PER_GROUP(MERGE_READS.out.bed.map{[it[0], it[1], \"1\"]}, chromsizes, \"bedgraph\")\n",
      "Nextflow operator map with content: ch_group_peak = MERGE_READS.out.bed.map{[it[0], anchors]}\n",
      "Nextflow operator map with content: MACS2_CALLPEAK(MERGE_READS.out.bed.map{[it[0], it[1], []]}, macs_gsize)\n",
      "Nextflow operator mix with content: ch_version = ch_version.mix(MACS2_CALLPEAK.out.versions)\n",
      "Nextflow operator map with content: ch_group_bdg = MACS2_CALLPEAK.out.bdg.map{[it[0], it[1].findAll{it.toString().contains('pileup')}]}\n",
      "Nextflow operator collect with content: anchor_peaks = MACS2_CALLPEAK.out.peak.map{it[1]}.collect()\n",
      "Nextflow operator collect with content: ATACQC(anchor_peaks, MERGE_READS.out.bed.map{it[1]}.collect(), gtf)\n",
      "Nextflow operator mix with content: ch_version = ch_version.mix(ATACQC.out.versions)\n",
      "Nextflow operator mix with content: ch_version = ch_version.mix(DUMP_READS_PER_GROUP.out.versions)\n",
      "Nextflow operator join with content: SHIFT_READS.out.bed.join(COVERAGE_SCALE.out.scale),\n",
      "Nextflow operator mix with content: ch_version = ch_version.mix(BEDTOOLS_GENOMECOV_PER_SAMPLE.out.versions)\n",
      "Nextflow operator mix with content: ch_version = ch_version.mix(UCSC_BEDGRAPHTOBIGWIG_PER_SAMPLE.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\hicar\\subworkflows\\local\\callatacpeak.nf\n",
      "Process RUN_CIRCOS in filepath ./git_repos\\hicar\\subworkflows\\local\\circos.nf\n",
      "Nextflow operator first with content: ).versions.first()\n",
      "Nextflow operator mix with content: ch_version = ch_version.mix(CIRCOS.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\hicar\\subworkflows\\local\\circos.nf\n",
      "Process COMPARTMENTS in filepath ./git_repos\\hicar\\subworkflows\\local\\compartments.nf\n",
      "Nextflow operator combine with content: ch_circos_files = HOMER_COMPARTMENTS.out.compartments.combine(cool_bin).map{[[id:it[0].id, bin:it[2]], it[1]]}\n",
      "Error: Unclosed process block in file: ./git_repos\\hicar\\subworkflows\\local\\compartments.nf\n",
      "Process COOLER in filepath ./git_repos\\hicar\\subworkflows\\local\\cooler.nf\n",
      "Nextflow operator map with content: ch_version = COOLER_CLOAD(valid_pairs.map{[it[0], it[2], it[3], it[1]]}, chromsizes).versions\n",
      "Nextflow operator map with content: .map{\n",
      "Nextflow operator groupTuple with content: .groupTuple(by:[0, 1])\n",
      "Nextflow operator map with content: .map{group, bin, cool -> [[id:group, bin:bin], cool]}\n",
      "Nextflow operator set with content: .set{ch_cooler}\n",
      "Nextflow operator map with content: COOLER_BALANCE(COOLER_MERGE.out.cool.map{[it[0], it[1], false]})\n",
      "Nextflow operator map with content: COOLER_DUMP_PER_GROUP(COOLER_MERGE.out.cool.map{[it[0], it[1], []]})\n",
      "Nextflow operator mix with content: ch_version = ch_version.mix(JUICER_PRE.out.versions)\n",
      "Nextflow operator map with content: COOLER_DUMP_PER_SAMPLE(COOLER_CLOAD.out.cool.map{ meta, cool, bin -> [[id:meta.id, group:meta.group, bin:bin], cool, []]})\n",
      "Nextflow operator mix with content: ch_version = ch_version.mix(DUMPREADS_PER_GROUP.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\hicar\\subworkflows\\local\\cooler.nf\n",
      "Process COVERAGE_SCALE in filepath ./git_repos\\hicar\\subworkflows\\local\\coveragescale.nf\n",
      "Nextflow operator collect with content: min_cnt = counts.map{[it[1].toInteger()]}.collect().map{it.min()}\n",
      "Error: Unclosed process block in file: ./git_repos\\hicar\\subworkflows\\local\\coveragescale.nf\n",
      "Process DA in filepath ./git_repos\\hicar\\subworkflows\\local\\differential_analysis.nf\n",
      "Nextflow operator map with content: ch_diffhicar = loops.map{meta, bin_size, peak -> [bin_size, peak]}\n",
      "Nextflow operator groupTuple with content: .groupTuple()\n",
      "Nextflow operator cross with content: .cross(samplebedpe.map{[it[0].bin, it[1]]}.groupTuple())\n",
      "Nextflow operator map with content: .map{ peak, long_bedpe ->\n",
      "Nextflow operator flatten with content: [peak[0], peak[1].flatten(), long_bedpe[1].flatten()] }//bin_size, meta, peak, long_bedpe\n",
      "Nextflow operator groupTuple with content: .groupTuple()\n",
      "Nextflow operator flatten with content: .map{[it[0], it[1].flatten().unique(), it[2].flatten()]}\n",
      "Nextflow operator filter with content: .filter{it[1].size > 1} // filter by the bedpe files. Single bedpe means single group, no need to do differential analysis\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(DIFFHICAR.out.versions)\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(DIFFHICAR.out.stats.collect().ifEmpty(null))\n",
      "Nextflow operator mix with content: ch_diff_files = ch_diff_files.mix(DIFFHICAR.out.diff)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(DIFFHIC.out.versions)\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(DIFFHIC.out.stats.collect().ifEmpty(null))\n",
      "Nextflow operator mix with content: ch_diff_files = ch_diff_files.mix(DIFFHIC.out.diff)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(DIFFSET.out.versions)\n",
      "Nextflow operator mix with content: ch_diff_files = ch_diff_files.mix(DIFFSET.out.diff)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(HICEXPLORER_DIFFHIC.out.versions)\n",
      "Nextflow operator mix with content: ch_diff_files = ch_diff_files.mix(HICEXPLORER_DIFFHIC.out.diff)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(DIFFHIC.out.versions)\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(DIFFHIC.out.stats.collect().ifEmpty(null))\n",
      "Error: Unclosed process block in file: ./git_repos\\hicar\\subworkflows\\local\\differential_analysis.nf\n",
      "Process HIPEAK in filepath ./git_repos\\hicar\\subworkflows\\local\\hipeak.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(R1_PEAK.out.versions)\n",
      "Nextflow operator map with content: distalpair = hdf5.map{meta, bed -> [meta.group, bed]}\n",
      "Nextflow operator groupTuple with content: .groupTuple()\n",
      "Nextflow operator map with content: grouped_reads_peak = peak.map{[it[0].id, it[1]]}\n",
      "Nextflow operator join with content: .join(R1_PEAK.out.peak.map{[it[0].id, it[1]]})\n",
      "Nextflow operator join with content: .join(distalpair)\n",
      "Nextflow operator map with content: .map{[[id:it[0]], it[1], it[2], it[3]]}\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(HI_PEAK.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\hicar\\subworkflows\\local\\hipeak.nf\n",
      "Process INTERACTIONS in filepath ./git_repos\\hicar\\subworkflows\\local\\interactions.nf\n",
      "Nextflow operator map with content: ch_annotation_files = MAPS.out.interactions.map{\n",
      "Nextflow operator map with content: ch_circos_files = MAPS.out.interactions.map{\n",
      "Nextflow operator map with content: ch_annotation_files = HICDCPLUS.out.interactions.map{\n",
      "Nextflow operator map with content: ch_circos_files = HICDCPLUS.out.interactions.map{\n",
      "Nextflow operator map with content: ch_annotation_files = PEAKACHU.out.interactions.map{\n",
      "Nextflow operator map with content: ch_circos_files = PEAKACHU.out.interactions.map{\n",
      "Nextflow operator map with content: ch_loops.map{\n",
      "Nextflow operator groupTuple with content: }.groupTuple()\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(MERGE_INTERACTIONS.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\hicar\\subworkflows\\local\\interactions.nf\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Process PAIRTOOLS_PAIRE in filepath ./git_repos\\hicar\\subworkflows\\local\\pairtools.nf\n",
      "Nextflow operator map with content: counts = PAIRTOOLS_PARSE.out.stat.map{\n",
      "Nextflow operator first with content: .first().replaceAll(/cis_20kb.\\s+/, '')\n",
      "Nextflow operator join with content: .join(COVERAGE_SCALE.out.scale)\n",
      "Nextflow operator map with content: rawstat.map{meta, stat -> [meta.id, meta, stat]}\n",
      "Nextflow operator join with content: .join(PAIRTOOLS_DEDUP.out.stat.map{meta, stat -> [meta.id, stat]})\n",
      "Nextflow operator map with content: .map{id, meta, raw, dedup -> [meta, raw, dedup ]}\n",
      "Nextflow operator set with content: .set{ reads_stat }\n",
      "Nextflow operator map with content: READS_SUMMARY(READS_STAT.out.stat.map{it[1]}\n",
      "Nextflow operator map with content: .mix(PAIRSPLOT.out.summary.map{it[1]})\n",
      "Nextflow operator collect with content: .mix(PAIRSPLOT.out.csv.map{it[1]}).collect())\n",
      "Error: Unclosed process block in file: ./git_repos\\hicar\\subworkflows\\local\\pairtools.nf\n",
      "Process PREPARE_GENOME in filepath ./git_repos\\hicar\\subworkflows\\local\\preparegenome.nf\n",
      "Nextflow operator map with content: ch_fasta = GUNZIP_FASTA.out.gunzip.map{it[1]}\n",
      "Nextflow operator map with content: ch_gtf = GUNZIP_GTF.out.gunzip.map{it[1]}\n",
      "Nextflow operator map with content: ch_gff = GUNZIP_GFF.out.gunzip.map{it[1]}\n",
      "Nextflow operator mix with content: ch_version = ch_version.mix(GFFREAD.out.versions)\n",
      "Nextflow operator map with content: ch_gene_bed = GUNZIP_GENE_BED.out.gunzip.map{it[1]}\n",
      "Nextflow operator mix with content: ch_version = ch_version.mix(GTF2BED.out.versions)\n",
      "Nextflow operator splitFasta with content: genome_size = ch_fasta.splitFasta( record: [id: true, seqString: true] )\n",
      "Nextflow operator map with content: .map { it.seqString.replaceAll('[^acgtACGT]','').length() }\n",
      "Nextflow operator sum with content: .sum()\n",
      "Nextflow operator ifEmpty with content: ch_blacklist = ch_blacklist.ifEmpty([])\n",
      "Nextflow operator mix with content: ch_version = ch_version.mix(GENOME_FILTER.out.versions)\n",
      "Nextflow operator mix with content: ch_version = ch_version.mix(COOLER_DIGEST.out.versions)\n",
      "Nextflow operator mix with content: ch_version = ch_version.mix(RE_CUTSITE.out.versions)\n",
      "Nextflow operator mix with content: ch_version = ch_version.mix(GENMAP_MAPPABILITY.out.versions)\n",
      "Nextflow operator map with content: GENMAP_MAPPABILITY.out.wig.map{[[id:'mappability'], it]},\n",
      "Nextflow operator map with content: ch_chrom_sizes).bw.map{it[1]}\n",
      "Nextflow operator mix with content: ch_version = ch_version.mix(UCSC_WIGTOBIGWIG.out.versions)\n",
      "Nextflow operator map with content: ch_bwa_index = params.bwa_index ? Channel.value(file(params.bwa_index)).map{ [['id':ucscname], it] } : BWA_INDEX ( ch_fasta.map{[['id':ucscname], it]} ).index\n",
      "Error: Unclosed process block in file: ./git_repos\\hicar\\subworkflows\\local\\preparegenome.nf\n",
      "Process TADS in filepath ./git_repos\\hicar\\subworkflows\\local\\tads.nf\n",
      "Nextflow operator map with content: ch_circos_files = COOLTOOLS_TADS.out.tads.map{[it[0], it[2]]}\n",
      "Nextflow operator map with content: ch_circos_files = HICEXPLORER_TADS.out.tads.map{[it[0], it[2]]}\n",
      "Nextflow operator combine with content: ch_circos_files = HOMER_TADS.out.tads.combine(cool_bin).map{[[id:it[0].id, bin:it[3]], it[2]]}\n",
      "Nextflow operator map with content: ch_circos_files = HICEXPLORER_TADS.out.tads.map{[it[0], it[2]]}\n",
      "Error: Unclosed process block in file: ./git_repos\\hicar\\subworkflows\\local\\tads.nf\n",
      "Process TFEA in filepath ./git_repos\\hicar\\subworkflows\\local\\tfea.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(HOMER_TFEA.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BIOC_ATACSEQTFEA.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(HOMER_FINDMOTIFSGENOME.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\hicar\\subworkflows\\local\\tfea.nf\n",
      "Process V4C in filepath ./git_repos\\hicar\\subworkflows\\local\\v4c.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(COOLTOOLS_V4C.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(HICEXPLORER_CHICPLOTVIEWPOINT.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BIOC_TRACKVIEWER.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BIOC_TRACKVIEWER.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\hicar\\subworkflows\\local\\v4c.nf\n",
      "Process COOLTOOLS_APACALLER in filepath ./git_repos\\hicar\\subworkflows\\local\\apa\\cooltools.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\hicar\\subworkflows\\local\\apa\\cooltools.nf\n",
      "Process JUICER_APACALLER in filepath ./git_repos\\hicar\\subworkflows\\local\\apa\\juicer.nf\n",
      "Nextflow operator map with content: ch_mergedloops = additional_param.map{[it[0], it[1]]}\n",
      "Nextflow operator map with content: ch_hic_loops = matrix.map{[it[0].bin, it[0], it[1]]}\n",
      "Nextflow operator combine with content: .combine(BIOC_SUBSETLOOPS.out.loops, by: 0)\n",
      "Nextflow operator map with content: .map{[it[1], it[2], it[3]]} // merge by bin\n",
      "Nextflow operator map with content: additional_param.map{[it[2]]}\n",
      "Error: Unclosed process block in file: ./git_repos\\hicar\\subworkflows\\local\\apa\\juicer.nf\n",
      "Process COOLTOOLS_COMPARTMENTS in filepath ./git_repos\\hicar\\subworkflows\\local\\compartments_caller\\cooltools.nf\n",
      "Nextflow operator combine with content: maxtix.combine(additional_param),\n",
      "Error: Unclosed process block in file: ./git_repos\\hicar\\subworkflows\\local\\compartments_caller\\cooltools.nf\n",
      "Process HICEXPLORER_COMPARTMENTS in filepath ./git_repos\\hicar\\subworkflows\\local\\compartments_caller\\hicexplorer.nf\n",
      "Nextflow operator mix with content: ch_version = ch_version.mix(HICEXPLORER_HICTRANSFORM.out.versions)\n",
      "Nextflow operator combine with content: .combine(HICEXPLORER_HICPCA.out.pca1, by: 0)\n",
      "Nextflow operator combine with content: .combine(HICEXPLORER_HICPCA.out.pca2, by: 0)\n",
      "Nextflow operator mix with content: ch_version = ch_version.mix(HICEXPLORER_HICPLOTMATRIX_PCA1.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\hicar\\subworkflows\\local\\compartments_caller\\hicexplorer.nf\n",
      "Process HOMER_COMPARTMENTS in filepath ./git_repos\\hicar\\subworkflows\\local\\compartments_caller\\homer.nf\n",
      "Nextflow operator collect with content: genome.map{it[0]}.collect()\n",
      "Nextflow operator collect with content: genome.map{it[0]}.collect()\n",
      "Nextflow operator mix with content: ch_version = ch_version.mix(HOMER_FINDHICCOMPARTMENTS.out.versions)\n",
      "Nextflow operator collect with content: genome.map{it[1]}.collect()\n",
      "Nextflow operator mix with content: ch_version = ch_version.mix(BEDGRAPH_TRIM.out.versions)\n",
      "Nextflow operator collect with content: genome.map{it[1]}.collect()\n",
      "Nextflow operator mix with content: ch_version = ch_version.mix(UCSC_BEDGRAPHTOBIGWIG_HOMER_COMPARTMENTS.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\hicar\\subworkflows\\local\\compartments_caller\\homer.nf\n",
      "Process JUICER_COMPARTMENTS in filepath ./git_repos\\hicar\\subworkflows\\local\\compartments_caller\\juicer.nf\n",
      "Nextflow operator combine with content: matrix.combine(additional_param),\n",
      "Nextflow operator collect with content: additional_param.map{it[1]}.collect()\n",
      "Nextflow operator mix with content: ch_version = ch_version.mix(WIG_TRIM.out.versions)\n",
      "Nextflow operator collect with content: additional_param.map{it[1]}.collect()\n",
      "Nextflow operator map with content: .map{\n",
      "Nextflow operator mix with content: ch_version = ch_version.mix(UCSC_BEDGRAPHTOBIGWIG_JUICER_EIGENVECTOR.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\hicar\\subworkflows\\local\\compartments_caller\\juicer.nf\n",
      "Process HICEXPLORER_DIFFHIC in filepath ./git_repos\\hicar\\subworkflows\\local\\diff_ana\\chicdiff.nf\n",
      "Nextflow operator combine with content: ch_diff = matrix.combine(HICEXPLORER_CHICDIFFERENTIALTEST.out.differential, by: 0)\n",
      "Error: Unclosed process block in file: ./git_repos\\hicar\\subworkflows\\local\\diff_ana\\chicdiff.nf\n",
      "Process HI_PEAK in filepath ./git_repos\\hicar\\subworkflows\\local\\hipeak\\call_hipeak.nf\n",
      "Nextflow operator map with content: r2_peaks = peaks.map{it[1]}\n",
      "Nextflow operator filter with content: chrom1 = chrom_size.splitCsv(sep:\"\\t\", header: false, strip: true).filter{ !(it[0] =~ /[_M]/) }.map{it[0]}\n",
      "Nextflow operator combine with content: ch_version = PREPARE_COUNTS(peaks.combine(chrom1)).versions\n",
      "Nextflow operator map with content: counts = PREPARE_COUNTS.out.counts.map{[it[0].id, it[1]]}\n",
      "Nextflow operator groupTuple with content: .groupTuple()\n",
      "Nextflow operator map with content: .map{[[id:it[0]], it[1]]}\n",
      "Nextflow operator combine with content: POST_COUNTS(counts.combine(mappability).combine(fasta).combine(digest_genome))\n",
      "Nextflow operator mix with content: ch_version = ch_version.mix(POST_COUNTS.out.versions)\n",
      "Nextflow operator mix with content: ch_version = ch_version.mix(CALL_HIPEAK.out.versions)\n",
      "Nextflow operator mix with content: ch_version = ch_version.mix(ASSIGN_TYPE.out.versions)\n",
      "Nextflow operator map with content: ch_bed = BEDPE2BED_HI(ASSIGN_TYPE.out.bedpe).bed.map{[it[0], 'hipeak', it[1]]}\n",
      "Nextflow operator map with content: .map{it[1]}\n",
      "Nextflow operator filter with content: .filter{ it.readLines().size > 1 }\n",
      "Nextflow operator collect with content: .collect()\n",
      "Nextflow operator map with content: .map{[\"HiPeak\", it]}\n",
      "Nextflow operator combine with content: .combine(r2_peaks), gtf, maps_3d_ext)\n",
      "Nextflow operator mix with content: ch_version = ch_version.mix(BIOC_CHIPPEAKANNO_HIPEAK.out.versions)\n",
      "Nextflow operator collect with content: hipeaks = ASSIGN_TYPE.out.peak.map{it[1]}.collect().filter{it.size()>1}\n",
      "Nextflow operator collect with content: peaks.map{it[3]}.collect())\n",
      "Nextflow operator mix with content: ch_version = ch_version.mix(DIFF_HIPEAK.out.versions)\n",
      "Nextflow operator map with content: ch_bed = ch_bed.mix(BEDPE2BED_DIFF(DIFF_HIPEAK.out.bedpe.map{[[id:'hipeak_diff'], it]}).bed.map{[it[0], 'hipeak_diff', it[1]]})\n",
      "Nextflow operator collect with content: .collect()\n",
      "Nextflow operator map with content: .map{[\"DiffHiPeak\", it]}\n",
      "Nextflow operator combine with content: .combine(r2_peaks), gtf, maps_3d_ext)\n",
      "Error: Unclosed process block in file: ./git_repos\\hicar\\subworkflows\\local\\hipeak\\call_hipeak.nf\n",
      "Process R1_PEAK in filepath ./git_repos\\hicar\\subworkflows\\local\\hipeak\\fragment_peak.nf\n",
      "Nextflow operator map with content: .map{meta, bed -> [meta.group, bed]}\n",
      "Nextflow operator groupTuple with content: .groupTuple()\n",
      "Nextflow operator map with content: .map{it -> [[id:it[0]], it[1]]} // id is group\n",
      "Nextflow operator set with content: .set{read4merge}\n",
      "Nextflow operator mix with content: ch_version = ch_version.mix(MERGE_R1READS.out.versions)\n",
      "Nextflow operator mix with content: ch_version = ch_version.mix(CALL_R1PEAK.out.versions)\n",
      "Nextflow operator collect with content: r1_peaks = CALL_R1PEAK.out.peak.map{it[1]}.collect()\n",
      "Nextflow operator mix with content: ch_version = ch_version.mix(DUMP_R1_READS_PER_SAMPLE.out.versions)\n",
      "Nextflow operator map with content: BEDTOOLS_GENOMECOV_PER_R1_SAMPLE(R1READS.out.bed.map{[it[0], it[1], \"1\"]}, chromsizes, \"bedgraph\")\n",
      "Nextflow operator mix with content: ch_version = ch_version.mix(BEDTOOLS_GENOMECOV_PER_R1_SAMPLE.out.versions)\n",
      "Nextflow operator mix with content: ch_version = ch_version.mix(UCSC_BEDGRAPHTOBIGWIG_PER_R1_SAMPLE.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\hicar\\subworkflows\\local\\hipeak\\fragment_peak.nf\n",
      "Process HICDCPLUS in filepath ./git_repos\\hicar\\subworkflows\\local\\interaction_caller\\hicdcplus.nf\n",
      "Nextflow operator map with content: bedpe.map{[ it[0].bin ]} // bin_size\n",
      "Nextflow operator unique with content: .unique() // unique the bin_size\n",
      "Nextflow operator combine with content: .combine(additional_param)\n",
      "Nextflow operator combine with content: ch_reads = reads.combine(HICDCPLUS_FEATURES.out.features)\n",
      "Nextflow operator map with content: .map{[[id:it[0].id, bin:it[2]],\n",
      "Nextflow operator combine with content: .combine(bedpe, by: 0)\n",
      "Nextflow operator combine with content: ch_loop = HICDCPLUS_CALL_LOOPS(ch_reads.combine(additional_param.map{[it[2]]})).interactions\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(HICDCPLUS_CALL_LOOPS.out.versions.ifEmpty([]))\n",
      "Error: Unclosed process block in file: ./git_repos\\hicar\\subworkflows\\local\\interaction_caller\\hicdcplus.nf\n",
      "Process HICEXPLORER_INTERACTIONS in filepath ./git_repos\\hicar\\subworkflows\\local\\interaction_caller\\hicexplorer.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\hicar\\subworkflows\\local\\interaction_caller\\hicexplorer.nf\n",
      "Process MAPS in filepath ./git_repos\\hicar\\subworkflows\\local\\interaction_caller\\maps.nf\n",
      "Nextflow operator map with content: bedpe.map{ [ it[0].bin ] }.unique(),                   // bin_size\n",
      "Nextflow operator map with content: additional_param.map{[ it[0], it[1], it[2], it[3] ]}   // site, fasta, chrom_size, mappability_bw\n",
      "Nextflow operator map with content: .map{ meta, reads, peaks ->\n",
      "Nextflow operator cross with content: .cross(bedpe.map{[it[0].id, it[0].bin, it[1]]})// group, bin, bedpe\n",
      "Nextflow operator map with content: .map{ short_bed, long_bedpe -> //[bin_size, group, macs2, long_bedpe, short_bed]\n",
      "Nextflow operator cross with content: background.cross(reads_peak)\n",
      "Nextflow operator map with content: .map{ background, reads -> //[group, bin_size, macs2, long_bedpe, short_bed, background]\n",
      "Nextflow operator set with content: .set{ maps_input }\n",
      "Nextflow operator map with content: additional_param.map{[ it[1] ]},            //chrom_size\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(MAPS_PEAK.out.versions)\n",
      "Nextflow operator collect with content: ch_multiqc_files = MAPS_PEAK.out.stats.collect().ifEmpty(null)\n",
      "Error: Unclosed process block in file: ./git_repos\\hicar\\subworkflows\\local\\interaction_caller\\maps.nf\n",
      "Process MAPS_MULTIENZYME in filepath ./git_repos\\hicar\\subworkflows\\local\\interaction_caller\\maps_multienzyme.nf\n",
      "Nextflow operator map with content: chrom_sizes = genome.map{[ it[2] ]}\n",
      "Nextflow operator map with content: mappability = genome.map{[ it[3] ]}\n",
      "Nextflow operator combine with content: ch_digest = cool_bin.combine(Channel.fromPath(params.maps_digest_file))\n",
      "Nextflow operator combine with content: cool_bin.combine(genome.map{[it[0], it[1]]}), // [bin, site, [fasta]]\n",
      "Nextflow operator combine with content: cool_bin.combine(genome.map{[it[0], it[1]]}), // [bin, site, [fasta]]\n",
      "Nextflow operator collect with content: seqlevelsstyle = SEQLEVELS_STYLE(MAPS_FEND.out.bed.map{it[1]}.collect().map{it[0]}).seqlevels_style\n",
      "Nextflow operator combine with content: ENSEMBL_UCSC_CONVERT2(cool_bin.combine(mappability))\n",
      "Nextflow operator mix with content: ch_version = ch_version.mix(ENSEMBL_UCSC_CONVERT1.out.versions)\n",
      "Nextflow operator map with content: UCSC_BIGWIGAVERAGEOVERBED(ENSEMBL_UCSC_CONVERT1.out.tab.map{[['id':'background', 'bin_size':it[0]], it[1]]},\n",
      "Nextflow operator map with content: ENSEMBL_UCSC_CONVERT2.out.tab.map{it[1]})\n",
      "Nextflow operator map with content: UCSC_BIGWIGAVERAGEOVERBED(MAPS_FEND.out.bed.map{[['id':'background', 'bin_size':it[0]], it[1]]}, mappability)\n",
      "Nextflow operator mix with content: ch_version = ch_version.mix(UCSC_BIGWIGAVERAGEOVERBED.out.versions)\n",
      "Nextflow operator cross with content: MAPS_MERGE(ch_digest.cross(UCSC_BIGWIGAVERAGEOVERBED.out.tab.map{[it[0].bin_size, it[1]]}).map{[it[0][0], it[0][1], it[1][1]]})\n",
      "Nextflow operator map with content: MAPS_FEATURE(MAPS_MERGE.out.map, chrom_sizes)\n",
      "Error: Unclosed process block in file: ./git_repos\\hicar\\subworkflows\\local\\interaction_caller\\maps_multienzyme.nf\n",
      "Process MAPS_PEAK in filepath ./git_repos\\hicar\\subworkflows\\local\\interaction_caller\\maps_peak.nf\n",
      "Nextflow operator mix with content: ch_version = ch_version.mix(MAPS_CALLPEAK.out.versions)\n",
      "Nextflow operator mix with content: ch_version = ch_version.mix(MAPS_RAW2BG2.out.versions)\n",
      "Nextflow operator mix with content: ch_version = ch_version.mix(COOLER_LOAD.out.versions)\n",
      "Nextflow operator map with content: .map{\n",
      "Nextflow operator set with content: .set{ch_cooler}\n",
      "Nextflow operator mix with content: ch_version = ch_version.mix(MAPS_REFORMAT.out.versions)\n",
      "Nextflow operator collect with content: MAPS_STATS(MAPS_CALLPEAK.out.summary.map{it[2]}.collect())\n",
      "Error: Unclosed process block in file: ./git_repos\\hicar\\subworkflows\\local\\interaction_caller\\maps_peak.nf\n",
      "Process PEAKACHU in filepath ./git_repos\\hicar\\subworkflows\\local\\interaction_caller\\peakachu.nf\n",
      "Nextflow operator combine with content: ch_loop = PEAKACHU_SCORE(cool.combine(PEAKACHU_MODEL.out.model.map{[it[0], file(it[1].trim(), checkIfExists:true)]}, by:0)).interactions\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(PEAKACHU_SCORE.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\hicar\\subworkflows\\local\\interaction_caller\\peakachu.nf\n",
      "Process COOLTOOLS_TADS in filepath ./git_repos\\hicar\\subworkflows\\local\\tads_caller\\cooltools.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\hicar\\subworkflows\\local\\tads_caller\\cooltools.nf\n",
      "Process HICEXPLORER_TADS in filepath ./git_repos\\hicar\\subworkflows\\local\\tads_caller\\hicexplorer.nf\n",
      "Nextflow operator join with content: cool.join(HICEXPLORER_HICFINDTADS.out.tads),\n",
      "Nextflow operator mix with content: ch_version = ch_version.mix(HICEXPLORER_HICPLOTTADS.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\hicar\\subworkflows\\local\\tads_caller\\hicexplorer.nf\n",
      "Process HOMER_TADS in filepath ./git_repos\\hicar\\subworkflows\\local\\tads_caller\\homer.nf\n",
      "Nextflow operator map with content: tagdir.map{\n",
      "Error: Unclosed process block in file: ./git_repos\\hicar\\subworkflows\\local\\tads_caller\\homer.nf\n",
      "Process HOMER_TFEA in filepath ./git_repos\\hicar\\subworkflows\\local\\tfea\\homer.nf\n",
      "Nextflow operator collect with content: seqlevelsstyle = SEQLEVELS_STYLE(bed.map{it[2]}.collect().map{it[0]}).seqlevels_style\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SEQLEVELS_STYLE.out.versions)\n",
      "Nextflow operator map with content: ch_bed = bed.map{[it[0].id, it[0], it[1], it[2]]}\n",
      "Nextflow operator combine with content: ch_new_bed = ch_bed.combine(ENSEMBL_UCSC_CONVERT(ch_bed.map{[it[0], it[3]]}).tab, by:0)\n",
      "Nextflow operator map with content: .map{[it[1], it[2], it[4]]}\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ENSEMBL_UCSC_CONVERT.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(HOMER_FINDMOTIFSGENOME.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\hicar\\subworkflows\\local\\tfea\\homer.nf\n",
      "Process COOLTOOLS_V4C in filepath ./git_repos\\hicar\\subworkflows\\local\\v4c\\cooltools.nf\n",
      "Nextflow operator map with content: matrix.map{it[3]},\n",
      "Nextflow operator map with content: matrix.map{[it[0], it[2]]}).versions\n",
      "Nextflow operator map with content: matrix.map{[it[0], it[1]]}\n",
      "Nextflow operator combine with content: .combine(BIOC_SUBSETLOOPS.out.bed, by: 0),\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(VIRTUAL4C_BY_COOLTOOLS.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\hicar\\subworkflows\\local\\v4c\\cooltools.nf\n",
      "Process HICAR in filepath ./git_repos\\hicar\\workflows\\hicar.nf\n",
      "Nextflow operator mix with content: ch_multiqc_files = ch_multiqc_files.mix(ch_multiqc_config)\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(ch_multiqc_custom_config.collect().ifEmpty([]))\n",
      "Nextflow operator map with content: .map{\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(CHECKSUMS.out.versions.ifEmpty(null))\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(PREPARE_GENOME.out.versions.ifEmpty(null))\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(FASTQC.out.versions.first().ifEmpty(null))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(FASTQC.out.zip.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(CUTADAPT.out.versions.ifEmpty(null))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(CUTADAPT.out.log.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(BWA_MEM.out.versions.ifEmpty(null))\n",
      "Nextflow operator map with content: .map{\n",
      "Nextflow operator groupTuple with content: .groupTuple(by: [0])\n",
      "Nextflow operator flatten with content: .map{[it[1][0], it[2].flatten()]}\n",
      "Nextflow operator set with content: .set{ mapped_bam }\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(SAMTOOLS_MERGE.out.versions.ifEmpty(null))\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(BAM_STAT.out.versions.ifEmpty(null))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(BAM_STAT.out.stats.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(BAM_STAT.out.flagstat.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(BAM_STAT.out.idxstats.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(PAIRTOOLS_PAIRE.out.versions.ifEmpty(null))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(PAIRTOOLS_PAIRE.out.stat.collect().ifEmpty([]))\n",
      "Nextflow operator combine with content: cool_bin.combine(PAIRTOOLS_PAIRE.out.pair)\n",
      "Nextflow operator map with content: .map{bin, meta, pair, px -> [meta, bin, pair, px]}\n",
      "Nextflow operator set with content: .set{cool_input}\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(COOLER.out.versions.ifEmpty(null))\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(ATAC_PEAK.out.versions.ifEmpty(null))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(ATAC_PEAK.out.stats.collect().ifEmpty([]))\n",
      "Nextflow operator map with content: ch_tfea_bed = ATAC_PEAK.out.peak.map{[it[0], 'R2', it[1]]}\n",
      "Nextflow operator combine with content: ch_loop_1d_peak = ATAC_PEAK.out.reads.combine(ATAC_PEAK.out.mergedpeak)\n",
      "Nextflow operator combine with content: .combine(PREPARE_GENOME.out.fasta)\n",
      "Nextflow operator combine with content: .combine(PREPARE_GENOME.out.chrom_sizes)\n",
      "Nextflow operator combine with content: .combine(PREPARE_GENOME.out.mappability)\n",
      "Nextflow operator collect with content: .collect()\n",
      "Nextflow operator combine with content: .combine(PREPARE_GENOME.out.fasta)\n",
      "Nextflow operator combine with content: .combine(PREPARE_GENOME.out.chrom_sizes)\n",
      "Nextflow operator combine with content: .combine(PREPARE_GENOME.out.mappability)\n",
      "Nextflow operator map with content: .map{ meta, cool ->\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Nextflow operator groupTuple with content: .groupTuple()\n",
      "Nextflow operator combine with content: .combine(RECENTER_PEAK.out.peak)\n",
      "Nextflow operator set with content: .set{ch_chic_explorer}\n",
      "Nextflow operator combine with content: HICEXPLORER_CHICVIEWPOINT(HICEXPLORER_CHICQUALITYCONTROL.out.referencepoints.combine(HICEXPLORER_CHICVIEWPOINTBACKGROUNDMODEL.out.background_model, by: 0))\n",
      "Nextflow operator combine with content: ch_comp_additional = PREPARE_GENOME.out.fasta.combine(PREPARE_GENOME.out.chrom_sizes)\n",
      "Nextflow operator map with content: PAIRTOOLS_PAIRE.out.homerpair.map{\n",
      "Nextflow operator groupTuple with content: }.groupTuple(by:0)\n",
      "Nextflow operator combine with content: ch_comp_additional = homer_genome.combine(PREPARE_GENOME.out.chrom_sizes)\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(CAT_CAT.out.versions.ifEmpty(null))\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(HOMER_MAKETAGDIRECTORY.out.versions.ifEmpty(null))\n",
      "Nextflow operator combine with content: ch_comp_additional  = ch_hic_tools.combine(PREPARE_GENOME.out.chrom_sizes)\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(COMPARTMENTS.out.versions.ifEmpty(null))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(COMPARTMENTS.out.mqc.collect().ifEmpty([]))\n",
      "Nextflow operator mix with content: ch_circos_files = ch_circos_files.mix(COMPARTMENTS.out.circos)\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(TADS.out.versions.ifEmpty(null))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(TADS.out.mqc.collect().ifEmpty([]))\n",
      "Nextflow operator mix with content: ch_circos_files = ch_circos_files.mix(TADS.out.circos)\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(INTERACTIONS.out.versions.ifEmpty(null))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(INTERACTIONS.out.mqc.collect().ifEmpty([]))\n",
      "Nextflow operator map with content: ch_annotation_files = ch_annotation_files.mix(INTERACTIONS.out.anno.map{[params.interactions_tool+'/'+it[0], it[1]]})\n",
      "Nextflow operator mix with content: ch_circos_files = ch_circos_files.mix(INTERACTIONS.out.circos)\n",
      "Nextflow operator mix with content: ch_de_files = ch_de_files.mix(INTERACTIONS.out.loops)\n",
      "Nextflow operator combine with content: ch_apa_additional = INTERACTIONS.out.mergedloops.combine(ch_apa_additional).unique()\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(APA.out.versions.ifEmpty(null))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(APA.out.mqc.collect().ifEmpty([]))\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(HIPEAK.out.versions.ifEmpty(null))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(HIPEAK.out.mqc.collect().ifEmpty([]))\n",
      "Nextflow operator map with content: ch_tfea_bed = ch_tfea_bed.mix(HIPEAK.out.fragmentPeak.map{[it[0], 'R1', it[1]]})\n",
      "Nextflow operator mix with content: ch_tfea_bed = ch_tfea_bed.mix(HIPEAK.out.bed4tfea)\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(DA.out.versions.ifEmpty(null))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(DA.out.mqc.collect().ifEmpty([]))\n",
      "Nextflow operator map with content: ch_annotation_files = ch_annotation_files.mix(DA.out.anno.map{[params.da_tool+'/'+it[1], it[2]]})\n",
      "Nextflow operator map with content: ch_v4c_files = ch_v4c_files.mix(DA.out.anno.map{[it[0], it[2]]})\n",
      "Nextflow operator groupTuple with content: ch_tfea_bed = ch_tfea_bed.map{[[id:it[1]], it[2]]}.groupTuple(by:0)\n",
      "Nextflow operator combine with content: .combine(\n",
      "Nextflow operator map with content: .map{[it[1]]}\n",
      "Nextflow operator collect with content: .collect().toList())\n",
      "Nextflow operator combine with content: .combine(PREPARE_GENOME.out.fasta)\n",
      "Nextflow operator combine with content: .combine(PREPARE_GENOME.out.gtf)\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(TFEA.out.versions.ifEmpty(null))\n",
      "Nextflow operator combine with content: BIOC_CHIPPEAKANNO(ch_annotation_files.combine(ATAC_PEAK.out.mergedpeak), PREPARE_GENOME.out.gtf, params.maps_3d_ext)\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(BIOC_CHIPPEAKANNO.out.versions.ifEmpty(null))\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(V4C.out.versions.ifEmpty(null))\n",
      "Nextflow operator mix with content: .mix(\n",
      "Nextflow operator map with content: .map{\n",
      "Nextflow operator groupTuple with content: .groupTuple())\n",
      "Nextflow operator groupTuple with content: .groupTuple()\n",
      "Nextflow operator filter with content: .filter{it[1].size()>1} // in case the ch_v4c files is empty\n",
      "Nextflow operator map with content: .map{bin, df -> [bin, df[0], df[1]]} // [bin, cool, bedpe]\n",
      "Nextflow operator combine with content: .combine(ATAC_PEAK.out.mergedpeak)\n",
      "Nextflow operator set with content: .set{ch_v4c}\n",
      "Nextflow operator collect with content: PAIRTOOLS_PAIRE.out.hdf5.collect{it[1]},\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(V4C.out.versions.ifEmpty(null))\n",
      "Nextflow operator groupTuple with content: ch_circos_files.groupTuple(),\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(RUN_CIRCOS.out.versions.ifEmpty(null))\n",
      "Nextflow operator map with content: INTERACTIONS.out.loops.map{[it[0].id+'.'+it[1]+'.contacts',\n",
      "Nextflow operator mix with content: .mix(ATAC_PEAK.out\n",
      "Nextflow operator map with content: .bws.map{[it[0].id+\"_R2\",\n",
      "Nextflow operator set with content: .set{ch_trackfiles} // collect track files for igv\n",
      "Nextflow operator collect with content: ch_trackfiles.collect{it.join('\\t')}\n",
      "Nextflow operator flatten with content: .flatten()\n",
      "Nextflow operator collectFile with content: .collectFile(\n",
      "Nextflow operator set with content: .set{ igv_track_files }\n",
      "Nextflow operator collectFile with content: ch_versions.unique().collectFile(name: 'collated_versions.yml')\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(CUSTOM_DUMPSOFTWAREVERSIONS.out.mqc_yml.collect().ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(CUSTOM_DUMPSOFTWAREVERSIONS.out.yml.collect().ifEmpty([]))\n",
      "Nextflow operator flatten with content: .flatten()\n",
      "Nextflow operator map with content: .map { it -> if (it) [ it.baseName, it ] }\n",
      "Nextflow operator groupTuple with content: .groupTuple()\n",
      "Nextflow operator map with content: .map { it[1][0] }\n",
      "Nextflow operator flatten with content: .flatten()\n",
      "Nextflow operator collect with content: .collect()\n",
      "Nextflow operator set with content: .set { ch_multiqc_files }\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_workflow_summary.collectFile(name: 'workflow_summary_mqc.yaml'))\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_methods_description.collectFile(name: 'methods_description_mqc.yaml'))\n",
      "Nextflow operator collect with content: ch_multiqc_files.collect(),\n",
      "Nextflow operator toList with content: ch_multiqc_config.toList(),\n",
      "Nextflow operator toList with content: ch_multiqc_custom_config.toList(),\n",
      "Nextflow operator toList with content: ch_multiqc_logo.toList()\n",
      "Nextflow operator toList with content: multiqc_report = MULTIQC.out.report.toList()\n",
      "Error: Unclosed process block in file: ./git_repos\\hicar\\workflows\\hicar.nf\n",
      "Process NFCORE_HLATYPING in filepath ./git_repos\\hlatyping\\main.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\hlatyping\\main.nf\n",
      "Process INPUT_CHECK in filepath ./git_repos\\hlatyping\\subworkflows\\local\\input_check.nf\n",
      "Nextflow operator splitCsv with content: .splitCsv ( header:true, sep:',' )\n",
      "Nextflow operator map with content: .map { create_fastq_channel(it) }\n",
      "Nextflow operator set with content: .set { reads }\n",
      "Error: Unclosed process block in file: ./git_repos\\hlatyping\\subworkflows\\local\\input_check.nf\n",
      "Process HLATYPING in filepath ./git_repos\\hlatyping\\workflows\\hlatyping.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(INPUT_CHECK.out.versions)\n",
      "Nextflow operator branch with content: .branch { meta, files ->\n",
      "Nextflow operator set with content: .set { ch_input_files }\n",
      "Nextflow operator map with content: .map {meta, reads, single_end ->\n",
      "Nextflow operator set with content: .set { ch_bam_pe_corrected }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(CHECK_PAIRED.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_COLLATEFASTQ.out.versions)\n",
      "Nextflow operator map with content: .map { meta, reads, reads_other, reads_singleton ->\n",
      "Nextflow operator set with content: .set { ch_filtered_bam2fq }\n",
      "Nextflow operator mix with content: .mix(ch_filtered_bam2fq)\n",
      "Nextflow operator map with content: .map { meta, reads ->\n",
      "Nextflow operator set with content: .set { ch_input_with_references }\n",
      "Nextflow operator mix with content: .mix(ch_filtered_bam2fq)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(FASTQC.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(YARA_INDEX.out.versions)\n",
      "Nextflow operator mix with content: .mix(ch_filtered_bam2fq)\n",
      "Nextflow operator cross with content: .cross(YARA_INDEX.out.index)\n",
      "Nextflow operator multiMap with content: .multiMap { reads, index ->\n",
      "Nextflow operator set with content: .set { ch_mapping_input }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(YARA_MAPPER.out.versions)\n",
      "Nextflow operator join with content: YARA_MAPPER.out.bam.join(YARA_MAPPER.out.bai)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(OPTITYPE.out.versions)\n",
      "Nextflow operator collectFile with content: ch_versions.unique().collectFile(name: 'collated_versions.yml')\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_workflow_summary.collectFile(name: 'workflow_summary_mqc.yaml'))\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_methods_description.collectFile(name: 'methods_description_mqc.yaml'))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(CUSTOM_DUMPSOFTWAREVERSIONS.out.mqc_yml.collect())\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(FASTQC.out.zip.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(OPTITYPE.out.output.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files.collect(),\n",
      "Nextflow operator collect with content: ch_multiqc_config.collect().ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_multiqc_custom_config.collect().ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_multiqc_logo.collect().ifEmpty([]),\n",
      "Nextflow operator toList with content: multiqc_report = MULTIQC.out.report.toList()\n",
      "Nextflow operator mix with content: ch_versions    = ch_versions.mix(MULTIQC.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\hlatyping\\workflows\\hlatyping.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\imcyto\\main.nf\n",
      "Process NFCORE_ISOSEQ in filepath ./git_repos\\isoseq\\main.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\isoseq\\main.nf\n",
      "Process INPUT_CHECK in filepath ./git_repos\\isoseq\\subworkflows\\local\\input_check.nf\n",
      "Nextflow operator splitCsv with content: .splitCsv ( header:true, sep:',' )\n",
      "Nextflow operator flatMap with content: .flatMap { create_pbccs_channels(it, chunk) }\n",
      "Nextflow operator set with content: .set { reads }\n",
      "Error: Unclosed process block in file: ./git_repos\\isoseq\\subworkflows\\local\\input_check.nf\n",
      "Process SET_CHUNK_NUM_CHANNEL in filepath ./git_repos\\isoseq\\subworkflows\\local\\set_chunk_num_channel.nf\n",
      "Nextflow operator toList with content: .from((1..chunk).step(1).toList()*n_samples)\n",
      "Nextflow operator set with content: .set { chunk_num }\n",
      "Error: Unclosed process block in file: ./git_repos\\isoseq\\subworkflows\\local\\set_chunk_num_channel.nf\n",
      "Process SET_VALUE_CHANNEL in filepath ./git_repos\\isoseq\\subworkflows\\local\\set_value_channel.nf\n",
      "Nextflow operator map with content: .map { it[1] }\n",
      "Nextflow operator set with content: .set { data }\n",
      "Nextflow operator set with content: .set { data }\n",
      "Error: Unclosed process block in file: ./git_repos\\isoseq\\subworkflows\\local\\set_value_channel.nf\n",
      "Process ISOSEQ in filepath ./git_repos\\isoseq\\workflows\\isoseq.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(INPUT_CHECK.out.versions)\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator set with content: .set { ch_pbccs_bam_updated }\n",
      "Nextflow operator map with content: GNU_SORT(SET_GTF_CHANNEL.out.data.map { it -> [ [id:'genome'], it ]  } )          // Sort GTF on sequence and start, uLTRA index fails with topological sort\n",
      "Nextflow operator map with content: ULTRA_INDEX(SET_FASTA_CHANNEL.out.data, GNU_SORT.out.sorted.map { it[1] })        // Index GTF file before alignment\n",
      "Nextflow operator map with content: .map { [ [id:it[0].id_former], it[1] ] }\n",
      "Nextflow operator groupTuple with content: .groupTuple()\n",
      "Nextflow operator set with content: .set { ch_tcollapse }\n",
      "Nextflow operator join with content: .join( GSTAMA_FILELIST.out.tsv )\n",
      "Nextflow operator set with content: .set { ch_tmerge_in }\n",
      "Nextflow operator map with content: GSTAMA_MERGE(ch_tmerge_in.map { [ it[0], it[1] ] }, ch_tmerge_in.map { it[2] }) // Merge all bed files from one sample into a uniq bed file\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(PBCCS.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(LIMA.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ISOSEQ3_REFINE.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BAMTOOLS_CONVERT.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GSTAMA_COLLAPSE.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GSTAMA_MERGE.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GSTAMA_POLYACLEANUP.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GNU_SORT.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ULTRA_INDEX.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ULTRA_ALIGN.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(MINIMAP2_ALIGN.out.versions)\n",
      "Nextflow operator collectFile with content: ch_versions.unique().collectFile(name: 'collated_versions.yml')\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_workflow_summary.collectFile(name: 'workflow_summary_mqc.yaml'))\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_methods_description.collectFile(name: 'methods_description_mqc.yaml'))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(PBCCS.out.report_json.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(LIMA.out.summary.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(LIMA.out.counts.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(CUSTOM_DUMPSOFTWAREVERSIONS.out.mqc_yml.collect())\n",
      "Nextflow operator collect with content: ch_multiqc_files.collect(),\n",
      "Nextflow operator toList with content: ch_multiqc_config.toList(),\n",
      "Nextflow operator toList with content: ch_multiqc_custom_config.toList(),\n",
      "Nextflow operator toList with content: ch_multiqc_logo.toList()\n",
      "Nextflow operator toList with content: multiqc_report = MULTIQC.out.report.toList()\n",
      "Error: Unclosed process block in file: ./git_repos\\isoseq\\workflows\\isoseq.nf\n",
      "Process NFCORE_LIVERCTANALYSIS in filepath ./git_repos\\liverctanalysis\\main.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\liverctanalysis\\main.nf\n",
      "Process INPUT_CHECK in filepath ./git_repos\\liverctanalysis\\subworkflows\\local\\input_check.nf\n",
      "Nextflow operator splitCsv with content: .splitCsv ( header:true, sep:',' )\n",
      "Nextflow operator map with content: .map { create_fastq_channels(it) }\n",
      "Nextflow operator set with content: .set { reads }\n",
      "Error: Unclosed process block in file: ./git_repos\\liverctanalysis\\subworkflows\\local\\input_check.nf\n",
      "Process LIVERCTANALYSIS in filepath ./git_repos\\liverctanalysis\\workflows\\liverctanalysis.nf\n",
      "Nextflow operator first with content: ch_software_versions = ch_software_versions.mix(FASTQC.out.version.first().ifEmpty(null))\n",
      "Nextflow operator map with content: .map { it -> if (it) [ it.baseName, it ] }\n",
      "Nextflow operator groupTuple with content: .groupTuple()\n",
      "Nextflow operator map with content: .map { it[1][0] }\n",
      "Nextflow operator flatten with content: .flatten()\n",
      "Nextflow operator collect with content: .collect()\n",
      "Nextflow operator set with content: .set { ch_software_versions }\n",
      "Nextflow operator collect with content: ch_software_versions.map { it }.collect()\n",
      "Nextflow operator mix with content: ch_multiqc_files = ch_multiqc_files.mix(Channel.from(ch_multiqc_config))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(ch_multiqc_custom_config.collect().ifEmpty([]))\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_workflow_summary.collectFile(name: 'workflow_summary_mqc.yaml'))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(GET_SOFTWARE_VERSIONS.out.yaml.collect())\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(FASTQC.out.zip.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files.collect()\n",
      "Nextflow operator toList with content: multiqc_report       = MULTIQC.out.report.toList()\n",
      "Nextflow operator ifEmpty with content: ch_software_versions = ch_software_versions.mix(MULTIQC.out.version.ifEmpty(null))\n",
      "Error: Unclosed process block in file: ./git_repos\\liverctanalysis\\workflows\\liverctanalysis.nf\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Process NFCORE_MAG in filepath ./git_repos\\mag\\main.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\mag\\main.nf\n",
      "Process ANCIENT_DNA_ASSEMBLY_VALIDATION in filepath ./git_repos\\mag\\subworkflows\\local\\ancient_dna.nf\n",
      "Nextflow operator map with content: PYDAMAGE_ANALYZE(input.map {item -> [item[0], item[2], item[3]]})\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(PYDAMAGE_ANALYZE.out.versions.first())\n",
      "Nextflow operator map with content: FAIDX(input.map { item -> [ item[0], item[1] ] }, [[],[]] )\n",
      "Nextflow operator join with content: freebayes_input = input.join(FAIDX.out.fai)  // [val(meta), path(contigs), path(bam), path(bam_index), path(fai)]\n",
      "Nextflow operator multiMap with content: .multiMap{\n",
      "Nextflow operator join with content: BCFTOOLS_VIEW(FREEBAYES.out.vcf.join(BCFTOOLS_INDEX_PRE.out.tbi), [], [], [])\n",
      "Nextflow operator join with content: .join(BCFTOOLS_INDEX_POST.out.tbi)\n",
      "Nextflow operator join with content: .join(input.map { item -> [ item[0], item[1] ] }))\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(FAIDX.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(FREEBAYES.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(BCFTOOLS_CONSENSUS.out.versions.first())\n",
      "Error: Unclosed process block in file: ./git_repos\\mag\\subworkflows\\local\\ancient_dna.nf\n",
      "Process BINNING in filepath ./git_repos\\mag\\subworkflows\\local\\binning.nf\n",
      "Nextflow operator map with content: .map { meta, assembly, bams, bais ->\n",
      "Nextflow operator map with content: .map { meta, depths ->\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(METABAT2_JGISUMMARIZEBAMCONTIGDEPTHS.out.versions.first())\n",
      "Nextflow operator map with content: .map { meta, assembly, bams, bais ->\n",
      "Nextflow operator join with content: .join( ch_metabat_depths, by: 0 )\n",
      "Nextflow operator map with content: .map { meta, assembly, bams, bais, depths ->\n",
      "Nextflow operator map with content: .map { meta, assembly, reads, depth ->\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(CONVERT_DEPTHS.out.versions.first())\n",
      "Nextflow operator mix with content: ch_final_bins_for_gunzip = ch_final_bins_for_gunzip.mix( METABAT2_METABAT2.out.fasta.transpose() )\n",
      "Nextflow operator mix with content: ch_binning_results_gzipped_final = ch_binning_results_gzipped_final.mix( METABAT2_METABAT2.out.fasta )\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(METABAT2_METABAT2.out.versions.first())\n",
      "Nextflow operator mix with content: ch_final_bins_for_gunzip = ch_final_bins_for_gunzip.mix( ADJUST_MAXBIN2_EXT.out.renamed_bins.transpose() )\n",
      "Nextflow operator mix with content: ch_binning_results_gzipped_final = ch_binning_results_gzipped_final.mix( ADJUST_MAXBIN2_EXT.out.renamed_bins )\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(MAXBIN2.out.versions)\n",
      "Nextflow operator map with content: .map { meta, bins, bams, bais ->\n",
      "Nextflow operator multiMap with content: .multiMap {\n",
      "Nextflow operator mix with content: ch_final_bins_for_gunzip = ch_final_bins_for_gunzip.mix( FASTA_BINNING_CONCOCT.out.bins.transpose() )\n",
      "Nextflow operator mix with content: ch_binning_results_gzipped_final = ch_binning_results_gzipped_final.mix( FASTA_BINNING_CONCOCT.out.bins )\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(FASTA_BINNING_CONCOCT.out.versions)\n",
      "Nextflow operator mix with content: ch_input_splitfasta = METABAT2_METABAT2.out.unbinned.mix(MAXBIN2.out.unbinned_fasta)\n",
      "Nextflow operator transpose with content: ch_split_fasta_results_transposed = SPLIT_FASTA.out.unbinned.transpose()\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SPLIT_FASTA.out.versions)\n",
      "Nextflow operator groupTuple with content: .groupTuple(by: 0)\n",
      "Nextflow operator groupTuple with content: .groupTuple(by: 0)\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(GUNZIP_BINS.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(GUNZIP_UNBINS.out.versions.first())\n",
      "Error: Unclosed process block in file: ./git_repos\\mag\\subworkflows\\local\\binning.nf\n",
      "Process BINNING_PREPARATION in filepath ./git_repos\\mag\\subworkflows\\local\\binning_preparation.nf\n",
      "Nextflow operator combine with content: .combine(reads)\n",
      "Nextflow operator map with content: ch_reads_bowtie2 = reads.map{ meta, reads -> [ meta.group, meta, reads ] }\n",
      "Nextflow operator map with content: .map { meta, assembly, index -> [ meta.group, meta, assembly, index ] }\n",
      "Nextflow operator combine with content: .combine(ch_reads_bowtie2, by: 0)\n",
      "Nextflow operator map with content: .map { group, assembly_meta, assembly, index, reads_meta, reads -> [ assembly_meta, assembly, index, reads_meta, reads ] }\n",
      "Nextflow operator map with content: ch_reads_bowtie2 = reads.map{ meta, reads -> [ meta.id, meta, reads ] }\n",
      "Nextflow operator map with content: .map { meta, assembly, index -> [ meta.id, meta, assembly, index ] }\n",
      "Nextflow operator combine with content: .combine(ch_reads_bowtie2, by: 0)\n",
      "Nextflow operator map with content: .map { id, assembly_meta, assembly, index, reads_meta, reads -> [ assembly_meta, assembly, index, reads_meta, reads ] }\n",
      "Nextflow operator groupTuple with content: .groupTuple(by: 0)\n",
      "Nextflow operator map with content: .map { meta, assembly, bams, bais -> [ meta, assembly.sort()[0], bams, bais ] }     // multiple symlinks to the same assembly -> use first of sorted list\n",
      "Nextflow operator map with content: bowtie2_assembly_multiqc = BOWTIE2_ASSEMBLY_ALIGN.out.log.map { assembly_meta, reads_meta, log -> [ log ] }\n",
      "Error: Unclosed process block in file: ./git_repos\\mag\\subworkflows\\local\\binning_preparation.nf\n",
      "Process BINNING_REFINEMENT in filepath ./git_repos\\mag\\subworkflows\\local\\binning_refinement.nf\n",
      "Nextflow operator map with content: .map { meta, bins ->\n",
      "Nextflow operator groupTuple with content: .groupTuple()\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator flatten with content: meta, bins -> [meta, bins.flatten()]\n",
      "Nextflow operator branch with content: .branch {\n",
      "Nextflow operator mix with content: .mix(DASTOOL_FASTATOCONTIG2BIN_METABAT2.out.fastatocontig2bin)\n",
      "Nextflow operator mix with content: .mix(DASTOOL_FASTATOCONTIG2BIN_MAXBIN2.out.fastatocontig2bin)\n",
      "Nextflow operator mix with content: .mix(DASTOOL_FASTATOCONTIG2BIN_CONCOCT.out.fastatocontig2bin)\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator groupTuple with content: .groupTuple(by: 0)\n",
      "Nextflow operator join with content: ch_input_for_dastool = ch_contigs_for_dastool.join(ch_fastatocontig2bin_for_dastool, by: 0)\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(DASTOOL_FASTATOCONTIG2BIN_METABAT2.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(DASTOOL_FASTATOCONTIG2BIN_MAXBIN2.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(DASTOOL_FASTATOCONTIG2BIN_CONCOCT.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(DASTOOL_DASTOOL.out.versions.first())\n",
      "Nextflow operator transpose with content: ch_dastool_bins_newmeta = DASTOOL_DASTOOL.out.bins.transpose()\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator groupTuple with content: .groupTuple()\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator map with content: .map {\n",
      "Error: Unclosed process block in file: ./git_repos\\mag\\subworkflows\\local\\binning_refinement.nf\n",
      "Process BUSCO_QC in filepath ./git_repos\\mag\\subworkflows\\local\\busco_qc.nf\n",
      "Nextflow operator map with content: .map{\n",
      "Nextflow operator map with content: .map{\n",
      "Nextflow operator collect with content: .collect()\n",
      "Nextflow operator map with content: .map{\n",
      "Nextflow operator collect with content: .collect()\n",
      "Nextflow operator flatten with content: ch_downloads = BUSCO.out.busco_downloads.groupTuple().map{lin,downloads -> downloads[0]}.toSortedList().flatten()\n",
      "Nextflow operator collect with content: busco_summary_domain = BUSCO.out.summary_domain.collect()\n",
      "Nextflow operator collect with content: busco_summary_specific = BUSCO.out.summary_specific.collect()\n",
      "Nextflow operator collect with content: busco_failed_bin = BUSCO.out.failed_bin.collect()\n",
      "Nextflow operator collect with content: BUSCO.out.summary_domain.map{it[1]}.collect().ifEmpty([]),\n",
      "Nextflow operator collect with content: BUSCO.out.summary_specific.map{it[1]}.collect().ifEmpty([]),\n",
      "Nextflow operator collect with content: BUSCO.out.failed_bin.map{it[1]}.collect().ifEmpty([])\n",
      "Nextflow operator map with content: failed_bin  = BUSCO.out.failed_bin.map{it[1]}\n",
      "Nextflow operator map with content: multiqc     = BUSCO.out.summary_domain.mix(BUSCO.out.summary_specific).map{it[1]}\n",
      "Error: Unclosed process block in file: ./git_repos\\mag\\subworkflows\\local\\busco_qc.nf\n",
      "Process CHECKM_QC in filepath ./git_repos\\mag\\subworkflows\\local\\checkm_qc.nf\n",
      "Nextflow operator multiMap with content: .multiMap {\n",
      "Nextflow operator join with content: ext: fa.extension.unique().join(\"\") // we set this in the pipeline to always `.fa` so this should be fine\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(CHECKM_LINEAGEWF.out.versions.first())\n",
      "Nextflow operator join with content: .join(CHECKM_LINEAGEWF.out.marker_file)\n",
      "Nextflow operator map with content: .map{\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(CHECKM_QA.out.versions.first())\n",
      "Nextflow operator collect with content: COMBINE_CHECKM_TSV ( CHECKM_QA.out.output.map{it[1]}.collect() )\n",
      "Error: Unclosed process block in file: ./git_repos\\mag\\subworkflows\\local\\checkm_qc.nf\n",
      "Process DEPTHS in filepath ./git_repos\\mag\\subworkflows\\local\\depths.nf\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator groupTuple with content: .groupTuple()\n",
      "Nextflow operator combine with content: .combine(depths, by: 0)\n",
      "Nextflow operator transpose with content: .transpose()\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator groupTuple with content: .groupTuple(by: [0,2])\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator flatten with content: [meta, bins.unique().flatten(), depth]\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(MAG_DEPTHS.out.versions)\n",
      "Nextflow operator collectFile with content: .collectFile(name:'sample_groups.tsv'){ meta, reads -> meta.id + '\\t' + meta.group + '\\n' }\n",
      "Nextflow operator map with content: .map { meta, bin_depths_file ->\n",
      "Nextflow operator collect with content: MAG_DEPTHS_PLOT ( ch_mag_depths_plot, ch_sample_groups.collect() )\n",
      "Nextflow operator collectFile with content: .collectFile(keepHeader: true) {\n",
      "Nextflow operator collect with content: MAG_DEPTHS_SUMMARY ( ch_mag_depth_out.collect() )\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix( MAG_DEPTHS_PLOT.out.versions )\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix( MAG_DEPTHS_SUMMARY.out.versions )\n",
      "Error: Unclosed process block in file: ./git_repos\\mag\\subworkflows\\local\\depths.nf\n",
      "Process DOMAIN_CLASSIFICATION in filepath ./git_repos\\mag\\subworkflows\\local\\domain_classification.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(TIARA.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\mag\\subworkflows\\local\\domain_classification.nf\n",
      "Process GTDBTK in filepath ./git_repos\\mag\\subworkflows\\local\\gtdbtk.nf\n",
      "Nextflow operator splitCsv with content: .splitCsv(header: true, sep: '\\t')\n",
      "Nextflow operator map with content: .map { row ->\n",
      "Nextflow operator splitCsv with content: .splitCsv(header: true, sep: '\\t')\n",
      "Nextflow operator map with content: .map { row ->\n",
      "Nextflow operator transpose with content: .transpose()\n",
      "Nextflow operator map with content: .map { meta, bin -> [bin.getName(), bin, meta]}\n",
      "Nextflow operator join with content: .join(ch_bin_metrics, failOnDuplicate: true)\n",
      "Nextflow operator map with content: .map { bin_name, bin, meta, completeness, contamination -> [meta, bin, completeness, contamination] }\n",
      "Nextflow operator branch with content: .branch {\n",
      "Nextflow operator map with content: .map{\n",
      "Nextflow operator last with content: [ it.toString().split('/').last(), it ]\n",
      "Nextflow operator collect with content: .collect()\n",
      "Nextflow operator count with content: .count()\n",
      "Nextflow operator map with content: .map{it == 0 ? log.warn(\"No contigs passed GTDB-TK min. completeness filters. GTDB-TK summary will execute but results will be empty!\") : \"\"}\n",
      "Nextflow operator groupTuple with content: ch_filtered_bins.passed.groupTuple(),\n",
      "Nextflow operator collect with content: ch_filtered_bins.discarded.map{it[1]}.collect().ifEmpty([]),\n",
      "Nextflow operator collect with content: GTDBTK_CLASSIFYWF.out.summary.map{it[1]}.collect().ifEmpty([]),\n",
      "Error: Unclosed process block in file: ./git_repos\\mag\\subworkflows\\local\\gtdbtk.nf\n",
      "Process GUNC_QC in filepath ./git_repos\\mag\\subworkflows\\local\\gunc_qc.nf\n",
      "Nextflow operator mix with content: ch_versions.mix( GUNC_DOWNLOADDB.out.versions )\n",
      "Nextflow operator mix with content: ch_versions.mix( GUNC_RUN.out.versions )\n",
      "Nextflow operator map with content: .map{it[1]}\n",
      "Nextflow operator collectFile with content: .collectFile(name: \"gunc_summary.tsv\", keepHeader: true, storeDir: \"${params.outdir}/GenomeBinning/QC/\")\n",
      "Nextflow operator combine with content: .combine(ch_checkm_table, by: 0)\n",
      "Nextflow operator mix with content: ch_versions.mix( GUNC_MERGECHECKM.out.versions )\n",
      "Nextflow operator map with content: .map{it[1]}\n",
      "Nextflow operator collectFile with content: .collectFile(name: \"gunc_checkm_summary.tsv\", keepHeader: true, storeDir: \"${params.outdir}/GenomeBinning/QC/\")\n",
      "Error: Unclosed process block in file: ./git_repos\\mag\\subworkflows\\local\\gunc_qc.nf\n",
      "Process INPUT_CHECK in filepath ./git_repos\\mag\\subworkflows\\local\\input_check.nf\n",
      "Nextflow operator splitCsv with content: .splitCsv(header: true)\n",
      "Nextflow operator map with content: .map { row ->\n",
      "Nextflow operator map with content: .map { id, run, group, sr1, sr2, lr ->\n",
      "Nextflow operator map with content: .map { id, run, group, sr1, sr2, lr ->\n",
      "Nextflow operator ifEmpty with content: .ifEmpty { exit 1, \"Cannot find any reads matching: ${params.input}\\nNB: Path needs to be enclosed in quotes!\\nIf this is single-end data, please specify --single_end on the command line.\" }\n",
      "Nextflow operator map with content: .map { row ->\n",
      "Nextflow operator splitCsv with content: .splitCsv(header: true)\n",
      "Nextflow operator map with content: .map { row ->\n",
      "Nextflow operator map with content: .map { id, group, assembler, fasta ->\n",
      "Nextflow operator groupTuple with content: .groupTuple(by: 0)\n",
      "Nextflow operator map with content: .map { id, run, group, sr1, sr2, lr -> if( run.size() != run.unique().size() ) { { error(\"ERROR: input samplesheet contains duplicated sample or run IDs (within a sample)! Check samplesheet for sample id: ${id}\") } } }\n",
      "Nextflow operator map with content: .map { id, run, group, sr1, sr2, lr -> params.coassemble_group ? group : id }\n",
      "Nextflow operator unique with content: .unique()\n",
      "Nextflow operator toList with content: .toList()\n",
      "Nextflow operator map with content: .map { id, group, assembler, assembly -> params.coassemble_group ? group : id }\n",
      "Nextflow operator unique with content: .unique()\n",
      "Nextflow operator toList with content: .toList()\n",
      "Nextflow operator cross with content: ch_read_ids.cross(ch_assembly_ids)\n",
      "Nextflow operator map with content: .map { ids1, ids2 ->\n",
      "Error: Unclosed process block in file: ./git_repos\\mag\\subworkflows\\local\\input_check.nf\n",
      "Process TIARA in filepath ./git_repos\\mag\\subworkflows\\local\\tiara.nf\n",
      "Nextflow operator map with content: .map { meta, bins ->\n",
      "Nextflow operator map with content: .map { meta, unbins ->\n",
      "Nextflow operator mix with content: ch_tiara_input = bins.mix(unbins)\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(TIARA_TIARA.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(DASTOOL_FASTATOCONTIG2BIN_TIARA.out.versions.first())\n",
      "Nextflow operator combine with content: .combine(ch_tiara_input, by: 0)\n",
      "Nextflow operator map with content: .map { meta, contig2bin, bins ->\n",
      "Nextflow operator combine with content: .combine( TIARA_TIARA.out.classifications, by: 0)\n",
      "Nextflow operator map with content: .map { meta_join, meta, contig2bin, bins, classifications ->\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(TIARA_CLASSIFY.out.versions.first())\n",
      "Nextflow operator map with content: .map { meta, bins ->\n",
      "Nextflow operator map with content: .map { meta, bins ->\n",
      "Nextflow operator map with content: .map { meta, bins ->\n",
      "Nextflow operator map with content: .map { meta, bins ->\n",
      "Nextflow operator map with content: .map { meta, bins ->\n",
      "Nextflow operator map with content: .map { meta, bins ->\n",
      "Nextflow operator mix with content: .mix(ch_prokarya_bins)\n",
      "Nextflow operator mix with content: .mix(ch_bacteria_bins)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Nextflow operator mix with content: .mix(ch_archaea_bins)\n",
      "Nextflow operator mix with content: .mix(ch_organelle_bins)\n",
      "Nextflow operator mix with content: .mix(ch_unknown_bins)\n",
      "Nextflow operator filter with content: .filter { meta, bins ->\n",
      "Nextflow operator map with content: .map { meta, bins ->\n",
      "Nextflow operator filter with content: .filter { meta, bins ->\n",
      "Nextflow operator map with content: .map { meta, bins ->\n",
      "Nextflow operator map with content: .map { meta, classification ->\n",
      "Nextflow operator collect with content: .collect()\n",
      "Error: Unclosed process block in file: ./git_repos\\mag\\subworkflows\\local\\tiara.nf\n",
      "Process VIRUS_IDENTIFICATION in filepath ./git_repos\\mag\\subworkflows\\local\\virus_identification.nf\n",
      "Nextflow operator mix with content: ch_versions.mix( GENOMAD_DOWNLOAD.out.versions )\n",
      "Nextflow operator mix with content: ch_versions.mix( GENOMAD_ENDTOEND.out.versions )\n",
      "Error: Unclosed process block in file: ./git_repos\\mag\\subworkflows\\local\\virus_identification.nf\n",
      "Process FASTA_BINNING_CONCOCT in filepath ./git_repos\\mag\\subworkflows\\nf-core\\fasta_binning_concoct\\main.nf\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(CONCOCT_CUTUPFASTA.out.versions.first())\n",
      "Nextflow operator join with content: .join( ch_bam, failOnMismatch: true )\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(CONCOCT_CONCOCTCOVERAGETABLE.out.versions.first())\n",
      "Nextflow operator join with content: .join(CONCOCT_CUTUPFASTA.out.fasta, failOnMismatch: true)\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(CONCOCT_CONCOCT.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix( CONCOCT_MERGECUTUPCLUSTERING.out.versions.first())\n",
      "Nextflow operator join with content: .join(CONCOCT_MERGECUTUPCLUSTERING.out.csv, failOnMismatch: true)\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(CONCOCT_EXTRACTFASTABINS.out.versions.first())\n",
      "Error: Unclosed process block in file: ./git_repos\\mag\\subworkflows\\nf-core\\fasta_binning_concoct\\main.nf\n",
      "Process MAG in filepath ./git_repos\\mag\\workflows\\mag.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(MMSEQS_DATABASES.out.versions)\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(FASTQC_RAW.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(FASTP.out.versions.first())\n",
      "Nextflow operator branch with content: .branch {\n",
      "Nextflow operator mix with content: ch_short_reads_prepped = ch_short_reads_prepped.mix(ADAPTERREMOVAL_SE.out.singles_truncated, ADAPTERREMOVAL_PE.out.paired_truncated)\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(ADAPTERREMOVAL_PE.out.versions.first(), ADAPTERREMOVAL_SE.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(BOWTIE2_HOST_REMOVAL_ALIGN.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(BOWTIE2_PHIX_REMOVAL_ALIGN.out.versions.first())\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(FASTQC_TRIMMED.out.versions)\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator groupTuple with content: .groupTuple()\n",
      "Nextflow operator branch with content: .branch {\n",
      "Nextflow operator flatten with content: CAT_FASTQ ( ch_short_reads_forcat.cat.map { meta, reads -> [ meta, reads.flatten() ]} )\n",
      "Nextflow operator map with content: .map { meta, reads ->\n",
      "Nextflow operator flatten with content: def new_reads = meta.single_end ? reads[0] : reads.flatten()\n",
      "Nextflow operator mix with content: ch_short_reads = CAT_FASTQ.out.reads.mix(ch_short_reads_catskipped)\n",
      "Nextflow operator first with content: ch_versions    = ch_versions.mix(CAT_FASTQ.out.versions.first())\n",
      "Nextflow operator filter with content: ch_short_reads.filter { ! it[0].single_end }\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SEQTK_MERGEPE.out.versions.first())\n",
      "Nextflow operator filter with content: .mix(ch_short_reads.filter { it[0].single_end })\n",
      "Nextflow operator map with content: .map { [ [ id: sprintf(\"group%s\", it[0].group), group: it[0].group, single_end: true ], it[1] ] }\n",
      "Nextflow operator groupTuple with content: .groupTuple()\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BBMAP_BBNORM.out.versions)\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(NANOPLOT_RAW.out.versions.first())\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(PORECHOP.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(NANOLYSE.out.versions.first())\n",
      "Nextflow operator map with content: .map { meta, sr -> [ meta.id, meta, sr ] }\n",
      "Nextflow operator map with content: .map { meta, lr -> [ meta.id, meta, lr ] }\n",
      "Nextflow operator join with content: .join(ch_short_reads_tmp, by: 0)\n",
      "Nextflow operator map with content: .map { id, meta_lr, lr, meta_sr, sr -> [ meta_lr, lr, sr[0], sr[1] ] }  // should not occur for single-end, since SPAdes (hybrid) does not support single-end\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(FILTLONG.out.versions.first())\n",
      "Nextflow operator collect with content: .collect()\n",
      "Nextflow operator map with content: .map{\n",
      "Nextflow operator collect with content: .collect()\n",
      "Nextflow operator map with content: .map{\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(CENTRIFUGE.out.versions.first())\n",
      "Nextflow operator collect with content: .collect()\n",
      "Nextflow operator map with content: .map{\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(KRAKEN2.out.versions.first())\n",
      "Nextflow operator mix with content: ch_tax_classifications = CENTRIFUGE.out.results_for_krona.mix(KRAKEN2.out.results_for_krona)\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(KRONA.out.versions.first())\n",
      "Nextflow operator map with content: .map { meta, reads -> [ meta.group, meta, reads ] }\n",
      "Nextflow operator groupTuple with content: .groupTuple(by: 0)\n",
      "Nextflow operator map with content: .map { group, metas, reads ->\n",
      "Nextflow operator collect with content: if ( assemble_as_single ) [ meta, reads.collect { it }, [] ]\n",
      "Nextflow operator collect with content: else [ meta, reads.collect { it[0] }, reads.collect { it[1] } ]\n",
      "Nextflow operator map with content: .map { meta, reads -> [ meta.group, meta, reads ] }\n",
      "Nextflow operator groupTuple with content: .groupTuple(by: 0)\n",
      "Nextflow operator map with content: .map { group, metas, reads ->\n",
      "Nextflow operator collect with content: [ meta, reads.collect { it } ]\n",
      "Nextflow operator filter with content: .filter { it[0].single_end }\n",
      "Nextflow operator map with content: .map { meta, reads -> [ meta, [ reads ], [] ] }\n",
      "Nextflow operator mix with content: .mix (\n",
      "Nextflow operator filter with content: .filter { ! it[0].single_end }\n",
      "Nextflow operator map with content: .map { meta, reads -> [ meta, [ reads[0] ], [ reads[1] ] ] }\n",
      "Nextflow operator map with content: .map { meta, assembly ->\n",
      "Nextflow operator mix with content: ch_assemblies = ch_assemblies.mix(ch_megahit_assemblies)\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(MEGAHIT.out.versions.first())\n",
      "Nextflow operator map with content: ch_short_reads_spades = ch_short_reads_grouped.map { [ it[0], it[1] ] }\n",
      "Nextflow operator filter with content: .filter { it[0].single_end }\n",
      "Nextflow operator filter with content: .filter { ! it[0].single_end }\n",
      "Nextflow operator mix with content: .mix(POOL_PAIRED_READS.out.reads)\n",
      "Nextflow operator map with content: .map { meta, assembly ->\n",
      "Nextflow operator mix with content: ch_assemblies = ch_assemblies.mix(ch_spades_assemblies)\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SPADES.out.versions.first())\n",
      "Nextflow operator map with content: .map { meta, reads -> [ meta.id, meta, reads ] }\n",
      "Nextflow operator map with content: .map { meta, reads -> [ meta.id, meta, reads ] }\n",
      "Nextflow operator combine with content: .combine(ch_short_reads_spades_tmp, by: 0)\n",
      "Nextflow operator map with content: .map { id, meta_long, long_reads, meta_short, short_reads -> [ meta_short, long_reads, short_reads ] }\n",
      "Nextflow operator map with content: .map { meta, assembly ->\n",
      "Nextflow operator mix with content: ch_assemblies = ch_assemblies.mix(ch_spadeshybrid_assemblies)\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SPADESHYBRID.out.versions.first())\n",
      "Nextflow operator branch with content: .branch { meta, assembly ->\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GUNZIP_ASSEMBLIES.out.versions)\n",
      "Nextflow operator mix with content: ch_assemblies = ch_assemblies.mix(ch_assemblies_split.ungzip, GUNZIP_ASSEMBLIES.out.gunzip)\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(QUAST.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(PRODIGAL.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(VIRUS_IDENTIFICATION.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(BINNING_PREPARATION.out.bowtie2_version.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(ANCIENT_DNA_ASSEMBLY_VALIDATION.out.versions.first())\n",
      "Nextflow operator join with content: .join(ANCIENT_DNA_ASSEMBLY_VALIDATION.out.contigs_recalled)\n",
      "Nextflow operator map with content: .map { it -> [ it[0], it[4], it[2], it[3] ] }, // [meta, contigs_recalled, bam, bais]\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BINNING.out.versions)\n",
      "Nextflow operator mix with content: ch_versions               = ch_versions.mix(DOMAIN_CLASSIFICATION.out.versions)\n",
      "Nextflow operator map with content: .map { meta, bins ->\n",
      "Nextflow operator map with content: .map { meta, bins ->\n",
      "Nextflow operator map with content: .map { meta, bins ->\n",
      "Nextflow operator map with content: .map { meta, bins ->\n",
      "Nextflow operator filter with content: .filter { meta, bins ->\n",
      "Nextflow operator filter with content: .filter { meta, bins ->\n",
      "Nextflow operator map with content: .map{ meta, contigs, bam, bai -> [ meta, contigs ] }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BINNING_REFINEMENT.out.versions)\n",
      "Nextflow operator mix with content: ch_input_for_postbinning_bins_unbins = ch_binning_results_bins.mix(ch_binning_results_unbins)\n",
      "Nextflow operator mix with content: ch_input_for_postbinning_bins_unbins = ch_refined_bins.mix(ch_refined_unbins)\n",
      "Nextflow operator mix with content: ch_all_bins = ch_binning_results_bins.mix(ch_refined_bins)\n",
      "Nextflow operator mix with content: ch_input_for_postbinning_bins_unbins = ch_all_bins.mix(ch_binning_results_unbins).mix(ch_refined_unbins)\n",
      "Nextflow operator mix with content: ch_input_for_postbinning_bins_unbins = ch_binning_results_bins.mix(ch_binning_results_unbins)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(DEPTHS.out.versions)\n",
      "Nextflow operator transpose with content: ch_input_bins_for_qc = ch_input_for_postbinning_bins_unbins.transpose()\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(BUSCO_QC.out.versions.first())\n",
      "Nextflow operator splitCsv with content: .splitCsv(sep: '\\t')\n",
      "Nextflow operator map with content: .map { bin, error -> if (!bin.contains(\".unbinned.\")) busco_failed_bins[bin] = error }\n",
      "Nextflow operator filter with content: .filter { meta, bins ->\n",
      "Nextflow operator groupTuple with content: ch_input_bins_for_checkm.groupTuple(),\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(CHECKM_QC.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix( GUNC_QC.out.versions )\n",
      "Nextflow operator filter with content: .filter { meta, bins ->\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix( GUNC_QC.out.versions )\n",
      "Nextflow operator groupTuple with content: .groupTuple()\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator flatten with content: def new_bins = bins.flatten()\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(QUAST_BINS.out.versions.first())\n",
      "Nextflow operator collectFile with content: .collectFile(keepHeader: true) {\n",
      "Nextflow operator collect with content: QUAST_BINS_SUMMARY ( ch_quast_bin_summary.collect() )\n",
      "Nextflow operator collectFile with content: .collectFile(keepHeader: true) {\n",
      "Nextflow operator collect with content: ch_cat_summary.collect()\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(CAT.out.versions.first())\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(CAT_SUMMARY.out.versions)\n",
      "Nextflow operator filter with content: .filter { meta, bins ->\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(GTDBTK.out.versions.first())\n",
      "Nextflow operator ifEmpty with content: ch_busco_summary.ifEmpty([]),\n",
      "Nextflow operator ifEmpty with content: ch_checkm_summary.ifEmpty([]),\n",
      "Nextflow operator ifEmpty with content: ch_quast_bins_summary.ifEmpty([]),\n",
      "Nextflow operator ifEmpty with content: ch_gtdbtk_summary.ifEmpty([]),\n",
      "Nextflow operator ifEmpty with content: ch_cat_global_summary.ifEmpty([])\n",
      "Nextflow operator transpose with content: ch_bins_for_prokka = ch_input_for_postbinning_bins_unbins.transpose()\n",
      "Nextflow operator map with content: .map { meta, bin ->\n",
      "Nextflow operator filter with content: .filter { meta, bin ->\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(PROKKA.out.versions.first())\n",
      "Nextflow operator transpose with content: ch_bins_for_metaeuk = ch_input_for_postbinning_bins_unbins.transpose()\n",
      "Nextflow operator filter with content: .filter { meta, bin ->\n",
      "Nextflow operator map with content: .map { meta, bin ->\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(METAEUK_EASYPREDICT.out.versions)\n",
      "Nextflow operator collectFile with content: ch_versions.unique().collectFile(name: 'collated_versions.yml')\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_workflow_summary.collectFile(name: 'workflow_summary_mqc.yaml'))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(CUSTOM_DUMPSOFTWAREVERSIONS.out.mqc_yml.collect())\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_methods_description.collectFile(name: 'methods_description_mqc.yaml'))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(FASTQC_RAW.out.zip.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(ADAPTERREMOVAL_PE.out.settings.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(ADAPTERREMOVAL_SE.out.settings.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(FASTP.out.json.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(FASTQC_TRIMMED.out.zip.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(BOWTIE2_HOST_REMOVAL_ALIGN.out.log.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(BOWTIE2_PHIX_REMOVAL_ALIGN.out.log.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(CENTRIFUGE.out.kreport.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(KRAKEN2.out.report.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(QUAST.out.report.collect().ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(QUAST_BINS.out.dir.collect().ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(BINNING_PREPARATION.out.bowtie2_assembly_multiqc.collect().ifEmpty([]))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(PROKKA.out.txt.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(BUSCO_QC.out.multiqc.collect().ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files.collect(),\n",
      "Nextflow operator toList with content: ch_multiqc_config.toList(),\n",
      "Nextflow operator toList with content: ch_multiqc_custom_config.toList(),\n",
      "Nextflow operator toList with content: ch_multiqc_logo.toList()\n",
      "Nextflow operator toList with content: multiqc_report = MULTIQC.out.report.toList()\n",
      "Error: Unclosed process block in file: ./git_repos\\mag\\workflows\\mag.nf\n",
      "Process NFCORE_MARSSEQ in filepath ./git_repos\\marsseq\\main.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\marsseq\\main.nf\n",
      "Process ALIGN_READS in filepath ./git_repos\\marsseq\\subworkflows\\local\\align_reads.nf\n",
      "Nextflow operator map with content: .map { meta, reads -> [ [ \"id\": meta.id, \"single_end\": true, \"filename\": reads.baseName ], reads ]}\n",
      "Nextflow operator map with content: .map { meta, reads -> [ meta, index ] }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BOWTIE2_ALIGN.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(QC_ALIGNED.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(CUT_SAM.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\marsseq\\subworkflows\\local\\align_reads.nf\n",
      "Process BUILD_REFERENCES in filepath ./git_repos\\marsseq\\subworkflows\\local\\build_references.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(DOWNLOAD_FASTA.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(DOWNLOAD_GTF.out.versions)\n",
      "Nextflow operator map with content: ch_fasta = GUNZIP_FASTA ( DOWNLOAD_FASTA.out.file ).gunzip.map { it[1] }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GUNZIP_FASTA.out.versions)\n",
      "Nextflow operator map with content: ch_gtf = GUNZIP_GTF ( DOWNLOAD_GTF.out.file ).gunzip.map { it[1] }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GUNZIP_GTF.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ERCC_CREATE.out.versions)\n",
      "Nextflow operator merge with content: ch_fastas = ch_fasta.merge(ERCC_CREATE.out.fasta)\n",
      "Nextflow operator map with content: .map{ it -> [ [\"id\": \"${fasta - '.fa'}\"], it ] }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(CAT_FASTA.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BOWTIE2_BUILD.out.versions)\n",
      "Nextflow operator map with content: ch_star_index = STAR_GENOMEGENERATE( ch_genome.map{ meta, fasta -> fasta }, ch_gtf.map{ meta, gtf -> gtf } ).index\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(STAR_GENOMEGENERATE.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\marsseq\\subworkflows\\local\\build_references.nf\n",
      "Process DEMULTIPLEX_READS in filepath ./git_repos\\marsseq\\subworkflows\\local\\demultiplex_reads.nf\n",
      "Nextflow operator splitCsv with content: .splitCsv( header:true, sep:'\\t' )\n",
      "Nextflow operator map with content: .map { row -> [ row['Amp_batch_ID'], row['Pool_barcode'] ] }\n",
      "Nextflow operator combine with content: .combine(reads)\n",
      "Nextflow operator map with content: .map { amp_batch, pool_barcode, meta, reads -> [\n",
      "Nextflow operator combine with content: .combine(wells_cells)\n",
      "Nextflow operator combine with content: .combine(gene_intervals)\n",
      "Nextflow operator combine with content: .combine(spike_seq)\n",
      "Nextflow operator combine with content: .combine(oligos)\n",
      "Nextflow operator combine with content: .combine(amp_batches)\n",
      "Nextflow operator combine with content: .combine(seq_batches)\n",
      "Nextflow operator combine with content: .combine(wells_cells)\n",
      "Nextflow operator combine with content: .combine(spike_concentrations)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(DEMULTIPLEX.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(QC_BATCH.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\marsseq\\subworkflows\\local\\demultiplex_reads.nf\n",
      "Process INPUT_CHECK in filepath ./git_repos\\marsseq\\subworkflows\\local\\input_check.nf\n",
      "Nextflow operator splitCsv with content: .splitCsv ( header:true, sep:',' )\n",
      "Nextflow operator map with content: .map { create_fastq_channel(it) }\n",
      "Nextflow operator set with content: .set { reads }\n",
      "Error: Unclosed process block in file: ./git_repos\\marsseq\\subworkflows\\local\\input_check.nf\n",
      "Process LABEL_READS in filepath ./git_repos\\marsseq\\subworkflows\\local\\label_reads.nf\n",
      "Nextflow operator collate with content: .map { meta, reads -> return [ meta, reads.sort().collate(2) ] }\n",
      "Nextflow operator transpose with content: .transpose()\n",
      "Nextflow operator combine with content: .combine(oligos)\n",
      "Nextflow operator combine with content: .combine(amp_batches)\n",
      "Nextflow operator combine with content: .combine(seq_batches)\n",
      "Error: Unclosed process block in file: ./git_repos\\marsseq\\subworkflows\\local\\label_reads.nf\n",
      "Process PREPARE_PIPELINE in filepath ./git_repos\\marsseq\\subworkflows\\local\\prepare_pipeline.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(PREPARE.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(FASTP_SPLIT.out.versions)\n",
      "Nextflow operator last with content: if ( ch_reads.last().length() % 2 != 0 ) {\n",
      "Error: Unclosed process block in file: ./git_repos\\marsseq\\subworkflows\\local\\prepare_pipeline.nf\n",
      "Process VELOCITY in filepath ./git_repos\\marsseq\\subworkflows\\local\\velocity.nf\n",
      "Nextflow operator first with content: ch_folder = reads.map { meta, reads -> [ meta, reads.first().Parent ] }\n",
      "Nextflow operator map with content: ch_index = reads.map { meta, reads -> [ meta, index ] }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(VELOCITY_CONVERT.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(VELOCITY_WHITELIST.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(VELOCITY_TRIM.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(VELOCITY_STARSOLO.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\marsseq\\subworkflows\\local\\velocity.nf\n",
      "Process BUILD_REFERENCES in filepath ./git_repos\\marsseq\\workflows\\build_references.nf\n",
      "Nextflow operator map with content: .map { meta, fasta -> fasta }\n",
      "Nextflow operator merge with content: ch_fastas = ch_fasta.merge(ch_ercc)\n",
      "Nextflow operator map with content: .map{ it -> [ [\"id\": \"${fasta - '.fa'}\"], it ] }\n",
      "Nextflow operator map with content: STAR_GENOMEGENERATE( ch_genome.map{ meta, fasta -> fasta }, ch_gtf.map{ meta, gtf -> gtf } )\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(STAR_GENOMEGENERATE.out.versions)\n",
      "Nextflow operator mix with content: .mix(DOWNLOAD_FASTA.out.versions)\n",
      "Nextflow operator mix with content: .mix(GUNZIP_FASTA.out.versions)\n",
      "Nextflow operator mix with content: .mix(GUNZIP_GTF.out.versions)\n",
      "Nextflow operator mix with content: .mix(ERCC_CREATE.out.versions)\n",
      "Nextflow operator mix with content: .mix(CAT_FASTA.out.versions)\n",
      "Nextflow operator mix with content: .mix(BOWTIE2_BUILD.out.versions)\n",
      "Nextflow operator unique with content: .unique()\n",
      "Nextflow operator collectFile with content: .collectFile(name: 'collated_versions.yml')\n",
      "Error: Unclosed process block in file: ./git_repos\\marsseq\\workflows\\build_references.nf\n",
      "Process MARSSEQ in filepath ./git_repos\\marsseq\\workflows\\marsseq.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(INPUT_CHECK.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(FASTQC.out.versions)\n",
      "Nextflow operator map with content: INPUT_CHECK.out.reads.map { it[0].amp_batches },\n",
      "Nextflow operator map with content: INPUT_CHECK.out.reads.map { it[0].seq_batches },\n",
      "Nextflow operator map with content: INPUT_CHECK.out.reads.map { it[0].well_cells },\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(PREPARE_PIPELINE.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(LABEL_READS.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ALIGN_READS.out.versions)\n",
      "Nextflow operator map with content: .map { meta, sam -> [ meta.id, sam ] }\n",
      "Nextflow operator groupTuple with content: .groupTuple(by: [0], sort: { it.name })\n",
      "Nextflow operator map with content: .map { batch, sams -> [ [ \"id\": batch ], sams ] }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(MERGE_READS.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(DEMULTIPLEX_READS.out.versions)\n",
      "Nextflow operator groupTuple with content: DEMULTIPLEX_READS.out.qc_rd.map { meta, rds -> [ [\"id\": meta.id], rds ] }.groupTuple(),\n",
      "Nextflow operator groupTuple with content: DEMULTIPLEX_READS.out.qc_pdf.map { meta, pdf -> [ [\"id\": meta.id], pdf ] }.groupTuple(),\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(QC_REPORT.out.versions)\n",
      "Nextflow operator map with content: INPUT_CHECK.out.reads.map { it[0].amp_batches },\n",
      "Nextflow operator map with content: INPUT_CHECK.out.reads.map { it[0].well_cells },\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(VELOCITY.out.versions)\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(VELOCITY.out.catadapt_multiqc.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(VELOCITY.out.star_multiqc.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collectFile with content: ch_versions.unique().collectFile(name: 'collated_versions.yml')\n",
      "Nextflow operator collect with content: .mix(PREPARE_PIPELINE.out.fastp_multiqc.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: .mix(ALIGN_READS.out.bowtie2_multiqc.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collectFile with content: .mix(ch_workflow_summary.collectFile(name: 'workflow_summary_mqc.yaml'))\n",
      "Nextflow operator collectFile with content: .mix(ch_methods_description.collectFile(name: 'methods_description_mqc.yaml'))\n",
      "Nextflow operator collect with content: .mix(CUSTOM_DUMPSOFTWAREVERSIONS.out.mqc_yml.collect())\n",
      "Nextflow operator collect with content: .mix(FASTQC.out.zip.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files.collect(),\n",
      "Nextflow operator toList with content: ch_multiqc_config.toList(),\n",
      "Nextflow operator toList with content: ch_multiqc_custom_config.toList(),\n",
      "Nextflow operator toList with content: ch_multiqc_logo.toList()\n",
      "Nextflow operator toList with content: multiqc_report = MULTIQC.out.report.toList()\n",
      "Error: Unclosed process block in file: ./git_repos\\marsseq\\workflows\\marsseq.nf\n",
      "Process NFCORE_MCMICRO in filepath ./git_repos\\mcmicro\\main.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\mcmicro\\main.nf\n",
      "Process INPUT_CHECK in filepath ./git_repos\\mcmicro\\subworkflows\\local\\input_check.nf\n",
      "Nextflow operator splitCsv with content: .splitCsv ( header:true, sep:',' )\n",
      "Nextflow operator map with content: .map { create_fastq_channel(it) }\n",
      "Nextflow operator groupTuple with content: .groupTuple(by: [0])\n",
      "Nextflow operator flatten with content: .map { meta, reads -> [ meta, reads.flatten() ] }\n",
      "Nextflow operator set with content: .set { images_merged }\n",
      "Nextflow operator splitCsv with content: .splitCsv ( header:true, sep:',' )\n",
      "Nextflow operator map with content: .map { create_fastq_channel(it) }\n",
      "Nextflow operator set with content: .set { input }\n",
      "Error: Unclosed process block in file: ./git_repos\\mcmicro\\subworkflows\\local\\input_check.nf\n",
      "Process MCMICRO in filepath ./git_repos\\mcmicro\\workflows\\mcmicro.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ASHLAR.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(DEEPCELL_MESMER.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(MCQUANT.out.versions)\n",
      "Nextflow operator collectFile with content: ch_versions.unique().collectFile(name: 'collated_versions.yml')\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_workflow_summary.collectFile(name: 'workflow_summary_mqc.yaml'))\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_methods_description.collectFile(name: 'methods_description_mqc.yaml'))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(CUSTOM_DUMPSOFTWAREVERSIONS.out.mqc_yml.collect())\n",
      "Nextflow operator collect with content: ch_multiqc_files.collect(),\n",
      "Nextflow operator toList with content: ch_multiqc_config.toList(),\n",
      "Nextflow operator toList with content: ch_multiqc_custom_config.toList(),\n",
      "Nextflow operator toList with content: ch_multiqc_logo.toList()\n",
      "Nextflow operator toList with content: multiqc_report = MULTIQC.out.report.toList()\n",
      "Error: Unclosed process block in file: ./git_repos\\mcmicro\\workflows\\mcmicro.nf\n",
      "Process NFCORE_METABOIGNITER in filepath ./git_repos\\metaboigniter\\main.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\metaboigniter\\main.nf\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Process ANNOTATION in filepath ./git_repos\\metaboigniter\\subworkflows\\local\\annotation.nf\n",
      "Nextflow operator map with content: OPENMS_METABOLITEADDUCTDECHARGER(quantification_information.map{it[0,1]})\n",
      "Nextflow operator join with content: .join(quantification_information.map{it[0,2]},by:[0])\n",
      "Nextflow operator first with content: ch_versions       = ch_versions.mix(OPENMS_METABOLITEADDUCTDECHARGER.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions       = ch_versions.mix(PYOPENMS_C13DETECTION.out.versions.first())\n",
      "Error: Unclosed process block in file: ./git_repos\\metaboigniter\\subworkflows\\local\\annotation.nf\n",
      "Process IDENTIFICATION in filepath ./git_repos\\metaboigniter\\subworkflows\\local\\identification.nf\n",
      "Nextflow operator collect with content: consensusxml_data.combine(mzml_files.filter{meta,file->meta.level == \"MS2\" | meta.level == \"MS12\"}.collect{it[1]}\n",
      "Nextflow operator map with content: .map { files ->\n",
      "Nextflow operator toList with content: }.toList()) | PYOPENMS_MSMAPPING\n",
      "Nextflow operator first with content: ch_versions       = ch_versions.mix(PYOPENMS_MSMAPPING.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions       = ch_versions.mix(OPENMS_FILEFILTER.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions       = ch_versions.mix(PYOPENMS_SPLITCONSENSUS.out.versions.first())\n",
      "Nextflow operator flatten with content: filtered_consensus = PYOPENMS_SPLITCONSENSUS.out.consensusxml.map{it[1]}.flatten().map{ file ->\n",
      "Nextflow operator filter with content: mzml_files.filter{ meta, file -> meta.level == \"MS2\" || meta.level == \"MS12\" }.map{tuple ->\n",
      "Nextflow operator collect with content: [tuple]}.collect().map { tuples ->\n",
      "Error: Unclosed process block in file: ./git_repos\\metaboigniter\\subworkflows\\local\\identification.nf\n",
      "Process LINKER in filepath ./git_repos\\metaboigniter\\subworkflows\\local\\linker.nf\n",
      "Nextflow operator map with content: quantification_information.map{it[0,1]} | PYOPENMS_EXTRACTFEATUREMZ\n",
      "Nextflow operator first with content: ch_versions       = ch_versions.mix(PYOPENMS_EXTRACTFEATUREMZ.out.versions.first())\n",
      "Nextflow operator collect with content: PYOPENMS_CONCTSV_MZ(Channel.of([id:\"merged_mz_range\"]).combine(PYOPENMS_EXTRACTFEATUREMZ.out.tsv.collect{it[1]}.toList()),\"tsv\",\"tsv\")\n",
      "Nextflow operator map with content: PYOPENMS_SPLITFEATUREXML(quantification_information.map{it[0,1]},\n",
      "Nextflow operator map with content: PYOPENMS_CALCULATEBOUNDRIES.out.tsv.map{it[1]}\n",
      "Nextflow operator collect with content: PYOPENMS_SPLITFEATUREXML.out.featurexml.collect{it[1]}.flatten().map{file ->\n",
      "Nextflow operator groupTuple with content: [[id:\"part\"+n1],file]}.groupTuple(by: 0).map{meta,files->[[meta][0], files.sort { a, b ->\n",
      "Nextflow operator combine with content: to_be_merged = Channel.of([id:\"Linked_data\"]).combine(\n",
      "Nextflow operator collect with content: OPENMS_FEATURELINKERUNLABELEDKDPARTS.out.consensusxml.map{it[1]}.collect()\n",
      "Nextflow operator map with content: .map { files ->\n",
      "Nextflow operator toList with content: }.toList())\n",
      "Nextflow operator map with content: consensusxml=   PYOPENMS_FILEMERGER.out.consensusxml.map{meta,consensusxml->[[id:consensusxml.baseName],consensusxml]}\n",
      "Nextflow operator collect with content: Channel.of([id:\"Linked_data\"]).combine(quantification_information.collect{it[1]}.map { files ->\n",
      "Nextflow operator toList with content: }.toList()) | OPENMS_FEATURELINKERUNLABELEDKD\n",
      "Nextflow operator map with content: consensusxml=OPENMS_FEATURELINKERUNLABELEDKD.out.consensusxml.map{meta,consensusxml->[[id:consensusxml.baseName],consensusxml]}\n",
      "Nextflow operator first with content: ch_versions       = ch_versions.mix(OPENMS_FEATURELINKERUNLABELEDKD.out.versions.first())\n",
      "Process MERGEMS1MS2 in filepath ./git_repos\\metaboigniter\\subworkflows\\local\\mergems1ms2.nf\n",
      "Nextflow operator filter with content: mzml_files.filter{meta,file->meta.level == \"MS2\"} | OPENMS_FILEFILTERMS2\n",
      "Nextflow operator first with content: ch_versions       = ch_versions.mix(OPENMS_FILEFILTERMS2.out.versions.first())\n",
      "Nextflow operator combine with content: mzml_files.filter{meta,file->meta.level != \"MS2\"}.combine(OPENMS_FILEFILTERMS2.out.mzml.map{it[1]}).groupTuple(by:[0,1]) | OPENMS_FILEMERGER\n",
      "Nextflow operator first with content: ch_versions       = ch_versions.mix(OPENMS_FILEMERGER.out.versions.first())\n",
      "Error: Unclosed process block in file: ./git_repos\\metaboigniter\\subworkflows\\local\\mergems1ms2.nf\n",
      "Process MS2QUERY in filepath ./git_repos\\metaboigniter\\subworkflows\\local\\ms2query.nf\n",
      "Nextflow operator collect with content: MS2QUERY_MODELDOWNLOADER.out.model_files.collect() |  MS2QUERY_CHECKMODELFILES\n",
      "Nextflow operator first with content: ch_versions       = ch_versions.mix(MS2QUERY_CHECKMODELFILES.out.versions.first())\n",
      "Nextflow operator collect with content: .collect() | MS2QUERY_CHECKMODELFILES\n",
      "Nextflow operator first with content: ch_versions       = ch_versions.mix(MS2QUERY_CHECKMODELFILES.out.versions.first())\n",
      "Nextflow operator mix with content: all_model_files=MS2QUERY_CHECKMODELFILES.out.sqlite.mix(\n",
      "Nextflow operator collect with content: ).collect().toList()\n",
      "Nextflow operator first with content: ch_versions       = ch_versions.mix(PYOPENMS_SPLITMGF.out.versions.first())\n",
      "Nextflow operator collect with content: mgf_channel = PYOPENMS_SPLITMGF.out.mgf.flatMap{id, values -> values.collect { [id, it]}}\n",
      "Nextflow operator combine with content: .map{it->[[id:it.baseName],it]}.combine(all_model_files) | MS2QUERY_MODELTRAIN\n",
      "Nextflow operator mix with content: all_model_files=MS2QUERY_MODELTRAIN.out.sqlite.mix(\n",
      "Nextflow operator collect with content: ).collect().toList()\n",
      "Nextflow operator first with content: ch_versions       = ch_versions.mix(MS2QUERY_MODELTRAIN.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions       = ch_versions.mix(MS2QUERY_SEARCH.out.versions.first())\n",
      "Nextflow operator map with content: output = MS2QUERY_SEARCH.out.csv.map{meta,file->[[id:\"ms2query_${meta.id}\"],file]}\n",
      "Nextflow operator groupTuple with content: output = output.groupTuple(by:0)\n",
      "Nextflow operator first with content: ch_versions       = ch_versions.mix(PYOPENMS_CONCTSVMS2QUERY.out.versions.first())\n",
      "Error: Unclosed process block in file: ./git_repos\\metaboigniter\\subworkflows\\local\\ms2query.nf\n",
      "Process MS2QUERY_MODELDOWNLOADER in filepath ./git_repos\\metaboigniter\\subworkflows\\local\\ms2query_modeldownloader.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\metaboigniter\\subworkflows\\local\\ms2query_modeldownloader.nf\n",
      "Process MS2QUERY in filepath ./git_repos\\metaboigniter\\subworkflows\\local\\ms2query_search.nf\n",
      "Nextflow operator collect with content: MS2QUERY_MODELDOWNLOADER.out.model_files.collect() |  MS2QUERY_CHECKMODELFILES\n",
      "Nextflow operator first with content: ch_versions       = ch_versions.mix(MS2QUERY_CHECKMODELFILES.out.versions.first())\n",
      "Nextflow operator collect with content: .collect() | MS2QUERY_CHECKMODELFILES\n",
      "Nextflow operator first with content: ch_versions       = ch_versions.mix(MS2QUERY_CHECKMODELFILES.out.versions.first())\n",
      "Nextflow operator mix with content: all_model_files=MS2QUERY_CHECKMODELFILES.out.sqlite.mix(\n",
      "Nextflow operator collect with content: ).collect().toList()\n",
      "Nextflow operator view with content: mgf_channel.view()\n",
      "Nextflow operator first with content: ch_versions       = ch_versions.mix(PYOPENMS_SPLITMGF.out.versions.first())\n",
      "Nextflow operator combine with content: .map{it->[[id:it.baseName],it]}.combine(all_model_files) | MS2QUERY_MODELTRAIN\n",
      "Nextflow operator mix with content: all_model_files=MS2QUERY_MODELTRAIN.out.sqlite.mix(\n",
      "Nextflow operator collect with content: ).collect().toList()\n",
      "Nextflow operator first with content: ch_versions       = ch_versions.mix(MS2QUERY_MODELTRAIN.out.versions.first())\n",
      "Nextflow operator view with content: PYOPENMS_SPLITMGF.out.mgf.view()\n",
      "Nextflow operator combine with content: PYOPENMS_SPLITMGF.out.mgf.map{it[1]}.flatten().map{it->[[id:it.baseName],it]}.combine(all_model_files) | MS2QUERY_SEARCH\n",
      "Nextflow operator first with content: ch_versions       = ch_versions.mix(MS2QUERY_SEARCH.out.versions.first())\n",
      "Nextflow operator collect with content: OPENMS_GNPSEXPORT.out.mgf.map{it[0]}.combine(MS2QUERY_SEARCH.out.csv.collect{it[1]}.toList()) | PYOPENMS_CONCTSV\n",
      "Nextflow operator first with content: ch_versions       = ch_versions.mix(PYOPENMS_CONCTSV.out.versions.first())\n",
      "Nextflow operator combine with content: PYOPENMS_EXPORT.out.tsv.combine(PYOPENMS_CONCTSV.out.csv.map{it[1]}) | MS2QUERY_EXPORT\n",
      "Nextflow operator first with content: ch_versions       = ch_versions.mix(MS2QUERY_EXPORT.out.versions.first())\n",
      "Error: Unclosed process block in file: ./git_repos\\metaboigniter\\subworkflows\\local\\ms2query_search.nf\n",
      "Process QUANTIFICATION in filepath ./git_repos\\metaboigniter\\subworkflows\\local\\quantification.nf\n",
      "Nextflow operator first with content: ch_versions       = ch_versions.mix(OPENMS_PEAKPICKERHIRES.out.versions.first())\n",
      "Nextflow operator collect with content: mzml_files.collect{it[0]}.toList().combine(mzml_files.collect{it[1]}.toList()) | OPENMS_MAPALIGNERPOSECLUSTERINGMZML\n",
      "Nextflow operator map with content: mzml_files = mzml_files.map{meta,mzml->\n",
      "Nextflow operator flatten with content: [[map_id:idd],meta,mzml]}.join(OPENMS_MAPALIGNERPOSECLUSTERINGMZML.out.mzml.map{it[1]}.flatten().map{mzml ->\n",
      "Nextflow operator map with content: [[map_id:idd], mzml]}).map{it[1,3]}\n",
      "Nextflow operator first with content: ch_versions       = ch_versions.mix(OPENMS_MAPALIGNERPOSECLUSTERINGMZML.out.versions.first())\n",
      "Nextflow operator filter with content: quantificaiton_data=mzml_files.filter{meta,file->meta.level == \"MS1\" | meta.level == \"MS12\"}\n",
      "Nextflow operator first with content: ch_versions       = ch_versions.mix(OPENMS_FEATUREFINDERMETABO.out.versions.first())\n",
      "Nextflow operator collect with content: quantified_features.collect{it[0]}.toList().combine(quantified_features.collect{it[1]}.toList()) | OPENMS_MAPALIGNERPOSECLUSTERING\n",
      "Nextflow operator first with content: ch_versions       = ch_versions.mix(OPENMS_MAPALIGNERPOSECLUSTERING.out.versions.first())\n",
      "Nextflow operator map with content: combined_data =quantified_features.map{meta,featurexml->\n",
      "Nextflow operator flatten with content: .join(OPENMS_MAPALIGNERPOSECLUSTERING.out.featurexml.map{it[1]}.flatten().map{featurexml ->\n",
      "Nextflow operator flatten with content: .join(OPENMS_MAPALIGNERPOSECLUSTERING.out.trafoxml.map{it[1]}.flatten().map{trafoxml ->\n",
      "Nextflow operator join with content: .join(quantificaiton_data.map{meta,mzml->\n",
      "Nextflow operator map with content: [[map_id:idd],mzml]}).map{it[1..4]}\n",
      "Nextflow operator map with content: combined_data.map{it[0,2,3]} | OPENMS_MAPRTTRANSFORMER\n",
      "Nextflow operator first with content: ch_versions       = ch_versions.mix(OPENMS_MAPRTTRANSFORMER.out.versions.first())\n",
      "Nextflow operator map with content: quantified_features=combined_data.map{it[0,1]}\n",
      "Error: Unclosed process block in file: ./git_repos\\metaboigniter\\subworkflows\\local\\quantification.nf\n",
      "Process REQUANTIFICATION in filepath ./git_repos\\metaboigniter\\subworkflows\\local\\requantification.nf\n",
      "Nextflow operator combine with content: quantified_features.combine(data_split.map{it[1]}) |  PYOPENMS_RELOADMAPS\n",
      "Nextflow operator first with content: ch_versions       = ch_versions.mix(PYOPENMS_RELOADMAPS.out.versions.first())\n",
      "Nextflow operator map with content: data_split.map{it[0,2]} | PYOPENMS_LIBBUILDER\n",
      "Nextflow operator first with content: ch_versions       = ch_versions.mix(PYOPENMS_LIBBUILDER.out.versions.first())\n",
      "Nextflow operator combine with content: quantificaiton_data.combine(PYOPENMS_LIBBUILDER.out.tsv.map{it[1]}) | OPENMS_FEATUREFINDERMETABOIDENT\n",
      "Nextflow operator first with content: ch_versions       = ch_versions.mix(OPENMS_FEATUREFINDERMETABOIDENT.out.versions.first())\n",
      "Nextflow operator join with content: quantified_features_complete.join(OPENMS_FEATUREFINDERMETABOIDENT.out.featurexml,by:[0]) | PYOPENMS_MERGE\n",
      "Nextflow operator first with content: ch_versions       = ch_versions.mix(PYOPENMS_MERGE.out.versions.first())\n",
      "Error: Unclosed process block in file: ./git_repos\\metaboigniter\\subworkflows\\local\\requantification.nf\n",
      "Process SIRIUS in filepath ./git_repos\\metaboigniter\\subworkflows\\local\\sirius.nf\n",
      "Nextflow operator first with content: ch_versions       = ch_versions.mix(PYOPENMS_SPLITMS.out.versions.first())\n",
      "Nextflow operator collect with content: ms_channel = PYOPENMS_SPLITMS.out.ms.flatMap{id, values -> values.collect { [id, it]}}\n",
      "Nextflow operator first with content: ch_versions       = ch_versions.mix(SIRIUS_SEARCH.out.versions.first())\n",
      "Nextflow operator map with content: output_sirius = output_sirius.map{meta,file->[[id:\"sirius_${meta.id}\"],file]}\n",
      "Nextflow operator map with content: output_fingerid = output_fingerid.map{meta,file->[[id:\"fingerid_${meta.id}\"],file]}\n",
      "Nextflow operator groupTuple with content: output_sirius = SIRIUS_SEARCH.out.sirius_tsv.groupTuple(by:0)\n",
      "Nextflow operator map with content: output_sirius = output_sirius.map{meta,file->[[id:\"sirius_${meta.id}\"],file]}\n",
      "Nextflow operator groupTuple with content: output_fingerid = SIRIUS_SEARCH.out.fingerid_tsv.groupTuple(by:0)\n",
      "Nextflow operator map with content: output_fingerid = output_fingerid.map{meta,file->[[id:\"fingerid_${meta.id}\"],file]}\n",
      "Nextflow operator first with content: ch_versions       = ch_versions.mix(PYOPENMS_CONCTSVSIRIUS.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions       = ch_versions.mix(PYOPENMS_CONCTSVFINGERID.out.versions.first())\n",
      "Error: Unclosed process block in file: ./git_repos\\metaboigniter\\subworkflows\\local\\sirius.nf\n",
      "Process METABOIGNITER in filepath ./git_repos\\metaboigniter\\workflows\\metaboigniter.nf\n",
      "Nextflow operator map with content: .fromSamplesheet(\"input\").map{\n",
      "Nextflow operator join with content: quantification_information = quantified_features.join(quantificaiton_data,by:[0])\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(QUANTIFICATION.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ANNOTATION.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(LINKER.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(REQUANTIFICATION.out.versions)\n",
      "Nextflow operator join with content: quantification_information = REQUANTIFICATION.out.quantified_features.join(quantificaiton_data,by:[0])\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ANNOTATION_REQ.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(LINKER_REQ.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(PYOPENMS_EXPORTQUANTIFICATION.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(IDENTIFICATION.out.versions)\n",
      "Nextflow operator ifEmpty with content: IDENTIFICATION.out.sirius.ifEmpty([[],[]]),\n",
      "Nextflow operator ifEmpty with content: IDENTIFICATION.out.fingerid.ifEmpty([[],[]]),\n",
      "Nextflow operator ifEmpty with content: IDENTIFICATION.out.ms2query.ifEmpty([[],[]]))\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(PYOPENMS_EXPORTIDENTIFICATION.out.versions)\n",
      "Nextflow operator collectFile with content: ch_versions.unique().collectFile(name: 'collated_versions.yml')\n",
      "Error: Unclosed process block in file: ./git_repos\\metaboigniter\\workflows\\metaboigniter.nf\n",
      "Process NFCORE_METAPEP in filepath ./git_repos\\metapep\\main.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\metapep\\main.nf\n",
      "Process PROCESS_INPUT in filepath ./git_repos\\metapep\\subworkflows\\local\\process_input.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(CHECK_SAMPLESHEET_CREATE_TABLES.out.versions)\n",
      "Nextflow operator first with content: log_prefix = UNIFY_MODEL_LENGTHS.out.log.baseName.tokenize(\"_\").flatten().first()\n",
      "Nextflow operator map with content: log_prefix.map{it ->\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(UNIFY_MODEL_LENGTHS.out.versions)\n",
      "Nextflow operator splitCsv with content: .splitCsv(sep:'\\t', header:true)\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator branch with content: .branch {\n",
      "Nextflow operator set with content: .set{ ch_microbiomes_branch }\n",
      "Nextflow operator map with content: .map { row ->\n",
      "Nextflow operator set with content: .set { ch_taxa_input }\n",
      "Nextflow operator dump with content: ch_taxa_input.dump(tag:\"taxa\")\n",
      "Nextflow operator map with content: .map { row ->\n",
      "Nextflow operator set with content: .set { ch_proteins_input }\n",
      "Nextflow operator dump with content: ch_proteins_input.dump(tag:\"proteins\")\n",
      "Nextflow operator map with content: .map { row ->\n",
      "Nextflow operator set with content: .set { ch_assembly_input }\n",
      "Nextflow operator dump with content: ch_assembly_input.dump(tag:\"assembly\")\n",
      "Nextflow operator branch with content: .branch {\n",
      "Nextflow operator set with content: .set{ ch_microbiomes_bins }\n",
      "Nextflow operator flatMap with content: .flatMap { row ->\n",
      "Nextflow operator collect with content: return bin_files.collect {\n",
      "Nextflow operator set with content: .set { ch_bins_folders_input }\n",
      "Nextflow operator dump with content: ch_bins_folders_input.dump(tag:\"bins_folders\")\n",
      "Nextflow operator map with content: .map { row ->\n",
      "Nextflow operator set with content: .set{ ch_microbiomes_bins_archives_packed }\n",
      "Nextflow operator dump with content: ch_microbiomes_bins_archives_packed.dump(tag:\"bins_archives\")\n",
      "Nextflow operator unique with content: ch_microbiomes_bins_archives_packed.unique()\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(UNPACK_BIN_ARCHIVES.out.versions)\n",
      "Nextflow operator dump with content: UNPACK_BIN_ARCHIVES.out.ch_microbiomes_bins_archives_unpacked.dump(tag:\"bins_archives\")\n",
      "Nextflow operator flatMap with content: .flatMap { meta, bin_files ->\n",
      "Nextflow operator collect with content: return bin_files.collect {\n",
      "Nextflow operator set with content: .set{ ch_bins_archives_input }\n",
      "Nextflow operator dump with content: ch_bins_archives_input.dump(tag:\"bins_archives\")\n",
      "Nextflow operator concat with content: ch_nucl_input           = ch_assembly_input.concat(ch_bins_archives_input, ch_bins_folders_input).unique()\n",
      "Nextflow operator dump with content: ch_nucl_input.dump(tag:\"nucl\")\n",
      "Nextflow operator splitCsv with content: .splitCsv(sep:'\\t', header:true)\n",
      "Nextflow operator map with content: .map { row ->\n",
      "Nextflow operator set with content: .set { ch_weights }\n",
      "Nextflow operator dump with content: ch_weights.dump(tag:\"weights\")\n",
      "Nextflow operator collect with content: peptide_lengths         = peptide_lengths.collect()\n",
      "Error: Unclosed process block in file: ./git_repos\\metapep\\subworkflows\\local\\process_input.nf\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Process METAPEP in filepath ./git_repos\\metapep\\workflows\\metapep.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(EPYTOPE_SHOW_SUPPORTED_MODELS.out.versions)\n",
      "Nextflow operator collectFile with content: CUSTOM_DUMPSOFTWAREVERSIONS (ch_versions.unique().collectFile(name: 'collated_versions.yml'))\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(PROCESS_INPUT.out.versions)\n",
      "Nextflow operator collect with content: PROCESS_INPUT.out.ch_taxa_input.map { meta, file -> meta.id }.collect(),\n",
      "Nextflow operator collect with content: PROCESS_INPUT.out.ch_taxa_input.map { meta, file -> file }.collect()\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(DOWNLOAD_PROTEINS.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(PRODIGAL.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(CREATE_PROTEIN_TSV.out.versions)\n",
      "Nextflow operator collect with content: PROCESS_INPUT.out.ch_weights.map { meta, file -> meta.id }.collect().ifEmpty([]),\n",
      "Nextflow operator collect with content: PROCESS_INPUT.out.ch_weights.map { meta, file -> file }.collect().ifEmpty([])\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ASSIGN_NUCL_ENTITY_WEIGHTS.out.versions)\n",
      "Nextflow operator flatMap with content: ch_pred_proteins_sorted = CREATE_PROTEIN_TSV.out.ch_pred_proteins.toSortedList( { a, b -> a[0].id <=> b[0].id } ).flatMap()\n",
      "Nextflow operator collect with content: ch_pred_proteins_sorted.collect { meta, file -> file }.ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_pred_proteins_sorted.collect { meta, file -> meta }.ifEmpty([]),\n",
      "Nextflow operator ifEmpty with content: DOWNLOAD_PROTEINS.out.ch_entrez_proteins.ifEmpty([]),\n",
      "Nextflow operator ifEmpty with content: DOWNLOAD_PROTEINS.out.ch_entrez_entities_proteins.ifEmpty([]),\n",
      "Nextflow operator ifEmpty with content: DOWNLOAD_PROTEINS.out.ch_entrez_microbiomes_entities.ifEmpty([]),\n",
      "Nextflow operator collect with content: PROCESS_INPUT.out.ch_proteins_input.collect { meta, file -> file }.ifEmpty([]),\n",
      "Nextflow operator collect with content: PROCESS_INPUT.out.ch_proteins_input.collect { meta, file -> meta }.ifEmpty([])\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GENERATE_PROTEIN_AND_ENTITY_IDS.out.versions)\n",
      "Nextflow operator ifEmpty with content: DOWNLOAD_PROTEINS.out.ch_entrez_microbiomes_entities.ifEmpty([]),\n",
      "Nextflow operator ifEmpty with content: ASSIGN_NUCL_ENTITY_WEIGHTS.out.ch_nucl_microbiomes_entities.ifEmpty([]),\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(FINALIZE_MICROBIOME_ENTITIES.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GENERATE_PEPTIDES.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(COLLECT_STATS.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SPLIT_PRED_TASKS.out.versions)\n",
      "Nextflow operator flatten with content: SPLIT_PRED_TASKS.out.ch_epitope_prediction_chunks.flatten()\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(PREDICT_EPITOPES.out.versions)\n",
      "Nextflow operator count with content: SPLIT_PRED_TASKS.out.ch_epitope_prediction_chunks.flatten().count()\n",
      "Nextflow operator combine with content: .combine(PREDICT_EPITOPES.out.ch_epitope_predictions.toSortedList().flatten())\n",
      "Nextflow operator flatten with content: .merge(PREDICT_EPITOPES.out.ch_epitope_prediction_warnings.toSortedList().flatten())\n",
      "Nextflow operator branch with content: .branch{count, predictions_file, warnings_file ->\n",
      "Nextflow operator set with content: }.set { ch_pred_merge_input }\n",
      "Nextflow operator buffer with content: ch_pred_merge_input.buffer.multiMap{predictions_file, warnings_file ->\n",
      "Nextflow operator set with content: }.set { ch_predictions_mergebuffer_input }\n",
      "Nextflow operator multiMap with content: ch_pred_merge_input.unbuffered.multiMap{predictions_file, warnings_file ->\n",
      "Nextflow operator set with content: }.set { ch_predictions_unbuffered }\n",
      "Nextflow operator buffer with content: ch_predictions_mergebuffer_input.predictions.buffer(size: params.pred_buffer_files, remainder: true),\n",
      "Nextflow operator buffer with content: ch_predictions_mergebuffer_input.warnings.buffer(size: params.pred_buffer_files, remainder: true)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(MERGE_PREDICTIONS_BUFFER.out.versions)\n",
      "Nextflow operator mix with content: ch_merge_predictions_input_pred = MERGE_PREDICTIONS_BUFFER.out.ch_predictions_merged_buffer.mix(ch_predictions_unbuffered.predictions)\n",
      "Nextflow operator mix with content: ch_merge_predictions_input_warn = MERGE_PREDICTIONS_BUFFER.out.ch_prediction_warnings_merged_buffer.mix(ch_predictions_unbuffered.warnings)\n",
      "Nextflow operator collect with content: ch_merge_predictions_input_pred.collect(),\n",
      "Nextflow operator collect with content: ch_merge_predictions_input_warn.collect()\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(MERGE_PREDICTIONS.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(PREPARE_SCORE_DISTRIBUTION.out.versions)\n",
      "Nextflow operator flatten with content: PREPARE_SCORE_DISTRIBUTION.out.ch_prep_prediction_scores.flatten(),\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(PLOT_SCORE_DISTRIBUTION.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(PREPARE_ENTITY_BINDING_RATIOS.out.versions)\n",
      "Nextflow operator flatten with content: PREPARE_ENTITY_BINDING_RATIOS.out.ch_prep_entity_binding_ratios.flatten(),\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(PLOT_ENTITY_BINDING_RATIOS.out.versions)\n",
      "Nextflow operator collectFile with content: ch_versions.unique().collectFile(name: 'collated_versions.yml')\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_workflow_summary.collectFile(name: 'workflow_summary_mqc.yaml'))\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_methods_description.collectFile(name: 'methods_description_mqc.yaml'))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(CUSTOM_DUMPSOFTWAREVERSIONS.out.mqc_yml.collect())\n",
      "Nextflow operator collect with content: ch_multiqc_files.collect(),\n",
      "Nextflow operator collect with content: ch_metapep_logo.collect(),\n",
      "Nextflow operator toList with content: ch_multiqc_config.toList(),\n",
      "Nextflow operator toList with content: ch_multiqc_custom_config.toList(),\n",
      "Nextflow operator toList with content: ch_multiqc_logo.toList()\n",
      "Nextflow operator toList with content: multiqc_report = MULTIQC.out.report.toList()\n",
      "Error: Unclosed process block in file: ./git_repos\\metapep\\workflows\\metapep.nf\n",
      "Process NFCORE_METATDENOVO in filepath ./git_repos\\metatdenovo\\main.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\metatdenovo\\main.nf\n",
      "Process EGGNOG in filepath ./git_repos\\metatdenovo\\subworkflows\\local\\eggnog.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix ( EGGNOG_DOWNLOAD.out.versions )\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix ( EGGNOG_MAPPER.out.versions )\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix ( EGGNOG_SUM.out.versions )\n",
      "Error: Unclosed process block in file: ./git_repos\\metatdenovo\\subworkflows\\local\\eggnog.nf\n",
      "Process SUB_EUKULELE in filepath ./git_repos\\metatdenovo\\subworkflows\\local\\eukulele.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix ( EUKULELE_SEARCH.out.versions )\n",
      "Nextflow operator map with content: FORMAT_TAX( EUKULELE_SEARCH.out.taxonomy_estimation.map { meta, taxonomy, dbname -> [ meta, taxonomy ] } )\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix ( FORMAT_TAX.out.versions )\n",
      "Nextflow operator join with content: .join(eukulele)\n",
      "Nextflow operator map with content: .map { meta, taxonomy, protein, dbname, database -> [ meta, dbname, taxonomy ] }\n",
      "Nextflow operator set with content: .set { ch_sum_taxonomy }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix ( SUM_TAXONOMY.out.versions )\n",
      "Error: Unclosed process block in file: ./git_repos\\metatdenovo\\subworkflows\\local\\eukulele.nf\n",
      "Process FASTQC_TRIMGALORE in filepath ./git_repos\\metatdenovo\\subworkflows\\local\\fastqc_trimgalore.nf\n",
      "Nextflow operator set with content: FASTQC ( reads ).html.set { fastqc_html }\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(FASTQC.out.versions.first())\n",
      "Nextflow operator set with content: TRIMGALORE ( reads ).reads.set { trim_reads }\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(TRIMGALORE.out.versions.first())\n",
      "Error: Unclosed process block in file: ./git_repos\\metatdenovo\\subworkflows\\local\\fastqc_trimgalore.nf\n",
      "Process HMMCLASSIFY in filepath ./git_repos\\metatdenovo\\subworkflows\\local\\hmmclassify.nf\n",
      "Nextflow operator map with content: .map { meta, hmm, seqdb -> [ [ id: \"${meta.id}.${hmm.baseName}\" ], hmm, seqdb, false, true, false ] }\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(HMMER_HMMSEARCH.out.versions.first())\n",
      "Nextflow operator map with content: .map { meta, hmm, seqdb -> meta }\n",
      "Nextflow operator distinct with content: .distinct()\n",
      "Nextflow operator collect with content: .combine ( HMMER_HMMSEARCH.out.target_summary.collect { meta, summary -> summary } )\n",
      "Nextflow operator map with content: .map { [ it[0], it[1..-1] ] }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(HMMRANK.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\metatdenovo\\subworkflows\\local\\hmmclassify.nf\n",
      "Process INPUT_CHECK in filepath ./git_repos\\metatdenovo\\subworkflows\\local\\input_check.nf\n",
      "Nextflow operator splitCsv with content: .splitCsv ( header:true, sep:',' )\n",
      "Nextflow operator map with content: .map { create_fastq_channel(it) }\n",
      "Nextflow operator set with content: .set { reads }\n",
      "Error: Unclosed process block in file: ./git_repos\\metatdenovo\\subworkflows\\local\\input_check.nf\n",
      "Process KOFAMSCAN in filepath ./git_repos\\metatdenovo\\subworkflows\\local\\kofamscan.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix ( KOFAMSCAN_DOWNLOAD.out.versions )\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(KOFAMSCAN_SCAN.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SUM_KOFAMSCAN.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\metatdenovo\\subworkflows\\local\\kofamscan.nf\n",
      "Process PRODIGAL in filepath ./git_repos\\metatdenovo\\subworkflows\\local\\prodigal.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(PRODIGAL_MODULE.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(FORMAT_PRODIGAL_GFF.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\metatdenovo\\subworkflows\\local\\prodigal.nf\n",
      "Process PROKKA_SUBSETS in filepath ./git_repos\\metatdenovo\\subworkflows\\local\\prokka_subsets.nf\n",
      "Nextflow operator map with content: PROKKA ( contigs.map{ meta, contigs -> contigs }.splitFasta(size: batchsize, file: true).map { contigs -> [ [ id: contigs.getBaseName() ], contigs] }, [], []  )\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(PROKKA.out.versions)\n",
      "Nextflow operator collect with content: ch_log  = PROKKA.out.txt.collect()\n",
      "Nextflow operator map with content: contigs.map{ meta, contigs -> [ id:\"${meta.id}.prokka\" ] }\n",
      "Nextflow operator collect with content: .combine(PROKKA.out.gff.collect { meta, gff -> gff }.map { [ it ] })\n",
      "Nextflow operator set with content: .set { ch_gff }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GFF_CAT.out.versions)\n",
      "Nextflow operator map with content: contigs.map{ meta, contigs -> [ id:\"${meta.id}.prokka\" ] }\n",
      "Nextflow operator collect with content: .combine(PROKKA.out.faa.collect { meta, protein -> protein }.map { [ it ] })\n",
      "Nextflow operator set with content: .set { ch_faa }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(FAA_CAT.out.versions)\n",
      "Nextflow operator map with content: contigs.map{ meta, contigs -> [ id:\"${meta.id}.prokka\" ] }\n",
      "Nextflow operator collect with content: .combine(PROKKA.out.ffn.collect { meta, fnn -> fnn }.map { [ it ] })\n",
      "Nextflow operator set with content: .set { ch_ffn }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(FFN_CAT.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(PROKKAGFF2TSV.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\metatdenovo\\subworkflows\\local\\prokka_subsets.nf\n",
      "Process TRANSDECODER in filepath ./git_repos\\metatdenovo\\subworkflows\\local\\transdecoder.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(LONGORF.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(PREDICT.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\metatdenovo\\subworkflows\\local\\transdecoder.nf\n",
      "Process BAM_SORT_STATS_SAMTOOLS in filepath ./git_repos\\metatdenovo\\subworkflows\\nf-core\\bam_sort_stats_samtools\\main.nf\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SAMTOOLS_SORT.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SAMTOOLS_INDEX.out.versions.first())\n",
      "Nextflow operator join with content: .join(SAMTOOLS_INDEX.out.bai, by: [0], remainder: true)\n",
      "Nextflow operator join with content: .join(SAMTOOLS_INDEX.out.csi, by: [0], remainder: true)\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator set with content: .set { ch_bam_bai }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BAM_STATS_SAMTOOLS.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\metatdenovo\\subworkflows\\nf-core\\bam_sort_stats_samtools\\main.nf\n",
      "Process BAM_STATS_SAMTOOLS in filepath ./git_repos\\metatdenovo\\subworkflows\\nf-core\\bam_stats_samtools\\main.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_STATS.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_FLAGSTAT.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_IDXSTATS.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\metatdenovo\\subworkflows\\nf-core\\bam_stats_samtools\\main.nf\n",
      "Process METATDENOVO in filepath ./git_repos\\metatdenovo\\workflows\\metatdenovo.nf\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator groupTuple with content: .groupTuple()\n",
      "Nextflow operator branch with content: .branch {\n",
      "Nextflow operator flatten with content: return [ meta, fastq.flatten() ]\n",
      "Nextflow operator flatten with content: return [ meta, fastq.flatten() ]\n",
      "Nextflow operator set with content: .set { ch_fastq }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(INPUT_CHECK.out.versions)\n",
      "Nextflow operator mix with content: .mix(ch_fastq.single)\n",
      "Nextflow operator set with content: .set { ch_cat_fastq }\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(CAT_FASTQ.out.versions.first())\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(FASTQC_TRIMGALORE.out.versions)\n",
      "Nextflow operator collect with content: ch_collect_stats = ch_cat_fastq.collect { meta, fasta -> meta.id }.map { [ [ id:\"${assembler}.${orf_caller}\" ], it ] }\n",
      "Nextflow operator map with content: .map { meta, samples -> [ meta, samples, [] ] }\n",
      "Nextflow operator set with content: .set { ch_collect_stats }\n",
      "Nextflow operator collect with content: .combine(FASTQC_TRIMGALORE.out.trim_log.collect { meta, report -> report }.map { [ it ] })\n",
      "Nextflow operator set with content: .set { ch_collect_stats }\n",
      "Nextflow operator collect with content: .combine(FASTQC_TRIMGALORE.out.trim_log.collect { meta, report -> report[0] }.map { [ it ] })\n",
      "Nextflow operator set with content: .set { ch_collect_stats }\n",
      "Nextflow operator collect with content: ch_bbduk_logs = BBMAP_BBDUK.out.log.collect { meta, log ->  log }.map { [ it ] }\n",
      "Nextflow operator first with content: ch_versions   = ch_versions.mix(BBMAP_BBDUK.out.versions.first())\n",
      "Nextflow operator combine with content: .combine(ch_bbduk_logs)\n",
      "Nextflow operator set with content: .set {ch_collect_stats}\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(BBMAP_BBDUK.out.log.collect{ meta, log -> log })\n",
      "Nextflow operator map with content: .map { meta, samples, report -> [ meta, samples, report, [] ] }\n",
      "Nextflow operator set with content: .set { ch_collect_stats }\n",
      "Nextflow operator mix with content: ch_versions    = ch_versions.mix(SEQTK_MERGEPE.out.versions)\n",
      "Nextflow operator collect with content: BBMAP_BBNORM(ch_single_end.collect { meta, fastq -> fastq }.map {[ [id:'all_samples', single_end:true], it ] } )\n",
      "Nextflow operator map with content: ch_se_reads_to_assembly = BBMAP_BBNORM.out.fastq.map { meta, fasta -> fasta }\n",
      "Nextflow operator mix with content: ch_versions    = ch_versions.mix(BBMAP_BBNORM.out.versions)\n",
      "Nextflow operator map with content: ch_se_reads_to_assembly = ch_single_end.map { meta, fastq -> fastq }\n",
      "Nextflow operator collect with content: BBMAP_BBNORM(ch_interleaved.collect { meta, fastq -> fastq }.map {[ [id:'all_samples', single_end:true], it ] } )\n",
      "Nextflow operator map with content: ch_pe_reads_to_assembly = BBMAP_BBNORM.out.fastq.map { meta, fasta -> fasta }\n",
      "Nextflow operator mix with content: ch_versions    = ch_versions.mix(BBMAP_BBNORM.out.versions)\n",
      "Nextflow operator map with content: ch_pe_reads_to_assembly = ch_interleaved.map { meta, fastq -> fastq }\n",
      "Nextflow operator set with content: .set { ch_assembly_contigs }\n",
      "Nextflow operator toList with content: ch_pe_reads_to_assembly.toList(),\n",
      "Nextflow operator toList with content: ch_se_reads_to_assembly.toList()\n",
      "Nextflow operator mix with content: ch_versions    = ch_versions.mix(WRITESPADESYAML.out.versions)\n",
      "Nextflow operator mix with content: .mix(ch_se_reads_to_assembly)\n",
      "Nextflow operator collect with content: .collect()\n",
      "Nextflow operator map with content: .map { [ [ id:'rnaspades' ], it, [], [] ] }\n",
      "Nextflow operator set with content: .set { ch_spades }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SPADES.out.versions)\n",
      "Nextflow operator mix with content: ch_versions    = ch_versions.mix(FORMATSPADES.out.versions)\n",
      "Nextflow operator toList with content: ch_pe_reads_to_assembly.toList(),\n",
      "Nextflow operator toList with content: ch_se_reads_to_assembly.toList(),\n",
      "Nextflow operator map with content: .map { [ [ id: 'megahit' ], it ] }\n",
      "Nextflow operator set with content: .set { ch_assembly_contigs }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(MEGAHIT_INTERLEAVED.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SEQTK_SEQ_CONTIG_FILTER.out.versions)\n",
      "Nextflow operator map with content: UNPIGZ_GFF(PROKKA_SUBSETS.out.gff.map { meta, gff -> [ [id: \"${params.orf_caller}.${meta}\"], gff ] })\n",
      "Nextflow operator mix with content: ch_versions      = ch_versions.mix(PROKKA_SUBSETS.out.versions)\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(PROKKA_SUBSETS.out.prokka_log.collect{ meta, log -> log } )\n",
      "Nextflow operator map with content: PRODIGAL( ch_assembly_contigs.map { meta, contigs -> [ [id: \"${assembler}.${orf_caller}\"], contigs  ] } )\n",
      "Nextflow operator map with content: UNPIGZ_GFF(PRODIGAL.out.gff.map { meta, gff -> [ [id: \"${meta.id}.${orf_caller}\"], gff ] })\n",
      "Nextflow operator mix with content: ch_versions     = ch_versions.mix(PRODIGAL.out.versions)\n",
      "Nextflow operator map with content: TRANSDECODER ( ch_assembly_contigs.map { meta, contigs -> [ [id: \"transdecoder.${meta.id}\" ], contigs ] } )\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(TRANSDECODER.out.versions)\n",
      "Nextflow operator set with content: .set { ch_gff }\n",
      "Nextflow operator set with content: .set { ch_protein }\n",
      "Nextflow operator map with content: BBMAP_INDEX(ch_assembly_contigs.map { meta, contigs -> contigs })\n",
      "Nextflow operator mix with content: ch_versions   = ch_versions.mix(BBMAP_INDEX.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BBMAP_ALIGN.out.versions)\n",
      "Nextflow operator combine with content: .combine(ch_protein)\n",
      "Nextflow operator map with content: .map { hmm, meta, protein ->[ [ id: \"${assembler}.${orf_caller}\" ], hmm, protein ] }\n",
      "Nextflow operator set with content: .set { ch_hmmclassify }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(HMMCLASSIFY.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BAM_SORT_STATS_SAMTOOLS.out.versions)\n",
      "Nextflow operator combine with content: .combine(ch_gff.map { meta, bam -> bam } )\n",
      "Nextflow operator set with content: .set { ch_featurecounts }\n",
      "Nextflow operator collect with content: .combine(BAM_SORT_STATS_SAMTOOLS.out.idxstats.collect { meta, idxstats -> idxstats }.map { [ it ] } )\n",
      "Nextflow operator set with content: .set { ch_collect_stats }\n",
      "Nextflow operator mix with content: ch_versions       = ch_versions.mix(FEATURECOUNTS_CDS.out.versions)\n",
      "Nextflow operator collect with content: .collect() { meta, featurecounts -> featurecounts }\n",
      "Nextflow operator map with content: .map { featurecounts -> [ [ id:\"${assembler}.${orf_caller}\" ], featurecounts ] }\n",
      "Nextflow operator set with content: .set { ch_collect_feature }\n",
      "Nextflow operator mix with content: ch_versions           = ch_versions.mix(COLLECT_FEATURECOUNTS.out.versions)\n",
      "Nextflow operator collect with content: ch_fcs_for_stats      = COLLECT_FEATURECOUNTS.out.counts.collect { meta, tsv -> tsv }.map { [ it ] }\n",
      "Nextflow operator map with content: ch_fcs_for_summary    = COLLECT_FEATURECOUNTS.out.counts.map { meta, tsv -> tsv }\n",
      "Nextflow operator combine with content: .combine(ch_fcs_for_stats)\n",
      "Nextflow operator set with content: .set { ch_collect_stats }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(EGGNOG.out.versions)\n",
      "Nextflow operator map with content: .map { meta, protein -> [ meta, [] ] }\n",
      "Nextflow operator set with content: .set { ch_merge_tables }\n",
      "Nextflow operator map with content: .map { meta, protein -> [ meta, protein ] }\n",
      "Nextflow operator set with content: .set { ch_kofamscan }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(KOFAMSCAN.out.versions)\n",
      "Nextflow operator collect with content: ch_kofamscan_summary = KOFAMSCAN.out.kofamscan_summary.collect().map { meta, tsv -> tsv }\n",
      "Nextflow operator combine with content: .combine( ch_kofamscan_summary )\n",
      "Nextflow operator set with content: .set { ch_merge_tables }\n",
      "Nextflow operator map with content: .map { meta, tsv -> [ meta, tsv, [] ] }\n",
      "Nextflow operator set with content: .set { ch_merge_tables }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(UNPIGZ_CONTIGS.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(TRANSRATE.out.versions)\n",
      "Nextflow operator map with content: .map { [ it, file(params.eukulele_dbpath) ] }\n",
      "Nextflow operator set with content: .set { ch_eukulele_db }\n",
      "Nextflow operator map with content: .map { [ [], it ] }\n",
      "Nextflow operator set with content: .set { ch_eukulele_db }\n",
      "Nextflow operator map with content: .map { meta, protein -> [ [ id:\"${meta.id}\" ], protein ] }\n",
      "Nextflow operator combine with content: .combine( ch_eukulele_db )\n",
      "Nextflow operator set with content: .set { ch_eukulele }\n",
      "Nextflow operator collect with content: ch_taxonomy_summary = SUB_EUKULELE.out.taxonomy_summary.collect().map { meta, tsv -> tsv }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SUB_EUKULELE.out.versions)\n",
      "Nextflow operator combine with content: .combine( ch_taxonomy_summary )\n",
      "Nextflow operator set with content: .set { ch_merge_tables }\n",
      "Nextflow operator map with content: .map { meta, tsv1, tsv2 -> [ meta, tsv1, tsv2, [] ] }\n",
      "Nextflow operator set with content: .set { ch_merge_tables }\n",
      "Nextflow operator collect with content: .combine(MERGE_TABLES.out.merged_table.collect{ meta, tblout -> tblout }.map { [ it ] })\n",
      "Nextflow operator set with content: .set { ch_collect_stats }\n",
      "Nextflow operator mix with content: ch_versions       = ch_versions.mix(MERGE_TABLES.out.versions)\n",
      "Nextflow operator map with content: .map { meta, samples, report, tsv, idxstats, counts -> [ meta, samples, report, tsv, idxstats, counts, [] ] }\n",
      "Nextflow operator set with content: .set { ch_collect_stats }\n",
      "Nextflow operator mix with content: ch_versions     = ch_versions.mix(COLLECT_STATS.out.versions)\n",
      "Nextflow operator collectFile with content: ch_versions.unique().collectFile(name: 'collated_versions.yml')\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(CUSTOM_DUMPSOFTWAREVERSIONS.out.mqc_yml.collect())\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(FASTQC_TRIMGALORE.out.trim_zip.collect{ meta, zip -> zip })\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(TRANSRATE.out.assembly_qc.collect{ meta, tbl -> tbl })\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(BAM_SORT_STATS_SAMTOOLS.out.idxstats.collect{ meta, idxstats -> idxstats })\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(FEATURECOUNTS_CDS.out.summary.collect{ meta, summary -> summary })\n",
      "Nextflow operator collect with content: ch_multiqc_files.collect(),\n",
      "Nextflow operator toList with content: ch_multiqc_config.toList(),\n",
      "Nextflow operator toList with content: ch_multiqc_custom_config.toList(),\n",
      "Nextflow operator toList with content: ch_multiqc_logo.toList()\n",
      "Nextflow operator toList with content: multiqc_report = MULTIQC.out.report.toList()\n",
      "Error: Unclosed process block in file: ./git_repos\\metatdenovo\\workflows\\metatdenovo.nf\n",
      "Process NFCORE_METHYLSEQ in filepath ./git_repos\\methylseq\\main.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\methylseq\\main.nf\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Process BISMARK in filepath ./git_repos\\methylseq\\subworkflows\\local\\bismark.nf\n",
      "Nextflow operator mix with content: versions = versions.mix(BISMARK_ALIGN.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(SAMTOOLS_SORT_ALIGNED.out.versions)\n",
      "Nextflow operator map with content: alignment_reports = BISMARK_ALIGN.out.report.map{ meta, report -> [ meta, report, [] ] }\n",
      "Nextflow operator join with content: alignment_reports = BISMARK_ALIGN.out.report.join(BISMARK_DEDUPLICATE.out.report)\n",
      "Nextflow operator mix with content: versions = versions.mix(BISMARK_DEDUPLICATE.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(BISMARK_METHYLATIONEXTRACTOR.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(BISMARK_COVERAGE2CYTOSINE.out.versions)\n",
      "Nextflow operator join with content: .join(BISMARK_METHYLATIONEXTRACTOR.out.report)\n",
      "Nextflow operator join with content: .join(BISMARK_METHYLATIONEXTRACTOR.out.mbias)\n",
      "Nextflow operator mix with content: versions = versions.mix(BISMARK_REPORT.out.versions)\n",
      "Nextflow operator collect with content: BISMARK_ALIGN.out.bam.collect{ it[1].name }.ifEmpty([]),\n",
      "Nextflow operator collect with content: alignment_reports.collect{ it[1] }.ifEmpty([]),\n",
      "Nextflow operator collect with content: alignment_reports.collect{ it[2] }.ifEmpty([]),\n",
      "Nextflow operator collect with content: BISMARK_METHYLATIONEXTRACTOR.out.report.collect{ it[1] }.ifEmpty([]),\n",
      "Nextflow operator collect with content: BISMARK_METHYLATIONEXTRACTOR.out.mbias.collect{ it[1] }.ifEmpty([])\n",
      "Nextflow operator mix with content: versions = versions.mix(BISMARK_SUMMARY.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(SAMTOOLS_SORT_DEDUPLICATED.out.versions)\n",
      "Nextflow operator ifEmpty with content: BISMARK_SUMMARY.out.summary.ifEmpty([])\n",
      "Nextflow operator collect with content: .mix(alignment_reports.collect{ it[1] })\n",
      "Nextflow operator collect with content: .mix(alignment_reports.collect{ it[2] })\n",
      "Nextflow operator collect with content: .mix(BISMARK_METHYLATIONEXTRACTOR.out.report.collect{ it[1] })\n",
      "Nextflow operator collect with content: .mix(BISMARK_METHYLATIONEXTRACTOR.out.mbias.collect{ it[1] })\n",
      "Nextflow operator collect with content: .mix(BISMARK_REPORT.out.report.collect{ it[1] })\n",
      "Nextflow operator set with content: .set{ multiqc_files }\n",
      "Error: Unclosed process block in file: ./git_repos\\methylseq\\subworkflows\\local\\bismark.nf\n",
      "Process BWAMETH in filepath ./git_repos\\methylseq\\subworkflows\\local\\bwameth.nf\n",
      "Nextflow operator mix with content: versions = versions.mix(BWAMETH_ALIGN.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(SAMTOOLS_SORT.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(SAMTOOLS_INDEX_ALIGNMENTS.out.versions)\n",
      "Nextflow operator join with content: SAMTOOLS_FLAGSTAT( BWAMETH_ALIGN.out.bam.join(SAMTOOLS_INDEX_ALIGNMENTS.out.bai) )\n",
      "Nextflow operator join with content: SAMTOOLS_STATS( BWAMETH_ALIGN.out.bam.join(SAMTOOLS_INDEX_ALIGNMENTS.out.bai), [] )\n",
      "Nextflow operator mix with content: versions = versions.mix(SAMTOOLS_FLAGSTAT.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(SAMTOOLS_STATS.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(PICARD_MARKDUPLICATES.out.versions)\n",
      "Nextflow operator join with content: alignments.join(bam_index),\n",
      "Nextflow operator join with content: alignments.join(bam_index),\n",
      "Nextflow operator mix with content: versions = versions.mix(METHYLDACKEL_EXTRACT.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(METHYLDACKEL_MBIAS.out.versions)\n",
      "Nextflow operator collect with content: picard_metrics.collect{ it[1] }\n",
      "Nextflow operator collect with content: .mix(SAMTOOLS_FLAGSTAT.out.flagstat.collect{ it[1] })\n",
      "Nextflow operator collect with content: .mix(SAMTOOLS_STATS.out.stats.collect{ it[1] })\n",
      "Nextflow operator collect with content: .mix(METHYLDACKEL_EXTRACT.out.bedgraph.collect{ it[1] })\n",
      "Nextflow operator collect with content: .mix(METHYLDACKEL_MBIAS.out.txt.collect{ it[1] })\n",
      "Nextflow operator set with content: .set{ multiqc_files }\n",
      "Error: Unclosed process block in file: ./git_repos\\methylseq\\subworkflows\\local\\bwameth.nf\n",
      "Process PREPARE_GENOME in filepath ./git_repos\\methylseq\\subworkflows\\local\\prepare_genome.nf\n",
      "Nextflow operator map with content: ch_bismark_index = UNTAR ( [ [:], file(params.bismark_index) ] ).untar.map { it[1] }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BISMARK_GENOMEPREPARATION.out.versions)\n",
      "Nextflow operator map with content: ch_bismark_index = UNTAR ( [ [:], file(params.bwa_meth_index) ] ).untar.map { it[1] }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BWAMETH_INDEX.out.versions)\n",
      "Nextflow operator map with content: ch_fasta_index = SAMTOOLS_FAIDX.out.fai.map{ return(it[1])}\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_FAIDX.out.versions)\n",
      "Nextflow operator ifEmpty with content: versions      = ch_versions.ifEmpty(null) // channel: [ versions.yml ]\n",
      "Error: Unclosed process block in file: ./git_repos\\methylseq\\subworkflows\\local\\prepare_genome.nf\n",
      "Process METHYLSEQ in filepath ./git_repos\\methylseq\\workflows\\methylseq.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(PREPARE_GENOME.out.versions)\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator groupTuple with content: .groupTuple()\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator join with content: meta_clone.id = parts.length > 1 ? parts[0..-2].join('_') : meta_clone.id\n",
      "Nextflow operator groupTuple with content: .groupTuple(by: [0])\n",
      "Nextflow operator branch with content: .branch {\n",
      "Nextflow operator flatten with content: return [ meta, fastq.flatten() ]\n",
      "Nextflow operator flatten with content: return [ meta, fastq.flatten() ]\n",
      "Nextflow operator set with content: .set { ch_fastq }\n",
      "Nextflow operator mix with content: .mix(ch_fastq.single)\n",
      "Nextflow operator set with content: .set { ch_cat_fastq }\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(CAT_FASTQ.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(FASTQC.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(TRIMGALORE.out.versions.first())\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BISMARK.out.versions.unique{ it.baseName })\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BWAMETH.out.versions.unique{ it.baseName })\n",
      "Nextflow operator toList with content: params.bamqc_regions_file ? Channel.fromPath( params.bamqc_regions_file, checkIfExists: true ).toList() : []\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(QUALIMAP_BAMQC.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(PRESEQ_LCEXTRAP.out.versions.first())\n",
      "Nextflow operator collectFile with content: ch_versions.unique().collectFile(name: 'collated_versions.yml')\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_workflow_summary.collectFile(name: 'workflow_summary_mqc.yaml'))\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_methods_description.collectFile(name: 'methods_description_mqc.yaml'))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(CUSTOM_DUMPSOFTWAREVERSIONS.out.mqc_yml.collect())\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(QUALIMAP_BAMQC.out.results.collect{ it[1] }.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(PRESEQ_LCEXTRAP.out.log.collect{ it[1] }.ifEmpty([]))\n",
      "Nextflow operator ifEmpty with content: ch_multiqc_files = ch_multiqc_files.mix(ch_aligner_mqc.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(TRIMGALORE.out.log.collect{ it[1] })\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(FASTQC.out.zip.collect{ it[1] }.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files.collect(),\n",
      "Nextflow operator toList with content: ch_multiqc_config.toList(),\n",
      "Nextflow operator toList with content: ch_multiqc_custom_config.toList(),\n",
      "Nextflow operator toList with content: ch_multiqc_logo.toList()\n",
      "Nextflow operator toList with content: multiqc_report = MULTIQC.out.report.toList()\n",
      "Nextflow operator mix with content: ch_versions    = ch_versions.mix(MULTIQC.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\methylseq\\workflows\\methylseq.nf\n",
      "Process NFCORE_MHCQUANT in filepath ./git_repos\\mhcquant\\main.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\mhcquant\\main.nf\n",
      "Process INCLUDE_PROTEINS in filepath ./git_repos\\mhcquant\\subworkflows\\local\\include_proteins.nf\n",
      "Nextflow operator splitCsv with content: .splitCsv(header: ['Sample', 'VCF_FileName'], sep:'\\t', skip: 1)\n",
      "Nextflow operator map with content: .map { col -> tuple(\"${col.Sample}\", file(\"${col.VCF_FileName}\"),) }\n",
      "Nextflow operator set with content: .set { ch_vcf_from_sheet }\n",
      "Nextflow operator map with content: .map{ it -> [it[0].sample, it[0], it[1]] }\n",
      "Nextflow operator combine with content: .combine( ch_vcf_from_sheet, by: 0 )\n",
      "Nextflow operator map with content: .map(it -> [it[1], it[2], it[3]])\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(GENERATE_PROTEINS_FROM_VCF.out.versions.first().ifEmpty(null))\n",
      "Error: Unclosed process block in file: ./git_repos\\mhcquant\\subworkflows\\local\\include_proteins.nf\n",
      "Process INPUT_CHECK in filepath ./git_repos\\mhcquant\\subworkflows\\local\\input_check.nf\n",
      "Nextflow operator splitCsv with content: .splitCsv ( header:true, sep:'\\t' )\n",
      "Nextflow operator map with content: .map { create_ms_channel(it) }\n",
      "Nextflow operator set with content: .set { ms_runs }\n",
      "Error: Unclosed process block in file: ./git_repos\\mhcquant\\subworkflows\\local\\input_check.nf\n",
      "Process MAP_ALIGNMENT in filepath ./git_repos\\mhcquant\\subworkflows\\local\\map_alignment.nf\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(OPENMS_MAPALIGNERIDENTIFICATION.out.versions.first().ifEmpty(null))\n",
      "Nextflow operator join with content: .join( OPENMS_MAPALIGNERIDENTIFICATION.out.trafoxml )\n",
      "Nextflow operator map with content: .map { groupMeta, meta, trafoxml -> [meta, trafoxml] }\n",
      "Nextflow operator transpose with content: .transpose()\n",
      "Nextflow operator set with content: .set { joined_trafos }\n",
      "Nextflow operator join with content: joined_trafos_mzmls = mzml.join(joined_trafos)\n",
      "Nextflow operator join with content: .join( merge_meta_map )\n",
      "Nextflow operator map with content: .map { group_meta, idxml, meta -> [meta, idxml] }\n",
      "Nextflow operator transpose with content: .transpose()\n",
      "Nextflow operator join with content: .join( joined_trafos )\n",
      "Nextflow operator set with content: .set { joined_trafos_ids }\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(OPENMS_MAPRTTRANSFORMERMZML.out.versions.first().ifEmpty(null))\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(OPENMS_MAPRTTRANSFORMERIDXML.out.versions.first().ifEmpty(null))\n",
      "Error: Unclosed process block in file: ./git_repos\\mhcquant\\subworkflows\\local\\map_alignment.nf\n",
      "Process PREDICT_CLASS1 in filepath ./git_repos\\mhcquant\\subworkflows\\local\\predict_class1.nf\n",
      "Nextflow operator map with content: alleles = peptides_class_1_alleles.map{ meta, alleles -> [[id:meta], alleles] }\n",
      "Nextflow operator join with content: MHCFLURRY_PREDICTPEPTIDESCLASS1(mztab.join(alleles))\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(MHCFLURRY_PREDICTPEPTIDESCLASS1.out.versions.first().ifEmpty(null))\n",
      "Nextflow operator combine with content: PREDICT_POSSIBLE_CLASS1_NEOEPITOPES(alleles.combine(ch_vcf_from_sheet, by:0))\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(PREDICT_POSSIBLE_CLASS1_NEOEPITOPES.out.versions.first().ifEmpty(null))\n",
      "Nextflow operator map with content: .map{ it -> [it[0].sample, it[0], it[1]] }\n",
      "Nextflow operator combine with content: .combine( ch_predicted_possible_neoepitopes, by:0)\n",
      "Nextflow operator map with content: .map( it -> [it[1], it[2], it[3]])\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(RESOLVE_FOUND_CLASS1_NEOEPITOPES.out.versions.first().ifEmpty(null))\n",
      "Nextflow operator join with content: MHCFLURRY_PREDICTNEOEPITOPESCLASS1(alleles.join(RESOLVE_FOUND_CLASS1_NEOEPITOPES.out.csv, by:0))\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(MHCFLURRY_PREDICTNEOEPITOPESCLASS1.out.versions.first().ifEmpty(null))\n",
      "Error: Unclosed process block in file: ./git_repos\\mhcquant\\subworkflows\\local\\predict_class1.nf\n",
      "Process PREDICT_CLASS2 in filepath ./git_repos\\mhcquant\\subworkflows\\local\\predict_class2.nf\n",
      "Nextflow operator map with content: alleles = peptides_class_2_alleles.map{meta, alleles -> [[id:meta], alleles]}\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(MHCNUGGETS_PEPTIDESCLASS2PRE.out.versions.first().ifEmpty(null))\n",
      "Nextflow operator join with content: .join(alleles)\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(MHCNUGGETS_PREDICTPEPTIDESCLASS2.out.versions.first().ifEmpty(null))\n",
      "Nextflow operator join with content: MHCNUGGETS_PEPTIDESCLASS2POST( MHCNUGGETS_PREDICTPEPTIDESCLASS2.out.csv.join(MHCNUGGETS_PEPTIDESCLASS2PRE.out.geneID, by:0) )\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(MHCNUGGETS_PEPTIDESCLASS2POST.out.versions.first().ifEmpty(null))\n",
      "Nextflow operator combine with content: PREDICT_POSSIBLE_CLASS2_NEOEPITOPES(alleles.combine(ch_vcf_from_sheet, by:0))\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(PREDICT_POSSIBLE_CLASS2_NEOEPITOPES.out.versions.first().ifEmpty(null))\n",
      "Nextflow operator map with content: .map{ it -> [it[0].sample, it[1]] }\n",
      "Nextflow operator combine with content: .combine( ch_predicted_possible_neoepitopes, by:0)\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(RESOLVE_FOUND_CLASS2_NEOEPITOPES.out.versions.first().ifEmpty(null))\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(MHCNUGGETS_NEOEPITOPESCLASS2PRE.out.versions.first().ifEmpty(null))\n",
      "Nextflow operator join with content: MHCNUGGETS_PREDICTNEOEPITOPESCLASS2(MHCNUGGETS_NEOEPITOPESCLASS2PRE.out.preprocessed.join(alleles, by:0))\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(MHCNUGGETS_PREDICTNEOEPITOPESCLASS2.out.versions.first().ifEmpty(null))\n",
      "Nextflow operator join with content: MHCNUGGETS_NEOEPITOPESCLASS2POST(RESOLVE_FOUND_CLASS2_NEOEPITOPES.out.csv.join(MHCNUGGETS_PREDICTNEOEPITOPESCLASS2.out.csv, by:0))\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(MHCNUGGETS_NEOEPITOPESCLASS2POST.out.versions.first().ifEmpty(null))\n",
      "Error: Unclosed process block in file: ./git_repos\\mhcquant\\subworkflows\\local\\predict_class2.nf\n",
      "Process PROCESS_FEATURE in filepath ./git_repos\\mhcquant\\subworkflows\\local\\process_feature.nf\n",
      "Nextflow operator map with content: .map { meta, featurexml -> [[id: meta.sample + '_' + meta.condition], featurexml] }\n",
      "Nextflow operator groupTuple with content: .groupTuple()\n",
      "Nextflow operator set with content: .set { ch_features_grouped }\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(OPENMS_FEATUREFINDERIDENTIFICATION.out.versions.first().ifEmpty(null))\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(OPENMS_FEATURELINKERUNLABELEDKD.out.versions.first().ifEmpty(null))\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(OPENMS_IDCONFLICTRESOLVER.out.versions.first().ifEmpty(null))\n",
      "Error: Unclosed process block in file: ./git_repos\\mhcquant\\subworkflows\\local\\process_feature.nf\n",
      "Process QUANT in filepath ./git_repos\\mhcquant\\subworkflows\\local\\quant.nf\n",
      "Nextflow operator join with content: .join( merge_meta_map )\n",
      "Nextflow operator join with content: .join( filter_q_value )\n",
      "Nextflow operator map with content: .map { group_meta, ripped, meta, fdrfiltered -> [meta, ripped, fdrfiltered] }\n",
      "Nextflow operator transpose with content: .transpose()\n",
      "Nextflow operator set with content: .set { ch_ripped_pout }\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(OPENMS_IDRIPPER.out.versions.ifEmpty(null))\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(OPENMS_IDSCORESWITCHER.out.versions.ifEmpty(null))\n",
      "Nextflow operator map with content: .map { meta, idxml -> [[id:meta.sample + '_' + meta.condition], [id:meta.id, file:idxml]] }\n",
      "Nextflow operator groupTuple with content: .groupTuple( sort: sortById )\n",
      "Nextflow operator map with content: .map { meta, idxml -> [meta, idxml.file] }\n",
      "Nextflow operator set with content: .set { ch_runs_to_be_aligned }\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(PYOPENMS_IDFILTER.out.versions.ifEmpty(null))\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix( MAP_ALIGNMENT.out.versions.ifEmpty(null) )\n",
      "Nextflow operator map with content: .map { meta, aligned_idxml -> [[id: meta.sample + '_' + meta.condition], aligned_idxml] }\n",
      "Nextflow operator groupTuple with content: .groupTuple())\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(OPENMS_IDMERGER_QUANT.out.versions.ifEmpty(null))\n",
      "Nextflow operator join with content: .join( MAP_ALIGNMENT.out.aligned_idxml )\n",
      "Nextflow operator map with content: .map { meta, mzml, idxml -> [[id: meta.sample + '_' + meta.condition], meta, [id:meta.id, file:mzml], [id:meta.id, file:idxml]] }\n",
      "Nextflow operator groupTuple with content: .groupTuple( sort: sortById )\n",
      "Nextflow operator map with content: .map { group_meta, meta, mzml, idxml -> [group_meta, meta, mzml.file, idxml.file] }\n",
      "Nextflow operator join with content: .join( OPENMS_IDMERGER_QUANT.out.idxml )\n",
      "Nextflow operator map with content: .map { group_meta, meta, mzml, idxml, merged_idxml -> [meta, mzml, idxml, merged_idxml] }\n",
      "Nextflow operator transpose with content: .transpose()\n",
      "Nextflow operator set with content: .set { ch_runs_to_be_quantified }\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(PROCESS_FEATURE.out.versions.ifEmpty(null))\n",
      "Error: Unclosed process block in file: ./git_repos\\mhcquant\\subworkflows\\local\\quant.nf\n",
      "Process REFINE_FDR in filepath ./git_repos\\mhcquant\\subworkflows\\local\\refine_fdr.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(OPENMS_MZTABEXPORTERPERC.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(OPENMS_MZTABEXPORTERPSM.out.versions)\n",
      "Nextflow operator join with content: .join( OPENMS_MZTABEXPORTERPSM.out.mztab, by:[0] )\n",
      "Nextflow operator map with content: .map{ it -> [it[0].sample, it[0], it[1], it[2]] }\n",
      "Nextflow operator combine with content: .combine( classI_alleles, by:0)\n",
      "Nextflow operator map with content: .map(it -> [it[1], it[2], it[3], it[4]])\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(MHCFLURRY_PREDICTPSMS.out.versions)\n",
      "Nextflow operator combine with content: OPENMS_IDFILTER_PSMS(psm_features.combine( MHCFLURRY_PREDICTPSMS.out.idxml, by: [0] ))\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(OPENMS_IDFILTER_PSMS.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(OPENMS_PERCOLATORADAPTER.out.versions)\n",
      "Nextflow operator flatMap with content: OPENMS_IDFILTER_REFINED(OPENMS_PERCOLATORADAPTER.out.idxml.flatMap { it -> [tuple(it[0], it[1], null)]})\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(OPENMS_IDFILTER_REFINED.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\mhcquant\\subworkflows\\local\\refine_fdr.nf\n",
      "Process MHCQUANT in filepath ./git_repos\\mhcquant\\workflows\\mhcquant.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(INPUT_CHECK.out.versions)\n",
      "Nextflow operator branch with content: .branch {\n",
      "Nextflow operator set with content: .set { branched_ms_files }\n",
      "Nextflow operator combine with content: .combine(INPUT_CHECK.out.ms_runs)\n",
      "Nextflow operator map with content: .map{ fasta, meta, ms_file -> [meta.subMap('id', 'sample', 'condition'), fasta] }\n",
      "Nextflow operator ifEmpty with content: .ifEmpty { exit 1, \"params.fasta was empty - no input file supplied\" }\n",
      "Nextflow operator set with content: .set { input_fasta }\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(INCLUDE_PROTEINS.out.versions.ifEmpty(null))\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(OPENMS_DECOYDATABASE.out.versions.ifEmpty(null))\n",
      "Nextflow operator map with content: ch_ms_files = branched_ms_files.mzml.map{ meta, mzml -> [meta, mzml[0]]}\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(THERMORAWFILEPARSER.out.versions.ifEmpty(null))\n",
      "Nextflow operator mix with content: ch_ms_files = ch_ms_files.mix(THERMORAWFILEPARSER.out.mzml)\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(TDF2MZML.out.versions.ifEmpty(null))\n",
      "Nextflow operator mix with content: ch_ms_files = ch_ms_files.mix(TDF2MZML.out.mzml)\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(OPENMS_PEAKPICKERHIRES.out.versions.ifEmpty(null))\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(OPENMS_FILEFILTER.out.versions.ifEmpty(null))\n",
      "Nextflow operator join with content: OPENMS_COMETADAPTER(ch_clean_mzml_file.join(ch_decoy_db, remainder:true))\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(DEEPLC.out.versions.ifEmpty(null))\n",
      "Nextflow operator join with content: MS2PIP(ch_comet_out_idxml.join(ch_clean_mzml_file))\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(MS2PIP.out.versions.ifEmpty(null))\n",
      "Nextflow operator join with content: OPENMS_PEPTIDEINDEXER(ch_comet_out_idxml_proceeding.join(ch_decoy_db))\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(OPENMS_PEPTIDEINDEXER.out.versions.ifEmpty(null))\n",
      "Nextflow operator map with content: .map { meta, idxml -> [[id: meta.sample + '_' + meta.condition], meta] }\n",
      "Nextflow operator groupTuple with content: .groupTuple( sort: sortById )\n",
      "Nextflow operator set with content: .set { merge_meta_map }\n",
      "Nextflow operator map with content: .map { meta, idxml -> [[id: meta.sample + '_' + meta.condition], idxml] }\n",
      "Nextflow operator groupTuple with content: .groupTuple()\n",
      "Nextflow operator set with content: .set { ch_runs_to_merge }\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(OPENMS_IDMERGER.out.versions.ifEmpty(null))\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(OPENMS_PSMFEATUREEXTRACTOR.out.versions.ifEmpty(null))\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(OPENMS_PERCOLATORADAPTER.out.versions.ifEmpty(null))\n",
      "Nextflow operator flatMap with content: OPENMS_IDFILTER_Q_VALUE(OPENMS_PERCOLATORADAPTER.out.idxml.flatMap { it -> [tuple(it[0], it[1], null)] })\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(OPENMS_IDFILTER_Q_VALUE.out.versions.ifEmpty(null))\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(REFINE_FDR.out.versions.ifEmpty(null))\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(QUANT.out.versions.ifEmpty(null))\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(OPENMS_TEXTEXPORTER.out.versions.ifEmpty(null))\n",
      "Nextflow operator map with content: OPENMS_TEXTEXPORTER.out.tsv.map {\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(OPENMS_MZTABEXPORTER.out.versions.ifEmpty(null))\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(PREDICT_CLASS1.out.versions.ifEmpty(null))\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(PREDICT_CLASS2.out.versions.ifEmpty(null))\n",
      "Nextflow operator map with content: ch_clean_mzml_file.map { meta, mzml -> [[id: meta.sample + '_' + meta.condition], mzml] }\n",
      "Nextflow operator groupTuple with content: .groupTuple()\n",
      "Nextflow operator join with content: .join(filter_q_value)\n",
      "Nextflow operator set with content: .set{ ch_ion_annotator_input }\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(PYOPENMS_IONANNOTATOR.out.versions.ifEmpty(null))\n",
      "Nextflow operator collectFile with content: ch_versions.unique().collectFile(name: 'collated_versions.yml')\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_workflow_summary.collectFile(name: 'workflow_summary_mqc.yaml'))\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_methods_description.collectFile(name: 'methods_description_mqc.yaml'))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(CUSTOM_DUMPSOFTWAREVERSIONS.out.mqc_yml.collect())\n",
      "Nextflow operator collect with content: ch_multiqc_files.collect(),\n",
      "Nextflow operator collect with content: ch_multiqc_config.collect().ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_multiqc_custom_config.collect().ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_multiqc_logo.collect().ifEmpty([])\n",
      "Nextflow operator toList with content: multiqc_report = MULTIQC.out.report.toList()\n",
      "Nextflow operator mix with content: ch_versions    = ch_versions.mix(MULTIQC.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\mhcquant\\workflows\\mhcquant.nf\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Process NFCORE_MOLKART in filepath ./git_repos\\molkart\\main.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\molkart\\main.nf\n",
      "Process INPUT_CHECK in filepath ./git_repos\\molkart\\subworkflows\\local\\input_check.nf\n",
      "Nextflow operator splitCsv with content: .splitCsv ( header:true, sep:',' )\n",
      "Nextflow operator map with content: .map { create_fastq_channel(it) }\n",
      "Nextflow operator set with content: .set { reads }\n",
      "Error: Unclosed process block in file: ./git_repos\\molkart\\subworkflows\\local\\input_check.nf\n",
      "Process MOLKART in filepath ./git_repos\\molkart\\workflows\\molkart.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(INPUT_CHECK.out.versions)\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(FASTQC.out.versions.first())\n",
      "Nextflow operator collectFile with content: ch_versions.unique().collectFile(name: 'collated_versions.yml')\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_workflow_summary.collectFile(name: 'workflow_summary_mqc.yaml'))\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_methods_description.collectFile(name: 'methods_description_mqc.yaml'))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(CUSTOM_DUMPSOFTWAREVERSIONS.out.mqc_yml.collect())\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(FASTQC.out.zip.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files.collect(),\n",
      "Nextflow operator toList with content: ch_multiqc_config.toList(),\n",
      "Nextflow operator toList with content: ch_multiqc_custom_config.toList(),\n",
      "Nextflow operator toList with content: ch_multiqc_logo.toList()\n",
      "Nextflow operator toList with content: multiqc_report = MULTIQC.out.report.toList()\n",
      "Error: Unclosed process block in file: ./git_repos\\molkart\\workflows\\molkart.nf\n",
      "Process NFCORE_MULTIPLESEQUENCEALIGN in filepath ./git_repos\\multiplesequencealign\\main.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\multiplesequencealign\\main.nf\n",
      "Process ALIGN in filepath ./git_repos\\multiplesequencealign\\subworkflows\\local\\align.nf\n",
      "Nextflow operator multiMap with content: .multiMap{ it ->\n",
      "Nextflow operator unique with content: COMPUTE_TREES(ch_fastas, ch_tools_split.tree.unique())\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(COMPUTE_TREES.out.versions)\n",
      "Nextflow operator combine with content: ch_fastas.combine(ch_tools)\n",
      "Nextflow operator map with content: .map{ metafasta, fasta, metatree, metaalign -> [metafasta+metatree , metaalign, fasta] }\n",
      "Nextflow operator set with content: .set { ch_fasta_tools }\n",
      "Nextflow operator join with content: .join(trees, by: [0], remainder:true )\n",
      "Nextflow operator map with content: .map { metafasta_tree, metaalign, fasta, tree -> [metafasta_tree + metaalign, fasta, tree]}\n",
      "Nextflow operator map with content: .map { meta, fasta, tree ->  tree ? [meta,fasta, tree] : [meta, fasta, []] }\n",
      "Nextflow operator branch with content: .branch {\n",
      "Nextflow operator set with content: .set { ch_fasta_trees }\n",
      "Nextflow operator multiMap with content: .multiMap{\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(CLUSTALO_ALIGN.out.versions.first())\n",
      "Nextflow operator multiMap with content: .multiMap{\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(FAMSA_ALIGN.out.versions.first())\n",
      "Nextflow operator mix with content: msa = msa.mix(FAMSA_ALIGN.out.alignment)\n",
      "Nextflow operator multiMap with content: .multiMap{\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(KALIGN_ALIGN.out.versions.first())\n",
      "Nextflow operator multiMap with content: .multiMap{\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(LEARNMSA_ALIGN.out.versions.first())\n",
      "Nextflow operator multiMap with content: .multiMap{\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(MAFFT.out.versions.first())\n",
      "Nextflow operator multiMap with content: .multiMap{\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(TCOFFEE_ALIGN.out.versions.first())\n",
      "Nextflow operator mix with content: msa = msa.mix(TCOFFEE_ALIGN.out.alignment)\n",
      "Nextflow operator map with content: ch_fasta_trees_3dcoffee = ch_fasta_trees.tcoffee3d.map{ meta, fasta, tree -> [meta[\"id\"], meta, fasta, tree] }\n",
      "Nextflow operator combine with content: .combine(ch_structures.map{ meta, template, structures -> [meta[\"id\"], template, structures]}, by: 0)\n",
      "Nextflow operator multiMap with content: .multiMap{\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(TCOFFEE3D_ALIGN.out.versions.first())\n",
      "Nextflow operator mix with content: msa = msa.mix(TCOFFEE3D_ALIGN.out.alignment)\n",
      "Nextflow operator multiMap with content: .multiMap{\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(REGRESSIVE_ALIGN.out.versions.first())\n",
      "Nextflow operator mix with content: msa = msa.mix(REGRESSIVE_ALIGN.out.alignment)\n",
      "Nextflow operator multiMap with content: .multiMap{\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(MUSCLE5_SUPER5.out.versions.first())\n",
      "Nextflow operator first with content: msa = msa.mix(MUSCLE5_SUPER5.out.alignment.first())\n",
      "Nextflow operator ifEmpty with content: versions         = ch_versions.ifEmpty(null) // channel: [ versions.yml ]\n",
      "Error: Unclosed process block in file: ./git_repos\\multiplesequencealign\\subworkflows\\local\\align.nf\n",
      "Process COMPUTE_TREES in filepath ./git_repos\\multiplesequencealign\\subworkflows\\local\\compute_trees.nf\n",
      "Nextflow operator combine with content: .combine(tree_tools)\n",
      "Nextflow operator map with content: .map{ metafasta, fasta, metatree -> [metafasta+metatree, fasta] }\n",
      "Nextflow operator branch with content: .branch{\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(FAMSA_GUIDETREE.out.versions.first())\n",
      "Nextflow operator mix with content: ch_trees = ch_trees.mix(CLUSTALO_GUIDETREE.out.tree)\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(CLUSTALO_GUIDETREE.out.versions.first())\n",
      "Nextflow operator ifEmpty with content: versions         = ch_versions.ifEmpty(null) // channel: [ versions.yml ]\n",
      "Error: Unclosed process block in file: ./git_repos\\multiplesequencealign\\subworkflows\\local\\compute_trees.nf\n",
      "Process EVALUATE in filepath ./git_repos\\multiplesequencealign\\subworkflows\\local\\evaluate.nf\n",
      "Nextflow operator map with content: alignment_and_ref = ch_references.map { meta,ref -> [ meta.id, ref ] }\n",
      "Nextflow operator cross with content: .cross (ch_msa.map { meta, aln -> [ meta.id, meta, aln ] })\n",
      "Nextflow operator map with content: .map { chref, chaln -> [ chaln[1], chaln[2], chref[1]  ] }\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(TCOFFEE_ALNCOMPARE_SP.out.versions.first())\n",
      "Nextflow operator map with content: ch_sp_summary = sp_scores.map{\n",
      "Nextflow operator collect with content: }.collect().map{\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(TCOFFEE_ALNCOMPARE_TC.out.versions.first())\n",
      "Nextflow operator map with content: ch_tc_summary = tc_scores.map{\n",
      "Nextflow operator collect with content: }.collect().map{\n",
      "Nextflow operator map with content: msa_str = ch_structures.map { meta, template, str -> [ meta.id, template, str ] }\n",
      "Nextflow operator cross with content: .cross (ch_msa.map { meta, aln -> [ meta.id, meta, aln ] })\n",
      "Nextflow operator multiMap with content: .multiMap { chstr, chaln ->\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(TCOFFEE_IRMSD.out.versions.first())\n",
      "Nextflow operator map with content: ch_irmsd_summary = tcoffee_irmsd_scores_tot.map{\n",
      "Nextflow operator collect with content: }.collect().map{\n",
      "Nextflow operator map with content: sp      = sp_csv.map{ meta, csv -> csv }\n",
      "Nextflow operator map with content: tc      = tc_csv.map{ meta, csv -> csv }\n",
      "Nextflow operator map with content: irmsd   = irmsd_csv.map{ meta, csv -> csv }\n",
      "Nextflow operator count with content: def number_of_evals = [params.calc_sp, params.calc_tc, params.calc_irmsd].count{ it == true }\n",
      "Nextflow operator collect with content: csvs_stats = sp.mix(tc).mix(irmsd).collect().map{ csvs -> [[id:\"summary_eval\"], csvs] }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(MERGE_EVAL.out.versions)\n",
      "Nextflow operator ifEmpty with content: versions                    = ch_versions.ifEmpty(null) // channel: [ versions.yml ]\n",
      "Error: Unclosed process block in file: ./git_repos\\multiplesequencealign\\subworkflows\\local\\evaluate.nf\n",
      "Process STATS in filepath ./git_repos\\multiplesequencealign\\subworkflows\\local\\stats.nf\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(TCOFFEE_SEQREFORMAT_SIM.out.versions.first())\n",
      "Nextflow operator map with content: ch_sim_summary = tcoffee_seqreformat_simtot.map{\n",
      "Nextflow operator collect with content: }.collect().map{\n",
      "Nextflow operator mix with content: sim_csv = sim_csv.mix(CONCAT_SIMSTATS.out.csv)\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(CALCULATE_SEQSTATS.out.versions.first())\n",
      "Nextflow operator map with content: ch_seqstats_summary = seqstats_summary.map{\n",
      "Nextflow operator collect with content: }.collect().map{\n",
      "Nextflow operator mix with content: seqstats_csv = seqstats_csv.mix(CONCAT_SEQSTATS.out.csv)\n",
      "Nextflow operator map with content: sim      = sim_csv.map{ meta, csv -> csv }\n",
      "Nextflow operator map with content: seqstats = seqstats_csv.map{ meta, csv -> csv }\n",
      "Nextflow operator collect with content: csvs_stats = sim.mix(seqstats).collect().map{ csvs -> [[id:\"summary_stats\"], csvs] }\n",
      "Nextflow operator count with content: def number_of_stats = [params.calc_sim, params.calc_seq_stats].count{ it == true }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(MERGE_STATS.out.versions)\n",
      "Nextflow operator ifEmpty with content: versions         = ch_versions.ifEmpty(null) // channel: [ versions.yml ]\n",
      "Error: Unclosed process block in file: ./git_repos\\multiplesequencealign\\subworkflows\\local\\stats.nf\n",
      "Process MULTIPLESEQUENCEALIGN in filepath ./git_repos\\multiplesequencealign\\workflows\\multiplesequencealign.nf\n",
      "Nextflow operator map with content: ch_tools = Channel.fromSamplesheet('tools').map {\n",
      "Nextflow operator map with content: ch_seqs       = ch_input.map{ meta,fasta,ref,str,template -> [ meta, file(fasta)    ]}\n",
      "Nextflow operator filter with content: ch_refs       = ch_input.filter{ it[2].size() > 0}.map{ meta,fasta,ref,str,template -> [ meta, file(ref)      ]}\n",
      "Nextflow operator filter with content: ch_templates  = ch_input.filter{ it[4].size() > 0}.map{ meta,fasta,ref,str,template -> [ meta, file(template) ]}\n",
      "Nextflow operator filter with content: ch_structures = ch_input.map{ meta,fasta,ref,str,template -> [ meta, str            ]}.filter{ it[1].size() > 0 }\n",
      "Nextflow operator branch with content: ch_structures.branch {\n",
      "Nextflow operator set with content: }.set { ch_structures }\n",
      "Nextflow operator mix with content: .mix( ch_structures.uncompressed )\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator collect with content: [ meta,file(dir).listFiles().collect() ]\n",
      "Nextflow operator set with content: .set { ch_structures }\n",
      "Nextflow operator join with content: ch_structures_template = ch_structures.join(ch_templates, by:0, remainder:true)\n",
      "Nextflow operator branch with content: ch_structures_template.branch{\n",
      "Nextflow operator set with content: }.set { ch_structures_branched }\n",
      "Nextflow operator map with content: .map{\n",
      "Nextflow operator map with content: .map{\n",
      "Nextflow operator mix with content: ch_templates_merged = forced_templates.mix( new_templates)\n",
      "Nextflow operator combine with content: ch_structures_template = ch_templates_merged.combine(ch_structures, by:0)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(STATS.out.versions)\n",
      "Nextflow operator mix with content: stats_summary = stats_summary.mix(STATS.out.stats_summary)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ALIGN.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(EVALUATE.out.versions)\n",
      "Nextflow operator mix with content: evaluation_summary = evaluation_summary.mix(EVALUATE.out.eval_summary)\n",
      "Nextflow operator map with content: stats_summary_csv = stats_summary.map{ meta, csv -> csv }\n",
      "Nextflow operator map with content: eval_summary_csv  = evaluation_summary.map{ meta, csv -> csv }\n",
      "Nextflow operator collect with content: stats_and_evaluation = eval_summary_csv.mix(stats_summary_csv).collect().map{ csvs -> [[id:\"summary_stats_eval\"], csvs] }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(MERGE_STATS_EVAL.out.versions)\n",
      "Nextflow operator collectFile with content: ch_versions.unique().collectFile(name: 'collated_versions.yml')\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ZIP.out.versions)\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_workflow_summary.collectFile(name: 'workflow_summary_mqc.yaml'))\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_methods_description.collectFile(name: 'methods_description_mqc.yaml'))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(CUSTOM_DUMPSOFTWAREVERSIONS.out.mqc_yml.collect())\n",
      "Nextflow operator collect with content: ch_multiqc_table = ch_multiqc_table.mix(PREPARE_MULTIQC.out.multiqc_table.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files.collect(),\n",
      "Nextflow operator toList with content: ch_multiqc_config.toList(),\n",
      "Nextflow operator toList with content: ch_multiqc_custom_config.toList(),\n",
      "Nextflow operator toList with content: ch_multiqc_logo.toList(),\n",
      "Nextflow operator toList with content: multiqc_report = MULTIQC.out.report.toList()\n",
      "Error: Unclosed process block in file: ./git_repos\\multiplesequencealign\\workflows\\multiplesequencealign.nf\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Process NFCORE_NANOSEQ in filepath ./git_repos\\nanoseq\\main.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\nanoseq\\main.nf\n",
      "Process ALIGN_GRAPHMAP2 in filepath ./git_repos\\nanoseq\\subworkflows\\local\\align_graphmap2.nf\n",
      "Nextflow operator cross with content: .cross(ch_fastq) { it -> it[-1] }\n",
      "Nextflow operator flatten with content: .flatten()\n",
      "Nextflow operator collate with content: .collate(12) // [fasta, fasta sizes, gtf, bed, fasta_index, annotation_string, meta, fastq, fasta, gtf, is_transcript, fasta_gtf_string]\n",
      "Nextflow operator map with content: .map { it -> [ it[6], it[7], it[0], it[1], it[2], it[3], it[10], it[4] ] } // [ sample, fastq, fasta, sizes, gtf, bed, is_transcripts, index ]\n",
      "Nextflow operator set with content: .set { ch_index }\n",
      "Error: Unclosed process block in file: ./git_repos\\nanoseq\\subworkflows\\local\\align_graphmap2.nf\n",
      "Process ALIGN_MINIMAP2 in filepath ./git_repos\\nanoseq\\subworkflows\\local\\align_minimap2.nf\n",
      "Nextflow operator cross with content: .cross(ch_fastq) { it -> it[-1] }\n",
      "Nextflow operator flatten with content: .flatten()\n",
      "Nextflow operator collate with content: .collate(13)\n",
      "Nextflow operator map with content: .map { it -> [ it[7], it[8], it[0], it[1], it[2], it[3], it[4], it[5] ] } // [ sample, fastq, fasta, sizes, gtf, bed, is_transcripts, index ]\n",
      "Nextflow operator set with content: .set { ch_index }\n",
      "Error: Unclosed process block in file: ./git_repos\\nanoseq\\subworkflows\\local\\align_minimap2.nf\n",
      "Process BAM_SORT_INDEX_SAMTOOLS in filepath ./git_repos\\nanoseq\\subworkflows\\local\\bam_sort_index_samtools.nf\n",
      "Nextflow operator join with content: .join( SAMTOOLS_SORT_INDEX.out.bam_bai )\n",
      "Nextflow operator map with content: .map { it -> [ it[0], it[1], it[2], it[4], it[5] ] }\n",
      "Nextflow operator set with content: .set { sortbam }\n",
      "Nextflow operator join with content: .join( SAMTOOLS_SORT.out.bam )\n",
      "Nextflow operator join with content: .join( SAMTOOLS_INDEX.out.bai )\n",
      "Nextflow operator map with content: .map { it -> [ it[0], it[1], it[2], it[4], it[5] ] }\n",
      "Nextflow operator set with content: .set { sortbam }\n",
      "Nextflow operator join with content: BAM_STATS_SAMTOOLS ( SAMTOOLS_SORT.out.bam.join(SAMTOOLS_INDEX.out.bai, by: [0]), ch_fasta )\n",
      "Nextflow operator join with content: .join ( BAM_STATS_SAMTOOLS.out.idxstats )\n",
      "Nextflow operator join with content: .join ( BAM_STATS_SAMTOOLS.out.flagstat )\n",
      "Nextflow operator map with content: .map  { it -> [ it[1], it[2], it[3] ] }\n",
      "Nextflow operator set with content: .set  { sortbam_stats_multiqc }\n",
      "Error: Unclosed process block in file: ./git_repos\\nanoseq\\subworkflows\\local\\bam_sort_index_samtools.nf\n",
      "Process BEDTOOLS_UCSC_BIGBED in filepath ./git_repos\\nanoseq\\subworkflows\\local\\bedtools_ucsc_bigbed.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\nanoseq\\subworkflows\\local\\bedtools_ucsc_bigbed.nf\n",
      "Process BEDTOOLS_UCSC_BIGWIG in filepath ./git_repos\\nanoseq\\subworkflows\\local\\bedtools_ucsc_bigwig.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\nanoseq\\subworkflows\\local\\bedtools_ucsc_bigwig.nf\n",
      "Process DIFFERENTIAL_DESEQ2_DEXSEQ in filepath ./git_repos\\nanoseq\\subworkflows\\local\\differential_deseq2_dexseq.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\nanoseq\\subworkflows\\local\\differential_deseq2_dexseq.nf\n",
      "Process INPUT_CHECK in filepath ./git_repos\\nanoseq\\subworkflows\\local\\input_check.nf\n",
      "Nextflow operator splitCsv with content: .splitCsv ( header:true, sep:',' )\n",
      "Nextflow operator map with content: .map { get_sample_info(it, params.genomes) }\n",
      "Nextflow operator map with content: .map { it -> [ it[0], it[2], it[3], it[4], it[5], it[6], it[1] , it[7] ] }\n",
      "Nextflow operator set with content: .set { ch_sample }\n",
      "Error: Unclosed process block in file: ./git_repos\\nanoseq\\subworkflows\\local\\input_check.nf\n",
      "Process PREPARE_GENOME in filepath ./git_repos\\nanoseq\\subworkflows\\local\\prepare_genome.nf\n",
      "Nextflow operator filter with content: .filter { it[2] }\n",
      "Nextflow operator map with content: .map { it -> [ it[2], it[5].toString() ] }  // [ fasta, annotation_str ]\n",
      "Nextflow operator unique with content: .unique()\n",
      "Nextflow operator set with content: .set { ch_fastq_sizes }\n",
      "Nextflow operator filter with content: .filter { it[3] }\n",
      "Nextflow operator map with content: .map { it -> [ it[3], it[5] ] }  // [ gtf, annotation_str ]\n",
      "Nextflow operator unique with content: .unique()\n",
      "Nextflow operator set with content: .set { ch_fastq_gtf }\n",
      "Nextflow operator join with content: .join(ch_gtf_bed, by: 1, remainder:true)\n",
      "Nextflow operator map with content: .map { it -> [ it[1], it[2], it[0] ] }\n",
      "Nextflow operator cross with content: .cross(ch_fastq) { it -> it[-1] }\n",
      "Nextflow operator flatten with content: .flatten()\n",
      "Nextflow operator collate with content: .collate(9)\n",
      "Nextflow operator map with content: .map { it -> [ it[5], it[0], it[6], it[1], it[7], it[8] ]} // [ fasta, sizes, gtf, bed, is_transcripts, annotation_str ]\n",
      "Nextflow operator unique with content: .unique()\n",
      "Nextflow operator set with content: .set { ch_fasta_index }\n",
      "Nextflow operator filter with content: .filter { it[2] }\n",
      "Nextflow operator map with content: .map { it -> [ it[0], it[2] ] }  // [ gtf, annotation_str ]\n",
      "Nextflow operator unique with content: .unique()\n",
      "Nextflow operator set with content: .set { ch_fasta }\n",
      "Error: Unclosed process block in file: ./git_repos\\nanoseq\\subworkflows\\local\\prepare_genome.nf\n",
      "Process QCFASTQ_NANOPLOT_FASTQC in filepath ./git_repos\\nanoseq\\subworkflows\\local\\qcfastq_nanoplot_fastqc.nf\n",
      "Nextflow operator map with content: .map { ch -> [ ch[0], ch[1] ] }\n",
      "Nextflow operator set with content: .set { ch_fastq }\n",
      "Nextflow operator map with content: .map { it -> [ it[1] ] }\n",
      "Nextflow operator set with content: .set { fastqc_zip_only }\n",
      "Nextflow operator map with content: .map { it -> [ it[1] ] }\n",
      "Nextflow operator set with content: .set { fastqc_html_only }\n",
      "Nextflow operator mix with content: fastqc_multiqc = fastqc_multiqc.mix( fastqc_zip_only, fastqc_html_only )\n",
      "Error: Unclosed process block in file: ./git_repos\\nanoseq\\subworkflows\\local\\qcfastq_nanoplot_fastqc.nf\n",
      "Process QUANTIFY_STRINGTIE_FEATURECOUNTS in filepath ./git_repos\\nanoseq\\subworkflows\\local\\quantify_stringtie_featurecounts.nf\n",
      "Nextflow operator map with content: .map  { it -> [ it[0], it[2], it[3] ] }\n",
      "Nextflow operator join with content: .join ( ch_sortbam )\n",
      "Nextflow operator set with content: .set  { ch_sample }\n",
      "Nextflow operator map with content: .map { it -> [ it[2] ] }\n",
      "Nextflow operator unique with content: .unique()\n",
      "Nextflow operator set with content: .set { ch_sample_gtf }\n",
      "Nextflow operator collect with content: STRINGTIE_MERGE ( ch_stringtie_gtf.collect(), ch_sample_gtf )\n",
      "Nextflow operator collect with content: .collect { it[-1]    }\n",
      "Nextflow operator set with content: .set     { ch_sample }\n",
      "Error: Unclosed process block in file: ./git_repos\\nanoseq\\subworkflows\\local\\quantify_stringtie_featurecounts.nf\n",
      "Process RNA_FUSIONS_JAFFAL in filepath ./git_repos\\nanoseq\\subworkflows\\local\\rna_fusions_jaffal.nf\n",
      "Nextflow operator map with content: .map { it -> [ it[1] ]}\n",
      "Nextflow operator set with content: .set { ch_jaffal_ref_dir }\n",
      "Nextflow operator view with content: ch_jaffal_ref_dir.view()\n",
      "Nextflow operator view with content: ch_sample_simple.view()\n",
      "Error: Unclosed process block in file: ./git_repos\\nanoseq\\subworkflows\\local\\rna_fusions_jaffal.nf\n",
      "Process RNA_MODIFICATION_XPORE_M6ANET in filepath ./git_repos\\nanoseq\\subworkflows\\local\\rna_modifications_xpore_m6anet.nf\n",
      "Nextflow operator join with content: .join(ch_nanopolish_sortbam)\n",
      "Nextflow operator map with content: .map { it -> [ it[0], it[2], it[3], it[7], it[6], it[8], it[9] ] }\n",
      "Nextflow operator set with content: .set { ch_nanopolish_input }\n",
      "Nextflow operator map with content: .map{ it -> it[1]+','+it[0].id }\n",
      "Nextflow operator set with content: .set{ ch_xpore_diffmod_inputs }\n",
      "Nextflow operator collect with content: XPORE_DIFFMOD{ ch_xpore_diffmod_inputs.collect() }\n",
      "Error: Unclosed process block in file: ./git_repos\\nanoseq\\subworkflows\\local\\rna_modifications_xpore_m6anet.nf\n",
      "Process SHORT_VARIANT_CALLING in filepath ./git_repos\\nanoseq\\subworkflows\\local\\short_variant_calling.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(medaka_version = MEDAKA_VARIANT.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(bgzip_version = MEDAKA_BGZIP_VCF.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(tabix_version = MEDAKA_TABIX_VCF.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(DEEPVARIANT.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(DEEPVARIANT_TABIX_VCF.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(DEEPVARIANT_TABIX_VCF.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(PEPPER_MARGIN_DEEPVARIANT.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\nanoseq\\subworkflows\\local\\short_variant_calling.nf\n",
      "Process STRUCTURAL_VARIANT_CALLING in filepath ./git_repos\\nanoseq\\subworkflows\\local\\structural_variant_calling.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SNIFFLES.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SNIFFLES_SORT_VCF.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SNIFFLES_TABIX_VCF.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(CUTESV.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(CUTESV_SORT_VCF.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(CUTESV_TABIX_VCF.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\nanoseq\\subworkflows\\local\\structural_variant_calling.nf\n",
      "Process BAM_STATS_SAMTOOLS in filepath ./git_repos\\nanoseq\\subworkflows\\nf-core\\bam_stats_samtools.nf\n",
      "Nextflow operator map with content: .map{ it -> it[1] }\n",
      "Nextflow operator set with content: .set{ ch_fasta }\n",
      "Error: Unclosed process block in file: ./git_repos\\nanoseq\\subworkflows\\nf-core\\bam_stats_samtools.nf\n",
      "Process NANOSEQ in filepath ./git_repos\\nanoseq\\workflows\\nanoseq.nf\n",
      "Nextflow operator set with content: .set { ch_input_path }\n",
      "Nextflow operator set with content: .set { ch_sample }\n",
      "Nextflow operator flatten with content: .flatten()\n",
      "Nextflow operator map with content: .map { it -> [ it, it.baseName.substring(0,it.baseName.lastIndexOf('.'))] }\n",
      "Nextflow operator join with content: .join(ch_sample, by: 1) // join on barcode\n",
      "Nextflow operator map with content: .map { it -> [ it[2], it[1], it[3], it[4], it[5], it[6] ] }\n",
      "Nextflow operator set with content: .set { ch_fastq }\n",
      "Nextflow operator ifEmpty with content: ch_software_versions = ch_software_versions.mix(QCAT.out.versions.ifEmpty(null))\n",
      "Nextflow operator map with content: .map { it -> if (it[6].toString().endsWith('.gz')) [ it[0], it[6], it[2], it[1], it[4], it[5] ] }\n",
      "Nextflow operator set with content: .set { ch_fastq }\n",
      "Nextflow operator map with content: .map { it -> [ it[0], it[1] ] }\n",
      "Nextflow operator set with content: .set { ch_fastq_nanolyse }\n",
      "Nextflow operator set with content: .set{ ch_nanolyse_fasta }\n",
      "Nextflow operator join with content: .join( ch_sample )\n",
      "Nextflow operator map with content: .map { it -> [ it[0], it[1], it[3], it[4], it[5], it[6] ]}\n",
      "Nextflow operator set with content: .set { ch_fastq }\n",
      "Nextflow operator first with content: ch_software_versions = ch_software_versions.mix(NANOLYSE.out.versions.first().ifEmpty(null))\n",
      "Nextflow operator first with content: ch_software_versions = ch_software_versions.mix(QCFASTQ_NANOPLOT_FASTQC.out.fastqc_version.first().ifEmpty(null))\n",
      "Nextflow operator ifEmpty with content: ch_fastqc_multiqc    = QCFASTQ_NANOPLOT_FASTQC.out.fastqc_multiqc.ifEmpty([])\n",
      "Nextflow operator first with content: ch_software_versions = ch_software_versions.mix(PREPARE_GENOME.out.samtools_version.first().ifEmpty(null))\n",
      "Nextflow operator first with content: ch_software_versions = ch_software_versions.mix(PREPARE_GENOME.out.gtf2bed_version.first().ifEmpty(null))\n",
      "Nextflow operator first with content: ch_software_versions = ch_software_versions.mix(ALIGN_MINIMAP2.out.minimap2_version.first().ifEmpty(null))\n",
      "Nextflow operator first with content: ch_software_versions = ch_software_versions.mix(ALIGN_GRAPHMAP2.out.graphmap2_version.first().ifEmpty(null))\n",
      "Nextflow operator first with content: ch_software_versions = ch_software_versions.mix(BAM_SORT_INDEX_SAMTOOLS.out.samtools_versions.first().ifEmpty(null))\n",
      "Nextflow operator ifEmpty with content: ch_samtools_multiqc  = BAM_SORT_INDEX_SAMTOOLS.out.sortbam_stats_multiqc.ifEmpty([])\n",
      "Nextflow operator map with content: SHORT_VARIANT_CALLING ( ch_view_sortbam, ch_fasta.map{ it [1] }, ch_fai.map{ it [1] } )\n",
      "Nextflow operator first with content: ch_software_versions = ch_software_versions.mix(SHORT_VARIANT_CALLING.out.ch_versions.first().ifEmpty(null))\n",
      "Nextflow operator map with content: STRUCTURAL_VARIANT_CALLING ( ch_view_sortbam, ch_fasta.map{ it [1] }, ch_fai.map{ it [1] } )\n",
      "Nextflow operator first with content: ch_software_versions = ch_software_versions.mix(STRUCTURAL_VARIANT_CALLING.out.ch_versions.first().ifEmpty(null))\n",
      "Nextflow operator first with content: ch_bedtools_version = ch_bedtools_version.mix(BEDTOOLS_UCSC_BIGWIG.out.bedtools_version.first().ifEmpty(null))\n",
      "Nextflow operator first with content: ch_software_versions = ch_software_versions.mix(BEDTOOLS_UCSC_BIGWIG.out.bedgraphtobigwig_version.first().ifEmpty(null))\n",
      "Nextflow operator first with content: ch_bedtools_version = ch_bedtools_version.mix(BEDTOOLS_UCSC_BIGBED.out.bedtools_version.first().ifEmpty(null))\n",
      "Nextflow operator first with content: ch_software_versions = ch_software_versions.mix(BEDTOOLS_UCSC_BIGBED.out.bed12tobigbed_version.first().ifEmpty(null))\n",
      "Nextflow operator first with content: ch_software_versions = ch_software_versions.mix(ch_bedtools_version.first().ifEmpty(null))\n",
      "Nextflow operator map with content: .map { it -> [ it[0], it[3] ] }\n",
      "Nextflow operator set with content: .set { ch_sortbam }\n",
      "Nextflow operator map with content: .map { it -> [ it[0], it[3], it[4] ] }\n",
      "Nextflow operator set with content: .set { ch_nanopolish_sortbam }\n",
      "Nextflow operator map with content: .map { it -> if (it[6].toString().endsWith('.bam')) [ it[0], it[6] ] }\n",
      "Nextflow operator set with content: .set { ch_sample_bam }\n",
      "Nextflow operator map with content: ch_sample.map{ it[2] }.unique().toList().set { fastas }\n",
      "Nextflow operator map with content: ch_sample.map{ it[3] }.unique().toList().set { gtfs }\n",
      "Nextflow operator map with content: .map { it -> [ it[2], it[3] ]}\n",
      "Nextflow operator unique with content: .unique()\n",
      "Nextflow operator set with content: .set { ch_sample_annotation }\n",
      "Nextflow operator collect with content: BAMBU ( ch_sample_annotation, ch_sortbam.collect{ it [1] } )\n",
      "Nextflow operator first with content: ch_software_versions = ch_software_versions.mix(BAMBU.out.versions.first().ifEmpty(null))\n",
      "Nextflow operator first with content: ch_software_versions                = ch_software_versions.mix(QUANTIFY_STRINGTIE_FEATURECOUNTS.out.stringtie2_version.first().ifEmpty(null))\n",
      "Nextflow operator first with content: ch_software_versions                = ch_software_versions.mix(QUANTIFY_STRINGTIE_FEATURECOUNTS.out.featurecounts_version.first().ifEmpty(null))\n",
      "Nextflow operator ifEmpty with content: ch_featurecounts_gene_multiqc       = QUANTIFY_STRINGTIE_FEATURECOUNTS.out.featurecounts_gene_multiqc.ifEmpty([])\n",
      "Nextflow operator ifEmpty with content: ch_featurecounts_transcript_multiqc = QUANTIFY_STRINGTIE_FEATURECOUNTS.out.featurecounts_transcript_multiqc.ifEmpty([])\n",
      "Nextflow operator first with content: ch_software_versions = ch_software_versions.mix(DIFFERENTIAL_DESEQ2_DEXSEQ.out.deseq2_version.first().ifEmpty(null))\n",
      "Nextflow operator first with content: ch_software_versions = ch_software_versions.mix(DIFFERENTIAL_DESEQ2_DEXSEQ.out.dexseq_version.first().ifEmpty(null))\n",
      "Nextflow operator map with content: .map { it -> [ it[0], it[1] ] }\n",
      "Nextflow operator set with content: .set { ch_fastq_simple }\n",
      "Nextflow operator collectFile with content: ch_software_versions.unique().collectFile()\n",
      "Nextflow operator collect with content: ch_multiqc_custom_config.collect().ifEmpty([]),\n",
      "Nextflow operator ifEmpty with content: ch_fastqc_multiqc.ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_samtools_multiqc.collect().ifEmpty([]),\n",
      "Nextflow operator ifEmpty with content: ch_featurecounts_gene_multiqc.ifEmpty([]),\n",
      "Nextflow operator ifEmpty with content: ch_featurecounts_transcript_multiqc.ifEmpty([]),\n",
      "Nextflow operator collect with content: CUSTOM_DUMPSOFTWAREVERSIONS.out.mqc_yml.collect(),\n",
      "Nextflow operator collectFile with content: ch_workflow_summary.collectFile(name: 'workflow_summary_mqc.yaml')\n",
      "Error: Unclosed process block in file: ./git_repos\\nanoseq\\workflows\\nanoseq.nf\n",
      "Process NFCORE_NANOSTRING in filepath ./git_repos\\nanostring\\main.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\nanostring\\main.nf\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Process INPUT_CHECK in filepath ./git_repos\\nanostring\\subworkflows\\local\\input_check.nf\n",
      "Nextflow operator splitCsv with content: .splitCsv ( header:true, sep:',' )\n",
      "Nextflow operator map with content: .map { create_rcc_channel(it) }\n",
      "Nextflow operator set with content: .set { counts }\n",
      "Error: Unclosed process block in file: ./git_repos\\nanostring\\subworkflows\\local\\input_check.nf\n",
      "Process NORMALIZE in filepath ./git_repos\\nanostring\\subworkflows\\local\\normalize.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\nanostring\\subworkflows\\local\\normalize.nf\n",
      "Process QUALITY_CONTROL in filepath ./git_repos\\nanostring\\subworkflows\\local\\quality_control.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\nanostring\\subworkflows\\local\\quality_control.nf\n",
      "Process NANOSTRING in filepath ./git_repos\\nanostring\\workflows\\nanostring.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(INPUT_CHECK.out.versions)\n",
      "Nextflow operator collect with content: .map { meta, path -> path}.collect()\n",
      "Nextflow operator set with content: .set{ rcc_files }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(QUALITY_CONTROL.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(NORMALIZE.out.versions)\n",
      "Nextflow operator mix with content: NORMALIZE.out.normalized_counts.mix(NORMALIZE.out.normalized_counts_wo_HK),\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(CREATE_ANNOTATED_TABLES.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(COMPUTE_GENE_SCORES.out.versions)\n",
      "Nextflow operator toList with content: ch_heatmap_genes_to_filter.toList()\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(CREATE_GENE_HEATMAP.out.versions)\n",
      "Nextflow operator collectFile with content: ch_versions.unique().collectFile(name: 'collated_versions.yml')\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_workflow_summary.collectFile(name: 'workflow_summary_mqc.yaml'))\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_methods_description.collectFile(name: 'methods_description_mqc.yaml'))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(QUALITY_CONTROL.out.nacho_qc_multiqc_metrics.collect())\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(CUSTOM_DUMPSOFTWAREVERSIONS.out.mqc_yml.collect())\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(COMPUTE_GENE_SCORES.out.scores_for_mqc.collect())\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(CREATE_ANNOTATED_TABLES.out.annotated_data_mqc.collect())\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(CREATE_GENE_HEATMAP.out.gene_heatmap.collect())\n",
      "Nextflow operator collect with content: ch_multiqc_files.collect(),\n",
      "Nextflow operator toList with content: ch_multiqc_config.toList(),\n",
      "Nextflow operator toList with content: ch_multiqc_custom_config.toList(),\n",
      "Nextflow operator toList with content: ch_multiqc_logo.toList()\n",
      "Nextflow operator toList with content: multiqc_report = MULTIQC.out.report.toList()\n",
      "Error: Unclosed process block in file: ./git_repos\\nanostring\\workflows\\nanostring.nf\n",
      "Process NFCORE_NASCENT in filepath ./git_repos\\nascent\\main.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\nascent\\main.nf\n",
      "Process COVERAGE_GRAPHS in filepath ./git_repos\\nascent\\subworkflows\\local\\coverage_graphs.nf\n",
      "Nextflow operator combine with content: ch_genomecov_bam = bam.combine(Channel.from(1))\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(BEDTOOLS_GENOMECOV_PLUS.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(BEDTOOLS_GENOMECOV_MINUS.out.versions.first())\n",
      "Nextflow operator join with content: bam.join(bai, by: [0], remainder: true).set { ch_bam_bai }\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(DEEPTOOLS_BAMCOVERAGE_PLUS.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(DEEPTOOLS_BAMCOVERAGE_MINUS.out.versions.first())\n",
      "Nextflow operator join with content: ch_plus_minus = DEEPTOOLS_BAMCOVERAGE_PLUS.out.bigwig.join(DEEPTOOLS_BAMCOVERAGE_MINUS.out.bigwig)\n",
      "Error: Unclosed process block in file: ./git_repos\\nascent\\subworkflows\\local\\coverage_graphs.nf\n",
      "Process GROHMM in filepath ./git_repos\\nascent\\subworkflows\\local\\grohmm.nf\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(GROHMM_PARAMETERTUNING.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(GROHMM_TRANSCRIPTCALLING.out.versions.first())\n",
      "Error: Unclosed process block in file: ./git_repos\\nascent\\subworkflows\\local\\grohmm.nf\n",
      "Process INPUT_CHECK in filepath ./git_repos\\nascent\\subworkflows\\local\\input_check.nf\n",
      "Nextflow operator splitCsv with content: .splitCsv ( header:true, sep:',' )\n",
      "Nextflow operator map with content: .map { create_fastq_channel(it) }\n",
      "Nextflow operator set with content: .set { reads }\n",
      "Error: Unclosed process block in file: ./git_repos\\nascent\\subworkflows\\local\\input_check.nf\n",
      "Process PREPARE_GENOME in filepath ./git_repos\\nascent\\subworkflows\\local\\prepare_genome.nf\n",
      "Nextflow operator map with content: ch_fasta = GUNZIP_FASTA ( [ [:], params.fasta ] ).gunzip.map { it[1] }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GUNZIP_FASTA.out.versions)\n",
      "Nextflow operator map with content: ch_fai = SAMTOOLS_FAIDX( [ [:], ch_fasta ] ).fai.map { it[1] }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_FAIDX.out.versions)\n",
      "Nextflow operator map with content: ch_gtf = GUNZIP_GTF ( [ [:], params.gtf ] ).gunzip.map { it[1] }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GUNZIP_GTF.out.versions)\n",
      "Nextflow operator map with content: ch_gff = GUNZIP_GFF ( [ [:], params.gff ] ).gunzip.map { it[1] }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GUNZIP_GFF.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GFFREAD.out.versions)\n",
      "Nextflow operator map with content: ch_gene_bed = GUNZIP_GENE_BED ( [ [:], params.gene_bed ] ).gunzip.map { it[1] }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GUNZIP_GENE_BED.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GTF2BED.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(CUSTOM_GETCHROMSIZES.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(UNTAR_BWA_INDEX.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BWA_INDEX.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(UNTAR_BWA_INDEX.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BWAMEM2_INDEX.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(UNTAR_DRAGMAP_INDEX.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(DRAGMAP_HASHTABLE.out.versions)\n",
      "Nextflow operator ifEmpty with content: versions = ch_versions.ifEmpty(null)\n",
      "Error: Unclosed process block in file: ./git_repos\\nascent\\subworkflows\\local\\prepare_genome.nf\n",
      "Process QUALITY_CONTROL in filepath ./git_repos\\nascent\\subworkflows\\local\\quality_control.nf\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(PRESEQ_CCURVE.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(PRESEQ_LCEXTRAP.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(RSEQC_READDISTRIBUTION.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(RSEQC_READDUPLICATION.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(RSEQC_INFEREXPERIMENT.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(BBMAP_PILEUP.out.versions.first())\n",
      "Error: Unclosed process block in file: ./git_repos\\nascent\\subworkflows\\local\\quality_control.nf\n",
      "Process TRANSCRIPT_INDENTIFICATION in filepath ./git_repos\\nascent\\subworkflows\\local\\transcript_identification.nf\n",
      "Nextflow operator mix with content: ch_identification_bed = ch_identification_bed.mix(GROHMM.out.bed)\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(GROHMM.out.versions.first())\n",
      "Nextflow operator mix with content: ch_identification_bed = ch_identification_bed.mix(HOMER_GROSEQ.out.bed)\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(HOMER_GROSEQ.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(PINTS_CALLER.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(CAT_CAT.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(BEDTOOLS_SORT.out.versions.first())\n",
      "Nextflow operator mix with content: ch_identification_bed = ch_identification_bed.mix(BEDTOOLS_MERGE.out.bed)\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(BEDTOOLS_MERGE.out.versions.first())\n",
      "Nextflow operator combine with content: BEDTOOLS_INTERSECT_FILTER ( ch_identification_bed.combine(ch_filter_bed), \"bed\" )\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(BEDTOOLS_INTERSECT_FILTER.out.versions.first())\n",
      "Nextflow operator combine with content: BEDTOOLS_INTERSECT ( ch_identification_bed.combine(ch_intersect_bed), \"bed\" )\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(BEDTOOLS_INTERSECT_FILTER.out.versions.first())\n",
      "Nextflow operator filter with content: .filter { meta, bed -> bed.size() > 0 }\n",
      "Nextflow operator set with content: .set { ch_identification_bed_clean }\n",
      "Error: Unclosed process block in file: ./git_repos\\nascent\\subworkflows\\local\\transcript_identification.nf\n",
      "Process ALIGN_BWA in filepath ./git_repos\\nascent\\subworkflows\\local\\align_bwa\\main.nf\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(BWA_MEM.out.versions.first())\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BAM_SORT_STATS_SAMTOOLS.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\nascent\\subworkflows\\local\\align_bwa\\main.nf\n",
      "Process ALIGN_BWAMEM2 in filepath ./git_repos\\nascent\\subworkflows\\local\\align_bwamem2\\main.nf\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(BWAMEM2_MEM.out.versions.first())\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BAM_SORT_STATS_SAMTOOLS.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\nascent\\subworkflows\\local\\align_bwamem2\\main.nf\n",
      "Process ALIGN_DRAGMAP in filepath ./git_repos\\nascent\\subworkflows\\local\\align_dragmap\\main.nf\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(DRAGMAP_ALIGN.out.versions.first())\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BAM_SORT_STATS_SAMTOOLS.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\nascent\\subworkflows\\local\\align_dragmap\\main.nf\n",
      "Process BAM_DEDUP_STATS_SAMTOOLS_UMITOOLS in filepath ./git_repos\\nascent\\subworkflows\\nf-core\\bam_dedup_stats_samtools_umitools\\main.nf\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(UMITOOLS_DEDUP.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SAMTOOLS_INDEX.out.versions.first())\n",
      "Nextflow operator join with content: .join(SAMTOOLS_INDEX.out.bai, by: [0], remainder: true)\n",
      "Nextflow operator join with content: .join(SAMTOOLS_INDEX.out.csi, by: [0], remainder: true)\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator set with content: .set { ch_bam_bai }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BAM_STATS_SAMTOOLS.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\nascent\\subworkflows\\nf-core\\bam_dedup_stats_samtools_umitools\\main.nf\n",
      "Process BAM_SORT_STATS_SAMTOOLS in filepath ./git_repos\\nascent\\subworkflows\\nf-core\\bam_sort_stats_samtools\\main.nf\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SAMTOOLS_SORT.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SAMTOOLS_INDEX.out.versions.first())\n",
      "Nextflow operator join with content: .join(SAMTOOLS_INDEX.out.bai, by: [0], remainder: true)\n",
      "Nextflow operator join with content: .join(SAMTOOLS_INDEX.out.csi, by: [0], remainder: true)\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator set with content: .set { ch_bam_bai }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BAM_STATS_SAMTOOLS.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\nascent\\subworkflows\\nf-core\\bam_sort_stats_samtools\\main.nf\n",
      "Process BAM_STATS_SAMTOOLS in filepath ./git_repos\\nascent\\subworkflows\\nf-core\\bam_stats_samtools\\main.nf\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SAMTOOLS_STATS.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SAMTOOLS_FLAGSTAT.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SAMTOOLS_IDXSTATS.out.versions.first())\n",
      "Error: Unclosed process block in file: ./git_repos\\nascent\\subworkflows\\nf-core\\bam_stats_samtools\\main.nf\n",
      "Process HOMER_GROSEQ in filepath ./git_repos\\nascent\\subworkflows\\nf-core\\homer\\groseq\\main.nf\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(HOMER_MAKETAGDIRECTORY.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(HOMER_MAKEUCSCFILE.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(HOMER_FINDPEAKS.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(HOMER_POS2BED.out.versions.first())\n",
      "Error: Unclosed process block in file: ./git_repos\\nascent\\subworkflows\\nf-core\\homer\\groseq\\main.nf\n",
      "Process NASCENT in filepath ./git_repos\\nascent\\workflows\\nascent.nf\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(PREPARE_GENOME.out.versions.first())\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(INPUT_CHECK.out.versions)\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(FASTQC.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(FASTP.out.versions.first())\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Nextflow operator first with content: ch_versions = ch_versions.mix(ALIGN_BWA.out.versions.first())\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ALIGN_BWAMEM2.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ALIGN_DRAGMAP.out.versions)\n",
      "Nextflow operator join with content: ch_genome_bam.join(ch_genome_bai, by: [0]),\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BAM_DEDUP_STATS_SAMTOOLS_UMITOOLS.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(QUALITY_CONTROL.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(COVERAGE_GRAPHS.out.versions)\n",
      "Nextflow operator map with content: ch_genome_bam.map {\n",
      "Nextflow operator join with content: fmeta.id = fmeta.id.split('_')[0..-3].join('_')\n",
      "Nextflow operator groupTuple with content: .groupTuple(by: [0])\n",
      "Nextflow operator flatten with content: .map { it ->  [ it[0], it[1].flatten() ] }\n",
      "Nextflow operator set with content: .set { ch_sort_bam }\n",
      "Nextflow operator collect with content: ch_grohmm_multiqc = TRANSCRIPT_INDENTIFICATION.out.grohmm_td_plot.collect()\n",
      "Nextflow operator mix with content: ch_homer_multiqc = ch_homer_multiqc.mix(TRANSCRIPT_INDENTIFICATION.out.homer_tagdir)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(TRANSCRIPT_INDENTIFICATION.out.versions)\n",
      "Nextflow operator combine with content: ch_sort_bam.combine(\n",
      "Nextflow operator map with content: ).saf.map { it[1] }\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SUBREAD_FEATURECOUNTS_PREDICTED.out.versions.first())\n",
      "Nextflow operator combine with content: ch_sort_bam.combine(PREPARE_GENOME.out.gtf)\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SUBREAD_FEATURECOUNTS_GENE.out.versions.first())\n",
      "Nextflow operator collectFile with content: ch_versions.unique().collectFile(name: 'collated_versions.yml')\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_workflow_summary.collectFile(name: 'workflow_summary_mqc.yaml'))\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_methods_description.collectFile(name: 'methods_description_mqc.yaml'))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(CUSTOM_DUMPSOFTWAREVERSIONS.out.mqc_yml.collect())\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(FASTQC.out.zip.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(ch_samtools_stats.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(ch_samtools_flagstat.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(ch_samtools_idxstats.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(QUALITY_CONTROL.out.preseq_ccurve.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(QUALITY_CONTROL.out.preseq_lcextrap.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(QUALITY_CONTROL.out.readdistribution_txt.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(QUALITY_CONTROL.out.readduplication_seq_xls.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(QUALITY_CONTROL.out.readduplication_pos_xls.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(QUALITY_CONTROL.out.inferexperiment_txt.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(ch_grohmm_multiqc.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(ch_homer_multiqc.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(SUBREAD_FEATURECOUNTS_PREDICTED.out.summary.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(SUBREAD_FEATURECOUNTS_GENE.out.summary.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files.collect(),\n",
      "Nextflow operator toList with content: ch_multiqc_config.toList(),\n",
      "Nextflow operator toList with content: ch_multiqc_custom_config.toList(),\n",
      "Nextflow operator toList with content: ch_multiqc_logo.toList()\n",
      "Nextflow operator toList with content: multiqc_report = MULTIQC.out.report.toList()\n",
      "Error: Unclosed process block in file: ./git_repos\\nascent\\workflows\\nascent.nf\n",
      "Process NFCORE_PANGENOME in filepath ./git_repos\\pangenome\\main.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\pangenome\\main.nf\n",
      "Process COMMUNITY in filepath ./git_repos\\pangenome\\subworkflows\\local\\community.nf\n",
      "Nextflow operator map with content: ch_wfmash_map = fasta.map{meta, fasta -> [ meta, fasta, [] ]}\n",
      "Nextflow operator join with content: ch_wfmash_map = ch_wfmash_map.join(gzi).join(fai)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(WFMASH_MAP_COMMUNITY.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(PAF2NET.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(NET2COMMUNITIES.out.versions)\n",
      "Nextflow operator combine with content: ch_txt_communities = fasta.combine(NET2COMMUNITIES.out.communities.flatten())\n",
      "Nextflow operator map with content: ch_txt_communities = ch_txt_communities.map{meta, fasta, community -> [ [ id: community.baseName.split(\"//.\")[-1] ], fasta, community ]}\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(EXTRACT_COMMUNITIES.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(TABIX_BGZIP.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_FAIDX.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\pangenome\\subworkflows\\local\\community.nf\n",
      "Process INPUT_CHECK in filepath ./git_repos\\pangenome\\subworkflows\\local\\input_check.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_FAIDX.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_FAIDX.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(TABIX_BGZIP.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\pangenome\\subworkflows\\local\\input_check.nf\n",
      "Process ODGI_QC in filepath ./git_repos\\pangenome\\subworkflows\\local\\odgi_qc.nf\n",
      "Nextflow operator mix with content: ODGI_STATS(seqwish.mix(sorted_graph))\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ODGI_STATS.out.versions)\n",
      "Nextflow operator map with content: ch_odgi_viz = sorted_graph.map{meta, gfa -> [ [ id: meta.id + \".viz\" ], gfa ]}\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ODGI_VIZ_COLOR.out.versions)\n",
      "Nextflow operator map with content: ch_odgi_viz_pos = sorted_graph.map{meta, gfa -> [ [ id: meta.id + \".viz_pos\" ], gfa ]}\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ODGI_VIZ_POS.out.versions)\n",
      "Nextflow operator map with content: ch_odgi_viz_depth = sorted_graph.map{meta, gfa -> [ [ id: meta.id + \".viz_depth\" ], gfa ]}\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ODGI_VIZ_DEPTH.out.versions)\n",
      "Nextflow operator map with content: ch_odgi_viz_inv = sorted_graph.map{meta, gfa -> [ [ id: meta.id + \".viz_inv\" ], gfa ]}\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ODGI_VIZ_INV.out.versions)\n",
      "Nextflow operator map with content: ch_odgi_viz_compr = sorted_graph.map{meta, gfa -> [ [ id: meta.id + \".viz_O\" ], gfa ]}\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ODGI_VIZ_COMPR.out.versions)\n",
      "Nextflow operator map with content: ch_odgi_viz_uncalled = sorted_graph.map{meta, gfa -> [ [ id: meta.id + \".viz_uncalled\" ], gfa ]}\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ODGI_VIZ_UNCALLED.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ODGI_LAYOUT.out.versions)\n",
      "Nextflow operator join with content: ODGI_DRAW_MULTIQC(sorted_graph.join(ODGI_LAYOUT.out.lay))\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ODGI_DRAW_MULTIQC.out.versions)\n",
      "Nextflow operator join with content: ODGI_DRAW_HEIGHT(sorted_graph.join(ODGI_LAYOUT.out.lay))\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ODGI_DRAW_HEIGHT.out.versions)\n",
      "Nextflow operator groupTuple with content: ch_graph_qc = ODGI_STATS.out.yaml.map{meta, ymls -> [ [ id: meta.id.replaceFirst(\".seqwish\", \"\").replaceFirst(\".gfaffix\", \"\") ], ymls ]}.groupTuple(by:0, size:2).map{meta, ymls -> [ meta, ymls[0], ymls[1] ]}\n",
      "Nextflow operator join with content: ch_graph_qc = ch_graph_qc.join(ODGI_VIZ_COLOR.out.png.map{meta, png -> [ [ id: meta.id.replaceFirst(\".gfaffix.*\", \"\") ], png ]})\n",
      "Nextflow operator join with content: ch_graph_qc = ch_graph_qc.join(ODGI_VIZ_POS.out.png.map{meta, png -> [ [ id: meta.id.replaceFirst(\".gfaffix.*\", \"\") ], png ]})\n",
      "Nextflow operator join with content: ch_graph_qc = ch_graph_qc.join(ODGI_VIZ_DEPTH.out.png.map{meta, png -> [ [ id: meta.id.replaceFirst(\".gfaffix.*\", \"\") ], png ]})\n",
      "Nextflow operator join with content: ch_graph_qc = ch_graph_qc.join(ODGI_VIZ_INV.out.png.map{meta, png -> [ [ id: meta.id.replaceFirst(\".gfaffix.*\", \"\") ], png ]})\n",
      "Nextflow operator join with content: ch_graph_qc = ch_graph_qc.join(ODGI_VIZ_COMPR.out.png.map{meta, png -> [ [ id: meta.id.replaceFirst(\".gfaffix.*\", \"\") ], png ]})\n",
      "Nextflow operator join with content: ch_graph_qc = ch_graph_qc.join(ODGI_VIZ_UNCALLED.out.png.map{meta, png -> [ [ id: meta.id.replaceFirst(\".gfaffix.*\", \"\") ], png ]})\n",
      "Nextflow operator join with content: ch_graph_qc = ch_graph_qc.join(ODGI_DRAW_MULTIQC.out.png.map{meta, png -> [ [ id: meta.id.replaceFirst(\".gfaffix.*\", \"\") ], png ]})\n",
      "Nextflow operator groupTuple with content: ch_graph_qc = ODGI_STATS.out.yaml.map{meta, yaml -> [ [ id: meta.id.replaceFirst(\".seqwish\", \".gfaffix\") ], yaml ]}.groupTuple(by:0, size:2).map{meta, yamls -> [ meta, yamls[0], yamls[1] ]}\n",
      "Nextflow operator join with content: ch_graph_qc = ch_graph_qc.join(ODGI_VIZ_COLOR.out.png.map{meta, png -> [ [ id: meta.id.replaceFirst(\".viz\", \"\") ], png ]})\n",
      "Nextflow operator join with content: ch_graph_qc = ch_graph_qc.join(ODGI_VIZ_POS.out.png.map{meta, png -> [ [ id: meta.id.replaceFirst(\".viz_pos\", \"\") ], png ]})\n",
      "Nextflow operator join with content: ch_graph_qc = ch_graph_qc.join(ODGI_VIZ_DEPTH.out.png.map{meta, png -> [ [ id: meta.id.replaceFirst(\".viz_depth\", \"\") ], png ]})\n",
      "Nextflow operator join with content: ch_graph_qc = ch_graph_qc.join(ODGI_VIZ_INV.out.png.map{meta, png -> [ [ id: meta.id.replaceFirst(\".viz_inv\", \"\") ], png ]})\n",
      "Nextflow operator join with content: ch_graph_qc = ch_graph_qc.join(ODGI_VIZ_COMPR.out.png.map{meta, png -> [ [ id: meta.id.replaceFirst(\".viz_O\", \"\") ], png ]})\n",
      "Nextflow operator join with content: ch_graph_qc = ch_graph_qc.join(ODGI_VIZ_UNCALLED.out.png.map{meta, png -> [ [ id: meta.id.replaceFirst(\".viz_uncalled\", \"\") ], png ]})\n",
      "Nextflow operator join with content: ch_graph_qc = ch_graph_qc.join(ODGI_DRAW_MULTIQC.out.png)\n",
      "Error: Unclosed process block in file: ./git_repos\\pangenome\\subworkflows\\local\\odgi_qc.nf\n",
      "Process PGGB in filepath ./git_repos\\pangenome\\subworkflows\\local\\pggb.nf\n",
      "Nextflow operator map with content: ch_wfmash_map_align = fasta.map{meta, fasta -> [ meta, fasta, []]}\n",
      "Nextflow operator join with content: ch_wfmash_map_align = ch_wfmash_map_align.join(gzi).join(fai)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(WFMASH_MAP_ALIGN.out.versions)\n",
      "Nextflow operator map with content: ch_wfmash_map = fasta.map{meta, fasta -> [ meta, fasta, [] ]}\n",
      "Nextflow operator join with content: ch_wfmash_map = ch_wfmash_map.join(gzi).join(fai)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(WFMASH_MAP.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SPLIT_APPROX_MAPPINGS_IN_CHUNKS.out.versions)\n",
      "Nextflow operator combine with content: ch_wfmash_align = fasta.combine(SPLIT_APPROX_MAPPINGS_IN_CHUNKS.out.pafs.transpose(), by:0).combine(gzi, by:0).combine(fai, by:0)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(WFMASH_ALIGN.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SEQWISH.out.versions)\n",
      "Nextflow operator map with content: ch_wfmash_map_align = fasta.map{meta, fasta -> [ meta, fasta, [] ]}\n",
      "Nextflow operator join with content: ch_wfmash_map_align = ch_wfmash_map_align.join(gzi).join(fai)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(WFMASH_MAP_ALIGN.out.versions)\n",
      "Nextflow operator join with content: ch_seqwish_input = WFMASH_MAP_ALIGN.out.paf.join(fasta)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SEQWISH.out.versions)\n",
      "Nextflow operator map with content: ch_wfmash_map = fasta.map{meta, fasta -> [ meta, fasta, [] ]}\n",
      "Nextflow operator join with content: ch_wfmash_map = ch_wfmash_map.join(gzi).join(fai)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(WFMASH_MAP.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SPLIT_APPROX_MAPPINGS_IN_CHUNKS.out.versions)\n",
      "Nextflow operator combine with content: ch_wfmash_align = fasta.combine(SPLIT_APPROX_MAPPINGS_IN_CHUNKS.out.pafs.transpose(), by:0).combine(gzi, by:0).combine(fai, by:0)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(WFMASH_ALIGN.out.versions)\n",
      "Nextflow operator groupTuple with content: SEQWISH(WFMASH_ALIGN.out.paf.groupTuple(by: 0, size: params.wfmash_chunks).join(fasta))\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SEQWISH.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GFAFFIX.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SMOOTHXG.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GFAFFIX.out.versions)\n",
      "Nextflow operator map with content: ch_gfaffix = GFAFFIX.out.gfa.map{meta, gfa -> [ [ id: gfa.baseName ], gfa ]}\n",
      "Nextflow operator map with content: ch_seqwish = SEQWISH.out.gfa.map{meta, gfa -> [ [ id: gfa.baseName ], gfa ]}\n",
      "Nextflow operator mix with content: ODGI_BUILD(ch_gfaffix.mix(ch_seqwish))\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ODGI_BUILD.out.versions)\n",
      "Nextflow operator map with content: ch_odgi_build = ODGI_BUILD.out.og.map{meta, gfa ->\n",
      "Nextflow operator filter with content: }}.filter{ it != null }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ODGI_UNCHOP.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ODGI_SORT.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ODGI_VIEW.out.versions)\n",
      "Nextflow operator map with content: ch_odgi_build_seqwish = ODGI_BUILD.out.og.map{meta, gfa ->\n",
      "Nextflow operator filter with content: }}.filter{ it != null }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ODGI_QC.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\pangenome\\subworkflows\\local\\pggb.nf\n",
      "Process PANGENOME in filepath ./git_repos\\pangenome\\workflows\\pangenome.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(INPUT_CHECK.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(COMMUNITY.out.versions)\n",
      "Nextflow operator join with content: ch_community_join = COMMUNITY.out.fasta_gz.join(COMMUNITY.out.gzi).join(COMMUNITY.out.fai)\n",
      "Nextflow operator map with content: ch_community_join.map{meta, fasta, gzi, fai -> [ meta, fasta ]},\n",
      "Nextflow operator map with content: ch_community_join.map{meta, fasta, gzi, fai -> [ meta, fai ]},\n",
      "Nextflow operator map with content: ch_community_join.map{meta, fasta, gzi, fai -> [ meta, gzi ]},\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(PGGB.out.versions)\n",
      "Nextflow operator groupTuple with content: ch_squeeze_in = PGGB.out.og.map{meta, og -> [ [ id:meta.id.replaceFirst(\".community.*\", \"\") ], og ]}.groupTuple(by:0)\n",
      "Nextflow operator map with content: ch_odgi_qc_in = ODGI_SQUEEZE.out.graph.map{meta, ogs -> [ [ id:meta.id + \".squeeze\" ], ogs ]}\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ODGI_QC.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ODGI_SQUEEZE.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ODGI_VIEW.out.versions)\n",
      "Nextflow operator map with content: ch_multiqc_in = PGGB.out.qc.map{meta, seqwish, gfaffix, viz, viz_pos, viz_depth, viz_inv, viz_O, viz_uncalled, draw -> [ meta, [ seqwish, gfaffix, viz, viz_pos, viz_depth, viz_inv, viz_O, viz_uncalled, draw ] ]}\n",
      "Nextflow operator toList with content: ch_multiqc_config.toList(),\n",
      "Nextflow operator toList with content: ch_multiqc_custom_config.toList(),\n",
      "Nextflow operator toList with content: ch_multiqc_logo.toList())\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(PGGB.out.versions)\n",
      "Nextflow operator flatten with content: ch_vcf_spec = Channel.from(params.vcf_spec).splitCsv().flatten()\n",
      "Nextflow operator combine with content: VG_DECONSTRUCT(PGGB.out.gfa.combine(ch_vcf_spec))\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(VG_DECONSTRUCT.out.versions)\n",
      "Nextflow operator combine with content: VG_DECONSTRUCT(ODGI_VIEW.out.gfa.combine(ch_vcf_spec))\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(VG_DECONSTRUCT.out.versions)\n",
      "Nextflow operator collectFile with content: ch_versions.unique().collectFile(name: 'collated_versions.yml')\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_workflow_summary.collectFile(name: 'workflow_summary_mqc.yaml'))\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_methods_description.collectFile(name: 'methods_description_mqc.yaml'))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(CUSTOM_DUMPSOFTWAREVERSIONS.out.mqc_yml.collect())\n",
      "Nextflow operator map with content: ch_multiqc_files = ch_multiqc_files.mix(PGGB.out.qc.map{return it[1..8]})\n",
      "Nextflow operator map with content: ch_multiqc_files = ch_multiqc_files.mix(ODGI_QC.out.qc.map{return it[1..8]})\n",
      "Nextflow operator mix with content: ch_multiqc_files = ch_multiqc_files.mix(VG_DECONSTRUCT.out.stats)\n",
      "Nextflow operator collect with content: ch_multiqc_files.collect(),\n",
      "Nextflow operator toList with content: ch_multiqc_config.toList(),\n",
      "Nextflow operator toList with content: ch_multiqc_custom_config.toList(),\n",
      "Nextflow operator toList with content: ch_multiqc_logo.toList()\n",
      "Nextflow operator toList with content: multiqc_report = MULTIQC.out.report.toList()\n",
      "Error: Unclosed process block in file: ./git_repos\\pangenome\\workflows\\pangenome.nf\n",
      "Process NFCORE_PATHOGENSURVEILLANCE in filepath ./git_repos\\pathogensurveillance\\main.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\pathogensurveillance\\main.nf\n",
      "Process ALIGN_READS_TO_REF in filepath ./git_repos\\pathogensurveillance\\subworkflows\\local\\align_reads_to_ref.nf\n",
      "Nextflow operator map with content: .map { [[id: \"${it[2].id}_${it[0].id}\", ref: it[2], sample: it[0]]] + it } // make composite ID for read/ref combos\n",
      "Nextflow operator map with content: ch_reads     = samp_ref_combo.map { [it[0], it[2]] }\n",
      "Nextflow operator map with content: ch_bwa_index = samp_ref_combo.map { [it[0], it[6]] }\n",
      "Nextflow operator map with content: ch_versions = ch_versions.mix(BWA_MEM.out.versions.toSortedList().map{it[0]})\n",
      "Nextflow operator map with content: ch_versions = ch_versions.mix(PICARD_ADDORREPLACEREADGROUPS.out.versions.toSortedList().map{it[0]})\n",
      "Nextflow operator map with content: ch_versions = ch_versions.mix(PICARD_SORTSAM_1.out.versions.toSortedList().map{it[0]})\n",
      "Nextflow operator map with content: ch_reference = samp_ref_combo.map { [it[0], it[4]] } // channel: [ val(ref_samp_meta), file(reference) ]\n",
      "Nextflow operator map with content: ch_ref_index = samp_ref_combo.map { [it[0], it[5]] } // channel: [ val(ref_samp_meta), file(ref_index) ]\n",
      "Nextflow operator join with content: .join(ch_reference)\n",
      "Nextflow operator join with content: .join(ch_ref_index)\n",
      "Nextflow operator map with content: picard_input.map { it[0..1] },\n",
      "Nextflow operator map with content: picard_input.map { it[2] },\n",
      "Nextflow operator map with content: picard_input.map { it[3] }\n",
      "Nextflow operator map with content: ch_versions = ch_versions.mix(PICARD_MARKDUPLICATES.out.versions.toSortedList().map{it[0]})\n",
      "Nextflow operator map with content: ch_versions = ch_versions.mix(SAMTOOLS_INDEX.out.versions.toSortedList().map{it[0]})\n",
      "Nextflow operator map with content: .map { [it[0].sample, it[0].ref, it[1]] }\n",
      "Nextflow operator map with content: .map { [it[0].sample, it[0].ref, it[1]] }\n",
      "Error: Unclosed process block in file: ./git_repos\\pathogensurveillance\\subworkflows\\local\\align_reads_to_ref.nf\n",
      "Process ASSIGN_REFERENCES in filepath ./git_repos\\pathogensurveillance\\subworkflows\\local\\assign_references.nf\n",
      "Nextflow operator map with content: .map { it[0..1] } // [val(meta), [file(fastq)]], possibly duplicated\n",
      "Nextflow operator combine with content: .combine(depth, by:0) // [val(meta), [file(fastq)], depth], possibly duplicated\n",
      "Nextflow operator map with content: .map { it[0..1] + [ Float.parseFloat(it[2]) > 2 ? 2 / Float.parseFloat(it[2]) : 1 ] }\n",
      "Nextflow operator unique with content: .unique() // [val(meta), [file(fastq)]], one per sample\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(KHMER_TRIMLOWABUND.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SOURMASH_SKETCH_READS.out.versions)\n",
      "Nextflow operator filter with content: .filter { it[3] != null }\n",
      "Nextflow operator map with content: .map { it[2..3] } // val(ref_meta), file(ref)\n",
      "Nextflow operator unique with content: .unique()\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SOURMASH_SKETCH_GENOME.out.versions)\n",
      "Nextflow operator map with content: .map { [it[2], it[4]] } // ref_meta, group_meta\n",
      "Nextflow operator combine with content: .combine(SOURMASH_SKETCH_GENOME.out.signatures, by: 0) // ref_meta, group_meta, sig\n",
      "Nextflow operator map with content: .map { it[1..2] }  // [group_meta, sig], possibly duplicated\n",
      "Nextflow operator unique with content: .unique()\n",
      "Nextflow operator groupTuple with content: .groupTuple() // group_meta, [sig]\n",
      "Nextflow operator combine with content: .combine(SOURMASH_SKETCH_READS.out.signatures, by: 0)\n",
      "Nextflow operator map with content: .map { [it[4], it[5]] }\n",
      "Nextflow operator groupTuple with content: .groupTuple() // group_meta, [sig]\n",
      "Nextflow operator combine with content: .combine(signatures.map { [it[0].id, it[1]] }, by: 0)\n",
      "Nextflow operator map with content: .map { [it[1], it[0], it[2]] } // meta, assem, sig\n",
      "Nextflow operator combine with content: .combine(sample_data, by: 0) // meta, assem, sig, fastq, ref_meta, ref, group_meta\n",
      "Nextflow operator map with content: .map { [it[6], it[2]] } // group_meta, sig\n",
      "Nextflow operator groupTuple with content: .groupTuple() // group_meta, [sig]\n",
      "Nextflow operator map with content: .map { [it[0], it[1].unique()] }\n",
      "Nextflow operator join with content: .join(assem_sigs)\n",
      "Nextflow operator join with content: .join(user_sigs, remainder: true)\n",
      "Nextflow operator map with content: .map { [it[0], it[1] + it[2] + (it[3] != null ? it[3] : [])] }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SOURMASH_COMPARE.out.versions)\n",
      "Nextflow operator collectFile with content: .collectFile() { item -> [ \"${item[4].id}.csv\", \"${item[0].id},${item[2].id ?: 'NA'}\\n\" ] }\n",
      "Nextflow operator map with content: .map {[[id: it.getSimpleName()], it]} // TODO this recreates the group_meta, but if other feilds besids \"id\" are added this will not preserve those\n",
      "Nextflow operator join with content: SOURMASH_COMPARE.out.csv.join(samp_ref_pairs)\n",
      "Nextflow operator splitText with content: .splitText( elem: 1 )\n",
      "Nextflow operator map with content: .map { [it[0], it[1].replace('\\n', '')] } // remove newline that splitText adds\n",
      "Nextflow operator splitCsv with content: .splitCsv( elem: 1 )\n",
      "Nextflow operator map with content: .map { [it[0].id] + it[1] } // [val(sample_id), val(group_id), val(reference_id)]\n",
      "Nextflow operator map with content: .map { [it[4].id, it[0].id, it[4], it[0]] }\n",
      "Nextflow operator combine with content: .combine(id_meta_key, by: 0..1)\n",
      "Nextflow operator map with content: .map { [it[4], it[3], it[2]] } // [val(meta), val(group_meta), val(ref_id)]\n",
      "Nextflow operator filter with content: .filter { it[3] != null }\n",
      "Nextflow operator map with content: .map { [it[2].id, it[2], it[3]] }\n",
      "Nextflow operator unique with content: .unique() // [val(ref_id), val(ref_meta), file(reference)]\n",
      "Nextflow operator map with content: .map { [it[0], [id: it[0]], it[1]] }\n",
      "Nextflow operator concat with content: .concat(user_refs) // [val(ref_id), val(ref_meta), file(reference)]\n",
      "Nextflow operator map with content: .map { [it[2]] + it[0..1] } // [val(ref_id), val(meta), val(group_meta)]\n",
      "Nextflow operator combine with content: .combine(all_refs, by: 0)  // [val(ref_id), val(meta), val(group_meta), val(ref_meta), file(reference)]\n",
      "Nextflow operator map with content: .map { it[1..4] } // [val(meta), val(group_meta), val(ref_meta), file(reference)]\n",
      "Nextflow operator map with content: .map { [it[0], it[4], it[1]] } // [val(meta), val(group_meta), [file(fastq)] ]\n",
      "Nextflow operator combine with content: .combine ( assigned_refs_with_seq, by: 0..1 ) // [val(meta), val(group_meta), [file(fastq)], val(ref_meta), file(reference)]\n",
      "Nextflow operator map with content: .map { [it[0], it[2], it[3], it[4], it[1]] } // [val(meta), [file(fastq)], val(ref_meta), file(reference), val(group_meta)]\n",
      "Error: Unclosed process block in file: ./git_repos\\pathogensurveillance\\subworkflows\\local\\assign_references.nf\n",
      "Process CALL_VARIANTS in filepath ./git_repos\\pathogensurveillance\\subworkflows\\local\\call_variants.nf\n",
      "Nextflow operator map with content: .map { [[id: \"${it[5].id}_${it[3].id}\", group: it[5], ref: it[3]], it[0], it[1], it[2], it[4], it[6], it[7]] }\n",
      "Nextflow operator groupTuple with content: .groupTuple()\n",
      "Nextflow operator map with content: .map { [it[0], it[4].sort()[0], it[5].sort()[0], it[6].sort()[0], it[1], it[2], it[3]] } // remove redundant reference genome paths\n",
      "Nextflow operator map with content: ch_ref = ch_ref_grouped.map { it[0..1] }\n",
      "Nextflow operator join with content: ch_ref_grouped = ch_ref_grouped.join(MAKEREGIONFILE.out.regions) // make inputs in same order\n",
      "Nextflow operator map with content: ch_ref_grouped.map { [it[0], it[5], it[6]] },\n",
      "Nextflow operator map with content: ch_ref_grouped.map { [it[0], it[1]] },\n",
      "Nextflow operator map with content: ch_ref_grouped.map { [it[0], it[2]] },\n",
      "Nextflow operator map with content: ch_ref_grouped.map { it[7] },\n",
      "Nextflow operator map with content: ch_versions = ch_versions.mix(GRAPHTYPER_GENOTYPE.out.versions.toSortedList().map{it[0]})\n",
      "Nextflow operator map with content: ch_versions = ch_versions.mix(GRAPHTYPER_VCFCONCATENATE.out.versions.toSortedList().map{it[0]})\n",
      "Nextflow operator map with content: ch_versions = ch_versions.mix(TABIX_TABIX.out.versions.toSortedList().map{it[0]})\n",
      "Nextflow operator map with content: ch_versions = ch_versions.mix(BGZIP_MAKE_GZIP.out.versions.toSortedList().map{it[0]})\n",
      "Nextflow operator join with content: .join(TABIX_TABIX.out.tbi) // [val(ref+group_meta), file(vcf), file(tbi)]\n",
      "Nextflow operator join with content: .join(ch_ref_grouped.map { it[0..3] })\n",
      "Nextflow operator join with content: .join(BGZIP_MAKE_GZIP.out.gzi) // [val(ref+group_meta), file(vcf), file(tbi), file(ref), file(samtools_fai), file(picard_dict), file(gzi)]]\n",
      "Nextflow operator map with content: vf_input.map { it[0..2] },\n",
      "Nextflow operator map with content: vf_input.map { it[3] },\n",
      "Nextflow operator map with content: vf_input.map { it[4] },\n",
      "Nextflow operator map with content: vf_input.map { it[5] },\n",
      "Nextflow operator map with content: vf_input.map { it[6] }\n",
      "Nextflow operator map with content: ch_versions = ch_versions.mix(GATK4_VARIANTFILTRATION.out.versions.toSortedList().map{it[0]})\n",
      "Nextflow operator join with content: VCFLIB_VCFFILTER ( GATK4_VARIANTFILTRATION.out.vcf.join(GATK4_VARIANTFILTRATION.out.tbi) )\n",
      "Nextflow operator map with content: ch_versions = ch_versions.mix(VCFLIB_VCFFILTER.out.versions.toSortedList().map{it[0]})\n",
      "Error: Unclosed process block in file: ./git_repos\\pathogensurveillance\\subworkflows\\local\\call_variants.nf\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Process COARSE_SAMPLE_TAXONOMY in filepath ./git_repos\\pathogensurveillance\\subworkflows\\local\\coarse_sample_taxonomy.nf\n",
      "Nextflow operator map with content: ch_versions = ch_versions.mix(BBMAP_SENDSKETCH.out.versions.toSortedList().map{it[0]})\n",
      "Nextflow operator map with content: .map { [it[0], it[2]] }\n",
      "Nextflow operator map with content: .map { [it[0], it[1]] }\n",
      "Nextflow operator map with content: ch_versions = ch_versions.mix(INITIALCLASSIFICATION.out.versions.toSortedList().map{it[0]})\n",
      "Error: Unclosed process block in file: ./git_repos\\pathogensurveillance\\subworkflows\\local\\coarse_sample_taxonomy.nf\n",
      "Process CORE_GENOME_PHYLOGENY in filepath ./git_repos\\pathogensurveillance\\subworkflows\\local\\core_genome_phylogeny.nf\n",
      "Nextflow operator groupTuple with content: .groupTuple(by: 2)\n",
      "Nextflow operator flatten with content: .map { [it[2], it[1] + it[3].flatten().unique()] } // [ val(group_meta), [ gff ] ]\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(PIRATE.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(REFORMATPIRATERESULTS.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(ALIGNFEATURESEQUENCES.out.versions.first())\n",
      "Nextflow operator join with content: SUBSETCOREGENES ( REFORMATPIRATERESULTS.out.gene_fam.join(RENAMECOREGENEHEADERS.out.feat_seqs) )\n",
      "Nextflow operator transpose with content: MAFFT_SMALL ( SUBSETCOREGENES.out.feat_seq.transpose(), [] )\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(MAFFT_SMALL.out.versions.first())\n",
      "Nextflow operator groupTuple with content: IQTREE2 ( MAFFT_SMALL.out.fas.groupTuple(), [] )\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(IQTREE2.out.versions.first())\n",
      "Error: Unclosed process block in file: ./git_repos\\pathogensurveillance\\subworkflows\\local\\core_genome_phylogeny.nf\n",
      "Process DOWNLOAD_REFERENCES in filepath ./git_repos\\pathogensurveillance\\subworkflows\\local\\download_references.nf\n",
      "Nextflow operator map with content: .map {it[1]}\n",
      "Nextflow operator splitText with content: .splitText()\n",
      "Nextflow operator map with content: .map { it.replace('\\n', '') }\n",
      "Nextflow operator collect with content: .collect()\n",
      "Nextflow operator toSortedList with content: .toSortedList()\n",
      "Nextflow operator flatten with content: .flatten()\n",
      "Nextflow operator unique with content: .unique()\n",
      "Nextflow operator map with content: ch_versions = ch_versions.mix(FIND_ASSEMBLIES.out.versions.toSortedList().map{it[0]})\n",
      "Nextflow operator map with content: .map { it[1] }\n",
      "Nextflow operator collect with content: .collect()\n",
      "Nextflow operator join with content: .join(ch_genera)\n",
      "Nextflow operator join with content: .join(ch_species),\n",
      "Nextflow operator map with content: .map {it[1]}\n",
      "Nextflow operator splitText with content: .splitText()\n",
      "Nextflow operator map with content: .map { it.replace('\\n', '') }\n",
      "Nextflow operator collect with content: .collect()\n",
      "Nextflow operator toSortedList with content: .toSortedList()\n",
      "Nextflow operator flatten with content: .flatten()\n",
      "Nextflow operator unique with content: .unique()\n",
      "Nextflow operator join with content: MAKEGFFWITHFASTA ( DOWNLOAD_ASSEMBLIES.out.sequence.join(DOWNLOAD_ASSEMBLIES.out.gff) )\n",
      "Nextflow operator map with content: .map { [[id: it[0]], it[1]] }\n",
      "Nextflow operator splitText with content: .splitText(elem: 1)\n",
      "Nextflow operator map with content: .map { [it[1].replace('\\n', ''), it[0]] } // [ val(genome_id), val(meta) ]\n",
      "Error: Unclosed process block in file: ./git_repos\\pathogensurveillance\\subworkflows\\local\\download_references.nf\n",
      "Process GENOME_ASSEMBLY in filepath ./git_repos\\pathogensurveillance\\subworkflows\\local\\genome_assembly.nf\n",
      "Nextflow operator filter with content: .filter { it[5] == \"Bacteria\" }\n",
      "Nextflow operator map with content: .map { it[0..1] }\n",
      "Nextflow operator unique with content: .unique()\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(FASTP.out.versions.first())\n",
      "Nextflow operator map with content: FASTP.out.reads.map { [it[0], it[1], [], []] }, // tuple val(meta), path(illumina), path(pacbio), path(nanopore)\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SPADES.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(FILTER_ASSEMBLY.out.versions.first())\n",
      "Nextflow operator combine with content: .combine(FILTER_ASSEMBLY.out.filtered, by: 0)\n",
      "Nextflow operator groupTuple with content: .groupTuple(by: 2)\n",
      "Nextflow operator map with content: .map { [it[2], it[3].sort()[0], it[6]] } // [val(ref_meta), val(ref), [file(assembly] ]\n",
      "Nextflow operator map with content: ch_ref_grouped.map { it[2] }, // consensus (one or more assemblies)\n",
      "Nextflow operator map with content: ch_ref_grouped.map { it[1] }, // fasta (reference, optional)\n",
      "Error: Unclosed process block in file: ./git_repos\\pathogensurveillance\\subworkflows\\local\\genome_assembly.nf\n",
      "Process INPUT_CHECK in filepath ./git_repos\\pathogensurveillance\\subworkflows\\local\\input_check.nf\n",
      "Nextflow operator splitCsv with content: .splitCsv ( header:true, sep:',' )\n",
      "Nextflow operator map with content: .map { create_reads_ref_channel(it) }\n",
      "Nextflow operator transpose with content: .transpose ( by: 4 ) // Duplicate rows for each group when there are multiple groups per sample\n",
      "Nextflow operator map with content: .map { it[0..3] + [[id: it[4]]] }\n",
      "Nextflow operator set with content: .set { sample_data }\n",
      "Error: Unclosed process block in file: ./git_repos\\pathogensurveillance\\subworkflows\\local\\input_check.nf\n",
      "Process MAKE_REFERENCE_INDEX in filepath ./git_repos\\pathogensurveillance\\subworkflows\\local\\make_ref_index.nf\n",
      "Nextflow operator map with content: ch_versions = ch_versions.mix (PICARD_CREATESEQUENCEDICTIONARY.out.versions.toSortedList().map{it[0]})\n",
      "Nextflow operator map with content: ch_versions = ch_versions.mix (SAMTOOLS_FAIDX.out.versions.toSortedList().map{it[0]})\n",
      "Nextflow operator map with content: ch_versions = ch_versions.mix (BWA_INDEX.out.versions.toSortedList().map{it[0]})\n",
      "Error: Unclosed process block in file: ./git_repos\\pathogensurveillance\\subworkflows\\local\\make_ref_index.nf\n",
      "Process VARIANT_CALLING_ANALYSIS in filepath ./git_repos\\pathogensurveillance\\subworkflows\\local\\variant_calling_analysis.nf\n",
      "Nextflow operator map with content: MAKE_REFERENCE_INDEX ( input.map { it[2..3] }.unique() )\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(MAKE_REFERENCE_INDEX.out.versions)\n",
      "Nextflow operator map with content: .map { [it[2], it[0], it[1], it[3], it[4]] } // [val(ref_meta), val(meta), [file(fastq)], file(reference), val(group_meta)]\n",
      "Nextflow operator combine with content: .combine(MAKE_REFERENCE_INDEX.out.samtools_fai, by: 0)\n",
      "Nextflow operator combine with content: .combine(MAKE_REFERENCE_INDEX.out.bwa_index, by: 0)\n",
      "Nextflow operator combine with content: .combine(MAKE_REFERENCE_INDEX.out.picard_dict, by: 0)\n",
      "Nextflow operator map with content: .map { [it[1], it[2], it[0], it[3], it[4], it[5], it[6], it[7]] } // [val(meta), [file(fastq)], val(ref_meta), file(reference), val(group_meta), fai, bwa, picard]\n",
      "Nextflow operator map with content: .map { it[0..3] + it[5..6] }\n",
      "Nextflow operator unique with content: .unique()\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ALIGN_READS_TO_REF.out.versions)\n",
      "Nextflow operator map with content: .map { [it[0], it[2], it[1]] + it[3..5] + [it[7]] } // [val(meta), val(ref_meta), [file(fastq)], file(reference), val(group_meta), fai, picard]\n",
      "Nextflow operator combine with content: .combine(ALIGN_READS_TO_REF.out.bam, by: 0..1)\n",
      "Nextflow operator combine with content: .combine(ALIGN_READS_TO_REF.out.bai, by: 0..1) // [val(meta), val(ref_meta), [file(fastq)], file(reference), val(group_meta), fai, picard, bam, bai]\n",
      "Nextflow operator map with content: .map { [it[0], it[7], it[8], it[1]] + it[3..6] }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(CALL_VARIANTS.out.versions)\n",
      "Nextflow operator map with content: ch_versions = ch_versions.mix(VCFTOTAB.out.versions.toSortedList().map{it[0]})\n",
      "Nextflow operator map with content: ch_versions = ch_versions.mix(VCFTOSNPALN.out.versions.toSortedList().map{it[0]})\n",
      "Nextflow operator map with content: ch_versions = ch_versions.mix(VARIANTREPORT.out.versions.toSortedList().map{it[0]})\n",
      "Nextflow operator map with content: .map { [it[0].group, it[0].ref, it[1]] }        // channel: [ group_meta, ref_meta, tree ]\n",
      "Error: Unclosed process block in file: ./git_repos\\pathogensurveillance\\subworkflows\\local\\variant_calling_analysis.nf\n",
      "Process VARIANT_CALLING_REPORT in filepath ./git_repos\\pathogensurveillance\\subworkflows\\local\\variant_calling_report.nf\n",
      "Nextflow operator map with content: ch_versions = ch_versions.mix(VCFTOTAB.out.versions.toSortedList().map{it[0]})\n",
      "Nextflow operator map with content: ch_versions = ch_versions.mix(VCFTOSNPALN.out.versions.toSortedList().map{it[0]})\n",
      "Nextflow operator map with content: ch_versions = ch_versions.mix(VARIANTREPORT.out.versions.toSortedList().map{it[0]})\n",
      "Error: Unclosed process block in file: ./git_repos\\pathogensurveillance\\subworkflows\\local\\variant_calling_report.nf\n",
      "Process PATHOGENSURVEILLANCE in filepath ./git_repos\\pathogensurveillance\\workflows\\pathogensurveillance.nf\n",
      "Nextflow operator map with content: .map { it[0..1] }\n",
      "Nextflow operator distinct with content: .distinct()\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(INPUT_CHECK.out.versions)\n",
      "Nextflow operator map with content: ch_versions = ch_versions.mix(FASTQC.out.versions.toSortedList().map{it[0]})\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(COARSE_SAMPLE_TAXONOMY.out.versions)\n",
      "Nextflow operator combine with content: .combine(COARSE_SAMPLE_TAXONOMY.out.kingdom, by: 0)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GENOME_ASSEMBLY.out.versions)\n",
      "Nextflow operator combine with content: .combine(DOWNLOAD_REFERENCES.out.gff, by: 0) // [ val(genome_id), val(meta), file(gff) ]\n",
      "Nextflow operator map with content: .map { it[1..2] } // [ val(meta), file(gff) ]\n",
      "Nextflow operator groupTuple with content: .groupTuple() // [ val(meta), [file(gff)] ]\n",
      "Nextflow operator combine with content: .combine(GENOME_ASSEMBLY.out.gff, by: 0) // [val(meta), [file(fastq)], val(ref_meta), file(reference), val(group_meta), file(gff)]\n",
      "Nextflow operator combine with content: .combine(ref_gffs, by: 0) // [val(meta), [file(fastq)], val(ref_meta), file(reference), val(group_meta), file(gff), [file(ref_gff)] ]\n",
      "Nextflow operator map with content: .map { [it[0], it[5], it[4], it[6]] } // [ val(meta), file(gff), val(group_meta), [file(ref_gff)] ]\n",
      "Nextflow operator groupTuple with content: .groupTuple() // [ group_meta, [ref_meta], [tree] ]\n",
      "Nextflow operator join with content: .join(ASSIGN_REFERENCES.out.ani_matrix) // [ group_meta, [ref_meta], [tree], ani_matrix ]\n",
      "Nextflow operator join with content: .join(CORE_GENOME_PHYLOGENY.out.phylogeny, remainder: true) // [ group_meta, [ref_meta], [snp_tree], ani_matrix, core_tree ]\n",
      "Nextflow operator collect with content: ch_versions.unique().collect(sort:true)\n",
      "Error: Unclosed process block in file: ./git_repos\\pathogensurveillance\\workflows\\pathogensurveillance.nf\n",
      "Process NFCORE_PHAGEANNOTATOR in filepath ./git_repos\\phageannotator\\main.nf\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(FASTQC.out.versions.first())\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(PHAGEANNOTATOR.out.versions)\n",
      "Nextflow operator collectFile with content: CUSTOM_DUMPSOFTWAREVERSIONS (ch_versions.unique().collectFile(name: 'collated_versions.yml'))\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_workflow_summary.collectFile(name: 'workflow_summary_mqc.yaml'))\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_methods_description.collectFile(name: 'methods_description_mqc.yaml'))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(CUSTOM_DUMPSOFTWAREVERSIONS.out.mqc_yml.collect())\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(FASTQC.out.zip.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files.collect(),\n",
      "Nextflow operator toList with content: ch_multiqc_config.toList(),\n",
      "Nextflow operator toList with content: ch_multiqc_custom_config.toList(),\n",
      "Nextflow operator toList with content: ch_multiqc_logo.toList()\n",
      "Nextflow operator toList with content: multiqc_report = MULTIQC.out.report.toList()\n",
      "Error: Unclosed process block in file: ./git_repos\\phageannotator\\main.nf\n",
      "Process FASTA_ALL_V_ALL_BLAST in filepath ./git_repos\\phageannotator\\subworkflows\\local\\fasta_all_v_all_blast\\main.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix( GUNZIP.out.versions )\n",
      "Nextflow operator map with content: ch_blast_db = BLAST_MAKEBLASTDB ( fasta.map{ it[1] } ).db\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix( BLAST_MAKEBLASTDB.out.versions )\n",
      "Nextflow operator mix with content: ch_versions = ch_versions = ch_versions.mix( BLAST_MAKEBLASTDB.out.versions )\n",
      "Error: Unclosed process block in file: ./git_repos\\phageannotator\\subworkflows\\local\\fasta_all_v_all_blast\\main.nf\n",
      "Process FASTA_MICRODIVERSITY_INSTRAIN in filepath ./git_repos\\phageannotator\\subworkflows\\local\\fasta_microdiversity_instrain\\main.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GUNZIP_PROTEINS.out.versions)\n",
      "Nextflow operator first with content: ch_stb_file_tsv_nometa = instrain_stb.map { it[1] }.first()\n",
      "Nextflow operator first with content: ch_fasta_nometa = fasta.map { it[1] }.first()\n",
      "Nextflow operator first with content: ch_proteins_fna_nometa = proteins_fna.map { it[1] }.first()\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(INSTRAIN_PROFILE.out.versions)\n",
      "Nextflow operator join with content: ch_bam_profiles_combined = bam.join(ch_instrain_profiles)\n",
      "Nextflow operator groupTuple with content: ch_instrain_compare_input = ch_bam_profiles_combined.map { [ [ id:'all_samples' ], it[1], it[2] ] }.groupTuple( sort: 'deep' )\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(INSTRAIN_COMPARE.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\phageannotator\\subworkflows\\local\\fasta_microdiversity_instrain\\main.nf\n",
      "Process FASTA_PHAGE_HOST_IPHOP in filepath ./git_repos\\phageannotator\\subworkflows\\local\\fasta_phage_host_iphop\\main.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(IPHOP_DOWNLOAD.out.versions)\n",
      "Nextflow operator map with content: ch_iphop_db = UNTAR ( [ [ id:'iphop_minimal_db' ], file(\"https://github.com/nf-core/test-datasets/raw/phageannotator/modules/nfcore/iphop/download/Test_db_rw.tar.gz\", checkIfExists: true) ] ).untar.map { it[1] }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(UNTAR.out.versions)\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(GUNZIP.out.versions.first())\n",
      "Nextflow operator map with content: ch_iphop_partial_input = UNTAR2 ( [ [ id:'iphop_partial_input' ], file(\"https://github.com/nf-core/test-datasets/raw/phageannotator/modules/nfcore/iphop/predict/iPHoP_data.tar.gz\", checkIfExists: true) ] ).untar.map { it[1] }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(UNTAR2.out.versions)\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(IPHOP_PREDICT.out.versions.first())\n",
      "Error: Unclosed process block in file: ./git_repos\\phageannotator\\subworkflows\\local\\fasta_phage_host_iphop\\main.nf\n",
      "Process FASTA_VIRUS_CLASSIFICATION_GENOMAD in filepath ./git_repos\\phageannotator\\subworkflows\\local\\fasta_virus_classification_genomad\\main.nf\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(GENOMAD_DOWNLOAD.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(GENOMAD_ENDTOEND.out.versions.first())\n",
      "Error: Unclosed process block in file: ./git_repos\\phageannotator\\subworkflows\\local\\fasta_virus_classification_genomad\\main.nf\n",
      "Process FASTA_VIRUS_QUALITY_CHECKV in filepath ./git_repos\\phageannotator\\subworkflows\\local\\fasta_virus_quality_checkv\\main.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(CHECKV_DOWNLOADDATABASE.out.versions)\n",
      "Nextflow operator map with content: ch_checkv_db = UNTAR ( [ [ id:'checkv_minimal_db' ], file(\"https://github.com/nf-core/test-datasets/raw/phageannotator/modules/nfcore/checkv/endtoend/checkv_minimal_db.tar\", checkIfExists: true) ] ).untar.map { it[1] }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(UNTAR.out.versions)\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(GUNZIP.out.versions.first())\n",
      "Nextflow operator collect with content: CHECKV_ENDTOEND ( virus_fasta_gz, ch_checkv_db.collect() )\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(CHECKV_ENDTOEND.out.versions.first())\n",
      "Error: Unclosed process block in file: ./git_repos\\phageannotator\\subworkflows\\local\\fasta_virus_quality_checkv\\main.nf\n",
      "Process FASTQ_FASTA_REFERENCE_CONTAINMENT_MASH in filepath ./git_repos\\phageannotator\\subworkflows\\local\\fastq_fasta_reference_containment_mash\\main.nf\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(MASH_SKETCH_ASSEMBLIES.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(MASH_SKETCH_REFERENCES.out.versions.first())\n",
      "Nextflow operator collect with content: ch_combined_sketch_msh = MASH_PASTE ( ch_assembly_sketch_msh, ch_reference_sketch_msh.collect() ).msh\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(MASH_PASTE.out.versions.first())\n",
      "Nextflow operator join with content: ch_mash_screen_input = fastq_gz.join( ch_combined_sketch_msh, by:0 )\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(MASH_SCREEN.out.versions.first())\n",
      "Error: Unclosed process block in file: ./git_repos\\phageannotator\\subworkflows\\local\\fastq_fasta_reference_containment_mash\\main.nf\n",
      "Process FASTQ_VIRUS_ENRICHMENT_VIROMEQC in filepath ./git_repos\\phageannotator\\subworkflows\\local\\fastq_virus_enrichment_viromeqc\\main.nf\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(VIROMEQC_INSTALL.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(VIROMEQC_VIROMEQC.out.versions.first())\n",
      "Error: Unclosed process block in file: ./git_repos\\phageannotator\\subworkflows\\local\\fastq_virus_enrichment_viromeqc\\main.nf\n",
      "Process BAM_SORT_STATS_SAMTOOLS in filepath ./git_repos\\phageannotator\\subworkflows\\nf-core\\bam_sort_stats_samtools\\main.nf\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SAMTOOLS_SORT.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SAMTOOLS_INDEX.out.versions.first())\n",
      "Nextflow operator join with content: .join(SAMTOOLS_INDEX.out.bai, by: [0], remainder: true)\n",
      "Nextflow operator join with content: .join(SAMTOOLS_INDEX.out.csi, by: [0], remainder: true)\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator set with content: .set { ch_bam_bai }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BAM_STATS_SAMTOOLS.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\phageannotator\\subworkflows\\nf-core\\bam_sort_stats_samtools\\main.nf\n",
      "Process BAM_STATS_SAMTOOLS in filepath ./git_repos\\phageannotator\\subworkflows\\nf-core\\bam_stats_samtools\\main.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_STATS.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_FLAGSTAT.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_IDXSTATS.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\phageannotator\\subworkflows\\nf-core\\bam_stats_samtools\\main.nf\n",
      "Process FASTQ_ALIGN_BOWTIE2 in filepath ./git_repos\\phageannotator\\subworkflows\\nf-core\\fastq_align_bowtie2\\main.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BOWTIE2_ALIGN.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BAM_SORT_STATS_SAMTOOLS.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\phageannotator\\subworkflows\\nf-core\\fastq_align_bowtie2\\main.nf\n",
      "Process INITIALISE in filepath ./git_repos\\phageannotator\\subworkflows\\nf-core\\initialise\\main.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\phageannotator\\subworkflows\\nf-core\\initialise\\main.nf\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Process PHAGEANNOTATOR in filepath ./git_repos\\phageannotator\\workflows\\phageannotator\\main.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(FASTQ_VIRUS_ENRICHMENT_VIROMEQC.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SEQKIT_SEQ.out.versions)\n",
      "Nextflow operator first with content: ch_containment_results_tsv = FASTQ_FASTA_REFERENCE_CONTAINMENT_MASH ( fastq_gz, ch_filtered_input_fasta_gz, ch_reference_virus_fasta_gz.first(), ch_reference_virus_sketch_msh ).mash_screen_results\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(FASTQ_FASTA_REFERENCE_CONTAINMENT_MASH.out.versions.first())\n",
      "Nextflow operator join with content: ch_append_screen_hits_input = ch_containment_results_tsv.join( ch_filtered_input_fasta_gz, by:0 )\n",
      "Nextflow operator first with content: ch_assembly_w_references_fasta_gz = APPENDSCREENHITS ( ch_append_screen_hits_input, ch_reference_virus_fasta_gz.first() ).assembly_w_screen_hits\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(APPENDSCREENHITS.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(FASTA_VIRUS_CLASSIFICATION_GENOMAD.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(FASTA_VIRUS_CLASSIFICATION_GENOMAD.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(FASTA_VIRUS_QUALITY_CHECKV.out.versions.first())\n",
      "Nextflow operator join with content: ch_quality_filter_viruses_input1 = FASTA_VIRUS_QUALITY_CHECKV.out.viruses_fna_gz.join(FASTA_VIRUS_QUALITY_CHECKV.out.proviruses_fna_gz)\n",
      "Nextflow operator join with content: ch_quality_filter_viruses_input2 = ch_quality_filter_viruses_input1.join(FASTA_VIRUS_QUALITY_CHECKV.out.quality_summary_tsv)\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(QUALITYFILTERVIRUSES.out.versions.first())\n",
      "Nextflow operator map with content: .map { [ [ id:'all_samples' ], it[1] ] }\n",
      "Nextflow operator groupTuple with content: .groupTuple( sort: 'deep' )\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix( FASTA_ALL_V_ALL_BLAST.out.versions )\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix( ANICLUSTER_ANICALC.out.versions )\n",
      "Nextflow operator join with content: ch_aniclust_input = ch_filtered_viruses_combined_fna_gz.join( ch_ani_tsv )\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix( ANICLUSTER_ANICLUST.out.versions )\n",
      "Nextflow operator join with content: ch_extractreps_input = ch_filtered_viruses_combined_fna_gz.join( ch_clusters_tsv )\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix( ANICLUSTER_EXTRACTREPS.out.versions )\n",
      "Nextflow operator first with content: ch_anicluster_reps_bt2 = BOWTIE2_BUILD ( ch_anicluster_reps_fasta_gz ).index.first()\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix( BOWTIE2_BUILD.out.versions )\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix( FASTQ_ALIGN_BOWTIE2.out.versions )\n",
      "Nextflow operator groupTuple with content: ch_combined_bams = ch_cluster_rep_alignment_bam.map { [ [ id:'all_samples' ], it[1] ] }.groupTuple( sort: 'deep' )\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix( COVERM_CONTIG.out.versions )\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix( GENOMAD_TAXONOMY.out.versions )\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix( FASTA_PHAGE_HOST_IPHOP.out.versions )\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix( GUNZIP.out.versions )\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix( BACPHLIP.out.versions )\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix( PRODIGAL_PRODIGALGV.out.versions )\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(INSTRAIN_STB.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions = ch_versions.mix(FASTA_MICRODIVERSITY_INSTRAIN.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\phageannotator\\workflows\\phageannotator\\main.nf\n",
      "Process NFCORE_PHYLOPLACE in filepath ./git_repos\\phyloplace\\main.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\phyloplace\\main.nf\n",
      "Process FASTA_NEWICK_EPANG_GAPPA in filepath ./git_repos\\phyloplace\\subworkflows\\nf-core\\fasta_newick_epang_gappa\\main.nf\n",
      "Nextflow operator filter with content: ch_hmmer_data = ch_pp_data.filter { it.data.alignmethod == 'hmmer' }\n",
      "Nextflow operator filter with content: ch_mafft_data = ch_pp_data.filter { it.data.alignmethod == 'mafft' }\n",
      "Nextflow operator filter with content: .filter { ! it.data.hmmfile }\n",
      "Nextflow operator map with content: .map { [ it.meta, it.data.refseqfile ] },\n",
      "Nextflow operator map with content: .mix(HMMER_HMMBUILD.out.hmm.map { [ it[0], it[1] ] })\n",
      "Nextflow operator mix with content: .mix(\n",
      "Nextflow operator filter with content: .filter { it.data.hmmfile }\n",
      "Nextflow operator map with content: .map { [ it.meta, it.data.hmmfile ] }\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(HMMER_HMMBUILD.out.versions.first())\n",
      "Nextflow operator filter with content: .filter { ! it.data.hmmfile }\n",
      "Nextflow operator map with content: .map { [ it.meta, it.data.refseqfile ] }\n",
      "Nextflow operator map with content: .mix(HMMER_UNALIGNREF.out.seqreformated.map { [ it[0], it[1] ] })\n",
      "Nextflow operator mix with content: .mix(\n",
      "Nextflow operator filter with content: .filter { it.data.hmmfile }\n",
      "Nextflow operator map with content: .map { [ it.meta, it.data.refseqfile ] }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(HMMER_UNALIGNREF.out.versions)\n",
      "Nextflow operator mix with content: .mix(ch_hmmer_unaligned)\n",
      "Nextflow operator groupTuple with content: .groupTuple(size: 2, sort: { a, b -> a =~ /\\.hmm/ ? 1 : -1 })\n",
      "Nextflow operator map with content: ch_hmmer_alignref.map { [ it[0], it[1][0] ] },\n",
      "Nextflow operator map with content: ch_hmmer_alignref.map { it[1][1] }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(HMMER_HMMALIGNREF.out.versions)\n",
      "Nextflow operator map with content: .mix(ch_hmmer_data.map { [ it.meta, it.data.queryseqfile ] })\n",
      "Nextflow operator mix with content: .mix(ch_hmm)\n",
      "Nextflow operator groupTuple with content: .groupTuple(size: 2, sort: { a, b -> a =~ /\\.hmm/ ? 1 : -1 })\n",
      "Nextflow operator map with content: ch_hmmer_alignquery.map { [ it[0], it[1][0] ] },\n",
      "Nextflow operator map with content: ch_hmmer_alignquery.map { it[1][1] }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(HMMER_HMMALIGNQUERY.out.versions)\n",
      "Nextflow operator map with content: HMMER_MASKREF ( HMMER_HMMALIGNREF.out.sthlm.map { [ it[0], it[1], [], [], [], [], [], [] ] }, [] )\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(HMMER_MASKREF.out.versions)\n",
      "Nextflow operator map with content: HMMER_MASKQUERY ( HMMER_HMMALIGNQUERY.out.sthlm.map { [ it[0], it[1], [], [], [], [], [], [] ] }, [] )\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(HMMER_MASKQUERY.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(HMMER_AFAFORMATREF.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(HMMER_AFAFORMATQUERY.out.versions)\n",
      "Nextflow operator map with content: ch_mafft_data.map { [ it.meta, it.data.refseqfile ] },\n",
      "Nextflow operator map with content: ch_mafft_data.map { [ it.data.queryseqfile ] }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(MAFFT.out.versions)\n",
      "Nextflow operator map with content: ch_mafft_data.map { [ it.meta, it.data.refseqfile ] }\n",
      "Nextflow operator join with content: .join(MAFFT.out.fas)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(EPANG_SPLIT.out.versions)\n",
      "Nextflow operator map with content: ch_epang_query = ch_pp_data.map { [ it.meta, it.data.model, it.data.refphylogeny ] }\n",
      "Nextflow operator join with content: .join ( HMMER_AFAFORMATQUERY.out.seqreformated )\n",
      "Nextflow operator join with content: .join ( HMMER_AFAFORMATREF.out.seqreformated )\n",
      "Nextflow operator mix with content: .mix(\n",
      "Nextflow operator map with content: ch_pp_data.map { [ it.meta, it.data.model, it.data.refphylogeny ] }\n",
      "Nextflow operator join with content: .join(EPANG_SPLIT.out.query.map { [ it[0], it[1] ] } )\n",
      "Nextflow operator join with content: .join(EPANG_SPLIT.out.reference.map { [ it[0], it[1] ] } )\n",
      "Nextflow operator map with content: .map { [ [ id:it[0].id, model:it[1] ], it[3], it[4], it[2] ] }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(EPANG_PLACE.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GAPPA_GRAFT.out.versions)\n",
      "Nextflow operator map with content: .map { [ [ id:it[0].id ], it[1] ] }\n",
      "Nextflow operator join with content: .join( ch_pp_data.map { [ it.meta, it.data.taxonomy ] } )\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GAPPA_ASSIGN.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GAPPA_HEATTREE.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\phyloplace\\subworkflows\\nf-core\\fasta_newick_epang_gappa\\main.nf\n",
      "Process PHYLOPLACE in filepath ./git_repos\\phyloplace\\workflows\\phyloplace.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(FASTA_NEWICK_EPANG_GAPPA.out.versions)\n",
      "Nextflow operator collectFile with content: ch_versions.unique().collectFile(name: 'collated_versions.yml')\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_workflow_summary.collectFile(name: 'workflow_summary_mqc.yaml'))\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_methods_description.collectFile(name: 'methods_description_mqc.yaml'))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(CUSTOM_DUMPSOFTWAREVERSIONS.out.mqc_yml.collect())\n",
      "Nextflow operator collect with content: ch_multiqc_files.collect(),\n",
      "Nextflow operator collect with content: ch_multiqc_config.collect().ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_multiqc_custom_config.collect().ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_multiqc_logo.collect().ifEmpty([])\n",
      "Nextflow operator toList with content: multiqc_report = MULTIQC.out.report.toList()\n",
      "Nextflow operator mix with content: ch_versions    = ch_versions.mix(MULTIQC.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\phyloplace\\workflows\\phyloplace.nf\n",
      "Process NFCORE_PIXELATOR in filepath ./git_repos\\pixelator\\main.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\pixelator\\main.nf\n",
      "Process GENERATE_REPORTS in filepath ./git_repos\\pixelator\\subworkflows\\local\\generate_reports.nf\n",
      "Nextflow operator map with content: .map { meta, data -> [ meta.id, meta] }\n",
      "Nextflow operator groupTuple with content: .groupTuple()\n",
      "Nextflow operator map with content: .map { id, data ->\n",
      "Nextflow operator map with content: .map { meta, data -> [ meta.id, data] }\n",
      "Nextflow operator map with content: ch_amplicon_col         = amplicon_data.map { meta, data -> [ meta.id, data] }\n",
      "Nextflow operator map with content: ch_preqc_col            = preqc_data.map { meta, data -> [ meta.id, data] }\n",
      "Nextflow operator map with content: ch_adapterqc_col        = adapterqc_data.map { meta, data -> [ meta.id, data] }\n",
      "Nextflow operator map with content: ch_demux_col            = demux_data.map { meta, data -> [ meta.id, data] }\n",
      "Nextflow operator map with content: ch_collapse_col         = collapse_data.map { meta, data -> [ meta.id, data] }\n",
      "Nextflow operator map with content: ch_graph_col            = graph_data.map { meta, data -> [meta.id, data] }\n",
      "Nextflow operator map with content: ch_annotate_col         = annotate_data.map { meta, data -> [meta.id, data] }\n",
      "Nextflow operator map with content: ch_analysis_col         = analysis_data.map { meta, data -> [meta.id, data] }\n",
      "Nextflow operator concat with content: .concat ( ch_panel_col )\n",
      "Nextflow operator concat with content: .concat ( ch_amplicon_col )\n",
      "Nextflow operator concat with content: .concat ( ch_preqc_col )\n",
      "Nextflow operator concat with content: .concat ( ch_adapterqc_col )\n",
      "Nextflow operator concat with content: .concat ( ch_demux_col )\n",
      "Nextflow operator concat with content: .concat ( ch_collapse_col )\n",
      "Nextflow operator concat with content: .concat ( ch_graph_col )\n",
      "Nextflow operator concat with content: .concat ( ch_annotate_col )\n",
      "Nextflow operator concat with content: .concat ( ch_analysis_col )\n",
      "Nextflow operator groupTuple with content: .groupTuple (size: 10)\n",
      "Nextflow operator map with content: ch_panel_files_grouped  = ch_report_data.map { id, data -> [ data[0], data[1], data[1] ? null : data[0].panel ] }\n",
      "Nextflow operator flatten with content: ch_amplicon_grouped     = ch_report_data.map { id, data -> data[2] ? data[2].flatten() : [] }\n",
      "Nextflow operator flatten with content: ch_preqc_grouped        = ch_report_data.map { id, data -> data[3] ? data[3].flatten() : [] }\n",
      "Nextflow operator flatten with content: ch_adapterqc_grouped    = ch_report_data.map { id, data -> data[4] ? data[4].flatten() : [] }\n",
      "Nextflow operator flatten with content: ch_demux_grouped        = ch_report_data.map { id, data -> data[5] ? data[5].flatten() : [] }\n",
      "Nextflow operator flatten with content: ch_collapse_grouped     = ch_report_data.map { id, data -> data[6] ? data[6].flatten() : [] }\n",
      "Nextflow operator flatten with content: ch_graph_grouped        = ch_report_data.map { id, data -> data[7] ? data[7].flatten() : [] }\n",
      "Nextflow operator flatten with content: ch_annotate_grouped     = ch_report_data.map { id, data -> data[8] ? data[8].flatten() : [] }\n",
      "Nextflow operator flatten with content: ch_analysis_grouped     = ch_report_data.map { id, data -> data[9] ? data[9].flatten() : [] }\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(PIXELATOR_REPORT.out.versions.first())\n",
      "Error: Unclosed process block in file: ./git_repos\\pixelator\\subworkflows\\local\\generate_reports.nf\n",
      "Process INPUT_CHECK in filepath ./git_repos\\pixelator\\subworkflows\\local\\input_check.nf\n",
      "Nextflow operator map with content: .map { check_channels(inputBaseDir, *it) }\n",
      "Nextflow operator splitText with content: .splitText()\n",
      "Nextflow operator map with content: .map( text -> text.trim())\n",
      "Nextflow operator reduce with content: .reduce( new HashSet() ) { prev, curr -> prev << curr }\n",
      "Nextflow operator splitText with content: .splitText()\n",
      "Nextflow operator map with content: .map( text -> text.trim())\n",
      "Nextflow operator reduce with content: .reduce( new HashSet() ) { prev, curr -> prev << curr }\n",
      "Nextflow operator map with content: .map { it -> it[0] }\n",
      "Nextflow operator combine with content: .combine(ch_panel_options)\n",
      "Nextflow operator combine with content: .combine(ch_design_options)\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator join with content: .join(ch_input)\n",
      "Nextflow operator map with content: .map { it -> [ it[0] ] + it[2..-1] }\n",
      "Nextflow operator map with content: reads = ch_checked_input.map { it -> [it[0]] + it[2..-1] }\n",
      "Nextflow operator map with content: panels = ch_checked_input.map { it -> [it[0], it[1]] }\n",
      "Error: Unclosed process block in file: ./git_repos\\pixelator\\subworkflows\\local\\input_check.nf\n",
      "Process PIXELATOR in filepath ./git_repos\\pixelator\\workflows\\pixelator.nf\n",
      "Nextflow operator groupTuple with content: .groupTuple()\n",
      "Nextflow operator branch with content: .branch {\n",
      "Nextflow operator flatten with content: return [ meta, fastq.flatten() ]\n",
      "Nextflow operator flatten with content: return [ meta, fastq.flatten() ]\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(PIXELATOR_COLLECT_METADATA.out.versions)\n",
      "Nextflow operator mix with content: .mix(ch_fastq_split.single)\n",
      "Nextflow operator map with content: .map { meta, data -> [ meta.id, data] }\n",
      "Nextflow operator groupTuple with content: .groupTuple()\n",
      "Nextflow operator map with content: .map { id, data ->\n",
      "Nextflow operator unique with content: def unique_panels = data.unique()\n",
      "Nextflow operator map with content: .map { meta, _ -> [meta.id, meta] }\n",
      "Nextflow operator join with content: .join(ch_checked_panel_files, failOnMismatch:true, failOnDuplicate:true)\n",
      "Nextflow operator map with content: .map { id, meta, panel_files -> [meta, panel_files] }\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(CAT_FASTQ.out.versions.first().ifEmpty(null))\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(RENAME_READS.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(PIXELATOR_AMPLICON.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(PIXELATOR_QC.out.versions.first())\n",
      "Nextflow operator join with content: .join(ch_cat_panel_files, failOnMismatch:true, failOnDuplicate:true)\n",
      "Nextflow operator map with content: .map { meta, fq, panel_file -> [meta, fq, panel_file, panel_file ? null : meta.panel ] }\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(PIXELATOR_DEMUX.out.versions.first())\n",
      "Nextflow operator join with content: .join(ch_cat_panel_files, failOnMismatch:true, failOnDuplicate:true)\n",
      "Nextflow operator map with content: .map { meta, demuxed, panel_file -> [meta, demuxed, panel_file, panel_file ? null : meta.panel ] }\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix( PIXELATOR_COLLAPSE.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(PIXELATOR_GRAPH.out.versions.first())\n",
      "Nextflow operator join with content: .join(ch_cat_panel_files, failOnMismatch:true, failOnDuplicate:true)\n",
      "Nextflow operator map with content: .map { meta, clustered, panel_file -> [meta, clustered, panel_file, panel_file ? null : meta.panel ] }\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix( PIXELATOR_ANNOTATE.out.versions.first() )\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(PIXELATOR_ANALYSIS.out.versions.first())\n",
      "Nextflow operator concat with content: .concat(PIXELATOR_AMPLICON.out.metadata)\n",
      "Nextflow operator groupTuple with content: .groupTuple(size: 2)\n",
      "Nextflow operator concat with content: .concat(PIXELATOR_QC.out.preqc_metadata)\n",
      "Nextflow operator groupTuple with content: .groupTuple(size: 2)\n",
      "Nextflow operator concat with content: .concat(PIXELATOR_QC.out.adapterqc_metadata)\n",
      "Nextflow operator groupTuple with content: .groupTuple(size: 2)\n",
      "Nextflow operator concat with content: .concat(PIXELATOR_DEMUX.out.metadata)\n",
      "Nextflow operator groupTuple with content: .groupTuple(size: 2)\n",
      "Nextflow operator concat with content: .concat(PIXELATOR_COLLAPSE.out.metadata)\n",
      "Nextflow operator groupTuple with content: .groupTuple(size: 2)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GENERATE_REPORTS.out.versions)\n",
      "Nextflow operator collectFile with content: ch_versions.unique().collectFile(name: 'collated_versions.yml')\n",
      "Error: Unclosed process block in file: ./git_repos\\pixelator\\workflows\\pixelator.nf\n",
      "Process NFCORE_PROTEINFOLD in filepath ./git_repos\\proteinfold\\main.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\proteinfold\\main.nf\n",
      "Process ARIA2_UNCOMPRESS in filepath ./git_repos\\proteinfold\\subworkflows\\local\\aria2_uncompress.nf\n",
      "Nextflow operator flatten with content: ch_db = UNTAR ( ARIA2.out.downloaded_file.flatten().map{ [ [:], it ] } ).untar.map{ it[1] }\n",
      "Nextflow operator flatten with content: ch_db = GUNZIP ( ARIA2.out.downloaded_file.flatten().map{ [ [:], it ] } ).gunzip.map { it[1] }\n",
      "Error: Unclosed process block in file: ./git_repos\\proteinfold\\subworkflows\\local\\aria2_uncompress.nf\n",
      "Process INPUT_CHECK in filepath ./git_repos\\proteinfold\\subworkflows\\local\\input_check.nf\n",
      "Nextflow operator splitCsv with content: .splitCsv ( header:true, sep:',' )\n",
      "Nextflow operator map with content: .map { create_fasta_channel(it) }\n",
      "Nextflow operator set with content: .set { fastas }\n",
      "Error: Unclosed process block in file: ./git_repos\\proteinfold\\subworkflows\\local\\input_check.nf\n",
      "Process PREPARE_ALPHAFOLD2_DBS in filepath ./git_repos\\proteinfold\\subworkflows\\local\\prepare_alphafold2_dbs.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ARIA2_BFD.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ARIA2_SMALL_BFD.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ARIA2_ALPHAFOLD2_PARAMS.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ARIA2_MGNIFY.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ARIA2_PDB70.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(DOWNLOAD_PDBMMCIF.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ARIA2_UNICLUST30.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ARIA2_UNIREF90.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ARIA2.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ARIA2_UNIPROT_SPROT.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ARIA2_UNIPROT_TREMBL.out.versions)\n",
      "Nextflow operator mix with content: ch_version =  ch_versions.mix(COMBINE_UNIPROT.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\proteinfold\\subworkflows\\local\\prepare_alphafold2_dbs.nf\n",
      "Process PREPARE_COLABFOLD_DBS in filepath ./git_repos\\proteinfold\\subworkflows\\local\\prepare_colabfold_dbs.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ARIA2_COLABFOLD_PARAMS.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ARIA2_COLABFOLD_DB.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(MMSEQS_TSV2EXPROFILEDB_COLABFOLDDB.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(MMSEQS_CREATEINDEX_COLABFOLDDB.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ARIA2_UNIREF30.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(MMSEQS_TSV2EXPROFILEDB_UNIPROT30.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(MMSEQS_CREATEINDEX_UNIPROT30.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\proteinfold\\subworkflows\\local\\prepare_colabfold_dbs.nf\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Process ALPHAFOLD2 in filepath ./git_repos\\proteinfold\\workflows\\alphafold2.nf\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator splitFasta with content: [ meta, fasta.splitFasta(file:true) ]\n",
      "Nextflow operator transpose with content: .transpose()\n",
      "Nextflow operator set with content: .set { ch_fasta }\n",
      "Nextflow operator set with content: .set { ch_fasta }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(INPUT_CHECK.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(PREPARE_ALPHAFOLD2_DBS.out.versions)\n",
      "Nextflow operator ifEmpty with content: PREPARE_ALPHAFOLD2_DBS.out.bfd.ifEmpty([]),\n",
      "Nextflow operator ifEmpty with content: PREPARE_ALPHAFOLD2_DBS.out.small_bfd.ifEmpty([]),\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(RUN_ALPHAFOLD2.out.versions)\n",
      "Nextflow operator collect with content: ch_multiqc_rep = RUN_ALPHAFOLD2.out.multiqc.collect()\n",
      "Nextflow operator ifEmpty with content: PREPARE_ALPHAFOLD2_DBS.out.bfd.ifEmpty([]),\n",
      "Nextflow operator ifEmpty with content: PREPARE_ALPHAFOLD2_DBS.out.small_bfd.ifEmpty([]),\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(RUN_ALPHAFOLD2_MSA.out.versions)\n",
      "Nextflow operator ifEmpty with content: PREPARE_ALPHAFOLD2_DBS.out.bfd.ifEmpty([]),\n",
      "Nextflow operator ifEmpty with content: PREPARE_ALPHAFOLD2_DBS.out.small_bfd.ifEmpty([]),\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(RUN_ALPHAFOLD2_PRED.out.versions)\n",
      "Nextflow operator collect with content: ch_multiqc_rep = RUN_ALPHAFOLD2_PRED.out.multiqc.collect()\n",
      "Nextflow operator collectFile with content: ch_versions.unique().collectFile(name: 'collated_versions.yml')\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_workflow_summary.collectFile(name: 'workflow_summary_mqc.yaml'))\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_methods_description.collectFile(name: 'methods_description_mqc.yaml'))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(CUSTOM_DUMPSOFTWAREVERSIONS.out.mqc_yml.collect())\n",
      "Nextflow operator mix with content: ch_multiqc_files = ch_multiqc_files.mix(ch_multiqc_rep)\n",
      "Nextflow operator collect with content: ch_multiqc_files.collect(),\n",
      "Nextflow operator toList with content: ch_multiqc_config.toList(),\n",
      "Nextflow operator toList with content: ch_multiqc_custom_config.toList(),\n",
      "Nextflow operator toList with content: ch_multiqc_logo.toList()\n",
      "Nextflow operator toList with content: multiqc_report = MULTIQC.out.report.toList()\n",
      "Error: Unclosed process block in file: ./git_repos\\proteinfold\\workflows\\alphafold2.nf\n",
      "Process COLABFOLD in filepath ./git_repos\\proteinfold\\workflows\\colabfold.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(INPUT_CHECK.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(PREPARE_COLABFOLD_DBS.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(MULTIFASTA_TO_CSV.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(COLABFOLD_BATCH.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(COLABFOLD_BATCH.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(MULTIFASTA_TO_CSV.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(MMSEQS_COLABFOLDSEARCH.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(MMSEQS_COLABFOLDSEARCH.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(COLABFOLD_BATCH.out.versions)\n",
      "Nextflow operator collectFile with content: ch_versions.unique().collectFile(name: 'collated_versions.yml')\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_workflow_summary.collectFile(name: 'workflow_summary_mqc.yaml'))\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_methods_description.collectFile(name: 'methods_description_mqc.yaml'))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(CUSTOM_DUMPSOFTWAREVERSIONS.out.mqc_yml.collect())\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(COLABFOLD_BATCH.out.multiqc.collect())\n",
      "Nextflow operator collect with content: ch_multiqc_files.collect(),\n",
      "Nextflow operator toList with content: ch_multiqc_config.toList(),\n",
      "Nextflow operator toList with content: ch_multiqc_custom_config.toList(),\n",
      "Nextflow operator toList with content: ch_multiqc_logo.toList()\n",
      "Nextflow operator toList with content: multiqc_report = MULTIQC.out.report.toList()\n",
      "Error: Unclosed process block in file: ./git_repos\\proteinfold\\workflows\\colabfold.nf\n",
      "Process NFCORE_QUANTMS in filepath ./git_repos\\quantms\\main.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\quantms\\main.nf\n",
      "Process CREATE_INPUT_CHANNEL in filepath ./git_repos\\quantms\\subworkflows\\local\\create_input_channel.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SDRFPARSING.out.version)\n",
      "Nextflow operator splitCsv with content: ch_config.splitCsv(header: true, sep: '\\t')\n",
      "Nextflow operator map with content: .map { create_meta_channel(it, is_sdrf, enzymes, files, wrapper) }\n",
      "Nextflow operator branch with content: .branch {\n",
      "Nextflow operator set with content: .set{result}\n",
      "Nextflow operator take with content: meta.mzml_id                        = file(filestr).name.take(file(filestr).name.lastIndexOf('.'))\n",
      "Nextflow operator take with content: filestr = (params.local_input_type ? filestr.take(filestr.lastIndexOf('.'))\n",
      "Error: Unclosed process block in file: ./git_repos\\quantms\\subworkflows\\local\\create_input_channel.nf\n",
      "Process DATABASESEARCHENGINES in filepath ./git_repos\\quantms\\subworkflows\\local\\databasesearchengines.nf\n",
      "Nextflow operator combine with content: SEARCHENGINEMSGF(ch_mzmls_search.combine(ch_searchengine_in_db))\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SEARCHENGINEMSGF.out.version)\n",
      "Nextflow operator mix with content: ch_id_msgf = ch_id_msgf.mix(SEARCHENGINEMSGF.out.id_files_msgf)\n",
      "Nextflow operator combine with content: SEARCHENGINECOMET(ch_mzmls_search.combine(ch_searchengine_in_db))\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SEARCHENGINECOMET.out.version)\n",
      "Nextflow operator mix with content: ch_id_comet = ch_id_comet.mix(SEARCHENGINECOMET.out.id_files_comet)\n",
      "Nextflow operator map with content: ch_meta_mzml_db = ch_mzmls_search.map{ metapart, mzml ->\n",
      "Nextflow operator groupTuple with content: ch_meta_mzml_db_chunked = ch_meta_mzml_db.groupTuple(by: [0,1])\n",
      "Nextflow operator combine with content: SEARCHENGINESAGE(ch_meta_mzml_db_chunked.combine(ch_searchengine_in_db))\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SEARCHENGINESAGE.out.version)\n",
      "Nextflow operator mix with content: ch_id_sage = ch_id_sage.mix(SEARCHENGINESAGE.out.id_files_sage.transpose())\n",
      "Nextflow operator mix with content: ch_id_files_idx = ch_id_msgf.mix(ch_id_comet).mix(ch_id_sage)\n",
      "Error: Unclosed process block in file: ./git_repos\\quantms\\subworkflows\\local\\databasesearchengines.nf\n",
      "Process FEATUREMAPPER in filepath ./git_repos\\quantms\\subworkflows\\local\\featuremapper.nf\n",
      "Nextflow operator mix with content: ch_version = ch_version.mix(ISOBARICANALYZER.out.version)\n",
      "Nextflow operator combine with content: IDMAPPER(ch_id_files.combine(ISOBARICANALYZER.out.id_files_consensusXML, by: 0))\n",
      "Nextflow operator mix with content: ch_version = ch_version.mix(IDMAPPER.out.version)\n",
      "Error: Unclosed process block in file: ./git_repos\\quantms\\subworkflows\\local\\featuremapper.nf\n",
      "Process FILE_PREPARATION in filepath ./git_repos\\quantms\\subworkflows\\local\\file_preparation.nf\n",
      "Nextflow operator branch with content: .branch {\n",
      "Nextflow operator set with content: }.set { ch_branched_input }\n",
      "Nextflow operator mix with content: compressed_files = ch_branched_input.dottar.mix(ch_branched_input.dotzip, ch_branched_input.gz)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(DECOMPRESS.out.version)\n",
      "Nextflow operator mix with content: ch_rawfiles = ch_branched_input.uncompressed.mix(DECOMPRESS.out.decompressed_files)\n",
      "Nextflow operator branch with content: .branch {\n",
      "Nextflow operator set with content: }.set { ch_branched_input }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(MZMLINDEXING.out.version)\n",
      "Nextflow operator mix with content: ch_results  = ch_results.mix(MZMLINDEXING.out.mzmls_indexed)\n",
      "Nextflow operator mix with content: ch_results = ch_results.mix(ch_branched_input.mzML)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(THERMORAWFILEPARSER.out.version)\n",
      "Nextflow operator mix with content: ch_results  = ch_results.mix(THERMORAWFILEPARSER.out.mzmls_converted)\n",
      "Nextflow operator map with content: ch_results.map{ it -> [it[0], it[1]] }.set{ indexed_mzml_bundle }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(TDF2MZML.out.version)\n",
      "Nextflow operator mix with content: ch_results = indexed_mzml_bundle.mix(TDF2MZML.out.mzmls_converted)\n",
      "Nextflow operator mix with content: ch_results = indexed_mzml_bundle.mix(ch_branched_input.dotd)\n",
      "Nextflow operator collect with content: ch_statistics = ch_statistics.mix(MZMLSTATISTICS.out.ms_statistics.collect())\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(MZMLSTATISTICS.out.version)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(OPENMSPEAKPICKER.out.version)\n",
      "Error: Unclosed process block in file: ./git_repos\\quantms\\subworkflows\\local\\file_preparation.nf\n",
      "Process ID in filepath ./git_repos\\quantms\\subworkflows\\local\\id.nf\n",
      "Nextflow operator ifEmpty with content: ch_software_versions = ch_software_versions.mix(DATABASESEARCHENGINES.out.versions.ifEmpty(null))\n",
      "Nextflow operator ifEmpty with content: ch_software_versions = ch_software_versions.mix(PSMRESCORING.out.versions.ifEmpty(null))\n",
      "Nextflow operator groupTuple with content: CONSENSUSID(PSMRESCORING.out.results.groupTuple(size: params.search_engines.split(\",\").size()))\n",
      "Nextflow operator ifEmpty with content: ch_software_versions = ch_software_versions.mix(CONSENSUSID.out.version.ifEmpty(null))\n",
      "Nextflow operator ifEmpty with content: ch_software_versions = ch_software_versions.mix(PSMFDRCONTROL.out.version.ifEmpty(null))\n",
      "Nextflow operator ifEmpty with content: ch_software_versions = ch_software_versions.mix(PHOSPHOSCORING.out.version.ifEmpty(null))\n",
      "Error: Unclosed process block in file: ./git_repos\\quantms\\subworkflows\\local\\id.nf\n",
      "Process INPUT_CHECK in filepath ./git_repos\\quantms\\subworkflows\\local\\input_check.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\quantms\\subworkflows\\local\\input_check.nf\n",
      "Process PHOSPHOSCORING in filepath ./git_repos\\quantms\\subworkflows\\local\\phosphoscoring.nf\n",
      "Nextflow operator combine with content: IDSCORESWITCHERFORLUCIPHOR(ch_id_files.combine(Channel.value(\"\\\"Posterior Error Probability_score\\\"\")))\n",
      "Nextflow operator mix with content: ch_version = ch_version.mix(IDSCORESWITCHERFORLUCIPHOR.out.version)\n",
      "Nextflow operator join with content: LUCIPHORADAPTER(ch_mzml_files.join(IDSCORESWITCHERFORLUCIPHOR.out.id_score_switcher))\n",
      "Nextflow operator mix with content: ch_version = ch_version.mix(LUCIPHORADAPTER.out.version)\n",
      "Error: Unclosed process block in file: ./git_repos\\quantms\\subworkflows\\local\\phosphoscoring.nf\n",
      "Process PROTEININFERENCE in filepath ./git_repos\\quantms\\subworkflows\\local\\proteininference.nf\n",
      "Nextflow operator mix with content: ch_version = ch_version.mix(EPIFANY.out.version)\n",
      "Nextflow operator mix with content: ch_version = ch_version.mix(PROTEININFERENCER.out.version)\n",
      "Nextflow operator mix with content: ch_version = ch_version.mix(IDFILTER.out.version)\n",
      "Nextflow operator multiMap with content: .multiMap{ it ->\n",
      "Nextflow operator set with content: .set{ ch_epi_results }\n",
      "Error: Unclosed process block in file: ./git_repos\\quantms\\subworkflows\\local\\proteininference.nf\n",
      "Process PROTEINQUANT in filepath ./git_repos\\quantms\\subworkflows\\local\\proteinquant.nf\n",
      "Nextflow operator mix with content: ch_version = ch_version.mix(IDCONFLICTRESOLVER.out.version)\n",
      "Nextflow operator mix with content: ch_version = ch_version.mix(PROTEINQUANTIFIER.out.version)\n",
      "Nextflow operator mix with content: ch_version = ch_version.mix(MSSTATSCONVERTER.out.version)\n",
      "Error: Unclosed process block in file: ./git_repos\\quantms\\subworkflows\\local\\proteinquant.nf\n",
      "Process PSMFDRCONTROL in filepath ./git_repos\\quantms\\subworkflows\\local\\psmfdrcontrol.nf\n",
      "Nextflow operator mix with content: ch_version = ch_version.mix(IDSCORESWITCHER.out.version)\n",
      "Nextflow operator mix with content: ch_version = ch_version.mix(FDRCONSENSUSID.out.version)\n",
      "Nextflow operator mix with content: ch_version = ch_version.mix(IDFILTER.out.version)\n",
      "Error: Unclosed process block in file: ./git_repos\\quantms\\subworkflows\\local\\psmfdrcontrol.nf\n",
      "Process PSMRESCORING in filepath ./git_repos\\quantms\\subworkflows\\local\\psmrescoring.nf\n",
      "Nextflow operator branch with content: ch_id_files.branch{ meta, filename ->\n",
      "Nextflow operator set with content: }.set{ch_id_files_branched}\n",
      "Nextflow operator mix with content: ch_id_files_feats = ch_id_files_branched.sage.mix(EXTRACTPSMFEATURES.out.id_files_feat)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(EXTRACTPSMFEATURES.out.version)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(PERCOLATOR.out.version)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(FDRIDPEP.out.version)\n",
      "Nextflow operator mix with content: IDPEP(ch_fdridpep.mix(ch_id_files))\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(IDPEP.out.version)\n",
      "Error: Unclosed process block in file: ./git_repos\\quantms\\subworkflows\\local\\psmrescoring.nf\n",
      "Process DIA in filepath ./git_repos\\quantms\\workflows\\dia.nf\n",
      "Nextflow operator set with content: Channel.fromPath(params.database).set { ch_searchdb }\n",
      "Nextflow operator multiMap with content: ch_file_preparation_results.multiMap {\n",
      "Nextflow operator set with content: .set { ch_result }\n",
      "Nextflow operator unique with content: meta = ch_result.meta.unique { it[0] }\n",
      "Nextflow operator ifEmpty with content: ch_software_versions = ch_software_versions.mix(DIANNCFG.out.version.ifEmpty(null))\n",
      "Nextflow operator combine with content: DIANN_PRELIMINARY_ANALYSIS(ch_file_preparation_results.combine(speclib))\n",
      "Nextflow operator ifEmpty with content: ch_software_versions = ch_software_versions.mix(DIANN_PRELIMINARY_ANALYSIS.out.version.ifEmpty(null))\n",
      "Nextflow operator collect with content: ch_result.ms_file.collect(),\n",
      "Nextflow operator collect with content: DIANN_PRELIMINARY_ANALYSIS.out.diann_quant.collect(),\n",
      "Nextflow operator ifEmpty with content: ch_software_versions = ch_software_versions.mix(ASSEMBLE_EMPIRICAL_LIBRARY.out.version.ifEmpty(null))\n",
      "Nextflow operator combine with content: .combine(ch_searchdb)\n",
      "Nextflow operator combine with content: .combine(ASSEMBLE_EMPIRICAL_LIBRARY.out.log)\n",
      "Nextflow operator combine with content: .combine(ASSEMBLE_EMPIRICAL_LIBRARY.out.empirical_library)\n",
      "Nextflow operator ifEmpty with content: ch_software_versions = ch_software_versions.mix(INDIVIDUAL_FINAL_ANALYSIS.out.version.ifEmpty(null))\n",
      "Nextflow operator map with content: .ms_file.map { msfile -> file(msfile).getName() }\n",
      "Nextflow operator collect with content: .collect()\n",
      "Nextflow operator set with content: .set { ms_file_names }\n",
      "Nextflow operator collect with content: INDIVIDUAL_FINAL_ANALYSIS.out.diann_quant.collect(), ch_searchdb)\n",
      "Nextflow operator ifEmpty with content: ch_software_versions = ch_software_versions.mix(DIANNSUMMARY.out.version.ifEmpty(null))\n",
      "Nextflow operator ifEmpty with content: ch_software_versions = ch_software_versions.mix(DIANNCONVERT.out.version.ifEmpty(null))\n",
      "Nextflow operator ifEmpty with content: ch_software_versions = ch_software_versions.mix(MSSTATS.out.version.ifEmpty(null))\n",
      "Error: Unclosed process block in file: ./git_repos\\quantms\\workflows\\dia.nf\n",
      "Process LFQ in filepath ./git_repos\\quantms\\workflows\\lfq.nf\n",
      "Nextflow operator ifEmpty with content: ch_software_versions = ch_software_versions.mix(ID.out.version.ifEmpty(null))\n",
      "Nextflow operator join with content: ch_file_preparation_results.join(ID.out.id_results)\n",
      "Nextflow operator multiMap with content: .multiMap { it ->\n",
      "Nextflow operator set with content: .set{ ch_plfq }\n",
      "Nextflow operator collect with content: PROTEOMICSLFQ(ch_plfq.mzmls.collect(),\n",
      "Nextflow operator collect with content: ch_plfq.ids.collect(),\n",
      "Nextflow operator ifEmpty with content: ch_software_versions = ch_software_versions.mix(PROTEOMICSLFQ.out.version.ifEmpty(null))\n",
      "Nextflow operator ifEmpty with content: ch_software_versions = ch_software_versions.mix(MSSTATS.out.version.ifEmpty(null))\n",
      "Nextflow operator map with content: .map { it -> it[1] }\n",
      "Nextflow operator set with content: .set { ch_pmultiqc_ids }\n",
      "Nextflow operator map with content: .map { it -> it[1] }\n",
      "Nextflow operator set with content: .set { ch_pmultiqc_consensus }\n",
      "Error: Unclosed process block in file: ./git_repos\\quantms\\workflows\\lfq.nf\n",
      "Process QUANTMS in filepath ./git_repos\\quantms\\workflows\\quantms.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(INPUT_CHECK.out.versions)\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(CREATE_INPUT_CHANNEL.out.version.ifEmpty(null))\n",
      "Nextflow operator mix with content: CREATE_INPUT_CHANNEL.out.ch_meta_config_iso.mix(CREATE_INPUT_CHANNEL.out.ch_meta_config_lfq).mix(CREATE_INPUT_CHANNEL.out.ch_meta_config_dia)\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(FILE_PREPARATION.out.version.ifEmpty(null))\n",
      "Nextflow operator branch with content: .branch {\n",
      "Nextflow operator set with content: .set{ch_fileprep_result}\n",
      "Nextflow operator first with content: CREATE_INPUT_CHANNEL.out.ch_meta_config_iso.mix(CREATE_INPUT_CHANNEL.out.ch_meta_config_lfq).first()         // Only run if iso or lfq have at least one file\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(DECOYDATABASE.out.version.ifEmpty(null))\n",
      "Nextflow operator mix with content: ch_ids_pmultiqc = ch_ids_pmultiqc.mix(TMT.out.ch_pmultiqc_ids)\n",
      "Nextflow operator mix with content: ch_consensus_pmultiqc = ch_consensus_pmultiqc.mix(TMT.out.ch_pmultiqc_consensus)\n",
      "Nextflow operator mix with content: ch_pipeline_results = ch_pipeline_results.mix(TMT.out.final_result)\n",
      "Nextflow operator mix with content: ch_msstats_in = ch_msstats_in.mix(TMT.out.msstats_in)\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(TMT.out.versions.ifEmpty(null))\n",
      "Nextflow operator mix with content: ch_ids_pmultiqc = ch_ids_pmultiqc.mix(LFQ.out.ch_pmultiqc_ids)\n",
      "Nextflow operator mix with content: ch_consensus_pmultiqc = ch_consensus_pmultiqc.mix(LFQ.out.ch_pmultiqc_consensus)\n",
      "Nextflow operator mix with content: ch_pipeline_results = ch_pipeline_results.mix(LFQ.out.final_result)\n",
      "Nextflow operator mix with content: ch_msstats_in = ch_msstats_in.mix(LFQ.out.msstats_in)\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(LFQ.out.versions.ifEmpty(null))\n",
      "Nextflow operator mix with content: ch_pipeline_results = ch_pipeline_results.mix(DIA.out.diann_report)\n",
      "Nextflow operator mix with content: ch_msstats_in = ch_msstats_in.mix(DIA.out.msstats_in)\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(DIA.out.versions.ifEmpty(null))\n",
      "Nextflow operator collectFile with content: ch_versions.unique().collectFile(name: 'collated_versions.yml')\n",
      "Nextflow operator mix with content: ch_multiqc_files = ch_multiqc_files.mix(Channel.from(ch_multiqc_config))\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_workflow_summary.collectFile(name: 'workflow_summary_mqc.yaml'))\n",
      "Nextflow operator mix with content: ch_multiqc_files = ch_multiqc_files.mix(FILE_PREPARATION.out.statistics)\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_methods_description.collectFile(name: 'methods_description_mqc.yaml'))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(CUSTOM_DUMPSOFTWAREVERSIONS.out.mqc_yml.collect())\n",
      "Nextflow operator collect with content: .combine(ch_pipeline_results.ifEmpty([]).combine(ch_multiqc_files.collect())\n",
      "Nextflow operator collect with content: .combine(ch_ids_pmultiqc.collect().ifEmpty([]))\n",
      "Nextflow operator collect with content: .combine(ch_consensus_pmultiqc.collect().ifEmpty([])))\n",
      "Nextflow operator combine with content: .combine(ch_msstats_in.ifEmpty([])),\n",
      "Nextflow operator toList with content: multiqc_report      = SUMMARYPIPELINE.out.ch_pmultiqc_report.toList()\n",
      "Error: Unclosed process block in file: ./git_repos\\quantms\\workflows\\quantms.nf\n",
      "Process TMT in filepath ./git_repos\\quantms\\workflows\\tmt.nf\n",
      "Nextflow operator ifEmpty with content: ch_software_versions = ch_software_versions.mix(ID.out.version.ifEmpty(null))\n",
      "Nextflow operator ifEmpty with content: ch_software_versions = ch_software_versions.mix(FEATUREMAPPER.out.version.ifEmpty(null))\n",
      "Nextflow operator collect with content: FILEMERGE(FEATUREMAPPER.out.id_map.collect())\n",
      "Nextflow operator ifEmpty with content: ch_software_versions = ch_software_versions.mix(FILEMERGE.out.version.ifEmpty(null))\n",
      "Nextflow operator ifEmpty with content: ch_software_versions = ch_software_versions.mix(PROTEININFERENCE.out.version.ifEmpty(null))\n",
      "Nextflow operator ifEmpty with content: ch_software_versions = ch_software_versions.mix(PROTEINQUANT.out.version.ifEmpty(null))\n",
      "Nextflow operator ifEmpty with content: ch_software_versions = ch_software_versions.mix(MSSTATSTMT.out.version.ifEmpty(null))\n",
      "Nextflow operator map with content: .map { it -> it[1] }\n",
      "Nextflow operator set with content: .set { ch_pmultiqc_ids }\n",
      "Nextflow operator map with content: .map { it -> it[1] }\n",
      "Nextflow operator set with content: .set { ch_pmultiqc_consensus }\n",
      "Error: Unclosed process block in file: ./git_repos\\quantms\\workflows\\tmt.nf\n",
      "Process NFCORE_RADSEQ in filepath ./git_repos\\radseq\\main.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\radseq\\main.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\radseq\\modules\\local\\rbmerge2fasta.nf\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Process BAM_INTERVALS_BEDTOOLS in filepath ./git_repos\\radseq\\subworkflows\\local\\bam_intervals_bedtools.nf\n",
      "Nextflow operator randomSample with content: ch_bam = params.subset_intervals_channel ? bam.randomSample(params.subset_intervals_channel, 234) : bam\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix (BEDTOOLS_BAMTOBED.out.versions)\n",
      "Nextflow operator map with content: ch_bed_to_merge = ch_bed.map {\n",
      "Nextflow operator groupTuple with content: .groupTuple()\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BEDOPS_MERGE_BED.out.versions)\n",
      "Nextflow operator first with content: ch_sorted_mbed = BEDTOOLS_SORT (ch_mbed, faidx.first()).sorted\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BEDTOOLS_SORT.out.versions)\n",
      "Nextflow operator combine with content: cov = BEDTOOLS_COVERAGE (ch_bed.combine(ch_sorted_mbed.map{it[1]}).map{meta,bed,mbed -> [meta,mbed,bed]}, faidx.first()).bed\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix (BEDTOOLS_COVERAGE.out.versions)\n",
      "Nextflow operator map with content: ch_cov_to_merge = cov.map {\n",
      "Nextflow operator groupTuple with content: .groupTuple()\n",
      "Nextflow operator first with content: ch_mcov = BEDTOOLS_MERGE_COV (ch_cov_to_merge, faidx.first()).cov\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix (BEDTOOLS_MERGE_COV.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix (BEDTOOLS_MAKEWINDOWS.out.versions)\n",
      "Nextflow operator join with content: ch_intersect = BEDTOOLS_INTERSECT (ch_mcov.join(ch_split_high_coverage), 'bed').intersect\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix (BEDTOOLS_INTERSECT.out.versions)\n",
      "Nextflow operator join with content: ch_createintervals = ch_mcov.join(ch_intersect).join(BEDTOOLS_MAKEWINDOWS.out.low_cov)\n",
      "Nextflow operator transpose with content: ch_intervals = CREATE_INTERVALS (ch_createintervals, read_lengths).intervals.transpose()\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix (CREATE_INTERVALS.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\radseq\\subworkflows\\local\\bam_intervals_bedtools.nf\n",
      "Process INPUT_CHECK in filepath ./git_repos\\radseq\\subworkflows\\local\\input_check.nf\n",
      "Nextflow operator splitCsv with content: .splitCsv ( header:true, sep:',' )\n",
      "Nextflow operator map with content: .map { create_fastq_channels(it) }\n",
      "Nextflow operator set with content: .set { reads }\n",
      "Error: Unclosed process block in file: ./git_repos\\radseq\\subworkflows\\local\\input_check.nf\n",
      "Process BAM_DEDUP_STATS_SAMTOOLS_UMITOOLS in filepath ./git_repos\\radseq\\subworkflows\\nf-core\\bam_dedup_stats_samtools_umitools\\main.nf\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(UMITOOLS_DEDUP.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SAMTOOLS_INDEX.out.versions.first())\n",
      "Nextflow operator join with content: .join(SAMTOOLS_INDEX.out.bai, by: [0], remainder: true)\n",
      "Nextflow operator join with content: .join(SAMTOOLS_INDEX.out.csi, by: [0], remainder: true)\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator set with content: .set { ch_bam_bai }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BAM_STATS_SAMTOOLS.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\radseq\\subworkflows\\nf-core\\bam_dedup_stats_samtools_umitools\\main.nf\n",
      "Process BAM_STATS_SAMTOOLS in filepath ./git_repos\\radseq\\subworkflows\\nf-core\\bam_stats_samtools\\main.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_STATS.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_FLAGSTAT.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_IDXSTATS.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\radseq\\subworkflows\\nf-core\\bam_stats_samtools\\main.nf\n",
      "Process RADSEQ in filepath ./git_repos\\radseq\\workflows\\radseq.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(INPUT_CHECK.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(PROCESS_RAD.out.versions)\n",
      "Nextflow operator map with content: .map{genome -> tuple (genome.simpleName, genome)}\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(DENOVO.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_FAIDX.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ALIGN.out.versions)\n",
      "Nextflow operator map with content: ch_bam_bai.map{meta, bam, bai -> [meta, bam]},\n",
      "Nextflow operator map with content: ch_faidx.map{it[1]},\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BAM_INTERVALS_BEDTOOLS.out.versions)\n",
      "Nextflow operator combine with content: .combine(ch_intervals.map{it[1]})\n",
      "Nextflow operator map with content: .map { meta, bam, bai, bed ->\n",
      "Nextflow operator map with content: ch_reference.map{it[1]},\n",
      "Nextflow operator map with content: ch_faidx.map{it[1]}\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BAM_VARIANT_CALLING_FREEBAYES.out.versions)\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(FASTQC.out.versions.first())\n",
      "Nextflow operator collectFile with content: ch_versions.unique().collectFile(name: 'collated_versions.yml')\n",
      "Nextflow operator mix with content: ch_multiqc_files = ch_multiqc_files.mix(Channel.from(ch_multiqc_config))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(ch_multiqc_custom_config.collect().ifEmpty([]))\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_workflow_summary.collectFile(name: 'workflow_summary_mqc.yaml'))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(CUSTOM_DUMPSOFTWAREVERSIONS.out.mqc_yml.collect())\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(FASTQC.out.zip.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(PROCESS_RAD.out.fastp_json.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(ALIGN.out.stats.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(ALIGN.out.flagstat.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(ALIGN.out.idxstats.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(BAM_VARIANT_CALLING_FREEBAYES.out.stats.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files.collect(),\n",
      "Nextflow operator toList with content: multiqc_report = MULTIQC.out.report.toList()\n",
      "Nextflow operator mix with content: ch_versions    = ch_versions.mix(MULTIQC.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\radseq\\workflows\\radseq.nf\n",
      "Process NFCORE_RANGELAND in filepath ./git_repos\\rangeland\\main.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\rangeland\\main.nf\n",
      "Process HIGHER_LEVEL in filepath ./git_repos\\rangeland\\subworkflows\\local\\higher_level.nf\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(HIGHER_LEVEL_CONFIG.out.versions.first().ifEmpty(null))\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(FORCE_HIGHER_LEVEL.out.versions.first().ifEmpty(null))\n",
      "Nextflow operator flatten with content: trend_files = FORCE_HIGHER_LEVEL.out.trend_files.flatten().map{ x -> [ x.simpleName.substring(12), x ] }\n",
      "Nextflow operator groupTuple with content: trend_files_mosaic = trend_files.groupTuple()\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(FORCE_MOSAIC.out.versions.first().ifEmpty(null))\n",
      "Nextflow operator filter with content: FORCE_PYRAMID( trend_files.filter { it[1].name.endsWith('.tif')  }.map { [ it[1].simpleName.substring(0,11), it[1] ] } .groupTuple() )\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(FORCE_PYRAMID.out.versions.first().ifEmpty(null))\n",
      "Error: Unclosed process block in file: ./git_repos\\rangeland\\subworkflows\\local\\higher_level.nf\n",
      "Process INPUT_CHECK in filepath ./git_repos\\rangeland\\subworkflows\\local\\input_check.nf\n",
      "Nextflow operator splitCsv with content: .splitCsv ( header:true, sep:',' )\n",
      "Nextflow operator map with content: .map { create_fastq_channel(it) }\n",
      "Nextflow operator set with content: .set { reads }\n",
      "Error: Unclosed process block in file: ./git_repos\\rangeland\\subworkflows\\local\\input_check.nf\n",
      "Process PREPROCESSING in filepath ./git_repos\\rangeland\\subworkflows\\local\\preprocessing.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(FORCE_GENERATE_TILE_ALLOW_LIST.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(FORCE_GENERATE_ANALYSIS_MASK.out.versions)\n",
      "Nextflow operator flatten with content: masks = FORCE_GENERATE_ANALYSIS_MASK.out.masks.flatten().map{ x -> [ extractDirectory(x), x ] }\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(PREPROCESS_CONFIG.out.versions.first().ifEmpty(null))\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(FORCE_PREPROCESS.out.versions.first().ifEmpty(null))\n",
      "Nextflow operator flatten with content: boa_tiles = FORCE_PREPROCESS.out.boa_tiles.flatten().map{ [ \"${extractDirectory(it)}_${it.simpleName}\", it ] }.groupTuple()\n",
      "Nextflow operator flatten with content: qai_tiles = FORCE_PREPROCESS.out.qai_tiles.flatten().map{ [ \"${extractDirectory(it)}_${it.simpleName}\", it ] }.groupTuple()\n",
      "Nextflow operator filter with content: boa_tiles_to_merge = boa_tiles.filter{ x -> x[1].size() > 1 }\n",
      "Nextflow operator map with content: .map{ [ it[0].substring( 0, 11 ), it[1] ] }\n",
      "Nextflow operator toSortedList with content: .toSortedList{ a,b -> a[1][0].simpleName <=> b[1][0].simpleName }\n",
      "Nextflow operator flatMap with content: .flatMap{it}\n",
      "Nextflow operator flatten with content: .groupTuple( remainder : true, size : params.group_size ).map{ [ it[0], it[1] .flatten() ] }\n",
      "Nextflow operator filter with content: qai_tiles_to_merge = qai_tiles.filter{ x -> x[1].size() > 1 }\n",
      "Nextflow operator map with content: .map{ [ it[0].substring( 0, 11 ), it[1] ] }\n",
      "Nextflow operator toSortedList with content: .toSortedList{ a,b -> a[1][0].simpleName <=> b[1][0].simpleName }\n",
      "Nextflow operator flatMap with content: .flatMap{it}\n",
      "Nextflow operator flatten with content: .groupTuple( remainder : true, size : params.group_size ).map{ [ it[0], it[1] .flatten() ] }\n",
      "Nextflow operator filter with content: boa_tiles_done = boa_tiles.filter{ x -> x[1].size() == 1 }.map{ x -> [ x[0] .substring( 0, 11 ), x[1][0] ] }\n",
      "Nextflow operator filter with content: qai_tiles_done = qai_tiles.filter{ x -> x[1].size() == 1 }.map{ x -> [ x[0] .substring( 0, 11 ), x[1][0] ] }\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(MERGE_BOA.out.versions.first().ifEmpty(null))\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(MERGE_QAI.out.versions.first().ifEmpty(null))\n",
      "Nextflow operator concat with content: .concat( boa_tiles_done ).groupTuple()\n",
      "Nextflow operator flatten with content: .map { [it[0], it[1].flatten() ] }\n",
      "Nextflow operator concat with content: .concat( qai_tiles_done ).groupTuple()\n",
      "Nextflow operator flatten with content: .map { [it[0], it[1].flatten() ] }\n",
      "Nextflow operator join with content: tiles_and_masks = boa_tiles.join( qai_tiles ).join( masks )\n",
      "Error: Unclosed process block in file: ./git_repos\\rangeland\\subworkflows\\local\\preprocessing.nf\n",
      "Process RANGELAND in filepath ./git_repos\\rangeland\\workflows\\rangeland.nf\n",
      "Nextflow operator map with content: base_path = UNTAR_INPUT.out.untar.map(it -> it[1])\n",
      "Nextflow operator flatten with content: data = base_path.map(it -> file(\"$it/*/*\", type: 'dir')).flatten()\n",
      "Nextflow operator filter with content: data = data.flatten().filter{ inRegion(it) }\n",
      "Nextflow operator mix with content: tar_versions = tar_versions.mix(UNTAR_INPUT.out.versions)\n",
      "Nextflow operator flatten with content: data = Channel.fromPath( \"${params.input}/*/*\", type: 'dir') .flatten()\n",
      "Nextflow operator filter with content: data = data.flatten().filter{ inRegion(it) }\n",
      "Nextflow operator map with content: dem = UNTAR_DEM.out.untar.map(it -> file(it[1]))\n",
      "Nextflow operator mix with content: tar_versions = tar_versions.mix(UNTAR_DEM.out.versions)\n",
      "Nextflow operator map with content: wvdb = UNTAR_WVDB.out.untar.map(it -> file(it[1]))\n",
      "Nextflow operator mix with content: tar_versions = tar_versions.mix(UNTAR_WVDB.out.versions)\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(tar_versions.first().ifEmpty(null))\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(PREPROCESSING.out.versions)\n",
      "Nextflow operator filter with content: preprocessed_data = PREPROCESSING.out.tiles_and_masks.filter { params.only_tile ? it[0] == params.only_tile : true }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(HIGHER_LEVEL.out.versions)\n",
      "Nextflow operator buffer with content: grouped_trend_data = HIGHER_LEVEL.out.trend_files.map{ it[1] }.flatten().buffer( size: Integer.MAX_VALUE, remainder: true )\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(CHECK_RESULTS.out.versions)\n",
      "Nextflow operator collectFile with content: ch_versions.unique().collectFile(name: 'collated_versions.yml')\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_workflow_summary.collectFile(name: 'workflow_summary_mqc.yaml'))\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_methods_description.collectFile(name: 'methods_description_mqc.yaml'))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(CUSTOM_DUMPSOFTWAREVERSIONS.out.mqc_yml.collect())\n",
      "Nextflow operator collect with content: ch_multiqc_files.collect(),\n",
      "Nextflow operator toList with content: ch_multiqc_config.toList(),\n",
      "Nextflow operator toList with content: ch_multiqc_custom_config.toList(),\n",
      "Nextflow operator toList with content: ch_multiqc_logo.toList()\n",
      "Nextflow operator toList with content: multiqc_report = MULTIQC.out.report.toList()\n",
      "Error: Unclosed process block in file: ./git_repos\\rangeland\\workflows\\rangeland.nf\n",
      "Process NFCORE_RAREDISEASE in filepath ./git_repos\\raredisease\\main.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\raredisease\\main.nf\n",
      "Process ALIGN in filepath ./git_repos\\raredisease\\subworkflows\\local\\align.nf\n",
      "Nextflow operator mix with content: ch_marked_bam = Channel.empty().mix(ALIGN_BWAMEM2.out.marked_bam, ALIGN_SENTIEON.out.marked_bam)\n",
      "Nextflow operator mix with content: ch_marked_bai = Channel.empty().mix(ALIGN_BWAMEM2.out.marked_bai, ALIGN_SENTIEON.out.marked_bai)\n",
      "Nextflow operator join with content: ch_bam_bai    = ch_marked_bam.join(ch_marked_bai, failOnMismatch:true, failOnDuplicate:true)\n",
      "Nextflow operator mix with content: ch_versions   = Channel.empty().mix(ALIGN_BWAMEM2.out.versions, ALIGN_SENTIEON.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\raredisease\\subworkflows\\local\\align.nf\n",
      "Process ANALYSE_MT in filepath ./git_repos\\raredisease\\subworkflows\\local\\analyse_MT.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(CONVERT_MT_BAM_TO_FASTQ.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ALIGN_AND_CALL_MT.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ALIGN_AND_CALL_MT_SHIFT.out.versions)\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(PICARD_LIFTOVERVCF.out.versions.first())\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(MERGE_ANNOTATE_MT.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\raredisease\\subworkflows\\local\\analyse_MT.nf\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Process ANNOTATE_CSQ_PLI in filepath ./git_repos\\raredisease\\subworkflows\\local\\annotate_consequence_pli.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ADD_MOST_SEVERE_CSQ.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ADD_MOST_SEVERE_PLI.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(TABIX_BGZIPTABIX.out.versions)\n",
      "Nextflow operator collect with content: vcf_ann  = TABIX_BGZIPTABIX.out.gz_tbi.map { meta, vcf, tbi -> return [ meta, vcf ] }.collect() // channel: [ val(meta), path(vcf) ]\n",
      "Error: Unclosed process block in file: ./git_repos\\raredisease\\subworkflows\\local\\annotate_consequence_pli.nf\n",
      "Process ANNOTATE_SNVS in filepath ./git_repos\\raredisease\\subworkflows\\local\\annotate_snvs.nf\n",
      "Nextflow operator map with content: .map { meta, vcf, tbi -> return [meta, vcf, tbi, []]}\n",
      "Nextflow operator set with content: .set { ch_vcf_in }\n",
      "Nextflow operator map with content: .map {meta, vcf ->\n",
      "Nextflow operator flatten with content: .flatten()\n",
      "Nextflow operator buffer with content: .buffer (size: 2)\n",
      "Nextflow operator set with content: .set { ch_upd_in }\n",
      "Nextflow operator join with content: .join(TABIX_BCFTOOLS_VIEW.out.tbi, failOnMismatch:true, failOnDuplicate:true)\n",
      "Nextflow operator collect with content: .collect()\n",
      "Nextflow operator combine with content: .combine(ch_split_intervals)\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator set with content: .set { ch_vcf_scatter_in }\n",
      "Nextflow operator combine with content: .combine(ANNOTATE_CADD.out.vcf.ifEmpty(\"null\")) // If CADD is not run then this channel will be empty, so assign a default value to allow filtering with branch operator\n",
      "Nextflow operator branch with content: .branch { it  ->                                // If CADD is run, then \"it\" will be [[meta],selvar.vcf,[meta],cadd.vcf], else [[meta],selvar.vcf,null]\n",
      "Nextflow operator set with content: .set { ch_for_mix }\n",
      "Nextflow operator mix with content: ch_vep_in = ch_for_mix.selvar.mix(ch_for_mix.cadd)\n",
      "Nextflow operator map with content: .map { meta, vcf -> [meta - meta.subMap('scatterid'), vcf] }\n",
      "Nextflow operator set with content: .set { ch_vep_out }\n",
      "Nextflow operator join with content: .join(TABIX_VEP.out.tbi, failOnMismatch:true)\n",
      "Nextflow operator groupTuple with content: .groupTuple()\n",
      "Nextflow operator map with content: .map { meta, vcfs, tbis ->\n",
      "Nextflow operator set with content: .set { ch_concat_in }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BCFTOOLS_ROH.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(RHOCALL_ANNOTATE.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ZIP_TABIX_ROHCALL.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(VCFANNO.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(UPD_SITES.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(UPD_REGIONS.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(CHROMOGRAPH_SITES.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(CHROMOGRAPH_REGIONS.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ZIP_TABIX_VCFANNO.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BCFTOOLS_VIEW.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(TABIX_BCFTOOLS_VIEW.out.versions)\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(GATK4_SELECTVARIANTS.out.versions.first())\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ANNOTATE_CADD.out.versions)\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(ENSEMBLVEP_SNV.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(TABIX_VEP.out.versions.first())\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BCFTOOLS_CONCAT.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(TABIX_BCFTOOLS_CONCAT.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\raredisease\\subworkflows\\local\\annotate_snvs.nf\n",
      "Process ANNOTATE_STRUCTURAL_VARIANTS in filepath ./git_repos\\raredisease\\subworkflows\\local\\annotate_structural_variants.nf\n",
      "Nextflow operator splitCsv with content: .splitCsv ( header:true )\n",
      "Nextflow operator multiMap with content: .multiMap { row ->\n",
      "Nextflow operator set with content: .set { ch_svdb_dbs }\n",
      "Nextflow operator toList with content: ch_svdb_dbs.in_occs.toList(),\n",
      "Nextflow operator toList with content: ch_svdb_dbs.in_frqs.toList(),\n",
      "Nextflow operator toList with content: ch_svdb_dbs.out_occs.toList(),\n",
      "Nextflow operator toList with content: ch_svdb_dbs.out_frqs.toList(),\n",
      "Nextflow operator toList with content: ch_svdb_dbs.vcf_dbs.toList()\n",
      "Nextflow operator map with content: PICARD_SORTVCF.out.vcf.map { meta, vcf -> return [meta,vcf,[]] }.set { ch_sortvcf }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SVDB_QUERY.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(PICARD_SORTVCF.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BCFTOOLS_VIEW.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ENSEMBLVEP_SV.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(TABIX_VEP.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\raredisease\\subworkflows\\local\\annotate_structural_variants.nf\n",
      "Process CALL_REPEAT_EXPANSIONS in filepath ./git_repos\\raredisease\\subworkflows\\local\\call_repeat_expansions.nf\n",
      "Nextflow operator map with content: EXPANSIONHUNTER.out.vcf.map{ meta, vcf -> [ meta, vcf, [] ]},\n",
      "Nextflow operator join with content: RENAMESAMPLE_EXP.out.vcf.join(TABIX_EXP_RENAME.out.tbi, failOnMismatch:true, failOnDuplicate:true),\n",
      "Nextflow operator collect with content: .collect{it[1]}\n",
      "Nextflow operator toList with content: .toList()\n",
      "Nextflow operator collect with content: .collect()\n",
      "Nextflow operator set with content: .set {ch_exp_vcfs}\n",
      "Nextflow operator combine with content: .combine(ch_exp_vcfs)\n",
      "Nextflow operator set with content: .set {ch_svdb_merge_input}\n",
      "Nextflow operator map with content: STRANGER.out.vcf.map{ meta, vcf -> [meta, vcf, [] ]},\n",
      "Nextflow operator join with content: ch_vcf_idx = COMPRESS_STRANGER.out.vcf.join(INDEX_STRANGER.out.tbi, failOnMismatch:true, failOnDuplicate:true)\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(EXPANSIONHUNTER.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(BCFTOOLS_REHEADER_EXP.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(RENAMESAMPLE_EXP.out.versions.first()    )\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(TABIX_EXP_RENAME.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SPLIT_MULTIALLELICS_EXP.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SVDB_MERGE_REPEATS.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(STRANGER.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(COMPRESS_STRANGER.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(INDEX_STRANGER.out.versions.first())\n",
      "Error: Unclosed process block in file: ./git_repos\\raredisease\\subworkflows\\local\\call_repeat_expansions.nf\n",
      "Process CALL_SNV in filepath ./git_repos\\raredisease\\subworkflows\\local\\call_snv.nf\n",
      "Nextflow operator mix with content: ch_vcf      = Channel.empty().mix(CALL_SNV_DEEPVARIANT.out.vcf, CALL_SNV_SENTIEON.out.vcf)\n",
      "Nextflow operator mix with content: ch_tabix    = Channel.empty().mix(CALL_SNV_DEEPVARIANT.out.tabix, CALL_SNV_SENTIEON.out.tabix)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(CALL_SNV_DEEPVARIANT.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(CALL_SNV_SENTIEON.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\raredisease\\subworkflows\\local\\call_snv.nf\n",
      "Process CALL_STRUCTURAL_VARIANTS in filepath ./git_repos\\raredisease\\subworkflows\\local\\call_structural_variants.nf\n",
      "Nextflow operator collect with content: .collect{it[1]}\n",
      "Nextflow operator set with content: .set{ manta_vcf }\n",
      "Nextflow operator collect with content: .collect{it[1]}\n",
      "Nextflow operator set with content: .set { tiddit_vcf }\n",
      "Nextflow operator collect with content: .collect{it[1]}\n",
      "Nextflow operator set with content: .set { gcnvcaller_vcf }\n",
      "Nextflow operator combine with content: .combine(manta_vcf)\n",
      "Nextflow operator toList with content: .toList()\n",
      "Nextflow operator set with content: .set { vcf_list }\n",
      "Nextflow operator combine with content: .combine(manta_vcf)\n",
      "Nextflow operator combine with content: .combine(gcnvcaller_vcf)\n",
      "Nextflow operator toList with content: .toList()\n",
      "Nextflow operator set with content: .set { vcf_list }\n",
      "Nextflow operator combine with content: .combine(vcf_list)\n",
      "Nextflow operator set with content: .set { merge_input_vcfs }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(CALL_SV_MANTA.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(CALL_SV_TIDDIT.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(CALL_SV_GERMLINECNVCALLER.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(TABIX_TABIX.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\raredisease\\subworkflows\\local\\call_structural_variants.nf\n",
      "Process CHECK_INPUT in filepath ./git_repos\\raredisease\\subworkflows\\local\\check_input.nf\n",
      "Nextflow operator splitCsv with content: .splitCsv ( header:true, sep:',' )\n",
      "Nextflow operator set with content: .set { sheet }\n",
      "Nextflow operator toList with content: .toList()\n",
      "Nextflow operator map with content: .map {create_case_channel(it)}\n",
      "Nextflow operator map with content: reads     = sheet.map { row -> [[row.sample.split('_')[0]], row] }\n",
      "Nextflow operator groupTuple with content: .groupTuple()\n",
      "Nextflow operator map with content: .map { meta, rows ->\n",
      "Nextflow operator transpose with content: .transpose()\n",
      "Nextflow operator map with content: .map { row, numLanes ->\n",
      "Nextflow operator map with content: samples   = sheet.map { create_samples_channel(it) }\n",
      "Error: Unclosed process block in file: ./git_repos\\raredisease\\subworkflows\\local\\check_input.nf\n",
      "Process GENS in filepath ./git_repos\\raredisease\\subworkflows\\local\\gens.nf\n",
      "Nextflow operator map with content: GENS_GENERATE (DENOISEREADCOUNTS.out.standardized_read_counts, ch_vcf.map { meta, vcf -> vcf }, ch_gnomad_pos)\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(COLLECTREADCOUNTS.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(DENOISEREADCOUNTS.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(GENS_GENERATE.out.versions.first())\n",
      "Error: Unclosed process block in file: ./git_repos\\raredisease\\subworkflows\\local\\gens.nf\n",
      "Process PEDDY_CHECK in filepath ./git_repos\\raredisease\\subworkflows\\local\\peddy_check.nf\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(PEDDY.out.versions.first())\n",
      "Error: Unclosed process block in file: ./git_repos\\raredisease\\subworkflows\\local\\peddy_check.nf\n",
      "Process PREPARE_REFERENCES in filepath ./git_repos\\raredisease\\subworkflows\\local\\prepare_references.nf\n",
      "Nextflow operator set with content: BWA_INDEX_GENOME(ch_genome_fasta).index.set{ch_bwa}\n",
      "Nextflow operator set with content: SENTIEON_BWAINDEX_GENOME(ch_genome_fasta).index.set{ch_sentieonbwa}\n",
      "Nextflow operator collect with content: ch_fai = Channel.empty().mix(ch_genome_fai, SAMTOOLS_FAIDX_GENOME.out.fai).collect()\n",
      "Nextflow operator collect with content: ch_mt_fasta_in = Channel.empty().mix(ch_mt_fasta, SAMTOOLS_EXTRACT_MT.out.fa).collect()\n",
      "Nextflow operator multiMap with content: .multiMap{ meta, files ->\n",
      "Nextflow operator set with content: .set {ch_shiftfasta_mtintervals}\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(TABIX_DBSNP.out.versions)\n",
      "Nextflow operator set with content: TABIX_PT(ch_target_bed).tbi.set { ch_tbi }\n",
      "Nextflow operator set with content: TABIX_PBT(ch_target_bed).gz_tbi.set { ch_bgzip_tbi }\n",
      "Nextflow operator collect with content: .collect{ it[1] }\n",
      "Nextflow operator map with content: .map { it ->\n",
      "Nextflow operator set with content: .set { ch_bait_intervals_cat_in }\n",
      "Nextflow operator set with content: GATK_PREPROCESS_WGS (ch_genome_fasta, ch_fai, GATK_SD.out.dict, [[],[]], [[],[]]).set {ch_preprocwgs}\n",
      "Nextflow operator set with content: GATK_PREPROCESS_WES (ch_genome_fasta, ch_fai, GATK_SD.out.dict, GATK_BILT.out.interval_list, [[],[]]).set {ch_preprocwes}\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BWA_INDEX_GENOME.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BWAMEM2_INDEX_GENOME.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SENTIEON_BWAINDEX_GENOME.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_FAIDX_GENOME.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GATK_SD.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GET_CHROM_SIZES.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_EXTRACT_MT.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_FAIDX_MT_SHIFT.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GATK_SD_MT_SHIFT.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GATK_SHIFTFASTA.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BWAMEM2_INDEX_MT_SHIFT.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SENTIEON_BWAINDEX_MT_SHIFT.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(TABIX_GNOMAD_AF.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(TABIX_PT.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(TABIX_PBT.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GATK_BILT.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GATK_ILT.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(CAT_CAT_BAIT.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(UNTAR_VEP_CACHE.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GATK_PREPROCESS_WGS.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GATK_PREPROCESS_WES.out.versions)\n",
      "Nextflow operator collect with content: genome_bwa_index      = Channel.empty().mix(ch_bwa, ch_sentieonbwa).collect()            // channel: [ val(meta), path(index) ]\n",
      "Nextflow operator collect with content: genome_bwamem2_index  = BWAMEM2_INDEX_GENOME.out.index.collect()                         // channel: [ val(meta), path(index) ]\n",
      "Nextflow operator collect with content: genome_chrom_sizes    = GET_CHROM_SIZES.out.sizes.collect()                              // channel: [ path(sizes) ]\n",
      "Nextflow operator collect with content: genome_dict           = GATK_SD.out.dict.collect()                                       // channel: [ path(dict) ]\n",
      "Nextflow operator mix with content: .mix(ch_preprocwgs.interval_list,ch_preprocwes.interval_list)// channel: [ path(intervals) ]\n",
      "Nextflow operator collect with content: mt_intervals          = ch_shiftfasta_mtintervals.intervals.collect()                    // channel: [ path(intervals) ]\n",
      "Nextflow operator collect with content: mtshift_intervals     = ch_shiftfasta_mtintervals.shift_intervals.collect()              // channel: [ path(intervals) ]\n",
      "Nextflow operator collect with content: mtshift_backchain     = GATK_SHIFTFASTA.out.shift_back_chain.collect()                   // channel: [ val(meta), path(backchain) ]\n",
      "Nextflow operator collect with content: mtshift_fai           = GATK_SHIFTFASTA.out.shift_fai.collect()                          // channel: [ val(meta), path(fai) ]\n",
      "Nextflow operator collect with content: mtshift_fasta         = GATK_SHIFTFASTA.out.shift_fa.collect()                           // channel: [ val(meta), path(fai) ]\n",
      "Nextflow operator collect with content: mtshift_dict          = GATK_SHIFTFASTA.out.dict.collect()                               // channel: [ path(dict) ]\n",
      "Nextflow operator collect with content: mtshift_bwa_index     = SENTIEON_BWAINDEX_MT_SHIFT.out.index.collect()                   // channel: [ val(meta), path(index) ]\n",
      "Nextflow operator collect with content: mtshift_bwamem2_index = BWAMEM2_INDEX_MT_SHIFT.out.index.collect()                       // channel: [ val(meta), path(index) ]\n",
      "Nextflow operator collect with content: gnomad_af_idx         = TABIX_GNOMAD_AF.out.tbi.collect()                                // channel: [ val(meta), path(fasta) ]\n",
      "Nextflow operator collect with content: known_dbsnp_tbi       = TABIX_DBSNP.out.tbi.collect()                                    // channel: [ val(meta), path(fasta) ]\n",
      "Nextflow operator collect with content: target_bed            = Channel.empty().mix(ch_tbi, ch_bgzip_tbi).collect()              // channel: [ val(meta), path(bed), path(tbi) ]\n",
      "Nextflow operator collect with content: bait_intervals        = CAT_CAT_BAIT.out.file_out.map{ meta, inter -> inter}.collect()   // channel: [ path(intervals) ]\n",
      "Nextflow operator collect with content: target_intervals      = GATK_BILT.out.interval_list.map{ meta, inter -> inter}.collect() // channel: [ path(interval_list) ]\n",
      "Nextflow operator collect with content: vep_resources         = UNTAR_VEP_CACHE.out.untar.map{meta, files -> [files]}.collect()  // channel: [ path(cache) ]\n",
      "Error: Unclosed process block in file: ./git_repos\\raredisease\\subworkflows\\local\\prepare_references.nf\n",
      "Process QC_BAM in filepath ./git_repos\\raredisease\\subworkflows\\local\\qc_bam.nf\n",
      "Nextflow operator combine with content: .combine(ch_bait_intervals)\n",
      "Nextflow operator combine with content: .combine(ch_target_intervals)\n",
      "Nextflow operator set with content: .set { ch_hsmetrics_in}\n",
      "Nextflow operator map with content: ch_bam_bai.map{ meta, bam, bai -> [meta, bam, bai, []]}.set{ch_mosdepth_in}\n",
      "Nextflow operator mix with content: ch_cov   = Channel.empty().mix(PICARD_COLLECTWGSMETRICS.out.metrics, SENTIEON_WGSMETRICS.out.wgs_metrics)\n",
      "Nextflow operator mix with content: ch_cov_y = Channel.empty().mix(PICARD_COLLECTWGSMETRICS_Y.out.metrics, SENTIEON_WGSMETRICS_Y.out.wgs_metrics)\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(PICARD_COLLECTMULTIPLEMETRICS.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(PICARD_COLLECTHSMETRICS.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(QUALIMAP_BAMQC.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(TIDDIT_COV.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(UCSC_WIGTOBIGWIG.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(MOSDEPTH.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(PICARD_COLLECTWGSMETRICS.out.versions.first(), SENTIEON_WGSMETRICS.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(PICARD_COLLECTWGSMETRICS_Y.out.versions.first(), SENTIEON_WGSMETRICS_Y.out.versions.first())\n",
      "Error: Unclosed process block in file: ./git_repos\\raredisease\\subworkflows\\local\\qc_bam.nf\n",
      "Process RANK_VARIANTS in filepath ./git_repos\\raredisease\\subworkflows\\local\\rank_variants.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GENMOD_ANNOTATE.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GENMOD_MODELS.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GENMOD_SCORE.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GENMOD_COMPOUND.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(TABIX_BGZIPTABIX.out.versions)\n",
      "Nextflow operator collect with content: vcf      = TABIX_BGZIPTABIX.out.gz_tbi.map { meta, vcf, tbi -> return [ meta, vcf ] }.collect() // channel: [ val(meta), path(vcf) ]\n",
      "Error: Unclosed process block in file: ./git_repos\\raredisease\\subworkflows\\local\\rank_variants.nf\n",
      "Process SCATTER_GENOME in filepath ./git_repos\\raredisease\\subworkflows\\local\\scatter_genome.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BUILD_BED.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GATK4_SPLITINTERVALS.out.versions)\n",
      "Nextflow operator collect with content: bed             = BUILD_BED.out.bed.collect()   // channel: [ val(meta), path(bed) ]\n",
      "Nextflow operator collate with content: split_intervals = GATK4_SPLITINTERVALS.out.split_intervals.map { meta, it -> it }.flatten().collate(1) // channel: [ val(meta), [ path(interval_lists) ] ]\n",
      "Error: Unclosed process block in file: ./git_repos\\raredisease\\subworkflows\\local\\scatter_genome.nf\n",
      "Process ALIGN_BWAMEM2 in filepath ./git_repos\\raredisease\\subworkflows\\local\\alignment\\align_bwamem2.nf\n",
      "Nextflow operator join with content: bam_sorted_indexed = BWAMEM2_MEM.out.bam.join(SAMTOOLS_INDEX_ALIGN.out.bai, failOnMismatch:true, failOnDuplicate:true)\n",
      "Nextflow operator map with content: .map{ meta, bam ->\n",
      "Nextflow operator groupTuple with content: .groupTuple()\n",
      "Nextflow operator branch with content: .branch{\n",
      "Nextflow operator set with content: .set{ bams }\n",
      "Nextflow operator mix with content: prepared_bam = bams.single.mix(SAMTOOLS_MERGE.out.bam)\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(BWAMEM2_MEM.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SAMTOOLS_INDEX_ALIGN.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SAMTOOLS_STATS.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SAMTOOLS_MERGE.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(MARKDUPLICATES.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SAMTOOLS_INDEX_MARKDUP.out.versions.first())\n",
      "Error: Unclosed process block in file: ./git_repos\\raredisease\\subworkflows\\local\\alignment\\align_bwamem2.nf\n",
      "Process ALIGN_SENTIEON in filepath ./git_repos\\raredisease\\subworkflows\\local\\alignment\\align_sentieon.nf\n",
      "Nextflow operator join with content: .join(SENTIEON_BWAMEM.out.bai, failOnMismatch:true, failOnDuplicate:true)\n",
      "Nextflow operator map with content: .map{ meta, bam, bai ->\n",
      "Nextflow operator groupTuple with content: .groupTuple()\n",
      "Nextflow operator branch with content: .branch{\n",
      "Nextflow operator set with content: .set{ merge_bams_in }\n",
      "Nextflow operator mix with content: ch_bam_bai = merge_bams_in.single.mix(SENTIEON_READWRITER.out.bam_bai)\n",
      "Nextflow operator join with content: .join(SENTIEON_LOCUSCOLLECTOR.out.score, failOnMismatch:true, failOnDuplicate:true)\n",
      "Nextflow operator join with content: .join(SENTIEON_LOCUSCOLLECTOR.out.score_idx, failOnMismatch:true, failOnDuplicate:true)\n",
      "Nextflow operator set with content: .set { ch_bam_bai_score }\n",
      "Nextflow operator join with content: .join(SENTIEON_DEDUP.out.bai, failOnMismatch:true, failOnDuplicate:true)\n",
      "Nextflow operator set with content: .set { ch_dedup_bam_bai }\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SENTIEON_BQSR.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SENTIEON_BWAMEM.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SENTIEON_DATAMETRICS.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SENTIEON_LOCUSCOLLECTOR.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SENTIEON_DEDUP.out.versions.first())\n",
      "Nextflow operator ifEmpty with content: recal_bam   = ch_bqsr_bam.ifEmpty(null)                          // channel: [ val(meta), path(bam) ]\n",
      "Nextflow operator ifEmpty with content: recal_bai   = ch_bqsr_bai.ifEmpty(null)                          // channel: [ val(meta), path(bai) ]\n",
      "Nextflow operator ifEmpty with content: recal_csv   = ch_bqsr_csv.ifEmpty(null)                          // channel: [ val(meta), path(csv) ]\n",
      "Nextflow operator ifEmpty with content: mq_metrics  = SENTIEON_DATAMETRICS.out.mq_metrics.ifEmpty(null)  // channel: [ val(meta), path(mq_metrics) ]\n",
      "Nextflow operator ifEmpty with content: qd_metrics  = SENTIEON_DATAMETRICS.out.qd_metrics.ifEmpty(null)  // channel: [ val(meta), path(qd_metrics) ]\n",
      "Nextflow operator ifEmpty with content: gc_metrics  = SENTIEON_DATAMETRICS.out.gc_metrics.ifEmpty(null)  // channel: [ val(meta), path(gc_metrics) ]\n",
      "Nextflow operator ifEmpty with content: gc_summary  = SENTIEON_DATAMETRICS.out.gc_summary.ifEmpty(null)  // channel: [ val(meta), path(gc_summary) ]\n",
      "Nextflow operator ifEmpty with content: aln_metrics = SENTIEON_DATAMETRICS.out.aln_metrics.ifEmpty(null) // channel: [ val(meta), path(aln_metrics) ]\n",
      "Nextflow operator ifEmpty with content: is_metrics  = SENTIEON_DATAMETRICS.out.is_metrics.ifEmpty(null)  // channel: [ val(meta), path(is_metrics) ]\n",
      "Error: Unclosed process block in file: ./git_repos\\raredisease\\subworkflows\\local\\alignment\\align_sentieon.nf\n",
      "Process ANNOTATE_CADD in filepath ./git_repos\\raredisease\\subworkflows\\local\\annotation\\annotate_cadd.nf\n",
      "Nextflow operator join with content: BCFTOOLS_VIEW(ch_vcf.join(ch_index), [], [], [])\n",
      "Nextflow operator join with content: .join(ch_index)\n",
      "Nextflow operator join with content: .join(CADD.out.tsv)\n",
      "Nextflow operator join with content: .join(TABIX_CADD.out.tbi)\n",
      "Nextflow operator combine with content: .combine(ch_header)\n",
      "Nextflow operator set with content: .set { ch_annotate_in }\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(BCFTOOLS_VIEW.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(TABIX_VIEW.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(CADD.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(TABIX_CADD.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(BCFTOOLS_ANNOTATE.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(TABIX_ANNOTATE.out.versions.first())\n",
      "Error: Unclosed process block in file: ./git_repos\\raredisease\\subworkflows\\local\\annotation\\annotate_cadd.nf\n",
      "Process ALIGN_AND_CALL_MT in filepath ./git_repos\\raredisease\\subworkflows\\local\\mitochondria\\align_and_call_MT.nf\n",
      "Nextflow operator mix with content: .mix(BWAMEM2_MEM_MT.out.bam, SENTIEON_BWAMEM_MT.out.bam)\n",
      "Nextflow operator join with content: .join(ch_ubam, failOnMismatch:true, failOnDuplicate:true)\n",
      "Nextflow operator set with content: .set {ch_bam_ubam}\n",
      "Nextflow operator join with content: ch_sort_index_bam        = SAMTOOLS_SORT_MT.out.bam.join(SAMTOOLS_INDEX_MT.out.bai, failOnMismatch:true, failOnDuplicate:true)\n",
      "Nextflow operator combine with content: ch_sort_index_bam_int_mt = ch_sort_index_bam.combine(ch_intervals)\n",
      "Nextflow operator join with content: ch_mutect_vcf = GATK4_MUTECT2_MT.out.vcf.join(GATK4_MUTECT2_MT.out.tbi, failOnMismatch:true, failOnDuplicate:true)\n",
      "Nextflow operator join with content: ch_mutect_out = ch_mutect_vcf.join(GATK4_MUTECT2_MT.out.stats, failOnMismatch:true, failOnDuplicate:true)\n",
      "Nextflow operator map with content: ch_to_filt    = ch_mutect_out.map {\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(BWAMEM2_MEM_MT.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(GATK4_MERGEBAMALIGNMENT_MT.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(PICARD_ADDORREPLACEREADGROUPS_MT.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(PICARD_MARKDUPLICATES_MT.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SAMTOOLS_SORT_MT.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SAMTOOLS_INDEX_MT.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(EKLIPSE_MT.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(MT_DELETION.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(GATK4_MUTECT2_MT.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(HAPLOCHECK_MT.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(GATK4_FILTERMUTECTCALLS_MT.out.versions.first())\n",
      "Error: Unclosed process block in file: ./git_repos\\raredisease\\subworkflows\\local\\mitochondria\\align_and_call_MT.nf\n",
      "Process CONVERT_MT_BAM_TO_FASTQ in filepath ./git_repos\\raredisease\\subworkflows\\local\\mitochondria\\convert_mt_bam_to_fastq.nf\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(GATK4_PRINTREADS_MT.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(GATK4_REVERTSAM_MT.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(GATK4_SAMTOFASTQ_MT.out.versions.first())\n",
      "Error: Unclosed process block in file: ./git_repos\\raredisease\\subworkflows\\local\\mitochondria\\convert_mt_bam_to_fastq.nf\n",
      "Process MERGE_ANNOTATE_MT in filepath ./git_repos\\raredisease\\subworkflows\\local\\mitochondria\\merge_annotate_MT.nf\n",
      "Nextflow operator join with content: .join(ch_vcf2, remainder: true)\n",
      "Nextflow operator map with content: .map{ meta, vcf1, vcf2 ->\n",
      "Nextflow operator join with content: .join(GATK4_MERGEVCFS_LIFT_UNLIFT_MT.out.tbi, failOnMismatch:true, failOnDuplicate:true)\n",
      "Nextflow operator set with content: .set { ch_filt_vcf }\n",
      "Nextflow operator join with content: .join(GATK4_VARIANTFILTRATION_MT.out.tbi, failOnMismatch:true, failOnDuplicate:true)\n",
      "Nextflow operator set with content: .set { ch_in_split }\n",
      "Nextflow operator join with content: .join(TABIX_TABIX_MT.out.tbi, failOnMismatch:true, failOnDuplicate:true)\n",
      "Nextflow operator set with content: .set { ch_in_remdup }\n",
      "Nextflow operator collect with content: .collect{it[1]}\n",
      "Nextflow operator ifEmpty with content: .ifEmpty([])\n",
      "Nextflow operator toList with content: .toList()\n",
      "Nextflow operator set with content: .set { file_list_vcf }\n",
      "Nextflow operator collect with content: .collect{it[1]}\n",
      "Nextflow operator ifEmpty with content: .ifEmpty([])\n",
      "Nextflow operator toList with content: .toList()\n",
      "Nextflow operator set with content: .set { file_list_tbi }\n",
      "Nextflow operator combine with content: .combine(file_list_vcf)\n",
      "Nextflow operator combine with content: .combine(file_list_tbi)\n",
      "Nextflow operator set with content: .set { ch_rem_dup_vcf_tbi }\n",
      "Nextflow operator branch with content: ch_rem_dup_vcf_tbi.branch {\n",
      "Nextflow operator set with content: }.set { ch_case_vcf }\n",
      "Nextflow operator mix with content: .mix(ch_case_vcf.single)\n",
      "Nextflow operator set with content: .set { ch_annotation_in }\n",
      "Nextflow operator combine with content: .combine(ANNOTATE_CADD.out.vcf.ifEmpty(\"null\"))\n",
      "Nextflow operator branch with content: .branch { it  ->\n",
      "Nextflow operator set with content: .set { ch_for_mix }\n",
      "Nextflow operator mix with content: ch_vep_in = ch_for_mix.merged.mix(ch_for_mix.cadd)\n",
      "Nextflow operator join with content: .join(TABIX_TABIX_MT3.out.tbi, failOnMismatch:true, failOnDuplicate:true)\n",
      "Nextflow operator map with content: .map { meta, vcf, tbi -> return [meta, vcf, tbi, []]}\n",
      "Nextflow operator set with content: .set { ch_in_vcfanno }\n",
      "Nextflow operator map with content: HMTNOTE_ANNOTATE.out.vcf.map{meta, vcf ->\n",
      "Nextflow operator set with content: .set { ch_hmtnote_reformatted }\n",
      "Nextflow operator map with content: ch_vcf_out = ZIP_TABIX_HMTNOTE.out.gz_tbi.map{meta, vcf, tbi -> return [meta, vcf] }\n",
      "Nextflow operator map with content: ch_tbi_out = ZIP_TABIX_HMTNOTE.out.gz_tbi.map{meta, vcf, tbi -> return [meta, tbi] }\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(GATK4_MERGEVCFS_LIFT_UNLIFT_MT.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(GATK4_VARIANTFILTRATION_MT.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SPLIT_MULTIALLELICS_MT.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(REMOVE_DUPLICATES_MT.out.versions.first())\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BCFTOOLS_MERGE_MT.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ANNOTATE_CADD.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ENSEMBLVEP_MT.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(VCFANNO_MT.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(HMTNOTE_ANNOTATE.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(HAPLOGREP2_CLASSIFY_MT.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\raredisease\\subworkflows\\local\\mitochondria\\merge_annotate_MT.nf\n",
      "Process CALL_SNV_DEEPVARIANT in filepath ./git_repos\\raredisease\\subworkflows\\local\\variant_calling\\call_snv_deepvariant.nf\n",
      "Nextflow operator map with content: ch_bam_bai.map { meta, bam, bai ->\n",
      "Nextflow operator set with content: .set { ch_deepvar_in }\n",
      "Nextflow operator collect with content: .collect{it[1]}\n",
      "Nextflow operator toList with content: .toList()\n",
      "Nextflow operator collect with content: .collect()\n",
      "Nextflow operator set with content: .set { ch_file_list }\n",
      "Nextflow operator combine with content: .combine(ch_file_list)\n",
      "Nextflow operator set with content: .set { ch_gvcfs }\n",
      "Nextflow operator map with content: .map{ meta, bcf ->\n",
      "Nextflow operator map with content: .map{ meta, vcf ->\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(DEEPVARIANT.out.versions.first())\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GLNEXUS.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SPLIT_MULTIALLELICS_GL.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(REMOVE_DUPLICATES_GL.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(TABIX_GL.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\raredisease\\subworkflows\\local\\variant_calling\\call_snv_deepvariant.nf\n",
      "Process CALL_SNV_SENTIEON in filepath ./git_repos\\raredisease\\subworkflows\\local\\variant_calling\\call_snv_sentieon.nf\n",
      "Nextflow operator join with content: BCF_FILTER_TWO.out.vcf.join(TABIX_BCFTOOLS.out.tbi, failOnMismatch:true, failOnDuplicate:true)\n",
      "Nextflow operator map with content: .map { meta,vcf,tbi -> return [vcf,tbi] }\n",
      "Nextflow operator set with content: .set { ch_vcf_idx }\n",
      "Nextflow operator combine with content: .combine(ch_vcf_idx)\n",
      "Nextflow operator groupTuple with content: .groupTuple()\n",
      "Nextflow operator branch with content: .branch{                                                                                                    // branch the channel into multiple channels (single, multiple) depending on size of list\n",
      "Nextflow operator set with content: .set{ ch_vcf_idx_merge_in }\n",
      "Nextflow operator map with content: .map{meta, bcf ->\n",
      "Nextflow operator mix with content: ch_vcf_idx_case =  ch_vcf_idx_merge_in.single.mix(ch_split_multi_in)\n",
      "Nextflow operator map with content: .map{meta, vcf ->\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SENTIEON_DNASCOPE.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SENTIEON_DNAMODELAPPLY.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(BCFTOOLS_MERGE.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SPLIT_MULTIALLELICS_SEN.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(REMOVE_DUPLICATES_SEN.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(TABIX_SEN.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(BCF_FILTER_ONE.out.versions.first())\n",
      "Error: Unclosed process block in file: ./git_repos\\raredisease\\subworkflows\\local\\variant_calling\\call_snv_sentieon.nf\n",
      "Process CALL_SV_GERMLINECNVCALLER in filepath ./git_repos\\raredisease\\subworkflows\\local\\variant_calling\\call_sv_germlinecnvcaller.nf\n",
      "Nextflow operator collect with content: input = ch_bam_bai.combine( ch_readcount_intervals.collect{ it[1] } )\n",
      "Nextflow operator map with content: .map({ meta, tsv -> return [meta, tsv, [], [] ]})\n",
      "Nextflow operator set with content: .set{ch_dgcp_in}\n",
      "Nextflow operator join with content: .join(GATK4_DETERMINEGERMLINECONTIGPLOIDY.out.calls)\n",
      "Nextflow operator combine with content: .combine(ch_gcnvcaller_model)\n",
      "Nextflow operator map with content: .map({ meta, tsv, calls, meta2, model -> return [meta, tsv, [], calls, model ]})\n",
      "Nextflow operator set with content: .set{ch_gcnvc_in}\n",
      "Nextflow operator toList with content: GATK4_GERMLINECNVCALLER.out.calls.toList()\n",
      "Nextflow operator flatMap with content: .flatMap {reduce_input(it)}\n",
      "Nextflow operator buffer with content: .buffer (size: 2)\n",
      "Nextflow operator collect with content: .combine(ch_gcnvcaller_model.collect{it[1]}.toList())\n",
      "Nextflow operator join with content: .join(GATK4_DETERMINEGERMLINECONTIGPLOIDY.out.calls)\n",
      "Nextflow operator set with content: .set {ch_postproc_in}\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GATK4_COLLECTREADCOUNTS.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GATK4_DETERMINEGERMLINECONTIGPLOIDY.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GATK4_GERMLINECNVCALLER.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GATK4_POSTPROCESSGERMLINECNVCALLS.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\raredisease\\subworkflows\\local\\variant_calling\\call_sv_germlinecnvcaller.nf\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Process CALL_SV_MANTA in filepath ./git_repos\\raredisease\\subworkflows\\local\\variant_calling\\call_sv_manta.nf\n",
      "Nextflow operator collect with content: ch_bam.collect{it[1]}\n",
      "Nextflow operator toList with content: .toList()\n",
      "Nextflow operator set with content: .set { bam_file_list }\n",
      "Nextflow operator collect with content: ch_bai.collect{it[1]}\n",
      "Nextflow operator toList with content: .toList()\n",
      "Nextflow operator set with content: .set { bai_file_list }\n",
      "Nextflow operator map with content: ch_bed.map {\n",
      "Nextflow operator set with content: .set { bed_input }\n",
      "Nextflow operator combine with content: ch_case_info.combine(bam_file_list)\n",
      "Nextflow operator combine with content: .combine(bai_file_list)\n",
      "Nextflow operator map with content: .map { it -> it + [ [], [] ] }\n",
      "Nextflow operator set with content: .set { manta_input }\n",
      "Nextflow operator combine with content: ch_case_info.combine(bam_file_list)\n",
      "Nextflow operator combine with content: .combine(bai_file_list)\n",
      "Nextflow operator combine with content: .combine(bed_input)\n",
      "Nextflow operator set with content: .set { manta_input }\n",
      "Error: Unclosed process block in file: ./git_repos\\raredisease\\subworkflows\\local\\variant_calling\\call_sv_manta.nf\n",
      "Process CALL_SV_TIDDIT in filepath ./git_repos\\raredisease\\subworkflows\\local\\variant_calling\\call_sv_tiddit.nf\n",
      "Nextflow operator collect with content: .collect{it[1]}\n",
      "Nextflow operator toList with content: .toList()\n",
      "Nextflow operator set with content: .set { vcf_file_list }\n",
      "Nextflow operator combine with content: .combine(vcf_file_list)\n",
      "Nextflow operator set with content: .set { merge_input_vcfs }\n",
      "Nextflow operator first with content: ch_versions = TIDDIT_SV.out.versions.first()\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SVDB_MERGE_TIDDIT.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\raredisease\\subworkflows\\local\\variant_calling\\call_sv_tiddit.nf\n",
      "Process RAREDISEASE in filepath ./git_repos\\raredisease\\workflows\\raredisease.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(CHECK_INPUT.out.versions)\n",
      "Nextflow operator collect with content: ch_genome_fasta             = Channel.fromPath(params.fasta).map { it -> [[id:it[0].simpleName], it] }.collect()\n",
      "Nextflow operator collect with content: ch_genome_fai               = params.fai            ? Channel.fromPath(params.fai).map {it -> [[id:it[0].simpleName], it]}.collect()\n",
      "Nextflow operator collect with content: ch_gnomad_af_tab            = params.gnomad_af      ? Channel.fromPath(params.gnomad_af).map{ it -> [[id:it[0].simpleName], it] }.collect()\n",
      "Nextflow operator collect with content: ch_dbsnp                    = params.known_dbsnp    ? Channel.fromPath(params.known_dbsnp).map{ it -> [[id:it[0].simpleName], it] }.collect()\n",
      "Nextflow operator collect with content: ch_mt_fasta                 = params.mt_fasta       ? Channel.fromPath(params.mt_fasta).map { it -> [[id:it[0].simpleName], it] }.collect()\n",
      "Nextflow operator collect with content: ch_target_bed_unprocessed   = params.target_bed     ? Channel.fromPath(params.target_bed).map{ it -> [[id:it[0].simpleName], it] }.collect()\n",
      "Nextflow operator collect with content: ch_vep_cache_unprocessed    = params.vep_cache      ? Channel.fromPath(params.vep_cache).map { it -> [[id:'vep_cache'], it] }.collect()\n",
      "Nextflow operator set with content: .set { ch_references }\n",
      "Nextflow operator collect with content: ch_cadd_header              = Channel.fromPath(\"$projectDir/assets/cadd_to_vcf_header_-1.0-.txt\", checkIfExists: true).collect()\n",
      "Nextflow operator collect with content: ch_cadd_resources           = params.cadd_resources                    ? Channel.fromPath(params.cadd_resources).collect()\n",
      "Nextflow operator collect with content: ch_call_interval            = params.call_interval                     ? Channel.fromPath(params.call_interval).collect()\n",
      "Nextflow operator collect with content: ch_dbsnp_tbi                = params.known_dbsnp_tbi                   ? Channel.fromPath(params.known_dbsnp_tbi).map {it -> [[id:it[0].simpleName], it]}.collect()\n",
      "Nextflow operator ifEmpty with content: : ch_references.known_dbsnp_tbi.ifEmpty([[],[]])\n",
      "Nextflow operator splitCsv with content: ch_gcnvcaller_model         = params.gcnvcaller_model                  ? Channel.fromPath(params.gcnvcaller_model).splitCsv ( header:true )\n",
      "Nextflow operator map with content: .map { row ->\n",
      "Nextflow operator collect with content: ch_genome_bwaindex          = params.bwa                               ? Channel.fromPath(params.bwa).map {it -> [[id:it[0].simpleName], it]}.collect()\n",
      "Nextflow operator collect with content: ch_genome_bwamem2index      = params.bwamem2                           ? Channel.fromPath(params.bwamem2).map {it -> [[id:it[0].simpleName], it]}.collect()\n",
      "Nextflow operator collect with content: ch_genome_dictionary        = params.sequence_dictionary               ? Channel.fromPath(params.sequence_dictionary).map {it -> [[id:it[0].simpleName], it]}.collect()\n",
      "Nextflow operator collect with content: ch_gnomad_afidx             = params.gnomad_af_idx                     ? Channel.fromPath(params.gnomad_af_idx).collect()\n",
      "Nextflow operator collect with content: ch_gnomad_af                = params.gnomad_af                         ? ch_gnomad_af_tab.join(ch_gnomad_afidx).map {meta, tab, idx -> [tab,idx]}.collect()\n",
      "Nextflow operator collect with content: ch_intervals_wgs            = params.intervals_wgs                     ? Channel.fromPath(params.intervals_wgs).collect()\n",
      "Nextflow operator collect with content: ch_intervals_y              = params.intervals_y                       ? Channel.fromPath(params.intervals_y).collect()\n",
      "Nextflow operator collect with content: ch_ml_model                 = params.variant_caller.equals(\"sentieon\") ? Channel.fromPath(params.ml_model).collect()\n",
      "Nextflow operator collect with content: ch_ploidy_model             = params.ploidy_model                      ? Channel.fromPath(params.ploidy_model).map{ it -> [[id:it[0].simpleName], it] }.collect()\n",
      "Nextflow operator collect with content: ch_readcount_intervals      = params.readcount_intervals               ? Channel.fromPath(params.readcount_intervals).collect()\n",
      "Nextflow operator collect with content: ch_reduced_penetrance       = params.reduced_penetrance                ? Channel.fromPath(params.reduced_penetrance).collect()\n",
      "Nextflow operator collect with content: ch_score_config_snv         = params.score_config_snv                  ? Channel.fromPath(params.score_config_snv).collect()\n",
      "Nextflow operator collect with content: ch_score_config_sv          = params.score_config_sv                   ? Channel.fromPath(params.score_config_sv).collect()\n",
      "Nextflow operator collect with content: ch_variant_catalog          = params.variant_catalog                   ? Channel.fromPath(params.variant_catalog).map { it -> [[id:it[0].simpleName],it]}.collect()\n",
      "Nextflow operator collect with content: ch_variant_consequences     = Channel.fromPath(\"$projectDir/assets/variant_consequences_v1.txt\", checkIfExists: true).collect()\n",
      "Nextflow operator collect with content: ch_vcfanno_resources        = params.vcfanno_resources                 ? Channel.fromPath(params.vcfanno_resources).splitText().map{it -> it.trim()}.collect()\n",
      "Nextflow operator collect with content: ch_vcfanno_lua              = params.vcfanno_lua                       ? Channel.fromPath(params.vcfanno_lua).collect()\n",
      "Nextflow operator collect with content: ch_vcfanno_toml             = params.vcfanno_toml                      ? Channel.fromPath(params.vcfanno_toml).collect()\n",
      "Nextflow operator collect with content: : ( params.vep_cache    ? Channel.fromPath(params.vep_cache).collect() : Channel.value([]) )\n",
      "Nextflow operator collect with content: ch_vep_filters              = params.vep_filters                       ? Channel.fromPath(params.vep_filters).collect()\n",
      "Nextflow operator mix with content: ch_versions                 = ch_versions.mix(ch_references.versions)\n",
      "Nextflow operator map with content: ch_pedfile = CHECK_INPUT.out.samples.toList().map { makePed(it) }\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(FASTQC.out.versions.first())\n",
      "Nextflow operator set with content: .set { ch_scatter }\n",
      "Nextflow operator set with content: .set { ch_mapped }\n",
      "Nextflow operator mix with content: ch_versions   = ch_versions.mix(ALIGN.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(QC_BAM.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(CALL_REPEAT_EXPANSIONS.out.versions)\n",
      "Nextflow operator collect with content: .collect{it[1]}\n",
      "Nextflow operator toList with content: .toList()\n",
      "Nextflow operator set with content: .set { ch_bam_list }\n",
      "Nextflow operator collect with content: .collect{it[2]}\n",
      "Nextflow operator toList with content: .toList()\n",
      "Nextflow operator set with content: .set { ch_bai_list }\n",
      "Nextflow operator combine with content: .combine(ch_bam_list)\n",
      "Nextflow operator combine with content: .combine(ch_bai_list)\n",
      "Nextflow operator set with content: .set { ch_bams_bais }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SMNCOPYNUMBERCALLER.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(CALL_SNV.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(CALL_STRUCTURAL_VARIANTS.out.versions)\n",
      "Nextflow operator join with content: CALL_SNV.out.vcf.join(CALL_SNV.out.tabix, failOnMismatch:true, failOnDuplicate:true),\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(PEDDY_CHECK.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GENS.out.versions)\n",
      "Nextflow operator set with content: ).set {ch_sv_annotate}\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ch_sv_annotate.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ANN_CSQ_PLI_SV.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(RANK_VARIANTS_SV.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(FILTER_VEP_SV.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ANALYSE_MT.out.versions)\n",
      "Nextflow operator join with content: ch_vcf = CALL_SNV.out.vcf.join(CALL_SNV.out.tabix, failOnMismatch:true, failOnDuplicate:true)\n",
      "Nextflow operator map with content: .map { meta, vcf, tbi -> return [meta, vcf, tbi, []]}\n",
      "Nextflow operator set with content: .set { ch_selvar_in }\n",
      "Nextflow operator join with content: ch_vcf = GATK4_SELECTVARIANTS.out.vcf.join(GATK4_SELECTVARIANTS.out.tbi, failOnMismatch:true, failOnDuplicate:true)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GATK4_SELECTVARIANTS.out.versions)\n",
      "Nextflow operator set with content: ).set {ch_snv_annotate}\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ch_snv_annotate.versions)\n",
      "Nextflow operator concat with content: .concat(ANALYSE_MT.out.vcf)\n",
      "Nextflow operator groupTuple with content: .groupTuple()\n",
      "Nextflow operator set with content: .set { ch_merged_vcf }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GATK4_MERGEVCFS.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ANN_CSQ_PLI_SNV.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(RANK_VARIANTS_SNV.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(FILTER_VEP_SNV.out.versions)\n",
      "Nextflow operator collectFile with content: ch_versions.unique().collectFile(name: 'collated_versions.yml')\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_workflow_summary.collectFile(name: 'workflow_summary_mqc.yaml'))\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_methods_description.collectFile(name: 'methods_description_mqc.yaml'))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(CUSTOM_DUMPSOFTWAREVERSIONS.out.mqc_yml.collect())\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(FASTQC.out.zip.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(QC_BAM.out.multiple_metrics.map{it[1]}.collect().ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(QC_BAM.out.hs_metrics.map{it[1]}.collect().ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(QC_BAM.out.qualimap_results.map{it[1]}.collect().ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(QC_BAM.out.global_dist.map{it[1]}.collect().ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(QC_BAM.out.cov.map{it[1]}.collect().ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(PEDDY_CHECK.out.ped.map{it[1]}.collect().ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(PEDDY_CHECK.out.csv.map{it[1]}.collect().ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files.collect(),\n",
      "Nextflow operator toList with content: ch_multiqc_config.toList(),\n",
      "Nextflow operator toList with content: ch_multiqc_custom_config.toList(),\n",
      "Nextflow operator toList with content: ch_multiqc_logo.toList()\n",
      "Nextflow operator toList with content: multiqc_report = MULTIQC.out.report.toList()\n",
      "Nextflow operator join with content: outfile.text = ['#family_id', 'sample_id', 'father', 'mother', 'sex', 'phenotype'].join('\\t')\n",
      "Nextflow operator join with content: sample_name        =  sample_tokenized.join(\"_\")\n",
      "Nextflow operator join with content: outfile.append('\\n' + [samples[i].case_id, sample_name, samples[i].paternal, samples[i].maternal, samples[i].sex, samples[i].phenotype].join('\\t'));\n",
      "Error: Unclosed process block in file: ./git_repos\\raredisease\\workflows\\raredisease.nf\n",
      "Process NFCORE_READSIMULATOR in filepath ./git_repos\\readsimulator\\main.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\readsimulator\\main.nf\n",
      "Process AMPLICON_WORKFLOW in filepath ./git_repos\\readsimulator\\subworkflows\\local\\amplicon_workflow.nf\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator mix with content: ch_versions  = ch_versions.mix(CRABS_DBIMPORT.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(CRABS_INSILICOPCR.out.versions)\n",
      "Nextflow operator combine with content: .combine ( ch_input )\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(ART_ILLUMINA.out.versions.first())\n",
      "Nextflow operator map with content: .map {\n",
      "Error: Unclosed process block in file: ./git_repos\\readsimulator\\subworkflows\\local\\amplicon_workflow.nf\n",
      "Process INPUT_CHECK in filepath ./git_repos\\readsimulator\\subworkflows\\local\\input_check.nf\n",
      "Nextflow operator splitCsv with content: .splitCsv ( header:true, sep:',' )\n",
      "Nextflow operator map with content: .map { create_fastq_channel(it) }\n",
      "Nextflow operator set with content: .set { reads }\n",
      "Error: Unclosed process block in file: ./git_repos\\readsimulator\\subworkflows\\local\\input_check.nf\n",
      "Process TARGET_CAPTURE_WORKFLOW in filepath ./git_repos\\readsimulator\\subworkflows\\local\\target_capture_workflow.nf\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(BOWTIE2_BUILD.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(BOWTIE2_ALIGN.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SAMTOOLS_INDEX.out.versions.first())\n",
      "Nextflow operator map with content: ch_capsim_input = ch_meta_fasta.map { it = it[1] }\n",
      "Nextflow operator combine with content: .combine ( BOWTIE2_ALIGN.out.aligned.map { it = it[1] } )\n",
      "Nextflow operator combine with content: .combine ( SAMTOOLS_INDEX.out.bai.map { it = it[1] } )\n",
      "Nextflow operator combine with content: .combine ( ch_input )\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(JAPSA_CAPSIM.out.versions.first())\n",
      "Nextflow operator map with content: .map {\n",
      "Error: Unclosed process block in file: ./git_repos\\readsimulator\\subworkflows\\local\\target_capture_workflow.nf\n",
      "Process READSIMULATOR in filepath ./git_repos\\readsimulator\\workflows\\readsimulator\\main.nf\n",
      "Nextflow operator ifEmpty with content: ch_accessions.ifEmpty([]),\n",
      "Nextflow operator ifEmpty with content: ch_taxids.ifEmpty([]),\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator ifEmpty with content: ch_fasta.ifEmpty([]),\n",
      "Nextflow operator first with content: ch_versions        = ch_versions.mix(AMPLICON_WORKFLOW.out.versions.first())\n",
      "Nextflow operator mix with content: ch_simulated_reads = ch_simulated_reads.mix(AMPLICON_WORKFLOW.out.reads)\n",
      "Nextflow operator ifEmpty with content: ch_probes.ifEmpty([])\n",
      "Nextflow operator first with content: ch_versions        = ch_versions.mix(TARGET_CAPTURE_WORKFLOW.out.versions.first())\n",
      "Nextflow operator mix with content: ch_simulated_reads = ch_simulated_reads.mix(TARGET_CAPTURE_WORKFLOW.out.reads)\n",
      "Nextflow operator combine with content: ch_input.combine(ch_fasta.ifEmpty([[]]))\n",
      "Nextflow operator first with content: ch_versions         = ch_versions.mix(INSILICOSEQ_GENERATE.out.versions.first())\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator mix with content: ch_simulated_reads  = ch_simulated_reads.mix(ch_metagenome_reads)\n",
      "Nextflow operator combine with content: ch_input.combine(ch_fasta)\n",
      "Nextflow operator first with content: ch_versions          = ch_versions.mix(WGSIM.out.versions.first())\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator mix with content: ch_simulated_reads  = ch_simulated_reads.mix(ch_wholegenome_reads)\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator groupTuple with content: .groupTuple(sort: 'deep')\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(FASTQC.out.versions.first())\n",
      "Nextflow operator collectFile with content: ch_versions.unique().collectFile(name: 'collated_versions.yml')\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_workflow_summary.collectFile(name: 'workflow_summary_mqc.yaml'))\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_methods_description.collectFile(name: 'methods_description_mqc.yaml'))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(CUSTOM_DUMPSOFTWAREVERSIONS.out.mqc_yml.collect())\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(FASTQC.out.zip.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files.collect(),\n",
      "Nextflow operator toList with content: ch_multiqc_config.toList(),\n",
      "Nextflow operator toList with content: ch_multiqc_custom_config.toList(),\n",
      "Nextflow operator toList with content: ch_multiqc_logo.toList()\n",
      "Nextflow operator toList with content: multiqc_report = MULTIQC.out.report.toList()\n",
      "Error: Unclosed process block in file: ./git_repos\\readsimulator\\workflows\\readsimulator\\main.nf\n",
      "Process NFCORE_RIBOSEQ in filepath ./git_repos\\riboseq\\main.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\riboseq\\main.nf\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Process INPUT_CHECK in filepath ./git_repos\\riboseq\\subworkflows\\local\\input_check.nf\n",
      "Nextflow operator splitCsv with content: .splitCsv ( header:true, sep:',' )\n",
      "Nextflow operator map with content: .map { create_fastq_channel(it) }\n",
      "Nextflow operator set with content: .set { reads }\n",
      "Error: Unclosed process block in file: ./git_repos\\riboseq\\subworkflows\\local\\input_check.nf\n",
      "Process RIBOSEQ in filepath ./git_repos\\riboseq\\workflows\\riboseq.nf\n",
      "Nextflow operator view with content: ch_fastq.view()\n",
      "Nextflow operator map with content: ch_gtf.map { [ [:], it ] }\n",
      "Nextflow operator map with content: ch_rRNA_fasta.map { [ [:], it ] },\n",
      "Nextflow operator first with content: ch_hisat2_index_rRNA = HISAT2_BUILD_rRNA.out.index.first()\n",
      "Nextflow operator map with content: ch_transcriptome_fasta.map { [ [:], it ] },\n",
      "Nextflow operator map with content: ch_gtf.map { [ [:], it ] },\n",
      "Nextflow operator first with content: ch_splicesites.txt.first()\n",
      "Nextflow operator first with content: ch_hisat2_index_transcriptome = HISAT2_BUILD_transcriptome.out.index.first()\n",
      "Nextflow operator view with content: ch_fastq.view()\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(FASTQC.out.versions.first())\n",
      "Nextflow operator collectFile with content: ch_versions.unique().collectFile(name: 'collated_versions.yml')\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(CUTADAPT.out.versions.first())\n",
      "Nextflow operator set with content: .set { hisat2_rRNA_results }\n",
      "Nextflow operator first with content: ch_splicesites.txt.first()\n",
      "Nextflow operator join with content: ch_transcriptome_sorted_bam.join(ch_transcriptome_sorted_bai, by: [0]),\n",
      "Nextflow operator map with content: ch_transcriptome_fasta.map { [ [:], it ] },\n",
      "Nextflow operator view with content: UMITOOLS_DEDUP.out.bam.view()\n",
      "Nextflow operator map with content: ch_bam_scale = UMITOOLS_DEDUP.out.bam.map {\n",
      "Nextflow operator view with content: ch_bam_scale.view()\n",
      "Nextflow operator first with content: CUSTOM_GETCHROMSIZES.out.sizes.first().map { it[1] },\n",
      "Nextflow operator first with content: CUSTOM_GETCHROMSIZES.out.sizes.first().map { it[1] },\n",
      "Nextflow operator combine with content: .combine(ch_gtf)\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_workflow_summary.collectFile(name: 'workflow_summary_mqc.yaml'))\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_methods_description.collectFile(name: 'methods_description_mqc.yaml'))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(CUSTOM_DUMPSOFTWAREVERSIONS.out.mqc_yml.collect())\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(FASTQC.out.zip.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(ch_hisat2_rRNA_multiqc.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(ch_hisat2_transcriptome_multiqc.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(ch_featurecounts_multiqc.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files.collect(),\n",
      "Nextflow operator toList with content: ch_multiqc_config.toList(),\n",
      "Nextflow operator toList with content: ch_multiqc_custom_config.toList(),\n",
      "Nextflow operator toList with content: ch_multiqc_logo.toList()\n",
      "Nextflow operator toList with content: multiqc_report = MULTIQC.out.report.toList()\n",
      "Error: Unclosed process block in file: ./git_repos\\riboseq\\workflows\\riboseq.nf\n",
      "Process NFCORE_RNADNAVAR in filepath ./git_repos\\rnadnavar\\main.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\rnadnavar\\main.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\rnadnavar\\conf\\modules\\variant_calling\\mutect2.config\n",
      "Error: Unclosed process block in file: ./git_repos\\rnadnavar\\modules\\local\\create_intervals_bed\\main.nf\n",
      "Process BAM_ALIGN in filepath ./git_repos\\rnadnavar\\subworkflows\\local\\bam_align\\main.nf\n",
      "Nextflow operator branch with content: input_sample_type = input_sample.branch{\n",
      "Nextflow operator mix with content: input_fastq = input_sample_type.fastq.mix(CONVERT_FASTQ_INPUT.out.reads)\n",
      "Nextflow operator collect with content: reports = reports.mix(FASTQC.out.zip.collect{ meta, logs -> logs })\n",
      "Nextflow operator first with content: versions = versions.mix(FASTQC.out.versions.first())\n",
      "Nextflow operator collect with content: reports = reports.mix(FASTP.out.json.collect{ meta, json -> json })\n",
      "Nextflow operator collect with content: reports = reports.mix(FASTP.out.html.collect{ meta, html -> html })\n",
      "Nextflow operator map with content: reads_for_alignment = FASTP.out.reads.map{ meta, reads ->\n",
      "Nextflow operator collate with content: read_files = reads.sort(false) { a,b -> a.getName().tokenize('.')[0] <=> b.getName().tokenize('.')[0] }.collate(2)\n",
      "Nextflow operator transpose with content: }.transpose()\n",
      "Nextflow operator mix with content: versions = versions.mix(FASTP.out.versions)\n",
      "Nextflow operator map with content: reads_for_alignment = reads_for_alignment.map{ meta, reads ->\n",
      "Nextflow operator branch with content: reads_for_alignment_status = reads_for_alignment.branch{\n",
      "Nextflow operator map with content: bam_mapped_dna = FASTQ_ALIGN_BWAMEM_MEM2_DRAGMAP.out.bam.map{ meta, bam ->\n",
      "Nextflow operator groupTuple with content: }.groupTuple()\n",
      "Nextflow operator dump with content: bam_mapped_dna.dump(tag:\"bam_mapped_dna\")\n",
      "Nextflow operator dump with content: reads_for_alignment_status.rna.dump(tag:\"reads_for_alignment_status.rna\")\n",
      "Nextflow operator map with content: bam_mapped_rna = FASTQ_ALIGN_STAR.out.bam.map{ meta, bam ->\n",
      "Nextflow operator groupTuple with content: }.groupTuple()\n",
      "Nextflow operator dump with content: bam_mapped_rna.dump(tag:\"bam_mapped_rna\")\n",
      "Nextflow operator collect with content: reports           = reports.mix(FASTQ_ALIGN_STAR.out.stats.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: reports           = reports.mix(FASTQ_ALIGN_STAR.out.log_final.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator mix with content: versions          = versions.mix(FASTQ_ALIGN_STAR.out.versions)\n",
      "Nextflow operator mix with content: bam_mapped = bam_mapped_dna.mix(bam_mapped_rna)\n",
      "Nextflow operator mix with content: versions = versions.mix(BAM_MERGE_INDEX_SAMTOOLS.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(BAM_TO_CRAM_MAPPING.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(CONVERT_FASTQ_INPUT.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(FASTQ_ALIGN_BWAMEM_MEM2_DRAGMAP.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(FASTQ_ALIGN_STAR.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\rnadnavar\\subworkflows\\local\\bam_align\\main.nf\n",
      "Process BAM_APPLYBQSR in filepath ./git_repos\\rnadnavar\\subworkflows\\local\\bam_applybqsr\\main.nf\n",
      "Nextflow operator combine with content: cram_intervals = cram.combine(intervals)\n",
      "Nextflow operator map with content: .map{ meta, cram, crai, recal, intervals, num_intervals -> [ meta + [ num_intervals:num_intervals ], cram, crai, recal, intervals ] }\n",
      "Nextflow operator map with content: GATK4_APPLYBQSR(cram_intervals, fasta, fasta_fai, dict.map{ meta, it -> [ it ] })\n",
      "Nextflow operator groupTuple with content: cram_to_merge = GATK4_APPLYBQSR.out.cram.map{ meta, cram -> [ groupKey(meta, meta.num_intervals), cram ] }.groupTuple()\n",
      "Nextflow operator map with content: .map{ meta, cram, crai -> [ meta - meta.subMap('num_intervals'), cram, crai ] }\n",
      "Nextflow operator mix with content: versions = versions.mix(GATK4_APPLYBQSR.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(CRAM_MERGE_INDEX_SAMTOOLS.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\rnadnavar\\subworkflows\\local\\bam_applybqsr\\main.nf\n",
      "Process BAM_BASERECALIBRATOR in filepath ./git_repos\\rnadnavar\\subworkflows\\local\\bam_baserecalibrator\\main.nf\n",
      "Nextflow operator combine with content: cram_intervals = cram.combine(intervals)\n",
      "Nextflow operator map with content: .map{ meta, cram, crai, intervals, num_intervals -> [ meta + [ num_intervals:num_intervals ], cram, crai, intervals ] }\n",
      "Nextflow operator map with content: GATK4_BASERECALIBRATOR(cram_intervals, fasta, fasta_fai, dict.map{ meta, it -> [ it ] }, known_sites, known_sites_tbi)\n",
      "Nextflow operator branch with content: table_to_merge = GATK4_BASERECALIBRATOR.out.table.map{ meta, table -> [ groupKey(meta, meta.num_intervals), table ] }.groupTuple().branch{\n",
      "Nextflow operator map with content: table_bqsr = GATK4_GATHERBQSRREPORTS.out.table.mix(table_to_merge.single.map{ meta, table -> [ meta, table[0] ] })\n",
      "Nextflow operator map with content: .map{ meta, table -> [ meta - meta.subMap('num_intervals'), table ] }\n",
      "Nextflow operator mix with content: versions = versions.mix(GATK4_BASERECALIBRATOR.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(GATK4_GATHERBQSRREPORTS.out.versions)\n",
      "Process BAM_CONVERT_SAMTOOLS in filepath ./git_repos\\rnadnavar\\subworkflows\\local\\bam_convert_samtools\\main.nf\n",
      "Nextflow operator join with content: .join(SAMTOOLS_VIEW_UNMAP_MAP.out.bam, failOnDuplicate: true, remainder: true)\n",
      "Nextflow operator join with content: .join(SAMTOOLS_VIEW_MAP_UNMAP.out.bam, failOnDuplicate: true, remainder: true)\n",
      "Nextflow operator map with content: .map{ meta, unmap_unmap, unmap_map, map_unmap -> [ meta, [ unmap_unmap, unmap_map, map_unmap ] ] }\n",
      "Nextflow operator join with content: .join(COLLATE_FASTQ_UNMAP.out.fastq, failOnDuplicate: true, failOnMismatch: true)\n",
      "Nextflow operator map with content: .map{ meta, mapped_reads, unmapped_reads -> [ meta, [ mapped_reads[0], mapped_reads[1], unmapped_reads[0], unmapped_reads[1] ] ] }\n",
      "Nextflow operator mix with content: versions = versions.mix(CAT_FASTQ.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(COLLATE_FASTQ_MAP.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(COLLATE_FASTQ_UNMAP.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(SAMTOOLS_MERGE_UNMAP.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(SAMTOOLS_VIEW_MAP_MAP.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(SAMTOOLS_VIEW_MAP_UNMAP.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(SAMTOOLS_VIEW_UNMAP_MAP.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(SAMTOOLS_VIEW_UNMAP_UNMAP.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\rnadnavar\\subworkflows\\local\\bam_convert_samtools\\main.nf\n",
      "Process BAM_GATK_PREPROCESSING in filepath ./git_repos\\rnadnavar\\subworkflows\\local\\bam_gatk_preprocessing\\main.nf\n",
      "Nextflow operator map with content: bam_mapped = bam_mapped.map { infoMap ->\n",
      "Nextflow operator map with content: cram_mapped = cram_mapped.map { infoMap ->\n",
      "Nextflow operator map with content: cram_for_markduplicates = params.step == 'mapping' || second_run ? bam_mapped : input_sample.map{ meta, input, index -> [ meta, input ] }\n",
      "Nextflow operator branch with content: input_markduplicates_convert = cram_for_markduplicates.branch{\n",
      "Nextflow operator mix with content: versions = versions.mix(BAM_TO_CRAM.out.versions)\n",
      "Nextflow operator mix with content: cram_skip_markduplicates = Channel.empty().mix(input_markduplicates_convert.cram, BAM_TO_CRAM.out.alignment_index)\n",
      "Nextflow operator collect with content: reports = reports.mix(CRAM_QC_NO_MD.out.reports.collect{ meta, report -> report })\n",
      "Nextflow operator mix with content: versions = versions.mix(CRAM_QC_NO_MD.out.versions)\n",
      "Nextflow operator dump with content: cram_for_markduplicates.dump(tag:\"cram_for_markduplicates\")\n",
      "Nextflow operator collect with content: reports = reports.mix(BAM_MARKDUPLICATES.out.reports.collect{ meta, report -> report })\n",
      "Nextflow operator mix with content: versions = versions.mix(BAM_MARKDUPLICATES.out.versions)\n",
      "Nextflow operator mix with content: ch_md_cram_for_restart = Channel.empty().mix(cram_markduplicates_no_spark)\n",
      "Nextflow operator map with content: .map{ meta, cram, crai -> [ meta + [data_type: \"cram\"], cram, crai ] }\n",
      "Nextflow operator mix with content: versions = versions.mix(CRAM_TO_BAM.out.versions)\n",
      "Nextflow operator mix with content: ch_md_cram_for_restart   = Channel.empty().mix(input_sample)\n",
      "Nextflow operator mix with content: cram_skip_markduplicates = Channel.empty().mix(input_sample)\n",
      "Nextflow operator branch with content: input_sncr_convert = input_sample.branch{\n",
      "Nextflow operator map with content: input_sncr_convert  = input_sncr_convert.bam.map{ meta, bam, bai, table -> [ meta, bam, bai ] }\n",
      "Nextflow operator mix with content: versions = versions.mix(BAM_TO_CRAM.out.versions)\n",
      "Nextflow operator map with content: .map{ meta, cram, crai -> [ meta + [data_type: \"cram\"], cram, crai ] }\n",
      "Nextflow operator mix with content: cram_skip_splitncigar = Channel.empty().mix(sncr_cram_from_bam, input_sncr_convert.cram)\n",
      "Nextflow operator map with content: .map{ meta, cram, crai -> [ meta + [data_type: \"cram\"], cram, crai ] }\n",
      "Nextflow operator dump with content: cram_skip_splitncigar.dump(tag:\"cram_skip_splitncigar\")\n",
      "Nextflow operator branch with content: cram_for_splitncigar_status = cram_skip_splitncigar.branch{\n",
      "Nextflow operator mix with content: cram_splitncigar_no_spark = BAM_SPLITNCIGARREADS.out.cram.mix(cram_for_splitncigar_status.dna)\n",
      "Nextflow operator mix with content: versions = versions.mix(BAM_SPLITNCIGARREADS.out.versions)\n",
      "Nextflow operator mix with content: ch_sncr_cram_for_restart = Channel.empty().mix(cram_splitncigar_no_spark).mix(cram_skip_splitncigar)\n",
      "Nextflow operator map with content: .map{ meta, cram, crai -> [ meta + [data_type: \"cram\"], cram, crai ] }\n",
      "Nextflow operator branch with content: input_prepare_recal_convert = input_sample.branch{\n",
      "Nextflow operator mix with content: versions = versions.mix(BAM_TO_CRAM.out.versions)\n",
      "Nextflow operator map with content: .map{ meta, cram, crai -> [ meta + [data_type: \"cram\"], cram, crai ] }\n",
      "Nextflow operator mix with content: ch_cram_for_bam_baserecalibrator = Channel.empty().mix(ch_cram_for_bam_baserecalibrator, input_prepare_recal_convert.cram)\n",
      "Nextflow operator mix with content: ch_cram_for_bam_baserecalibrator = Channel.empty().mix(ch_sncr_cram_for_restart, cram_skip_splitncigar )\n",
      "Nextflow operator map with content: .map{ meta, cram, crai -> [ meta + [data_type: \"cram\"], cram, crai ] }\n",
      "Nextflow operator dump with content: ch_cram_for_bam_baserecalibrator.dump(tag:\"ch_cram_for_bam_baserecalibrator\")\n",
      "Nextflow operator mix with content: versions = versions.mix(BAM_BASERECALIBRATOR.out.versions)\n",
      "Nextflow operator mix with content: ch_table_bqsr = Channel.empty().mix(\n",
      "Nextflow operator collect with content: reports = reports.mix(ch_table_bqsr.collect{ meta, table -> table })\n",
      "Nextflow operator join with content: cram_applybqsr = ch_cram_for_bam_baserecalibrator.join(ch_table_bqsr, failOnDuplicate: true, failOnMismatch: true)\n",
      "Nextflow operator dump with content: cram_applybqsr = cram_applybqsr.dump(tag:\"cram_applybqsr\")\n",
      "Nextflow operator join with content: CHANNEL_BASERECALIBRATOR_CREATE_CSV(ch_sncr_cram_for_restart.join(ch_table_bqsr, failOnDuplicate: true), params.tools, params.skip_tools, params.save_output_as_bam, params.outdir)\n",
      "Nextflow operator branch with content: input_recal_convert = input_sample.branch{\n",
      "Nextflow operator map with content: input_only_table = input_recal_convert.bam.map{ meta, bam, bai, table -> [ meta, table ] }\n",
      "Nextflow operator map with content: input_only_bam   = input_recal_convert.bam.map{ meta, bam, bai, table -> [ meta, bam, bai ] }\n",
      "Nextflow operator mix with content: versions = versions.mix(BAM_TO_CRAM.out.versions)\n",
      "Nextflow operator mix with content: cram_applybqsr = Channel.empty().mix(\n",
      "Nextflow operator join with content: BAM_TO_CRAM.out.alignment_index.join(input_only_table, failOnDuplicate: true, failOnMismatch: true),\n",
      "Nextflow operator map with content: .map{ meta, cram, crai, table -> [ meta + [data_type: \"cram\"], cram, crai, table ]}\n",
      "Nextflow operator dump with content: cram_applybqsr.dump(tag:\"cram_applybqsr\")\n",
      "Nextflow operator mix with content: versions = versions.mix(BAM_APPLYBQSR.out.versions)\n",
      "Nextflow operator mix with content: cram_variant_calling = Channel.empty().mix(\n",
      "Nextflow operator collect with content: reports = reports.mix(CRAM_QC_RECAL.out.reports.collect{ meta, report -> report })\n",
      "Nextflow operator mix with content: versions = versions.mix(CRAM_QC_RECAL.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(CRAM_TO_BAM_RECAL.out.versions)\n",
      "Nextflow operator mix with content: cram_variant_calling = Channel.empty().mix(\n",
      "Nextflow operator map with content: input_recal_convert.cram.map{ meta, cram, crai, table -> [ meta, cram, crai ] })\n",
      "Nextflow operator mix with content: cram_variant_calling = Channel.empty().mix(ch_cram_for_bam_baserecalibrator)\n",
      "Nextflow operator branch with content: input_variant_calling_convert = input_sample.branch{\n",
      "Nextflow operator mix with content: versions = versions.mix(BAM_TO_CRAM.out.versions)\n",
      "Nextflow operator mix with content: cram_variant_calling = Channel.empty().mix(BAM_TO_CRAM.out.alignment_index, input_variant_calling_convert.cram)\n",
      "Nextflow operator map with content: cram_variant_calling = cram_variant_calling.map{meta, cram, crai ->\n",
      "Nextflow operator dump with content: cram_variant_calling.dump(tag:\"cram_variant_calling\")\n",
      "Error: Unclosed process block in file: ./git_repos\\rnadnavar\\subworkflows\\local\\bam_gatk_preprocessing\\main.nf\n",
      "Process BAM_MARKDUPLICATES in filepath ./git_repos\\rnadnavar\\subworkflows\\local\\bam_markduplicates\\main.nf\n",
      "Nextflow operator join with content: cram = GATK4_MARKDUPLICATES.out.cram.join(GATK4_MARKDUPLICATES.out.crai, failOnDuplicate: true, failOnMismatch: true)\n",
      "Nextflow operator mix with content: reports = reports.mix(GATK4_MARKDUPLICATES.out.metrics)\n",
      "Nextflow operator mix with content: reports = reports.mix(CRAM_QC_MOSDEPTH_SAMTOOLS.out.reports)\n",
      "Nextflow operator mix with content: versions = versions.mix(GATK4_MARKDUPLICATES.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(CRAM_QC_MOSDEPTH_SAMTOOLS.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\rnadnavar\\subworkflows\\local\\bam_markduplicates\\main.nf\n",
      "Process BAM_MERGE_INDEX_SAMTOOLS in filepath ./git_repos\\rnadnavar\\subworkflows\\local\\bam_merge_index_samtools\\main.nf\n",
      "Nextflow operator branch with content: bam_to_merge = bam.branch{ meta, bam ->\n",
      "Nextflow operator mix with content: bam_all = MERGE_BAM.out.bam.mix(bam_to_merge.single)\n",
      "Nextflow operator join with content: bam_bai = bam_all.join(INDEX_MERGE_BAM.out.bai, failOnDuplicate: true, failOnMismatch: true)\n",
      "Nextflow operator mix with content: versions = versions.mix(INDEX_MERGE_BAM.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(MERGE_BAM.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\rnadnavar\\subworkflows\\local\\bam_merge_index_samtools\\main.nf\n",
      "Process BAM_SPLITNCIGARREADS in filepath ./git_repos\\rnadnavar\\subworkflows\\local\\bam_splitncigarreads\\main.nf\n",
      "Nextflow operator combine with content: cram_intervals = cram.combine(intervals)\n",
      "Nextflow operator map with content: .map{ meta, cram, crai, intervals, num_intervals -> [ meta + [ num_intervals:num_intervals ], cram, crai, intervals ] }\n",
      "Nextflow operator map with content: dict.map{ meta, it ->  it  }\n",
      "Nextflow operator groupTuple with content: cram_to_merge = GATK4_SPLITNCIGARREADS.out.cram.map{ meta, cram -> [ groupKey(meta, meta.num_intervals), cram ] }.groupTuple()\n",
      "Nextflow operator map with content: .map{ meta, cram, crai -> [ meta - meta.subMap('num_intervals'), cram, crai ] }\n",
      "Nextflow operator mix with content: versions = versions.mix(GATK4_SPLITNCIGARREADS.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(CRAM_MERGE_INDEX_SAMTOOLS.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\rnadnavar\\subworkflows\\local\\bam_splitncigarreads\\main.nf\n",
      "Process BAM_VARIANT_CALLING in filepath ./git_repos\\rnadnavar\\subworkflows\\local\\bam_variant_calling\\main.nf\n",
      "Nextflow operator branch with content: cram_variant_calling_status = cram_variant_calling.branch{\n",
      "Nextflow operator map with content: cram_variant_calling_normal_to_cross = cram_variant_calling_status.normal.map{ meta, cram, crai -> [ meta.patient, meta, cram, crai ] }\n",
      "Nextflow operator map with content: cram_variant_calling_pair_to_cross = cram_variant_calling_status.tumor.map{ meta, cram, crai -> [ meta.patient, meta, cram, crai ] }\n",
      "Nextflow operator groupTuple with content: cram_variant_calling_tumor_grouped = cram_variant_calling_pair_to_cross.groupTuple()\n",
      "Nextflow operator join with content: cram_variant_calling_tumor_joined = cram_variant_calling_tumor_grouped.join(cram_variant_calling_normal_to_cross, failOnDuplicate: true, remainder: true)\n",
      "Nextflow operator filter with content: cram_variant_calling_tumor_filtered = cram_variant_calling_tumor_joined.filter{ it ->  !(it.last()) }\n",
      "Nextflow operator map with content: cram_variant_calling_tumor_only = cram_variant_calling_tumor_filtered.transpose().map{ it -> [it[1], it[2], it[3]] }\n",
      "Nextflow operator join with content: cram_variant_calling_normal_joined = cram_variant_calling_normal_to_cross.join(cram_variant_calling_tumor_grouped, failOnDuplicate: true, remainder: true)\n",
      "Nextflow operator filter with content: cram_variant_calling_normal_filtered = cram_variant_calling_normal_joined.filter{ it ->  !(it.last()) }\n",
      "Nextflow operator map with content: cram_variant_calling_status_normal = cram_variant_calling_normal_filtered.map{ it -> [it[1], it[2], it[3]] }\n",
      "Nextflow operator cross with content: cram_variant_calling_pair = cram_variant_calling_normal_to_cross.cross(cram_variant_calling_pair_to_cross)\n",
      "Nextflow operator map with content: .map { normal, tumor ->\n",
      "Nextflow operator dump with content: cram_variant_calling_pair.dump(tag:\"cram_variant_calling_pair\")\n",
      "Nextflow operator mix with content: vcf_to_normalise            = Channel.empty().mix(BAM_VARIANT_CALLING_SOMATIC.out.vcf_all)\n",
      "Nextflow operator mix with content: contamination_table_mutect2 = Channel.empty().mix(BAM_VARIANT_CALLING_SOMATIC.out.contamination_table_mutect2)\n",
      "Nextflow operator mix with content: segmentation_table_mutect2  = Channel.empty().mix(BAM_VARIANT_CALLING_SOMATIC.out.segmentation_table_mutect2)\n",
      "Nextflow operator mix with content: artifact_priors_mutect2     = Channel.empty().mix(BAM_VARIANT_CALLING_SOMATIC.out.artifact_priors_mutect2)\n",
      "Nextflow operator mix with content: versions = versions.mix(BAM_VARIANT_CALLING_SOMATIC.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(VCF_QC_BCFTOOLS_VCFTOOLS.out.versions)\n",
      "Nextflow operator collect with content: reports = reports.mix(VCF_QC_BCFTOOLS_VCFTOOLS.out.bcftools_stats.collect{ meta, stats -> stats })\n",
      "Nextflow operator collect with content: reports = reports.mix(VCF_QC_BCFTOOLS_VCFTOOLS.out.vcftools_tstv_counts.collect{ meta, counts -> counts })\n",
      "Nextflow operator collect with content: reports = reports.mix(VCF_QC_BCFTOOLS_VCFTOOLS.out.vcftools_tstv_qual.collect{ meta, qual -> qual })\n",
      "Nextflow operator collect with content: reports = reports.mix(VCF_QC_BCFTOOLS_VCFTOOLS.out.vcftools_filter_summary.collect{ meta, summary -> summary })\n",
      "Error: Unclosed process block in file: ./git_repos\\rnadnavar\\subworkflows\\local\\bam_variant_calling\\main.nf\n",
      "Process BAM_VARIANT_CALLING_FREEBAYES in filepath ./git_repos\\rnadnavar\\subworkflows\\local\\bam_variant_calling_freebayes\\main.nf\n",
      "Nextflow operator combine with content: cram_intervals = cram.combine(intervals)\n",
      "Nextflow operator map with content: .map{ meta, cram1, crai1, cram2, crai2, intervals, num_intervals -> [ meta + [ num_intervals:num_intervals ], cram1, crai1, cram2, crai2, intervals ]}\n",
      "Nextflow operator branch with content: bcftools_vcf_out = BCFTOOLS_SORT.out.vcf.branch{\n",
      "Nextflow operator groupTuple with content: vcf_to_merge = bcftools_vcf_out.intervals.map{ meta, vcf -> [ groupKey(meta, meta.num_intervals), vcf ]}.groupTuple()\n",
      "Nextflow operator mix with content: vcf = MERGE_FREEBAYES.out.vcf.mix(bcftools_vcf_out.no_intervals)\n",
      "Nextflow operator map with content: .map{ meta, vcf -> [ meta - meta.subMap('num_intervals') + [ variantcaller:'freebayes' ], vcf ] }\n",
      "Nextflow operator mix with content: versions = versions.mix(BCFTOOLS_SORT.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(MERGE_FREEBAYES.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(FREEBAYES.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(TABIX_VC_FREEBAYES.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\rnadnavar\\subworkflows\\local\\bam_variant_calling_freebayes\\main.nf\n",
      "Process BAM_VARIANT_CALLING_PRE_POST_PROCESSING in filepath ./git_repos\\rnadnavar\\subworkflows\\local\\bam_variant_calling_pre_post_processing\\main.nf\n",
      "Nextflow operator mix with content: versions             = versions.mix(BAM_GATK_PREPROCESSING.out.versions)\n",
      "Nextflow operator mix with content: reports              = reports.mix(BAM_GATK_PREPROCESSING.out.reports)\n",
      "Nextflow operator mix with content: versions                      = versions.mix(BAM_VARIANT_CALLING.out.versions)\n",
      "Nextflow operator mix with content: reports                       = reports.mix(BAM_VARIANT_CALLING.out.reports)\n",
      "Nextflow operator map with content: fasta.map{ it -> [ [ id:'fasta' ], it ] },\n",
      "Nextflow operator mix with content: versions                      = versions.mix(VCF_NORMALISE.out.versions)\n",
      "Nextflow operator dump with content: vcf_to_annotate.dump(tag:\"vcf_to_annotate\")\n",
      "Nextflow operator map with content: vcf_to_annotate.map{meta, vcf -> [ meta + [ file_name: vcf.baseName ], vcf ] },\n",
      "Nextflow operator mix with content: versions                      = versions.mix(VCF_ANNOTATE.out.versions)\n",
      "Nextflow operator mix with content: reports                       = reports.mix(VCF_ANNOTATE.out.reports)\n",
      "Nextflow operator dump with content: vcf_to_consensus.dump(tag:\"vcf_to_consensus0\")\n",
      "Nextflow operator mix with content: versions                      = versions.mix(VCF_CONSENSUS.out.versions)\n",
      "Nextflow operator dump with content: maf_to_filter.dump(tag:\"maf_to_filter0\")\n",
      "Nextflow operator mix with content: versions     = versions.mix(MAF_FILTERING.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\rnadnavar\\subworkflows\\local\\bam_variant_calling_pre_post_processing\\main.nf\n",
      "Process BAM_VARIANT_CALLING_SOMATIC in filepath ./git_repos\\rnadnavar\\subworkflows\\local\\bam_variant_calling_somatic\\main.nf\n",
      "Nextflow operator dump with content: cram.dump(tag:\"sage_cram\")\n",
      "Nextflow operator map with content: fasta.map{ it -> [ [ id:'fasta' ], it ] },\n",
      "Nextflow operator map with content: fasta_fai.map{ it -> [ [ id:'fasta_fai' ], it ] },\n",
      "Nextflow operator mix with content: versions   = versions.mix(BAM_VARIANT_CALLING_SOMATIC_SAGE.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(BAM_VARIANT_CALLING_SOMATIC_MANTA.out.versions)\n",
      "Nextflow operator join with content: cram.join(BAM_VARIANT_CALLING_SOMATIC_MANTA.out.candidate_small_indels_vcf, failOnDuplicate: true, failOnMismatch: true).join(BAM_VARIANT_CALLING_SOMATIC_MANTA.out.candidate_small_indels_vcf_tbi, failOnDuplicate: true, failOnMismatch: true) :\n",
      "Nextflow operator map with content: cram.map{ meta, normal_cram, normal_crai, tumor_cram, tumor_crai -> [ meta, normal_cram, normal_crai, tumor_cram, tumor_crai, [], [] ] }\n",
      "Nextflow operator mix with content: vcf_strelka = Channel.empty().mix(BAM_VARIANT_CALLING_SOMATIC_STRELKA.out.vcf)\n",
      "Nextflow operator mix with content: versions = versions.mix(BAM_VARIANT_CALLING_SOMATIC_STRELKA.out.versions)\n",
      "Nextflow operator map with content: mutect_cram = cram.map { meta, normal_cram, normal_crai, tumor_cram, tumor_crai -> [ meta, [ normal_cram, tumor_cram ], [ normal_crai, tumor_crai ] ] }\n",
      "Nextflow operator map with content: fasta.map{ it -> [ [ id:'fasta' ], it ] },\n",
      "Nextflow operator map with content: fasta_fai.map{ it -> [ [ id:'fasta_fai' ], it ] },\n",
      "Nextflow operator mix with content: versions                    = versions.mix(BAM_VARIANT_CALLING_SOMATIC_MUTECT2.out.versions)\n",
      "Nextflow operator mix with content: vcf_all = Channel.empty().mix(\n",
      "Error: Unclosed process block in file: ./git_repos\\rnadnavar\\subworkflows\\local\\bam_variant_calling_somatic\\main.nf\n",
      "Process BAM_VARIANT_CALLING_SOMATIC_MANTA in filepath ./git_repos\\rnadnavar\\subworkflows\\local\\bam_variant_calling_somatic_manta\\main.nf\n",
      "Nextflow operator combine with content: cram_intervals = cram.combine(intervals).map{ it ->\n",
      "Nextflow operator map with content: vcf = Channel.empty().mix(diploid_sv_vcf, somatic_sv_vcf).map{ meta, vcf -> [ meta + [ variantcaller:'manta' ], vcf ] }\n",
      "Nextflow operator mix with content: versions = versions.mix(MANTA_SOMATIC.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\rnadnavar\\subworkflows\\local\\bam_variant_calling_somatic_manta\\main.nf\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Process BAM_VARIANT_CALLING_SOMATIC_MUTECT2 in filepath ./git_repos\\rnadnavar\\subworkflows\\local\\bam_variant_calling_somatic_mutect2\\main.nf\n",
      "Nextflow operator combine with content: input_intervals = input.combine(intervals)\n",
      "Nextflow operator map with content: .map{ meta, input_list, input_index_list, intervals, num_intervals -> [ meta + [ num_intervals:num_intervals ], input_list, input_index_list, intervals ] }\n",
      "Nextflow operator map with content: ch_normal_cram = input.map{ meta, cram, crai -> [ meta - meta.subMap('tumor_id') + [id:meta.patient], cram[0], crai[0] ] }.unique()\n",
      "Nextflow operator map with content: ch_tumor_cram = input.map{ meta, cram, crai -> [ meta - meta.subMap('tumor_id') + [id:meta.patient], cram[1], crai[1] ] }\n",
      "Nextflow operator groupTuple with content: ch_tn_cram = ch_normal_cram.mix(ch_tumor_cram).groupTuple()\n",
      "Nextflow operator combine with content: ch_tn_intervals = ch_tn_cram.combine(intervals)\n",
      "Nextflow operator map with content: .map{ meta, cram, crai, intervals, num_intervals -> [ meta + [ num_intervals:num_intervals ], cram, crai, intervals ] }\n",
      "Nextflow operator branch with content: vcf_branch = MUTECT2_PAIRED.out.vcf.branch{\n",
      "Nextflow operator branch with content: tbi_branch = MUTECT2_PAIRED.out.tbi.branch{\n",
      "Nextflow operator branch with content: stats_branch = MUTECT2_PAIRED.out.stats.branch{\n",
      "Nextflow operator branch with content: f1r2_branch = MUTECT2_PAIRED.out.f1r2.branch{\n",
      "Nextflow operator groupTuple with content: vcf_to_merge = vcf_branch.intervals.map{ meta, vcf -> [ groupKey(meta, meta.num_intervals), vcf ] }.groupTuple()\n",
      "Nextflow operator groupTuple with content: stats_to_merge = stats_branch.intervals.map{ meta, stats -> [ groupKey(meta, meta.num_intervals), stats ] }.groupTuple()\n",
      "Nextflow operator groupTuple with content: f1r2_to_merge = f1r2_branch.intervals.map{ meta, f1r2 -> [ groupKey(meta, meta.num_intervals), f1r2 ] }.groupTuple()\n",
      "Nextflow operator map with content: vcf = Channel.empty().mix(MERGE_MUTECT2.out.vcf, vcf_branch.no_intervals).map{ meta, vcf -> [ meta - meta.subMap('normal_id', 'tumor_id', 'num_intervals'), vcf ]}\n",
      "Nextflow operator map with content: tbi = Channel.empty().mix(MERGE_MUTECT2.out.tbi, tbi_branch.no_intervals).map{ meta, tbi -> [ meta - meta.subMap('normal_id', 'tumor_id', 'num_intervals'), tbi ]}\n",
      "Nextflow operator map with content: stats = Channel.empty().mix(MERGEMUTECTSTATS.out.stats, stats_branch.no_intervals).map{ meta, stats -> [ meta - meta.subMap('normal_id', 'tumor_id', 'num_intervals'), stats ]}\n",
      "Nextflow operator map with content: f1r2 = Channel.empty().mix(f1r2_to_merge, f1r2_branch.no_intervals).map{ meta, f1r2 -> [ meta - meta.subMap('normal_id', 'tumor_id', 'num_intervals'), f1r2 ]}\n",
      "Nextflow operator multiMap with content: pileup = input_intervals.multiMap{  meta, input_list, input_index_list, intervals ->\n",
      "Nextflow operator map with content: pileup_normal = pileup.normal.map{ meta, cram, crai, intervals -> [ meta - meta.subMap('tumor_id') + [ id:meta.normal_id ], cram, crai, intervals] }.unique()\n",
      "Nextflow operator map with content: pileup_tumor = pileup.tumor.map{ meta, cram, crai, intervals -> [ meta + [ id:meta.tumor_id ], cram, crai, intervals ] }\n",
      "Nextflow operator branch with content: pileup_table_normal_branch = GETPILEUPSUMMARIES_NORMAL.out.table.branch{\n",
      "Nextflow operator branch with content: pileup_table_tumor_branch = GETPILEUPSUMMARIES_TUMOR.out.table.branch{\n",
      "Nextflow operator groupTuple with content: pileup_table_normal_to_merge = pileup_table_normal_branch.intervals.map{ meta, table -> [ groupKey(meta, meta.num_intervals), table ] }.groupTuple()\n",
      "Nextflow operator groupTuple with content: pileup_table_tumor_to_merge = pileup_table_tumor_branch.intervals.map{ meta, table -> [ groupKey(meta, meta.num_intervals), table ] }.groupTuple()\n",
      "Nextflow operator map with content: GATHERPILEUPSUMMARIES_NORMAL(pileup_table_normal_to_merge, dict.map{ meta, dict -> [ dict ] })\n",
      "Nextflow operator map with content: GATHERPILEUPSUMMARIES_TUMOR(pileup_table_tumor_to_merge, dict.map{ meta, dict -> [ dict ] })\n",
      "Nextflow operator map with content: pileup_table_tumor = Channel.empty().mix(GATHERPILEUPSUMMARIES_TUMOR.out.table, pileup_table_tumor_branch.no_intervals).map{meta, table -> [ meta - meta.subMap('normal_id', 'tumor_id', 'num_intervals') + [id:meta.patient], meta.id, table ] }\n",
      "Nextflow operator map with content: pileup_table_normal= Channel.empty().mix(GATHERPILEUPSUMMARIES_NORMAL.out.table, pileup_table_normal_branch.no_intervals).map{meta, table -> [ meta - meta.subMap('normal_id', 'tumor_id', 'num_intervals') + [id:meta.patient], meta.id, table ] }\n",
      "Nextflow operator combine with content: ch_calculatecontamination_in_tables = pileup_table_tumor.combine(\n",
      "Nextflow operator map with content: pileup_table_normal, by:0).map{\n",
      "Nextflow operator groupTuple with content: ch_seg_to_filtermutectcalls = CALCULATECONTAMINATION.out.segmentation.map{ meta, seg -> [ meta - meta.subMap('tumor_id') + [id: meta.patient], seg]}.groupTuple()\n",
      "Nextflow operator groupTuple with content: ch_cont_to_filtermutectcalls = CALCULATECONTAMINATION.out.contamination.map{ meta, cont -> [ meta - meta.subMap('tumor_id') + [id: meta.patient], cont]}.groupTuple()\n",
      "Nextflow operator join with content: vcf_to_filter = vcf.join(tbi, failOnDuplicate: true, failOnMismatch: true)\n",
      "Nextflow operator join with content: .join(stats, failOnDuplicate: true, failOnMismatch: true)\n",
      "Nextflow operator join with content: .join(LEARNREADORIENTATIONMODEL.out.artifactprior, failOnDuplicate: true, failOnMismatch: true)\n",
      "Nextflow operator join with content: .join(ch_seg_to_filtermutectcalls)\n",
      "Nextflow operator join with content: .join(ch_cont_to_filtermutectcalls)\n",
      "Nextflow operator map with content: .map{ meta, vcf, tbi, stats, orientation, seg, cont -> [ meta, vcf, tbi, stats, orientation, seg, cont, [] ] }\n",
      "Nextflow operator mix with content: versions = versions.mix(CALCULATECONTAMINATION.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(GETPILEUPSUMMARIES_NORMAL.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(GETPILEUPSUMMARIES_TUMOR.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(GATHERPILEUPSUMMARIES_NORMAL.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(GATHERPILEUPSUMMARIES_TUMOR.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(LEARNREADORIENTATIONMODEL.out.versions)\n",
      "Nextflow operator join with content: vcf_to_filter = vcf.join(tbi, failOnDuplicate: true, failOnMismatch: true)\n",
      "Nextflow operator join with content: .join(stats, failOnDuplicate: true, failOnMismatch: true)\n",
      "Nextflow operator map with content: .map{ meta, vcf, tbi, stats -> [ meta, vcf, tbi, stats, [], [], [], [] ] }\n",
      "Nextflow operator map with content: .map{ meta, vcf -> [ meta + [ variantcaller:'mutect2' ], vcf ] }\n",
      "Nextflow operator mix with content: versions = versions.mix(MERGE_MUTECT2.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(FILTERMUTECTCALLS.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(MERGEMUTECTSTATS.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(MUTECT2_PAIRED.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\rnadnavar\\subworkflows\\local\\bam_variant_calling_somatic_mutect2\\main.nf\n",
      "Process BAM_VARIANT_CALLING_SOMATIC_SAGE in filepath ./git_repos\\rnadnavar\\subworkflows\\local\\bam_variant_calling_somatic_sage\\main.nf\n",
      "Nextflow operator combine with content: cram_intervals = cram.combine(intervals)\n",
      "Nextflow operator map with content: .map{ meta, cram1, crai1, cram2, crai2, intervals, num_intervals -> [ meta + [ num_intervals:num_intervals ], cram1, crai1, cram2, crai2, intervals ]}\n",
      "Nextflow operator branch with content: bcftools_vcf_out = BCFTOOLS_SORT.out.vcf.branch{\n",
      "Nextflow operator groupTuple with content: vcf_to_merge = bcftools_vcf_out.intervals.map{ meta, vcf -> [ groupKey(meta, meta.num_intervals), vcf ]}.groupTuple()\n",
      "Nextflow operator mix with content: vcf = MERGE_SAGE.out.vcf.mix(bcftools_vcf_out.no_intervals)\n",
      "Nextflow operator map with content: .map{ meta, vcf -> [ meta - meta.subMap('normal_id', 'tumor_id','num_intervals') + [ variantcaller:'sage' ], vcf ] }\n",
      "Nextflow operator mix with content: versions = versions.mix(BCFTOOLS_SORT.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(MERGE_SAGE.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(SAGE.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(TABIX_VC_SAGE.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\rnadnavar\\subworkflows\\local\\bam_variant_calling_somatic_sage\\main.nf\n",
      "Process BAM_VARIANT_CALLING_SOMATIC_STRELKA in filepath ./git_repos\\rnadnavar\\subworkflows\\local\\bam_variant_calling_somatic_strelka\\main.nf\n",
      "Nextflow operator combine with content: cram_intervals = cram.combine(intervals)\n",
      "Nextflow operator map with content: .map{ meta, normal_cram, normal_crai, tumor_cram, tumor_crai, manta_vcf, manta_tbi, intervals_gz_tbi, num_intervals -> [ meta + [ num_intervals:0 ], normal_cram, normal_crai, tumor_cram, tumor_crai, manta_vcf, manta_tbi, [], [] ] }\n",
      "Nextflow operator combine with content: cram_intervals = cram.combine(intervals)\n",
      "Nextflow operator map with content: .map{ meta, normal_cram, normal_crai, tumor_cram, tumor_crai, manta_vcf, manta_tbi, intervals_gz_tbi, num_intervals -> [ meta + [ num_intervals:num_intervals ], normal_cram, normal_crai, tumor_cram, tumor_crai, manta_vcf, manta_tbi, intervals_gz_tbi[0], intervals_gz_tbi[1] ] }\n",
      "Nextflow operator branch with content: vcf_indels = STRELKA_SOMATIC.out.vcf_indels.branch{\n",
      "Nextflow operator branch with content: vcf_snvs = STRELKA_SOMATIC.out.vcf_snvs.branch{\n",
      "Nextflow operator mix with content: .mix(STRELKA_SOMATIC.out.vcf_snvs)\n",
      "Nextflow operator map with content: .map{ meta, vcf -> [ groupKey(meta, meta.num_intervals * 2), vcf ]}\n",
      "Nextflow operator groupTuple with content: .groupTuple()\n",
      "Nextflow operator dump with content: vcfs_to_merge.dump(tag:\"vcfs_to_merge\")\n",
      "Nextflow operator map with content: .map{ meta, vcf ->\n",
      "Nextflow operator mix with content: versions = versions.mix(MERGE_STRELKA.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(STRELKA_SOMATIC.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\rnadnavar\\subworkflows\\local\\bam_variant_calling_somatic_strelka\\main.nf\n",
      "Process CHANNEL_ALIGN_CREATE_CSV in filepath ./git_repos\\rnadnavar\\subworkflows\\local\\channel_align_create_csv\\main.nf\n",
      "Nextflow operator collectFile with content: bam_indexed.collectFile(keepHeader: true, skip: 1, sort: true, storeDir: \"${params.outdir}/csv\") { meta, bam, bai ->\n",
      "Process CHANNEL_APPLYBQSR_CREATE_CSV in filepath ./git_repos\\rnadnavar\\subworkflows\\local\\channel_applybqsr_create_csv\\main.nf\n",
      "Nextflow operator collectFile with content: cram_recalibrated_index.collectFile(keepHeader: true, skip: 1, sort: true, storeDir: \"${params.outdir}/csv\") { meta, file, index ->\n",
      "Process CHANNEL_BASERECALIBRATOR_CREATE_CSV in filepath ./git_repos\\rnadnavar\\subworkflows\\local\\channel_baserecalibrator_create_csv\\main.nf\n",
      "Nextflow operator collectFile with content: cram_table_bqsr.collectFile(keepHeader: true, skip: 1, sort: true, storeDir: \"${outdir}/csv\") { meta, cram, crai, table ->\n",
      "Nextflow operator collectFile with content: cram_table_bqsr.collectFile(keepHeader: true, skip: 1, sort: true, storeDir: \"${outdir}/csv\") { meta, cram, crai, table ->\n",
      "Process CHANNEL_CONSENSUS_CREATE_CSV in filepath ./git_repos\\rnadnavar\\subworkflows\\local\\channel_consensus_create_csv\\main.nf\n",
      "Nextflow operator collectFile with content: maf_to_csv.collectFile(keepHeader: true, skip: 1,sort: true, storeDir: \"${params.outdir}/csv\"){ meta, maf, variantcaller ->\n",
      "Process CHANNEL_MARKDUPLICATES_CREATE_CSV in filepath ./git_repos\\rnadnavar\\subworkflows\\local\\channel_markduplicates_create_csv\\main.nf\n",
      "Nextflow operator collectFile with content: cram_markduplicates.collectFile(keepHeader: true, skip: 1, sort: true, storeDir: \"${outdir}/csv\") { meta, file, index ->\n",
      "Process CHANNEL_SPLITNCIGARREADS_CREATE_CSV in filepath ./git_repos\\rnadnavar\\subworkflows\\local\\channel_splitncigarreads_create_csv\\main.nf\n",
      "Nextflow operator collectFile with content: cram_splitncigarreads.collectFile(keepHeader: true, skip: 1, sort: true, storeDir: \"${outdir}/csv\") { meta, file, index ->\n",
      "Process CHANNEL_VARIANT_CALLING_CREATE_CSV in filepath ./git_repos\\rnadnavar\\subworkflows\\local\\channel_variant_calling_create_csv\\main.nf\n",
      "Nextflow operator collectFile with content: vcf_to_csv.collectFile(keepHeader: true, skip: 1,sort: true, storeDir: \"${params.outdir}/csv\"){ meta, vcf ->\n",
      "Process CRAM_MERGE_INDEX_SAMTOOLS in filepath ./git_repos\\rnadnavar\\subworkflows\\local\\cram_merge_index_samtools\\main.nf\n",
      "Nextflow operator branch with content: cram_to_merge = cram.branch{ meta, cram ->\n",
      "Nextflow operator map with content: MERGE_CRAM(cram_to_merge.multiple, fasta.map{ it -> [ [ id:'fasta' ], it ] }, fasta_fai.map{ it -> [ [ id:'fasta_fai' ], it ] })\n",
      "Nextflow operator mix with content: cram_all = MERGE_CRAM.out.cram.mix(cram_to_merge.single)\n",
      "Nextflow operator join with content: cram_crai = cram_all.join(INDEX_CRAM.out.crai, failOnDuplicate: true, failOnMismatch: true)\n",
      "Nextflow operator first with content: versions = versions.mix(INDEX_CRAM.out.versions.first())\n",
      "Nextflow operator first with content: versions = versions.mix(MERGE_CRAM.out.versions.first())\n",
      "Error: Unclosed process block in file: ./git_repos\\rnadnavar\\subworkflows\\local\\cram_merge_index_samtools\\main.nf\n",
      "Process CRAM_QC_MOSDEPTH_SAMTOOLS in filepath ./git_repos\\rnadnavar\\subworkflows\\local\\cram_qc_mosdepth_samtools\\main.nf\n",
      "Nextflow operator map with content: SAMTOOLS_STATS(cram, fasta.map{ it -> [ [ id:'fasta' ], it ] })\n",
      "Nextflow operator combine with content: MOSDEPTH(cram.combine(intervals.map{ meta, bed -> [ bed?:[] ] }), fasta.map{ it -> [ [ id:'fasta' ], it ] })\n",
      "Nextflow operator mix with content: reports = reports.mix(SAMTOOLS_STATS.out.stats)\n",
      "Nextflow operator mix with content: reports = reports.mix(MOSDEPTH.out.global_txt)\n",
      "Nextflow operator mix with content: reports = reports.mix(MOSDEPTH.out.regions_txt)\n",
      "Nextflow operator mix with content: versions = versions.mix(MOSDEPTH.out.versions)\n",
      "Nextflow operator first with content: versions = versions.mix(SAMTOOLS_STATS.out.versions.first())\n",
      "Error: Unclosed process block in file: ./git_repos\\rnadnavar\\subworkflows\\local\\cram_qc_mosdepth_samtools\\main.nf\n",
      "Process FASTQ_ALIGN_BWAMEM_MEM2_DRAGMAP in filepath ./git_repos\\rnadnavar\\subworkflows\\local\\fastq_align_bwamem_mem2_dragmap\\main.nf\n",
      "Nextflow operator map with content: BWAMEM1_MEM(reads, index.map{ it -> [ [ id:'index' ], it ] }, sort) // If aligner is bwa-mem\n",
      "Nextflow operator map with content: BWAMEM2_MEM(reads, index.map{ it -> [ [ id:'index' ], it ] }, sort) // If aligner is bwa-mem2\n",
      "Nextflow operator map with content: DRAGMAP_ALIGN(reads, index.map{ it -> [ [ id:'index' ], it ] }, sort) // If aligner is dragmap\n",
      "Nextflow operator mix with content: bam = bam.mix(BWAMEM1_MEM.out.bam)\n",
      "Nextflow operator mix with content: bam = bam.mix(BWAMEM2_MEM.out.bam)\n",
      "Nextflow operator mix with content: bam = bam.mix(DRAGMAP_ALIGN.out.bam)\n",
      "Nextflow operator mix with content: reports = reports.mix(DRAGMAP_ALIGN.out.log)\n",
      "Nextflow operator mix with content: versions = versions.mix(BWAMEM1_MEM.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(BWAMEM2_MEM.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(DRAGMAP_ALIGN.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\rnadnavar\\subworkflows\\local\\fastq_align_bwamem_mem2_dragmap\\main.nf\n",
      "Process MAF_FILTERING in filepath ./git_repos\\rnadnavar\\subworkflows\\local\\maf_filtering\\main.nf\n",
      "Nextflow operator mix with content: versions = versions.mix(FILTERING.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\rnadnavar\\subworkflows\\local\\maf_filtering\\main.nf\n",
      "Process MAF_FILTERING_RNA in filepath ./git_repos\\rnadnavar\\subworkflows\\local\\maf_rna_filtering\\main.nf\n",
      "Nextflow operator dump with content: maf_to_filter.dump(tag:\"maf_to_filter\")\n",
      "Nextflow operator dump with content: maf_to_filter_realigned.dump(tag:\"maf_to_filter_realigned\")\n",
      "Nextflow operator map with content: .map{meta, maf -> [meta.patient, meta, maf]}\n",
      "Nextflow operator dump with content: maf_to_cross_first_pass.dump(tag:\"[STEP9: FILTERING] maf_to_cross_first_pass\")\n",
      "Nextflow operator map with content: .map{meta, maf -> [meta.patient, meta, maf]}\n",
      "Nextflow operator dump with content: maf_to_cross_second_pass.dump(tag:\"[STEP9: FILTERING] maf_to_cross_second_pass\")\n",
      "Nextflow operator cross with content: .cross(maf_to_cross_second_pass).dump(tag:\"[STEP9: FILTERING] maf_to_cross_crossed_pass\")\n",
      "Nextflow operator cross with content: .cross(maf_to_cross_second_pass)\n",
      "Nextflow operator map with content: .map{first, second ->\n",
      "Nextflow operator dump with content: maf_to_filter.dump(tag:\"maf_to_filter2\")\n",
      "Nextflow operator map with content: maf_crossed = maf_to_filter.map{meta, maf -> [meta, maf, []]}\n",
      "Nextflow operator dump with content: maf_crossed.dump(tag:\"[STEP9: FILTERING] maf_crossed\")\n",
      "Nextflow operator mix with content: versions = versions.mix(RNA_FILTERING.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\rnadnavar\\subworkflows\\local\\maf_rna_filtering\\main.nf\n",
      "Process POST_VARIANTCALLING in filepath ./git_repos\\rnadnavar\\subworkflows\\local\\post_variant_calling\\main.nf\n",
      "Nextflow operator mix with content: vcfs = vcfs.mix(CONCATENATE_GERMLINE_VCFS.out.vcfs)\n",
      "Nextflow operator mix with content: versions = versions.mix(CONCATENATE_GERMLINE_VCFS.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\rnadnavar\\subworkflows\\local\\post_variant_calling\\main.nf\n",
      "Process PREPARE_GENOME in filepath ./git_repos\\rnadnavar\\subworkflows\\local\\prepare_genome\\main.nf\n",
      "Nextflow operator map with content: fasta    = fasta.map{ fasta -> [ [ id:fasta.baseName ], fasta ] }\n",
      "Nextflow operator collect with content: bwa         = BWAMEM1_INDEX.out.index.map{ meta, index -> [index] }.collect()       // path: bwa/*\n",
      "Nextflow operator collect with content: bwamem2     = BWAMEM2_INDEX.out.index.map{ meta, index -> [index] }.collect()       // path: bwamem2/*\n",
      "Nextflow operator collect with content: hashtable   = DRAGMAP_HASHTABLE.out.hashmap.map{ meta, index -> [index] }.collect() // path: dragmap/*\n",
      "Nextflow operator mix with content: versions = versions.mix(BWAMEM1_INDEX.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(BWAMEM2_INDEX.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(DRAGMAP_HASHTABLE.out.versions)\n",
      "Nextflow operator flatten with content: TABIX_DBSNP(dbsnp.flatten().map{ it -> [ [ id:it.baseName ], it ] })\n",
      "Nextflow operator flatten with content: TABIX_GERMLINE_RESOURCE(germline_resource.flatten().map{ it -> [ [ id:it.baseName ], it ] })\n",
      "Nextflow operator flatten with content: TABIX_KNOWN_SNPS(known_snps.flatten().map{ it -> [ [ id:it.baseName ], it ] } )\n",
      "Nextflow operator flatten with content: TABIX_KNOWN_INDELS(known_indels.flatten().map{ it -> [ [ id:it.baseName ], it ] } )\n",
      "Nextflow operator flatten with content: TABIX_PON(pon.flatten().map{ it -> [ [ id:it.baseName ], it ] })\n",
      "Nextflow operator map with content: Channel.fromPath(params.gtf).map{ it -> [[id:it[0].baseName], it] }\n",
      "Nextflow operator collect with content: ch_gtf = GUNZIP_GTF.out.gunzip.map{ meta, gtf -> [gtf] }.collect()\n",
      "Nextflow operator mix with content: versions = versions.mix(GUNZIP_GTF.out.versions)\n",
      "Nextflow operator collect with content: ch_gtf = Channel.fromPath(params.gtf).collect()\n",
      "Nextflow operator map with content: Channel.fromPath(params.gff).map{ it -> [[id:it[0].baseName], it] }\n",
      "Nextflow operator collect with content: ch_gff = GUNZIP_GFF.out.gunzip.map{ meta, gff -> [gff] }.collect()\n",
      "Nextflow operator mix with content: versions = versions.mix(GUNZIP_GFF.out.versions)\n",
      "Nextflow operator collect with content: ch_gff = Channel.fromPath(params.gff).collect()\n",
      "Nextflow operator set with content: .set { ch_gtf }\n",
      "Nextflow operator mix with content: versions = versions.mix(GFFREAD.out.versions)\n",
      "Nextflow operator map with content: Channel.fromPath(params.exon_bed).map{ it -> [[id:it[0].baseName], it] }\n",
      "Nextflow operator collect with content: ch_gene_bed = GUNZIP_GENE_BED.out.gunzip.map{ meta, bed -> [bed] }.collect()\n",
      "Nextflow operator mix with content: versions = versions.mix(GUNZIP_GENE_BED.out.versions)\n",
      "Nextflow operator collect with content: ch_exon_bed = Channel.fromPath(params.exon_bed).collect()\n",
      "Nextflow operator collect with content: ch_exon_bed = GTF2BED ( ch_gtf ).bed.collect()\n",
      "Nextflow operator mix with content: versions = versions.mix(GTF2BED.out.versions)\n",
      "Nextflow operator map with content: Channel.fromPath(params.star_index).map{ it -> [[id:it[0].baseName], it] }\n",
      "Nextflow operator collect with content: ch_star_index = UNTAR_STAR_INDEX.out.untar.map{ meta, star_index -> [star_index] }.collect()\n",
      "Nextflow operator mix with content: versions   = versions.mix(UNTAR_STAR_INDEX.out.versions)\n",
      "Nextflow operator collect with content: ch_star_index = Channel.fromPath(params.star_index).collect()\n",
      "Nextflow operator map with content: STAR_GENOMEGENERATE ( fasta.map{meta, fasta -> fasta},ch_gtf )\n",
      "Nextflow operator mix with content: versions      = versions.mix(STAR_GENOMEGENERATE.out.versions)\n",
      "Nextflow operator collect with content: ch_splicesites  = Channel.fromPath(params.splicesites).collect().map{ it -> [ [ id:'null' ], it ]}\n",
      "Nextflow operator map with content: HISAT2_EXTRACTSPLICESITES ( ch_gtf.map{ it -> [ [ id:'null' ], it ]} )\n",
      "Nextflow operator mix with content: versions = versions.mix(HISAT2_EXTRACTSPLICESITES.out.versions)\n",
      "Nextflow operator collect with content: ch_hisat2_index  = Channel.fromPath(params.hisat2_index).collect().map{it -> [ [ id:\"hisat2_index\" ], it ]}\n",
      "Nextflow operator map with content: ch_gtf.map{ it -> [ [ id:'null' ], it ]},\n",
      "Nextflow operator mix with content: versions = versions.mix(HISAT2_BUILD.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(SAMTOOLS_FAIDX.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(GATK4_CREATESEQUENCEDICTIONARY.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(TABIX_DBSNP.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(TABIX_GERMLINE_RESOURCE.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(TABIX_KNOWN_SNPS.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(TABIX_KNOWN_INDELS.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(TABIX_PON.out.versions)\n",
      "Nextflow operator collect with content: dbsnp_tbi             = TABIX_DBSNP.out.tbi.map{ meta, tbi -> [tbi] }.collect()               // path: dbsnb.vcf.gz.tbi\n",
      "Nextflow operator map with content: fasta_fai             = SAMTOOLS_FAIDX.out.fai.map{ meta, fai -> [fai] }                      // path: genome.fasta.fai\n",
      "Nextflow operator collect with content: germline_resource_tbi = TABIX_GERMLINE_RESOURCE.out.tbi.map{ meta, tbi -> [tbi] }.collect()   // path: germline_resource.vcf.gz.tbi\n",
      "Nextflow operator collect with content: known_snps_tbi        = TABIX_KNOWN_SNPS.out.tbi.map{ meta, tbi -> [tbi] }.collect()          // path: {known_indels*}.vcf.gz.tbi\n",
      "Nextflow operator collect with content: known_indels_tbi      = TABIX_KNOWN_INDELS.out.tbi.map{ meta, tbi -> [tbi] }.collect()        // path: {known_indels*}.vcf.gz.tbi\n",
      "Nextflow operator collect with content: pon_tbi               = TABIX_PON.out.tbi.map{ meta, tbi -> [tbi] }.collect()                 // path: pon.vcf.gz.tbi\n",
      "Error: Unclosed process block in file: ./git_repos\\rnadnavar\\subworkflows\\local\\prepare_genome\\main.nf\n",
      "Process PREPARE_INTERVALS in filepath ./git_repos\\rnadnavar\\subworkflows\\local\\prepare_intervals\\main.nf\n",
      "Nextflow operator map with content: BUILD_INTERVALS(fasta_fai.map{it -> [ [ id:it.baseName ], it ] })\n",
      "Nextflow operator map with content: CREATE_INTERVALS_BED(intervals_combined.map{ meta, path -> path }).bed\n",
      "Nextflow operator mix with content: versions = versions.mix(BUILD_INTERVALS.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(CREATE_INTERVALS_BED.out.versions)\n",
      "Nextflow operator map with content: intervals_combined = Channel.fromPath(file(intervals)).map{it -> [ [ id:it.baseName ], it ] }\n",
      "Nextflow operator mix with content: versions = versions.mix(CREATE_INTERVALS_BED.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(GATK4_INTERVALLISTTOBED.out.versions)\n",
      "Nextflow operator flatten with content: intervals_bed = intervals_bed.flatten()\n",
      "Nextflow operator map with content: .map{ intervalFile ->\n",
      "Nextflow operator toInteger with content: start = fields[1].toInteger()\n",
      "Nextflow operator toInteger with content: end = fields[2].toInteger()\n",
      "Nextflow operator toSortedList with content: }.toSortedList({ a, b -> b[0] <=> a[0] })\n",
      "Nextflow operator collate with content: .flatten().collate(2).map{ duration, intervalFile -> intervalFile }.collect()\n",
      "Nextflow operator map with content: .map{ it -> [ it, it.size() ] }\n",
      "Nextflow operator transpose with content: .transpose()\n",
      "Nextflow operator map with content: TABIX_BGZIPTABIX_INTERVAL_SPLIT(intervals_bed.map{ file, num_intervals -> [ [ id:file.baseName], file ] })\n",
      "Nextflow operator map with content: intervals_bed_gz_tbi = TABIX_BGZIPTABIX_INTERVAL_SPLIT.out.gz_tbi.map{ meta, bed, tbi -> [ bed, tbi ] }.toList()\n",
      "Nextflow operator map with content: .map{ it -> [ it, it.size() ] }\n",
      "Nextflow operator transpose with content: .transpose()\n",
      "Nextflow operator mix with content: versions = versions.mix(TABIX_BGZIPTABIX_INTERVAL_SPLIT.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(TABIX_BGZIPTABIX_INTERVAL_COMBINED.out.versions)\n",
      "Nextflow operator collect with content: intervals_bed_gz_tbi_combined = TABIX_BGZIPTABIX_INTERVAL_COMBINED.out.gz_tbi.map{meta, gz, tbi -> [gz, tbi] }.collect()\n",
      "Nextflow operator map with content: intervals_bed_gz_tbi_and_num_intervals = intervals_bed_gz_tbi.map{ intervals, num_intervals ->\n",
      "Nextflow operator collect with content: intervals_combined.map{meta, bed -> bed }.collect()\n",
      "Nextflow operator collect with content: intervals_bed_combined.map{it -> [ [ id:it.baseName ], it ]}.collect() :\n",
      "Nextflow operator map with content: intervals_and_num_intervals   = intervals_bed.map{ interval, num_intervals ->\n",
      "Error: Unclosed process block in file: ./git_repos\\rnadnavar\\subworkflows\\local\\prepare_intervals\\main.nf\n",
      "Process PREPARE_REFERENCE_AND_INTERVALS in filepath ./git_repos\\rnadnavar\\subworkflows\\local\\prepare_reference_and_intervals\\main.nf\n",
      "Nextflow operator collect with content: dbsnp              = params.dbsnp              ? Channel.fromPath(params.dbsnp).collect()                    : Channel.value([])\n",
      "Nextflow operator collect with content: known_snps         = params.known_snps         ? Channel.fromPath(params.known_snps).collect()               : Channel.value([])\n",
      "Nextflow operator collect with content: fasta              = params.fasta              ? Channel.fromPath(params.fasta).collect()                    : Channel.empty()\n",
      "Nextflow operator collect with content: fasta_fai          = params.fasta_fai          ? Channel.fromPath(params.fasta_fai).collect()                : Channel.empty()\n",
      "Nextflow operator collect with content: germline_resource  = params.germline_resource  ? Channel.fromPath(params.germline_resource).collect()        : Channel.value([]) //Mutec2 does not require a germline resource, so set to optional input\n",
      "Nextflow operator collect with content: known_indels       = params.known_indels       ? Channel.fromPath(params.known_indels).collect()             : Channel.value([])\n",
      "Nextflow operator collect with content: pon                = params.pon                ? Channel.fromPath(params.pon).collect()                      : Channel.value([]) //PON is optional for Mutect2 (but highly recommended)\n",
      "Nextflow operator collect with content: whitelist          = params.whitelist          ? Channel.fromPath(params.whitelist).collect()                : Channel.value([]) // whitelist optional for filtering\n",
      "Nextflow operator mix with content: versions = versions.mix(PREPARE_GENOME.out.versions)\n",
      "Nextflow operator collect with content: bwa                    = params.fasta                   ? params.bwa                        ? Channel.fromPath(params.bwa).collect()                           : PREPARE_GENOME.out.bwa                   : []\n",
      "Nextflow operator collect with content: bwamem2                = params.fasta                   ? params.bwamem2                    ? Channel.fromPath(params.bwamem2).collect()                       : PREPARE_GENOME.out.bwamem2               : []\n",
      "Nextflow operator collect with content: dragmap                = params.fasta                   ? params.dragmap                    ? Channel.fromPath(params.dragmap).collect()                       : PREPARE_GENOME.out.hashtable             : []\n",
      "Nextflow operator collect with content: hisat2_index           = params.fasta                   ? params.hisat2_index               ? Channel.fromPath(params.hisat2_index).map{ it -> [ [id:'ht_idx'], it ] }.collect()                   : PREPARE_GENOME.out.hisat2_index : []\n",
      "Nextflow operator collect with content: splicesites            = params.fasta                   ? params.splicesites                ? Channel.fromPath(params.splicesites).collect()                   : PREPARE_GENOME.out.splicesites           : []\n",
      "Nextflow operator collect with content: dict                   = params.dict                    ? Channel.fromPath(params.dict).map{ it -> [ [id:'dict'], it ] }.collect()                             : PREPARE_GENOME.out.dict\n",
      "Nextflow operator collect with content: fasta_fai              = params.fasta                   ? params.fasta_fai                  ? Channel.fromPath(params.fasta_fai).collect()                     : PREPARE_GENOME.out.fasta_fai             : []\n",
      "Nextflow operator collect with content: dbsnp_tbi              = params.dbsnp                   ? params.dbsnp_tbi                  ? Channel.fromPath(params.dbsnp_tbi).collect()                     : PREPARE_GENOME.out.dbsnp_tbi             : Channel.value([])\n",
      "Nextflow operator collect with content: germline_resource_tbi  = params.germline_resource       ? params.germline_resource_tbi      ? Channel.fromPath(params.germline_resource_tbi).collect()         : PREPARE_GENOME.out.germline_resource_tbi : []\n",
      "Nextflow operator collect with content: known_indels_tbi       = params.known_indels            ? params.known_indels_tbi           ? Channel.fromPath(params.known_indels_tbi).collect()              : PREPARE_GENOME.out.known_indels_tbi      : Channel.value([])\n",
      "Nextflow operator collect with content: known_snps_tbi         = params.known_snps              ? params.known_snps_tbi             ? Channel.fromPath(params.known_snps_tbi).collect()                : PREPARE_GENOME.out.known_snps_tbi        : Channel.value([])\n",
      "Nextflow operator collect with content: pon_tbi                = params.pon                     ? params.pon_tbi                    ? Channel.fromPath(params.pon_tbi).collect()                       : PREPARE_GENOME.out.pon_tbi               : []\n",
      "Nextflow operator collect with content: known_sites_indels     = dbsnp.concat(known_indels).collect()\n",
      "Nextflow operator collect with content: known_sites_indels_tbi = dbsnp_tbi.concat(known_indels_tbi).collect()\n",
      "Nextflow operator collect with content: known_sites_snps       = dbsnp.concat(known_snps).collect()\n",
      "Nextflow operator collect with content: known_sites_snps_tbi   = dbsnp_tbi.concat(known_snps_tbi).collect()\n",
      "Nextflow operator mix with content: versions = versions.mix(PREPARE_INTERVALS.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\rnadnavar\\subworkflows\\local\\prepare_reference_and_intervals\\main.nf\n",
      "Process BAM_EXTRACT_READS_HISAT2_ALIGN in filepath ./git_repos\\rnadnavar\\subworkflows\\local\\prepare_second_run\\main.nf\n",
      "Nextflow operator branch with content: input_elements_status = input_sample.branch{\n",
      "Nextflow operator dump with content: input_elements_status.norealign.dump(tag:\"input_elements_status.norealign\")\n",
      "Nextflow operator map with content: dna_mafs = input_elements_status.norealign.map{meta, vcf -> [meta + [ consensus: meta.variantcaller ==~ /(\\S+)?(?i)consensus(\\S+)/ ], vcf, meta.variantcaller]}\n",
      "Nextflow operator branch with content: dna_mafs_consensus = dna_mafs.branch{\n",
      "Nextflow operator map with content: dna_varcall_mafs  = dna_mafs_consensus.noconsensus.map{meta, maf, vc -> [meta, [maf], [vc]]}\n",
      "Nextflow operator dump with content: dna_mafs.dump(tag:\"dna_mafs\")\n",
      "Nextflow operator dump with content: dna_consensus_maf.dump(tag:\"dna_consensus_maf\")\n",
      "Nextflow operator dump with content: maf_with_candidates.dump(tag:\"maf_with_candidates\")\n",
      "Nextflow operator dump with content: cram_to_realign.dump(tag:\"cram_to_realign\")\n",
      "Nextflow operator dump with content: reads_to_realign.dump(tag:'reads_to_realign0')\n",
      "Nextflow operator branch with content: reads_to_realign_branch = reads_to_realign.branch{\n",
      "Nextflow operator branch with content: maf_with_candidates_branch = maf_with_candidates.branch{\n",
      "Nextflow operator dump with content: maf_with_candidates_branch.realign.dump(tag:'maf_with_candidates0')\n",
      "Nextflow operator map with content: maf_with_candidates_tumor = maf_with_candidates_branch.realign.map{\n",
      "Nextflow operator map with content: maf_with_candidates_normal = maf_with_candidates_branch.realign.map{\n",
      "Nextflow operator mix with content: maf_with_candidates_to_realign = maf_with_candidates_tumor.mix(maf_with_candidates_normal)\n",
      "Nextflow operator map with content: reads_to_realign_and_join = reads_to_realign_branch.realign.map{meta, cram, crai ->\n",
      "Nextflow operator dump with content: reads_to_realign_and_join.dump(tag:'reads_to_realign1')\n",
      "Nextflow operator dump with content: maf_with_candidates_to_realign.dump(tag:'maf_with_candidates_to_realign')\n",
      "Nextflow operator join with content: cram_to_realign = reads_to_realign_and_join.join(maf_with_candidates_to_realign)\n",
      "Nextflow operator dump with content: cram_to_realign.dump(tag:\"cram_to_realign\")\n",
      "Nextflow operator map with content: maf_to_bed = cram_to_realign.map{meta, cram, crai, maf -> [meta + [cram_file:cram, crai_file:crai, maf_file:maf], maf]}\n",
      "Nextflow operator map with content: cram_to_extract = MAF2BED.out.bed.map{meta, bed -> [meta, meta.cram_file, meta.crai_file, bed]}\n",
      "Nextflow operator map with content: cram_to_filter = SAMTOOLS_EXTRACT_READ_IDS.out.read_ids.map{meta, readsid -> [meta, meta.cram_file, readsid]}\n",
      "Nextflow operator join with content: bam_to_fq = PICARD_FILTERSAMREADS.out.bam.join(PICARD_FILTERSAMREADS.out.bai)\n",
      "Nextflow operator map with content: fasta.map{it -> [ [ id:\"fasta\" ], it ]}, // fasta\n",
      "Nextflow operator map with content: fasta_fai.map{it -> [ [ id:\"fasta_fai\" ], it ]},  // fasta_fai\n",
      "Nextflow operator map with content: reads_for_realignment.map{meta, reads -> [meta + [single_end:se], reads]},\n",
      "Nextflow operator map with content: fasta.map{it -> [ [ id:\"fasta\" ], it ]}\n",
      "Nextflow operator join with content: bam_mapped = FASTQ_ALIGN_HISAT2.out.bam.join(FASTQ_ALIGN_HISAT2.out.bai).map{meta,bam,bai -> [meta + [ id:meta.sample, data_type:\"bam\"], bam, bai]}\n",
      "Nextflow operator dump with content: bam_mapped.dump(tag:\"bam_mapped_REALIGN\")\n",
      "Error: Unclosed process block in file: ./git_repos\\rnadnavar\\subworkflows\\local\\prepare_second_run\\main.nf\n",
      "Process SAMPLESHEET_TO_CHANNEL in filepath ./git_repos\\rnadnavar\\subworkflows\\local\\samplesheet_to_channel\\main.nf\n",
      "Nextflow operator dump with content: ch_from_samplesheet.dump(tag:\"ch_from_samplesheet\")\n",
      "Nextflow operator map with content: .map{ meta, fastq_1, fastq_2, table, cram, crai, bam, bai, vcf, variantcaller, maf ->\n",
      "Nextflow operator tap with content: .tap{ ch_with_patient_sample } // save the channel\n",
      "Nextflow operator groupTuple with content: .groupTuple() //group by patient_sample to get all lanes\n",
      "Nextflow operator map with content: .map { patient_sample, ch_items ->\n",
      "Nextflow operator combine with content: .combine(ch_with_patient_sample, by: 0) // for each entry add numLanes\n",
      "Nextflow operator map with content: .map { patient_sample, num_lanes, ch_items ->\n",
      "Nextflow operator toInteger with content: meta           = meta + [num_lanes: num_lanes.toInteger(), read_group: read_group.toString(), data_type: 'fastq', size: 1]\n",
      "Nextflow operator toInteger with content: if (cram)  return [ meta + [num_lanes: num_lanes.toInteger(), read_group: read_group.toString(), data_type: 'cram', size: 1], cram, crai, maf ]\n",
      "Nextflow operator toInteger with content: else if (bam) return [ meta + [num_lanes: num_lanes.toInteger(), read_group: read_group.toString(), data_type: 'bam', size: 1], bam, bai, maf ]\n",
      "Nextflow operator toInteger with content: meta            = meta + [num_lanes: num_lanes.toInteger(), read_group: read_group.toString(), data_type: 'bam', size: 1]\n",
      "Nextflow operator dump with content: input_sample.dump(tag:\"input_sample\")\n",
      "Nextflow operator filter with content: input_sample.filter{ it[0].status == 1 }.ifEmpty{ // In this case, the sample-sheet contains no tumor-samples\n",
      "Nextflow operator join with content: error('The sample-sheet only contains normal-samples, but the following tools, which were requested with \"--tools\", expect at least one tumor-sample : ' + tools_tumor_asked.join(\", \"))\n",
      "Nextflow operator filter with content: input_sample.filter{ it[0].status == 0 }.ifEmpty{ // In this case, the sample-sheet contains no normal/germline-samples\n",
      "Nextflow operator join with content: error('The sample-sheet only contains tumor-samples, but the following tools, which were requested by the option \"tools\", expect at least one normal-sample at the moment: ' + requested_tools_requiring_normal_samples.join(\", \"))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Process VCF_ANNOTATE in filepath ./git_repos\\rnadnavar\\subworkflows\\local\\vcf_annotate\\main.nf\n",
      "Nextflow operator map with content: fasta = (params.vep_include_fasta) ? fasta.map{ fasta -> [ [ id:fasta.baseName ], fasta ] } : [[id: 'null'], []]\n",
      "Nextflow operator collect with content: vep_cache          = params.vep_cache    ? params.use_annotation_cache_keys ? Channel.fromPath(\"${params.vep_cache}/${params.vep_cache_version}_${params.vep_genome}\").collect() : Channel.fromPath(params.vep_cache).collect()    : []\n",
      "Nextflow operator map with content: vcf_for_vep = vcf.map{ meta, vcf -> [ meta, vcf, [] ] }\n",
      "Nextflow operator mix with content: reports  = reports.mix(VCF_ANNOTATE_ENSEMBLVEP.out.reports)\n",
      "Nextflow operator map with content: vcf_ann  = vcf_ann.mix(VCF_ANNOTATE_ENSEMBLVEP.out.vcf_tbi).map{meta, vcf, tbi -> [meta +[data_type:\"vcf\"], vcf, tbi]}\n",
      "Nextflow operator mix with content: tab_ann  = tab_ann.mix(VCF_ANNOTATE_ENSEMBLVEP.out.tab)\n",
      "Nextflow operator mix with content: json_ann = json_ann.mix(VCF_ANNOTATE_ENSEMBLVEP.out.json)\n",
      "Nextflow operator mix with content: versions = versions.mix(VCF_ANNOTATE_ENSEMBLVEP.out.versions)\n",
      "Nextflow operator map with content: CHANNEL_ANNOTATE_CREATE_CSV(vcf_ann.map{meta, vcf, tbi -> [meta, vcf]}, \"annotated\")\n",
      "Nextflow operator map with content: vcf_ann = vcf.map{metaVCF -> [metaVCF[0] + [data_type:\"vcf\"], metaVCF[1]]}\n",
      "Error: Unclosed process block in file: ./git_repos\\rnadnavar\\subworkflows\\local\\vcf_annotate\\main.nf\n",
      "Process VCF_CONSENSUS in filepath ./git_repos\\rnadnavar\\subworkflows\\local\\vcf_consensus\\main.nf\n",
      "Nextflow operator branch with content: vcf_to_consensus_type = vcf_to_consensus.branch{\n",
      "Nextflow operator map with content: VCF2MAF(vcf_to_consensus_type.vcf.map{metaVCF -> [metaVCF[0], metaVCF[1]]},\n",
      "Nextflow operator mix with content: maf_to_consensus = VCF2MAF.out.maf.mix(vcf_to_consensus_type.maf)\n",
      "Nextflow operator mix with content: versions         = versions.mix(VCF2MAF.out.versions)\n",
      "Nextflow operator dump with content: maf_to_consensus.dump(tag:\"maf_to_consensus0\")\n",
      "Nextflow operator map with content: maf_to_consensus = maf_to_consensus.map{ meta, maf ->\n",
      "Nextflow operator count with content: def ncallers   = toolsllist.count('sage') +\n",
      "Nextflow operator count with content: toolsllist.count('strelka') +\n",
      "Nextflow operator count with content: toolsllist.count('mutect2')\n",
      "Nextflow operator groupTuple with content: .groupTuple()\n",
      "Nextflow operator dump with content: maf_to_consensus.dump(tag:\"maf_to_consensus1\")\n",
      "Nextflow operator branch with content: mafs_from_varcal   = maf_to_consensus.branch{\n",
      "Nextflow operator branch with content: maf_from_consensus = consensus_maf.branch{\n",
      "Nextflow operator map with content: maf_from_consensus_rna = maf_from_consensus.rna.map{meta, maf -> [meta, maf, ['ConsensusRNA']]}\n",
      "Nextflow operator map with content: maf_from_consensus_dna = maf_from_consensus.dna.map{meta, maf -> [meta, maf, ['ConsensusDNA']]}\n",
      "Nextflow operator mix with content: .mix(maf_from_consensus_rna)\n",
      "Nextflow operator mix with content: .mix(mafs_from_varcal_dna)\n",
      "Nextflow operator mix with content: .mix(mafs_from_varcal_rna)\n",
      "Nextflow operator transpose with content: .transpose(),\n",
      "Nextflow operator map with content: maf_consensus_status_dna_to_cross = maf_from_consensus_dna.map{\n",
      "Nextflow operator map with content: maf_consensus_status_rna_to_cross = maf_from_consensus_rna.map{\n",
      "Nextflow operator map with content: mafs_status_dna_to_cross = mafs_from_varcal_dna.map{\n",
      "Nextflow operator map with content: mafs_status_rna_to_cross = mafs_from_varcal_rna.map{\n",
      "Nextflow operator cross with content: .cross(maf_consensus_status_rna_to_cross)\n",
      "Nextflow operator map with content: .map { dna, rna ->\n",
      "Nextflow operator cross with content: .cross(maf_consensus_status_dna_to_cross)\n",
      "Nextflow operator map with content: .map { rna, dna ->\n",
      "Nextflow operator dump with content: mafs_dna_crossed_with_rna_rescue.mix(mafs_rna_crossed_with_dna_rescue).dump(tag:\"mafs_to_rescue\")\n",
      "Nextflow operator mix with content: RUN_CONSENSUS_RESCUE ( mafs_dna_crossed_with_rna_rescue.mix(mafs_rna_crossed_with_dna_rescue) )\n",
      "Nextflow operator branch with content: maf_from_rescue = RUN_CONSENSUS_RESCUE.out.maf.branch{\n",
      "Nextflow operator map with content: maf_from_consensus_dna = maf_from_rescue.dna.map{meta, maf -> [meta, maf, ['ConsensusDNA']]}\n",
      "Nextflow operator map with content: maf_from_consensus_rna = maf_from_rescue.rna.map{meta, maf -> [meta, maf, ['ConsensusRNA']]}\n",
      "Nextflow operator mix with content: consensus_maf = maf_from_consensus_dna.mix(maf_from_consensus_rna)\n",
      "Nextflow operator mix with content: .mix(maf_from_consensus_rna)\n",
      "Nextflow operator mix with content: .mix(mafs_from_varcal_dna)\n",
      "Nextflow operator dump with content: .mix(mafs_from_varcal_rna).transpose().dump(tag:'rescued')\n",
      "Nextflow operator mix with content: .mix(maf_from_consensus_rna)\n",
      "Nextflow operator mix with content: .mix(mafs_from_varcal_dna)\n",
      "Nextflow operator mix with content: .mix(mafs_from_varcal_rna)\n",
      "Nextflow operator transpose with content: .transpose(),\n",
      "Nextflow operator branch with content: vcf_to_consensus_type = vcf_to_consensus.branch{\n",
      "Nextflow operator map with content: VCF2MAF(vcf_to_consensus_type.vcf.map{metaVCF -> [metaVCF[0], metaVCF[1]]},\n",
      "Nextflow operator mix with content: consensus_maf    = VCF2MAF.out.maf.mix(vcf_to_consensus_type.maf)\n",
      "Nextflow operator mix with content: versions         = versions.mix(VCF2MAF.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\rnadnavar\\subworkflows\\local\\vcf_consensus\\main.nf\n",
      "Process VCF_NORMALISE in filepath ./git_repos\\rnadnavar\\subworkflows\\local\\vcf_normalise\\main.nf\n",
      "Nextflow operator mix with content: vcf_decomposed = vcf_decomposed.mix(VT_DECOMPOSE.out.vcf)\n",
      "Nextflow operator first with content: version = version.mix(VT_DECOMPOSE.out.versions.first())\n",
      "Nextflow operator first with content: version = version.mix(VT_NORMALISE.out.versions.first())\n",
      "Error: Unclosed process block in file: ./git_repos\\rnadnavar\\subworkflows\\local\\vcf_normalise\\main.nf\n",
      "Process VCF_QC_BCFTOOLS_VCFTOOLS in filepath ./git_repos\\rnadnavar\\subworkflows\\local\\vcf_qc_bcftools_vcftools\\main.nf\n",
      "Nextflow operator map with content: BCFTOOLS_STATS(vcf.map{ meta, vcf -> [ meta, vcf, [] ] }, [[:],[]], [[:],[]], [[:],[]], [[:],[]], [[:],[]])\n",
      "Nextflow operator mix with content: versions = versions.mix(BCFTOOLS_STATS.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(VCFTOOLS_TSTV_COUNT.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\rnadnavar\\subworkflows\\local\\vcf_qc_bcftools_vcftools\\main.nf\n",
      "Process ANNOTATION_ENSEMBLVEP in filepath ./git_repos\\rnadnavar\\subworkflows\\nf-core\\annotation\\ensemblvep\\main.nf\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(ENSEMBLVEP.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(TABIX_BGZIPTABIX.out.versions.first())\n",
      "Error: Unclosed process block in file: ./git_repos\\rnadnavar\\subworkflows\\nf-core\\annotation\\ensemblvep\\main.nf\n",
      "Process BAM_MARKDUPLICATES_PICARD in filepath ./git_repos\\rnadnavar\\subworkflows\\nf-core\\bam_markduplicates_picard\\main.nf\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(PICARD_MARKDUPLICATES.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SAMTOOLS_INDEX.out.versions.first())\n",
      "Nextflow operator join with content: .join(SAMTOOLS_INDEX.out.bai, by: [0], remainder: true)\n",
      "Nextflow operator join with content: .join(SAMTOOLS_INDEX.out.csi, by: [0], remainder: true)\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BAM_STATS_SAMTOOLS.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\rnadnavar\\subworkflows\\nf-core\\bam_markduplicates_picard\\main.nf\n",
      "Process BAM_SORT_STATS_SAMTOOLS in filepath ./git_repos\\rnadnavar\\subworkflows\\nf-core\\bam_sort_stats_samtools\\main.nf\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SAMTOOLS_SORT.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SAMTOOLS_INDEX.out.versions.first())\n",
      "Nextflow operator join with content: .join(SAMTOOLS_INDEX.out.bai, by: [0], remainder: true)\n",
      "Nextflow operator join with content: .join(SAMTOOLS_INDEX.out.csi, by: [0], remainder: true)\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator set with content: .set { ch_bam_bai }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BAM_STATS_SAMTOOLS.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\rnadnavar\\subworkflows\\nf-core\\bam_sort_stats_samtools\\main.nf\n",
      "Process BAM_STATS_SAMTOOLS in filepath ./git_repos\\rnadnavar\\subworkflows\\nf-core\\bam_stats_samtools\\main.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_STATS.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_FLAGSTAT.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_IDXSTATS.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\rnadnavar\\subworkflows\\nf-core\\bam_stats_samtools\\main.nf\n",
      "Process BAM_TUMOR_NORMAL_SOMATIC_VARIANT_CALLING_GATK in filepath ./git_repos\\rnadnavar\\subworkflows\\nf-core\\bam_tumor_normal_somatic_variant_calling_gatk\\main.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(MUTECT2.out.versions)\n",
      "Nextflow operator collect with content: LEARNREADORIENTATIONMODEL (MUTECT2.out.f1r2.collect())\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(LEARNREADORIENTATIONMODEL.out.versions)\n",
      "Nextflow operator combine with content: ch_pileup_tumor_input = ch_input.combine(ch_interval_file).map {\n",
      "Nextflow operator combine with content: ch_pileup_normal_input = ch_input.combine(ch_interval_file).map {\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(GETPILEUPSUMMARIES_TUMOR.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(GETPILEUPSUMMARIES_NORMAL.out.versions.first())\n",
      "Nextflow operator collect with content: ch_pileup_tumor = GETPILEUPSUMMARIES_TUMOR.out.table.collect()\n",
      "Nextflow operator collect with content: ch_pileup_normal = GETPILEUPSUMMARIES_NORMAL.out.table.collect()\n",
      "Nextflow operator join with content: ch_calccon_in = ch_pileup_tumor.join(ch_pileup_normal, failOnDuplicate: true, failOnMismatch: true)\n",
      "Nextflow operator mix with content: ch_versions   = ch_versions.mix(CALCULATECONTAMINATION.out.versions)\n",
      "Nextflow operator collect with content: ch_vcf           = MUTECT2.out.vcf.collect()\n",
      "Nextflow operator collect with content: ch_tbi           = MUTECT2.out.tbi.collect()\n",
      "Nextflow operator collect with content: ch_stats         = MUTECT2.out.stats.collect()\n",
      "Nextflow operator collect with content: ch_orientation   = LEARNREADORIENTATIONMODEL.out.artifactprior.collect()\n",
      "Nextflow operator collect with content: ch_segment       = CALCULATECONTAMINATION.out.segmentation.collect()\n",
      "Nextflow operator collect with content: ch_contamination = CALCULATECONTAMINATION.out.contamination.collect()\n",
      "Nextflow operator join with content: .join(ch_tbi, failOnDuplicate: true, failOnMismatch: true)\n",
      "Nextflow operator join with content: .join(ch_stats, failOnDuplicate: true, failOnMismatch: true)\n",
      "Nextflow operator join with content: .join(ch_orientation, failOnDuplicate: true, failOnMismatch: true)\n",
      "Nextflow operator join with content: .join(ch_segment, failOnDuplicate: true, failOnMismatch: true)\n",
      "Nextflow operator join with content: .join(ch_contamination, failOnDuplicate: true, failOnMismatch: true)\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(FILTERMUTECTCALLS.out.versions.first())\n",
      "Nextflow operator collect with content: mutect2_vcf            = MUTECT2.out.vcf.collect()                             // channel: [ val(meta), path(vcf) ]\n",
      "Nextflow operator collect with content: mutect2_tbi            = MUTECT2.out.tbi.collect()                             // channel: [ val(meta), path(tbi) ]\n",
      "Nextflow operator collect with content: mutect2_stats          = MUTECT2.out.stats.collect()                           // channel: [ val(meta), path(stats) ]\n",
      "Nextflow operator collect with content: mutect2_f1r2           = MUTECT2.out.f1r2.collect()                            // channel: [ val(meta), path(f1r2) ]\n",
      "Nextflow operator collect with content: artifact_priors        = LEARNREADORIENTATIONMODEL.out.artifactprior.collect() // channel: [ val(meta), path(artifactprior) ]\n",
      "Nextflow operator collect with content: pileup_table_tumor     = GETPILEUPSUMMARIES_TUMOR.out.table.collect()          // channel: [ val(meta), path(table) ]\n",
      "Nextflow operator collect with content: pileup_table_normal    = GETPILEUPSUMMARIES_NORMAL.out.table.collect()         // channel: [ val(meta), path(table) ]\n",
      "Nextflow operator collect with content: contamination_table    = CALCULATECONTAMINATION.out.contamination.collect()    // channel: [ val(meta), path(table) ]\n",
      "Nextflow operator collect with content: segmentation_table     = CALCULATECONTAMINATION.out.segmentation.collect()     // channel: [ val(meta), path(table) ]\n",
      "Nextflow operator collect with content: filtered_vcf           = FILTERMUTECTCALLS.out.vcf.collect()                   // channel: [ val(meta), path(vcf) ]\n",
      "Nextflow operator collect with content: filtered_tbi           = FILTERMUTECTCALLS.out.tbi.collect()                   // channel: [ val(meta), path(tbi) ]\n",
      "Nextflow operator collect with content: filtered_stats         = FILTERMUTECTCALLS.out.stats.collect()                 // channel: [ val(meta), path(stats) ]\n",
      "Error: Unclosed process block in file: ./git_repos\\rnadnavar\\subworkflows\\nf-core\\bam_tumor_normal_somatic_variant_calling_gatk\\main.nf\n",
      "Process FASTQ_ALIGN_BWA in filepath ./git_repos\\rnadnavar\\subworkflows\\nf-core\\fastq_align_bwa\\main.nf\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(BWA_MEM.out.versions.first())\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BAM_SORT_STATS_SAMTOOLS.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\rnadnavar\\subworkflows\\nf-core\\fastq_align_bwa\\main.nf\n",
      "Process FASTQ_ALIGN_HISAT2 in filepath ./git_repos\\rnadnavar\\subworkflows\\nf-core\\fastq_align_hisat2\\main.nf\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(HISAT2_ALIGN.out.versions.first())\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BAM_SORT_STATS_SAMTOOLS.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\rnadnavar\\subworkflows\\nf-core\\fastq_align_hisat2\\main.nf\n",
      "Process FASTQ_ALIGN_STAR in filepath ./git_repos\\rnadnavar\\subworkflows\\nf-core\\fastq_align_star\\main.nf\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(STAR_ALIGN.out.versions.first())\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BAM_SORT_STATS_SAMTOOLS.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\rnadnavar\\subworkflows\\nf-core\\fastq_align_star\\main.nf\n",
      "Process GATK4_MAPPING in filepath ./git_repos\\rnadnavar\\subworkflows\\nf-core\\gatk4\\mapping\\main.nf\n",
      "Nextflow operator mix with content: ch_bam_mapped = ch_bam_mapped.mix(BWAMEM1_MEM.out.bam)\n",
      "Nextflow operator mix with content: ch_bam_mapped = ch_bam_mapped.mix(BWAMEM2_MEM.out.bam)\n",
      "Nextflow operator mix with content: ch_bam_mapped = ch_bam_mapped.mix(DRAGMAP_ALIGN.out.bam)\n",
      "Nextflow operator mix with content: ch_reports = ch_reports.mix(DRAGMAP_ALIGN.out.log)\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(BWAMEM1_MEM.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(BWAMEM2_MEM.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(DRAGMAP_ALIGN.out.versions.first())\n",
      "Error: Unclosed process block in file: ./git_repos\\rnadnavar\\subworkflows\\nf-core\\gatk4\\mapping\\main.nf\n",
      "Process BAM_MARKDUPLICATES in filepath ./git_repos\\rnadnavar\\subworkflows\\nf-core\\gatk4\\markduplicates\\main.nf\n",
      "Nextflow operator join with content: cram = GATK4_MARKDUPLICATES.out.cram.join(GATK4_MARKDUPLICATES.out.crai, failOnDuplicate: true, failOnMismatch: true)\n",
      "Nextflow operator mix with content: reports = reports.mix(GATK4_MARKDUPLICATES.out.metrics)\n",
      "Nextflow operator mix with content: reports = reports.mix(CRAM_QC_MOSDEPTH_SAMTOOLS.out.reports)\n",
      "Nextflow operator mix with content: versions = versions.mix(GATK4_MARKDUPLICATES.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(CRAM_QC_MOSDEPTH_SAMTOOLS.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\rnadnavar\\subworkflows\\nf-core\\gatk4\\markduplicates\\main.nf\n",
      "Process MARKDUPLICATES_SPARK in filepath ./git_repos\\rnadnavar\\subworkflows\\nf-core\\gatk4\\markduplicates_spark\\main.nf\n",
      "Nextflow operator join with content: .join(INDEX_MARKDUPLICATES.out.crai)\n",
      "Nextflow operator mix with content: qc_reports = qc_reports.mix(GATK4_ESTIMATELIBRARYCOMPLEXITY.out.metrics,\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(GATK4_ESTIMATELIBRARYCOMPLEXITY.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(GATK4_MARKDUPLICATES_SPARK.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(INDEX_MARKDUPLICATES.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(BAM_TO_CRAM.out.versions.first())\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_CRAMTOBAM.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\rnadnavar\\subworkflows\\nf-core\\gatk4\\markduplicates_spark\\main.nf\n",
      "Process BAM_BASERECALIBRATOR in filepath ./git_repos\\rnadnavar\\subworkflows\\nf-core\\gatk4\\prepare_recalibration\\main.nf\n",
      "Nextflow operator combine with content: cram_intervals = cram.combine(intervals)\n",
      "Nextflow operator map with content: .map{ meta, cram, crai, intervals, num_intervals -> [ meta + [ num_intervals:num_intervals ], cram, crai, intervals ] }\n",
      "Nextflow operator branch with content: table_to_merge = GATK4_BASERECALIBRATOR.out.table.map{ meta, table -> [ groupKey(meta, meta.num_intervals), table ] }.groupTuple().branch{\n",
      "Nextflow operator map with content: table_bqsr = GATK4_GATHERBQSRREPORTS.out.table.mix(table_to_merge.single.map{ meta, table -> [ meta, table[0] ] })\n",
      "Nextflow operator map with content: .map{ meta, table -> [ meta - meta.subMap('num_intervals'), table ] }\n",
      "Nextflow operator mix with content: versions = versions.mix(GATK4_BASERECALIBRATOR.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(GATK4_GATHERBQSRREPORTS.out.versions)\n",
      "Process RECALIBRATE in filepath ./git_repos\\rnadnavar\\subworkflows\\nf-core\\gatk4\\recalibrate\\main.nf\n",
      "Nextflow operator combine with content: cram_intervals = cram.combine(intervals)\n",
      "Nextflow operator map with content: .map{ meta, cram, crai, recal, intervals, num_intervals ->\n",
      "Nextflow operator map with content: ch_cram_recal_out = MERGE_INDEX_CRAM.out.cram_crai.map{ meta, cram, crai ->\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(APPLYBQSR.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(MERGE_INDEX_CRAM.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\rnadnavar\\subworkflows\\nf-core\\gatk4\\recalibrate\\main.nf\n",
      "Process RECALIBRATE_SPARK in filepath ./git_repos\\rnadnavar\\subworkflows\\nf-core\\gatk4\\recalibrate_spark\\main.nf\n",
      "Nextflow operator combine with content: cram_intervals = cram.combine(intervals)\n",
      "Nextflow operator map with content: .map{ meta, cram, crai, recal, intervals, num_intervals ->\n",
      "Nextflow operator map with content: ch_cram_recal_out = MERGE_INDEX_CRAM.out.cram_crai.map{ meta, cram, crai ->\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(APPLYBQSR_SPARK.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(MERGE_INDEX_CRAM.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\rnadnavar\\subworkflows\\nf-core\\gatk4\\recalibrate_spark\\main.nf\n",
      "Process GATK_TUMOR_NORMAL_SOMATIC_VARIANT_CALLING in filepath ./git_repos\\rnadnavar\\subworkflows\\nf-core\\gatk4\\tumor_normal_somatic_variant_calling\\main.nf\n",
      "Nextflow operator map with content: ch_input = input.map{ meta, input, input_index, intervals ->\n",
      "Nextflow operator branch with content: MUTECT2.out.vcf.branch{\n",
      "Nextflow operator set with content: }.set{ mutect2_vcf_branch }\n",
      "Nextflow operator branch with content: MUTECT2.out.tbi.branch{\n",
      "Nextflow operator set with content: }.set{ mutect2_tbi_branch }\n",
      "Nextflow operator branch with content: MUTECT2.out.stats.branch{\n",
      "Nextflow operator set with content: }.set{ mutect2_stats_branch }\n",
      "Nextflow operator branch with content: MUTECT2.out.f1r2.branch{\n",
      "Nextflow operator set with content: }.set{ mutect2_f1r2_branch }\n",
      "Nextflow operator dump with content: mutect2_vcf_branch.intervals.dump(tag:'[STEP3] GATK4 - mutect2_vcf_branch.intervals')\n",
      "Nextflow operator unique with content: .unique()  // Potential bug is nextflow were some files are duplicated\n",
      "Nextflow operator groupTuple with content: .groupTuple()\n",
      "Nextflow operator dump with content: mutect2_to_merge.dump(tag:'[STEP3] GATK4 - mutect2_to_merge')\n",
      "Nextflow operator mix with content: mutect2_vcf = Channel.empty().mix(\n",
      "Nextflow operator dump with content: mutect2_vcf.dump(tag:'[STEP3] GATK4 - mutect2')\n",
      "Nextflow operator mix with content: mutect2_tbi = Channel.empty().mix(\n",
      "Nextflow operator dump with content: mutect2_tbi.dump(tag:'[STEP3] GATK4 - mutect2 tbi')\n",
      "Nextflow operator groupTuple with content: .groupTuple())\n",
      "Nextflow operator mix with content: mutect2_stats = MERGEMUTECTSTATS.out.stats.mix(\n",
      "Nextflow operator dump with content: mutect2_stats.dump(tag:'[STEP3] GATK4 - MUTECTSTATS')\n",
      "Nextflow operator map with content: artifactprior = mutect2_vcf.map{meta, vcf -> [meta, file(\"${meta.id}_PRIOR_NO_ARTPRIOR\")]}\n",
      "Nextflow operator groupTuple with content: .groupTuple()\n",
      "Nextflow operator mix with content: .mix(mutect2_f1r2_branch.no_intervals)\n",
      "Nextflow operator mix with content: ch_versions   = ch_versions.mix(LEARNREADORIENTATIONMODEL.out.versions)\n",
      "Nextflow operator map with content: artifactprior = mutect2_vcf.map{meta, vcf -> [meta, file(\"${meta.id}_NO_ARTPRIOR\")]}\n",
      "Nextflow operator map with content: segmentation = mutect2_vcf.map{meta, vcf -> [meta, file(\"${meta.id}_PRIOR_NO_SEG\")]}\n",
      "Nextflow operator map with content: contamination =  mutect2_vcf.map{meta, vcf -> [meta, file(\"${meta.id}_PRIOR_NO_TABLE\")]}\n",
      "Nextflow operator multiMap with content: pileup = input.multiMap{  meta, input_list, input_index_list, intervals ->\n",
      "Nextflow operator map with content: GETPILEUPSUMMARIES_TUMOR ( pileup.tumor.map{\n",
      "Nextflow operator map with content: GETPILEUPSUMMARIES_NORMAL ( pileup.normal.map{\n",
      "Nextflow operator branch with content: GETPILEUPSUMMARIES_NORMAL.out.table.branch{\n",
      "Nextflow operator branch with content: GETPILEUPSUMMARIES_TUMOR.out.table.branch{\n",
      "Nextflow operator groupTuple with content: GETPILEUPSUMMARIES_NORMAL.out.table.groupTuple(),\n",
      "Nextflow operator mix with content: gather_table_normal = Channel.empty().mix(\n",
      "Nextflow operator map with content: pileup_table_normal.no_intervals).map{ meta, table ->\n",
      "Nextflow operator groupTuple with content: GETPILEUPSUMMARIES_TUMOR.out.table.groupTuple(),\n",
      "Nextflow operator mix with content: gather_table_tumor = Channel.empty().mix(\n",
      "Nextflow operator map with content: pileup_table_tumor.no_intervals).map{ meta, table ->\n",
      "Nextflow operator join with content: CALCULATECONTAMINATION( gather_table_tumor.join(gather_table_normal) )\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GETPILEUPSUMMARIES_NORMAL.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GETPILEUPSUMMARIES_TUMOR.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GATHERPILEUPSUMMARIES_NORMAL.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GATHERPILEUPSUMMARIES_TUMOR.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(CALCULATECONTAMINATION.out.versions)\n",
      "Nextflow operator map with content: segmentation = mutect2_vcf.map{meta, vcf -> [meta, file(\"${meta.id}_NO_SEG\")]}\n",
      "Nextflow operator map with content: contamination =  mutect2_vcf.map{meta, vcf -> [meta, file(\"${meta.id}_NO_TABLE\")]}\n",
      "Nextflow operator dump with content: mutect2_vcf.dump(tag:\"mutect2_vcf FOR FMC\")\n",
      "Nextflow operator dump with content: contamination.dump(tag:\"CONTAMINATION FOR FMC\")\n",
      "Nextflow operator dump with content: artifactprior.dump(tag:\"artifactprior FOR FMC\")\n",
      "Nextflow operator dump with content: segmentation.dump(tag:\"segmentation FOR FMC\")\n",
      "Nextflow operator join with content: ch_filtermutect    = mutect2_vcf.join(mutect2_tbi)\n",
      "Nextflow operator join with content: .join(mutect2_stats)\n",
      "Nextflow operator join with content: .join(artifactprior)\n",
      "Nextflow operator join with content: .join(segmentation)\n",
      "Nextflow operator join with content: .join(contamination)\n",
      "Nextflow operator map with content: ch_filtermutect_in = ch_filtermutect.map{ meta, vcf, tbi, stats, orientation, seg, cont ->\n",
      "Nextflow operator dump with content: ch_filtermutect.dump(tag:\"[STEP3] GATK4 - input FILTERMUTECTCALLS\")\n",
      "Nextflow operator map with content: filtered_vcf = FILTERMUTECTCALLS.out.vcf.map{ meta, vcf ->\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(MERGE_MUTECT2.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(FILTERMUTECTCALLS.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(MERGEMUTECTSTATS.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(MUTECT2.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\rnadnavar\\subworkflows\\nf-core\\gatk4\\tumor_normal_somatic_variant_calling\\main.nf\n",
      "Process GATK_TUMOR_ONLY_SOMATIC_VARIANT_CALLING in filepath ./git_repos\\rnadnavar\\subworkflows\\nf-core\\gatk4\\tumor_only_somatic_variant_calling\\main.nf\n",
      "Nextflow operator branch with content: MUTECT2.out.vcf.branch{\n",
      "Nextflow operator set with content: }.set{ mutect2_vcf_branch }\n",
      "Nextflow operator branch with content: MUTECT2.out.tbi.branch{\n",
      "Nextflow operator set with content: }.set{ mutect2_tbi_branch }\n",
      "Nextflow operator branch with content: MUTECT2.out.stats.branch{\n",
      "Nextflow operator set with content: }.set{ mutect2_stats_branch }\n",
      "Nextflow operator branch with content: MUTECT2.out.f1r2.branch{\n",
      "Nextflow operator set with content: }.set{ mutect2_f1r2_branch }\n",
      "Nextflow operator map with content: .map{ meta, vcf ->\n",
      "Nextflow operator groupTuple with content: }.groupTuple(),\n",
      "Nextflow operator mix with content: mutect2_vcf = Channel.empty().mix(\n",
      "Nextflow operator mix with content: mutect2_tbi = Channel.empty().mix(\n",
      "Nextflow operator map with content: .map{ meta, stats ->\n",
      "Nextflow operator groupTuple with content: }.groupTuple())\n",
      "Nextflow operator mix with content: mutect2_stats = Channel.empty().mix(\n",
      "Nextflow operator mix with content: Channel.empty().mix(\n",
      "Nextflow operator map with content: .map{ meta, f1r2 ->\n",
      "Nextflow operator groupTuple with content: }.groupTuple(),\n",
      "Nextflow operator branch with content: GETPILEUPSUMMARIES.out.table.branch{\n",
      "Nextflow operator map with content: .map{ meta, table ->\n",
      "Nextflow operator groupTuple with content: }.groupTuple(),\n",
      "Nextflow operator mix with content: pileup_table = Channel.empty().mix(\n",
      "Nextflow operator map with content: table_contamination = pileup_table.map{meta, table -> [meta, table, []]}\n",
      "Nextflow operator join with content: ch_filtermutect    = mutect2_vcf.join(mutect2_tbi)\n",
      "Nextflow operator join with content: .join(mutect2_stats)\n",
      "Nextflow operator join with content: .join(LEARNREADORIENTATIONMODEL.out.artifactprior)\n",
      "Nextflow operator join with content: .join(CALCULATECONTAMINATION.out.segmentation)\n",
      "Nextflow operator join with content: .join(CALCULATECONTAMINATION.out.contamination)\n",
      "Nextflow operator map with content: ch_filtermutect_in = ch_filtermutect.map{ meta, vcf, tbi, stats, artifactprior, seg, cont -> [meta, vcf, tbi, stats, artifactprior, seg, cont, []] }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(MERGE_MUTECT2.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(CALCULATECONTAMINATION.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(FILTERMUTECTCALLS.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GETPILEUPSUMMARIES.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GATHERPILEUPSUMMARIES.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(LEARNREADORIENTATIONMODEL.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(MERGEMUTECTSTATS.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(MUTECT2.out.versions)\n",
      "Nextflow operator map with content: filtered_vcf        = FILTERMUTECTCALLS.out.vcf.map{ meta, vcf -> [[\n",
      "Error: Unclosed process block in file: ./git_repos\\rnadnavar\\subworkflows\\nf-core\\gatk4\\tumor_only_somatic_variant_calling\\main.nf\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Process VCF_ANNOTATE_ENSEMBLVEP in filepath ./git_repos\\rnadnavar\\subworkflows\\nf-core\\vcf_annotate_ensemblvep\\main.nf\n",
      "Nextflow operator join with content: ch_vcf_tbi = ENSEMBLVEP_VEP.out.vcf.join(TABIX_TABIX.out.tbi, failOnDuplicate: true, failOnMismatch: true)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ENSEMBLVEP_VEP.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(TABIX_TABIX.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\rnadnavar\\subworkflows\\nf-core\\vcf_annotate_ensemblvep\\main.nf\n",
      "Process RNADNAVAR in filepath ./git_repos\\rnadnavar\\workflows\\rnadnavar.nf\n",
      "Nextflow operator collect with content: vep_cache    = ENSEMBLVEP_DOWNLOAD.out.cache.collect().map{ meta, cache -> [ cache ] }\n",
      "Nextflow operator mix with content: versions = versions.mix(ENSEMBLVEP_DOWNLOAD.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(PREPARE_REFERENCE_AND_INTERVALS.out.versions)\n",
      "Nextflow operator map with content: intervals_and_num_intervals = intervals.map{ interval, num_intervals ->\n",
      "Nextflow operator mix with content: reports = reports.mix(BAM_ALIGN.out.reports)\n",
      "Nextflow operator mix with content: versions = versions.mix(BAM_ALIGN.out.versions)\n",
      "Nextflow operator mix with content: reports      = reports.mix(BAM_PROCESSING.out.reports)\n",
      "Nextflow operator mix with content: versions     = versions.mix(BAM_PROCESSING.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(PREPARE_REALIGNMENT.out.versions)\n",
      "Nextflow operator mix with content: reports                = reports.mix(REALIGNMENT.out.reports)\n",
      "Nextflow operator mix with content: versions               = versions.mix(REALIGNMENT.out.versions)\n",
      "Nextflow operator dump with content: filtered_maf.dump(tag:\"filtered_maf\")\n",
      "Nextflow operator dump with content: realigned_filtered_maf.dump(tag:\"realigned_filtered_maf\")\n",
      "Nextflow operator branch with content: filtered_maf.branch{  dna:  it[0].status < 2\n",
      "Nextflow operator branch with content: realigned_filtered_maf.branch{  dna:  it[0].status < 2\n",
      "Nextflow operator mix with content: versions = versions.mix(MAF_FILTERING_RNA.out.versions)\n",
      "Nextflow operator collectFile with content: CUSTOM_DUMPSOFTWAREVERSIONS(versions.unique().collectFile(name: 'collated_versions.yml'))\n",
      "Nextflow operator collect with content: version_yaml = CUSTOM_DUMPSOFTWAREVERSIONS.out.mqc_yml.collect()\n",
      "Nextflow operator mix with content: multiqc_files = multiqc_files.mix(version_yaml)\n",
      "Nextflow operator collectFile with content: multiqc_files = multiqc_files.mix(ch_workflow_summary.collectFile(name: 'workflow_summary_mqc.yaml'))\n",
      "Nextflow operator collectFile with content: multiqc_files = multiqc_files.mix(ch_methods_description.collectFile(name: 'methods_description_mqc.yaml'))\n",
      "Nextflow operator collect with content: multiqc_files = multiqc_files.mix(reports.collect().ifEmpty([]))\n",
      "Nextflow operator collect with content: MULTIQC(multiqc_files.collect(), ch_multiqc_config.collect().ifEmpty([]), ch_multiqc_custom_config.collect().ifEmpty([]), ch_multiqc_logo.collect().ifEmpty([]))\n",
      "Nextflow operator toList with content: multiqc_report = MULTIQC.out.report.toList()\n",
      "Nextflow operator mix with content: versions = versions.mix(MULTIQC.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\rnadnavar\\workflows\\rnadnavar.nf\n",
      "Process NFCORE_RNAFUSION in filepath ./git_repos\\rnafusion\\main.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\rnafusion\\main.nf\n",
      "Process ARRIBA_WORKFLOW in filepath ./git_repos\\rnafusion\\subworkflows\\local\\arriba_workflow.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(STAR_FOR_ARRIBA.out.versions)\n",
      "Nextflow operator combine with content: ch_arriba_fusions = reads.combine( Channel.value( file( params.arriba_fusions, checkIfExists: true ) ) )\n",
      "Nextflow operator map with content: .map { meta, reads, fusions -> [ meta, fusions ] }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ARRIBA.out.versions)\n",
      "Nextflow operator map with content: ch_arriba_fusion_fail = ARRIBA.out.fusions_fail.map{ meta, file -> return file}\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_SORT_FOR_ARRIBA.out.versions )\n",
      "Nextflow operator map with content: SAMTOOLS_VIEW_FOR_ARRIBA(SAMTOOLS_SORT_FOR_ARRIBA.out.bam.map { meta, bam -> [ meta, bam, [] ] }, ch_fasta, [])\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_VIEW_FOR_ARRIBA.out.versions )\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_INDEX_FOR_ARRIBA.out.versions )\n",
      "Nextflow operator combine with content: ch_arriba_fusions       = reads.combine(Channel.value( file(ch_dummy_file, checkIfExists:true ) ) )\n",
      "Nextflow operator map with content: .map { meta, reads, fusions -> [ meta, fusions ] }\n",
      "Error: Unclosed process block in file: ./git_repos\\rnafusion\\subworkflows\\local\\arriba_workflow.nf\n",
      "Process FUSIONCATCHER_WORKFLOW in filepath ./git_repos\\rnafusion\\subworkflows\\local\\fusioncatcher_workflow.nf\n",
      "Nextflow operator combine with content: ch_fusioncatcher_fusions = reads.combine(Channel.value(file(params.fusioncatcher_fusions, checkIfExists:true)))\n",
      "Nextflow operator map with content: .map { meta, reads, fusions -> [ meta, fusions ] }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(FUSIONCATCHER.out.versions)\n",
      "Nextflow operator combine with content: ch_fusioncatcher_fusions = reads.combine(Channel.value(file(ch_dummy_file, checkIfExists:true)))\n",
      "Nextflow operator map with content: .map { meta, reads, fusions -> [ meta, fusions ] }\n",
      "Error: Unclosed process block in file: ./git_repos\\rnafusion\\subworkflows\\local\\fusioncatcher_workflow.nf\n",
      "Process FUSIONINSPECTOR_WORKFLOW in filepath ./git_repos\\rnafusion\\subworkflows\\local\\fusioninspector_workflow.nf\n",
      "Nextflow operator branch with content: .branch{\n",
      "Nextflow operator combine with content: ch_whitelist = ch_fusion_list.fusions.combine(Channel.value(file(params.whitelist, checkIfExists:true)))\n",
      "Nextflow operator map with content: .map { meta, fusions, whitelist -> [ meta, [fusions, whitelist] ] }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(CAT_CAT.out.versions)\n",
      "Nextflow operator join with content: ch_reads_fusion = reads.join(ch_fusion_list.fusions )\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(FUSIONINSPECTOR.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(AGAT_CONVERTSPGFF2TSV.out.versions)\n",
      "Nextflow operator join with content: fusion_data = FUSIONINSPECTOR.out.tsv_coding_effect.join(AGAT_CONVERTSPGFF2TSV.out.tsv).join(fusionreport_out).join(fusionreport_csv)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(VCF_COLLECT.out.versions)\n",
      "Nextflow operator join with content: ch_bam_sorted_indexed_fusions = bam_sorted_indexed.join(FUSIONINSPECTOR.out.tsv)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ARRIBA_VISUALISATION.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\rnafusion\\subworkflows\\local\\fusioninspector_workflow.nf\n",
      "Process FUSIONREPORT_WORKFLOW in filepath ./git_repos\\rnafusion\\subworkflows\\local\\fusionreport_workflow.nf\n",
      "Nextflow operator join with content: .join(arriba_fusions, remainder: true)\n",
      "Nextflow operator join with content: .join(starfusion_fusions, remainder: true)\n",
      "Nextflow operator join with content: .join(fusioncatcher_fusions, remainder: true)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(FUSIONREPORT.out.versions)\n",
      "Nextflow operator combine with content: ch_fusion_list = reads.combine(Channel.value(file(params.fusioninspector_fusions, checkIfExists:true)))\n",
      "Nextflow operator map with content: .map { meta, reads, fusions -> [ meta, fusions ] }\n",
      "Nextflow operator ifEmpty with content: report                   = ch_report.ifEmpty(null)\n",
      "Nextflow operator ifEmpty with content: csv                      = ch_csv.ifEmpty(null)\n",
      "Error: Unclosed process block in file: ./git_repos\\rnafusion\\subworkflows\\local\\fusionreport_workflow.nf\n",
      "Process INPUT_CHECK in filepath ./git_repos\\rnafusion\\subworkflows\\local\\input_check.nf\n",
      "Nextflow operator splitCsv with content: .splitCsv ( header:true, sep:',' )\n",
      "Nextflow operator map with content: .map { create_fastq_channel(it) }\n",
      "Nextflow operator set with content: .set { reads }\n",
      "Error: Unclosed process block in file: ./git_repos\\rnafusion\\subworkflows\\local\\input_check.nf\n",
      "Process QC_WORKFLOW in filepath ./git_repos\\rnafusion\\subworkflows\\local\\qc_workflow.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(PICARD_COLLECTRNASEQMETRICS.out.versions)\n",
      "Nextflow operator mix with content: ch_rnaseq_metrics = Channel.empty().mix(PICARD_COLLECTRNASEQMETRICS.out.metrics)\n",
      "Nextflow operator map with content: GATK4_MARKDUPLICATES(ch_bam_sorted, ch_fasta.map { meta, fasta -> [ fasta ]}, ch_fai.map { meta, fasta_fai -> [ fasta_fai ]})\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GATK4_MARKDUPLICATES.out.versions)\n",
      "Nextflow operator mix with content: ch_duplicate_metrics = Channel.empty().mix(GATK4_MARKDUPLICATES.out.metrics)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(PICARD_COLLECTINSERTSIZEMETRICS.out.versions)\n",
      "Nextflow operator mix with content: ch_insertsize_metrics = Channel.empty().mix(PICARD_COLLECTINSERTSIZEMETRICS.out.metrics)\n",
      "Error: Unclosed process block in file: ./git_repos\\rnafusion\\subworkflows\\local\\qc_workflow.nf\n",
      "Process STARFUSION_WORKFLOW in filepath ./git_repos\\rnafusion\\subworkflows\\local\\starfusion_workflow.nf\n",
      "Nextflow operator combine with content: ch_starfusion_fusions = reads.combine(Channel.value(file(params.starfusion_fusions, checkIfExists:true)))\n",
      "Nextflow operator map with content: .map { meta, reads, fusions -> [ meta, fusions ] }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(STAR_FOR_STARFUSION.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_INDEX_FOR_STARFUSION.out.versions)\n",
      "Nextflow operator join with content: bam_sorted_indexed = STAR_FOR_STARFUSION.out.bam_sorted.join(SAMTOOLS_INDEX_FOR_STARFUSION.out.bai)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_VIEW_FOR_STARFUSION.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_INDEX_FOR_STARFUSION_CRAM.out.versions)\n",
      "Nextflow operator join with content: reads_junction = reads.join(STAR_FOR_STARFUSION.out.junction )\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(STARFUSION.out.versions)\n",
      "Nextflow operator combine with content: ch_starfusion_fusions = reads.combine(Channel.value(file(ch_dummy_file, checkIfExists:true)))\n",
      "Nextflow operator map with content: .map { meta, reads, fusions -> [ meta, fusions ] }\n",
      "Nextflow operator ifEmpty with content: ch_bam_sorted         = ch_align.ifEmpty([[],[]])\n",
      "Nextflow operator ifEmpty with content: ch_bam_sorted_indexed = bam_sorted_indexed.ifEmpty([[],[],[]])\n",
      "Error: Unclosed process block in file: ./git_repos\\rnafusion\\subworkflows\\local\\starfusion_workflow.nf\n",
      "Process STRINGTIE_WORKFLOW in filepath ./git_repos\\rnafusion\\subworkflows\\local\\stringtie_workflow.nf\n",
      "Nextflow operator map with content: STRINGTIE_STRINGTIE(bam_sorted, ch_chrgtf.map { meta, gtf -> [ gtf ]})\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(STRINGTIE_STRINGTIE.out.versions)\n",
      "Nextflow operator map with content: .map { it -> it[1] }\n",
      "Nextflow operator set with content: .set { stringtie_gtf }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(STRINGTIE_STRINGTIE.out.versions)\n",
      "Nextflow operator map with content: STRINGTIE_MERGE (stringtie_gtf, ch_chrgtf.map { meta, gtf -> [ gtf ]})\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(STRINGTIE_MERGE.out.versions)\n",
      "Nextflow operator ifEmpty with content: stringtie_gtf = ch_stringtie_gtf.ifEmpty(null)\n",
      "Error: Unclosed process block in file: ./git_repos\\rnafusion\\subworkflows\\local\\stringtie_workflow.nf\n",
      "Process TRIM_WORKFLOW in filepath ./git_repos\\rnafusion\\subworkflows\\local\\trim_workflow.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(FASTP.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(FASTQC_FOR_FASTP.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\rnafusion\\subworkflows\\local\\trim_workflow.nf\n",
      "Process BUILD_REFERENCES in filepath ./git_repos\\rnafusion\\workflows\\build_references.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\rnafusion\\workflows\\build_references.nf\n",
      "Process RNAFUSION in filepath ./git_repos\\rnafusion\\workflows\\rnafusion.nf\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator join with content: meta_clone.id = meta_clone.id.split('_')[0..-2].join('_')\n",
      "Nextflow operator groupTuple with content: .groupTuple(by: [0])\n",
      "Nextflow operator branch with content: .branch {\n",
      "Nextflow operator flatten with content: return [ meta, fastq.flatten() ]\n",
      "Nextflow operator flatten with content: return [ meta, fastq.flatten() ]\n",
      "Nextflow operator set with content: .set { ch_fastq }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(INPUT_CHECK.out.versions)\n",
      "Nextflow operator mix with content: .mix(ch_fastq.single)\n",
      "Nextflow operator set with content: .set { ch_cat_fastq }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(CAT_FASTQ.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(FASTQC.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(TRIM_WORKFLOW.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ARRIBA_WORKFLOW.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(STARFUSION_WORKFLOW.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(FUSIONCATCHER_WORKFLOW.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(STRINGTIE_WORKFLOW.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(FUSIONREPORT_WORKFLOW.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(FUSIONINSPECTOR_WORKFLOW.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(QC_WORKFLOW.out.versions)\n",
      "Nextflow operator collectFile with content: ch_versions.unique().collectFile(name: 'collated_versions.yml')\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_workflow_summary.collectFile(name: 'workflow_summary_mqc.yaml'))\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_methods_description.collectFile(name: 'methods_description_mqc.yaml'))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(CUSTOM_DUMPSOFTWAREVERSIONS.out.mqc_yml.collect())\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(FASTQC.out.zip.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(TRIM_WORKFLOW.out.ch_fastp_html.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(TRIM_WORKFLOW.out.ch_fastp_json.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(TRIM_WORKFLOW.out.ch_fastqc_trimmed.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(STARFUSION_WORKFLOW.out.star_stats.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(STARFUSION_WORKFLOW.out.star_gene_count.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(QC_WORKFLOW.out.rnaseq_metrics.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(QC_WORKFLOW.out.duplicate_metrics.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(QC_WORKFLOW.out.insertsize_metrics.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(FUSIONINSPECTOR_WORKFLOW.out.ch_arriba_visualisation.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files.collect(),\n",
      "Nextflow operator toList with content: ch_multiqc_config.toList(),\n",
      "Nextflow operator toList with content: ch_multiqc_custom_config.toList(),\n",
      "Nextflow operator toList with content: ch_multiqc_logo.toList()\n",
      "Nextflow operator toList with content: multiqc_report       = MULTIQC.out.report.toList()\n",
      "Nextflow operator mix with content: ch_versions          = ch_versions.mix(MULTIQC.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\rnafusion\\workflows\\rnafusion.nf\n",
      "Process NFCORE_RNASEQ in filepath ./git_repos\\rnaseq\\main.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\rnaseq\\main.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\rnaseq\\conf\\modules.config\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Process ALIGN_STAR in filepath ./git_repos\\rnaseq\\subworkflows\\local\\align_star\\main.nf\n",
      "Nextflow operator first with content: ch_versions       = ch_versions.mix(STAR_ALIGN_IGENOMES.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions       = ch_versions.mix(STAR_ALIGN.out.versions.first())\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BAM_SORT_STATS_SAMTOOLS.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\rnaseq\\subworkflows\\local\\align_star\\main.nf\n",
      "Process PREPARE_GENOME in filepath ./git_repos\\rnaseq\\subworkflows\\local\\prepare_genome\\main.nf\n",
      "Nextflow operator map with content: ch_fasta    = GUNZIP_FASTA ( [ [:], fasta ] ).gunzip.map { it[1] }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GUNZIP_FASTA.out.versions)\n",
      "Nextflow operator map with content: ch_gtf      = GUNZIP_GTF ( [ [:], gtf ] ).gunzip.map { it[1] }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GUNZIP_GTF.out.versions)\n",
      "Nextflow operator map with content: ch_gff      = GUNZIP_GFF ( [ [:], gff ] ).gunzip.map { it[1] }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GUNZIP_GFF.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GFFREAD.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GTF_FILTER.out.versions)\n",
      "Nextflow operator map with content: ch_add_fasta = GUNZIP_ADDITIONAL_FASTA ( [ [:], additional_fasta ] ).gunzip.map { it[1] }\n",
      "Nextflow operator mix with content: ch_versions  = ch_versions.mix(GUNZIP_ADDITIONAL_FASTA.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(CAT_ADDITIONAL_FASTA.out.versions)\n",
      "Nextflow operator map with content: ch_gene_bed = GUNZIP_GENE_BED ( [ [:], gene_bed ] ).gunzip.map { it[1] }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GUNZIP_GENE_BED.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GTF2BED.out.versions)\n",
      "Nextflow operator map with content: ch_transcript_fasta = GUNZIP_TRANSCRIPT_FASTA ( [ [:], transcript_fasta ] ).gunzip.map { it[1] }\n",
      "Nextflow operator mix with content: ch_versions         = ch_versions.mix(GUNZIP_TRANSCRIPT_FASTA.out.versions)\n",
      "Nextflow operator mix with content: ch_versions         = ch_versions.mix(PREPROCESS_TRANSCRIPTS_FASTA_GENCODE.out.versions)\n",
      "Nextflow operator mix with content: ch_versions         = ch_versions.mix(MAKE_TRANSCRIPTS_FASTA.out.versions)\n",
      "Nextflow operator map with content: CUSTOM_GETCHROMSIZES ( ch_fasta.map { [ [:], it ] } )\n",
      "Nextflow operator map with content: ch_fai         = CUSTOM_GETCHROMSIZES.out.fai.map { it[1] }\n",
      "Nextflow operator map with content: ch_chrom_sizes = CUSTOM_GETCHROMSIZES.out.sizes.map { it[1] }\n",
      "Nextflow operator mix with content: ch_versions    = ch_versions.mix(CUSTOM_GETCHROMSIZES.out.versions)\n",
      "Nextflow operator map with content: ch_bbsplit_index = UNTAR_BBSPLIT_INDEX ( [ [:], bbsplit_index ] ).untar.map { it[1] }\n",
      "Nextflow operator mix with content: ch_versions      = ch_versions.mix(UNTAR_BBSPLIT_INDEX.out.versions)\n",
      "Nextflow operator splitCsv with content: .splitCsv() // Read in 2 column csv file: short_name,path_to_fasta\n",
      "Nextflow operator flatMap with content: .flatMap { id, fasta -> [ [ 'id', id ], [ 'fasta', file(fasta, checkIfExists: true) ] ] } // Flatten entries to be able to groupTuple by a common key\n",
      "Nextflow operator groupTuple with content: .groupTuple()\n",
      "Nextflow operator map with content: .map { it -> it[1] } // Get rid of keys and keep grouped values\n",
      "Nextflow operator collect with content: .collect { [ it ] } // Collect entries as a list to pass as \"tuple val(short_names), path(path_to_fasta)\" to module\n",
      "Nextflow operator set with content: .set { ch_bbsplit_fasta_list }\n",
      "Nextflow operator mix with content: ch_versions      = ch_versions.mix(BBMAP_BBSPLIT.out.versions)\n",
      "Nextflow operator map with content: ch_star_index = UNTAR_STAR_INDEX ( [ [:], star_index ] ).untar.map { it[1] }\n",
      "Nextflow operator mix with content: ch_versions   = ch_versions.mix(UNTAR_STAR_INDEX.out.versions)\n",
      "Nextflow operator mix with content: ch_versions   = ch_versions.mix(STAR_GENOMEGENERATE_IGENOMES.out.versions)\n",
      "Nextflow operator map with content: ch_star_index = STAR_GENOMEGENERATE ( ch_fasta.map { [ [:], it ] }, ch_gtf.map { [ [:], it ] } ).index.map { it[1] }\n",
      "Nextflow operator mix with content: ch_versions   = ch_versions.mix(STAR_GENOMEGENERATE.out.versions)\n",
      "Nextflow operator map with content: ch_rsem_index = UNTAR_RSEM_INDEX ( [ [:], rsem_index ] ).untar.map { it[1] }\n",
      "Nextflow operator mix with content: ch_versions   = ch_versions.mix(UNTAR_RSEM_INDEX.out.versions)\n",
      "Nextflow operator mix with content: ch_versions   = ch_versions.mix(RSEM_PREPAREREFERENCE_GENOME.out.versions)\n",
      "Nextflow operator map with content: ch_splicesites = HISAT2_EXTRACTSPLICESITES ( ch_gtf.map { [ [:], it ] } ).txt.map { it[1] }\n",
      "Nextflow operator mix with content: ch_versions    = ch_versions.mix(HISAT2_EXTRACTSPLICESITES.out.versions)\n",
      "Nextflow operator map with content: ch_hisat2_index = UNTAR_HISAT2_INDEX ( [ [:], hisat2_index ] ).untar.map { it[1] }\n",
      "Nextflow operator mix with content: ch_versions     = ch_versions.mix(UNTAR_HISAT2_INDEX.out.versions)\n",
      "Nextflow operator map with content: ch_hisat2_index = HISAT2_BUILD ( ch_fasta.map { [ [:], it ] }, ch_gtf.map { [ [:], it ] }, ch_splicesites.map { [ [:], it ] } ).index.map { it[1] }\n",
      "Nextflow operator mix with content: ch_versions     = ch_versions.mix(HISAT2_BUILD.out.versions)\n",
      "Nextflow operator map with content: ch_salmon_index = UNTAR_SALMON_INDEX ( [ [:], salmon_index ] ).untar.map { it[1] }\n",
      "Nextflow operator mix with content: ch_versions     = ch_versions.mix(UNTAR_SALMON_INDEX.out.versions)\n",
      "Nextflow operator mix with content: ch_versions     = ch_versions.mix(SALMON_INDEX.out.versions)\n",
      "Nextflow operator mix with content: ch_versions     = ch_versions.mix(UNTAR_KALLISTO_INDEX.out.versions)\n",
      "Nextflow operator map with content: ch_kallisto_index = KALLISTO_INDEX ( ch_transcript_fasta.map{[ [:], it]} ).index\n",
      "Nextflow operator mix with content: ch_versions     = ch_versions.mix(KALLISTO_INDEX.out.versions)\n",
      "Nextflow operator ifEmpty with content: versions         = ch_versions.ifEmpty(null) // channel: [ versions.yml ]\n",
      "Error: Unclosed process block in file: ./git_repos\\rnaseq\\subworkflows\\local\\prepare_genome\\main.nf\n",
      "Process QUANTIFY_PSEUDO_ALIGNMENT in filepath ./git_repos\\rnaseq\\subworkflows\\local\\quantify_pseudo\\main.nf\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SALMON_QUANT.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(KALLISTO_QUANT.out.versions.first())\n",
      "Nextflow operator collect with content: TX2GENE ( ch_pseudo_results.collect{it[1]}, pseudo_aligner, gtf )\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(TX2GENE.out.versions)\n",
      "Nextflow operator collect with content: TXIMPORT ( ch_pseudo_results.collect{it[1]}, TX2GENE.out.tsv.collect(), pseudo_aligner )\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(TXIMPORT.out.versions)\n",
      "Nextflow operator collect with content: TX2GENE.out.tsv.collect()\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SE_GENE.out.versions)\n",
      "Nextflow operator collect with content: TX2GENE.out.tsv.collect()\n",
      "Nextflow operator collect with content: TX2GENE.out.tsv.collect()\n",
      "Nextflow operator collect with content: TX2GENE.out.tsv.collect()\n",
      "Error: Unclosed process block in file: ./git_repos\\rnaseq\\subworkflows\\local\\quantify_pseudo\\main.nf\n",
      "Process QUANTIFY_RSEM in filepath ./git_repos\\rnaseq\\subworkflows\\local\\quantify_rsem\\main.nf\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(RSEM_CALCULATEEXPRESSION.out.versions.first())\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BAM_SORT_STATS_SAMTOOLS.out.versions)\n",
      "Nextflow operator collect with content: RSEM_CALCULATEEXPRESSION.out.counts_gene.collect{it[1]},       // [meta, counts]: Collect the second element (counts files) in the channel across all samples\n",
      "Nextflow operator collect with content: RSEM_CALCULATEEXPRESSION.out.counts_transcript.collect{it[1]}\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(RSEM_MERGE_COUNTS.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\rnaseq\\subworkflows\\local\\quantify_rsem\\main.nf\n",
      "Process BAM_DEDUP_STATS_SAMTOOLS_UMITOOLS in filepath ./git_repos\\rnaseq\\subworkflows\\nf-core\\bam_dedup_stats_samtools_umitools\\main.nf\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(UMITOOLS_DEDUP.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SAMTOOLS_INDEX.out.versions.first())\n",
      "Nextflow operator join with content: .join(SAMTOOLS_INDEX.out.bai, by: [0], remainder: true)\n",
      "Nextflow operator join with content: .join(SAMTOOLS_INDEX.out.csi, by: [0], remainder: true)\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BAM_STATS_SAMTOOLS.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\rnaseq\\subworkflows\\nf-core\\bam_dedup_stats_samtools_umitools\\main.nf\n",
      "Process BAM_MARKDUPLICATES_PICARD in filepath ./git_repos\\rnaseq\\subworkflows\\nf-core\\bam_markduplicates_picard\\main.nf\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(PICARD_MARKDUPLICATES.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SAMTOOLS_INDEX.out.versions.first())\n",
      "Nextflow operator join with content: .join(SAMTOOLS_INDEX.out.bai, by: [0], remainder: true)\n",
      "Nextflow operator join with content: .join(SAMTOOLS_INDEX.out.csi, by: [0], remainder: true)\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BAM_STATS_SAMTOOLS.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\rnaseq\\subworkflows\\nf-core\\bam_markduplicates_picard\\main.nf\n",
      "Process BAM_RSEQC in filepath ./git_repos\\rnaseq\\subworkflows\\nf-core\\bam_rseqc\\main.nf\n",
      "Nextflow operator map with content: .map { [ it[0], it[1] ] }\n",
      "Nextflow operator set with content: .set { ch_bam }\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(RSEQC_BAMSTAT.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(RSEQC_INNERDISTANCE.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(RSEQC_INFEREXPERIMENT.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(RSEQC_JUNCTIONANNOTATION.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(RSEQC_JUNCTIONSATURATION.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(RSEQC_READDISTRIBUTION.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(RSEQC_READDUPLICATION.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(RSEQC_TIN.out.versions.first())\n",
      "Error: Unclosed process block in file: ./git_repos\\rnaseq\\subworkflows\\nf-core\\bam_rseqc\\main.nf\n",
      "Process BAM_SORT_STATS_SAMTOOLS in filepath ./git_repos\\rnaseq\\subworkflows\\nf-core\\bam_sort_stats_samtools\\main.nf\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SAMTOOLS_SORT.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SAMTOOLS_INDEX.out.versions.first())\n",
      "Nextflow operator join with content: .join(SAMTOOLS_INDEX.out.bai, by: [0], remainder: true)\n",
      "Nextflow operator join with content: .join(SAMTOOLS_INDEX.out.csi, by: [0], remainder: true)\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator set with content: .set { ch_bam_bai }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BAM_STATS_SAMTOOLS.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\rnaseq\\subworkflows\\nf-core\\bam_sort_stats_samtools\\main.nf\n",
      "Process BAM_STATS_SAMTOOLS in filepath ./git_repos\\rnaseq\\subworkflows\\nf-core\\bam_stats_samtools\\main.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_STATS.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_FLAGSTAT.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_IDXSTATS.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\rnaseq\\subworkflows\\nf-core\\bam_stats_samtools\\main.nf\n",
      "Process BEDGRAPH_BEDCLIP_BEDGRAPHTOBIGWIG in filepath ./git_repos\\rnaseq\\subworkflows\\nf-core\\bedgraph_bedclip_bedgraphtobigwig\\main.nf\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(UCSC_BEDCLIP.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(UCSC_BEDGRAPHTOBIGWIG.out.versions.first())\n",
      "Error: Unclosed process block in file: ./git_repos\\rnaseq\\subworkflows\\nf-core\\bedgraph_bedclip_bedgraphtobigwig\\main.nf\n",
      "Process FASTQ_ALIGN_HISAT2 in filepath ./git_repos\\rnaseq\\subworkflows\\nf-core\\fastq_align_hisat2\\main.nf\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(HISAT2_ALIGN.out.versions.first())\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BAM_SORT_STATS_SAMTOOLS.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\rnaseq\\subworkflows\\nf-core\\fastq_align_hisat2\\main.nf\n",
      "Process FASTQ_FASTQC_UMITOOLS_FASTP in filepath ./git_repos\\rnaseq\\subworkflows\\nf-core\\fastq_fastqc_umitools_fastp\\main.nf\n",
      "Nextflow operator first with content: ch_versions     = ch_versions.mix(FASTQC_RAW.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(UMITOOLS_EXTRACT.out.versions.first())\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator set with content: .set { umi_reads }\n",
      "Nextflow operator first with content: ch_versions       = ch_versions.mix(FASTP.out.versions.first())\n",
      "Nextflow operator join with content: .join(trim_json)\n",
      "Nextflow operator map with content: .map { meta, reads, json -> [ meta, reads, getFastpReadsAfterFiltering(json) ] }\n",
      "Nextflow operator set with content: .set { ch_num_trimmed_reads }\n",
      "Nextflow operator filter with content: .filter { meta, reads, num_reads -> num_reads >= min_trimmed_reads.toLong() }\n",
      "Nextflow operator map with content: .map { meta, reads, num_reads -> [ meta, reads ] }\n",
      "Nextflow operator set with content: .set { trim_reads }\n",
      "Nextflow operator map with content: .map { meta, reads, num_reads -> [ meta, num_reads ] }\n",
      "Nextflow operator set with content: .set { trim_read_count }\n",
      "Nextflow operator first with content: ch_versions      = ch_versions.mix(FASTQC_TRIM.out.versions.first())\n",
      "Nextflow operator ifEmpty with content: versions = ch_versions.ifEmpty(null) // channel: [ versions.yml ]\n",
      "Error: Unclosed process block in file: ./git_repos\\rnaseq\\subworkflows\\nf-core\\fastq_fastqc_umitools_fastp\\main.nf\n",
      "Process FASTQ_FASTQC_UMITOOLS_TRIMGALORE in filepath ./git_repos\\rnaseq\\subworkflows\\nf-core\\fastq_fastqc_umitools_trimgalore\\main.nf\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(FASTQC.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(UMITOOLS_EXTRACT.out.versions.first())\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator set with content: .set { umi_reads }\n",
      "Nextflow operator first with content: ch_versions   = ch_versions.mix(TRIMGALORE.out.versions.first())\n",
      "Nextflow operator join with content: .join(trim_log, remainder: true)\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator set with content: .set { ch_num_trimmed_reads }\n",
      "Nextflow operator filter with content: .filter { meta, reads, num_reads -> num_reads >= min_trimmed_reads.toFloat() }\n",
      "Nextflow operator map with content: .map { meta, reads, num_reads -> [ meta, reads ] }\n",
      "Nextflow operator set with content: .set { trim_reads }\n",
      "Nextflow operator map with content: .map { meta, reads, num_reads -> [ meta, num_reads ] }\n",
      "Nextflow operator set with content: .set { trim_read_count }\n",
      "Nextflow operator ifEmpty with content: versions = ch_versions.ifEmpty(null) // channel: [ versions.yml ]\n",
      "Error: Unclosed process block in file: ./git_repos\\rnaseq\\subworkflows\\nf-core\\fastq_fastqc_umitools_trimgalore\\main.nf\n",
      "Process FASTQ_SUBSAMPLE_FQ_SALMON in filepath ./git_repos\\rnaseq\\subworkflows\\nf-core\\fastq_subsample_fq_salmon\\main.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SALMON_INDEX.out.versions)\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(FQ_SUBSAMPLE.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SALMON_QUANT.out.versions.first())\n",
      "Error: Unclosed process block in file: ./git_repos\\rnaseq\\subworkflows\\nf-core\\fastq_subsample_fq_salmon\\main.nf\n",
      "Process RNASEQ in filepath ./git_repos\\rnaseq\\workflows\\rnaseq.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(PREPARE_GENOME.out.versions)\n",
      "Nextflow operator map with content: .map { WorkflowRnaseq.checkMaxContigSize(it, log) }\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator groupTuple with content: .groupTuple()\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator branch with content: .branch {\n",
      "Nextflow operator flatten with content: return [ meta, fastqs.flatten() ]\n",
      "Nextflow operator flatten with content: return [ meta, fastqs.flatten() ]\n",
      "Nextflow operator set with content: .set { ch_fastq }\n",
      "Nextflow operator mix with content: .mix(ch_fastq.single)\n",
      "Nextflow operator set with content: .set { ch_cat_fastq }\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(CAT_FASTQ.out.versions.first().ifEmpty(null))\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(FASTQ_FASTQC_UMITOOLS_TRIMGALORE.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(FASTQ_FASTQC_UMITOOLS_FASTP.out.versions)\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator collect with content: .collect()\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator set with content: .set { ch_fail_trimming_multiqc }\n",
      "Nextflow operator set with content: .set { ch_filtered_reads }\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(BBMAP_BBSPLIT.out.versions.first())\n",
      "Nextflow operator collect with content: ch_sortmerna_fastas = Channel.from(ch_ribo_db.readLines()).map { row -> file(row, checkIfExists: true) }.collect()\n",
      "Nextflow operator set with content: .set { ch_filtered_reads }\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SORTMERNA.out.versions.first())\n",
      "Nextflow operator branch with content: .branch {\n",
      "Nextflow operator set with content: .set { ch_strand_fastq }\n",
      "Nextflow operator combine with content: .combine(ch_strand_fastq.auto_strand)\n",
      "Nextflow operator first with content: .map { it.first() }\n",
      "Nextflow operator first with content: .first()\n",
      "Nextflow operator set with content: .set { ch_genome_fasta }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(FASTQ_SUBSAMPLE_FQ_SALMON.out.versions)\n",
      "Nextflow operator join with content: .join(ch_strand_fastq.auto_strand)\n",
      "Nextflow operator map with content: .map { meta, json, reads ->\n",
      "Nextflow operator mix with content: .mix(ch_strand_fastq.known_strand)\n",
      "Nextflow operator set with content: .set { ch_strand_inferred_filtered_fastq }\n",
      "Nextflow operator map with content: PREPARE_GENOME.out.star_index.map { [ [:], it ] },\n",
      "Nextflow operator map with content: PREPARE_GENOME.out.gtf.map { [ [:], it ] },\n",
      "Nextflow operator map with content: PREPARE_GENOME.out.fasta.map { [ [:], it ] }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ALIGN_STAR.out.versions)\n",
      "Nextflow operator join with content: ch_genome_bam.join(ch_genome_bam_index, by: [0]),\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BAM_DEDUP_STATS_SAMTOOLS_UMITOOLS_GENOME.out.versions)\n",
      "Nextflow operator map with content: PREPARE_GENOME.out.fasta.map { [ [:], it ] }\n",
      "Nextflow operator join with content: ch_transcriptome_sorted_bam.join(ch_transcriptome_sorted_bai, by: [0]),\n",
      "Nextflow operator branch with content: .branch {\n",
      "Nextflow operator set with content: .set { ch_umitools_dedup_bam }\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(UMITOOLS_PREPAREFORSALMON.out.versions.first())\n",
      "Nextflow operator mix with content: .mix(UMITOOLS_PREPAREFORSALMON.out.bam)\n",
      "Nextflow operator set with content: .set { ch_transcriptome_bam }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(QUANTIFY_STAR_SALMON.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(DESEQ2_QC_STAR_SALMON.out.versions)\n",
      "Nextflow operator map with content: PREPARE_GENOME.out.fasta.map { [ [:], it ] }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(QUANTIFY_RSEM.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(DESEQ2_QC_RSEM.out.versions)\n",
      "Nextflow operator map with content: PREPARE_GENOME.out.hisat2_index.map { [ [:], it ] },\n",
      "Nextflow operator map with content: PREPARE_GENOME.out.splicesites.map { [ [:], it ] },\n",
      "Nextflow operator map with content: PREPARE_GENOME.out.fasta.map { [ [:], it ] }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(FASTQ_ALIGN_HISAT2.out.versions)\n",
      "Nextflow operator join with content: ch_genome_bam.join(ch_genome_bam_index, by: [0]),\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BAM_DEDUP_STATS_SAMTOOLS_UMITOOLS_GENOME.out.versions)\n",
      "Nextflow operator map with content: .map { meta, align_log -> [ meta ] + WorkflowRnaseq.getStarPercentMapped(params, align_log) }\n",
      "Nextflow operator set with content: .set { ch_percent_mapped }\n",
      "Nextflow operator join with content: .join(ch_percent_mapped, by: [0])\n",
      "Nextflow operator map with content: .map { meta, ofile, mapped, pass -> if (pass) [ meta, ofile ] }\n",
      "Nextflow operator set with content: .set { ch_genome_bam }\n",
      "Nextflow operator join with content: .join(ch_percent_mapped, by: [0])\n",
      "Nextflow operator map with content: .map { meta, ofile, mapped, pass -> if (pass) [ meta, ofile ] }\n",
      "Nextflow operator set with content: .set { ch_genome_bam_index }\n",
      "Nextflow operator branch with content: .branch { meta, mapped, pass ->\n",
      "Nextflow operator set with content: .set { ch_pass_fail_mapped }\n",
      "Nextflow operator collect with content: .collect()\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator set with content: .set { ch_fail_mapping_multiqc }\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(PRESEQ_LCEXTRAP.out.versions.first())\n",
      "Nextflow operator map with content: PREPARE_GENOME.out.fasta.map { [ [:], it ] },\n",
      "Nextflow operator map with content: PREPARE_GENOME.out.fai.map { [ [:], it ] }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BAM_MARKDUPLICATES_PICARD.out.versions)\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(STRINGTIE_STRINGTIE.out.versions.first())\n",
      "Nextflow operator map with content: .map { WorkflowRnaseq.biotypeInGtf(it, biotype, log) }\n",
      "Nextflow operator set with content: .set { biotype_in_gtf }\n",
      "Nextflow operator combine with content: .combine(PREPARE_GENOME.out.gtf)\n",
      "Nextflow operator combine with content: .combine(biotype_in_gtf)\n",
      "Nextflow operator filter with content: .filter { it[-1] }\n",
      "Nextflow operator map with content: .map { it[0..<it.size()-1] }\n",
      "Nextflow operator set with content: .set { ch_featurecounts }\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SUBREAD_FEATURECOUNTS.out.versions.first())\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Nextflow operator first with content: ch_versions = ch_versions.mix(MULTIQC_CUSTOM_BIOTYPE.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(BEDTOOLS_GENOMECOV.out.versions.first())\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BEDGRAPH_BEDCLIP_BEDGRAPHTOBIGWIG_FORWARD.out.versions)\n",
      "Nextflow operator map with content: PREPARE_GENOME.out.gtf.map { [ [:], it ] }\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(QUALIMAP_RNASEQ.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(DUPRADAR.out.versions.first())\n",
      "Nextflow operator join with content: ch_genome_bam.join(ch_genome_bam_index, by: [0]),\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BAM_RSEQC.out.versions)\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator join with content: return [ \"$meta.id\\t$meta.strandedness\\t${inferred_strand.join('\\t')}\" ]\n",
      "Nextflow operator collect with content: .collect()\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator set with content: .set { ch_fail_strand_multiqc }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(QUANTIFY_PSEUDO_ALIGNMENT.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(DESEQ2_QC_PSEUDO.out.versions)\n",
      "Nextflow operator collectFile with content: ch_versions.unique().collectFile(name: 'collated_versions.yml')\n",
      "Nextflow operator collect with content: ch_multiqc_custom_config.collect().ifEmpty([]),\n",
      "Nextflow operator collect with content: CUSTOM_DUMPSOFTWAREVERSIONS.out.mqc_yml.collect(),\n",
      "Nextflow operator collectFile with content: ch_workflow_summary.collectFile(name: 'workflow_summary_mqc.yaml'),\n",
      "Nextflow operator collectFile with content: ch_methods_description.collectFile(name: 'methods_description_mqc.yaml'),\n",
      "Nextflow operator collect with content: ch_multiqc_logo.collect().ifEmpty([]),\n",
      "Nextflow operator collectFile with content: ch_fail_trimming_multiqc.collectFile(name: 'fail_trimmed_samples_mqc.tsv').ifEmpty([]),\n",
      "Nextflow operator collectFile with content: ch_fail_mapping_multiqc.collectFile(name: 'fail_mapped_samples_mqc.tsv').ifEmpty([]),\n",
      "Nextflow operator collectFile with content: ch_fail_strand_multiqc.collectFile(name: 'fail_strand_check_mqc.tsv').ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_fastqc_raw_multiqc.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_fastqc_trim_multiqc.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_trim_log_multiqc.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_sortmerna_multiqc.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_star_multiqc.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_hisat2_multiqc.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_rsem_multiqc.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_pseudo_multiqc.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_samtools_stats.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_samtools_flagstat.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_samtools_idxstats.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_markduplicates_multiqc.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_featurecounts_multiqc.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_aligner_pca_multiqc.collect().ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_aligner_clustering_multiqc.collect().ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_pseudoaligner_pca_multiqc.collect().ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_pseudoaligner_clustering_multiqc.collect().ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_preseq_multiqc.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_qualimap_multiqc.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_dupradar_multiqc.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_bamstat_multiqc.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_inferexperiment_multiqc.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_innerdistance_multiqc.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_junctionannotation_multiqc.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_junctionsaturation_multiqc.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_readdistribution_multiqc.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_readduplication_multiqc.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_tin_multiqc.collect{it[1]}.ifEmpty([])\n",
      "Nextflow operator toList with content: multiqc_report = MULTIQC.out.report.toList()\n",
      "Error: Unclosed process block in file: ./git_repos\\rnaseq\\workflows\\rnaseq.nf\n",
      "Process NFCORE_RNASPLICE in filepath ./git_repos\\rnasplice\\main.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\rnasplice\\main.nf\n",
      "Process ALIGN_STAR in filepath ./git_repos\\rnasplice\\subworkflows\\local\\align_star.nf\n",
      "Nextflow operator first with content: ch_versions       = ch_versions.mix(STAR_ALIGN_IGENOMES.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions       = ch_versions.mix(STAR_ALIGN.out.versions.first())\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BAM_SORT_STATS_SAMTOOLS.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\rnasplice\\subworkflows\\local\\align_star.nf\n",
      "Process CONTRASTS_CHECK in filepath ./git_repos\\rnasplice\\subworkflows\\local\\contrasts_check.nf\n",
      "Nextflow operator splitCsv with content: .splitCsv ( header:true, sep:',' )\n",
      "Nextflow operator set with content: .set { contrasts }\n",
      "Error: Unclosed process block in file: ./git_repos\\rnasplice\\subworkflows\\local\\contrasts_check.nf\n",
      "Process DEXSEQ_DEU in filepath ./git_repos\\rnasplice\\subworkflows\\local\\dexseq_deu.nf\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(DEXSEQ_ANNOTATION.out.versions.first())\n",
      "Nextflow operator combine with content: ch_genome_bam_gff = ch_genome_bam.combine(ch_dexseq_gff)\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(DEXSEQ_COUNT.out.versions.first())\n",
      "Nextflow operator collect with content: DEXSEQ_COUNT.out.dexseq_clean_txt.map{ it[1] }.collect(),\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(DEXSEQ_EXON.out.versions)\n",
      "Nextflow operator collect with content: dexseq_clean_txt        = DEXSEQ_COUNT.out.dexseq_clean_txt.map{ it[1] }.collect()\n",
      "Error: Unclosed process block in file: ./git_repos\\rnasplice\\subworkflows\\local\\dexseq_deu.nf\n",
      "Process DRIMSEQ_DEXSEQ_DTU in filepath ./git_repos\\rnasplice\\subworkflows\\local\\drimseq_dexseq_dtu.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(DRIMSEQ_FILTER.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(DEXSEQ_DTU.out.versions)\n",
      "Nextflow operator flatten with content: .flatten()\n",
      "Nextflow operator map with content: .map { it ->\n",
      "Nextflow operator flatten with content: .flatten()\n",
      "Nextflow operator map with content: .map { it ->\n",
      "Nextflow operator join with content: ch_dexseq_feature_gene_tsv = ch_dexseq_feature_tsv.join(ch_dexseq_gene_tsv)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(STAGER.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\rnasplice\\subworkflows\\local\\drimseq_dexseq_dtu.nf\n",
      "Process EDGER_DEU in filepath ./git_repos\\rnasplice\\subworkflows\\local\\edger_deu.nf\n",
      "Nextflow operator combine with content: ch_feature_counts = ch_genome_bam.combine(SUBREAD_FLATTENGTF.out.saf)\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SUBREAD_FEATURECOUNTS.out.versions.first())\n",
      "Nextflow operator collect with content: SUBREAD_FEATURECOUNTS.out.counts.collect({it[1]}),\n",
      "Error: Unclosed process block in file: ./git_repos\\rnasplice\\subworkflows\\local\\edger_deu.nf\n",
      "Process INPUT_CHECK in filepath ./git_repos\\rnasplice\\subworkflows\\local\\input_check.nf\n",
      "Nextflow operator splitCsv with content: .splitCsv ( header:true, sep:',' )\n",
      "Nextflow operator map with content: .map { create_fastq_channel(it) }\n",
      "Nextflow operator set with content: .set { reads }\n",
      "Nextflow operator splitCsv with content: .splitCsv ( header:true, sep:',' )\n",
      "Nextflow operator map with content: .map { create_genome_bam_channel(it) }\n",
      "Nextflow operator set with content: .set { reads }\n",
      "Nextflow operator splitCsv with content: .splitCsv ( header:true, sep:',' )\n",
      "Nextflow operator map with content: .map { create_transcriptome_bam_channel(it) }\n",
      "Nextflow operator set with content: .set { reads }\n",
      "Nextflow operator splitCsv with content: .splitCsv ( header:true, sep:',' )\n",
      "Nextflow operator map with content: .map { create_salmon_results_channel(it) }\n",
      "Nextflow operator set with content: .set { reads }\n",
      "Error: Unclosed process block in file: ./git_repos\\rnasplice\\subworkflows\\local\\input_check.nf\n",
      "Process PREPARE_GENOME in filepath ./git_repos\\rnasplice\\subworkflows\\local\\prepare_genome.nf\n",
      "Nextflow operator map with content: ch_fasta    = GUNZIP_FASTA ( [ [:], fasta ] ).gunzip.map { it[1] }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GUNZIP_FASTA.out.versions)\n",
      "Nextflow operator map with content: ch_gtf      = GUNZIP_GTF ( [ [:], gtf ] ).gunzip.map { it[1] }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GUNZIP_GTF.out.versions)\n",
      "Nextflow operator map with content: ch_gff      = GUNZIP_GFF ( [ [:], gff ] ).gunzip.map { it[1] }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GUNZIP_GFF.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GFFREAD.out.versions)\n",
      "Nextflow operator map with content: ch_transcript_fasta = GUNZIP_TRANSCRIPT_FASTA ( [ [:], transcript_fasta ] ).gunzip.map { it[1] }\n",
      "Nextflow operator mix with content: ch_versions         = ch_versions.mix(GUNZIP_TRANSCRIPT_FASTA.out.versions)\n",
      "Nextflow operator mix with content: ch_versions         = ch_versions.mix(PREPROCESS_TRANSCRIPTS_FASTA_GENCODE.out.versions)\n",
      "Nextflow operator mix with content: ch_versions         = ch_versions.mix(GTF_GENE_FILTER.out.versions)\n",
      "Nextflow operator mix with content: ch_versions         = ch_versions.mix(MAKE_TRANSCRIPTS_FASTA.out.versions)\n",
      "Nextflow operator map with content: CUSTOM_GETCHROMSIZES ( ch_fasta.map { [ [:], it ] } )\n",
      "Nextflow operator map with content: ch_fai         = CUSTOM_GETCHROMSIZES.out.fai.map { it[1] }\n",
      "Nextflow operator map with content: ch_chrom_sizes = CUSTOM_GETCHROMSIZES.out.sizes.map { it[1] }\n",
      "Nextflow operator mix with content: ch_versions    = ch_versions.mix(CUSTOM_GETCHROMSIZES.out.versions)\n",
      "Nextflow operator map with content: ch_star_index = UNTAR_STAR_INDEX ( [ [:], star_index ] ).untar.map { it[1] }\n",
      "Nextflow operator mix with content: ch_versions   = ch_versions.mix(UNTAR_STAR_INDEX.out.versions)\n",
      "Nextflow operator mix with content: ch_versions   = ch_versions.mix(STAR_GENOMEGENERATE_IGENOMES.out.versions)\n",
      "Nextflow operator mix with content: ch_versions   = ch_versions.mix(STAR_GENOMEGENERATE.out.versions)\n",
      "Nextflow operator map with content: ch_salmon_index = UNTAR_SALMON_INDEX ( [ [:], salmon_index ] ).untar.map { it[1] }\n",
      "Nextflow operator mix with content: ch_versions     = ch_versions.mix(UNTAR_SALMON_INDEX.out.versions)\n",
      "Nextflow operator mix with content: ch_versions     = ch_versions.mix(SALMON_INDEX.out.versions)\n",
      "Nextflow operator map with content: ch_dexseq_gff = GUNZIP_GFF_DEXSEQ ( [ [:], gff_dexseq ] ).gunzip.map { it[1] }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GUNZIP_GFF_DEXSEQ.out.versions)\n",
      "Nextflow operator map with content: ch_suppa_tpm = GUNZIP_SUPPA_TPM ( [ [:], suppa_tpm ] ).gunzip.map { it[1] }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GUNZIP_SUPPA_TPM.out.versions)\n",
      "Nextflow operator ifEmpty with content: versions         = ch_versions.ifEmpty(null) // channel: [ versions.yml ]\n",
      "Error: Unclosed process block in file: ./git_repos\\rnasplice\\subworkflows\\local\\prepare_genome.nf\n",
      "Process RMATS in filepath ./git_repos\\rnasplice\\subworkflows\\local\\rmats.nf\n",
      "Nextflow operator splitCsv with content: ch_contrasts = ch_contrastsheet.splitCsv(header:true)\n",
      "Nextflow operator map with content: .map { it -> [it['treatment'], it] }\n",
      "Nextflow operator combine with content: .combine ( ch_genome_bam_conditions, by: 0 )\n",
      "Nextflow operator map with content: .map { it -> it[1] + ['meta1': it[2], 'bam1': it[3]] }\n",
      "Nextflow operator map with content: .map { it -> [it['control'], it] }\n",
      "Nextflow operator combine with content: .combine ( ch_genome_bam_conditions, by: 0 )\n",
      "Nextflow operator map with content: .map { it -> it[1] + ['meta2': it[2], 'bam2': it[3]] }\n",
      "Nextflow operator map with content: ch_bam = ch_contrasts.map { [ it.contrast, it.treatment, it.bam1 ] }\n",
      "Nextflow operator map with content: .map { it -> [it['contrast'], it] }\n",
      "Nextflow operator combine with content: .combine ( ch_bamlist, by: 0 )\n",
      "Nextflow operator map with content: .map { it -> it[1] + ['bam1_text': it[2]] }\n",
      "Nextflow operator map with content: ch_contrasts_bamlist = ch_contrasts.map { [ it.contrast, it.treatment, it.meta1, it.bam1, it.bam1_text ] }\n",
      "Nextflow operator map with content: .map { it -> [it['contrast'], it] }\n",
      "Nextflow operator combine with content: .combine ( ch_rmats_temp, by: 0 )\n",
      "Nextflow operator map with content: .map { it -> it[1] + ['rmats_temp': it[2]] }\n",
      "Nextflow operator map with content: ch_contrasts_bamlist = ch_contrasts.map { [ it.contrast, it.treatment, it.meta1, it.bam1, it.bam1_text, it.rmats_temp ] }\n",
      "Nextflow operator mix with content: ch_versions         = ch_versions.mix(RMATS_POST_SINGLE.out.versions)\n",
      "Nextflow operator map with content: ch_bam = ch_contrasts.map { [ it.contrast, it.treatment, it.control, it.bam1, it.bam2 ] }\n",
      "Nextflow operator map with content: .map { it -> [it['contrast'], it] }\n",
      "Nextflow operator combine with content: .combine ( ch_bamlist, by: 0 )\n",
      "Nextflow operator map with content: .map { it -> it[1] + ['bam1_text': it[2], 'bam2_text': it[3]] }\n",
      "Nextflow operator map with content: ch_contrasts_bamlist = ch_contrasts.map { [ it.contrast, it.treatment, it.control, it.meta1, it.meta2, it.bam1, it.bam2, it.bam1_text, it.bam2_text ] }\n",
      "Nextflow operator map with content: .map { it -> [it['contrast'], it] }\n",
      "Nextflow operator combine with content: .combine ( ch_rmats_temp, by: 0 )\n",
      "Nextflow operator map with content: .map { it -> it[1] + ['rmats_temp': it[2]] }\n",
      "Nextflow operator map with content: ch_contrasts_bamlist = ch_contrasts.map { [ it.contrast, it.treatment, it.control, it.meta1, it.meta2, it.bam1, it.bam2, it.bam1_text, it.bam2_text, it.rmats_temp ] }\n",
      "Nextflow operator mix with content: ch_versions         = ch_versions.mix(RMATS_POST.out.versions)\n",
      "Nextflow operator ifEmpty with content: versions = ch_versions.ifEmpty(null)    //    channel: [ versions.yml ]\n",
      "Error: Unclosed process block in file: ./git_repos\\rnasplice\\subworkflows\\local\\rmats.nf\n",
      "Process SUPPA in filepath ./git_repos\\rnasplice\\subworkflows\\local\\suppa.nf\n",
      "Nextflow operator splitCsv with content: ch_suppa_local_contrasts = ch_contrastsheet.splitCsv(header:true)\n",
      "Nextflow operator flatten with content: .flatten()\n",
      "Nextflow operator map with content: .map { [it.baseName, it ] }\n",
      "Nextflow operator map with content: .map { it -> [it['treatment'], it] }\n",
      "Nextflow operator combine with content: .combine ( ch_suppa_tpm_conditions, by: 0 )\n",
      "Nextflow operator map with content: .map { it -> it[1] + ['tpm1': it[2]] }\n",
      "Nextflow operator map with content: .map { it -> [it['control'], it] }\n",
      "Nextflow operator combine with content: .combine ( ch_suppa_tpm_conditions, by: 0 )\n",
      "Nextflow operator map with content: .map { it -> it[1] + ['tpm2': it[2]] }\n",
      "Nextflow operator flatten with content: .flatten()\n",
      "Nextflow operator map with content: .map { [ it.baseName.toString().replaceAll(\"local_\", \"\"), it ] }\n",
      "Nextflow operator map with content: .map { it -> [it['treatment'], it] }\n",
      "Nextflow operator combine with content: .combine ( ch_suppa_psi_conditions, by: 0 )\n",
      "Nextflow operator map with content: .map { it -> it[1] + ['psi1': it[2]] }\n",
      "Nextflow operator map with content: .map { it -> [it['control'], it] }\n",
      "Nextflow operator combine with content: .combine ( ch_suppa_psi_conditions, by: 0 )\n",
      "Nextflow operator map with content: .map { it -> it[1] + ['psi2': it[2]] }\n",
      "Nextflow operator map with content: ch_split_suppa_tpms = ch_suppa_local_contrasts.map { [ it.treatment, it.control, it.tpm1, it.tpm2 ] }\n",
      "Nextflow operator map with content: ch_split_suppa_local_psi = ch_suppa_local_contrasts.map { [ it.treatment, it.control, it.psi1, it.psi2 ] }\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Nextflow operator splitCsv with content: ch_suppa_isoform_contrasts = ch_contrastsheet.splitCsv(header:true)\n",
      "Nextflow operator flatten with content: .flatten()\n",
      "Nextflow operator map with content: .map { [it.baseName, it ] }\n",
      "Nextflow operator map with content: .map { it -> [it['treatment'], it] }\n",
      "Nextflow operator combine with content: .combine ( ch_suppa_tpm_conditions, by: 0)\n",
      "Nextflow operator map with content: .map { it -> it[1] + ['tpm1': it[2]] }\n",
      "Nextflow operator map with content: .map { it -> [it['control'], it] }\n",
      "Nextflow operator combine with content: .combine ( ch_suppa_tpm_conditions, by: 0)\n",
      "Nextflow operator map with content: .map { it -> it[1] + ['tpm2': it[2]] }\n",
      "Nextflow operator flatten with content: .flatten()\n",
      "Nextflow operator map with content: .map { [ it.baseName.toString().replaceAll(\"transcript_\", \"\"), it ] }\n",
      "Nextflow operator map with content: .map { it -> [it['treatment'], it] }\n",
      "Nextflow operator combine with content: .combine ( ch_suppa_psi_conditions, by: 0 )\n",
      "Nextflow operator map with content: .map { it -> it[1] + ['psi1': it[2]] }\n",
      "Nextflow operator map with content: .map { it -> [it['control'], it] }\n",
      "Nextflow operator combine with content: .combine ( ch_suppa_psi_conditions, by: 0 )\n",
      "Nextflow operator map with content: .map { it -> it[1] + ['psi2': it[2]] }\n",
      "Nextflow operator map with content: ch_split_suppa_tpms = ch_suppa_isoform_contrasts.map { [ it.treatment, it.control, it.tpm1, it.tpm2 ] }\n",
      "Nextflow operator map with content: ch_split_suppa_isoform_psi = ch_suppa_isoform_contrasts.map { [ it.treatment, it.control, it.psi1, it.psi2 ] }\n",
      "Nextflow operator ifEmpty with content: versions = ch_versions.ifEmpty(null)                         // channel: [ versions.yml ]\n",
      "Error: Unclosed process block in file: ./git_repos\\rnasplice\\subworkflows\\local\\suppa.nf\n",
      "Process TX2GENE_TXIMPORT in filepath ./git_repos\\rnasplice\\subworkflows\\local\\tx2gene_tximport.nf\n",
      "Nextflow operator collect with content: salmon_results   // tuple [meta, salmon path] when starting from salmon or path: \"salmon/*\" (SALMON_QUANT.out.results.collect{it[1]}) other cases\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator branch with content: .branch{\n",
      "Nextflow operator set with content: .set{ salmon_results }\n",
      "Nextflow operator mix with content: salmon_results = salmon_results.dir.mix(UNTAR.out.untar)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GFFREAD_TX2GENE.out.versions)\n",
      "Nextflow operator collect with content: TXIMPORT ( salmon_results.collect{it[1]}, tx2gene )\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(TXIMPORT.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\rnasplice\\subworkflows\\local\\tx2gene_tximport.nf\n",
      "Process VISUALISE_MISO in filepath ./git_repos\\rnasplice\\subworkflows\\local\\visualise_miso.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GTF_2_GFF3.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(MISO_INDEX.out.versions)\n",
      "Nextflow operator join with content: ch_bam_join = ch_genome_bam.join(ch_genome_bai)\n",
      "Nextflow operator collect with content: ch_miso_index =  MISO_INDEX.out.miso_index.collect()\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(MISO_RUN.out.versions)\n",
      "Nextflow operator collect with content: ch_bams = ch_genome_bam.collect({it[1]})\n",
      "Nextflow operator collect with content: ch_miso_run = MISO_RUN.out.miso.map{it[1]}.collect()\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(MISO_SETTINGS.out.versions)\n",
      "Nextflow operator collect with content: def miso_genes_list = miso_genes ? miso_genes.split(',').collect{ it.trim() } : [\"\"]\n",
      "Nextflow operator splitCsv with content: .splitCsv()\n",
      "Nextflow operator concat with content: ch_miso_genes_list.concat( ch_miso_genes_file )\n",
      "Nextflow operator set with content: .set{ ch_miso_genes }\n",
      "Nextflow operator splitCsv with content: .splitCsv()\n",
      "Nextflow operator set with content: .set{ ch_miso_genes }\n",
      "Nextflow operator combine with content: ch_miso_input = MISO_SETTINGS.out.miso_settings.combine(ch_miso_genes)\n",
      "Nextflow operator map with content: .map { [it[1], it[2] ] }\n",
      "Nextflow operator collect with content: .collect()\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(MISO_SASHIMI.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\rnasplice\\subworkflows\\local\\visualise_miso.nf\n",
      "Process BAM_SORT_STATS_SAMTOOLS in filepath ./git_repos\\rnasplice\\subworkflows\\nf-core\\bam_sort_stats_samtools\\main.nf\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SAMTOOLS_SORT.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SAMTOOLS_INDEX.out.versions.first())\n",
      "Nextflow operator join with content: .join(SAMTOOLS_INDEX.out.bai, by: [0], remainder: true)\n",
      "Nextflow operator join with content: .join(SAMTOOLS_INDEX.out.csi, by: [0], remainder: true)\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator set with content: .set { ch_bam_bai }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BAM_STATS_SAMTOOLS.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\rnasplice\\subworkflows\\nf-core\\bam_sort_stats_samtools\\main.nf\n",
      "Process BAM_STATS_SAMTOOLS in filepath ./git_repos\\rnasplice\\subworkflows\\nf-core\\bam_stats_samtools\\main.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_STATS.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_FLAGSTAT.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_IDXSTATS.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\rnasplice\\subworkflows\\nf-core\\bam_stats_samtools\\main.nf\n",
      "Process BEDGRAPH_BEDCLIP_BEDGRAPHTOBIGWIG in filepath ./git_repos\\rnasplice\\subworkflows\\nf-core\\bedgraph_bedclip_bedgraphtobigwig\\main.nf\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(UCSC_BEDCLIP.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(UCSC_BEDGRAPHTOBIGWIG.out.versions.first())\n",
      "Error: Unclosed process block in file: ./git_repos\\rnasplice\\subworkflows\\nf-core\\bedgraph_bedclip_bedgraphtobigwig\\main.nf\n",
      "Process FASTQ_FASTQC_UMITOOLS_TRIMGALORE in filepath ./git_repos\\rnasplice\\subworkflows\\nf-core\\fastq_fastqc_umitools_trimgalore\\main.nf\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(FASTQC.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(UMITOOLS_EXTRACT.out.versions.first())\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator set with content: .set { umi_reads }\n",
      "Nextflow operator first with content: ch_versions   = ch_versions.mix(TRIMGALORE.out.versions.first())\n",
      "Nextflow operator join with content: .join(trim_log, remainder: true)\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator set with content: .set { ch_num_trimmed_reads }\n",
      "Nextflow operator filter with content: .filter { meta, reads, num_reads -> num_reads >= min_trimmed_reads.toFloat() }\n",
      "Nextflow operator map with content: .map { meta, reads, num_reads -> [ meta, reads ] }\n",
      "Nextflow operator set with content: .set { trim_reads }\n",
      "Nextflow operator map with content: .map { meta, reads, num_reads -> [ meta, num_reads ] }\n",
      "Nextflow operator set with content: .set { trim_read_count }\n",
      "Nextflow operator ifEmpty with content: versions = ch_versions.ifEmpty(null) // channel: [ versions.yml ]\n",
      "Error: Unclosed process block in file: ./git_repos\\rnasplice\\subworkflows\\nf-core\\fastq_fastqc_umitools_trimgalore\\main.nf\n",
      "Process RNASPLICE in filepath ./git_repos\\rnasplice\\workflows\\rnasplice.nf\n",
      "Nextflow operator map with content: ch_meta_fasta = PREPARE_GENOME.out.fasta.map { [ [:], it ] }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(PREPARE_GENOME.out.versions)\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator groupTuple with content: .groupTuple()\n",
      "Nextflow operator branch with content: .branch {\n",
      "Nextflow operator flatten with content: return [ meta, fastq.flatten() ]\n",
      "Nextflow operator flatten with content: return [ meta, fastq.flatten() ]\n",
      "Nextflow operator set with content: .set { ch_fastq }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(INPUT_CHECK.out.versions)\n",
      "Nextflow operator set with content: .set { ch_genome_bam }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(INPUT_CHECK.out.versions)\n",
      "Nextflow operator set with content: .set { ch_transcriptome_bam }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(INPUT_CHECK.out.versions)\n",
      "Nextflow operator set with content: .set { ch_salmon_results }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(INPUT_CHECK.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(CONTRASTS_CHECK.out.versions)\n",
      "Nextflow operator mix with content: .mix(ch_fastq.single)\n",
      "Nextflow operator set with content: .set { ch_cat_fastq }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(CAT_FASTQ.out.versions)\n",
      "Nextflow operator mix with content: ch_versions        = ch_versions.mix(FASTQ_FASTQC_UMITOOLS_TRIMGALORE.out.versions)\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator collect with content: .collect()\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator set with content: .set { ch_fail_trimming_multiqc }\n",
      "Nextflow operator mix with content: ch_versions          = ch_versions.mix(BAM_SORT_STATS_SAMTOOLS.out.versions)\n",
      "Nextflow operator mix with content: ch_versions                 = ch_versions.mix(BAM_SORT_STATS_SAMTOOLS.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ALIGN_STAR.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(DEXSEQ_DEU.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(EDGER_DEU.out.versions)\n",
      "Nextflow operator map with content: .map { meta, bam -> [meta.condition, meta, bam] }\n",
      "Nextflow operator groupTuple with content: .groupTuple(by:0)\n",
      "Nextflow operator set with content: .set { ch_genome_bam_conditions }\n",
      "Nextflow operator map with content: .map { meta, bam -> [meta.condition, meta, bam] }\n",
      "Nextflow operator groupTuple with content: .groupTuple(by:0)\n",
      "Nextflow operator set with content: .set { ch_genome_bam_conditions }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(RMATS.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(VISUALISE_MISO.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SALMON_QUANT_STAR.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(TX2GENE_TXIMPORT_STAR_SALMON.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(DRIMSEQ_DEXSEQ_DTU_STAR_SALMON.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SUPPA_STAR_SALMON.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SALMON_QUANT_SALMON.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(TX2GENE_TXIMPORT_SALMON.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(TX2GENE_TXIMPORT_SALMON.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(DRIMSEQ_DEXSEQ_DTU_SALMON.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SUPPA_SALMON.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BEDTOOLS_GENOMECOV.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BEDGRAPH_TO_BIGWIG_FORWARD.out.versions)\n",
      "Nextflow operator collectFile with content: ch_versions.unique().collectFile(name: 'collated_versions.yml')\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(ch_multiqc_custom_config.collect().ifEmpty([]))\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_workflow_summary.collectFile(name: 'workflow_summary_mqc.yaml'))\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_methods_description.collectFile(name: 'methods_description_mqc.yaml'))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(CUSTOM_DUMPSOFTWAREVERSIONS.out.mqc_yml.collect())\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(FASTQ_FASTQC_UMITOOLS_TRIMGALORE.out.fastqc_zip.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(FASTQ_FASTQC_UMITOOLS_TRIMGALORE.out.trim_zip.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(FASTQ_FASTQC_UMITOOLS_TRIMGALORE.out.trim_log.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_fail_trimming_multiqc.collectFile(name: 'fail_trimmed_samples_mqc.tsv').ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(ch_salmon_results.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(ch_star_multiqc.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(ch_samtools_stats.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(ch_samtools_flagstat.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(ch_samtools_idxstats.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(EDGER_DEU.out.featureCounts_summary.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files.collect(),\n",
      "Nextflow operator toList with content: ch_multiqc_config.toList(),\n",
      "Nextflow operator toList with content: ch_multiqc_custom_config.toList(),\n",
      "Nextflow operator toList with content: ch_multiqc_logo.toList()\n",
      "Nextflow operator toList with content: multiqc_report = MULTIQC.out.report.toList()\n",
      "Error: Unclosed process block in file: ./git_repos\\rnasplice\\workflows\\rnasplice.nf\n",
      "Process NFCORE_RNAVAR in filepath ./git_repos\\rnavar\\main.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\rnavar\\main.nf\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Process ANNOTATE in filepath ./git_repos\\rnavar\\subworkflows\\local\\annotate.nf\n",
      "Nextflow operator mix with content: ch_vcf_ann  = ch_vcf_ann.mix(SNPEFF_ANNOTATE.out.vcf_tbi)\n",
      "Nextflow operator mix with content: ch_reports  = ch_reports.mix(SNPEFF_ANNOTATE.out.reports)\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SNPEFF_ANNOTATE.out.versions.first())\n",
      "Nextflow operator map with content: vcf_ann_for_merge = SNPEFF_ANNOTATE.out.vcf_tbi.map{ meta, vcf, tbi -> [meta, vcf] }\n",
      "Nextflow operator mix with content: ch_vcf_ann  = ch_vcf_ann.mix(MERGE_ANNOTATE.out.vcf_tbi)\n",
      "Nextflow operator mix with content: ch_reports  = ch_reports.mix(MERGE_ANNOTATE.out.reports)\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(MERGE_ANNOTATE.out.versions.first())\n",
      "Nextflow operator mix with content: ch_vcf_ann  = ch_vcf_ann.mix(ENSEMBLVEP_ANNOTATE.out.vcf_tbi)\n",
      "Nextflow operator mix with content: ch_reports  = ch_reports.mix(ENSEMBLVEP_ANNOTATE.out.reports)\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(ENSEMBLVEP_ANNOTATE.out.versions.first())\n",
      "Error: Unclosed process block in file: ./git_repos\\rnavar\\subworkflows\\local\\annotate.nf\n",
      "Process INPUT_CHECK in filepath ./git_repos\\rnavar\\subworkflows\\local\\input_check.nf\n",
      "Nextflow operator splitCsv with content: .splitCsv ( header:true, sep:',' )\n",
      "Nextflow operator map with content: .map { create_fastq_channel(it) }\n",
      "Nextflow operator set with content: .set { reads }\n",
      "Error: Unclosed process block in file: ./git_repos\\rnavar\\subworkflows\\local\\input_check.nf\n",
      "Process PREPARE_GENOME in filepath ./git_repos\\rnavar\\subworkflows\\local\\prepare_genome.nf\n",
      "Nextflow operator map with content: Channel.fromPath(params.fasta).map{ it -> [[id:it[0].baseName], it] }\n",
      "Nextflow operator collect with content: ch_fasta = GUNZIP_FASTA.out.gunzip.map{ meta, fasta -> [fasta] }.collect()\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GUNZIP_FASTA.out.versions)\n",
      "Nextflow operator collect with content: ch_fasta = Channel.fromPath(params.fasta).collect()\n",
      "Nextflow operator map with content: Channel.fromPath(params.gtf).map{ it -> [[id:it[0].baseName], it] }\n",
      "Nextflow operator collect with content: ch_gtf = GUNZIP_GTF.out.gunzip.map{ meta, gtf -> [gtf] }.collect()\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GUNZIP_GTF.out.versions)\n",
      "Nextflow operator collect with content: ch_gtf = Channel.fromPath(params.gtf).collect()\n",
      "Nextflow operator map with content: Channel.fromPath(params.gff).map{ it -> [[id:it[0].baseName], it] }\n",
      "Nextflow operator collect with content: ch_gff = GUNZIP_GFF.out.gunzip.map{ meta, gff -> [gff] }.collect()\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GUNZIP_GFF.out.versions)\n",
      "Nextflow operator collect with content: ch_gff = Channel.fromPath(params.gff).collect()\n",
      "Nextflow operator set with content: .set { ch_gtf }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GFFREAD.out.versions)\n",
      "Nextflow operator map with content: Channel.fromPath(params.exon_bed).map{ it -> [[id:it[0].baseName], it] }\n",
      "Nextflow operator collect with content: ch_gene_bed = GUNZIP_GENE_BED.out.gunzip.map{ meta, bed -> [bed] }.collect()\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GUNZIP_GENE_BED.out.versions)\n",
      "Nextflow operator collect with content: ch_gene_bed = Channel.fromPath(params.exon_bed).collect()\n",
      "Nextflow operator collect with content: ch_exon_bed = GTF2BED ( ch_gtf ).bed.collect()\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GTF2BED.out.versions)\n",
      "Nextflow operator collect with content: if (params.fasta_fai) ch_fasta_fai = Channel.fromPath(params.fasta_fai).collect()\n",
      "Nextflow operator map with content: ch_fasta.map{ it -> [[id:it[0].getName()], it]}\n",
      "Nextflow operator collect with content: ch_fasta_fai    = SAMTOOLS_FAIDX.out.fai.map{ meta, fai -> [fai] }.collect()\n",
      "Nextflow operator mix with content: ch_versions     = ch_versions.mix(SAMTOOLS_FAIDX.out.versions)\n",
      "Nextflow operator collect with content: if (params.dict) ch_fasta_dict = Channel.fromPath(params.dict).collect()\n",
      "Nextflow operator map with content: Channel.fromPath(params.star_index).map{ it -> [[id:it[0].baseName], it] }\n",
      "Nextflow operator collect with content: ch_star_index = UNTAR_STAR_INDEX.out.untar.map{ meta, star_index -> [star_index] }.collect()\n",
      "Nextflow operator mix with content: ch_versions   = ch_versions.mix(UNTAR_STAR_INDEX.out.versions)\n",
      "Nextflow operator collect with content: ch_star_index = Channel.fromPath(params.star_index).collect()\n",
      "Nextflow operator set with content: .set { ch_star_index }\n",
      "Nextflow operator mix with content: ch_versions     = ch_versions.mix(STAR_GENOMEGENERATE.out.versions)\n",
      "Nextflow operator ifEmpty with content: versions         = ch_versions.ifEmpty(null) // channel: [ versions.yml ]\n",
      "Error: Unclosed process block in file: ./git_repos\\rnavar\\subworkflows\\local\\prepare_genome.nf\n",
      "Process ALIGN_STAR in filepath ./git_repos\\rnavar\\subworkflows\\nf-core\\align_star.nf\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(STAR_ALIGN.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(BAM_SORT_SAMTOOLS.out.versions.first())\n",
      "Error: Unclosed process block in file: ./git_repos\\rnavar\\subworkflows\\nf-core\\align_star.nf\n",
      "Process BAM_SORT_SAMTOOLS in filepath ./git_repos\\rnavar\\subworkflows\\nf-core\\bam_sort_samtools.nf\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SAMTOOLS_SORT.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SAMTOOLS_INDEX.out.versions.first())\n",
      "Nextflow operator join with content: .join(SAMTOOLS_INDEX.out.bai, by: [0], remainder: true)\n",
      "Nextflow operator join with content: .join(SAMTOOLS_INDEX.out.csi, by: [0], remainder: true)\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator set with content: .set { ch_bam_bai }\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(BAM_STATS_SAMTOOLS.out.versions.first())\n",
      "Error: Unclosed process block in file: ./git_repos\\rnavar\\subworkflows\\nf-core\\bam_sort_samtools.nf\n",
      "Process BAM_STATS_SAMTOOLS in filepath ./git_repos\\rnavar\\subworkflows\\nf-core\\bam_stats_samtools.nf\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SAMTOOLS_STATS.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SAMTOOLS_FLAGSTAT.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SAMTOOLS_IDXSTATS.out.versions.first())\n",
      "Error: Unclosed process block in file: ./git_repos\\rnavar\\subworkflows\\nf-core\\bam_stats_samtools.nf\n",
      "Process ENSEMBLVEP_ANNOTATE in filepath ./git_repos\\rnavar\\subworkflows\\nf-core\\ensemblvep_annotate.nf\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(ENSEMBLVEP.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(TABIX_BGZIPTABIX.out.versions.first())\n",
      "Error: Unclosed process block in file: ./git_repos\\rnavar\\subworkflows\\nf-core\\ensemblvep_annotate.nf\n",
      "Process MARKDUPLICATES in filepath ./git_repos\\rnavar\\subworkflows\\nf-core\\markduplicates.nf\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(GATK4_MARKDUPLICATES.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SAMTOOLS_INDEX.out.versions.first())\n",
      "Nextflow operator join with content: .join(SAMTOOLS_INDEX.out.bai, by: [0], remainder: true)\n",
      "Nextflow operator join with content: .join(SAMTOOLS_INDEX.out.csi, by: [0], remainder: true)\n",
      "Nextflow operator map with content: .map{meta, bam, bai, csi ->\n",
      "Nextflow operator set with content: .set{ch_bam_bai}\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(BAM_STATS_SAMTOOLS.out.versions.first())\n",
      "Error: Unclosed process block in file: ./git_repos\\rnavar\\subworkflows\\nf-core\\markduplicates.nf\n",
      "Process RECALIBRATE in filepath ./git_repos\\rnavar\\subworkflows\\nf-core\\recalibrate.nf\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(APPLYBQSR.out.versions.first())\n",
      "Nextflow operator join with content: .join(SAMTOOLS_INDEX.out.bai, by: [0], remainder: true)\n",
      "Nextflow operator join with content: .join(SAMTOOLS_INDEX.out.csi, by: [0], remainder: true)\n",
      "Nextflow operator map with content: .map{meta, bam, bai, csi ->\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SAMTOOLS_INDEX.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SAMTOOLS_STATS.out.versions.first())\n",
      "Error: Unclosed process block in file: ./git_repos\\rnavar\\subworkflows\\nf-core\\recalibrate.nf\n",
      "Process SNPEFF_ANNOTATE in filepath ./git_repos\\rnavar\\subworkflows\\nf-core\\snpeff_annotate.nf\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SNPEFF.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(TABIX_BGZIPTABIX.out.versions.first())\n",
      "Error: Unclosed process block in file: ./git_repos\\rnavar\\subworkflows\\nf-core\\snpeff_annotate.nf\n",
      "Process SPLITNCIGAR in filepath ./git_repos\\rnavar\\subworkflows\\nf-core\\splitncigar.nf\n",
      "Nextflow operator combine with content: bam.combine(intervals)\n",
      "Nextflow operator map with content: .map{ meta, bam, bai, intervals ->\n",
      "Nextflow operator set with content: }.set{bam_interval}\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(GATK4_SPLITNCIGARREADS.out.versions.first())\n",
      "Nextflow operator map with content: .map{ meta, bam ->\n",
      "Nextflow operator groupTuple with content: }.groupTuple().set{bam_splitncigar_interval}\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SAMTOOLS_MERGE.out.versions.first())\n",
      "Nextflow operator join with content: .join(SAMTOOLS_INDEX.out.bai, by: [0], remainder: true)\n",
      "Nextflow operator join with content: .join(SAMTOOLS_INDEX.out.csi, by: [0], remainder: true)\n",
      "Nextflow operator map with content: .map{meta, bam, bai, csi ->\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SAMTOOLS_INDEX.out.versions.first())\n",
      "Error: Unclosed process block in file: ./git_repos\\rnavar\\subworkflows\\nf-core\\splitncigar.nf\n",
      "Process RNAVAR in filepath ./git_repos\\rnavar\\workflows\\rnavar.nf\n",
      "Nextflow operator combine with content: ch_genome_bed = Channel.from([id:'genome.bed']).combine(PREPARE_GENOME.out.exon_bed)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(PREPARE_GENOME.out.versions)\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator join with content: meta_clone.id = meta_clone.id.split('_')[0..-2].join('_')\n",
      "Nextflow operator groupTuple with content: .groupTuple(by: [0])\n",
      "Nextflow operator branch with content: .branch {\n",
      "Nextflow operator flatten with content: return [ meta, fastq.flatten() ]\n",
      "Nextflow operator flatten with content: return [ meta, fastq.flatten() ]\n",
      "Nextflow operator set with content: .set { ch_fastq }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(INPUT_CHECK.out.versions)\n",
      "Nextflow operator mix with content: .mix(ch_fastq.single)\n",
      "Nextflow operator set with content: .set { ch_cat_fastq }\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(CAT_FASTQ.out.versions.first().ifEmpty(null))\n",
      "Nextflow operator collect with content: ch_reports  = ch_reports.mix(FASTQC.out.zip.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(FASTQC.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(GATK4_BEDTOINTERVALLIST.out.versions.first().ifEmpty(null))\n",
      "Nextflow operator flatten with content: ch_interval_list_split = GATK4_INTERVALLISTTOOLS.out.interval_list.map{ meta, bed -> [bed] }.flatten()\n",
      "Nextflow operator collect with content: ch_reports           = ch_reports.mix(ALIGN_STAR.out.stats.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_reports           = ch_reports.mix(ALIGN_STAR.out.log_final.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator first with content: ch_versions          = ch_versions.mix(ALIGN_STAR.out.versions.first().ifEmpty(null))\n",
      "Nextflow operator collect with content: ch_reports                = ch_reports.mix(MARKDUPLICATES.out.stats.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_reports                = ch_reports.mix(MARKDUPLICATES.out.metrics.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator first with content: ch_versions               = ch_versions.mix(MARKDUPLICATES.out.versions.first().ifEmpty(null))\n",
      "Nextflow operator first with content: ch_versions             = ch_versions.mix(SPLITNCIGAR.out.versions.first().ifEmpty(null))\n",
      "Nextflow operator collect with content: ch_known_sites     = ch_dbsnp.concat(ch_known_indels).collect()\n",
      "Nextflow operator collect with content: ch_known_sites_tbi = ch_dbsnp_tbi.concat(ch_known_indels_tbi).collect()\n",
      "Nextflow operator flatten with content: ch_interval_list_recalib = ch_interval_list.map{ meta, bed -> [bed] }.flatten()\n",
      "Nextflow operator combine with content: ch_splitncigar_bam_bai.combine(ch_interval_list_recalib)\n",
      "Nextflow operator map with content: .map{ meta, bam, bai, interval -> [ meta, bam, bai, interval]\n",
      "Nextflow operator set with content: }.set{ch_splitncigar_bam_bai_interval}\n",
      "Nextflow operator map with content: ch_reports  = ch_reports.mix(ch_bqsr_table.map{ meta, table -> table})\n",
      "Nextflow operator first with content: ch_versions     = ch_versions.mix(GATK4_BASERECALIBRATOR.out.versions.first().ifEmpty(null))\n",
      "Nextflow operator join with content: ch_bam_applybqsr       = ch_splitncigar_bam_bai.join(ch_bqsr_table, by: [0])\n",
      "Nextflow operator flatten with content: ch_interval_list_applybqsr = ch_interval_list.map{ meta, bed -> [bed] }.flatten()\n",
      "Nextflow operator combine with content: ch_bam_applybqsr.combine(ch_interval_list_applybqsr)\n",
      "Nextflow operator map with content: .map{ meta, bam, bai, table, interval -> [ meta, bam, bai, table, interval]\n",
      "Nextflow operator set with content: }.set{ch_applybqsr_bam_bai_interval}\n",
      "Nextflow operator collect with content: ch_reports  = ch_reports.mix(RECALIBRATE.out.qc.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(RECALIBRATE.out.versions.first().ifEmpty(null))\n",
      "Nextflow operator combine with content: ch_haplotypecaller_interval_bam = ch_bam_variant_calling.combine(ch_interval_list_split)\n",
      "Nextflow operator map with content: .map{ meta, bam, bai, interval_list ->\n",
      "Nextflow operator map with content: .map{ meta, vcf ->\n",
      "Nextflow operator groupTuple with content: .groupTuple()\n",
      "Nextflow operator first with content: ch_versions  = ch_versions.mix(GATK4_HAPLOTYPECALLER.out.versions.first().ifEmpty(null))\n",
      "Nextflow operator first with content: ch_versions  = ch_versions.mix(GATK4_MERGEVCFS.out.versions.first().ifEmpty(null))\n",
      "Nextflow operator join with content: .join(TABIX.out.tbi, by: [0], remainder: true)\n",
      "Nextflow operator join with content: .join(TABIX.out.csi, by: [0], remainder: true)\n",
      "Nextflow operator map with content: .map{meta, vcf, tbi, csi ->\n",
      "Nextflow operator first with content: ch_versions     = ch_versions.mix(TABIX.out.versions.first().ifEmpty(null))\n",
      "Nextflow operator first with content: ch_versions     = ch_versions.mix(GATK4_VARIANTFILTRATION.out.versions.first().ifEmpty(null))\n",
      "Nextflow operator mix with content: ch_reports  = ch_reports.mix(ANNOTATE.out.reports)\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(ANNOTATE.out.versions.first().ifEmpty(null))\n",
      "Nextflow operator collectFile with content: CUSTOM_DUMPSOFTWAREVERSIONS (ch_versions.unique().collectFile(name: 'collated_versions.yml'))\n",
      "Nextflow operator collect with content: ch_version_yaml = CUSTOM_DUMPSOFTWAREVERSIONS.out.mqc_yml.collect()\n",
      "Nextflow operator mix with content: ch_multiqc_files    =  Channel.empty().mix(ch_version_yaml,\n",
      "Nextflow operator collect with content: ch_multiqc_custom_config.collect().ifEmpty([]),\n",
      "Nextflow operator collectFile with content: ch_workflow_summary.collectFile(name: 'workflow_summary_mqc.yaml'),\n",
      "Nextflow operator collect with content: ch_reports.collect(),\n",
      "Nextflow operator collect with content: MULTIQC (ch_multiqc_files.collect())\n",
      "Nextflow operator toList with content: multiqc_report = MULTIQC.out.report.toList()\n",
      "Error: Unclosed process block in file: ./git_repos\\rnavar\\workflows\\rnavar.nf\n",
      "Process NFCORE_SAMMYSEQ in filepath ./git_repos\\sammyseq\\main.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\sammyseq\\main.nf\n",
      "Process CAT_FRACTIONS in filepath ./git_repos\\sammyseq\\subworkflows\\local\\cat_fractions.nf\n",
      "Nextflow operator map with content: all_files = starting_files.map { meta, fastq ->\n",
      "Nextflow operator groupTuple with content: all_files.groupTuple( by:0 )\n",
      "Nextflow operator map with content: .map{ expID, meta , fastq ->\n",
      "Nextflow operator flatten with content: [ [meta2], fastq.flatten().toList() ]\n",
      "Nextflow operator set with content: .set { reads_to_merge }\n",
      "Nextflow operator filter with content: .filter { meta, fastq -> fastq.size() > 1 }\n",
      "Nextflow operator set with content: .set { ch_reads_to_process_in_CAT_FASTQ }\n",
      "Nextflow operator flatMap with content: .flatMap { meta, fastq  ->\n",
      "Nextflow operator collect with content: meta.collect { m -> [m,fastq]}\n",
      "Nextflow operator set with content: .set { ch_to_CAT }\n",
      "Nextflow operator set with content: ).reads.set { cat_fastq_output }\n",
      "Nextflow operator map with content: ch_cat_adjusted = CAT_FASTQ.out.reads.map { meta, fastq ->\n",
      "Nextflow operator mix with content: merged_reads = starting_files.mix(ch_cat_adjusted)\n",
      "Error: Unclosed process block in file: ./git_repos\\sammyseq\\subworkflows\\local\\cat_fractions.nf\n",
      "Process FAIDX_SUBWORKFLOW in filepath ./git_repos\\sammyseq\\subworkflows\\local\\faidx_cut_chr_size.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\sammyseq\\subworkflows\\local\\faidx_cut_chr_size.nf\n",
      "Process INPUT_CHECK in filepath ./git_repos\\sammyseq\\subworkflows\\local\\input_check.nf\n",
      "Nextflow operator splitCsv with content: .splitCsv ( header:true, sep:',' )\n",
      "Nextflow operator map with content: .map { create_fastq_channel(it) }\n",
      "Nextflow operator set with content: .set { reads }\n",
      "Nextflow operator map with content: .map { meta, fastq ->\n",
      "Nextflow operator groupTuple with content: .groupTuple( by:0 )\n",
      "Nextflow operator map with content: .map{ expID, meta , fastq ->\n",
      "Nextflow operator flatten with content: [ [meta], fastq.flatten() ]\n",
      "Nextflow operator set with content: .set { reads_to_merge }\n",
      "Error: Unclosed process block in file: ./git_repos\\sammyseq\\subworkflows\\local\\input_check.nf\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Process PREPARE_GENOME in filepath ./git_repos\\sammyseq\\subworkflows\\local\\prepare_genome.nf\n",
      "Nextflow operator map with content: ch_fasta    = GUNZIP_FASTA ( [ [:], params.fasta ] ).gunzip.map{ it[1] }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GUNZIP_FASTA.out.versions)\n",
      "Nextflow operator mix with content: ch_versions  = ch_versions.mix(UNTAR_BWA_INDEX.out.versions)\n",
      "Nextflow operator mix with content: ch_versions  = ch_versions.mix(BWA_INDEX.out.versions)\n",
      "Nextflow operator ifEmpty with content: versions    = ch_versions.ifEmpty(null) // channel: [ versions.yml ]\n",
      "Error: Unclosed process block in file: ./git_repos\\sammyseq\\subworkflows\\local\\prepare_genome.nf\n",
      "Process BAM_MARKDUPLICATES_PICARD in filepath ./git_repos\\sammyseq\\subworkflows\\nf-core\\bam_markduplicates_picard\\main.nf\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(PICARD_MARKDUPLICATES.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SAMTOOLS_INDEX.out.versions.first())\n",
      "Nextflow operator join with content: .join(SAMTOOLS_INDEX.out.bai, by: [0], remainder: true)\n",
      "Nextflow operator join with content: .join(SAMTOOLS_INDEX.out.csi, by: [0], remainder: true)\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BAM_STATS_SAMTOOLS.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\sammyseq\\subworkflows\\nf-core\\bam_markduplicates_picard\\main.nf\n",
      "Process BAM_STATS_SAMTOOLS in filepath ./git_repos\\sammyseq\\subworkflows\\nf-core\\bam_stats_samtools\\main.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_STATS.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_FLAGSTAT.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_IDXSTATS.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\sammyseq\\subworkflows\\nf-core\\bam_stats_samtools\\main.nf\n",
      "Process FASTQ_ALIGN_BWAALN in filepath ./git_repos\\sammyseq\\subworkflows\\nf-core\\fastq_align_bwaaln\\main.nf\n",
      "Nextflow operator combine with content: .combine(ch_index)\n",
      "Nextflow operator map with content: .map{\n",
      "Nextflow operator multiMap with content: .multiMap {\n",
      "Nextflow operator map with content: ch_reads_newid = ch_prepped_input.map{ meta, reads, meta_index, index -> [ meta, reads ] }\n",
      "Nextflow operator map with content: ch_index_newid = ch_prepped_input.map{ meta, reads, meta_index, index -> [ meta, index ] }\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix( BWA_ALN.out.versions.first() )\n",
      "Nextflow operator join with content: .join ( BWA_ALN.out.sai )\n",
      "Nextflow operator branch with content: .branch {\n",
      "Nextflow operator join with content: .join ( ch_index_newid )\n",
      "Nextflow operator multiMap with content: .multiMap {\n",
      "Nextflow operator join with content: .join ( ch_index_newid )\n",
      "Nextflow operator multiMap with content: .multiMap {\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix( BWA_SAMPE.out.versions.first() )\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix( BWA_SAMSE.out.versions.first() )\n",
      "Nextflow operator mix with content: ch_bam_for_index = BWA_SAMPE.out.bam.mix( BWA_SAMSE.out.bam )\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SAMTOOLS_INDEX.out.versions.first())\n",
      "Nextflow operator map with content: ch_bam_for_emit = ch_bam_for_index.map{ meta, bam -> [meta - meta.subMap('key_read_ref'), bam] }\n",
      "Nextflow operator map with content: ch_bai_for_emit = SAMTOOLS_INDEX.out.bai.map{ meta, bai -> [meta - meta.subMap('key_read_ref'), bai] }\n",
      "Nextflow operator map with content: ch_csi_for_emit = SAMTOOLS_INDEX.out.csi.map{ meta, csi -> [meta - meta.subMap('key_read_ref'), csi] }\n",
      "Error: Unclosed process block in file: ./git_repos\\sammyseq\\subworkflows\\nf-core\\fastq_align_bwaaln\\main.nf\n",
      "Process SAMMYSEQ in filepath ./git_repos\\sammyseq\\workflows\\sammyseq.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(PREPARE_GENOME.out.versions)\n",
      "Nextflow operator map with content: .map{ meta, path ->\n",
      "Nextflow operator groupTuple with content: .groupTuple()\n",
      "Nextflow operator filter with content: .filter{ it[1].size() == 1 }\n",
      "Nextflow operator map with content: .map{ id, meta, path ->\n",
      "Nextflow operator map with content: .map{ meta, path ->\n",
      "Nextflow operator groupTuple with content: .groupTuple()\n",
      "Nextflow operator filter with content: .filter{ it[1].size() >= 2 } //filtra per numero di meta presenti dopo il tupla se hai due meta vuol dire che devi unire due campioni\n",
      "Nextflow operator map with content: .map{ id, meta, path ->\n",
      "Nextflow operator flatten with content: def flatPath = path.flatten()\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(INPUT_CHECK.out.versions)\n",
      "Nextflow operator set with content: ).reads.set { cat_lane_output }\n",
      "Nextflow operator mix with content: ch_starter = cat_lane_output.mix(ch_notmerge_lane)\n",
      "Nextflow operator map with content: .map{ meta, path ->\n",
      "Nextflow operator mix with content: ch_fastqc_in = ch_fastqc_trim.mix(merged_reads)\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(FASTQC.out.versions.first())\n",
      "Nextflow operator collect with content: ch_fai_for_cut = SAMTOOLS_FAIDX.out.fai.collect()\n",
      "Nextflow operator collect with content: SAMTOOLS_FAIDX.out.fai.collect()\n",
      "Nextflow operator join with content: .join(BAM_MARKDUPLICATES_PICARD.out.bai, by: [0], remainder: true)\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator map with content: ch_fai_path = SAMTOOLS_FAIDX.out.fai.map { it[1] }\n",
      "Nextflow operator map with content: ch_fasta_path = ch_fasta_meta.map { it[1] }\n",
      "Nextflow operator view with content: ch_bam_input.view()\n",
      "Nextflow operator splitCsv with content: .splitCsv(header : true)\n",
      "Nextflow operator map with content: .map{ row ->\n",
      "Nextflow operator set with content: .set { comparisons_ch_s1 }\n",
      "Nextflow operator splitCsv with content: .splitCsv(header : true)\n",
      "Nextflow operator map with content: .map{ row ->\n",
      "Nextflow operator set with content: .set { comparisons_ch_s2 }\n",
      "Nextflow operator map with content: .map {meta, bam ->\n",
      "Nextflow operator set with content: .set { ch_bam_reformat }\n",
      "Nextflow operator combine with content: .combine(ch_bam_reformat , by:0)\n",
      "Nextflow operator map with content: .map { sample1, comparison, bam ->\n",
      "Nextflow operator set with content: .set { bam1_comparison }\n",
      "Nextflow operator combine with content: .combine(ch_bam_reformat, by:0)\n",
      "Nextflow operator map with content: .map{ sample2, comparison, bam ->\n",
      "Nextflow operator set with content: .set{ bam2_comparison }\n",
      "Nextflow operator join with content: .join(bam2_comparison, remainder: false, by: 0 )\n",
      "Nextflow operator set with content: .set{comparisons_merge_ch}\n",
      "Nextflow operator collectFile with content: ch_versions.unique().collectFile(name: 'collated_versions.yml')\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_workflow_summary.collectFile(name: 'workflow_summary_mqc.yaml'))\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_methods_description.collectFile(name: 'methods_description_mqc.yaml'))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(CUSTOM_DUMPSOFTWAREVERSIONS.out.mqc_yml.collect())\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(FASTQC.out.zip.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(BAM_MARKDUPLICATES_PICARD.out.stats.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(BAM_MARKDUPLICATES_PICARD.out.metrics.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(BAM_MARKDUPLICATES_PICARD.out.flagstat.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(BAM_MARKDUPLICATES_PICARD.out.idxstats.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files.collect(),\n",
      "Nextflow operator toList with content: ch_multiqc_config.toList(),\n",
      "Nextflow operator toList with content: ch_multiqc_custom_config.toList(),\n",
      "Nextflow operator toList with content: ch_multiqc_logo.toList()\n",
      "Nextflow operator toList with content: multiqc_report = MULTIQC.out.report.toList()\n",
      "Error: Unclosed process block in file: ./git_repos\\sammyseq\\workflows\\sammyseq.nf\n",
      "Process NFCORE_SAREK in filepath ./git_repos\\sarek\\main.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\sarek\\main.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\sarek\\conf\\modules\\mutect2.config\n",
      "Error: Unclosed process block in file: ./git_repos\\sarek\\modules\\local\\create_intervals_bed\\main.nf\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Process BAM_APPLYBQSR in filepath ./git_repos\\sarek\\subworkflows\\local\\bam_applybqsr\\main.nf\n",
      "Nextflow operator combine with content: cram_intervals = cram.combine(intervals)\n",
      "Nextflow operator map with content: .map{ meta, cram, crai, recal, intervals, num_intervals -> [ meta + [ num_intervals:num_intervals ], cram, crai, recal, intervals ] }\n",
      "Nextflow operator map with content: GATK4_APPLYBQSR(cram_intervals, fasta, fasta_fai, dict.map{ meta, it -> [ it ] })\n",
      "Nextflow operator groupTuple with content: cram_to_merge = GATK4_APPLYBQSR.out.cram.map{ meta, cram -> [ groupKey(meta, meta.num_intervals), cram ] }.groupTuple()\n",
      "Nextflow operator map with content: .map{ meta, cram, crai -> [ meta - meta.subMap('num_intervals'), cram, crai ] }\n",
      "Nextflow operator mix with content: versions = versions.mix(GATK4_APPLYBQSR.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(CRAM_MERGE_INDEX_SAMTOOLS.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\sarek\\subworkflows\\local\\bam_applybqsr\\main.nf\n",
      "Process BAM_APPLYBQSR_SPARK in filepath ./git_repos\\sarek\\subworkflows\\local\\bam_applybqsr_spark\\main.nf\n",
      "Nextflow operator combine with content: cram_intervals = cram.combine(intervals)\n",
      "Nextflow operator map with content: .map{ meta, cram, crai, recal, intervals, num_intervals -> [ meta + [ num_intervals:num_intervals ], cram, crai, recal, intervals ] }\n",
      "Nextflow operator map with content: GATK4SPARK_APPLYBQSR(cram_intervals, fasta, fasta_fai, dict.map{ meta, it -> [ it ] })\n",
      "Nextflow operator groupTuple with content: cram_to_merge = GATK4SPARK_APPLYBQSR.out.cram.map{ meta, cram -> [ groupKey(meta, meta.num_intervals), cram ] }.groupTuple()\n",
      "Nextflow operator map with content: .map{ meta, cram, crai -> [ meta - meta.subMap('num_intervals'), cram, crai ] }\n",
      "Nextflow operator mix with content: versions = versions.mix(GATK4SPARK_APPLYBQSR.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(CRAM_MERGE_INDEX_SAMTOOLS.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\sarek\\subworkflows\\local\\bam_applybqsr_spark\\main.nf\n",
      "Process BAM_BASERECALIBRATOR in filepath ./git_repos\\sarek\\subworkflows\\local\\bam_baserecalibrator\\main.nf\n",
      "Nextflow operator combine with content: cram_intervals = cram.combine(intervals)\n",
      "Nextflow operator map with content: .map{ meta, cram, crai, intervals, num_intervals -> [ meta + [ num_intervals:num_intervals ], cram, crai, intervals ] }\n",
      "Nextflow operator map with content: GATK4_BASERECALIBRATOR(cram_intervals, fasta, fasta_fai, dict.map{ meta, it -> [ it ] }, known_sites, known_sites_tbi)\n",
      "Nextflow operator branch with content: table_to_merge = GATK4_BASERECALIBRATOR.out.table.map{ meta, table -> [ groupKey(meta, meta.num_intervals), table ] }.groupTuple().branch{\n",
      "Nextflow operator map with content: table_bqsr = GATK4_GATHERBQSRREPORTS.out.table.mix(table_to_merge.single.map{ meta, table -> [ meta, table[0] ] })\n",
      "Nextflow operator map with content: .map{ meta, table -> [ meta - meta.subMap('num_intervals'), table ] }\n",
      "Nextflow operator mix with content: versions = versions.mix(GATK4_BASERECALIBRATOR.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(GATK4_GATHERBQSRREPORTS.out.versions)\n",
      "Process BAM_BASERECALIBRATOR_SPARK in filepath ./git_repos\\sarek\\subworkflows\\local\\bam_baserecalibrator_spark\\main.nf\n",
      "Nextflow operator combine with content: cram_intervals = cram.combine(intervals)\n",
      "Nextflow operator map with content: .map{ meta, cram, crai, intervals, num_intervals -> [ meta + [ num_intervals:num_intervals ], cram, crai, intervals ] }\n",
      "Nextflow operator map with content: GATK4SPARK_BASERECALIBRATOR(cram_intervals, fasta, fasta_fai, dict.map{ meta, it -> [ it ] }, known_sites, known_sites_tbi)\n",
      "Nextflow operator branch with content: table_to_merge = GATK4SPARK_BASERECALIBRATOR.out.table.map{ meta, table -> [ groupKey(meta, meta.num_intervals), table ] }.groupTuple().branch{\n",
      "Nextflow operator map with content: table_bqsr = GATK4_GATHERBQSRREPORTS.out.table.mix(table_to_merge.single.map{ meta, table -> [ meta, table[0] ] })\n",
      "Nextflow operator map with content: .map{ meta, table -> [ meta - meta.subMap('num_intervals'), table ] }\n",
      "Nextflow operator mix with content: versions = versions.mix(GATK4SPARK_BASERECALIBRATOR.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(GATK4_GATHERBQSRREPORTS.out.versions)\n",
      "Process BAM_CONVERT_SAMTOOLS in filepath ./git_repos\\sarek\\subworkflows\\local\\bam_convert_samtools\\main.nf\n",
      "Nextflow operator join with content: .join(SAMTOOLS_VIEW_UNMAP_MAP.out.bam, failOnDuplicate: true, remainder: true)\n",
      "Nextflow operator join with content: .join(SAMTOOLS_VIEW_MAP_UNMAP.out.bam, failOnDuplicate: true, remainder: true)\n",
      "Nextflow operator map with content: .map{ meta, unmap_unmap, unmap_map, map_unmap -> [ meta, [ unmap_unmap, unmap_map, map_unmap ] ] }\n",
      "Nextflow operator join with content: .join(COLLATE_FASTQ_UNMAP.out.fastq, failOnDuplicate: true, failOnMismatch: true)\n",
      "Nextflow operator map with content: .map{ meta, mapped_reads, unmapped_reads -> [ meta, [ mapped_reads[0], mapped_reads[1], unmapped_reads[0], unmapped_reads[1] ] ] }\n",
      "Nextflow operator mix with content: versions = versions.mix(CAT_FASTQ.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(COLLATE_FASTQ_MAP.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(COLLATE_FASTQ_UNMAP.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(SAMTOOLS_MERGE_UNMAP.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(SAMTOOLS_VIEW_MAP_MAP.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(SAMTOOLS_VIEW_MAP_UNMAP.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(SAMTOOLS_VIEW_UNMAP_MAP.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(SAMTOOLS_VIEW_UNMAP_UNMAP.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\sarek\\subworkflows\\local\\bam_convert_samtools\\main.nf\n",
      "Process BAM_JOINT_CALLING_GERMLINE_GATK in filepath ./git_repos\\sarek\\subworkflows\\local\\bam_joint_calling_germline_gatk\\main.nf\n",
      "Nextflow operator map with content: .map{ meta, gvcf, tbi, intervals -> [ [ id:'joint_variant_calling', intervals_name:intervals.simpleName, num_intervals:meta.num_intervals ], gvcf, tbi, intervals ] }\n",
      "Nextflow operator groupTuple with content: .groupTuple(by:3) //join on interval file\n",
      "Nextflow operator map with content: .map{ meta_list, gvcf, tbi, intervals ->\n",
      "Nextflow operator map with content: genotype_input = GATK4_GENOMICSDBIMPORT.out.genomicsdb.map{ meta, genomicsdb -> [ meta, genomicsdb, [], [], [] ] }\n",
      "Nextflow operator map with content: GATK4_GENOTYPEGVCFS(genotype_input, fasta, fai, dict.map{ meta, dict -> [ dict ] }, dbsnp, dbsnp_tbi)\n",
      "Nextflow operator groupTuple with content: gvcf_to_merge = BCFTOOLS_SORT.out.vcf.map{ meta, vcf -> [ meta.subMap('num_intervals') + [ id:'joint_variant_calling', patient:'all_samples', variantcaller:'haplotypecaller' ], vcf ]}.groupTuple()\n",
      "Nextflow operator join with content: vqsr_input = MERGE_GENOTYPEGVCFS.out.vcf.join(MERGE_GENOTYPEGVCFS.out.tbi, failOnDuplicate: true)\n",
      "Nextflow operator collect with content: indels_resource_label = known_indels_vqsr.mix(dbsnp_vqsr).collect()\n",
      "Nextflow operator collect with content: snps_resource_label = known_snps_vqsr.mix(dbsnp_vqsr).collect()\n",
      "Nextflow operator map with content: dict.map{ meta, dict -> [ dict ] })\n",
      "Nextflow operator map with content: dict.map{ meta, dict -> [ dict ] })\n",
      "Nextflow operator join with content: vqsr_input_snp = vqsr_input.join(VARIANTRECALIBRATOR_SNP.out.recal, failOnDuplicate: true)\n",
      "Nextflow operator join with content: .join(VARIANTRECALIBRATOR_SNP.out.idx, failOnDuplicate: true)\n",
      "Nextflow operator join with content: .join(VARIANTRECALIBRATOR_SNP.out.tranches, failOnDuplicate: true)\n",
      "Nextflow operator map with content: .map{ meta, vcf, tbi, recal, index, tranche -> [ meta + [ id:'recalibrated_joint_variant_calling' ], vcf, tbi, recal, index, tranche ] }\n",
      "Nextflow operator map with content: dict.map{ meta, dict -> [ dict ] })\n",
      "Nextflow operator join with content: vqsr_input_indel = GATK4_APPLYVQSR_SNP.out.vcf.join(GATK4_APPLYVQSR_SNP.out.tbi).map{ meta, vcf, tbi -> [ meta + [ id:'joint_variant_calling' ], vcf, tbi ]}\n",
      "Nextflow operator join with content: .join(VARIANTRECALIBRATOR_INDEL.out.recal, failOnDuplicate: true)\n",
      "Nextflow operator join with content: .join(VARIANTRECALIBRATOR_INDEL.out.idx, failOnDuplicate: true)\n",
      "Nextflow operator join with content: .join(VARIANTRECALIBRATOR_INDEL.out.tranches, failOnDuplicate: true)\n",
      "Nextflow operator map with content: .map{ meta, vcf, tbi, recal, index, tranche -> [ meta + [ id:'recalibrated_joint_variant_calling' ], vcf, tbi, recal, index, tranche ] }\n",
      "Nextflow operator map with content: dict.map{ meta, dict -> [ dict ] })\n",
      "Nextflow operator map with content: merge_vcf_for_join = MERGE_GENOTYPEGVCFS.out.vcf.map{meta, vcf -> [[id: 'joint_variant_calling'] , vcf]}\n",
      "Nextflow operator map with content: merge_tbi_for_join = MERGE_GENOTYPEGVCFS.out.tbi.map{meta, tbi -> [[id: 'joint_variant_calling'] , tbi]}\n",
      "Nextflow operator ifEmpty with content: vqsr_vcf_for_join = GATK4_APPLYVQSR_INDEL.out.vcf.ifEmpty([[:], []]).map{meta, vcf -> [[id: 'joint_variant_calling'] , vcf]}\n",
      "Nextflow operator ifEmpty with content: vqsr_tbi_for_join = GATK4_APPLYVQSR_INDEL.out.tbi.ifEmpty([[:], []]).map{meta, tbi -> [[id: 'joint_variant_calling'] , tbi]}\n",
      "Nextflow operator join with content: genotype_vcf = merge_vcf_for_join.join(vqsr_vcf_for_join, remainder: true).map{\n",
      "Nextflow operator join with content: genotype_index = merge_tbi_for_join.join(vqsr_tbi_for_join, remainder: true).map{\n",
      "Nextflow operator mix with content: versions = versions.mix(GATK4_GENOMICSDBIMPORT.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(GATK4_GENOTYPEGVCFS.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(VARIANTRECALIBRATOR_SNP.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(GATK4_APPLYVQSR_SNP.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\sarek\\subworkflows\\local\\bam_joint_calling_germline_gatk\\main.nf\n",
      "Process BAM_JOINT_CALLING_GERMLINE_SENTIEON in filepath ./git_repos\\sarek\\subworkflows\\local\\bam_joint_calling_germline_sentieon\\main.nf\n",
      "Nextflow operator map with content: .map{ meta, gvcf, tbi, intervals -> [ [ id:'joint_variant_calling', intervals_name:intervals.simpleName, num_intervals:meta.num_intervals ], gvcf, tbi, intervals ] }\n",
      "Nextflow operator groupTuple with content: .groupTuple(by:[0, 3])\n",
      "Nextflow operator groupTuple with content: gvcf_to_merge = BCFTOOLS_SORT.out.vcf.map{ meta, vcf -> [ meta.subMap('num_intervals') + [ id:'joint_variant_calling', patient:'all_samples', variantcaller:variant_caller ], vcf ]}.groupTuple()\n",
      "Nextflow operator map with content: merged_vcf = MERGE_GENOTYPEGVCFS.out.vcf.map{meta, vcf -> [[id: 'joint_variant_calling'] , vcf]}\n",
      "Nextflow operator map with content: merged_tbi = MERGE_GENOTYPEGVCFS.out.tbi.map{meta, tbi -> [[id: 'joint_variant_calling'] , tbi]}\n",
      "Nextflow operator map with content: genotype_vcf   = merged_vcf.map{\n",
      "Nextflow operator map with content: genotype_index = merged_tbi.map{\n",
      "Nextflow operator join with content: vqsr_input = MERGE_GENOTYPEGVCFS.out.vcf.join(MERGE_GENOTYPEGVCFS.out.tbi, failOnDuplicate: true)\n",
      "Nextflow operator collect with content: indels_resource_label = known_indels_vqsr.mix(dbsnp_vqsr).collect()\n",
      "Nextflow operator collect with content: snps_resource_label = known_snps_vqsr.mix(dbsnp_vqsr).collect()\n",
      "Nextflow operator join with content: vqsr_input_snp = vqsr_input.join(SENTIEON_VARCAL_SNP.out.recal, failOnDuplicate: true)\n",
      "Nextflow operator join with content: .join(SENTIEON_VARCAL_SNP.out.idx, failOnDuplicate: true)\n",
      "Nextflow operator join with content: .join(SENTIEON_VARCAL_SNP.out.tranches, failOnDuplicate: true)\n",
      "Nextflow operator map with content: .map{ meta, vcf, tbi, recal, index, tranche -> [ meta + [ id:'recalibrated_joint_variant_calling' ], vcf, tbi, recal, index, tranche ] }\n",
      "Nextflow operator map with content: fasta.map{ fasta -> [ [ id:fasta.baseName ], fasta ] },\n",
      "Nextflow operator map with content: fai.map{ fai -> [ [ id:fai.baseName ], fai ] })\n",
      "Nextflow operator join with content: vqsr_input_indel = SENTIEON_APPLYVARCAL_SNP.out.vcf.join(SENTIEON_APPLYVARCAL_SNP.out.tbi).map{ meta, vcf, tbi -> [ meta + [ id:'joint_variant_calling' ], vcf, tbi ]}\n",
      "Nextflow operator join with content: .join(SENTIEON_VARCAL_INDEL.out.recal, failOnDuplicate: true)\n",
      "Nextflow operator join with content: .join(SENTIEON_VARCAL_INDEL.out.idx, failOnDuplicate: true)\n",
      "Nextflow operator join with content: .join(SENTIEON_VARCAL_INDEL.out.tranches, failOnDuplicate: true)\n",
      "Nextflow operator map with content: .map{ meta, vcf, tbi, recal, index, tranche -> [ meta + [ id:'recalibrated_joint_variant_calling' ], vcf, tbi, recal, index, tranche ] }\n",
      "Nextflow operator map with content: fasta.map{ fasta -> [ [ id:fasta.baseName ], fasta ] },\n",
      "Nextflow operator map with content: fai.map{ fai -> [ [ id:fai.baseName ], fai ] })\n",
      "Nextflow operator ifEmpty with content: vqsr_vcf_for_join = SENTIEON_APPLYVARCAL_INDEL.out.vcf.ifEmpty([[:], []]).map{meta, vcf -> [[id: 'joint_variant_calling'] , vcf]}\n",
      "Nextflow operator ifEmpty with content: vqsr_tbi_for_join = SENTIEON_APPLYVARCAL_INDEL.out.tbi.ifEmpty([[:], []]).map{meta, tbi -> [[id: 'joint_variant_calling'] , tbi]}\n",
      "Nextflow operator join with content: genotype_vcf = merged_vcf.join(vqsr_vcf_for_join, remainder: true).map{\n",
      "Nextflow operator join with content: genotype_index = merged_tbi.join(vqsr_tbi_for_join, remainder: true).map{\n",
      "Nextflow operator mix with content: versions = versions.mix(SENTIEON_VARCAL_SNP.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(SENTIEON_VARCAL_INDEL.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(SENTIEON_APPLYVARCAL_INDEL.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(SENTIEON_GVCFTYPER.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\sarek\\subworkflows\\local\\bam_joint_calling_germline_sentieon\\main.nf\n",
      "Process BAM_MARKDUPLICATES in filepath ./git_repos\\sarek\\subworkflows\\local\\bam_markduplicates\\main.nf\n",
      "Nextflow operator join with content: cram = GATK4_MARKDUPLICATES.out.cram.join(GATK4_MARKDUPLICATES.out.crai, failOnDuplicate: true, failOnMismatch: true)\n",
      "Nextflow operator mix with content: reports = reports.mix(GATK4_MARKDUPLICATES.out.metrics)\n",
      "Nextflow operator mix with content: reports = reports.mix(CRAM_QC_MOSDEPTH_SAMTOOLS.out.reports)\n",
      "Nextflow operator mix with content: versions = versions.mix(GATK4_MARKDUPLICATES.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(CRAM_QC_MOSDEPTH_SAMTOOLS.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\sarek\\subworkflows\\local\\bam_markduplicates\\main.nf\n",
      "Process BAM_MARKDUPLICATES_SPARK in filepath ./git_repos\\sarek\\subworkflows\\local\\bam_markduplicates_spark\\main.nf\n",
      "Nextflow operator join with content: cram = GATK4SPARK_MARKDUPLICATES.out.output.join(INDEX_MARKDUPLICATES.out.crai, failOnDuplicate: true, failOnMismatch: true)\n",
      "Nextflow operator mix with content: reports = reports.mix(GATK4_ESTIMATELIBRARYCOMPLEXITY.out.metrics)\n",
      "Nextflow operator mix with content: reports = reports.mix(CRAM_QC_MOSDEPTH_SAMTOOLS.out.reports)\n",
      "Nextflow operator mix with content: versions = versions.mix(GATK4_ESTIMATELIBRARYCOMPLEXITY.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(GATK4SPARK_MARKDUPLICATES.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(INDEX_MARKDUPLICATES.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(CRAM_QC_MOSDEPTH_SAMTOOLS.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\sarek\\subworkflows\\local\\bam_markduplicates_spark\\main.nf\n",
      "Process BAM_MERGE_INDEX_SAMTOOLS in filepath ./git_repos\\sarek\\subworkflows\\local\\bam_merge_index_samtools\\main.nf\n",
      "Nextflow operator branch with content: bam_to_merge = bam.branch{ meta, bam ->\n",
      "Nextflow operator mix with content: bam_all = MERGE_BAM.out.bam.mix(bam_to_merge.single)\n",
      "Nextflow operator join with content: bam_bai = bam_all.join(INDEX_MERGE_BAM.out.bai, failOnDuplicate: true, failOnMismatch: true)\n",
      "Nextflow operator mix with content: versions = versions.mix(INDEX_MERGE_BAM.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(MERGE_BAM.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\sarek\\subworkflows\\local\\bam_merge_index_samtools\\main.nf\n",
      "Process BAM_SENTIEON_DEDUP in filepath ./git_repos\\sarek\\subworkflows\\local\\bam_sentieon_dedup\\main.nf\n",
      "Nextflow operator map with content: bam = bam.map{ meta, bam -> [ meta - meta.subMap('data_type'), bam ] }\n",
      "Nextflow operator map with content: bai = bai.map{ meta, bai -> [ meta - meta.subMap('data_type'), bai ] }\n",
      "Nextflow operator join with content: bam_bai = bam.join(bai, failOnMismatch:true, failOnDuplicate:true)\n",
      "Nextflow operator map with content: SENTIEON_DEDUP(bam_bai, fasta.map{fa -> [[:], fa]}, fasta_fai.map{fai -> [[:], fai]})\n",
      "Nextflow operator join with content: cram = SENTIEON_DEDUP.out.cram.join(SENTIEON_DEDUP.out.crai, failOnDuplicate: true, failOnMismatch: true)\n",
      "Nextflow operator mix with content: reports = reports.mix(SENTIEON_DEDUP.out.metrics)\n",
      "Nextflow operator mix with content: reports = reports.mix(SENTIEON_DEDUP.out.metrics_multiqc_tsv)\n",
      "Nextflow operator mix with content: reports = reports.mix(SENTIEON_DEDUP.out.score)\n",
      "Nextflow operator mix with content: reports = reports.mix(CRAM_QC_MOSDEPTH_SAMTOOLS.out.reports)\n",
      "Nextflow operator mix with content: versions = versions.mix(SENTIEON_DEDUP.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(CRAM_QC_MOSDEPTH_SAMTOOLS.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\sarek\\subworkflows\\local\\bam_sentieon_dedup\\main.nf\n",
      "Process BAM_VARIANT_CALLING_CNVKIT in filepath ./git_repos\\sarek\\subworkflows\\local\\bam_variant_calling_cnvkit\\main.nf\n",
      "Nextflow operator join with content: ch_genemetrics = CNVKIT_BATCH.out.cnr.join(CNVKIT_BATCH.out.cns).map{ meta, cnr, cns -> [meta, cnr, cns[2]]}\n",
      "Nextflow operator mix with content: versions = versions.mix(CNVKIT_BATCH.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(CNVKIT_GENEMETRICS.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\sarek\\subworkflows\\local\\bam_variant_calling_cnvkit\\main.nf\n",
      "Process BAM_VARIANT_CALLING_DEEPVARIANT in filepath ./git_repos\\sarek\\subworkflows\\local\\bam_variant_calling_deepvariant\\main.nf\n",
      "Nextflow operator combine with content: cram_intervals = cram.combine(intervals)\n",
      "Nextflow operator map with content: .map{ meta, cram, crai, intervals, num_intervals -> [ meta + [ num_intervals:num_intervals ], cram, crai, intervals ]}\n",
      "Nextflow operator map with content: DEEPVARIANT(cram_intervals, fasta.map{ fasta -> [ [ id:fasta.baseName ], fasta ] }, fasta_fai.map{ fasta_fai -> [ [ id:fasta_fai.baseName ], fasta_fai ] }, [ [ id:'null' ], [] ])\n",
      "Nextflow operator branch with content: vcf_out = DEEPVARIANT.out.vcf.branch{\n",
      "Nextflow operator branch with content: gvcf_out = DEEPVARIANT.out.gvcf.branch{\n",
      "Nextflow operator groupTuple with content: gvcf_to_merge = gvcf_out.intervals.map{ meta, vcf -> [ groupKey(meta, meta.num_intervals), vcf ]}.groupTuple()\n",
      "Nextflow operator groupTuple with content: vcf_to_merge = vcf_out.intervals.map{ meta, vcf -> [ groupKey(meta, meta.num_intervals), vcf ]}.groupTuple()\n",
      "Nextflow operator mix with content: gvcf = Channel.empty().mix(MERGE_DEEPVARIANT_GVCF.out.vcf, gvcf_out.no_intervals)\n",
      "Nextflow operator map with content: .map{ meta, vcf -> [ meta - meta.subMap('num_intervals') + [ variantcaller:'deepvariant' ], vcf ] }\n",
      "Nextflow operator mix with content: vcf = Channel.empty().mix(MERGE_DEEPVARIANT_VCF.out.vcf, vcf_out.no_intervals)\n",
      "Nextflow operator map with content: .map{ meta, vcf -> [ meta - meta.subMap('num_intervals') + [ variantcaller:'deepvariant' ], vcf ] }\n",
      "Nextflow operator mix with content: versions = versions.mix(DEEPVARIANT.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(MERGE_DEEPVARIANT_GVCF.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(MERGE_DEEPVARIANT_VCF.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\sarek\\subworkflows\\local\\bam_variant_calling_deepvariant\\main.nf\n",
      "Process BAM_VARIANT_CALLING_FREEBAYES in filepath ./git_repos\\sarek\\subworkflows\\local\\bam_variant_calling_freebayes\\main.nf\n",
      "Nextflow operator combine with content: cram_intervals = cram.combine(intervals)\n",
      "Nextflow operator map with content: .map{ meta, cram1, crai1, cram2, crai2, intervals, num_intervals -> [ meta + [ num_intervals:num_intervals ], cram1, crai1, cram2, crai2, intervals ]}\n",
      "Nextflow operator branch with content: bcftools_vcf_out = BCFTOOLS_SORT.out.vcf.branch{\n",
      "Nextflow operator groupTuple with content: vcf_to_merge = bcftools_vcf_out.intervals.map{ meta, vcf -> [ groupKey(meta, meta.num_intervals), vcf ]}.groupTuple()\n",
      "Nextflow operator mix with content: vcf = MERGE_FREEBAYES.out.vcf.mix(bcftools_vcf_out.no_intervals)\n",
      "Nextflow operator map with content: .map{ meta, vcf -> [ meta - meta.subMap('num_intervals') + [ variantcaller:'freebayes' ], vcf ] }\n",
      "Nextflow operator mix with content: versions = versions.mix(BCFTOOLS_SORT.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(MERGE_FREEBAYES.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(FREEBAYES.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(TABIX_VC_FREEBAYES.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\sarek\\subworkflows\\local\\bam_variant_calling_freebayes\\main.nf\n",
      "Process BAM_VARIANT_CALLING_GERMLINE_ALL in filepath ./git_repos\\sarek\\subworkflows\\local\\bam_variant_calling_germline_all\\main.nf\n",
      "Nextflow operator mix with content: versions = versions.mix(BAM_VARIANT_CALLING_MPILEUP.out.versions)\n",
      "Nextflow operator map with content: cram.map{ meta, cram, crai -> [ meta, [], cram ] },\n",
      "Nextflow operator map with content: fasta.map{ it -> [[id:it[0].baseName], it] },\n",
      "Nextflow operator map with content: fasta_fai.map{ it -> [[id:it[0].baseName], it] },\n",
      "Nextflow operator map with content: intervals_bed_combined.map{ it -> [[id:it[0].baseName], it] },\n",
      "Nextflow operator mix with content: versions = versions.mix(BAM_VARIANT_CALLING_CNVKIT.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(BAM_VARIANT_CALLING_DEEPVARIANT.out.versions)\n",
      "Nextflow operator map with content: cram.map{ meta, cram, crai -> [ meta, cram, crai, [], [] ] },\n",
      "Nextflow operator mix with content: versions = versions.mix(BAM_VARIANT_CALLING_FREEBAYES.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(BAM_VARIANT_CALLING_HAPLOTYPECALLER.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(BAM_JOINT_CALLING_GERMLINE_GATK.out.versions)\n",
      "Nextflow operator join with content: vcf_haplotypecaller.join(tbi_haplotypecaller, failOnDuplicate: true, failOnMismatch: true),\n",
      "Nextflow operator map with content: dict.map{ meta, dict -> [ dict ] },\n",
      "Nextflow operator collect with content: known_sites_indels.concat(known_sites_snps).flatten().unique().collect(),\n",
      "Nextflow operator collect with content: known_sites_indels_tbi.concat(known_sites_snps_tbi).flatten().unique().collect())\n",
      "Nextflow operator mix with content: versions = versions.mix(VCF_VARIANT_FILTERING_GATK.out.versions)\n",
      "Nextflow operator map with content: fasta.map{ it -> [ [ id:'fasta' ], it ] },\n",
      "Nextflow operator map with content: fasta_fai.map{ it -> [ [ id:'fasta_fai' ], it ] },\n",
      "Nextflow operator mix with content: versions = versions.mix(BAM_VARIANT_CALLING_GERMLINE_MANTA.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(BAM_VARIANT_CALLING_SENTIEON_DNASCOPE.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(BAM_JOINT_CALLING_GERMLINE_SENTIEON.out.versions)\n",
      "Nextflow operator join with content: vcf_sentieon_dnascope.join(vcf_tbi_sentieon_dnascope, failOnDuplicate: true, failOnMismatch: true),\n",
      "Nextflow operator map with content: fasta.map{ fasta -> [ [ id:fasta.baseName ], fasta ] },\n",
      "Nextflow operator map with content: fasta_fai.map{ fai -> [ [ id:fai.baseName ], fai ] },\n",
      "Nextflow operator map with content: sentieon_dnascope_model.map{ model -> [ [ id:model.baseName ], model ] })\n",
      "Nextflow operator mix with content: versions = versions.mix(SENTIEON_DNAMODELAPPLY.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(BAM_VARIANT_CALLING_SENTIEON_HAPLOTYPER.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(BAM_JOINT_CALLING_GERMLINE_SENTIEON.out.versions)\n",
      "Nextflow operator join with content: vcf_sentieon_haplotyper.join(vcf_tbi_sentieon_haplotyper, failOnDuplicate: true, failOnMismatch: true),\n",
      "Nextflow operator map with content: dict.map{ meta, dict -> [ dict ] },\n",
      "Nextflow operator collect with content: known_sites_indels.concat(known_sites_snps).flatten().unique().collect(),\n",
      "Nextflow operator collect with content: known_sites_indels_tbi.concat(known_sites_snps_tbi).flatten().unique().collect())\n",
      "Nextflow operator mix with content: versions = versions.mix(SENTIEON_HAPLOTYPER_VCF_VARIANT_FILTERING_GATK.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(BAM_VARIANT_CALLING_SINGLE_STRELKA.out.versions)\n",
      "Nextflow operator map with content: fasta.map{ it -> [ [ id:'fasta' ], it ] },\n",
      "Nextflow operator mix with content: versions = versions.mix(BAM_VARIANT_CALLING_SINGLE_TIDDIT.out.versions)\n",
      "Nextflow operator mix with content: vcf_all = Channel.empty().mix(\n",
      "Error: Unclosed process block in file: ./git_repos\\sarek\\subworkflows\\local\\bam_variant_calling_germline_all\\main.nf\n",
      "Process BAM_VARIANT_CALLING_GERMLINE_MANTA in filepath ./git_repos\\sarek\\subworkflows\\local\\bam_variant_calling_germline_manta\\main.nf\n",
      "Nextflow operator combine with content: cram_intervals = cram.combine(intervals).map{ it ->\n",
      "Nextflow operator map with content: vcf = diploid_sv_vcf.map{ meta, vcf -> [ meta + [ variantcaller:'manta' ], vcf ] }\n",
      "Nextflow operator mix with content: versions = versions.mix(MANTA_GERMLINE.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\sarek\\subworkflows\\local\\bam_variant_calling_germline_manta\\main.nf\n",
      "Process BAM_VARIANT_CALLING_HAPLOTYPECALLER in filepath ./git_repos\\sarek\\subworkflows\\local\\bam_variant_calling_haplotypecaller\\main.nf\n",
      "Nextflow operator combine with content: cram_intervals = cram.combine(intervals)\n",
      "Nextflow operator map with content: .map{ meta, cram, crai, intervals, num_intervals -> [ meta + [ interval_name:intervals.simpleName, num_intervals:num_intervals, variantcaller:'haplotypecaller' ], cram, crai, intervals, [] ] }\n",
      "Nextflow operator map with content: GATK4_HAPLOTYPECALLER(cram_intervals, fasta, fasta_fai, dict.map{ meta, dict -> [ dict ] }, dbsnp, dbsnp_tbi)\n",
      "Nextflow operator join with content: .join(GATK4_HAPLOTYPECALLER.out.tbi, failOnMismatch: true)\n",
      "Nextflow operator join with content: .join(cram_intervals, failOnMismatch: true)\n",
      "Nextflow operator map with content: .map{ meta, gvcf, tbi, cram, crai, intervals, dragstr_model -> [ meta, gvcf, tbi, intervals ] }\n",
      "Nextflow operator map with content: haplotypecaller_vcf = GATK4_HAPLOTYPECALLER.out.vcf.map{\n",
      "Nextflow operator branch with content: .branch{\n",
      "Nextflow operator map with content: haplotypecaller_tbi = GATK4_HAPLOTYPECALLER.out.tbi.map{\n",
      "Nextflow operator branch with content: }.branch{\n",
      "Nextflow operator map with content: haplotypecaller_bam = GATK4_HAPLOTYPECALLER.out.bam.map{\n",
      "Nextflow operator branch with content: }.branch{\n",
      "Nextflow operator groupTuple with content: MERGE_HAPLOTYPECALLER(haplotypecaller_vcf.intervals.map{ meta, vcf -> [ groupKey(meta, meta.num_intervals), vcf ] }.groupTuple(), dict)\n",
      "Nextflow operator mix with content: haplotypecaller_vcf = Channel.empty().mix(\n",
      "Nextflow operator mix with content: haplotypecaller_tbi = Channel.empty().mix(\n",
      "Nextflow operator map with content: .map{ meta, bam -> [ groupKey(meta, meta.num_intervals), bam ] }\n",
      "Nextflow operator groupTuple with content: .groupTuple()\n",
      "Nextflow operator mix with content: .mix(haplotypecaller_bam.no_intervals))\n",
      "Nextflow operator mix with content: versions = versions.mix(GATK4_HAPLOTYPECALLER.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(MERGE_HAPLOTYPECALLER.out.versions)\n",
      "Nextflow operator map with content: vcf = haplotypecaller_vcf.map{ meta, vcf -> [ meta - meta.subMap('num_intervals'), vcf ] }\n",
      "Nextflow operator map with content: tbi = haplotypecaller_tbi.map{ meta, tbi -> [ meta - meta.subMap('num_intervals'), tbi ] }\n",
      "Error: Unclosed process block in file: ./git_repos\\sarek\\subworkflows\\local\\bam_variant_calling_haplotypecaller\\main.nf\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Process BAM_VARIANT_CALLING_MPILEUP in filepath ./git_repos\\sarek\\subworkflows\\local\\bam_variant_calling_mpileup\\main.nf\n",
      "Nextflow operator combine with content: cram_intervals = cram.combine(intervals)\n",
      "Nextflow operator map with content: .map{ meta, cram, crai, intervals, num_intervals -> [ meta + [ num_intervals:num_intervals ], cram, intervals ] }\n",
      "Nextflow operator map with content: BCFTOOLS_MPILEUP(cram_intervals, fasta.map{ it -> [[id:it[0].baseName], it] }, keep_bcftools_mpileup)\n",
      "Nextflow operator branch with content: vcf_mpileup = BCFTOOLS_MPILEUP.out.vcf.branch{\n",
      "Nextflow operator branch with content: mpileup_samtools = SAMTOOLS_MPILEUP.out.mpileup.branch{\n",
      "Nextflow operator groupTuple with content: mpileup_to_merge = mpileup_samtools.intervals.map{ meta, pileup -> [ groupKey(meta, meta.num_intervals), pileup ] }.groupTuple(sort:true)\n",
      "Nextflow operator groupTuple with content: vcf_to_merge = vcf_mpileup.intervals.map{ meta, vcf -> [ groupKey(meta, meta.num_intervals), vcf ] }.groupTuple()\n",
      "Nextflow operator mix with content: mpileup = CAT_MPILEUP.out.file_out.mix(mpileup_samtools.no_intervals)\n",
      "Nextflow operator map with content: .map{ meta, mpileup -> [ meta - meta.subMap('num_intervals') + [ variantcaller:'samtools' ], mpileup ] }\n",
      "Nextflow operator mix with content: vcf = MERGE_BCFTOOLS_MPILEUP.out.vcf.mix(vcf_mpileup.no_intervals)\n",
      "Nextflow operator map with content: .map{ meta, vcf -> [ meta - meta.subMap('num_intervals') + [ variantcaller:'bcftools' ], vcf ] }\n",
      "Nextflow operator mix with content: versions = versions.mix(SAMTOOLS_MPILEUP.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(BCFTOOLS_MPILEUP.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(CAT_MPILEUP.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(MERGE_BCFTOOLS_MPILEUP.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\sarek\\subworkflows\\local\\bam_variant_calling_mpileup\\main.nf\n",
      "Process BAM_VARIANT_CALLING_SENTIEON_DNASCOPE in filepath ./git_repos\\sarek\\subworkflows\\local\\bam_variant_calling_sentieon_dnascope\\main.nf\n",
      "Nextflow operator combine with content: cram_intervals_for_sentieon = cram.combine(intervals)\n",
      "Nextflow operator map with content: .map{ meta, cram, crai, intervals, num_intervals -> [\n",
      "Nextflow operator map with content: fasta.map{it -> [[:], it]},\n",
      "Nextflow operator map with content: fasta_fai.map{it -> [[:], it]},\n",
      "Nextflow operator map with content: dbsnp.map{it -> [[:], it]},\n",
      "Nextflow operator map with content: dbsnp_tbi.map{it -> [[:], it]},\n",
      "Nextflow operator map with content: sentieon_dnascope_model.map{it -> [[:], it]},\n",
      "Nextflow operator join with content: .join(SENTIEON_DNASCOPE.out.gvcf_tbi, failOnMismatch: true)\n",
      "Nextflow operator join with content: .join(cram_intervals_for_sentieon, failOnMismatch: true)\n",
      "Nextflow operator map with content: .map{ meta, gvcf, tbi, cram, crai, intervals -> [ meta, gvcf, tbi, intervals ] }\n",
      "Nextflow operator map with content: dnascope_vcf_branch = SENTIEON_DNASCOPE.out.vcf.map{\n",
      "Nextflow operator branch with content: .branch{\n",
      "Nextflow operator map with content: dnascope_vcf_tbi_branch = SENTIEON_DNASCOPE.out.vcf_tbi.map{\n",
      "Nextflow operator branch with content: .branch{\n",
      "Nextflow operator map with content: haplotyper_gvcf_branch = SENTIEON_DNASCOPE.out.gvcf.map{\n",
      "Nextflow operator branch with content: .branch{\n",
      "Nextflow operator map with content: haplotyper_gvcf_tbi_branch = SENTIEON_DNASCOPE.out.gvcf_tbi.map{\n",
      "Nextflow operator branch with content: .branch{\n",
      "Nextflow operator map with content: vcfs_for_merging = dnascope_vcf_branch.intervals.map{\n",
      "Nextflow operator map with content: vcfs_for_merging = vcfs_for_merging.map{\n",
      "Nextflow operator groupTuple with content: vcf]}.groupTuple()\n",
      "Nextflow operator mix with content: dnascope_vcf = Channel.empty().mix(\n",
      "Nextflow operator mix with content: haplotyper_tbi = Channel.empty().mix(\n",
      "Nextflow operator map with content: vcf = dnascope_vcf.map{ meta, vcf -> [ meta - meta.subMap('num_intervals'), vcf ] }\n",
      "Nextflow operator map with content: vcf_tbi = haplotyper_tbi.map{ meta, tbi -> [ meta - meta.subMap('num_intervals'), tbi ] }\n",
      "Nextflow operator map with content: gvcfs_for_merging = haplotyper_gvcf_branch.intervals.map{\n",
      "Nextflow operator map with content: gvcfs_for_merging = gvcfs_for_merging.map{\n",
      "Nextflow operator groupTuple with content: }.groupTuple()\n",
      "Nextflow operator mix with content: gvcf = Channel.empty().mix(\n",
      "Nextflow operator mix with content: gvcf_tbi = Channel.empty().mix(\n",
      "Nextflow operator mix with content: versions = versions.mix(SENTIEON_DNASCOPE.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(MERGE_SENTIEON_DNASCOPE_VCFS.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(MERGE_SENTIEON_DNASCOPE_GVCFS.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\sarek\\subworkflows\\local\\bam_variant_calling_sentieon_dnascope\\main.nf\n",
      "Process BAM_VARIANT_CALLING_SENTIEON_HAPLOTYPER in filepath ./git_repos\\sarek\\subworkflows\\local\\bam_variant_calling_sentieon_haplotyper\\main.nf\n",
      "Nextflow operator combine with content: cram_intervals_for_sentieon = cram.combine(intervals)\n",
      "Nextflow operator map with content: .map{ meta, cram, crai, intervals, num_intervals -> [\n",
      "Nextflow operator join with content: .join(SENTIEON_HAPLOTYPER.out.gvcf_tbi, failOnMismatch: true)\n",
      "Nextflow operator join with content: .join(cram_intervals_for_sentieon, failOnMismatch: true)\n",
      "Nextflow operator map with content: .map{ meta, gvcf, tbi, cram, crai, intervals -> [ meta, gvcf, tbi, intervals ] }\n",
      "Nextflow operator map with content: haplotyper_vcf_branch = SENTIEON_HAPLOTYPER.out.vcf.map{\n",
      "Nextflow operator branch with content: .branch{\n",
      "Nextflow operator map with content: haplotyper_vcf_tbi_branch = SENTIEON_HAPLOTYPER.out.vcf_tbi.map{\n",
      "Nextflow operator branch with content: .branch{\n",
      "Nextflow operator map with content: haplotyper_gvcf_branch = SENTIEON_HAPLOTYPER.out.gvcf.map{\n",
      "Nextflow operator branch with content: .branch{\n",
      "Nextflow operator map with content: haplotyper_gvcf_tbi_branch = SENTIEON_HAPLOTYPER.out.gvcf_tbi.map{\n",
      "Nextflow operator branch with content: .branch{\n",
      "Nextflow operator map with content: vcfs_for_merging = haplotyper_vcf_branch.intervals.map{\n",
      "Nextflow operator map with content: vcfs_for_merging = vcfs_for_merging.map{\n",
      "Nextflow operator groupTuple with content: vcf]}.groupTuple()\n",
      "Nextflow operator mix with content: haplotyper_vcf = Channel.empty().mix(\n",
      "Nextflow operator mix with content: haplotyper_tbi = Channel.empty().mix(\n",
      "Nextflow operator map with content: vcf = haplotyper_vcf.map{ meta, vcf -> [ meta - meta.subMap('num_intervals'), vcf ] }\n",
      "Nextflow operator map with content: vcf_tbi = haplotyper_tbi.map{ meta, tbi -> [ meta - meta.subMap('num_intervals'), tbi ] }\n",
      "Nextflow operator map with content: gvcfs_for_merging = haplotyper_gvcf_branch.intervals.map{\n",
      "Nextflow operator map with content: gvcfs_for_merging = gvcfs_for_merging.map{\n",
      "Nextflow operator groupTuple with content: }.groupTuple()\n",
      "Nextflow operator mix with content: gvcf = Channel.empty().mix(\n",
      "Nextflow operator mix with content: gvcf_tbi = Channel.empty().mix(\n",
      "Nextflow operator mix with content: versions = versions.mix(SENTIEON_HAPLOTYPER.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(MERGE_SENTIEON_HAPLOTYPER_VCFS.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(MERGE_SENTIEON_HAPLOTYPER_GVCFS.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\sarek\\subworkflows\\local\\bam_variant_calling_sentieon_haplotyper\\main.nf\n",
      "Process BAM_VARIANT_CALLING_SINGLE_STRELKA in filepath ./git_repos\\sarek\\subworkflows\\local\\bam_variant_calling_single_strelka\\main.nf\n",
      "Nextflow operator combine with content: cram_intervals = cram.combine(intervals)\n",
      "Nextflow operator map with content: .map{ meta, cram, crai, intervals, intervals_index, num_intervals -> [ meta + [ num_intervals:num_intervals ], cram, crai, intervals, intervals_index ] }\n",
      "Nextflow operator branch with content: genome_vcf = STRELKA_SINGLE.out.genome_vcf.branch{\n",
      "Nextflow operator branch with content: vcf = STRELKA_SINGLE.out.vcf.branch{\n",
      "Nextflow operator groupTuple with content: genome_vcf_to_merge = genome_vcf.intervals.map{ meta, vcf -> [ groupKey(meta, meta.num_intervals), vcf ]}.groupTuple()\n",
      "Nextflow operator groupTuple with content: vcf_to_merge        = vcf.intervals.map{ meta, vcf -> [ groupKey(meta, meta.num_intervals), vcf ]}.groupTuple()\n",
      "Nextflow operator mix with content: vcf = Channel.empty().mix(MERGE_STRELKA.out.vcf, vcf.no_intervals)\n",
      "Nextflow operator map with content: .map{ meta, vcf -> [ meta - meta.subMap('num_intervals') + [ variantcaller:'strelka' ], vcf ] }\n",
      "Nextflow operator mix with content: versions = versions.mix(MERGE_STRELKA.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(MERGE_STRELKA_GENOME.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(STRELKA_SINGLE.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\sarek\\subworkflows\\local\\bam_variant_calling_single_strelka\\main.nf\n",
      "Process BAM_VARIANT_CALLING_SINGLE_TIDDIT in filepath ./git_repos\\sarek\\subworkflows\\local\\bam_variant_calling_single_tiddit\\main.nf\n",
      "Nextflow operator map with content: vcf    = TABIX_BGZIP_TIDDIT_SV.out.gz_tbi.map{ meta, gz, tbi -> [ meta + [ variantcaller: 'tiddit'], gz ] }\n",
      "Nextflow operator mix with content: versions = versions.mix(TABIX_BGZIP_TIDDIT_SV.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(TIDDIT_SV.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\sarek\\subworkflows\\local\\bam_variant_calling_single_tiddit\\main.nf\n",
      "Process BAM_VARIANT_CALLING_SOMATIC_ALL in filepath ./git_repos\\sarek\\subworkflows\\local\\bam_variant_calling_somatic_all\\main.nf\n",
      "Nextflow operator mix with content: versions = versions.mix(BAM_VARIANT_CALLING_SOMATIC_ASCAT.out.versions)\n",
      "Nextflow operator map with content: cram_normal = cram.map{ meta, normal_cram, normal_crai, tumor_cram, tumor_crai -> [ meta, normal_cram, normal_crai ] }\n",
      "Nextflow operator map with content: cram_tumor = cram.map{ meta, normal_cram, normal_crai, tumor_cram, tumor_crai -> [ meta, tumor_cram, tumor_crai ] }\n",
      "Nextflow operator cross with content: mpileup_pair = mpileup_normal.cross(mpileup_tumor).map{ normal, tumor -> [ normal[0], normal[1], tumor[1], [], [], [], [] ] }\n",
      "Nextflow operator mix with content: versions = versions.mix(MPILEUP_NORMAL.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(MPILEUP_TUMOR.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(BAM_VARIANT_CALLING_SOMATIC_CONTROLFREEC.out.versions)\n",
      "Nextflow operator map with content: cram.map{ meta, normal_cram, normal_crai, tumor_cram, tumor_crai -> [ meta, tumor_cram, normal_cram ] },\n",
      "Nextflow operator map with content: fasta.map{ it -> [[id:it[0].baseName], it] },\n",
      "Nextflow operator map with content: fasta_fai.map{ it -> [[id:it[0].baseName], it] },\n",
      "Nextflow operator map with content: intervals_bed_combined.map{ it -> [[id:it[0].baseName], it] },\n",
      "Nextflow operator mix with content: versions = versions.mix(BAM_VARIANT_CALLING_CNVKIT.out.versions)\n",
      "Nextflow operator mix with content: versions   = versions.mix(BAM_VARIANT_CALLING_FREEBAYES.out.versions)\n",
      "Nextflow operator map with content: fasta.map{ it -> [ [ id:'fasta' ], it ] },\n",
      "Nextflow operator map with content: fasta_fai.map{ it -> [ [ id:'fasta_fai' ], it ] },\n",
      "Nextflow operator mix with content: versions = versions.mix(BAM_VARIANT_CALLING_SOMATIC_MANTA.out.versions)\n",
      "Nextflow operator join with content: cram.join(BAM_VARIANT_CALLING_SOMATIC_MANTA.out.candidate_small_indels_vcf, failOnDuplicate: true, failOnMismatch: true).join(BAM_VARIANT_CALLING_SOMATIC_MANTA.out.candidate_small_indels_vcf_tbi, failOnDuplicate: true, failOnMismatch: true) :\n",
      "Nextflow operator map with content: cram.map{ meta, normal_cram, normal_crai, tumor_cram, tumor_crai -> [ meta, normal_cram, normal_crai, tumor_cram, tumor_crai, [], [] ] }\n",
      "Nextflow operator mix with content: vcf_strelka = Channel.empty().mix(BAM_VARIANT_CALLING_SOMATIC_STRELKA.out.vcf)\n",
      "Nextflow operator mix with content: versions = versions.mix(BAM_VARIANT_CALLING_SOMATIC_STRELKA.out.versions)\n",
      "Nextflow operator combine with content: MSISENSORPRO_MSISOMATIC(cram.combine(intervals_bed_combined), fasta, msisensorpro_scan)\n",
      "Nextflow operator mix with content: versions = versions.mix(MSISENSORPRO_MSISOMATIC.out.versions)\n",
      "Nextflow operator mix with content: out_msisensorpro = out_msisensorpro.mix(MSISENSORPRO_MSISOMATIC.out.output_report)\n",
      "Nextflow operator map with content: cram.map{ meta, normal_cram, normal_crai, tumor_cram, tumor_crai ->\n",
      "Nextflow operator map with content: fasta.map{ it -> [ [ id:'fasta' ], it ] },\n",
      "Nextflow operator map with content: fasta_fai.map{ it -> [ [ id:'fasta_fai' ], it ] },\n",
      "Nextflow operator mix with content: versions = versions.mix(BAM_VARIANT_CALLING_SOMATIC_MUTECT2.out.versions)\n",
      "Nextflow operator map with content: cram.map{ meta, normal_cram, normal_crai, tumor_cram, tumor_crai -> [ meta, normal_cram, normal_crai ] },\n",
      "Nextflow operator map with content: cram.map{ meta, normal_cram, normal_crai, tumor_cram, tumor_crai -> [ meta, tumor_cram, tumor_crai ] },\n",
      "Nextflow operator map with content: fasta.map{ it -> [ [ id:'fasta' ], it ] },\n",
      "Nextflow operator mix with content: versions = versions.mix(BAM_VARIANT_CALLING_SOMATIC_TIDDIT.out.versions)\n",
      "Nextflow operator mix with content: vcf_all = Channel.empty().mix(\n",
      "Error: Unclosed process block in file: ./git_repos\\sarek\\subworkflows\\local\\bam_variant_calling_somatic_all\\main.nf\n",
      "Process BAM_VARIANT_CALLING_SOMATIC_ASCAT in filepath ./git_repos\\sarek\\subworkflows\\local\\bam_variant_calling_somatic_ascat\\main.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ASCAT.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\sarek\\subworkflows\\local\\bam_variant_calling_somatic_ascat\\main.nf\n",
      "Process BAM_VARIANT_CALLING_SOMATIC_CONTROLFREEC in filepath ./git_repos\\sarek\\subworkflows\\local\\bam_variant_calling_somatic_controlfreec\\main.nf\n",
      "Nextflow operator join with content: ASSESS_SIGNIFICANCE(FREEC_SOMATIC.out.CNV.join(FREEC_SOMATIC.out.ratio, failOnDuplicate: true, failOnMismatch: true))\n",
      "Nextflow operator join with content: MAKEGRAPH(FREEC_SOMATIC.out.ratio.join(FREEC_SOMATIC.out.BAF, failOnDuplicate: true, failOnMismatch: true))\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(FREEC_SOMATIC.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ASSESS_SIGNIFICANCE.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(FREEC2BED.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(FREEC2CIRCOS.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(MAKEGRAPH.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\sarek\\subworkflows\\local\\bam_variant_calling_somatic_controlfreec\\main.nf\n",
      "Process BAM_VARIANT_CALLING_SOMATIC_MANTA in filepath ./git_repos\\sarek\\subworkflows\\local\\bam_variant_calling_somatic_manta\\main.nf\n",
      "Nextflow operator combine with content: cram_intervals = cram.combine(intervals).map{ it ->\n",
      "Nextflow operator map with content: vcf = Channel.empty().mix(diploid_sv_vcf, somatic_sv_vcf).map{ meta, vcf -> [ meta + [ variantcaller:'manta' ], vcf ] }\n",
      "Nextflow operator mix with content: versions = versions.mix(MANTA_SOMATIC.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\sarek\\subworkflows\\local\\bam_variant_calling_somatic_manta\\main.nf\n",
      "Process BAM_VARIANT_CALLING_SOMATIC_MUTECT2 in filepath ./git_repos\\sarek\\subworkflows\\local\\bam_variant_calling_somatic_mutect2\\main.nf\n",
      "Nextflow operator combine with content: input_intervals = input.combine(intervals)\n",
      "Nextflow operator map with content: .map{ meta, input_list, input_index_list, intervals, num_intervals -> [ meta + [ num_intervals:num_intervals ], input_list, input_index_list, intervals ] }\n",
      "Nextflow operator multiMap with content: ch_cram = input.multiMap{ meta, cram, crai ->\n",
      "Nextflow operator groupTuple with content: ch_tn_cram =  ch_cram.normal.unique().mix(ch_cram.tumor).groupTuple()\n",
      "Nextflow operator combine with content: ch_tn_intervals = ch_tn_cram.combine(intervals)\n",
      "Nextflow operator map with content: .map{ meta, cram, crai, intervals, num_intervals -> [ meta + [ num_intervals:num_intervals ], cram, crai, intervals ] }\n",
      "Nextflow operator branch with content: vcf_branch = MUTECT2_PAIRED.out.vcf.branch{\n",
      "Nextflow operator branch with content: tbi_branch = MUTECT2_PAIRED.out.tbi.branch{\n",
      "Nextflow operator branch with content: stats_branch = MUTECT2_PAIRED.out.stats.branch{\n",
      "Nextflow operator branch with content: f1r2_branch = MUTECT2_PAIRED.out.f1r2.branch{\n",
      "Nextflow operator groupTuple with content: vcf_to_merge = vcf_branch.intervals.map{ meta, vcf -> [ groupKey(meta, meta.num_intervals), vcf ] }.groupTuple()\n",
      "Nextflow operator groupTuple with content: stats_to_merge = stats_branch.intervals.map{ meta, stats -> [ groupKey(meta, meta.num_intervals), stats ] }.groupTuple()\n",
      "Nextflow operator groupTuple with content: f1r2_to_merge = f1r2_branch.intervals.map{ meta, f1r2 -> [ groupKey(meta, meta.num_intervals), f1r2 ] }.groupTuple()\n",
      "Nextflow operator map with content: vcf = Channel.empty().mix(MERGE_MUTECT2.out.vcf, vcf_branch.no_intervals).map{ meta, vcf ->\n",
      "Nextflow operator map with content: tbi = Channel.empty().mix(MERGE_MUTECT2.out.tbi, tbi_branch.no_intervals).map{ meta, tbi->\n",
      "Nextflow operator map with content: stats = Channel.empty().mix(MERGEMUTECTSTATS.out.stats, stats_branch.no_intervals).map{ meta, stats ->\n",
      "Nextflow operator map with content: f1r2 = Channel.empty().mix(f1r2_to_merge, f1r2_branch.no_intervals).map{ meta, f1r2->\n",
      "Nextflow operator multiMap with content: pileup = input_intervals.multiMap{  meta, input_list, input_index_list, intervals ->\n",
      "Nextflow operator map with content: pileup_normal = pileup.normal.map{ meta, cram, crai, intervals -> [ meta - meta.subMap('tumor_id') + [ id:meta.normal_id ], cram, crai, intervals] }.unique()\n",
      "Nextflow operator map with content: pileup_tumor = pileup.tumor.map{ meta, cram, crai, intervals -> [ meta - meta.subMap('normal_id') + [ id:meta.tumor_id ], cram, crai, intervals ] }\n",
      "Nextflow operator branch with content: pileup_table_normal_branch = GETPILEUPSUMMARIES_NORMAL.out.table.branch{\n",
      "Nextflow operator branch with content: pileup_table_tumor_branch = GETPILEUPSUMMARIES_TUMOR.out.table.branch{\n",
      "Nextflow operator groupTuple with content: pileup_table_normal_to_merge = pileup_table_normal_branch.intervals.map{ meta, table -> [ groupKey(meta, meta.num_intervals), table ] }.groupTuple()\n",
      "Nextflow operator groupTuple with content: pileup_table_tumor_to_merge = pileup_table_tumor_branch.intervals.map{ meta, table -> [ groupKey(meta, meta.num_intervals), table ] }.groupTuple()\n",
      "Nextflow operator map with content: GATHERPILEUPSUMMARIES_NORMAL(pileup_table_normal_to_merge, dict.map{ meta, dict -> [ dict ] })\n",
      "Nextflow operator map with content: GATHERPILEUPSUMMARIES_TUMOR(pileup_table_tumor_to_merge, dict.map{ meta, dict -> [ dict ] })\n",
      "Nextflow operator map with content: pileup_table_tumor  = Channel.empty().mix(GATHERPILEUPSUMMARIES_TUMOR.out.table, pileup_table_tumor_branch.no_intervals).map{meta, table -> [ meta - meta.subMap('normal_id', 'tumor_id', 'num_intervals') + [id:meta.patient], meta.id, table ] }\n",
      "Nextflow operator map with content: pileup_table_normal = Channel.empty().mix(GATHERPILEUPSUMMARIES_NORMAL.out.table, pileup_table_normal_branch.no_intervals).map{meta, table -> [ meta - meta.subMap('normal_id', 'tumor_id', 'num_intervals') + [id:meta.patient], meta.id, table ] }\n",
      "Nextflow operator combine with content: ch_calculatecontamination_in_tables = pileup_table_tumor.combine(\n",
      "Nextflow operator map with content: pileup_table_normal, by:0).map{\n",
      "Nextflow operator groupTuple with content: calculatecontamination_out_seg = CALCULATECONTAMINATION.out.segmentation.map{ meta, seg -> [ meta + [id: meta.patient], seg]}.groupTuple()\n",
      "Nextflow operator groupTuple with content: calculatecontamination_out_cont = CALCULATECONTAMINATION.out.contamination.map{ meta, cont -> [ meta + [id: meta.patient], cont]}.groupTuple()\n",
      "Nextflow operator join with content: vcf_to_filter = vcf.join(tbi, failOnDuplicate: true, failOnMismatch: true)\n",
      "Nextflow operator join with content: .join(stats, failOnDuplicate: true, failOnMismatch: true)\n",
      "Nextflow operator join with content: .join(LEARNREADORIENTATIONMODEL.out.artifactprior, failOnDuplicate: true, failOnMismatch: true)\n",
      "Nextflow operator join with content: .join(calculatecontamination_out_seg)\n",
      "Nextflow operator join with content: .join(calculatecontamination_out_cont)\n",
      "Nextflow operator map with content: .map{ meta, vcf, tbi, stats, orientation, seg, cont -> [ meta, vcf, tbi, stats, orientation, seg, cont, [] ] }\n",
      "Nextflow operator map with content: .map{ meta, vcf -> [ meta + [ variantcaller:'mutect2' ], vcf ] }\n",
      "Nextflow operator mix with content: versions = versions.mix(MERGE_MUTECT2.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(CALCULATECONTAMINATION.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(FILTERMUTECTCALLS.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(GETPILEUPSUMMARIES_NORMAL.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(GETPILEUPSUMMARIES_TUMOR.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(GATHERPILEUPSUMMARIES_NORMAL.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(GATHERPILEUPSUMMARIES_TUMOR.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(LEARNREADORIENTATIONMODEL.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(MERGEMUTECTSTATS.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(MUTECT2_PAIRED.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\sarek\\subworkflows\\local\\bam_variant_calling_somatic_mutect2\\main.nf\n",
      "Process BAM_VARIANT_CALLING_SOMATIC_STRELKA in filepath ./git_repos\\sarek\\subworkflows\\local\\bam_variant_calling_somatic_strelka\\main.nf\n",
      "Nextflow operator combine with content: cram_intervals = cram.combine(intervals)\n",
      "Nextflow operator map with content: .map{ meta, normal_cram, normal_crai, tumor_cram, tumor_crai, manta_vcf, manta_tbi, intervals, intervals_index, num_intervals -> [ meta + [ num_intervals:num_intervals ], normal_cram, normal_crai, tumor_cram, tumor_crai, manta_vcf, manta_tbi, intervals, intervals_index ] }\n",
      "Nextflow operator branch with content: vcf_indels = STRELKA_SOMATIC.out.vcf_indels.branch{\n",
      "Nextflow operator branch with content: vcf_snvs = STRELKA_SOMATIC.out.vcf_snvs.branch{\n",
      "Nextflow operator groupTuple with content: vcf_indels_to_merge = vcf_indels.intervals.map{ meta, vcf -> [ groupKey(meta, meta.num_intervals), vcf ]}.groupTuple()\n",
      "Nextflow operator groupTuple with content: vcf_snvs_to_merge = vcf_snvs.intervals.map{ meta, vcf -> [ groupKey(meta, meta.num_intervals), vcf ]}.groupTuple()\n",
      "Nextflow operator mix with content: vcf = Channel.empty().mix(MERGE_STRELKA_INDELS.out.vcf, MERGE_STRELKA_SNVS.out.vcf, vcf_indels.no_intervals, vcf_snvs.no_intervals)\n",
      "Nextflow operator map with content: .map{ meta, vcf -> [ meta - meta.subMap('num_intervals') + [ variantcaller:'strelka' ], vcf ] }\n",
      "Nextflow operator mix with content: versions = versions.mix(MERGE_STRELKA_SNVS.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(MERGE_STRELKA_INDELS.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(STRELKA_SOMATIC.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\sarek\\subworkflows\\local\\bam_variant_calling_somatic_strelka\\main.nf\n",
      "Process BAM_VARIANT_CALLING_SOMATIC_TIDDIT in filepath ./git_repos\\sarek\\subworkflows\\local\\bam_variant_calling_somatic_tiddit\\main.nf\n",
      "Nextflow operator join with content: SVDB_MERGE(TIDDIT_NORMAL.out.vcf.join(TIDDIT_TUMOR.out.vcf, failOnDuplicate: true, failOnMismatch: true).map{ meta, vcf_normal, vcf_tumor -> [ meta, [vcf_normal, vcf_tumor] ] }, false)\n",
      "Nextflow operator mix with content: versions = versions.mix(TIDDIT_NORMAL.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(TIDDIT_TUMOR.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(SVDB_MERGE.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\sarek\\subworkflows\\local\\bam_variant_calling_somatic_tiddit\\main.nf\n",
      "Process BAM_VARIANT_CALLING_TUMOR_ONLY_ALL in filepath ./git_repos\\sarek\\subworkflows\\local\\bam_variant_calling_tumor_only_all\\main.nf\n",
      "Nextflow operator mix with content: versions = versions.mix(BAM_VARIANT_CALLING_MPILEUP.out.versions)\n",
      "Nextflow operator map with content: BAM_VARIANT_CALLING_MPILEUP.out.mpileup.map{ meta, pileup_tumor -> [ meta, [], pileup_tumor, [], [], [], [] ] },\n",
      "Nextflow operator mix with content: versions = versions.mix(BAM_VARIANT_CALLING_TUMOR_ONLY_CONTROLFREEC.out.versions)\n",
      "Nextflow operator map with content: cram.map{ meta, cram, crai -> [ meta, cram, [] ] },\n",
      "Nextflow operator map with content: fasta.map{ it -> [[id:it[0].baseName], it] },\n",
      "Nextflow operator map with content: fasta_fai.map{ it -> [[id:it[0].baseName], it] },\n",
      "Nextflow operator map with content: cnvkit_reference.map{ it -> [[id:it[0].baseName], it] }\n",
      "Nextflow operator mix with content: versions = versions.mix(BAM_VARIANT_CALLING_CNVKIT.out.versions)\n",
      "Nextflow operator map with content: cram.map{ meta, cram, crai -> [ meta, cram, crai, [], [] ] },\n",
      "Nextflow operator mix with content: versions = versions.mix(BAM_VARIANT_CALLING_FREEBAYES.out.versions)\n",
      "Nextflow operator map with content: cram.map{ meta, cram, crai ->\n",
      "Nextflow operator map with content: fasta.map{ it -> [ [ id:'fasta' ], it ] },\n",
      "Nextflow operator map with content: fasta_fai.map{ it -> [ [ id:'fasta_fai' ], it ] },\n",
      "Nextflow operator mix with content: versions = versions.mix(BAM_VARIANT_CALLING_TUMOR_ONLY_MUTECT2.out.versions)\n",
      "Nextflow operator map with content: fasta.map{ it -> [ [ id:'fasta' ], it ] },\n",
      "Nextflow operator map with content: fasta_fai.map{ it -> [ [ id:'fasta_fai' ], it ] },\n",
      "Nextflow operator mix with content: versions = versions.mix(BAM_VARIANT_CALLING_TUMOR_ONLY_MANTA.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(BAM_VARIANT_CALLING_SINGLE_STRELKA.out.versions)\n",
      "Nextflow operator map with content: fasta.map{ it -> [ [ id:'fasta' ], it ] },\n",
      "Nextflow operator mix with content: versions = versions.mix(BAM_VARIANT_CALLING_SINGLE_TIDDIT.out.versions)\n",
      "Nextflow operator mix with content: vcf_all = Channel.empty().mix(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Error: Unclosed process block in file: ./git_repos\\sarek\\subworkflows\\local\\bam_variant_calling_tumor_only_all\\main.nf\n",
      "Process BAM_VARIANT_CALLING_TUMOR_ONLY_CONTROLFREEC in filepath ./git_repos\\sarek\\subworkflows\\local\\bam_variant_calling_tumor_only_controlfreec\\main.nf\n",
      "Nextflow operator join with content: ASSESS_SIGNIFICANCE(FREEC_TUMORONLY.out.CNV.join(FREEC_TUMORONLY.out.ratio, failOnDuplicate: true, failOnMismatch: true))\n",
      "Nextflow operator join with content: MAKEGRAPH(FREEC_TUMORONLY.out.ratio.join(FREEC_TUMORONLY.out.BAF, failOnDuplicate: true, failOnMismatch: true))\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(FREEC_TUMORONLY.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ASSESS_SIGNIFICANCE.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(FREEC2BED.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(FREEC2CIRCOS.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(MAKEGRAPH.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\sarek\\subworkflows\\local\\bam_variant_calling_tumor_only_controlfreec\\main.nf\n",
      "Process BAM_VARIANT_CALLING_TUMOR_ONLY_MANTA in filepath ./git_repos\\sarek\\subworkflows\\local\\bam_variant_calling_tumor_only_manta\\main.nf\n",
      "Nextflow operator combine with content: cram_intervals = cram.combine(intervals).map{ it ->\n",
      "Nextflow operator map with content: vcf = tumor_sv_vcf.map{ meta, vcf -> [ meta + [ variantcaller:'manta' ], vcf ] }\n",
      "Nextflow operator mix with content: versions = versions.mix(MANTA_TUMORONLY.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\sarek\\subworkflows\\local\\bam_variant_calling_tumor_only_manta\\main.nf\n",
      "Process BAM_VARIANT_CALLING_TUMOR_ONLY_MUTECT2 in filepath ./git_repos\\sarek\\subworkflows\\local\\bam_variant_calling_tumor_only_mutect2\\main.nf\n",
      "Nextflow operator combine with content: input_intervals = input.combine(intervals)\n",
      "Nextflow operator map with content: .map{ meta, input, index, intervals, num_intervals -> [ meta + [ num_intervals:num_intervals ], input, index, intervals ] }\n",
      "Nextflow operator map with content: .map{ meta, input, index -> [ meta - meta.subMap('sample') + [ id:meta.patient ], input, index ] }\n",
      "Nextflow operator groupTuple with content: .groupTuple()\n",
      "Nextflow operator combine with content: input_joint_intervals = input_joint.combine(intervals)\n",
      "Nextflow operator map with content: .map{ meta, cram, crai, intervals, num_intervals -> [ meta + [ num_intervals:num_intervals ], cram, crai, intervals ] }\n",
      "Nextflow operator branch with content: vcf_branch = MUTECT2.out.vcf.branch{\n",
      "Nextflow operator branch with content: tbi_branch = MUTECT2.out.tbi.branch{\n",
      "Nextflow operator branch with content: stats_branch = MUTECT2.out.stats.branch{\n",
      "Nextflow operator branch with content: f1r2_branch = MUTECT2.out.f1r2.branch{\n",
      "Nextflow operator groupTuple with content: vcf_to_merge = vcf_branch.intervals.map{ meta, vcf -> [ groupKey(meta, meta.num_intervals), vcf ] }.groupTuple()\n",
      "Nextflow operator groupTuple with content: stats_to_merge = stats_branch.intervals.map{ meta, stats -> [ groupKey(meta, meta.num_intervals), stats ] }.groupTuple()\n",
      "Nextflow operator groupTuple with content: f1r2_to_merge = f1r2_branch.intervals.map{ meta, f1r2 -> [ groupKey(meta, meta.num_intervals), f1r2 ] }.groupTuple()\n",
      "Nextflow operator map with content: vcf   = Channel.empty().mix(MERGE_MUTECT2.out.vcf, vcf_branch.no_intervals).map{ meta, vcf -> [ meta - meta.subMap('num_intervals'), vcf ] }\n",
      "Nextflow operator map with content: tbi   = Channel.empty().mix(MERGE_MUTECT2.out.tbi, tbi_branch.no_intervals).map{ meta, tbi -> [ meta - meta.subMap('num_intervals'), tbi ] }\n",
      "Nextflow operator map with content: stats = Channel.empty().mix(MERGEMUTECTSTATS.out.stats, stats_branch.no_intervals).map{ meta, stats -> [ meta - meta.subMap('num_intervals'), stats ] }\n",
      "Nextflow operator map with content: f1r2  = Channel.empty().mix(f1r2_to_merge, f1r2_branch.no_intervals).map{ meta, f1r2 -> [ meta - meta.subMap('num_intervals'), f1r2 ] }\n",
      "Nextflow operator map with content: pileup_input = input_intervals.map{ meta, cram, crai, intervals -> [ meta + [ id:meta.sample ], cram, crai, intervals] }.unique()\n",
      "Nextflow operator branch with content: pileup_table_branch = GETPILEUPSUMMARIES.out.table.branch{\n",
      "Nextflow operator groupTuple with content: pileup_table_to_merge = pileup_table_branch.intervals.map{ meta, table -> [ groupKey(meta, meta.num_intervals), table ] }.groupTuple()\n",
      "Nextflow operator map with content: GATHERPILEUPSUMMARIES(pileup_table_to_merge, dict.map{ meta, dict -> [ dict ] })\n",
      "Nextflow operator map with content: pileup_table = Channel.empty().mix(GATHERPILEUPSUMMARIES.out.table, pileup_table_branch.no_intervals).map{meta, table -> [ meta - meta.subMap('num_intervals') + [id:meta.sample], table ] }\n",
      "Nextflow operator map with content: CALCULATECONTAMINATION(pileup_table.map{ meta, table -> [ meta, table, [] ] })\n",
      "Nextflow operator groupTuple with content: calculatecontamination_out_seg = CALCULATECONTAMINATION.out.segmentation.map{ meta, seg -> [ meta - meta.subMap('sample', 'num_intervals') + [id:meta.patient], seg ] }.groupTuple()\n",
      "Nextflow operator groupTuple with content: calculatecontamination_out_cont = CALCULATECONTAMINATION.out.contamination.map{ meta, cont -> [ meta - meta.subMap('sample', 'num_intervals') + [id:meta.patient], cont ] }.groupTuple()\n",
      "Nextflow operator map with content: calculatecontamination_out_seg = CALCULATECONTAMINATION.out.segmentation.map{ meta, seg -> [ meta - meta.subMap('num_intervals'), seg ] }\n",
      "Nextflow operator map with content: calculatecontamination_out_cont = CALCULATECONTAMINATION.out.contamination.map{ meta, cont -> [ meta - meta.subMap('num_intervals'), cont ] }\n",
      "Nextflow operator join with content: vcf_to_filter = vcf.join(tbi, failOnDuplicate: true, failOnMismatch: true)\n",
      "Nextflow operator join with content: .join(stats, failOnDuplicate: true, failOnMismatch: true)\n",
      "Nextflow operator join with content: .join(LEARNREADORIENTATIONMODEL.out.artifactprior, failOnDuplicate: true, failOnMismatch: true)\n",
      "Nextflow operator join with content: .join(calculatecontamination_out_seg)\n",
      "Nextflow operator join with content: .join(calculatecontamination_out_cont)\n",
      "Nextflow operator map with content: .map{ meta, vcf, tbi, stats, artifactprior, seg, cont -> [ meta, vcf, tbi, stats, artifactprior, seg, cont, [] ] }\n",
      "Nextflow operator map with content: .map{ meta, vcf -> [ meta + [ variantcaller:'mutect2' ], vcf ] }\n",
      "Nextflow operator mix with content: versions = versions.mix(MERGE_MUTECT2.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(CALCULATECONTAMINATION.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(FILTERMUTECTCALLS.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(GETPILEUPSUMMARIES.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(GATHERPILEUPSUMMARIES.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(LEARNREADORIENTATIONMODEL.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(MERGEMUTECTSTATS.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(MUTECT2.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\sarek\\subworkflows\\local\\bam_variant_calling_tumor_only_mutect2\\main.nf\n",
      "Process CHANNEL_ALIGN_CREATE_CSV in filepath ./git_repos\\sarek\\subworkflows\\local\\channel_align_create_csv\\main.nf\n",
      "Nextflow operator collectFile with content: bam_indexed.collectFile(keepHeader: true, skip: 1, sort: true, storeDir: \"${params.outdir}/csv\") { meta, bam, bai ->\n",
      "Process CHANNEL_APPLYBQSR_CREATE_CSV in filepath ./git_repos\\sarek\\subworkflows\\local\\channel_applybqsr_create_csv\\main.nf\n",
      "Nextflow operator collectFile with content: cram_recalibrated_index.collectFile(keepHeader: true, skip: 1, sort: true, storeDir: \"${params.outdir}/csv\") { meta, file, index ->\n",
      "Process CHANNEL_BASERECALIBRATOR_CREATE_CSV in filepath ./git_repos\\sarek\\subworkflows\\local\\channel_baserecalibrator_create_csv\\main.nf\n",
      "Nextflow operator collectFile with content: cram_table_bqsr.collectFile(keepHeader: true, skip: 1, sort: true, storeDir: \"${outdir}/csv\") { meta, cram, crai, table ->\n",
      "Nextflow operator collectFile with content: cram_table_bqsr.collectFile(keepHeader: true, skip: 1, sort: true, storeDir: \"${outdir}/csv\") { meta, cram, crai, table ->\n",
      "Nextflow operator collectFile with content: cram_table_bqsr.collectFile(keepHeader: true, skip: 1, sort: true, storeDir: \"${outdir}/csv\") { meta, cram, crai, table ->\n",
      "Process CHANNEL_MARKDUPLICATES_CREATE_CSV in filepath ./git_repos\\sarek\\subworkflows\\local\\channel_markduplicates_create_csv\\main.nf\n",
      "Nextflow operator collectFile with content: cram_markduplicates.collectFile(keepHeader: true, skip: 1, sort: true, storeDir: \"${outdir}/csv\") { meta, file, index ->\n",
      "Process CHANNEL_VARIANT_CALLING_CREATE_CSV in filepath ./git_repos\\sarek\\subworkflows\\local\\channel_variant_calling_create_csv\\main.nf\n",
      "Nextflow operator collectFile with content: vcf_to_annotate.collectFile(keepHeader: true, skip: 1,sort: true, storeDir: \"${params.outdir}/csv\"){ meta, vcf ->\n",
      "Process CRAM_MERGE_INDEX_SAMTOOLS in filepath ./git_repos\\sarek\\subworkflows\\local\\cram_merge_index_samtools\\main.nf\n",
      "Nextflow operator branch with content: cram_to_merge = cram.branch{ meta, cram ->\n",
      "Nextflow operator map with content: MERGE_CRAM(cram_to_merge.multiple, fasta.map{ it -> [ [ id:'fasta' ], it ] }, fasta_fai.map{ it -> [ [ id:'fasta_fai' ], it ] })\n",
      "Nextflow operator mix with content: cram_all = MERGE_CRAM.out.cram.mix(cram_to_merge.single)\n",
      "Nextflow operator join with content: cram_crai = cram_all.join(INDEX_CRAM.out.crai, failOnDuplicate: true, failOnMismatch: true)\n",
      "Nextflow operator first with content: versions = versions.mix(INDEX_CRAM.out.versions.first())\n",
      "Nextflow operator first with content: versions = versions.mix(MERGE_CRAM.out.versions.first())\n",
      "Error: Unclosed process block in file: ./git_repos\\sarek\\subworkflows\\local\\cram_merge_index_samtools\\main.nf\n",
      "Process CRAM_QC_MOSDEPTH_SAMTOOLS in filepath ./git_repos\\sarek\\subworkflows\\local\\cram_qc_mosdepth_samtools\\main.nf\n",
      "Nextflow operator map with content: SAMTOOLS_STATS(cram, fasta.map{ it -> [ [ id:'fasta' ], it ] })\n",
      "Nextflow operator combine with content: MOSDEPTH(cram.combine(intervals.map{ meta, bed -> [ bed?:[] ] }), fasta.map{ it -> [ [ id:'fasta' ], it ] })\n",
      "Nextflow operator mix with content: reports = reports.mix(SAMTOOLS_STATS.out.stats)\n",
      "Nextflow operator mix with content: reports = reports.mix(MOSDEPTH.out.global_txt)\n",
      "Nextflow operator mix with content: reports = reports.mix(MOSDEPTH.out.regions_txt)\n",
      "Nextflow operator mix with content: versions = versions.mix(MOSDEPTH.out.versions)\n",
      "Nextflow operator first with content: versions = versions.mix(SAMTOOLS_STATS.out.versions.first())\n",
      "Error: Unclosed process block in file: ./git_repos\\sarek\\subworkflows\\local\\cram_qc_mosdepth_samtools\\main.nf\n",
      "Process CRAM_SAMPLEQC in filepath ./git_repos\\sarek\\subworkflows\\local\\cram_sampleqc\\main.nf\n",
      "Nextflow operator map with content: ch_ngscheckmate_bed = ngscheckmate_bed.map{bed -> [[id: \"ngscheckmate\"], bed]}\n",
      "Nextflow operator map with content: ch_fasta            = fasta.map{fasta -> [[id: \"genome\"], fasta]}\n",
      "Nextflow operator map with content: BAM_NGSCHECKMATE ( ch_cram.map{meta, cram, crai -> [meta, cram]}, ch_ngscheckmate_bed, ch_fasta)\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(BAM_NGSCHECKMATE.out.versions.first())\n",
      "Error: Unclosed process block in file: ./git_repos\\sarek\\subworkflows\\local\\cram_sampleqc\\main.nf\n",
      "Process DOWNLOAD_CACHE_SNPEFF_VEP in filepath ./git_repos\\sarek\\subworkflows\\local\\download_cache_snpeff_vep\\main.nf\n",
      "Nextflow operator mix with content: versions = versions.mix(ENSEMBLVEP_DOWNLOAD.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(SNPEFF_DOWNLOAD.out.versions)\n",
      "Nextflow operator collect with content: ensemblvep_cache = ENSEMBLVEP_DOWNLOAD.out.cache.collect()  // channel: [ meta, cache ]\n",
      "Nextflow operator collect with content: snpeff_cache     = SNPEFF_DOWNLOAD.out.cache.collect()      // channel: [ meta, cache ]\n",
      "Error: Unclosed process block in file: ./git_repos\\sarek\\subworkflows\\local\\download_cache_snpeff_vep\\main.nf\n",
      "Process FASTQ_ALIGN_BWAMEM_MEM2_DRAGMAP_SENTIEON in filepath ./git_repos\\sarek\\subworkflows\\local\\fastq_align_bwamem_mem2_dragmap_sentieon\\main.nf\n",
      "Nextflow operator map with content: BWAMEM1_MEM(reads, index.map{ it -> [ [ id:'index' ], it ] }, sort) // If aligner is bwa-mem\n",
      "Nextflow operator map with content: BWAMEM2_MEM(reads, index.map{ it -> [ [ id:'index' ], it ] }, sort) // If aligner is bwa-mem2\n",
      "Nextflow operator map with content: DRAGMAP_ALIGN(reads, index.map{ it -> [ [ id:'index' ], it ] }, sort) // If aligner is dragmap\n",
      "Nextflow operator map with content: SENTIEON_BWAMEM(reads, index.map{ it -> [ [ id:'index' ], it ] }, fasta.map{fa -> [[:], fa]}, fasta_fai.map{fai -> [[:], fai]}) // If aligner is sentieon-bwamem\n",
      "Nextflow operator mix with content: bam = bam.mix(BWAMEM1_MEM.out.bam)\n",
      "Nextflow operator mix with content: bam = bam.mix(BWAMEM2_MEM.out.bam)\n",
      "Nextflow operator mix with content: bam = bam.mix(DRAGMAP_ALIGN.out.bam)\n",
      "Nextflow operator map with content: bam = bam.mix(SENTIEON_BWAMEM.out.bam_and_bai.map{ meta, bam, bai -> [ meta, bam ] })\n",
      "Nextflow operator map with content: bai = SENTIEON_BWAMEM.out.bam_and_bai.map{ meta, bam, bai -> [ meta, bai ] }\n",
      "Nextflow operator mix with content: reports = reports.mix(DRAGMAP_ALIGN.out.log)\n",
      "Nextflow operator mix with content: versions = versions.mix(BWAMEM1_MEM.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(BWAMEM2_MEM.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(DRAGMAP_ALIGN.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(SENTIEON_BWAMEM.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\sarek\\subworkflows\\local\\fastq_align_bwamem_mem2_dragmap_sentieon\\main.nf\n",
      "Process FASTQ_CREATE_UMI_CONSENSUS_FGBIO in filepath ./git_repos\\sarek\\subworkflows\\local\\fastq_create_umi_consensus_fgbio\\main.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BAM2FASTQ.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ALIGN_UMI.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(CALLUMICONSENSUS.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(FASTQTOBAM.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GROUPREADSBYUMI.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMBLASTER.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\sarek\\subworkflows\\local\\fastq_create_umi_consensus_fgbio\\main.nf\n",
      "Process INITIALIZE_ANNOTATION_CACHE in filepath ./git_repos\\sarek\\subworkflows\\local\\initialize_annotation_cache\\main.nf\n",
      "Nextflow operator collect with content: snpeff_cache = Channel.fromPath(file(\"${snpeff_cache}/${snpeff_annotation_cache_key}\"), checkIfExists: true).collect()\n",
      "Nextflow operator map with content: .map{ cache -> [ [ id:\"${snpeff_genome}.${snpeff_db}\" ], cache ] }\n",
      "Nextflow operator collect with content: ensemblvep_cache = Channel.fromPath(file(\"${vep_cache}/${vep_annotation_cache_key}\"), checkIfExists: true).collect()\n",
      "Error: Unclosed process block in file: ./git_repos\\sarek\\subworkflows\\local\\initialize_annotation_cache\\main.nf\n",
      "Process POST_VARIANTCALLING in filepath ./git_repos\\sarek\\subworkflows\\local\\post_variantcalling\\main.nf\n",
      "Nextflow operator mix with content: vcfs = vcfs.mix(CONCATENATE_GERMLINE_VCFS.out.vcfs)\n",
      "Nextflow operator mix with content: versions = versions.mix(CONCATENATE_GERMLINE_VCFS.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\sarek\\subworkflows\\local\\post_variantcalling\\main.nf\n",
      "Process PREPARE_GENOME in filepath ./git_repos\\sarek\\subworkflows\\local\\prepare_genome\\main.nf\n",
      "Nextflow operator map with content: fasta = fasta.map{ fasta -> [ [ id:fasta.baseName ], fasta ] }\n",
      "Nextflow operator flatten with content: TABIX_DBSNP(dbsnp.flatten().map{ it -> [ [ id:it.baseName ], it ] })\n",
      "Nextflow operator flatten with content: TABIX_GERMLINE_RESOURCE(germline_resource.flatten().map{ it -> [ [ id:it.baseName ], it ] })\n",
      "Nextflow operator flatten with content: TABIX_KNOWN_SNPS(known_snps.flatten().map{ it -> [ [ id:it.baseName ], it ] } )\n",
      "Nextflow operator flatten with content: TABIX_KNOWN_INDELS(known_indels.flatten().map{ it -> [ [ id:it.baseName ], it ] } )\n",
      "Nextflow operator flatten with content: TABIX_PON(pon.flatten().map{ it -> [ [ id:it.baseName ], it ] })\n",
      "Nextflow operator map with content: UNZIP_ALLELES(ascat_alleles.map{ it -> [[id:it[0].baseName], it]})\n",
      "Nextflow operator map with content: allele_files = UNZIP_ALLELES.out.unzipped_archive.map{ it[1] }\n",
      "Nextflow operator mix with content: versions = versions.mix(UNZIP_ALLELES.out.versions)\n",
      "Nextflow operator map with content: UNZIP_LOCI(ascat_loci.map{ it -> [[id:it[0].baseName], it]})\n",
      "Nextflow operator map with content: loci_files = UNZIP_LOCI.out.unzipped_archive.map{ it[1] }\n",
      "Nextflow operator mix with content: versions = versions.mix(UNZIP_LOCI.out.versions)\n",
      "Nextflow operator map with content: UNZIP_GC(ascat_loci_gc.map{ it -> [[id:it[0].baseName], it]})\n",
      "Nextflow operator map with content: gc_file = UNZIP_GC.out.unzipped_archive.map{ it[1] }\n",
      "Nextflow operator mix with content: versions = versions.mix(UNZIP_GC.out.versions)\n",
      "Nextflow operator map with content: UNZIP_RT(ascat_loci_rt.map{ it -> [[id:it[0].baseName], it]})\n",
      "Nextflow operator map with content: rt_file = UNZIP_RT.out.unzipped_archive.map{ it[1] }\n",
      "Nextflow operator mix with content: versions = versions.mix(UNZIP_RT.out.versions)\n",
      "Nextflow operator map with content: UNTAR_CHR_DIR(chr_dir.map{ it -> [ [ id:'chr_dir' ], it ] })\n",
      "Nextflow operator map with content: chr_files = UNTAR_CHR_DIR.out.untar.map{ it[1] }\n",
      "Nextflow operator mix with content: versions = versions.mix(UNTAR_CHR_DIR.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(SAMTOOLS_FAIDX.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(BWAMEM1_INDEX.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(BWAMEM2_INDEX.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(DRAGMAP_HASHTABLE.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(GATK4_CREATESEQUENCEDICTIONARY.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(MSISENSORPRO_SCAN.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(TABIX_DBSNP.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(TABIX_GERMLINE_RESOURCE.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(TABIX_KNOWN_SNPS.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(TABIX_KNOWN_INDELS.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(TABIX_PON.out.versions)\n",
      "Nextflow operator collect with content: bwa                   = BWAMEM1_INDEX.out.index.map{ meta, index -> [index] }.collect()       // path: bwa/*\n",
      "Nextflow operator collect with content: bwamem2               = BWAMEM2_INDEX.out.index.map{ meta, index -> [index] }.collect()       // path: bwamem2/*\n",
      "Nextflow operator collect with content: hashtable             = DRAGMAP_HASHTABLE.out.hashmap.map{ meta, index -> [index] }.collect() // path: dragmap/*\n",
      "Nextflow operator collect with content: dbsnp_tbi             = TABIX_DBSNP.out.tbi.map{ meta, tbi -> [tbi] }.collect()               // path: dbsnb.vcf.gz.tbi\n",
      "Nextflow operator map with content: fasta_fai             = SAMTOOLS_FAIDX.out.fai.map{ meta, fai -> [fai] }                      // path: genome.fasta.fai\n",
      "Nextflow operator collect with content: germline_resource_tbi = TABIX_GERMLINE_RESOURCE.out.tbi.map{ meta, tbi -> [tbi] }.collect()   // path: germline_resource.vcf.gz.tbi\n",
      "Nextflow operator collect with content: known_snps_tbi        = TABIX_KNOWN_SNPS.out.tbi.map{ meta, tbi -> [tbi] }.collect()          // path: {known_indels*}.vcf.gz.tbi\n",
      "Nextflow operator collect with content: known_indels_tbi      = TABIX_KNOWN_INDELS.out.tbi.map{ meta, tbi -> [tbi] }.collect()        // path: {known_indels*}.vcf.gz.tbi\n",
      "Nextflow operator map with content: msisensorpro_scan     = MSISENSORPRO_SCAN.out.list.map{ meta, list -> [list] }                // path: genome_msi.list\n",
      "Nextflow operator collect with content: pon_tbi               = TABIX_PON.out.tbi.map{ meta, tbi -> [tbi] }.collect()                 // path: pon.vcf.gz.tbi\n",
      "Error: Unclosed process block in file: ./git_repos\\sarek\\subworkflows\\local\\prepare_genome\\main.nf\n",
      "Process PREPARE_INTERVALS in filepath ./git_repos\\sarek\\subworkflows\\local\\prepare_intervals\\main.nf\n",
      "Nextflow operator map with content: intervals_bed        = Channel.fromPath(file(\"${params.outdir}/no_intervals.bed\")).map{ it -> [ it, 0 ] }\n",
      "Nextflow operator collect with content: intervals_bed_gz_tbi = Channel.fromPath(file(\"${params.outdir}/no_intervals.bed.{gz,gz.tbi}\")).collect().map{ it -> [ it, 0 ] }\n",
      "Nextflow operator map with content: intervals_combined   = Channel.fromPath(file(\"${params.outdir}/no_intervals.bed\")).map{ it -> [ [ id:it.simpleName ], it ] }\n",
      "Nextflow operator map with content: BUILD_INTERVALS(fasta_fai.map{it -> [ [ id:it.baseName ], it ] })\n",
      "Nextflow operator map with content: CREATE_INTERVALS_BED(intervals_combined.map{ meta, path -> path }).bed\n",
      "Nextflow operator mix with content: versions = versions.mix(BUILD_INTERVALS.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(CREATE_INTERVALS_BED.out.versions)\n",
      "Nextflow operator map with content: intervals_combined = Channel.fromPath(file(intervals)).map{it -> [ [ id:it.baseName ], it ] }\n",
      "Nextflow operator mix with content: versions = versions.mix(CREATE_INTERVALS_BED.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(GATK4_INTERVALLISTTOBED.out.versions)\n",
      "Nextflow operator flatten with content: intervals_bed = intervals_bed.flatten()\n",
      "Nextflow operator map with content: .map{ intervalFile ->\n",
      "Nextflow operator toInteger with content: start = fields[1].toInteger()\n",
      "Nextflow operator toInteger with content: end = fields[2].toInteger()\n",
      "Nextflow operator toSortedList with content: }.toSortedList({ a, b -> b[0] <=> a[0] })\n",
      "Nextflow operator collate with content: .flatten().collate(2).map{ duration, intervalFile -> intervalFile }.collect()\n",
      "Nextflow operator map with content: .map{ it -> [ it, it.size() ] }\n",
      "Nextflow operator transpose with content: .transpose()\n",
      "Nextflow operator map with content: TABIX_BGZIPTABIX_INTERVAL_SPLIT(intervals_bed.map{ file, num_intervals -> [ [ id:file.baseName], file ] })\n",
      "Nextflow operator map with content: intervals_bed_gz_tbi = TABIX_BGZIPTABIX_INTERVAL_SPLIT.out.gz_tbi.map{ meta, bed, tbi -> [ bed, tbi ] }.toList()\n",
      "Nextflow operator map with content: .map{ it -> [ it, it.size() ] }\n",
      "Nextflow operator transpose with content: .transpose()\n",
      "Nextflow operator mix with content: versions = versions.mix(TABIX_BGZIPTABIX_INTERVAL_SPLIT.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(TABIX_BGZIPTABIX_INTERVAL_COMBINED.out.versions)\n",
      "Nextflow operator collect with content: intervals_bed_combined        = intervals_combined.map{meta, bed -> bed }.collect()\n",
      "Nextflow operator collect with content: intervals_bed_gz_tbi_combined = TABIX_BGZIPTABIX_INTERVAL_COMBINED.out.gz_tbi.map{meta, gz, tbi -> [gz, tbi] }.collect()\n",
      "Error: Unclosed process block in file: ./git_repos\\sarek\\subworkflows\\local\\prepare_intervals\\main.nf\n",
      "Process PREPARE_REFERENCE_CNVKIT in filepath ./git_repos\\sarek\\subworkflows\\local\\prepare_reference_cnvkit\\main.nf\n",
      "Nextflow operator flatten with content: CNVKIT_ANTITARGET(intervals_bed_combined.flatten().map{ it -> [ [ id:'intervals' ], it ] })\n",
      "Nextflow operator map with content: CNVKIT_REFERENCE(fasta, intervals_bed_combined, CNVKIT_ANTITARGET.out.bed.map{ meta, bed -> [ bed ] } )\n",
      "Nextflow operator mix with content: versions = versions.mix(CNVKIT_ANTITARGET.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(CNVKIT_REFERENCE.out.versions)\n",
      "Nextflow operator collect with content: cnvkit_reference = CNVKIT_REFERENCE.out.cnn.collect()\n",
      "Error: Unclosed process block in file: ./git_repos\\sarek\\subworkflows\\local\\prepare_reference_cnvkit\\main.nf\n",
      "Process SAMPLESHEET_TO_CHANNEL in filepath ./git_repos\\sarek\\subworkflows\\local\\samplesheet_to_channel\\main.nf\n",
      "Nextflow operator dump with content: ch_from_samplesheet.dump(tag:\"ch_from_samplesheet\")\n",
      "Nextflow operator map with content: input_sample = ch_from_samplesheet.map{ meta, fastq_1, fastq_2, table, cram, crai, bam, bai, vcf, variantcaller ->\n",
      "Nextflow operator tap with content: }.tap{ ch_with_patient_sample } // save the channel\n",
      "Nextflow operator groupTuple with content: .groupTuple() //group by patient_sample to get all lanes\n",
      "Nextflow operator map with content: .map { patient_sample, ch_items ->\n",
      "Nextflow operator combine with content: }.combine(ch_with_patient_sample, by: 0) // for each entry add numLanes\n",
      "Nextflow operator map with content: .map { patient_sample, num_lanes, ch_items ->\n",
      "Nextflow operator toInteger with content: meta           = meta - meta.subMap('lane') + [num_lanes: num_lanes.toInteger(), read_group: read_group.toString(), data_type: 'fastq', size: 1]\n",
      "Nextflow operator toInteger with content: meta            = meta - meta.subMap('lane') + [num_lanes: num_lanes.toInteger(), read_group: read_group.toString(), data_type: 'bam', size: 1]\n",
      "Nextflow operator filter with content: input_sample.filter{ it[0].status == 1 }.ifEmpty{ // In this case, the sample-sheet contains no tumor-samples\n",
      "Nextflow operator join with content: error('The sample-sheet only contains normal-samples, but the following tools, which were requested with \"--tools\", expect at least one tumor-sample : ' + tools_tumor_asked.join(\", \"))\n",
      "Nextflow operator filter with content: input_sample.filter{ it[0].status == 0 }.ifEmpty{ // In this case, the sample-sheet contains no normal/germline-samples\n",
      "Nextflow operator join with content: error('The sample-sheet only contains tumor-samples, but the following tools, which were requested by the option \"tools\", expect at least one normal-sample : ' + requested_tools_requiring_normal_samples.join(\", \"))\n",
      "Nextflow operator map with content: input_sample.map{\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Process VCF_ANNOTATE_ALL in filepath ./git_repos\\sarek\\subworkflows\\local\\vcf_annotate_all\\main.nf\n",
      "Nextflow operator mix with content: vcf_ann = vcf_ann.mix(VCF_ANNOTATE_BCFTOOLS.out.vcf_tbi)\n",
      "Nextflow operator mix with content: versions = versions.mix(VCF_ANNOTATE_BCFTOOLS.out.versions)\n",
      "Nextflow operator mix with content: reports = reports.mix(VCF_ANNOTATE_SNPEFF.out.reports)\n",
      "Nextflow operator mix with content: vcf_ann = vcf_ann.mix(VCF_ANNOTATE_SNPEFF.out.vcf_tbi)\n",
      "Nextflow operator mix with content: versions = versions.mix(VCF_ANNOTATE_SNPEFF.out.versions)\n",
      "Nextflow operator map with content: vcf_ann_for_merge = VCF_ANNOTATE_SNPEFF.out.vcf_tbi.map{ meta, vcf, tbi -> [ meta, vcf, [] ] }\n",
      "Nextflow operator mix with content: reports = reports.mix(VCF_ANNOTATE_MERGE.out.reports)\n",
      "Nextflow operator mix with content: vcf_ann = vcf_ann.mix(VCF_ANNOTATE_MERGE.out.vcf_tbi)\n",
      "Nextflow operator mix with content: versions = versions.mix(VCF_ANNOTATE_MERGE.out.versions)\n",
      "Nextflow operator map with content: vcf_for_vep = vcf.map{ meta, vcf -> [ meta, vcf, [] ] }\n",
      "Nextflow operator mix with content: reports  = reports.mix(VCF_ANNOTATE_ENSEMBLVEP.out.reports)\n",
      "Nextflow operator mix with content: vcf_ann  = vcf_ann.mix(VCF_ANNOTATE_ENSEMBLVEP.out.vcf_tbi)\n",
      "Nextflow operator mix with content: tab_ann  = tab_ann.mix(VCF_ANNOTATE_ENSEMBLVEP.out.tab)\n",
      "Nextflow operator mix with content: json_ann = json_ann.mix(VCF_ANNOTATE_ENSEMBLVEP.out.json)\n",
      "Nextflow operator mix with content: versions = versions.mix(VCF_ANNOTATE_ENSEMBLVEP.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\sarek\\subworkflows\\local\\vcf_annotate_all\\main.nf\n",
      "Process VCF_ANNOTATE_BCFTOOLS in filepath ./git_repos\\sarek\\subworkflows\\local\\vcf_annotate_bcftools\\main.nf\n",
      "Nextflow operator join with content: ch_vcf_tbi = BCFTOOLS_ANNOTATE.out.vcf.join(TABIX_TABIX.out.tbi, failOnDuplicate: true, failOnMismatch: true)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BCFTOOLS_ANNOTATE.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(TABIX_TABIX.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\sarek\\subworkflows\\local\\vcf_annotate_bcftools\\main.nf\n",
      "Process CONCATENATE_GERMLINE_VCFS in filepath ./git_repos\\sarek\\subworkflows\\local\\vcf_concatenate_germline\\main.nf\n",
      "Nextflow operator groupTuple with content: germline_vcfs_with_tbis = TABIX_EXT_VCF.out.gz_tbi.map{ meta, vcf, tbi -> [ meta.subMap('id'), vcf, tbi ] }.groupTuple()\n",
      "Nextflow operator mix with content: versions = versions.mix(ADD_INFO_TO_VCF.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(TABIX_EXT_VCF.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(GERMLINE_VCFS_CONCAT.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(GERMLINE_VCFS_CONCAT.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(TABIX_GERMLINE_VCFS_CONCAT_SORT.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\sarek\\subworkflows\\local\\vcf_concatenate_germline\\main.nf\n",
      "Process VCF_QC_BCFTOOLS_VCFTOOLS in filepath ./git_repos\\sarek\\subworkflows\\local\\vcf_qc_bcftools_vcftools\\main.nf\n",
      "Nextflow operator map with content: BCFTOOLS_STATS(vcf.map{ meta, vcf -> [ meta, vcf, [] ] }, [[:],[]], [[:],[]], [[:],[]], [[:],[]], [[:],[]])\n",
      "Nextflow operator mix with content: versions = versions.mix(BCFTOOLS_STATS.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(VCFTOOLS_TSTV_COUNT.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\sarek\\subworkflows\\local\\vcf_qc_bcftools_vcftools\\main.nf\n",
      "Process VCF_VARIANT_FILTERING_GATK in filepath ./git_repos\\sarek\\subworkflows\\local\\vcf_variant_filtering_gatk\\main.nf\n",
      "Nextflow operator combine with content: cnn_in = vcf.combine(intervals_bed_combined).map{ meta, vcf, tbi, intervals -> [ meta, vcf, tbi, [], intervals ] }\n",
      "Nextflow operator combine with content: FILTERVARIANTTRANCHES(CNNSCOREVARIANTS.out.vcf.join(CNNSCOREVARIANTS.out.tbi, failOnDuplicate: true, failOnMismatch: true).combine(intervals_bed_combined), known_sites, known_sites_tbi, fasta, fasta_fai, dict)\n",
      "Nextflow operator map with content: .map{ meta, vcf -> [ meta - meta.subMap('num_intervals'), vcf ] }\n",
      "Nextflow operator mix with content: versions = versions.mix(CNNSCOREVARIANTS.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(FILTERVARIANTTRANCHES.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\sarek\\subworkflows\\local\\vcf_variant_filtering_gatk\\main.nf\n",
      "Process BAM_NGSCHECKMATE in filepath ./git_repos\\sarek\\subworkflows\\nf-core\\bam_ngscheckmate\\main.nf\n",
      "Nextflow operator collect with content: ch_input_bed = ch_input.combine(ch_snp_bed.collect())\n",
      "Nextflow operator map with content: .map{ input_meta, input_file, bed_meta, bed_file ->\n",
      "Nextflow operator collect with content: BCFTOOLS_MPILEUP (ch_input_bed, ch_fasta.collect(), false)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BCFTOOLS_MPILEUP.out.versions)\n",
      "Nextflow operator map with content: .map{meta, vcf -> vcf}    // discard individual metas\n",
      "Nextflow operator collect with content: .collect()                // group into one channel\n",
      "Nextflow operator map with content: .map{files -> [files]}    // make the channel into [vcf1, vcf2, ...]\n",
      "Nextflow operator set with content: .set {ch_collected_vcfs}\n",
      "Nextflow operator map with content: .map{meta, bed -> meta} // use the snp_bed file meta as the meta for the merged channel\n",
      "Nextflow operator combine with content: .combine(ch_collected_vcfs) // add the vcf files after the meta, now looks like [meta, [vcf1, vcf2, ... ] ]\n",
      "Nextflow operator set with content: .set {ch_vcfs}\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(NGSCHECKMATE_NCM.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\sarek\\subworkflows\\nf-core\\bam_ngscheckmate\\main.nf\n",
      "Process VCF_ANNOTATE_ENSEMBLVEP in filepath ./git_repos\\sarek\\subworkflows\\nf-core\\vcf_annotate_ensemblvep\\main.nf\n",
      "Nextflow operator join with content: ch_vcf_tbi = ENSEMBLVEP_VEP.out.vcf.join(TABIX_TABIX.out.tbi, failOnDuplicate: true, failOnMismatch: true)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ENSEMBLVEP_VEP.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(TABIX_TABIX.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\sarek\\subworkflows\\nf-core\\vcf_annotate_ensemblvep\\main.nf\n",
      "Process VCF_ANNOTATE_SNPEFF in filepath ./git_repos\\sarek\\subworkflows\\nf-core\\vcf_annotate_snpeff\\main.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SNPEFF_SNPEFF.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(TABIX_BGZIPTABIX.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\sarek\\subworkflows\\nf-core\\vcf_annotate_snpeff\\main.nf\n",
      "Process SAREK in filepath ./git_repos\\sarek\\workflows\\sarek.nf\n",
      "Nextflow operator map with content: vep_cache    = DOWNLOAD_CACHE_SNPEFF_VEP.out.ensemblvep_cache.map{ meta, cache -> [ cache ] }\n",
      "Nextflow operator mix with content: versions = versions.mix(DOWNLOAD_CACHE_SNPEFF_VEP.out.versions)\n",
      "Nextflow operator collect with content: dict       = params.dict        ? Channel.fromPath(params.dict).map{ it -> [ [id:'dict'], it ] }.collect()\n",
      "Nextflow operator collect with content: fasta_fai  = params.fasta_fai   ? Channel.fromPath(params.fasta_fai).collect()\n",
      "Nextflow operator collect with content: bwa        = params.bwa         ? Channel.fromPath(params.bwa).collect()\n",
      "Nextflow operator collect with content: bwamem2    = params.bwamem2     ? Channel.fromPath(params.bwamem2).collect()\n",
      "Nextflow operator collect with content: dragmap    = params.dragmap     ? Channel.fromPath(params.dragmap).collect()\n",
      "Nextflow operator collect with content: dbsnp_tbi              = params.dbsnp                   ? params.dbsnp_tbi             ? Channel.fromPath(params.dbsnp_tbi).collect()             : PREPARE_GENOME.out.dbsnp_tbi             : Channel.value([])\n",
      "Nextflow operator collect with content: germline_resource_tbi  = params.germline_resource       ? params.germline_resource_tbi ? Channel.fromPath(params.germline_resource_tbi).collect() : PREPARE_GENOME.out.germline_resource_tbi : [] //do not change to Channel.value([]), the check for its existence then fails for Getpileupsumamries\n",
      "Nextflow operator collect with content: known_indels_tbi       = params.known_indels            ? params.known_indels_tbi      ? Channel.fromPath(params.known_indels_tbi).collect()      : PREPARE_GENOME.out.known_indels_tbi      : Channel.value([])\n",
      "Nextflow operator collect with content: known_snps_tbi         = params.known_snps              ? params.known_snps_tbi        ? Channel.fromPath(params.known_snps_tbi).collect()        : PREPARE_GENOME.out.known_snps_tbi        : Channel.value([])\n",
      "Nextflow operator collect with content: pon_tbi                = params.pon                     ? params.pon_tbi               ? Channel.fromPath(params.pon_tbi).collect()               : PREPARE_GENOME.out.pon_tbi               : Channel.value([])\n",
      "Nextflow operator collect with content: known_sites_indels     = dbsnp.concat(known_indels).collect()\n",
      "Nextflow operator collect with content: known_sites_indels_tbi = dbsnp_tbi.concat(known_indels_tbi).collect()\n",
      "Nextflow operator collect with content: known_sites_snps       = dbsnp.concat(known_snps).collect()\n",
      "Nextflow operator collect with content: known_sites_snps_tbi   = dbsnp_tbi.concat(known_snps_tbi).collect()\n",
      "Nextflow operator collect with content: intervals_bed_combined.map{it -> [ [ id:it.baseName ], it ]}.collect() :\n",
      "Nextflow operator map with content: intervals_and_num_intervals = intervals.map{ interval, num_intervals ->\n",
      "Nextflow operator map with content: intervals_bed_gz_tbi_and_num_intervals = intervals_bed_gz_tbi.map{ intervals, num_intervals ->\n",
      "Nextflow operator collect with content: cnvkit_reference = Channel.fromPath(params.cnvkit_reference).collect()\n",
      "Nextflow operator mix with content: versions = versions.mix(PREPARE_REFERENCE_CNVKIT.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(PREPARE_GENOME.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(PREPARE_INTERVALS.out.versions)\n",
      "Nextflow operator branch with content: input_sample_type = input_sample.branch{\n",
      "Nextflow operator mix with content: input_fastq = input_sample_type.fastq.mix(CONVERT_FASTQ_INPUT.out.reads)\n",
      "Nextflow operator collect with content: reports = reports.mix(FASTQC.out.zip.collect{ meta, logs -> logs })\n",
      "Nextflow operator first with content: versions = versions.mix(FASTQC.out.versions.first())\n",
      "Nextflow operator map with content: bam_converted_from_fastq = FASTQ_CREATE_UMI_CONSENSUS_FGBIO.out.consensusbam.map{ meta, bam -> [ meta, bam, [] ] }\n",
      "Nextflow operator mix with content: versions = versions.mix(CONVERT_FASTQ_UMI.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(FASTQ_CREATE_UMI_CONSENSUS_FGBIO.out.versions)\n",
      "Nextflow operator collect with content: reports = reports.mix(FASTP.out.json.collect{ meta, json -> json })\n",
      "Nextflow operator collect with content: reports = reports.mix(FASTP.out.html.collect{ meta, html -> html })\n",
      "Nextflow operator map with content: reads_for_alignment = FASTP.out.reads.map{ meta, reads ->\n",
      "Nextflow operator collate with content: read_files = reads.sort(false) { a,b -> a.getName().tokenize('.')[0] <=> b.getName().tokenize('.')[0] }.collate(2)\n",
      "Nextflow operator transpose with content: }.transpose()\n",
      "Nextflow operator mix with content: versions = versions.mix(FASTP.out.versions)\n",
      "Nextflow operator map with content: reads_for_alignment = reads_for_alignment.map{ meta, reads ->\n",
      "Nextflow operator map with content: bam_mapped = FASTQ_ALIGN_BWAMEM_MEM2_DRAGMAP_SENTIEON.out.bam.map{ meta, bam ->\n",
      "Nextflow operator groupTuple with content: }.groupTuple()\n",
      "Nextflow operator map with content: bai_mapped = FASTQ_ALIGN_BWAMEM_MEM2_DRAGMAP_SENTIEON.out.bai.map{ meta, bai ->\n",
      "Nextflow operator groupTuple with content: }.groupTuple()\n",
      "Nextflow operator mix with content: versions = versions.mix(BAM_MERGE_INDEX_SAMTOOLS.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(BAM_TO_CRAM_MAPPING.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(CONVERT_FASTQ_INPUT.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(FASTQ_ALIGN_BWAMEM_MEM2_DRAGMAP_SENTIEON.out.versions)\n",
      "Nextflow operator map with content: cram_for_markduplicates = params.step == 'mapping' ? bam_mapped : input_sample.map{ meta, input, index -> [ meta, input ] }\n",
      "Nextflow operator branch with content: input_markduplicates_convert = input_sample.branch{\n",
      "Nextflow operator mix with content: versions = versions.mix(BAM_TO_CRAM.out.versions)\n",
      "Nextflow operator mix with content: cram_skip_markduplicates = Channel.empty().mix(input_markduplicates_convert.cram, BAM_TO_CRAM.out.alignment_index)\n",
      "Nextflow operator collect with content: reports = reports.mix(CRAM_QC_NO_MD.out.reports.collect{ meta, report -> report })\n",
      "Nextflow operator mix with content: versions = versions.mix(CRAM_QC_NO_MD.out.versions)\n",
      "Nextflow operator map with content: dict.map{ meta, dict -> [ dict ] },\n",
      "Nextflow operator collect with content: reports = reports.mix(BAM_MARKDUPLICATES_SPARK.out.reports.collect{ meta, report -> report })\n",
      "Nextflow operator mix with content: versions = versions.mix(BAM_MARKDUPLICATES_SPARK.out.versions)\n",
      "Nextflow operator map with content: crai_for_markduplicates = params.step == 'mapping' ? bai_mapped : input_sample.map{ meta, input, index -> [ meta, index ] }\n",
      "Nextflow operator collect with content: reports = reports.mix(BAM_SENTIEON_DEDUP.out.reports.collect{ meta, report -> report })\n",
      "Nextflow operator mix with content: versions = versions.mix(BAM_SENTIEON_DEDUP.out.versions)\n",
      "Nextflow operator collect with content: reports = reports.mix(BAM_MARKDUPLICATES.out.reports.collect{ meta, report -> report })\n",
      "Nextflow operator mix with content: versions = versions.mix(BAM_MARKDUPLICATES.out.versions)\n",
      "Nextflow operator mix with content: ch_md_cram_for_restart = Channel.empty().mix(cram_markduplicates_no_spark, cram_markduplicates_spark, cram_sentieon_dedup)\n",
      "Nextflow operator map with content: .map{ meta, cram, crai -> [ meta + [data_type: \"cram\"], cram, crai ] }\n",
      "Nextflow operator mix with content: versions = versions.mix(CRAM_TO_BAM.out.versions)\n",
      "Nextflow operator branch with content: input_prepare_recal_convert = input_sample.branch{\n",
      "Nextflow operator mix with content: versions = versions.mix(BAM_TO_CRAM.out.versions)\n",
      "Nextflow operator map with content: .map{ meta, cram, crai -> [ meta + [data_type: \"cram\"], cram, crai ] }\n",
      "Nextflow operator mix with content: ch_cram_for_bam_baserecalibrator = Channel.empty().mix(ch_cram_from_bam, input_prepare_recal_convert.cram)\n",
      "Nextflow operator mix with content: ch_cram_for_bam_baserecalibrator = Channel.empty().mix(ch_md_cram_for_restart, cram_skip_markduplicates )\n",
      "Nextflow operator map with content: .map{ meta, cram, crai -> [ meta + [data_type: \"cram\"], cram, crai ] }\n",
      "Nextflow operator mix with content: versions = versions.mix(BAM_BASERECALIBRATOR_SPARK.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(BAM_BASERECALIBRATOR.out.versions)\n",
      "Nextflow operator mix with content: ch_table_bqsr = Channel.empty().mix(\n",
      "Nextflow operator collect with content: reports = reports.mix(ch_table_bqsr.collect{ meta, table -> table })\n",
      "Nextflow operator join with content: cram_applybqsr = ch_cram_for_bam_baserecalibrator.join(ch_table_bqsr, failOnDuplicate: true, failOnMismatch: true)\n",
      "Nextflow operator join with content: CHANNEL_BASERECALIBRATOR_CREATE_CSV(ch_md_cram_for_restart.join(ch_table_bqsr, failOnDuplicate: true), params.tools, params.skip_tools, params.save_output_as_bam, params.outdir)\n",
      "Nextflow operator branch with content: input_recal_convert = input_sample.branch{\n",
      "Nextflow operator map with content: input_only_table = input_recal_convert.bam.map{ meta, bam, bai, table -> [ meta, table ] }\n",
      "Nextflow operator map with content: input_only_bam   = input_recal_convert.bam.map{ meta, bam, bai, table -> [ meta, bam, bai ] }\n",
      "Nextflow operator mix with content: versions = versions.mix(BAM_TO_CRAM.out.versions)\n",
      "Nextflow operator mix with content: cram_applybqsr = Channel.empty().mix(\n",
      "Nextflow operator join with content: BAM_TO_CRAM.out.alignment_index.join(input_only_table, failOnDuplicate: true, failOnMismatch: true),\n",
      "Nextflow operator map with content: .map{ meta, cram, crai, table -> [ meta + [data_type: \"cram\"], cram, crai, table ]}\n",
      "Nextflow operator mix with content: versions = versions.mix(BAM_APPLYBQSR_SPARK.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(BAM_APPLYBQSR.out.versions)\n",
      "Nextflow operator mix with content: cram_variant_calling = Channel.empty().mix(\n",
      "Nextflow operator collect with content: reports = reports.mix(CRAM_QC_RECAL.out.reports.collect{ meta, report -> report })\n",
      "Nextflow operator mix with content: versions = versions.mix(CRAM_QC_RECAL.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(CRAM_TO_BAM_RECAL.out.versions)\n",
      "Nextflow operator mix with content: cram_variant_calling = Channel.empty().mix(\n",
      "Nextflow operator map with content: input_recal_convert.cram.map{ meta, cram, crai, table -> [ meta, cram, crai ] })\n",
      "Nextflow operator mix with content: cram_variant_calling = Channel.empty().mix(ch_cram_for_bam_baserecalibrator)\n",
      "Nextflow operator branch with content: input_variant_calling_convert = input_sample.branch{\n",
      "Nextflow operator mix with content: versions = versions.mix(BAM_TO_CRAM.out.versions)\n",
      "Nextflow operator mix with content: cram_variant_calling = Channel.empty().mix(BAM_TO_CRAM.out.alignment_index, input_variant_calling_convert.cram)\n",
      "Nextflow operator branch with content: cram_variant_calling_status = cram_variant_calling.branch{\n",
      "Nextflow operator map with content: cram_variant_calling_normal_to_cross = cram_variant_calling_status.normal.map{ meta, cram, crai -> [ meta.patient, meta, cram, crai ] }\n",
      "Nextflow operator map with content: cram_variant_calling_pair_to_cross = cram_variant_calling_status.tumor.map{ meta, cram, crai -> [ meta.patient, meta, cram, crai ] }\n",
      "Nextflow operator groupTuple with content: cram_variant_calling_tumor_grouped = cram_variant_calling_pair_to_cross.groupTuple()\n",
      "Nextflow operator join with content: cram_variant_calling_tumor_joined = cram_variant_calling_tumor_grouped.join(cram_variant_calling_normal_to_cross, failOnDuplicate: true, remainder: true)\n",
      "Nextflow operator filter with content: cram_variant_calling_tumor_filtered = cram_variant_calling_tumor_joined.filter{ it ->  !(it.last()) }\n",
      "Nextflow operator map with content: cram_variant_calling_tumor_only = cram_variant_calling_tumor_filtered.transpose().map{ it -> [it[1], it[2], it[3]] }\n",
      "Nextflow operator join with content: cram_variant_calling_normal_joined = cram_variant_calling_normal_to_cross.join(cram_variant_calling_tumor_grouped, failOnDuplicate: true, remainder: true)\n",
      "Nextflow operator filter with content: cram_variant_calling_normal_filtered = cram_variant_calling_normal_joined.filter{ it ->  !(it.last()) }\n",
      "Nextflow operator map with content: cram_variant_calling_status_normal = cram_variant_calling_normal_filtered.map{ it -> [it[1], it[2], it[3]] }\n",
      "Nextflow operator cross with content: cram_variant_calling_pair = cram_variant_calling_normal_to_cross.cross(cram_variant_calling_pair_to_cross)\n",
      "Nextflow operator map with content: .map { normal, tumor ->\n",
      "Nextflow operator mix with content: vcf_to_annotate = vcf_to_annotate.mix(BAM_VARIANT_CALLING_GERMLINE_ALL.out.vcf_deepvariant)\n",
      "Nextflow operator mix with content: vcf_to_annotate = vcf_to_annotate.mix(BAM_VARIANT_CALLING_GERMLINE_ALL.out.vcf_freebayes)\n",
      "Nextflow operator mix with content: vcf_to_annotate = vcf_to_annotate.mix(BAM_VARIANT_CALLING_GERMLINE_ALL.out.vcf_haplotypecaller)\n",
      "Nextflow operator mix with content: vcf_to_annotate = vcf_to_annotate.mix(BAM_VARIANT_CALLING_GERMLINE_ALL.out.vcf_manta)\n",
      "Nextflow operator mix with content: vcf_to_annotate = vcf_to_annotate.mix(BAM_VARIANT_CALLING_GERMLINE_ALL.out.vcf_sentieon_dnascope)\n",
      "Nextflow operator mix with content: vcf_to_annotate = vcf_to_annotate.mix(BAM_VARIANT_CALLING_GERMLINE_ALL.out.vcf_sentieon_haplotyper)\n",
      "Nextflow operator mix with content: vcf_to_annotate = vcf_to_annotate.mix(BAM_VARIANT_CALLING_GERMLINE_ALL.out.vcf_strelka)\n",
      "Nextflow operator mix with content: vcf_to_annotate = vcf_to_annotate.mix(BAM_VARIANT_CALLING_GERMLINE_ALL.out.vcf_tiddit)\n",
      "Nextflow operator mix with content: vcf_to_annotate = vcf_to_annotate.mix(BAM_VARIANT_CALLING_GERMLINE_ALL.out.vcf_mpileup)\n",
      "Nextflow operator mix with content: vcf_to_annotate = vcf_to_annotate.mix(BAM_VARIANT_CALLING_TUMOR_ONLY_ALL.out.vcf_all)\n",
      "Nextflow operator mix with content: vcf_to_annotate = vcf_to_annotate.mix(BAM_VARIANT_CALLING_SOMATIC_ALL.out.vcf_all)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Nextflow operator collect with content: reports = reports.mix(VCF_QC_BCFTOOLS_VCFTOOLS.out.bcftools_stats.collect{ meta, stats -> stats })\n",
      "Nextflow operator collect with content: reports = reports.mix(VCF_QC_BCFTOOLS_VCFTOOLS.out.vcftools_tstv_counts.collect{ meta, counts -> counts })\n",
      "Nextflow operator collect with content: reports = reports.mix(VCF_QC_BCFTOOLS_VCFTOOLS.out.vcftools_tstv_qual.collect{ meta, qual -> qual })\n",
      "Nextflow operator collect with content: reports = reports.mix(VCF_QC_BCFTOOLS_VCFTOOLS.out.vcftools_filter_summary.collect{ meta, summary -> summary })\n",
      "Nextflow operator mix with content: versions = versions.mix(BAM_VARIANT_CALLING_GERMLINE_ALL.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(BAM_VARIANT_CALLING_SOMATIC_ALL.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(BAM_VARIANT_CALLING_TUMOR_ONLY_ALL.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(POST_VARIANTCALLING.out.versions)\n",
      "Nextflow operator mix with content: versions = versions.mix(VCF_QC_BCFTOOLS_VCFTOOLS.out.versions)\n",
      "Nextflow operator map with content: vep_fasta = (params.vep_include_fasta) ? fasta.map{ fasta -> [ [ id:fasta.baseName ], fasta ] } : [[id: 'null'], []]\n",
      "Nextflow operator map with content: vcf_to_annotate.map{meta, vcf -> [ meta + [ file_name: vcf.baseName ], vcf ] },\n",
      "Nextflow operator mix with content: versions = versions.mix(VCF_ANNOTATE_ALL.out.versions)\n",
      "Nextflow operator mix with content: reports = reports.mix(VCF_ANNOTATE_ALL.out.reports)\n",
      "Nextflow operator collectFile with content: CUSTOM_DUMPSOFTWAREVERSIONS(versions.unique().collectFile(name: 'collated_versions.yml'))\n",
      "Nextflow operator collect with content: version_yaml = CUSTOM_DUMPSOFTWAREVERSIONS.out.mqc_yml.collect()\n",
      "Nextflow operator mix with content: multiqc_files = multiqc_files.mix(version_yaml)\n",
      "Nextflow operator collectFile with content: multiqc_files = multiqc_files.mix(ch_workflow_summary.collectFile(name: 'workflow_summary_mqc.yaml'))\n",
      "Nextflow operator collectFile with content: multiqc_files = multiqc_files.mix(ch_methods_description.collectFile(name: 'methods_description_mqc.yaml'))\n",
      "Nextflow operator collect with content: multiqc_files = multiqc_files.mix(reports.collect().ifEmpty([]))\n",
      "Nextflow operator collect with content: MULTIQC(multiqc_files.collect(), ch_multiqc_config.collect().ifEmpty([]), ch_multiqc_custom_config.collect().ifEmpty([]), ch_multiqc_logo.collect().ifEmpty([]))\n",
      "Nextflow operator toList with content: multiqc_report = MULTIQC.out.report.toList()\n",
      "Nextflow operator mix with content: versions = versions.mix(MULTIQC.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\sarek\\workflows\\sarek.nf\n",
      "Process INPUT_CHECK in filepath ./git_repos\\scflow\\subworkflows\\local\\input_check.nf\n",
      "Nextflow operator splitCsv with content: .splitCsv ( header:true, sep:',' )\n",
      "Nextflow operator map with content: .map { create_fastq_channels(it) }\n",
      "Nextflow operator set with content: .set { reads }\n",
      "Error: Unclosed process block in file: ./git_repos\\scflow\\subworkflows\\local\\input_check.nf\n",
      "Process SCFLOW in filepath ./git_repos\\scflow\\workflows\\scflow.nf\n",
      "Nextflow operator splitCsv with content: SCFLOW_CHECKINPUTS.out.checked_manifest.splitCsv(\n",
      "Nextflow operator map with content: .map { row -> tuple(row.key, row.filepath) },\n",
      "Nextflow operator collect with content: SCFLOW_QC.out.qc_summary.collect()\n",
      "Nextflow operator collect with content: SCFLOW_QC.out.qc_sce.collect(),\n",
      "Nextflow operator splitCsv with content: SCFLOW_FINALIZE.out.celltypes.splitCsv (\n",
      "Nextflow operator map with content: .map { row -> tuple(row.celltype, row.n_cells) },\n",
      "Error: Unclosed process block in file: ./git_repos\\scflow\\workflows\\scflow.nf\n",
      "Process NFCORE_SCRNASEQ in filepath ./git_repos\\scrnaseq\\main.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\scrnaseq\\main.nf\n",
      "Process SCRNASEQ_ALEVIN in filepath ./git_repos\\scrnaseq\\subworkflows\\local\\alevin.nf\n",
      "Nextflow operator collect with content: salmon_index = SIMPLEAF_INDEX.out.index.collect()\n",
      "Nextflow operator collect with content: transcript_tsv = SIMPLEAF_INDEX.out.transcript_tsv.collect()\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SIMPLEAF_INDEX.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SIMPLEAF_QUANT.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ALEVINQC.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\scrnaseq\\subworkflows\\local\\alevin.nf\n",
      "Process CELLRANGER_ALIGN in filepath ./git_repos\\scrnaseq\\subworkflows\\local\\align_cellranger.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(CELLRANGER_MKGTF.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(CELLRANGER_MKREF.out.versions)\n",
      "Nextflow operator map with content: ch_fastq.map{ meta, reads -> [meta + [\"chemistry\": protocol, \"gem\": meta.id, \"samples\": [meta.id]], reads] },\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(CELLRANGER_COUNT.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\scrnaseq\\subworkflows\\local\\align_cellranger.nf\n",
      "Process CELLRANGERARC_ALIGN in filepath ./git_repos\\scrnaseq\\subworkflows\\local\\align_cellrangerarc.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(CELLRANGERARC_MKGTF.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(CELLRANGERARC_MKREF.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(CELLRANGERARC_COUNT.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\scrnaseq\\subworkflows\\local\\align_cellrangerarc.nf\n",
      "Process UNIVERSC_ALIGN in filepath ./git_repos\\scrnaseq\\subworkflows\\local\\align_universc.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(CELLRANGER_MKGTF.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(CELLRANGER_MKREF.out.versions)\n",
      "Nextflow operator map with content: ch_fastq.map{ meta, reads -> [\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(UNIVERSC.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\scrnaseq\\subworkflows\\local\\align_universc.nf\n",
      "Process FASTQC_CHECK in filepath ./git_repos\\scrnaseq\\subworkflows\\local\\fastqc.nf\n",
      "Nextflow operator map with content: ch_fastq.map { ch -> [ ch[0], ch[n] ] }.set { ch_fastq }\n",
      "Nextflow operator map with content: .map { it -> [ it[1] ] }\n",
      "Nextflow operator set with content: .set { fastqc_zip_only }\n",
      "Nextflow operator map with content: .map { it -> [ it[1] ] }\n",
      "Nextflow operator set with content: .set { fastqc_html_only }\n",
      "Nextflow operator mix with content: fastqc_multiqc = fastqc_multiqc.mix( fastqc_zip_only, fastqc_html_only )\n",
      "Error: Unclosed process block in file: ./git_repos\\scrnaseq\\subworkflows\\local\\fastqc.nf\n",
      "Process INPUT_CHECK in filepath ./git_repos\\scrnaseq\\subworkflows\\local\\input_check.nf\n",
      "Nextflow operator splitCsv with content: .splitCsv ( header:true, sep:',' )\n",
      "Nextflow operator map with content: .map { create_fastq_channel(it) }\n",
      "Nextflow operator groupTuple with content: .groupTuple(by: [0])\n",
      "Nextflow operator flatten with content: .map { meta, sample_type, sub_sample, reads -> [ meta, sample_type.flatten(), sub_sample.flatten(), reads.flatten() ] }\n",
      "Nextflow operator set with content: .set { reads }\n",
      "Nextflow operator flatten with content: .map { meta, reads -> [ meta, reads.flatten() ] }\n",
      "Nextflow operator set with content: .set { reads }\n",
      "Error: Unclosed process block in file: ./git_repos\\scrnaseq\\subworkflows\\local\\input_check.nf\n",
      "Process KALLISTO_BUSTOOLS in filepath ./git_repos\\scrnaseq\\subworkflows\\local\\kallisto_bustools.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GENE_MAP.out.versions)\n",
      "Nextflow operator collect with content: txp2gene = KALLISTOBUSTOOLS_REF.out.t2g.collect()\n",
      "Nextflow operator collect with content: kallisto_index = KALLISTOBUSTOOLS_REF.out.index.collect()\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(KALLISTOBUSTOOLS_REF.out.versions)\n",
      "Nextflow operator ifEmpty with content: t1c = KALLISTOBUSTOOLS_REF.out.cdna_t2c.ifEmpty{ [] }\n",
      "Nextflow operator ifEmpty with content: t2c = KALLISTOBUSTOOLS_REF.out.intron_t2c.ifEmpty{ [] }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(KALLISTOBUSTOOLS_COUNT.out.versions)\n",
      "Nextflow operator count with content: counts = KALLISTOBUSTOOLS_COUNT.out.count\n",
      "Nextflow operator collect with content: txp2gene = txp2gene.collect()\n",
      "Error: Unclosed process block in file: ./git_repos\\scrnaseq\\subworkflows\\local\\kallisto_bustools.nf\n",
      "Process MTX_CONVERSION in filepath ./git_repos\\scrnaseq\\subworkflows\\local\\mtx_conversion.nf\n",
      "Nextflow operator map with content: mtx_matrices = mtx_matrices.map { meta, mtx_files ->\n",
      "Nextflow operator filter with content: .filter { meta, mtx_files -> mtx_files } // Remove any that are missing the relevant files\n",
      "Nextflow operator collect with content: MTX_TO_H5AD.out.h5ad.collect(), // gather all sample-specific files\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(MTX_TO_H5AD.out.versions, MTX_TO_SEURAT.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\scrnaseq\\subworkflows\\local\\mtx_conversion.nf\n",
      "Process STARSOLO in filepath ./git_repos\\scrnaseq\\subworkflows\\local\\starsolo.nf\n",
      "Nextflow operator map with content: genome_fasta.map{ f -> [[id: f.baseName], f]},\n",
      "Nextflow operator map with content: gtf.map{ g -> [[id: g.baseName], g]}\n",
      "Nextflow operator collect with content: star_index = STAR_GENOMEGENERATE.out.index.collect()\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(STAR_GENOMEGENERATE.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(STAR_ALIGN.out.versions)\n",
      "Nextflow operator map with content: star_index  = star_index.map{ meta, index -> index}\n",
      "Error: Unclosed process block in file: ./git_repos\\scrnaseq\\subworkflows\\local\\starsolo.nf\n",
      "Process SCRNASEQ in filepath ./git_repos\\scrnaseq\\workflows\\scrnaseq.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(INPUT_CHECK.out.versions)\n",
      "Nextflow operator mix with content: ch_versions       = ch_versions.mix(FASTQC_CHECK.out.fastqc_version)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(KALLISTO_BUSTOOLS.out.ch_versions)\n",
      "Nextflow operator mix with content: ch_mtx_matrices = ch_mtx_matrices.mix(KALLISTO_BUSTOOLS.out.counts)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SCRNASEQ_ALEVIN.out.ch_versions)\n",
      "Nextflow operator mix with content: ch_mtx_matrices = ch_mtx_matrices.mix(SCRNASEQ_ALEVIN.out.alevin_results)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(STARSOLO.out.ch_versions)\n",
      "Nextflow operator mix with content: ch_mtx_matrices = ch_mtx_matrices.mix(STARSOLO.out.star_counts)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(CELLRANGER_ALIGN.out.ch_versions)\n",
      "Nextflow operator mix with content: ch_mtx_matrices = ch_mtx_matrices.mix(CELLRANGER_ALIGN.out.cellranger_out)\n",
      "Nextflow operator map with content: ch_multiqc_cellranger = CELLRANGER_ALIGN.out.cellranger_out.map{\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(UNIVERSC_ALIGN.out.ch_versions)\n",
      "Nextflow operator mix with content: ch_mtx_matrices = ch_mtx_matrices.mix(UNIVERSC_ALIGN.out.universc_out)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(CELLRANGERARC_ALIGN.out.ch_versions)\n",
      "Nextflow operator mix with content: ch_mtx_matrices = ch_mtx_matrices.mix(CELLRANGERARC_ALIGN.out.cellranger_arc_out)\n",
      "Nextflow operator mix with content: ch_versions.mix(MTX_CONVERSION.out.ch_versions)\n",
      "Nextflow operator collectFile with content: ch_versions.unique().collectFile(name: 'collated_versions.yml')\n",
      "Nextflow operator dump with content: ch_multiqc_fastqc.dump(tag: 'fastqc', pretty: true)\n",
      "Nextflow operator dump with content: ch_multiqc_alevin.dump(tag: 'alevin', pretty: true)\n",
      "Nextflow operator dump with content: ch_multiqc_star.dump(tag: 'star', pretty: true)\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_workflow_summary.collectFile(name: 'workflow_summary_mqc.yaml'))\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_methods_description.collectFile(name: 'methods_description_mqc.yaml'))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(CUSTOM_DUMPSOFTWAREVERSIONS.out.mqc_yml.collect())\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(ch_multiqc_fastqc.collect{ meta, qcfile -> qcfile }.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(ch_multiqc_alevin.collect{ meta, qcfile -> qcfile }.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(ch_multiqc_star.collect{ meta, qcfile -> qcfile }.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(ch_multiqc_cellranger.collect().ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files.collect(),\n",
      "Nextflow operator toList with content: ch_multiqc_config.toList(),\n",
      "Nextflow operator toList with content: ch_multiqc_custom_config.toList(),\n",
      "Nextflow operator toList with content: ch_multiqc_logo.toList()\n",
      "Nextflow operator toList with content: multiqc_report = MULTIQC.out.report.toList()\n",
      "Error: Unclosed process block in file: ./git_repos\\scrnaseq\\workflows\\scrnaseq.nf\n",
      "Process NFCORE_SMRNASEQ in filepath ./git_repos\\smrnaseq\\main.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\smrnaseq\\main.nf\n",
      "Process CONTAMINANT_FILTER in filepath ./git_repos\\smrnaseq\\subworkflows\\local\\contaminant_filter.nf\n",
      "Nextflow operator set with content: reads.set { rrna_reads }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(INDEX_RRNA.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(MAP_RRNA.out.versions)\n",
      "Nextflow operator ifEmpty with content: ch_filter_stats = ch_filter_stats.mix(MAP_RRNA.out.stats.ifEmpty(null))\n",
      "Nextflow operator set with content: MAP_RRNA.out.unmapped.set { rrna_reads }\n",
      "Nextflow operator set with content: rrna_reads.set { trna_reads }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(INDEX_TRNA.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(MAP_TRNA.out.versions)\n",
      "Nextflow operator ifEmpty with content: ch_filter_stats = ch_filter_stats.mix(MAP_TRNA.out.stats.ifEmpty(null))\n",
      "Nextflow operator set with content: MAP_TRNA.out.unmapped.set { trna_reads }\n",
      "Nextflow operator set with content: trna_reads.set { cdna_reads }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BLAT_CDNA.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(INDEX_CDNA.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(MAP_CDNA.out.versions)\n",
      "Nextflow operator ifEmpty with content: ch_filter_stats = ch_filter_stats.mix(MAP_CDNA.out.stats.ifEmpty(null))\n",
      "Nextflow operator set with content: MAP_CDNA.out.unmapped.set { cdna_reads }\n",
      "Nextflow operator set with content: cdna_reads.set { ncrna_reads }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BLAT_NCRNA.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(INDEX_NCRNA.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(MAP_NCRNA.out.versions)\n",
      "Nextflow operator ifEmpty with content: ch_filter_stats = ch_filter_stats.mix(MAP_NCRNA.out.stats.ifEmpty(null))\n",
      "Nextflow operator set with content: MAP_NCRNA.out.unmapped.set { ncrna_reads }\n",
      "Nextflow operator set with content: ncrna_reads.set { pirna_reads }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BLAT_PIRNA.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(INDEX_PIRNA.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(MAP_PIRNA.out.versions)\n",
      "Nextflow operator ifEmpty with content: ch_filter_stats = ch_filter_stats.mix(MAP_PIRNA.out.stats.ifEmpty(null))\n",
      "Nextflow operator set with content: MAP_PIRNA.out.unmapped.set { pirna_reads }\n",
      "Nextflow operator set with content: pirna_reads.set { other_cont_reads }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BLAT_OTHER.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(INDEX_OTHER.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(MAP_OTHER.out.versions)\n",
      "Nextflow operator ifEmpty with content: ch_filter_stats = ch_filter_stats.mix(MAP_OTHER.out.stats.ifEmpty(null))\n",
      "Nextflow operator set with content: MAP_OTHER.out.unmapped.set { other_cont_reads }\n",
      "Nextflow operator collect with content: FILTER_STATS ( other_cont_reads, ch_filter_stats.collect() )\n",
      "Error: Unclosed process block in file: ./git_repos\\smrnaseq\\subworkflows\\local\\contaminant_filter.nf\n",
      "Process FASTQC_FASTP in filepath ./git_repos\\smrnaseq\\subworkflows\\local\\fastqc_fastp.nf\n",
      "Nextflow operator map with content: adapterseq      = reads.map { meta, _ -> [meta, null] }\n",
      "Nextflow operator first with content: ch_versions     = ch_versions.mix(FASTQC_RAW.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions       = ch_versions.mix(FASTP.out.versions.first())\n",
      "Nextflow operator join with content: .join(trim_json)\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator set with content: .set { trim_reads }\n",
      "Nextflow operator map with content: .map { meta, json -> [meta, getFastpAdapterSequence(json)] }\n",
      "Nextflow operator set with content: .set { adapterseq }\n",
      "Nextflow operator first with content: ch_versions      = ch_versions.mix(FASTQC_TRIM.out.versions.first())\n",
      "Nextflow operator ifEmpty with content: versions = ch_versions.ifEmpty(null) // channel: [ versions.yml ]\n",
      "Error: Unclosed process block in file: ./git_repos\\smrnaseq\\subworkflows\\local\\fastqc_fastp.nf\n",
      "Process GENOME_QUANT in filepath ./git_repos\\smrnaseq\\subworkflows\\local\\genome_quant.nf\n",
      "Nextflow operator mix with content: ch_versions     = ch_versions.mix(INDEX_GENOME.out.versions)\n",
      "Nextflow operator ifEmpty with content: bowtie_index    = Channel.fromPath(\"${index}**ebwt\", checkIfExists: true).ifEmpty { exit 1, \"Bowtie1 index directory not found: ${index}\" }\n",
      "Nextflow operator collect with content: BOWTIE_MAP_GENOME ( reads, bowtie_index.collect() )\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BOWTIE_MAP_GENOME.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BAM_SORT_STATS_SAMTOOLS.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\smrnaseq\\subworkflows\\local\\genome_quant.nf\n",
      "Process INPUT_CHECK in filepath ./git_repos\\smrnaseq\\subworkflows\\local\\input_check.nf\n",
      "Nextflow operator splitCsv with content: .splitCsv ( header:true, sep:',' )\n",
      "Nextflow operator map with content: .map { create_fastq_channel(it) }\n",
      "Nextflow operator set with content: .set { reads }\n",
      "Error: Unclosed process block in file: ./git_repos\\smrnaseq\\subworkflows\\local\\input_check.nf\n",
      "Process MIRDEEP2 in filepath ./git_repos\\smrnaseq\\subworkflows\\local\\mirdeep2.nf\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(MIRDEEP2_PIGZ.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(MIRDEEP2_MAPPER.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(MIRDEEP2_RUN.out.versions.first())\n",
      "Error: Unclosed process block in file: ./git_repos\\smrnaseq\\subworkflows\\local\\mirdeep2.nf\n",
      "Process MIRNA_QUANT in filepath ./git_repos\\smrnaseq\\subworkflows\\local\\mirna_quant.nf\n",
      "Nextflow operator set with content: PARSE_MATURE ( mature ).parsed_fasta.set { mirna_parsed }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(PARSE_MATURE.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(FORMAT_MATURE.out.versions)\n",
      "Nextflow operator set with content: PARSE_HAIRPIN ( hairpin ).parsed_fasta.set { hairpin_parsed }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(PARSE_HAIRPIN.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(FORMAT_HAIRPIN.out.versions)\n",
      "Nextflow operator set with content: INDEX_MATURE ( FORMAT_MATURE.out.formatted_fasta ).index.set { mature_bowtie }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(INDEX_MATURE.out.versions)\n",
      "Nextflow operator set with content: INDEX_HAIRPIN ( FORMAT_HAIRPIN.out.formatted_fasta ).index.set { hairpin_bowtie }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(INDEX_HAIRPIN.out.versions)\n",
      "Nextflow operator map with content: .map { add_suffix(it, \"mature\") }\n",
      "Nextflow operator dump with content: .dump (tag:'msux')\n",
      "Nextflow operator set with content: .set { reads_mirna }\n",
      "Nextflow operator collect with content: BOWTIE_MAP_MATURE ( reads_mirna, mature_bowtie.collect() )\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BOWTIE_MAP_MATURE.out.versions)\n",
      "Nextflow operator map with content: .map { add_suffix(it, \"hairpin\") }\n",
      "Nextflow operator dump with content: .dump (tag:'hsux')\n",
      "Nextflow operator set with content: .set { reads_hairpin }\n",
      "Nextflow operator collect with content: BOWTIE_MAP_HAIRPIN ( reads_hairpin, hairpin_bowtie.collect() )\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BOWTIE_MAP_HAIRPIN.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BAM_STATS_MATURE.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BAM_STATS_HAIRPIN.out.versions)\n",
      "Nextflow operator collect with content: BAM_STATS_MATURE.out.idxstats.collect{it[1]}\n",
      "Nextflow operator collect with content: .mix(BAM_STATS_HAIRPIN.out.idxstats.collect{it[1]})\n",
      "Nextflow operator dump with content: .dump(tag:'edger')\n",
      "Nextflow operator flatten with content: .flatten()\n",
      "Nextflow operator collect with content: .collect()\n",
      "Nextflow operator set with content: .set { edger_input }\n",
      "Nextflow operator map with content: .map { add_suffix(it, \"seqcluster\") }\n",
      "Nextflow operator dump with content: .dump (tag:'ssux')\n",
      "Nextflow operator set with content: .set { reads_seqcluster }\n",
      "Nextflow operator set with content: SEQCLUSTER_SEQUENCES ( reads_seqcluster ).collapsed.set { reads_collapsed }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SEQCLUSTER_SEQUENCES.out.versions)\n",
      "Nextflow operator collect with content: BOWTIE_MAP_SEQCLUSTER ( reads_collapsed, hairpin_bowtie.collect() )\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BOWTIE_MAP_SEQCLUSTER.out.versions)\n",
      "Nextflow operator collect with content: MIRTOP_QUANT ( BOWTIE_MAP_SEQCLUSTER.out.bam.collect{it[1]}, FORMAT_HAIRPIN.out.formatted_fasta.collect{it[1]}, gtf )\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(MIRTOP_QUANT.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(TABLE_MERGE.out.versions)\n",
      "Nextflow operator map with content: .map { add_suffix(it, \"genome\") }\n",
      "Nextflow operator dump with content: .dump (tag:'gsux')\n",
      "Nextflow operator set with content: .set { reads_genome }\n",
      "Error: Unclosed process block in file: ./git_repos\\smrnaseq\\subworkflows\\local\\mirna_quant.nf\n",
      "Process MIRTRACE in filepath ./git_repos\\smrnaseq\\subworkflows\\local\\mirtrace.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\smrnaseq\\subworkflows\\local\\mirtrace.nf\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Process BAM_SORT_STATS_SAMTOOLS in filepath ./git_repos\\smrnaseq\\subworkflows\\nf-core\\bam_sort_stats_samtools\\main.nf\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SAMTOOLS_SORT.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SAMTOOLS_INDEX.out.versions.first())\n",
      "Nextflow operator join with content: .join(SAMTOOLS_INDEX.out.bai, by: [0], remainder: true)\n",
      "Nextflow operator join with content: .join(SAMTOOLS_INDEX.out.csi, by: [0], remainder: true)\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator set with content: .set { ch_bam_bai }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BAM_STATS_SAMTOOLS.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\smrnaseq\\subworkflows\\nf-core\\bam_sort_stats_samtools\\main.nf\n",
      "Process BAM_STATS_SAMTOOLS in filepath ./git_repos\\smrnaseq\\subworkflows\\nf-core\\bam_stats_samtools\\main.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_STATS.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_FLAGSTAT.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_IDXSTATS.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\smrnaseq\\subworkflows\\nf-core\\bam_stats_samtools\\main.nf\n",
      "Process SMRNASEQ in filepath ./git_repos\\smrnaseq\\workflows\\smrnaseq.nf\n",
      "Nextflow operator dump with content: .dump(tag: 'group')\n",
      "Nextflow operator branch with content: .branch {\n",
      "Nextflow operator flatten with content: return [ meta, fastq.flatten() ]\n",
      "Nextflow operator flatten with content: return [ meta, fastq.flatten() ]\n",
      "Nextflow operator set with content: .set { ch_fastq }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(INPUT_CHECK.out.versions)\n",
      "Nextflow operator mix with content: .mix(ch_fastq.single)\n",
      "Nextflow operator set with content: .set { ch_cat_fastq }\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(CAT_FASTQ.out.versions.first().ifEmpty(null))\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(FASTQC_FASTP.out.versions)\n",
      "Nextflow operator join with content: .join( FASTQC_FASTP.out.reads )\n",
      "Nextflow operator map with content: .map { meta, adapterseq, reads -> [adapterseq, meta.id, reads] }\n",
      "Nextflow operator groupTuple with content: .groupTuple()\n",
      "Nextflow operator set with content: .set { ch_mirtrace_inputs }\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(MIRTRACE.out.versions.ifEmpty(null))\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(CONTAMINANT_FILTER.out.versions)\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(MIRNA_QUANT.out.versions.ifEmpty(null))\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GENOME_QUANT.out.versions)\n",
      "Nextflow operator collect with content: GENOME_QUANT.out.index.collect(),\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(MIRDEEP2.out.versions)\n",
      "Nextflow operator collectFile with content: ch_versions.unique().collectFile(name: 'collated_versions.yml')\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(CUSTOM_DUMPSOFTWAREVERSIONS.out.mqc_yml.collect())\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_workflow_summary.collectFile(name: 'workflow_summary_mqc.yaml'))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(FASTQC_FASTP.out.fastqc_raw_zip.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(FASTQC_FASTP.out.trim_json.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(contamination_stats.collect().ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(genome_stats.collect({it[1]}).ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(MIRNA_QUANT.out.mature_stats.collect({it[1]}).ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(MIRNA_QUANT.out.hairpin_stats.collect({it[1]}).ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(MIRNA_QUANT.out.mirtop_logs.collect().ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(MIRTRACE.out.results.collect().ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files.collect(),\n",
      "Nextflow operator toList with content: ch_multiqc_config.toList(),\n",
      "Nextflow operator toList with content: ch_multiqc_custom_config.toList(),\n",
      "Nextflow operator toList with content: ch_multiqc_logo.toList()\n",
      "Nextflow operator toList with content: multiqc_report = MULTIQC.out.report.toList()\n",
      "Error: Unclosed process block in file: ./git_repos\\smrnaseq\\workflows\\smrnaseq.nf\n",
      "Process NFCORE_SPATIALTRANSCRIPTOMICS in filepath ./git_repos\\spatialtranscriptomics\\main.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\spatialtranscriptomics\\main.nf\n",
      "Process INPUT_CHECK in filepath ./git_repos\\spatialtranscriptomics\\subworkflows\\local\\input_check.nf\n",
      "Nextflow operator splitCsv with content: .splitCsv ( header: true, sep: ',')\n",
      "Nextflow operator branch with content: .branch   {\n",
      "Nextflow operator map with content: .map { it -> [it, it.fastq_dir]}\n",
      "Nextflow operator branch with content: .branch {\n",
      "Nextflow operator mix with content: .mix ( ch_spaceranger.dir )\n",
      "Nextflow operator map with content: .map { meta, dir -> meta + [fastq_dir: dir] }\n",
      "Nextflow operator map with content: ch_spaceranger_input = ch_spaceranger_combined.map { create_channel_spaceranger(it) }\n",
      "Nextflow operator map with content: .map    { create_channel_downstream_tar(it) }\n",
      "Nextflow operator branch with content: .branch {\n",
      "Nextflow operator mix with content: .mix ( ch_downstream.dir )\n",
      "Nextflow operator map with content: .map { meta, dir -> [sample: meta.id, spaceranger_dir: dir] }\n",
      "Nextflow operator map with content: ch_downstream_input = ch_downstream_combined.map { create_channel_downstream(it) }\n",
      "Error: Unclosed process block in file: ./git_repos\\spatialtranscriptomics\\subworkflows\\local\\input_check.nf\n",
      "Process SPACERANGER in filepath ./git_repos\\spatialtranscriptomics\\subworkflows\\local\\spaceranger.nf\n",
      "Nextflow operator map with content: ch_reference = SPACERANGER_UNTAR_REFERENCE.out.untar.map({meta, ref -> ref})\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SPACERANGER_UNTAR_REFERENCE.out.versions)\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SPACERANGER_COUNT.out.versions.first())\n",
      "Error: Unclosed process block in file: ./git_repos\\spatialtranscriptomics\\subworkflows\\local\\spaceranger.nf\n",
      "Process ST_DOWNSTREAM in filepath ./git_repos\\spatialtranscriptomics\\subworkflows\\local\\st_downstream.nf\n",
      "Nextflow operator collect with content: report_template = Channel.fromPath(\"${projectDir}/assets/_extensions\").collect()\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ST_QUALITY_CONTROLS.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ST_CLUSTERING.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ST_SPATIAL_DE.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\spatialtranscriptomics\\subworkflows\\local\\st_downstream.nf\n",
      "Process ST in filepath ./git_repos\\spatialtranscriptomics\\workflows\\spatialtranscriptomics.nf\n",
      "Nextflow operator map with content: INPUT_CHECK.out.ch_spaceranger_input.map{ it -> [it[0] /* meta */, it[1] /* reads */]}\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(FASTQC.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SPACERANGER.out.versions)\n",
      "Nextflow operator concat with content: ch_downstream_input = INPUT_CHECK.out.ch_downstream_input.concat(SPACERANGER.out.sr_dir).map{\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ST_READ_DATA.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ST_DOWNSTREAM.out.versions)\n",
      "Nextflow operator collectFile with content: ch_versions.unique().collectFile(name: 'collated_versions.yml')\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_workflow_summary.collectFile(name: 'workflow_summary_mqc.yaml').mix(\n",
      "Nextflow operator collectFile with content: ch_methods_description.collectFile(name: 'methods_description_mqc.yaml'),\n",
      "Nextflow operator collect with content: CUSTOM_DUMPSOFTWAREVERSIONS.out.mqc_yml.collect(),\n",
      "Nextflow operator collect with content: FASTQC.out.zip.collect{ meta, qcfile -> qcfile }\n",
      "Nextflow operator collect with content: ch_multiqc_files.collect(),\n",
      "Nextflow operator toList with content: ch_multiqc_config.toList(),\n",
      "Nextflow operator toList with content: ch_multiqc_custom_config.toList(),\n",
      "Nextflow operator toList with content: ch_multiqc_logo.toList()\n",
      "Nextflow operator toList with content: multiqc_report = MULTIQC.out.report.toList()\n",
      "Error: Unclosed process block in file: ./git_repos\\spatialtranscriptomics\\workflows\\spatialtranscriptomics.nf\n",
      "Process NFCORE_SPINNINGJENNY in filepath ./git_repos\\spinningjenny\\main.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\spinningjenny\\main.nf\n",
      "Process INPUT_CHECK in filepath ./git_repos\\spinningjenny\\subworkflows\\local\\input_check.nf\n",
      "Nextflow operator splitCsv with content: .splitCsv ( header:true, sep:',' )\n",
      "Nextflow operator map with content: .map { create_fastq_channel(it) }\n",
      "Nextflow operator set with content: .set { reads }\n",
      "Error: Unclosed process block in file: ./git_repos\\spinningjenny\\subworkflows\\local\\input_check.nf\n",
      "Process SPINNINGJENNY in filepath ./git_repos\\spinningjenny\\workflows\\spinningjenny.nf\n",
      "Nextflow operator map with content: .map { line ->\n",
      "Nextflow operator set with content: }.set{ pipe_params}\n",
      "Nextflow operator map with content: pipe_params.map {\n",
      "Nextflow operator set with content: }.transpose().set{reshaped_pars}\n",
      "Nextflow operator combine with content: xml_files.combine(Experiments).combine(n_batches).map{\n",
      "Nextflow operator set with content: }.set{data_for_model}\n",
      "Nextflow operator map with content: res_model.map{\n",
      "Nextflow operator groupTuple with content: }.groupTuple().set{files_pieces}\n",
      "Error: Unclosed process block in file: ./git_repos\\spinningjenny\\workflows\\spinningjenny.nf\n",
      "Process NFCORE_SSDS in filepath ./git_repos\\ssds\\main.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\ssds\\main.nf\n",
      "Process ALIGN_SS in filepath ./git_repos\\ssds\\subworkflows\\local\\align_ss.nf\n",
      "Nextflow operator map with content: ch_unprocessed = FILTER_SUPP_ALIGNMENTS.out.bam.map {return [[id:\"${it[0].id}.unprocessed_SSDS\", single_end: it[0].single_end], it[1]]}\n",
      "Nextflow operator map with content: PICARD_SORTSAM_QRY (FILTER_SUPP_ALIGNMENTS.out.bam.map {return [[id:it[0].id.replaceAll(\".unprocessed_SSDS\",\"\"), single_end: it[0].single_end], it[1]]}, 'queryname')\n",
      "Nextflow operator collect with content: ch_ssds_bam_unsorted = PARSESSDS.out.bam.collect{it[1]}.flatten().map {\n",
      "Nextflow operator groupTuple with content: .groupTuple()\n",
      "Nextflow operator collect with content: ch_ssds_bed_unsorted = PARSESSDS.out.bed.collect{it[1]}.flatten().map {\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Process INPUT_CHECK in filepath ./git_repos\\ssds\\subworkflows\\local\\index_bams.nf\n",
      "Nextflow operator splitCsv with content: .splitCsv ( header:true, sep:',' )\n",
      "Nextflow operator map with content: .map { create_fastq_channels(it) }\n",
      "Nextflow operator set with content: .set { reads }\n",
      "Error: Unclosed process block in file: ./git_repos\\ssds\\subworkflows\\local\\index_bams.nf\n",
      "Process INPUT_CHECK in filepath ./git_repos\\ssds\\subworkflows\\local\\input_check.nf\n",
      "Nextflow operator splitCsv with content: .splitCsv ( header:true, sep:',' )\n",
      "Nextflow operator map with content: .map { create_fastq_channels(it) }\n",
      "Nextflow operator set with content: .set { reads }\n",
      "Error: Unclosed process block in file: ./git_repos\\ssds\\subworkflows\\local\\input_check.nf\n",
      "Process MAKE_SSDS_BIGWIGS in filepath ./git_repos\\ssds\\subworkflows\\local\\make_ssds_bigwigs.nf\n",
      "Nextflow operator flatten with content: ch_coverage_bg = GENERATE_SSDS_COVERAGE.out.bedgraph.flatten().map {\n",
      "Nextflow operator groupTuple with content: .groupTuple()\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(GENERATE_SSDS_COVERAGE.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(UCSC_BEDGRAPHTOBIGWIG.out.versions.first())\n",
      "Error: Unclosed process block in file: ./git_repos\\ssds\\subworkflows\\local\\make_ssds_bigwigs.nf\n",
      "Process MARKDUPLICATES_AND_INDEX in filepath ./git_repos\\ssds\\subworkflows\\local\\markduplicates_and_index_bam.nf\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(PICARD_SORTSAM.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(PICARD_MARKDUPLICATES.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SAMTOOLS_INDEX.out.versions.first())\n",
      "Error: Unclosed process block in file: ./git_repos\\ssds\\subworkflows\\local\\markduplicates_and_index_bam.nf\n",
      "Process BAM_SORT_SAMTOOLS in filepath ./git_repos\\ssds\\subworkflows\\local\\sort_and_index.nf\n",
      "Nextflow operator join with content: .join(SAMTOOLS_INDEX.out.bai, by: [0], remainder: true)\n",
      "Nextflow operator join with content: .join(SAMTOOLS_INDEX.out.csi, by: [0], remainder: true)\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator set with content: .set { ch_bam_bai }\n",
      "Error: Unclosed process block in file: ./git_repos\\ssds\\subworkflows\\local\\sort_and_index.nf\n",
      "Process SORT_AND_INDEX_BAM in filepath ./git_repos\\ssds\\subworkflows\\local\\sort_and_index_bam.nf\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SAMTOOLS_SORTSAM.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SAMTOOLS_INDEX.out.versions.first())\n",
      "Error: Unclosed process block in file: ./git_repos\\ssds\\subworkflows\\local\\sort_and_index_bam.nf\n",
      "Process SSDS in filepath ./git_repos\\ssds\\workflows\\ssds.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(INPUT_CHECK.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(FASTQC.out.versions)\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(BWA_INDEX.out.versions.ifEmpty(null))\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(ALIGN_SS.out.versions.ifEmpty(null))\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(BEDTOOLS_MAKEWINDOWS.out.versions.ifEmpty(null))\n",
      "Nextflow operator filter with content: .filter {file(it[1]).size() > 40000}\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(MAKE_SSDS_BIGWIGS.out.versions.ifEmpty(null))\n",
      "Nextflow operator map with content: ch_bams_for_spot = ALIGN_SS.out.bam.map { [[\"id\":it[0].name, \"single-end\":it[0].\"single-end\"], it[1], it[2], it[1].name, it[0].type ] }\n",
      "Nextflow operator groupTuple with content: .groupTuple(by:0)\n",
      "Nextflow operator collect with content: ch_intervals_for_spot = ch_spot_intervals.map{[[genome:it[0].genome],it[1]]}.groupTuple(by:0).collect()\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(CALCULATESPOT.out.versions.ifEmpty(null))\n",
      "Nextflow operator map with content: ch_fingerprint_bams = ALIGN_SS.out.bam.map { [[\"id\":it[0].name, \"single-end\":it[0].\"single-end\"], it[1], it[2]] }\n",
      "Nextflow operator groupTuple with content: .groupTuple(by:0)\n",
      "Nextflow operator ifEmpty with content: ch_versions = ch_versions.mix(DEEPTOOLS_PLOTFINGERPRINT.out.versions.ifEmpty(null))\n",
      "Nextflow operator collectFile with content: ch_versions.unique().collectFile(name: 'collated_versions.yml')\n",
      "Nextflow operator mix with content: ch_multiqc_files = ch_multiqc_files.mix(Channel.from(ch_multiqc_config))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(ch_multiqc_custom_config.collect().ifEmpty([]))\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_workflow_summary.collectFile(name: 'workflow_summary_mqc.yaml'))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(CUSTOM_DUMPSOFTWAREVERSIONS.out.mqc_yml.collect())\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(FASTQC.out.zip.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(ALIGN_SS.out.reports.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(CALCULATESPOT.out.report.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(DEEPTOOLS_PLOTFINGERPRINT.out.metrics.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(DEEPTOOLS_PLOTFINGERPRINT.out.matrix.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files.collect()\n",
      "Nextflow operator toList with content: multiqc_report = MULTIQC_SSDS.out.report.toList()\n",
      "Nextflow operator mix with content: ch_versions    = ch_versions.mix(MULTIQC_SSDS.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\ssds\\workflows\\ssds.nf\n",
      "Process NFCORE_TAXPROFILER in filepath ./git_repos\\taxprofiler\\main.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\taxprofiler\\main.nf\n",
      "Process DB_CHECK in filepath ./git_repos\\taxprofiler\\subworkflows\\local\\db_check.nf\n",
      "Nextflow operator splitCsv with content: .splitCsv ( header:false, sep:',' )\n",
      "Nextflow operator first with content: .first()\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator splitCsv with content: .splitCsv ( header:true, sep:',' )\n",
      "Nextflow operator map with content: .map { row ->\n",
      "Nextflow operator splitCsv with content: .splitCsv ( header:true, sep:',' )\n",
      "Nextflow operator map with content: .map {[it.tool, it.db_name] }\n",
      "Nextflow operator groupTuple with content: .groupTuple()\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator unique with content: def unique_names = db_name.unique(false)\n",
      "Nextflow operator branch with content: .branch { db_meta, db ->\n",
      "Nextflow operator filter with content: .filter { db_meta, db -> params[\"run_${db_meta.tool}\"] }\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(UNTAR.out.versions.first())\n",
      "Nextflow operator mix with content: ch_final_dbs = ch_dbs_for_untar.skip.mix( UNTAR.out.untar )\n",
      "Nextflow operator join with content: if ( !row.keySet().containsAll(expected_headers) ) error(\"[nf-core/taxprofiler] ERROR: Invalid database input sheet - malformed column names. Please check input TSV. Column names should be: ${expected_headers.join(\", \")}\")\n",
      "Error: Unclosed process block in file: ./git_repos\\taxprofiler\\subworkflows\\local\\db_check.nf\n",
      "Process INPUT_CHECK in filepath ./git_repos\\taxprofiler\\subworkflows\\local\\input_check.nf\n",
      "Nextflow operator splitCsv with content: .splitCsv ( header:true, sep:',' )\n",
      "Nextflow operator map with content: .map{\n",
      "Nextflow operator groupTuple with content: .groupTuple()\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator transpose with content: .transpose(by: 0)\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator branch with content: .branch { row ->\n",
      "Nextflow operator map with content: .map { create_fastq_channel(it) }\n",
      "Nextflow operator map with content: .map { create_fastq_channel(it) }\n",
      "Nextflow operator map with content: .map { create_fasta_channel(it) }\n",
      "Error: Unclosed process block in file: ./git_repos\\taxprofiler\\subworkflows\\local\\input_check.nf\n",
      "Process LONGREAD_HOSTREMOVAL in filepath ./git_repos\\taxprofiler\\subworkflows\\local\\longread_hostremoval.nf\n",
      "Nextflow operator map with content: ch_minimap2_index = MINIMAP2_INDEX ( [ [], reference ] ).index.map { it[1] }\n",
      "Nextflow operator mix with content: ch_versions       = ch_versions.mix( MINIMAP2_INDEX.out.versions )\n",
      "Nextflow operator first with content: ch_versions        = ch_versions.mix( MINIMAP2_ALIGN.out.versions.first() )\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator first with content: ch_versions      = ch_versions.mix( SAMTOOLS_VIEW.out.versions.first() )\n",
      "Nextflow operator first with content: ch_versions      = ch_versions.mix( SAMTOOLS_FASTQ.out.versions.first() )\n",
      "Nextflow operator first with content: ch_versions      = ch_versions.mix( SAMTOOLS_INDEX.out.versions.first() )\n",
      "Nextflow operator join with content: .join(SAMTOOLS_INDEX.out.bai)\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SAMTOOLS_STATS.out.versions.first())\n",
      "Nextflow operator mix with content: ch_multiqc_files = ch_multiqc_files.mix( SAMTOOLS_STATS.out.stats )\n",
      "Error: Unclosed process block in file: ./git_repos\\taxprofiler\\subworkflows\\local\\longread_hostremoval.nf\n",
      "Process LONGREAD_PREPROCESSING in filepath ./git_repos\\taxprofiler\\subworkflows\\local\\longread_preprocessing.nf\n",
      "Nextflow operator map with content: .map { meta, reads -> [ meta + [single_end: 1], reads ] }\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(PORECHOP_PORECHOP.out.versions.first())\n",
      "Nextflow operator mix with content: ch_multiqc_files = ch_multiqc_files.mix( PORECHOP_PORECHOP.out.log )\n",
      "Nextflow operator map with content: ch_processed_reads = FILTLONG ( reads.map { meta, reads -> [meta, [], reads ] } )\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(FILTLONG.out.versions.first())\n",
      "Nextflow operator mix with content: ch_multiqc_files = ch_multiqc_files.mix( FILTLONG.out.log )\n",
      "Nextflow operator map with content: .map { meta, reads -> [ meta + [single_end: 1], reads ] }\n",
      "Nextflow operator map with content: ch_processed_reads = FILTLONG ( ch_clipped_reads.map { meta, reads -> [ meta, [], reads ] } ).reads\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(PORECHOP_PORECHOP.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(FILTLONG.out.versions.first())\n",
      "Nextflow operator mix with content: ch_multiqc_files = ch_multiqc_files.mix( PORECHOP_PORECHOP.out.log )\n",
      "Nextflow operator mix with content: ch_multiqc_files = ch_multiqc_files.mix( FILTLONG.out.log )\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix( FASTQC_PROCESSED.out.versions )\n",
      "Nextflow operator mix with content: ch_multiqc_files = ch_multiqc_files.mix( FASTQC_PROCESSED.out.zip )\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix( FALCO_PROCESSED.out.versions )\n",
      "Nextflow operator mix with content: ch_multiqc_files = ch_multiqc_files.mix( FALCO_PROCESSED.out.txt )\n",
      "Error: Unclosed process block in file: ./git_repos\\taxprofiler\\subworkflows\\local\\longread_preprocessing.nf\n",
      "Process PROFILING in filepath ./git_repos\\taxprofiler\\subworkflows\\local\\profiling.nf\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator combine with content: .combine(databases)\n",
      "Nextflow operator branch with content: .branch {\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator groupTuple with content: .groupTuple(by: [0,2])\n",
      "Nextflow operator multiMap with content: .multiMap {\n",
      "Nextflow operator flatten with content: reads: [ meta, reads.flatten() ]\n",
      "Nextflow operator transpose with content: .transpose()\n",
      "Nextflow operator map with content: .map{\n",
      "Nextflow operator mix with content: ch_multiqc_files       = ch_multiqc_files.mix( MALT_RUN.out.log )\n",
      "Nextflow operator first with content: ch_versions            = ch_versions.mix( MALT_RUN.out.versions.first(), MEGAN_RMA2INFO_TSV.out.versions.first() )\n",
      "Nextflow operator mix with content: ch_raw_classifications = ch_raw_classifications.mix( ch_maltrun_for_megan )\n",
      "Nextflow operator mix with content: ch_raw_profiles        = ch_raw_profiles.mix( MEGAN_RMA2INFO_TSV.out.txt )\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator multiMap with content: .multiMap {\n",
      "Nextflow operator mix with content: ch_multiqc_files       = ch_multiqc_files.mix( KRAKEN2_KRAKEN2.out.report )\n",
      "Nextflow operator first with content: ch_versions            = ch_versions.mix( KRAKEN2_KRAKEN2.out.versions.first() )\n",
      "Nextflow operator mix with content: ch_raw_classifications = ch_raw_classifications.mix( KRAKEN2_KRAKEN2.out.classified_reads_assignment )\n",
      "Nextflow operator mix with content: ch_raw_profiles        = ch_raw_profiles.mix(\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator filter with content: .filter {\n",
      "Nextflow operator filter with content: .filter { meta, db -> meta.tool == 'bracken' }\n",
      "Nextflow operator map with content: .map { meta, db -> [meta.db_name, meta, db] }\n",
      "Nextflow operator map with content: .map { meta, report -> [meta.db_name, meta, report] }\n",
      "Nextflow operator combine with content: .combine(ch_bracken_databases, by: 0)\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator multiMap with content: .multiMap { key, meta, report, db_meta, db ->\n",
      "Nextflow operator first with content: ch_versions     = ch_versions.mix(BRACKEN_BRACKEN.out.versions.first())\n",
      "Nextflow operator mix with content: ch_raw_profiles = ch_raw_profiles.mix(BRACKEN_BRACKEN.out.reports)\n",
      "Nextflow operator filter with content: .filter{\n",
      "Nextflow operator multiMap with content: .multiMap {\n",
      "Nextflow operator first with content: ch_versions            = ch_versions.mix( CENTRIFUGE_CENTRIFUGE.out.versions.first() )\n",
      "Nextflow operator mix with content: ch_raw_classifications = ch_raw_classifications.mix( CENTRIFUGE_CENTRIFUGE.out.results )\n",
      "Nextflow operator filter with content: .filter { meta, db -> meta.tool == 'centrifuge' }\n",
      "Nextflow operator map with content: .map { meta, db -> [meta.db_name, meta, db] }\n",
      "Nextflow operator first with content: ch_versions            = ch_versions.mix( CENTRIFUGE_KREPORT.out.versions.first() )\n",
      "Nextflow operator mix with content: ch_raw_profiles        = ch_raw_profiles.mix( CENTRIFUGE_KREPORT.out.kreport )\n",
      "Nextflow operator mix with content: ch_multiqc_files       = ch_multiqc_files.mix( CENTRIFUGE_KREPORT.out.kreport )\n",
      "Nextflow operator multiMap with content: .multiMap {\n",
      "Nextflow operator first with content: ch_versions        = ch_versions.mix( METAPHLAN_METAPHLAN.out.versions.first() )\n",
      "Nextflow operator mix with content: ch_raw_profiles    = ch_raw_profiles.mix( METAPHLAN_METAPHLAN.out.profile )\n",
      "Nextflow operator multiMap with content: .multiMap {\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix( KAIJU_KAIJU.out.versions.first() )\n",
      "Nextflow operator mix with content: ch_raw_classifications = ch_raw_classifications.mix( KAIJU_KAIJU.out.results )\n",
      "Nextflow operator filter with content: .filter { meta, db -> meta.tool == 'kaiju' }\n",
      "Nextflow operator map with content: .map { meta, db -> [meta.db_name, meta, db] }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix( KAIJU_KAIJU2TABLE_SINGLE.out.versions )\n",
      "Nextflow operator mix with content: ch_multiqc_files = ch_multiqc_files.mix( KAIJU_KAIJU2TABLE_SINGLE.out.summary )\n",
      "Nextflow operator mix with content: ch_raw_profiles  = ch_raw_profiles.mix( KAIJU_KAIJU2TABLE_SINGLE.out.summary )\n",
      "Nextflow operator multiMap with content: .multiMap {\n",
      "Nextflow operator first with content: ch_versions        = ch_versions.mix( DIAMOND_BLASTX.out.versions.first() )\n",
      "Nextflow operator mix with content: ch_raw_profiles    = ch_raw_profiles.mix( DIAMOND_BLASTX.out.tsv )\n",
      "Nextflow operator mix with content: ch_multiqc_files   = ch_multiqc_files.mix( DIAMOND_BLASTX.out.log )\n",
      "Nextflow operator filter with content: .filter{\n",
      "Nextflow operator multiMap with content: .multiMap {\n",
      "Nextflow operator first with content: ch_versions        = ch_versions.mix( MOTUS_PROFILE.out.versions.first() )\n",
      "Nextflow operator mix with content: ch_raw_profiles    = ch_raw_profiles.mix( MOTUS_PROFILE.out.out )\n",
      "Nextflow operator mix with content: ch_multiqc_files   = ch_multiqc_files.mix( MOTUS_PROFILE.out.log )\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator groupTuple with content: .groupTuple(by: [0,2,3])\n",
      "Nextflow operator flatMap with content: .flatMap { single_meta, reads, db_meta, db ->\n",
      "Nextflow operator collate with content: def batches = reads.collate(params.krakenuniq_batch_size)\n",
      "Nextflow operator collect with content: return batches.collect { batch -> [ single_meta + db_meta, batch.flatten(), db ]}\n",
      "Nextflow operator multiMap with content: .multiMap {\n",
      "Nextflow operator mix with content: ch_multiqc_files       = ch_multiqc_files.mix( KRAKENUNIQ_PRELOADEDKRAKENUNIQ.out.report )\n",
      "Nextflow operator first with content: ch_versions            = ch_versions.mix( KRAKENUNIQ_PRELOADEDKRAKENUNIQ.out.versions.first() )\n",
      "Nextflow operator mix with content: ch_raw_classifications = ch_raw_classifications.mix( KRAKENUNIQ_PRELOADEDKRAKENUNIQ.out.classified_assignment )\n",
      "Nextflow operator mix with content: ch_raw_profiles        = ch_raw_profiles.mix( KRAKENUNIQ_PRELOADEDKRAKENUNIQ.out.report )\n",
      "Nextflow operator filter with content: .filter {\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator multiMap with content: .multiMap  {\n",
      "Nextflow operator first with content: ch_versions            = ch_versions.mix( KMCP_SEARCH.out.versions.first() )\n",
      "Nextflow operator mix with content: ch_raw_classifications = ch_raw_classifications.mix(KMCP_SEARCH.out.result)\n",
      "Nextflow operator filter with content: .filter { meta, db -> meta.tool == 'kmcp' }\n",
      "Nextflow operator map with content: .map { meta, db -> [meta.db_name, meta, db] }\n",
      "Nextflow operator map with content: .map { meta, report -> [meta.db_name, meta, report] }\n",
      "Nextflow operator combine with content: .combine(ch_database_for_kmcp_profile, by: 0)\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator multiMap with content: .multiMap { key, meta, report, db_meta, db ->\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix( KMCP_PROFILE.out.versions.first() )\n",
      "Nextflow operator mix with content: ch_raw_profiles    = ch_raw_profiles.mix( KMCP_PROFILE.out.profile )\n",
      "Nextflow operator mix with content: ch_multiqc_files   = ch_multiqc_files.mix( KMCP_PROFILE.out.profile )\n",
      "Nextflow operator filter with content: .filter {\n",
      "Nextflow operator multiMap with content: .multiMap {\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix( GANON_CLASSIFY.out.versions.first() )\n",
      "Nextflow operator filter with content: .filter { meta, db -> meta.tool == \"ganon\" }\n",
      "Nextflow operator map with content: .map { meta, db -> [meta.db_name, meta, db] }\n",
      "Nextflow operator first with content: ch_versions            = ch_versions.mix( GANON_REPORT.out.versions.first() )\n",
      "Nextflow operator mix with content: ch_raw_profiles        = ch_raw_profiles.mix( GANON_REPORT.out.tre )\n",
      "Nextflow operator mix with content: ch_raw_classifications = ch_raw_classifications.mix( GANON_CLASSIFY.out.all )\n",
      "Nextflow operator first with content: motus_version   = params.run_motus ? MOTUS_PROFILE.out.versions.first() : Channel.empty()\n",
      "Error: Unclosed process block in file: ./git_repos\\taxprofiler\\subworkflows\\local\\profiling.nf\n",
      "Process SHORTREAD_ADAPTERREMOVAL in filepath ./git_repos\\taxprofiler\\subworkflows\\local\\shortread_adapterremoval.nf\n",
      "Nextflow operator branch with content: .branch{\n",
      "Nextflow operator mix with content: .mix(\n",
      "Nextflow operator map with content: .map { meta, reads ->\n",
      "Nextflow operator groupTuple with content: .groupTuple()\n",
      "Nextflow operator flatten with content: .map { meta, fastq -> [meta, fastq.flatten()] }\n",
      "Nextflow operator mix with content: .mix(ADAPTERREMOVAL_SINGLE.out.singles_truncated)\n",
      "Nextflow operator mix with content: .mix(\n",
      "Nextflow operator map with content: .map { meta, reads ->\n",
      "Nextflow operator groupTuple with content: .groupTuple()\n",
      "Nextflow operator flatten with content: .map { meta, fastq -> [meta, fastq.flatten()] }\n",
      "Nextflow operator mix with content: .mix(ADAPTERREMOVAL_SINGLE.out.singles_truncated)\n",
      "Nextflow operator mix with content: .mix(ADAPTERREMOVAL_SINGLE.out.singles_truncated)\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix( ADAPTERREMOVAL_SINGLE.out.versions.first() )\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix( ADAPTERREMOVAL_PAIRED.out.versions.first() )\n",
      "Nextflow operator mix with content: ch_multiqc_files = ch_multiqc_files.mix(\n",
      "Error: Unclosed process block in file: ./git_repos\\taxprofiler\\subworkflows\\local\\shortread_adapterremoval.nf\n",
      "Process SHORTREAD_COMPLEXITYFILTERING in filepath ./git_repos\\taxprofiler\\subworkflows\\local\\shortread_complexityfiltering.nf\n",
      "Nextflow operator first with content: ch_versions        =  ch_versions.mix( BBMAP_BBDUK.out.versions.first() )\n",
      "Nextflow operator mix with content: ch_multiqc_files   =  ch_multiqc_files.mix( BBMAP_BBDUK.out.log )\n",
      "Nextflow operator first with content: ch_versions        =  ch_versions.mix( PRINSEQPLUSPLUS.out.versions.first() )\n",
      "Error: Unclosed process block in file: ./git_repos\\taxprofiler\\subworkflows\\local\\shortread_complexityfiltering.nf\n",
      "Process SHORTREAD_FASTP in filepath ./git_repos\\taxprofiler\\subworkflows\\local\\shortread_fastp.nf\n",
      "Nextflow operator branch with content: .branch{\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator flatten with content: [ meta + [single_end:true], [ reads ].flatten() ]\n",
      "Nextflow operator mix with content: ch_fastp_reads_prepped = ch_fastp_reads_prepped_pe.mix( FASTP_SINGLE.out.reads )\n",
      "Nextflow operator mix with content: .mix( FASTP_SINGLE.out.reads )\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(FASTP_SINGLE.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(FASTP_PAIRED.out.versions.first())\n",
      "Nextflow operator mix with content: ch_multiqc_files = ch_multiqc_files.mix( FASTP_SINGLE.out.json )\n",
      "Nextflow operator mix with content: ch_multiqc_files = ch_multiqc_files.mix( FASTP_PAIRED.out.json )\n",
      "Error: Unclosed process block in file: ./git_repos\\taxprofiler\\subworkflows\\local\\shortread_fastp.nf\n",
      "Process SHORTREAD_HOSTREMOVAL in filepath ./git_repos\\taxprofiler\\subworkflows\\local\\shortread_hostremoval.nf\n",
      "Nextflow operator mix with content: ch_versions      = ch_versions.mix( BOWTIE2_BUILD.out.versions )\n",
      "Nextflow operator first with content: ch_bowtie2_index = index.first()\n",
      "Nextflow operator first with content: ch_versions      = ch_versions.mix( BOWTIE2_ALIGN.out.versions.first() )\n",
      "Nextflow operator mix with content: ch_multiqc_files = ch_multiqc_files.mix( BOWTIE2_ALIGN.out.log )\n",
      "Nextflow operator first with content: ch_versions      = ch_versions.mix( SAMTOOLS_INDEX.out.versions.first() )\n",
      "Nextflow operator join with content: .join(SAMTOOLS_INDEX.out.bai, remainder: true)\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SAMTOOLS_STATS.out.versions.first())\n",
      "Nextflow operator mix with content: ch_multiqc_files = ch_multiqc_files.mix( SAMTOOLS_STATS.out.stats )\n",
      "Error: Unclosed process block in file: ./git_repos\\taxprofiler\\subworkflows\\local\\shortread_hostremoval.nf\n",
      "Process SHORTREAD_PREPROCESSING in filepath ./git_repos\\taxprofiler\\subworkflows\\local\\shortread_preprocessing.nf\n",
      "Nextflow operator mix with content: ch_versions        =  ch_versions.mix( SHORTREAD_FASTP.out.versions )\n",
      "Nextflow operator mix with content: ch_multiqc_files   =  ch_multiqc_files.mix( SHORTREAD_FASTP.out.mqc )\n",
      "Nextflow operator mix with content: ch_versions        = ch_versions.mix( SHORTREAD_ADAPTERREMOVAL.out.versions )\n",
      "Nextflow operator mix with content: ch_multiqc_files   = ch_multiqc_files.mix( SHORTREAD_ADAPTERREMOVAL.out.mqc )\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix( FASTQC_PROCESSED.out.versions )\n",
      "Nextflow operator mix with content: ch_multiqc_files = ch_multiqc_files.mix( FASTQC_PROCESSED.out.zip )\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix( FALCO_PROCESSED.out.versions )\n",
      "Nextflow operator mix with content: ch_multiqc_files = ch_multiqc_files.mix( FALCO_PROCESSED.out.txt )\n",
      "Error: Unclosed process block in file: ./git_repos\\taxprofiler\\subworkflows\\local\\shortread_preprocessing.nf\n",
      "Process STANDARDISATION_PROFILES in filepath ./git_repos\\taxprofiler\\subworkflows\\local\\standardisation_profiles.nf\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator groupTuple with content: .groupTuple ()\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator flatten with content: [ meta, profiles.flatten() ]\n",
      "Nextflow operator collect with content: ch_taxpasta_tax_dir = params.taxpasta_taxonomy_dir ? Channel.fromPath(params.taxpasta_taxonomy_dir, checkIfExists: true).collect() : []\n",
      "Nextflow operator branch with content: .branch {\n",
      "Nextflow operator merge with content: TAXPASTA_MERGE       (ch_input_for_taxpasta.merge      , ch_taxpasta_tax_dir, [])\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix( TAXPASTA_MERGE.out.versions.first() )\n",
      "Nextflow operator first with content: ch_version = ch_versions.mix( TAXPASTA_STANDARDISE.out.versions.first() )\n",
      "Nextflow operator branch with content: .branch {\n",
      "Nextflow operator branch with content: .branch {\n",
      "Nextflow operator branch with content: .branch {\n",
      "Nextflow operator map with content: .map { [it[0]['db_name'], it[1]] }\n",
      "Nextflow operator groupTuple with content: .groupTuple()\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator map with content: .map { [it[0]['db_name'], it[1]] }\n",
      "Nextflow operator groupTuple with content: .groupTuple(sort: {-it.size()} )\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator mix with content: ch_multiqc_files = ch_multiqc_files.mix( KRAKENTOOLS_COMBINEKREPORTS_CENTRIFUGE.out.txt )\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix( KRAKENTOOLS_COMBINEKREPORTS_CENTRIFUGE.out.versions )\n",
      "Nextflow operator map with content: .map { [it[0]['db_name'], it[1]] }\n",
      "Nextflow operator groupTuple with content: .groupTuple()\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator mix with content: ch_multiqc_files = ch_multiqc_files.mix( KAIJU_KAIJU2TABLE_COMBINED.out.summary )\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix( KAIJU_KAIJU2TABLE_COMBINED.out.versions )\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator groupTuple with content: .groupTuple(sort: {-it.size()})\n",
      "Nextflow operator mix with content: ch_multiqc_files = ch_multiqc_files.mix( KRAKENTOOLS_COMBINEKREPORTS_KRAKEN.out.txt )\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix( KRAKENTOOLS_COMBINEKREPORTS_KRAKEN.out.versions )\n",
      "Nextflow operator map with content: .map { [it[0]['db_name'], it[1]] }\n",
      "Nextflow operator groupTuple with content: .groupTuple()\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator mix with content: ch_multiqc_files = ch_multiqc_files.mix( METAPHLAN_MERGEMETAPHLANTABLES.out.txt )\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix( METAPHLAN_MERGEMETAPHLANTABLES.out.versions )\n",
      "Nextflow operator map with content: .map { [it[0]['db_name'], it[1]] }\n",
      "Nextflow operator groupTuple with content: .groupTuple()\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix( MOTUS_MERGE.out.versions )\n",
      "Nextflow operator map with content: .map { [it[0]['db_name'], it[1]] }\n",
      "Nextflow operator groupTuple with content: .groupTuple()\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator mix with content: ch_multiqc_files = ch_multiqc_files.mix( GANON_TABLE.out.txt )\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix( GANON_TABLE.out.versions )\n",
      "Error: Unclosed process block in file: ./git_repos\\taxprofiler\\subworkflows\\local\\standardisation_profiles.nf\n",
      "Process VISUALIZATION_KRONA in filepath ./git_repos\\taxprofiler\\subworkflows\\local\\visualization_krona.nf\n",
      "Nextflow operator branch with content: .branch {\n",
      "Nextflow operator branch with content: .branch {\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator mix with content: .mix( ch_input_profiles.centrifuge )\n",
      "Nextflow operator mix with content: ch_krona_text = ch_krona_text.mix( KRAKENTOOLS_KREPORT2KRONA.out.txt )\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix( KRAKENTOOLS_KREPORT2KRONA.out.versions.first() )\n",
      "Nextflow operator map with content: .map{ meta, profiles -> [[meta['tool'], meta['db_name']], meta, profiles] }\n",
      "Nextflow operator combine with content: .combine( databases.map{ meta, db -> [[meta['tool'], meta['db_name']], db] }, by: 0 )\n",
      "Nextflow operator multiMap with content: .multiMap{\n",
      "Nextflow operator mix with content: ch_krona_text = ch_krona_text.mix( KAIJU_KAIJU2KRONA.out.txt )\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix( KAIJU_KAIJU2KRONA.out.versions.first() )\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix( KRONA_CLEANUP.out.versions.first() )\n",
      "Nextflow operator map with content: .map{[[id: it[0]['db_name'], tool: it[0]['tool']], it[1]]}\n",
      "Nextflow operator groupTuple with content: .groupTuple()\n",
      "Nextflow operator mix with content: ch_krona_html = ch_krona_html.mix( KRONA_KTIMPORTTEXT.out.html )\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix( KRONA_KTIMPORTTEXT.out.versions.first() )\n",
      "Nextflow operator map with content: .map{[[id: it[0]['db_name'], tool: it[0]['tool']], it[1]]}\n",
      "Nextflow operator groupTuple with content: .groupTuple()\n",
      "Nextflow operator mix with content: ch_krona_html.mix( KRONA_KTIMPORTTAXONOMY.out.html )\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix( MEGAN_RMA2INFO_KRONA.out.versions.first() )\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix( KRONA_KTIMPORTTAXONOMY.out.versions.first() )\n",
      "Error: Unclosed process block in file: ./git_repos\\taxprofiler\\subworkflows\\local\\visualization_krona.nf\n",
      "Process TAXPROFILER in filepath ./git_repos\\taxprofiler\\workflows\\taxprofiler.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(INPUT_CHECK.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(DB_CHECK.out.versions)\n",
      "Nextflow operator mix with content: ch_input_for_fastqc = INPUT_CHECK.out.fastq.mix( INPUT_CHECK.out.nanopore )\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(FALCO.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(FASTQC.out.versions.first())\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix( SHORTREAD_PREPROCESSING.out.versions )\n",
      "Nextflow operator map with content: .map { it -> [ it[0], [it[1]] ] }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix( LONGREAD_PREPROCESSING.out.versions )\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix( SHORTREAD_COMPLEXITYFILTERING.out.versions )\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SHORTREAD_HOSTREMOVAL.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(LONGREAD_HOSTREMOVAL.out.versions)\n",
      "Nextflow operator mix with content: .mix( ch_longreads_hostremoved )\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator groupTuple with content: .groupTuple()\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator flatten with content: [ meta, reads.flatten() ]\n",
      "Nextflow operator branch with content: .branch {\n",
      "Nextflow operator mix with content: .mix( ch_reads_for_cat_branch.skip )\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator flatten with content: [ meta, [ reads ].flatten() ]\n",
      "Nextflow operator mix with content: .mix( INPUT_CHECK.out.fasta )\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(MERGE_RUNS.out.versions)\n",
      "Nextflow operator mix with content: .mix( ch_longreads_hostremoved, INPUT_CHECK.out.fasta )\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix( PROFILING.out.versions )\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix( VISUALIZATION_KRONA.out.versions )\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix( STANDARDISATION_PROFILES.out.versions )\n",
      "Nextflow operator collectFile with content: ch_versions.unique().collectFile(name: 'collated_versions.yml')\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_workflow_summary.collectFile(name: 'workflow_summary_mqc.yaml'))\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_methods_description.collectFile(name: 'methods_description_mqc.yaml'))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(CUSTOM_DUMPSOFTWAREVERSIONS.out.mqc_yml.collect())\n",
      "Nextflow operator mix with content: ch_multiqc_files = ch_multiqc_files.mix(FALCO.out.txt\n",
      "Nextflow operator map with content: .map { meta, reports -> reports }\n",
      "Nextflow operator flatten with content: .flatten()\n",
      "Nextflow operator filter with content: .filter { path -> path.name.endsWith('_data.txt')}\n",
      "Nextflow operator ifEmpty with content: .ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(FASTQC.out.zip.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix( SHORTREAD_PREPROCESSING.out.mqc.collect{it[1]}.ifEmpty([]) )\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix( LONGREAD_PREPROCESSING.out.mqc.collect{it[1]}.ifEmpty([]) )\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix( SHORTREAD_COMPLEXITYFILTERING.out.mqc.collect{it[1]}.ifEmpty([]) )\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(SHORTREAD_HOSTREMOVAL.out.mqc.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(LONGREAD_HOSTREMOVAL.out.mqc.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix( PROFILING.out.mqc.collect{it[1]}.ifEmpty([]) )\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix( STANDARDISATION_PROFILES.out.mqc.collect{it[1]}.ifEmpty([]) )\n",
      "Nextflow operator collect with content: ch_multiqc_files.collect(),\n",
      "Nextflow operator toList with content: ch_multiqc_config.toList(),\n",
      "Nextflow operator toList with content: ch_multiqc_custom_config.toList(),\n",
      "Nextflow operator toList with content: ch_multiqc_logo.toList()\n",
      "Nextflow operator toList with content: multiqc_report = MULTIQC.out.report.toList()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Error: Unclosed process block in file: ./git_repos\\taxprofiler\\workflows\\taxprofiler.nf\n",
      "Process NFCORE_VARIANTCATALOGUE in filepath ./git_repos\\variantcatalogue\\main.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\variantcatalogue\\main.nf\n",
      "Process INPUT_CHECK in filepath ./git_repos\\variantcatalogue\\subworkflows\\local\\input_check.nf\n",
      "Nextflow operator splitCsv with content: .splitCsv ( header:true, sep:',' )\n",
      "Nextflow operator map with content: .map { create_fastq_channel(it) }\n",
      "Nextflow operator set with content: .set { reads }\n",
      "Error: Unclosed process block in file: ./git_repos\\variantcatalogue\\subworkflows\\local\\input_check.nf\n",
      "Process VARIANTCATALOGUE in filepath ./git_repos\\variantcatalogue\\workflows\\variantcatalogue.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(INPUT_CHECK.out.versions)\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(FASTQC.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(TRIMMOMATIC.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(BWA_INDEX.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(BWA_MEM.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SAMTOOLS_INDEX.out.versions.first())\n",
      "Nextflow operator join with content: ch_bam = BWA_MEM.out.bam.join(SAMTOOLS_INDEX.out.bai)       // channel: [mandatory] [ val(meta), path(bam), path(bai) ]\n",
      "Nextflow operator map with content: ch_bam.map { meta, bam, bai ->\n",
      "Nextflow operator set with content: .set { ch_mosdepth_in }\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(MOSDEPTH.out.versions.first())\n",
      "Nextflow operator join with content: .join(SAMTOOLS_INDEX.out.bai),\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(PICARD_COLLECTWGSMETRICS.out.versions.first())\n",
      "Nextflow operator join with content: Picard_CollectAlignmentSummaryMetrics ( BWA_MEM.out.bam.join(SAMTOOLS_INDEX.out.bai))\n",
      "Nextflow operator join with content: Picard_QualityScoreDistribution       ( BWA_MEM.out.bam.join(SAMTOOLS_INDEX.out.bai))\n",
      "Nextflow operator join with content: ch_bam = BWA_MEM.out.bam.join(SAMTOOLS_INDEX.out.bai)       // channel: [mandatory] [ val(meta), path(bam), path(bai) ]\n",
      "Nextflow operator map with content: ch_bam.map { meta, bam, bai ->\n",
      "Nextflow operator set with content: .set { ch_deepvar_in }\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(DEEPVARIANT.out.versions.first())\n",
      "Nextflow operator collect with content: .collect{it[1]}\n",
      "Nextflow operator toList with content: .toList()\n",
      "Nextflow operator collect with content: .collect()\n",
      "Nextflow operator set with content: .set { ch_file_list }\n",
      "Nextflow operator combine with content: .combine(ch_file_list)\n",
      "Nextflow operator set with content: .set { ch_gvcfs }\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(GLNEXUS.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(TABIX_TABIX.out.versions.first())\n",
      "Nextflow operator join with content: BCFTOOLS_NORM     (GLNEXUS.out.bcf.join(TABIX_TABIX.out.csi), file(params.reference))\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(BCFTOOLS_NORM.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(BCFTOOLS_ANNOTATE.out.versions.first())\n",
      "Nextflow operator join with content: GATK4_PRINTREADS                   (BWA_MEM.out.bam.join(SAMTOOLS_INDEX.out.bai), reference_MT, file(params.reference_MT_index), file(params.reference_MT_dict))\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(GATK4_PRINTREADS.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(GATK4_SAMTOFASTQ.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(GATK4_MARKDUPLICATES_MT.out.versions.first())\n",
      "Nextflow operator join with content: PICARD_COLLECTHSMETRICS_MT         (GATK4_MARKDUPLICATES_MT.out.bam.join(GATK4_MARKDUPLICATES_MT.out.bai), reference_MT, reference_MT_index, file(params.non_control_region_interval_list))\n",
      "Nextflow operator join with content: PICARD_COLLECTHSMETRICS_MT_shifted (GATK4_MARKDUPLICATES_MT_shifted.out.bam.join(GATK4_MARKDUPLICATES_MT_shifted.out.bai), reference_MT_shifted, reference_MT_shifted_index, file(params.control_region_shifted_reference_interval_list))\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(PICARD_COLLECTHSMETRICS_MT.out.versions.first())\n",
      "Nextflow operator join with content: ch_bam_MT = GATK4_MARKDUPLICATES_MT.out.bam.join(GATK4_MARKDUPLICATES_MT.out.bai)\n",
      "Nextflow operator map with content: ch_bam_MT.map { meta, bam, bai ->\n",
      "Nextflow operator set with content: .set { ch_mutect2_in }\n",
      "Nextflow operator join with content: ch_bam_MT_shifted = GATK4_MARKDUPLICATES_MT_shifted.out.bam.join(GATK4_MARKDUPLICATES_MT_shifted.out.bai)\n",
      "Nextflow operator map with content: ch_bam_MT_shifted.map { meta, bam, bai ->\n",
      "Nextflow operator set with content: .set { ch_mutect2_shifted_in }\n",
      "Nextflow operator join with content: MT_Liftover            (GATK4_MUTECT2_shifted.out.vcf.join(GATK4_MUTECT2_shifted.out.tbi), reference_MT, reference_MT_dict, BWA_INDEX_MT.out.index, file(params.ShiftBack_chain))\n",
      "Nextflow operator join with content: GATK4_MERGEVCFS        (MT_Liftover.out.lifted_vcf.join(GATK4_MUTECT2.out.vcf), reference_MT_dict)\n",
      "Nextflow operator join with content: BCFTOOLS_NORM_MT       (GATK4_MERGEVCFS.out.vcf.join(TABIX_TABIX_MT.out.tbi), file(params.reference_MT))\n",
      "Nextflow operator join with content: GATK4_MERGEMUTECTSTATS (GATK4_MUTECT2.out.stats.join(GATK4_MUTECT2_shifted.out.stats))\n",
      "Nextflow operator join with content: .join(TABIX_TABIX_MT2.out.tbi)\n",
      "Nextflow operator map with content: ch_vcf_MT_filtered.map { meta, vcf, tbi ->\n",
      "Nextflow operator set with content: .set { ch_vcf_MT_filtered_in }\n",
      "Nextflow operator join with content: GATK4_VARIANTFILTRATION(GATK4_LEFTALIGNANDTRIMVARIANTS.out.vcf.join(GATK4_LEFTALIGNANDTRIMVARIANTS.out.tbi), file(params.reference_MT), file(params.reference_MT_index), file(params.reference_MT_dict))\n",
      "Nextflow operator join with content: shift_back (PICARD_COLLECTHSMETRICS_MT_shifted.out.tsv.join(PICARD_COLLECTHSMETRICS_MT.out.tsv))\n",
      "Nextflow operator collect with content: MT_Step1_input_tsv(shift_back.out.Sample_MT_Step1_input_tsv.collect())\n",
      "Nextflow operator collect with content: MT_Step2_participant_data(GATK4_VARIANTFILTRATION.out.sample_MT_Step2_participant_data.collect(), GATK4_VARIANTFILTRATION.out.Sample_list.collect())\n",
      "Nextflow operator collect with content: MT_Step3_metadata(MT_Step3_metadata_sample.out.collect())\n",
      "Nextflow operator collectFile with content: ch_versions.unique().collectFile(name: 'collated_versions.yml')\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_workflow_summary.collectFile(name: 'workflow_summary_mqc.yaml'))\n",
      "Nextflow operator collectFile with content: ch_multiqc_files = ch_multiqc_files.mix(ch_methods_description.collectFile(name: 'methods_description_mqc.yaml'))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(CUSTOM_DUMPSOFTWAREVERSIONS.out.mqc_yml.collect())\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(FASTQC.out.zip.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(MOSDEPTH.out.global_txt.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(PICARD_COLLECTWGSMETRICS.out.metrics.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(Picard_CollectAlignmentSummaryMetrics.out.report.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files = ch_multiqc_files.mix(Picard_QualityScoreDistribution.out.report.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files.collect(),\n",
      "Nextflow operator toList with content: ch_multiqc_config.toList(),\n",
      "Nextflow operator toList with content: ch_multiqc_custom_config.toList(),\n",
      "Nextflow operator toList with content: ch_multiqc_logo.toList()\n",
      "Nextflow operator toList with content: multiqc_report = MULTIQC.out.report.toList()\n",
      "Error: Unclosed process block in file: ./git_repos\\variantcatalogue\\workflows\\variantcatalogue.nf\n",
      "Process NFCORE_VIRALINTEGRATION in filepath ./git_repos\\viralintegration\\main.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\viralintegration\\main.nf\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Process INPUT_CHECK in filepath ./git_repos\\viralintegration\\subworkflows\\local\\input_check.nf\n",
      "Nextflow operator splitCsv with content: .splitCsv ( header:true, sep:',' )\n",
      "Nextflow operator map with content: .map { create_fastq_channel(it) }\n",
      "Nextflow operator set with content: .set { reads }\n",
      "Error: Unclosed process block in file: ./git_repos\\viralintegration\\subworkflows\\local\\input_check.nf\n",
      "Process VIRALINTEGRATION in filepath ./git_repos\\viralintegration\\workflows\\viralintegration.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(INPUT_CHECK.out.versions)\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(FASTQC.out.versions.first())\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(STAR_GENOMEGENERATE_HOST.out.versions)\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(STAR_ALIGN_HOST.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(TRIMMOMATIC.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(POLYA_STRIPPER.out.versions.first())\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(CAT_FASTA.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(STAR_GENOMEGENERATE_PLUS.out.versions)\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(STAR_ALIGN_PLUS.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SAMTOOLS_SORT_PLUS.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SAMTOOLS_INDEX_PLUS.out.versions.first())\n",
      "Nextflow operator join with content: .join(SAMTOOLS_INDEX_PLUS.out.bai, by: [0], remainder: true)\n",
      "Nextflow operator join with content: .join(STAR_ALIGN_PLUS.out.junction)\n",
      "Nextflow operator set with content: .set { ch_bam_bai_junction }\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(INSERTION_SITE_CANDIDATES.out.versions.first())\n",
      "Nextflow operator join with content: .join(SAMTOOLS_INDEX_PLUS.out.bai, by: [0], remainder: true)\n",
      "Nextflow operator join with content: .join(ABRIDGED_TSV.out.filtered_abridged)\n",
      "Nextflow operator set with content: .set { ch_bam_bai_filtered }\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(VIRUS_REPORT.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(EXTRACT_CHIMERIC_GENOMIC_TARGETS.out.versions.first())\n",
      "Nextflow operator join with content: .join(EXTRACT_CHIMERIC_GENOMIC_TARGETS.out.fasta_extract)\n",
      "Nextflow operator set with content: .set { ch_unaligned_fastq_fasta }\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(STAR_ALIGN_VALIDATE.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SAMTOOLS_SORT_VALIDATE.out.versions.first())\n",
      "Nextflow operator join with content: SAMTOOLS_SORT_VALIDATE.out.bam.join(\n",
      "Nextflow operator set with content: .set { ch_validate_bam_bai }\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(REMOVE_DUPLICATES.out.versions.first())\n",
      "Nextflow operator join with content: .join(EXTRACT_CHIMERIC_GENOMIC_TARGETS.out.gtf_extract, by: [0])\n",
      "Nextflow operator set with content: .set { ch_validate_bam_bai_gtf }\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(CHIMERIC_CONTIG_EVIDENCE_ANALYZER.out.versions.first())\n",
      "Nextflow operator join with content: .join(CHIMERIC_CONTIG_EVIDENCE_ANALYZER.out.evidence_bai, by: [0])\n",
      "Nextflow operator join with content: .join(ABRIDGED_TSV.out.filtered_abridged, by: [0])\n",
      "Nextflow operator join with content: .join(CHIMERIC_CONTIG_EVIDENCE_ANALYZER.out.evidence_counts, by: [0])\n",
      "Nextflow operator join with content: .join(EXTRACT_CHIMERIC_GENOMIC_TARGETS.out.gtf_extract, by: [0])\n",
      "Nextflow operator join with content: .join(EXTRACT_CHIMERIC_GENOMIC_TARGETS.out.fasta_extract, by: [0])\n",
      "Nextflow operator join with content: .join(VIRUS_REPORT.out.genome_abundance_plot, by: [0])\n",
      "Nextflow operator join with content: .join(VIRUS_REPORT.out.read_counts_image, by: [0])\n",
      "Nextflow operator join with content: .join(VIRUS_REPORT.out.read_counts_log_image, by: [0])\n",
      "Nextflow operator set with content: .set { ch_summary_report }\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SUMMARY_REPORT.out.versions.first())\n",
      "Nextflow operator collectFile with content: ch_versions.unique().collectFile(name: 'collated_versions.yml')\n",
      "Nextflow operator collectFile with content: .mix(ch_workflow_summary.collectFile(name: 'workflow_summary_mqc.yaml'))\n",
      "Nextflow operator collectFile with content: .mix(ch_methods_description.collectFile(name: 'methods_description_mqc.yaml'))\n",
      "Nextflow operator collect with content: .mix(CUSTOM_DUMPSOFTWAREVERSIONS.out.mqc_yml.collect())\n",
      "Nextflow operator collect with content: .mix(FASTQC.out.zip.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: .mix(STAR_ALIGN_HOST.out.log_final.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: .mix(TRIMMOMATIC.out.mqc_log.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: .mix(STAR_ALIGN_PLUS.out.log_final.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: .mix(STAR_ALIGN_VALIDATE.out.log_final.collect{it[1]}.ifEmpty([]))\n",
      "Nextflow operator collect with content: ch_multiqc_files.collect(),\n",
      "Nextflow operator toList with content: ch_multiqc_config.toList(),\n",
      "Nextflow operator toList with content: ch_multiqc_custom_config.toList(),\n",
      "Nextflow operator toList with content: ch_multiqc_logo.toList()\n",
      "Nextflow operator toList with content: multiqc_report = MULTIQC.out.report.toList()\n",
      "Error: Unclosed process block in file: ./git_repos\\viralintegration\\workflows\\viralintegration.nf\n",
      "Process NFCORE_VIRALRECON in filepath ./git_repos\\viralrecon\\main.nf\n",
      "Error: Unclosed process block in file: ./git_repos\\viralrecon\\main.nf\n",
      "Process ASSEMBLY_MINIA in filepath ./git_repos\\viralrecon\\subworkflows\\local\\assembly_minia.nf\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(MINIA.out.versions.first())\n",
      "Nextflow operator filter with content: .filter { meta, contig -> contig.size() > 0 }\n",
      "Nextflow operator set with content: .set { ch_contigs }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ASSEMBLY_QC.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\viralrecon\\subworkflows\\local\\assembly_minia.nf\n",
      "Process ASSEMBLY_QC in filepath ./git_repos\\viralrecon\\subworkflows\\local\\assembly_qc.nf\n",
      "Nextflow operator first with content: ch_versions  = ch_versions.mix(BLAST_BLASTN.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions         = ch_versions.mix(FILTER_BLASTN.out.versions.first())\n",
      "Nextflow operator collect with content: scaffolds.collect{ it[1] },\n",
      "Nextflow operator mix with content: ch_versions      = ch_versions.mix(QUAST.out.versions)\n",
      "Nextflow operator first with content: ch_versions       = ch_versions.mix(ABACAS.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions           = ch_versions.mix(PLASMIDID.out.versions.first())\n",
      "Error: Unclosed process block in file: ./git_repos\\viralrecon\\subworkflows\\local\\assembly_qc.nf\n",
      "Process ASSEMBLY_SPADES in filepath ./git_repos\\viralrecon\\subworkflows\\local\\assembly_spades.nf\n",
      "Nextflow operator filter with content: .filter { meta, illumina, pacbio, nanopore -> !meta.single_end }\n",
      "Nextflow operator set with content: .set { ch_reads }\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SPADES.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(GUNZIP_SCAFFOLDS.out.versions.first())\n",
      "Nextflow operator filter with content: .filter { meta, scaffold -> scaffold.size() > 0 }\n",
      "Nextflow operator set with content: .set { ch_scaffolds }\n",
      "Nextflow operator filter with content: .filter { meta, gfa -> gfa.size() > 0 }\n",
      "Nextflow operator set with content: .set { ch_gfa }\n",
      "Nextflow operator first with content: ch_versions    = ch_versions.mix(BANDAGE_IMAGE.out.versions.first())\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ASSEMBLY_QC.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\viralrecon\\subworkflows\\local\\assembly_spades.nf\n",
      "Process ASSEMBLY_UNICYCLER in filepath ./git_repos\\viralrecon\\subworkflows\\local\\assembly_unicycler.nf\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(UNICYCLER.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(GUNZIP_SCAFFOLDS.out.versions.first())\n",
      "Nextflow operator filter with content: .filter { meta, scaffold -> scaffold.size() > 0 }\n",
      "Nextflow operator set with content: .set { ch_scaffolds }\n",
      "Nextflow operator filter with content: .filter { meta, gfa -> gfa.size() > 0 }\n",
      "Nextflow operator set with content: .set { ch_gfa }\n",
      "Nextflow operator first with content: ch_versions    = ch_versions.mix(BANDAGE_IMAGE.out.versions.first())\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(ASSEMBLY_QC.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\viralrecon\\subworkflows\\local\\assembly_unicycler.nf\n",
      "Process BAM_TRIM_PRIMERS_IVAR in filepath ./git_repos\\viralrecon\\subworkflows\\local\\bam_trim_primers_ivar.nf\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(IVAR_TRIM.out.versions.first())\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BAM_SORT_STATS_SAMTOOLS.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\viralrecon\\subworkflows\\local\\bam_trim_primers_ivar.nf\n",
      "Process CONSENSUS_BCFTOOLS in filepath ./git_repos\\viralrecon\\subworkflows\\local\\consensus_bcftools.nf\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(BCFTOOLS_FILTER.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(TABIX_TABIX.out.versions.first())\n",
      "Nextflow operator join with content: bam.join(BCFTOOLS_FILTER.out.vcf, by: [0]),\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(MAKE_BED_MASK.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(BEDTOOLS_MERGE.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(BEDTOOLS_MASKFASTA.out.versions.first())\n",
      "Nextflow operator join with content: BCFTOOLS_FILTER.out.vcf.join(TABIX_TABIX.out.tbi, by: [0]).join(BEDTOOLS_MASKFASTA.out.fasta, by: [0])\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(BCFTOOLS_CONSENSUS.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(RENAME_FASTA_HEADER.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(CONSENSUS_QC.out.versions.first())\n",
      "Error: Unclosed process block in file: ./git_repos\\viralrecon\\subworkflows\\local\\consensus_bcftools.nf\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Process CONSENSUS_IVAR in filepath ./git_repos\\viralrecon\\subworkflows\\local\\consensus_ivar.nf\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(IVAR_CONSENSUS.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(CONSENSUS_QC.out.versions.first())\n",
      "Error: Unclosed process block in file: ./git_repos\\viralrecon\\subworkflows\\local\\consensus_ivar.nf\n",
      "Process CONSENSUS_QC in filepath ./git_repos\\viralrecon\\subworkflows\\local\\consensus_qc.nf\n",
      "Nextflow operator collect with content: consensus.collect{ it[1] },\n",
      "Nextflow operator mix with content: ch_versions      = ch_versions.mix(QUAST.out.versions)\n",
      "Nextflow operator first with content: ch_versions        = ch_versions.mix(PANGOLIN.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions         = ch_versions.mix(NEXTCLADE_RUN.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions  = ch_versions.mix(PLOT_BASE_DENSITY.out.versions.first())\n",
      "Error: Unclosed process block in file: ./git_repos\\viralrecon\\subworkflows\\local\\consensus_qc.nf\n",
      "Process FASTQ_TRIM_FASTP_FASTQC in filepath ./git_repos\\viralrecon\\subworkflows\\local\\fastq_trim_fastp_fastqc.nf\n",
      "Nextflow operator first with content: ch_versions     = ch_versions.mix(FASTQC_RAW.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions       = ch_versions.mix(FASTP.out.versions.first())\n",
      "Nextflow operator join with content: .join(trim_json)\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator set with content: .set { trim_reads }\n",
      "Nextflow operator first with content: ch_versions      = ch_versions.mix(FASTQC_TRIM.out.versions.first())\n",
      "Nextflow operator ifEmpty with content: versions = ch_versions.ifEmpty(null) // channel: [ versions.yml ]\n",
      "Error: Unclosed process block in file: ./git_repos\\viralrecon\\subworkflows\\local\\fastq_trim_fastp_fastqc.nf\n",
      "Process FILTER_BAM_SAMTOOLS in filepath ./git_repos\\viralrecon\\subworkflows\\local\\filter_bam_samtools.nf\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SAMTOOLS_VIEW.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SAMTOOLS_INDEX.out.versions.first())\n",
      "Nextflow operator join with content: SAMTOOLS_VIEW.out.bam.join(SAMTOOLS_INDEX.out.bai, by: [0]),\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BAM_STATS_SAMTOOLS.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\viralrecon\\subworkflows\\local\\filter_bam_samtools.nf\n",
      "Process INPUT_CHECK in filepath ./git_repos\\viralrecon\\subworkflows\\local\\input_check.nf\n",
      "Nextflow operator splitCsv with content: .splitCsv ( header:true, sep:',' )\n",
      "Nextflow operator map with content: .map { create_fastq_channel(it) }\n",
      "Nextflow operator set with content: .set { sample_info }\n",
      "Nextflow operator splitCsv with content: .splitCsv ( header:true, sep:',' )\n",
      "Nextflow operator map with content: .map { row -> [ row.barcode, row.sample ] }\n",
      "Nextflow operator set with content: .set { sample_info }\n",
      "Error: Unclosed process block in file: ./git_repos\\viralrecon\\subworkflows\\local\\input_check.nf\n",
      "Process PREPARE_GENOME in filepath ./git_repos\\viralrecon\\subworkflows\\local\\prepare_genome_illumina.nf\n",
      "Nextflow operator map with content: ch_fasta    = GUNZIP_FASTA.out.gunzip.map { it[1] }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GUNZIP_FASTA.out.versions)\n",
      "Nextflow operator map with content: ch_gff      = GUNZIP_GFF.out.gunzip.map { it[1] }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GUNZIP_GFF.out.versions)\n",
      "Nextflow operator map with content: ch_fasta.map { [ [:], it ] }\n",
      "Nextflow operator map with content: ch_fai         = CUSTOM_GETCHROMSIZES.out.fai.map { it[1] }\n",
      "Nextflow operator map with content: ch_chrom_sizes = CUSTOM_GETCHROMSIZES.out.sizes.map { it[1] }\n",
      "Nextflow operator mix with content: ch_versions    = ch_versions.mix(CUSTOM_GETCHROMSIZES.out.versions)\n",
      "Nextflow operator map with content: ch_kraken2_db = UNTAR_KRAKEN2_DB.out.untar.map { it[1] }\n",
      "Nextflow operator mix with content: ch_versions   = ch_versions.mix(UNTAR_KRAKEN2_DB.out.versions)\n",
      "Nextflow operator first with content: ch_kraken2_db = KRAKEN2_BUILD.out.db.first()\n",
      "Nextflow operator mix with content: ch_versions   = ch_versions.mix(KRAKEN2_BUILD.out.versions)\n",
      "Nextflow operator map with content: ch_primer_bed = GUNZIP_PRIMER_BED.out.gunzip.map { it[1] }\n",
      "Nextflow operator mix with content: ch_versions   = ch_versions.mix(GUNZIP_PRIMER_BED.out.versions)\n",
      "Nextflow operator mix with content: ch_versions             = ch_versions.mix(COLLAPSE_PRIMERS.out.versions)\n",
      "Nextflow operator map with content: ch_primer_fasta = GUNZIP_PRIMER_FASTA.out.gunzip.map { it[1] }\n",
      "Nextflow operator mix with content: ch_versions     = ch_versions.mix(GUNZIP_PRIMER_FASTA.out.versions)\n",
      "Nextflow operator mix with content: ch_versions     = ch_versions.mix(BEDTOOLS_GETFASTA.out.versions)\n",
      "Nextflow operator map with content: ch_bowtie2_index = UNTAR_BOWTIE2_INDEX.out.untar.map { it[1] }\n",
      "Nextflow operator mix with content: ch_versions      = ch_versions.mix(UNTAR_BOWTIE2_INDEX.out.versions)\n",
      "Nextflow operator map with content: ch_fasta.map { [ [:], it ] }\n",
      "Nextflow operator mix with content: ch_versions      = ch_versions.mix(BOWTIE2_BUILD.out.versions)\n",
      "Nextflow operator map with content: ch_nextclade_db = UNTAR_NEXTCLADE_DB.out.untar.map { it[1] }\n",
      "Nextflow operator mix with content: ch_versions     = ch_versions.mix(UNTAR_NEXTCLADE_DB.out.versions)\n",
      "Nextflow operator mix with content: ch_versions     = ch_versions.mix(NEXTCLADE_DATASETGET.out.versions)\n",
      "Nextflow operator map with content: ch_blast_db = UNTAR_BLAST_DB.out.untar.map { it[1] }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(UNTAR_BLAST_DB.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BLAST_MAKEBLASTDB.out.versions)\n",
      "Nextflow operator mix with content: ch_versions      = ch_versions.mix(SNPEFF_BUILD.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\viralrecon\\subworkflows\\local\\prepare_genome_illumina.nf\n",
      "Process PREPARE_GENOME in filepath ./git_repos\\viralrecon\\subworkflows\\local\\prepare_genome_nanopore.nf\n",
      "Nextflow operator map with content: ch_fasta    = GUNZIP_FASTA.out.gunzip.map { it[1] }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GUNZIP_FASTA.out.versions)\n",
      "Nextflow operator map with content: ch_gff      = GUNZIP_GFF.out.gunzip.map { it[1] }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(GUNZIP_GFF.out.versions)\n",
      "Nextflow operator map with content: ch_fasta.map { [ [:], it ] }\n",
      "Nextflow operator map with content: ch_fai         = CUSTOM_GETCHROMSIZES.out.fai.map { it[1] }\n",
      "Nextflow operator map with content: ch_chrom_sizes = CUSTOM_GETCHROMSIZES.out.sizes.map { it[1] }\n",
      "Nextflow operator mix with content: ch_versions    = ch_versions.mix(CUSTOM_GETCHROMSIZES.out.versions)\n",
      "Nextflow operator map with content: ch_primer_bed = GUNZIP_PRIMER_BED.out.gunzip.map { it[1] }\n",
      "Nextflow operator mix with content: ch_versions   = ch_versions.mix(GUNZIP_PRIMER_BED.out.versions)\n",
      "Nextflow operator mix with content: ch_versions             = ch_versions.mix(COLLAPSE_PRIMERS.out.versions)\n",
      "Nextflow operator map with content: ch_nextclade_db = UNTAR.out.untar.map { it[1] }\n",
      "Nextflow operator mix with content: ch_versions     = ch_versions.mix(UNTAR.out.versions)\n",
      "Nextflow operator mix with content: ch_versions     = ch_versions.mix(NEXTCLADE_DATASETGET.out.versions)\n",
      "Nextflow operator mix with content: ch_versions      = ch_versions.mix(SNPEFF_BUILD.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\viralrecon\\subworkflows\\local\\prepare_genome_nanopore.nf\n",
      "Process SNPEFF_SNPSIFT in filepath ./git_repos\\viralrecon\\subworkflows\\local\\snpeff_snpsift.nf\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SNPEFF_ANN.out.versions.first())\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(VCF_BGZIP_TABIX_STATS.out.versions)\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SNPSIFT_EXTRACTFIELDS.out.versions.first())\n",
      "Error: Unclosed process block in file: ./git_repos\\viralrecon\\subworkflows\\local\\snpeff_snpsift.nf\n",
      "Process VARIANTS_BCFTOOLS in filepath ./git_repos\\viralrecon\\subworkflows\\local\\variants_bcftools.nf\n",
      "Nextflow operator map with content: bam.map{ meta, bam_file -> [ meta, bam_file, [] ] },\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(BCFTOOLS_MPILEUP.out.versions.first())\n",
      "Nextflow operator join with content: .join(BCFTOOLS_MPILEUP.out.tbi)\n",
      "Nextflow operator join with content: .join(BCFTOOLS_MPILEUP.out.stats)\n",
      "Nextflow operator filter with content: .filter { meta, vcf, tbi, stats -> WorkflowCommons.getNumVariantsFromBCFToolsStats(stats) > 0 }\n",
      "Nextflow operator set with content: .set { ch_vcf_tbi_stats }\n",
      "Nextflow operator map with content: .map { meta, vcf, tbi, stats -> [ meta, vcf ] }\n",
      "Nextflow operator set with content: .set { ch_vcf }\n",
      "Nextflow operator map with content: .map { meta, vcf, tbi, stats -> [ meta, tbi ] }\n",
      "Nextflow operator set with content: .set { ch_tbi }\n",
      "Nextflow operator map with content: .map { meta, vcf, tbi, stats -> [ meta, stats ] }\n",
      "Nextflow operator set with content: .set { ch_stats }\n",
      "Nextflow operator join with content: ch_vcf.join(ch_tbi, by: [0]),\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(BCFTOOLS_NORM.out.versions.first())\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(VCF_TABIX_STATS.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(VARIANTS_QC.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\viralrecon\\subworkflows\\local\\variants_bcftools.nf\n",
      "Process VARIANTS_IVAR in filepath ./git_repos\\viralrecon\\subworkflows\\local\\variants_ivar.nf\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(IVAR_VARIANTS.out.versions.first())\n",
      "Nextflow operator filter with content: .filter { meta, tsv -> WorkflowCommons.getNumLinesInFile(tsv) > 1 }\n",
      "Nextflow operator set with content: .set { ch_ivar_tsv }\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(IVAR_VARIANTS_TO_VCF.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(BCFTOOLS_SORT.out.versions.first())\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(VCF_TABIX_STATS.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(VARIANTS_QC.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\viralrecon\\subworkflows\\local\\variants_ivar.nf\n",
      "Process VARIANTS_LONG_TABLE in filepath ./git_repos\\viralrecon\\subworkflows\\local\\variants_long_table.nf\n",
      "Nextflow operator join with content: vcf.join(tbi, by: [0]),\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(BCFTOOLS_QUERY.out.versions.first())\n",
      "Nextflow operator collect with content: BCFTOOLS_QUERY.out.txt.collect{it[1]},\n",
      "Nextflow operator collect with content: snpsift.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: pangolin.collect{it[1]}.ifEmpty([])\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(MAKE_VARIANTS_LONG_TABLE.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\viralrecon\\subworkflows\\local\\variants_long_table.nf\n",
      "Process VARIANTS_QC in filepath ./git_repos\\viralrecon\\subworkflows\\local\\variants_qc.nf\n",
      "Nextflow operator mix with content: ch_versions     = ch_versions.mix(SNPEFF_SNPSIFT.out.versions)\n",
      "Nextflow operator join with content: .join(vcf, by: [0])\n",
      "Nextflow operator join with content: .join(stats, by: [0])\n",
      "Nextflow operator map with content: .map { meta, bam, vcf, stats ->\n",
      "Nextflow operator set with content: .set { ch_asciigenome }\n",
      "Nextflow operator first with content: ch_versions        = ch_versions.mix(ASCIIGENOME.out.versions.first())\n",
      "Error: Unclosed process block in file: ./git_repos\\viralrecon\\subworkflows\\local\\variants_qc.nf\n",
      "Process VCF_BGZIP_TABIX_STATS in filepath ./git_repos\\viralrecon\\subworkflows\\local\\vcf_bgzip_tabix_stats.nf\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(TABIX_BGZIP.out.versions.first())\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(VCF_TABIX_STATS.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\viralrecon\\subworkflows\\local\\vcf_bgzip_tabix_stats.nf\n",
      "Process VCF_TABIX_STATS in filepath ./git_repos\\viralrecon\\subworkflows\\local\\vcf_tabix_stats.nf\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(TABIX_TABIX.out.versions.first())\n",
      "Nextflow operator join with content: vcf.join(TABIX_TABIX.out.tbi, by: [0]),\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(BCFTOOLS_STATS.out.versions.first())\n",
      "Error: Unclosed process block in file: ./git_repos\\viralrecon\\subworkflows\\local\\vcf_tabix_stats.nf\n",
      "Process BAM_MARKDUPLICATES_PICARD in filepath ./git_repos\\viralrecon\\subworkflows\\nf-core\\bam_markduplicates_picard\\main.nf\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(PICARD_MARKDUPLICATES.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SAMTOOLS_INDEX.out.versions.first())\n",
      "Nextflow operator join with content: .join(SAMTOOLS_INDEX.out.bai, by: [0], remainder: true)\n",
      "Nextflow operator join with content: .join(SAMTOOLS_INDEX.out.csi, by: [0], remainder: true)\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BAM_STATS_SAMTOOLS.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\viralrecon\\subworkflows\\nf-core\\bam_markduplicates_picard\\main.nf\n",
      "Process BAM_SORT_STATS_SAMTOOLS in filepath ./git_repos\\viralrecon\\subworkflows\\nf-core\\bam_sort_stats_samtools\\main.nf\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SAMTOOLS_SORT.out.versions.first())\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(SAMTOOLS_INDEX.out.versions.first())\n",
      "Nextflow operator join with content: .join(SAMTOOLS_INDEX.out.bai, by: [0], remainder: true)\n",
      "Nextflow operator join with content: .join(SAMTOOLS_INDEX.out.csi, by: [0], remainder: true)\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator set with content: .set { ch_bam_bai }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BAM_STATS_SAMTOOLS.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\viralrecon\\subworkflows\\nf-core\\bam_sort_stats_samtools\\main.nf\n",
      "Process BAM_STATS_SAMTOOLS in filepath ./git_repos\\viralrecon\\subworkflows\\nf-core\\bam_stats_samtools\\main.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_STATS.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_FLAGSTAT.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(SAMTOOLS_IDXSTATS.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\viralrecon\\subworkflows\\nf-core\\bam_stats_samtools\\main.nf\n",
      "Process FASTQ_ALIGN_BOWTIE2 in filepath ./git_repos\\viralrecon\\subworkflows\\nf-core\\fastq_align_bowtie2\\main.nf\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(BOWTIE2_ALIGN.out.versions.first())\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(BAM_SORT_STATS_SAMTOOLS.out.versions)\n",
      "Error: Unclosed process block in file: ./git_repos\\viralrecon\\subworkflows\\nf-core\\fastq_align_bowtie2\\main.nf\n",
      "Process ILLUMINA in filepath ./git_repos\\viralrecon\\workflows\\illumina.nf\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(PREPARE_GENOME.out.versions)\n",
      "Nextflow operator map with content: .map { WorkflowIllumina.isMultiFasta(it, log) }\n",
      "Nextflow operator map with content: .map { WorkflowCommons.checkPrimerSuffixes(it, params.primer_left_suffix, params.primer_right_suffix, log) }\n",
      "Nextflow operator map with content: .map { [ WorkflowCommons.getColFromFile(it, col=0, uniqify=true, sep='\\t') ] }\n",
      "Nextflow operator set with content: .set { ch_bed_contigs }\n",
      "Nextflow operator map with content: .map { [ WorkflowCommons.getColFromFile(it, col=0, uniqify=true, sep='\\t') ] }\n",
      "Nextflow operator concat with content: .concat(ch_bed_contigs)\n",
      "Nextflow operator collect with content: .collect()\n",
      "Nextflow operator map with content: .map { fai, bed -> WorkflowCommons.checkContigsInBED(fai, bed, log) }\n",
      "Nextflow operator map with content: .map { WorkflowIllumina.checkIfSwiftProtocol(it, 'covid19genome', log) }\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator join with content: meta.id = meta.id.split('_')[0..-2].join('_')\n",
      "Nextflow operator groupTuple with content: .groupTuple(by: [0])\n",
      "Nextflow operator branch with content: .branch {\n",
      "Nextflow operator flatten with content: return [ meta, fastq.flatten() ]\n",
      "Nextflow operator flatten with content: return [ meta, fastq.flatten() ]\n",
      "Nextflow operator set with content: .set { ch_fastq }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(INPUT_CHECK.out.versions)\n",
      "Nextflow operator mix with content: .mix(ch_fastq.single)\n",
      "Nextflow operator set with content: .set { ch_cat_fastq }\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(CAT_FASTQ.out.versions.first().ifEmpty(null))\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(FASTQ_TRIM_FASTP_FASTQC.out.versions)\n",
      "Nextflow operator join with content: .join(FASTQ_TRIM_FASTP_FASTQC.out.trim_json)\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator set with content: .set { ch_pass_fail_reads }\n",
      "Nextflow operator map with content: .map { meta, reads, json, pass -> if (pass) [ meta, reads ] }\n",
      "Nextflow operator set with content: .set { ch_variants_fastq }\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator collect with content: .collect()\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator set with content: .set { ch_fail_reads_multiqc }\n",
      "Nextflow operator first with content: ch_versions        = ch_versions.mix(KRAKEN2_KRAKEN2.out.versions.first().ifEmpty(null))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Nextflow operator mix with content: ch_versions                 = ch_versions.mix(FASTQ_ALIGN_BOWTIE2.out.versions)\n",
      "Nextflow operator map with content: .map { meta, flagstat -> [ meta ] + WorkflowIllumina.getFlagstatMappedReads(flagstat, params) }\n",
      "Nextflow operator set with content: .set { ch_mapped_reads }\n",
      "Nextflow operator join with content: .join(ch_mapped_reads, by: [0])\n",
      "Nextflow operator map with content: .map { meta, ofile, mapped, pass -> if (pass) [ meta, ofile ] }\n",
      "Nextflow operator set with content: .set { ch_bam }\n",
      "Nextflow operator join with content: .join(ch_mapped_reads, by: [0])\n",
      "Nextflow operator map with content: .map { meta, ofile, mapped, pass -> if (pass) [ meta, ofile ] }\n",
      "Nextflow operator set with content: .set { ch_bai }\n",
      "Nextflow operator branch with content: .branch { meta, mapped, pass ->\n",
      "Nextflow operator set with content: .set { ch_pass_fail_mapped }\n",
      "Nextflow operator collect with content: .collect()\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator set with content: .set { ch_fail_mapping_multiqc }\n",
      "Nextflow operator join with content: ch_bam.join(ch_bai, by: [0]),\n",
      "Nextflow operator mix with content: ch_versions                   = ch_versions.mix(BAM_TRIM_PRIMERS_IVAR.out.versions)\n",
      "Nextflow operator mix with content: ch_versions                        = ch_versions.mix(BAM_MARKDUPLICATES_PICARD.out.versions)\n",
      "Nextflow operator join with content: ch_bam.join(ch_bai, by: [0]),\n",
      "Nextflow operator map with content: PREPARE_GENOME.out.fasta.map { [ [:], it ] },\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(PICARD_COLLECTMULTIPLEMETRICS.out.versions.first().ifEmpty(null))\n",
      "Nextflow operator join with content: ch_bam.join(ch_bai, by: [0]),\n",
      "Nextflow operator first with content: ch_versions         = ch_versions.mix(MOSDEPTH_GENOME.out.versions.first().ifEmpty(null))\n",
      "Nextflow operator collect with content: MOSDEPTH_GENOME.out.regions_bed.collect { it[1] }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(PLOT_MOSDEPTH_REGIONS_GENOME.out.versions)\n",
      "Nextflow operator join with content: ch_bam.join(ch_bai, by: [0]),\n",
      "Nextflow operator map with content: PREPARE_GENOME.out.primer_collapsed_bed.map { [ [:], it ] },\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(MOSDEPTH_AMPLICON.out.versions.first().ifEmpty(null))\n",
      "Nextflow operator collect with content: MOSDEPTH_AMPLICON.out.regions_bed.collect { it[1] }\n",
      "Nextflow operator mix with content: ch_versions                 = ch_versions.mix(PLOT_MOSDEPTH_REGIONS_AMPLICON.out.versions)\n",
      "Nextflow operator mix with content: ch_versions               = ch_versions.mix(VARIANTS_IVAR.out.versions)\n",
      "Nextflow operator mix with content: ch_versions               = ch_versions.mix(VARIANTS_BCFTOOLS.out.versions)\n",
      "Nextflow operator mix with content: ch_versions         = ch_versions.mix(CONSENSUS_IVAR.out.versions)\n",
      "Nextflow operator mix with content: ch_versions         = ch_versions.mix(CONSENSUS_BCFTOOLS.out.versions)\n",
      "Nextflow operator map with content: .map { meta, csv ->\n",
      "Nextflow operator collect with content: .collect()\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator set with content: .set { ch_nextclade_multiqc }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(VARIANTS_LONG_TABLE.out.versions)\n",
      "Nextflow operator first with content: ch_versions         = ch_versions.mix(CUTADAPT.out.versions.first().ifEmpty(null))\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(FASTQC.out.versions.first().ifEmpty(null))\n",
      "Nextflow operator map with content: ch_assembly_fastq.map { meta, fastq -> [ meta, fastq, [], [] ] },\n",
      "Nextflow operator mix with content: ch_versions             = ch_versions.mix(ASSEMBLY_SPADES.out.versions)\n",
      "Nextflow operator map with content: ch_assembly_fastq.map { meta, fastq -> [ meta, fastq, [] ] },\n",
      "Nextflow operator mix with content: ch_versions                = ch_versions.mix(ASSEMBLY_UNICYCLER.out.versions)\n",
      "Nextflow operator mix with content: ch_versions            = ch_versions.mix(ASSEMBLY_MINIA.out.versions)\n",
      "Nextflow operator collectFile with content: ch_versions.unique().collectFile(name: 'collated_versions.yml')\n",
      "Nextflow operator collect with content: CUSTOM_DUMPSOFTWAREVERSIONS.out.mqc_yml.collect(),\n",
      "Nextflow operator collectFile with content: ch_workflow_summary.collectFile(name: 'workflow_summary_mqc.yaml'),\n",
      "Nextflow operator collectFile with content: ch_fail_reads_multiqc.collectFile(name: 'fail_mapped_reads_mqc.tsv').ifEmpty([]),\n",
      "Nextflow operator collectFile with content: ch_fail_mapping_multiqc.collectFile(name: 'fail_mapped_samples_mqc.tsv').ifEmpty([]),\n",
      "Nextflow operator ifEmpty with content: ch_amplicon_heatmap_multiqc.ifEmpty([]),\n",
      "Nextflow operator collect with content: FASTQ_TRIM_FASTP_FASTQC.out.fastqc_raw_zip.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: FASTQ_TRIM_FASTP_FASTQC.out.trim_json.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_kraken2_multiqc.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_bowtie2_flagstat_multiqc.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_bowtie2_multiqc.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_ivar_trim_flagstat_multiqc.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_markduplicates_flagstat_multiqc.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_mosdepth_multiqc.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_ivar_counts_multiqc.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_bcftools_stats_multiqc.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_snpeff_multiqc.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_quast_multiqc.collect().ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_pangolin_multiqc.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collectFile with content: ch_nextclade_multiqc.collectFile(name: 'nextclade_clade_mqc.tsv').ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_cutadapt_multiqc.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_spades_quast_multiqc.collect().ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_unicycler_quast_multiqc.collect().ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_minia_quast_multiqc.collect().ifEmpty([])\n",
      "Nextflow operator toList with content: multiqc_report = MULTIQC.out.report.toList()\n",
      "Error: Unclosed process block in file: ./git_repos\\viralrecon\\workflows\\illumina.nf\n",
      "Process NANOPORE in filepath ./git_repos\\viralrecon\\workflows\\nanopore.nf\n",
      "Nextflow operator map with content: Channel.of(ch_sequencing_summary).map { [ [:], it ] }\n",
      "Nextflow operator mix with content: ch_versions       = ch_versions.mix(PYCOQC.out.versions)\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(PREPARE_GENOME.out.versions)\n",
      "Nextflow operator map with content: .map { WorkflowCommons.checkPrimerSuffixes(it, params.primer_left_suffix, params.primer_right_suffix, log) }\n",
      "Nextflow operator map with content: .map { [ WorkflowCommons.getColFromFile(it, col=0, uniqify=true, sep='\\t') ] }\n",
      "Nextflow operator set with content: .set { ch_bed_contigs }\n",
      "Nextflow operator map with content: .map { [ WorkflowCommons.getColFromFile(it, col=0, uniqify=true, sep='\\t') ] }\n",
      "Nextflow operator concat with content: .concat(ch_bed_contigs)\n",
      "Nextflow operator collect with content: .collect()\n",
      "Nextflow operator map with content: .map { fai, bed -> WorkflowCommons.checkContigsInBED(fai, bed, log) }\n",
      "Nextflow operator filter with content: .filter( ~/.*barcode[0-9]{1,4}$/ )\n",
      "Nextflow operator map with content: .map { dir ->\n",
      "Nextflow operator countFastq with content: count += x.countFastq()\n",
      "Nextflow operator set with content: .set { ch_fastq_dirs }\n",
      "Nextflow operator join with content: .join(ch_fastq_dirs, remainder: true)\n",
      "Nextflow operator set with content: .set { ch_fastq_dirs }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(INPUT_CHECK.out.versions)\n",
      "Nextflow operator filter with content: .filter { it[1] == null }\n",
      "Nextflow operator filter with content: .filter { it[-1] >= params.min_barcode_reads }\n",
      "Nextflow operator map with content: .map { it -> [ \"${it[0]}\\t${it[-1]}\" ] }\n",
      "Nextflow operator collect with content: .collect()\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator set with content: .set { ch_custom_no_sample_name_multiqc }\n",
      "Nextflow operator filter with content: .filter { it[-1] == null }\n",
      "Nextflow operator map with content: .map { it -> [ \"${it[1]}\\t${it[0]}\" ] }\n",
      "Nextflow operator collect with content: .collect()\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator set with content: .set { ch_custom_no_barcodes_multiqc }\n",
      "Nextflow operator filter with content: .filter { (it[1] != null)  }\n",
      "Nextflow operator filter with content: .filter { (it[-1] != null) }\n",
      "Nextflow operator set with content: .set { ch_fastq_dirs }\n",
      "Nextflow operator map with content: .map { barcode, dir, count -> [ barcode, barcode, dir, count ] }\n",
      "Nextflow operator set with content: .set { ch_fastq_dirs }\n",
      "Nextflow operator map with content: .map { it -> [ 'SAMPLE_1', 'single_barcode', it, 10000000 ] }\n",
      "Nextflow operator set with content: .set{ ch_fastq_dirs }\n",
      "Nextflow operator branch with content: .branch { barcode, sample, dir, count  ->\n",
      "Nextflow operator set with content: .set { ch_pass_fail_barcode_count }\n",
      "Nextflow operator collect with content: .collect()\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator set with content: .set { ch_custom_fail_barcodes_count_multiqc }\n",
      "Nextflow operator filter with content: .filter { it[-1] > params.min_barcode_reads }\n",
      "Nextflow operator map with content: .map { barcode, sample, dir, count -> [ [ id: sample, barcode:barcode ], dir ] }\n",
      "Nextflow operator set with content: .set { ch_fastq_dirs }\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(ARTIC_GUPPYPLEX.out.versions.first().ifEmpty(null))\n",
      "Nextflow operator branch with content: .branch { meta, fastq  ->\n",
      "Nextflow operator countFastq with content: def count = fastq.countFastq()\n",
      "Nextflow operator set with content: .set { ch_pass_fail_guppyplex_count }\n",
      "Nextflow operator collect with content: .collect()\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator set with content: .set { ch_custom_fail_guppyplex_count_multiqc }\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(NANOPLOT.out.versions.first().ifEmpty(null))\n",
      "Nextflow operator countFastq with content: ARTIC_GUPPYPLEX.out.fastq.filter { it[-1].countFastq() > params.min_guppyplex_reads },\n",
      "Nextflow operator collect with content: PREPARE_GENOME.out.fasta.collect(),\n",
      "Nextflow operator collect with content: PREPARE_GENOME.out.primer_bed.collect(),\n",
      "Nextflow operator collect with content: ch_medaka_model.collect().ifEmpty([]),\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(ARTIC_MINION.out.versions.first().ifEmpty(null))\n",
      "Nextflow operator join with content: ARTIC_MINION.out.vcf.join(ARTIC_MINION.out.tbi, by: [0]),\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(VCFLIB_VCFUNIQ.out.versions.first().ifEmpty(null))\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(TABIX_TABIX.out.versions.first().ifEmpty(null))\n",
      "Nextflow operator join with content: VCFLIB_VCFUNIQ.out.vcf.join(TABIX_TABIX.out.tbi, by: [0]),\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(BCFTOOLS_STATS.out.versions.first().ifEmpty(null))\n",
      "Nextflow operator join with content: ARTIC_MINION.out.bam.join(ARTIC_MINION.out.bai, by: [0]),\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(FILTER_BAM_SAMTOOLS.out.versions)\n",
      "Nextflow operator join with content: ARTIC_MINION.out.bam_primertrimmed.join(ARTIC_MINION.out.bai_primertrimmed, by: [0]),\n",
      "Nextflow operator first with content: ch_versions          = ch_versions.mix(MOSDEPTH_GENOME.out.versions.first().ifEmpty(null))\n",
      "Nextflow operator collect with content: MOSDEPTH_GENOME.out.regions_bed.collect { it[1] }\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(PLOT_MOSDEPTH_REGIONS_GENOME.out.versions)\n",
      "Nextflow operator join with content: ARTIC_MINION.out.bam_primertrimmed.join(ARTIC_MINION.out.bai_primertrimmed, by: [0]),\n",
      "Nextflow operator collect with content: PREPARE_GENOME.out.primer_collapsed_bed.map { [ [:], it ] }.collect(),\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(MOSDEPTH_AMPLICON.out.versions.first().ifEmpty(null))\n",
      "Nextflow operator collect with content: MOSDEPTH_AMPLICON.out.regions_bed.collect { it[1] }\n",
      "Nextflow operator mix with content: ch_versions                 = ch_versions.mix(PLOT_MOSDEPTH_REGIONS_AMPLICON.out.versions)\n",
      "Nextflow operator first with content: ch_versions         = ch_versions.mix(PANGOLIN.out.versions.first().ifEmpty(null))\n",
      "Nextflow operator collect with content: PREPARE_GENOME.out.nextclade_db.collect()\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(NEXTCLADE_RUN.out.versions.first().ifEmpty(null))\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator collect with content: .collect()\n",
      "Nextflow operator map with content: .map {\n",
      "Nextflow operator set with content: .set { ch_nextclade_multiqc }\n",
      "Nextflow operator collect with content: ARTIC_MINION.out.fasta.collect{ it[1] },\n",
      "Nextflow operator collect with content: PREPARE_GENOME.out.fasta.collect(),\n",
      "Nextflow operator mix with content: ch_versions      = ch_versions.mix(QUAST.out.versions)\n",
      "Nextflow operator collect with content: PREPARE_GENOME.out.snpeff_db.collect(),\n",
      "Nextflow operator collect with content: PREPARE_GENOME.out.snpeff_config.collect(),\n",
      "Nextflow operator collect with content: PREPARE_GENOME.out.fasta.collect()\n",
      "Nextflow operator mix with content: ch_versions       = ch_versions.mix(SNPEFF_SNPSIFT.out.versions)\n",
      "Nextflow operator join with content: .join(VCFLIB_VCFUNIQ.out.vcf, by: [0])\n",
      "Nextflow operator join with content: .join(BCFTOOLS_STATS.out.stats, by: [0])\n",
      "Nextflow operator map with content: .map { meta, bam, vcf, stats ->\n",
      "Nextflow operator set with content: .set { ch_asciigenome }\n",
      "Nextflow operator collect with content: PREPARE_GENOME.out.fasta.collect(),\n",
      "Nextflow operator collect with content: PREPARE_GENOME.out.chrom_sizes.collect(),\n",
      "Nextflow operator collect with content: PREPARE_GENOME.out.primer_bed.collect(),\n",
      "Nextflow operator first with content: ch_versions = ch_versions.mix(ASCIIGENOME.out.versions.first().ifEmpty(null))\n",
      "Nextflow operator mix with content: ch_versions = ch_versions.mix(VARIANTS_LONG_TABLE.out.versions)\n",
      "Nextflow operator collectFile with content: ch_versions.unique().collectFile(name: 'collated_versions.yml')\n",
      "Nextflow operator collect with content: CUSTOM_DUMPSOFTWAREVERSIONS.out.mqc_yml.collect(),\n",
      "Nextflow operator collectFile with content: ch_workflow_summary.collectFile(name: 'workflow_summary_mqc.yaml'),\n",
      "Nextflow operator collectFile with content: ch_custom_no_sample_name_multiqc.collectFile(name: 'fail_barcodes_no_sample_mqc.tsv').ifEmpty([]),\n",
      "Nextflow operator collectFile with content: ch_custom_no_barcodes_multiqc.collectFile(name: 'fail_no_barcode_samples_mqc.tsv').ifEmpty([]),\n",
      "Nextflow operator collectFile with content: ch_custom_fail_barcodes_count_multiqc.collectFile(name: 'fail_barcode_count_samples_mqc.tsv').ifEmpty([]),\n",
      "Nextflow operator collectFile with content: ch_custom_fail_guppyplex_count_multiqc.collectFile(name: 'fail_guppyplex_count_samples_mqc.tsv').ifEmpty([]),\n",
      "Nextflow operator ifEmpty with content: ch_amplicon_heatmap_multiqc.ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_pycoqc_multiqc.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: ARTIC_MINION.out.json.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: FILTER_BAM_SAMTOOLS.out.flagstat.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: BCFTOOLS_STATS.out.stats.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_mosdepth_multiqc.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_quast_multiqc.collect().ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_snpeff_multiqc.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collect with content: ch_pangolin_multiqc.collect{it[1]}.ifEmpty([]),\n",
      "Nextflow operator collectFile with content: ch_nextclade_multiqc.collectFile(name: 'nextclade_clade_mqc.tsv').ifEmpty([])\n",
      "Nextflow operator toList with content: multiqc_report = MULTIQC.out.report.toList()\n",
      "Error: Unclosed process block in file: ./git_repos\\viralrecon\\workflows\\nanopore.nf\n"
     ]
    }
   ],
   "source": [
    "directory_path = './git_repos'\n",
    "\n",
    "df_operators = pd.read_csv('./data/nextflow_operators.csv')\n",
    "pattern_list = [re.escape(str(pattern)) for pattern in df_operators['operator']]\n",
    "\n",
    "output_file_name = 'nextflow_operators_in_workflow'\n",
    "\n",
    "find_operators_in_processes(directory_path, pattern_list, output_file_name)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "operator\n",
       "none            45222\n",
       "mix              2450\n",
       "map              1954\n",
       "collect          1192\n",
       "first             847\n",
       "set               750\n",
       "join              601\n",
       "ifEmpty           339\n",
       "combine           331\n",
       "groupTuple        268\n",
       "collectFile       248\n",
       "toList            232\n",
       "branch            192\n",
       "flatten           167\n",
       "filter            148\n",
       "dump              113\n",
       "splitCsv           99\n",
       "multiMap           63\n",
       "transpose          46\n",
       "unique             33\n",
       "concat             30\n",
       "flatMap            24\n",
       "cross              20\n",
       "subscribe          19\n",
       "toInteger          19\n",
       "toSortedList       15\n",
       "count              12\n",
       "collate            12\n",
       "view               11\n",
       "take                8\n",
       "splitText           8\n",
       "tap                 8\n",
       "countFastq          6\n",
       "buffer              6\n",
       "splitFasta          6\n",
       "countLines          5\n",
       "merge               4\n",
       "last                3\n",
       "countFasta          3\n",
       "reduce              3\n",
       "min                 2\n",
       "distinct            2\n",
       "splitFastq          1\n",
       "sum                 1\n",
       "max                 1\n",
       "randomSample        1\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "df = pd.read_csv('./results/nextflow_operators_in_workflow.csv')\n",
    "df['operator'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pipeline</th>\n",
       "      <th>workflow</th>\n",
       "      <th>operator</th>\n",
       "      <th>file_path</th>\n",
       "      <th>line_number</th>\n",
       "      <th>line_content</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>airrflow</td>\n",
       "      <td>ASSEMBLED_INPUT_CHECK</td>\n",
       "      <td>splitCsv</td>\n",
       "      <td>./git_repos\\airrflow\\subworkflows\\local\\assemb...</td>\n",
       "      <td>22</td>\n",
       "      <td>.splitCsv(header: true, sep:'\\t')</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>airrflow</td>\n",
       "      <td>ASSEMBLED_INPUT_CHECK</td>\n",
       "      <td>map</td>\n",
       "      <td>./git_repos\\airrflow\\subworkflows\\local\\assemb...</td>\n",
       "      <td>23</td>\n",
       "      <td>.map { get_meta(it) }</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>airrflow</td>\n",
       "      <td>ASSEMBLED_INPUT_CHECK</td>\n",
       "      <td>branch</td>\n",
       "      <td>./git_repos\\airrflow\\subworkflows\\local\\assemb...</td>\n",
       "      <td>24</td>\n",
       "      <td>.branch { it -&gt;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>airrflow</td>\n",
       "      <td>ASSEMBLED_INPUT_CHECK</td>\n",
       "      <td>set</td>\n",
       "      <td>./git_repos\\airrflow\\subworkflows\\local\\assemb...</td>\n",
       "      <td>28</td>\n",
       "      <td>.set{ ch_metadata }</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80</th>\n",
       "      <td>airrflow</td>\n",
       "      <td>BULK_QC_AND_FILTER</td>\n",
       "      <td>collect</td>\n",
       "      <td>./git_repos\\airrflow\\subworkflows\\local\\bulk_q...</td>\n",
       "      <td>23</td>\n",
       "      <td>ch_imgt.collect()</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55500</th>\n",
       "      <td>viralrecon</td>\n",
       "      <td>NANOPORE</td>\n",
       "      <td>collect</td>\n",
       "      <td>./git_repos\\viralrecon\\workflows\\nanopore.nf</td>\n",
       "      <td>530</td>\n",
       "      <td>ch_quast_multiqc.collect().ifEmpty([]),</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55501</th>\n",
       "      <td>viralrecon</td>\n",
       "      <td>NANOPORE</td>\n",
       "      <td>collect</td>\n",
       "      <td>./git_repos\\viralrecon\\workflows\\nanopore.nf</td>\n",
       "      <td>531</td>\n",
       "      <td>ch_snpeff_multiqc.collect{it[1]}.ifEmpty([]),</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55502</th>\n",
       "      <td>viralrecon</td>\n",
       "      <td>NANOPORE</td>\n",
       "      <td>collect</td>\n",
       "      <td>./git_repos\\viralrecon\\workflows\\nanopore.nf</td>\n",
       "      <td>532</td>\n",
       "      <td>ch_pangolin_multiqc.collect{it[1]}.ifEmpty([]),</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55503</th>\n",
       "      <td>viralrecon</td>\n",
       "      <td>NANOPORE</td>\n",
       "      <td>collectFile</td>\n",
       "      <td>./git_repos\\viralrecon\\workflows\\nanopore.nf</td>\n",
       "      <td>533</td>\n",
       "      <td>ch_nextclade_multiqc.collectFile(name: 'nextcl...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55505</th>\n",
       "      <td>viralrecon</td>\n",
       "      <td>NANOPORE</td>\n",
       "      <td>toList</td>\n",
       "      <td>./git_repos\\viralrecon\\workflows\\nanopore.nf</td>\n",
       "      <td>535</td>\n",
       "      <td>multiqc_report = MULTIQC.out.report.toList()</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10303 rows  6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         pipeline               workflow     operator  \\\n",
       "24       airrflow  ASSEMBLED_INPUT_CHECK     splitCsv   \n",
       "25       airrflow  ASSEMBLED_INPUT_CHECK          map   \n",
       "26       airrflow  ASSEMBLED_INPUT_CHECK       branch   \n",
       "30       airrflow  ASSEMBLED_INPUT_CHECK          set   \n",
       "80       airrflow     BULK_QC_AND_FILTER      collect   \n",
       "...           ...                    ...          ...   \n",
       "55500  viralrecon               NANOPORE      collect   \n",
       "55501  viralrecon               NANOPORE      collect   \n",
       "55502  viralrecon               NANOPORE      collect   \n",
       "55503  viralrecon               NANOPORE  collectFile   \n",
       "55505  viralrecon               NANOPORE       toList   \n",
       "\n",
       "                                               file_path line_number  \\\n",
       "24     ./git_repos\\airrflow\\subworkflows\\local\\assemb...          22   \n",
       "25     ./git_repos\\airrflow\\subworkflows\\local\\assemb...          23   \n",
       "26     ./git_repos\\airrflow\\subworkflows\\local\\assemb...          24   \n",
       "30     ./git_repos\\airrflow\\subworkflows\\local\\assemb...          28   \n",
       "80     ./git_repos\\airrflow\\subworkflows\\local\\bulk_q...          23   \n",
       "...                                                  ...         ...   \n",
       "55500       ./git_repos\\viralrecon\\workflows\\nanopore.nf         530   \n",
       "55501       ./git_repos\\viralrecon\\workflows\\nanopore.nf         531   \n",
       "55502       ./git_repos\\viralrecon\\workflows\\nanopore.nf         532   \n",
       "55503       ./git_repos\\viralrecon\\workflows\\nanopore.nf         533   \n",
       "55505       ./git_repos\\viralrecon\\workflows\\nanopore.nf         535   \n",
       "\n",
       "                                            line_content  \n",
       "24                     .splitCsv(header: true, sep:'\\t')  \n",
       "25                                 .map { get_meta(it) }  \n",
       "26                                       .branch { it ->  \n",
       "30                                   .set{ ch_metadata }  \n",
       "80                                     ch_imgt.collect()  \n",
       "...                                                  ...  \n",
       "55500            ch_quast_multiqc.collect().ifEmpty([]),  \n",
       "55501      ch_snpeff_multiqc.collect{it[1]}.ifEmpty([]),  \n",
       "55502    ch_pangolin_multiqc.collect{it[1]}.ifEmpty([]),  \n",
       "55503  ch_nextclade_multiqc.collectFile(name: 'nextcl...  \n",
       "55505       multiqc_report = MULTIQC.out.report.toList()  \n",
       "\n",
       "[10303 rows x 6 columns]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "non_none_operator_rows = df[df['operator'] != 'none']\n",
    "non_none_operator_rows"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>operator</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>count</th>\n",
       "      <td>665.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mean</th>\n",
       "      <td>15.493233</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>std</th>\n",
       "      <td>21.733262</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>min</th>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25%</th>\n",
       "      <td>3.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50%</th>\n",
       "      <td>7.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75%</th>\n",
       "      <td>18.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>max</th>\n",
       "      <td>171.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         operator\n",
       "count  665.000000\n",
       "mean    15.493233\n",
       "std     21.733262\n",
       "min      1.000000\n",
       "25%      3.000000\n",
       "50%      7.000000\n",
       "75%     18.000000\n",
       "max    171.000000"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "op_per_workflow = non_none_operator_rows.groupby(['file_path', 'workflow']).agg({'operator': 'count'})\n",
    "op_per_workflow.describe()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAuoAAAG4CAYAAAATycLHAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjcuMiwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy8pXeV/AAAACXBIWXMAAA9hAAAPYQGoP6dpAACIk0lEQVR4nO3dd1QU19sH8O/Qm4DYEeyi2Au2WACjJvYWE1usP0sSe+yJLcVYo4kmsbcYNfYYNZqoYIkYRVFBQUVRESsq0pR63z84O+8ubAMWdki+n3P2ODh37jwzOzP7zJ07M5IQQoCIiIiIiBTFwtwBEBERERFRTkzUiYiIiIgUiIk6EREREZECMVEnIiIiIlIgJupERERERArERJ2IiIiISIGYqBMRERERKRATdSIiIiIiBWKiTkRERESkQEzU/wUCAwMhSRIkSYKfn5/OcqoykiQVXnCFwJjl8vPzk8sEBgZqLTN37ly5zNy5cwsmWAX5ry1vYdi9eze6deuG8uXLw9bW1qj9kujfgMcTZbp79678vVSqVMlk9b5+/RrLli1DmzZtULJkSVhZWWn9/v+teUdhsjJVRffu3cNvv/2GP/74A7dv38aTJ0+QmpqKMmXKoHz58nj77bfRtWtXNGnSxFSzJCJSjEGDBuHnn382dxhERAUqLi4Obdq0QWhoqLlD+U/Id6IeGxuLuXPnYvXq1UhPT88x/t69e7h37x7Onj2LL7/8Eh06dMDSpUtRp06d/M6a/oXUz7qFEGaMpGiZO3cu5s2bBwCYM2cOW7QK2fbt2zWS9KZNm6JWrVpwdHQEAFSvXt1coVE++Pn54eTJkwCAgIAAXhkhAjBt2jQ5Sbe0tES7du1QsWJFWFtbA8g6/pHp5CtRv379Ot59911ER0f/f4VWVmjevDkqVqwIW1tbPHz4EGfPnkV8fDwA4M8//4SPjw+2bt2K9957L3/RExEpwObNm+XhefPmYfbs2WaMhoioYKSnp+OXX36R/z5+/Dh8fX3NGNG/X54T9evXr6NVq1Z4+fIlAMDa2hqTJ0/GpEmTULJkSY2yKSkp+PXXXzF58mQ8e/YMKSkp+OCDD7BlyxYMGDAgf0tA/3mmanmfO3fuf6ol+r+2vAXp0qVL8vDw4cPNGAmRefB48t9w8+ZNJCUlAQCqVavGJL0Q5Olm0jdv3uCDDz6Qk3QHBwf89ddfmD9/fo4kHQBsbW0xaNAgXLlyRb4EnJmZiVGjRuHGjRv5CJ+IyPxUx0IAKFeunBkjISIqODzWFb48Jerz589HWFiY/PfPP/9s1FlVuXLl8Ndff8HJyQkAkJSUhJEjR+YlBCIixVC/P8fCgg/TIqJ/p7S0NHmYx7pCInIpKSlJlChRQgAQAES3bt1yW4VYunSpPD0A8c8//2iMHzNmjDxu5MiRRte7efNmebpGjRrpLRsbGyuWLFki2rVrJzw8PIStra1wcXER3t7e4uOPPxYXLlwwOL85c+bI85szZ44QQojk5GSxbt060b59e+Hp6Smsra0FABESEqIxbVxcnNi2bZsYOXKkaNq0qShRooSwtrYWxYoVE1WrVhX9+vUTO3fuFBkZGQbjCAgIkOPw9fXVWU59nefXxo0b5boGDx5ssHxUVJRcvmLFihrj1OM35hMVFZXr5fL19ZXLBAQEaC2j7ftUN3jw4FzFaWj93L17V/z444+ib9++onbt2sLZ2VlYWVkJNzc3UbduXTF69GgRFBSkZ61qLldu4zC0vNmlpqaKDRs2iO7du4sKFSoIOzs7UaxYMeHl5SWGDRsm/vzzT4N1CCFExYoVc3yX0dHR4vPPPxf16tUTLi4uwsHBQdSoUUOMGTNG3L1716h68+LIkSNi6NChonr16qJYsWLCzs5OVKhQQfTo0UNs2LBBpKamGrUc+j7Zt/e8yMzMFDt37hR9+/YVVapUEY6OjsLR0VFUqVJF9OvXT+zatUtkZmYarEfbfhAdHS1mzZol6tevL4oXLy6v+3HjxokbN27kOtZjx46JUaNGiVq1aonixYsLGxsbUa5cOdGhQwexYsUKkZycbLAObfv05cuXxbhx40Tt2rVF8eLFBQDRvXv3HNMGBweL+fPni86dO4vKlSsLR0dHYW1tLUqXLi1atGghZs6cKe7du2f0/A19Nm7cqLOesLAwMXnyZNGgQQNRokQJeV34+vqKBQsWiNjYWIPrQtuxNj09XWzfvl1069ZNVK5cWdjZ2QkAYt++fRrTpqamip9//ln07NlTXheqGOrWrSs6deoklixZIkJDQw3GYYgxxxNdvxt79+4VXbp0EZ6ensLGxkaUKlVKtG/fXvz8889GbdeGPHr0SJ6vh4eH3rLly5c3uqz6MeDBgwc6yxXksTMyMlLMnDlTNGjQQJQsWVJIkiTq168vT6Pvtze75ORk0bVrV7l8mTJlREhIiEYduf2dyW3ekd99pkaNGvL8IiMjdZbr37+/Rmz6yg4aNEgut3XrVq1lwsPDxZQpU0SzZs3kfM7Z2VlUrVpVNGvWTHz00Udi37594tWrV0atB3W5ztg2bdqksXAnT57M9UwTEhKEk5OTXMeQIUM0xp87d04eV7x4cZGSkmJUve3bt5en+/bbb3WWW7lypXBxcdG7sUmSJIYNG6Z33tkPTNevXxe1a9fWWp96or5nzx5ha2tr1EbfoEGDHIlpdkzUi16iPnnyZCFJklHT9+vXTyQlJRlcrtzGkZtE/dy5c6Jq1aoG59G+fXvx9OlTvXVl/7HZt2+f3v3R3t5eHDx4UG+dufXkyRPx9ttvG1ye6tWr6zxpL6xE/ebNm6Jhw4YG59O4cWO9PzZC5NwPDhw4IFxdXXXWaWtrK1asWGFUnPfv3xd+fn4G43R3dxenTp3SW1f2fXrOnDnC0tIyR13ZE/UmTZoY9Z1YW1uLhQsXGjV/Qx9tiXpaWpoYO3as1pjVP66urmLTpk1610X2Y21MTIxo3bq11vrUE/UbN24Ib29vo5fj1q1beuMwJC+JelxcnOjWrZveuN59912jTu4MUU/gdC3rzZs3jV4vd+7ckctUq1ZN53wL8ti5evVq+SRN/ZOXRD0uLk5ju6pcubK87IWRqJtqnxk1apRcdu3atTrLqZ+QGSpr6IRszpw5wsrKyqj1M2DAAL3rQZtc30waEBAgD3t4eKBNmza5rQJOTk7o3r27fOdw9hfQNGvWDNWrV8etW7fw8uVL/PHHH+jevbveOh8/fowTJ04AyHpcUL9+/bSWmzhxIpYvXy7/XaJECTRv3hzu7u548+YNQkJCEBYWBiEENmzYgIcPH+LQoUMGL/E8f/4c7777Lu7fvw87Ozu0bt0aFStWREJCAs6dO6dR9unTp0hJSQGQtQ5r1aqFsmXLwsHBAYmJiQgPD8elS5cghMDly5fRunVrXL58GSVKlNAbQ1FVvnx5fPLJJwCAH374Qf5/1f9l5+zsXChxZdeuXTu525Y+Qghs3LgRr1+/BgCtL3qIjo6GEAKSJKFGjRqoUaMGSpQoAWtrazx//hwhISG4ffs2gKxH/7169QoHDx7MUVfPnj1Rp04dnD9/HhcuXAAANGnSROvjsZo3b57rZQaAU6dOoWPHjkhOTpb/r2nTpqhduzZSU1Nx7tw5Oda//voLrVq1wpkzZ1CqVCmDdR8/fhyjRo1CRkYGKlSogBYtWsDZ2RlRUVEIDAxEeno6Xr9+jffffx9hYWGoXLlynpZB3ZMnT9CyZUs5ZgCoWrUqmjVrBltbW1y/fh3//PMPAODWrVvw9/fHkSNH0LJlS416Bg8ejOfPnwPQv93mZ78NDw+Hr68vnj17Jv9f3bp10aBBA0iShJCQEPkxaRcvXkTLli1x6tQpeHl5Gaw7ODgYn332GVJTU+Hm5gY/Pz+4ubnh/v37CAwMRGpqKlJSUjB27FhYWFjg448/1hvn22+/jUePHgHI2uYbNGiA2rVrw8HBATExMTh16hQSEhLw8OFDtG/fHn/88Qf8/f0Nxrl48WL50aNVq1ZF06ZN4eDggLt378qPg1O5f/8+gKz7omrXro1q1arBxcUFQgg8evQI//zzD2JjY5GWloZp06YBAKZOnZpjnqrvcN++fXj48CEAoEePHihfvnyOst7e3hp/Z2Zmonfv3jhw4ID8f+rrNzo6GgEBAUhNTUVcXByGDBmCly9fYsKECQbXRUpKCrp164aLFy/CysoKb731FqpVq4Y3b95o3NCckJCAdu3ayU9ks7CwQMOGDeHt7Q0nJyckJycjJiYGV65cQWxsrMH5FoSMjAz07t0bx48fh42NDd566y1UrVoVb968wenTp+Xv8siRI5g0aRJ++umnfM3Pz89PvicuICAA1apVy1FG28vwjCmr67GdBXns3LVrl7ztli9fHm+99RZcXFzw8OFDvHjxwuD06h49eoR3330XV69eBZB1jDl69KjcB93Z2VneJ2JiYrB//34AgLu7O3r27KlRV15+Z0y5z/j5+WH16tUAsr67//3vfznK3Lp1CzExMRr/p6tsVFQU7t27ByDr5tnsx4Dly5fLxycAKFmyJJo3b45y5cpBkiS8ePECERERCA8PR0ZGhvErRV1uM3v1M8PevXvn+sxA5fvvv9c4y4iOjtYYP3fuXHnce++9Z7C+b7/9Vi7foUMHrWXWr18vl3FychI//fST1hbzEydOaJxt6Wp5UW9BUJ1Nvffee+LZs2ca5TIyMjQuoR84cEB88803elsw7ty5I9555x25/uHDh+ssW9Rb1PMbozHTmKJF3VgzZ86U63FwcBAXL17MUWbRokVi48aNObYVdadOnRLVqlWT6/r55591ls1L7MZM8+LFC419oWrVqlpbmLdu3Srs7e3lcl27dtU5X/XWCVtbW+Ho6Kj1EndYWJjGvIcOHWrUchnSsWNHje/nl19+yVHmwoULokqVKnI5T09P8fLlS511mnLfUklJSRH169eX6y1VqpQ4evRojnJHjx4VJUuWlMs1atRIZ5cd9f3AxsZGABCTJk0Sb9680SgXExMj2rZtq/E9hYeHa60zKSlJo+W2Xbt2WrvMvHr1SowePVouV65cOREXF6e1TvX1aWVlJVxcXHJ06xBC5Ij7o48+EocOHdLZApueni42btwoHB0dBZDVsn7nzh2tZYUw7riR3cKFCzXinzp1ao44Hz16JDp06KCxjOfOndNan/qxVvU74+vrq/VKq2o+y5Ytk6epVauWiIiI0Fp3ZmamOH/+vPjoo4/E/fv3jVo+XXLboq66qtyxY8ccrZRpaWli8uTJcllJkgxeWTZk+/btcn39+vXTWkbVHcLOzk6+6q+r7IcffijXp607REEfO62srISNjY1Ys2ZNjmOn+vZm6Lc3MjJSVK5cWS7TsmVLvcc6Y/MNFWOOjabcZ9S7OZUrV07r/NasWaNxXNVXVj1v/N///qcxLi0tTaMr+DfffKPz2Pv8+XOxYcMGvVfydMn1r4p6835+kpkTJ05ofDFnzpzRGB8ZGSmPs7OzM9ivp3HjxnL5LVu25BgfHx8vX+K1tLQ02GXn+vXr8iWlEiVKaO16oH5gArJOEIzpU26s1NRUUa9ePXkdvHjxQms5JurKSdS3bt2q8eOyZ8+ePNWjEhUVJW+HTZs21VmuoBL12bNny2VcXV319u3du3evxnehax9T/7GRJEn88ccfOus8ePCgXNbJyUmkpaUZtWy6ZD/u/PbbbzrLRkVFaXTJmTdvns6ypty3VDZs2CDXaW1tLYKDg3WWPX/+vMaxefPmzVrLZe8qNXr0aJ11Jicny8cfAKJPnz5ay33xxRdymXfffdfgd6Te33PBggVay6jHKElSnrpY6rNjxw6NpECX3Cbqr1690ujWOWnSJJ1l37x5o9FVx9/fX2s59WMtAFG3bl2DXUF69+4tl//rr78Mxm0KuU3UAYjWrVvr3F4yMzM11o+ubcVYxiRwqsS6bdu24t1339Vb1tPTU65PW3eIgj52Arr7S6vT99sbEhIiypQpI4/v1KmTwW3L1Il6QewzNWvWlMtoazRQnZDZ29uLefPm6S2rfkKWvVEnNDRUHteyZUt9qyFfcnXLbnx8vMbTDdzc3HIzuYbixYtr/J39Uk3VqlXlSyhv3rzBnj17dNZ148YNXLx4EUDWoyKzX4oBgA0bNiAuLg4AMGTIEINddry9vTF48GAAWd1ajhw5on+BkHUJxJR3QVtbW8vPmX/z5g3OnDljsrrJ9M6dO6dx6ezLL79Er1698lVnpUqV5O4BFy5ckF8cVhiEEFizZo3896xZs1ChQgWd5Xv27ImOHTvKfxtzqbpLly549913dY7v1KkTypYtCwBITExERESEMaHrpLokCgBdu3ZFt27ddJatVKkSZs6cKf+9atWqQn1brnqso0ePRuPGjXWWbdKkCUaMGCH/bcy6L1asGBYuXKhzvL29PZYuXSr/vX///hzdJNLS0rBy5UoAWd0rVq1aBSsr/T0qv/nmG7kLl/qLU3Tp06dPnrpY6vPee+/J3diOHTtmsnq3bduGxMREAECZMmXw1Vdf6Sxra2srrzsg69K7MY8rXrhwIezt7fWWUT9OGNONwlyWLVumc3uRJAlDhw6V/1Z17cursmXLombNmgCyunpkX9c3b96Uu0P4+fnJx11tZW/fvi13K6pevXqO7hCFcexs2rRpvt5Dc/LkSfj6+uLJkycAgIEDB+K3334zuG2ZWkHsM+pdkdS7a6uoui21aNFC4/dHX9ns9QKFt5/lqo96QkKCxt+q12PnRfa+vtoSkIEDB8r9u3/55ReNnVbd1q1b5eEePXpo7Ud8+PBhebhv375Gxdi2bVv5x/LMmTN6k6569erl6KtojLi4OJw7dw7Xrl3D8+fPkZiYiMzMTHm8emJy+fJldO3aNdfzoIIXHR2Nnj174s2bNwCAAQMG4LPPPjNq2vv37+P8+fO4efMm4uLi8Pr1a42EMCoqCkDWwf/KlSto3bq16RdAi/DwcDx+/BhAVhKmOnHV53//+x/++OMPANr7e2bXp08fveMlSUL9+vXlOO7evYs6deoYrFcX9QPxsGHDDJYfOnQoZsyYgczMTPkHW/VjX5ASExMRHBws/21MrP/73//kH/gLFy4gKSlJ7zG6e/fuBu/3ePvtt+Hh4YEHDx4gLS0NZ86cQY8ePeTxwcHBePr0KYCsH72KFSsajNPd3R01a9ZEeHg4wsLCEBcXB1dXV53ljT1eZxcaGopLly7h7t27iI+Pl+8LUlGdLISGhiIzM9MkjSyq+6SArLgNJT1NmzZF3bp15fsMAgICUKNGDZ3lixcvjnfeecdgHOpJ4Y8//qhx0qcUVapU0XvyCQANGzaUh+/evZvvefr5+cm/qdnXtfrxyt/fHzY2NvLf+spq659eGMfOvO4XQNZJd79+/eTfq/Hjx2PZsmVa76cqaAWxz/j5+WHVqlXy+FGjRsnjbt68Kd934u/vj8aNG6NYsWJISEjIUTb7CZm7u7vGfNT3sxMnTiA8PDxPeaAhuUrUixUrpvG36u1UeaE6g1LR9oPRt29fTJw4EWlpaQgICMDDhw9zrCgg64xMZeDAgVrnFxQUJA9v2bJFvhlCnwcPHsjDqi9LF0MHHG11T58+Hbt3787xA6KLuW76If2SkpLQrVs3+cDcokULrF+/3uB0QUFBmD59Ok6fPm10S21hbgMhISHycM2aNY26KVL9hsvHjx/r3GdV6tata7BO9fm+evXKYHldYmJi5KQSAN566y2D05QqVQpeXl7yj/ulS5cKJVG/cuWKfOORk5MT6tWrZ3CaBg0awNHREUlJScjIyMCVK1f0LqMxN31JkoRmzZrJx8KQkBCNRF39uBobG4sxY8YYrBOAfHVTCIGYmBi9iXpuj62bN2/G/PnzcfPmTaPKp6Wl4dWrVzmu8uaF+j6T/eZjXVq2bCknHeo3hGrToEEDo04oPvjgA/kYtGbNGly4cAGDBw/GO++8UyjbrzEKc99XyZ7AjR49Wh6nSo4dHBzQtGlTWFpawtnZGfHx8TnKqp/wa0vUC+PYmdv9QmX9+vXyDfxA1pXfzz//PE91mUJB7DPq30n2k57sJ2SWlpZo3bo1Dh8+nKOsoe/Zw8MDb731Fs6ePYv4+Hj4+PhgwIAB6NmzJ1q1apUjZ86rXCXqzs7OsLKykru/5PbOYnXqb7cCtHejKVGiBN555x0cPHgQmZmZ2L59Oz799FONMmfPnsWdO3cAAKVLl0b79u1z1JOYmKhxNeDnn3/Od7zZ5eayR0hICN5++22DdWaX/YoGmZ8QAgMHDsTly5cBZJ1h79u3D7a2tnqn27BhA/73v//luitFYW4D6k8aMaalFMi6dGlnZye31MTGxur9sXFxcTFYp/qTPdRftpFb6stjb2+P0qVLGzVdpUqV5ES9sE6U1GP19PQ0KjmzsLCAp6en0bHquxSvztPTU2tcAOSWKSCrC2Je3jRtqmOrEALDhw/Hxo0bcx1DQkKCSRL1vOwzlSpVkocNfWfGrov27dtj4sSJWLZsGYCs3xxVQlSyZEm0bNkSfn5+6N27t8b3W5gKc99X0ZfAnTx5EkDWCbyqNb1Vq1ZaEzhV2ex1qhTGsTMvXS1iYmI0umf+9NNPGicg5lAQ+0yZMmVQs2ZNRERE4MmTJxot3dlPyICs7/Dw4cM6y6rKaLNhwwb4+/vj0aNHSE5Oxtq1a7F27VpYWlqibt26aNOmDTp27Ih27doZ7BaoS66v9amvSPW3k+ZW9ml1fUHqLeTa+jOqd3vp27ev1hVhijNx9b752hjbryslJQW9e/eWf5zKlCmDOXPmIDAwENHR0UhKSkJmZiZE1o2+Gj866l1iSBlmzpwpX51xcnLC77//jjJlyuidJjw8HKNGjZKT9Hr16uH777/HhQsX8PTpU7nri+qjftm0MLcB9ateuenmpl7W0IlFYV5qLYzlMZXCiNXBwSHfdSrp2Lp27VqN42WXLl3w888/IywsDC9fvkRKSorGfqX+m2Oq/Sov31tuvrPc9B/+9ttvceDAgRxXVWJjY/Hbb79h4sSJqFixInr37i0/fq4wmaObRZkyZeQk7OnTp7h+/TqAnN0hVFTJmXrZW7duyVeYvLy8tCbThbH/5qUvuYWFhcZJv2qZzKmg9hn171G9ZVx1ktWyZUv5RFBXn3ZjEvUaNWrgypUrmDhxokaDc0ZGBi5fvozvv/8eHTt2RMWKFbF27VrDC6dFrhN19UsTqmcN54X6tBUrVoSHh4fWcur9KENCQhAeHi6PS0tLw65du+S/dXV7yf7lx8XFaRywjfkY02fMGHv27JH7HHt4eODKlSuYO3cufH194eHhAQcHB40D2L+lFf3feJKxdetWLFiwAEDWAfCXX34xqovCsmXL5OSkU6dOCA4OxtixY+Hj44NSpUrBzs5Oo7y5tgH1ez1y081NvaypLv2ZQlFansKIVf3ZznmtU/3YOmHChFwfV4UQOn8Ac2vJkiXy8Ndff43ff/8dAwcORO3ateHq6qrR5xgomP0qL99bQW5fXbt2xd9//40HDx5g69atGDVqFGrVqiWPF0Jg7969aNy4sdFdhYo6bUmZenKmnuBpS/aMSd6UeqwpV64ctmzZIifrK1aswPjx400+n9woqH1G2/d848YNrSdkjRo1kq/wqMqqP2td1wmZSqlSpfDtt9/i8ePHOH36NL7++mt07NhRo0v3w4cPMXLkSIwbN86oZVSX60RdfeEePHiAU6dO5XqmSUlJ+O2337TWmZ2dnZ3GTZzqLehHjhyRL3t4eXmhSZMmWutwdXXV6Ipw69atXMdsKsePH5eHJ06caLD11RwtHcZQvyRpqEUMME3Lm5IEBQVpXEL85ptv9D5BRJ36NvDll1/meGlLdubaBtQvrapePmLI06dP5Uu3QNZldqVQX57Xr18b3Y1Fff0X1vKox6p6OZYhmZmZGvfSGIrV2O9UvVz2OtWPX+Y8rkZHR8vzL168uNaXGKmLj4/PdddDY+RlnymM7at8+fIYMGAAVq1ahWvXriE6OhpffvmlfKL1/PlzTJo0qUDmrTTaEjhV8u3k5KSRRzRs2DBHAmeo3zKg7GPngAEDNJL177//3qiXbRWUgtpnsndzyt7gqp53WlpaolWrVhpljfmes7O2tkarVq0wc+ZMHD58GLGxsThy5Ah8fX3lMitWrMj1E4xynaj36dNHoy/ft99+m9sqsGbNGo3LHep32Wqj3lK+bds2+UdLvSuMrtZ0FfU3NR49ejRX8ZqSep/O2rVrGyyflxOhwqB+Fqt6O6M+qhs//g3u37+Pnj17yjcBDxkyxGBioC4328CrV6/kt8XpUxCXkdWfuBAREWHUPSl///23PFy2bFm9rRCFrXz58hr90s+ePWtwmufPn2v0u27UqFGBxJZd/fr1YWlpCSDr0rAx+8+VK1fkliZLS0vUr19fb3n1G0F1EUJoXP3MvvzNmjWTh0+ePGn0jfGmpr5P1ahRw2Bf0DNnzhh18pPb/Up9nzFm+wI095nC2r48PDzw+eefa1yK//PPP832/RUm9aTr5MmTEEJodIdQ33ZUNxpqK5u9LnVKP3YOGDAAmzdvlpP17777DhMnTiyw+elTUPtM6dKl5W5OsbGxuHbtmsYJmY+Pj0Z5VeKevSxgfKKenbW1Nd555x38+eefGjdP//7777mqJ9eJuqOjo8arpH/77Tfs3bvX6Onv3buH2bNny3+3atXK4NMH/P395eeU3r17F2fPnkVCQoLG62YNPU+0S5cu8vCqVas0zlwLk3r/MEOXni9evJjvZ8cWFPVXuV+5csXgj97OnTuNqle924cpbh4yNdUTXlTPnm3VqlWuH32Wm21g3bp1Rq2Hglhv3t7e8jPMMzIyNK5m6aLeR9iY18MXNvWYNm3aZLD85s2b5W5b7u7ueh+dZ0rZf0iMiVV93Tdt2tRgf88DBw4YvNJ17Ngx+fKvtbV1jqcytGzZUn5iS2JiYp77YOZXbvYpwLjnVAO536/atm0rD+/YscPg78ylS5c0TsQLe59R/11MS0vL1wMiiorsCdyePXvw6NEjANrXvypJU5VVnRR6eXmhXLlyWudRFI6dAwcO1EjWly9fbpZkvSD3mexXT1QnWa1atcpxMp+9rCkSdRUbGxuNB52o8gdj5enBsZ999plGP7cPP/zQqJbfx48fo0OHDnJruoODg1EHdgsLC/Tr10/+e+vWrdi7dy9ev34NIOsu7SpVquitY9SoUfIPyoMHD/Dxxx/n6pF4qkcZ5Zd6nOrdf7JLTk7GyJEjTTLPguDt7S23qj969Ah//vmnzrKHDh3CoUOHjKpX/TFWqgRBKYQQGDBgAK5cuQIg62Rl3759Ofq+GmLsNnDr1i3MmzfPqDoLYr1JkqSxDX7xxRd66z506JBGS4G5nyagjfrVu3379ul9kVl0dLTGyzdGjRpVqDfAqcf6ww8/6L2yEhISIj92DjBu3SckJGD69Ok6x79+/RqTJ0+W/+7evXuOJ03Y2tpqXDafOXNmrq6e5fYHS5fKlSvL301YWBhu376ts+yvv/6KgwcPGlVvbver/v37axwX9e2/qampGDt2rPy3v7+/yU4Eje3Wpd7VwMLCIl8vMSxK1JM79e9IW9KXm7IqReXYOXDgQGzatEkjWS/sLlAFuc+of0dr1qzRe0LWsGFDOUdUL1ujRg2dJ2QvX740+v479X0t10/syesrTUNDQzVer21tbS1mzpwpYmNjc5RNSUkRmzdvFqVLl9Z4LbSu11xrc+XKFXnaEiVKaLza+ccffzSqjuyvL+7cubMIDw/XWjYzM1OcPXtWfPLJJ8LBwUEkJCTkKJOX17YfOXJEYx0sWbJEpKena5S5deuWaNmypQAgHB0d5fKDBw/WWqexr/RVX3ZTGDZsmFyfp6enuHbtmsb4zMxMsWXLFuHg4CBsbW11vsZYneq1zQDEwoULjYrDmOUy5lXghr7PadOmyeOdnZ1FWFiYUfFlN336dLme4sWLiyNHjuQoc+zYMeHu7p5jG9i4caPWOs+dOyeXKVu2rIiLizMYhzHb74sXL+TXagMQNWrUECEhITnK7dixQyPOrl276pyv+muwo6KiDMY5ePBgg8ufGx07dpTrc3JyEjt37sxR5tKlS6J69eoa2/fLly911mnqfUuIrONm/fr1Nb7XEydO5Ch37NgxjWNro0aNRGpqqtY61fcDGxsbAUB8+umn4s2bNxrlHj58KN5++22Nstn3b5WEhARRu3ZtjX1jzZo1IiUlRWv52NhYsXbtWtGoUSPx6aefai2Tl/XZvHlzeZqGDRuKiIgIjfEZGRli5cqVwtraWlhaWgo7OzuD2+GCBQs0fi8yMzMNxrFw4UKN+D///PMc6+Lx48eiU6dOchkrKytx7tw5rfWp/27p+g3Irlq1aqJv377i0KFDOr+H69evi0aNGsl1t2/f3qi6dTHmeJLbZYmKijLqdyO3du7cqfEdARDFihXL8VssRNZ24+rqmqP89u3b9c5DCcdOIYxbh1u2bBEWFhZyuYkTJ+qsz9h8Q8WYfdnU+4zK06dPc3xvAMQ///yjtXzXrl1zlB01apTO+jdu3CiqVKkiFi1aJO7cuaO1zOvXr8WyZcuEJElynX///bfeuLPL20MdAdSpUwdnzpxBx44d5bfWzZ8/H4sWLZLfUGdjY4NHjx7h7NmzGpdYbWxssGXLFnzwwQdGz69evXry26ieP38uX8KwtrbG+++/b1QdQ4YMwZ07d/Dll18CyDqLPXz4MOrUqYM6derA2dkZSUlJiImJQUhIiPxSDlN655134OvrK/d3mzx5Mn744Qf5ruNbt27h7NmzyMjIQPny5TF+/Phc9X8uTLNmzcKvv/6KpKQkREdHo0GDBvD19UWVKlUQHx+Ps2fP4v79+7C0tMTq1as1br7UpXfv3nIr5/Tp03HkyBHUrl1b42bgzz77zCTPPM4t9detV69e3ejL582bN9e4h2LixIlYv349nj17hpcvX+Ldd99Fo0aNUKtWLUiShEuXLuHatWsAsraX0qVLG3z2f5MmTVChQgXcv38fjx8/Rs2aNdGhQweULFlSbmls0qRJrvY5IOvGvG3btqFjx45ITk7GjRs30KhRIzRr1gy1atVCamoq/vnnH40bCatXr27UC5/MZePGjWjZsiVu376NxMREvP/++6hevTqaNWsGGxsbhIeH49y5c/IVN0dHR2zfvl3vS3kKgo2NDbZv3w5fX188e/YMjx8/Rtu2bVG/fn00aNAAQNbbilVXeICsy/rbt283eIMykPVklM8++wxLly7Fxo0b0bZtWxQvXhz3799HYGCgRn/lxYsXa1xFVefk5IQDBw6gXbt2iIqKQnx8PEaOHIkpU6agRYsWKF++PCRJwosXLxAeHo4bN27IrVCmvMT/1VdfoUOHDsjMzERISAjq1q2Lli1bokqVKkhMTMTp06flVrKvv/4aa9asMXijdq9evTBjxgwIIXDo0CHUq1cPb731lsY9On379tXopjR58mScOXNGbiH96quv8NNPP8Hf3x/FixdHdHQ0AgICcqxf9f7++ZWWloYdO3Zgx44dsLe3R7169VClShU4Ozvj5cuXuH37Ni5evCiXt7e313hqzr+dtq4MrVu3lu8LUWdhYYE2bdpodLXVVYe6onTs/PDDDyGEwNChQ5GZmSm/pXTp0qWFMv+C2mdKlSqFWrVqaTyG0tnZWefLovz9/XP0Hzf0Pd+5cwdTp07F1KlTUaFCBdSrVw+lS5eGEAKPHz/GuXPnNG5cHzBggFEv29OQq7ReiydPnojRo0cLKysrrWcu2T/t2rUTly9fztO81Fs3VJ9u3brlup5ff/1Vbq005tO0adMcLU5C5K1FXYisM0P1lgxtn1q1aolr164Z1QJhrhZ1IYT4448/hIODg87lcHZ2Fnv27DG6ZSQtLU34+/vrXTfZWxKMWS5TtKgbu71k/2j73s6ePStKliypd7oePXqIuLg4o1uUDx06pHHlwlAcudl+g4KCRJUqVYzav58+faq3LnO3qAuRtQ+2bdvW4PJUq1ZNnD9/3mB9BbFvqdy4cUM0bNjQYKyNGjUSkZGReuvKvh/89ttvGldGs39sbGzE8uXLjYrz+fPnok+fPhotR/o+rq6uYtOmTVrryuv6/Omnn/T+FllYWIjZs2eLzMxMo7fDzz//XO9yaNsm09LSxJgxY4SlpaXeaV1cXAxu03lpUa9Tp47Rx6fKlSvnuoVPm6LUoi6EELVq1dJYD4sXL9ZZ9ttvv9UoW6NGDaPnY85jpxC5W4ebN2/WaFnXdsWrIFrUhTDtPqPu448/1pi+c+fOOsteunQpx/wePXqks/yuXbuMPt5ZWFiIjz/+WOfVTn3y3KKuUrp0afz000+YNm0a9u/fjyNHjiAyMhJPnz5FWloaSpUqBQ8PD7Rt2xbdunXTePpKbg0YMAAzZ87U6BNk6Gkv2rz//vvo3r07duzYgaNHj+LChQt49uwZEhMT4ejoiPLly8Pb2xutW7dGp06d4OXlleeYtSlTpgzOnj2LdevWYceOHQgLC0NycjJKly6NGjVq4IMPPsCAAQPg4OCA8+fPm3Tepvbuu+8iIiICS5YswdGjRxEdHQ1LS0tUqFABXbt2xUcffYQKFSrg7t27RtVnZWWFo0ePYsOGDdizZw9CQ0Px4sULpKamFuyCFLIWLVrg2rVrWL58OX7//Xf57brlypVD48aNMXDgQHTt2jVXdXbq1AkXL17EypUrcebMGdy7dw+JiYm5fvupNs2bN0d4eDi2bt2K/fv34/Lly3j69Cmsra1RtmxZtGrVCv369UOHDh3yPa/CUKZMGRw/fhxHjx7Fjh07cObMGTx+/BhpaWkoXbo0GjZsiB49emDgwIFGtU4XJC8vLwQHB2P37t3Ys2cPzp8/j6dPnwLIOv42a9YM7733Hnr37p3rPvTdunXD1atXsWrVKhw8eBDR0dFISUmBh4cH3nnnHYwZM8boftNubm7YuXMnwsLCsH37dgQGBiIqKgrPnz+HhYUFXF1dUa1aNTRq1Ajt2rVD+/btc7wzIL9Gjx6Nli1bYtmyZQgICMDDhw9hb2+P8uXLo23bthg2bJjGUyaM8eWXX6Jly5bYsGEDgoOD8eTJE4M3rFpZWWHFihUYPXo0NmzYgOPHjyM6OhoJCQlwc3ODl5cXOnXqhBEjRhj1evncunz5Ms6dO4eAgACcP39efn50cnIyHBwcULZsWTRo0ADdunXD+++/b/BNyv9G/v7+Gi2t+lpOs1/5yc3NhUXp2Dlo0CAAkFvWly5dCkmSsHjx4gKfd0HtM/7+/vjxxx/lv/V9d/Xr10fx4sXlFvAaNWrINwVr895778n36P3999+4cuUK7ty5I/fGcHFxgZeXF1q1aoVBgwbpvCppiCRM8StORESK5+fnJ3cbDAgIMNnLhoiIqGDk6akvRERERERUsJioExEREREpEBN1IiIiIiIFYqJORERERKRATNSJiIiIiBSIiToRERERkQLx8YwEAMjMzMTDhw9RrFixXD+LmYiIiExDCIGEhAS4u7vDwoLtqf91+X7hEf07PHz4EJ6enuYOg4iIiABER0fDw8PD3GGQmTFRJwBAsWLFAGQdGJydnc0cDRER0X9TfHw8PD095d9l+m9jok4AIHd3cXZ2ZqJORERkZuyGSgBvJiUiIiIiUiQm6kRERERECsREnYiIiIhIgZioExEREREpEBN1IiIiIiIFYqJORERERKRATNSJiIiIiBSIiToRERERkQIxUSciIiIiUqB/xZtJnzx5goMHDyI2NhaVK1dG165dYW9vb+6wiIiIiIjyTPGJenh4OObMmQNJkrB69Wq4urpqjD9w4AD69++P169fy//n6emJAwcOoF69eoUcLRERERGRaSg+Ud+/fz92796NNm3a5EjSnz59ioEDByI5OVnj/+/fv4+uXbvi+vXrcHR0LMRoSZv79+8jNjbW5PWWLFkSFSpUMHm9REREREqg+ET9+PHjkCQJXbp0yTHuxx9/RGJiIqysrLBo0SK8/fbbOHr0KKZPn44HDx5g7dq1mDBhQuEHTbL79+/D29s7x8mUKTg4OCA8PJzJOhEREf0rKT5Rv3//PgCgfv36Ocbt3bsXkiRh0KBBckJet25d3Lp1C2vXrsWBAweYqJtZbGwskpOTsfWzL+BdsbLJ6g2/F4WBX89GbGwsE3UiIiL6V1J8ov7s2TMAQKlSpTT+PzY2FteuXQMA9O/fX2Nct27dsHbtWnk8mZ93xcpo5FXT3GEQERERFRmKfzyjqsvEmzdvNP7/zJkzEELAxsYGLVu21BhXrlw5AEBcXFyhxEhEREREZGqKT9Td3NwA/H8XGJXjx48DAHx8fGBra6sxLj09HQDg5ORUCBESEREREZme4hN1Vd/0bdu2yf/3+vVr7Nq1C5IkoW3btjmmuXfvHgCgTJkyhRMkEREREZGJKT5R79u3L4QQ+P3339G3b1+sXLkSHTp0wNOnTyFJEvr165djmn/++QcAUKVKlcIOl4iIiIjIJBSfqA8aNAitWrWCEAK7du3C+PHjcfbsWQDA0KFDUbNmzhsUVU+D0dbaTkRERERUFCg+UbewsMAff/yBSZMmwcPDA1ZWVvD09MSsWbPw008/5Sj/+++/4+7duwCA9u3bF3K0RERERESmofjHMwKAo6MjlixZgiVLlhgs26pVK0RFRQEAKlasWNChEREREREViCKRqOdG8eLFUbx4cXOHQURERESUL4rv+pKYmGjuEIiIiIiICp3iW9Td3NzQsGFD+Pr6wtfXF61bt4azs7O5wyIiIiIiKlCKT9TT09MRHByM4OBgLF26FBYWFqhfvz78/Pzg6+uLNm3awMXFxdxhEhERERGZlOIT9WXLliEwMBCnT5/GixcvkJGRgUuXLiEkJATLli2DJEmoV6+eRuLOPupEREREVNQpPlEfP348xo8fDwC4cuUKTp48KSfuz58/hxACly9fxpUrV/Ddd99BkiTUqVNHTtx79uxp5iUgIiIiIso9xSfq6urXr4/69etj3LhxAICrV6/KifupU6fkxD00NBShoaFYuXIl0tPTzRw1EREREVHuKf6pL/rUq1cPY8eOxZ49e3D9+nXMmjVL7q8uhIAQwswREhERERHlTZFqUVf3/PlzuTU9MDAQ169fz5GcV61a1YwREhERERHlXZFJ1HUl5gDkf6tUqQI/Pz/4+fnB398f5cuXN2fIRERERER5pvhEffz48QgICNDaYs7EnIiIiIj+rRSfqK9YsQKSJEEIwcSciIiIiP4ziszNpJIkwcHBQf7Y2dmZOyQiIiIiogKj+Bb1YcOG4eTJk7h9+zbCwsJw7do1/PDDD5AkCbVq1ZJb2H19fVGiRAlzh0tEREREZBKKT9TXrVsHAHj48CECAgLkG0ojIyOZuBMRERHRv5biE3UVd3d3DBgwAAMGDAAAPHr0SE7cAwICtCbutWvXxpUrV8wcORERERFR7hWZPurZlStXDv3798fq1atx8+ZNxMTEYNasWXB2doYQApmZmQgLCzN3mEREREREeVJkWtS1uXnzpvxc9ZMnT+Lx48cAID8lhoiIiIioqCpSibquxByARmJerVo1+Pr6ws/PzwxREhERERHln+IT9bVr18p90XUl5l5eXnJi7ufnh3LlypkjVCIiIiIik1F8oj5q1KgcXVlq1qwpJ+a+vr4oW7asGSMkIiIiIjI9xSfqAODt7a2RmJcuXdrcIRERERERFSjFJ+pPnz5FyZIlzR0GEREREVGhUvzjGZmkExEREdF/keJb1HVJT0/Hy5cvAQDFixeHlVWRXRQiIiIiohwU36KuLjw8HGPHjoW3tzfs7OxQtmxZlC1bFnZ2dvD29sa4ceNw/fp1c4dJRERERJRvRSZRnzFjBurVq4cff/wRN27cQGZmJoQQ8ltIb9y4gR9++AH169fHzJkzzR0uEREREVG+FIn+ImPHjsWPP/4oP6LR29sbzZo1Q9myZSGEwJMnT3D+/Hlcv34dGRkZWLhwIZKSkvDdd9+ZOXIiIiIiorxRfKL+999/44cffoAkSahVqxbWrFmDt956S2vZoKAgjB49GqGhoVi5ciU++OADnWWJiIiIiJRM8V1fVq9eDQCoXLky/v77b72Jd4sWLXDq1ClUqVIFALBq1apCiZGIiIiIyNQUn6ifPn0akiRh+vTpcHFxMVjexcUF06ZNgxACp0+fLoQIiYiIiIhMT/GJ+uPHjwEADRs2NHqaRo0aAQCePHlSIDERERERERU0xSfqdnZ2AICkpCSjp0lMTAQA2NraFkhMREREREQFTfGJeuXKlQEABw4cMHqa33//HQDkvupEREREREWN4hP1Tp06QQiBlStX4vjx4wbLHz9+HCtWrIAkSejUqVMhREhEREREZHqKT9QnTJgAZ2dnpKWloWPHjvjkk09w8eJFZGZmymUyMzNx8eJFfPzxx+jYsSPS0tLg7OyMCRMmmC9wIiIiIqJ8UPxz1EuWLImdO3eiW7duSE1NxapVq7Bq1SrY2NjAzc0NkiTh+fPnSE1NBQAIIWBjY4Ndu3ahRIkSZo6eiIiIiChvFN+iDgAdOnTAuXPn4OPjAyEEhBBISUnBo0eP8PDhQ6SkpMj/36RJE/zzzz9o166ducMmIiIiIsozxbeoqzRo0ADnz5/HhQsXcOzYMYSFheHFixcAADc3N9SpUwft2rVDkyZNzBwpEREREVH+FZlEXaVJkyZMxomIiIjoX69IdH0hIiIiIvqvYaKuR3x8PHbs2IFPP/0Uvr6+qFatGlxcXGBjY4PSpUvDz88PixYtwvPnz42q78iRI+jVqxc8PDxga2sLDw8P9OrVC0eOHDE6puTkZCxevBhNmzaFm5sbnJyc4O3tjcmTJ+P+/ft5XVQiIiIiUhhJCCHMHQSAAksyK1SokOdpjx07hvbt2xssV7JkSWzduhXvvPOO1vFCCIwePRpr1qzRWcfIkSOxatUqSJKks8zt27fRuXNn3LhxQ+t4FxcXbNu2LU/Pj4+Pj4eLiwtevXoFZ2fnXE+vy6VLl9C4cWNcXPMzGnnVNF29NyPQeOSHuHjxIho1amSyeomIiMypoH6PqWhSTB911RtITUmSJKSnp+erDk9PT/j7+6Nx48bw9PREuXLlkJmZiQcPHmD37t3Yu3cvYmNj0a1bN1y4cAH16tXLUcfnn38uJ+kNGzbE1KlTUbVqVdy+fRuLFi1CSEgI1qxZg1KlSuGrr77SGkdiYiK6dOkiJ+kjRoxA3759YW9vj4CAAHzzzTd49eoV+vTpg6CgIK1xEBEREVHRoZgWdQsL0/fCkSQJGRkZeZ4+IyMDlpaWesvs378fPXv2BAD06tULe/bs0RgfGRkJb29vpKenw8fHB6dOnYK9vb08Pjk5Gb6+vggODoaVlRUiIiJQtWrVHPOZO3cu5s2bBwBYtGgRpkyZojE+KCgIbdq0QXp6Ovz9/XHixIlcLStb1ImIiMyPLeqkTjEt6hs3bjR3CDkYStIBoEePHqhZsyYiIiJw6tSpHOOXLVsmt+qvWLFCI0kHAAcHB6xYsQItWrRAeno6li9fjhUrVmiUSUtLw3fffQcA8Pb2xqeffppjPi1atMDw4cOxevVqBAQE4OLFi2jcuLHRy0pEREREyqKYRH3w4MHmDiHPHB0dAQBv3rzR+H8hBH777TcAQM2aNdG8eXOt0zdv3hw1atTAjRs3sH//fnz//fcafdUDAwMRFxcHIGs96br6MGTIEKxevRoAsHfvXibqREREREWYYp76MmnSJHz66ad4+vSpuUPJlfDwcFy+fBlAVjKuLioqCjExMQAAX19fvfWoxj948AB3797VGHf69Okc5bTx8fGRTxrOnDljVPxEREREpEyKSdSXL1+O5cuXIzY2VuP/K1eujKpVqyIyMtJMkeWUnJyMW7du4dtvv4W/v7/cD378+PEa5cLDw+Xh7El8durj1afLTT1WVlZy//bsdRARERFR0aKYri+63Lt3D5IkITU11axxbNq0CUOHDtU5fvLkyRgwYIDG/0VHR8vDHh4eeuv39PTUOp36346OjnB1dTVYz9WrV/Hs2TOkpKTA1tZWb3kiIiIiUibFJOoODg54/fp1jhZ1pWvQoAFWrVqFZs2a5RiXkJAgDzs5OemtR9VlBch6FKO2egzVoa0eXYl6SkoKUlJS5L/j4+MN1k1EREREhUcxXV+qVasGANiyZQsU8sRIDT169EBoaChCQ0Nx/vx5bN++HT179sTly5cxYMAAHDx4MMc06jeX2tjY6K1fPaF+/fq11noM1WGoHnXffPMNXFxc5I96iz4RERERmZ9iWtR79uyJq1evYuPGjfjjjz9QpUoVWFtby+OHDh2q0VpsDEmScPz4cZPE5+rqqtHtpEmTJujbty9+/vlnDB48GN27d8f69esxZMgQuYydnZ08bKjrjnrrdvZHOKrqMab7j7561M2YMQOTJk2S/46Pj2eyTkRERKQgiknUp02bhj///BNBQUF49OgRHj16JI8TQuDChQtG1yVJEoQQGo84LCgffvghDh48iJ07d2LMmDHo3r07ihcvDgAoVqyYXC57d5bskpKS5OHsXVxU9Riqw1A96mxtbdl/nYiIiEjBFJOo29nZ4eTJk9i1axeOHTuGmJgYpKSk4OTJk5AkCY0bN851i3ph6d69O3bu3ImkpCT88ccf6N+/PwDNG0gfPHigtw71G0izt2x7eHjgn3/+QVJSEuLi4vTeUKqqp1SpUkzEiYiIiIowxSTqQNbjBfv164d+/frJ/6d6uc+mTZtQq1Ytc4WmV6lSpeThe/fuycPq8UZEROitQ328t7e3xrhatWphz549cjldL05KT0/H7du3tdZBREREREWLYm4mLcpULzUCNLubVK5cGe7u7gCAkydP6q3j1KlTAIDy5cujUqVKGuNatWolD+urJzg4WO760rJlS+OCJyIiIiJFUnyiHhUVhTt37sDLyyvX06anpxdARDnt2rVLHq5bt648LEkSunfvDiCrJfzcuXNapz937pzcot69e/ccfev9/Pzg4uICANi8ebPOp+Js2rRJHu7Zs2fuF4SIiIiIFEPxifqZM2dQsWJFWFnlrpdOWloaevfuna95b9q0SeMRi9osW7YMhw8fBgBUqlRJo/UbACZMmCDHPnbs2ByPTHz9+jXGjh0LIKvrz4QJE3LMw8bGBuPGjQOQ9cbRJUuW5CgTFBSE9evXAwB8fX3RpEkTI5aQiIiIiJRKUX3UtRk2bBhcXV3RuXNno6dJS0tDjx49cOTIkXzNe+7cufj000/Ru3dvtGrVClWrVoWTkxMSEhIQGhqKX375BX///TeArGR67dq1OU4ovLy8MHnyZCxYsADBwcFo2bIlpk2bhqpVq+L27dtYuHAhQkJCAABTpkxB9erVtcYyZcoU/Prrr7h58yamTp2KyMhI9O3bF/b29ggICMD8+fORnp4Oe3t7LF++PF/LTURERETmJwklvl1IjYWFBezt7XHkyBG0bt3aYPnU1FR0794dR48ehSRJyMjIyPO8K1WqpHFzqC4eHh7YsGED2rdvr3V8ZmYmRowYgQ0bNuisY/jw4VizZo1886w2kZGR6NSpE27duqV1vLOzM3755Rd06dLFYMzZxcfHw8XFBa9evYKzs3Oup9fl0qVLaNy4MS6u+RmNvGqart6bEWg88kNcvHgRjRo1Mlm9RERE5lRQv8dUNCm+60uLFi3w+vVrdO3aFZcvX9Zb9s2bN+jSpQuOHj0KIOulPvlx/PhxrFq1Ch988AHq1auHMmXKwMrKCk5OTqhatSp69+6NjRs34saNGzqTdCDrZGP9+vU4dOgQunfvDnd3d9jY2MDd3R3du3fH4cOHsW7dOr1JOpD19taQkBAsXLgQPj4+cHV1hYODA2rUqIGJEyfi6tWreUrSiYiIiEh5FN+i/urVK7Rp0wahoaEoXbo0Tp8+rbV7iCqZP3HiBABg1qxZmDdvXmGHW2SxRZ2IiMj82KJO6hTfou7i4oKjR4+iSpUqePr0KTp06KDxOEQg622cnTp1kpP0uXPnMkknIiIioiJN8Yk6AJQtWxZ//fUXypUrh/v376NDhw54/vw5ACAxMRGdOnWSny/+5ZdfYvbs2eYMl4iIiIgo34pEog5kvTzo6NGjcHV1RUREBDp27IiYmBi88847OH36NABg/vz5+Oyzz8wcKRERERFR/hWZRB0A6tSpg4MHD8LBwQEXL15E9erVERQUBABYsGABpk+fbuYIiYiIiIhMo0gl6kDWU2D27NkDa2trvHnzBpIkYcmSJZg6daq5QyMiIiIiMhnFvPDo1KlTRpe1s7PDiBEj8MMPP6BXr17w8fHROX2bNm1MFSIRERERUaFRTKLu5+cHSZJyNY0kSdi7dy/27t2rc3x6eropwiMiIiIiKlSKSdQBQOGPdCciIiIiKjSKSdQDAgLMHQIRERERkWIoJlH39fU1dwhERERERIqhmERdl/v37wMAnJyc4ObmZuZoiIiIiIgKh+Ifz1ipUiVUrlwZO3bsMHcoRERERESFRvGJur29PQCgSZMmZo6EiIiIiKjwKD5RL1++PAAgIyPDzJEQERERERUexSfqHTp0AACcOXPGzJEQERERERUexSfq48ePh729PZYsWYKYmBhzh0NEREREVCgUn6hXr14d27ZtQ3JyMpo3b45t27YhNTXV3GERERERERUoxT+esW3btgCAUqVKISoqCh9++CGGDx+O6tWro3jx4rC0tNQ5rSRJOH78eGGFSkRERERkMopP1AMDAyFJkvy3EAIpKSkICwvTOY0kSRBCaExHRERERFSUKD5Rb9OmDRNuIiIiIvrPUXyiHhgYaO4QiIiIiIgKneJvJiUiIiIi+i9iok5EREREpECK7/qijRACd+7cwYsXLwAAbm5uqFy5MiwseN5BRERERP8ORSpRP3r0KFauXInAwEAkJydrjHNwcIC/vz/GjBkjv82UiIiIiKioKhJN0KmpqejXrx86deqEw4cPIykpCUIIjU9SUhIOHTqEjh07on///nwpEhEREREVaUWiRb1///7Yt28fhBCwsrJC+/bt0axZM5QtWxZCCDx58gTnz5/HX3/9hbS0NPz6669IT0/Hzp07zR06EREREVGeKD5RP3ToEPbu3QtJkuDv748NGzagYsWKWsvev38fw4YNw4kTJ7Bnzx4cPnwYnTp1KuSIiYiIiIjyT/FdXzZt2gQAqF+/Po4cOaIzSQeAChUq4I8//kCDBg0AABs3biyECImIiIiITE/xifq5c+cgSRI+/fRTWFtbGyxvbW2NyZMnQwiBc+fOFUKERERERESmp/hE/dmzZwCAWrVqGT1NzZo1AQCxsbEFEhMRERERUUFTfKLu6OgIAHj+/LnR06ier+7g4FAgMRERERERFTTFJ+o1atQAAPz6669GT7Njxw6NaYmIiIiIihrFJ+rdunWDEAIbN26UbyzVZ9OmTdi4cSMkSUKPHj0KPD4iIiIiooKg+ER97NixKFeuHIQQGD58ODp37ow9e/bgwYMHSEtLQ1paGh48eIA9e/agU6dOGD58OIQQcHd3x5gxY8wdPhERERFRnij+OeqOjo44ePAg2rVrh5cvX+LIkSM4cuSIzvJCCBQvXhwHDx5kH3UiIiIiKrIU36IOAA0bNkRoaCh69+4NCwsLCCG0fiwsLPDee+/h6tWrqF+/vrnDJiIiIiLKM8W3qKu4u7tj165dePz4MQICAhAWFiY/3cXNzQ116tSBn58fypUrZ+ZIiYiIiIjyr8gk6iply5ZFv379zB0GEREREVGBKhJdX4iIiIiI/msUn6hPmzYNhw8fRkJCgrlDISIiIiIqNIpP1BcvXoyuXbvCzc0NTZo0wZQpU3Dw4EG8evXK3KERERERERUYxfdRd3V1RVxcHDIyMnDx4kVcunQJ3377LSwsLFCvXj34+fnB19cXbdq0gaurq7nDJSIiIiIyCcUn6i9evMCVK1cQGBiIwMBAnD59Gi9evEBGRgZCQkJw+fJlLF++HJIkoW7duhqJu5ubm7nDJyIiIiLKE8Un6gBQv3591K9fH+PHjwcAXL16VSNxf/78OYQQuHLlCq5evYrvv/8ekiShTp06uHz5snmDJyIiIiLKgyKRqGdXr1491KtXD+PGjQMAhIaGyon7sWPHkJCQACEEQkNDzRwpEREREVHeKP5mUkOeP3+OW7duyZ+kpCRIkmTusIiIiIiI8qXItag/f/4cJ0+elFvQr1+/DiEEAMj/VqlSBX5+fvD39zdnqEREREREeab4RD03ibkqOS9fvrw5QyYiIiIiyjfFJ+qlS5eWh5mYExEREdF/heITdSGE3Oe8e/fumDNnDho0aGDeoIiIiIiICliRuJlUCAEhBA4cOAAfHx/5UY179+7F8+fPzR0eEREREZHJKb5FPTo6Wu6fHhgYiNu3byM0NBShoaFYuXIlJElCrVq15K4wvr6+KFGihLnDJiIiIiLKF8Un6uXLl8eAAQMwYMAAAMDDhw81EvfIyEiEhYUhLCwMP/zwQ47EvVevXmZeAiIiIiKi3FN8op6du7s7+vfvj/79+wMAHj16pJG437p1C9euXUNYWBh+/PFHpKenmzliIiIiIqLcK3KJenblypVD48aNkZCQgPj4eDx58gQJCQkA/v8pMURERERERU2RTNRv3rwpt6CfPHkSjx8/lsepJ+dVq1Y1R3hERERERPlWJBJ1YxPz6tWrw9fXV+6f7u7unu95X7p0CUeOHMHp06cRFhaGp0+fwtraGu7u7njrrbcwfPhwtG7d2uj6jhw5gjVr1uD8+fN49uwZSpUqhaZNm2LkyJF49913jaojOTkZP/zwA3bt2oXIyEikpqbC09MTnTt3xrhx41ChQoW8Li4RERERKYTiE/Xy5cvrTMxr1KghJ+a+vr4oV66cSeft6+uLU6dO5fj/1NRU3Lp1C7du3cLmzZvx4YcfYt26dbCxsdFZlxACo0ePxpo1azT+PyYmBvv27cO+ffswcuRIrFq1Sn5uvDa3b99G586dcePGDY3/j4iIQEREBNatW4dt27ahU6dOuVxaIiIiIlISxSfqjx49koe9vb01EvMyZcoU6LxjYmIAZN3A2qdPH7Ru3RoVKlRARkYGgoKCsHTpUsTExODnn39Geno6tm3bprOuzz//XE7SGzZsiKlTp6Jq1aq4ffs2Fi1ahJCQEKxZswalSpXCV199pbWOxMREdOnSRU7SR4wYgb59+8Le3h4BAQH45ptv8OrVK/Tp0wdBQUGoV6+eidcIERERERUWSSj8jssxY8bIiXmpUqUKdd5dunTBoEGD0Lt3b1haWuYYHxsbi5YtW+LmzZsAgFOnTmntBhMZGQlvb2+kp6fDx8cHp06dgr29vTw+OTkZvr6+CA4OhpWVFSIiIrT2r587dy7mzZsHAFi0aBGmTJmiMT4oKAht2rRBeno6/P39ceLECaOXNT4+Hi4uLnj16hWcnZ2Nns6QS5cuoXHjxri45mc08qppunpvRqDxyA9x8eJFNGrUyGT1EhERmVNB/R5T0aT4N5OuXLkS7733XqEn6QBw8OBBvP/++1qTdAAoWbIkli5dKv+9e/dureWWLVsmPyZyxYoVGkk6ADg4OGDFihUAgPT0dCxfvjxHHWlpafjuu+8AZF1Z+PTTT3OUadGiBYYPHw4ACAgIwMWLFw0sIREREREpleITdaXz8/OTh2/fvp1jvBACv/32GwCgZs2aaN68udZ6mjdvjho1agAA9u/fn+PRkoGBgYiLiwMADB48GBYW2r+6IUOGyMN79+41djGIiIiISGGYqOdTamqqPKwteY6KipL7uvv6+uqtSzX+wYMHuHv3rsa406dP5yinjY+PDxwdHQEAZ86c0R88ERERESkWE/V8OnnypDxcs2bOPtjh4eF6x6tTH68+XW7qsbKykvu3Z6+DiIiIiIoOJur5kJmZiQULFsh/v//++znKREdHy8MeHh566/P09NQ6nfrfjo6OcHV1NaqeZ8+eISUlRW9ZIiIiIlImxT+eUcmWLVuG8+fPAwB69uwJHx+fHGUSEhLkYScnJ731qbqsAFmPYtRWj6E6tNVja2ubo0xKSopGEh8fH2+wXiIiIiIqPGxRz6OTJ09i+vTpAIDSpUvjp59+0lruzZs38rC+FyIB0EioX79+rbUeQ3UYqkflm2++gYuLi/xRb80nIiIiIvNjop4H165dQ8+ePZGeng5bW1vs3LlT58uX7Ozs5GH1G0+1UW/hzv4IR1U9huowVI/KjBkz8OrVK/mTvasNEREREZkXu77kUlRUFDp06ICXL1/C0tIS27dv1/sUlmLFisnD2buzZJeUlCQPZ+/ioqrHUB2G6lGxtbXV2iWGiIiIiJRBMS3qjRo1QuPGjREVFaXx//fv38f9+/eRkZFhpsj+38OHD9GuXTs8fPgQkiRhw4YN6Nmzp95p1G8gffDggd6y6q3a2buiqOpJSkqSn6duqJ5SpUoxGSciIiIqohSTqF++fBmXL1/O0ae6UqVKqFKlCm7cuGGmyLLExsaiffv2uHPnDoCsN4wOGjTI4HS1atWShyMiIvSWVR/v7e2dp3rS09PlFy9lr4OIiIiIig7FJOqSJAHIeuRhdtnf0lnYXr16hXfeeQfXr18HACxYsACffPKJUdNWrlwZ7u7uADSfua7NqVOnAADly5dHpUqVNMa1atVKHtZXT3BwsNz1pWXLlkbFSERERETKo5hE3cXFBUDO54ebW3JyMjp37oxLly4BAD777DNMmzbN6OklSUL37t0BZLWEnzt3Tmu5c+fOyS3l3bt3l09cVPz8/OR1tHnzZp0nL5s2bZKHDXXLISIiIiLlUkyiXrduXQDAV199hYiIiBx90rMnroUhNTUVPXv2xN9//w0AGD9+PL766qtc1zNhwgRYWWXdtzt27Ngc3Xtev36NsWPHAsh6s+iECRNy1GFjY4Nx48YByHrj6JIlS3KUCQoKwvr16wEAvr6+aNKkSa5jJSIiIiJlUMxTX/73v//h9OnTOHfuHGrXrq0xTgiBOnXq5LpOSZKQnp6e55j69euHP//8EwDQtm1bDB8+HGFhYTrL29jYwMvLK8f/e3l5YfLkyViwYAGCg4PRsmVLTJs2DVWrVsXt27excOFChISEAACmTJmC6tWra61/ypQp+PXXX3Hz5k1MnToVkZGR6Nu3L+zt7REQEID58+cjPT0d9vb2WL58eZ6Xm4iIiIjMTxLm7gCuZurUqVi2bJnJnvAiSVK+6sptK37FihVx9+5dreMyMzMxYsQIbNiwQef0w4cPx5o1a2BhoftCR2RkJDp16oRbt25pHe/s7IxffvkFXbp0yVXs8fHxcHFxwatXr+Ds7JyrafW5dOkSGjdujItrfkYjr5qmq/dmBBqP/BAXL15Eo0aNTFYvERGRORXU7zEVTYppUQeARYsWYdy4cQgICEBMTAxSUlIwb948SJKE0aNHo3Tp0uYOMc8sLCywfv169O7dG2vWrMGFCxcQGxuLkiVLokmTJhg1ahQ6duxosJ5q1aohJCQEP/zwA3bt2oXIyEikpqbC09MTnTp1wvjx41GxYsVCWCIiIiIiKkiKalHXxsLCApIkITQ0VOMRhWRabFEnIiIyP7aokzpFtahrU6FCBUiSBBsbG3OHQkRERERUaBSfqOvq801ERERE9G+mmMczEhERERHR/1N8i7q6Fy9eYOPGjTh27BjCwsLw4sULAICbmxvq1KmDdu3aYejQoXBzczNzpERERERE+VNkEvXVq1dj8uTJSE5OBgCNN3PGxMTg4cOH+PPPPzF37lwsXboUI0eONFeoRERERET5ViQS9QULFuCzzz6Tk3MXFxc0bNgQZcuWhRACT548QUhICF69eoWkpCR89NFHiIuLw9SpU80cORERERFR3ig+UQ8LC8OsWbMghEC5cuWwePFi9OnTB9bW1hrl0tPTsWvXLkyZMgUPHz7E559/js6dO+d4yykRERERUVGg+JtJV65ciYyMDJQqVQpBQUHo379/jiQdAKysrNCvXz8EBQWhdOnSyMjIwMqVK80QMRERERFR/ik+UT9x4gQkScKMGTNQoUIFg+U9PT0xbdo0CCFw/PjxQoiQiIiIiMj0FJ+ox8TEAADeeusto6dp2bIlAODhw4cFEhMRERERUUFTfKJuaWkJIKsPurFUZS0sFL94RERERERaKT6TVXV3yU03FlVZY7rKEBEREREpkeIT9fbt20MIgSVLliA0NNRg+atXr2Lx4sWQJAkdOnQohAiJiIiIiExP8Yn6hAkTYGtri8TERLRq1QqLFy9GbGxsjnKxsbFYtGgR2rRpg8TERNja2mLChAmFHzARERERkQko/jnqFStWxOrVqzF06FAkJiZi+vTpmDFjBipVqoQyZcpAkiQ8fvwYd+/ehRACQghIkoTVq1ez6wsRERERFVmKT9QBYNCgQShRogRGjx6NmJgYCCFw584dREVFAYD8xlIAcHd3x5o1a9CpUydzhUtERERElG9FIlEHgM6dOyMqKgr79u3DsWPHEBYWhhcvXgAA3NzcUKdOHbRr1w49e/aElVWRWSwiIiIiIq2KVEZrZWWFPn36oE+fPuYOhYiIiIioQCn+ZlIiIiIiov8iJupERERERArERJ2IiIiISIGYqBMRERERKRATdSIiIiIiBWKiTkRERESkQEzUiYiIiIgUiIk6EREREZECMVEnIiIiIlIgxb+ZdNiwYQCAjh078o2kRERERPSfofhEffPmzQCADz74wMyREBEREREVHsV3fSlVqhQAoEyZMmaOhIiIiIio8Cg+Ua9VqxYA4N69e2aOhIiIiIio8Cg+UR84cCCEEHIXGCIiIiKi/wLFJ+pDhw7F22+/jd9++w3z5s2DEMLcIRERERERFTjF30x6+vRpTJ48Gc+ePcMXX3yBHTt24IMPPkC9evVQvHhxWFpa6p2+TZs2hRQpEREREZHpKD5R9/PzgyRJ8t83b97El19+adS0kiQhPT29oEIjIiIiIiowik/UAbC7CxERERH95yg+UQ8ICDB3CEREREREhU7xibqvr6+5QyAiIiIiKnSKf+oLEREREdF/ERN1IiIiIiIFUnzXF3Xx8fHYvXs3goKC8PjxYyQnJ2PDhg2oWLGiXObhw4eIi4uDnZ0dqlSpYsZoiYiIiIjyrsgk6j/88AM+++wzJCQkAMh6EowkSUhKStIod/LkSQwYMAB2dnZ48OAB3NzczBEuEREREVG+FImuL3PnzsW4ceMQHx8PGxsbNG7cWGfZDz74AOXKlUNKSgr27NlTiFESEREREZmO4hP1kJAQ+QVHAwcOxOPHj3H+/Hmd5S0sLNCnTx8IIfDXX38VVphERERERCal+ER9xYoVEEKgRYsW2LJlC1xcXAxO06JFCwBAaGhoQYdHRERERFQgFJ+onzx5EpIkYcyYMUZPU6lSJQBATExMAUVFRERERFSwFJ+oP3r0CABQo0YNo6extbUFAKSkpBRITEREREREBU3xibqNjQ0AIC0tzehpVMm9q6trQYRERERERFTgFJ+oe3h4AACuXbtm9DR//vknAKBatWoFEhMRERERUUFTfKLetm1bCCGwceNGo8rfuXMH69evhyRJaN++fQFHR0RERERUMBSfqI8ZMwZWVlb4+++/MXfuXL1lg4OD0aFDByQmJsLW1hajRo0qnCCJiIiIiExM8Ym6l5cXZs2aBSEEvvzySzRr1gyLFi2Sxx85cgQLFy7E22+/jWbNmiEqKgqSJGHBggUoV66cGSMnIiIiIso7K3MHYIxZs2YhLS0N8+fPx4ULFxAcHAxJkgAAU6ZMkcsJISBJEmbPno1x48aZK1wiIiIionxTfIu6yhdffIFz586hV69esLe3hxBC42NtbY2OHTvi9OnTmDNnjrnDJSIiIiLKlyLRoq7i4+OD3bt3Iz09HdevX8fTp0+RkZGBEiVKoHbt2rC3tzd3iEREREREJlGkEnUVKysr1KtXz9xhEBEREREVmCLT9YWIiIiI6L+kyLWoP3nyBIGBgQgLC8OLFy8AAG5ubqhTpw78/PxQpkwZM0dIRERERJR/RSZRj46OxuTJk7F//36kp6drLWNpaYmePXti8eLFqFChgknm+/TpU5w/fx7nz5/HhQsXcOHCBTx//hwAMHjwYGzatClX9R05cgRr1qzB+fPn8ezZM5QqVQpNmzbFyJEj8e677xpVR3JyMn744Qfs2rULkZGRSE1NhaenJzp37oxx48aZbNmJiIiIyHyKRKJ++vRpdO3aFQkJCRBC6CyXnp6O3bt34+jRozh48CBatWqV73mbqoVeCIHRo0djzZo1Gv8fExODffv2Yd++fRg5ciRWrVolP3pSm9u3b6Nz5864ceOGxv9HREQgIiIC69atw7Zt29CpUyeTxE1ERERE5qH4PuoxMTHo2rUr4uPjIYRAx44dsWvXLty7dw9v3rzBmzdvcO/ePezevRudOnWCEALx8fHo2rUrHj58aNJYPD090aFDhzxN+/nnn8tJesOGDbF9+3acP38e27dvR8OGDQEAa9aswaxZs3TWkZiYiC5dushJ+ogRI3D8+HGcPXsWX3/9NZycnPDq1Sv06dMHV69ezVOcRERERKQMim9RX7BgAeLj42FpaYmNGzdi4MCBOcp4enrC09MTvXr1wrZt2zBo0CDEx8djwYIF+P777/M1/9mzZ6NJkyZo0qQJypQpg7t376Jy5cq5qiMyMlJ+m6qPjw9OnTolP0qySZMm6NatG3x9fREcHIyFCxdi6NChqFq1ao56lixZgoiICADAokWLNF721KJFC/j7+6NNmzZITk7GhAkTcOLEibwuNhERERGZmeJb1A8fPgxJkjBixAitSXp2/fv3x8iRIyGEwKFDh/I9/3nz5qFLly756gKzbNkyuV/9ihUrcjzv3cHBAStWrACQ1X1n+fLlOepIS0vDd999BwDw9vbGp59+mqNMixYtMHz4cABAQEAALl68mOeYiYiIiMi8FJ+oq7qv9OnTx+hpVGVN3fUlL4QQ+O233wAANWvWRPPmzbWWa968OWrUqAEA2L9/f46++IGBgYiLiwOQdROrhYX2r27IkCHy8N69e/MZPRERERGZi+IT9eLFiwMAXFxcjJ5GVVY1rTlFRUUhJiYGAODr66u3rGr8gwcPcPfuXY1xp0+fzlFOGx8fHzg6OgIAzpw5k5eQiYiIiEgBFJ+o+/j4AABCQ0ONnkZVVjWtOYWHh8vDNWvW1FtWfbz6dLmpx8rKSu7fnr0OIiIiIio6FJ+ojxs3DkIILFq0CMnJyQbLJycnY+HChZAkCWPHji2ECPWLjo6Whz08PPSW9fT01Dqd+t+Ojo5wdXU1qp5nz54hJSUlN+ESERERkUIoPlFv164d5syZg/DwcPj5+eHy5cs6y165cgX+/v64ceMG5syZg/bt2xdeoDokJCTIw05OTnrLqrqsAFmPYtRWj6E6DNWjkpKSgvj4eI0PERERESmHYh7P+MUXX+gcJ0kSfHx8EBwcjMaNG6Nu3bpo0qQJSpcuDUmS8OTJE1y4cCFHl5cvvvgCs2fPLpT4dXnz5o08bGNjo7esra2tPPz69Wut9Riqw1A9Kt988w3mzZtnsC4iIiIiMg/FJOpz587V+0ZOICthF0IgNDRUa591IQQkSUJwcDCCg4MBwOyJup2dnTycmpqqt6x6N5Xsj3BU1WOoDkP1qMyYMQOTJk2S/46Pj9foekNERERE5qWYRB1AjkcS5qWcsXUUlmLFisnDurqhqCQlJcnD2bu4qOoxVIehelRsbW01Wt6JiIiISFkU00c9MzOzQD7mpn4D6YMHD/SWVb+BNHvrtqqepKQk+XnqhuopVaoUk3EiIiKiIkoxifq/Va1ateThiIgIvWXVx3t7e+epnvT0dNy+fVtrHURERERUdDBRL2CVK1eGu7s7AODkyZN6y546dQoAUL58eVSqVEljXKtWreRhffUEBwfLXV9atmyZl5CJiIiISAGYqBcwSZLQvXt3AFkt4efOndNa7ty5c3JLeffu3XPcWOvn5ye/cXXz5s06++Jv2rRJHu7Zs2d+wyciIiIiMymSiXp8fDxiYmJw//59gx8lmDBhAqyssu7bHTt2bI5HJr5+/Vp+OZOVlRUmTJiQow4bGxuMGzcOQNYbR5csWZKjTFBQENavXw8A8PX1RZMmTUy5GERERERUiBT11Bd9/vzzT/z00084ffo0Xr58adQ0kiQhPT09X/M9c+YMIiMj5b9jY2Pl4cjISI0WbAAYMmRIjjq8vLwwefJkLFiwAMHBwWjZsiWmTZuGqlWr4vbt21i4cCFCQkIAAFOmTEH16tW1xjJlyhT8+uuvuHnzJqZOnYrIyEj07dsX9vb2CAgIwPz585Geng57e3ssX748X8tNREREROYlCaU9z1CL0aNHY+3atQBy9/hFSZKQkZGRr3kPGTIEmzdvNrq8rvgyMzMxYsQIbNiwQee0w4cPx5o1a2BhoftCR2RkJDp16oRbt25pHe/s7IxffvkFXbp0MTpmIOsqhYuLC169egVnZ+dcTavPpUuX0LhxY1xc8zMaedU0Xb03I9B45Ie4ePEiGjVqZLJ6iYiIzKmgfo+paFJ8i/rq1auxZs0aAFnPEu/Zsyfq168PV1dXvQmt0lhYWGD9+vXo3bs31qxZgwsXLiA2NhYlS5ZEkyZNMGrUKHTs2NFgPdWqVUNISAh++OEH7Nq1C5GRkUhNTYWnpyc6deqE8ePHo2LFioWwRERERERUkBTfot64cWOEhITA29sbJ06cQJkyZcwd0r8SW9SJiIjMjy3qpE7xTdIRERGQJAlz585lkk5ERERE/xmK7/ri5OSEN2/e6LzBkv7bwsPDTV5nyZIlUaFCBZPXS0RERJQbik/Uvb29cfr0aTx69AgNGjQwdzikEI+ex8LCwgIDBw40ed0ODg4IDw9nsk5ERERmpfhEfeTIkTh16hS2b99u1M2W9N8Ql5iIzMxMbP3sC3hXrGyyesPvRWHg17MRGxvLRJ2IiIjMSvGJev/+/bFv3z788ssvaNq0KcaMGWPukEhBvCtWNulNqkRERERKofhEHQB++eUXTJ48GRMmTMCOHTvw/vvvw8vLCw4ODganbdOmTSFESERERERkWkUiUbe2tkb9+vXh5uaGoKAgBAUFGTWdKd5MSkRERERkDopP1NPT09GvXz/s3bsXQO7eTEpEREREVFQpPlFftWoV9uzZAwCoVKkSBg8ejHr16hW5N5MSEREREeWG4hP1devWAQCaN2+O48ePw97e3swREREREREVPMU3SUdGRkKSJMyYMYNJOhERERH9Zyg+UXd0dAQAeHp6mjkSIiIiIqLCo/hEvX79+gCAe/fumTkSIiIiIqLCo/hEffTo0RBCYP369eYOhYiIiIio0Cg+Ue/VqxdGjx6NgwcPYvLkycjIyDB3SEREREREBU7xT33ZsmULmjdvjpCQECxbtgy7d+9Gr169jH4z6aBBgwohSiIiIiIi01J8oj5kyBBIkiT/HR0dje+++86oaSVJYqJOREREREWS4hN1gG8jJSIiIqL/HsUn6lFRUeYOgYiIiIio0Ck+Ua9YsaK5QyAiIiIiKnSKf+oLEREREdF/ERN1IiIiIiIFYqJORERERKRAiu+jXqVKlTxPK0kSbt++bcJoiIiIiIgKh+IT9bt37xpdVpIkjUc5qj9/nYiIiIioKFF8oj548GCDZZKSknDjxg2EhoZCkiQ0bNgQdevWLYToiIiIiIgKhuIT9Y0bNxpdNiwsDMOGDcO1a9cwY8YM9O7duwAjIyIiIiIqOP+qm0nr1KmDEydOoFy5chg0aBAiIiLMHRIRERERUZ78qxJ1AHBycsKkSZPw+vVrLF682NzhEBERERHlyb8uUQeAJk2aAACOHz9u5kiIiIiIiPLmX5moq5788uTJEzNHQkRERESUN//KRP3o0aMAABcXFzNHQkRERESUN/+6RH3Hjh345ptvIEkSWrVqZe5wiIiIiIjyRPGPZxw2bJjBMpmZmXj58iUuXbqEhw8fQggBKysrTJ8+vRAiJCIiIiIyPcUn6ps2bTL6DaOqvunOzs5Yt24dfHx8CjI0IiIiIqICo/hEvUKFCgYTdQsLCxQrVgyVK1eGr68vBg4ciJIlSxZShEREREREpqf4RP3u3bvmDoGIiIiIqND9624mJSIiIiL6N2CiTkRERESkQEzUiYiIiIgUiIk6EREREZECKepmUktLS5PWJ0kS0tPTTVonEREREVFhUFSirnoOOhERERHRf52iEvXBgwfna3ohBA4fPoznz58z6SciIiKiIk1RifrGjRvzPO3+/fsxZ84cPH/+XP4/T09PU4RFRERERFToivzNpIcPH4aPjw969+6NsLAwCCFQtmxZrFixAjdv3jR3eEREREREeaKoFvXc+OuvvzB79mycP38eQFa3l9KlS2PatGn46KOPYGdnZ+YIiYiIiIjyrsgl6idPnsSsWbPw999/A8hK0EuUKIEpU6ZgzJgxcHBwMHOERERERET5V2QS9bNnz2L27NkICAgAkJWgu7q6YtKkSZgwYQKcnJzMHCH9m4SHh5u8zpIlS6JChQomr5eIiIj+nRSfqF+4cAGzZ8/Gn3/+CSArQS9WrBgmTJiASZMmwcXFxcwR0r/Jo+exsLCwwMCBA01et4ODA8LDw5msExERkVEUm6hfvnwZc+bMwcGDBwFkJeiOjo4YM2YMpkyZAjc3NzNHSP9GcYmJyMzMxNbPvoB3xcomqzf8XhQGfj0bsbGxTNSJiIjIKIpL1K9du4Y5c+Zg3759ALISdHt7e3z00UeYPn06SpYsaeYI6b/Au2JlNPKqae4wiIiI6D9MUYl6//79sXPnTgghIISAra0tRo0ahenTp6Ns2bLmDo+IiIiIqNAoKlHfsWOHPFyqVCmMHz8eHh4ecv/0vBg0aJApQiMiIiIiKlSKStQBQJIkAEBsbCxmzZqV77qYqBMRERFRUaS4RF0IYe4QiIiIiIjMTlGJuuoZ6URERERE/3WKStR9fX3NHQIRERERkSJYmDsAypv79+9j8uTJ8Pb2hqOjI9zc3NC0aVMsWbIEycnJ5g6PiIiIiPJJUS3qZJxDhw5hwIABePXqlfx/ycnJuHDhAi5cuIB169bh8OHDqFKlihmjJCIiIqL8YKJexFy5cgXvv/8+kpOT4eTkhBkzZsDf3x+vX7/Gjh07sHbtWty4cQOdO3fGhQsX4OTkZO6QSU14eLjJ6yxZsiTfdkpERPQvxES9iJkwYQKSk5NhZWWFP//8Ey1atJDHtW3bFtWrV8fUqVMRERGBb7/9FrNnzzZjtKTy6HksLCwsMHDgQJPXbWdnh927d6NcuXImrZcnAERERObFRL0IuXDhAgIDAwEAw4cP10jSVT799FNs3LgR4eHhWL58OWbMmAFra+tCjpSyi0tMRGZmJrZ+9gW8K1Y2Wb2nr4Zg0o/L0aVLF5PVqcITACIiIvNiol6E7N+/Xx4eOnSo1jIWFhYYNGgQZsyYgZcvXyIwMBDt27cvpAjJEO+KldHIq6bJ6gu/d7fInQA4ODggPDycyToREZEBTNSLkNOnTwMAHB0d0bhxY53l1B9zeebMGSbq/wFF5QQg/F4UBn49G7GxsUzUiYiIDGCiXoSobkSsVq0arKx0f3U1a/5/wlYQNy/Sf4epTwCIiIjIeEzUi4g3b94gNjYWAODh4aG3bPHixeHo6IikpCRER0cXRnhEuVIQJ5ApKSmwtbU1eb1FrU/9/fv35WOFKRW19UCkjvsFFVVM1IuIhIQEediYRy6qEvXExESt41NSUpCSkiL/rXome3x8fD4j1aSa/8Wb4Uh8/dpk9Ybfi2K9RbDeoGtXIElSgTz9xkKSkCmEyeu1s7PDli1bUKZMGZPWa2FhgczMTJPW+eTJEwweNAiv37wxab1A0VoPrJf1qivI/cLe3h4XLlyAp6enyepU/Q6LAjieUdEjCW4JRUJ0dLR81v7hhx9iy5YtestXqFAB0dHRqFq1KiIjI3OMnzt3LubNm1cgsRIREVH+REdHG7yCTv9+bFEvIuzs7OTh1NRUg+VVreX29vZax8+YMQOTJk2S/87MzMSLFy9QokQJSJKUz2izxMfHw9PTE9HR0XB2djZJnURFCfcB+i/j9p83QggkJCTA3d3d3KGQAjBRLyKKFSsmD+vqzqIuKSkJgO5uMra2tjn687q6uuY9QD2cnZ15kKb/NO4D9F/G7T/3XFxczB0CKYSFuQMg49jZ2aFkyZIAgAcPHugt+/LlSzlRN2W/OSIiIiIqPEzUixBvb28AQGRkJNLT03WWi4iIyDENERERERUtTNSLkFatWgHI6tZy8eJFneVOnjwpD7ds2bLA49LF1tYWc+bMKZBH5hEVBdwH6L+M2z9R/vGpL0XI+fPn0axZMwDAqFGjsGrVqhxlMjMzUadOHYSHh8PV1RVPnz6FtbV1YYdKRERERPnEFvUipGnTpmjdujUAYP369QgKCspRZunSpfLLZMaPH88knYiIiKiIYot6ERMSEoKWLVvi9evXcHJywsyZM+Hv74/Xr19jx44dWLNmDQDAy8sLwcHBGk+LISIiIqKig4l6EfT7779j4MCBOt8i6uXlhUOHDqFatWqFHBkRERERmQq7vhRBXbt2xdWrVzFx4kR4eXnBwcEBrq6u8PHxwcKFCxESEmLWJP3+/fuYPHkyvL294ejoCDc3NzRt2hRLlixBcnKy2eIi0ubSpUuYP38+OnbsCE9PT9ja2sLJyQleXl4YMmQITp8+nav6jhw5gl69esHDwwO2trbw8PBAr169cOTIEaPrSE5OxuLFi9G0aVO4ubnByckJ3t7emDx5Mu7fv5/bRSTKk6lTp0KSJPkTGBhocBpu/0QmJohM6ODBg8LFxUUA0PqpUaOGuH37trnDJBJCCNGmTRud26r658MPPxQpKSl668rMzBQjR47UW8/IkSNFZmam3noiIyNFjRo1dNbh4uIiDh06ZMrVQJTD5cuXhZWVlca2FxAQoLM8t3+igsFEnUzm8uXLwsHBQQAQTk5O4uuvvxZnz54Vx48fFyNGjJAPtDVr1hQJCQnmDpdIVK1aVQAQ7u7uYvz48WL37t3i/PnzIigoSHz77beifPny8nbbr18/vXXNnDlTLtuwYUOxfft2cf78ebF9+3bRsGFDedxnn32ms46EhARRs2ZNueyIESPE8ePHxdmzZ8XXX38tnJycBADh4OAgrly5YurVQSSEECIjI0M0adJEABClS5c2KlHn9k9UMJiok8n4+fkJAMLKykqcPXs2x/hFixbJB+B58+aZIUIiTZ07dxa//vqrSE9P1zr+2bNnwsvLS95uT506pbXcrVu35NZHHx8fkZycrDE+KSlJ+Pj4yPtHZGSk1nrmzJkjz2vRokU5xp89e1aej7+/fy6Xlsg4y5YtkxtVZsyYYTBR5/ZPVHCYqJNJnD9/Xj7Ajho1SmuZjIwM4e3tLQCI4sWLi9TU1EKOkij3fv/9d3nbHjdunNYyH3/8sVwmKChIa5mgoCC5zJgxY3KMT01NFa6urgKA8Pb2FhkZGVrrGTVqlFxPcHBw3heMSIv79+/LLdcBAQEaybOuRJ3bP1HB4c2kZBL79++Xh4cOHaq1jIWFBQYNGgQAePnypVE3JhGZm5+fnzx8+/btHOOFEPjtt98AADVr1kTz5s211tO8eXPUqFEDQNb+IrI9cCswMBBxcXEAgMGDB8PCQvvheciQIfLw3r17jV0MIqN8/PHHSExMxODBgzW2fV24/RMVLCbqZBKqJ2M4OjqicePGOsv5+vrKw2fOnCnwuIjyKzU1VR7WljxERUUhJiYGgOb2rY1q/IMHD3D37l2NcepPl9FXj4+PDxwdHQFwHyLT2rlzJw4ePAg3NzcsXrzYqGm4/RMVLCbqZBKqt6FWq1YNVlZWOsvVrFkzxzRESnby5El5WH37VVHfjrWNV6dv+ze2HisrK1StWlVrHUR5FRcXh/HjxwMAFi5ciFKlShk1Hbd/ooLFRJ3y7c2bN4iNjQUAeHh46C1bvHhxuTUkOjq6wGMjyo/MzEwsWLBA/vv999/PUUZ9Oza0/Xt6emqdTv1vR0dHuLq6GlXPs2fPkJKSorcskTGmTp2Kx48f46233sLw4cONno7bP1HBYqJO+ZaQkCAPOzk5GSyvStQTExMLLCYiU1i2bBnOnz8PAOjZsyd8fHxylMnN9q/a9oGc27+qntzsQ9rqIcqtM2fOYN26dbCyssKqVasgSZLR03L7JypYTNQp3968eSMP29jYGCxva2sLAHj9+nWBxUSUXydPnsT06dMBAKVLl8ZPP/2ktVxutn/Vtg/k3P5V9eRmH9JWD1FupKamYuTIkRBCYOLEiahbt26upuf2T1SwmKhTvtnZ2cnD6jfe6aK6VGlvb19gMRHlx7Vr19CzZ0+kp6fD1tYWO3fuRJkyZbSWzc32r36ZPvv2r6onN/uQtnqIcmP+/PkIDw9HhQoVMGfOnFxPz+2fqGAxUad8K1asmDxszGXIpKQkAMZd4iQqbFFRUejQoQNevnwJS0tLbN++Xe9TKHKz/au2fSDn9q+qJzf7kLZ6iIwVERGBb775BgCwYsUKjS4lxuL2T1SwdD+eg8hIdnZ2KFmyJGJjY/HgwQO9ZV++fCkfZNVvLCJSgocPH6Jdu3Z4+PAhJEnChg0b0LNnT73TqN9AZ2j7V7+BLvv27+HhgX/++QdJSUmIi4vTe0Odqp5SpUppdAMgyo1ly5YhNTUVVapUQXJyMnbs2JGjTFhYmDx84sQJPH78GADQtWtXODo6cvsnKmBM1MkkvL29cfr0aURGRiI9PV3nIxojIiI0piFSitjYWLRv3x537twBkNXCqHpBlz61atWSh9W3b230bf+1atXCnj175HK6XhyTnp4uv3iJ+xDlh6oLyZ07d9CvXz+D5b/88kt5OCoqCo6Ojtz+iQoYu76QSbRq1QpA1iXJixcv6iyn/kzqli1bFnhcRMZ49eoV3nnnHVy/fh0AsGDBAnzyySdGTVu5cmW4u7sD0Ny+tTl16hQAoHz58qhUqZLGONU+ZKie4OBg+aoU9yEyN27/RAWLiTqZRI8ePeThjRs3ai2TmZmJLVu2AABcXV3h7+9fGKER6ZWcnIzOnTvj0qVLAIDPPvsM06ZNM3p6SZLQvXt3AFktgefOndNa7ty5c3KLYvfu3XM8As/Pzw8uLi4AgM2bN+d4xbrKpk2b5GFD3XKI9Nm0aROEEHo/6jeYBgQEyP+vSrS5/RMVMEFkIq1btxYAhJWVlTh79myO8YsWLRIABAAxZ86cwg+QKJuUlBTRoUMHebscP358nuq5ceOGsLKyEgCEj4+PSE5O1hifnJwsfHx85P3j5s2bWuuZNWuWHMuiRYtyjD979qw8H19f3zzFSpQbc+bMkbfJgIAArWW4/RMVHCbqZDKXLl0S9vb2AoBwcnIS8+fPF0FBQeLEiRNi5MiR8gHYy8tLxMfHmztcItGrVy95u2zbtq24evWqCA0N1fm5ceOGzrqmT58u19WwYUOxY8cOceHCBbFjxw7RsGFDedyMGTN01hEfHy+8vLzksiNHjhQnTpwQQUFBYv78+cLJyUkAEPb29iIkJKQA1giRJmMSdSG4/RMVFCbqZFIHDhwQzs7O8oE2+8fLy0vcunXL3GESCSGEzu1U16dixYo668rIyBDDhg3TO/3w4cNFRkaG3phu3bolqlevrrMOZ2dn8fvvv5t4TRBpZ2yizu2fqGCwjzqZVNeuXXH16lVMnDgRXl5ecHBwgKurK3x8fLBw4UKEhISgWrVq5g6TyOQsLCywfv16HDp0CN27d4e7uztsbGzg7u6O7t274/Dhw1i3bh0sLPQfdqtVq4aQkBAsXLgQPj4+cHV1hYODA2rUqIGJEyfi6tWr6NKlSyEtFZFxuP0TFQxJCB13bBARERERkdmwRZ2IiIiISIGYqBMRERERKRATdSIiIiIiBWKiTkRERESkQEzUiYiIiIgUiIk6EREREZECMVEnIiIiIlIgJupERERERArERJ2IiIiISIGYqBMRERERKRATdSIiIiIiBWKiTkSKJUkSJEnC3LlzzR2Kov38889o06YNihcvDgsLC0iShAYNGpg7LCIiyicm6kQKFBgYKCepkiShb9++BqcZMmSIXJ7+O6ZOnYpBgwbh9OnTiIuLgxDC3CEREZGJMFEnKgJ27tyJ0NBQc4dBChMdHY1vv/0WANC8eXMcPHgQV65cQWhoKPbs2WPm6P477t69K58kb9q0ydzhENG/iJW5AyAiw4QQmDNnDvbu3WvuUEhBAgICkJGRAQBYt24dateubeaIiIjIlNiiTqRwJUuWBADs27cPly5dMnM0pCQxMTHysJeXlxkjISKigsBEnUjhxo8fD1tbWwDAnDlzzBwNKUlKSoo8bG1tbcZIiIioIDBRJ1I4Dw8PjBw5EgBw8OBB/PPPP3mqp1KlSpAkCUOGDNFbTnVTaqVKlXKM09YXd+/evejQoQNKly4NR0dH1K9fHytWrEBaWpo8nRAC27Ztg5+fH0qXLg0HBwc0atQIq1atytXNj8eOHUO3bt1Qrlw52NnZoUqVKhgzZgwePHhg1PQ3btzAuHHjULt2bbi4uMDe3h5VqlTB0KFD9V6tUL+5NzAwEJmZmdiwYQP8/f1RpkwZWFhYGFyvuty9excTJ05E7dq1UaxYMTg4OKB69eoYNWqUzvsSVN/lvHnz5P9Tv/lYkiTcvXs317FkZmZi69at6NSpE8qWLQsbGxuUKlUK/v7++PHHH5Gamqpz2rlz52rczBwXF4c5c+agdu3acHJygpubG/z8/PDLL78YFUt6ejrWr1+PTp06wd3dHba2tihZsiTatGmD5cuX482bNzqn9fPzgyRJ8PPzAwDcunULY8aMQfXq1eHg4JBj/Tx69Ag//vgj3nvvPVSvXh2Ojo6wtbVF+fLl0b17d/z666/IzMzUOi9JklC5cmX576FDh+b4LrQ9tciU6/rVq1f48ssv0bBhQ7i6umrtK79v3z706NEDHh4esLW1RbFixVClShW0bt0as2bNwvnz53XOj4jMSBCR4gQEBAgAAoDYuHGjePjwobC3txcARIcOHbROM3jwYHkabSpWrCgAiMGDB+udt6qeihUr5hgXFRWlEddHH30k/53906tXL5Geni7evHkj3nvvPZ3lRowYoTMWVZk5c+aIuXPn6qzD2dlZnDx5Uu9yffHFF8LKykpnHZIkidmzZ2udVv37+OOPP0S7du1yTG9ovWqzefNmYWtrqzMmS0tLMX/+/BzTqb5LfZ+oqKhcxfL8+XPRsmVLvXV6e3uLu3fvap1+zpw5crk7d+6IqlWr6qznvffeE2lpaTpjiYyMFLVq1dIbS/Xq1cXNmze1Tu/r6ysACF9fX7F//37h6Oioc/2kp6cLCwsLg+uzffv2IiEhIce8DE2n2n4Lal3fvHlTVKpUKcf0GzdulJevT58+BmNs3Lixzu+DiMyHiTqRAmVP1IUQYtKkSfL/nT59Osc0hZ2oN2vWTAAQnTp1Env37hUXL14U+/fvl/8fgFi7dq0YO3asACD69+8vDh48KC5evCh27NghatasqZH8aqMa7+PjIwCIGjVqiPXr14sLFy6IY8eOiVGjRslJVrFixXQmNrNmzZLreuutt8S6detEUFCQCA4OFr/88oto0aKFPP7777/PMb3691GvXj0BQHTr1k1e7sOHD4sdO3boXa/ZHTx4UEiSJAAIJycnMWfOHHH69GkRFBQkli5dKkqWLCnP88cff9SY9saNGyI0NFTjRCk0NFTjk5qaanQs6enpGuvA19dX7Nq1SwQHB4sDBw6IHj16yOOqVq2qNWFVTx6bNGkiLCwsxOjRo8WxY8fEhQsXxPr164WXl5dcZuzYsVpjefjwoShTpoz8nX766afijz/+EJcuXRIBAQFixowZwsHBQQAQVapUEXFxcTnqUCXqlStXFk5OTqJUqVJiwYIF4u+//xbnzp0TK1asEM+ePRNCCJGWliYsLCxE27ZtxeLFi8WRI0fExYsXRWBgoNiwYYPGehk0aFCOeYWGhoqjR4/KZb766qsc38WTJ08KbF3Xq1dPWFtbi7Fjx4q//vpLBAcHi+3bt4uzZ88KIYRYsWKFXLZVq1Zi06ZN4vTp0yIkJEQcP35cfPfdd+Ldd98VTZs2NW5jIaJCxUSdSIG0JepPnz6VWwb9/f1zTFPYiToAMWHChBxlkpKS5Ba+kiVLCkmSxPLly3OUe/TokShWrJic9GqjPq9GjRppTVq2bNmi0VKb3fnz5+Vk/vPPP9c6n4yMDDFw4EA5OXz58qXGePXvA4CYNWuW1nqMlZqaKsqXLy8n6SEhITnK3L17V5QrV04AEA4ODnJiqU49YcuPlStXaiSjmZmZOcrMnDlTLjN16lS9sQAQ27Zty1EmPj5e1K9fXwAQFhYW4urVqznKdOnSRQAQnp6e4vbt21rjvXTpkrwvaPtOVYk6AOHu7i7u3bunc9kzMzPFrVu3dI4XQojZs2cLIOuqi7ZW/OxXmvQx9bq2sLAQf/75p875tW7dWj6x1ncV4/nz53rjJiLzYKJOpEDaEnUhhJg2bZr8/ydOnNCYprATdU9PT52ttuqJRPPmzXXOa9CgQQKAKF68uNbx6olfcHCwzno6duwoAAgrKyvx8OFDjXG9e/eWL+1rS4pUXr58KXdDWbt2rcY49e/Dy8tLpKen66zHGL/++qtc3zfffKOz3NatW+VyixYtyjHeVIm6t7e3fGIVHx+vtUx6erp8FaR48eLizZs3OmPp0qWLznn9888/crmPP/5YY1xoaKg87rffftMb89SpU+VEPDv1RH3Lli166zFGenq6fIVjyZIlOcbnJlE39boeNmyY3vlVr15dABATJ07Uv5BEpEi8mZSoCJkyZQqKFSsGAJg1a5ZZY+nVq5fOJ43Uq1dPHv7ggw901lG/fn0AwMuXLxEXF6ezXN26ddG4cWOd44cNGwYg6wbEwMBA+f/T0tLwxx9/AADee+89vW9tdXV1Rd26dQEAQUFBOst98MEHsLS01DneGMeOHQOQdSOiKnZt+vTpAxcXF41pTO3hw4cIDw8HALz//vvy9pWdpaUlhg4dCiDr+9J3862qnDZNmzaVn/eefZl+++03AICDgwM6d+6sN+42bdrI8UdHR2stY2Njgz59+uitJ7vMzEw8fPgQN27cQFhYGMLCwhAeHg4PDw8AwJUrV3JVn7qCWNcDBgzQO89y5coBAH7//XfExsbmJWwiMiMm6kRFSIkSJTBhwgQAwN9//42jR4+aLRZ9z+12dXXNdbmEhASd5Zo0aaI3lqZNm8rDYWFh8vD169eRnJwMAJgxY0aOp3Fk/wQHBwMAHj9+rHNe6icheaWKsVKlSihdurTOcjY2NmjYsGGO5TIl9XqbNWumt6z6eH3xGPt93bp1S+PpJqr1n5ycDCsrK73fVZcuXeTpdH1f1atXh52dnd5YAEAIga1bt8Lf3x9OTk4oX748atasibp168qfy5cvA0C+kt2CWNeGtsfBgwcDACIjI1GtWjUMGzYM27dvN/pJSURkXkzUiYqYSZMmyQnu7NmzzRaHg4ODznEWFha5Lqd6w6Y2+pJZAChTpow8/OLFC3n46dOneqfTRZXca1O8ePE81alOFaN63LqULVtWYxpTU6/XUDyqWAzFY+z3JYTAy5cv5f839fdlzHf15s0bdO7cGR9++CECAwPx+vVrveUNjdenINa1oWUcNmwYZs6cCSsrK7x69QobN25E//794enpiWrVqmHy5Mm4c+eOkUtARIXNytwBEFHuuLq6YtKkSZg9ezbOnz+PgwcParQu/hvp67Kij3ryv3jxYrz77rtGTefo6KhzXH67vagzZrlELp4zn1+G4jE2lrzWo/q+KleujAMHDhg1L1V5bYz5rr7++mu5e5Svry8++eQTNGrUCGXLloW9vb18MtmmTRucPn3aZN+Hqda1scs4cuRI/PLLLzh+/DjOnTuH5ORk3L59G0uXLsX333+P77//HqNHjzZqnkRUeJioExVBEyZMwHfffYfnz59j9uzZBvvzAv/feq3rxS0qSUlJJonRlJ48eWL0eDc3N3m4RIkS8nBaWhrq1Klj+uDyQBWjvi42KqplU1+ugojFmHh0rWdt5Tw9PXWOV7WcS5Kk0SKs+r6ePHmCmjVrwsqqYH+ihBBYt24dAKBVq1Y4ceKExlUedeot/3lVEOvaWBUrVsTMmTMxc+ZMpKWl4fz589i1axdWr16NN2/e4OOPP0azZs3krlZEpAzs+kJUBBUrVgxTpkwBAISEhGD//v1GTQMYTjhu3LiR7/hM7cKFC0aPV0/Ga9euDRsbGwDAn3/+WTDB5YEqxrt37+rt7pGWloaQkBCNaQoqFgAG33qr/vZKffEY+31Vr15d/n4AyElicnIy/v77b711mMKLFy/khPn999/XmaQnJibq3S+MveJTEOs6L6ytrdGyZUssX74c27ZtA5B10rJ7926TzoeI8o+JOlERNWbMGLkv8Jw5cwxeKld1D7h06ZLOsmFhYTpfW29OoaGhcsKqzYYNGwBkdQNQvTYeyOof//bbbwMAAgMDFfOa9Hbt2gHISo5UsWuze/duvHr1SmMaU3N3d4e3tzcAYNeuXTpv6s3IyJBfS1+8eHE0atRIZ52bN2/WOS44OFi+OTL7MnXv3l0eXrRokVHx50d6ero8rO++hPXr1yMtLU3nePUbVlNSUnSWK4h1nV+q/QPI342yRFQwmKgTFVGOjo6YNm0agKxE9vDhw3rL+/r6Ash6RNz27dtzjE9ISND7qEBzGzlypNZuOdu2bZOXvUePHvLj6FQ+++wzucWzb9++uH37ts55ZGRkYNu2bQX+RIyePXvC3d0dADB//nytj/yLjo7G5MmTAWSdcOh75GF+ffLJJwCAZ8+eYezYsVpP5ObNm4fr168DAEaMGAFbW1ud9R04cAA7d+7M8f+JiYkYOXIkgKyuWKNGjdIY36RJE3To0AEAcPjwYcyZM0dv3Hfv3tW6LRurVKlS8o3ZO3bs0HgCjcqFCxfw+eef662nRIkS8pUBfdsXYPp1bcjWrVs1TkiyU7/SpKuvPxGZkTke3k5E+ul64VF2ycnJ8tsr1T/aPH36VDg7OwsAws7OTsybN0+cO3dO/PPPP+KHH34QVatWFXZ2dqJhw4ZGvfBIX1zq8QcEBOgst3HjRrlcVFRUjvGqcT4+PgKAqFmzpti4caMIDg4Wx48fFx999JH81tFixYpprUMIzRfEODk5ifHjx4tDhw6JS5cuiaCgILF9+3Yxbtw44e7uLgCI0NDQPC1Pbhw8eFBIkiTHNG/ePHHmzBlx7tw58e2334rSpUvL8/zxxx8NLld+ZH+tvZ+fn9i1a5e4ePGiOHjwoOjVq5c8zpjX2vv4+AhLS0vx8ccfixMnTojg4GCxYcMGUaNGDbnM2LFjtcYSExOjsU03a9ZMrF69Wpw9e1ZcunRJ/PXXX2Lp0qWiffv2wtLSUvTu3TtHHaoXHvn6+hpc9k8++USeV9OmTcX27dvFhQsXxLFjx8SkSZOEnZ2dKFmypPDy8tJbZ8uWLQUAUaJECbFt2zZx/fp1cevWLXHr1i2Nt36ael0bAkCUKVNGfPTRR+Lnn3+W1+Mff/whJk2aJOzt7eVtMDo62mB9RFS4mKgTKZCxiboQQqxYscKoRF0IIXbu3CksLS1zlFcl7zt37jT6zaSFmajPmTMnxyvq1T/Ozs4iMDBQz1oSYtmyZfKbR/V9bGxscrxSviASdSGE2LRpk96YLC0txfz583VOb6pEXYisV8irkk1dH29vb3H37l2Dsdy5c0dUrlxZZz29e/fW+zr7u3fviiZNmhj8rgCIoUOH5pg+N4l6XFycaNCggc763dzcxMmTJw3WqX7ilf0zZ86cAlvXhhizDl1dXcXRo0cN1kVEhY9dX4iKuBEjRuh9woa6Pn364OzZs+jZsydKlSoFGxsbeHp6YvDgwQgODs71WxwL09y5c3HkyBF07twZZcqUgY2NDSpVqoSPP/4Y165dk7v26DJhwgTcvn0bs2bNQvPmzVGyZElYWVnB0dERXl5e6N27N1atWoWYmBhUq1atUJZp8ODBiIiIwPjx4+Ht7Q1HR0fY29ujatWqGDFiBEJCQjBjxoxCicXNzQ2nTp3Czz//jHfffRdlypSBtbU1SpQoAT8/P6xcuRKXL19GxYoVDdZVuXJlXLx4ETNnzoS3tzccHBzg4uKCNm3aYOvWrdi9e7feJ7pUrFgR//zzD/bt24e+ffuicuXKcHBwgLW1NUqVKoW33noLn376KU6ePIn169fna7ldXFzw999/48svv0TdunVhZ2cHJycneHt7Y/Lkybhy5Yr8FlR9OnfujOPHj6N79+5wd3fX+dZewLTr2pCIiAisWLECPXr0QK1atVCiRAlYWVmhePHiaN68OebOnYsbN27IXY6ISFkkIQrxIb1ERPSvNHfuXMybNw8ACvXZ70RE/2ZsUSciIiIiUiAm6kRERERECsREnYiIiIhIgZioExEREREpEBN1IiIiIiIF4lNfiIiIiIgUiC3qREREREQKxESdiIiIiEiBmKgTERERESkQE3UiIiIiIgViok5EREREpEBM1ImIiIiIFIiJOhERERGRAjFRJyIiIiJSICbqREREREQK9H+U1PCQXYbFMAAAAABJRU5ErkJggg==",
      "text/plain": [
       "<Figure size 640x480 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "import matplotlib.pyplot as plt\n",
    "\n",
    "# Plotting\n",
    "plt.rc('font', size=20)\n",
    "plt.hist(op_per_workflow.groupby('workflow')['operator'].sum(), bins=20, color='pink', edgecolor='black')\n",
    "plt.title('Overall utilization of operators in workflows')\n",
    "plt.xlabel('Number of operators')\n",
    "plt.ylabel('Number of workflows')\n",
    "plt.tight_layout()\n",
    "plt.savefig('./plots/number_of_operators_workflows.png', format=\"png\", bbox_inches='tight')\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import re\n",
    "import csv\n",
    "import pandas as pd\n",
    "\n",
    "def find_operators_in_processes(directory, pattern_list, output_file_name):\n",
    "    result_list = []\n",
    "\n",
    "    for pipeline in os.listdir(directory):\n",
    "        subdirectory_path = os.path.join(directory, pipeline)\n",
    "        folder_name = os.path.basename(subdirectory_path)\n",
    "\n",
    "        for root, dirs, files in os.walk(subdirectory_path):\n",
    "            for filename in files:\n",
    "                file_path = os.path.join(root, filename)\n",
    "                \n",
    "                if filename.endswith('.nf') or filename.endswith('.config'):\n",
    "                    with open(file_path, 'r', encoding='utf-8', errors='ignore') as file:\n",
    "                        lines = file.readlines()\n",
    "                    \n",
    "                    process_flag = False\n",
    "                    process_name = None\n",
    "                    brace_stack = []\n",
    "\n",
    "                    process_pattern = r'process\\s+(\\w+)\\s*{'\n",
    "\n",
    "                    for line_num, line in enumerate(lines):\n",
    "                        stripped_line = line.strip()\n",
    "                        \n",
    "                        if line.strip().startswith('#') or line.strip().startswith('//') or line.strip().startswith('*'):\n",
    "                                continue\n",
    "\n",
    "                        if re.search(process_pattern, stripped_line):\n",
    "                            process_flag = True\n",
    "                            process_match = re.search(process_pattern, stripped_line)\n",
    "                            if process_match:\n",
    "                                process_name = process_match.group(1)\n",
    "                            print(f'Process {process_name} in filepath {file_path}')\n",
    "                            brace_stack.append('{')\n",
    "\n",
    "                        if '{' in stripped_line:\n",
    "                            brace_stack.append('{')\n",
    "\n",
    "                        if '}' in stripped_line:\n",
    "                            if brace_stack:\n",
    "                                brace_stack.pop()\n",
    "                                if not brace_stack and process_flag:\n",
    "                                    process_flag = False\n",
    "\n",
    "                        if process_flag:\n",
    "                            pattern_found = False\n",
    "                            for pattern in pattern_list:\n",
    "                                regex_pattern = re.compile(r'\\.' + re.escape(pattern) + r'(?!\\w)' , re.IGNORECASE)\n",
    "                                if regex_pattern.search(stripped_line):\n",
    "                                    result_list.append([folder_name, process_name, pattern, file_path, line_num+1, stripped_line])\n",
    "                                    print(f'Nextflow operator {pattern} with content: {stripped_line}')\n",
    "                                    pattern_found = True\n",
    "                                    break\n",
    "                            if not pattern_found:\n",
    "                                result_list.append([folder_name, process_name, 'none', file_path, 'none', 'none'])\n",
    "\n",
    "                    if brace_stack:\n",
    "                        print(\"Error: Unclosed process block in file:\", file_path)\n",
    "\n",
    "    with open(f'./results/{output_file_name}.csv', 'w', newline='', encoding='utf-8') as csvfile:\n",
    "        writer = csv.writer(csvfile)\n",
    "        writer.writerow(['pipeline', 'process', 'operator', 'file_path', 'line_number', 'line_content'])\n",
    "        writer.writerows(result_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "\n",
    "df = pd.read_csv('./results/nextflow_operators_in_processes.csv')\n",
    "#df = pd.read_csv('./results/all_operators_in_scripts_new.csv')\n",
    "\n",
    "pd.set_option('display.max_colwidth', None)\n",
    "#pd.set_option('display.max_rows', None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "# drop duplica\n",
    "df = df.drop_duplicates().reset_index()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>index</th>\n",
       "      <th>pipeline</th>\n",
       "      <th>process_name</th>\n",
       "      <th>operator</th>\n",
       "      <th>file_path</th>\n",
       "      <th>line_number</th>\n",
       "      <th>line_content</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>airrflow</td>\n",
       "      <td>FASTQC_POSTASSEMBLY</td>\n",
       "      <td>none</td>\n",
       "      <td>./git_repos\\airrflow\\modules\\local\\fastqc_postassembly.nf</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>30</td>\n",
       "      <td>airrflow</td>\n",
       "      <td>FETCH_DATABASES</td>\n",
       "      <td>none</td>\n",
       "      <td>./git_repos\\airrflow\\modules\\local\\fetch_databases.nf</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>63</td>\n",
       "      <td>airrflow</td>\n",
       "      <td>GUNZIP</td>\n",
       "      <td>none</td>\n",
       "      <td>./git_repos\\airrflow\\modules\\local\\gunzip.nf</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>90</td>\n",
       "      <td>airrflow</td>\n",
       "      <td>MERGE_UMI</td>\n",
       "      <td>none</td>\n",
       "      <td>./git_repos\\airrflow\\modules\\local\\merge_UMI.nf</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>119</td>\n",
       "      <td>airrflow</td>\n",
       "      <td>PARSE_LOGS</td>\n",
       "      <td>none</td>\n",
       "      <td>./git_repos\\airrflow\\modules\\local\\parse_logs.nf</td>\n",
       "      <td>none</td>\n",
       "      <td>none</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   index  pipeline         process_name operator  \\\n",
       "0      0  airrflow  FASTQC_POSTASSEMBLY     none   \n",
       "1     30  airrflow      FETCH_DATABASES     none   \n",
       "2     63  airrflow               GUNZIP     none   \n",
       "3     90  airrflow            MERGE_UMI     none   \n",
       "4    119  airrflow           PARSE_LOGS     none   \n",
       "\n",
       "                                                   file_path line_number  \\\n",
       "0  ./git_repos\\airrflow\\modules\\local\\fastqc_postassembly.nf        none   \n",
       "1      ./git_repos\\airrflow\\modules\\local\\fetch_databases.nf        none   \n",
       "2               ./git_repos\\airrflow\\modules\\local\\gunzip.nf        none   \n",
       "3            ./git_repos\\airrflow\\modules\\local\\merge_UMI.nf        none   \n",
       "4           ./git_repos\\airrflow\\modules\\local\\parse_logs.nf        none   \n",
       "\n",
       "  line_content  \n",
       "0         none  \n",
       "1         none  \n",
       "2         none  \n",
       "3         none  \n",
       "4         none  "
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "operator\n",
       "none           3114\n",
       "collect         610\n",
       "join            357\n",
       "set             123\n",
       "map              86\n",
       "mix              56\n",
       "ifEmpty          40\n",
       "filter           36\n",
       "groupTuple       33\n",
       "dump             33\n",
       "flatten          31\n",
       "toList           23\n",
       "count            22\n",
       "first            17\n",
       "branch           13\n",
       "collectFile      12\n",
       "combine          12\n",
       "concat           12\n",
       "splitCsv         10\n",
       "transpose         6\n",
       "collate           6\n",
       "unique            6\n",
       "toInteger         6\n",
       "max               5\n",
       "merge             4\n",
       "tap               4\n",
       "buffer            3\n",
       "subscribe         2\n",
       "min               2\n",
       "splitText         2\n",
       "take              2\n",
       "multiMap          2\n",
       "view              1\n",
       "flatMap           1\n",
       "cross             1\n",
       "last              1\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df['operator'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pipeline</th>\n",
       "      <th>process_name</th>\n",
       "      <th>operator_count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>airrflow</td>\n",
       "      <td>ADD_META_TO_TAB</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>airrflow</td>\n",
       "      <td>AIRRFLOW_REPORT</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>airrflow</td>\n",
       "      <td>CHANGEO_ASSIGNGENES</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>airrflow</td>\n",
       "      <td>CHANGEO_CONVERTDB_FASTA</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>airrflow</td>\n",
       "      <td>CHANGEO_CREATEGERMLINES</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3076</th>\n",
       "      <td>viralrecon</td>\n",
       "      <td>TABIX_BGZIP</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3077</th>\n",
       "      <td>viralrecon</td>\n",
       "      <td>TABIX_TABIX</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3078</th>\n",
       "      <td>viralrecon</td>\n",
       "      <td>UNICYCLER</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3079</th>\n",
       "      <td>viralrecon</td>\n",
       "      <td>UNTAR</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3080</th>\n",
       "      <td>viralrecon</td>\n",
       "      <td>VCFLIB_VCFUNIQ</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3081 rows  3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        pipeline             process_name  operator_count\n",
       "0       airrflow          ADD_META_TO_TAB               1\n",
       "1       airrflow          AIRRFLOW_REPORT               1\n",
       "2       airrflow      CHANGEO_ASSIGNGENES               1\n",
       "3       airrflow  CHANGEO_CONVERTDB_FASTA               1\n",
       "4       airrflow  CHANGEO_CREATEGERMLINES               1\n",
       "...          ...                      ...             ...\n",
       "3076  viralrecon              TABIX_BGZIP               1\n",
       "3077  viralrecon              TABIX_TABIX               1\n",
       "3078  viralrecon                UNICYCLER               1\n",
       "3079  viralrecon                    UNTAR               1\n",
       "3080  viralrecon           VCFLIB_VCFUNIQ               1\n",
       "\n",
       "[3081 rows x 3 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "operators_in_all_process = df.groupby(['pipeline', 'process_name']).size().reset_index(name='operator_count')\n",
    "operators_in_all_process"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Processes without an operator get the entry 'none'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>index</th>\n",
       "      <th>pipeline</th>\n",
       "      <th>process_name</th>\n",
       "      <th>operator</th>\n",
       "      <th>file_path</th>\n",
       "      <th>line_number</th>\n",
       "      <th>line_content</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>427</td>\n",
       "      <td>airrflow</td>\n",
       "      <td>CHANGEO_MAKEDB</td>\n",
       "      <td>collect</td>\n",
       "      <td>./git_repos\\airrflow\\modules\\local\\changeo\\changeo_makedb.nf</td>\n",
       "      <td>14</td>\n",
       "      <td>path(igblast) // igblast fasta from ch_igblast_db_for_process_igblast.mix(ch_igblast_db_for_process_igblast_mix).collect()</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>547</td>\n",
       "      <td>airrflow</td>\n",
       "      <td>COLLAPSE_DUPLICATES</td>\n",
       "      <td>join</td>\n",
       "      <td>./git_repos\\airrflow\\modules\\local\\enchantr\\collapse_duplicates.nf</td>\n",
       "      <td>25</td>\n",
       "      <td>echo \"${tabs.join('\\n')}\" &gt; tabs.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>590</td>\n",
       "      <td>airrflow</td>\n",
       "      <td>DEFINE_CLONES</td>\n",
       "      <td>join</td>\n",
       "      <td>./git_repos\\airrflow\\modules\\local\\enchantr\\define_clones.nf</td>\n",
       "      <td>46</td>\n",
       "      <td>def thr = threshold.join(\"\")</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>595</td>\n",
       "      <td>airrflow</td>\n",
       "      <td>DEFINE_CLONES</td>\n",
       "      <td>join</td>\n",
       "      <td>./git_repos\\airrflow\\modules\\local\\enchantr\\define_clones.nf</td>\n",
       "      <td>51</td>\n",
       "      <td>input = tabs.join(',')</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>641</td>\n",
       "      <td>airrflow</td>\n",
       "      <td>DETECT_CONTAMINATION</td>\n",
       "      <td>join</td>\n",
       "      <td>./git_repos\\airrflow\\modules\\local\\enchantr\\detect_contamination.nf</td>\n",
       "      <td>26</td>\n",
       "      <td>echo \"${tabs.join('\\n')}\" &gt; tabs.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4661</th>\n",
       "      <td>141057</td>\n",
       "      <td>viralrecon</td>\n",
       "      <td>FASTQC</td>\n",
       "      <td>collect</td>\n",
       "      <td>./git_repos\\viralrecon\\modules\\nf-core\\fastqc\\main.nf</td>\n",
       "      <td>25</td>\n",
       "      <td>def old_new_pairs = reads instanceof Path || reads.size() == 1 ? [[ reads, \"${prefix}.${reads.extension}\" ]] : reads.withIndex().collect { entry, index -&gt; [ entry, \"${prefix}_${index + 1}.${entry.extension}\" ] }</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4662</th>\n",
       "      <td>141058</td>\n",
       "      <td>viralrecon</td>\n",
       "      <td>FASTQC</td>\n",
       "      <td>join</td>\n",
       "      <td>./git_repos\\viralrecon\\modules\\nf-core\\fastqc\\main.nf</td>\n",
       "      <td>26</td>\n",
       "      <td>def rename_to = old_new_pairs*.join(' ').join(' ')</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4663</th>\n",
       "      <td>141059</td>\n",
       "      <td>viralrecon</td>\n",
       "      <td>FASTQC</td>\n",
       "      <td>collect</td>\n",
       "      <td>./git_repos\\viralrecon\\modules\\nf-core\\fastqc\\main.nf</td>\n",
       "      <td>27</td>\n",
       "      <td>def renamed_files = old_new_pairs.collect{ old_name, new_name -&gt; new_name }.join(' ')</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4670</th>\n",
       "      <td>141344</td>\n",
       "      <td>viralrecon</td>\n",
       "      <td>MINIA</td>\n",
       "      <td>join</td>\n",
       "      <td>./git_repos\\viralrecon\\modules\\nf-core\\minia\\main.nf</td>\n",
       "      <td>25</td>\n",
       "      <td>def read_list = reads.join(\",\")</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4681</th>\n",
       "      <td>141845</td>\n",
       "      <td>viralrecon</td>\n",
       "      <td>QUAST</td>\n",
       "      <td>join</td>\n",
       "      <td>./git_repos\\viralrecon\\modules\\nf-core\\quast\\main.nf</td>\n",
       "      <td>36</td>\n",
       "      <td>${consensus.join(' ')}</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1580 rows  7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       index    pipeline          process_name operator  \\\n",
       "14       427    airrflow        CHANGEO_MAKEDB  collect   \n",
       "18       547    airrflow   COLLAPSE_DUPLICATES     join   \n",
       "20       590    airrflow         DEFINE_CLONES     join   \n",
       "21       595    airrflow         DEFINE_CLONES     join   \n",
       "23       641    airrflow  DETECT_CONTAMINATION     join   \n",
       "...      ...         ...                   ...      ...   \n",
       "4661  141057  viralrecon                FASTQC  collect   \n",
       "4662  141058  viralrecon                FASTQC     join   \n",
       "4663  141059  viralrecon                FASTQC  collect   \n",
       "4670  141344  viralrecon                 MINIA     join   \n",
       "4681  141845  viralrecon                 QUAST     join   \n",
       "\n",
       "                                                                file_path  \\\n",
       "14           ./git_repos\\airrflow\\modules\\local\\changeo\\changeo_makedb.nf   \n",
       "18     ./git_repos\\airrflow\\modules\\local\\enchantr\\collapse_duplicates.nf   \n",
       "20           ./git_repos\\airrflow\\modules\\local\\enchantr\\define_clones.nf   \n",
       "21           ./git_repos\\airrflow\\modules\\local\\enchantr\\define_clones.nf   \n",
       "23    ./git_repos\\airrflow\\modules\\local\\enchantr\\detect_contamination.nf   \n",
       "...                                                                   ...   \n",
       "4661                ./git_repos\\viralrecon\\modules\\nf-core\\fastqc\\main.nf   \n",
       "4662                ./git_repos\\viralrecon\\modules\\nf-core\\fastqc\\main.nf   \n",
       "4663                ./git_repos\\viralrecon\\modules\\nf-core\\fastqc\\main.nf   \n",
       "4670                 ./git_repos\\viralrecon\\modules\\nf-core\\minia\\main.nf   \n",
       "4681                 ./git_repos\\viralrecon\\modules\\nf-core\\quast\\main.nf   \n",
       "\n",
       "     line_number  \\\n",
       "14            14   \n",
       "18            25   \n",
       "20            46   \n",
       "21            51   \n",
       "23            26   \n",
       "...          ...   \n",
       "4661          25   \n",
       "4662          26   \n",
       "4663          27   \n",
       "4670          25   \n",
       "4681          36   \n",
       "\n",
       "                                                                                                                                                                                                             line_content  \n",
       "14                                                                                             path(igblast) // igblast fasta from ch_igblast_db_for_process_igblast.mix(ch_igblast_db_for_process_igblast_mix).collect()  \n",
       "18                                                                                                                                                                                   echo \"${tabs.join('\\n')}\" > tabs.txt  \n",
       "20                                                                                                                                                                                           def thr = threshold.join(\"\")  \n",
       "21                                                                                                                                                                                                 input = tabs.join(',')  \n",
       "23                                                                                                                                                                                   echo \"${tabs.join('\\n')}\" > tabs.txt  \n",
       "...                                                                                                                                                                                                                   ...  \n",
       "4661  def old_new_pairs = reads instanceof Path || reads.size() == 1 ? [[ reads, \"${prefix}.${reads.extension}\" ]] : reads.withIndex().collect { entry, index -> [ entry, \"${prefix}_${index + 1}.${entry.extension}\" ] }  \n",
       "4662                                                                                                                                                                   def rename_to = old_new_pairs*.join(' ').join(' ')  \n",
       "4663                                                                                                                                def renamed_files = old_new_pairs.collect{ old_name, new_name -> new_name }.join(' ')  \n",
       "4670                                                                                                                                                                                      def read_list = reads.join(\",\")  \n",
       "4681                                                                                                                                                                                               ${consensus.join(' ')}  \n",
       "\n",
       "[1580 rows x 7 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "non_none_operator_rows = df[df['operator'] != 'none']\n",
    "non_none_operator_rows"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pipeline</th>\n",
       "      <th>process_name</th>\n",
       "      <th>operator_count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>airrflow</td>\n",
       "      <td>CHANGEO_MAKEDB</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>airrflow</td>\n",
       "      <td>COLLAPSE_DUPLICATES</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>airrflow</td>\n",
       "      <td>DEFINE_CLONES</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>airrflow</td>\n",
       "      <td>DETECT_CONTAMINATION</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>airrflow</td>\n",
       "      <td>FASTQC</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>587</th>\n",
       "      <td>viralrecon</td>\n",
       "      <td>CAT_FASTQ</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>588</th>\n",
       "      <td>viralrecon</td>\n",
       "      <td>FASTQC</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>589</th>\n",
       "      <td>viralrecon</td>\n",
       "      <td>MINIA</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>590</th>\n",
       "      <td>viralrecon</td>\n",
       "      <td>PLOT_MOSDEPTH_REGIONS</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>591</th>\n",
       "      <td>viralrecon</td>\n",
       "      <td>QUAST</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>592 rows  3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       pipeline           process_name  operator_count\n",
       "0      airrflow         CHANGEO_MAKEDB               1\n",
       "1      airrflow    COLLAPSE_DUPLICATES               1\n",
       "2      airrflow          DEFINE_CLONES               2\n",
       "3      airrflow   DETECT_CONTAMINATION               1\n",
       "4      airrflow                 FASTQC               3\n",
       "..          ...                    ...             ...\n",
       "587  viralrecon              CAT_FASTQ               5\n",
       "588  viralrecon                 FASTQC               3\n",
       "589  viralrecon                  MINIA               1\n",
       "590  viralrecon  PLOT_MOSDEPTH_REGIONS               1\n",
       "591  viralrecon                  QUAST               1\n",
       "\n",
       "[592 rows x 3 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "operators_in_process = non_none_operator_rows.groupby(['pipeline', 'process_name']).size().reset_index(name='operator_count')\n",
    "operators_in_process"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAm0AAAHLCAYAAACapq/QAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjcuMiwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy8pXeV/AAAACXBIWXMAAA9hAAAPYQGoP6dpAACC50lEQVR4nO3dd1QU1/838PfQe1EEFbGLYjdiiyJqLLGLxhYTsZsYe43RryUmtpioMcVg7Ik90cQSNcYuqKjYO3YUEQuIKPU+f/Ds/HbZCrOIuO/XOXvO7M5tMzu7+9k7M/dKQggBIiIiInqjWeV3A4iIiIjIOAZtRERERAUAgzYiIiKiAoBBGxEREVEBwKCNiIiIqABg0EZERERUADBoIyIiIioAGLQRERERFQAM2oiIiIgKAAZtFqpPnz6QJAmSJGHFihX53Zy30pMnT/Dll1+iXr168PT0hLW1Nfc5WYzSpUvLx/utW7fyuzlEb4UCH7Q1adJE/mKQJAklS5ZESkqKSXk///xzOV+fPn3ytqFkUW7duoWaNWti6tSpOH78OJ49e4bMzMz8bhYRERVgNvndAHO7e/cufvnlFwwfPjy/m0IWbPDgwbh79y4AwNHREc2bN4evry+sra0BAAEBAfnZPMqFW7duoUyZMgCAUqVKsfeIiF67ty5oA4BZs2ZhwIABcHJyyu+mkAWKjY3F7t27AQD29vY4c+YMKlSokM+tIiKigq7Anx7VJTY2Fj/88EN+N4Ms1KlTp+TloKAgBmxkkW7dugUhBIQQKF26dH43h+it8FYFbfXq1ZOX58yZg8TExHxsDVmqp0+fysvFihXLx5YQEdHb5K0K2j7++GNUrFgRQNade/Pnz8/nFpElSktLk5etrN6qjxgREeWjt+oXxdraGtOmTZOff/fdd3jy5ImiMvfv3y/fYdqkSROT8qjfzZqTNCdPnsTAgQPh7+8PZ2dnuLm5ISgoCKtWrYIQQquM//77D126dEG5cuXg4OCAokWLIiQkBIcOHcrxdgJZge7cuXNRt25dFClSBI6OjihbtiwGDBiAkydP5ri8yMhIjBo1CjVr1kSRIkVgZ2eHokWLIjg4GHPmzNHokdJH17AB0dHRmDRpEmrVqoUiRYrAysoKNWvWzHH7sktKSsL333+PVq1aoUSJEnBwcICnpyeqVq2KoUOH4tixY3rzqh8nffv2lV9fuXKlxnstSZLGMZpb8fHxmD17NoKDg1GsWDHY29vDy8sLtWrVwrhx43Dx4kWjZdy6dUtuk/rpq3379uHDDz9EuXLl4OjoCC8vLzRq1Ajff/89Xr16laN2vnjxAj///DPat2+PUqVKwcnJCa6urqhQoQL69euHvXv3Gi1jxYoVWnd5Z2RkYN26dejYsSPKli0LR0dHSJKELVu2aOR9+fIltmzZguHDh6NRo0bw8fGBnZ0dXFxcULp0aXTu3BnLli1Damqq0fpVNyEAwO3bt7XeV2OfeSEENm7ciJ49e6JcuXJwcXGBi4sLypUrhw8//BCbNm3S+TnPTv2O+f379wMAHjx4gJkzZ6Ju3booWrQorK2t4eHhoZX37t27mD59Oho3bgwfHx/Y29vD1dUVZcqUQd26ddG3b1+sXbsW8fHxRtthjClDfujalidPnmDOnDmoU6cOvLy85O+h/v374/z584rbpU7f9/CgQYNQsWJFuLi4wNPTE4GBgfj666+RkJBgtEwl7w8A3LlzB1OmTEH9+vXl49XHxwf169fH1KlT5RuccuLGjRuYNm0aGjduDF9fXzg4OMDJyQlly5ZFp06dsGjRIsTFxRktxxyfZ5W9e/eif//+qFatGjw8PGBrawsvLy8EBASgSZMmmDRpEg4ePKjxJzg7cx7PQghs3rwZoaGh8Pf3h7u7OxwcHODn54dOnTph5cqVSE9PN2nbLl++jPHjx6N+/frw8vKCnZ0d3N3dUb58edSvXx9DhgzBli1bcnc2UBRwwcHBAoAAIH7++WeRmZkpqlWrJr82YcIEvXknTJggpwsNDdWZZt++fXKa4OBgk9qkSm9o92ZP8/XXXwtra2uN19UfvXv3FpmZmUIIIZKSkkT79u31ppUkSXz33XcG2xgaGiqnX758uYiIiBC+vr56y7SyshITJ040afufPHkiunTporcs1cPDw0Ns3LjRYFmlSpWS09+8eVP88ssvwsHBQausGjVqmNQ2fbZu3SqKFi1qtM0ffvihePHihVZ+9ePE2GPq1KmK2rp06VLh7u5usA5ra2sxcuRIkZ6errecmzdvyulLlSolUlNTxSeffGKwXH9/f3HhwgWT2rlhwwaT9mm7du3Es2fP9JazfPlyjc9pTEyMCAoK0lnW5s2b5XxHjx4VLi4uJr0npUuXFqdOnTJavykPXa5evSpq1aplNG/t2rXF9evXDe5X9e+8ffv2iS1btghPT0+tstzd3TXyLV68WDg6Opq0DQ0bNjTYBlNk/+yasi2HDx82+D1kbW0twsLCFLdNJfv7Nn36dGFlZaW3/mLFiokDBw4YLDO3748QWb8Dur7f1B8ODg5i1qxZJm3fq1evxGeffSZsbGyMvue2trYiMTFRb1nm+jw/f/5cdOjQweTP05IlS3SWY87j+cyZM6JmzZpGy6lYsaLR77+pU6eatL8BiF69ehksS5e37u5RSZIwY8YMdOrUCQCwaNEijBo1Cj4+PvnbMAN+/vlnTJo0CQBQu3ZtVKtWDUIIHDp0CDdu3AAArFq1ChUrVsTnn3+OLl26YNeuXbCzs0OjRo1QtmxZJCYm4r///sPjx48hhMDo0aNRp04dNGrUyGj9t2/fxujRo/H06VM4OzujWbNm8PHxQWxsLPbt24cXL14gMzMTs2bNQnp6OubOnau3rNjYWDRr1gyXLl2SXwsICEDNmjXh6uqKuLg4HD58GPHx8Xj27Bm6deuG1atXo1evXkbbuXHjRowfPx4A4Ovri3fffRfu7u64f/++oh7V9evXo1evXsjIyACQ1WPbqFEjlC9fHklJSTh06BDu378PAFizZg1u3ryJvXv3wsHBQS7D19cXn332GYCsf1n//fcfAKBSpUp47733NOqrW7durts6b948jBs3Tn5ub2+P4OBglCxZEk+fPsW+ffvw5MkTZGRkYMGCBbh9+zb++OMPgz1AKhMmTMDixYsBAFWrVkWtWrUgSRJOnTol93BcvXoVzZo1Q3h4OMqWLau3rPnz52PMmDFyz5GrqysaNGgAPz8/ZGRk4OLFi4iMjIQQAtu2bUNwcDDCw8ON3vGdkpKCDh064OTJk7CxscG7776L8uXL49WrVxo3gABZ1xYmJSUBALy9vVGlShWUKFECzs7OSE5OxvXr13H8+HGkp6fj1q1bCA4OxqlTp1C+fHmNcgICAvDZZ5/h+fPnWLVqlbw9vXv3NrpPAeDSpUsIDg7Go0eP5NeqVauGmjVrQpIkREVF4dy5cwCyenkaNmyIgwcPwt/f32jZ4eHhmDZtGtLS0lC4cGE0btwYXl5eiIuLQ1RUlJxuy5Yt+OSTT+Tnbm5uaNCgAUqUKAEbGxskJCTg6tWrOH/+vMFex7x0/vx5TJw4EUlJSfD29kZQUBAKFy6MmJgY7N27Fy9fvkRGRgY++eQTVK1aFQ0aNDBr/d9//z2mTp0KAChXrhzq1asHe3t7XLhwAcePHweQ1WPWpk0b7N2716TPsanvDwAMHToUP/74o/xc9V1ctGhR+bs4KSkJr169wsSJE/Hw4UODlwAlJSWhZcuWiIiIkF9zcnJCo0aNUKJECQghEBMTg5MnT+Lx48dIS0uTvwOzM+fn+eOPP8bff/8tPy9fvjxq1aqFQoUKIS0tDY8ePcK5c+cMDqljzuP54MGDaN++vdzrZWNjg8DAQFSsWBG2tra4desWDh8+jFevXuHKlSt49913ERERoXPYpgULFmD69Onycy8vL9SvXx/FihWDJEl48uQJLl++jEuXLund10blOMx7w2TvaVOpU6eO/PqIESN05n1Tetrs7OyEr6+vOHLkiEaa9PR0MXbsWDmdh4eHmDZtmgAgmjRpIm7fvq2R/tmzZ6Jp06Zy+iZNmuitX72nzc7OTgBZvUjZ/yE9e/ZM9OjRQ6O9e/fu1VlmRkaGRv3vvPOOiIyM1Er38uVLMW3aNCFJkgAgnJ2dxY0bN3SWqf5v3cbGRtjZ2YmwsDC511Hl1atXerfVkOvXr2v0xgQGBoqrV69qbde3336r8Q982LBhesvM3jNkLuHh4Rq9se+//7548OCBRppXr16JcePGabxf3377rc7y1HvabG1tBQBRuHBh8c8//2il3bFjh0ZvQZMmTbTeA5U9e/bI+8rW1lZ89dVXIikpSStdVFSUqFy5slzmp59+qrM89f2p+gcbHByss/dG/Tg4evSo+OKLL8S5c+d0liuEEA8fPhQff/yxXP57772nN232nklTpKSkiBo1asj5ihQpInbt2qWVbteuXcLLy0vjs5OamqqzTPXvPBsbGyFJkpgxY4ZWevV9od6GoUOH6uwtFiKrF2TDhg0Gz1CYKqc9bfb29sLa2lp8++23Ii0tTSPdnTt3RNWqVeW0TZs2Vdw+IbS/hx0cHMTq1au10h09elRjeypWrChevnxpdJtMfX/Wr1+v0ZbevXuLhIQEjfQJCQnio48+0ki3adMmvdvWvXt3OZ21tbWYPn26zs9hRkaG2Lt3r+jYsaPOHjJzfp6joqLk9S4uLmLHjh162x8dHS2++uor8ffff2utM9fx/ODBA+Hj4yOX1bNnT3Hv3j2tdLGxsSIkJEROV61aNa2zGGlpaaJw4cJymlmzZun9DD9+/FgsW7ZMzJkzR+/26/PWBm07d+7U+DK4e/euVt43JWhzdHQUFy9e1JkuIyNDBAQEaKSvUqWK3i+MO3fuyD9skiSJ+/fv60ynHrQBEG3atBEZGRl629CqVSs5bZ06dXSmW7VqlZymZs2a4vnz53q3XwghpkyZIqf/5JNPdKZR/6IEIH777TeDZeZU79695bLLli0rnj59qjftd999J6e1srLSG2jmVdDWuHFjudz69euLlJQUvWmHDx8up3Vzc9N52kM9CFFtU/Y/DuoOHjwoB9oAxPbt27XSZGRkiAoVKpj8fj148EB4e3vLPwi6PqfZT09Wq1ZNJCcnGyw3p1q3bi2Xr++zmJugbdmyZRqB8YkTJ/SmPX78uMZplZUrV+pMp/6dB0B89dVXBtvw/PlzOa2fn5/eYNvcchq0ARC//PKL3vLOnTsnH3+GvttyQr1uAGLdunV60165ckU4OTnJaX/88UeTtsnY+5ORkSHKlCkjp+/SpYve9ygzM1N07NhRTluuXDmd39v//vuvRhvWrl1rsA2G2mbOz/OiRYvksiZNmpSrNpnzeO7Xr59c1oABAwymTU9P1+iUyH6snDt3Tl5njssL9HlrgzYhhMZ1L4MGDdLK+6YEbaNGjTJYnnpwA0DnPw916j/uW7du1ZlGPWiTJMnodTRXrlzR+ME+ffq0Vhr1awIOHTpksDwhsnrcPDw8BJB1fYeuLx/1L/66desaLTMnnj59Kuzt7eXy//jjD4PpMzIyRJUqVeT0n3/+uc50eRG0Xbx4UeMYOHnypMH0SUlJGj03ixcv1kqTPWjr3bu30Xb06tVLTt+pUyet9Vu2bJHXG+q1Ujdr1iw5j65ewexBm6F/57ml3tPx/fff60yTm6CtXr16ch5DvbMqn376qUZgrov6d56vr69Wr1R2MTExcvqaNWua1G5zyGnQVq1aNaNl1q1b1+h3W06oH1eNGzc2mn7SpElG92VO359//vlHTm9nZ2c0GL13757cMw5A7Ny5UyvN+++/L6/v3r270e3Sx9yf56+//lpet2DBgly1yVzHc1xcnPz97+7ubrSTQQghIiIi5Lrbt2+vse7IkSMGvxvN5a26ezS7r776Sl5evny5fH3Ym6ZLly4G11etWlVednR0ROvWrU1Of/PmTaP1N2zYEOXKlTOYxt/fX+Makn379mmsf/DgAU6fPg0g6/ouU66lc3BwkMtMSEgwemdYjx49jJaZE+Hh4fI8tYULF0aHDh0MpreyskK/fv3k59n3QV5Sr6tGjRp45513DKZ3dnZGz549debXx5RrtEJDQ+Xl/fv3a93tuGPHDnnZ1PerWbNm8vLhw4cNpvX09ESrVq1MKlddcnIy9u7di4ULF2Ly5MkYMWIEhg4dKj/Wrl0rp1Udx0olJSXhxIkT8nP1Y0efAQMGyMuRkZF48eKFwfRdunSBjY3hS5NVd4IDwLlz53Dw4EGj7cgPXbt2NZqmVq1a8rK5pxHL6fF/5swZo3fAm/L+qN9x2bp1a6NjO/r6+uL999+Xn2f/bKekpMh3rQLAsGHDDJZniLk/zyVLlpSXV65cafT41sVcx/OePXvk7/927drBxcXFaJ569erJ1+kZ2ra9e/dqXNdtTm/djQjqGjdujBYtWuDff/9FWloapk+fjpUrV+Z3s7RUqVLF4HpPT0952d/f3+iXgHp6U24prl+/vtE0ANCgQQOEh4cDgNZFtOoXuwohMHToUJPKjI6Olpfv3r2L6tWr601bu3Ztk8o0lfo21KtXz+h+BbICXPX8QgiTLvJXSr2t6m0wpGHDhli0aBEAaF2kn50kSRqDU+tTr149SJIEIQSePXumMR8noHkcbN++3aQASH0YBWPDGdSsWTNHY989efIEU6ZMwapVq/D8+XOT8phjuAsg60dddbGxi4uLwWNbpWbNmnB2dsaLFy+QkZGBM2fO4N1339Wb3pTPhK2tLUJCQrBmzRpkZGSgefPm+OCDD/DBBx8gODgYhQsXNn2j8lC1atWMplFvqynDb+SEKd+DFSpUQOHCheUbvk6fPo2mTZvqTW/K+5Pbz/bWrVsBaH+2T58+LQ/N4+TkZNLnWh9zf57btGkDFxcXJCUlISoqChUrVkTfvn3Rtm1b1K5dG7a2tkbLN9fxrL5tV69eNfk3S/V9//TpU7x48QLOzs4AgBIlSuDdd99FeHg4EhMTERgYiF69eiEkJASNGjWCq6urSeUb81YHbUBWb9u///4LAPjtt9/w+eefv3GTdbu7uxtcrx5MGEubPb2hMW5U1P8hGOLn5ycvq98JB0C+u1K1rH4XlKmM/WstUqRIjss0RH0bSpUqZVIe9fHMUlNT8fz5c7i5uZm1XboobauxQMTT09Okf5pubm5wd3fHs2fP5HapB23qx0H2MdNMYc5j4Pbt22jcuDHu3LmTozaYGtwZo/6e+fn5mRRsWllZwc/PD5cvXwZg/H0zdX/Mnz8fp06dwuXLl5GWloa1a9di7dq1kCQJlSpVQlBQEFq2bIm2bdtq3BX9Opny3ab+o27Kd1tO5OR78PHjxwC0vwezM+X9Mfdn++HDh/Kyn5+fSX9G9TH357lQoUJYvnw5evXqhdTUVMTExOCrr77CV199BUdHR9StWxfBwcHo0KGDwYDXHMez+rZFRkYiMjIyV9unCtoAYNmyZWjatCkePHiA5ORkLFmyBEuWLIG1tTWqVauGxo0bo3Xr1mjevHmu35e3+vQokDW8guq0V2ZmpnxL95skJz01edGrY2yYBRX1gzP7D5s5/vUaG7hQ1SVuLqrhIADNbTMkezpz/cAbo7Stxtpp6jFgrFylx4E5j4FevXrJAZubmxvGjBmDXbt24ebNm0hKSkJGRoY8N6b6KabMzMzcNT6b3Lxn2dMae99M3R/e3t6IjIzEtGnTULx4cfl1IQQuXbqEsLAwfPDBByhevDhmz56d++EIFHgdPdaGmON7MDtT3h9zf7bVn5vyR8yQvPg8f/DBBzhx4gS6du0KOzs7+fWXL1/iwIED+PLLLxEYGIjAwEC9pz7NcTznxW9WxYoVcebMGYwaNQqFChWSX8/IyMDp06fx/fffo3Xr1ihVqhSWLFmSqzrf+qANAGbMmCF/IWzatMls16zoYq4v/NcpOTnZpHTq1x9k7+pV/xLp1KmT/GOYk4dqtPvXRf0LzdRrK7KnM1eXtzFK22qsnaYeA8bKVT8OTp8+neNjwFzXKYWHh+PIkSNyG48dO4Z58+ahZcuWKF26NJydnTV6vvIi+M7Ne5Y9rTmPLxcXF3lE/cjISHz77bfo1KkTvLy85DRPnz7FxIkT0aVLF5NmZ3ibmON7MDfM/dlWf64eEOZGXn2eq1Wrhg0bNuDRo0fYtm0bxo8fjwYNGmj0pJ48eRJNmzbFxo0bdZah9HhW37YFCxbk6jdLvcdTpUiRIvjuu+8QGxuLQ4cO4euvv0br1q01zsjcv38fgwYNwvDhww3uf10sImirXr26fJGrEAL/+9//TM6rfhCZMoWFua+zeB1MPX2knk79gwFAY/Dia9eumadheUz91IWp++D27dvysp2d3WsL2pS2Nfv7ld3Tp09NClwSExM1jvE39ThQDW4MAH369EGlSpUMplffV+ai/p7dvXvXpCAoMzNT4zogY+9bblhZWSEwMBCjR4/G5s2b8fDhQxw+fFgekBwA/vrrL/zxxx9mr/tNZurnytzvj7k/2+qfwbt375o89ZIuef15dnNzQ9u2bTFnzhyEh4cjPj4eK1askIOhzMxMDBkyBC9fvtRbRm6P57zeNltbWzRq1AhffPEFduzYgfj4eOzcuRPBwcFymkWLFuX4tKxFBG0A8OWXX8La2hoAsG3bNoPzSKpT/1FWXcdgiGpk84JE/YJMQ44ePSovZ797Uf0i3gsXLuDevXvmaVweUr8T7fjx4yadElL13qjyv65TOuptVd0MYox6W43dbSqEMOkzcezYMTn48PDw0LieDYDGRc+7du0yqZ15Qf16FWM3+gAw6Q60nL7XNWrUkL9zkpKSTPpuOHPmjNyLYm1tjRo1auSoztywsrJCw4YN8eeff2rcmas+ar0lMOV78Nq1a/LvgCRJGp/L3DL3Z7tmzZrydVzJyckm/9bp8ro/z25ubggNDcXevXthb28PIOuaPVN/owDTj+fXvW22trZo1aoVdu/erXHTjeqGElNZTNBWsWJFfPTRR/LzyZMnm5RPNekxAFy/ft1od/OGDRty38h8Eh4ernEXpy5Xr17V+OBkv2OqTJkyGjd4LFiwwKxtzAvvvvuu/MXw6NEjbN++3WB6IYTG3cfqt7bnNfW6oqKicObMGYPpX758iXXr1unMr8/q1auNplmxYoW8rJoYW127du3k5TVr1pg0CXVeUD/1aey01/37900KUNQvaDblIngXFxcEBgbKz9X3nT7Lly+Xl+vWrZuja+GUkiQJbdu2lZ+rX9BuCXJ6/NeoUUPjTv3cUv9s7tixw+hnJjY2Fjt37tSZH8ia2k79+/mHH37Iddvy6/NcpkwZjT9buTkWjR3PrVq1km8GuH79OrZt25bL1uaMnZ0dWrRoobddxlhM0AYAU6dOlU937tmzBwcOHDCax83NDRUrVgSQdXr0999/15s2Kioq1xcX5ichBIYPH673erzMzEwMHz5c7mEJDAzU2QMwYcIEeXnhwoXYs2ePyW2IjY3NYauV8/DwQPfu3eXn48aNM3iK8Mcff5SDJSsrKwwcODDP26hSqVIlNG7cWH4+bNgwg4HD//73P/kL1s3NDR9++KHROn777TeD//QPHTqkMZ6Z+phiKl26dJHn7kxOTsZHH31k8l1+SUlJuRq3SRf1eVH/+usvvekyMjIwaNAgebwmQzw8PORgMC4uzqTtGjx4sLz8448/4uzZs3rTRkVFyXO/AtCYW1GJ58+fmzyfqPrpOXPfrf2mO3jwINavX693/bVr1zT+jOo6/nOjZcuWco91SkoKRo4cqTet6rta9X6WK1cOzZs310o3evRoeXndunUaf+BywtyfZ1OH00lPT8eDBw/k5+rHormOZ19fX42OnE8++QQxMTEmlZuZmal15/DTp09NvqZd0edM0dC8bwBDMyLo8sknn2iMgq16GBq5/ssvv5TTeXh46Bztf8eOHcLLy0tjhH1Du9eUNCo5nZVh6tSpcvqpU6fqTKNr7tFevXrpnHu0Z8+eGu39999/dZaZnp4umjVrplHu7Nmzdc5TJ0TWdCRr1qwRTZs2FR988IHONKaMqq5EdHS0xtyjDRo0ENHR0RppMjIyxPfff68x7+ebMPdo+/btxcOHDzXSpKSkiC+++ELj/crp3KO6Rlj/559/NOYebdy4sd7pY/7991+NdtavX18cP35c73adOXNGfP7558LT01PnPKG52Z+XLl3SmMFjzJgxWlNfPXjwQJ4SyNnZ2aTPWKVKleR069evN9qO7HOPFi1aVOfcvXv27JGn/wFMn3t03759Rtuwb98+UbRoUTFlyhRx/vx5nWnS09PFb7/9pvH99fvvvxst25CczohgyraY8t2WE+qfE0Nzjx47dkxjeypUqGDS3KOmbJMQ2nOPDhgwQGuE/sTERNG3b1+NdIbmHu3ataucTjX3qK45OlVzj3bq1Enn3KPm/Dz36dNHNGrUSKxYsUI8efJEZ/64uDiN+YDd3Nw0PrvmPJ7v378vihUrJqcpVqyY2Lhxo94pHe/duycWLFggKlSoIBYtWqSxbvny5aJs2bJi7ty5eqc4fPnypZg/f77Gd5OhqQN1sbig7d69e8LBwSFHQduzZ89E8eLF5bSSJIlGjRqJAQMGiI8++kjjizz7lDv6vElB27Rp0+QfZBcXF9GhQwcxcOBA0aFDB42ABoAYPXq0wbrj4+NFrVq1NPI4OTmJZs2aiT59+ohBgwaJrl27iho1amhMxdKlSxed5eV10CaEEOvWrdP4UrKxsRFNmzYVAwYMED179hQlSpTQ2J769evr/cIWIu+CNiGE+OabbzTaYm9vL95//30xcOBA8cEHH2hMXQVAhISE6A2usk/LNGrUKPl59erVRWhoqAgNDRXVqlXTKNPb29votGdhYWEa+1T1Q9etWzcxePBg0bt3b9GyZUuNQAWA2YI2ITTnlVV9IXfo0EEMGDBANG/eXP6z4urqKhYvXmzSZ0x9GiNbW1vRpk0bMXz4cDFmzBj5kd3FixdFkSJFNNpSo0YNef+qB3Wq/XvlyhW9bchN0KZevo+Pj3j//fdFnz59xIABA0S7du00frgAiKCgIL0/XKYqaEHbwoUL5eVy5cqJjz76SPTr109jKjLV91lERITeMnMTtAkhxGeffaZRj6urq/xd3LFjR+Hq6qqxfuTIkQbLS0hI0Jj2C8j6c9KyZUvRv39/0a9fP9GqVSuNSc71zb1srs+z+u+OtbW1qFy5svjggw/EwIEDxUcffSSaNGkify5Vj7CwMI0yzH08Hz9+XOt709vbW7Rv314MGDBA9OvXT3Ts2FGUK1dOI9jSFbSpl1GyZEnRrl070a9fP9G3b1/RunVrjT++QFZHSU5ZXNAmhBAjR47U2HGm/BicPHlS641Vf9jZ2ckTCKu/ro8paVTyOmhbvny5CA8P1whMsz+srKzEuHHjTJqcNzk5WXzyyScak18bejg6OoqZM2fqLOt1BG1CCLF161bh4+NjtK09e/bU+W9VXV4GbUII8euvvwo3NzeD7bS2thYjRowQ6enpesvJHrSlpqaKgQMHGiy3fPnyOgMrXfbu3asx2bSxR5UqVURMTIxWObndny9evBAtW7Y0WGeJEiXE4cOHTf6MJSQkiMqVKxssU5crV65o/ZnR9XjnnXeMBsQ5DQqOHj1q8mcRgPjggw9EYmKi0XKNKWhBm6p89R/m7I+iRYsabWdugzYhhJgxY4bW2ZrsDwcHB73fl9klJyeLgQMHagVc+so19L6b4/M8dOhQk/O7urpqBWxC5M3xfOvWLfHee++ZXKaPj4/WGYmNGzcaPHbUH1ZWVmLIkCF6e9MNKbBBW0JCgli7dq1GL4iDg4OwtbUVRYoUEcHBwWLOnDkiPj5eK+/Dhw81TocAEM2bNxchISHC19dX2NnZCV9fXxESEiL++ecfOV98fLz44osvRLVq1YSLi4twcnIS/v7+4rPPPhMXLlwQL168EHPnztUot1KlSmLMmDHi9u3bGm0w9iWv7nUEbUII8ejRI/H111+L2rVri8KFCwt7e3tRunRp0bdvXxEZGWm03uxu3rwpZsyYIZo2bSp8fX2Fg4ODsLOzE0WKFBH16tUTgwYNEuvXrxcJCQl6y3hdQZsQWadrFy5cKFq0aCGKFy8u7OzshLu7uwgICBBDhgwRR48eNamcvA7ahMh6r2bOnCmCgoKEj4+PsLW1FYUKFRI1atQQY8aMERcuXDBahr4J0Hfv3i26d+8uypQpIxwcHEShQoXEu+++K+bPn2+wh1GXjIwMsWnTJtG3b18REBAgPD09hbW1tXB1dRXly5cX7dq1EzNnzhRRUVF6y1CyPzMyMsTq1atF8+bNReHChYWtra0oVqyYaNiwofjuu+/kUzQ5+YwlJSWJb775RjRu3FgUKVJEo8fY0Oc5IyNDrF+/XnTr1k2ULl1aODk5CScnJ1G6dGnRvXt3sXHjRpP+FOUmKHjy5InYsGGDGD58uAgKChLFixcX9vb2wsbGRhQqVEjUqVNHDBs2TBw7dsyk8kxREIM2IbKCgr59+4ry5csLJycn4e7uLt555x0xY8YMvT1R6pQEbUJkBRCTJ08WderUEV5eXsLGxkZ4eXmJunXriv/9739avyWmuHTpkpg4caKoW7eu8Pb2FjY2NsLJyUmUK1dOhISEiJ9++kk8fvzYaDnm+DxfvHhRfP/99+LDDz8UNWrUEJ6ensLGxkY4ODgIX19f0bJlSzFv3jytSz/U5dXxHB4eLkaNGiVq164tvL29ha2trXB0dBQlSpQQTZs2FePGjRP//vuvSEtL05k/NjZWrFq1SgwePFjUr19feHt7Czs7O/l3r2HDhmLChAkmfT/rIwlRMEdQ3LNnj8YdGPp4eXnht99+0zvJtBACn3zyCcLCwvSWMWjQICxevNjgLf/R0dFo27Ytrly5onO9u7s71qxZgzZt2hhtM9HroD5vaKlSpcw+ATfRm079O72A/hSShSnQd4/6+fmhd+/eWLhwIf78809ERETgyJEjWL9+Pbp27Qpra2vEx8ejQ4cOeu/amjx5shyw1apVC2vXrsXx48exdu1aefycsLAwgwPyJiUloV27dnLANnDgQPz3338IDw/H119/DRcXFyQkJKBr164G7x4jIiIi0ivXfXT5zNC1OiqbN2+Wu6k7d+6stf7atWvyufHAwECtu8tevHghAgMDBZB1cbq+a03Uu+znzp2rtT48PFyup2nTpiZuIVHe0nd6lMhSwITT2kRvkgLb06YaadyQTp06ydPX6BrxfP78+fIUH4sWLdKa3NfJyQmLFi0CkDVujK4BY9PS0rBw4UIAQEBAAMaMGaOVpkGDBujfvz8AYN++fTh58qTRthMRERGpK7BBm6lUI4q/evVK43UhhDzoZqVKlTSmYVJXv359eXDdLVu2aF33sH//fjx79gwAEBoaqjESuzr1ydD//PPPHG8HERERWba3Omi7dOkSTp8+DQBaE0bfvHlTHv1YfQJXXVTr7927p3Wx9qFDh7TS6RIYGCgHkIcPHzap/UREREQqNvndAHNLTk5GTEwMtm7dirlz58qTgI8YMUIj3aVLl+Tl7AFddurrL126pDFJtqnl2NjYoFy5cjh79qxGHlNkZmbi/v37cHV1fW0TlNPbT33KLiEEEhMT87E1RPmLx/+bQQiB58+fo3jx4nrPXFmytyJoW7FiBfr27at3/dixY9GrVy+N1+7evSsvlyhRwmD5fn5+OvOpP3d2doaHh4fRcs6ePYtHjx4hJSVFnqw8u5SUFI25EGNiYlC5cmWDZRMpcefOHbi7u+d3M4jyDY//N8vdu3eN/jZborciaNOnZs2aWLx4MerVq6e1Tr2XwcXFxWA5qtOaQNbwHrrKMVaGrnL0BW2zZs3C9OnTtV6/e/cu3NzcjNZDRERUECUmJsLPzw+urq753ZQ30lsRtHXq1AmBgYEAgJcvXyI6OhobNmzA5s2b0atXLyxYsADt2rXTyKN+Y4KdnZ3B8tWDq5cvX+osx1gZxspRN3HiRIwePVp+rjqI3dzcGLQREdFbj5cC6fZWBG0eHh4apybr1KmDHj16YPXq1QgNDUXHjh2xdOlSjTs4HRwc5OXU1FSD5aufqsw+LIiqHGNlGCtHnb29vd5eOCIiIrJMb/VVfh9//DG6du2KzMxMDB06FE+fPpXXqXe9Zj/lmd2LFy/k5eynQVXlGCvDWDlEREREhrzVQRsAdOzYEUBWwPTPP//Ir6tf4Hjv3j2DZajffKB+U4J6OS9evJDHazNWTpEiRdiTRkRERDny1gdtRYoUkZdv374tL6vfjXn58mWDZaivDwgI0Fhnajnp6emIjo7WWQYRERGRMW990KYaQBfQPCVZpkwZFC9eHABw4MABg2WopsDy9fVF6dKlNdY1atRIXjZUzokTJ+TTow0bNjSt8URERET/31sftG3cuFFerlatmrwsSZJ86vTy5cs4evSozvxHjx6Ve9A6duyodUdLkyZN5PF9Vq5cqTXNlcqKFSvk5ZCQkJxvCBEREVm0Ahu0rVixQms+0ezmz5+PHTt2AABKly6t0SsGACNHjoSNTdYNtMOGDdMahuPly5cYNmwYgKwZDUaOHKlVh52dHYYPHw4ga3aEefPmaaWJiIjA0qVLAWRNdVWnTh0TtpCIiIjo/0hCX9fQG6506dJ4/vw5unTpgkaNGqFcuXJwcXHB8+fPce7cOfz+++84cuQIgKzAavv27WjevLlWORMnTsTs2bMBALVq1cKECRNQrlw5REdHY86cOYiKipLTzZw5U2dbnj9/jsDAQFy9ehUAMGjQIPTo0QOOjo7Yt28fZs6ciaSkJDg6OiI8PBw1a9bM0bYmJibC3d0dCQkJHKeNiIjeWvy9M6xAB23qNxboU6JECSxbtgwtWrTQuT4zMxMDBw7EsmXL9JbRv39/hIWFGZwH7fr162jTpg2uXbumc72bmxt+//13rUF+TcGDmIiILAF/7wwrsEFbdHQ09uzZg3379uHSpUt4+PAhHj9+DAcHB/j4+KBmzZpo164dunXrBicnJ6Pl7dixA2FhYYiMjER8fDy8vLxQp04dDB48GK1btzapTS9evMCPP/6IjRs34vr160hNTYWfnx/atGmDESNGoFSpUrnaVh7ERERkCfh7Z1iBDdosCQ9iIiKyBPy9M6zA3ohAREREZEkYtBEREREVAG/FhPGUZXq2MeSMmcoz40RERAUGe9qIiIiICgAGbUREREQFAIM2IiIiogKAQRsRERFRAcCgjYiIiKgAYNBGREREVAAwaCMiIiIqABi0ERERERUAioK2tLQ0XLx4ERcvXkRKSorW+levXmHMmDHw8/ODo6MjKleujB9++EFJlUREREQWSdGMCJs3b0bPnj3h5eWFu3fvaq0PCQnB7t27oZqT/vLlyxgxYgSuXbuGhQsXKqmaiIiIyKIo6mnbtWsXhBDo3Lkz7OzsNNZt374du3btAgCUKFECISEh8PX1hRACP/zwAyIiIpRUTURERGRRFAVtp06dgiRJaNy4sda65cuXAwD8/f1x4cIF/PHHHzh//jwCAgIAAL/++quSqomIiIgsiqKgLS4uDgBQtmxZjdczMzOxZ88eSJKEoUOHwtXVFQDg7u6OoUOHQgiB8PBwJVUTERERWRRFQVt8fDwAwMHBQeP106dPIzExEQDQtm1bjXVVq1YFAJ3XwBERERGRboqCNtV1bKrgTeXgwYMAsq5lK126tMY6Va9bRkaGkqqJiIiILIqioE0VkB07dkzj9a1bt+q91u3JkycAgCJFiiipmoiIiMiiKAramjZtCiEEFi1ahEuXLgEA/v77b+zfvx8A0KZNG60858+fBwAUK1ZMSdVEREREFkVR0DZs2DDY2dkhLi4OVatWhZeXF0JCQiCEgK+vL7p06aKVZ/fu3ZAkCYGBgUqqJiIiIrIoioK2ChUqYPXq1XBycoIQAk+ePIEQAh4eHli7dq3W2G2xsbH4999/AQAtWrRQUjURERGRRVE0IwIAdO3aFcHBwdi+fTtiY2NRrFgxdOjQAYUKFdJKe/bsWXz44YcAgGbNmimtmoiIiMhiSEI1xxS9sRITE+Hu7o6EhAS4ubnpTTddknJU7lS+9URE9AYx9ffOUik6PUpEREREr4fi06PqXr16hZMnTyI2NhbJycno2LEjI2UiIiIiMzBL0Hb37l1MnjwZ69evR1pamvz6uXPnULlyZfn50qVL8csvv8Dd3V2+i5SIiIiIjFN8evT48eOoVasWfvvtN6SmpkIIAX2XyXXo0AFnz57F3r17sXv3bqVVExEREVkMRUFbQkICOnbsiCdPnqBo0aL46aefcO7cOb3pixQpgtatWwMAtm/frqRqIiIiIoui6PTookWL8PDhQ3h5eSEiIgIlS5Y0mqdFixb466+/cPz4cSVVExEREVkURT1tqjlGR48ebVLABgBVqlQBAERHRyupmoiIiMiiKArarl27BgA6J4bXx8PDA0DWWCxEREREZBpFQdvLly8BAM7OzibnSUpKAgA4ODgoqZqIiIjIoigK2ooUKQIga8gPU508eRIAUKxYMSVVExEREVkURUFb3bp1AQD//POPSekzMjIQFhYGSZLQqFEjJVUTERERWRRFQVvPnj0hhMCyZcsQFRVlMG1mZiY++eQTXLx4EQDw0UcfKamaiIiIyKIoCtq6dOmCd999FykpKXjvvffw448/Ii4uTl4vSRIePnyI1atXIzAwEMuWLYMkSXj//ffRpEkTpW0nIiIishiS0Dd9gYni4+PRuHFjXL58WWtaKjs7O6SmpsrPhRCoVq0aDhw4IN9FSsYlJibC3d0dCQkJBudynZ7DacGmKnvriYiIzMrU3ztLpXgaKy8vL5w4cQKfffYZ7O3t5WmshBBISUmRl21sbDBo0CCEh4czYCMiIiLKIcU9beoeP36MXbt24cSJE4iLi0NGRgYKFy6MWrVqoXXr1ihevLi5qrIo7GkjIiJLwJ42wxRNY5Vd4cKF8eGHH+LDDz80Z7FEREREFk/x6VEiIiIiyntm7WnTJSUlBYcPH0Z8fDzKlCkjj+1GRERERKZTFLTdvn0bP/74IwDgiy++0LrB4OjRo/jggw/w4MED+bV33nkHf/zxh8kTzBMRERGRwtOjmzdvxrx587B3716tgO358+fo1KkTHjx4oHFH6cmTJ9G2bVukp6crqZqIiIjIoigK2v79919IkoROnTpprQsLC5MH2h0+fDj++usvDBkyBABw8eJFrFy5UknVRERERBZFUdB248YNAEDt2rW11m3YsAGSJCEkJAQLFixA+/bt8cMPP6Br164QQmDTpk1KqiYiIiKyKIqCNlVPmo+Pj8briYmJOHXqFACgb9++Gut69OgBADhz5oySqomIiIgsiqKg7fnz5wCAjIwMjdePHDmCjIwMWFtba80x6ufnBwB48uSJkqqJiIiILIqioM3d3R0AcP/+fY3X9+/fDwCoUaMGnJ2ddeZ1cHBQUjURERGRRVEUtFWtWhVA1l2kKhkZGfL1bE2bNtXKExMTA0D7lCoRERER6adonLaQkBAcOHAAq1evho+PD4KCgrB69Wrcvn0bkiShW7duWnlOnDgBABynjYiIiCgHFPW0DR48GAEBARBCYN68eejYsaN8V2j79u0RGBiolWfz5s2QJEnrWjciIiIi0k9R0GZvb4///vsPnTt3ho2NDYQQsLW1xccff4zVq1drpT948CAuXrwIAGjRooWSqomIiIgsiuK5R4sWLYpNmzYhJSUFT548QeHChWFnZ6czrZ+fH/bt2wcAqFOnjtKqiYiIiCyG2SaMt7e3R7FixQymKVOmDMqUKWOuKomIiIgshqLTo0RERET0eigO2pKTk5GcnKx3/aJFixAUFISAgAC0adMG27ZtU1olERERkcVRFLRt3boVrq6uKF68uDw7grp+/fph5MiRCA8Px5UrV7Br1y507NgRc+fOVVItERERkcVRFLTt2rULQgh06tQJrq6uGusOHz6MFStWAACcnJxQq1YtODg4QAiByZMn48KFC0qqJiIiIrIoioK2o0eP6p35ICwsDABQvHhxXLp0CSdPnsTly5fh5+eHjIwM/PLLL0qqJiIiIrIoioK2uLg4AECFChW01u3cuROSJGHYsGEoUaIEgKwhP4YNGwYhBA4cOKCkaiIiIiKLoihoe/ToEQDAxcVF4/WLFy8iPj4eANChQweNdapZEm7duqWkaiIiIiKLoihos7a2BgA8efJE4/VDhw4BAIoUKYJKlSpprPP09AQAvHr1SknVRERERBZFUdDm6+sLADh9+rTG69u3b4ckSQgKCtLKk5CQAADw8vJSUjURERGRRVEUtAUFBUEIgR9++EE+HRoZGYmdO3cCAFq1aqWV59KlSwCypr8iIiIiItMoCtqGDBkCKysr3Lx5E2XLlkVgYCCCg4ORnp4OT09PdO/eXSvP3r17IUkSatasqaRqIiIiIouiKGh755138M0330CSJCQlJeHUqVN49eoVbG1tsWTJEq2x2xISErB9+3YAQIsWLZRUTURERGRRFE8YP2rUKDRv3hybNm1CbGwsihUrhp49e6JixYpaaffv3486deoAAJo3b660aiIiIiKLIQkhRH43ggxLTEyEu7s7EhIS4ObmpjfddEnKUblT+dYTEdEbxNTfO0uleMJ4IiIiIsp7ik+PZhcTE4PY2FgkJycjMDAQjo6O5q6CiIiIyOKYpaft+fPnmDp1Kvz8/FCyZEnUrVsXTZo0wc2bNzXSrVu3Dt26dcPAgQPNUS0RERGRxVDc03b9+nW0bt0aN27cgPrlcZKO66saNGiAjz/+GJmZmQgNDUWjRo2UVk9ERERkERT1tKWkpKBt27aIjo6Gk5MTxo8fj23btulNX6pUKTRt2hQA8PfffyupmoiIiMiiKOppW7x4Ma5duwZnZ2ccOnTIpAFzW7dujT179iAiIkJJ1UREREQWRVFP259//glJkjBixAiTZzioXr06AODatWtKqiYiIiKyKIqCtosXLwIAWrZsaXKewoULAwCePXumpGoiIiIii6IoaHv+/DkAwN3d3eQ8r169AgDY2toqqZqIiIjIoigK2lS9Zg8fPjQ5z7lz5wAAPj4+SqomIiIisiiKgjbVdWz//fefyXmWLVsGSZJQr149JVUTERERWRRFQVvnzp0hhMAvv/yC27dvG00/ffp0HDt2DADQvXt3JVUTERERWRRFQVufPn0QEBCA58+fIzg4GNu3b9caYDczMxOHDh1C+/bt8eWXX0KSJNSpUwcdOnRQ3HgiIiIiS6FonDZra2v8/fffaNiwIe7cuYMOHTrAyclJXt++fXs8fPgQycnJAAAhBIoXL46NGzcqazURERGRhVE892i5cuVw+vRptG3bFkIIvHjxAkBWgHbjxg28ePECQggIIdCyZUtERkbCz89PccOJiIiILIniuUcBoGjRoti6dSsuXLiAv/76CydOnEBcXBwyMjJQuHBh1KpVCx07dkRgYKA5qiMiIiKyOGYJ2lSqVKmCKlWqmLNIIiIiIoIZTo8SERERUd5j0EZERERUACgK2q5evYqyZcvC398f9+/fN5o+JiYG5cuXR7ly5Uwa182YU6dOYebMmWjdujX8/Pxgb28PFxcX+Pv7o0+fPjh06FCOytu5cyc6d+6MEiVKwN7eHiVKlEDnzp2xc+dOk8tITk7GN998g7p166JQoUJwcXFBQEAAxo4dizt37uR0E4mIiIgAAJJQH1gth2bMmIGpU6fi/fffx44dO0zK06ZNG+zatQszZ87EhAkTcls1goODcfDgQaPpPv74Y/z666+ws7PTm0YIgU8++QRhYWF60wwaNAiLFy+GJEl600RHR6Nt27a4cuWKzvXu7u5Ys2YN2rRpY7Td6hITE+Hu7o6EhAS4ubnpTTfdQNt0mZr7t56IiMjsTP29s1SKetp27doFSZLQvn17k/N07NgRQgiTgzx9YmJiAADFixfHiBEjsGnTJhw/fhwRERH47rvv4OvrCwBYvXo1+vTpY7CsyZMnywFbrVq1sHbtWhw/fhxr165FrVq1AABhYWH43//+p7eMpKQktGvXTg7YBg4ciP/++w/h4eH4+uuv4eLigoSEBHTt2hVnz55VtO1ERERkeRT1tJUsWRIxMTE4ePAgGjZsaFKeI0eOICgoCKVKlcLNmzdzWzXatWuH3r17o0uXLrC2ttZaHx8fj4YNG+Lq1asAgIMHDyIoKEgr3fXr1xEQEID09HQEBgbi4MGDcHR0lNcnJycjODgYJ06cgI2NDS5fvoxy5cpplTNt2jRMnz4dADB37lyMGzdOY31ERAQaN26M9PR0NG3aFHv37jV5W9nTRkREloA9bYYp6mmLi4sDALi4uJicR5U2NjZWSdXYtm0bunXrpjNgAwAvLy98++238vNNmzbpTDd//nykp6cDABYtWqQRsAGAk5MTFi1aBABIT0/HggULtMpIS0vDwoULAQABAQEYM2aMVpoGDRqgf//+AIB9+/bh5MmTRraQiIiI6P8oCtrc3d0B5CwAU6VVn+4qrzRp0kRejo6O1lovhMBff/0FAKhUqRLq16+vs5z69eujYsWKAIAtW7Yge+fk/v378ezZMwBAaGgorKx071b107R//vmnqZtBREREpCxoq1ChAgDk6O7Kf/75BwB0nmI0t9TUVHlZVyB18+ZN+dq44OBgg2Wp1t+7dw+3bt3SWKd+l6qhcgIDA+Hs7AwAOHz4sOHGExEREalRFLS1atUKQgiEhYXh0qVLRtNfuHABS5YsgSRJeP/995VUbZIDBw7Iy5UqVdJar95mXevVqa/Pvq2mlmNjYyMHq6bsLyIiIiIVRUHbp59+CmdnZ7x69QrNmjXD1q1b9ab9+++/0bx5c7x8+RKOjo747LPPlFRtVGZmJmbPni0/79atm1aau3fvysslSpQwWJ76JPfq+dSfOzs7w8PDw6RyHj16hJSUFJ1pUlJSkJiYqPEgIiIiy6Zo7lEvLy8sXrwYH3/8MeLi4tCpUyeUKVMGQUFBKFasGCRJwv3793Ho0CHcvHkTQghIkoSff/4ZPj4+5toGnebPn4/jx48DAEJCQnROVv/8+XN52djNFKrTmkDW8B66yjHlhozs5djb22ulmTVrlnwnKhERERFghgnje/XqhYyMDAwZMgTJycm4ceOG1lAeqgv3nZ2d8fPPP+Ojjz5SWq1BBw4cwOeffw4A8Pb2xs8//6wz3atXr+RlQ4PvAtAIrl6+fKmzHGNlGCtHZeLEiRg9erT8PDExUaOnj4iIiCyPWeYe7d27N65du4YJEyagWrVqALICNVXPWvXq1TFp0iRcv349zwO2CxcuICQkBOnp6bC3t8eGDRv09uo5ODjIy+o3Leiifioz+7AgqnKMlWGsHBV7e3u4ublpPIiIiMiyKe5pUylWrBhmzZqFWbNmIT09HU+ePAEAFCpUCDY2ZqvGoJs3b6Jly5Z4+vQprK2tsXbtWoN3c7q6usrL2U95ZvfixQt5OftpUFU5xsowVg4RERGRPmbpacvOxsYG3t7e8Pb2fm0B2/3799G8eXPcv38fkiRh2bJlCAkJMZhH/eaDe/fuGUyrfvNB9lOVqnJevHghj9dmrJwiRYrovJ6NiIiISJc8Cdpet/j4eLRo0QI3btwAkDWzQe/evY3mq1y5srx8+fJlg2nV1wcEBOSqnPT0dHmQ3+xlEBERERlitm6wjIwMbNmyBXv27MH58+c1To9WrVoVzZs3R6dOnfROO5VbCQkJaNWqFS5evAgAmD17tsnDiZQpUwbFixfH/fv3NcZ00+XgwYMAAF9fX5QuXVpjXaNGjeTlAwcO6J1Z4cSJE/LpUVPnaiUiIiICzNTTtnPnTpQpUwbdunVDWFgYwsPDcenSJVy6dAnh4eEICwtDt27dULZsWezatcscVQLImsy9bdu2OHXqFABg0qRJmDBhgsn5JUlCx44dAWT1kB09elRnuqNHj8o9aB07doSUbWL2Jk2ayFN6rVy5UmuaK5UVK1bIy8ZO3RIRERGpUxy0rV69Gu3atUNMTIx8x2ipUqVQv3591KtXD6VKlQKQdTfp3bt30bZtW/z++++KG56amoqQkBAcOXIEADBixAh89dVXOS5n5MiR8nV3w4YN0xqG4+XLlxg2bBiArGv1Ro4cqVWGnZ0dhg8fDiBrpoN58+ZppYmIiMDSpUsBZE11VadOnRy3lYiIiCyXJPR1C5ng9u3bqFSpElJSUuDs7IyJEydiwIAB8Pb21kj36NEj/Prrr5g1axaSkpLg4OCAy5cvo2TJkrlueJcuXeRJ15s1a4YFCxZo9YCps7Ozg7+/v851EydOlGdPqFWrFiZMmIBy5cohOjoac+bMQVRUlJxu5syZOst4/vw5AgMDcfXqVQDAoEGD0KNHDzg6OmLfvn2YOXMmkpKS4OjoiPDwcNSsWdPkbU1MTIS7uzsSEhIMDv8x3cD26zI19289ERGR2Zn6e2epFAVto0ePxoIFC+Di4oKDBw8aDUTOnDmDoKAgvHjxAiNHjsS3336b26oNBmi6lCpVSmuid5XMzEwMHDgQy5Yt05u/f//+CAsL0znxvMr169fRpk0bXLt2Ted6Nzc3/P7772jXrl2O2s6gjYiILAGDNsMUnR7dvXs3JEnCuHHjTOo5qlGjBsaOHQshhFmvbVPKysoKS5cuxfbt29GxY0cUL14cdnZ2KF68ODp27IgdO3bg119/NRiwAUD58uURFRWFOXPmIDAwEB4eHnByckLFihUxatQonD17NscBGxERERGgsKfNzc0NL168wOHDh9GgQQOT8kRERKBhw4ZwcXHhROgmYk8bERFZAva0Gaaopy0jIwMAcjSMhyptZmamkqqJiIiILIqioM3X1xcAEB4ebnIeVdrixYsrqZqIiIjIoigK2po2bQohBGbPno379+8bTX/v3j3Mnj0bkiShWbNmSqomIiIisiiKgrZhw4bBysoKjx49Qr169bBx40b5lKm6jIwMbNiwAQ0aNEBcXBysrKwwdOhQJVUTERERWRRF01hVrVoVM2bMwKRJk3D//n306NEDHh4eqFWrFnx8fCBJEmJjYxEVFYVnz57JMwXMmDEDVatWNcsGEBEREVkCxXOPTpw4Ee7u7hg/fjySk5Px9OlT7Nu3TyONKlhzcnLCN998g08//VRptUREREQWxSwTxg8ZMgTdunXD8uXLDU4Y37dvX3h5eZmjSiIiIiKLomicNno9OE4bERFZAo7TZpiiGxGsrKxgY2ODuXPnmqs9RERERKSDoqDNzs4OQggEBQWZqz1EREREpIOioE01QG5OZkQgIiIiopxTFLQ1btwYAHDq1CmzNIaIiIiIdFM8uK61tTXmzZvHyd+JiIiI8pCioK127dpYtGgRbt++jeDg4BzNQUpEREREplM0Tlu/fv0AABUrVsSZM2cQFBQEPz8/VK9eHZ6engavdZMkCUuXLlVSPREREZHFUDROm5WVFSS1scFURUlGxgsTQkCSJJ3zlJI2jtNGRESWgOO0Gaaop61kyZJGAzQiIiIiUk5R0Hbr1i0zNYOIiIiIDFF0IwIRERERvR4M2oiIiIgKAEWnR/VJT0/H06dPAQCenp6wscmTaoiIiIgshtl62i5cuIBhw4YhICAADg4OKFq0KIoWLQoHBwcEBARg2LBhOH/+vLmqIyIiIrIoioO2zMxMjBw5EjVr1sRPP/2EK1euIDMzE0IICCGQmZmJK1eu4KeffkKtWrUwatQoZGZmmqPtRERERBZD8XnLHj164I8//pDHaKtSpQrq1q0LHx8fCCEQFxeHyMhInD9/HhkZGfj+++9x//59rF+/XnHjiYiIiCyFoqBtzZo12LRpEyRJQo0aNRAWFoY6deroTHvixAkMHjwYUVFR2LRpE9atW4cePXooqZ6IiIjIYig6PbpkyRIAgL+/Pw4fPqw3YAOAwMBAHDx4EBUrVoQQAr/88ouSqomIiIgsiqKg7ezZs5AkCRMmTICzs7PR9M7OzpgwYQIA4MyZM0qqJiIiIrIoioK21NRUAED16tVNzqNKm5aWpqRqIiIiIouiKGgrVaoUACAhIcHkPImJiRp5iYiIiMg4RUFbly5dIITAH3/8YXIe1Y0LISEhSqomIiIisiiKgrbRo0ejbNmy+OWXX7Bhwwaj6Tdt2oRffvkFZcqUwdixY5VUTURERGRRFAVt7u7u2LNnD9555x307NkTnTp1wpYtWxATE4O0tDSkp6cjJiYGW7ZsQUhICLp374533nkH//33H9zd3c21DURERERvPUmoRsXNBWtra3lZCAFJkgymNyWNJElIT0/PbZPeSomJiXB3d0dCQgLc3Nz0pptuZN9mNzX3bz0REZHZmfp7Z6kUDa6bPd4zJf5TECMSERERWSxFQdvUqVPN1Q4iIiIiMoBBGxEREVEBoOhGBCIiIiJ6PRi0ERERERUADNqIiIiICgAGbUREREQFAIM2IiIiogKAQRsRERFRAcCgjYiIiKgAYNBGREREVAAwaCMiIiIqAEwK2jp37owuXbrg3r17ed0eIiIiItLBpKBty5Yt2LJlCxITEzUzW1nBxsYGFy9ezJPGEREREVGWHJ0eFUKY9BoRERERmZdJQZurqysA4OHDh3naGCIiIiLSzaSgrVKlSgCAhQsXIikpSWu9JEnmbRURERERabAxJdGHH36IyMhIbNu2DYUKFYKPjw9sbW3l9S1bttR4bgpJkhAdHZ2z1hIRERFZKJOCtmHDhuHIkSPYtGkT0tPTERMTI68TQmg8NxV754iIiIhMZ1LQZmVlhQ0bNiAiIgJ79uxBTEwMUlJSsHLlSkiShA4dOsDDwyOPm0pERERkuSSh4PZPKysrSJKEc+fOoXLlyuZsF6lJTEyEu7s7EhIS4Obmpjfd9Bz2Xk7lnb9ERPQGMfX3zlJxRgQiIiKiAsCk06P6ZGZmmqsdRERERGQAe9qIiIiICgBFPW3qMjIysGXLFuzZswfnz5/HkydPAACFChVC1apV0bx5c3Tq1AnW1tbmqpKIiIjIYpglaNu5cycGDRqkNRQIkDW0R3h4OMLCwlCiRAmEhYWhVatW5qiWiIiIyGIoPj26evVqtGvXDjExMRBCQAiBUqVKoX79+qhXrx5KlSoFICuIu3v3Ltq2bYvff/9dccOJiIiILImioO327dsYNGgQMjMz4eTkhK+++gqxsbG4ceMGwsPDERERgRs3biA2NhZff/01XFxckJmZiYEDB+LOnTvm2gYiIiKit56ioG3hwoVISUmBi4sLDh06hC+++ALe3t5a6YoUKYKJEyfi0KFDcHFxQUpKChYuXKikaiIiIiKLoiho2717NyRJwrhx41CzZk2j6WvUqIGxY8dCCIFdu3YpqZqIiIjIoigK2lSnOJs3b25ynhYtWmjkJSIiIiLjFAVtGRkZAJCjYTxUaTkwLxEREZHpFAVtvr6+AIDw8HCT86jSFi9eXEnVRERERBZFUdDWtGlTCCEwe/Zs3L9/32j6e/fuYfbs2ZAkCc2aNVNSNREREZFFURS0DRs2DFZWVnj06BHq1auHjRs3yqdM1WVkZGDDhg1o0KAB4uLiYGVlhaFDhyqpmoiIiMiiKJoRoWrVqpgxYwYmTZqE+/fvo0ePHvDw8ECtWrXg4+MDSZIQGxuLqKgoPHv2TJ4lYcaMGahatapZNoCIiIjIEiiexmrixIlwd3fH+PHjkZycjKdPn2Lfvn0aaVTBmpOTE7755ht8+umnSqslIiIisihmmXt0yJAh6NatG5YvX25wwvi+ffvCy8vLHFUSERERWRRJqLrB6I2VmJgId3d3JCQkwM3NTW+66ZKUo3Kn8q0nIqI3iKm/d5ZK8YTxRERERJT3GLQRERERFQAM2oiIiIgKAAZtRERERAUAgzYiIiKiAoBBGxEREVEBwKCNiIiIqABg0EZERERUADBoIyIiIioAFE1j9eWXXwIA6tWrh1atWpmlQURERESkTVHQNm3aNEiShM2bN5urPURERESkg6LTo4ULFwYAlCxZ0iyNISIiIiLdFAVt5cuXBwDExsaapTFEREREpJuioK179+4QQmDDhg3mag8RERER6aAoaBsyZAhq1KiBVatWYeXKleZqExERERFlo+hGhNjYWPz666/o378/+vXrh99//x0ffvghqlevDk9PT1hbWxvMz2vhiIiIiEyjKGgrXbo0JEkCAAgh8N9//+G///4zKa8kSUhPT1dSPREREZHFUDy4rhACQgiNZVMfSsXFxWHbtm2YMmUKWrduDS8vL0iSBEmS0KdPnxyXt3PnTnTu3BklSpSAvb09SpQogc6dO2Pnzp0ml5GcnIxvvvkGdevWRaFCheDi4oKAgACMHTsWd+7cyXGbiIiIiACFPW3Lly83VztyxcfHxyzlCCHwySefICwsTOP1mJgYbN68GZs3b8agQYOwePFiuWdRl+joaLRt2xZXrlzReP3y5cu4fPkyfv31V6xZswZt2rQxS7uJiIjIcigK2kJDQ83VDsX8/PwQEBCA3bt35zjv5MmT5YCtVq1aGD9+PMqVK4fo6GjMnTsXUVFRCAsLQ5EiRfDVV1/pLCMpKQnt2rWTA7aBAweiR48ecHR0xL59+zBr1iwkJCSga9euiIiIQPXq1XO/sURERGRxFAVt+W3KlCmoU6cO6tSpAx8fH9y6dQtlypTJURnXr1/H3LlzAQCBgYE4ePAgHB0dAQB16tRBhw4dEBwcjBMnTmDOnDno27cvypUrp1XOvHnzcPnyZQDA3LlzMW7cOHldgwYN0LRpUzRu3BjJyckYOXIk9u7dm9vNzhPTDfQg6jLVDKe3iYiIyHQFesL46dOno127dopOk86fP1++IWLRokVywKbi5OSERYsWAQDS09OxYMECrTLS0tKwcOFCAEBAQADGjBmjlaZBgwbo378/AGDfvn04efJkrttMRERElsdsQVtmZib27t2Lr7/+GsOGDUP//v3x4MEDjTSpqalITk5GSkqKuapVRAiBv/76CwBQqVIl1K9fX2e6+vXro2LFigCALVu2aN1EsX//fjx79gxA1iljKyvdu1X95og///xTYeuJiIjIkpglaNu+fTsqVKiAFi1aYMqUKfjpp5+wYsUKPH36VCPd0qVL4erqCm9vb7x48cIcVSty8+ZNxMTEAACCg4MNplWtv3fvHm7duqWx7tChQ1rpdAkMDISzszMA4PDhw7lpMhEREVkoxUHbr7/+ig4dOuDmzZsQQqBw4cJ6h/Po378/PDw8kJSUhM2bNyutWrFLly7Jy5UqVTKYVn29er6clGNjYyNfD5e9DCIiIiJDFAVt169fx2effQYAaNasGS5evIi4uDi96e3s7NClSxcIIXJ1l6e53b17V14uUaKEwbR+fn4686k/d3Z2hoeHh0nlPHr0SO9p4pSUFCQmJmo8iIiIyLIpCtoWLFiAtLQ0VKlSBTt27DDaWwUAQUFBAIDTp08rqdosnj9/Li+7uLgYTKs6rQlkDe+hqxxjZRgrR2XWrFlwd3eXH+oBIxEREVkmRUHbf//9B0mSMHLkSNjZ2ZmUR3V68E2YHeDVq1fysrH229vby8svX77UWY4p+8BQOSoTJ05EQkKC/Mjes0dERESWR9E4bapgombNmibnUfU0JScnK6naLBwcHOTl1NRUg2nVT2VmHxZEVY6xMoyVo2Jvb68R3BEREREp6mlTnyzeVI8ePQIAuLm5KanaLFxdXeVlfacqVdTvds1+GlRVjrEyjJVDREREpI+ioK148eIAgKtXr5qc58CBAwCA0qVLK6naLNRvPrh3757BtOqnKLNfY6Yq58WLF/J4bcbKKVKkCHvTiIiIyGSKgrbGjRtDCIE1a9aYlD4+Ph6//PILJElCs2bNlFRtFpUrV5aXVVNQ6aO+PiAgIFflpKenIzo6WmcZRERERIYouqZt0KBBWLp0KXbs2IHly5ejb9++etPeu3cPnTt3Rnx8PGxsbDBo0CAlVZtFmTJlULx4cdy/f1/uAdTn4MGDAABfX1+tXsJGjRrJywcOHNA7s8KJEyfk06MNGzZU0PI3S07mLeWcpURERLmjqKetTp06+OSTTyCEwIABA9C1a1ds2LBBXn/27FmsX78e/fv3R8WKFXHy5ElIkoQxY8agfPnyihuvlCRJ6NixI4CsHrKjR4/qTHf06FG5B61jx47ytXwqTZo0gbu7OwBg5cqVeq/xW7FihbwcEhKitPlERERkQRTPiLBo0SJ8/PHHEELgzz//RM+ePeWgplevXvjwww+xYsUKvHz5EkIIhIaGYubMmYobbi4jR46EjU1Wh+OwYcO0huF4+fIlhg0bBiBrRoORI0dqlWFnZ4fhw4cDyJrpYN68eVppIiIisHTpUgBZU13VqVPHnJtBREREbzlFp0cBwNraGitXrkSHDh0wa9YsnDp1Sme6ypUrY/LkyejRo4fSKmWHDx/G9evX5efx8fHy8vXr1zV6tgDNCdtV/P39MXbsWMyePRsnTpxAw4YNMWHCBJQrVw7R0dGYM2cOoqKiAADjxo1DhQoVdLZl3LhxWL9+Pa5evYrx48fj+vXr6NGjBxwdHbFv3z7MnDkT6enpcHR0xIIFCxRvOxEREVkWSeRkvA4T3L9/HydOnEBcXBwyMjJQuHBh1KpVSx5U15z69OmDlStXmpxe36ZmZmZi4MCBWLZsmd68/fv3R1hYGKys9HdOXr9+HW3atMG1a9d0rndzc8Pvv/+Odu3amdxmAEhMTIS7uzsSEhIMDpWSk2vLAM3ry15XXl7TRkRE+pj6e2epFPe0ZVe8eHF06NDB3MXmKSsrKyxduhRdunRBWFgYIiMjER8fDy8vL9SpUweDBw9G69atjZZTvnx5REVF4ccff8TGjRtx/fp1pKamws/PD23atMGIESNQqlSp17BFRERE9LYxe08bmR972oiIyBKwp80ws/a0paWl4dSpUzh//jyePHkCAChUqBCqVq2Kd955B7a2tuasjoiIiMhimCVoS0pKwowZM7B06VI8ffpUZxpPT0/0798fkydP1pg+ioiIiIiMUzzkx6VLl1ClShXMmzcPT548gRBC5+PJkyeYN28eqlWrhitXrpij7UREREQWQ1FP27Nnz9C8eXM8ePAAAFC1alWEhoaibt268PHxgRACcXFxiIyMxMqVK3Hu3DncuXMHzZs3x/nz5+UBaYmIiIjIMEU9bXPmzMGDBw8gSRJmzJiBM2fOYMyYMQgKCoK/vz8qVqyIoKAgjB49GqdPn8ZXX30FIGtYkDlz5phlA4iIiIgsgaKgbcuWLZAkCd27d8ekSZO0pndSJ0kSvvjiC3Tv3h1CCGzevFlJ1UREREQWRVHQdvv2bQBAaGioyXlUsxKo8hIRERGRcYqCNtVdoN7e3ibnUaV1cXFRUjURERGRRVEUtFWrVg0A9E7bpIsqrSovERERERmnKGgbPHgwhBBYsGABMjMzjabPzMzE/PnzIUkSBg0apKRqIiIiIouiKGjr2rUr+vbti6NHj6JTp06IjY3Vm/bhw4fo3Lkzjh07hj59+qB79+5KqiYiIiKyKCaN07Zq1Sq964KDg3H+/Hls27YNZcuWRcuWLVGnTh14e3tDkiQ8fPgQkZGR2L17N1JSUlCnTh0EBwdj1apV6N27t9k2hIiIiOhtZtKE8VZWVgaH81ARQuhNl32dJElIT0/PQVMtFyeMJyIiS8AJ4w0zeUYEE2I7o+lMLYOIiIiINJkUtN28eTOv20FEREREBpgUtJUqVSqv20FEREREBii6e5SIiIiIXg8GbUREREQFAIM2IiIiogLALEHbpUuXMGrUKAQGBqJQoUKwtbWFtbW1wYeNjck3rhIRERFZPMWR0+zZszFlyhRkZGRwSA8iIiKiPKIoaNu4cSO++OILAFkD8AYFBaFGjRrw8PCAlRXPvBIRERGZi6KgbeHChQAAX19f7NixA9WqVTNLo4iIiIhIk6LusLNnz0KSJMyYMYMBGxEREVEeUhS02draAgBq1qxpjrYQERERkR6KgjZ/f38AwOPHj83SGCIiIiLSTVHQFhoaCiEEtmzZYqbmEBEREZEuioK2/v37IygoCGFhYdi6dau52kRERERE2Si6e9TW1hZ//fUXQkNDERISgh49eqBbt27w9/eHk5OT0fwlS5ZUUj0RERGRxVA8uK6HhweGDx+Oo0ePYu3atVi7dq1J+SRJQnp6utLqiYiIiCyC4hFwR44ciZYtWyI+Ph5CiBw9iIiIiMg0inrafvvtN3z//fcAADc3N4SEhKB69eqcEYGIiIjIzBQFbYsWLQIAVKpUCfv27YOPj49ZGkVEREREmhR1h12+fBmSJGHatGkM2IiIiIjykFlmRFANsktEREREeUNR0FapUiUAQGxsrFkaQ0RERES6KQra+vbtCyGEycN8EBEREVHuKJ4RoV27dvjtt9/www8/mKtNRERERJSNortHDx48iOHDh+PRo0cYMWIE1qxZgx49epg8I0Ljxo2VVE9ERERkMRQFbU2aNIEkSfLzY8eO4dixYybl5YwIRERERKZTPI0VZzYgIiIiynuKgrZ9+/aZqx1EREREZICioC04ONhc7SAiIiIiAzhBKBEREVEBwKCNiIiIqABg0EZERERUACi6pq1Zs2a5zitJEv777z8l1RMRERFZDEVB2/79+yFJksFhP9THcQP+b4iQ7K8TERERkX6KgrbGjRsbDb5evHiBa9euISEhAZIkwd/fH8WKFVNSLREREZHFUdzTZgohBLZv344RI0bgyZMn+PXXX9GoUSMlVRMRERFZlNdyI4IkSWjXrh0OHz4Ma2trhISEICYm5nVUTURERPRWeK13jxYrVgyjR4/G48ePMXfu3NdZNREREVGB9tqH/FCdFt2+ffvrrpqIiIiowHrtQZudnR0A4P79+6+7aiIiIqIC67UHbYcPHwYAODk5ve6qiYiIiAqs1xq0RURE4Msvv4QkSahbt+7rrJqIiIioQFM05MeXX35pNE1mZiaePn2KEydO4NixY8jMzIQkSRg1apSSqomIiIgsiqKgbdq0aTma2UAIARsbG8ydOxctWrRQUjURERGRRVEUtAEwOIUVkDVGm6urK8qUKYPg4GAMGjQIlStXVlotERERkUVRFLRlZmaaqx1EREREZIDinjai/DA9B6flAWCqkR5hIiKiN91rH/KDiIiIiHKOQRsRERFRAWDW06PPnz/HzZs38fz5c2RkZBhN37hxY3NWT0RERPTWMkvQtmTJEvz00084e/asyXkkSUJ6ero5qiciIiJ66ykK2jIyMtClSxds3boVgPHhP4iIiIgodxQFbYsXL8bff/8NAPDx8UHfvn1Ru3ZtFCpUCFZWvFyOiIiIyFwUBW2rVq0CAFSuXBmHDh2Cp6enWRpFRERERJoUdYddunQJkiThf//7HwM2IiIiojxklnOYFStWNEcxRERERKSHoqCtQoUKAIAnT56YpTFEREREpJuioK1Hjx4QQmDbtm3mag8RERER6aAoaBs+fDiqV6+On3/+GYcOHTJXm4iIiIgoG0VBm729PXbv3o3atWujRYsWGD9+PE6fPo1Xr16Zq31EREREBIVDflhbW8vLQgh8++23+Pbbb03KyxkRiIiIiEynKGjLPgMCZ0SgnJguSTlKP5XHFxERWTBFQdvUqVPN1Q4iIiIiMoBBGxEREVEBwAlCiYiIiAoABm1EREREBQCDNiIiIqICgEEbERERUQHAoI2IiIioAGDQRkRERFQAMGgjIiIiKgAYtBEREREVAAzaiIiIiAoABm1EREREBQCDNiIiIqICgEEbERERUQHAoC0P3LlzB2PHjkVAQACcnZ1RqFAh1K1bF/PmzUNycnJ+N4+IiIgKIJv8bsDbZvv27ejVqxcSEhLk15KTkxEZGYnIyEj8+uuv2LFjB8qWLZuPrbRs0yUpR+mnCmGWvEREREowaDOjM2fOoFu3bkhOToaLiwsmTpyIpk2b4uXLl1i3bh2WLFmCK1euoG3btoiMjISLi0t+N5lILwaoRERvFgZtZjRy5EgkJyfDxsYGu3fvRoMGDeR1zZo1Q4UKFTB+/HhcvnwZ3333HaZMmZKPraXXLSdBEAMgIiLKjkGbmURGRmL//v0AgP79+2sEbCpjxozB8uXLcenSJSxYsAATJ06Era3ta24pFTQF8XRuQeulK2jtJSLLxKDNTLZs2SIv9+3bV2caKysr9O7dGxMnTsTTp0+xf/9+tGjR4jW1kOjtxsCLiN52vHvUTA4dOgQAcHZ2Ru3atfWmCw4OlpcPHz6c5+0iIiKitwODNjO5dOkSAKB8+fKwsdHfgVmpUiWtPERERETG8PSoGbx69Qrx8fEAgBIlShhM6+npCWdnZ7x48QJ3797VmSYlJQUpKSnyc9XwIYmJiYbbkZNGZyvvdeXNjzoLYt6C1l5z5p3l7m5yvolqQ+sUhPYCmm1WIrf76XXVac56yXKoPluCly/oJAnuGcUePXoEb29vAED37t2xbt06g+l9fHwQFxeHqlWr4ty5c1rrp02bhunTp+dJW4mIiN50d+/eNdoJYonY02YGr1793/90Ozs7o+nt7e0BAC9fvtS5fuLEiRg9erT8/NmzZyhVqhTu3LkD9xz+07UkiYmJ8PPzw927d+Hm5pbfzXljcT+ZhvvJNNxPpuF+Mo0QAs+fP0fx4sXzuylvJAZtZuDg4CAvp6amGk2vOvXp6Oioc729vb0c2Klzd3fnh90Ebm5u3E8m4H4yDfeTabifTMP9ZBw7J/TjjQhm4OrqKi8nJSUZTf/ixQsA4IwIREREZDIGbWbg4OAALy8vAMC9e/cMpn369KkctPn5+eV524iIiOjtwKDNTAICAgAA169fR3p6ut50ly9f1spjjL29PaZOnarzlCn9H+4n03A/mYb7yTTcT6bhfiJz4N2jZvLFF19g1qxZAICjR4+iXr16OtPNnj0bEydOBADs2rULLVu2fG1tJCIiooKLPW1m0qlTJ3l5+fLlOtNkZmZi1apVAAAPDw80bdr0dTSNiIiI3gIM2sykbt26CAoKAgAsXboUERERWmm+/fZbeRaEESNGcLJ4IiIiMhlPj5pRVFQUGjZsiJcvX8LFxQVffPEFmjZtipcvX2LdunUICwsDAPj7++PEiRMad50SERERGcKgzcy2bt2Kjz76SO+UU/7+/ti+fTvKly//mltGREREBRlPj5pZ+/btcfbsWYwaNQr+/v5wcnKCh4cHAgMDMWfOHERFRZkcsN25cwdjx45FQEAAnJ2dUahQIdStWxfz5s1DcnJyHm/Jm02SJJMeTZo0ye+m5pm4uDhs27YNU6ZMQevWreHl5SVvd58+fXJc3s6dO9G5c2eUKFEC9vb2KFGiBDp37oydO3eav/GvkTn204oVK0w+5lasWJGn25NXTp06hZkzZ6J169bw8/ODvb09XFxc4O/vjz59+uDQoUM5Ku9tPZ5y6tGjR1i+fDl69+6NqlWrwtXVFXZ2dihatCjef/99/PLLL3pnxzHF+PHjNY6//fv3m6/x9OYR9Ebatm2bcHd3FwB0PipWrCiio6Pzu5n5Rt9+yf4IDg7O76bmGUPbHRoaanI5mZmZYtCgQQbLGzRokMjMzMy7jclD5thPy5cvN/mYW758eZ5uT15o3LixSdv28ccfi5SUFINlve3HU06EhYUJa2tro/u1QoUK4syZMzku//Tp08LGxkajrH379pl/Q+iNwWms3kBnzpxBt27dkJycDBcXF0ycOFHj2rglS5bgypUraNu2LSIjIy16ZoVPP/0UQ4YM0bve2dn5NbYm//j5+SEgIAC7d+/Ocd7JkyfL11vWqlUL48ePR7ly5RAdHY25c+ciKioKYWFhKFKkCL766itzN/21UrKfVHbt2mVwXsSCOMl1TEwMAKB48eLo2rUrgoKCULJkSWRkZCAiIgLffvstYmJisHr1aqSnp2PNmjV6y7Kk48mYhw8fIiMjA3Z2dmjXrh1atmyJgIAAuLq6Ijo6GkuWLMHu3btx7do1NG/eHKdOnTL5+MnMzMTAgQORnp4Ob29vxMXF5fHW0Bshv6NG0takSRMBQNjY2Ijw8HCt9XPnzpX/VU2fPj0fWpj/VNs/derU/G5KvpkyZYrYunWriI2NFUIIcfPmzRz3IF27dk3+px4YGCiSk5M11r948UIEBgbKx+P169fNvRl5zhz7Sb2n7ebNm3nX2HzStm1bsX79epGenq5z/aNHj4S/v7+8Dw4ePKgznSUcTznx3XffiQkTJoi4uDi9aUaPHi3v1379+plc9vz58wUAUalSJTFx4kT2tFkIBm1vmOPHj8sfvsGDB+tMk5GRIQICAgQA4enpKVJTU19zK/MfgzZtuQlGhgwZIueJiIjQmSYiIkJOM3ToUDO2OH8waMudrVu3yvtg+PDhOtMU1OPp559/Fk+fPs2XulNSUkSxYsUEAOHh4WHSaeM7d+4IFxcXOUibOnUqgzYLwRsR3jBbtmyRl/v27aszjZWVFXr37g0gay5TXnhKuSGEwF9//QUAqFSpEurXr68zXf369VGxYkUAWcen4A3nFkn9pp7o6Git9QX5ePr0009RrFgx9OjRA//88w8yMjJeW912dnZo2LAhAODZs2d4/Pix0TxDhgxBUlISQkND3+qbrUgbg7Y3jOoOLWdnZ9SuXVtvuuDgYHn58OHDed4uevvcvHlTvpZJ/XjSRbX+3r17uHXrVl43jd5Aqamp8rKVlfZPR0E+niRJwqtXr7B+/Xq0adMGfn5+GD9+PC5evPha6k9JSZGXde1bdRs2bMC2bdtQqFAhfPPNN3ndNHrDMGh7w6hmTChfvjxsbPTfJ1KpUiWtPJZo48aNqFixIhwdHeHq6ooKFSogNDQU+/bty++mvfHUjxv140kXHm//p0+fPvDx8YGdnR28vLxQv359TJ48WQ5Y3lYHDhyQl3UdLwX5eLpx4wamT58uD8f04MEDfPPNN6hSpQrq1KmDH3/8EU+ePMmTutPS0uQZdLy9vVGoUCG9aZ89e4YRI0YAAObMmYMiRYrkSZvozcWg7Q3y6tUrxMfHAzB+B5qnp6d8Z+Tdu3fzvG1vqosXL+Lq1at49eoVkpKScP36daxatQrNmjVDSEgIEhIS8ruJbyz148bY8ebn56cznyU6cOAA4uLikJaWhsePH+PYsWP4+uuvUb58efzyyy/53bw8kZmZidmzZ8vPu3XrppWmIB9PpUuXxpQpU3Dt2jUcOXIEgwcPhqenJwDgxIkTGDp0KIoVK4YPPvgAW7duRXp6utnqDgsLk7/3u3btajDt+PHjERsbi3fffRf9+/c3Wxuo4GDQ9gZ5/vy5vGzKMB6qoC0pKSnP2vSmcnJyQo8ePbBkyRIcOnQIUVFR2L17NyZNmoTChQsDyLpepmPHjkhLS8vn1r6ZcnK8qQ+dYonHGwCULVsWY8eOxR9//IHjx4/j+PHjWLduHbp27SqfXvvkk0/k4S7eJvPnz8fx48cBACEhIQgMDNRK87YcT++++y4WL16MBw8eYOPGjWjfvj1sbW2RmpqKP/74Ax06dECJEiUwZswYnD17VlFdN27cwKRJkwBAnvpQn8OHD+PXX3+FjY0NFi9eDEmSFNVNBRPHaXuDvHr1Sl62s7Mzmt7e3h4AFI2mXVDFxMTAw8ND6/UWLVpg2LBhaN26NaKionDgwAH8/PPPGD58+Otv5BsuJ8eb6lgDLPN4CwkJQWhoqNYPZZ06ddC9e3ds27YNnTt3RlpaGkaNGoUOHTqgaNGi+dRa8zpw4AA+//xzAFmn737++Wed6d6248ne3h4ffPABPvjgA8THx2PNmjVYtWoVTp48iYcPH+K7777Dd999h1q1aiE0NBS9e/eWe+dMkZycjM6dO8tnAxYtWqR3/L/U1FQMGjQIQgiMGjUK1apVM8s2UsHDnrY3iIODg7ysftGvPqqLVx0dHfOsTW8qXQGbio+PDzZt2iT/cCxatOg1tapgycnxpn6htCUeb+7u7gZ7Ntq1a4epU6cCyPoxXrp06etqWp66cOECQkJCkJ6eDnt7e2zYsAE+Pj46077Nx5OXlxeGDx+OEydO4OLFi5gwYYJ8CjgqKgojR46U75w1RXp6Orp27YozZ84AAAYPHmxwSrWZM2fi0qVLKFmypHyckWVi0PYGcXV1lZdNOWXw4sULAKadSrU0ZcuWRYsWLQAA169fx/379/O5RW+enBxvqmMN4PGmz8CBA+XATv2i/YLq5s2baNmyJZ4+fQpra2usXbvW4F2hlnI8FS1aFGXKlMn1zBdCCPTp0wc7duwAkHUd208//aQ3/eXLlzFr1iwAWX9ALWWWF9KNp0ffIA4ODvDy8kJ8fDzu3btnMO3Tp0/lLz71i3rp/1SuXBnbt28HkHU61dDUQ5ZI/UfH2PGmfrE4jzfdvL294eXlhUePHhX4O0nv37+P5s2b4/79+5AkCcuWLUNISIjBPG/z8ZSWloYdO3Zg1apV2LZtm0ZPoq+vLz766CO0bNnSpLI+++wz/P777wCA1q1b4/fffzc4zMf8+fORmpqKsmXLIjk5GevWrdNKc/78eXl57969iI2NBQC0b9+eQd5bhkHbGyYgIACHDh3C9evXkZ6ernfYj8uXL2vkIW1vwqCdb7LKlSvLy+rHky483kzzNhxz8fHxaNGiBW7cuAEgq3dHNZi3IW/j8XTs2DGsXr0a69at0xj01tHREZ06dUJoaChatGhhdGw1lQkTJsjXBDZu3Bh//PEHbG1tDeZRnUq+ceMGevbsabSOGTNmyMs3b95k0PaWYdD2hmnUqBEOHTqEFy9e4OTJk6hXr57OdOqnX1SjaZMm9YEx2cumrUyZMihevDju379v9HTewYMHAWT1KpQuXfo1tK7giYuLk3/YC+rxlpCQgFatWsmfndmzZ+Ozzz4zKe/bcjzdunULv/32G1avXo2rV69qrHv33XcRGhqK7t27w93dPUflfvXVV5g7dy6ArBtYtm3bViCu56M3TP7NoEW6HDt2LEdzj3p4eFjk3KPGREdHC1tbWwFAlC1bNr+b81rkZk7NTz/9NEdzRQ4ZMsSMLc4fudlPppgxY4Zc7owZM8xW7uvy4sUL0bBhQ3kbJk2alOMyCurx9OzZM/Hrr7+Kxo0bC0mS5PYBECVLlhSTJk0SV69ezXX5CxYskMurVq2aePz4sRlbLzj3qAVh0PYGCgoKEgCEjY2NCA8P11o/d+5ci54w/e+//xZpaWl618fGxopatWrJ++jbb799ja3LP7kJRq5cuSJsbGwEABEYGCiSk5M11icnJ4vAwED5eFTyw/WmyOl+unnzpjh16pTBNFu3bhV2dnYCgHBwcBD37t0zU2tfj5SUFNGyZUt5v4wYMSJX5RTU48nBwUEjUHNychIfffSR2LNnj0kTuBuybNkyORD09/cXsbGxZmr1/2HQZjl4evQNtHDhQjRs2BAvX75Ey5Yt8cUXX6Bp06Z4+fIl1q1bJw/e6e/vjzFjxuRza1+/YcOGIS0tDV26dEGDBg1QunRpODo6Ij4+Hvv378fixYvl01SNGjUy+fROQXP48GFcv35dfq4aVR3IumN2xYoVGul1DSng7++PsWPHYvbs2Thx4gQaNmyICRMmoFy5coiOjsacOXMQFRUFABg3bhwqVKiQJ9uSl5Tup1u3bqFp06Zo0KAB2rdvj5o1a8Lb2xtCCNy4cQObNm3Cpk2b5OvZ5s2bB19f3zzbnrzQs2dP7N69GwDQrFkz9O/fX+Pi9uzs7Ozg7++v9XpBPZ5evXoFSZIQFBSE0NBQdOvWzSx3tW7ZsgUDBw6EEAJubm5YuHAhHj16hEePHunNU6ZMGV6HRvrlc9BIevz999/Czc1N49+f+sPf319cu3Ytv5uZL0qVKqV3v6g/unTpIp4+fZrfzc0zoaGhJu0H1UOfjIwM0a9fP4N5+/fvLzIyMl7j1pmP0v20b98+k/I5OTmJX375JR+2ULmc7B8AolSpUnrLKojH09SpU0V0dLTZy83psYdc9pSxp81ysKftDdW+fXucPXsWCxcuxPbt23Hv3j3Y2dmhfPny6Nq1K4YOHQonJ6f8bma+WLlyJQ4cOICIiAjcuHED8fHxSExMhIuLC/z8/OSLhRs0aJDfTS0QrKyssHTpUnTp0gVhYWGIjIxEfHw8vLy8UKdOHQwePBitW7fO72bmm9q1a+O3335DREQETpw4gQcPHiA+Ph7p6enw9PRElSpV8N5772HAgAHw9vbO7+bmu4J4PE2bNi2/m0BkEkmIt+AedSIiIqK3HGdEICIiIioAGLQRERERFQAM2oiIiIgKAAZtRERERAUAgzYiIiKiAoBBGxEREVEBwKCNiIiIqABg0EZERERUADBoIyIiIioAGLQRERERFQAM2oiIiIgKAAZtRG+5W7duQZIkSJKEFStW5Hdz3lgZGRlYuHAh6tatCzc3N3mfderUKb+bRkQEALDJ7wYQEb0JevbsiY0bN+Z3M4iI9GJPGxFZvPDwcDlga9u2Lf7991+cPXsW586dw/fff5/PrbMc+/fvl3s49+/fn9/NIXrjsKeNiCzenj17AADW1tZYs2YN3Nzc8rlFRETa2NNGRBYvJiYGAODj48OAjYjeWAzaiMjipaSkAABsbW3zuSVERPoxaCOLMW3aNPl6GQB49eoVvvnmG7zzzjtwdXWFq6sr6tatix9++AHp6el6y1GVMW3aNIP1NWnSBJIkoUmTJlrrsl+7I4TA0qVL0ahRIxQuXBhubm6oW7cuVq9erZEvNTUVixcvRv369VGoUCG4urqiYcOG2LBhQ472xcaNG9G8eXN4e3vD0dERlSpVwueff46nT5+alP/48eMYOHAg/P394eLiAmdnZ1SqVAmfffYZrl27pjffihUr5O2+desWUlJSsGDBAtSvXx9eXl4m7Vd9zp07h0GDBqFChQpwcnKCq6srqlSpglGjRuHWrVs686jasnLlSgDA7du35dfUj5WcSk1NxU8//YSmTZuiSJEisLOzQ9GiRdGmTRv89ttvyMzM1Ju3T58+kCQJpUuXBpDVCzh69Gj4+/vDyckJRYoUQZs2bfDPP/+Y1Jbk5GQsWLAATZs2hY+PD+zs7ODt7Y2WLVti+fLlyMjI0Ju3dOnSkCQJffr0AQCcPHkSffr0QZkyZWBvb6+1f27cuIFvv/0W7du3R+nSpeHo6AhHR0eUKlUK3bt3x86dO3XWo7rDuWnTpvJrTZs21XovdN39bM59/eDBA0yYMAFVqlSBq6ur1rV1GRkZWLFiBVq1aoWiRYvCzs4OHh4eqFChAt577z3MnDkTFy9e1FsfkWKCyEJMnTpVABAARGxsrKhRo4b8PPujffv2IiMjQ2c5qjRTp041WF9wcLAAIIKDg7XW7du3Ty5n9+7don379nrbMnz4cCGEEE+ePBGNGzfWm+7rr7/W2Y6bN2/KaZYvXy769eunt4xixYqJCxcu6N2mtLQ08emnn+rND0DY2tqKsLAwnfmXL18up4uMjBQ1a9bUym9sv+oyc+ZMYWVlpbdN9vb2YuXKlVr5DG2H6pFTt27dEgEBAQbLbNSokXj8+LHO/KGhoQKAKFWqlIiMjBTe3t56yxkxYoTBthw/flz4+voabEvdunVFbGyszvylSpUSAERoaKj4+eefhY2Njd79c+PGDZP250cffSTS0tI06lE/Rg09li9fnmf7OiIiQnh5eWnl37dvnxBCiOfPn4ugoCCjbezSpYvB94RICQZtZDHUg7Z3331X2NnZieHDh4t///1XnDx5UqxZs0bjB2Dx4sU6yzF30FavXj0BQPTq1Uts375dnDx5Uqxdu1ZUrFhRTvPvv/+KDh06CBsbG/Hpp5+K3bt3i5MnT4qlS5eK4sWLCwDC2tpanD9/Xqsu9R/EOnXqyD/Ua9euFSdOnBA7duwQ3bt3l9OUKFFCJCQk6Nym3r17y+lat24tfvvtN3H8+HERGRkplixZIqpUqSKv//vvv7Xyqwdt1atXF5Ikid69e8vbvXnzZrFjxw6D+zW7H3/8US6zSJEiYt68eSIiIkIcPnxYTJs2TTg7OwsAQpIksX37do28586dE+fOnRMdO3YUAETx4sXl11SPnHj+/LkoW7as3J5OnTqJv//+W5w4cUJs3LhRPiYAiAYNGoj09HStMlSBRJEiRUTp0qWFvb29+Pzzz8XBgwfFsWPHxPfffy+KFSsml/Ptt9/qbMvZs2flbff29hZTp04Ve/bsEVFRUWLXrl3is88+k4OwevXqidTUVK0yVEFb5cqVhbW1tShdurT44Ycf5P07a9YsOe21a9eEnZ2daN++vfj+++/Fnj17xKlTp8SePXvETz/9pHFsTJkyRaOe1NRUce7cObFs2TI5zbJly7Tei6dPn+bJvi5cuLAoXry4cHFxEZMmTRL79+8Xx48fF0uXLhWXL18WQggxZswYubx27dqJtWvXiiNHjoiTJ0+KnTt3ijlz5oigoCDxwQcfmHSsEOUGgzayGOpBm62trfwPWt3jx4+Fj4+PHFToYu6gDYBYsGCBVpoHDx4INzc3+QdckiSxefNmrXRnzpyRe5lUvXLqsvditGnTRqunQwghvvzySznN2LFjtdZv2rRJXr9kyRKd2/zy5UvRrFkzAUCULl1aqx71oA2AWLp0qc5yTBUXFyecnJzkgOvOnTtaaU6dOiUHL76+vjqDE/UeFyXGjh0rb9vkyZO11mdmZopevXrJaX766Se9bVEdpwcOHNBKExMTI0qUKCEACCcnJ/Hw4UOteqpXry4AiBo1aohHjx7pbO8///wjHzu//vqr1npV0AZAVKtWTSNoyi4pKUncv39f7/rMzEzRp08fAUA4OzuLZ8+eaaVR/1zo+nyqM/e+dnFxEadPn9Zbn5+fnwBgNCjT16tHZA4M2shiqAdto0eP1pvu888/l9Pp+mHJi542fdR/VHr06KE3neq0aa1atbTWqQdt9vb2IiYmRmcZGRkZomrVqgKA8PT0FK9evdJYX7t2bQFAhISE6G2HEEJcvHhRo4dQnXrQ1qxZM4PlmGLOnDlyeWvXrtWb7quvvpLTbdiwQWu9OYK2V69eCQ8PD7lnSlfPjhBCJCQkiMKFC8vp9LUFgBg6dKje+tavXy+nmzt3rsa6rVu3yuvOnDljsN3dunUTAETDhg211qkHbQcPHjRYjikeP34srK2tBQCxadMmrfWmBm15sa+//PJLg223tbUVAMTChQsNbyRRHuKNCGSRevXqpXdd7dq15eWbN2/meVt69Oihd1316tXl5e7du+tNV6NGDQBZF4Ib0rJlSxQvXlznOisrK4SGhgIAnj59ilOnTsnrYmJicPLkSQBAt27dDNYREBAALy8vAEBERITedIbeA1Opxlfz8PBAly5d9KYbMGCAVh5zO3nyJJ49ewYg6wJ3a2trnenc3NzkfXjx4kU8ePBAb5l9+/bVuy4kJAQeHh4AtLfpr7/+AgBUrFhR4xjSpXHjxgCAyMhIvTcl+Pn5ISgoyGA52aWlpeHevXu4dOkSzp8/j/Pnz+P+/fsoXLgwAODMmTM5Kk9dXuxrY8djsWLFAADr169HcnJyLlpNpByDNrJIlSpV0ruuUKFC8vLz58/zvC3+/v5616l+lE1NZ6y9derUMbi+bt268vL58+fl5RMnTsjLPXv21LqrL/sjPj4eABAbG6u3LmPBhClUbaxVq5bB4Tp8fHzkOwTVt8uc1MutV6+ewbTq6/W1x87OzuA+srW1Ra1atXSWoXq/rly5YvS9Gjp0KICsuzCfPHmisy5T36u0tDT8+OOPqF+/PlxcXODn54fKlSujWrVq8iMuLg4A5GMkN8y9r11cXFC2bFmD5aj+0ISHh6NMmTIYOnQoNm/ejEePHpnabCLFGLSRRXJyctK7zsrq/z4WhoZDeN1tMSWdoeENAMDb29vgeh8fH3lZ/Qdc9UObU4Z6JDw9PXNVpjpVG9XbrU/RokU18pibernG2qNqi6H2FCpUCDY2hietUdWTvQxzv1+mvFdPnjxBgwYNMHToUBw7dgypqakG0798+TJXbVTVpWKOfa3+50if//3vf+jXrx8kSUJcXBx+/PFHdO7cGT4+PqhWrRqmTp2Khw8fmrYBRLnEaayILIixcceEEDpfVw9ef//9d5N7Xgz92Os7pZUbpoynpm/b8kJu93NOyjBUjur9atiwIRYvXmy0HBV9p85Nea9GjBghn0Lv1KkT+vXrh+rVq8Pb2xsODg7y9pQsWRJ379412/thjn1tyvbZ2tpi6dKlGDNmDNauXYu9e/fixIkTSE1NlU//fvfdd/jtt9/QsWNHk9tPlBMM2ohySJIkCCGM9mq9ePHiNbXIdMZ6AtR7aNRPE6uuQwKytr9q1armb1wuFCpUCA8ePDB4GlZFte3q22XutqjExsYaPJ2t/j7oa8/jx4+RkZFhMKBQvV/ZyyhcuDAePnyIR48evZb3KjExEevXrwcAfPjhh/j999/1pjV1AGdDzL2vc6Jy5cqYMWMGZsyYgZcvX+LIkSNYs2YNVq1ahaSkJPTs2RPR0dHyNXBE5sTTo0Q55OrqCsDwj09mZqbBmQHyS2RkpMnr1X/sVddOAcDu3bvN37BcUrUxKioKaWlpetPFxcXh9u3bGnnyqi0AcOzYMYNpjx8/rjOfutTUVIMX66enp+P06dM6y1C9X1evXpW3Oy9du3ZN3v+Gbqy5cuUKkpKS9K43dQYKc+/r3HJ0dETz5s2xbNkyfPPNNwCyTvtu27bNrPUQqTBoI8qhMmXKANC8OD+7HTt2ICEh4XU1yWS7d+/WewddZmamPJ2Tp6cn3nnnHXld+fLlUblyZQDAunXrcOfOnbxvrAmaN28OAHj27Bn++OMPvemWLl0qnyZT5TG32rVry9dGrVy5Uu/1kM+fP5enHatcubLBHhnV+6HL5s2b5T8O2bepQ4cO8vLcuXNNar8S6tO+GbqO0dipWgcHB3lZNR+sLnmxr5V677335GUlN1kQGcKgjSiHgoODAWT9wz9y5IjW+gcPHmD48OGvu1kmSUlJweDBg3X+yM2ePRvnzp0DAPTr1w/29vYa6ydPngwga87Wzp07G7xrLiUlBT/99BNevXplxtZr69u3r3yDxpgxY3D37l2tNGfOnMHMmTMBAL6+vujUqVOetMXe3l4eWuTChQuYPn26VhohBIYOHSr/qKvu3NTn559/xuHDh7Vej42NxdixYwFk3aCiurNRpUuXLggICJDLWLp0qcF6zp8/j61btxpMY0j58uXlXrJVq1bpTLNt2zYsWrTIYDnqQVV0dLTedHmxrw158uQJ/v77b4PXx6n3QKv+2BGZXT6ND0f02qkPrmuIsQE+z58/L0//4+npKebPny8iIyPFkSNHxNy5c4WPj48oXLiwqFChgkmD6xoaRFR9MNqbN2/matvUB9cNDAyUB/Rdt26dOHnypPjnn39Ejx495DQlSpTQOaiwEJqDkXp5eYlJkyaJ3bt3i6ioKHH48GGxcuVKMWDAAFGoUCEBQDx//jxX25MT6tNYeXt7i++++04cPXpUHDlyREyfPl24uLgIQPc0Vtm3S+mMCImJiRpTK4WEhIitW7eKkydPik2bNokmTZrI60yZxqpUqVLCwcFBTJw4URw6dEgcP35c/PDDD/LUZYDhaaxU2w5AtGrVSqxcuVIcPXpUft9nzpwp3n33XQFAjBkzRqsM9blHjWnbtq1GXX/++ac8TVr//v2FtbW1qFChgihSpIjBMlUzPZQpU0Zs2bJFXLp0SVy7dk1cu3ZNJCYm5sm+Nva+qz5DpUuXFqNHjxbr168XR48eFSdOnBBbt24VgwYNkmeWKFGihEhKSjK6v4hyg0EbWQxzBW1CCPHdd9/JabI/PD09xYEDB0yeEeF1Bm3Lly+XpxLS9TA2YXx6eroYP368PKq9oYezs7NITk7O1fbk1Ndff52rCeNVzBW0CZG1vytVqmRw3zRs2NDkCeN1TWKueuiatkzdmTNn5D8Pxh7Tp0/Xyp+ToO3OnTuiZMmSessvWbKkuHDhgtEyf/rpJ71lZJ8w3pz72hBTJ7T39fUVp06dMrqviHKLp0eJcmHUqFHYuXMnWrVqBU9PT9jb26NMmTL47LPPcPr0aXmU+TfR8uXLsWbNGjRp0gSFCxeGvb09/P39MX78eFy4cEG+dk0Xa2trzJkzBxcvXsSYMWNQq1YteHp6wtraGq6urqhSpQp69eqFlStX4sGDB3B0dHwt2/TFF18gKioKAwcORLly5eDo6AhnZ2cEBARgxIgRuHz5Mnr37v1a2lK6dGmcOXMGP/zwA4KDg1G4cGHY2trCx8cH77//PlavXo2DBw+adCdjYGAgTp06heHDh6NcuXJwcHBA4cKF8f7772PHjh1YuHChwfzVq1fHxYsXsXLlSnTq1Al+fn5wcHCAnZ0dihUrhiZNmmDy5Mk4efIkpkyZomi7/fz8cOrUKYwbNw7+/v6wt7eHu7s7atSogalTp+L06dMGjy2VTz/9FH/88QdatmwJb29vg2PVmXNfG1KqVCmcPn0a33zzDVq3bo2KFSvCw8MDNjY28PLyQnBwMObNm4dLly5p3LRDZG6SEK9x8CIiIjKoT58+WLlyJUqVKoVbt27ld3OI6A3CnjYiIiKiAoBBGxEREVEBwKCNiIiIqABg0EZERERUADBoIyIiIioAePcoERERUQHAnjYiIiKiAoBBGxEREVEBwKCNiIiIqABg0EZERERUADBoIyIiIioAGLQRERERFQAM2oiIiIgKAAZtRERERAXA/wNA6nIIBnqz9AAAAABJRU5ErkJggg==",
      "text/plain": [
       "<Figure size 600x500 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "import matplotlib.pyplot as plt\n",
    "\n",
    "counts = operators_in_process['operator_count'].value_counts().sort_index()\n",
    "bins = counts.index\n",
    "\n",
    "#bar_width = 0.5\n",
    "plt.figure(figsize=(6, 5))\n",
    "\n",
    "plt.rc('font', size=20)\n",
    "plt.bar(bins, counts, color='maroon')\n",
    "plt.xticks([0, 5, 10, 15, 20, 25], [\"0\", \"5\", \"10\", \"15\", \"20\", \">24\"])\n",
    "plt.xlim([0, 30]) \n",
    "plt.title('Number of operators in processes')\n",
    "plt.xlabel('number of operators')\n",
    "plt.ylabel('number of processes')\n",
    "plt.tight_layout()\n",
    "plt.savefig(f'./plots/operators_in_processes.png', format=\"png\", bbox_inches='tight')\n",
    "plt.show()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
